// google-ads-api-client/dist/protos/bindings/google/longrunning/operations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Operations = exports.OperationInfo = exports.WaitOperationRequest = exports.DeleteOperationRequest = exports.CancelOperationRequest = exports.ListOperationsResponse = exports.ListOperationsRequest = exports.GetOperationRequest = exports.Operation = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "google/longrunning/operations.proto" (package "google.longrunning", syntax proto3)
// tslint:disable
//
// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const empty_1 = require("../protobuf/empty");
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const duration_1 = require("../protobuf/duration");
const status_1 = require("../rpc/status");
const any_1 = require("../protobuf/any");
// @generated message type with reflection information, may provide speed optimized methods
class Operation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.longrunning.Operation", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "metadata", kind: "message", T: () => any_1.Any },
            { no: 3, name: "done", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "error", kind: "message", oneof: "result", T: () => status_1.Status },
            { no: 5, name: "response", kind: "message", oneof: "result", T: () => any_1.Any }
        ]);
    }
    create(value) {
        const message = { name: "", done: false, result: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* google.protobuf.Any metadata */ 2:
                    message.metadata = any_1.Any.internalBinaryRead(reader, reader.uint32(), options, message.metadata);
                    break;
                case /* bool done */ 3:
                    message.done = reader.bool();
                    break;
                case /* google.rpc.Status error */ 4:
                    message.result = {
                        oneofKind: "error",
                        error: status_1.Status.internalBinaryRead(reader, reader.uint32(), options, message.result.error)
                    };
                    break;
                case /* google.protobuf.Any response */ 5:
                    message.result = {
                        oneofKind: "response",
                        response: any_1.Any.internalBinaryRead(reader, reader.uint32(), options, message.result.response)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.protobuf.Any metadata = 2; */
        if (message.metadata)
            any_1.Any.internalBinaryWrite(message.metadata, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool done = 3; */
        if (message.done !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.done);
        /* google.rpc.Status error = 4; */
        if (message.result.oneofKind === "error")
            status_1.Status.internalBinaryWrite(message.result.error, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.protobuf.Any response = 5; */
        if (message.result.oneofKind === "response")
            any_1.Any.internalBinaryWrite(message.result.response, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.longrunning.Operation
 */
exports.Operation = new Operation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GetOperationRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.longrunning.GetOperationRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { name: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.longrunning.GetOperationRequest
 */
exports.GetOperationRequest = new GetOperationRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListOperationsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.longrunning.ListOperationsRequest", [
            { no: 4, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 1, name: "filter", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "page_size", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { name: "", filter: "", pageSize: 0, pageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 4:
                    message.name = reader.string();
                    break;
                case /* string filter */ 1:
                    message.filter = reader.string();
                    break;
                case /* int32 page_size */ 2:
                    message.pageSize = reader.int32();
                    break;
                case /* string page_token */ 3:
                    message.pageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 4; */
        if (message.name !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.name);
        /* string filter = 1; */
        if (message.filter !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.filter);
        /* int32 page_size = 2; */
        if (message.pageSize !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.pageSize);
        /* string page_token = 3; */
        if (message.pageToken !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.pageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.longrunning.ListOperationsRequest
 */
exports.ListOperationsRequest = new ListOperationsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListOperationsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.longrunning.ListOperationsResponse", [
            { no: 1, name: "operations", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.Operation },
            { no: 2, name: "next_page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { operations: [], nextPageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.longrunning.Operation operations */ 1:
                    message.operations.push(exports.Operation.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string next_page_token */ 2:
                    message.nextPageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.longrunning.Operation operations = 1; */
        for (let i = 0; i < message.operations.length; i++)
            exports.Operation.internalBinaryWrite(message.operations[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string next_page_token = 2; */
        if (message.nextPageToken !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.nextPageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.longrunning.ListOperationsResponse
 */
exports.ListOperationsResponse = new ListOperationsResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CancelOperationRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.longrunning.CancelOperationRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { name: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.longrunning.CancelOperationRequest
 */
exports.CancelOperationRequest = new CancelOperationRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DeleteOperationRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.longrunning.DeleteOperationRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { name: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.longrunning.DeleteOperationRequest
 */
exports.DeleteOperationRequest = new DeleteOperationRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class WaitOperationRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.longrunning.WaitOperationRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "timeout", kind: "message", T: () => duration_1.Duration }
        ]);
    }
    create(value) {
        const message = { name: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* google.protobuf.Duration timeout */ 2:
                    message.timeout = duration_1.Duration.internalBinaryRead(reader, reader.uint32(), options, message.timeout);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.protobuf.Duration timeout = 2; */
        if (message.timeout)
            duration_1.Duration.internalBinaryWrite(message.timeout, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.longrunning.WaitOperationRequest
 */
exports.WaitOperationRequest = new WaitOperationRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class OperationInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.longrunning.OperationInfo", [
            { no: 1, name: "response_type", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "metadata_type", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { responseType: "", metadataType: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string response_type */ 1:
                    message.responseType = reader.string();
                    break;
                case /* string metadata_type */ 2:
                    message.metadataType = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string response_type = 1; */
        if (message.responseType !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.responseType);
        /* string metadata_type = 2; */
        if (message.metadataType !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.metadataType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.longrunning.OperationInfo
 */
exports.OperationInfo = new OperationInfo$Type();
/**
 * @generated ServiceType for protobuf service google.longrunning.Operations
 */
exports.Operations = new runtime_rpc_1.ServiceType("google.longrunning.Operations", [
    { name: "ListOperations", options: { "google.api.http": { get: "/v1/{name=operations}" }, "google.api.method_signature": ["name,filter"] }, I: exports.ListOperationsRequest, O: exports.ListOperationsResponse },
    { name: "GetOperation", options: { "google.api.http": { get: "/v1/{name=operations/**}" }, "google.api.method_signature": ["name"] }, I: exports.GetOperationRequest, O: exports.Operation },
    { name: "DeleteOperation", options: { "google.api.http": { delete: "/v1/{name=operations/**}" }, "google.api.method_signature": ["name"] }, I: exports.DeleteOperationRequest, O: empty_1.Empty },
    { name: "CancelOperation", options: { "google.api.http": { post: "/v1/{name=operations/**}:cancel", body: "*" }, "google.api.method_signature": ["name"] }, I: exports.CancelOperationRequest, O: empty_1.Empty },
    { name: "WaitOperation", options: {}, I: exports.WaitOperationRequest, O: exports.Operation }
], { "google.api.default_host": "longrunning.googleapis.com" });
