// google-ads-api-client/dist/protos/bindings/google/api/context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContextRule = exports.Context = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
// @generated message type with reflection information, may provide speed optimized methods
class Context$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.Context", [
            { no: 1, name: "rules", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.ContextRule }
        ]);
    }
    create(value) {
        const message = { rules: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.ContextRule rules */ 1:
                    message.rules.push(exports.ContextRule.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.ContextRule rules = 1; */
        for (let i = 0; i < message.rules.length; i++)
            exports.ContextRule.internalBinaryWrite(message.rules[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.Context
 */
exports.Context = new Context$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ContextRule$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.ContextRule", [
            { no: 1, name: "selector", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "requested", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "provided", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "allowed_request_extensions", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "allowed_response_extensions", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { selector: "", requested: [], provided: [], allowedRequestExtensions: [], allowedResponseExtensions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string selector */ 1:
                    message.selector = reader.string();
                    break;
                case /* repeated string requested */ 2:
                    message.requested.push(reader.string());
                    break;
                case /* repeated string provided */ 3:
                    message.provided.push(reader.string());
                    break;
                case /* repeated string allowed_request_extensions */ 4:
                    message.allowedRequestExtensions.push(reader.string());
                    break;
                case /* repeated string allowed_response_extensions */ 5:
                    message.allowedResponseExtensions.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string selector = 1; */
        if (message.selector !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.selector);
        /* repeated string requested = 2; */
        for (let i = 0; i < message.requested.length; i++)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.requested[i]);
        /* repeated string provided = 3; */
        for (let i = 0; i < message.provided.length; i++)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.provided[i]);
        /* repeated string allowed_request_extensions = 4; */
        for (let i = 0; i < message.allowedRequestExtensions.length; i++)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.allowedRequestExtensions[i]);
        /* repeated string allowed_response_extensions = 5; */
        for (let i = 0; i < message.allowedResponseExtensions.length; i++)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.allowedResponseExtensions[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.ContextRule
 */
exports.ContextRule = new ContextRule$Type();
