// google-ads-api-client/dist/protos/bindings/google/api/documentation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Page = exports.DocumentationRule = exports.Documentation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
// @generated message type with reflection information, may provide speed optimized methods
class Documentation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.Documentation", [
            { no: 1, name: "summary", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "pages", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.Page },
            { no: 3, name: "rules", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.DocumentationRule },
            { no: 4, name: "documentation_root_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "service_root_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "overview", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { summary: "", pages: [], rules: [], documentationRootUrl: "", serviceRootUrl: "", overview: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string summary */ 1:
                    message.summary = reader.string();
                    break;
                case /* repeated google.api.Page pages */ 5:
                    message.pages.push(exports.Page.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.api.DocumentationRule rules */ 3:
                    message.rules.push(exports.DocumentationRule.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string documentation_root_url */ 4:
                    message.documentationRootUrl = reader.string();
                    break;
                case /* string service_root_url */ 6:
                    message.serviceRootUrl = reader.string();
                    break;
                case /* string overview */ 2:
                    message.overview = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string summary = 1; */
        if (message.summary !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.summary);
        /* repeated google.api.Page pages = 5; */
        for (let i = 0; i < message.pages.length; i++)
            exports.Page.internalBinaryWrite(message.pages[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.api.DocumentationRule rules = 3; */
        for (let i = 0; i < message.rules.length; i++)
            exports.DocumentationRule.internalBinaryWrite(message.rules[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string documentation_root_url = 4; */
        if (message.documentationRootUrl !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.documentationRootUrl);
        /* string service_root_url = 6; */
        if (message.serviceRootUrl !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.serviceRootUrl);
        /* string overview = 2; */
        if (message.overview !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.overview);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.Documentation
 */
exports.Documentation = new Documentation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DocumentationRule$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.DocumentationRule", [
            { no: 1, name: "selector", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "deprecation_description", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { selector: "", description: "", deprecationDescription: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string selector */ 1:
                    message.selector = reader.string();
                    break;
                case /* string description */ 2:
                    message.description = reader.string();
                    break;
                case /* string deprecation_description */ 3:
                    message.deprecationDescription = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string selector = 1; */
        if (message.selector !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.selector);
        /* string description = 2; */
        if (message.description !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.description);
        /* string deprecation_description = 3; */
        if (message.deprecationDescription !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.deprecationDescription);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.DocumentationRule
 */
exports.DocumentationRule = new DocumentationRule$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Page$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.Page", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "content", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "subpages", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.Page }
        ]);
    }
    create(value) {
        const message = { name: "", content: "", subpages: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* string content */ 2:
                    message.content = reader.string();
                    break;
                case /* repeated google.api.Page subpages */ 3:
                    message.subpages.push(exports.Page.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* string content = 2; */
        if (message.content !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.content);
        /* repeated google.api.Page subpages = 3; */
        for (let i = 0; i < message.subpages.length; i++)
            exports.Page.internalBinaryWrite(message.subpages[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.Page
 */
exports.Page = new Page$Type();
