// google-ads-api-client/dist/protos/bindings/google/api/usage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageRule = exports.Usage = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
// @generated message type with reflection information, may provide speed optimized methods
class Usage$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.Usage", [
            { no: 1, name: "requirements", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "rules", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.UsageRule },
            { no: 7, name: "producer_notification_channel", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { requirements: [], rules: [], producerNotificationChannel: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated string requirements */ 1:
                    message.requirements.push(reader.string());
                    break;
                case /* repeated google.api.UsageRule rules */ 6:
                    message.rules.push(exports.UsageRule.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string producer_notification_channel */ 7:
                    message.producerNotificationChannel = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated string requirements = 1; */
        for (let i = 0; i < message.requirements.length; i++)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.requirements[i]);
        /* repeated google.api.UsageRule rules = 6; */
        for (let i = 0; i < message.rules.length; i++)
            exports.UsageRule.internalBinaryWrite(message.rules[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string producer_notification_channel = 7; */
        if (message.producerNotificationChannel !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.producerNotificationChannel);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.Usage
 */
exports.Usage = new Usage$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UsageRule$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.UsageRule", [
            { no: 1, name: "selector", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "allow_unregistered_calls", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "skip_service_control", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { selector: "", allowUnregisteredCalls: false, skipServiceControl: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string selector */ 1:
                    message.selector = reader.string();
                    break;
                case /* bool allow_unregistered_calls */ 2:
                    message.allowUnregisteredCalls = reader.bool();
                    break;
                case /* bool skip_service_control */ 3:
                    message.skipServiceControl = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string selector = 1; */
        if (message.selector !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.selector);
        /* bool allow_unregistered_calls = 2; */
        if (message.allowUnregisteredCalls !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.allowUnregisteredCalls);
        /* bool skip_service_control = 3; */
        if (message.skipServiceControl !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.skipServiceControl);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.UsageRule
 */
exports.UsageRule = new UsageRule$Type();
