// google-ads-api-client/dist/protos/bindings/ads/enums/access_role.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessRoleEnum = exports.AccessRoleEnum_AccessRole = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible access role of a user.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AccessRoleEnum.AccessRole
 */
var AccessRoleEnum_AccessRole;
(function (AccessRoleEnum_AccessRole) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccessRoleEnum_AccessRole[AccessRoleEnum_AccessRole["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccessRoleEnum_AccessRole[AccessRoleEnum_AccessRole["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Owns its account and can control the addition of other users.
     *
     * @generated from protobuf enum value: ADMIN = 2;
     */
    AccessRoleEnum_AccessRole[AccessRoleEnum_AccessRole["ADMIN"] = 2] = "ADMIN";
    /**
     * Can modify campaigns, but can't affect other users.
     *
     * @generated from protobuf enum value: STANDARD = 3;
     */
    AccessRoleEnum_AccessRole[AccessRoleEnum_AccessRole["STANDARD"] = 3] = "STANDARD";
    /**
     * Can view campaigns and account changes, but cannot make edits.
     *
     * @generated from protobuf enum value: READ_ONLY = 4;
     */
    AccessRoleEnum_AccessRole[AccessRoleEnum_AccessRole["READ_ONLY"] = 4] = "READ_ONLY";
    /**
     * Role for \"email only\" access. Represents an email recipient rather than
     * a true User entity.
     *
     * @generated from protobuf enum value: EMAIL_ONLY = 5;
     */
    AccessRoleEnum_AccessRole[AccessRoleEnum_AccessRole["EMAIL_ONLY"] = 5] = "EMAIL_ONLY";
})(AccessRoleEnum_AccessRole = exports.AccessRoleEnum_AccessRole || (exports.AccessRoleEnum_AccessRole = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccessRoleEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AccessRoleEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AccessRoleEnum
 */
exports.AccessRoleEnum = new AccessRoleEnum$Type();
