// google-ads-api-client/dist/protos/bindings/ads/enums/access_invitation_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessInvitationStatusEnum = exports.AccessInvitationStatusEnum_AccessInvitationStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible access invitation status of a user
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AccessInvitationStatusEnum.AccessInvitationStatus
 */
var AccessInvitationStatusEnum_AccessInvitationStatus;
(function (AccessInvitationStatusEnum_AccessInvitationStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccessInvitationStatusEnum_AccessInvitationStatus[AccessInvitationStatusEnum_AccessInvitationStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccessInvitationStatusEnum_AccessInvitationStatus[AccessInvitationStatusEnum_AccessInvitationStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The initial state of an invitation, before being acted upon by anyone.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    AccessInvitationStatusEnum_AccessInvitationStatus[AccessInvitationStatusEnum_AccessInvitationStatus["PENDING"] = 2] = "PENDING";
    /**
     * Invitation process was terminated by the email recipient. No new user was
     * created.
     *
     * @generated from protobuf enum value: DECLINED = 3;
     */
    AccessInvitationStatusEnum_AccessInvitationStatus[AccessInvitationStatusEnum_AccessInvitationStatus["DECLINED"] = 3] = "DECLINED";
    /**
     * Invitation URLs expired without being acted upon. No new user can be
     * created.  Invitations expire 20 days after creation.
     *
     * @generated from protobuf enum value: EXPIRED = 4;
     */
    AccessInvitationStatusEnum_AccessInvitationStatus[AccessInvitationStatusEnum_AccessInvitationStatus["EXPIRED"] = 4] = "EXPIRED";
})(AccessInvitationStatusEnum_AccessInvitationStatus = exports.AccessInvitationStatusEnum_AccessInvitationStatus || (exports.AccessInvitationStatusEnum_AccessInvitationStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccessInvitationStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AccessInvitationStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AccessInvitationStatusEnum
 */
exports.AccessInvitationStatusEnum = new AccessInvitationStatusEnum$Type();
