// google-ads-api-client/dist/protos/bindings/ads/enums/access_reason.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessReasonEnum = exports.AccessReasonEnum_AccessReason = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible access reasons.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AccessReasonEnum.AccessReason
 */
var AccessReasonEnum_AccessReason;
(function (AccessReasonEnum_AccessReason) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccessReasonEnum_AccessReason[AccessReasonEnum_AccessReason["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccessReasonEnum_AccessReason[AccessReasonEnum_AccessReason["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The resource is owned by the user.
     *
     * @generated from protobuf enum value: OWNED = 2;
     */
    AccessReasonEnum_AccessReason[AccessReasonEnum_AccessReason["OWNED"] = 2] = "OWNED";
    /**
     * The resource is shared to the user.
     *
     * @generated from protobuf enum value: SHARED = 3;
     */
    AccessReasonEnum_AccessReason[AccessReasonEnum_AccessReason["SHARED"] = 3] = "SHARED";
    /**
     * The resource is licensed to the user.
     *
     * @generated from protobuf enum value: LICENSED = 4;
     */
    AccessReasonEnum_AccessReason[AccessReasonEnum_AccessReason["LICENSED"] = 4] = "LICENSED";
    /**
     * The user subscribed to the resource.
     *
     * @generated from protobuf enum value: SUBSCRIBED = 5;
     */
    AccessReasonEnum_AccessReason[AccessReasonEnum_AccessReason["SUBSCRIBED"] = 5] = "SUBSCRIBED";
    /**
     * The resource is accessible to the user.
     *
     * @generated from protobuf enum value: AFFILIATED = 6;
     */
    AccessReasonEnum_AccessReason[AccessReasonEnum_AccessReason["AFFILIATED"] = 6] = "AFFILIATED";
})(AccessReasonEnum_AccessReason = exports.AccessReasonEnum_AccessReason || (exports.AccessReasonEnum_AccessReason = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccessReasonEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AccessReasonEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AccessReasonEnum
 */
exports.AccessReasonEnum = new AccessReasonEnum$Type();
