// google-ads-api-client/dist/protos/bindings/ads/enums/account_budget_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountBudgetStatusEnum = exports.AccountBudgetStatusEnum_AccountBudgetStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of an AccountBudget.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AccountBudgetStatusEnum.AccountBudgetStatus
 */
var AccountBudgetStatusEnum_AccountBudgetStatus;
(function (AccountBudgetStatusEnum_AccountBudgetStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccountBudgetStatusEnum_AccountBudgetStatus[AccountBudgetStatusEnum_AccountBudgetStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccountBudgetStatusEnum_AccountBudgetStatus[AccountBudgetStatusEnum_AccountBudgetStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The account budget is pending approval.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    AccountBudgetStatusEnum_AccountBudgetStatus[AccountBudgetStatusEnum_AccountBudgetStatus["PENDING"] = 2] = "PENDING";
    /**
     * The account budget has been approved.
     *
     * @generated from protobuf enum value: APPROVED = 3;
     */
    AccountBudgetStatusEnum_AccountBudgetStatus[AccountBudgetStatusEnum_AccountBudgetStatus["APPROVED"] = 3] = "APPROVED";
    /**
     * The account budget has been cancelled by the user.
     *
     * @generated from protobuf enum value: CANCELLED = 4;
     */
    AccountBudgetStatusEnum_AccountBudgetStatus[AccountBudgetStatusEnum_AccountBudgetStatus["CANCELLED"] = 4] = "CANCELLED";
})(AccountBudgetStatusEnum_AccountBudgetStatus = exports.AccountBudgetStatusEnum_AccountBudgetStatus || (exports.AccountBudgetStatusEnum_AccountBudgetStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccountBudgetStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AccountBudgetStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AccountBudgetStatusEnum
 */
exports.AccountBudgetStatusEnum = new AccountBudgetStatusEnum$Type();
