// google-ads-api-client/dist/protos/bindings/ads/enums/account_budget_proposal_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountBudgetProposalStatusEnum = exports.AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of an AccountBudgetProposal.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AccountBudgetProposalStatusEnum.AccountBudgetProposalStatus
 */
var AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus;
(function (AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus[AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus[AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The proposal is pending approval.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus[AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus["PENDING"] = 2] = "PENDING";
    /**
     * The proposal has been approved but the corresponding billing setup
     * has not.  This can occur for proposals that set up the first budget
     * when signing up for billing or when performing a change of bill-to
     * operation.
     *
     * @generated from protobuf enum value: APPROVED_HELD = 3;
     */
    AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus[AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus["APPROVED_HELD"] = 3] = "APPROVED_HELD";
    /**
     * The proposal has been approved.
     *
     * @generated from protobuf enum value: APPROVED = 4;
     */
    AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus[AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus["APPROVED"] = 4] = "APPROVED";
    /**
     * The proposal has been cancelled by the user.
     *
     * @generated from protobuf enum value: CANCELLED = 5;
     */
    AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus[AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus["CANCELLED"] = 5] = "CANCELLED";
    /**
     * The proposal has been rejected by the user, for example, by rejecting an
     * acceptance email.
     *
     * @generated from protobuf enum value: REJECTED = 6;
     */
    AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus[AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus["REJECTED"] = 6] = "REJECTED";
})(AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus = exports.AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus || (exports.AccountBudgetProposalStatusEnum_AccountBudgetProposalStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccountBudgetProposalStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AccountBudgetProposalStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AccountBudgetProposalStatusEnum
 */
exports.AccountBudgetProposalStatusEnum = new AccountBudgetProposalStatusEnum$Type();
