// google-ads-api-client/dist/protos/bindings/ads/enums/account_budget_proposal_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountBudgetProposalTypeEnum = exports.AccountBudgetProposalTypeEnum_AccountBudgetProposalType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible types of an AccountBudgetProposal.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AccountBudgetProposalTypeEnum.AccountBudgetProposalType
 */
var AccountBudgetProposalTypeEnum_AccountBudgetProposalType;
(function (AccountBudgetProposalTypeEnum_AccountBudgetProposalType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccountBudgetProposalTypeEnum_AccountBudgetProposalType[AccountBudgetProposalTypeEnum_AccountBudgetProposalType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccountBudgetProposalTypeEnum_AccountBudgetProposalType[AccountBudgetProposalTypeEnum_AccountBudgetProposalType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Identifies a request to create a new budget.
     *
     * @generated from protobuf enum value: CREATE = 2;
     */
    AccountBudgetProposalTypeEnum_AccountBudgetProposalType[AccountBudgetProposalTypeEnum_AccountBudgetProposalType["CREATE"] = 2] = "CREATE";
    /**
     * Identifies a request to edit an existing budget.
     *
     * @generated from protobuf enum value: UPDATE = 3;
     */
    AccountBudgetProposalTypeEnum_AccountBudgetProposalType[AccountBudgetProposalTypeEnum_AccountBudgetProposalType["UPDATE"] = 3] = "UPDATE";
    /**
     * Identifies a request to end a budget that has already started.
     *
     * @generated from protobuf enum value: END = 4;
     */
    AccountBudgetProposalTypeEnum_AccountBudgetProposalType[AccountBudgetProposalTypeEnum_AccountBudgetProposalType["END"] = 4] = "END";
    /**
     * Identifies a request to remove a budget that hasn't started yet.
     *
     * @generated from protobuf enum value: REMOVE = 5;
     */
    AccountBudgetProposalTypeEnum_AccountBudgetProposalType[AccountBudgetProposalTypeEnum_AccountBudgetProposalType["REMOVE"] = 5] = "REMOVE";
})(AccountBudgetProposalTypeEnum_AccountBudgetProposalType = exports.AccountBudgetProposalTypeEnum_AccountBudgetProposalType || (exports.AccountBudgetProposalTypeEnum_AccountBudgetProposalType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccountBudgetProposalTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AccountBudgetProposalTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AccountBudgetProposalTypeEnum
 */
exports.AccountBudgetProposalTypeEnum = new AccountBudgetProposalTypeEnum$Type();
