// google-ads-api-client/dist/protos/bindings/ads/enums/account_link_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountLinkStatusEnum = exports.AccountLinkStatusEnum_AccountLinkStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Describes the possible statuses for a link between a Google Ads customer
 * and another account.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AccountLinkStatusEnum.AccountLinkStatus
 */
var AccountLinkStatusEnum_AccountLinkStatus;
(function (AccountLinkStatusEnum_AccountLinkStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccountLinkStatusEnum_AccountLinkStatus[AccountLinkStatusEnum_AccountLinkStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccountLinkStatusEnum_AccountLinkStatus[AccountLinkStatusEnum_AccountLinkStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The link is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    AccountLinkStatusEnum_AccountLinkStatus[AccountLinkStatusEnum_AccountLinkStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The link is removed/disabled.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    AccountLinkStatusEnum_AccountLinkStatus[AccountLinkStatusEnum_AccountLinkStatus["REMOVED"] = 3] = "REMOVED";
    /**
     * The link to the other account has been requested. A user on the other
     * account may now approve the link by setting the status to ENABLED.
     *
     * @generated from protobuf enum value: REQUESTED = 4;
     */
    AccountLinkStatusEnum_AccountLinkStatus[AccountLinkStatusEnum_AccountLinkStatus["REQUESTED"] = 4] = "REQUESTED";
    /**
     * This link has been requested by a user on the other account. It may be
     * approved by a user on this account by setting the status to ENABLED.
     *
     * @generated from protobuf enum value: PENDING_APPROVAL = 5;
     */
    AccountLinkStatusEnum_AccountLinkStatus[AccountLinkStatusEnum_AccountLinkStatus["PENDING_APPROVAL"] = 5] = "PENDING_APPROVAL";
    /**
     * The link is rejected by the approver.
     *
     * @generated from protobuf enum value: REJECTED = 6;
     */
    AccountLinkStatusEnum_AccountLinkStatus[AccountLinkStatusEnum_AccountLinkStatus["REJECTED"] = 6] = "REJECTED";
    /**
     * The link is revoked by the user who requested the link.
     *
     * @generated from protobuf enum value: REVOKED = 7;
     */
    AccountLinkStatusEnum_AccountLinkStatus[AccountLinkStatusEnum_AccountLinkStatus["REVOKED"] = 7] = "REVOKED";
})(AccountLinkStatusEnum_AccountLinkStatus = exports.AccountLinkStatusEnum_AccountLinkStatus || (exports.AccountLinkStatusEnum_AccountLinkStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccountLinkStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AccountLinkStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AccountLinkStatusEnum
 */
exports.AccountLinkStatusEnum = new AccountLinkStatusEnum$Type();
