// google-ads-api-client/dist/protos/bindings/ads/enums/ad_customizer_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdCustomizerPlaceholderFieldEnum = exports.AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Ad Customizers placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdCustomizerPlaceholderFieldEnum.AdCustomizerPlaceholderField
 */
var AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField;
(function (AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField[AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField[AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: INT64. Integer value to be inserted.
     *
     * @generated from protobuf enum value: INTEGER = 2;
     */
    AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField[AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField["INTEGER"] = 2] = "INTEGER";
    /**
     * Data Type: STRING. Price value to be inserted.
     *
     * @generated from protobuf enum value: PRICE = 3;
     */
    AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField[AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField["PRICE"] = 3] = "PRICE";
    /**
     * Data Type: DATE_TIME. Date value to be inserted.
     *
     * @generated from protobuf enum value: DATE = 4;
     */
    AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField[AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField["DATE"] = 4] = "DATE";
    /**
     * Data Type: STRING. String value to be inserted.
     *
     * @generated from protobuf enum value: STRING = 5;
     */
    AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField[AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField["STRING"] = 5] = "STRING";
})(AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField = exports.AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField || (exports.AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdCustomizerPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdCustomizerPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdCustomizerPlaceholderFieldEnum
 */
exports.AdCustomizerPlaceholderFieldEnum = new AdCustomizerPlaceholderFieldEnum$Type();
