// google-ads-api-client/dist/protos/bindings/ads/enums/ad_destination_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdDestinationTypeEnum = exports.AdDestinationTypeEnum_AdDestinationType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates Google Ads destination types
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdDestinationTypeEnum.AdDestinationType
 */
var AdDestinationTypeEnum_AdDestinationType;
(function (AdDestinationTypeEnum_AdDestinationType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Ads that don't intend to drive users off from ads to other destinations
     *
     * @generated from protobuf enum value: NOT_APPLICABLE = 2;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["NOT_APPLICABLE"] = 2] = "NOT_APPLICABLE";
    /**
     * Website
     *
     * @generated from protobuf enum value: WEBSITE = 3;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["WEBSITE"] = 3] = "WEBSITE";
    /**
     * App Deep Link
     *
     * @generated from protobuf enum value: APP_DEEP_LINK = 4;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["APP_DEEP_LINK"] = 4] = "APP_DEEP_LINK";
    /**
     * iOS App Store or Play Store
     *
     * @generated from protobuf enum value: APP_STORE = 5;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["APP_STORE"] = 5] = "APP_STORE";
    /**
     * Call Dialer
     *
     * @generated from protobuf enum value: PHONE_CALL = 6;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["PHONE_CALL"] = 6] = "PHONE_CALL";
    /**
     * Map App
     *
     * @generated from protobuf enum value: MAP_DIRECTIONS = 7;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["MAP_DIRECTIONS"] = 7] = "MAP_DIRECTIONS";
    /**
     * Location Dedicated Page
     *
     * @generated from protobuf enum value: LOCATION_LISTING = 8;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["LOCATION_LISTING"] = 8] = "LOCATION_LISTING";
    /**
     * Text Message
     *
     * @generated from protobuf enum value: MESSAGE = 9;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["MESSAGE"] = 9] = "MESSAGE";
    /**
     * Lead Generation Form
     *
     * @generated from protobuf enum value: LEAD_FORM = 10;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["LEAD_FORM"] = 10] = "LEAD_FORM";
    /**
     * YouTube
     *
     * @generated from protobuf enum value: YOUTUBE = 11;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["YOUTUBE"] = 11] = "YOUTUBE";
    /**
     * Ad Destination for Conversions with keys unknown
     *
     * @generated from protobuf enum value: UNMODELED_FOR_CONVERSIONS = 12;
     */
    AdDestinationTypeEnum_AdDestinationType[AdDestinationTypeEnum_AdDestinationType["UNMODELED_FOR_CONVERSIONS"] = 12] = "UNMODELED_FOR_CONVERSIONS";
})(AdDestinationTypeEnum_AdDestinationType = exports.AdDestinationTypeEnum_AdDestinationType || (exports.AdDestinationTypeEnum_AdDestinationType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdDestinationTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdDestinationTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdDestinationTypeEnum
 */
exports.AdDestinationTypeEnum = new AdDestinationTypeEnum$Type();
