// google-ads-api-client/dist/protos/bindings/ads/enums/ad_group_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupStatusEnum = exports.AdGroupStatusEnum_AdGroupStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of an ad group.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdGroupStatusEnum.AdGroupStatus
 */
var AdGroupStatusEnum_AdGroupStatus;
(function (AdGroupStatusEnum_AdGroupStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupStatusEnum_AdGroupStatus[AdGroupStatusEnum_AdGroupStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupStatusEnum_AdGroupStatus[AdGroupStatusEnum_AdGroupStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The ad group is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    AdGroupStatusEnum_AdGroupStatus[AdGroupStatusEnum_AdGroupStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The ad group is paused.
     *
     * @generated from protobuf enum value: PAUSED = 3;
     */
    AdGroupStatusEnum_AdGroupStatus[AdGroupStatusEnum_AdGroupStatus["PAUSED"] = 3] = "PAUSED";
    /**
     * The ad group is removed.
     *
     * @generated from protobuf enum value: REMOVED = 4;
     */
    AdGroupStatusEnum_AdGroupStatus[AdGroupStatusEnum_AdGroupStatus["REMOVED"] = 4] = "REMOVED";
})(AdGroupStatusEnum_AdGroupStatus = exports.AdGroupStatusEnum_AdGroupStatus || (exports.AdGroupStatusEnum_AdGroupStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdGroupStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdGroupStatusEnum
 */
exports.AdGroupStatusEnum = new AdGroupStatusEnum$Type();
