// google-ads-api-client/dist/protos/bindings/ads/enums/ad_group_ad_rotation_mode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupAdRotationModeEnum = exports.AdGroupAdRotationModeEnum_AdGroupAdRotationMode = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible ad rotation modes of an ad group.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdGroupAdRotationModeEnum.AdGroupAdRotationMode
 */
var AdGroupAdRotationModeEnum_AdGroupAdRotationMode;
(function (AdGroupAdRotationModeEnum_AdGroupAdRotationMode) {
    /**
     * The ad rotation mode has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupAdRotationModeEnum_AdGroupAdRotationMode[AdGroupAdRotationModeEnum_AdGroupAdRotationMode["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupAdRotationModeEnum_AdGroupAdRotationMode[AdGroupAdRotationModeEnum_AdGroupAdRotationMode["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Optimize ad group ads based on clicks or conversions.
     *
     * @generated from protobuf enum value: OPTIMIZE = 2;
     */
    AdGroupAdRotationModeEnum_AdGroupAdRotationMode[AdGroupAdRotationModeEnum_AdGroupAdRotationMode["OPTIMIZE"] = 2] = "OPTIMIZE";
    /**
     * Rotate evenly forever.
     *
     * @generated from protobuf enum value: ROTATE_FOREVER = 3;
     */
    AdGroupAdRotationModeEnum_AdGroupAdRotationMode[AdGroupAdRotationModeEnum_AdGroupAdRotationMode["ROTATE_FOREVER"] = 3] = "ROTATE_FOREVER";
})(AdGroupAdRotationModeEnum_AdGroupAdRotationMode = exports.AdGroupAdRotationModeEnum_AdGroupAdRotationMode || (exports.AdGroupAdRotationModeEnum_AdGroupAdRotationMode = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupAdRotationModeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdGroupAdRotationModeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdGroupAdRotationModeEnum
 */
exports.AdGroupAdRotationModeEnum = new AdGroupAdRotationModeEnum$Type();
