// google-ads-api-client/dist/protos/bindings/ads/enums/ad_group_criterion_approval_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupCriterionApprovalStatusEnum = exports.AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates AdGroupCriterion approval statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdGroupCriterionApprovalStatusEnum.AdGroupCriterionApprovalStatus
 */
var AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus;
(function (AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus[AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus[AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Approved.
     *
     * @generated from protobuf enum value: APPROVED = 2;
     */
    AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus[AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus["APPROVED"] = 2] = "APPROVED";
    /**
     * Disapproved.
     *
     * @generated from protobuf enum value: DISAPPROVED = 3;
     */
    AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus[AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus["DISAPPROVED"] = 3] = "DISAPPROVED";
    /**
     * Pending Review.
     *
     * @generated from protobuf enum value: PENDING_REVIEW = 4;
     */
    AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus[AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus["PENDING_REVIEW"] = 4] = "PENDING_REVIEW";
    /**
     * Under review.
     *
     * @generated from protobuf enum value: UNDER_REVIEW = 5;
     */
    AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus[AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus["UNDER_REVIEW"] = 5] = "UNDER_REVIEW";
})(AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus = exports.AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus || (exports.AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupCriterionApprovalStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdGroupCriterionApprovalStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdGroupCriterionApprovalStatusEnum
 */
exports.AdGroupCriterionApprovalStatusEnum = new AdGroupCriterionApprovalStatusEnum$Type();
