// google-ads-api-client/dist/protos/bindings/ads/enums/ad_group_ad_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupAdStatusEnum = exports.AdGroupAdStatusEnum_AdGroupAdStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of an AdGroupAd.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdGroupAdStatusEnum.AdGroupAdStatus
 */
var AdGroupAdStatusEnum_AdGroupAdStatus;
(function (AdGroupAdStatusEnum_AdGroupAdStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupAdStatusEnum_AdGroupAdStatus[AdGroupAdStatusEnum_AdGroupAdStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupAdStatusEnum_AdGroupAdStatus[AdGroupAdStatusEnum_AdGroupAdStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The ad group ad is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    AdGroupAdStatusEnum_AdGroupAdStatus[AdGroupAdStatusEnum_AdGroupAdStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The ad group ad is paused.
     *
     * @generated from protobuf enum value: PAUSED = 3;
     */
    AdGroupAdStatusEnum_AdGroupAdStatus[AdGroupAdStatusEnum_AdGroupAdStatus["PAUSED"] = 3] = "PAUSED";
    /**
     * The ad group ad is removed.
     *
     * @generated from protobuf enum value: REMOVED = 4;
     */
    AdGroupAdStatusEnum_AdGroupAdStatus[AdGroupAdStatusEnum_AdGroupAdStatus["REMOVED"] = 4] = "REMOVED";
})(AdGroupAdStatusEnum_AdGroupAdStatus = exports.AdGroupAdStatusEnum_AdGroupAdStatus || (exports.AdGroupAdStatusEnum_AdGroupAdStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupAdStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdGroupAdStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdGroupAdStatusEnum
 */
exports.AdGroupAdStatusEnum = new AdGroupAdStatusEnum$Type();
