// google-ads-api-client/dist/protos/bindings/ads/enums/ad_group_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupTypeEnum = exports.AdGroupTypeEnum_AdGroupType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the possible types of an ad group.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdGroupTypeEnum.AdGroupType
 */
var AdGroupTypeEnum_AdGroupType;
(function (AdGroupTypeEnum_AdGroupType) {
    /**
     * The type has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The default ad group type for Search campaigns.
     *
     * @generated from protobuf enum value: SEARCH_STANDARD = 2;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["SEARCH_STANDARD"] = 2] = "SEARCH_STANDARD";
    /**
     * The default ad group type for Display campaigns.
     *
     * @generated from protobuf enum value: DISPLAY_STANDARD = 3;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["DISPLAY_STANDARD"] = 3] = "DISPLAY_STANDARD";
    /**
     * The ad group type for Shopping campaigns serving standard product ads.
     *
     * @generated from protobuf enum value: SHOPPING_PRODUCT_ADS = 4;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["SHOPPING_PRODUCT_ADS"] = 4] = "SHOPPING_PRODUCT_ADS";
    /**
     * The default ad group type for Hotel campaigns.
     *
     * @generated from protobuf enum value: HOTEL_ADS = 6;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["HOTEL_ADS"] = 6] = "HOTEL_ADS";
    /**
     * The type for ad groups in Smart Shopping campaigns.
     *
     * @generated from protobuf enum value: SHOPPING_SMART_ADS = 7;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["SHOPPING_SMART_ADS"] = 7] = "SHOPPING_SMART_ADS";
    /**
     * Short unskippable in-stream video ads.
     *
     * @generated from protobuf enum value: VIDEO_BUMPER = 8;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["VIDEO_BUMPER"] = 8] = "VIDEO_BUMPER";
    /**
     * TrueView (skippable) in-stream video ads.
     *
     * @generated from protobuf enum value: VIDEO_TRUE_VIEW_IN_STREAM = 9;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["VIDEO_TRUE_VIEW_IN_STREAM"] = 9] = "VIDEO_TRUE_VIEW_IN_STREAM";
    /**
     * TrueView in-display video ads.
     *
     * @generated from protobuf enum value: VIDEO_TRUE_VIEW_IN_DISPLAY = 10;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["VIDEO_TRUE_VIEW_IN_DISPLAY"] = 10] = "VIDEO_TRUE_VIEW_IN_DISPLAY";
    /**
     * Unskippable in-stream video ads.
     *
     * @generated from protobuf enum value: VIDEO_NON_SKIPPABLE_IN_STREAM = 11;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["VIDEO_NON_SKIPPABLE_IN_STREAM"] = 11] = "VIDEO_NON_SKIPPABLE_IN_STREAM";
    /**
     * Outstream video ads.
     *
     * @generated from protobuf enum value: VIDEO_OUTSTREAM = 12;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["VIDEO_OUTSTREAM"] = 12] = "VIDEO_OUTSTREAM";
    /**
     * Ad group type for Dynamic Search Ads ad groups.
     *
     * @generated from protobuf enum value: SEARCH_DYNAMIC_ADS = 13;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["SEARCH_DYNAMIC_ADS"] = 13] = "SEARCH_DYNAMIC_ADS";
    /**
     * The type for ad groups in Shopping Comparison Listing campaigns.
     *
     * @generated from protobuf enum value: SHOPPING_COMPARISON_LISTING_ADS = 14;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["SHOPPING_COMPARISON_LISTING_ADS"] = 14] = "SHOPPING_COMPARISON_LISTING_ADS";
    /**
     * The ad group type for Promoted Hotel ad groups.
     *
     * @generated from protobuf enum value: PROMOTED_HOTEL_ADS = 15;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["PROMOTED_HOTEL_ADS"] = 15] = "PROMOTED_HOTEL_ADS";
    /**
     * Video responsive ad groups.
     *
     * @generated from protobuf enum value: VIDEO_RESPONSIVE = 16;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["VIDEO_RESPONSIVE"] = 16] = "VIDEO_RESPONSIVE";
    /**
     * Video efficient reach ad groups.
     *
     * @generated from protobuf enum value: VIDEO_EFFICIENT_REACH = 17;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["VIDEO_EFFICIENT_REACH"] = 17] = "VIDEO_EFFICIENT_REACH";
    /**
     * Ad group type for Smart campaigns.
     *
     * @generated from protobuf enum value: SMART_CAMPAIGN_ADS = 18;
     */
    AdGroupTypeEnum_AdGroupType[AdGroupTypeEnum_AdGroupType["SMART_CAMPAIGN_ADS"] = 18] = "SMART_CAMPAIGN_ADS";
})(AdGroupTypeEnum_AdGroupType = exports.AdGroupTypeEnum_AdGroupType || (exports.AdGroupTypeEnum_AdGroupType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdGroupTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdGroupTypeEnum
 */
exports.AdGroupTypeEnum = new AdGroupTypeEnum$Type();
