// google-ads-api-client/dist/protos/bindings/ads/enums/ad_network_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdNetworkTypeEnum = exports.AdNetworkTypeEnum_AdNetworkType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates Google Ads network types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdNetworkTypeEnum.AdNetworkType
 */
var AdNetworkTypeEnum_AdNetworkType;
(function (AdNetworkTypeEnum_AdNetworkType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdNetworkTypeEnum_AdNetworkType[AdNetworkTypeEnum_AdNetworkType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdNetworkTypeEnum_AdNetworkType[AdNetworkTypeEnum_AdNetworkType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Google search.
     *
     * @generated from protobuf enum value: SEARCH = 2;
     */
    AdNetworkTypeEnum_AdNetworkType[AdNetworkTypeEnum_AdNetworkType["SEARCH"] = 2] = "SEARCH";
    /**
     * Search partners.
     *
     * @generated from protobuf enum value: SEARCH_PARTNERS = 3;
     */
    AdNetworkTypeEnum_AdNetworkType[AdNetworkTypeEnum_AdNetworkType["SEARCH_PARTNERS"] = 3] = "SEARCH_PARTNERS";
    /**
     * Display Network.
     *
     * @generated from protobuf enum value: CONTENT = 4;
     */
    AdNetworkTypeEnum_AdNetworkType[AdNetworkTypeEnum_AdNetworkType["CONTENT"] = 4] = "CONTENT";
    /**
     * YouTube Search.
     *
     * @generated from protobuf enum value: YOUTUBE_SEARCH = 5;
     */
    AdNetworkTypeEnum_AdNetworkType[AdNetworkTypeEnum_AdNetworkType["YOUTUBE_SEARCH"] = 5] = "YOUTUBE_SEARCH";
    /**
     * YouTube Videos
     *
     * @generated from protobuf enum value: YOUTUBE_WATCH = 6;
     */
    AdNetworkTypeEnum_AdNetworkType[AdNetworkTypeEnum_AdNetworkType["YOUTUBE_WATCH"] = 6] = "YOUTUBE_WATCH";
    /**
     * Cross-network.
     *
     * @generated from protobuf enum value: MIXED = 7;
     */
    AdNetworkTypeEnum_AdNetworkType[AdNetworkTypeEnum_AdNetworkType["MIXED"] = 7] = "MIXED";
})(AdNetworkTypeEnum_AdNetworkType = exports.AdNetworkTypeEnum_AdNetworkType || (exports.AdNetworkTypeEnum_AdNetworkType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdNetworkTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdNetworkTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdNetworkTypeEnum
 */
exports.AdNetworkTypeEnum = new AdNetworkTypeEnum$Type();
