// google-ads-api-client/dist/protos/bindings/ads/enums/ad_serving_optimization_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdServingOptimizationStatusEnum = exports.AdServingOptimizationStatusEnum_AdServingOptimizationStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible serving statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdServingOptimizationStatusEnum.AdServingOptimizationStatus
 */
var AdServingOptimizationStatusEnum_AdServingOptimizationStatus;
(function (AdServingOptimizationStatusEnum_AdServingOptimizationStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdServingOptimizationStatusEnum_AdServingOptimizationStatus[AdServingOptimizationStatusEnum_AdServingOptimizationStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdServingOptimizationStatusEnum_AdServingOptimizationStatus[AdServingOptimizationStatusEnum_AdServingOptimizationStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Ad serving is optimized based on CTR for the campaign.
     *
     * @generated from protobuf enum value: OPTIMIZE = 2;
     */
    AdServingOptimizationStatusEnum_AdServingOptimizationStatus[AdServingOptimizationStatusEnum_AdServingOptimizationStatus["OPTIMIZE"] = 2] = "OPTIMIZE";
    /**
     * Ad serving is optimized based on CTR * Conversion for the campaign. If
     * the campaign is not in the conversion optimizer bidding strategy, it will
     * default to OPTIMIZED.
     *
     * @generated from protobuf enum value: CONVERSION_OPTIMIZE = 3;
     */
    AdServingOptimizationStatusEnum_AdServingOptimizationStatus[AdServingOptimizationStatusEnum_AdServingOptimizationStatus["CONVERSION_OPTIMIZE"] = 3] = "CONVERSION_OPTIMIZE";
    /**
     * Ads are rotated evenly for 90 days, then optimized for clicks.
     *
     * @generated from protobuf enum value: ROTATE = 4;
     */
    AdServingOptimizationStatusEnum_AdServingOptimizationStatus[AdServingOptimizationStatusEnum_AdServingOptimizationStatus["ROTATE"] = 4] = "ROTATE";
    /**
     * Show lower performing ads more evenly with higher performing ads, and do
     * not optimize.
     *
     * @generated from protobuf enum value: ROTATE_INDEFINITELY = 5;
     */
    AdServingOptimizationStatusEnum_AdServingOptimizationStatus[AdServingOptimizationStatusEnum_AdServingOptimizationStatus["ROTATE_INDEFINITELY"] = 5] = "ROTATE_INDEFINITELY";
    /**
     * Ad serving optimization status is not available.
     *
     * @generated from protobuf enum value: UNAVAILABLE = 6;
     */
    AdServingOptimizationStatusEnum_AdServingOptimizationStatus[AdServingOptimizationStatusEnum_AdServingOptimizationStatus["UNAVAILABLE"] = 6] = "UNAVAILABLE";
})(AdServingOptimizationStatusEnum_AdServingOptimizationStatus = exports.AdServingOptimizationStatusEnum_AdServingOptimizationStatus || (exports.AdServingOptimizationStatusEnum_AdServingOptimizationStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdServingOptimizationStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdServingOptimizationStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdServingOptimizationStatusEnum
 */
exports.AdServingOptimizationStatusEnum = new AdServingOptimizationStatusEnum$Type();
