// google-ads-api-client/dist/protos/bindings/ads/enums/ad_strength.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdStrengthEnum = exports.AdStrengthEnum_AdStrength = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the possible ad strengths.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdStrengthEnum.AdStrength
 */
var AdStrengthEnum_AdStrength;
(function (AdStrengthEnum_AdStrength) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdStrengthEnum_AdStrength[AdStrengthEnum_AdStrength["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdStrengthEnum_AdStrength[AdStrengthEnum_AdStrength["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The ad strength is currently pending.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    AdStrengthEnum_AdStrength[AdStrengthEnum_AdStrength["PENDING"] = 2] = "PENDING";
    /**
     * No ads could be generated.
     *
     * @generated from protobuf enum value: NO_ADS = 3;
     */
    AdStrengthEnum_AdStrength[AdStrengthEnum_AdStrength["NO_ADS"] = 3] = "NO_ADS";
    /**
     * Poor strength.
     *
     * @generated from protobuf enum value: POOR = 4;
     */
    AdStrengthEnum_AdStrength[AdStrengthEnum_AdStrength["POOR"] = 4] = "POOR";
    /**
     * Average strength.
     *
     * @generated from protobuf enum value: AVERAGE = 5;
     */
    AdStrengthEnum_AdStrength[AdStrengthEnum_AdStrength["AVERAGE"] = 5] = "AVERAGE";
    /**
     * Good strength.
     *
     * @generated from protobuf enum value: GOOD = 6;
     */
    AdStrengthEnum_AdStrength[AdStrengthEnum_AdStrength["GOOD"] = 6] = "GOOD";
    /**
     * Excellent strength.
     *
     * @generated from protobuf enum value: EXCELLENT = 7;
     */
    AdStrengthEnum_AdStrength[AdStrengthEnum_AdStrength["EXCELLENT"] = 7] = "EXCELLENT";
})(AdStrengthEnum_AdStrength = exports.AdStrengthEnum_AdStrength || (exports.AdStrengthEnum_AdStrength = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdStrengthEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdStrengthEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdStrengthEnum
 */
exports.AdStrengthEnum = new AdStrengthEnum$Type();
