// google-ads-api-client/dist/protos/bindings/ads/enums/ad_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdTypeEnum = exports.AdTypeEnum_AdType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible types of an ad.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdTypeEnum.AdType
 */
var AdTypeEnum_AdType;
(function (AdTypeEnum_AdType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The ad is a text ad.
     *
     * @generated from protobuf enum value: TEXT_AD = 2;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["TEXT_AD"] = 2] = "TEXT_AD";
    /**
     * The ad is an expanded text ad.
     *
     * @generated from protobuf enum value: EXPANDED_TEXT_AD = 3;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["EXPANDED_TEXT_AD"] = 3] = "EXPANDED_TEXT_AD";
    /**
     * The ad is an expanded dynamic search ad.
     *
     * @generated from protobuf enum value: EXPANDED_DYNAMIC_SEARCH_AD = 7;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["EXPANDED_DYNAMIC_SEARCH_AD"] = 7] = "EXPANDED_DYNAMIC_SEARCH_AD";
    /**
     * The ad is a hotel ad.
     *
     * @generated from protobuf enum value: HOTEL_AD = 8;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["HOTEL_AD"] = 8] = "HOTEL_AD";
    /**
     * The ad is a Smart Shopping ad.
     *
     * @generated from protobuf enum value: SHOPPING_SMART_AD = 9;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["SHOPPING_SMART_AD"] = 9] = "SHOPPING_SMART_AD";
    /**
     * The ad is a standard Shopping ad.
     *
     * @generated from protobuf enum value: SHOPPING_PRODUCT_AD = 10;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["SHOPPING_PRODUCT_AD"] = 10] = "SHOPPING_PRODUCT_AD";
    /**
     * The ad is a video ad.
     *
     * @generated from protobuf enum value: VIDEO_AD = 12;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["VIDEO_AD"] = 12] = "VIDEO_AD";
    /**
     * This ad is a Gmail ad.
     *
     * @generated from protobuf enum value: GMAIL_AD = 13;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["GMAIL_AD"] = 13] = "GMAIL_AD";
    /**
     * This ad is an Image ad.
     *
     * @generated from protobuf enum value: IMAGE_AD = 14;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["IMAGE_AD"] = 14] = "IMAGE_AD";
    /**
     * The ad is a responsive search ad.
     *
     * @generated from protobuf enum value: RESPONSIVE_SEARCH_AD = 15;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["RESPONSIVE_SEARCH_AD"] = 15] = "RESPONSIVE_SEARCH_AD";
    /**
     * The ad is a legacy responsive display ad.
     *
     * @generated from protobuf enum value: LEGACY_RESPONSIVE_DISPLAY_AD = 16;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["LEGACY_RESPONSIVE_DISPLAY_AD"] = 16] = "LEGACY_RESPONSIVE_DISPLAY_AD";
    /**
     * The ad is an app ad.
     *
     * @generated from protobuf enum value: APP_AD = 17;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["APP_AD"] = 17] = "APP_AD";
    /**
     * The ad is a legacy app install ad.
     *
     * @generated from protobuf enum value: LEGACY_APP_INSTALL_AD = 18;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["LEGACY_APP_INSTALL_AD"] = 18] = "LEGACY_APP_INSTALL_AD";
    /**
     * The ad is a responsive display ad.
     *
     * @generated from protobuf enum value: RESPONSIVE_DISPLAY_AD = 19;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["RESPONSIVE_DISPLAY_AD"] = 19] = "RESPONSIVE_DISPLAY_AD";
    /**
     * The ad is a local ad.
     *
     * @generated from protobuf enum value: LOCAL_AD = 20;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["LOCAL_AD"] = 20] = "LOCAL_AD";
    /**
     * The ad is a display upload ad with the HTML5_UPLOAD_AD product type.
     *
     * @generated from protobuf enum value: HTML5_UPLOAD_AD = 21;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["HTML5_UPLOAD_AD"] = 21] = "HTML5_UPLOAD_AD";
    /**
     * The ad is a display upload ad with one of the DYNAMIC_HTML5_* product
     * types.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_AD = 22;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["DYNAMIC_HTML5_AD"] = 22] = "DYNAMIC_HTML5_AD";
    /**
     * The ad is an app engagement ad.
     *
     * @generated from protobuf enum value: APP_ENGAGEMENT_AD = 23;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["APP_ENGAGEMENT_AD"] = 23] = "APP_ENGAGEMENT_AD";
    /**
     * The ad is a Shopping Comparison Listing ad.
     *
     * @generated from protobuf enum value: SHOPPING_COMPARISON_LISTING_AD = 24;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["SHOPPING_COMPARISON_LISTING_AD"] = 24] = "SHOPPING_COMPARISON_LISTING_AD";
    /**
     * Video bumper ad.
     *
     * @generated from protobuf enum value: VIDEO_BUMPER_AD = 25;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["VIDEO_BUMPER_AD"] = 25] = "VIDEO_BUMPER_AD";
    /**
     * Video non-skippable in-stream ad.
     *
     * @generated from protobuf enum value: VIDEO_NON_SKIPPABLE_IN_STREAM_AD = 26;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["VIDEO_NON_SKIPPABLE_IN_STREAM_AD"] = 26] = "VIDEO_NON_SKIPPABLE_IN_STREAM_AD";
    /**
     * Video outstream ad.
     *
     * @generated from protobuf enum value: VIDEO_OUTSTREAM_AD = 27;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["VIDEO_OUTSTREAM_AD"] = 27] = "VIDEO_OUTSTREAM_AD";
    /**
     * Video TrueView in-stream ad.
     *
     * @generated from protobuf enum value: VIDEO_TRUEVIEW_IN_STREAM_AD = 29;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["VIDEO_TRUEVIEW_IN_STREAM_AD"] = 29] = "VIDEO_TRUEVIEW_IN_STREAM_AD";
    /**
     * Video responsive ad.
     *
     * @generated from protobuf enum value: VIDEO_RESPONSIVE_AD = 30;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["VIDEO_RESPONSIVE_AD"] = 30] = "VIDEO_RESPONSIVE_AD";
    /**
     * Smart campaign ad.
     *
     * @generated from protobuf enum value: SMART_CAMPAIGN_AD = 31;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["SMART_CAMPAIGN_AD"] = 31] = "SMART_CAMPAIGN_AD";
    /**
     * Call ad.
     *
     * @generated from protobuf enum value: CALL_AD = 32;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["CALL_AD"] = 32] = "CALL_AD";
    /**
     * Universal app pre-registration ad.
     *
     * @generated from protobuf enum value: APP_PRE_REGISTRATION_AD = 33;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["APP_PRE_REGISTRATION_AD"] = 33] = "APP_PRE_REGISTRATION_AD";
    /**
     * In-feed video ad.
     *
     * @generated from protobuf enum value: IN_FEED_VIDEO_AD = 34;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["IN_FEED_VIDEO_AD"] = 34] = "IN_FEED_VIDEO_AD";
    /**
     * Discovery multi asset ad.
     *
     * @generated from protobuf enum value: DISCOVERY_MULTI_ASSET_AD = 35;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["DISCOVERY_MULTI_ASSET_AD"] = 35] = "DISCOVERY_MULTI_ASSET_AD";
    /**
     * Discovery carousel ad.
     *
     * @generated from protobuf enum value: DISCOVERY_CAROUSEL_AD = 36;
     */
    AdTypeEnum_AdType[AdTypeEnum_AdType["DISCOVERY_CAROUSEL_AD"] = 36] = "DISCOVERY_CAROUSEL_AD";
})(AdTypeEnum_AdType = exports.AdTypeEnum_AdType || (exports.AdTypeEnum_AdType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdTypeEnum
 */
exports.AdTypeEnum = new AdTypeEnum$Type();
