// google-ads-api-client/dist/protos/bindings/ads/enums/advertising_channel_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvertisingChannelTypeEnum = exports.AdvertisingChannelTypeEnum_AdvertisingChannelType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the various advertising channel types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdvertisingChannelTypeEnum.AdvertisingChannelType
 */
var AdvertisingChannelTypeEnum_AdvertisingChannelType;
(function (AdvertisingChannelTypeEnum_AdvertisingChannelType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Search Network. Includes display bundled, and Search+ campaigns.
     *
     * @generated from protobuf enum value: SEARCH = 2;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["SEARCH"] = 2] = "SEARCH";
    /**
     * Google Display Network only.
     *
     * @generated from protobuf enum value: DISPLAY = 3;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["DISPLAY"] = 3] = "DISPLAY";
    /**
     * Shopping campaigns serve on the shopping property
     * and on google.com search results.
     *
     * @generated from protobuf enum value: SHOPPING = 4;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["SHOPPING"] = 4] = "SHOPPING";
    /**
     * Hotel Ads campaigns.
     *
     * @generated from protobuf enum value: HOTEL = 5;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["HOTEL"] = 5] = "HOTEL";
    /**
     * Video campaigns.
     *
     * @generated from protobuf enum value: VIDEO = 6;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["VIDEO"] = 6] = "VIDEO";
    /**
     * App Campaigns, and App Campaigns for Engagement, that run
     * across multiple channels.
     *
     * @generated from protobuf enum value: MULTI_CHANNEL = 7;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["MULTI_CHANNEL"] = 7] = "MULTI_CHANNEL";
    /**
     * Local ads campaigns.
     *
     * @generated from protobuf enum value: LOCAL = 8;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["LOCAL"] = 8] = "LOCAL";
    /**
     * Smart campaigns.
     *
     * @generated from protobuf enum value: SMART = 9;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["SMART"] = 9] = "SMART";
    /**
     * Performance Max campaigns.
     *
     * @generated from protobuf enum value: PERFORMANCE_MAX = 10;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["PERFORMANCE_MAX"] = 10] = "PERFORMANCE_MAX";
    /**
     * Local services campaigns.
     *
     * @generated from protobuf enum value: LOCAL_SERVICES = 11;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["LOCAL_SERVICES"] = 11] = "LOCAL_SERVICES";
    /**
     * Discovery campaigns.
     *
     * @generated from protobuf enum value: DISCOVERY = 12;
     */
    AdvertisingChannelTypeEnum_AdvertisingChannelType[AdvertisingChannelTypeEnum_AdvertisingChannelType["DISCOVERY"] = 12] = "DISCOVERY";
})(AdvertisingChannelTypeEnum_AdvertisingChannelType = exports.AdvertisingChannelTypeEnum_AdvertisingChannelType || (exports.AdvertisingChannelTypeEnum_AdvertisingChannelType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdvertisingChannelTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdvertisingChannelTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdvertisingChannelTypeEnum
 */
exports.AdvertisingChannelTypeEnum = new AdvertisingChannelTypeEnum$Type();
