// google-ads-api-client/dist/protos/bindings/ads/enums/advertising_channel_sub_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvertisingChannelSubTypeEnum = exports.AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the different channel subtypes.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AdvertisingChannelSubTypeEnum.AdvertisingChannelSubType
 */
var AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType;
(function (AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used as a return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Mobile app campaigns for Search.
     *
     * @generated from protobuf enum value: SEARCH_MOBILE_APP = 2;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["SEARCH_MOBILE_APP"] = 2] = "SEARCH_MOBILE_APP";
    /**
     * Mobile app campaigns for Display.
     *
     * @generated from protobuf enum value: DISPLAY_MOBILE_APP = 3;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["DISPLAY_MOBILE_APP"] = 3] = "DISPLAY_MOBILE_APP";
    /**
     * AdWords express campaigns for search.
     *
     * @generated from protobuf enum value: SEARCH_EXPRESS = 4;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["SEARCH_EXPRESS"] = 4] = "SEARCH_EXPRESS";
    /**
     * AdWords Express campaigns for display.
     *
     * @generated from protobuf enum value: DISPLAY_EXPRESS = 5;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["DISPLAY_EXPRESS"] = 5] = "DISPLAY_EXPRESS";
    /**
     * Smart Shopping campaigns.
     *
     * @generated from protobuf enum value: SHOPPING_SMART_ADS = 6;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["SHOPPING_SMART_ADS"] = 6] = "SHOPPING_SMART_ADS";
    /**
     * Gmail Ad campaigns.
     *
     * @generated from protobuf enum value: DISPLAY_GMAIL_AD = 7;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["DISPLAY_GMAIL_AD"] = 7] = "DISPLAY_GMAIL_AD";
    /**
     * Smart display campaigns.
     *
     * @generated from protobuf enum value: DISPLAY_SMART_CAMPAIGN = 8;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["DISPLAY_SMART_CAMPAIGN"] = 8] = "DISPLAY_SMART_CAMPAIGN";
    /**
     * Video Outstream campaigns.
     *
     * @generated from protobuf enum value: VIDEO_OUTSTREAM = 9;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["VIDEO_OUTSTREAM"] = 9] = "VIDEO_OUTSTREAM";
    /**
     * Video TrueView for Action campaigns.
     *
     * @generated from protobuf enum value: VIDEO_ACTION = 10;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["VIDEO_ACTION"] = 10] = "VIDEO_ACTION";
    /**
     * Video campaigns with non-skippable video ads.
     *
     * @generated from protobuf enum value: VIDEO_NON_SKIPPABLE = 11;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["VIDEO_NON_SKIPPABLE"] = 11] = "VIDEO_NON_SKIPPABLE";
    /**
     * App Campaign that lets you easily promote your Android or iOS app
     * across Google's top properties including Search, Play, YouTube, and the
     * Google Display Network.
     *
     * @generated from protobuf enum value: APP_CAMPAIGN = 12;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["APP_CAMPAIGN"] = 12] = "APP_CAMPAIGN";
    /**
     * App Campaign for engagement, focused on driving re-engagement with the
     * app across several of Google's top properties including Search, YouTube,
     * and the Google Display Network.
     *
     * @generated from protobuf enum value: APP_CAMPAIGN_FOR_ENGAGEMENT = 13;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["APP_CAMPAIGN_FOR_ENGAGEMENT"] = 13] = "APP_CAMPAIGN_FOR_ENGAGEMENT";
    /**
     * Campaigns specialized for local advertising.
     *
     * @generated from protobuf enum value: LOCAL_CAMPAIGN = 14;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["LOCAL_CAMPAIGN"] = 14] = "LOCAL_CAMPAIGN";
    /**
     * Shopping Comparison Listing campaigns.
     *
     * @generated from protobuf enum value: SHOPPING_COMPARISON_LISTING_ADS = 15;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["SHOPPING_COMPARISON_LISTING_ADS"] = 15] = "SHOPPING_COMPARISON_LISTING_ADS";
    /**
     * Standard Smart campaigns.
     *
     * @generated from protobuf enum value: SMART_CAMPAIGN = 16;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["SMART_CAMPAIGN"] = 16] = "SMART_CAMPAIGN";
    /**
     * Video campaigns with sequence video ads.
     *
     * @generated from protobuf enum value: VIDEO_SEQUENCE = 17;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["VIDEO_SEQUENCE"] = 17] = "VIDEO_SEQUENCE";
    /**
     * App Campaign for pre registration, specialized for advertising mobile
     * app pre-registration, that targets multiple advertising channels across
     * Google Play, YouTube and Display Network. See
     * https://support.google.com/google-ads/answer/9441344 to learn more.
     *
     * @generated from protobuf enum value: APP_CAMPAIGN_FOR_PRE_REGISTRATION = 18;
     */
    AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType[AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType["APP_CAMPAIGN_FOR_PRE_REGISTRATION"] = 18] = "APP_CAMPAIGN_FOR_PRE_REGISTRATION";
})(AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType = exports.AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType || (exports.AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdvertisingChannelSubTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AdvertisingChannelSubTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AdvertisingChannelSubTypeEnum
 */
exports.AdvertisingChannelSubTypeEnum = new AdvertisingChannelSubTypeEnum$Type();
