// google-ads-api-client/dist/protos/bindings/ads/enums/affiliate_location_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AffiliateLocationPlaceholderFieldEnum = exports.AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Affiliate Location placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AffiliateLocationPlaceholderFieldEnum.AffiliateLocationPlaceholderField
 */
var AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField;
(function (AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. The name of the business.
     *
     * @generated from protobuf enum value: BUSINESS_NAME = 2;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["BUSINESS_NAME"] = 2] = "BUSINESS_NAME";
    /**
     * Data Type: STRING. Line 1 of the business address.
     *
     * @generated from protobuf enum value: ADDRESS_LINE_1 = 3;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["ADDRESS_LINE_1"] = 3] = "ADDRESS_LINE_1";
    /**
     * Data Type: STRING. Line 2 of the business address.
     *
     * @generated from protobuf enum value: ADDRESS_LINE_2 = 4;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["ADDRESS_LINE_2"] = 4] = "ADDRESS_LINE_2";
    /**
     * Data Type: STRING. City of the business address.
     *
     * @generated from protobuf enum value: CITY = 5;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["CITY"] = 5] = "CITY";
    /**
     * Data Type: STRING. Province of the business address.
     *
     * @generated from protobuf enum value: PROVINCE = 6;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["PROVINCE"] = 6] = "PROVINCE";
    /**
     * Data Type: STRING. Postal code of the business address.
     *
     * @generated from protobuf enum value: POSTAL_CODE = 7;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["POSTAL_CODE"] = 7] = "POSTAL_CODE";
    /**
     * Data Type: STRING. Country code of the business address.
     *
     * @generated from protobuf enum value: COUNTRY_CODE = 8;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["COUNTRY_CODE"] = 8] = "COUNTRY_CODE";
    /**
     * Data Type: STRING. Phone number of the business.
     *
     * @generated from protobuf enum value: PHONE_NUMBER = 9;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["PHONE_NUMBER"] = 9] = "PHONE_NUMBER";
    /**
     * Data Type: STRING. Language code of the business.
     *
     * @generated from protobuf enum value: LANGUAGE_CODE = 10;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["LANGUAGE_CODE"] = 10] = "LANGUAGE_CODE";
    /**
     * Data Type: INT64. ID of the chain.
     *
     * @generated from protobuf enum value: CHAIN_ID = 11;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["CHAIN_ID"] = 11] = "CHAIN_ID";
    /**
     * Data Type: STRING. Name of the chain.
     *
     * @generated from protobuf enum value: CHAIN_NAME = 12;
     */
    AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField[AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField["CHAIN_NAME"] = 12] = "CHAIN_NAME";
})(AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField = exports.AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField || (exports.AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AffiliateLocationPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AffiliateLocationPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AffiliateLocationPlaceholderFieldEnum
 */
exports.AffiliateLocationPlaceholderFieldEnum = new AffiliateLocationPlaceholderFieldEnum$Type();
