// google-ads-api-client/dist/protos/bindings/ads/enums/affiliate_location_feed_relationship_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AffiliateLocationFeedRelationshipTypeEnum = exports.AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for a relationship type for an affiliate location feed.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AffiliateLocationFeedRelationshipTypeEnum.AffiliateLocationFeedRelationshipType
 */
var AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType;
(function (AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType[AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType[AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * General retailer relationship.
     *
     * @generated from protobuf enum value: GENERAL_RETAILER = 2;
     */
    AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType[AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType["GENERAL_RETAILER"] = 2] = "GENERAL_RETAILER";
})(AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType = exports.AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType || (exports.AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AffiliateLocationFeedRelationshipTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AffiliateLocationFeedRelationshipTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AffiliateLocationFeedRelationshipTypeEnum
 */
exports.AffiliateLocationFeedRelationshipTypeEnum = new AffiliateLocationFeedRelationshipTypeEnum$Type();
