// google-ads-api-client/dist/protos/bindings/ads/enums/age_range_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgeRangeTypeEnum = exports.AgeRangeTypeEnum_AgeRangeType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of demographic age ranges (for example, between 18 and 24 years
 * old).
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AgeRangeTypeEnum.AgeRangeType
 */
var AgeRangeTypeEnum_AgeRangeType;
(function (AgeRangeTypeEnum_AgeRangeType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Between 18 and 24 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_18_24 = 503001;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["AGE_RANGE_18_24"] = 503001] = "AGE_RANGE_18_24";
    /**
     * Between 25 and 34 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_25_34 = 503002;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["AGE_RANGE_25_34"] = 503002] = "AGE_RANGE_25_34";
    /**
     * Between 35 and 44 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_35_44 = 503003;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["AGE_RANGE_35_44"] = 503003] = "AGE_RANGE_35_44";
    /**
     * Between 45 and 54 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_45_54 = 503004;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["AGE_RANGE_45_54"] = 503004] = "AGE_RANGE_45_54";
    /**
     * Between 55 and 64 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_55_64 = 503005;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["AGE_RANGE_55_64"] = 503005] = "AGE_RANGE_55_64";
    /**
     * 65 years old and beyond.
     *
     * @generated from protobuf enum value: AGE_RANGE_65_UP = 503006;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["AGE_RANGE_65_UP"] = 503006] = "AGE_RANGE_65_UP";
    /**
     * Undetermined age range.
     *
     * @generated from protobuf enum value: AGE_RANGE_UNDETERMINED = 503999;
     */
    AgeRangeTypeEnum_AgeRangeType[AgeRangeTypeEnum_AgeRangeType["AGE_RANGE_UNDETERMINED"] = 503999] = "AGE_RANGE_UNDETERMINED";
})(AgeRangeTypeEnum_AgeRangeType = exports.AgeRangeTypeEnum_AgeRangeType || (exports.AgeRangeTypeEnum_AgeRangeType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AgeRangeTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AgeRangeTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AgeRangeTypeEnum
 */
exports.AgeRangeTypeEnum = new AgeRangeTypeEnum$Type();
