// google-ads-api-client/dist/protos/bindings/ads/enums/app_campaign_app_store.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCampaignAppStoreEnum = exports.AppCampaignAppStoreEnum_AppCampaignAppStore = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing app campaign app store.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AppCampaignAppStoreEnum.AppCampaignAppStore
 */
var AppCampaignAppStoreEnum_AppCampaignAppStore;
(function (AppCampaignAppStoreEnum_AppCampaignAppStore) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AppCampaignAppStoreEnum_AppCampaignAppStore[AppCampaignAppStoreEnum_AppCampaignAppStore["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AppCampaignAppStoreEnum_AppCampaignAppStore[AppCampaignAppStoreEnum_AppCampaignAppStore["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Apple app store.
     *
     * @generated from protobuf enum value: APPLE_APP_STORE = 2;
     */
    AppCampaignAppStoreEnum_AppCampaignAppStore[AppCampaignAppStoreEnum_AppCampaignAppStore["APPLE_APP_STORE"] = 2] = "APPLE_APP_STORE";
    /**
     * Google play.
     *
     * @generated from protobuf enum value: GOOGLE_APP_STORE = 3;
     */
    AppCampaignAppStoreEnum_AppCampaignAppStore[AppCampaignAppStoreEnum_AppCampaignAppStore["GOOGLE_APP_STORE"] = 3] = "GOOGLE_APP_STORE";
})(AppCampaignAppStoreEnum_AppCampaignAppStore = exports.AppCampaignAppStoreEnum_AppCampaignAppStore || (exports.AppCampaignAppStoreEnum_AppCampaignAppStore = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AppCampaignAppStoreEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AppCampaignAppStoreEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AppCampaignAppStoreEnum
 */
exports.AppCampaignAppStoreEnum = new AppCampaignAppStoreEnum$Type();
