// google-ads-api-client/dist/protos/bindings/ads/enums/app_store.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppStoreEnum = exports.AppStoreEnum_AppStore = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * App store type in an app extension.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AppStoreEnum.AppStore
 */
var AppStoreEnum_AppStore;
(function (AppStoreEnum_AppStore) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AppStoreEnum_AppStore[AppStoreEnum_AppStore["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AppStoreEnum_AppStore[AppStoreEnum_AppStore["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Apple iTunes.
     *
     * @generated from protobuf enum value: APPLE_ITUNES = 2;
     */
    AppStoreEnum_AppStore[AppStoreEnum_AppStore["APPLE_ITUNES"] = 2] = "APPLE_ITUNES";
    /**
     * Google Play.
     *
     * @generated from protobuf enum value: GOOGLE_PLAY = 3;
     */
    AppStoreEnum_AppStore[AppStoreEnum_AppStore["GOOGLE_PLAY"] = 3] = "GOOGLE_PLAY";
})(AppStoreEnum_AppStore = exports.AppStoreEnum_AppStore || (exports.AppStoreEnum_AppStore = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AppStoreEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AppStoreEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AppStoreEnum
 */
exports.AppStoreEnum = new AppStoreEnum$Type();
