// google-ads-api-client/dist/protos/bindings/ads/enums/app_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppPlaceholderFieldEnum = exports.AppPlaceholderFieldEnum_AppPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for App placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AppPlaceholderFieldEnum.AppPlaceholderField
 */
var AppPlaceholderFieldEnum_AppPlaceholderField;
(function (AppPlaceholderFieldEnum_AppPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: INT64. The application store that the target application
     * belongs to. Valid values are: 1 = Apple iTunes Store; 2 = Google Play
     * Store.
     *
     * @generated from protobuf enum value: STORE = 2;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["STORE"] = 2] = "STORE";
    /**
     * Data Type: STRING. The store-specific ID for the target application.
     *
     * @generated from protobuf enum value: ID = 3;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["ID"] = 3] = "ID";
    /**
     * Data Type: STRING. The visible text displayed when the link is rendered
     * in an ad.
     *
     * @generated from protobuf enum value: LINK_TEXT = 4;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["LINK_TEXT"] = 4] = "LINK_TEXT";
    /**
     * Data Type: STRING. The destination URL of the in-app link.
     *
     * @generated from protobuf enum value: URL = 5;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["URL"] = 5] = "URL";
    /**
     * Data Type: URL_LIST. Final URLs for the in-app link when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_URLS = 6;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["FINAL_URLS"] = 6] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final Mobile URLs for the in-app link when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 7;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["FINAL_MOBILE_URLS"] = 7] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the in-app link when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 8;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["TRACKING_URL"] = 8] = "TRACKING_URL";
    /**
     * Data Type: STRING. Final URL suffix for the in-app link when using
     * parallel tracking.
     *
     * @generated from protobuf enum value: FINAL_URL_SUFFIX = 9;
     */
    AppPlaceholderFieldEnum_AppPlaceholderField[AppPlaceholderFieldEnum_AppPlaceholderField["FINAL_URL_SUFFIX"] = 9] = "FINAL_URL_SUFFIX";
})(AppPlaceholderFieldEnum_AppPlaceholderField = exports.AppPlaceholderFieldEnum_AppPlaceholderField || (exports.AppPlaceholderFieldEnum_AppPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AppPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AppPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AppPlaceholderFieldEnum
 */
exports.AppPlaceholderFieldEnum = new AppPlaceholderFieldEnum$Type();
