// google-ads-api-client/dist/protos/bindings/ads/enums/app_url_operating_system_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppUrlOperatingSystemTypeEnum = exports.AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Operating System
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AppUrlOperatingSystemTypeEnum.AppUrlOperatingSystemType
 */
var AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType;
(function (AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType[AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType[AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The Apple IOS operating system.
     *
     * @generated from protobuf enum value: IOS = 2;
     */
    AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType[AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType["IOS"] = 2] = "IOS";
    /**
     * The Android operating system.
     *
     * @generated from protobuf enum value: ANDROID = 3;
     */
    AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType[AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType["ANDROID"] = 3] = "ANDROID";
})(AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType = exports.AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType || (exports.AppUrlOperatingSystemTypeEnum_AppUrlOperatingSystemType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AppUrlOperatingSystemTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AppUrlOperatingSystemTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AppUrlOperatingSystemTypeEnum
 */
exports.AppUrlOperatingSystemTypeEnum = new AppUrlOperatingSystemTypeEnum$Type();
