// google-ads-api-client/dist/protos/bindings/ads/enums/app_campaign_bidding_strategy_goal_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCampaignBiddingStrategyGoalTypeEnum = exports.AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Goal type of App campaign BiddingStrategy.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AppCampaignBiddingStrategyGoalTypeEnum.AppCampaignBiddingStrategyGoalType
 */
var AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType;
(function (AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType[AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType[AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Aim to maximize the number of app installs. The cpa bid is the
     * target cost per install.
     *
     * @generated from protobuf enum value: OPTIMIZE_INSTALLS_TARGET_INSTALL_COST = 2;
     */
    AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType[AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType["OPTIMIZE_INSTALLS_TARGET_INSTALL_COST"] = 2] = "OPTIMIZE_INSTALLS_TARGET_INSTALL_COST";
    /**
     * Aim to maximize the long term number of selected in-app conversions from
     * app installs. The cpa bid is the target cost per install.
     *
     * @generated from protobuf enum value: OPTIMIZE_IN_APP_CONVERSIONS_TARGET_INSTALL_COST = 3;
     */
    AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType[AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType["OPTIMIZE_IN_APP_CONVERSIONS_TARGET_INSTALL_COST"] = 3] = "OPTIMIZE_IN_APP_CONVERSIONS_TARGET_INSTALL_COST";
    /**
     * Aim to maximize the long term number of selected in-app conversions from
     * app installs. The cpa bid is the target cost per in-app conversion. Note
     * that the actual cpa may seem higher than the target cpa at first, since
     * the long term conversions haven't happened yet.
     *
     * @generated from protobuf enum value: OPTIMIZE_IN_APP_CONVERSIONS_TARGET_CONVERSION_COST = 4;
     */
    AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType[AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType["OPTIMIZE_IN_APP_CONVERSIONS_TARGET_CONVERSION_COST"] = 4] = "OPTIMIZE_IN_APP_CONVERSIONS_TARGET_CONVERSION_COST";
    /**
     * Aim to maximize all conversions' value, for example, install + selected
     * in-app conversions while achieving or exceeding target return on
     * advertising spend.
     *
     * @generated from protobuf enum value: OPTIMIZE_RETURN_ON_ADVERTISING_SPEND = 5;
     */
    AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType[AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType["OPTIMIZE_RETURN_ON_ADVERTISING_SPEND"] = 5] = "OPTIMIZE_RETURN_ON_ADVERTISING_SPEND";
    /**
     * Aim to maximize the pre-registration of the app.
     *
     * @generated from protobuf enum value: OPTIMIZE_PRE_REGISTRATION_CONVERSION_VOLUME = 6;
     */
    AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType[AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType["OPTIMIZE_PRE_REGISTRATION_CONVERSION_VOLUME"] = 6] = "OPTIMIZE_PRE_REGISTRATION_CONVERSION_VOLUME";
    /**
     * Aim to maximize installation of the app without target cost-per-install.
     *
     * @generated from protobuf enum value: OPTIMIZE_INSTALLS_WITHOUT_TARGET_INSTALL_COST = 7;
     */
    AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType[AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType["OPTIMIZE_INSTALLS_WITHOUT_TARGET_INSTALL_COST"] = 7] = "OPTIMIZE_INSTALLS_WITHOUT_TARGET_INSTALL_COST";
})(AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType = exports.AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType || (exports.AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AppCampaignBiddingStrategyGoalTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AppCampaignBiddingStrategyGoalTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AppCampaignBiddingStrategyGoalTypeEnum
 */
exports.AppCampaignBiddingStrategyGoalTypeEnum = new AppCampaignBiddingStrategyGoalTypeEnum$Type();
