// google-ads-api-client/dist/protos/bindings/ads/enums/app_payment_model_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppPaymentModelTypeEnum = exports.AppPaymentModelTypeEnum_AppPaymentModelType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible app payment models.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AppPaymentModelTypeEnum.AppPaymentModelType
 */
var AppPaymentModelTypeEnum_AppPaymentModelType;
(function (AppPaymentModelTypeEnum_AppPaymentModelType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AppPaymentModelTypeEnum_AppPaymentModelType[AppPaymentModelTypeEnum_AppPaymentModelType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AppPaymentModelTypeEnum_AppPaymentModelType[AppPaymentModelTypeEnum_AppPaymentModelType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Represents paid-for apps.
     *
     * @generated from protobuf enum value: PAID = 30;
     */
    AppPaymentModelTypeEnum_AppPaymentModelType[AppPaymentModelTypeEnum_AppPaymentModelType["PAID"] = 30] = "PAID";
})(AppPaymentModelTypeEnum_AppPaymentModelType = exports.AppPaymentModelTypeEnum_AppPaymentModelType || (exports.AppPaymentModelTypeEnum_AppPaymentModelType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AppPaymentModelTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AppPaymentModelTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AppPaymentModelTypeEnum
 */
exports.AppPaymentModelTypeEnum = new AppPaymentModelTypeEnum$Type();
