// google-ads-api-client/dist/protos/bindings/ads/enums/asset_link_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetLinkStatusEnum = exports.AssetLinkStatusEnum_AssetLinkStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing statuses of an asset link.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetLinkStatusEnum.AssetLinkStatus
 */
var AssetLinkStatusEnum_AssetLinkStatus;
(function (AssetLinkStatusEnum_AssetLinkStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetLinkStatusEnum_AssetLinkStatus[AssetLinkStatusEnum_AssetLinkStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetLinkStatusEnum_AssetLinkStatus[AssetLinkStatusEnum_AssetLinkStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Asset link is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    AssetLinkStatusEnum_AssetLinkStatus[AssetLinkStatusEnum_AssetLinkStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Asset link has been removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    AssetLinkStatusEnum_AssetLinkStatus[AssetLinkStatusEnum_AssetLinkStatus["REMOVED"] = 3] = "REMOVED";
    /**
     * Asset link is paused.
     *
     * @generated from protobuf enum value: PAUSED = 4;
     */
    AssetLinkStatusEnum_AssetLinkStatus[AssetLinkStatusEnum_AssetLinkStatus["PAUSED"] = 4] = "PAUSED";
})(AssetLinkStatusEnum_AssetLinkStatus = exports.AssetLinkStatusEnum_AssetLinkStatus || (exports.AssetLinkStatusEnum_AssetLinkStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetLinkStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetLinkStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetLinkStatusEnum
 */
exports.AssetLinkStatusEnum = new AssetLinkStatusEnum$Type();
