// google-ads-api-client/dist/protos/bindings/ads/enums/asset_group_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetGroupStatusEnum = exports.AssetGroupStatusEnum_AssetGroupStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of an asset group.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetGroupStatusEnum.AssetGroupStatus
 */
var AssetGroupStatusEnum_AssetGroupStatus;
(function (AssetGroupStatusEnum_AssetGroupStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetGroupStatusEnum_AssetGroupStatus[AssetGroupStatusEnum_AssetGroupStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetGroupStatusEnum_AssetGroupStatus[AssetGroupStatusEnum_AssetGroupStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The asset group is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    AssetGroupStatusEnum_AssetGroupStatus[AssetGroupStatusEnum_AssetGroupStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The asset group is paused.
     *
     * @generated from protobuf enum value: PAUSED = 3;
     */
    AssetGroupStatusEnum_AssetGroupStatus[AssetGroupStatusEnum_AssetGroupStatus["PAUSED"] = 3] = "PAUSED";
    /**
     * The asset group is removed.
     *
     * @generated from protobuf enum value: REMOVED = 4;
     */
    AssetGroupStatusEnum_AssetGroupStatus[AssetGroupStatusEnum_AssetGroupStatus["REMOVED"] = 4] = "REMOVED";
})(AssetGroupStatusEnum_AssetGroupStatus = exports.AssetGroupStatusEnum_AssetGroupStatus || (exports.AssetGroupStatusEnum_AssetGroupStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetGroupStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetGroupStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetGroupStatusEnum
 */
exports.AssetGroupStatusEnum = new AssetGroupStatusEnum$Type();
