// google-ads-api-client/dist/protos/bindings/ads/enums/asset_performance_label.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetPerformanceLabelEnum = exports.AssetPerformanceLabelEnum_AssetPerformanceLabel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the possible performance labels of an asset, usually
 * computed in the context of a linkage.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetPerformanceLabelEnum.AssetPerformanceLabel
 */
var AssetPerformanceLabelEnum_AssetPerformanceLabel;
(function (AssetPerformanceLabelEnum_AssetPerformanceLabel) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetPerformanceLabelEnum_AssetPerformanceLabel[AssetPerformanceLabelEnum_AssetPerformanceLabel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetPerformanceLabelEnum_AssetPerformanceLabel[AssetPerformanceLabelEnum_AssetPerformanceLabel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * This asset does not yet have any performance informantion. This may be
     * because it is still under review.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    AssetPerformanceLabelEnum_AssetPerformanceLabel[AssetPerformanceLabelEnum_AssetPerformanceLabel["PENDING"] = 2] = "PENDING";
    /**
     * The asset has started getting impressions but the stats are not
     * statistically significant enough to get an asset performance label.
     *
     * @generated from protobuf enum value: LEARNING = 3;
     */
    AssetPerformanceLabelEnum_AssetPerformanceLabel[AssetPerformanceLabelEnum_AssetPerformanceLabel["LEARNING"] = 3] = "LEARNING";
    /**
     * Worst performing assets.
     *
     * @generated from protobuf enum value: LOW = 4;
     */
    AssetPerformanceLabelEnum_AssetPerformanceLabel[AssetPerformanceLabelEnum_AssetPerformanceLabel["LOW"] = 4] = "LOW";
    /**
     * Good performing assets.
     *
     * @generated from protobuf enum value: GOOD = 5;
     */
    AssetPerformanceLabelEnum_AssetPerformanceLabel[AssetPerformanceLabelEnum_AssetPerformanceLabel["GOOD"] = 5] = "GOOD";
    /**
     * Best performing assets.
     *
     * @generated from protobuf enum value: BEST = 6;
     */
    AssetPerformanceLabelEnum_AssetPerformanceLabel[AssetPerformanceLabelEnum_AssetPerformanceLabel["BEST"] = 6] = "BEST";
})(AssetPerformanceLabelEnum_AssetPerformanceLabel = exports.AssetPerformanceLabelEnum_AssetPerformanceLabel || (exports.AssetPerformanceLabelEnum_AssetPerformanceLabel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetPerformanceLabelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetPerformanceLabelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetPerformanceLabelEnum
 */
exports.AssetPerformanceLabelEnum = new AssetPerformanceLabelEnum$Type();
