// google-ads-api-client/dist/protos/bindings/ads/enums/asset_set_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSetTypeEnum = exports.AssetSetTypeEnum_AssetSetType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible types of an asset set.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetSetTypeEnum.AssetSetType
 */
var AssetSetTypeEnum_AssetSetType;
(function (AssetSetTypeEnum_AssetSetType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Page asset set.
     *
     * @generated from protobuf enum value: PAGE_FEED = 2;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["PAGE_FEED"] = 2] = "PAGE_FEED";
    /**
     * Dynamic education asset set.
     *
     * @generated from protobuf enum value: DYNAMIC_EDUCATION = 3;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["DYNAMIC_EDUCATION"] = 3] = "DYNAMIC_EDUCATION";
    /**
     * Google Merchant Center asset set.
     *
     * @generated from protobuf enum value: MERCHANT_CENTER_FEED = 4;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["MERCHANT_CENTER_FEED"] = 4] = "MERCHANT_CENTER_FEED";
    /**
     * Dynamic real estate asset set.
     *
     * @generated from protobuf enum value: DYNAMIC_REAL_ESTATE = 5;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["DYNAMIC_REAL_ESTATE"] = 5] = "DYNAMIC_REAL_ESTATE";
    /**
     * Dynamic custom asset set.
     *
     * @generated from protobuf enum value: DYNAMIC_CUSTOM = 6;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["DYNAMIC_CUSTOM"] = 6] = "DYNAMIC_CUSTOM";
    /**
     * Dynamic hotels and rentals asset set.
     *
     * @generated from protobuf enum value: DYNAMIC_HOTELS_AND_RENTALS = 7;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["DYNAMIC_HOTELS_AND_RENTALS"] = 7] = "DYNAMIC_HOTELS_AND_RENTALS";
    /**
     * Dynamic flights asset set.
     *
     * @generated from protobuf enum value: DYNAMIC_FLIGHTS = 8;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["DYNAMIC_FLIGHTS"] = 8] = "DYNAMIC_FLIGHTS";
    /**
     * Dynamic travel asset set.
     *
     * @generated from protobuf enum value: DYNAMIC_TRAVEL = 9;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["DYNAMIC_TRAVEL"] = 9] = "DYNAMIC_TRAVEL";
    /**
     * Dynamic local asset set.
     *
     * @generated from protobuf enum value: DYNAMIC_LOCAL = 10;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["DYNAMIC_LOCAL"] = 10] = "DYNAMIC_LOCAL";
    /**
     * Dynamic jobs asset set.
     *
     * @generated from protobuf enum value: DYNAMIC_JOBS = 11;
     */
    AssetSetTypeEnum_AssetSetType[AssetSetTypeEnum_AssetSetType["DYNAMIC_JOBS"] = 11] = "DYNAMIC_JOBS";
})(AssetSetTypeEnum_AssetSetType = exports.AssetSetTypeEnum_AssetSetType || (exports.AssetSetTypeEnum_AssetSetType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetSetTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetSetTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetSetTypeEnum
 */
exports.AssetSetTypeEnum = new AssetSetTypeEnum$Type();
