// google-ads-api-client/dist/protos/bindings/ads/enums/asset_set_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSetStatusEnum = exports.AssetSetStatusEnum_AssetSetStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of an asset set.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetSetStatusEnum.AssetSetStatus
 */
var AssetSetStatusEnum_AssetSetStatus;
(function (AssetSetStatusEnum_AssetSetStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetSetStatusEnum_AssetSetStatus[AssetSetStatusEnum_AssetSetStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetSetStatusEnum_AssetSetStatus[AssetSetStatusEnum_AssetSetStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The asset set is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    AssetSetStatusEnum_AssetSetStatus[AssetSetStatusEnum_AssetSetStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The asset set is removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    AssetSetStatusEnum_AssetSetStatus[AssetSetStatusEnum_AssetSetStatus["REMOVED"] = 3] = "REMOVED";
})(AssetSetStatusEnum_AssetSetStatus = exports.AssetSetStatusEnum_AssetSetStatus || (exports.AssetSetStatusEnum_AssetSetStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetSetStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetSetStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetSetStatusEnum
 */
exports.AssetSetStatusEnum = new AssetSetStatusEnum$Type();
