// google-ads-api-client/dist/protos/bindings/ads/enums/asset_source.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSourceEnum = exports.AssetSourceEnum_AssetSource = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible source of asset.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetSourceEnum.AssetSource
 */
var AssetSourceEnum_AssetSource;
(function (AssetSourceEnum_AssetSource) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetSourceEnum_AssetSource[AssetSourceEnum_AssetSource["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetSourceEnum_AssetSource[AssetSourceEnum_AssetSource["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The asset or asset link is provided by advertiser.
     *
     * @generated from protobuf enum value: ADVERTISER = 2;
     */
    AssetSourceEnum_AssetSource[AssetSourceEnum_AssetSource["ADVERTISER"] = 2] = "ADVERTISER";
    /**
     * The asset or asset link is generated by Google.
     *
     * @generated from protobuf enum value: AUTOMATICALLY_CREATED = 3;
     */
    AssetSourceEnum_AssetSource[AssetSourceEnum_AssetSource["AUTOMATICALLY_CREATED"] = 3] = "AUTOMATICALLY_CREATED";
})(AssetSourceEnum_AssetSource = exports.AssetSourceEnum_AssetSource || (exports.AssetSourceEnum_AssetSource = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetSourceEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetSourceEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetSourceEnum
 */
exports.AssetSourceEnum = new AssetSourceEnum$Type();
