// google-ads-api-client/dist/protos/bindings/ads/enums/asset_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetTypeEnum = exports.AssetTypeEnum_AssetType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible types of asset.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetTypeEnum.AssetType
 */
var AssetTypeEnum_AssetType;
(function (AssetTypeEnum_AssetType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * YouTube video asset.
     *
     * @generated from protobuf enum value: YOUTUBE_VIDEO = 2;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["YOUTUBE_VIDEO"] = 2] = "YOUTUBE_VIDEO";
    /**
     * Media bundle asset.
     *
     * @generated from protobuf enum value: MEDIA_BUNDLE = 3;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["MEDIA_BUNDLE"] = 3] = "MEDIA_BUNDLE";
    /**
     * Image asset.
     *
     * @generated from protobuf enum value: IMAGE = 4;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["IMAGE"] = 4] = "IMAGE";
    /**
     * Text asset.
     *
     * @generated from protobuf enum value: TEXT = 5;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["TEXT"] = 5] = "TEXT";
    /**
     * Lead form asset.
     *
     * @generated from protobuf enum value: LEAD_FORM = 6;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["LEAD_FORM"] = 6] = "LEAD_FORM";
    /**
     * Book on Google asset.
     *
     * @generated from protobuf enum value: BOOK_ON_GOOGLE = 7;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["BOOK_ON_GOOGLE"] = 7] = "BOOK_ON_GOOGLE";
    /**
     * Promotion asset.
     *
     * @generated from protobuf enum value: PROMOTION = 8;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["PROMOTION"] = 8] = "PROMOTION";
    /**
     * Callout asset.
     *
     * @generated from protobuf enum value: CALLOUT = 9;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["CALLOUT"] = 9] = "CALLOUT";
    /**
     * Structured Snippet asset.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPET = 10;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["STRUCTURED_SNIPPET"] = 10] = "STRUCTURED_SNIPPET";
    /**
     * Sitelink asset.
     *
     * @generated from protobuf enum value: SITELINK = 11;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["SITELINK"] = 11] = "SITELINK";
    /**
     * Page Feed asset.
     *
     * @generated from protobuf enum value: PAGE_FEED = 12;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["PAGE_FEED"] = 12] = "PAGE_FEED";
    /**
     * Dynamic Education asset.
     *
     * @generated from protobuf enum value: DYNAMIC_EDUCATION = 13;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DYNAMIC_EDUCATION"] = 13] = "DYNAMIC_EDUCATION";
    /**
     * Mobile app asset.
     *
     * @generated from protobuf enum value: MOBILE_APP = 14;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["MOBILE_APP"] = 14] = "MOBILE_APP";
    /**
     * Hotel callout asset.
     *
     * @generated from protobuf enum value: HOTEL_CALLOUT = 15;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["HOTEL_CALLOUT"] = 15] = "HOTEL_CALLOUT";
    /**
     * Call asset.
     *
     * @generated from protobuf enum value: CALL = 16;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["CALL"] = 16] = "CALL";
    /**
     * Price asset.
     *
     * @generated from protobuf enum value: PRICE = 17;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["PRICE"] = 17] = "PRICE";
    /**
     * Call to action asset.
     *
     * @generated from protobuf enum value: CALL_TO_ACTION = 18;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["CALL_TO_ACTION"] = 18] = "CALL_TO_ACTION";
    /**
     * Dynamic real estate asset.
     *
     * @generated from protobuf enum value: DYNAMIC_REAL_ESTATE = 19;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DYNAMIC_REAL_ESTATE"] = 19] = "DYNAMIC_REAL_ESTATE";
    /**
     * Dynamic custom asset.
     *
     * @generated from protobuf enum value: DYNAMIC_CUSTOM = 20;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DYNAMIC_CUSTOM"] = 20] = "DYNAMIC_CUSTOM";
    /**
     * Dynamic hotels and rentals asset.
     *
     * @generated from protobuf enum value: DYNAMIC_HOTELS_AND_RENTALS = 21;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DYNAMIC_HOTELS_AND_RENTALS"] = 21] = "DYNAMIC_HOTELS_AND_RENTALS";
    /**
     * Dynamic flights asset.
     *
     * @generated from protobuf enum value: DYNAMIC_FLIGHTS = 22;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DYNAMIC_FLIGHTS"] = 22] = "DYNAMIC_FLIGHTS";
    /**
     * Discovery Carousel Card asset.
     *
     * @generated from protobuf enum value: DISCOVERY_CAROUSEL_CARD = 23;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DISCOVERY_CAROUSEL_CARD"] = 23] = "DISCOVERY_CAROUSEL_CARD";
    /**
     * Dynamic travel asset.
     *
     * @generated from protobuf enum value: DYNAMIC_TRAVEL = 24;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DYNAMIC_TRAVEL"] = 24] = "DYNAMIC_TRAVEL";
    /**
     * Dynamic local asset.
     *
     * @generated from protobuf enum value: DYNAMIC_LOCAL = 25;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DYNAMIC_LOCAL"] = 25] = "DYNAMIC_LOCAL";
    /**
     * Dynamic jobs asset.
     *
     * @generated from protobuf enum value: DYNAMIC_JOBS = 26;
     */
    AssetTypeEnum_AssetType[AssetTypeEnum_AssetType["DYNAMIC_JOBS"] = 26] = "DYNAMIC_JOBS";
})(AssetTypeEnum_AssetType = exports.AssetTypeEnum_AssetType || (exports.AssetTypeEnum_AssetType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetTypeEnum
 */
exports.AssetTypeEnum = new AssetTypeEnum$Type();
