// google-ads-api-client/dist/protos/bindings/ads/enums/asset_field_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetFieldTypeEnum = exports.AssetFieldTypeEnum_AssetFieldType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the possible placements of an asset.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetFieldTypeEnum.AssetFieldType
 */
var AssetFieldTypeEnum_AssetFieldType;
(function (AssetFieldTypeEnum_AssetFieldType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The asset is linked for use as a headline.
     *
     * @generated from protobuf enum value: HEADLINE = 2;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["HEADLINE"] = 2] = "HEADLINE";
    /**
     * The asset is linked for use as a description.
     *
     * @generated from protobuf enum value: DESCRIPTION = 3;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["DESCRIPTION"] = 3] = "DESCRIPTION";
    /**
     * The asset is linked for use as mandatory ad text.
     *
     * @generated from protobuf enum value: MANDATORY_AD_TEXT = 4;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["MANDATORY_AD_TEXT"] = 4] = "MANDATORY_AD_TEXT";
    /**
     * The asset is linked for use as a marketing image.
     *
     * @generated from protobuf enum value: MARKETING_IMAGE = 5;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["MARKETING_IMAGE"] = 5] = "MARKETING_IMAGE";
    /**
     * The asset is linked for use as a media bundle.
     *
     * @generated from protobuf enum value: MEDIA_BUNDLE = 6;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["MEDIA_BUNDLE"] = 6] = "MEDIA_BUNDLE";
    /**
     * The asset is linked for use as a YouTube video.
     *
     * @generated from protobuf enum value: YOUTUBE_VIDEO = 7;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["YOUTUBE_VIDEO"] = 7] = "YOUTUBE_VIDEO";
    /**
     * The asset is linked to indicate that a hotels campaign is "Book on
     * Google" enabled.
     *
     * @generated from protobuf enum value: BOOK_ON_GOOGLE = 8;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["BOOK_ON_GOOGLE"] = 8] = "BOOK_ON_GOOGLE";
    /**
     * The asset is linked for use as a Lead Form extension.
     *
     * @generated from protobuf enum value: LEAD_FORM = 9;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["LEAD_FORM"] = 9] = "LEAD_FORM";
    /**
     * The asset is linked for use as a Promotion extension.
     *
     * @generated from protobuf enum value: PROMOTION = 10;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["PROMOTION"] = 10] = "PROMOTION";
    /**
     * The asset is linked for use as a Callout extension.
     *
     * @generated from protobuf enum value: CALLOUT = 11;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["CALLOUT"] = 11] = "CALLOUT";
    /**
     * The asset is linked for use as a Structured Snippet extension.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPET = 12;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["STRUCTURED_SNIPPET"] = 12] = "STRUCTURED_SNIPPET";
    /**
     * The asset is linked for use as a Sitelink extension.
     *
     * @generated from protobuf enum value: SITELINK = 13;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["SITELINK"] = 13] = "SITELINK";
    /**
     * The asset is linked for use as a Mobile App extension.
     *
     * @generated from protobuf enum value: MOBILE_APP = 14;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["MOBILE_APP"] = 14] = "MOBILE_APP";
    /**
     * The asset is linked for use as a Hotel Callout extension.
     *
     * @generated from protobuf enum value: HOTEL_CALLOUT = 15;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["HOTEL_CALLOUT"] = 15] = "HOTEL_CALLOUT";
    /**
     * The asset is linked for use as a Call extension.
     *
     * @generated from protobuf enum value: CALL = 16;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["CALL"] = 16] = "CALL";
    /**
     * The asset is linked for use as a Price extension.
     *
     * @generated from protobuf enum value: PRICE = 24;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["PRICE"] = 24] = "PRICE";
    /**
     * The asset is linked for use as a long headline.
     *
     * @generated from protobuf enum value: LONG_HEADLINE = 17;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["LONG_HEADLINE"] = 17] = "LONG_HEADLINE";
    /**
     * The asset is linked for use as a business name.
     *
     * @generated from protobuf enum value: BUSINESS_NAME = 18;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["BUSINESS_NAME"] = 18] = "BUSINESS_NAME";
    /**
     * The asset is linked for use as a square marketing image.
     *
     * @generated from protobuf enum value: SQUARE_MARKETING_IMAGE = 19;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["SQUARE_MARKETING_IMAGE"] = 19] = "SQUARE_MARKETING_IMAGE";
    /**
     * The asset is linked for use as a portrait marketing image.
     *
     * @generated from protobuf enum value: PORTRAIT_MARKETING_IMAGE = 20;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["PORTRAIT_MARKETING_IMAGE"] = 20] = "PORTRAIT_MARKETING_IMAGE";
    /**
     * The asset is linked for use as a logo.
     *
     * @generated from protobuf enum value: LOGO = 21;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["LOGO"] = 21] = "LOGO";
    /**
     * The asset is linked for use as a landscape logo.
     *
     * @generated from protobuf enum value: LANDSCAPE_LOGO = 22;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["LANDSCAPE_LOGO"] = 22] = "LANDSCAPE_LOGO";
    /**
     * The asset is linked for use as a non YouTube logo.
     *
     * @generated from protobuf enum value: VIDEO = 23;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["VIDEO"] = 23] = "VIDEO";
    /**
     * The asset is linked for use to select a call-to-action.
     *
     * @generated from protobuf enum value: CALL_TO_ACTION_SELECTION = 25;
     */
    AssetFieldTypeEnum_AssetFieldType[AssetFieldTypeEnum_AssetFieldType["CALL_TO_ACTION_SELECTION"] = 25] = "CALL_TO_ACTION_SELECTION";
})(AssetFieldTypeEnum_AssetFieldType = exports.AssetFieldTypeEnum_AssetFieldType || (exports.AssetFieldTypeEnum_AssetFieldType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetFieldTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetFieldTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetFieldTypeEnum
 */
exports.AssetFieldTypeEnum = new AssetFieldTypeEnum$Type();
