// google-ads-api-client/dist/protos/bindings/ads/enums/asset_set_link_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSetLinkStatusEnum = exports.AssetSetLinkStatusEnum_AssetSetLinkStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of the linkage between asset set and its container.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AssetSetLinkStatusEnum.AssetSetLinkStatus
 */
var AssetSetLinkStatusEnum_AssetSetLinkStatus;
(function (AssetSetLinkStatusEnum_AssetSetLinkStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetSetLinkStatusEnum_AssetSetLinkStatus[AssetSetLinkStatusEnum_AssetSetLinkStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetSetLinkStatusEnum_AssetSetLinkStatus[AssetSetLinkStatusEnum_AssetSetLinkStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The linkage between asset set and its container is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    AssetSetLinkStatusEnum_AssetSetLinkStatus[AssetSetLinkStatusEnum_AssetSetLinkStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The linkage between asset set and its container is removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    AssetSetLinkStatusEnum_AssetSetLinkStatus[AssetSetLinkStatusEnum_AssetSetLinkStatus["REMOVED"] = 3] = "REMOVED";
})(AssetSetLinkStatusEnum_AssetSetLinkStatus = exports.AssetSetLinkStatusEnum_AssetSetLinkStatus || (exports.AssetSetLinkStatusEnum_AssetSetLinkStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetSetLinkStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AssetSetLinkStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AssetSetLinkStatusEnum
 */
exports.AssetSetLinkStatusEnum = new AssetSetLinkStatusEnum$Type();
