// google-ads-api-client/dist/protos/bindings/ads/enums/async_action_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AsyncActionStatusEnum = exports.AsyncActionStatusEnum_AsyncActionStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The async action status of the experiment.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AsyncActionStatusEnum.AsyncActionStatus
 */
var AsyncActionStatusEnum_AsyncActionStatus;
(function (AsyncActionStatusEnum_AsyncActionStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AsyncActionStatusEnum_AsyncActionStatus[AsyncActionStatusEnum_AsyncActionStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AsyncActionStatusEnum_AsyncActionStatus[AsyncActionStatusEnum_AsyncActionStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Action has not started.
     *
     * @generated from protobuf enum value: NOT_STARTED = 2;
     */
    AsyncActionStatusEnum_AsyncActionStatus[AsyncActionStatusEnum_AsyncActionStatus["NOT_STARTED"] = 2] = "NOT_STARTED";
    /**
     * Action is in progress.
     *
     * @generated from protobuf enum value: IN_PROGRESS = 3;
     */
    AsyncActionStatusEnum_AsyncActionStatus[AsyncActionStatusEnum_AsyncActionStatus["IN_PROGRESS"] = 3] = "IN_PROGRESS";
    /**
     * Action has completed successfully.
     *
     * @generated from protobuf enum value: COMPLETED = 4;
     */
    AsyncActionStatusEnum_AsyncActionStatus[AsyncActionStatusEnum_AsyncActionStatus["COMPLETED"] = 4] = "COMPLETED";
    /**
     * Action has failed.
     *
     * @generated from protobuf enum value: FAILED = 5;
     */
    AsyncActionStatusEnum_AsyncActionStatus[AsyncActionStatusEnum_AsyncActionStatus["FAILED"] = 5] = "FAILED";
    /**
     * Action has completed successfully with warnings.
     *
     * @generated from protobuf enum value: COMPLETED_WITH_WARNING = 6;
     */
    AsyncActionStatusEnum_AsyncActionStatus[AsyncActionStatusEnum_AsyncActionStatus["COMPLETED_WITH_WARNING"] = 6] = "COMPLETED_WITH_WARNING";
})(AsyncActionStatusEnum_AsyncActionStatus = exports.AsyncActionStatusEnum_AsyncActionStatus || (exports.AsyncActionStatusEnum_AsyncActionStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AsyncActionStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AsyncActionStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AsyncActionStatusEnum
 */
exports.AsyncActionStatusEnum = new AsyncActionStatusEnum$Type();
