// google-ads-api-client/dist/protos/bindings/ads/enums/attribution_model.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttributionModelEnum = exports.AttributionModelEnum_AttributionModel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The attribution model that describes how to distribute credit for a
 * particular conversion across potentially many prior interactions.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AttributionModelEnum.AttributionModel
 */
var AttributionModelEnum_AttributionModel;
(function (AttributionModelEnum_AttributionModel) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Uses external attribution.
     *
     * @generated from protobuf enum value: EXTERNAL = 100;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["EXTERNAL"] = 100] = "EXTERNAL";
    /**
     * Attributes all credit for a conversion to its last click.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_LAST_CLICK = 101;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["GOOGLE_ADS_LAST_CLICK"] = 101] = "GOOGLE_ADS_LAST_CLICK";
    /**
     * Attributes all credit for a conversion to its first click using Google
     * Search attribution.
     *
     * @generated from protobuf enum value: GOOGLE_SEARCH_ATTRIBUTION_FIRST_CLICK = 102;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["GOOGLE_SEARCH_ATTRIBUTION_FIRST_CLICK"] = 102] = "GOOGLE_SEARCH_ATTRIBUTION_FIRST_CLICK";
    /**
     * Attributes credit for a conversion equally across all of its clicks using
     * Google Search attribution.
     *
     * @generated from protobuf enum value: GOOGLE_SEARCH_ATTRIBUTION_LINEAR = 103;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["GOOGLE_SEARCH_ATTRIBUTION_LINEAR"] = 103] = "GOOGLE_SEARCH_ATTRIBUTION_LINEAR";
    /**
     * Attributes exponentially more credit for a conversion to its more recent
     * clicks using Google Search attribution (half-life is 1 week).
     *
     * @generated from protobuf enum value: GOOGLE_SEARCH_ATTRIBUTION_TIME_DECAY = 104;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["GOOGLE_SEARCH_ATTRIBUTION_TIME_DECAY"] = 104] = "GOOGLE_SEARCH_ATTRIBUTION_TIME_DECAY";
    /**
     * Attributes 40% of the credit for a conversion to its first and last
     * clicks. Remaining 20% is evenly distributed across all other clicks. This
     * uses Google Search attribution.
     *
     * @generated from protobuf enum value: GOOGLE_SEARCH_ATTRIBUTION_POSITION_BASED = 105;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["GOOGLE_SEARCH_ATTRIBUTION_POSITION_BASED"] = 105] = "GOOGLE_SEARCH_ATTRIBUTION_POSITION_BASED";
    /**
     * Flexible model that uses machine learning to determine the appropriate
     * distribution of credit among clicks using Google Search attribution.
     *
     * @generated from protobuf enum value: GOOGLE_SEARCH_ATTRIBUTION_DATA_DRIVEN = 106;
     */
    AttributionModelEnum_AttributionModel[AttributionModelEnum_AttributionModel["GOOGLE_SEARCH_ATTRIBUTION_DATA_DRIVEN"] = 106] = "GOOGLE_SEARCH_ATTRIBUTION_DATA_DRIVEN";
})(AttributionModelEnum_AttributionModel = exports.AttributionModelEnum_AttributionModel || (exports.AttributionModelEnum_AttributionModel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AttributionModelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AttributionModelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AttributionModelEnum
 */
exports.AttributionModelEnum = new AttributionModelEnum$Type();
