// google-ads-api-client/dist/protos/bindings/ads/enums/audience_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AudienceStatusEnum = exports.AudienceStatusEnum_AudienceStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible audience status types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AudienceStatusEnum.AudienceStatus
 */
var AudienceStatusEnum_AudienceStatus;
(function (AudienceStatusEnum_AudienceStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AudienceStatusEnum_AudienceStatus[AudienceStatusEnum_AudienceStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AudienceStatusEnum_AudienceStatus[AudienceStatusEnum_AudienceStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Enabled status - audience is enabled and can be targeted.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    AudienceStatusEnum_AudienceStatus[AudienceStatusEnum_AudienceStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Removed status - audience is removed and cannot be used for
     * targeting.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    AudienceStatusEnum_AudienceStatus[AudienceStatusEnum_AudienceStatus["REMOVED"] = 3] = "REMOVED";
})(AudienceStatusEnum_AudienceStatus = exports.AudienceStatusEnum_AudienceStatus || (exports.AudienceStatusEnum_AudienceStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AudienceStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AudienceStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AudienceStatusEnum
 */
exports.AudienceStatusEnum = new AudienceStatusEnum$Type();
