// google-ads-api-client/dist/protos/bindings/ads/enums/audience_insights_dimension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AudienceInsightsDimensionEnum = exports.AudienceInsightsDimensionEnum_AudienceInsightsDimension = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible audience dimensions for use in generating insights.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.AudienceInsightsDimensionEnum.AudienceInsightsDimension
 */
var AudienceInsightsDimensionEnum_AudienceInsightsDimension;
(function (AudienceInsightsDimensionEnum_AudienceInsightsDimension) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A Product & Service category.
     *
     * @generated from protobuf enum value: CATEGORY = 2;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["CATEGORY"] = 2] = "CATEGORY";
    /**
     * A Knowledge Graph entity.
     *
     * @generated from protobuf enum value: KNOWLEDGE_GRAPH = 3;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["KNOWLEDGE_GRAPH"] = 3] = "KNOWLEDGE_GRAPH";
    /**
     * A country, represented by a geo target.
     *
     * @generated from protobuf enum value: GEO_TARGET_COUNTRY = 4;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["GEO_TARGET_COUNTRY"] = 4] = "GEO_TARGET_COUNTRY";
    /**
     * A geographic location within a country.
     *
     * @generated from protobuf enum value: SUB_COUNTRY_LOCATION = 5;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["SUB_COUNTRY_LOCATION"] = 5] = "SUB_COUNTRY_LOCATION";
    /**
     * A YouTube channel.
     *
     * @generated from protobuf enum value: YOUTUBE_CHANNEL = 6;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["YOUTUBE_CHANNEL"] = 6] = "YOUTUBE_CHANNEL";
    /**
     * A YouTube Dynamic Lineup.
     *
     * @generated from protobuf enum value: YOUTUBE_DYNAMIC_LINEUP = 7;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["YOUTUBE_DYNAMIC_LINEUP"] = 7] = "YOUTUBE_DYNAMIC_LINEUP";
    /**
     * An Affinity UserInterest.
     *
     * @generated from protobuf enum value: AFFINITY_USER_INTEREST = 8;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["AFFINITY_USER_INTEREST"] = 8] = "AFFINITY_USER_INTEREST";
    /**
     * An In-Market UserInterest.
     *
     * @generated from protobuf enum value: IN_MARKET_USER_INTEREST = 9;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["IN_MARKET_USER_INTEREST"] = 9] = "IN_MARKET_USER_INTEREST";
    /**
     * A Parental Status value (parent, or not a parent).
     *
     * @generated from protobuf enum value: PARENTAL_STATUS = 10;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["PARENTAL_STATUS"] = 10] = "PARENTAL_STATUS";
    /**
     * A household income percentile range.
     *
     * @generated from protobuf enum value: INCOME_RANGE = 11;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["INCOME_RANGE"] = 11] = "INCOME_RANGE";
    /**
     * An age range.
     *
     * @generated from protobuf enum value: AGE_RANGE = 12;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["AGE_RANGE"] = 12] = "AGE_RANGE";
    /**
     * A gender.
     *
     * @generated from protobuf enum value: GENDER = 13;
     */
    AudienceInsightsDimensionEnum_AudienceInsightsDimension[AudienceInsightsDimensionEnum_AudienceInsightsDimension["GENDER"] = 13] = "GENDER";
})(AudienceInsightsDimensionEnum_AudienceInsightsDimension = exports.AudienceInsightsDimensionEnum_AudienceInsightsDimension || (exports.AudienceInsightsDimensionEnum_AudienceInsightsDimension = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AudienceInsightsDimensionEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.AudienceInsightsDimensionEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.AudienceInsightsDimensionEnum
 */
exports.AudienceInsightsDimensionEnum = new AudienceInsightsDimensionEnum$Type();
