// google-ads-api-client/dist/protos/bindings/ads/enums/batch_job_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchJobStatusEnum = exports.BatchJobStatusEnum_BatchJobStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The batch job statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BatchJobStatusEnum.BatchJobStatus
 */
var BatchJobStatusEnum_BatchJobStatus;
(function (BatchJobStatusEnum_BatchJobStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BatchJobStatusEnum_BatchJobStatus[BatchJobStatusEnum_BatchJobStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BatchJobStatusEnum_BatchJobStatus[BatchJobStatusEnum_BatchJobStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The job is not currently running.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    BatchJobStatusEnum_BatchJobStatus[BatchJobStatusEnum_BatchJobStatus["PENDING"] = 2] = "PENDING";
    /**
     * The job is running.
     *
     * @generated from protobuf enum value: RUNNING = 3;
     */
    BatchJobStatusEnum_BatchJobStatus[BatchJobStatusEnum_BatchJobStatus["RUNNING"] = 3] = "RUNNING";
    /**
     * The job is done.
     *
     * @generated from protobuf enum value: DONE = 4;
     */
    BatchJobStatusEnum_BatchJobStatus[BatchJobStatusEnum_BatchJobStatus["DONE"] = 4] = "DONE";
})(BatchJobStatusEnum_BatchJobStatus = exports.BatchJobStatusEnum_BatchJobStatus || (exports.BatchJobStatusEnum_BatchJobStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BatchJobStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BatchJobStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BatchJobStatusEnum
 */
exports.BatchJobStatusEnum = new BatchJobStatusEnum$Type();
