// google-ads-api-client/dist/protos/bindings/ads/enums/bid_modifier_source.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BidModifierSourceEnum = exports.BidModifierSourceEnum_BidModifierSource = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible bid modifier sources.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BidModifierSourceEnum.BidModifierSource
 */
var BidModifierSourceEnum_BidModifierSource;
(function (BidModifierSourceEnum_BidModifierSource) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BidModifierSourceEnum_BidModifierSource[BidModifierSourceEnum_BidModifierSource["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BidModifierSourceEnum_BidModifierSource[BidModifierSourceEnum_BidModifierSource["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The bid modifier is specified at the campaign level, on the campaign
     * level criterion.
     *
     * @generated from protobuf enum value: CAMPAIGN = 2;
     */
    BidModifierSourceEnum_BidModifierSource[BidModifierSourceEnum_BidModifierSource["CAMPAIGN"] = 2] = "CAMPAIGN";
    /**
     * The bid modifier is specified (overridden) at the ad group level.
     *
     * @generated from protobuf enum value: AD_GROUP = 3;
     */
    BidModifierSourceEnum_BidModifierSource[BidModifierSourceEnum_BidModifierSource["AD_GROUP"] = 3] = "AD_GROUP";
})(BidModifierSourceEnum_BidModifierSource = exports.BidModifierSourceEnum_BidModifierSource || (exports.BidModifierSourceEnum_BidModifierSource = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BidModifierSourceEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BidModifierSourceEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BidModifierSourceEnum
 */
exports.BidModifierSourceEnum = new BidModifierSourceEnum$Type();
