// google-ads-api-client/dist/protos/bindings/ads/enums/bidding_source.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BiddingSourceEnum = exports.BiddingSourceEnum_BiddingSource = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Indicates where a bid or target is defined. For example, an ad group
 * criterion may define a cpc bid directly, or it can inherit its cpc bid from
 * the ad group.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource
 */
var BiddingSourceEnum_BiddingSource;
(function (BiddingSourceEnum_BiddingSource) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BiddingSourceEnum_BiddingSource[BiddingSourceEnum_BiddingSource["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BiddingSourceEnum_BiddingSource[BiddingSourceEnum_BiddingSource["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Effective bid or target is inherited from campaign bidding strategy.
     *
     * @generated from protobuf enum value: CAMPAIGN_BIDDING_STRATEGY = 5;
     */
    BiddingSourceEnum_BiddingSource[BiddingSourceEnum_BiddingSource["CAMPAIGN_BIDDING_STRATEGY"] = 5] = "CAMPAIGN_BIDDING_STRATEGY";
    /**
     * The bid or target is defined on the ad group.
     *
     * @generated from protobuf enum value: AD_GROUP = 6;
     */
    BiddingSourceEnum_BiddingSource[BiddingSourceEnum_BiddingSource["AD_GROUP"] = 6] = "AD_GROUP";
    /**
     * The bid or target is defined on the ad group criterion.
     *
     * @generated from protobuf enum value: AD_GROUP_CRITERION = 7;
     */
    BiddingSourceEnum_BiddingSource[BiddingSourceEnum_BiddingSource["AD_GROUP_CRITERION"] = 7] = "AD_GROUP_CRITERION";
})(BiddingSourceEnum_BiddingSource = exports.BiddingSourceEnum_BiddingSource || (exports.BiddingSourceEnum_BiddingSource = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BiddingSourceEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BiddingSourceEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BiddingSourceEnum
 */
exports.BiddingSourceEnum = new BiddingSourceEnum$Type();
