// google-ads-api-client/dist/protos/bindings/ads/enums/bidding_strategy_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BiddingStrategyStatusEnum = exports.BiddingStrategyStatusEnum_BiddingStrategyStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of a BiddingStrategy.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BiddingStrategyStatusEnum.BiddingStrategyStatus
 */
var BiddingStrategyStatusEnum_BiddingStrategyStatus;
(function (BiddingStrategyStatusEnum_BiddingStrategyStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BiddingStrategyStatusEnum_BiddingStrategyStatus[BiddingStrategyStatusEnum_BiddingStrategyStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BiddingStrategyStatusEnum_BiddingStrategyStatus[BiddingStrategyStatusEnum_BiddingStrategyStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The bidding strategy is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    BiddingStrategyStatusEnum_BiddingStrategyStatus[BiddingStrategyStatusEnum_BiddingStrategyStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The bidding strategy is removed.
     *
     * @generated from protobuf enum value: REMOVED = 4;
     */
    BiddingStrategyStatusEnum_BiddingStrategyStatus[BiddingStrategyStatusEnum_BiddingStrategyStatus["REMOVED"] = 4] = "REMOVED";
})(BiddingStrategyStatusEnum_BiddingStrategyStatus = exports.BiddingStrategyStatusEnum_BiddingStrategyStatus || (exports.BiddingStrategyStatusEnum_BiddingStrategyStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BiddingStrategyStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BiddingStrategyStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BiddingStrategyStatusEnum
 */
exports.BiddingStrategyStatusEnum = new BiddingStrategyStatusEnum$Type();
