// google-ads-api-client/dist/protos/bindings/ads/enums/bidding_strategy_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BiddingStrategyTypeEnum = exports.BiddingStrategyTypeEnum_BiddingStrategyType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible bidding strategy types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BiddingStrategyTypeEnum.BiddingStrategyType
 */
var BiddingStrategyTypeEnum_BiddingStrategyType;
(function (BiddingStrategyTypeEnum_BiddingStrategyType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Commission is an automatic bidding strategy in which the advertiser pays
     * a certain portion of the conversion value.
     *
     * @generated from protobuf enum value: COMMISSION = 16;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["COMMISSION"] = 16] = "COMMISSION";
    /**
     * Enhanced CPC is a bidding strategy that raises bids for clicks
     * that seem more likely to lead to a conversion and lowers
     * them for clicks where they seem less likely.
     *
     * @generated from protobuf enum value: ENHANCED_CPC = 2;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["ENHANCED_CPC"] = 2] = "ENHANCED_CPC";
    /**
     * Used for return value only. Indicates that a campaign does not have a
     * bidding strategy. This prevents the campaign from serving. For example,
     * a campaign may be attached to a manager bidding strategy and the serving
     * account is subsequently unlinked from the manager account. In this case
     * the campaign will automatically be detached from the now inaccessible
     * manager bidding strategy and transition to the INVALID bidding strategy
     * type.
     *
     * @generated from protobuf enum value: INVALID = 17;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["INVALID"] = 17] = "INVALID";
    /**
     * Manual bidding strategy that allows advertiser to set the bid per
     * advertiser-specified action.
     *
     * @generated from protobuf enum value: MANUAL_CPA = 18;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["MANUAL_CPA"] = 18] = "MANUAL_CPA";
    /**
     * Manual click based bidding where user pays per click.
     *
     * @generated from protobuf enum value: MANUAL_CPC = 3;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["MANUAL_CPC"] = 3] = "MANUAL_CPC";
    /**
     * Manual impression based bidding
     * where user pays per thousand impressions.
     *
     * @generated from protobuf enum value: MANUAL_CPM = 4;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["MANUAL_CPM"] = 4] = "MANUAL_CPM";
    /**
     * A bidding strategy that pays a configurable amount per video view.
     *
     * @generated from protobuf enum value: MANUAL_CPV = 13;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["MANUAL_CPV"] = 13] = "MANUAL_CPV";
    /**
     * A bidding strategy that automatically maximizes number of conversions
     * given a daily budget.
     *
     * @generated from protobuf enum value: MAXIMIZE_CONVERSIONS = 10;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["MAXIMIZE_CONVERSIONS"] = 10] = "MAXIMIZE_CONVERSIONS";
    /**
     * An automated bidding strategy that automatically sets bids to maximize
     * revenue while spending your budget.
     *
     * @generated from protobuf enum value: MAXIMIZE_CONVERSION_VALUE = 11;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["MAXIMIZE_CONVERSION_VALUE"] = 11] = "MAXIMIZE_CONVERSION_VALUE";
    /**
     * Page-One Promoted bidding scheme, which sets max cpc bids to
     * target impressions on page one or page one promoted slots on google.com.
     * This enum value is deprecated.
     *
     * @generated from protobuf enum value: PAGE_ONE_PROMOTED = 5;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["PAGE_ONE_PROMOTED"] = 5] = "PAGE_ONE_PROMOTED";
    /**
     * Percent Cpc is bidding strategy where bids are a fraction of the
     * advertised price for some good or service.
     *
     * @generated from protobuf enum value: PERCENT_CPC = 12;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["PERCENT_CPC"] = 12] = "PERCENT_CPC";
    /**
     * Target CPA is an automated bid strategy that sets bids
     * to help get as many conversions as possible
     * at the target cost-per-acquisition (CPA) you set.
     *
     * @generated from protobuf enum value: TARGET_CPA = 6;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["TARGET_CPA"] = 6] = "TARGET_CPA";
    /**
     * Target CPM is an automated bid strategy that sets bids to help get
     * as many impressions as possible at the target cost per one thousand
     * impressions (CPM) you set.
     *
     * @generated from protobuf enum value: TARGET_CPM = 14;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["TARGET_CPM"] = 14] = "TARGET_CPM";
    /**
     * An automated bidding strategy that sets bids so that a certain percentage
     * of search ads are shown at the top of the first page (or other targeted
     * location).
     *
     * @generated from protobuf enum value: TARGET_IMPRESSION_SHARE = 15;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["TARGET_IMPRESSION_SHARE"] = 15] = "TARGET_IMPRESSION_SHARE";
    /**
     * Target Outrank Share is an automated bidding strategy that sets bids
     * based on the target fraction of auctions where the advertiser
     * should outrank a specific competitor.
     * This enum value is deprecated.
     *
     * @generated from protobuf enum value: TARGET_OUTRANK_SHARE = 7;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["TARGET_OUTRANK_SHARE"] = 7] = "TARGET_OUTRANK_SHARE";
    /**
     * Target ROAS is an automated bidding strategy
     * that helps you maximize revenue while averaging
     * a specific target Return On Average Spend (ROAS).
     *
     * @generated from protobuf enum value: TARGET_ROAS = 8;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["TARGET_ROAS"] = 8] = "TARGET_ROAS";
    /**
     * Target Spend is an automated bid strategy that sets your bids
     * to help get as many clicks as possible within your budget.
     *
     * @generated from protobuf enum value: TARGET_SPEND = 9;
     */
    BiddingStrategyTypeEnum_BiddingStrategyType[BiddingStrategyTypeEnum_BiddingStrategyType["TARGET_SPEND"] = 9] = "TARGET_SPEND";
})(BiddingStrategyTypeEnum_BiddingStrategyType = exports.BiddingStrategyTypeEnum_BiddingStrategyType || (exports.BiddingStrategyTypeEnum_BiddingStrategyType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BiddingStrategyTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BiddingStrategyTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BiddingStrategyTypeEnum
 */
exports.BiddingStrategyTypeEnum = new BiddingStrategyTypeEnum$Type();
