// google-ads-api-client/dist/protos/bindings/ads/enums/bidding_strategy_system_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BiddingStrategySystemStatusEnum = exports.BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible system statuses of a BiddingStrategy.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BiddingStrategySystemStatusEnum.BiddingStrategySystemStatus
 */
var BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus;
(function (BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus) {
    /**
     * Signals that an unexpected error occurred, for example, no bidding
     * strategy type was found, or no status information was found.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The bid strategy is active, and AdWords cannot find any specific issues
     * with the strategy.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The bid strategy is learning because it has been recently created or
     * recently reactivated.
     *
     * @generated from protobuf enum value: LEARNING_NEW = 3;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LEARNING_NEW"] = 3] = "LEARNING_NEW";
    /**
     * The bid strategy is learning because of a recent setting change.
     *
     * @generated from protobuf enum value: LEARNING_SETTING_CHANGE = 4;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LEARNING_SETTING_CHANGE"] = 4] = "LEARNING_SETTING_CHANGE";
    /**
     * The bid strategy is learning because of a recent budget change.
     *
     * @generated from protobuf enum value: LEARNING_BUDGET_CHANGE = 5;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LEARNING_BUDGET_CHANGE"] = 5] = "LEARNING_BUDGET_CHANGE";
    /**
     * The bid strategy is learning because of recent change in number of
     * campaigns, ad groups or keywords attached to it.
     *
     * @generated from protobuf enum value: LEARNING_COMPOSITION_CHANGE = 6;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LEARNING_COMPOSITION_CHANGE"] = 6] = "LEARNING_COMPOSITION_CHANGE";
    /**
     * The bid strategy depends on conversion reporting and the customer
     * recently modified conversion types that were relevant to the
     * bid strategy.
     *
     * @generated from protobuf enum value: LEARNING_CONVERSION_TYPE_CHANGE = 7;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LEARNING_CONVERSION_TYPE_CHANGE"] = 7] = "LEARNING_CONVERSION_TYPE_CHANGE";
    /**
     * The bid strategy depends on conversion reporting and the customer
     * recently changed their conversion settings.
     *
     * @generated from protobuf enum value: LEARNING_CONVERSION_SETTING_CHANGE = 8;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LEARNING_CONVERSION_SETTING_CHANGE"] = 8] = "LEARNING_CONVERSION_SETTING_CHANGE";
    /**
     * The bid strategy is limited by its bid ceiling.
     *
     * @generated from protobuf enum value: LIMITED_BY_CPC_BID_CEILING = 9;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LIMITED_BY_CPC_BID_CEILING"] = 9] = "LIMITED_BY_CPC_BID_CEILING";
    /**
     * The bid strategy is limited by its bid floor.
     *
     * @generated from protobuf enum value: LIMITED_BY_CPC_BID_FLOOR = 10;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LIMITED_BY_CPC_BID_FLOOR"] = 10] = "LIMITED_BY_CPC_BID_FLOOR";
    /**
     * The bid strategy is limited because there was not enough conversion
     * traffic over the past weeks.
     *
     * @generated from protobuf enum value: LIMITED_BY_DATA = 11;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LIMITED_BY_DATA"] = 11] = "LIMITED_BY_DATA";
    /**
     * A significant fraction of keywords in this bid strategy are limited by
     * budget.
     *
     * @generated from protobuf enum value: LIMITED_BY_BUDGET = 12;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LIMITED_BY_BUDGET"] = 12] = "LIMITED_BY_BUDGET";
    /**
     * The bid strategy cannot reach its target spend because its spend has
     * been de-prioritized.
     *
     * @generated from protobuf enum value: LIMITED_BY_LOW_PRIORITY_SPEND = 13;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LIMITED_BY_LOW_PRIORITY_SPEND"] = 13] = "LIMITED_BY_LOW_PRIORITY_SPEND";
    /**
     * A significant fraction of keywords in this bid strategy have a low
     * Quality Score.
     *
     * @generated from protobuf enum value: LIMITED_BY_LOW_QUALITY = 14;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LIMITED_BY_LOW_QUALITY"] = 14] = "LIMITED_BY_LOW_QUALITY";
    /**
     * The bid strategy cannot fully spend its budget because of narrow
     * targeting.
     *
     * @generated from protobuf enum value: LIMITED_BY_INVENTORY = 15;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["LIMITED_BY_INVENTORY"] = 15] = "LIMITED_BY_INVENTORY";
    /**
     * Missing conversion tracking (no pings present) and/or remarketing lists
     * for SSC.
     *
     * @generated from protobuf enum value: MISCONFIGURED_ZERO_ELIGIBILITY = 16;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MISCONFIGURED_ZERO_ELIGIBILITY"] = 16] = "MISCONFIGURED_ZERO_ELIGIBILITY";
    /**
     * The bid strategy depends on conversion reporting and the customer is
     * lacking conversion types that might be reported against this strategy.
     *
     * @generated from protobuf enum value: MISCONFIGURED_CONVERSION_TYPES = 17;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MISCONFIGURED_CONVERSION_TYPES"] = 17] = "MISCONFIGURED_CONVERSION_TYPES";
    /**
     * The bid strategy depends on conversion reporting and the customer's
     * conversion settings are misconfigured.
     *
     * @generated from protobuf enum value: MISCONFIGURED_CONVERSION_SETTINGS = 18;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MISCONFIGURED_CONVERSION_SETTINGS"] = 18] = "MISCONFIGURED_CONVERSION_SETTINGS";
    /**
     * There are campaigns outside the bid strategy that share budgets with
     * campaigns included in the strategy.
     *
     * @generated from protobuf enum value: MISCONFIGURED_SHARED_BUDGET = 19;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MISCONFIGURED_SHARED_BUDGET"] = 19] = "MISCONFIGURED_SHARED_BUDGET";
    /**
     * The campaign has an invalid strategy type and is not serving.
     *
     * @generated from protobuf enum value: MISCONFIGURED_STRATEGY_TYPE = 20;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MISCONFIGURED_STRATEGY_TYPE"] = 20] = "MISCONFIGURED_STRATEGY_TYPE";
    /**
     * The bid strategy is not active. Either there are no active campaigns,
     * ad groups or keywords attached to the bid strategy. Or there are no
     * active budgets connected to the bid strategy.
     *
     * @generated from protobuf enum value: PAUSED = 21;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["PAUSED"] = 21] = "PAUSED";
    /**
     * This bid strategy currently does not support status reporting.
     *
     * @generated from protobuf enum value: UNAVAILABLE = 22;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["UNAVAILABLE"] = 22] = "UNAVAILABLE";
    /**
     * There were multiple LEARNING_* system statuses for this bid strategy
     * during the time in question.
     *
     * @generated from protobuf enum value: MULTIPLE_LEARNING = 23;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MULTIPLE_LEARNING"] = 23] = "MULTIPLE_LEARNING";
    /**
     * There were multiple LIMITED_* system statuses for this bid strategy
     * during the time in question.
     *
     * @generated from protobuf enum value: MULTIPLE_LIMITED = 24;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MULTIPLE_LIMITED"] = 24] = "MULTIPLE_LIMITED";
    /**
     * There were multiple MISCONFIGURED_* system statuses for this bid strategy
     * during the time in question.
     *
     * @generated from protobuf enum value: MULTIPLE_MISCONFIGURED = 25;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MULTIPLE_MISCONFIGURED"] = 25] = "MULTIPLE_MISCONFIGURED";
    /**
     * There were multiple system statuses for this bid strategy during the
     * time in question.
     *
     * @generated from protobuf enum value: MULTIPLE = 26;
     */
    BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus[BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus["MULTIPLE"] = 26] = "MULTIPLE";
})(BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus = exports.BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus || (exports.BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BiddingStrategySystemStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BiddingStrategySystemStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BiddingStrategySystemStatusEnum
 */
exports.BiddingStrategySystemStatusEnum = new BiddingStrategySystemStatusEnum$Type();
