// google-ads-api-client/dist/protos/bindings/ads/enums/billing_setup_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingSetupStatusEnum = exports.BillingSetupStatusEnum_BillingSetupStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of a BillingSetup.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BillingSetupStatusEnum.BillingSetupStatus
 */
var BillingSetupStatusEnum_BillingSetupStatus;
(function (BillingSetupStatusEnum_BillingSetupStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BillingSetupStatusEnum_BillingSetupStatus[BillingSetupStatusEnum_BillingSetupStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BillingSetupStatusEnum_BillingSetupStatus[BillingSetupStatusEnum_BillingSetupStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The billing setup is pending approval.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    BillingSetupStatusEnum_BillingSetupStatus[BillingSetupStatusEnum_BillingSetupStatus["PENDING"] = 2] = "PENDING";
    /**
     * The billing setup has been approved but the corresponding first budget
     * has not.  This can only occur for billing setups configured for monthly
     * invoicing.
     *
     * @generated from protobuf enum value: APPROVED_HELD = 3;
     */
    BillingSetupStatusEnum_BillingSetupStatus[BillingSetupStatusEnum_BillingSetupStatus["APPROVED_HELD"] = 3] = "APPROVED_HELD";
    /**
     * The billing setup has been approved.
     *
     * @generated from protobuf enum value: APPROVED = 4;
     */
    BillingSetupStatusEnum_BillingSetupStatus[BillingSetupStatusEnum_BillingSetupStatus["APPROVED"] = 4] = "APPROVED";
    /**
     * The billing setup was cancelled by the user prior to approval.
     *
     * @generated from protobuf enum value: CANCELLED = 5;
     */
    BillingSetupStatusEnum_BillingSetupStatus[BillingSetupStatusEnum_BillingSetupStatus["CANCELLED"] = 5] = "CANCELLED";
})(BillingSetupStatusEnum_BillingSetupStatus = exports.BillingSetupStatusEnum_BillingSetupStatus || (exports.BillingSetupStatusEnum_BillingSetupStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BillingSetupStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BillingSetupStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BillingSetupStatusEnum
 */
exports.BillingSetupStatusEnum = new BillingSetupStatusEnum$Type();
