// google-ads-api-client/dist/protos/bindings/ads/enums/brand_safety_suitability.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BrandSafetySuitabilityEnum = exports.BrandSafetySuitabilityEnum_BrandSafetySuitability = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * 3-Tier brand safety suitability control.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BrandSafetySuitabilityEnum.BrandSafetySuitability
 */
var BrandSafetySuitabilityEnum_BrandSafetySuitability;
(function (BrandSafetySuitabilityEnum_BrandSafetySuitability) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BrandSafetySuitabilityEnum_BrandSafetySuitability[BrandSafetySuitabilityEnum_BrandSafetySuitability["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BrandSafetySuitabilityEnum_BrandSafetySuitability[BrandSafetySuitabilityEnum_BrandSafetySuitability["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * This option lets you show ads across all inventory on YouTube and video
     * partners that meet our standards for monetization. This option may be an
     * appropriate choice for brands that want maximum access to the full
     * breadth of videos eligible for ads, including, for example, videos that
     * have strong profanity in the context of comedy or a documentary, or
     * excessive violence as featured in video games.
     *
     * @generated from protobuf enum value: EXPANDED_INVENTORY = 2;
     */
    BrandSafetySuitabilityEnum_BrandSafetySuitability[BrandSafetySuitabilityEnum_BrandSafetySuitability["EXPANDED_INVENTORY"] = 2] = "EXPANDED_INVENTORY";
    /**
     * This option lets you show ads across a wide range of content that's
     * appropriate for most brands, such as popular music videos, documentaries,
     * and movie trailers. The content you can show ads on is based on YouTube's
     * advertiser-friendly content guidelines that take into account, for
     * example, the strength or frequency of profanity, or the appropriateness
     * of subject matter like sensitive events. Ads won't show, for example, on
     * content with repeated strong profanity, strong sexual content, or graphic
     * violence.
     *
     * @generated from protobuf enum value: STANDARD_INVENTORY = 3;
     */
    BrandSafetySuitabilityEnum_BrandSafetySuitability[BrandSafetySuitabilityEnum_BrandSafetySuitability["STANDARD_INVENTORY"] = 3] = "STANDARD_INVENTORY";
    /**
     * This option lets you show ads on a reduced range of content that's
     * appropriate for brands with particularly strict guidelines around
     * inappropriate language and sexual suggestiveness; above and beyond what
     * YouTube's advertiser-friendly content guidelines address. The videos
     * accessible in this sensitive category meet heightened requirements,
     * especially for inappropriate language and sexual suggestiveness. For
     * example, your ads will be excluded from showing on some of YouTube's most
     * popular music videos and other pop culture content across YouTube and
     * Google video partners.
     *
     * @generated from protobuf enum value: LIMITED_INVENTORY = 4;
     */
    BrandSafetySuitabilityEnum_BrandSafetySuitability[BrandSafetySuitabilityEnum_BrandSafetySuitability["LIMITED_INVENTORY"] = 4] = "LIMITED_INVENTORY";
})(BrandSafetySuitabilityEnum_BrandSafetySuitability = exports.BrandSafetySuitabilityEnum_BrandSafetySuitability || (exports.BrandSafetySuitabilityEnum_BrandSafetySuitability = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BrandSafetySuitabilityEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BrandSafetySuitabilityEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BrandSafetySuitabilityEnum
 */
exports.BrandSafetySuitabilityEnum = new BrandSafetySuitabilityEnum$Type();
