// google-ads-api-client/dist/protos/bindings/ads/enums/budget_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetTypeEnum = exports.BudgetTypeEnum_BudgetType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible Budget types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BudgetTypeEnum.BudgetType
 */
var BudgetTypeEnum_BudgetType;
(function (BudgetTypeEnum_BudgetType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BudgetTypeEnum_BudgetType[BudgetTypeEnum_BudgetType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BudgetTypeEnum_BudgetType[BudgetTypeEnum_BudgetType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Budget type for standard Google Ads usage.
     * Caps daily spend at two times the specified budget amount.
     * Full details: https://support.google.com/google-ads/answer/6385083
     *
     * @generated from protobuf enum value: STANDARD = 2;
     */
    BudgetTypeEnum_BudgetType[BudgetTypeEnum_BudgetType["STANDARD"] = 2] = "STANDARD";
    /**
     * Budget type with a fixed cost-per-acquisition (conversion).
     * Full details: https://support.google.com/google-ads/answer/7528254
     *
     * This type is only supported by campaigns with
     * AdvertisingChannelType.DISPLAY (excluding
     * AdvertisingChannelSubType.DISPLAY_GMAIL),
     * BiddingStrategyType.TARGET_CPA and PaymentMode.CONVERSIONS.
     *
     * @generated from protobuf enum value: FIXED_CPA = 4;
     */
    BudgetTypeEnum_BudgetType[BudgetTypeEnum_BudgetType["FIXED_CPA"] = 4] = "FIXED_CPA";
    /**
     * Budget type for Smart Campaign.
     * Full details: https://support.google.com/google-ads/answer/7653509
     *
     * This type is only supported by campaigns with
     * AdvertisingChannelType.SMART and
     * AdvertisingChannelSubType.SMART_CAMPAIGN.
     *
     * @generated from protobuf enum value: SMART_CAMPAIGN = 5;
     */
    BudgetTypeEnum_BudgetType[BudgetTypeEnum_BudgetType["SMART_CAMPAIGN"] = 5] = "SMART_CAMPAIGN";
    /**
     * Budget type for Local Services Campaign.
     * Full details: https://support.google.com/localservices/answer/7434558
     *
     * This type is only supported by campaigns with
     * AdvertisingChannelType.LOCAL_SERVICES.
     *
     * @generated from protobuf enum value: LOCAL_SERVICES = 6;
     */
    BudgetTypeEnum_BudgetType[BudgetTypeEnum_BudgetType["LOCAL_SERVICES"] = 6] = "LOCAL_SERVICES";
})(BudgetTypeEnum_BudgetType = exports.BudgetTypeEnum_BudgetType || (exports.BudgetTypeEnum_BudgetType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BudgetTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BudgetTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BudgetTypeEnum
 */
exports.BudgetTypeEnum = new BudgetTypeEnum$Type();
