// google-ads-api-client/dist/protos/bindings/ads/enums/budget_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetStatusEnum = exports.BudgetStatusEnum_BudgetStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a Budget.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BudgetStatusEnum.BudgetStatus
 */
var BudgetStatusEnum_BudgetStatus;
(function (BudgetStatusEnum_BudgetStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BudgetStatusEnum_BudgetStatus[BudgetStatusEnum_BudgetStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BudgetStatusEnum_BudgetStatus[BudgetStatusEnum_BudgetStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Budget is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    BudgetStatusEnum_BudgetStatus[BudgetStatusEnum_BudgetStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Budget is removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    BudgetStatusEnum_BudgetStatus[BudgetStatusEnum_BudgetStatus["REMOVED"] = 3] = "REMOVED";
})(BudgetStatusEnum_BudgetStatus = exports.BudgetStatusEnum_BudgetStatus || (exports.BudgetStatusEnum_BudgetStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BudgetStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BudgetStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BudgetStatusEnum
 */
exports.BudgetStatusEnum = new BudgetStatusEnum$Type();
