// google-ads-api-client/dist/protos/bindings/ads/enums/budget_period.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetPeriodEnum = exports.BudgetPeriodEnum_BudgetPeriod = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible period of a Budget.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BudgetPeriodEnum.BudgetPeriod
 */
var BudgetPeriodEnum_BudgetPeriod;
(function (BudgetPeriodEnum_BudgetPeriod) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BudgetPeriodEnum_BudgetPeriod[BudgetPeriodEnum_BudgetPeriod["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BudgetPeriodEnum_BudgetPeriod[BudgetPeriodEnum_BudgetPeriod["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Daily budget.
     *
     * @generated from protobuf enum value: DAILY = 2;
     */
    BudgetPeriodEnum_BudgetPeriod[BudgetPeriodEnum_BudgetPeriod["DAILY"] = 2] = "DAILY";
    /**
     * Custom budget, added back in V5.
     * Custom bugdet can be used with total_amount to specify lifetime budget
     * limit. See: https://support.google.com/google-ads/answer/6385083 for more
     * info.
     *
     * @generated from protobuf enum value: CUSTOM_PERIOD = 5;
     */
    BudgetPeriodEnum_BudgetPeriod[BudgetPeriodEnum_BudgetPeriod["CUSTOM_PERIOD"] = 5] = "CUSTOM_PERIOD";
})(BudgetPeriodEnum_BudgetPeriod = exports.BudgetPeriodEnum_BudgetPeriod || (exports.BudgetPeriodEnum_BudgetPeriod = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BudgetPeriodEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BudgetPeriodEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BudgetPeriodEnum
 */
exports.BudgetPeriodEnum = new BudgetPeriodEnum$Type();
