// google-ads-api-client/dist/protos/bindings/ads/enums/budget_delivery_method.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetDeliveryMethodEnum = exports.BudgetDeliveryMethodEnum_BudgetDeliveryMethod = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible delivery methods of a Budget.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.BudgetDeliveryMethodEnum.BudgetDeliveryMethod
 */
var BudgetDeliveryMethodEnum_BudgetDeliveryMethod;
(function (BudgetDeliveryMethodEnum_BudgetDeliveryMethod) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BudgetDeliveryMethodEnum_BudgetDeliveryMethod[BudgetDeliveryMethodEnum_BudgetDeliveryMethod["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BudgetDeliveryMethodEnum_BudgetDeliveryMethod[BudgetDeliveryMethodEnum_BudgetDeliveryMethod["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The budget server will throttle serving evenly across
     * the entire time period.
     *
     * @generated from protobuf enum value: STANDARD = 2;
     */
    BudgetDeliveryMethodEnum_BudgetDeliveryMethod[BudgetDeliveryMethodEnum_BudgetDeliveryMethod["STANDARD"] = 2] = "STANDARD";
    /**
     * The budget server will not throttle serving,
     * and ads will serve as fast as possible.
     *
     * @generated from protobuf enum value: ACCELERATED = 3;
     */
    BudgetDeliveryMethodEnum_BudgetDeliveryMethod[BudgetDeliveryMethodEnum_BudgetDeliveryMethod["ACCELERATED"] = 3] = "ACCELERATED";
})(BudgetDeliveryMethodEnum_BudgetDeliveryMethod = exports.BudgetDeliveryMethodEnum_BudgetDeliveryMethod || (exports.BudgetDeliveryMethodEnum_BudgetDeliveryMethod = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BudgetDeliveryMethodEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.BudgetDeliveryMethodEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.BudgetDeliveryMethodEnum
 */
exports.BudgetDeliveryMethodEnum = new BudgetDeliveryMethodEnum$Type();
