// google-ads-api-client/dist/protos/bindings/ads/enums/call_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallPlaceholderFieldEnum = exports.CallPlaceholderFieldEnum_CallPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Call placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CallPlaceholderFieldEnum.CallPlaceholderField
 */
var CallPlaceholderFieldEnum_CallPlaceholderField;
(function (CallPlaceholderFieldEnum_CallPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CallPlaceholderFieldEnum_CallPlaceholderField[CallPlaceholderFieldEnum_CallPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CallPlaceholderFieldEnum_CallPlaceholderField[CallPlaceholderFieldEnum_CallPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. The advertiser's phone number to append to the ad.
     *
     * @generated from protobuf enum value: PHONE_NUMBER = 2;
     */
    CallPlaceholderFieldEnum_CallPlaceholderField[CallPlaceholderFieldEnum_CallPlaceholderField["PHONE_NUMBER"] = 2] = "PHONE_NUMBER";
    /**
     * Data Type: STRING. Uppercase two-letter country code of the advertiser's
     * phone number.
     *
     * @generated from protobuf enum value: COUNTRY_CODE = 3;
     */
    CallPlaceholderFieldEnum_CallPlaceholderField[CallPlaceholderFieldEnum_CallPlaceholderField["COUNTRY_CODE"] = 3] = "COUNTRY_CODE";
    /**
     * Data Type: BOOLEAN. Indicates whether call tracking is enabled. Default:
     * true.
     *
     * @generated from protobuf enum value: TRACKED = 4;
     */
    CallPlaceholderFieldEnum_CallPlaceholderField[CallPlaceholderFieldEnum_CallPlaceholderField["TRACKED"] = 4] = "TRACKED";
    /**
     * Data Type: INT64. The ID of an AdCallMetricsConversion object. This
     * object contains the phoneCallDurationfield which is the minimum duration
     * (in seconds) of a call to be considered a conversion.
     *
     * @generated from protobuf enum value: CONVERSION_TYPE_ID = 5;
     */
    CallPlaceholderFieldEnum_CallPlaceholderField[CallPlaceholderFieldEnum_CallPlaceholderField["CONVERSION_TYPE_ID"] = 5] = "CONVERSION_TYPE_ID";
    /**
     * Data Type: STRING. Indicates whether this call extension uses its own
     * call conversion setting or follows the account level setting.
     * Valid values are: USE_ACCOUNT_LEVEL_CALL_CONVERSION_ACTION and
     * USE_RESOURCE_LEVEL_CALL_CONVERSION_ACTION.
     *
     * @generated from protobuf enum value: CONVERSION_REPORTING_STATE = 6;
     */
    CallPlaceholderFieldEnum_CallPlaceholderField[CallPlaceholderFieldEnum_CallPlaceholderField["CONVERSION_REPORTING_STATE"] = 6] = "CONVERSION_REPORTING_STATE";
})(CallPlaceholderFieldEnum_CallPlaceholderField = exports.CallPlaceholderFieldEnum_CallPlaceholderField || (exports.CallPlaceholderFieldEnum_CallPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CallPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CallPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CallPlaceholderFieldEnum
 */
exports.CallPlaceholderFieldEnum = new CallPlaceholderFieldEnum$Type();
