// google-ads-api-client/dist/protos/bindings/ads/enums/call_tracking_display_location.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallTrackingDisplayLocationEnum = exports.CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible call tracking display locations.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CallTrackingDisplayLocationEnum.CallTrackingDisplayLocation
 */
var CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation;
(function (CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation[CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation[CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The phone call placed from the ad.
     *
     * @generated from protobuf enum value: AD = 2;
     */
    CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation[CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation["AD"] = 2] = "AD";
    /**
     * The phone call placed from the landing page ad points to.
     *
     * @generated from protobuf enum value: LANDING_PAGE = 3;
     */
    CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation[CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation["LANDING_PAGE"] = 3] = "LANDING_PAGE";
})(CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation = exports.CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation || (exports.CallTrackingDisplayLocationEnum_CallTrackingDisplayLocation = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CallTrackingDisplayLocationEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CallTrackingDisplayLocationEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CallTrackingDisplayLocationEnum
 */
exports.CallTrackingDisplayLocationEnum = new CallTrackingDisplayLocationEnum$Type();
