// google-ads-api-client/dist/protos/bindings/ads/enums/call_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallTypeEnum = exports.CallTypeEnum_CallType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible types of property from where the call was made.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CallTypeEnum.CallType
 */
var CallTypeEnum_CallType;
(function (CallTypeEnum_CallType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CallTypeEnum_CallType[CallTypeEnum_CallType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CallTypeEnum_CallType[CallTypeEnum_CallType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The phone call was manually dialed.
     *
     * @generated from protobuf enum value: MANUALLY_DIALED = 2;
     */
    CallTypeEnum_CallType[CallTypeEnum_CallType["MANUALLY_DIALED"] = 2] = "MANUALLY_DIALED";
    /**
     * The phone call was a mobile click-to-call.
     *
     * @generated from protobuf enum value: HIGH_END_MOBILE_SEARCH = 3;
     */
    CallTypeEnum_CallType[CallTypeEnum_CallType["HIGH_END_MOBILE_SEARCH"] = 3] = "HIGH_END_MOBILE_SEARCH";
})(CallTypeEnum_CallType = exports.CallTypeEnum_CallType || (exports.CallTypeEnum_CallType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CallTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CallTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CallTypeEnum
 */
exports.CallTypeEnum = new CallTypeEnum$Type();
