// google-ads-api-client/dist/protos/bindings/ads/enums/call_conversion_reporting_state.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallConversionReportingStateEnum = exports.CallConversionReportingStateEnum_CallConversionReportingState = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible data types for a call conversion action state.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState
 */
var CallConversionReportingStateEnum_CallConversionReportingState;
(function (CallConversionReportingStateEnum_CallConversionReportingState) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CallConversionReportingStateEnum_CallConversionReportingState[CallConversionReportingStateEnum_CallConversionReportingState["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CallConversionReportingStateEnum_CallConversionReportingState[CallConversionReportingStateEnum_CallConversionReportingState["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Call conversion action is disabled.
     *
     * @generated from protobuf enum value: DISABLED = 2;
     */
    CallConversionReportingStateEnum_CallConversionReportingState[CallConversionReportingStateEnum_CallConversionReportingState["DISABLED"] = 2] = "DISABLED";
    /**
     * Call conversion action will use call conversion type set at the
     * account level.
     *
     * @generated from protobuf enum value: USE_ACCOUNT_LEVEL_CALL_CONVERSION_ACTION = 3;
     */
    CallConversionReportingStateEnum_CallConversionReportingState[CallConversionReportingStateEnum_CallConversionReportingState["USE_ACCOUNT_LEVEL_CALL_CONVERSION_ACTION"] = 3] = "USE_ACCOUNT_LEVEL_CALL_CONVERSION_ACTION";
    /**
     * Call conversion action will use call conversion type set at the resource
     * (call only ads/call extensions) level.
     *
     * @generated from protobuf enum value: USE_RESOURCE_LEVEL_CALL_CONVERSION_ACTION = 4;
     */
    CallConversionReportingStateEnum_CallConversionReportingState[CallConversionReportingStateEnum_CallConversionReportingState["USE_RESOURCE_LEVEL_CALL_CONVERSION_ACTION"] = 4] = "USE_RESOURCE_LEVEL_CALL_CONVERSION_ACTION";
})(CallConversionReportingStateEnum_CallConversionReportingState = exports.CallConversionReportingStateEnum_CallConversionReportingState || (exports.CallConversionReportingStateEnum_CallConversionReportingState = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CallConversionReportingStateEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CallConversionReportingStateEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CallConversionReportingStateEnum
 */
exports.CallConversionReportingStateEnum = new CallConversionReportingStateEnum$Type();
