// google-ads-api-client/dist/protos/bindings/ads/enums/call_to_action_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallToActionTypeEnum = exports.CallToActionTypeEnum_CallToActionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible types of call to action.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CallToActionTypeEnum.CallToActionType
 */
var CallToActionTypeEnum_CallToActionType;
(function (CallToActionTypeEnum_CallToActionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The call to action type is learn more.
     *
     * @generated from protobuf enum value: LEARN_MORE = 2;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["LEARN_MORE"] = 2] = "LEARN_MORE";
    /**
     * The call to action type is get quote.
     *
     * @generated from protobuf enum value: GET_QUOTE = 3;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["GET_QUOTE"] = 3] = "GET_QUOTE";
    /**
     * The call to action type is apply now.
     *
     * @generated from protobuf enum value: APPLY_NOW = 4;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["APPLY_NOW"] = 4] = "APPLY_NOW";
    /**
     * The call to action type is sign up.
     *
     * @generated from protobuf enum value: SIGN_UP = 5;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["SIGN_UP"] = 5] = "SIGN_UP";
    /**
     * The call to action type is contact us.
     *
     * @generated from protobuf enum value: CONTACT_US = 6;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["CONTACT_US"] = 6] = "CONTACT_US";
    /**
     * The call to action type is subscribe.
     *
     * @generated from protobuf enum value: SUBSCRIBE = 7;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["SUBSCRIBE"] = 7] = "SUBSCRIBE";
    /**
     * The call to action type is download.
     *
     * @generated from protobuf enum value: DOWNLOAD = 8;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["DOWNLOAD"] = 8] = "DOWNLOAD";
    /**
     * The call to action type is book now.
     *
     * @generated from protobuf enum value: BOOK_NOW = 9;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["BOOK_NOW"] = 9] = "BOOK_NOW";
    /**
     * The call to action type is shop now.
     *
     * @generated from protobuf enum value: SHOP_NOW = 10;
     */
    CallToActionTypeEnum_CallToActionType[CallToActionTypeEnum_CallToActionType["SHOP_NOW"] = 10] = "SHOP_NOW";
})(CallToActionTypeEnum_CallToActionType = exports.CallToActionTypeEnum_CallToActionType || (exports.CallToActionTypeEnum_CallToActionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CallToActionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CallToActionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CallToActionTypeEnum
 */
exports.CallToActionTypeEnum = new CallToActionTypeEnum$Type();
