// google-ads-api-client/dist/protos/bindings/ads/enums/campaign_experiment_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignExperimentTypeEnum = exports.CampaignExperimentTypeEnum_CampaignExperimentType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Indicates if this campaign is a normal campaign,
 * a draft campaign, or an experiment campaign.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CampaignExperimentTypeEnum.CampaignExperimentType
 */
var CampaignExperimentTypeEnum_CampaignExperimentType;
(function (CampaignExperimentTypeEnum_CampaignExperimentType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignExperimentTypeEnum_CampaignExperimentType[CampaignExperimentTypeEnum_CampaignExperimentType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignExperimentTypeEnum_CampaignExperimentType[CampaignExperimentTypeEnum_CampaignExperimentType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * This is a regular campaign.
     *
     * @generated from protobuf enum value: BASE = 2;
     */
    CampaignExperimentTypeEnum_CampaignExperimentType[CampaignExperimentTypeEnum_CampaignExperimentType["BASE"] = 2] = "BASE";
    /**
     * This is a draft version of a campaign.
     * It has some modifications from a base campaign,
     * but it does not serve or accrue metrics.
     *
     * @generated from protobuf enum value: DRAFT = 3;
     */
    CampaignExperimentTypeEnum_CampaignExperimentType[CampaignExperimentTypeEnum_CampaignExperimentType["DRAFT"] = 3] = "DRAFT";
    /**
     * This is an experiment version of a campaign.
     * It has some modifications from a base campaign,
     * and a percentage of traffic is being diverted
     * from the BASE campaign to this experiment campaign.
     *
     * @generated from protobuf enum value: EXPERIMENT = 4;
     */
    CampaignExperimentTypeEnum_CampaignExperimentType[CampaignExperimentTypeEnum_CampaignExperimentType["EXPERIMENT"] = 4] = "EXPERIMENT";
})(CampaignExperimentTypeEnum_CampaignExperimentType = exports.CampaignExperimentTypeEnum_CampaignExperimentType || (exports.CampaignExperimentTypeEnum_CampaignExperimentType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignExperimentTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CampaignExperimentTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CampaignExperimentTypeEnum
 */
exports.CampaignExperimentTypeEnum = new CampaignExperimentTypeEnum$Type();
