// google-ads-api-client/dist/protos/bindings/ads/enums/campaign_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignStatusEnum = exports.CampaignStatusEnum_CampaignStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a campaign.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CampaignStatusEnum.CampaignStatus
 */
var CampaignStatusEnum_CampaignStatus;
(function (CampaignStatusEnum_CampaignStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignStatusEnum_CampaignStatus[CampaignStatusEnum_CampaignStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignStatusEnum_CampaignStatus[CampaignStatusEnum_CampaignStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Campaign is currently serving ads depending on budget information.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CampaignStatusEnum_CampaignStatus[CampaignStatusEnum_CampaignStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Campaign has been paused by the user.
     *
     * @generated from protobuf enum value: PAUSED = 3;
     */
    CampaignStatusEnum_CampaignStatus[CampaignStatusEnum_CampaignStatus["PAUSED"] = 3] = "PAUSED";
    /**
     * Campaign has been removed.
     *
     * @generated from protobuf enum value: REMOVED = 4;
     */
    CampaignStatusEnum_CampaignStatus[CampaignStatusEnum_CampaignStatus["REMOVED"] = 4] = "REMOVED";
})(CampaignStatusEnum_CampaignStatus = exports.CampaignStatusEnum_CampaignStatus || (exports.CampaignStatusEnum_CampaignStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CampaignStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CampaignStatusEnum
 */
exports.CampaignStatusEnum = new CampaignStatusEnum$Type();
