// google-ads-api-client/dist/protos/bindings/ads/enums/campaign_experiment_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignExperimentStatusEnum = exports.CampaignExperimentStatusEnum_CampaignExperimentStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a campaign experiment.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CampaignExperimentStatusEnum.CampaignExperimentStatus
 */
var CampaignExperimentStatusEnum_CampaignExperimentStatus;
(function (CampaignExperimentStatusEnum_CampaignExperimentStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The experiment campaign is being initialized.
     *
     * @generated from protobuf enum value: INITIALIZING = 2;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["INITIALIZING"] = 2] = "INITIALIZING";
    /**
     * Initialization of the experiment campaign failed.
     *
     * @generated from protobuf enum value: INITIALIZATION_FAILED = 8;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["INITIALIZATION_FAILED"] = 8] = "INITIALIZATION_FAILED";
    /**
     * The experiment campaign is fully initialized. The experiment is currently
     * running, scheduled to run in the future or has ended based on its
     * end date. An experiment with the status INITIALIZING will be updated to
     * ENABLED when it is fully created.
     *
     * @generated from protobuf enum value: ENABLED = 3;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["ENABLED"] = 3] = "ENABLED";
    /**
     * The experiment campaign was graduated to a stand-alone
     * campaign, existing independently of the experiment.
     *
     * @generated from protobuf enum value: GRADUATED = 4;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["GRADUATED"] = 4] = "GRADUATED";
    /**
     * The experiment is removed.
     *
     * @generated from protobuf enum value: REMOVED = 5;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["REMOVED"] = 5] = "REMOVED";
    /**
     * The experiment's changes are being applied to the original campaign.
     * The long running operation returned by the promote method can be polled
     * to see the status of the promotion.
     *
     * @generated from protobuf enum value: PROMOTING = 6;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["PROMOTING"] = 6] = "PROMOTING";
    /**
     * Promote of the experiment campaign failed.
     *
     * @generated from protobuf enum value: PROMOTION_FAILED = 9;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["PROMOTION_FAILED"] = 9] = "PROMOTION_FAILED";
    /**
     * The changes of the experiment are promoted to their original campaign.
     *
     * @generated from protobuf enum value: PROMOTED = 7;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["PROMOTED"] = 7] = "PROMOTED";
    /**
     * The experiment was ended manually. It did not end based on its end date.
     *
     * @generated from protobuf enum value: ENDED_MANUALLY = 10;
     */
    CampaignExperimentStatusEnum_CampaignExperimentStatus[CampaignExperimentStatusEnum_CampaignExperimentStatus["ENDED_MANUALLY"] = 10] = "ENDED_MANUALLY";
})(CampaignExperimentStatusEnum_CampaignExperimentStatus = exports.CampaignExperimentStatusEnum_CampaignExperimentStatus || (exports.CampaignExperimentStatusEnum_CampaignExperimentStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignExperimentStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CampaignExperimentStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CampaignExperimentStatusEnum
 */
exports.CampaignExperimentStatusEnum = new CampaignExperimentStatusEnum$Type();
