// google-ads-api-client/dist/protos/bindings/ads/enums/campaign_shared_set_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignSharedSetStatusEnum = exports.CampaignSharedSetStatusEnum_CampaignSharedSetStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the possible campaign shared set statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CampaignSharedSetStatusEnum.CampaignSharedSetStatus
 */
var CampaignSharedSetStatusEnum_CampaignSharedSetStatus;
(function (CampaignSharedSetStatusEnum_CampaignSharedSetStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignSharedSetStatusEnum_CampaignSharedSetStatus[CampaignSharedSetStatusEnum_CampaignSharedSetStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignSharedSetStatusEnum_CampaignSharedSetStatus[CampaignSharedSetStatusEnum_CampaignSharedSetStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The campaign shared set is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CampaignSharedSetStatusEnum_CampaignSharedSetStatus[CampaignSharedSetStatusEnum_CampaignSharedSetStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The campaign shared set is removed and can no longer be used.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    CampaignSharedSetStatusEnum_CampaignSharedSetStatus[CampaignSharedSetStatusEnum_CampaignSharedSetStatus["REMOVED"] = 3] = "REMOVED";
})(CampaignSharedSetStatusEnum_CampaignSharedSetStatus = exports.CampaignSharedSetStatusEnum_CampaignSharedSetStatus || (exports.CampaignSharedSetStatusEnum_CampaignSharedSetStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignSharedSetStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CampaignSharedSetStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CampaignSharedSetStatusEnum
 */
exports.CampaignSharedSetStatusEnum = new CampaignSharedSetStatusEnum$Type();
