// google-ads-api-client/dist/protos/bindings/ads/enums/campaign_criterion_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignCriterionStatusEnum = exports.CampaignCriterionStatusEnum_CampaignCriterionStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of a CampaignCriterion.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CampaignCriterionStatusEnum.CampaignCriterionStatus
 */
var CampaignCriterionStatusEnum_CampaignCriterionStatus;
(function (CampaignCriterionStatusEnum_CampaignCriterionStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignCriterionStatusEnum_CampaignCriterionStatus[CampaignCriterionStatusEnum_CampaignCriterionStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignCriterionStatusEnum_CampaignCriterionStatus[CampaignCriterionStatusEnum_CampaignCriterionStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The campaign criterion is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CampaignCriterionStatusEnum_CampaignCriterionStatus[CampaignCriterionStatusEnum_CampaignCriterionStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The campaign criterion is paused.
     *
     * @generated from protobuf enum value: PAUSED = 3;
     */
    CampaignCriterionStatusEnum_CampaignCriterionStatus[CampaignCriterionStatusEnum_CampaignCriterionStatus["PAUSED"] = 3] = "PAUSED";
    /**
     * The campaign criterion is removed.
     *
     * @generated from protobuf enum value: REMOVED = 4;
     */
    CampaignCriterionStatusEnum_CampaignCriterionStatus[CampaignCriterionStatusEnum_CampaignCriterionStatus["REMOVED"] = 4] = "REMOVED";
})(CampaignCriterionStatusEnum_CampaignCriterionStatus = exports.CampaignCriterionStatusEnum_CampaignCriterionStatus || (exports.CampaignCriterionStatusEnum_CampaignCriterionStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignCriterionStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CampaignCriterionStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CampaignCriterionStatusEnum
 */
exports.CampaignCriterionStatusEnum = new CampaignCriterionStatusEnum$Type();
