// google-ads-api-client/dist/protos/bindings/ads/enums/campaign_draft_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignDraftStatusEnum = exports.CampaignDraftStatusEnum_CampaignDraftStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a campaign draft.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CampaignDraftStatusEnum.CampaignDraftStatus
 */
var CampaignDraftStatusEnum_CampaignDraftStatus;
(function (CampaignDraftStatusEnum_CampaignDraftStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignDraftStatusEnum_CampaignDraftStatus[CampaignDraftStatusEnum_CampaignDraftStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignDraftStatusEnum_CampaignDraftStatus[CampaignDraftStatusEnum_CampaignDraftStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Initial state of the draft, the advertiser can start adding changes with
     * no effect on serving.
     *
     * @generated from protobuf enum value: PROPOSED = 2;
     */
    CampaignDraftStatusEnum_CampaignDraftStatus[CampaignDraftStatusEnum_CampaignDraftStatus["PROPOSED"] = 2] = "PROPOSED";
    /**
     * The campaign draft is removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    CampaignDraftStatusEnum_CampaignDraftStatus[CampaignDraftStatusEnum_CampaignDraftStatus["REMOVED"] = 3] = "REMOVED";
    /**
     * Advertiser requested to promote draft's changes back into the original
     * campaign. Advertiser can poll the long running operation returned by
     * the promote action to see the status of the promotion.
     *
     * @generated from protobuf enum value: PROMOTING = 5;
     */
    CampaignDraftStatusEnum_CampaignDraftStatus[CampaignDraftStatusEnum_CampaignDraftStatus["PROMOTING"] = 5] = "PROMOTING";
    /**
     * The process to merge changes in the draft back to the original campaign
     * has completed successfully.
     *
     * @generated from protobuf enum value: PROMOTED = 4;
     */
    CampaignDraftStatusEnum_CampaignDraftStatus[CampaignDraftStatusEnum_CampaignDraftStatus["PROMOTED"] = 4] = "PROMOTED";
    /**
     * The promotion failed after it was partially applied. Promote cannot be
     * attempted again safely, so the issue must be corrected in the original
     * campaign.
     *
     * @generated from protobuf enum value: PROMOTE_FAILED = 6;
     */
    CampaignDraftStatusEnum_CampaignDraftStatus[CampaignDraftStatusEnum_CampaignDraftStatus["PROMOTE_FAILED"] = 6] = "PROMOTE_FAILED";
})(CampaignDraftStatusEnum_CampaignDraftStatus = exports.CampaignDraftStatusEnum_CampaignDraftStatus || (exports.CampaignDraftStatusEnum_CampaignDraftStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignDraftStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CampaignDraftStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CampaignDraftStatusEnum
 */
exports.CampaignDraftStatusEnum = new CampaignDraftStatusEnum$Type();
