// google-ads-api-client/dist/protos/bindings/ads/enums/campaign_experiment_traffic_split_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignExperimentTrafficSplitTypeEnum = exports.CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum of strategies for splitting traffic between base and experiment
 * campaigns in campaign experiment.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CampaignExperimentTrafficSplitTypeEnum.CampaignExperimentTrafficSplitType
 */
var CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType;
(function (CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType[CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType[CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Traffic is randomly assigned to the base or experiment arm for each
     * query, independent of previous assignments for the same user.
     *
     * @generated from protobuf enum value: RANDOM_QUERY = 2;
     */
    CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType[CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType["RANDOM_QUERY"] = 2] = "RANDOM_QUERY";
    /**
     * Traffic is split using cookies to keep users in the same arm (base or
     * experiment) of the experiment.
     *
     * @generated from protobuf enum value: COOKIE = 3;
     */
    CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType[CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType["COOKIE"] = 3] = "COOKIE";
})(CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType = exports.CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType || (exports.CampaignExperimentTrafficSplitTypeEnum_CampaignExperimentTrafficSplitType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignExperimentTrafficSplitTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CampaignExperimentTrafficSplitTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CampaignExperimentTrafficSplitTypeEnum
 */
exports.CampaignExperimentTrafficSplitTypeEnum = new CampaignExperimentTrafficSplitTypeEnum$Type();
