// google-ads-api-client/dist/protos/bindings/ads/enums/campaign_serving_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignServingStatusEnum = exports.CampaignServingStatusEnum_CampaignServingStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible serving statuses of a campaign.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CampaignServingStatusEnum.CampaignServingStatus
 */
var CampaignServingStatusEnum_CampaignServingStatus;
(function (CampaignServingStatusEnum_CampaignServingStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignServingStatusEnum_CampaignServingStatus[CampaignServingStatusEnum_CampaignServingStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignServingStatusEnum_CampaignServingStatus[CampaignServingStatusEnum_CampaignServingStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Serving.
     *
     * @generated from protobuf enum value: SERVING = 2;
     */
    CampaignServingStatusEnum_CampaignServingStatus[CampaignServingStatusEnum_CampaignServingStatus["SERVING"] = 2] = "SERVING";
    /**
     * None.
     *
     * @generated from protobuf enum value: NONE = 3;
     */
    CampaignServingStatusEnum_CampaignServingStatus[CampaignServingStatusEnum_CampaignServingStatus["NONE"] = 3] = "NONE";
    /**
     * Ended.
     *
     * @generated from protobuf enum value: ENDED = 4;
     */
    CampaignServingStatusEnum_CampaignServingStatus[CampaignServingStatusEnum_CampaignServingStatus["ENDED"] = 4] = "ENDED";
    /**
     * Pending.
     *
     * @generated from protobuf enum value: PENDING = 5;
     */
    CampaignServingStatusEnum_CampaignServingStatus[CampaignServingStatusEnum_CampaignServingStatus["PENDING"] = 5] = "PENDING";
    /**
     * Suspended.
     *
     * @generated from protobuf enum value: SUSPENDED = 6;
     */
    CampaignServingStatusEnum_CampaignServingStatus[CampaignServingStatusEnum_CampaignServingStatus["SUSPENDED"] = 6] = "SUSPENDED";
})(CampaignServingStatusEnum_CampaignServingStatus = exports.CampaignServingStatusEnum_CampaignServingStatus || (exports.CampaignServingStatusEnum_CampaignServingStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignServingStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CampaignServingStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CampaignServingStatusEnum
 */
exports.CampaignServingStatusEnum = new CampaignServingStatusEnum$Type();
