// google-ads-api-client/dist/protos/bindings/ads/enums/change_event_resource_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeEventResourceTypeEnum = exports.ChangeEventResourceTypeEnum_ChangeEventResourceType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the resource types support by the ChangeEvent resource.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ChangeEventResourceTypeEnum.ChangeEventResourceType
 */
var ChangeEventResourceTypeEnum_ChangeEventResourceType;
(function (ChangeEventResourceTypeEnum_ChangeEventResourceType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents an unclassified resource unknown
     * in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An Ad resource change.
     *
     * @generated from protobuf enum value: AD = 2;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["AD"] = 2] = "AD";
    /**
     * An AdGroup resource change.
     *
     * @generated from protobuf enum value: AD_GROUP = 3;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["AD_GROUP"] = 3] = "AD_GROUP";
    /**
     * An AdGroupCriterion resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_CRITERION = 4;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["AD_GROUP_CRITERION"] = 4] = "AD_GROUP_CRITERION";
    /**
     * A Campaign resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN = 5;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["CAMPAIGN"] = 5] = "CAMPAIGN";
    /**
     * A CampaignBudget resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_BUDGET = 6;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["CAMPAIGN_BUDGET"] = 6] = "CAMPAIGN_BUDGET";
    /**
     * An AdGroupBidModifier resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_BID_MODIFIER = 7;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["AD_GROUP_BID_MODIFIER"] = 7] = "AD_GROUP_BID_MODIFIER";
    /**
     * A CampaignCriterion resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_CRITERION = 8;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["CAMPAIGN_CRITERION"] = 8] = "CAMPAIGN_CRITERION";
    /**
     * A Feed resource change.
     *
     * @generated from protobuf enum value: FEED = 9;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["FEED"] = 9] = "FEED";
    /**
     * A FeedItem resource change.
     *
     * @generated from protobuf enum value: FEED_ITEM = 10;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["FEED_ITEM"] = 10] = "FEED_ITEM";
    /**
     * A CampaignFeed resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_FEED = 11;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["CAMPAIGN_FEED"] = 11] = "CAMPAIGN_FEED";
    /**
     * An AdGroupFeed resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_FEED = 12;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["AD_GROUP_FEED"] = 12] = "AD_GROUP_FEED";
    /**
     * An AdGroupAd resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_AD = 13;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["AD_GROUP_AD"] = 13] = "AD_GROUP_AD";
    /**
     * An Asset resource change.
     *
     * @generated from protobuf enum value: ASSET = 14;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["ASSET"] = 14] = "ASSET";
    /**
     * A CustomerAsset resource change.
     *
     * @generated from protobuf enum value: CUSTOMER_ASSET = 15;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["CUSTOMER_ASSET"] = 15] = "CUSTOMER_ASSET";
    /**
     * A CampaignAsset resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_ASSET = 16;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["CAMPAIGN_ASSET"] = 16] = "CAMPAIGN_ASSET";
    /**
     * An AdGroupAsset resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_ASSET = 17;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["AD_GROUP_ASSET"] = 17] = "AD_GROUP_ASSET";
    /**
     * An AssetSet resource change.
     *
     * @generated from protobuf enum value: ASSET_SET = 18;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["ASSET_SET"] = 18] = "ASSET_SET";
    /**
     * An AssetSetAsset resource change.
     *
     * @generated from protobuf enum value: ASSET_SET_ASSET = 19;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["ASSET_SET_ASSET"] = 19] = "ASSET_SET_ASSET";
    /**
     * A CampaignAssetSet resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_ASSET_SET = 20;
     */
    ChangeEventResourceTypeEnum_ChangeEventResourceType[ChangeEventResourceTypeEnum_ChangeEventResourceType["CAMPAIGN_ASSET_SET"] = 20] = "CAMPAIGN_ASSET_SET";
})(ChangeEventResourceTypeEnum_ChangeEventResourceType = exports.ChangeEventResourceTypeEnum_ChangeEventResourceType || (exports.ChangeEventResourceTypeEnum_ChangeEventResourceType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ChangeEventResourceTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ChangeEventResourceTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ChangeEventResourceTypeEnum
 */
exports.ChangeEventResourceTypeEnum = new ChangeEventResourceTypeEnum$Type();
