// google-ads-api-client/dist/protos/bindings/ads/enums/change_client_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeClientTypeEnum = exports.ChangeClientTypeEnum_ChangeClientType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The source that the change_event resource was made through.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ChangeClientTypeEnum.ChangeClientType
 */
var ChangeClientTypeEnum_ChangeClientType;
(function (ChangeClientTypeEnum_ChangeClientType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents an unclassified client type
     * unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Changes made through the "ads.google.com".
     * For example, changes made through campaign management.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_WEB_CLIENT = 2;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["GOOGLE_ADS_WEB_CLIENT"] = 2] = "GOOGLE_ADS_WEB_CLIENT";
    /**
     * Changes made through Google Ads automated rules.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_AUTOMATED_RULE = 3;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["GOOGLE_ADS_AUTOMATED_RULE"] = 3] = "GOOGLE_ADS_AUTOMATED_RULE";
    /**
     * Changes made through Google Ads scripts.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_SCRIPTS = 4;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["GOOGLE_ADS_SCRIPTS"] = 4] = "GOOGLE_ADS_SCRIPTS";
    /**
     * Changes made by Google Ads bulk upload.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_BULK_UPLOAD = 5;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["GOOGLE_ADS_BULK_UPLOAD"] = 5] = "GOOGLE_ADS_BULK_UPLOAD";
    /**
     * Changes made by Google Ads API.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_API = 6;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["GOOGLE_ADS_API"] = 6] = "GOOGLE_ADS_API";
    /**
     * Changes made by Google Ads Editor.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_EDITOR = 7;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["GOOGLE_ADS_EDITOR"] = 7] = "GOOGLE_ADS_EDITOR";
    /**
     * Changes made by Google Ads mobile app.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_MOBILE_APP = 8;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["GOOGLE_ADS_MOBILE_APP"] = 8] = "GOOGLE_ADS_MOBILE_APP";
    /**
     * Changes made through Google Ads recommendations.
     *
     * @generated from protobuf enum value: GOOGLE_ADS_RECOMMENDATIONS = 9;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["GOOGLE_ADS_RECOMMENDATIONS"] = 9] = "GOOGLE_ADS_RECOMMENDATIONS";
    /**
     * Changes made through Search Ads 360 Sync.
     *
     * @generated from protobuf enum value: SEARCH_ADS_360_SYNC = 10;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["SEARCH_ADS_360_SYNC"] = 10] = "SEARCH_ADS_360_SYNC";
    /**
     * Changes made through Search Ads 360 Post.
     *
     * @generated from protobuf enum value: SEARCH_ADS_360_POST = 11;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["SEARCH_ADS_360_POST"] = 11] = "SEARCH_ADS_360_POST";
    /**
     * Changes made through internal tools.
     * For example, when a user sets a URL template on an entity like a
     * Campaign, it's automatically wrapped with the SA360 Clickserver URL.
     *
     * @generated from protobuf enum value: INTERNAL_TOOL = 12;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["INTERNAL_TOOL"] = 12] = "INTERNAL_TOOL";
    /**
     * Types of changes that are not categorized, for example,
     * changes made by coupon redemption through Google Ads.
     *
     * @generated from protobuf enum value: OTHER = 13;
     */
    ChangeClientTypeEnum_ChangeClientType[ChangeClientTypeEnum_ChangeClientType["OTHER"] = 13] = "OTHER";
})(ChangeClientTypeEnum_ChangeClientType = exports.ChangeClientTypeEnum_ChangeClientType || (exports.ChangeClientTypeEnum_ChangeClientType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ChangeClientTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ChangeClientTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ChangeClientTypeEnum
 */
exports.ChangeClientTypeEnum = new ChangeClientTypeEnum$Type();
