// google-ads-api-client/dist/protos/bindings/ads/enums/change_status_operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeStatusOperationEnum = exports.ChangeStatusOperationEnum_ChangeStatusOperation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Status of the changed resource
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ChangeStatusOperationEnum.ChangeStatusOperation
 */
var ChangeStatusOperationEnum_ChangeStatusOperation;
(function (ChangeStatusOperationEnum_ChangeStatusOperation) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ChangeStatusOperationEnum_ChangeStatusOperation[ChangeStatusOperationEnum_ChangeStatusOperation["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents an unclassified resource unknown
     * in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ChangeStatusOperationEnum_ChangeStatusOperation[ChangeStatusOperationEnum_ChangeStatusOperation["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The resource was created.
     *
     * @generated from protobuf enum value: ADDED = 2;
     */
    ChangeStatusOperationEnum_ChangeStatusOperation[ChangeStatusOperationEnum_ChangeStatusOperation["ADDED"] = 2] = "ADDED";
    /**
     * The resource was modified.
     *
     * @generated from protobuf enum value: CHANGED = 3;
     */
    ChangeStatusOperationEnum_ChangeStatusOperation[ChangeStatusOperationEnum_ChangeStatusOperation["CHANGED"] = 3] = "CHANGED";
    /**
     * The resource was removed.
     *
     * @generated from protobuf enum value: REMOVED = 4;
     */
    ChangeStatusOperationEnum_ChangeStatusOperation[ChangeStatusOperationEnum_ChangeStatusOperation["REMOVED"] = 4] = "REMOVED";
})(ChangeStatusOperationEnum_ChangeStatusOperation = exports.ChangeStatusOperationEnum_ChangeStatusOperation || (exports.ChangeStatusOperationEnum_ChangeStatusOperation = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ChangeStatusOperationEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ChangeStatusOperationEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ChangeStatusOperationEnum
 */
exports.ChangeStatusOperationEnum = new ChangeStatusOperationEnum$Type();
