// google-ads-api-client/dist/protos/bindings/ads/enums/change_status_resource_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeStatusResourceTypeEnum = exports.ChangeStatusResourceTypeEnum_ChangeStatusResourceType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the resource types support by the ChangeStatus resource.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ChangeStatusResourceTypeEnum.ChangeStatusResourceType
 */
var ChangeStatusResourceTypeEnum_ChangeStatusResourceType;
(function (ChangeStatusResourceTypeEnum_ChangeStatusResourceType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents an unclassified resource unknown
     * in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An AdGroup resource change.
     *
     * @generated from protobuf enum value: AD_GROUP = 3;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["AD_GROUP"] = 3] = "AD_GROUP";
    /**
     * An AdGroupAd resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_AD = 4;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["AD_GROUP_AD"] = 4] = "AD_GROUP_AD";
    /**
     * An AdGroupCriterion resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_CRITERION = 5;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["AD_GROUP_CRITERION"] = 5] = "AD_GROUP_CRITERION";
    /**
     * A Campaign resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN = 6;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["CAMPAIGN"] = 6] = "CAMPAIGN";
    /**
     * A CampaignCriterion resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_CRITERION = 7;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["CAMPAIGN_CRITERION"] = 7] = "CAMPAIGN_CRITERION";
    /**
     * A Feed resource change.
     *
     * @generated from protobuf enum value: FEED = 9;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["FEED"] = 9] = "FEED";
    /**
     * A FeedItem resource change.
     *
     * @generated from protobuf enum value: FEED_ITEM = 10;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["FEED_ITEM"] = 10] = "FEED_ITEM";
    /**
     * An AdGroupFeed resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_FEED = 11;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["AD_GROUP_FEED"] = 11] = "AD_GROUP_FEED";
    /**
     * A CampaignFeed resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_FEED = 12;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["CAMPAIGN_FEED"] = 12] = "CAMPAIGN_FEED";
    /**
     * An AdGroupBidModifier resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_BID_MODIFIER = 13;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["AD_GROUP_BID_MODIFIER"] = 13] = "AD_GROUP_BID_MODIFIER";
    /**
     * A SharedSet resource change.
     *
     * @generated from protobuf enum value: SHARED_SET = 14;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["SHARED_SET"] = 14] = "SHARED_SET";
    /**
     * A CampaignSharedSet resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_SHARED_SET = 15;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["CAMPAIGN_SHARED_SET"] = 15] = "CAMPAIGN_SHARED_SET";
    /**
     * An Asset resource change.
     *
     * @generated from protobuf enum value: ASSET = 16;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["ASSET"] = 16] = "ASSET";
    /**
     * A CustomerAsset resource change.
     *
     * @generated from protobuf enum value: CUSTOMER_ASSET = 17;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["CUSTOMER_ASSET"] = 17] = "CUSTOMER_ASSET";
    /**
     * A CampaignAsset resource change.
     *
     * @generated from protobuf enum value: CAMPAIGN_ASSET = 18;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["CAMPAIGN_ASSET"] = 18] = "CAMPAIGN_ASSET";
    /**
     * An AdGroupAsset resource change.
     *
     * @generated from protobuf enum value: AD_GROUP_ASSET = 19;
     */
    ChangeStatusResourceTypeEnum_ChangeStatusResourceType[ChangeStatusResourceTypeEnum_ChangeStatusResourceType["AD_GROUP_ASSET"] = 19] = "AD_GROUP_ASSET";
})(ChangeStatusResourceTypeEnum_ChangeStatusResourceType = exports.ChangeStatusResourceTypeEnum_ChangeStatusResourceType || (exports.ChangeStatusResourceTypeEnum_ChangeStatusResourceType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ChangeStatusResourceTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ChangeStatusResourceTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ChangeStatusResourceTypeEnum
 */
exports.ChangeStatusResourceTypeEnum = new ChangeStatusResourceTypeEnum$Type();
