// google-ads-api-client/dist/protos/bindings/ads/enums/click_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClickTypeEnum = exports.ClickTypeEnum_ClickType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates Google Ads click types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ClickTypeEnum.ClickType
 */
var ClickTypeEnum_ClickType;
(function (ClickTypeEnum_ClickType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * App engagement ad deep link.
     *
     * @generated from protobuf enum value: APP_DEEPLINK = 2;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["APP_DEEPLINK"] = 2] = "APP_DEEPLINK";
    /**
     * Breadcrumbs.
     *
     * @generated from protobuf enum value: BREADCRUMBS = 3;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["BREADCRUMBS"] = 3] = "BREADCRUMBS";
    /**
     * Broadband Plan.
     *
     * @generated from protobuf enum value: BROADBAND_PLAN = 4;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["BROADBAND_PLAN"] = 4] = "BROADBAND_PLAN";
    /**
     * Manually dialed phone calls.
     *
     * @generated from protobuf enum value: CALL_TRACKING = 5;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["CALL_TRACKING"] = 5] = "CALL_TRACKING";
    /**
     * Phone calls.
     *
     * @generated from protobuf enum value: CALLS = 6;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["CALLS"] = 6] = "CALLS";
    /**
     * Click on engagement ad.
     *
     * @generated from protobuf enum value: CLICK_ON_ENGAGEMENT_AD = 7;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["CLICK_ON_ENGAGEMENT_AD"] = 7] = "CLICK_ON_ENGAGEMENT_AD";
    /**
     * Driving direction.
     *
     * @generated from protobuf enum value: GET_DIRECTIONS = 8;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["GET_DIRECTIONS"] = 8] = "GET_DIRECTIONS";
    /**
     * Get location details.
     *
     * @generated from protobuf enum value: LOCATION_EXPANSION = 9;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["LOCATION_EXPANSION"] = 9] = "LOCATION_EXPANSION";
    /**
     * Call.
     *
     * @generated from protobuf enum value: LOCATION_FORMAT_CALL = 10;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["LOCATION_FORMAT_CALL"] = 10] = "LOCATION_FORMAT_CALL";
    /**
     * Directions.
     *
     * @generated from protobuf enum value: LOCATION_FORMAT_DIRECTIONS = 11;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["LOCATION_FORMAT_DIRECTIONS"] = 11] = "LOCATION_FORMAT_DIRECTIONS";
    /**
     * Image(s).
     *
     * @generated from protobuf enum value: LOCATION_FORMAT_IMAGE = 12;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["LOCATION_FORMAT_IMAGE"] = 12] = "LOCATION_FORMAT_IMAGE";
    /**
     * Go to landing page.
     *
     * @generated from protobuf enum value: LOCATION_FORMAT_LANDING_PAGE = 13;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["LOCATION_FORMAT_LANDING_PAGE"] = 13] = "LOCATION_FORMAT_LANDING_PAGE";
    /**
     * Map.
     *
     * @generated from protobuf enum value: LOCATION_FORMAT_MAP = 14;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["LOCATION_FORMAT_MAP"] = 14] = "LOCATION_FORMAT_MAP";
    /**
     * Go to store info.
     *
     * @generated from protobuf enum value: LOCATION_FORMAT_STORE_INFO = 15;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["LOCATION_FORMAT_STORE_INFO"] = 15] = "LOCATION_FORMAT_STORE_INFO";
    /**
     * Text.
     *
     * @generated from protobuf enum value: LOCATION_FORMAT_TEXT = 16;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["LOCATION_FORMAT_TEXT"] = 16] = "LOCATION_FORMAT_TEXT";
    /**
     * Mobile phone calls.
     *
     * @generated from protobuf enum value: MOBILE_CALL_TRACKING = 17;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["MOBILE_CALL_TRACKING"] = 17] = "MOBILE_CALL_TRACKING";
    /**
     * Print offer.
     *
     * @generated from protobuf enum value: OFFER_PRINTS = 18;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["OFFER_PRINTS"] = 18] = "OFFER_PRINTS";
    /**
     * Other.
     *
     * @generated from protobuf enum value: OTHER = 19;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["OTHER"] = 19] = "OTHER";
    /**
     * Product plusbox offer.
     *
     * @generated from protobuf enum value: PRODUCT_EXTENSION_CLICKS = 20;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRODUCT_EXTENSION_CLICKS"] = 20] = "PRODUCT_EXTENSION_CLICKS";
    /**
     * Shopping - Product - Online.
     *
     * @generated from protobuf enum value: PRODUCT_LISTING_AD_CLICKS = 21;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRODUCT_LISTING_AD_CLICKS"] = 21] = "PRODUCT_LISTING_AD_CLICKS";
    /**
     * Sitelink.
     *
     * @generated from protobuf enum value: SITELINKS = 22;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SITELINKS"] = 22] = "SITELINKS";
    /**
     * Show nearby locations.
     *
     * @generated from protobuf enum value: STORE_LOCATOR = 23;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["STORE_LOCATOR"] = 23] = "STORE_LOCATOR";
    /**
     * Headline.
     *
     * @generated from protobuf enum value: URL_CLICKS = 25;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["URL_CLICKS"] = 25] = "URL_CLICKS";
    /**
     * App store.
     *
     * @generated from protobuf enum value: VIDEO_APP_STORE_CLICKS = 26;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["VIDEO_APP_STORE_CLICKS"] = 26] = "VIDEO_APP_STORE_CLICKS";
    /**
     * Call-to-Action overlay.
     *
     * @generated from protobuf enum value: VIDEO_CALL_TO_ACTION_CLICKS = 27;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["VIDEO_CALL_TO_ACTION_CLICKS"] = 27] = "VIDEO_CALL_TO_ACTION_CLICKS";
    /**
     * Cards.
     *
     * @generated from protobuf enum value: VIDEO_CARD_ACTION_HEADLINE_CLICKS = 28;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["VIDEO_CARD_ACTION_HEADLINE_CLICKS"] = 28] = "VIDEO_CARD_ACTION_HEADLINE_CLICKS";
    /**
     * End cap.
     *
     * @generated from protobuf enum value: VIDEO_END_CAP_CLICKS = 29;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["VIDEO_END_CAP_CLICKS"] = 29] = "VIDEO_END_CAP_CLICKS";
    /**
     * Website.
     *
     * @generated from protobuf enum value: VIDEO_WEBSITE_CLICKS = 30;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["VIDEO_WEBSITE_CLICKS"] = 30] = "VIDEO_WEBSITE_CLICKS";
    /**
     * Visual Sitelinks.
     *
     * @generated from protobuf enum value: VISUAL_SITELINKS = 31;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["VISUAL_SITELINKS"] = 31] = "VISUAL_SITELINKS";
    /**
     * Wireless Plan.
     *
     * @generated from protobuf enum value: WIRELESS_PLAN = 32;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["WIRELESS_PLAN"] = 32] = "WIRELESS_PLAN";
    /**
     * Shopping - Product - Local.
     *
     * @generated from protobuf enum value: PRODUCT_LISTING_AD_LOCAL = 33;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRODUCT_LISTING_AD_LOCAL"] = 33] = "PRODUCT_LISTING_AD_LOCAL";
    /**
     * Shopping - Product - MultiChannel Local.
     *
     * @generated from protobuf enum value: PRODUCT_LISTING_AD_MULTICHANNEL_LOCAL = 34;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRODUCT_LISTING_AD_MULTICHANNEL_LOCAL"] = 34] = "PRODUCT_LISTING_AD_MULTICHANNEL_LOCAL";
    /**
     * Shopping - Product - MultiChannel Online.
     *
     * @generated from protobuf enum value: PRODUCT_LISTING_AD_MULTICHANNEL_ONLINE = 35;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRODUCT_LISTING_AD_MULTICHANNEL_ONLINE"] = 35] = "PRODUCT_LISTING_AD_MULTICHANNEL_ONLINE";
    /**
     * Shopping - Product - Coupon.
     *
     * @generated from protobuf enum value: PRODUCT_LISTING_ADS_COUPON = 36;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRODUCT_LISTING_ADS_COUPON"] = 36] = "PRODUCT_LISTING_ADS_COUPON";
    /**
     * Shopping - Product - Sell on Google.
     *
     * @generated from protobuf enum value: PRODUCT_LISTING_AD_TRANSACTABLE = 37;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRODUCT_LISTING_AD_TRANSACTABLE"] = 37] = "PRODUCT_LISTING_AD_TRANSACTABLE";
    /**
     * Shopping - Product - App engagement ad deep link.
     *
     * @generated from protobuf enum value: PRODUCT_AD_APP_DEEPLINK = 38;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRODUCT_AD_APP_DEEPLINK"] = 38] = "PRODUCT_AD_APP_DEEPLINK";
    /**
     * Shopping - Showcase - Category.
     *
     * @generated from protobuf enum value: SHOWCASE_AD_CATEGORY_LINK = 39;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SHOWCASE_AD_CATEGORY_LINK"] = 39] = "SHOWCASE_AD_CATEGORY_LINK";
    /**
     * Shopping - Showcase - Local storefront.
     *
     * @generated from protobuf enum value: SHOWCASE_AD_LOCAL_STOREFRONT_LINK = 40;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SHOWCASE_AD_LOCAL_STOREFRONT_LINK"] = 40] = "SHOWCASE_AD_LOCAL_STOREFRONT_LINK";
    /**
     * Shopping - Showcase - Online product.
     *
     * @generated from protobuf enum value: SHOWCASE_AD_ONLINE_PRODUCT_LINK = 42;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SHOWCASE_AD_ONLINE_PRODUCT_LINK"] = 42] = "SHOWCASE_AD_ONLINE_PRODUCT_LINK";
    /**
     * Shopping - Showcase - Local product.
     *
     * @generated from protobuf enum value: SHOWCASE_AD_LOCAL_PRODUCT_LINK = 43;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SHOWCASE_AD_LOCAL_PRODUCT_LINK"] = 43] = "SHOWCASE_AD_LOCAL_PRODUCT_LINK";
    /**
     * Promotion Extension.
     *
     * @generated from protobuf enum value: PROMOTION_EXTENSION = 44;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PROMOTION_EXTENSION"] = 44] = "PROMOTION_EXTENSION";
    /**
     * Ad Headline.
     *
     * @generated from protobuf enum value: SWIPEABLE_GALLERY_AD_HEADLINE = 45;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SWIPEABLE_GALLERY_AD_HEADLINE"] = 45] = "SWIPEABLE_GALLERY_AD_HEADLINE";
    /**
     * Swipes.
     *
     * @generated from protobuf enum value: SWIPEABLE_GALLERY_AD_SWIPES = 46;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SWIPEABLE_GALLERY_AD_SWIPES"] = 46] = "SWIPEABLE_GALLERY_AD_SWIPES";
    /**
     * See More.
     *
     * @generated from protobuf enum value: SWIPEABLE_GALLERY_AD_SEE_MORE = 47;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SWIPEABLE_GALLERY_AD_SEE_MORE"] = 47] = "SWIPEABLE_GALLERY_AD_SEE_MORE";
    /**
     * Sitelink 1.
     *
     * @generated from protobuf enum value: SWIPEABLE_GALLERY_AD_SITELINK_ONE = 48;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SWIPEABLE_GALLERY_AD_SITELINK_ONE"] = 48] = "SWIPEABLE_GALLERY_AD_SITELINK_ONE";
    /**
     * Sitelink 2.
     *
     * @generated from protobuf enum value: SWIPEABLE_GALLERY_AD_SITELINK_TWO = 49;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SWIPEABLE_GALLERY_AD_SITELINK_TWO"] = 49] = "SWIPEABLE_GALLERY_AD_SITELINK_TWO";
    /**
     * Sitelink 3.
     *
     * @generated from protobuf enum value: SWIPEABLE_GALLERY_AD_SITELINK_THREE = 50;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SWIPEABLE_GALLERY_AD_SITELINK_THREE"] = 50] = "SWIPEABLE_GALLERY_AD_SITELINK_THREE";
    /**
     * Sitelink 4.
     *
     * @generated from protobuf enum value: SWIPEABLE_GALLERY_AD_SITELINK_FOUR = 51;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SWIPEABLE_GALLERY_AD_SITELINK_FOUR"] = 51] = "SWIPEABLE_GALLERY_AD_SITELINK_FOUR";
    /**
     * Sitelink 5.
     *
     * @generated from protobuf enum value: SWIPEABLE_GALLERY_AD_SITELINK_FIVE = 52;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SWIPEABLE_GALLERY_AD_SITELINK_FIVE"] = 52] = "SWIPEABLE_GALLERY_AD_SITELINK_FIVE";
    /**
     * Hotel price.
     *
     * @generated from protobuf enum value: HOTEL_PRICE = 53;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["HOTEL_PRICE"] = 53] = "HOTEL_PRICE";
    /**
     * Price Extension.
     *
     * @generated from protobuf enum value: PRICE_EXTENSION = 54;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["PRICE_EXTENSION"] = 54] = "PRICE_EXTENSION";
    /**
     * Book on Google hotel room selection.
     *
     * @generated from protobuf enum value: HOTEL_BOOK_ON_GOOGLE_ROOM_SELECTION = 55;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["HOTEL_BOOK_ON_GOOGLE_ROOM_SELECTION"] = 55] = "HOTEL_BOOK_ON_GOOGLE_ROOM_SELECTION";
    /**
     * Shopping - Comparison Listing.
     *
     * @generated from protobuf enum value: SHOPPING_COMPARISON_LISTING = 56;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["SHOPPING_COMPARISON_LISTING"] = 56] = "SHOPPING_COMPARISON_LISTING";
    /**
     * Cross-network. From Performance Max and Discovery Campaigns.
     *
     * @generated from protobuf enum value: CROSS_NETWORK = 57;
     */
    ClickTypeEnum_ClickType[ClickTypeEnum_ClickType["CROSS_NETWORK"] = 57] = "CROSS_NETWORK";
})(ClickTypeEnum_ClickType = exports.ClickTypeEnum_ClickType || (exports.ClickTypeEnum_ClickType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ClickTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ClickTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ClickTypeEnum
 */
exports.ClickTypeEnum = new ClickTypeEnum$Type();
