// google-ads-api-client/dist/protos/bindings/ads/enums/combined_audience_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CombinedAudienceStatusEnum = exports.CombinedAudienceStatusEnum_CombinedAudienceStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible combined audience status types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CombinedAudienceStatusEnum.CombinedAudienceStatus
 */
var CombinedAudienceStatusEnum_CombinedAudienceStatus;
(function (CombinedAudienceStatusEnum_CombinedAudienceStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CombinedAudienceStatusEnum_CombinedAudienceStatus[CombinedAudienceStatusEnum_CombinedAudienceStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CombinedAudienceStatusEnum_CombinedAudienceStatus[CombinedAudienceStatusEnum_CombinedAudienceStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Enabled status - combined audience is enabled and can be targeted.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CombinedAudienceStatusEnum_CombinedAudienceStatus[CombinedAudienceStatusEnum_CombinedAudienceStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Removed status - combined audience is removed and cannot be used for
     * targeting.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    CombinedAudienceStatusEnum_CombinedAudienceStatus[CombinedAudienceStatusEnum_CombinedAudienceStatus["REMOVED"] = 3] = "REMOVED";
})(CombinedAudienceStatusEnum_CombinedAudienceStatus = exports.CombinedAudienceStatusEnum_CombinedAudienceStatus || (exports.CombinedAudienceStatusEnum_CombinedAudienceStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CombinedAudienceStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CombinedAudienceStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CombinedAudienceStatusEnum
 */
exports.CombinedAudienceStatusEnum = new CombinedAudienceStatusEnum$Type();
