// google-ads-api-client/dist/protos/bindings/ads/enums/content_label_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentLabelTypeEnum = exports.ContentLabelTypeEnum_ContentLabelType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the content label types supported by ContentLabel criterion.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ContentLabelTypeEnum.ContentLabelType
 */
var ContentLabelTypeEnum_ContentLabelType;
(function (ContentLabelTypeEnum_ContentLabelType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Sexually suggestive content.
     *
     * @generated from protobuf enum value: SEXUALLY_SUGGESTIVE = 2;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["SEXUALLY_SUGGESTIVE"] = 2] = "SEXUALLY_SUGGESTIVE";
    /**
     * Below the fold placement.
     *
     * @generated from protobuf enum value: BELOW_THE_FOLD = 3;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["BELOW_THE_FOLD"] = 3] = "BELOW_THE_FOLD";
    /**
     * Parked domain.
     *
     * @generated from protobuf enum value: PARKED_DOMAIN = 4;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["PARKED_DOMAIN"] = 4] = "PARKED_DOMAIN";
    /**
     * Juvenile, gross & bizarre content.
     *
     * @generated from protobuf enum value: JUVENILE = 6;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["JUVENILE"] = 6] = "JUVENILE";
    /**
     * Profanity & rough language.
     *
     * @generated from protobuf enum value: PROFANITY = 7;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["PROFANITY"] = 7] = "PROFANITY";
    /**
     * Death & tragedy.
     *
     * @generated from protobuf enum value: TRAGEDY = 8;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["TRAGEDY"] = 8] = "TRAGEDY";
    /**
     * Video.
     *
     * @generated from protobuf enum value: VIDEO = 9;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["VIDEO"] = 9] = "VIDEO";
    /**
     * Content rating: G.
     *
     * @generated from protobuf enum value: VIDEO_RATING_DV_G = 10;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["VIDEO_RATING_DV_G"] = 10] = "VIDEO_RATING_DV_G";
    /**
     * Content rating: PG.
     *
     * @generated from protobuf enum value: VIDEO_RATING_DV_PG = 11;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["VIDEO_RATING_DV_PG"] = 11] = "VIDEO_RATING_DV_PG";
    /**
     * Content rating: T.
     *
     * @generated from protobuf enum value: VIDEO_RATING_DV_T = 12;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["VIDEO_RATING_DV_T"] = 12] = "VIDEO_RATING_DV_T";
    /**
     * Content rating: MA.
     *
     * @generated from protobuf enum value: VIDEO_RATING_DV_MA = 13;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["VIDEO_RATING_DV_MA"] = 13] = "VIDEO_RATING_DV_MA";
    /**
     * Content rating: not yet rated.
     *
     * @generated from protobuf enum value: VIDEO_NOT_YET_RATED = 14;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["VIDEO_NOT_YET_RATED"] = 14] = "VIDEO_NOT_YET_RATED";
    /**
     * Embedded video.
     *
     * @generated from protobuf enum value: EMBEDDED_VIDEO = 15;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["EMBEDDED_VIDEO"] = 15] = "EMBEDDED_VIDEO";
    /**
     * Live streaming video.
     *
     * @generated from protobuf enum value: LIVE_STREAMING_VIDEO = 16;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["LIVE_STREAMING_VIDEO"] = 16] = "LIVE_STREAMING_VIDEO";
    /**
     * Sensitive social issues.
     *
     * @generated from protobuf enum value: SOCIAL_ISSUES = 17;
     */
    ContentLabelTypeEnum_ContentLabelType[ContentLabelTypeEnum_ContentLabelType["SOCIAL_ISSUES"] = 17] = "SOCIAL_ISSUES";
})(ContentLabelTypeEnum_ContentLabelType = exports.ContentLabelTypeEnum_ContentLabelType || (exports.ContentLabelTypeEnum_ContentLabelType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ContentLabelTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ContentLabelTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ContentLabelTypeEnum
 */
exports.ContentLabelTypeEnum = new ContentLabelTypeEnum$Type();
