// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_value_rule_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionValueRuleStatusEnum = exports.ConversionValueRuleStatusEnum_ConversionValueRuleStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a conversion value rule.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionValueRuleStatusEnum.ConversionValueRuleStatus
 */
var ConversionValueRuleStatusEnum_ConversionValueRuleStatus;
(function (ConversionValueRuleStatusEnum_ConversionValueRuleStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionValueRuleStatusEnum_ConversionValueRuleStatus[ConversionValueRuleStatusEnum_ConversionValueRuleStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionValueRuleStatusEnum_ConversionValueRuleStatus[ConversionValueRuleStatusEnum_ConversionValueRuleStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Conversion Value Rule is enabled and can be applied.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    ConversionValueRuleStatusEnum_ConversionValueRuleStatus[ConversionValueRuleStatusEnum_ConversionValueRuleStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Conversion Value Rule is permanently deleted and can't be applied.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    ConversionValueRuleStatusEnum_ConversionValueRuleStatus[ConversionValueRuleStatusEnum_ConversionValueRuleStatus["REMOVED"] = 3] = "REMOVED";
    /**
     * Conversion Value Rule is paused, but can be re-enabled.
     *
     * @generated from protobuf enum value: PAUSED = 4;
     */
    ConversionValueRuleStatusEnum_ConversionValueRuleStatus[ConversionValueRuleStatusEnum_ConversionValueRuleStatus["PAUSED"] = 4] = "PAUSED";
})(ConversionValueRuleStatusEnum_ConversionValueRuleStatus = exports.ConversionValueRuleStatusEnum_ConversionValueRuleStatus || (exports.ConversionValueRuleStatusEnum_ConversionValueRuleStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionValueRuleStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionValueRuleStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionValueRuleStatusEnum
 */
exports.ConversionValueRuleStatusEnum = new ConversionValueRuleStatusEnum$Type();
