// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_adjustment_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionAdjustmentTypeEnum = exports.ConversionAdjustmentTypeEnum_ConversionAdjustmentType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The different actions advertisers can take to adjust the conversions that
 * they already reported. Retractions negate a conversion. Restatements change
 * the value of a conversion.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionAdjustmentTypeEnum.ConversionAdjustmentType
 */
var ConversionAdjustmentTypeEnum_ConversionAdjustmentType;
(function (ConversionAdjustmentTypeEnum_ConversionAdjustmentType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionAdjustmentTypeEnum_ConversionAdjustmentType[ConversionAdjustmentTypeEnum_ConversionAdjustmentType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionAdjustmentTypeEnum_ConversionAdjustmentType[ConversionAdjustmentTypeEnum_ConversionAdjustmentType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Negates a conversion so that its total value and count are both zero.
     *
     * @generated from protobuf enum value: RETRACTION = 2;
     */
    ConversionAdjustmentTypeEnum_ConversionAdjustmentType[ConversionAdjustmentTypeEnum_ConversionAdjustmentType["RETRACTION"] = 2] = "RETRACTION";
    /**
     * Changes the value of a conversion.
     *
     * @generated from protobuf enum value: RESTATEMENT = 3;
     */
    ConversionAdjustmentTypeEnum_ConversionAdjustmentType[ConversionAdjustmentTypeEnum_ConversionAdjustmentType["RESTATEMENT"] = 3] = "RESTATEMENT";
    /**
     * Supplements an existing conversion with provided user identifiers and
     * user agent, which can be used by Google to enhance the conversion count.
     *
     * @generated from protobuf enum value: ENHANCEMENT = 4;
     */
    ConversionAdjustmentTypeEnum_ConversionAdjustmentType[ConversionAdjustmentTypeEnum_ConversionAdjustmentType["ENHANCEMENT"] = 4] = "ENHANCEMENT";
})(ConversionAdjustmentTypeEnum_ConversionAdjustmentType = exports.ConversionAdjustmentTypeEnum_ConversionAdjustmentType || (exports.ConversionAdjustmentTypeEnum_ConversionAdjustmentType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionAdjustmentTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionAdjustmentTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionAdjustmentTypeEnum
 */
exports.ConversionAdjustmentTypeEnum = new ConversionAdjustmentTypeEnum$Type();
