// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_environment_enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionEnvironmentEnum = exports.ConversionEnvironmentEnum_ConversionEnvironment = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Conversion environment of the uploaded conversion.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionEnvironmentEnum.ConversionEnvironment
 */
var ConversionEnvironmentEnum_ConversionEnvironment;
(function (ConversionEnvironmentEnum_ConversionEnvironment) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionEnvironmentEnum_ConversionEnvironment[ConversionEnvironmentEnum_ConversionEnvironment["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionEnvironmentEnum_ConversionEnvironment[ConversionEnvironmentEnum_ConversionEnvironment["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The conversion was recorded on an app.
     *
     * @generated from protobuf enum value: APP = 2;
     */
    ConversionEnvironmentEnum_ConversionEnvironment[ConversionEnvironmentEnum_ConversionEnvironment["APP"] = 2] = "APP";
    /**
     * The conversion was recorded on a website.
     *
     * @generated from protobuf enum value: WEB = 3;
     */
    ConversionEnvironmentEnum_ConversionEnvironment[ConversionEnvironmentEnum_ConversionEnvironment["WEB"] = 3] = "WEB";
})(ConversionEnvironmentEnum_ConversionEnvironment = exports.ConversionEnvironmentEnum_ConversionEnvironment || (exports.ConversionEnvironmentEnum_ConversionEnvironment = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionEnvironmentEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionEnvironmentEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionEnvironmentEnum
 */
exports.ConversionEnvironmentEnum = new ConversionEnvironmentEnum$Type();
