// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_value_rule_primary_dimension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionValueRulePrimaryDimensionEnum = exports.ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Identifies the primary dimension for conversion value rule stats.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionValueRulePrimaryDimensionEnum.ConversionValueRulePrimaryDimension
 */
var ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension;
(function (ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * For no-value-rule-applied conversions after value rule is enabled.
     *
     * @generated from protobuf enum value: NO_RULE_APPLIED = 2;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["NO_RULE_APPLIED"] = 2] = "NO_RULE_APPLIED";
    /**
     * Below are for value-rule-applied conversions:
     * The original stats.
     *
     * @generated from protobuf enum value: ORIGINAL = 3;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["ORIGINAL"] = 3] = "ORIGINAL";
    /**
     * When a new or returning customer condition is satisfied.
     *
     * @generated from protobuf enum value: NEW_VS_RETURNING_USER = 4;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["NEW_VS_RETURNING_USER"] = 4] = "NEW_VS_RETURNING_USER";
    /**
     * When a query-time Geo location condition is satisfied.
     *
     * @generated from protobuf enum value: GEO_LOCATION = 5;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["GEO_LOCATION"] = 5] = "GEO_LOCATION";
    /**
     * When a query-time browsing device condition is satisfied.
     *
     * @generated from protobuf enum value: DEVICE = 6;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["DEVICE"] = 6] = "DEVICE";
    /**
     * When a query-time audience condition is satisfied.
     *
     * @generated from protobuf enum value: AUDIENCE = 7;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["AUDIENCE"] = 7] = "AUDIENCE";
    /**
     * When multiple rules are applied.
     *
     * @generated from protobuf enum value: MULTIPLE = 8;
     */
    ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension[ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension["MULTIPLE"] = 8] = "MULTIPLE";
})(ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension = exports.ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension || (exports.ConversionValueRulePrimaryDimensionEnum_ConversionValueRulePrimaryDimension = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionValueRulePrimaryDimensionEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionValueRulePrimaryDimensionEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionValueRulePrimaryDimensionEnum
 */
exports.ConversionValueRulePrimaryDimensionEnum = new ConversionValueRulePrimaryDimensionEnum$Type();
