// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_origin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionOriginEnum = exports.ConversionOriginEnum_ConversionOrigin = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible places where a conversion can occur.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionOriginEnum.ConversionOrigin
 */
var ConversionOriginEnum_ConversionOrigin;
(function (ConversionOriginEnum_ConversionOrigin) {
    /**
     * The conversion origin has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionOriginEnum_ConversionOrigin[ConversionOriginEnum_ConversionOrigin["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The conversion origin is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionOriginEnum_ConversionOrigin[ConversionOriginEnum_ConversionOrigin["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Conversion that occurs when a user visits a website or takes an action
     * there after viewing an ad.
     *
     * @generated from protobuf enum value: WEBSITE = 2;
     */
    ConversionOriginEnum_ConversionOrigin[ConversionOriginEnum_ConversionOrigin["WEBSITE"] = 2] = "WEBSITE";
    /**
     * Conversions reported by an offline pipeline which collects local actions
     * from Google-hosted pages (for example, Google Maps, Google Place Page,
     * etc) and attributes them to relevant ad events.
     *
     * @generated from protobuf enum value: GOOGLE_HOSTED = 3;
     */
    ConversionOriginEnum_ConversionOrigin[ConversionOriginEnum_ConversionOrigin["GOOGLE_HOSTED"] = 3] = "GOOGLE_HOSTED";
    /**
     * Conversion that occurs when a user performs an action through any app
     * platforms.
     *
     * @generated from protobuf enum value: APP = 4;
     */
    ConversionOriginEnum_ConversionOrigin[ConversionOriginEnum_ConversionOrigin["APP"] = 4] = "APP";
    /**
     * Conversion that occurs when a user makes a call from ads.
     *
     * @generated from protobuf enum value: CALL_FROM_ADS = 5;
     */
    ConversionOriginEnum_ConversionOrigin[ConversionOriginEnum_ConversionOrigin["CALL_FROM_ADS"] = 5] = "CALL_FROM_ADS";
    /**
     * Conversion that occurs when a user visits or makes a purchase at a
     * physical store.
     *
     * @generated from protobuf enum value: STORE = 6;
     */
    ConversionOriginEnum_ConversionOrigin[ConversionOriginEnum_ConversionOrigin["STORE"] = 6] = "STORE";
    /**
     * Conversion that occurs on YouTube.
     *
     * @generated from protobuf enum value: YOUTUBE_HOSTED = 7;
     */
    ConversionOriginEnum_ConversionOrigin[ConversionOriginEnum_ConversionOrigin["YOUTUBE_HOSTED"] = 7] = "YOUTUBE_HOSTED";
})(ConversionOriginEnum_ConversionOrigin = exports.ConversionOriginEnum_ConversionOrigin || (exports.ConversionOriginEnum_ConversionOrigin = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionOriginEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionOriginEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionOriginEnum
 */
exports.ConversionOriginEnum = new ConversionOriginEnum$Type();
