// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_action_counting_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionActionCountingTypeEnum = exports.ConversionActionCountingTypeEnum_ConversionActionCountingType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Indicates how conversions for this action will be counted. For more
 * information, see https://support.google.com/google-ads/answer/3438531.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionActionCountingTypeEnum.ConversionActionCountingType
 */
var ConversionActionCountingTypeEnum_ConversionActionCountingType;
(function (ConversionActionCountingTypeEnum_ConversionActionCountingType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionActionCountingTypeEnum_ConversionActionCountingType[ConversionActionCountingTypeEnum_ConversionActionCountingType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionActionCountingTypeEnum_ConversionActionCountingType[ConversionActionCountingTypeEnum_ConversionActionCountingType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Count only one conversion per click.
     *
     * @generated from protobuf enum value: ONE_PER_CLICK = 2;
     */
    ConversionActionCountingTypeEnum_ConversionActionCountingType[ConversionActionCountingTypeEnum_ConversionActionCountingType["ONE_PER_CLICK"] = 2] = "ONE_PER_CLICK";
    /**
     * Count all conversions per click.
     *
     * @generated from protobuf enum value: MANY_PER_CLICK = 3;
     */
    ConversionActionCountingTypeEnum_ConversionActionCountingType[ConversionActionCountingTypeEnum_ConversionActionCountingType["MANY_PER_CLICK"] = 3] = "MANY_PER_CLICK";
})(ConversionActionCountingTypeEnum_ConversionActionCountingType = exports.ConversionActionCountingTypeEnum_ConversionActionCountingType || (exports.ConversionActionCountingTypeEnum_ConversionActionCountingType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionActionCountingTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionActionCountingTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionActionCountingTypeEnum
 */
exports.ConversionActionCountingTypeEnum = new ConversionActionCountingTypeEnum$Type();
