// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_tracking_status_enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionTrackingStatusEnum = exports.ConversionTrackingStatusEnum_ConversionTrackingStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Conversion Tracking status of the customer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionTrackingStatusEnum.ConversionTrackingStatus
 */
var ConversionTrackingStatusEnum_ConversionTrackingStatus;
(function (ConversionTrackingStatusEnum_ConversionTrackingStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionTrackingStatusEnum_ConversionTrackingStatus[ConversionTrackingStatusEnum_ConversionTrackingStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionTrackingStatusEnum_ConversionTrackingStatus[ConversionTrackingStatusEnum_ConversionTrackingStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Customer does not use any conversion tracking.
     *
     * @generated from protobuf enum value: NOT_CONVERSION_TRACKED = 2;
     */
    ConversionTrackingStatusEnum_ConversionTrackingStatus[ConversionTrackingStatusEnum_ConversionTrackingStatus["NOT_CONVERSION_TRACKED"] = 2] = "NOT_CONVERSION_TRACKED";
    /**
     * The conversion actions are created and managed by this customer.
     *
     * @generated from protobuf enum value: CONVERSION_TRACKING_MANAGED_BY_SELF = 3;
     */
    ConversionTrackingStatusEnum_ConversionTrackingStatus[ConversionTrackingStatusEnum_ConversionTrackingStatus["CONVERSION_TRACKING_MANAGED_BY_SELF"] = 3] = "CONVERSION_TRACKING_MANAGED_BY_SELF";
    /**
     * The conversion actions are created and managed by the manager specified
     * in the request's `login-customer-id`.
     *
     * @generated from protobuf enum value: CONVERSION_TRACKING_MANAGED_BY_THIS_MANAGER = 4;
     */
    ConversionTrackingStatusEnum_ConversionTrackingStatus[ConversionTrackingStatusEnum_ConversionTrackingStatus["CONVERSION_TRACKING_MANAGED_BY_THIS_MANAGER"] = 4] = "CONVERSION_TRACKING_MANAGED_BY_THIS_MANAGER";
    /**
     * The conversion actions are created and managed by a manager different
     * from the customer or manager specified in the request's
     * `login-customer-id`.
     *
     * @generated from protobuf enum value: CONVERSION_TRACKING_MANAGED_BY_ANOTHER_MANAGER = 5;
     */
    ConversionTrackingStatusEnum_ConversionTrackingStatus[ConversionTrackingStatusEnum_ConversionTrackingStatus["CONVERSION_TRACKING_MANAGED_BY_ANOTHER_MANAGER"] = 5] = "CONVERSION_TRACKING_MANAGED_BY_ANOTHER_MANAGER";
})(ConversionTrackingStatusEnum_ConversionTrackingStatus = exports.ConversionTrackingStatusEnum_ConversionTrackingStatus || (exports.ConversionTrackingStatusEnum_ConversionTrackingStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionTrackingStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionTrackingStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionTrackingStatusEnum
 */
exports.ConversionTrackingStatusEnum = new ConversionTrackingStatusEnum$Type();
