// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_action_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionActionStatusEnum = exports.ConversionActionStatusEnum_ConversionActionStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a conversion action.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionActionStatusEnum.ConversionActionStatus
 */
var ConversionActionStatusEnum_ConversionActionStatus;
(function (ConversionActionStatusEnum_ConversionActionStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionActionStatusEnum_ConversionActionStatus[ConversionActionStatusEnum_ConversionActionStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionActionStatusEnum_ConversionActionStatus[ConversionActionStatusEnum_ConversionActionStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Conversions will be recorded.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    ConversionActionStatusEnum_ConversionActionStatus[ConversionActionStatusEnum_ConversionActionStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Conversions will not be recorded.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    ConversionActionStatusEnum_ConversionActionStatus[ConversionActionStatusEnum_ConversionActionStatus["REMOVED"] = 3] = "REMOVED";
    /**
     * Conversions will not be recorded and the conversion action will not
     * appear in the UI.
     *
     * @generated from protobuf enum value: HIDDEN = 4;
     */
    ConversionActionStatusEnum_ConversionActionStatus[ConversionActionStatusEnum_ConversionActionStatus["HIDDEN"] = 4] = "HIDDEN";
})(ConversionActionStatusEnum_ConversionActionStatus = exports.ConversionActionStatusEnum_ConversionActionStatus || (exports.ConversionActionStatusEnum_ConversionActionStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionActionStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionActionStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionActionStatusEnum
 */
exports.ConversionActionStatusEnum = new ConversionActionStatusEnum$Type();
