// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_custom_variable_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionCustomVariableStatusEnum = exports.ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a conversion custom variable.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionCustomVariableStatusEnum.ConversionCustomVariableStatus
 */
var ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus;
(function (ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus[ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus[ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The conversion custom variable is pending activation and will not
     * accrue stats until set to ENABLED.
     *
     * This status can't be used in CREATE and UPDATE requests.
     *
     * @generated from protobuf enum value: ACTIVATION_NEEDED = 2;
     */
    ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus[ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus["ACTIVATION_NEEDED"] = 2] = "ACTIVATION_NEEDED";
    /**
     * The conversion custom variable is enabled and will accrue stats.
     *
     * @generated from protobuf enum value: ENABLED = 3;
     */
    ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus[ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus["ENABLED"] = 3] = "ENABLED";
    /**
     * The conversion custom variable is paused and will not accrue stats
     * until set to ENABLED again.
     *
     * @generated from protobuf enum value: PAUSED = 4;
     */
    ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus[ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus["PAUSED"] = 4] = "PAUSED";
})(ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus = exports.ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus || (exports.ConversionCustomVariableStatusEnum_ConversionCustomVariableStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionCustomVariableStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionCustomVariableStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionCustomVariableStatusEnum
 */
exports.ConversionCustomVariableStatusEnum = new ConversionCustomVariableStatusEnum$Type();
