// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_attribution_event_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionAttributionEventTypeEnum = exports.ConversionAttributionEventTypeEnum_ConversionAttributionEventType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The event type of conversions that are attributed to.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionAttributionEventTypeEnum.ConversionAttributionEventType
 */
var ConversionAttributionEventTypeEnum_ConversionAttributionEventType;
(function (ConversionAttributionEventTypeEnum_ConversionAttributionEventType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionAttributionEventTypeEnum_ConversionAttributionEventType[ConversionAttributionEventTypeEnum_ConversionAttributionEventType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionAttributionEventTypeEnum_ConversionAttributionEventType[ConversionAttributionEventTypeEnum_ConversionAttributionEventType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The conversion is attributed to an impression.
     *
     * @generated from protobuf enum value: IMPRESSION = 2;
     */
    ConversionAttributionEventTypeEnum_ConversionAttributionEventType[ConversionAttributionEventTypeEnum_ConversionAttributionEventType["IMPRESSION"] = 2] = "IMPRESSION";
    /**
     * The conversion is attributed to an interaction.
     *
     * @generated from protobuf enum value: INTERACTION = 3;
     */
    ConversionAttributionEventTypeEnum_ConversionAttributionEventType[ConversionAttributionEventTypeEnum_ConversionAttributionEventType["INTERACTION"] = 3] = "INTERACTION";
})(ConversionAttributionEventTypeEnum_ConversionAttributionEventType = exports.ConversionAttributionEventTypeEnum_ConversionAttributionEventType || (exports.ConversionAttributionEventTypeEnum_ConversionAttributionEventType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionAttributionEventTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionAttributionEventTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionAttributionEventTypeEnum
 */
exports.ConversionAttributionEventTypeEnum = new ConversionAttributionEventTypeEnum$Type();
