// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_action_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionActionTypeEnum = exports.ConversionActionTypeEnum_ConversionActionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible types of a conversion action.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionActionTypeEnum.ConversionActionType
 */
var ConversionActionTypeEnum_ConversionActionType;
(function (ConversionActionTypeEnum_ConversionActionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Conversions that occur when a user clicks on an ad's call extension.
     *
     * @generated from protobuf enum value: AD_CALL = 2;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["AD_CALL"] = 2] = "AD_CALL";
    /**
     * Conversions that occur when a user on a mobile device clicks a phone
     * number.
     *
     * @generated from protobuf enum value: CLICK_TO_CALL = 3;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["CLICK_TO_CALL"] = 3] = "CLICK_TO_CALL";
    /**
     * Conversions that occur when a user downloads a mobile app from the Google
     * Play Store.
     *
     * @generated from protobuf enum value: GOOGLE_PLAY_DOWNLOAD = 4;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["GOOGLE_PLAY_DOWNLOAD"] = 4] = "GOOGLE_PLAY_DOWNLOAD";
    /**
     * Conversions that occur when a user makes a purchase in an app through
     * Android billing.
     *
     * @generated from protobuf enum value: GOOGLE_PLAY_IN_APP_PURCHASE = 5;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["GOOGLE_PLAY_IN_APP_PURCHASE"] = 5] = "GOOGLE_PLAY_IN_APP_PURCHASE";
    /**
     * Call conversions that are tracked by the advertiser and uploaded.
     *
     * @generated from protobuf enum value: UPLOAD_CALLS = 6;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["UPLOAD_CALLS"] = 6] = "UPLOAD_CALLS";
    /**
     * Conversions that are tracked by the advertiser and uploaded with
     * attributed clicks.
     *
     * @generated from protobuf enum value: UPLOAD_CLICKS = 7;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["UPLOAD_CLICKS"] = 7] = "UPLOAD_CLICKS";
    /**
     * Conversions that occur on a webpage.
     *
     * @generated from protobuf enum value: WEBPAGE = 8;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["WEBPAGE"] = 8] = "WEBPAGE";
    /**
     * Conversions that occur when a user calls a dynamically-generated phone
     * number from an advertiser's website.
     *
     * @generated from protobuf enum value: WEBSITE_CALL = 9;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["WEBSITE_CALL"] = 9] = "WEBSITE_CALL";
    /**
     * Store Sales conversion based on first-party or third-party merchant
     * data uploads.
     * Only customers on the allowlist can use store sales direct upload types.
     *
     * @generated from protobuf enum value: STORE_SALES_DIRECT_UPLOAD = 10;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["STORE_SALES_DIRECT_UPLOAD"] = 10] = "STORE_SALES_DIRECT_UPLOAD";
    /**
     * Store Sales conversion based on first-party or third-party merchant
     * data uploads and/or from in-store purchases using cards from payment
     * networks.
     * Only customers on the allowlist can use store sales types.
     * Read only.
     *
     * @generated from protobuf enum value: STORE_SALES = 11;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["STORE_SALES"] = 11] = "STORE_SALES";
    /**
     * Android app first open conversions tracked through Firebase.
     *
     * @generated from protobuf enum value: FIREBASE_ANDROID_FIRST_OPEN = 12;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["FIREBASE_ANDROID_FIRST_OPEN"] = 12] = "FIREBASE_ANDROID_FIRST_OPEN";
    /**
     * Android app in app purchase conversions tracked through Firebase.
     *
     * @generated from protobuf enum value: FIREBASE_ANDROID_IN_APP_PURCHASE = 13;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["FIREBASE_ANDROID_IN_APP_PURCHASE"] = 13] = "FIREBASE_ANDROID_IN_APP_PURCHASE";
    /**
     * Android app custom conversions tracked through Firebase.
     *
     * @generated from protobuf enum value: FIREBASE_ANDROID_CUSTOM = 14;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["FIREBASE_ANDROID_CUSTOM"] = 14] = "FIREBASE_ANDROID_CUSTOM";
    /**
     * iOS app first open conversions tracked through Firebase.
     *
     * @generated from protobuf enum value: FIREBASE_IOS_FIRST_OPEN = 15;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["FIREBASE_IOS_FIRST_OPEN"] = 15] = "FIREBASE_IOS_FIRST_OPEN";
    /**
     * iOS app in app purchase conversions tracked through Firebase.
     *
     * @generated from protobuf enum value: FIREBASE_IOS_IN_APP_PURCHASE = 16;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["FIREBASE_IOS_IN_APP_PURCHASE"] = 16] = "FIREBASE_IOS_IN_APP_PURCHASE";
    /**
     * iOS app custom conversions tracked through Firebase.
     *
     * @generated from protobuf enum value: FIREBASE_IOS_CUSTOM = 17;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["FIREBASE_IOS_CUSTOM"] = 17] = "FIREBASE_IOS_CUSTOM";
    /**
     * Android app first open conversions tracked through Third Party App
     * Analytics.
     *
     * @generated from protobuf enum value: THIRD_PARTY_APP_ANALYTICS_ANDROID_FIRST_OPEN = 18;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["THIRD_PARTY_APP_ANALYTICS_ANDROID_FIRST_OPEN"] = 18] = "THIRD_PARTY_APP_ANALYTICS_ANDROID_FIRST_OPEN";
    /**
     * Android app in app purchase conversions tracked through Third Party App
     * Analytics.
     *
     * @generated from protobuf enum value: THIRD_PARTY_APP_ANALYTICS_ANDROID_IN_APP_PURCHASE = 19;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["THIRD_PARTY_APP_ANALYTICS_ANDROID_IN_APP_PURCHASE"] = 19] = "THIRD_PARTY_APP_ANALYTICS_ANDROID_IN_APP_PURCHASE";
    /**
     * Android app custom conversions tracked through Third Party App Analytics.
     *
     * @generated from protobuf enum value: THIRD_PARTY_APP_ANALYTICS_ANDROID_CUSTOM = 20;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["THIRD_PARTY_APP_ANALYTICS_ANDROID_CUSTOM"] = 20] = "THIRD_PARTY_APP_ANALYTICS_ANDROID_CUSTOM";
    /**
     * iOS app first open conversions tracked through Third Party App Analytics.
     *
     * @generated from protobuf enum value: THIRD_PARTY_APP_ANALYTICS_IOS_FIRST_OPEN = 21;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["THIRD_PARTY_APP_ANALYTICS_IOS_FIRST_OPEN"] = 21] = "THIRD_PARTY_APP_ANALYTICS_IOS_FIRST_OPEN";
    /**
     * iOS app in app purchase conversions tracked through Third Party App
     * Analytics.
     *
     * @generated from protobuf enum value: THIRD_PARTY_APP_ANALYTICS_IOS_IN_APP_PURCHASE = 22;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["THIRD_PARTY_APP_ANALYTICS_IOS_IN_APP_PURCHASE"] = 22] = "THIRD_PARTY_APP_ANALYTICS_IOS_IN_APP_PURCHASE";
    /**
     * iOS app custom conversions tracked through Third Party App Analytics.
     *
     * @generated from protobuf enum value: THIRD_PARTY_APP_ANALYTICS_IOS_CUSTOM = 23;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["THIRD_PARTY_APP_ANALYTICS_IOS_CUSTOM"] = 23] = "THIRD_PARTY_APP_ANALYTICS_IOS_CUSTOM";
    /**
     * Conversions that occur when a user pre-registers a mobile app from the
     * Google Play Store. Read only.
     *
     * @generated from protobuf enum value: ANDROID_APP_PRE_REGISTRATION = 24;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["ANDROID_APP_PRE_REGISTRATION"] = 24] = "ANDROID_APP_PRE_REGISTRATION";
    /**
     * Conversions that track all Google Play downloads which aren't tracked
     * by an app-specific type. Read only.
     *
     * @generated from protobuf enum value: ANDROID_INSTALLS_ALL_OTHER_APPS = 25;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["ANDROID_INSTALLS_ALL_OTHER_APPS"] = 25] = "ANDROID_INSTALLS_ALL_OTHER_APPS";
    /**
     * Floodlight activity that counts the number of times that users have
     * visited a particular webpage after seeing or clicking on one of
     * an advertiser's ads. Read only.
     *
     * @generated from protobuf enum value: FLOODLIGHT_ACTION = 26;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["FLOODLIGHT_ACTION"] = 26] = "FLOODLIGHT_ACTION";
    /**
     * Floodlight activity that tracks the number of sales made or the number
     * of items purchased. Can also capture the total value of each sale.
     * Read only.
     *
     * @generated from protobuf enum value: FLOODLIGHT_TRANSACTION = 27;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["FLOODLIGHT_TRANSACTION"] = 27] = "FLOODLIGHT_TRANSACTION";
    /**
     * Conversions that track local actions from Google's products and
     * services after interacting with an ad. Read only.
     *
     * @generated from protobuf enum value: GOOGLE_HOSTED = 28;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["GOOGLE_HOSTED"] = 28] = "GOOGLE_HOSTED";
    /**
     * Conversions reported when a user submits a lead form. Read only.
     *
     * @generated from protobuf enum value: LEAD_FORM_SUBMIT = 29;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["LEAD_FORM_SUBMIT"] = 29] = "LEAD_FORM_SUBMIT";
    /**
     * Conversions that come from Salesforce. Read only.
     *
     * @generated from protobuf enum value: SALESFORCE = 30;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["SALESFORCE"] = 30] = "SALESFORCE";
    /**
     * Conversions imported from Search Ads 360 Floodlight data. Read only.
     *
     * @generated from protobuf enum value: SEARCH_ADS_360 = 31;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["SEARCH_ADS_360"] = 31] = "SEARCH_ADS_360";
    /**
     * Call conversions that occur on Smart campaign Ads without call tracking
     * setup, using Smart campaign custom criteria. Read only.
     *
     * @generated from protobuf enum value: SMART_CAMPAIGN_AD_CLICKS_TO_CALL = 32;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["SMART_CAMPAIGN_AD_CLICKS_TO_CALL"] = 32] = "SMART_CAMPAIGN_AD_CLICKS_TO_CALL";
    /**
     * The user clicks on a call element within Google Maps. Smart campaign
     * only. Read only.
     *
     * @generated from protobuf enum value: SMART_CAMPAIGN_MAP_CLICKS_TO_CALL = 33;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["SMART_CAMPAIGN_MAP_CLICKS_TO_CALL"] = 33] = "SMART_CAMPAIGN_MAP_CLICKS_TO_CALL";
    /**
     * The user requests directions to a business location within Google Maps.
     * Smart campaign only. Read only.
     *
     * @generated from protobuf enum value: SMART_CAMPAIGN_MAP_DIRECTIONS = 34;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["SMART_CAMPAIGN_MAP_DIRECTIONS"] = 34] = "SMART_CAMPAIGN_MAP_DIRECTIONS";
    /**
     * Call conversions that occur on Smart campaign Ads with call tracking
     * setup, using Smart campaign custom criteria. Read only.
     *
     * @generated from protobuf enum value: SMART_CAMPAIGN_TRACKED_CALLS = 35;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["SMART_CAMPAIGN_TRACKED_CALLS"] = 35] = "SMART_CAMPAIGN_TRACKED_CALLS";
    /**
     * Conversions that occur when a user visits an advertiser's retail store.
     * Read only.
     *
     * @generated from protobuf enum value: STORE_VISITS = 36;
     */
    ConversionActionTypeEnum_ConversionActionType[ConversionActionTypeEnum_ConversionActionType["STORE_VISITS"] = 36] = "STORE_VISITS";
})(ConversionActionTypeEnum_ConversionActionType = exports.ConversionActionTypeEnum_ConversionActionType || (exports.ConversionActionTypeEnum_ConversionActionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionActionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionActionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionActionTypeEnum
 */
exports.ConversionActionTypeEnum = new ConversionActionTypeEnum$Type();
