// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_value_rule_set_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionValueRuleSetStatusEnum = exports.ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a conversion value rule set.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionValueRuleSetStatusEnum.ConversionValueRuleSetStatus
 */
var ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus;
(function (ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus[ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus[ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Conversion Value Rule Set is enabled and can be applied.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus[ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Conversion Value Rule Set is permanently deleted and can't be applied.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus[ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus["REMOVED"] = 3] = "REMOVED";
    /**
     * Conversion Value Rule Set is paused and won't be applied. It can be
     * enabled again.
     *
     * @generated from protobuf enum value: PAUSED = 4;
     */
    ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus[ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus["PAUSED"] = 4] = "PAUSED";
})(ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus = exports.ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus || (exports.ConversionValueRuleSetStatusEnum_ConversionValueRuleSetStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionValueRuleSetStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionValueRuleSetStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionValueRuleSetStatusEnum
 */
exports.ConversionValueRuleSetStatusEnum = new ConversionValueRuleSetStatusEnum$Type();
