// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_lag_bucket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionLagBucketEnum = exports.ConversionLagBucketEnum_ConversionLagBucket = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum representing the number of days between impression and conversion.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionLagBucketEnum.ConversionLagBucket
 */
var ConversionLagBucketEnum_ConversionLagBucket;
(function (ConversionLagBucketEnum_ConversionLagBucket) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Conversion lag bucket from 0 to 1 day. 0 day is included, 1 day is not.
     *
     * @generated from protobuf enum value: LESS_THAN_ONE_DAY = 2;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["LESS_THAN_ONE_DAY"] = 2] = "LESS_THAN_ONE_DAY";
    /**
     * Conversion lag bucket from 1 to 2 days. 1 day is included, 2 days is not.
     *
     * @generated from protobuf enum value: ONE_TO_TWO_DAYS = 3;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["ONE_TO_TWO_DAYS"] = 3] = "ONE_TO_TWO_DAYS";
    /**
     * Conversion lag bucket from 2 to 3 days. 2 days is included,
     * 3 days is not.
     *
     * @generated from protobuf enum value: TWO_TO_THREE_DAYS = 4;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["TWO_TO_THREE_DAYS"] = 4] = "TWO_TO_THREE_DAYS";
    /**
     * Conversion lag bucket from 3 to 4 days. 3 days is included,
     * 4 days is not.
     *
     * @generated from protobuf enum value: THREE_TO_FOUR_DAYS = 5;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["THREE_TO_FOUR_DAYS"] = 5] = "THREE_TO_FOUR_DAYS";
    /**
     * Conversion lag bucket from 4 to 5 days. 4 days is included,
     * 5 days is not.
     *
     * @generated from protobuf enum value: FOUR_TO_FIVE_DAYS = 6;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["FOUR_TO_FIVE_DAYS"] = 6] = "FOUR_TO_FIVE_DAYS";
    /**
     * Conversion lag bucket from 5 to 6 days. 5 days is included,
     * 6 days is not.
     *
     * @generated from protobuf enum value: FIVE_TO_SIX_DAYS = 7;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["FIVE_TO_SIX_DAYS"] = 7] = "FIVE_TO_SIX_DAYS";
    /**
     * Conversion lag bucket from 6 to 7 days. 6 days is included,
     * 7 days is not.
     *
     * @generated from protobuf enum value: SIX_TO_SEVEN_DAYS = 8;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["SIX_TO_SEVEN_DAYS"] = 8] = "SIX_TO_SEVEN_DAYS";
    /**
     * Conversion lag bucket from 7 to 8 days. 7 days is included,
     * 8 days is not.
     *
     * @generated from protobuf enum value: SEVEN_TO_EIGHT_DAYS = 9;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["SEVEN_TO_EIGHT_DAYS"] = 9] = "SEVEN_TO_EIGHT_DAYS";
    /**
     * Conversion lag bucket from 8 to 9 days. 8 days is included,
     * 9 days is not.
     *
     * @generated from protobuf enum value: EIGHT_TO_NINE_DAYS = 10;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["EIGHT_TO_NINE_DAYS"] = 10] = "EIGHT_TO_NINE_DAYS";
    /**
     * Conversion lag bucket from 9 to 10 days. 9 days is included,
     * 10 days is not.
     *
     * @generated from protobuf enum value: NINE_TO_TEN_DAYS = 11;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["NINE_TO_TEN_DAYS"] = 11] = "NINE_TO_TEN_DAYS";
    /**
     * Conversion lag bucket from 10 to 11 days. 10 days is included,
     * 11 days is not.
     *
     * @generated from protobuf enum value: TEN_TO_ELEVEN_DAYS = 12;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["TEN_TO_ELEVEN_DAYS"] = 12] = "TEN_TO_ELEVEN_DAYS";
    /**
     * Conversion lag bucket from 11 to 12 days. 11 days is included,
     * 12 days is not.
     *
     * @generated from protobuf enum value: ELEVEN_TO_TWELVE_DAYS = 13;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["ELEVEN_TO_TWELVE_DAYS"] = 13] = "ELEVEN_TO_TWELVE_DAYS";
    /**
     * Conversion lag bucket from 12 to 13 days. 12 days is included,
     * 13 days is not.
     *
     * @generated from protobuf enum value: TWELVE_TO_THIRTEEN_DAYS = 14;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["TWELVE_TO_THIRTEEN_DAYS"] = 14] = "TWELVE_TO_THIRTEEN_DAYS";
    /**
     * Conversion lag bucket from 13 to 14 days. 13 days is included,
     * 14 days is not.
     *
     * @generated from protobuf enum value: THIRTEEN_TO_FOURTEEN_DAYS = 15;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["THIRTEEN_TO_FOURTEEN_DAYS"] = 15] = "THIRTEEN_TO_FOURTEEN_DAYS";
    /**
     * Conversion lag bucket from 14 to 21 days. 14 days is included,
     * 21 days is not.
     *
     * @generated from protobuf enum value: FOURTEEN_TO_TWENTY_ONE_DAYS = 16;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["FOURTEEN_TO_TWENTY_ONE_DAYS"] = 16] = "FOURTEEN_TO_TWENTY_ONE_DAYS";
    /**
     * Conversion lag bucket from 21 to 30 days. 21 days is included,
     * 30 days is not.
     *
     * @generated from protobuf enum value: TWENTY_ONE_TO_THIRTY_DAYS = 17;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["TWENTY_ONE_TO_THIRTY_DAYS"] = 17] = "TWENTY_ONE_TO_THIRTY_DAYS";
    /**
     * Conversion lag bucket from 30 to 45 days. 30 days is included,
     * 45 days is not.
     *
     * @generated from protobuf enum value: THIRTY_TO_FORTY_FIVE_DAYS = 18;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["THIRTY_TO_FORTY_FIVE_DAYS"] = 18] = "THIRTY_TO_FORTY_FIVE_DAYS";
    /**
     * Conversion lag bucket from 45 to 60 days. 45 days is included,
     * 60 days is not.
     *
     * @generated from protobuf enum value: FORTY_FIVE_TO_SIXTY_DAYS = 19;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["FORTY_FIVE_TO_SIXTY_DAYS"] = 19] = "FORTY_FIVE_TO_SIXTY_DAYS";
    /**
     * Conversion lag bucket from 60 to 90 days. 60 days is included,
     * 90 days is not.
     *
     * @generated from protobuf enum value: SIXTY_TO_NINETY_DAYS = 20;
     */
    ConversionLagBucketEnum_ConversionLagBucket[ConversionLagBucketEnum_ConversionLagBucket["SIXTY_TO_NINETY_DAYS"] = 20] = "SIXTY_TO_NINETY_DAYS";
})(ConversionLagBucketEnum_ConversionLagBucket = exports.ConversionLagBucketEnum_ConversionLagBucket || (exports.ConversionLagBucketEnum_ConversionLagBucket = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionLagBucketEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionLagBucketEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionLagBucketEnum
 */
exports.ConversionLagBucketEnum = new ConversionLagBucketEnum$Type();
