// google-ads-api-client/dist/protos/bindings/ads/enums/conversion_action_category.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionActionCategoryEnum = exports.ConversionActionCategoryEnum_ConversionActionCategory = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The category of conversions that are associated with a ConversionAction.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ConversionActionCategoryEnum.ConversionActionCategory
 */
var ConversionActionCategoryEnum_ConversionActionCategory;
(function (ConversionActionCategoryEnum_ConversionActionCategory) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Default category.
     *
     * @generated from protobuf enum value: DEFAULT = 2;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["DEFAULT"] = 2] = "DEFAULT";
    /**
     * User visiting a page.
     *
     * @generated from protobuf enum value: PAGE_VIEW = 3;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["PAGE_VIEW"] = 3] = "PAGE_VIEW";
    /**
     * Purchase, sales, or "order placed" event.
     *
     * @generated from protobuf enum value: PURCHASE = 4;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["PURCHASE"] = 4] = "PURCHASE";
    /**
     * Signup user action.
     *
     * @generated from protobuf enum value: SIGNUP = 5;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["SIGNUP"] = 5] = "SIGNUP";
    /**
     * Software download action (as for an app).
     *
     * @generated from protobuf enum value: DOWNLOAD = 7;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["DOWNLOAD"] = 7] = "DOWNLOAD";
    /**
     * The addition of items to a shopping cart or bag on an advertiser site.
     *
     * @generated from protobuf enum value: ADD_TO_CART = 8;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["ADD_TO_CART"] = 8] = "ADD_TO_CART";
    /**
     * When someone enters the checkout flow on an advertiser site.
     *
     * @generated from protobuf enum value: BEGIN_CHECKOUT = 9;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["BEGIN_CHECKOUT"] = 9] = "BEGIN_CHECKOUT";
    /**
     * The start of a paid subscription for a product or service.
     *
     * @generated from protobuf enum value: SUBSCRIBE_PAID = 10;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["SUBSCRIBE_PAID"] = 10] = "SUBSCRIBE_PAID";
    /**
     * A call to indicate interest in an advertiser's offering.
     *
     * @generated from protobuf enum value: PHONE_CALL_LEAD = 11;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["PHONE_CALL_LEAD"] = 11] = "PHONE_CALL_LEAD";
    /**
     * A lead conversion imported from an external source into Google Ads.
     *
     * @generated from protobuf enum value: IMPORTED_LEAD = 12;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["IMPORTED_LEAD"] = 12] = "IMPORTED_LEAD";
    /**
     * A submission of a form on an advertiser site indicating business
     * interest.
     *
     * @generated from protobuf enum value: SUBMIT_LEAD_FORM = 13;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["SUBMIT_LEAD_FORM"] = 13] = "SUBMIT_LEAD_FORM";
    /**
     * A booking of an appointment with an advertiser's business.
     *
     * @generated from protobuf enum value: BOOK_APPOINTMENT = 14;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["BOOK_APPOINTMENT"] = 14] = "BOOK_APPOINTMENT";
    /**
     * A quote or price estimate request.
     *
     * @generated from protobuf enum value: REQUEST_QUOTE = 15;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["REQUEST_QUOTE"] = 15] = "REQUEST_QUOTE";
    /**
     * A search for an advertiser's business location with intention to visit.
     *
     * @generated from protobuf enum value: GET_DIRECTIONS = 16;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["GET_DIRECTIONS"] = 16] = "GET_DIRECTIONS";
    /**
     * A click to an advertiser's partner's site.
     *
     * @generated from protobuf enum value: OUTBOUND_CLICK = 17;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["OUTBOUND_CLICK"] = 17] = "OUTBOUND_CLICK";
    /**
     * A call, SMS, email, chat or other type of contact to an advertiser.
     *
     * @generated from protobuf enum value: CONTACT = 18;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["CONTACT"] = 18] = "CONTACT";
    /**
     * A website engagement event such as long site time or a Google Analytics
     * (GA) Smart Goal. Intended to be used for GA, Firebase, GA Gold goal
     * imports.
     *
     * @generated from protobuf enum value: ENGAGEMENT = 19;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["ENGAGEMENT"] = 19] = "ENGAGEMENT";
    /**
     * A visit to a physical store location.
     *
     * @generated from protobuf enum value: STORE_VISIT = 20;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["STORE_VISIT"] = 20] = "STORE_VISIT";
    /**
     * A sale occurring in a physical store.
     *
     * @generated from protobuf enum value: STORE_SALE = 21;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["STORE_SALE"] = 21] = "STORE_SALE";
    /**
     * A lead conversion imported from an external source into Google Ads,
     * that has been further qualified by the advertiser (marketing/sales team).
     * In the lead-to-sale journey, advertisers get leads, then act on them
     * by reaching out to the consumer. If the consumer is interested and
     * may end up buying their product, the advertiser marks such leads as
     * "qualified leads".
     *
     * @generated from protobuf enum value: QUALIFIED_LEAD = 22;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["QUALIFIED_LEAD"] = 22] = "QUALIFIED_LEAD";
    /**
     * A lead conversion imported from an external source into Google Ads, that
     * has further completed a chosen stage as defined by the lead gen
     * advertiser.
     *
     * @generated from protobuf enum value: CONVERTED_LEAD = 23;
     */
    ConversionActionCategoryEnum_ConversionActionCategory[ConversionActionCategoryEnum_ConversionActionCategory["CONVERTED_LEAD"] = 23] = "CONVERTED_LEAD";
})(ConversionActionCategoryEnum_ConversionActionCategory = exports.ConversionActionCategoryEnum_ConversionActionCategory || (exports.ConversionActionCategoryEnum_ConversionActionCategory = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionActionCategoryEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ConversionActionCategoryEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ConversionActionCategoryEnum
 */
exports.ConversionActionCategoryEnum = new ConversionActionCategoryEnum$Type();
