// google-ads-api-client/dist/protos/bindings/ads/enums/criterion_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CriterionTypeEnum = exports.CriterionTypeEnum_CriterionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible criterion types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CriterionTypeEnum.CriterionType
 */
var CriterionTypeEnum_CriterionType;
(function (CriterionTypeEnum_CriterionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Keyword, for example, 'mars cruise'.
     *
     * @generated from protobuf enum value: KEYWORD = 2;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["KEYWORD"] = 2] = "KEYWORD";
    /**
     * Placement, also known as Website, for example, 'www.flowers4sale.com'
     *
     * @generated from protobuf enum value: PLACEMENT = 3;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["PLACEMENT"] = 3] = "PLACEMENT";
    /**
     * Mobile application categories to target.
     *
     * @generated from protobuf enum value: MOBILE_APP_CATEGORY = 4;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["MOBILE_APP_CATEGORY"] = 4] = "MOBILE_APP_CATEGORY";
    /**
     * Mobile applications to target.
     *
     * @generated from protobuf enum value: MOBILE_APPLICATION = 5;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["MOBILE_APPLICATION"] = 5] = "MOBILE_APPLICATION";
    /**
     * Devices to target.
     *
     * @generated from protobuf enum value: DEVICE = 6;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["DEVICE"] = 6] = "DEVICE";
    /**
     * Locations to target.
     *
     * @generated from protobuf enum value: LOCATION = 7;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["LOCATION"] = 7] = "LOCATION";
    /**
     * Listing groups to target.
     *
     * @generated from protobuf enum value: LISTING_GROUP = 8;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["LISTING_GROUP"] = 8] = "LISTING_GROUP";
    /**
     * Ad Schedule.
     *
     * @generated from protobuf enum value: AD_SCHEDULE = 9;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["AD_SCHEDULE"] = 9] = "AD_SCHEDULE";
    /**
     * Age range.
     *
     * @generated from protobuf enum value: AGE_RANGE = 10;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["AGE_RANGE"] = 10] = "AGE_RANGE";
    /**
     * Gender.
     *
     * @generated from protobuf enum value: GENDER = 11;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["GENDER"] = 11] = "GENDER";
    /**
     * Income Range.
     *
     * @generated from protobuf enum value: INCOME_RANGE = 12;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["INCOME_RANGE"] = 12] = "INCOME_RANGE";
    /**
     * Parental status.
     *
     * @generated from protobuf enum value: PARENTAL_STATUS = 13;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["PARENTAL_STATUS"] = 13] = "PARENTAL_STATUS";
    /**
     * YouTube Video.
     *
     * @generated from protobuf enum value: YOUTUBE_VIDEO = 14;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["YOUTUBE_VIDEO"] = 14] = "YOUTUBE_VIDEO";
    /**
     * YouTube Channel.
     *
     * @generated from protobuf enum value: YOUTUBE_CHANNEL = 15;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["YOUTUBE_CHANNEL"] = 15] = "YOUTUBE_CHANNEL";
    /**
     * User list.
     *
     * @generated from protobuf enum value: USER_LIST = 16;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["USER_LIST"] = 16] = "USER_LIST";
    /**
     * Proximity.
     *
     * @generated from protobuf enum value: PROXIMITY = 17;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["PROXIMITY"] = 17] = "PROXIMITY";
    /**
     * A topic target on the display network (for example, "Pets & Animals").
     *
     * @generated from protobuf enum value: TOPIC = 18;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["TOPIC"] = 18] = "TOPIC";
    /**
     * Listing scope to target.
     *
     * @generated from protobuf enum value: LISTING_SCOPE = 19;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["LISTING_SCOPE"] = 19] = "LISTING_SCOPE";
    /**
     * Language.
     *
     * @generated from protobuf enum value: LANGUAGE = 20;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["LANGUAGE"] = 20] = "LANGUAGE";
    /**
     * IpBlock.
     *
     * @generated from protobuf enum value: IP_BLOCK = 21;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["IP_BLOCK"] = 21] = "IP_BLOCK";
    /**
     * Content Label for category exclusion.
     *
     * @generated from protobuf enum value: CONTENT_LABEL = 22;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["CONTENT_LABEL"] = 22] = "CONTENT_LABEL";
    /**
     * Carrier.
     *
     * @generated from protobuf enum value: CARRIER = 23;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["CARRIER"] = 23] = "CARRIER";
    /**
     * A category the user is interested in.
     *
     * @generated from protobuf enum value: USER_INTEREST = 24;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["USER_INTEREST"] = 24] = "USER_INTEREST";
    /**
     * Webpage criterion for dynamic search ads.
     *
     * @generated from protobuf enum value: WEBPAGE = 25;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["WEBPAGE"] = 25] = "WEBPAGE";
    /**
     * Operating system version.
     *
     * @generated from protobuf enum value: OPERATING_SYSTEM_VERSION = 26;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["OPERATING_SYSTEM_VERSION"] = 26] = "OPERATING_SYSTEM_VERSION";
    /**
     * App payment model.
     *
     * @generated from protobuf enum value: APP_PAYMENT_MODEL = 27;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["APP_PAYMENT_MODEL"] = 27] = "APP_PAYMENT_MODEL";
    /**
     * Mobile device.
     *
     * @generated from protobuf enum value: MOBILE_DEVICE = 28;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["MOBILE_DEVICE"] = 28] = "MOBILE_DEVICE";
    /**
     * Custom affinity.
     *
     * @generated from protobuf enum value: CUSTOM_AFFINITY = 29;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["CUSTOM_AFFINITY"] = 29] = "CUSTOM_AFFINITY";
    /**
     * Custom intent.
     *
     * @generated from protobuf enum value: CUSTOM_INTENT = 30;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["CUSTOM_INTENT"] = 30] = "CUSTOM_INTENT";
    /**
     * Location group.
     *
     * @generated from protobuf enum value: LOCATION_GROUP = 31;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["LOCATION_GROUP"] = 31] = "LOCATION_GROUP";
    /**
     * Custom audience
     *
     * @generated from protobuf enum value: CUSTOM_AUDIENCE = 32;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["CUSTOM_AUDIENCE"] = 32] = "CUSTOM_AUDIENCE";
    /**
     * Combined audience
     *
     * @generated from protobuf enum value: COMBINED_AUDIENCE = 33;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["COMBINED_AUDIENCE"] = 33] = "COMBINED_AUDIENCE";
    /**
     * Smart Campaign keyword theme
     *
     * @generated from protobuf enum value: KEYWORD_THEME = 34;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["KEYWORD_THEME"] = 34] = "KEYWORD_THEME";
    /**
     * Audience
     *
     * @generated from protobuf enum value: AUDIENCE = 35;
     */
    CriterionTypeEnum_CriterionType[CriterionTypeEnum_CriterionType["AUDIENCE"] = 35] = "AUDIENCE";
})(CriterionTypeEnum_CriterionType = exports.CriterionTypeEnum_CriterionType || (exports.CriterionTypeEnum_CriterionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CriterionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CriterionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CriterionTypeEnum
 */
exports.CriterionTypeEnum = new CriterionTypeEnum$Type();
