// google-ads-api-client/dist/protos/bindings/ads/enums/criterion_system_serving_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CriterionSystemServingStatusEnum = exports.CriterionSystemServingStatusEnum_CriterionSystemServingStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates criterion system serving statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CriterionSystemServingStatusEnum.CriterionSystemServingStatus
 */
var CriterionSystemServingStatusEnum_CriterionSystemServingStatus;
(function (CriterionSystemServingStatusEnum_CriterionSystemServingStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CriterionSystemServingStatusEnum_CriterionSystemServingStatus[CriterionSystemServingStatusEnum_CriterionSystemServingStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CriterionSystemServingStatusEnum_CriterionSystemServingStatus[CriterionSystemServingStatusEnum_CriterionSystemServingStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Eligible.
     *
     * @generated from protobuf enum value: ELIGIBLE = 2;
     */
    CriterionSystemServingStatusEnum_CriterionSystemServingStatus[CriterionSystemServingStatusEnum_CriterionSystemServingStatus["ELIGIBLE"] = 2] = "ELIGIBLE";
    /**
     * Low search volume.
     *
     * @generated from protobuf enum value: RARELY_SERVED = 3;
     */
    CriterionSystemServingStatusEnum_CriterionSystemServingStatus[CriterionSystemServingStatusEnum_CriterionSystemServingStatus["RARELY_SERVED"] = 3] = "RARELY_SERVED";
})(CriterionSystemServingStatusEnum_CriterionSystemServingStatus = exports.CriterionSystemServingStatusEnum_CriterionSystemServingStatus || (exports.CriterionSystemServingStatusEnum_CriterionSystemServingStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CriterionSystemServingStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CriterionSystemServingStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CriterionSystemServingStatusEnum
 */
exports.CriterionSystemServingStatusEnum = new CriterionSystemServingStatusEnum$Type();
