// google-ads-api-client/dist/protos/bindings/ads/enums/criterion_category_channel_availability_mode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CriterionCategoryChannelAvailabilityModeEnum = exports.CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing the possible CriterionCategoryChannelAvailabilityMode.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CriterionCategoryChannelAvailabilityModeEnum.CriterionCategoryChannelAvailabilityMode
 */
var CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode;
(function (CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode[CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode[CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The category is available to campaigns of all channel types and subtypes.
     *
     * @generated from protobuf enum value: ALL_CHANNELS = 2;
     */
    CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode[CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode["ALL_CHANNELS"] = 2] = "ALL_CHANNELS";
    /**
     * The category is available to campaigns of a specific channel type,
     * including all subtypes under it.
     *
     * @generated from protobuf enum value: CHANNEL_TYPE_AND_ALL_SUBTYPES = 3;
     */
    CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode[CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode["CHANNEL_TYPE_AND_ALL_SUBTYPES"] = 3] = "CHANNEL_TYPE_AND_ALL_SUBTYPES";
    /**
     * The category is available to campaigns of a specific channel type and
     * subtype(s).
     *
     * @generated from protobuf enum value: CHANNEL_TYPE_AND_SUBSET_SUBTYPES = 4;
     */
    CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode[CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode["CHANNEL_TYPE_AND_SUBSET_SUBTYPES"] = 4] = "CHANNEL_TYPE_AND_SUBSET_SUBTYPES";
})(CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode = exports.CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode || (exports.CriterionCategoryChannelAvailabilityModeEnum_CriterionCategoryChannelAvailabilityMode = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CriterionCategoryChannelAvailabilityModeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CriterionCategoryChannelAvailabilityModeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CriterionCategoryChannelAvailabilityModeEnum
 */
exports.CriterionCategoryChannelAvailabilityModeEnum = new CriterionCategoryChannelAvailabilityModeEnum$Type();
