// google-ads-api-client/dist/protos/bindings/ads/enums/criterion_category_locale_availability_mode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CriterionCategoryLocaleAvailabilityModeEnum = exports.CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing the possible CriterionCategoryLocaleAvailabilityMode.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CriterionCategoryLocaleAvailabilityModeEnum.CriterionCategoryLocaleAvailabilityMode
 */
var CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode;
(function (CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode[CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode[CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The category is available to campaigns of all locales.
     *
     * @generated from protobuf enum value: ALL_LOCALES = 2;
     */
    CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode[CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode["ALL_LOCALES"] = 2] = "ALL_LOCALES";
    /**
     * The category is available to campaigns within a list of countries,
     * regardless of language.
     *
     * @generated from protobuf enum value: COUNTRY_AND_ALL_LANGUAGES = 3;
     */
    CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode[CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode["COUNTRY_AND_ALL_LANGUAGES"] = 3] = "COUNTRY_AND_ALL_LANGUAGES";
    /**
     * The category is available to campaigns within a list of languages,
     * regardless of country.
     *
     * @generated from protobuf enum value: LANGUAGE_AND_ALL_COUNTRIES = 4;
     */
    CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode[CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode["LANGUAGE_AND_ALL_COUNTRIES"] = 4] = "LANGUAGE_AND_ALL_COUNTRIES";
    /**
     * The category is available to campaigns within a list of country, language
     * pairs.
     *
     * @generated from protobuf enum value: COUNTRY_AND_LANGUAGE = 5;
     */
    CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode[CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode["COUNTRY_AND_LANGUAGE"] = 5] = "COUNTRY_AND_LANGUAGE";
})(CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode = exports.CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode || (exports.CriterionCategoryLocaleAvailabilityModeEnum_CriterionCategoryLocaleAvailabilityMode = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CriterionCategoryLocaleAvailabilityModeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CriterionCategoryLocaleAvailabilityModeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CriterionCategoryLocaleAvailabilityModeEnum
 */
exports.CriterionCategoryLocaleAvailabilityModeEnum = new CriterionCategoryLocaleAvailabilityModeEnum$Type();
