// google-ads-api-client/dist/protos/bindings/ads/enums/custom_conversion_goal_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomConversionGoalStatusEnum = exports.CustomConversionGoalStatusEnum_CustomConversionGoalStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of a custom conversion goal.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomConversionGoalStatusEnum.CustomConversionGoalStatus
 */
var CustomConversionGoalStatusEnum_CustomConversionGoalStatus;
(function (CustomConversionGoalStatusEnum_CustomConversionGoalStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomConversionGoalStatusEnum_CustomConversionGoalStatus[CustomConversionGoalStatusEnum_CustomConversionGoalStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomConversionGoalStatusEnum_CustomConversionGoalStatus[CustomConversionGoalStatusEnum_CustomConversionGoalStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The custom conversion goal is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CustomConversionGoalStatusEnum_CustomConversionGoalStatus[CustomConversionGoalStatusEnum_CustomConversionGoalStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The custom conversion goal is removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    CustomConversionGoalStatusEnum_CustomConversionGoalStatus[CustomConversionGoalStatusEnum_CustomConversionGoalStatus["REMOVED"] = 3] = "REMOVED";
})(CustomConversionGoalStatusEnum_CustomConversionGoalStatus = exports.CustomConversionGoalStatusEnum_CustomConversionGoalStatus || (exports.CustomConversionGoalStatusEnum_CustomConversionGoalStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomConversionGoalStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomConversionGoalStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomConversionGoalStatusEnum
 */
exports.CustomConversionGoalStatusEnum = new CustomConversionGoalStatusEnum$Type();
