// google-ads-api-client/dist/protos/bindings/ads/enums/custom_audience_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAudienceStatusEnum = exports.CustomAudienceStatusEnum_CustomAudienceStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible custom audience statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomAudienceStatusEnum.CustomAudienceStatus
 */
var CustomAudienceStatusEnum_CustomAudienceStatus;
(function (CustomAudienceStatusEnum_CustomAudienceStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomAudienceStatusEnum_CustomAudienceStatus[CustomAudienceStatusEnum_CustomAudienceStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomAudienceStatusEnum_CustomAudienceStatus[CustomAudienceStatusEnum_CustomAudienceStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Enabled status - custom audience is enabled and can be targeted.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CustomAudienceStatusEnum_CustomAudienceStatus[CustomAudienceStatusEnum_CustomAudienceStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Removed status - custom audience is removed and cannot be used for
     * targeting.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    CustomAudienceStatusEnum_CustomAudienceStatus[CustomAudienceStatusEnum_CustomAudienceStatus["REMOVED"] = 3] = "REMOVED";
})(CustomAudienceStatusEnum_CustomAudienceStatus = exports.CustomAudienceStatusEnum_CustomAudienceStatus || (exports.CustomAudienceStatusEnum_CustomAudienceStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomAudienceStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomAudienceStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomAudienceStatusEnum
 */
exports.CustomAudienceStatusEnum = new CustomAudienceStatusEnum$Type();
