// google-ads-api-client/dist/protos/bindings/ads/enums/customizer_value_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomizerValueStatusEnum = exports.CustomizerValueStatusEnum_CustomizerValueStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of a customizer value.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomizerValueStatusEnum.CustomizerValueStatus
 */
var CustomizerValueStatusEnum_CustomizerValueStatus;
(function (CustomizerValueStatusEnum_CustomizerValueStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomizerValueStatusEnum_CustomizerValueStatus[CustomizerValueStatusEnum_CustomizerValueStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomizerValueStatusEnum_CustomizerValueStatus[CustomizerValueStatusEnum_CustomizerValueStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The customizer value is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CustomizerValueStatusEnum_CustomizerValueStatus[CustomizerValueStatusEnum_CustomizerValueStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The customizer value is removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    CustomizerValueStatusEnum_CustomizerValueStatus[CustomizerValueStatusEnum_CustomizerValueStatus["REMOVED"] = 3] = "REMOVED";
})(CustomizerValueStatusEnum_CustomizerValueStatus = exports.CustomizerValueStatusEnum_CustomizerValueStatus || (exports.CustomizerValueStatusEnum_CustomizerValueStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomizerValueStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomizerValueStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomizerValueStatusEnum
 */
exports.CustomizerValueStatusEnum = new CustomizerValueStatusEnum$Type();
