// google-ads-api-client/dist/protos/bindings/ads/enums/customer_match_upload_key_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerMatchUploadKeyTypeEnum = exports.CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible customer match upload key types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomerMatchUploadKeyTypeEnum.CustomerMatchUploadKeyType
 */
var CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType;
(function (CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType[CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType[CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Members are matched from customer info such as email address, phone
     * number or physical address.
     *
     * @generated from protobuf enum value: CONTACT_INFO = 2;
     */
    CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType[CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType["CONTACT_INFO"] = 2] = "CONTACT_INFO";
    /**
     * Members are matched from a user id generated and assigned by the
     * advertiser.
     *
     * @generated from protobuf enum value: CRM_ID = 3;
     */
    CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType[CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType["CRM_ID"] = 3] = "CRM_ID";
    /**
     * Members are matched from mobile advertising ids.
     *
     * @generated from protobuf enum value: MOBILE_ADVERTISING_ID = 4;
     */
    CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType[CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType["MOBILE_ADVERTISING_ID"] = 4] = "MOBILE_ADVERTISING_ID";
})(CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType = exports.CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType || (exports.CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomerMatchUploadKeyTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomerMatchUploadKeyTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomerMatchUploadKeyTypeEnum
 */
exports.CustomerMatchUploadKeyTypeEnum = new CustomerMatchUploadKeyTypeEnum$Type();
