// google-ads-api-client/dist/protos/bindings/ads/enums/custom_interest_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomInterestTypeEnum = exports.CustomInterestTypeEnum_CustomInterestType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible custom interest types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomInterestTypeEnum.CustomInterestType
 */
var CustomInterestTypeEnum_CustomInterestType;
(function (CustomInterestTypeEnum_CustomInterestType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomInterestTypeEnum_CustomInterestType[CustomInterestTypeEnum_CustomInterestType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomInterestTypeEnum_CustomInterestType[CustomInterestTypeEnum_CustomInterestType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Allows brand advertisers to define custom affinity audience lists.
     *
     * @generated from protobuf enum value: CUSTOM_AFFINITY = 2;
     */
    CustomInterestTypeEnum_CustomInterestType[CustomInterestTypeEnum_CustomInterestType["CUSTOM_AFFINITY"] = 2] = "CUSTOM_AFFINITY";
    /**
     * Allows advertisers to define custom intent audience lists.
     *
     * @generated from protobuf enum value: CUSTOM_INTENT = 3;
     */
    CustomInterestTypeEnum_CustomInterestType[CustomInterestTypeEnum_CustomInterestType["CUSTOM_INTENT"] = 3] = "CUSTOM_INTENT";
})(CustomInterestTypeEnum_CustomInterestType = exports.CustomInterestTypeEnum_CustomInterestType || (exports.CustomInterestTypeEnum_CustomInterestType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomInterestTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomInterestTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomInterestTypeEnum
 */
exports.CustomInterestTypeEnum = new CustomInterestTypeEnum$Type();
