// google-ads-api-client/dist/protos/bindings/ads/enums/customizer_attribute_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomizerAttributeStatusEnum = exports.CustomizerAttributeStatusEnum_CustomizerAttributeStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of a customizer attribute.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomizerAttributeStatusEnum.CustomizerAttributeStatus
 */
var CustomizerAttributeStatusEnum_CustomizerAttributeStatus;
(function (CustomizerAttributeStatusEnum_CustomizerAttributeStatus) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomizerAttributeStatusEnum_CustomizerAttributeStatus[CustomizerAttributeStatusEnum_CustomizerAttributeStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomizerAttributeStatusEnum_CustomizerAttributeStatus[CustomizerAttributeStatusEnum_CustomizerAttributeStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The customizer attribute is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CustomizerAttributeStatusEnum_CustomizerAttributeStatus[CustomizerAttributeStatusEnum_CustomizerAttributeStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The customizer attribute is removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    CustomizerAttributeStatusEnum_CustomizerAttributeStatus[CustomizerAttributeStatusEnum_CustomizerAttributeStatus["REMOVED"] = 3] = "REMOVED";
})(CustomizerAttributeStatusEnum_CustomizerAttributeStatus = exports.CustomizerAttributeStatusEnum_CustomizerAttributeStatus || (exports.CustomizerAttributeStatusEnum_CustomizerAttributeStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomizerAttributeStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomizerAttributeStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomizerAttributeStatusEnum
 */
exports.CustomizerAttributeStatusEnum = new CustomizerAttributeStatusEnum$Type();
