// google-ads-api-client/dist/protos/bindings/ads/enums/custom_audience_member_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAudienceMemberTypeEnum = exports.CustomAudienceMemberTypeEnum_CustomAudienceMemberType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible custom audience member types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomAudienceMemberTypeEnum.CustomAudienceMemberType
 */
var CustomAudienceMemberTypeEnum_CustomAudienceMemberType;
(function (CustomAudienceMemberTypeEnum_CustomAudienceMemberType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomAudienceMemberTypeEnum_CustomAudienceMemberType[CustomAudienceMemberTypeEnum_CustomAudienceMemberType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomAudienceMemberTypeEnum_CustomAudienceMemberType[CustomAudienceMemberTypeEnum_CustomAudienceMemberType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Users whose interests or actions are described by a keyword.
     *
     * @generated from protobuf enum value: KEYWORD = 2;
     */
    CustomAudienceMemberTypeEnum_CustomAudienceMemberType[CustomAudienceMemberTypeEnum_CustomAudienceMemberType["KEYWORD"] = 2] = "KEYWORD";
    /**
     * Users who have interests related to the website's content.
     *
     * @generated from protobuf enum value: URL = 3;
     */
    CustomAudienceMemberTypeEnum_CustomAudienceMemberType[CustomAudienceMemberTypeEnum_CustomAudienceMemberType["URL"] = 3] = "URL";
    /**
     * Users who visit place types described by a place category.
     *
     * @generated from protobuf enum value: PLACE_CATEGORY = 4;
     */
    CustomAudienceMemberTypeEnum_CustomAudienceMemberType[CustomAudienceMemberTypeEnum_CustomAudienceMemberType["PLACE_CATEGORY"] = 4] = "PLACE_CATEGORY";
    /**
     * Users who have installed a mobile app.
     *
     * @generated from protobuf enum value: APP = 5;
     */
    CustomAudienceMemberTypeEnum_CustomAudienceMemberType[CustomAudienceMemberTypeEnum_CustomAudienceMemberType["APP"] = 5] = "APP";
})(CustomAudienceMemberTypeEnum_CustomAudienceMemberType = exports.CustomAudienceMemberTypeEnum_CustomAudienceMemberType || (exports.CustomAudienceMemberTypeEnum_CustomAudienceMemberType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomAudienceMemberTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomAudienceMemberTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomAudienceMemberTypeEnum
 */
exports.CustomAudienceMemberTypeEnum = new CustomAudienceMemberTypeEnum$Type();
