// google-ads-api-client/dist/protos/bindings/ads/enums/customizer_attribute_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomizerAttributeTypeEnum = exports.CustomizerAttributeTypeEnum_CustomizerAttributeType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible types of a customizer attribute.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomizerAttributeTypeEnum.CustomizerAttributeType
 */
var CustomizerAttributeTypeEnum_CustomizerAttributeType;
(function (CustomizerAttributeTypeEnum_CustomizerAttributeType) {
    /**
     * The status has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomizerAttributeTypeEnum_CustomizerAttributeType[CustomizerAttributeTypeEnum_CustomizerAttributeType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomizerAttributeTypeEnum_CustomizerAttributeType[CustomizerAttributeTypeEnum_CustomizerAttributeType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Text customizer.
     *
     * @generated from protobuf enum value: TEXT = 2;
     */
    CustomizerAttributeTypeEnum_CustomizerAttributeType[CustomizerAttributeTypeEnum_CustomizerAttributeType["TEXT"] = 2] = "TEXT";
    /**
     * Number customizer.
     *
     * @generated from protobuf enum value: NUMBER = 3;
     */
    CustomizerAttributeTypeEnum_CustomizerAttributeType[CustomizerAttributeTypeEnum_CustomizerAttributeType["NUMBER"] = 3] = "NUMBER";
    /**
     * Price customizer consisting of a number and a currency.
     *
     * @generated from protobuf enum value: PRICE = 4;
     */
    CustomizerAttributeTypeEnum_CustomizerAttributeType[CustomizerAttributeTypeEnum_CustomizerAttributeType["PRICE"] = 4] = "PRICE";
    /**
     * Percentage customizer consisting of a number and a '%'.
     *
     * @generated from protobuf enum value: PERCENT = 5;
     */
    CustomizerAttributeTypeEnum_CustomizerAttributeType[CustomizerAttributeTypeEnum_CustomizerAttributeType["PERCENT"] = 5] = "PERCENT";
})(CustomizerAttributeTypeEnum_CustomizerAttributeType = exports.CustomizerAttributeTypeEnum_CustomizerAttributeType || (exports.CustomizerAttributeTypeEnum_CustomizerAttributeType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomizerAttributeTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomizerAttributeTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomizerAttributeTypeEnum
 */
exports.CustomizerAttributeTypeEnum = new CustomizerAttributeTypeEnum$Type();
