// google-ads-api-client/dist/protos/bindings/ads/enums/custom_interest_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomInterestStatusEnum = exports.CustomInterestStatusEnum_CustomInterestStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible custom interest types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomInterestStatusEnum.CustomInterestStatus
 */
var CustomInterestStatusEnum_CustomInterestStatus;
(function (CustomInterestStatusEnum_CustomInterestStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomInterestStatusEnum_CustomInterestStatus[CustomInterestStatusEnum_CustomInterestStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomInterestStatusEnum_CustomInterestStatus[CustomInterestStatusEnum_CustomInterestStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Enabled status - custom interest is enabled and can be targeted to.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CustomInterestStatusEnum_CustomInterestStatus[CustomInterestStatusEnum_CustomInterestStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Removed status - custom interest is removed and cannot be used for
     * targeting.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    CustomInterestStatusEnum_CustomInterestStatus[CustomInterestStatusEnum_CustomInterestStatus["REMOVED"] = 3] = "REMOVED";
})(CustomInterestStatusEnum_CustomInterestStatus = exports.CustomInterestStatusEnum_CustomInterestStatus || (exports.CustomInterestStatusEnum_CustomInterestStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomInterestStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomInterestStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomInterestStatusEnum
 */
exports.CustomInterestStatusEnum = new CustomInterestStatusEnum$Type();
