// google-ads-api-client/dist/protos/bindings/ads/enums/custom_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomPlaceholderFieldEnum = exports.CustomPlaceholderFieldEnum_CustomPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Custom placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomPlaceholderFieldEnum.CustomPlaceholderField
 */
var CustomPlaceholderFieldEnum_CustomPlaceholderField;
(function (CustomPlaceholderFieldEnum_CustomPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Required. Combination ID and ID2 must be unique per
     * offer.
     *
     * @generated from protobuf enum value: ID = 2;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["ID"] = 2] = "ID";
    /**
     * Data Type: STRING. Combination ID and ID2 must be unique per offer.
     *
     * @generated from protobuf enum value: ID2 = 3;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["ID2"] = 3] = "ID2";
    /**
     * Data Type: STRING. Required. Main headline with product name to be shown
     * in dynamic ad.
     *
     * @generated from protobuf enum value: ITEM_TITLE = 4;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["ITEM_TITLE"] = 4] = "ITEM_TITLE";
    /**
     * Data Type: STRING. Optional text to be shown in the image ad.
     *
     * @generated from protobuf enum value: ITEM_SUBTITLE = 5;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["ITEM_SUBTITLE"] = 5] = "ITEM_SUBTITLE";
    /**
     * Data Type: STRING. Optional description of the product to be shown in the
     * ad.
     *
     * @generated from protobuf enum value: ITEM_DESCRIPTION = 6;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["ITEM_DESCRIPTION"] = 6] = "ITEM_DESCRIPTION";
    /**
     * Data Type: STRING. Full address of your offer or service, including
     * postal code. This will be used to identify the closest product to the
     * user when there are multiple offers in the feed that are relevant to the
     * user.
     *
     * @generated from protobuf enum value: ITEM_ADDRESS = 7;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["ITEM_ADDRESS"] = 7] = "ITEM_ADDRESS";
    /**
     * Data Type: STRING. Price to be shown in the ad.
     * Example: "100.00 USD"
     *
     * @generated from protobuf enum value: PRICE = 8;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["PRICE"] = 8] = "PRICE";
    /**
     * Data Type: STRING. Formatted price to be shown in the ad.
     * Example: "Starting at $100.00 USD", "$80 - $100"
     *
     * @generated from protobuf enum value: FORMATTED_PRICE = 9;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["FORMATTED_PRICE"] = 9] = "FORMATTED_PRICE";
    /**
     * Data Type: STRING. Sale price to be shown in the ad.
     * Example: "80.00 USD"
     *
     * @generated from protobuf enum value: SALE_PRICE = 10;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["SALE_PRICE"] = 10] = "SALE_PRICE";
    /**
     * Data Type: STRING. Formatted sale price to be shown in the ad.
     * Example: "On sale for $80.00", "$60 - $80"
     *
     * @generated from protobuf enum value: FORMATTED_SALE_PRICE = 11;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["FORMATTED_SALE_PRICE"] = 11] = "FORMATTED_SALE_PRICE";
    /**
     * Data Type: URL. Image to be displayed in the ad. Highly recommended for
     * image ads.
     *
     * @generated from protobuf enum value: IMAGE_URL = 12;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["IMAGE_URL"] = 12] = "IMAGE_URL";
    /**
     * Data Type: STRING. Used as a recommendation engine signal to serve items
     * in the same category.
     *
     * @generated from protobuf enum value: ITEM_CATEGORY = 13;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["ITEM_CATEGORY"] = 13] = "ITEM_CATEGORY";
    /**
     * Data Type: URL_LIST. Final URLs for the ad when using Upgraded
     * URLs. User will be redirected to these URLs when they click on an ad, or
     * when they click on a specific product for ads that have multiple
     * products.
     *
     * @generated from protobuf enum value: FINAL_URLS = 14;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["FINAL_URLS"] = 14] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 15;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["FINAL_MOBILE_URLS"] = 15] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 16;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["TRACKING_URL"] = 16] = "TRACKING_URL";
    /**
     * Data Type: STRING_LIST. Keywords used for product retrieval.
     *
     * @generated from protobuf enum value: CONTEXTUAL_KEYWORDS = 17;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["CONTEXTUAL_KEYWORDS"] = 17] = "CONTEXTUAL_KEYWORDS";
    /**
     * Data Type: STRING. Android app link. Must be formatted as:
     * android-app://{package_id}/{scheme}/{host_path}.
     * The components are defined as follows:
     * package_id: app ID as specified in Google Play.
     * scheme: the scheme to pass to the application. Can be HTTP, or a custom
     *   scheme.
     * host_path: identifies the specific content within your application.
     *
     * @generated from protobuf enum value: ANDROID_APP_LINK = 18;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["ANDROID_APP_LINK"] = 18] = "ANDROID_APP_LINK";
    /**
     * Data Type: STRING_LIST. List of recommended IDs to show together with
     * this item.
     *
     * @generated from protobuf enum value: SIMILAR_IDS = 19;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["SIMILAR_IDS"] = 19] = "SIMILAR_IDS";
    /**
     * Data Type: STRING. iOS app link.
     *
     * @generated from protobuf enum value: IOS_APP_LINK = 20;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["IOS_APP_LINK"] = 20] = "IOS_APP_LINK";
    /**
     * Data Type: INT64. iOS app store ID.
     *
     * @generated from protobuf enum value: IOS_APP_STORE_ID = 21;
     */
    CustomPlaceholderFieldEnum_CustomPlaceholderField[CustomPlaceholderFieldEnum_CustomPlaceholderField["IOS_APP_STORE_ID"] = 21] = "IOS_APP_STORE_ID";
})(CustomPlaceholderFieldEnum_CustomPlaceholderField = exports.CustomPlaceholderFieldEnum_CustomPlaceholderField || (exports.CustomPlaceholderFieldEnum_CustomPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomPlaceholderFieldEnum
 */
exports.CustomPlaceholderFieldEnum = new CustomPlaceholderFieldEnum$Type();
