// google-ads-api-client/dist/protos/bindings/ads/enums/custom_audience_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAudienceTypeEnum = exports.CustomAudienceTypeEnum_CustomAudienceType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible custom audience types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomAudienceTypeEnum.CustomAudienceType
 */
var CustomAudienceTypeEnum_CustomAudienceType;
(function (CustomAudienceTypeEnum_CustomAudienceType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomAudienceTypeEnum_CustomAudienceType[CustomAudienceTypeEnum_CustomAudienceType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomAudienceTypeEnum_CustomAudienceType[CustomAudienceTypeEnum_CustomAudienceType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Google Ads will auto-select the best interpretation at serving
     * time.
     *
     * @generated from protobuf enum value: AUTO = 2;
     */
    CustomAudienceTypeEnum_CustomAudienceType[CustomAudienceTypeEnum_CustomAudienceType["AUTO"] = 2] = "AUTO";
    /**
     * Matches users by their interests.
     *
     * @generated from protobuf enum value: INTEREST = 3;
     */
    CustomAudienceTypeEnum_CustomAudienceType[CustomAudienceTypeEnum_CustomAudienceType["INTEREST"] = 3] = "INTEREST";
    /**
     * Matches users by topics they are researching or products they are
     * considering for purchase.
     *
     * @generated from protobuf enum value: PURCHASE_INTENT = 4;
     */
    CustomAudienceTypeEnum_CustomAudienceType[CustomAudienceTypeEnum_CustomAudienceType["PURCHASE_INTENT"] = 4] = "PURCHASE_INTENT";
    /**
     * Matches users by what they searched on Google Search.
     *
     * @generated from protobuf enum value: SEARCH = 5;
     */
    CustomAudienceTypeEnum_CustomAudienceType[CustomAudienceTypeEnum_CustomAudienceType["SEARCH"] = 5] = "SEARCH";
})(CustomAudienceTypeEnum_CustomAudienceType = exports.CustomAudienceTypeEnum_CustomAudienceType || (exports.CustomAudienceTypeEnum_CustomAudienceType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomAudienceTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomAudienceTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomAudienceTypeEnum
 */
exports.CustomAudienceTypeEnum = new CustomAudienceTypeEnum$Type();
