// google-ads-api-client/dist/protos/bindings/ads/enums/customer_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerStatusEnum = exports.CustomerStatusEnum_CustomerStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a customer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomerStatusEnum.CustomerStatus
 */
var CustomerStatusEnum_CustomerStatus;
(function (CustomerStatusEnum_CustomerStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomerStatusEnum_CustomerStatus[CustomerStatusEnum_CustomerStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomerStatusEnum_CustomerStatus[CustomerStatusEnum_CustomerStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Indicates an active account able to serve ads.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    CustomerStatusEnum_CustomerStatus[CustomerStatusEnum_CustomerStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Indicates a canceled account unable to serve ads.
     * Can be reactivated by an admin user.
     *
     * @generated from protobuf enum value: CANCELED = 3;
     */
    CustomerStatusEnum_CustomerStatus[CustomerStatusEnum_CustomerStatus["CANCELED"] = 3] = "CANCELED";
    /**
     * Indicates a suspended account unable to serve ads.
     * May only be activated by Google support.
     *
     * @generated from protobuf enum value: SUSPENDED = 4;
     */
    CustomerStatusEnum_CustomerStatus[CustomerStatusEnum_CustomerStatus["SUSPENDED"] = 4] = "SUSPENDED";
    /**
     * Indicates a closed account unable to serve ads.
     * Test account will also have CLOSED status.
     * Status is permanent and may not be reopened.
     *
     * @generated from protobuf enum value: CLOSED = 5;
     */
    CustomerStatusEnum_CustomerStatus[CustomerStatusEnum_CustomerStatus["CLOSED"] = 5] = "CLOSED";
})(CustomerStatusEnum_CustomerStatus = exports.CustomerStatusEnum_CustomerStatus || (exports.CustomerStatusEnum_CustomerStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomerStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomerStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomerStatusEnum
 */
exports.CustomerStatusEnum = new CustomerStatusEnum$Type();
