// google-ads-api-client/dist/protos/bindings/ads/enums/custom_interest_member_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomInterestMemberTypeEnum = exports.CustomInterestMemberTypeEnum_CustomInterestMemberType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible custom interest member types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.CustomInterestMemberTypeEnum.CustomInterestMemberType
 */
var CustomInterestMemberTypeEnum_CustomInterestMemberType;
(function (CustomInterestMemberTypeEnum_CustomInterestMemberType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomInterestMemberTypeEnum_CustomInterestMemberType[CustomInterestMemberTypeEnum_CustomInterestMemberType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomInterestMemberTypeEnum_CustomInterestMemberType[CustomInterestMemberTypeEnum_CustomInterestMemberType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Custom interest member type KEYWORD.
     *
     * @generated from protobuf enum value: KEYWORD = 2;
     */
    CustomInterestMemberTypeEnum_CustomInterestMemberType[CustomInterestMemberTypeEnum_CustomInterestMemberType["KEYWORD"] = 2] = "KEYWORD";
    /**
     * Custom interest member type URL.
     *
     * @generated from protobuf enum value: URL = 3;
     */
    CustomInterestMemberTypeEnum_CustomInterestMemberType[CustomInterestMemberTypeEnum_CustomInterestMemberType["URL"] = 3] = "URL";
})(CustomInterestMemberTypeEnum_CustomInterestMemberType = exports.CustomInterestMemberTypeEnum_CustomInterestMemberType || (exports.CustomInterestMemberTypeEnum_CustomInterestMemberType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomInterestMemberTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.CustomInterestMemberTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.CustomInterestMemberTypeEnum
 */
exports.CustomInterestMemberTypeEnum = new CustomInterestMemberTypeEnum$Type();
