// google-ads-api-client/dist/protos/bindings/ads/enums/data_driven_model_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataDrivenModelStatusEnum = exports.DataDrivenModelStatusEnum_DataDrivenModelStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates data driven model statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.DataDrivenModelStatusEnum.DataDrivenModelStatus
 */
var DataDrivenModelStatusEnum_DataDrivenModelStatus;
(function (DataDrivenModelStatusEnum_DataDrivenModelStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DataDrivenModelStatusEnum_DataDrivenModelStatus[DataDrivenModelStatusEnum_DataDrivenModelStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DataDrivenModelStatusEnum_DataDrivenModelStatus[DataDrivenModelStatusEnum_DataDrivenModelStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The data driven model is available.
     *
     * @generated from protobuf enum value: AVAILABLE = 2;
     */
    DataDrivenModelStatusEnum_DataDrivenModelStatus[DataDrivenModelStatusEnum_DataDrivenModelStatus["AVAILABLE"] = 2] = "AVAILABLE";
    /**
     * The data driven model is stale. It hasn't been updated for at least 7
     * days. It is still being used, but will become expired if it does not get
     * updated for 30 days.
     *
     * @generated from protobuf enum value: STALE = 3;
     */
    DataDrivenModelStatusEnum_DataDrivenModelStatus[DataDrivenModelStatusEnum_DataDrivenModelStatus["STALE"] = 3] = "STALE";
    /**
     * The data driven model expired. It hasn't been updated for at least 30
     * days and cannot be used. Most commonly this is because there hasn't been
     * the required number of events in a recent 30-day period.
     *
     * @generated from protobuf enum value: EXPIRED = 4;
     */
    DataDrivenModelStatusEnum_DataDrivenModelStatus[DataDrivenModelStatusEnum_DataDrivenModelStatus["EXPIRED"] = 4] = "EXPIRED";
    /**
     * The data driven model has never been generated. Most commonly this is
     * because there has never been the required number of events in any 30-day
     * period.
     *
     * @generated from protobuf enum value: NEVER_GENERATED = 5;
     */
    DataDrivenModelStatusEnum_DataDrivenModelStatus[DataDrivenModelStatusEnum_DataDrivenModelStatus["NEVER_GENERATED"] = 5] = "NEVER_GENERATED";
})(DataDrivenModelStatusEnum_DataDrivenModelStatus = exports.DataDrivenModelStatusEnum_DataDrivenModelStatus || (exports.DataDrivenModelStatusEnum_DataDrivenModelStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DataDrivenModelStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.DataDrivenModelStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.DataDrivenModelStatusEnum
 */
exports.DataDrivenModelStatusEnum = new DataDrivenModelStatusEnum$Type();
