// google-ads-api-client/dist/protos/bindings/ads/enums/day_of_week.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DayOfWeekEnum = exports.DayOfWeekEnum_DayOfWeek = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates days of the week, for example, "Monday".
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.DayOfWeekEnum.DayOfWeek
 */
var DayOfWeekEnum_DayOfWeek;
(function (DayOfWeekEnum_DayOfWeek) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Monday.
     *
     * @generated from protobuf enum value: MONDAY = 2;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["MONDAY"] = 2] = "MONDAY";
    /**
     * Tuesday.
     *
     * @generated from protobuf enum value: TUESDAY = 3;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["TUESDAY"] = 3] = "TUESDAY";
    /**
     * Wednesday.
     *
     * @generated from protobuf enum value: WEDNESDAY = 4;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["WEDNESDAY"] = 4] = "WEDNESDAY";
    /**
     * Thursday.
     *
     * @generated from protobuf enum value: THURSDAY = 5;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["THURSDAY"] = 5] = "THURSDAY";
    /**
     * Friday.
     *
     * @generated from protobuf enum value: FRIDAY = 6;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["FRIDAY"] = 6] = "FRIDAY";
    /**
     * Saturday.
     *
     * @generated from protobuf enum value: SATURDAY = 7;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["SATURDAY"] = 7] = "SATURDAY";
    /**
     * Sunday.
     *
     * @generated from protobuf enum value: SUNDAY = 8;
     */
    DayOfWeekEnum_DayOfWeek[DayOfWeekEnum_DayOfWeek["SUNDAY"] = 8] = "SUNDAY";
})(DayOfWeekEnum_DayOfWeek = exports.DayOfWeekEnum_DayOfWeek || (exports.DayOfWeekEnum_DayOfWeek = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DayOfWeekEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.DayOfWeekEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.DayOfWeekEnum
 */
exports.DayOfWeekEnum = new DayOfWeekEnum$Type();
