// google-ads-api-client/dist/protos/bindings/ads/enums/device.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceEnum = exports.DeviceEnum_Device = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates Google Ads devices available for targeting.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.DeviceEnum.Device
 */
var DeviceEnum_Device;
(function (DeviceEnum_Device) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DeviceEnum_Device[DeviceEnum_Device["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DeviceEnum_Device[DeviceEnum_Device["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Mobile devices with full browsers.
     *
     * @generated from protobuf enum value: MOBILE = 2;
     */
    DeviceEnum_Device[DeviceEnum_Device["MOBILE"] = 2] = "MOBILE";
    /**
     * Tablets with full browsers.
     *
     * @generated from protobuf enum value: TABLET = 3;
     */
    DeviceEnum_Device[DeviceEnum_Device["TABLET"] = 3] = "TABLET";
    /**
     * Computers.
     *
     * @generated from protobuf enum value: DESKTOP = 4;
     */
    DeviceEnum_Device[DeviceEnum_Device["DESKTOP"] = 4] = "DESKTOP";
    /**
     * Smart TVs and game consoles.
     *
     * @generated from protobuf enum value: CONNECTED_TV = 6;
     */
    DeviceEnum_Device[DeviceEnum_Device["CONNECTED_TV"] = 6] = "CONNECTED_TV";
    /**
     * Other device types.
     *
     * @generated from protobuf enum value: OTHER = 5;
     */
    DeviceEnum_Device[DeviceEnum_Device["OTHER"] = 5] = "OTHER";
})(DeviceEnum_Device = exports.DeviceEnum_Device || (exports.DeviceEnum_Device = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DeviceEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.DeviceEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.DeviceEnum
 */
exports.DeviceEnum = new DeviceEnum$Type();
