// google-ads-api-client/dist/protos/bindings/ads/enums/display_ad_format_setting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisplayAdFormatSettingEnum = exports.DisplayAdFormatSettingEnum_DisplayAdFormatSetting = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates display ad format settings.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum.DisplayAdFormatSetting
 */
var DisplayAdFormatSettingEnum_DisplayAdFormatSetting;
(function (DisplayAdFormatSettingEnum_DisplayAdFormatSetting) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DisplayAdFormatSettingEnum_DisplayAdFormatSetting[DisplayAdFormatSettingEnum_DisplayAdFormatSetting["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DisplayAdFormatSettingEnum_DisplayAdFormatSetting[DisplayAdFormatSettingEnum_DisplayAdFormatSetting["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Text, image and native formats.
     *
     * @generated from protobuf enum value: ALL_FORMATS = 2;
     */
    DisplayAdFormatSettingEnum_DisplayAdFormatSetting[DisplayAdFormatSettingEnum_DisplayAdFormatSetting["ALL_FORMATS"] = 2] = "ALL_FORMATS";
    /**
     * Text and image formats.
     *
     * @generated from protobuf enum value: NON_NATIVE = 3;
     */
    DisplayAdFormatSettingEnum_DisplayAdFormatSetting[DisplayAdFormatSettingEnum_DisplayAdFormatSetting["NON_NATIVE"] = 3] = "NON_NATIVE";
    /**
     * Native format, for example, the format rendering is controlled by the
     * publisher and not by Google.
     *
     * @generated from protobuf enum value: NATIVE = 4;
     */
    DisplayAdFormatSettingEnum_DisplayAdFormatSetting[DisplayAdFormatSettingEnum_DisplayAdFormatSetting["NATIVE"] = 4] = "NATIVE";
})(DisplayAdFormatSettingEnum_DisplayAdFormatSetting = exports.DisplayAdFormatSettingEnum_DisplayAdFormatSetting || (exports.DisplayAdFormatSettingEnum_DisplayAdFormatSetting = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DisplayAdFormatSettingEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum
 */
exports.DisplayAdFormatSettingEnum = new DisplayAdFormatSettingEnum$Type();
