// google-ads-api-client/dist/protos/bindings/ads/enums/display_upload_product_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisplayUploadProductTypeEnum = exports.DisplayUploadProductTypeEnum_DisplayUploadProductType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates display upload product types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.DisplayUploadProductTypeEnum.DisplayUploadProductType
 */
var DisplayUploadProductTypeEnum_DisplayUploadProductType;
(function (DisplayUploadProductTypeEnum_DisplayUploadProductType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * HTML5 upload ad. This product type requires the upload_media_bundle
     * field in DisplayUploadAdInfo to be set.
     *
     * @generated from protobuf enum value: HTML5_UPLOAD_AD = 2;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["HTML5_UPLOAD_AD"] = 2] = "HTML5_UPLOAD_AD";
    /**
     * Dynamic HTML5 education ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in an education campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_EDUCATION_AD = 3;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_EDUCATION_AD"] = 3] = "DYNAMIC_HTML5_EDUCATION_AD";
    /**
     * Dynamic HTML5 flight ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in a flight campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_FLIGHT_AD = 4;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_FLIGHT_AD"] = 4] = "DYNAMIC_HTML5_FLIGHT_AD";
    /**
     * Dynamic HTML5 hotel and rental ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in a hotel campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_HOTEL_RENTAL_AD = 5;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_HOTEL_RENTAL_AD"] = 5] = "DYNAMIC_HTML5_HOTEL_RENTAL_AD";
    /**
     * Dynamic HTML5 job ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in a job campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_JOB_AD = 6;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_JOB_AD"] = 6] = "DYNAMIC_HTML5_JOB_AD";
    /**
     * Dynamic HTML5 local ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in a local campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_LOCAL_AD = 7;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_LOCAL_AD"] = 7] = "DYNAMIC_HTML5_LOCAL_AD";
    /**
     * Dynamic HTML5 real estate ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in a real estate campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_REAL_ESTATE_AD = 8;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_REAL_ESTATE_AD"] = 8] = "DYNAMIC_HTML5_REAL_ESTATE_AD";
    /**
     * Dynamic HTML5 custom ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in a custom campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_CUSTOM_AD = 9;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_CUSTOM_AD"] = 9] = "DYNAMIC_HTML5_CUSTOM_AD";
    /**
     * Dynamic HTML5 travel ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in a travel campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_TRAVEL_AD = 10;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_TRAVEL_AD"] = 10] = "DYNAMIC_HTML5_TRAVEL_AD";
    /**
     * Dynamic HTML5 hotel ad. This product type requires the
     * upload_media_bundle field in DisplayUploadAdInfo to be set. Can only be
     * used in a hotel campaign.
     *
     * @generated from protobuf enum value: DYNAMIC_HTML5_HOTEL_AD = 11;
     */
    DisplayUploadProductTypeEnum_DisplayUploadProductType[DisplayUploadProductTypeEnum_DisplayUploadProductType["DYNAMIC_HTML5_HOTEL_AD"] = 11] = "DYNAMIC_HTML5_HOTEL_AD";
})(DisplayUploadProductTypeEnum_DisplayUploadProductType = exports.DisplayUploadProductTypeEnum_DisplayUploadProductType || (exports.DisplayUploadProductTypeEnum_DisplayUploadProductType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DisplayUploadProductTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.DisplayUploadProductTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.DisplayUploadProductTypeEnum
 */
exports.DisplayUploadProductTypeEnum = new DisplayUploadProductTypeEnum$Type();
