// google-ads-api-client/dist/protos/bindings/ads/enums/distance_bucket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistanceBucketEnum = exports.DistanceBucketEnum_DistanceBucket = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The distance bucket for a user's distance from an advertiser's location
 * extension.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.DistanceBucketEnum.DistanceBucket
 */
var DistanceBucketEnum_DistanceBucket;
(function (DistanceBucketEnum_DistanceBucket) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * User was within 700m of the location.
     *
     * @generated from protobuf enum value: WITHIN_700M = 2;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_700M"] = 2] = "WITHIN_700M";
    /**
     * User was within 1KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_1KM = 3;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_1KM"] = 3] = "WITHIN_1KM";
    /**
     * User was within 5KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_5KM = 4;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_5KM"] = 4] = "WITHIN_5KM";
    /**
     * User was within 10KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_10KM = 5;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_10KM"] = 5] = "WITHIN_10KM";
    /**
     * User was within 15KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_15KM = 6;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_15KM"] = 6] = "WITHIN_15KM";
    /**
     * User was within 20KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_20KM = 7;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_20KM"] = 7] = "WITHIN_20KM";
    /**
     * User was within 25KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_25KM = 8;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_25KM"] = 8] = "WITHIN_25KM";
    /**
     * User was within 30KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_30KM = 9;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_30KM"] = 9] = "WITHIN_30KM";
    /**
     * User was within 35KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_35KM = 10;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_35KM"] = 10] = "WITHIN_35KM";
    /**
     * User was within 40KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_40KM = 11;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_40KM"] = 11] = "WITHIN_40KM";
    /**
     * User was within 45KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_45KM = 12;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_45KM"] = 12] = "WITHIN_45KM";
    /**
     * User was within 50KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_50KM = 13;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_50KM"] = 13] = "WITHIN_50KM";
    /**
     * User was within 55KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_55KM = 14;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_55KM"] = 14] = "WITHIN_55KM";
    /**
     * User was within 60KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_60KM = 15;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_60KM"] = 15] = "WITHIN_60KM";
    /**
     * User was within 65KM of the location.
     *
     * @generated from protobuf enum value: WITHIN_65KM = 16;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_65KM"] = 16] = "WITHIN_65KM";
    /**
     * User was beyond 65KM of the location.
     *
     * @generated from protobuf enum value: BEYOND_65KM = 17;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["BEYOND_65KM"] = 17] = "BEYOND_65KM";
    /**
     * User was within 0.7 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_0_7MILES = 18;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_0_7MILES"] = 18] = "WITHIN_0_7MILES";
    /**
     * User was within 1 mile of the location.
     *
     * @generated from protobuf enum value: WITHIN_1MILE = 19;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_1MILE"] = 19] = "WITHIN_1MILE";
    /**
     * User was within 5 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_5MILES = 20;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_5MILES"] = 20] = "WITHIN_5MILES";
    /**
     * User was within 10 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_10MILES = 21;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_10MILES"] = 21] = "WITHIN_10MILES";
    /**
     * User was within 15 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_15MILES = 22;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_15MILES"] = 22] = "WITHIN_15MILES";
    /**
     * User was within 20 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_20MILES = 23;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_20MILES"] = 23] = "WITHIN_20MILES";
    /**
     * User was within 25 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_25MILES = 24;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_25MILES"] = 24] = "WITHIN_25MILES";
    /**
     * User was within 30 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_30MILES = 25;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_30MILES"] = 25] = "WITHIN_30MILES";
    /**
     * User was within 35 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_35MILES = 26;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_35MILES"] = 26] = "WITHIN_35MILES";
    /**
     * User was within 40 miles of the location.
     *
     * @generated from protobuf enum value: WITHIN_40MILES = 27;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["WITHIN_40MILES"] = 27] = "WITHIN_40MILES";
    /**
     * User was beyond 40 miles of the location.
     *
     * @generated from protobuf enum value: BEYOND_40MILES = 28;
     */
    DistanceBucketEnum_DistanceBucket[DistanceBucketEnum_DistanceBucket["BEYOND_40MILES"] = 28] = "BEYOND_40MILES";
})(DistanceBucketEnum_DistanceBucket = exports.DistanceBucketEnum_DistanceBucket || (exports.DistanceBucketEnum_DistanceBucket = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DistanceBucketEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.DistanceBucketEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.DistanceBucketEnum
 */
exports.DistanceBucketEnum = new DistanceBucketEnum$Type();
