// google-ads-api-client/dist/protos/bindings/ads/enums/dsa_page_feed_criterion_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DsaPageFeedCriterionFieldEnum = exports.DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Dynamic Search Ad Page Feed criterion fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.DsaPageFeedCriterionFieldEnum.DsaPageFeedCriterionField
 */
var DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField;
(function (DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField[DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField[DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: URL or URL_LIST. URL of the web page you want to target.
     *
     * @generated from protobuf enum value: PAGE_URL = 2;
     */
    DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField[DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField["PAGE_URL"] = 2] = "PAGE_URL";
    /**
     * Data Type: STRING_LIST. The labels that will help you target ads within
     * your page feed.
     *
     * @generated from protobuf enum value: LABEL = 3;
     */
    DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField[DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField["LABEL"] = 3] = "LABEL";
})(DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField = exports.DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField || (exports.DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DsaPageFeedCriterionFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.DsaPageFeedCriterionFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.DsaPageFeedCriterionFieldEnum
 */
exports.DsaPageFeedCriterionFieldEnum = new DsaPageFeedCriterionFieldEnum$Type();
