// google-ads-api-client/dist/protos/bindings/ads/enums/education_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EducationPlaceholderFieldEnum = exports.EducationPlaceholderFieldEnum_EducationPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Education placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.EducationPlaceholderFieldEnum.EducationPlaceholderField
 */
var EducationPlaceholderFieldEnum_EducationPlaceholderField;
(function (EducationPlaceholderFieldEnum_EducationPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Required. Combination of PROGRAM ID and LOCATION ID
     * must be unique per offer.
     *
     * @generated from protobuf enum value: PROGRAM_ID = 2;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["PROGRAM_ID"] = 2] = "PROGRAM_ID";
    /**
     * Data Type: STRING. Combination of PROGRAM ID and LOCATION ID must be
     * unique per offer.
     *
     * @generated from protobuf enum value: LOCATION_ID = 3;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["LOCATION_ID"] = 3] = "LOCATION_ID";
    /**
     * Data Type: STRING. Required. Main headline with program name to be shown
     * in dynamic ad.
     *
     * @generated from protobuf enum value: PROGRAM_NAME = 4;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["PROGRAM_NAME"] = 4] = "PROGRAM_NAME";
    /**
     * Data Type: STRING. Area of study that can be shown in dynamic ad.
     *
     * @generated from protobuf enum value: AREA_OF_STUDY = 5;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["AREA_OF_STUDY"] = 5] = "AREA_OF_STUDY";
    /**
     * Data Type: STRING. Description of program that can be shown in dynamic
     * ad.
     *
     * @generated from protobuf enum value: PROGRAM_DESCRIPTION = 6;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["PROGRAM_DESCRIPTION"] = 6] = "PROGRAM_DESCRIPTION";
    /**
     * Data Type: STRING. Name of school that can be shown in dynamic ad.
     *
     * @generated from protobuf enum value: SCHOOL_NAME = 7;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["SCHOOL_NAME"] = 7] = "SCHOOL_NAME";
    /**
     * Data Type: STRING. Complete school address, including postal code.
     *
     * @generated from protobuf enum value: ADDRESS = 8;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["ADDRESS"] = 8] = "ADDRESS";
    /**
     * Data Type: URL. Image to be displayed in ads.
     *
     * @generated from protobuf enum value: THUMBNAIL_IMAGE_URL = 9;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["THUMBNAIL_IMAGE_URL"] = 9] = "THUMBNAIL_IMAGE_URL";
    /**
     * Data Type: URL. Alternative hosted file of image to be used in the ad.
     *
     * @generated from protobuf enum value: ALTERNATIVE_THUMBNAIL_IMAGE_URL = 10;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["ALTERNATIVE_THUMBNAIL_IMAGE_URL"] = 10] = "ALTERNATIVE_THUMBNAIL_IMAGE_URL";
    /**
     * Data Type: URL_LIST. Required. Final URLs to be used in ad when using
     * Upgraded URLs; the more specific the better (for example, the individual
     * URL of a specific program and its location).
     *
     * @generated from protobuf enum value: FINAL_URLS = 11;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["FINAL_URLS"] = 11] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 12;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["FINAL_MOBILE_URLS"] = 12] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 13;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["TRACKING_URL"] = 13] = "TRACKING_URL";
    /**
     * Data Type: STRING_LIST. Keywords used for product retrieval.
     *
     * @generated from protobuf enum value: CONTEXTUAL_KEYWORDS = 14;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["CONTEXTUAL_KEYWORDS"] = 14] = "CONTEXTUAL_KEYWORDS";
    /**
     * Data Type: STRING. Android app link. Must be formatted as:
     * android-app://{package_id}/{scheme}/{host_path}.
     * The components are defined as follows:
     * package_id: app ID as specified in Google Play.
     * scheme: the scheme to pass to the application. Can be HTTP, or a custom
     *   scheme.
     * host_path: identifies the specific content within your application.
     *
     * @generated from protobuf enum value: ANDROID_APP_LINK = 15;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["ANDROID_APP_LINK"] = 15] = "ANDROID_APP_LINK";
    /**
     * Data Type: STRING_LIST. List of recommended program IDs to show together
     * with this item.
     *
     * @generated from protobuf enum value: SIMILAR_PROGRAM_IDS = 16;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["SIMILAR_PROGRAM_IDS"] = 16] = "SIMILAR_PROGRAM_IDS";
    /**
     * Data Type: STRING. iOS app link.
     *
     * @generated from protobuf enum value: IOS_APP_LINK = 17;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["IOS_APP_LINK"] = 17] = "IOS_APP_LINK";
    /**
     * Data Type: INT64. iOS app store ID.
     *
     * @generated from protobuf enum value: IOS_APP_STORE_ID = 18;
     */
    EducationPlaceholderFieldEnum_EducationPlaceholderField[EducationPlaceholderFieldEnum_EducationPlaceholderField["IOS_APP_STORE_ID"] = 18] = "IOS_APP_STORE_ID";
})(EducationPlaceholderFieldEnum_EducationPlaceholderField = exports.EducationPlaceholderFieldEnum_EducationPlaceholderField || (exports.EducationPlaceholderFieldEnum_EducationPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class EducationPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.EducationPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.EducationPlaceholderFieldEnum
 */
exports.EducationPlaceholderFieldEnum = new EducationPlaceholderFieldEnum$Type();
