// google-ads-api-client/dist/protos/bindings/ads/enums/experiment_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExperimentTypeEnum = exports.ExperimentTypeEnum_ExperimentType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of the experiment.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ExperimentTypeEnum.ExperimentType
 */
var ExperimentTypeEnum_ExperimentType;
(function (ExperimentTypeEnum_ExperimentType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * This is a DISPLAY_AND_VIDEO_360 experiment.
     *
     * @generated from protobuf enum value: DISPLAY_AND_VIDEO_360 = 2;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["DISPLAY_AND_VIDEO_360"] = 2] = "DISPLAY_AND_VIDEO_360";
    /**
     * This is an ad variation experiment.
     *
     * @generated from protobuf enum value: AD_VARIATION = 3;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["AD_VARIATION"] = 3] = "AD_VARIATION";
    /**
     * A custom experiment consisting of Video campaigns.
     *
     * @generated from protobuf enum value: YOUTUBE_CUSTOM = 5;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["YOUTUBE_CUSTOM"] = 5] = "YOUTUBE_CUSTOM";
    /**
     * A custom experiment consisting of display campaigns.
     *
     * @generated from protobuf enum value: DISPLAY_CUSTOM = 6;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["DISPLAY_CUSTOM"] = 6] = "DISPLAY_CUSTOM";
    /**
     * A custom experiment consisting of search campaigns.
     *
     * @generated from protobuf enum value: SEARCH_CUSTOM = 7;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["SEARCH_CUSTOM"] = 7] = "SEARCH_CUSTOM";
    /**
     * An experiment that compares bidding strategies for display campaigns.
     *
     * @generated from protobuf enum value: DISPLAY_AUTOMATED_BIDDING_STRATEGY = 8;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["DISPLAY_AUTOMATED_BIDDING_STRATEGY"] = 8] = "DISPLAY_AUTOMATED_BIDDING_STRATEGY";
    /**
     * An experiment that compares bidding strategies for search campaigns."
     *
     * @generated from protobuf enum value: SEARCH_AUTOMATED_BIDDING_STRATEGY = 9;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["SEARCH_AUTOMATED_BIDDING_STRATEGY"] = 9] = "SEARCH_AUTOMATED_BIDDING_STRATEGY";
    /**
     * An experiment that compares bidding strategies for shopping campaigns.
     *
     * @generated from protobuf enum value: SHOPPING_AUTOMATED_BIDDING_STRATEGY = 10;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["SHOPPING_AUTOMATED_BIDDING_STRATEGY"] = 10] = "SHOPPING_AUTOMATED_BIDDING_STRATEGY";
    /**
     * DEPRECATED. A smart matching experiment with search campaigns.
     *
     * @generated from protobuf enum value: SMART_MATCHING = 11;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["SMART_MATCHING"] = 11] = "SMART_MATCHING";
    /**
     * A custom experiment consisting of hotel campaigns.
     *
     * @generated from protobuf enum value: HOTEL_CUSTOM = 12;
     */
    ExperimentTypeEnum_ExperimentType[ExperimentTypeEnum_ExperimentType["HOTEL_CUSTOM"] = 12] = "HOTEL_CUSTOM";
})(ExperimentTypeEnum_ExperimentType = exports.ExperimentTypeEnum_ExperimentType || (exports.ExperimentTypeEnum_ExperimentType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ExperimentTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ExperimentTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ExperimentTypeEnum
 */
exports.ExperimentTypeEnum = new ExperimentTypeEnum$Type();
