// google-ads-api-client/dist/protos/bindings/ads/enums/external_conversion_source.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalConversionSourceEnum = exports.ExternalConversionSourceEnum_ExternalConversionSource = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The external conversion source that is associated with a ConversionAction.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ExternalConversionSourceEnum.ExternalConversionSource
 */
var ExternalConversionSourceEnum_ExternalConversionSource;
(function (ExternalConversionSourceEnum_ExternalConversionSource) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Conversion that occurs when a user navigates to a particular webpage
     * after viewing an ad; Displayed in Google Ads UI as 'Website'.
     *
     * @generated from protobuf enum value: WEBPAGE = 2;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["WEBPAGE"] = 2] = "WEBPAGE";
    /**
     * Conversion that comes from linked Google Analytics goal or transaction;
     * Displayed in Google Ads UI as 'Analytics'.
     *
     * @generated from protobuf enum value: ANALYTICS = 3;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["ANALYTICS"] = 3] = "ANALYTICS";
    /**
     * Website conversion that is uploaded through ConversionUploadService;
     * Displayed in Google Ads UI as 'Import from clicks'.
     *
     * @generated from protobuf enum value: UPLOAD = 4;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["UPLOAD"] = 4] = "UPLOAD";
    /**
     * Conversion that occurs when a user clicks on a call extension directly on
     * an ad; Displayed in Google Ads UI as 'Calls from ads'.
     *
     * @generated from protobuf enum value: AD_CALL_METRICS = 5;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["AD_CALL_METRICS"] = 5] = "AD_CALL_METRICS";
    /**
     * Conversion that occurs when a user calls a dynamically-generated phone
     * number (by installed javascript) from an advertiser's website after
     * clicking on an ad; Displayed in Google Ads UI as 'Calls from website'.
     *
     * @generated from protobuf enum value: WEBSITE_CALL_METRICS = 6;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["WEBSITE_CALL_METRICS"] = 6] = "WEBSITE_CALL_METRICS";
    /**
     * Conversion that occurs when a user visits an advertiser's retail store
     * after clicking on a Google ad;
     * Displayed in Google Ads UI as 'Store visits'.
     *
     * @generated from protobuf enum value: STORE_VISITS = 7;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["STORE_VISITS"] = 7] = "STORE_VISITS";
    /**
     * Conversion that occurs when a user takes an in-app action such as a
     * purchase in an Android app;
     * Displayed in Google Ads UI as 'Android in-app action'.
     *
     * @generated from protobuf enum value: ANDROID_IN_APP = 8;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["ANDROID_IN_APP"] = 8] = "ANDROID_IN_APP";
    /**
     * Conversion that occurs when a user takes an in-app action such as a
     * purchase in an iOS app;
     * Displayed in Google Ads UI as 'iOS in-app action'.
     *
     * @generated from protobuf enum value: IOS_IN_APP = 9;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["IOS_IN_APP"] = 9] = "IOS_IN_APP";
    /**
     * Conversion that occurs when a user opens an iOS app for the first time;
     * Displayed in Google Ads UI as 'iOS app install (first open)'.
     *
     * @generated from protobuf enum value: IOS_FIRST_OPEN = 10;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["IOS_FIRST_OPEN"] = 10] = "IOS_FIRST_OPEN";
    /**
     * Legacy app conversions that do not have an AppPlatform provided;
     * Displayed in Google Ads UI as 'Mobile app'.
     *
     * @generated from protobuf enum value: APP_UNSPECIFIED = 11;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["APP_UNSPECIFIED"] = 11] = "APP_UNSPECIFIED";
    /**
     * Conversion that occurs when a user opens an Android app for the first
     * time; Displayed in Google Ads UI as 'Android app install (first open)'.
     *
     * @generated from protobuf enum value: ANDROID_FIRST_OPEN = 12;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["ANDROID_FIRST_OPEN"] = 12] = "ANDROID_FIRST_OPEN";
    /**
     * Call conversion that is uploaded through ConversionUploadService;
     * Displayed in Google Ads UI as 'Import from calls'.
     *
     * @generated from protobuf enum value: UPLOAD_CALLS = 13;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["UPLOAD_CALLS"] = 13] = "UPLOAD_CALLS";
    /**
     * Conversion that comes from a linked Firebase event;
     * Displayed in Google Ads UI as 'Firebase'.
     *
     * @generated from protobuf enum value: FIREBASE = 14;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["FIREBASE"] = 14] = "FIREBASE";
    /**
     * Conversion that occurs when a user clicks on a mobile phone number;
     * Displayed in Google Ads UI as 'Phone number clicks'.
     *
     * @generated from protobuf enum value: CLICK_TO_CALL = 15;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["CLICK_TO_CALL"] = 15] = "CLICK_TO_CALL";
    /**
     * Conversion that comes from Salesforce;
     * Displayed in Google Ads UI as 'Salesforce.com'.
     *
     * @generated from protobuf enum value: SALESFORCE = 16;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["SALESFORCE"] = 16] = "SALESFORCE";
    /**
     * Conversion that comes from in-store purchases recorded by CRM;
     * Displayed in Google Ads UI as 'Store sales (data partner)'.
     *
     * @generated from protobuf enum value: STORE_SALES_CRM = 17;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["STORE_SALES_CRM"] = 17] = "STORE_SALES_CRM";
    /**
     * Conversion that comes from in-store purchases from payment network;
     * Displayed in Google Ads UI as 'Store sales (payment network)'.
     *
     * @generated from protobuf enum value: STORE_SALES_PAYMENT_NETWORK = 18;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["STORE_SALES_PAYMENT_NETWORK"] = 18] = "STORE_SALES_PAYMENT_NETWORK";
    /**
     * Codeless Google Play conversion;
     * Displayed in Google Ads UI as 'Google Play'.
     *
     * @generated from protobuf enum value: GOOGLE_PLAY = 19;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["GOOGLE_PLAY"] = 19] = "GOOGLE_PLAY";
    /**
     * Conversion that comes from a linked third-party app analytics event;
     * Displayed in Google Ads UI as 'Third-party app analytics'.
     *
     * @generated from protobuf enum value: THIRD_PARTY_APP_ANALYTICS = 20;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["THIRD_PARTY_APP_ANALYTICS"] = 20] = "THIRD_PARTY_APP_ANALYTICS";
    /**
     * Conversion that is controlled by Google Attribution.
     *
     * @generated from protobuf enum value: GOOGLE_ATTRIBUTION = 21;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["GOOGLE_ATTRIBUTION"] = 21] = "GOOGLE_ATTRIBUTION";
    /**
     * Store Sales conversion based on first-party or third-party merchant data
     * uploads. Displayed in Google Ads UI as 'Store sales (direct upload)'.
     *
     * @generated from protobuf enum value: STORE_SALES_DIRECT_UPLOAD = 23;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["STORE_SALES_DIRECT_UPLOAD"] = 23] = "STORE_SALES_DIRECT_UPLOAD";
    /**
     * Store Sales conversion based on first-party or third-party merchant
     * data uploads and/or from in-store purchases using cards from payment
     * networks. Displayed in Google Ads UI as 'Store sales'.
     *
     * @generated from protobuf enum value: STORE_SALES = 24;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["STORE_SALES"] = 24] = "STORE_SALES";
    /**
     * Conversions imported from Search Ads 360 Floodlight data.
     *
     * @generated from protobuf enum value: SEARCH_ADS_360 = 25;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["SEARCH_ADS_360"] = 25] = "SEARCH_ADS_360";
    /**
     * Conversions that track local actions from Google's products and services
     * after interacting with an ad.
     *
     * @generated from protobuf enum value: GOOGLE_HOSTED = 27;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["GOOGLE_HOSTED"] = 27] = "GOOGLE_HOSTED";
    /**
     * Conversions reported by Floodlight tags.
     *
     * @generated from protobuf enum value: FLOODLIGHT = 29;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["FLOODLIGHT"] = 29] = "FLOODLIGHT";
    /**
     * Conversions that come from Google Analytics specifically for Search Ads
     * 360. Displayed in Google Ads UI as Analytics (SA360).
     *
     * @generated from protobuf enum value: ANALYTICS_SEARCH_ADS_360 = 31;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["ANALYTICS_SEARCH_ADS_360"] = 31] = "ANALYTICS_SEARCH_ADS_360";
    /**
     * Conversion that comes from a linked Firebase event for Search Ads 360.
     *
     * @generated from protobuf enum value: FIREBASE_SEARCH_ADS_360 = 33;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["FIREBASE_SEARCH_ADS_360"] = 33] = "FIREBASE_SEARCH_ADS_360";
    /**
     * Conversion that is reported by Floodlight for DV360.
     *
     * @generated from protobuf enum value: DISPLAY_AND_VIDEO_360_FLOODLIGHT = 34;
     */
    ExternalConversionSourceEnum_ExternalConversionSource[ExternalConversionSourceEnum_ExternalConversionSource["DISPLAY_AND_VIDEO_360_FLOODLIGHT"] = 34] = "DISPLAY_AND_VIDEO_360_FLOODLIGHT";
})(ExternalConversionSourceEnum_ExternalConversionSource = exports.ExternalConversionSourceEnum_ExternalConversionSource || (exports.ExternalConversionSourceEnum_ExternalConversionSource = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ExternalConversionSourceEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ExternalConversionSourceEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ExternalConversionSourceEnum
 */
exports.ExternalConversionSourceEnum = new ExternalConversionSourceEnum$Type();
