// google-ads-api-client/dist/protos/bindings/ads/enums/extension_setting_device.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionSettingDeviceEnum = exports.ExtensionSettingDeviceEnum_ExtensionSettingDevice = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible device types for an extension setting.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ExtensionSettingDeviceEnum.ExtensionSettingDevice
 */
var ExtensionSettingDeviceEnum_ExtensionSettingDevice;
(function (ExtensionSettingDeviceEnum_ExtensionSettingDevice) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ExtensionSettingDeviceEnum_ExtensionSettingDevice[ExtensionSettingDeviceEnum_ExtensionSettingDevice["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ExtensionSettingDeviceEnum_ExtensionSettingDevice[ExtensionSettingDeviceEnum_ExtensionSettingDevice["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Mobile. The extensions in the extension setting will only serve on
     * mobile devices.
     *
     * @generated from protobuf enum value: MOBILE = 2;
     */
    ExtensionSettingDeviceEnum_ExtensionSettingDevice[ExtensionSettingDeviceEnum_ExtensionSettingDevice["MOBILE"] = 2] = "MOBILE";
    /**
     * Desktop. The extensions in the extension setting will only serve on
     * desktop devices.
     *
     * @generated from protobuf enum value: DESKTOP = 3;
     */
    ExtensionSettingDeviceEnum_ExtensionSettingDevice[ExtensionSettingDeviceEnum_ExtensionSettingDevice["DESKTOP"] = 3] = "DESKTOP";
})(ExtensionSettingDeviceEnum_ExtensionSettingDevice = exports.ExtensionSettingDeviceEnum_ExtensionSettingDevice || (exports.ExtensionSettingDeviceEnum_ExtensionSettingDevice = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ExtensionSettingDeviceEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ExtensionSettingDeviceEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ExtensionSettingDeviceEnum
 */
exports.ExtensionSettingDeviceEnum = new ExtensionSettingDeviceEnum$Type();
