// google-ads-api-client/dist/protos/bindings/ads/enums/extension_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionTypeEnum = exports.ExtensionTypeEnum_ExtensionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible data types for an extension in an extension setting.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ExtensionTypeEnum.ExtensionType
 */
var ExtensionTypeEnum_ExtensionType;
(function (ExtensionTypeEnum_ExtensionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * None.
     *
     * @generated from protobuf enum value: NONE = 2;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["NONE"] = 2] = "NONE";
    /**
     * App.
     *
     * @generated from protobuf enum value: APP = 3;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["APP"] = 3] = "APP";
    /**
     * Call.
     *
     * @generated from protobuf enum value: CALL = 4;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["CALL"] = 4] = "CALL";
    /**
     * Callout.
     *
     * @generated from protobuf enum value: CALLOUT = 5;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["CALLOUT"] = 5] = "CALLOUT";
    /**
     * Message.
     *
     * @generated from protobuf enum value: MESSAGE = 6;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["MESSAGE"] = 6] = "MESSAGE";
    /**
     * Price.
     *
     * @generated from protobuf enum value: PRICE = 7;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["PRICE"] = 7] = "PRICE";
    /**
     * Promotion.
     *
     * @generated from protobuf enum value: PROMOTION = 8;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["PROMOTION"] = 8] = "PROMOTION";
    /**
     * Sitelink.
     *
     * @generated from protobuf enum value: SITELINK = 10;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["SITELINK"] = 10] = "SITELINK";
    /**
     * Structured snippet.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPET = 11;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["STRUCTURED_SNIPPET"] = 11] = "STRUCTURED_SNIPPET";
    /**
     * Location.
     *
     * @generated from protobuf enum value: LOCATION = 12;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["LOCATION"] = 12] = "LOCATION";
    /**
     * Affiliate location.
     *
     * @generated from protobuf enum value: AFFILIATE_LOCATION = 13;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["AFFILIATE_LOCATION"] = 13] = "AFFILIATE_LOCATION";
    /**
     * Hotel callout
     *
     * @generated from protobuf enum value: HOTEL_CALLOUT = 15;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["HOTEL_CALLOUT"] = 15] = "HOTEL_CALLOUT";
    /**
     * Image.
     *
     * @generated from protobuf enum value: IMAGE = 16;
     */
    ExtensionTypeEnum_ExtensionType[ExtensionTypeEnum_ExtensionType["IMAGE"] = 16] = "IMAGE";
})(ExtensionTypeEnum_ExtensionType = exports.ExtensionTypeEnum_ExtensionType || (exports.ExtensionTypeEnum_ExtensionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ExtensionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ExtensionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ExtensionTypeEnum
 */
exports.ExtensionTypeEnum = new ExtensionTypeEnum$Type();
