// google-ads-api-client/dist/protos/bindings/ads/enums/feed_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedStatusEnum = exports.FeedStatusEnum_FeedStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a feed.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedStatusEnum.FeedStatus
 */
var FeedStatusEnum_FeedStatus;
(function (FeedStatusEnum_FeedStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedStatusEnum_FeedStatus[FeedStatusEnum_FeedStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedStatusEnum_FeedStatus[FeedStatusEnum_FeedStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Feed is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    FeedStatusEnum_FeedStatus[FeedStatusEnum_FeedStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Feed has been removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    FeedStatusEnum_FeedStatus[FeedStatusEnum_FeedStatus["REMOVED"] = 3] = "REMOVED";
})(FeedStatusEnum_FeedStatus = exports.FeedStatusEnum_FeedStatus || (exports.FeedStatusEnum_FeedStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedStatusEnum
 */
exports.FeedStatusEnum = new FeedStatusEnum$Type();
