// google-ads-api-client/dist/protos/bindings/ads/enums/feed_item_quality_disapproval_reason.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemQualityDisapprovalReasonEnum = exports.FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible quality evaluation disapproval reasons of a feed item.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedItemQualityDisapprovalReasonEnum.FeedItemQualityDisapprovalReason
 */
var FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason;
(function (FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Price contains repetitive headers.
     *
     * @generated from protobuf enum value: PRICE_TABLE_REPETITIVE_HEADERS = 2;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_REPETITIVE_HEADERS"] = 2] = "PRICE_TABLE_REPETITIVE_HEADERS";
    /**
     * Price contains repetitive description.
     *
     * @generated from protobuf enum value: PRICE_TABLE_REPETITIVE_DESCRIPTION = 3;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_REPETITIVE_DESCRIPTION"] = 3] = "PRICE_TABLE_REPETITIVE_DESCRIPTION";
    /**
     * Price contains inconsistent items.
     *
     * @generated from protobuf enum value: PRICE_TABLE_INCONSISTENT_ROWS = 4;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_INCONSISTENT_ROWS"] = 4] = "PRICE_TABLE_INCONSISTENT_ROWS";
    /**
     * Price contains qualifiers in description.
     *
     * @generated from protobuf enum value: PRICE_DESCRIPTION_HAS_PRICE_QUALIFIERS = 5;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_DESCRIPTION_HAS_PRICE_QUALIFIERS"] = 5] = "PRICE_DESCRIPTION_HAS_PRICE_QUALIFIERS";
    /**
     * Price contains an unsupported language.
     *
     * @generated from protobuf enum value: PRICE_UNSUPPORTED_LANGUAGE = 6;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_UNSUPPORTED_LANGUAGE"] = 6] = "PRICE_UNSUPPORTED_LANGUAGE";
    /**
     * Price item header is not relevant to the price type.
     *
     * @generated from protobuf enum value: PRICE_TABLE_ROW_HEADER_TABLE_TYPE_MISMATCH = 7;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_ROW_HEADER_TABLE_TYPE_MISMATCH"] = 7] = "PRICE_TABLE_ROW_HEADER_TABLE_TYPE_MISMATCH";
    /**
     * Price item header has promotional text.
     *
     * @generated from protobuf enum value: PRICE_TABLE_ROW_HEADER_HAS_PROMOTIONAL_TEXT = 8;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_ROW_HEADER_HAS_PROMOTIONAL_TEXT"] = 8] = "PRICE_TABLE_ROW_HEADER_HAS_PROMOTIONAL_TEXT";
    /**
     * Price item description is not relevant to the item header.
     *
     * @generated from protobuf enum value: PRICE_TABLE_ROW_DESCRIPTION_NOT_RELEVANT = 9;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_ROW_DESCRIPTION_NOT_RELEVANT"] = 9] = "PRICE_TABLE_ROW_DESCRIPTION_NOT_RELEVANT";
    /**
     * Price item description contains promotional text.
     *
     * @generated from protobuf enum value: PRICE_TABLE_ROW_DESCRIPTION_HAS_PROMOTIONAL_TEXT = 10;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_ROW_DESCRIPTION_HAS_PROMOTIONAL_TEXT"] = 10] = "PRICE_TABLE_ROW_DESCRIPTION_HAS_PROMOTIONAL_TEXT";
    /**
     * Price item header and description are repetitive.
     *
     * @generated from protobuf enum value: PRICE_TABLE_ROW_HEADER_DESCRIPTION_REPETITIVE = 11;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_ROW_HEADER_DESCRIPTION_REPETITIVE"] = 11] = "PRICE_TABLE_ROW_HEADER_DESCRIPTION_REPETITIVE";
    /**
     * Price item is in a foreign language, nonsense, or can't be rated.
     *
     * @generated from protobuf enum value: PRICE_TABLE_ROW_UNRATEABLE = 12;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_ROW_UNRATEABLE"] = 12] = "PRICE_TABLE_ROW_UNRATEABLE";
    /**
     * Price item price is invalid or inaccurate.
     *
     * @generated from protobuf enum value: PRICE_TABLE_ROW_PRICE_INVALID = 13;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_ROW_PRICE_INVALID"] = 13] = "PRICE_TABLE_ROW_PRICE_INVALID";
    /**
     * Price item URL is invalid or irrelevant.
     *
     * @generated from protobuf enum value: PRICE_TABLE_ROW_URL_INVALID = 14;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_TABLE_ROW_URL_INVALID"] = 14] = "PRICE_TABLE_ROW_URL_INVALID";
    /**
     * Price item header or description has price.
     *
     * @generated from protobuf enum value: PRICE_HEADER_OR_DESCRIPTION_HAS_PRICE = 15;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["PRICE_HEADER_OR_DESCRIPTION_HAS_PRICE"] = 15] = "PRICE_HEADER_OR_DESCRIPTION_HAS_PRICE";
    /**
     * Structured snippet values do not match the header.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPETS_HEADER_POLICY_VIOLATED = 16;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["STRUCTURED_SNIPPETS_HEADER_POLICY_VIOLATED"] = 16] = "STRUCTURED_SNIPPETS_HEADER_POLICY_VIOLATED";
    /**
     * Structured snippet values are repeated.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPETS_REPEATED_VALUES = 17;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["STRUCTURED_SNIPPETS_REPEATED_VALUES"] = 17] = "STRUCTURED_SNIPPETS_REPEATED_VALUES";
    /**
     * Structured snippet values violate editorial guidelines like punctuation.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPETS_EDITORIAL_GUIDELINES = 18;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["STRUCTURED_SNIPPETS_EDITORIAL_GUIDELINES"] = 18] = "STRUCTURED_SNIPPETS_EDITORIAL_GUIDELINES";
    /**
     * Structured snippet contain promotional text.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPETS_HAS_PROMOTIONAL_TEXT = 19;
     */
    FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason[FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason["STRUCTURED_SNIPPETS_HAS_PROMOTIONAL_TEXT"] = 19] = "STRUCTURED_SNIPPETS_HAS_PROMOTIONAL_TEXT";
})(FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason = exports.FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason || (exports.FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemQualityDisapprovalReasonEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedItemQualityDisapprovalReasonEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedItemQualityDisapprovalReasonEnum
 */
exports.FeedItemQualityDisapprovalReasonEnum = new FeedItemQualityDisapprovalReasonEnum$Type();
