// google-ads-api-client/dist/protos/bindings/ads/enums/feed_mapping_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedMappingStatusEnum = exports.FeedMappingStatusEnum_FeedMappingStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a feed mapping.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedMappingStatusEnum.FeedMappingStatus
 */
var FeedMappingStatusEnum_FeedMappingStatus;
(function (FeedMappingStatusEnum_FeedMappingStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedMappingStatusEnum_FeedMappingStatus[FeedMappingStatusEnum_FeedMappingStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedMappingStatusEnum_FeedMappingStatus[FeedMappingStatusEnum_FeedMappingStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Feed mapping is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    FeedMappingStatusEnum_FeedMappingStatus[FeedMappingStatusEnum_FeedMappingStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Feed mapping has been removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    FeedMappingStatusEnum_FeedMappingStatus[FeedMappingStatusEnum_FeedMappingStatus["REMOVED"] = 3] = "REMOVED";
})(FeedMappingStatusEnum_FeedMappingStatus = exports.FeedMappingStatusEnum_FeedMappingStatus || (exports.FeedMappingStatusEnum_FeedMappingStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedMappingStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedMappingStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedMappingStatusEnum
 */
exports.FeedMappingStatusEnum = new FeedMappingStatusEnum$Type();
