// google-ads-api-client/dist/protos/bindings/ads/enums/feed_item_target_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemTargetStatusEnum = exports.FeedItemTargetStatusEnum_FeedItemTargetStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a feed item target.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedItemTargetStatusEnum.FeedItemTargetStatus
 */
var FeedItemTargetStatusEnum_FeedItemTargetStatus;
(function (FeedItemTargetStatusEnum_FeedItemTargetStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemTargetStatusEnum_FeedItemTargetStatus[FeedItemTargetStatusEnum_FeedItemTargetStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemTargetStatusEnum_FeedItemTargetStatus[FeedItemTargetStatusEnum_FeedItemTargetStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Feed item target is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    FeedItemTargetStatusEnum_FeedItemTargetStatus[FeedItemTargetStatusEnum_FeedItemTargetStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Feed item target has been removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    FeedItemTargetStatusEnum_FeedItemTargetStatus[FeedItemTargetStatusEnum_FeedItemTargetStatus["REMOVED"] = 3] = "REMOVED";
})(FeedItemTargetStatusEnum_FeedItemTargetStatus = exports.FeedItemTargetStatusEnum_FeedItemTargetStatus || (exports.FeedItemTargetStatusEnum_FeedItemTargetStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemTargetStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedItemTargetStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedItemTargetStatusEnum
 */
exports.FeedItemTargetStatusEnum = new FeedItemTargetStatusEnum$Type();
