// google-ads-api-client/dist/protos/bindings/ads/enums/feed_mapping_criterion_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedMappingCriterionTypeEnum = exports.FeedMappingCriterionTypeEnum_FeedMappingCriterionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible placeholder types for a feed mapping.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedMappingCriterionTypeEnum.FeedMappingCriterionType
 */
var FeedMappingCriterionTypeEnum_FeedMappingCriterionType;
(function (FeedMappingCriterionTypeEnum_FeedMappingCriterionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedMappingCriterionTypeEnum_FeedMappingCriterionType[FeedMappingCriterionTypeEnum_FeedMappingCriterionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedMappingCriterionTypeEnum_FeedMappingCriterionType[FeedMappingCriterionTypeEnum_FeedMappingCriterionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Allows campaign targeting at locations within a location feed.
     *
     * @generated from protobuf enum value: LOCATION_EXTENSION_TARGETING = 4;
     */
    FeedMappingCriterionTypeEnum_FeedMappingCriterionType[FeedMappingCriterionTypeEnum_FeedMappingCriterionType["LOCATION_EXTENSION_TARGETING"] = 4] = "LOCATION_EXTENSION_TARGETING";
    /**
     * Allows url targeting for your dynamic search ads within a page feed.
     *
     * @generated from protobuf enum value: DSA_PAGE_FEED = 3;
     */
    FeedMappingCriterionTypeEnum_FeedMappingCriterionType[FeedMappingCriterionTypeEnum_FeedMappingCriterionType["DSA_PAGE_FEED"] = 3] = "DSA_PAGE_FEED";
})(FeedMappingCriterionTypeEnum_FeedMappingCriterionType = exports.FeedMappingCriterionTypeEnum_FeedMappingCriterionType || (exports.FeedMappingCriterionTypeEnum_FeedMappingCriterionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedMappingCriterionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedMappingCriterionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedMappingCriterionTypeEnum
 */
exports.FeedMappingCriterionTypeEnum = new FeedMappingCriterionTypeEnum$Type();
