// google-ads-api-client/dist/protos/bindings/ads/enums/feed_item_validation_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemValidationStatusEnum = exports.FeedItemValidationStatusEnum_FeedItemValidationStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible validation statuses of a feed item.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedItemValidationStatusEnum.FeedItemValidationStatus
 */
var FeedItemValidationStatusEnum_FeedItemValidationStatus;
(function (FeedItemValidationStatusEnum_FeedItemValidationStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemValidationStatusEnum_FeedItemValidationStatus[FeedItemValidationStatusEnum_FeedItemValidationStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemValidationStatusEnum_FeedItemValidationStatus[FeedItemValidationStatusEnum_FeedItemValidationStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Validation pending.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    FeedItemValidationStatusEnum_FeedItemValidationStatus[FeedItemValidationStatusEnum_FeedItemValidationStatus["PENDING"] = 2] = "PENDING";
    /**
     * An error was found.
     *
     * @generated from protobuf enum value: INVALID = 3;
     */
    FeedItemValidationStatusEnum_FeedItemValidationStatus[FeedItemValidationStatusEnum_FeedItemValidationStatus["INVALID"] = 3] = "INVALID";
    /**
     * Feed item is semantically well-formed.
     *
     * @generated from protobuf enum value: VALID = 4;
     */
    FeedItemValidationStatusEnum_FeedItemValidationStatus[FeedItemValidationStatusEnum_FeedItemValidationStatus["VALID"] = 4] = "VALID";
})(FeedItemValidationStatusEnum_FeedItemValidationStatus = exports.FeedItemValidationStatusEnum_FeedItemValidationStatus || (exports.FeedItemValidationStatusEnum_FeedItemValidationStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemValidationStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedItemValidationStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedItemValidationStatusEnum
 */
exports.FeedItemValidationStatusEnum = new FeedItemValidationStatusEnum$Type();
