// google-ads-api-client/dist/protos/bindings/ads/enums/feed_link_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedLinkStatusEnum = exports.FeedLinkStatusEnum_FeedLinkStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a feed link.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedLinkStatusEnum.FeedLinkStatus
 */
var FeedLinkStatusEnum_FeedLinkStatus;
(function (FeedLinkStatusEnum_FeedLinkStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedLinkStatusEnum_FeedLinkStatus[FeedLinkStatusEnum_FeedLinkStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedLinkStatusEnum_FeedLinkStatus[FeedLinkStatusEnum_FeedLinkStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Feed link is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    FeedLinkStatusEnum_FeedLinkStatus[FeedLinkStatusEnum_FeedLinkStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Feed link has been removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    FeedLinkStatusEnum_FeedLinkStatus[FeedLinkStatusEnum_FeedLinkStatus["REMOVED"] = 3] = "REMOVED";
})(FeedLinkStatusEnum_FeedLinkStatus = exports.FeedLinkStatusEnum_FeedLinkStatus || (exports.FeedLinkStatusEnum_FeedLinkStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedLinkStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedLinkStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedLinkStatusEnum
 */
exports.FeedLinkStatusEnum = new FeedLinkStatusEnum$Type();
