// google-ads-api-client/dist/protos/bindings/ads/enums/feed_item_set_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemSetStatusEnum = exports.FeedItemSetStatusEnum_FeedItemSetStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a feed item set.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedItemSetStatusEnum.FeedItemSetStatus
 */
var FeedItemSetStatusEnum_FeedItemSetStatus;
(function (FeedItemSetStatusEnum_FeedItemSetStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemSetStatusEnum_FeedItemSetStatus[FeedItemSetStatusEnum_FeedItemSetStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemSetStatusEnum_FeedItemSetStatus[FeedItemSetStatusEnum_FeedItemSetStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Feed item set is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    FeedItemSetStatusEnum_FeedItemSetStatus[FeedItemSetStatusEnum_FeedItemSetStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Feed item set has been removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    FeedItemSetStatusEnum_FeedItemSetStatus[FeedItemSetStatusEnum_FeedItemSetStatus["REMOVED"] = 3] = "REMOVED";
})(FeedItemSetStatusEnum_FeedItemSetStatus = exports.FeedItemSetStatusEnum_FeedItemSetStatus || (exports.FeedItemSetStatusEnum_FeedItemSetStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemSetStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedItemSetStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedItemSetStatusEnum
 */
exports.FeedItemSetStatusEnum = new FeedItemSetStatusEnum$Type();
