// google-ads-api-client/dist/protos/bindings/ads/enums/feed_attribute_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedAttributeTypeEnum = exports.FeedAttributeTypeEnum_FeedAttributeType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible data types for a feed attribute.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedAttributeTypeEnum.FeedAttributeType
 */
var FeedAttributeTypeEnum_FeedAttributeType;
(function (FeedAttributeTypeEnum_FeedAttributeType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Int64.
     *
     * @generated from protobuf enum value: INT64 = 2;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["INT64"] = 2] = "INT64";
    /**
     * Double.
     *
     * @generated from protobuf enum value: DOUBLE = 3;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["DOUBLE"] = 3] = "DOUBLE";
    /**
     * String.
     *
     * @generated from protobuf enum value: STRING = 4;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["STRING"] = 4] = "STRING";
    /**
     * Boolean.
     *
     * @generated from protobuf enum value: BOOLEAN = 5;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["BOOLEAN"] = 5] = "BOOLEAN";
    /**
     * Url.
     *
     * @generated from protobuf enum value: URL = 6;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["URL"] = 6] = "URL";
    /**
     * Datetime.
     *
     * @generated from protobuf enum value: DATE_TIME = 7;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["DATE_TIME"] = 7] = "DATE_TIME";
    /**
     * Int64 list.
     *
     * @generated from protobuf enum value: INT64_LIST = 8;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["INT64_LIST"] = 8] = "INT64_LIST";
    /**
     * Double (8 bytes) list.
     *
     * @generated from protobuf enum value: DOUBLE_LIST = 9;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["DOUBLE_LIST"] = 9] = "DOUBLE_LIST";
    /**
     * String list.
     *
     * @generated from protobuf enum value: STRING_LIST = 10;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["STRING_LIST"] = 10] = "STRING_LIST";
    /**
     * Boolean list.
     *
     * @generated from protobuf enum value: BOOLEAN_LIST = 11;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["BOOLEAN_LIST"] = 11] = "BOOLEAN_LIST";
    /**
     * Url list.
     *
     * @generated from protobuf enum value: URL_LIST = 12;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["URL_LIST"] = 12] = "URL_LIST";
    /**
     * Datetime list.
     *
     * @generated from protobuf enum value: DATE_TIME_LIST = 13;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["DATE_TIME_LIST"] = 13] = "DATE_TIME_LIST";
    /**
     * Price.
     *
     * @generated from protobuf enum value: PRICE = 14;
     */
    FeedAttributeTypeEnum_FeedAttributeType[FeedAttributeTypeEnum_FeedAttributeType["PRICE"] = 14] = "PRICE";
})(FeedAttributeTypeEnum_FeedAttributeType = exports.FeedAttributeTypeEnum_FeedAttributeType || (exports.FeedAttributeTypeEnum_FeedAttributeType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedAttributeTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedAttributeTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedAttributeTypeEnum
 */
exports.FeedAttributeTypeEnum = new FeedAttributeTypeEnum$Type();
