// google-ads-api-client/dist/protos/bindings/ads/enums/feed_item_quality_approval_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemQualityApprovalStatusEnum = exports.FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible quality evaluation approval statuses of a feed item.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedItemQualityApprovalStatusEnum.FeedItemQualityApprovalStatus
 */
var FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus;
(function (FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus[FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus[FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Meets all quality expectations.
     *
     * @generated from protobuf enum value: APPROVED = 2;
     */
    FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus[FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus["APPROVED"] = 2] = "APPROVED";
    /**
     * Does not meet some quality expectations. The specific reason is found in
     * the quality_disapproval_reasons field.
     *
     * @generated from protobuf enum value: DISAPPROVED = 3;
     */
    FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus[FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus["DISAPPROVED"] = 3] = "DISAPPROVED";
})(FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus = exports.FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus || (exports.FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemQualityApprovalStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedItemQualityApprovalStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedItemQualityApprovalStatusEnum
 */
exports.FeedItemQualityApprovalStatusEnum = new FeedItemQualityApprovalStatusEnum$Type();
