// google-ads-api-client/dist/protos/bindings/ads/enums/feed_origin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedOriginEnum = exports.FeedOriginEnum_FeedOrigin = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for a feed origin.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedOriginEnum.FeedOrigin
 */
var FeedOriginEnum_FeedOrigin;
(function (FeedOriginEnum_FeedOrigin) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedOriginEnum_FeedOrigin[FeedOriginEnum_FeedOrigin["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedOriginEnum_FeedOrigin[FeedOriginEnum_FeedOrigin["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The FeedAttributes for this Feed are managed by the
     * user. Users can add FeedAttributes to this Feed.
     *
     * @generated from protobuf enum value: USER = 2;
     */
    FeedOriginEnum_FeedOrigin[FeedOriginEnum_FeedOrigin["USER"] = 2] = "USER";
    /**
     * The FeedAttributes for an GOOGLE Feed are created by Google. A feed of
     * this type is maintained by Google and will have the correct attributes
     * for the placeholder type of the feed.
     *
     * @generated from protobuf enum value: GOOGLE = 3;
     */
    FeedOriginEnum_FeedOrigin[FeedOriginEnum_FeedOrigin["GOOGLE"] = 3] = "GOOGLE";
})(FeedOriginEnum_FeedOrigin = exports.FeedOriginEnum_FeedOrigin || (exports.FeedOriginEnum_FeedOrigin = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedOriginEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedOriginEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedOriginEnum
 */
exports.FeedOriginEnum = new FeedOriginEnum$Type();
