// google-ads-api-client/dist/protos/bindings/ads/enums/feed_item_target_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemTargetTypeEnum = exports.FeedItemTargetTypeEnum_FeedItemTargetType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible type of a feed item target.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedItemTargetTypeEnum.FeedItemTargetType
 */
var FeedItemTargetTypeEnum_FeedItemTargetType;
(function (FeedItemTargetTypeEnum_FeedItemTargetType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemTargetTypeEnum_FeedItemTargetType[FeedItemTargetTypeEnum_FeedItemTargetType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemTargetTypeEnum_FeedItemTargetType[FeedItemTargetTypeEnum_FeedItemTargetType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Feed item targets a campaign.
     *
     * @generated from protobuf enum value: CAMPAIGN = 2;
     */
    FeedItemTargetTypeEnum_FeedItemTargetType[FeedItemTargetTypeEnum_FeedItemTargetType["CAMPAIGN"] = 2] = "CAMPAIGN";
    /**
     * Feed item targets an ad group.
     *
     * @generated from protobuf enum value: AD_GROUP = 3;
     */
    FeedItemTargetTypeEnum_FeedItemTargetType[FeedItemTargetTypeEnum_FeedItemTargetType["AD_GROUP"] = 3] = "AD_GROUP";
    /**
     * Feed item targets a criterion.
     *
     * @generated from protobuf enum value: CRITERION = 4;
     */
    FeedItemTargetTypeEnum_FeedItemTargetType[FeedItemTargetTypeEnum_FeedItemTargetType["CRITERION"] = 4] = "CRITERION";
})(FeedItemTargetTypeEnum_FeedItemTargetType = exports.FeedItemTargetTypeEnum_FeedItemTargetType || (exports.FeedItemTargetTypeEnum_FeedItemTargetType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemTargetTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedItemTargetTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedItemTargetTypeEnum
 */
exports.FeedItemTargetTypeEnum = new FeedItemTargetTypeEnum$Type();
