// google-ads-api-client/dist/protos/bindings/ads/enums/feed_item_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemStatusEnum = exports.FeedItemStatusEnum_FeedItemStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a feed item.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedItemStatusEnum.FeedItemStatus
 */
var FeedItemStatusEnum_FeedItemStatus;
(function (FeedItemStatusEnum_FeedItemStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemStatusEnum_FeedItemStatus[FeedItemStatusEnum_FeedItemStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemStatusEnum_FeedItemStatus[FeedItemStatusEnum_FeedItemStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Feed item is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    FeedItemStatusEnum_FeedItemStatus[FeedItemStatusEnum_FeedItemStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Feed item has been removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    FeedItemStatusEnum_FeedItemStatus[FeedItemStatusEnum_FeedItemStatus["REMOVED"] = 3] = "REMOVED";
})(FeedItemStatusEnum_FeedItemStatus = exports.FeedItemStatusEnum_FeedItemStatus || (exports.FeedItemStatusEnum_FeedItemStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedItemStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedItemStatusEnum
 */
exports.FeedItemStatusEnum = new FeedItemStatusEnum$Type();
