// google-ads-api-client/dist/protos/bindings/ads/enums/feed_item_target_device.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemTargetDeviceEnum = exports.FeedItemTargetDeviceEnum_FeedItemTargetDevice = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible data types for a feed item target device.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum.FeedItemTargetDevice
 */
var FeedItemTargetDeviceEnum_FeedItemTargetDevice;
(function (FeedItemTargetDeviceEnum_FeedItemTargetDevice) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemTargetDeviceEnum_FeedItemTargetDevice[FeedItemTargetDeviceEnum_FeedItemTargetDevice["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemTargetDeviceEnum_FeedItemTargetDevice[FeedItemTargetDeviceEnum_FeedItemTargetDevice["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Mobile.
     *
     * @generated from protobuf enum value: MOBILE = 2;
     */
    FeedItemTargetDeviceEnum_FeedItemTargetDevice[FeedItemTargetDeviceEnum_FeedItemTargetDevice["MOBILE"] = 2] = "MOBILE";
})(FeedItemTargetDeviceEnum_FeedItemTargetDevice = exports.FeedItemTargetDeviceEnum_FeedItemTargetDevice || (exports.FeedItemTargetDeviceEnum_FeedItemTargetDevice = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemTargetDeviceEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum
 */
exports.FeedItemTargetDeviceEnum = new FeedItemTargetDeviceEnum$Type();
