// google-ads-api-client/dist/protos/bindings/ads/enums/flight_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlightPlaceholderFieldEnum = exports.FlightPlaceholderFieldEnum_FlightPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Flight placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FlightPlaceholderFieldEnum.FlightPlaceholderField
 */
var FlightPlaceholderFieldEnum_FlightPlaceholderField;
(function (FlightPlaceholderFieldEnum_FlightPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Required. Destination id. Example: PAR, LON.
     * For feed items that only have destination id, destination id must be a
     * unique key. For feed items that have both destination id and origin id,
     * then the combination must be a unique key.
     *
     * @generated from protobuf enum value: DESTINATION_ID = 2;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["DESTINATION_ID"] = 2] = "DESTINATION_ID";
    /**
     * Data Type: STRING. Origin id. Example: PAR, LON.
     * Optional. Combination of destination id and origin id must be unique per
     * offer.
     *
     * @generated from protobuf enum value: ORIGIN_ID = 3;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["ORIGIN_ID"] = 3] = "ORIGIN_ID";
    /**
     * Data Type: STRING. Required. Main headline with product name to be shown
     * in dynamic ad.
     *
     * @generated from protobuf enum value: FLIGHT_DESCRIPTION = 4;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["FLIGHT_DESCRIPTION"] = 4] = "FLIGHT_DESCRIPTION";
    /**
     * Data Type: STRING. Shorter names are recommended.
     *
     * @generated from protobuf enum value: ORIGIN_NAME = 5;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["ORIGIN_NAME"] = 5] = "ORIGIN_NAME";
    /**
     * Data Type: STRING. Shorter names are recommended.
     *
     * @generated from protobuf enum value: DESTINATION_NAME = 6;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["DESTINATION_NAME"] = 6] = "DESTINATION_NAME";
    /**
     * Data Type: STRING. Price to be shown in the ad.
     * Example: "100.00 USD"
     *
     * @generated from protobuf enum value: FLIGHT_PRICE = 7;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["FLIGHT_PRICE"] = 7] = "FLIGHT_PRICE";
    /**
     * Data Type: STRING. Formatted price to be shown in the ad.
     * Example: "Starting at $100.00 USD", "$80 - $100"
     *
     * @generated from protobuf enum value: FORMATTED_PRICE = 8;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["FORMATTED_PRICE"] = 8] = "FORMATTED_PRICE";
    /**
     * Data Type: STRING. Sale price to be shown in the ad.
     * Example: "80.00 USD"
     *
     * @generated from protobuf enum value: FLIGHT_SALE_PRICE = 9;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["FLIGHT_SALE_PRICE"] = 9] = "FLIGHT_SALE_PRICE";
    /**
     * Data Type: STRING. Formatted sale price to be shown in the ad.
     * Example: "On sale for $80.00", "$60 - $80"
     *
     * @generated from protobuf enum value: FORMATTED_SALE_PRICE = 10;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["FORMATTED_SALE_PRICE"] = 10] = "FORMATTED_SALE_PRICE";
    /**
     * Data Type: URL. Image to be displayed in the ad.
     *
     * @generated from protobuf enum value: IMAGE_URL = 11;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["IMAGE_URL"] = 11] = "IMAGE_URL";
    /**
     * Data Type: URL_LIST. Required. Final URLs for the ad when using Upgraded
     * URLs. User will be redirected to these URLs when they click on an ad, or
     * when they click on a specific flight for ads that show multiple
     * flights.
     *
     * @generated from protobuf enum value: FINAL_URLS = 12;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["FINAL_URLS"] = 12] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 13;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["FINAL_MOBILE_URLS"] = 13] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 14;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["TRACKING_URL"] = 14] = "TRACKING_URL";
    /**
     * Data Type: STRING. Android app link. Must be formatted as:
     * android-app://{package_id}/{scheme}/{host_path}.
     * The components are defined as follows:
     * package_id: app ID as specified in Google Play.
     * scheme: the scheme to pass to the application. Can be HTTP, or a custom
     *   scheme.
     * host_path: identifies the specific content within your application.
     *
     * @generated from protobuf enum value: ANDROID_APP_LINK = 15;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["ANDROID_APP_LINK"] = 15] = "ANDROID_APP_LINK";
    /**
     * Data Type: STRING_LIST. List of recommended destination IDs to show
     * together with this item.
     *
     * @generated from protobuf enum value: SIMILAR_DESTINATION_IDS = 16;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["SIMILAR_DESTINATION_IDS"] = 16] = "SIMILAR_DESTINATION_IDS";
    /**
     * Data Type: STRING. iOS app link.
     *
     * @generated from protobuf enum value: IOS_APP_LINK = 17;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["IOS_APP_LINK"] = 17] = "IOS_APP_LINK";
    /**
     * Data Type: INT64. iOS app store ID.
     *
     * @generated from protobuf enum value: IOS_APP_STORE_ID = 18;
     */
    FlightPlaceholderFieldEnum_FlightPlaceholderField[FlightPlaceholderFieldEnum_FlightPlaceholderField["IOS_APP_STORE_ID"] = 18] = "IOS_APP_STORE_ID";
})(FlightPlaceholderFieldEnum_FlightPlaceholderField = exports.FlightPlaceholderFieldEnum_FlightPlaceholderField || (exports.FlightPlaceholderFieldEnum_FlightPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FlightPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FlightPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FlightPlaceholderFieldEnum
 */
exports.FlightPlaceholderFieldEnum = new FlightPlaceholderFieldEnum$Type();
