// google-ads-api-client/dist/protos/bindings/ads/enums/frequency_cap_time_unit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrequencyCapTimeUnitEnum = exports.FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Unit of time the cap is defined at (for example, day, week).
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FrequencyCapTimeUnitEnum.FrequencyCapTimeUnit
 */
var FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit;
(function (FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit[FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit[FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The cap would define limit per one day.
     *
     * @generated from protobuf enum value: DAY = 2;
     */
    FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit[FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit["DAY"] = 2] = "DAY";
    /**
     * The cap would define limit per one week.
     *
     * @generated from protobuf enum value: WEEK = 3;
     */
    FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit[FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit["WEEK"] = 3] = "WEEK";
    /**
     * The cap would define limit per one month.
     *
     * @generated from protobuf enum value: MONTH = 4;
     */
    FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit[FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit["MONTH"] = 4] = "MONTH";
})(FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit = exports.FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit || (exports.FrequencyCapTimeUnitEnum_FrequencyCapTimeUnit = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FrequencyCapTimeUnitEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FrequencyCapTimeUnitEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FrequencyCapTimeUnitEnum
 */
exports.FrequencyCapTimeUnitEnum = new FrequencyCapTimeUnitEnum$Type();
