// google-ads-api-client/dist/protos/bindings/ads/enums/frequency_cap_event_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrequencyCapEventTypeEnum = exports.FrequencyCapEventTypeEnum_FrequencyCapEventType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of event that the cap applies to (for example, impression).
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FrequencyCapEventTypeEnum.FrequencyCapEventType
 */
var FrequencyCapEventTypeEnum_FrequencyCapEventType;
(function (FrequencyCapEventTypeEnum_FrequencyCapEventType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FrequencyCapEventTypeEnum_FrequencyCapEventType[FrequencyCapEventTypeEnum_FrequencyCapEventType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FrequencyCapEventTypeEnum_FrequencyCapEventType[FrequencyCapEventTypeEnum_FrequencyCapEventType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The cap applies on ad impressions.
     *
     * @generated from protobuf enum value: IMPRESSION = 2;
     */
    FrequencyCapEventTypeEnum_FrequencyCapEventType[FrequencyCapEventTypeEnum_FrequencyCapEventType["IMPRESSION"] = 2] = "IMPRESSION";
    /**
     * The cap applies on video ad views.
     *
     * @generated from protobuf enum value: VIDEO_VIEW = 3;
     */
    FrequencyCapEventTypeEnum_FrequencyCapEventType[FrequencyCapEventTypeEnum_FrequencyCapEventType["VIDEO_VIEW"] = 3] = "VIDEO_VIEW";
})(FrequencyCapEventTypeEnum_FrequencyCapEventType = exports.FrequencyCapEventTypeEnum_FrequencyCapEventType || (exports.FrequencyCapEventTypeEnum_FrequencyCapEventType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FrequencyCapEventTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FrequencyCapEventTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FrequencyCapEventTypeEnum
 */
exports.FrequencyCapEventTypeEnum = new FrequencyCapEventTypeEnum$Type();
