// google-ads-api-client/dist/protos/bindings/ads/enums/frequency_cap_level.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrequencyCapLevelEnum = exports.FrequencyCapLevelEnum_FrequencyCapLevel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The level on which the cap is to be applied (e.g ad group ad, ad group).
 * Cap is applied to all the resources of this level.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.FrequencyCapLevelEnum.FrequencyCapLevel
 */
var FrequencyCapLevelEnum_FrequencyCapLevel;
(function (FrequencyCapLevelEnum_FrequencyCapLevel) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FrequencyCapLevelEnum_FrequencyCapLevel[FrequencyCapLevelEnum_FrequencyCapLevel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FrequencyCapLevelEnum_FrequencyCapLevel[FrequencyCapLevelEnum_FrequencyCapLevel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The cap is applied at the ad group ad level.
     *
     * @generated from protobuf enum value: AD_GROUP_AD = 2;
     */
    FrequencyCapLevelEnum_FrequencyCapLevel[FrequencyCapLevelEnum_FrequencyCapLevel["AD_GROUP_AD"] = 2] = "AD_GROUP_AD";
    /**
     * The cap is applied at the ad group level.
     *
     * @generated from protobuf enum value: AD_GROUP = 3;
     */
    FrequencyCapLevelEnum_FrequencyCapLevel[FrequencyCapLevelEnum_FrequencyCapLevel["AD_GROUP"] = 3] = "AD_GROUP";
    /**
     * The cap is applied at the campaign level.
     *
     * @generated from protobuf enum value: CAMPAIGN = 4;
     */
    FrequencyCapLevelEnum_FrequencyCapLevel[FrequencyCapLevelEnum_FrequencyCapLevel["CAMPAIGN"] = 4] = "CAMPAIGN";
})(FrequencyCapLevelEnum_FrequencyCapLevel = exports.FrequencyCapLevelEnum_FrequencyCapLevel || (exports.FrequencyCapLevelEnum_FrequencyCapLevel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FrequencyCapLevelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.FrequencyCapLevelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.FrequencyCapLevelEnum
 */
exports.FrequencyCapLevelEnum = new FrequencyCapLevelEnum$Type();
