// google-ads-api-client/dist/protos/bindings/ads/enums/gender_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenderTypeEnum = exports.GenderTypeEnum_GenderType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of demographic genders (for example, female).
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.GenderTypeEnum.GenderType
 */
var GenderTypeEnum_GenderType;
(function (GenderTypeEnum_GenderType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GenderTypeEnum_GenderType[GenderTypeEnum_GenderType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GenderTypeEnum_GenderType[GenderTypeEnum_GenderType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Male.
     *
     * @generated from protobuf enum value: MALE = 10;
     */
    GenderTypeEnum_GenderType[GenderTypeEnum_GenderType["MALE"] = 10] = "MALE";
    /**
     * Female.
     *
     * @generated from protobuf enum value: FEMALE = 11;
     */
    GenderTypeEnum_GenderType[GenderTypeEnum_GenderType["FEMALE"] = 11] = "FEMALE";
    /**
     * Undetermined gender.
     *
     * @generated from protobuf enum value: UNDETERMINED = 20;
     */
    GenderTypeEnum_GenderType[GenderTypeEnum_GenderType["UNDETERMINED"] = 20] = "UNDETERMINED";
})(GenderTypeEnum_GenderType = exports.GenderTypeEnum_GenderType || (exports.GenderTypeEnum_GenderType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GenderTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.GenderTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.GenderTypeEnum
 */
exports.GenderTypeEnum = new GenderTypeEnum$Type();
