// google-ads-api-client/dist/protos/bindings/ads/enums/geo_targeting_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeoTargetingTypeEnum = exports.GeoTargetingTypeEnum_GeoTargetingType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible geo targeting types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.GeoTargetingTypeEnum.GeoTargetingType
 */
var GeoTargetingTypeEnum_GeoTargetingType;
(function (GeoTargetingTypeEnum_GeoTargetingType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GeoTargetingTypeEnum_GeoTargetingType[GeoTargetingTypeEnum_GeoTargetingType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GeoTargetingTypeEnum_GeoTargetingType[GeoTargetingTypeEnum_GeoTargetingType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Location the user is interested in while making the query.
     *
     * @generated from protobuf enum value: AREA_OF_INTEREST = 2;
     */
    GeoTargetingTypeEnum_GeoTargetingType[GeoTargetingTypeEnum_GeoTargetingType["AREA_OF_INTEREST"] = 2] = "AREA_OF_INTEREST";
    /**
     * Location of the user issuing the query.
     *
     * @generated from protobuf enum value: LOCATION_OF_PRESENCE = 3;
     */
    GeoTargetingTypeEnum_GeoTargetingType[GeoTargetingTypeEnum_GeoTargetingType["LOCATION_OF_PRESENCE"] = 3] = "LOCATION_OF_PRESENCE";
})(GeoTargetingTypeEnum_GeoTargetingType = exports.GeoTargetingTypeEnum_GeoTargetingType || (exports.GeoTargetingTypeEnum_GeoTargetingType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GeoTargetingTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.GeoTargetingTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.GeoTargetingTypeEnum
 */
exports.GeoTargetingTypeEnum = new GeoTargetingTypeEnum$Type();
