// google-ads-api-client/dist/protos/bindings/ads/enums/geo_targeting_restriction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeoTargetingRestrictionEnum = exports.GeoTargetingRestrictionEnum_GeoTargetingRestriction = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * A restriction used to determine if the request context's
 * geo should be matched.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.GeoTargetingRestrictionEnum.GeoTargetingRestriction
 */
var GeoTargetingRestrictionEnum_GeoTargetingRestriction;
(function (GeoTargetingRestrictionEnum_GeoTargetingRestriction) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GeoTargetingRestrictionEnum_GeoTargetingRestriction[GeoTargetingRestrictionEnum_GeoTargetingRestriction["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GeoTargetingRestrictionEnum_GeoTargetingRestriction[GeoTargetingRestrictionEnum_GeoTargetingRestriction["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Indicates that request context should match the physical location of
     * the user.
     *
     * @generated from protobuf enum value: LOCATION_OF_PRESENCE = 2;
     */
    GeoTargetingRestrictionEnum_GeoTargetingRestriction[GeoTargetingRestrictionEnum_GeoTargetingRestriction["LOCATION_OF_PRESENCE"] = 2] = "LOCATION_OF_PRESENCE";
})(GeoTargetingRestrictionEnum_GeoTargetingRestriction = exports.GeoTargetingRestrictionEnum_GeoTargetingRestriction || (exports.GeoTargetingRestrictionEnum_GeoTargetingRestriction = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GeoTargetingRestrictionEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.GeoTargetingRestrictionEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.GeoTargetingRestrictionEnum
 */
exports.GeoTargetingRestrictionEnum = new GeoTargetingRestrictionEnum$Type();
