// google-ads-api-client/dist/protos/bindings/ads/enums/geo_target_constant_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeoTargetConstantStatusEnum = exports.GeoTargetConstantStatusEnum_GeoTargetConstantStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of a geo target constant.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.GeoTargetConstantStatusEnum.GeoTargetConstantStatus
 */
var GeoTargetConstantStatusEnum_GeoTargetConstantStatus;
(function (GeoTargetConstantStatusEnum_GeoTargetConstantStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GeoTargetConstantStatusEnum_GeoTargetConstantStatus[GeoTargetConstantStatusEnum_GeoTargetConstantStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GeoTargetConstantStatusEnum_GeoTargetConstantStatus[GeoTargetConstantStatusEnum_GeoTargetConstantStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The geo target constant is valid.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    GeoTargetConstantStatusEnum_GeoTargetConstantStatus[GeoTargetConstantStatusEnum_GeoTargetConstantStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The geo target constant is obsolete and will be removed.
     *
     * @generated from protobuf enum value: REMOVAL_PLANNED = 3;
     */
    GeoTargetConstantStatusEnum_GeoTargetConstantStatus[GeoTargetConstantStatusEnum_GeoTargetConstantStatus["REMOVAL_PLANNED"] = 3] = "REMOVAL_PLANNED";
})(GeoTargetConstantStatusEnum_GeoTargetConstantStatus = exports.GeoTargetConstantStatusEnum_GeoTargetConstantStatus || (exports.GeoTargetConstantStatusEnum_GeoTargetConstantStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GeoTargetConstantStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.GeoTargetConstantStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.GeoTargetConstantStatusEnum
 */
exports.GeoTargetConstantStatusEnum = new GeoTargetConstantStatusEnum$Type();
