// google-ads-api-client/dist/protos/bindings/ads/enums/goal_config_level.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoalConfigLevelEnum = exports.GoalConfigLevelEnum_GoalConfigLevel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible goal config levels. Campaigns automatically inherit the
 * effective conversion account's customer goals unless they have been
 * configured with their own set of campaign goals.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.GoalConfigLevelEnum.GoalConfigLevel
 */
var GoalConfigLevelEnum_GoalConfigLevel;
(function (GoalConfigLevelEnum_GoalConfigLevel) {
    /**
     * The goal config level has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GoalConfigLevelEnum_GoalConfigLevel[GoalConfigLevelEnum_GoalConfigLevel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The goal config level is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GoalConfigLevelEnum_GoalConfigLevel[GoalConfigLevelEnum_GoalConfigLevel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The goal config is defined at the customer level.
     *
     * @generated from protobuf enum value: CUSTOMER = 2;
     */
    GoalConfigLevelEnum_GoalConfigLevel[GoalConfigLevelEnum_GoalConfigLevel["CUSTOMER"] = 2] = "CUSTOMER";
    /**
     * The goal config is defined at the campaign level.
     *
     * @generated from protobuf enum value: CAMPAIGN = 3;
     */
    GoalConfigLevelEnum_GoalConfigLevel[GoalConfigLevelEnum_GoalConfigLevel["CAMPAIGN"] = 3] = "CAMPAIGN";
})(GoalConfigLevelEnum_GoalConfigLevel = exports.GoalConfigLevelEnum_GoalConfigLevel || (exports.GoalConfigLevelEnum_GoalConfigLevel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GoalConfigLevelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.GoalConfigLevelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.GoalConfigLevelEnum
 */
exports.GoalConfigLevelEnum = new GoalConfigLevelEnum$Type();
