// google-ads-api-client/dist/protos/bindings/ads/enums/google_ads_field_category.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleAdsFieldCategoryEnum = exports.GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The category of the artifact.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.GoogleAdsFieldCategoryEnum.GoogleAdsFieldCategory
 */
var GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory;
(function (GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory) {
    /**
     * Unspecified
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory[GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Unknown
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory[GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The described artifact is a resource.
     *
     * @generated from protobuf enum value: RESOURCE = 2;
     */
    GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory[GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory["RESOURCE"] = 2] = "RESOURCE";
    /**
     * The described artifact is a field and is an attribute of a resource.
     * Including a resource attribute field in a query may segment the query if
     * the resource to which it is attributed segments the resource found in
     * the FROM clause.
     *
     * @generated from protobuf enum value: ATTRIBUTE = 3;
     */
    GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory[GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory["ATTRIBUTE"] = 3] = "ATTRIBUTE";
    /**
     * The described artifact is a field and always segments search queries.
     *
     * @generated from protobuf enum value: SEGMENT = 5;
     */
    GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory[GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory["SEGMENT"] = 5] = "SEGMENT";
    /**
     * The described artifact is a field and is a metric. It never segments
     * search queries.
     *
     * @generated from protobuf enum value: METRIC = 6;
     */
    GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory[GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory["METRIC"] = 6] = "METRIC";
})(GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory = exports.GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory || (exports.GoogleAdsFieldCategoryEnum_GoogleAdsFieldCategory = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GoogleAdsFieldCategoryEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.GoogleAdsFieldCategoryEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.GoogleAdsFieldCategoryEnum
 */
exports.GoogleAdsFieldCategoryEnum = new GoogleAdsFieldCategoryEnum$Type();
