// google-ads-api-client/dist/protos/bindings/ads/enums/google_voice_call_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleVoiceCallStatusEnum = exports.GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a google voice call.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.GoogleVoiceCallStatusEnum.GoogleVoiceCallStatus
 */
var GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus;
(function (GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus[GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus[GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The call was missed.
     *
     * @generated from protobuf enum value: MISSED = 2;
     */
    GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus[GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus["MISSED"] = 2] = "MISSED";
    /**
     * The call was received.
     *
     * @generated from protobuf enum value: RECEIVED = 3;
     */
    GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus[GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus["RECEIVED"] = 3] = "RECEIVED";
})(GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus = exports.GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus || (exports.GoogleVoiceCallStatusEnum_GoogleVoiceCallStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GoogleVoiceCallStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.GoogleVoiceCallStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.GoogleVoiceCallStatusEnum
 */
exports.GoogleVoiceCallStatusEnum = new GoogleVoiceCallStatusEnum$Type();
