// google-ads-api-client/dist/protos/bindings/ads/enums/google_ads_field_data_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleAdsFieldDataTypeEnum = exports.GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * These are the various types a GoogleAdsService artifact may take on.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.GoogleAdsFieldDataTypeEnum.GoogleAdsFieldDataType
 */
var GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType;
(function (GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType) {
    /**
     * Unspecified
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Unknown
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Maps to google.protobuf.BoolValue
     *
     * Applicable operators:  =, !=
     *
     * @generated from protobuf enum value: BOOLEAN = 2;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["BOOLEAN"] = 2] = "BOOLEAN";
    /**
     * Maps to google.protobuf.StringValue. It can be compared using the set of
     * operators specific to dates however.
     *
     * Applicable operators:  =, <, >, <=, >=, BETWEEN, DURING, and IN
     *
     * @generated from protobuf enum value: DATE = 3;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["DATE"] = 3] = "DATE";
    /**
     * Maps to google.protobuf.DoubleValue
     *
     * Applicable operators:  =, !=, <, >, IN, NOT IN
     *
     * @generated from protobuf enum value: DOUBLE = 4;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["DOUBLE"] = 4] = "DOUBLE";
    /**
     * Maps to an enum. It's specific definition can be found at type_url.
     *
     * Applicable operators:  =, !=, IN, NOT IN
     *
     * @generated from protobuf enum value: ENUM = 5;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["ENUM"] = 5] = "ENUM";
    /**
     * Maps to google.protobuf.FloatValue
     *
     * Applicable operators:  =, !=, <, >, IN, NOT IN
     *
     * @generated from protobuf enum value: FLOAT = 6;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["FLOAT"] = 6] = "FLOAT";
    /**
     * Maps to google.protobuf.Int32Value
     *
     * Applicable operators:  =, !=, <, >, <=, >=, BETWEEN, IN, NOT IN
     *
     * @generated from protobuf enum value: INT32 = 7;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["INT32"] = 7] = "INT32";
    /**
     * Maps to google.protobuf.Int64Value
     *
     * Applicable operators:  =, !=, <, >, <=, >=, BETWEEN, IN, NOT IN
     *
     * @generated from protobuf enum value: INT64 = 8;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["INT64"] = 8] = "INT64";
    /**
     * Maps to a protocol buffer message type. The data type's details can be
     * found in type_url.
     *
     * No operators work with MESSAGE fields.
     *
     * @generated from protobuf enum value: MESSAGE = 9;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["MESSAGE"] = 9] = "MESSAGE";
    /**
     * Maps to google.protobuf.StringValue. Represents the resource name
     * (unique id) of a resource or one of its foreign keys.
     *
     * No operators work with RESOURCE_NAME fields.
     *
     * @generated from protobuf enum value: RESOURCE_NAME = 10;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["RESOURCE_NAME"] = 10] = "RESOURCE_NAME";
    /**
     * Maps to google.protobuf.StringValue.
     *
     * Applicable operators:  =, !=, LIKE, NOT LIKE, IN, NOT IN
     *
     * @generated from protobuf enum value: STRING = 11;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["STRING"] = 11] = "STRING";
    /**
     * Maps to google.protobuf.UInt64Value
     *
     * Applicable operators:  =, !=, <, >, <=, >=, BETWEEN, IN, NOT IN
     *
     * @generated from protobuf enum value: UINT64 = 12;
     */
    GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType[GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType["UINT64"] = 12] = "UINT64";
})(GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType = exports.GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType || (exports.GoogleAdsFieldDataTypeEnum_GoogleAdsFieldDataType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GoogleAdsFieldDataTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.GoogleAdsFieldDataTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.GoogleAdsFieldDataTypeEnum
 */
exports.GoogleAdsFieldDataTypeEnum = new GoogleAdsFieldDataTypeEnum$Type();
