// google-ads-api-client/dist/protos/bindings/ads/enums/hotel_reconciliation_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HotelReconciliationStatusEnum = exports.HotelReconciliationStatusEnum_HotelReconciliationStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Status of the hotel booking reconciliation.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.HotelReconciliationStatusEnum.HotelReconciliationStatus
 */
var HotelReconciliationStatusEnum_HotelReconciliationStatus;
(function (HotelReconciliationStatusEnum_HotelReconciliationStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    HotelReconciliationStatusEnum_HotelReconciliationStatus[HotelReconciliationStatusEnum_HotelReconciliationStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    HotelReconciliationStatusEnum_HotelReconciliationStatus[HotelReconciliationStatusEnum_HotelReconciliationStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Bookings are for a future date, or a stay is underway but the check-out
     * date hasn't passed. An active reservation can't be reconciled.
     *
     * @generated from protobuf enum value: RESERVATION_ENABLED = 2;
     */
    HotelReconciliationStatusEnum_HotelReconciliationStatus[HotelReconciliationStatusEnum_HotelReconciliationStatus["RESERVATION_ENABLED"] = 2] = "RESERVATION_ENABLED";
    /**
     * Check-out has already taken place, or the booked dates have passed
     * without cancellation. Bookings that are not reconciled within 45 days of
     * the check-out date are billed based on the original booking price.
     *
     * @generated from protobuf enum value: RECONCILIATION_NEEDED = 3;
     */
    HotelReconciliationStatusEnum_HotelReconciliationStatus[HotelReconciliationStatusEnum_HotelReconciliationStatus["RECONCILIATION_NEEDED"] = 3] = "RECONCILIATION_NEEDED";
    /**
     * These bookings have been reconciled. Reconciled bookings are billed 45
     * days after the check-out date.
     *
     * @generated from protobuf enum value: RECONCILED = 4;
     */
    HotelReconciliationStatusEnum_HotelReconciliationStatus[HotelReconciliationStatusEnum_HotelReconciliationStatus["RECONCILED"] = 4] = "RECONCILED";
    /**
     * This booking was marked as canceled. Canceled stays with a value greater
     * than zero (due to minimum stay rules or cancellation fees) are billed 45
     * days after the check-out date.
     *
     * @generated from protobuf enum value: CANCELED = 5;
     */
    HotelReconciliationStatusEnum_HotelReconciliationStatus[HotelReconciliationStatusEnum_HotelReconciliationStatus["CANCELED"] = 5] = "CANCELED";
})(HotelReconciliationStatusEnum_HotelReconciliationStatus = exports.HotelReconciliationStatusEnum_HotelReconciliationStatus || (exports.HotelReconciliationStatusEnum_HotelReconciliationStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class HotelReconciliationStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.HotelReconciliationStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.HotelReconciliationStatusEnum
 */
exports.HotelReconciliationStatusEnum = new HotelReconciliationStatusEnum$Type();
