// google-ads-api-client/dist/protos/bindings/ads/enums/hotel_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HotelPlaceholderFieldEnum = exports.HotelPlaceholderFieldEnum_HotelPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Hotel placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.HotelPlaceholderFieldEnum.HotelPlaceholderField
 */
var HotelPlaceholderFieldEnum_HotelPlaceholderField;
(function (HotelPlaceholderFieldEnum_HotelPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Required. Unique ID.
     *
     * @generated from protobuf enum value: PROPERTY_ID = 2;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["PROPERTY_ID"] = 2] = "PROPERTY_ID";
    /**
     * Data Type: STRING. Required. Main headline with property name to be shown
     * in dynamic ad.
     *
     * @generated from protobuf enum value: PROPERTY_NAME = 3;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["PROPERTY_NAME"] = 3] = "PROPERTY_NAME";
    /**
     * Data Type: STRING. Name of destination to be shown in dynamic ad.
     *
     * @generated from protobuf enum value: DESTINATION_NAME = 4;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["DESTINATION_NAME"] = 4] = "DESTINATION_NAME";
    /**
     * Data Type: STRING. Description of destination to be shown in dynamic ad.
     *
     * @generated from protobuf enum value: DESCRIPTION = 5;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["DESCRIPTION"] = 5] = "DESCRIPTION";
    /**
     * Data Type: STRING. Complete property address, including postal code.
     *
     * @generated from protobuf enum value: ADDRESS = 6;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["ADDRESS"] = 6] = "ADDRESS";
    /**
     * Data Type: STRING. Price to be shown in the ad.
     * Example: "100.00 USD"
     *
     * @generated from protobuf enum value: PRICE = 7;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["PRICE"] = 7] = "PRICE";
    /**
     * Data Type: STRING. Formatted price to be shown in the ad.
     * Example: "Starting at $100.00 USD", "$80 - $100"
     *
     * @generated from protobuf enum value: FORMATTED_PRICE = 8;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["FORMATTED_PRICE"] = 8] = "FORMATTED_PRICE";
    /**
     * Data Type: STRING. Sale price to be shown in the ad.
     * Example: "80.00 USD"
     *
     * @generated from protobuf enum value: SALE_PRICE = 9;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["SALE_PRICE"] = 9] = "SALE_PRICE";
    /**
     * Data Type: STRING. Formatted sale price to be shown in the ad.
     * Example: "On sale for $80.00", "$60 - $80"
     *
     * @generated from protobuf enum value: FORMATTED_SALE_PRICE = 10;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["FORMATTED_SALE_PRICE"] = 10] = "FORMATTED_SALE_PRICE";
    /**
     * Data Type: URL. Image to be displayed in the ad.
     *
     * @generated from protobuf enum value: IMAGE_URL = 11;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["IMAGE_URL"] = 11] = "IMAGE_URL";
    /**
     * Data Type: STRING. Category of property used to group like items together
     * for recommendation engine.
     *
     * @generated from protobuf enum value: CATEGORY = 12;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["CATEGORY"] = 12] = "CATEGORY";
    /**
     * Data Type: INT64. Star rating (1 to 5) used to group like items
     * together for recommendation engine.
     *
     * @generated from protobuf enum value: STAR_RATING = 13;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["STAR_RATING"] = 13] = "STAR_RATING";
    /**
     * Data Type: STRING_LIST. Keywords used for product retrieval.
     *
     * @generated from protobuf enum value: CONTEXTUAL_KEYWORDS = 14;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["CONTEXTUAL_KEYWORDS"] = 14] = "CONTEXTUAL_KEYWORDS";
    /**
     * Data Type: URL_LIST. Required. Final URLs for the ad when using Upgraded
     * URLs. User will be redirected to these URLs when they click on an ad, or
     * when they click on a specific flight for ads that show multiple
     * flights.
     *
     * @generated from protobuf enum value: FINAL_URLS = 15;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["FINAL_URLS"] = 15] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 16;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["FINAL_MOBILE_URLS"] = 16] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 17;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["TRACKING_URL"] = 17] = "TRACKING_URL";
    /**
     * Data Type: STRING. Android app link. Must be formatted as:
     * android-app://{package_id}/{scheme}/{host_path}.
     * The components are defined as follows:
     * package_id: app ID as specified in Google Play.
     * scheme: the scheme to pass to the application. Can be HTTP, or a custom
     *   scheme.
     * host_path: identifies the specific content within your application.
     *
     * @generated from protobuf enum value: ANDROID_APP_LINK = 18;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["ANDROID_APP_LINK"] = 18] = "ANDROID_APP_LINK";
    /**
     * Data Type: STRING_LIST. List of recommended property IDs to show together
     * with this item.
     *
     * @generated from protobuf enum value: SIMILAR_PROPERTY_IDS = 19;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["SIMILAR_PROPERTY_IDS"] = 19] = "SIMILAR_PROPERTY_IDS";
    /**
     * Data Type: STRING. iOS app link.
     *
     * @generated from protobuf enum value: IOS_APP_LINK = 20;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["IOS_APP_LINK"] = 20] = "IOS_APP_LINK";
    /**
     * Data Type: INT64. iOS app store ID.
     *
     * @generated from protobuf enum value: IOS_APP_STORE_ID = 21;
     */
    HotelPlaceholderFieldEnum_HotelPlaceholderField[HotelPlaceholderFieldEnum_HotelPlaceholderField["IOS_APP_STORE_ID"] = 21] = "IOS_APP_STORE_ID";
})(HotelPlaceholderFieldEnum_HotelPlaceholderField = exports.HotelPlaceholderFieldEnum_HotelPlaceholderField || (exports.HotelPlaceholderFieldEnum_HotelPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class HotelPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.HotelPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.HotelPlaceholderFieldEnum
 */
exports.HotelPlaceholderFieldEnum = new HotelPlaceholderFieldEnum$Type();
