// google-ads-api-client/dist/protos/bindings/ads/enums/hotel_date_selection_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HotelDateSelectionTypeEnum = exports.HotelDateSelectionTypeEnum_HotelDateSelectionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible hotel date selection types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.HotelDateSelectionTypeEnum.HotelDateSelectionType
 */
var HotelDateSelectionTypeEnum_HotelDateSelectionType;
(function (HotelDateSelectionTypeEnum_HotelDateSelectionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    HotelDateSelectionTypeEnum_HotelDateSelectionType[HotelDateSelectionTypeEnum_HotelDateSelectionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    HotelDateSelectionTypeEnum_HotelDateSelectionType[HotelDateSelectionTypeEnum_HotelDateSelectionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Dates selected by default.
     *
     * @generated from protobuf enum value: DEFAULT_SELECTION = 50;
     */
    HotelDateSelectionTypeEnum_HotelDateSelectionType[HotelDateSelectionTypeEnum_HotelDateSelectionType["DEFAULT_SELECTION"] = 50] = "DEFAULT_SELECTION";
    /**
     * Dates selected by the user.
     *
     * @generated from protobuf enum value: USER_SELECTED = 51;
     */
    HotelDateSelectionTypeEnum_HotelDateSelectionType[HotelDateSelectionTypeEnum_HotelDateSelectionType["USER_SELECTED"] = 51] = "USER_SELECTED";
})(HotelDateSelectionTypeEnum_HotelDateSelectionType = exports.HotelDateSelectionTypeEnum_HotelDateSelectionType || (exports.HotelDateSelectionTypeEnum_HotelDateSelectionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class HotelDateSelectionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.HotelDateSelectionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.HotelDateSelectionTypeEnum
 */
exports.HotelDateSelectionTypeEnum = new HotelDateSelectionTypeEnum$Type();
