// google-ads-api-client/dist/protos/bindings/ads/enums/hotel_price_bucket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HotelPriceBucketEnum = exports.HotelPriceBucketEnum_HotelPriceBucket = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible hotel price buckets.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.HotelPriceBucketEnum.HotelPriceBucket
 */
var HotelPriceBucketEnum_HotelPriceBucket;
(function (HotelPriceBucketEnum_HotelPriceBucket) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    HotelPriceBucketEnum_HotelPriceBucket[HotelPriceBucketEnum_HotelPriceBucket["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    HotelPriceBucketEnum_HotelPriceBucket[HotelPriceBucketEnum_HotelPriceBucket["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Uniquely lowest price. Partner has the lowest price, and no other
     * partners are within a small variance of that price.
     *
     * @generated from protobuf enum value: LOWEST_UNIQUE = 2;
     */
    HotelPriceBucketEnum_HotelPriceBucket[HotelPriceBucketEnum_HotelPriceBucket["LOWEST_UNIQUE"] = 2] = "LOWEST_UNIQUE";
    /**
     * Tied for lowest price. Partner is within a small variance of the lowest
     * price.
     *
     * @generated from protobuf enum value: LOWEST_TIED = 3;
     */
    HotelPriceBucketEnum_HotelPriceBucket[HotelPriceBucketEnum_HotelPriceBucket["LOWEST_TIED"] = 3] = "LOWEST_TIED";
    /**
     * Not lowest price. Partner is not within a small variance of the lowest
     * price.
     *
     * @generated from protobuf enum value: NOT_LOWEST = 4;
     */
    HotelPriceBucketEnum_HotelPriceBucket[HotelPriceBucketEnum_HotelPriceBucket["NOT_LOWEST"] = 4] = "NOT_LOWEST";
    /**
     * Partner was the only one shown.
     *
     * @generated from protobuf enum value: ONLY_PARTNER_SHOWN = 5;
     */
    HotelPriceBucketEnum_HotelPriceBucket[HotelPriceBucketEnum_HotelPriceBucket["ONLY_PARTNER_SHOWN"] = 5] = "ONLY_PARTNER_SHOWN";
})(HotelPriceBucketEnum_HotelPriceBucket = exports.HotelPriceBucketEnum_HotelPriceBucket || (exports.HotelPriceBucketEnum_HotelPriceBucket = {}));
// @generated message type with reflection information, may provide speed optimized methods
class HotelPriceBucketEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.HotelPriceBucketEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.HotelPriceBucketEnum
 */
exports.HotelPriceBucketEnum = new HotelPriceBucketEnum$Type();
