// google-ads-api-client/dist/protos/bindings/ads/enums/hotel_rate_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HotelRateTypeEnum = exports.HotelRateTypeEnum_HotelRateType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible hotel rate types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.HotelRateTypeEnum.HotelRateType
 */
var HotelRateTypeEnum_HotelRateType;
(function (HotelRateTypeEnum_HotelRateType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    HotelRateTypeEnum_HotelRateType[HotelRateTypeEnum_HotelRateType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    HotelRateTypeEnum_HotelRateType[HotelRateTypeEnum_HotelRateType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Rate type information is unavailable.
     *
     * @generated from protobuf enum value: UNAVAILABLE = 2;
     */
    HotelRateTypeEnum_HotelRateType[HotelRateTypeEnum_HotelRateType["UNAVAILABLE"] = 2] = "UNAVAILABLE";
    /**
     * Rates available to everyone.
     *
     * @generated from protobuf enum value: PUBLIC_RATE = 3;
     */
    HotelRateTypeEnum_HotelRateType[HotelRateTypeEnum_HotelRateType["PUBLIC_RATE"] = 3] = "PUBLIC_RATE";
    /**
     * A membership program rate is available and satisfies basic requirements
     * like having a public rate available. UI treatment will strikethrough the
     * public rate and indicate that a discount is available to the user. For
     * more on Qualified Rates, visit
     * https://developers.google.com/hotels/hotel-ads/dev-guide/qualified-rates
     *
     * @generated from protobuf enum value: QUALIFIED_RATE = 4;
     */
    HotelRateTypeEnum_HotelRateType[HotelRateTypeEnum_HotelRateType["QUALIFIED_RATE"] = 4] = "QUALIFIED_RATE";
    /**
     * Rates available to users that satisfy some eligibility criteria, for
     * example, all signed-in users, 20% of mobile users, all mobile users in
     * Canada, etc.
     *
     * @generated from protobuf enum value: PRIVATE_RATE = 5;
     */
    HotelRateTypeEnum_HotelRateType[HotelRateTypeEnum_HotelRateType["PRIVATE_RATE"] = 5] = "PRIVATE_RATE";
})(HotelRateTypeEnum_HotelRateType = exports.HotelRateTypeEnum_HotelRateType || (exports.HotelRateTypeEnum_HotelRateType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class HotelRateTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.HotelRateTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.HotelRateTypeEnum
 */
exports.HotelRateTypeEnum = new HotelRateTypeEnum$Type();
