// google-ads-api-client/dist/protos/bindings/ads/enums/income_range_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncomeRangeTypeEnum = exports.IncomeRangeTypeEnum_IncomeRangeType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of demographic income ranges (for example, between 0% to 50%).
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.IncomeRangeTypeEnum.IncomeRangeType
 */
var IncomeRangeTypeEnum_IncomeRangeType;
(function (IncomeRangeTypeEnum_IncomeRangeType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * 0%-50%.
     *
     * @generated from protobuf enum value: INCOME_RANGE_0_50 = 510001;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["INCOME_RANGE_0_50"] = 510001] = "INCOME_RANGE_0_50";
    /**
     * 50% to 60%.
     *
     * @generated from protobuf enum value: INCOME_RANGE_50_60 = 510002;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["INCOME_RANGE_50_60"] = 510002] = "INCOME_RANGE_50_60";
    /**
     * 60% to 70%.
     *
     * @generated from protobuf enum value: INCOME_RANGE_60_70 = 510003;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["INCOME_RANGE_60_70"] = 510003] = "INCOME_RANGE_60_70";
    /**
     * 70% to 80%.
     *
     * @generated from protobuf enum value: INCOME_RANGE_70_80 = 510004;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["INCOME_RANGE_70_80"] = 510004] = "INCOME_RANGE_70_80";
    /**
     * 80% to 90%.
     *
     * @generated from protobuf enum value: INCOME_RANGE_80_90 = 510005;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["INCOME_RANGE_80_90"] = 510005] = "INCOME_RANGE_80_90";
    /**
     * Greater than 90%.
     *
     * @generated from protobuf enum value: INCOME_RANGE_90_UP = 510006;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["INCOME_RANGE_90_UP"] = 510006] = "INCOME_RANGE_90_UP";
    /**
     * Undetermined income range.
     *
     * @generated from protobuf enum value: INCOME_RANGE_UNDETERMINED = 510000;
     */
    IncomeRangeTypeEnum_IncomeRangeType[IncomeRangeTypeEnum_IncomeRangeType["INCOME_RANGE_UNDETERMINED"] = 510000] = "INCOME_RANGE_UNDETERMINED";
})(IncomeRangeTypeEnum_IncomeRangeType = exports.IncomeRangeTypeEnum_IncomeRangeType || (exports.IncomeRangeTypeEnum_IncomeRangeType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class IncomeRangeTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.IncomeRangeTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.IncomeRangeTypeEnum
 */
exports.IncomeRangeTypeEnum = new IncomeRangeTypeEnum$Type();
