// google-ads-api-client/dist/protos/bindings/ads/enums/interaction_event_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InteractionEventTypeEnum = exports.InteractionEventTypeEnum_InteractionEventType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible types of payable and free interactions.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.InteractionEventTypeEnum.InteractionEventType
 */
var InteractionEventTypeEnum_InteractionEventType;
(function (InteractionEventTypeEnum_InteractionEventType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    InteractionEventTypeEnum_InteractionEventType[InteractionEventTypeEnum_InteractionEventType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    InteractionEventTypeEnum_InteractionEventType[InteractionEventTypeEnum_InteractionEventType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Click to site. In most cases, this interaction navigates to an external
     * location, usually the advertiser's landing page. This is also the default
     * InteractionEventType for click events.
     *
     * @generated from protobuf enum value: CLICK = 2;
     */
    InteractionEventTypeEnum_InteractionEventType[InteractionEventTypeEnum_InteractionEventType["CLICK"] = 2] = "CLICK";
    /**
     * The user's expressed intent to engage with the ad in-place.
     *
     * @generated from protobuf enum value: ENGAGEMENT = 3;
     */
    InteractionEventTypeEnum_InteractionEventType[InteractionEventTypeEnum_InteractionEventType["ENGAGEMENT"] = 3] = "ENGAGEMENT";
    /**
     * User viewed a video ad.
     *
     * @generated from protobuf enum value: VIDEO_VIEW = 4;
     */
    InteractionEventTypeEnum_InteractionEventType[InteractionEventTypeEnum_InteractionEventType["VIDEO_VIEW"] = 4] = "VIDEO_VIEW";
    /**
     * The default InteractionEventType for ad conversion events.
     * This is used when an ad conversion row does NOT indicate
     * that the free interactions (for example, the ad conversions)
     * should be 'promoted' and reported as part of the core metrics.
     * These are simply other (ad) conversions.
     *
     * @generated from protobuf enum value: NONE = 5;
     */
    InteractionEventTypeEnum_InteractionEventType[InteractionEventTypeEnum_InteractionEventType["NONE"] = 5] = "NONE";
})(InteractionEventTypeEnum_InteractionEventType = exports.InteractionEventTypeEnum_InteractionEventType || (exports.InteractionEventTypeEnum_InteractionEventType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class InteractionEventTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.InteractionEventTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.InteractionEventTypeEnum
 */
exports.InteractionEventTypeEnum = new InteractionEventTypeEnum$Type();
