// google-ads-api-client/dist/protos/bindings/ads/enums/interaction_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InteractionTypeEnum = exports.InteractionTypeEnum_InteractionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible interaction types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.InteractionTypeEnum.InteractionType
 */
var InteractionTypeEnum_InteractionType;
(function (InteractionTypeEnum_InteractionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    InteractionTypeEnum_InteractionType[InteractionTypeEnum_InteractionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    InteractionTypeEnum_InteractionType[InteractionTypeEnum_InteractionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Calls.
     *
     * @generated from protobuf enum value: CALLS = 8000;
     */
    InteractionTypeEnum_InteractionType[InteractionTypeEnum_InteractionType["CALLS"] = 8000] = "CALLS";
})(InteractionTypeEnum_InteractionType = exports.InteractionTypeEnum_InteractionType || (exports.InteractionTypeEnum_InteractionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class InteractionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.InteractionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.InteractionTypeEnum
 */
exports.InteractionTypeEnum = new InteractionTypeEnum$Type();
