// google-ads-api-client/dist/protos/bindings/ads/enums/invoice_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceTypeEnum = exports.InvoiceTypeEnum_InvoiceType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible type of invoices.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.InvoiceTypeEnum.InvoiceType
 */
var InvoiceTypeEnum_InvoiceType;
(function (InvoiceTypeEnum_InvoiceType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    InvoiceTypeEnum_InvoiceType[InvoiceTypeEnum_InvoiceType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    InvoiceTypeEnum_InvoiceType[InvoiceTypeEnum_InvoiceType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An invoice with a negative amount. The account receives a credit.
     *
     * @generated from protobuf enum value: CREDIT_MEMO = 2;
     */
    InvoiceTypeEnum_InvoiceType[InvoiceTypeEnum_InvoiceType["CREDIT_MEMO"] = 2] = "CREDIT_MEMO";
    /**
     * An invoice with a positive amount. The account owes a balance.
     *
     * @generated from protobuf enum value: INVOICE = 3;
     */
    InvoiceTypeEnum_InvoiceType[InvoiceTypeEnum_InvoiceType["INVOICE"] = 3] = "INVOICE";
})(InvoiceTypeEnum_InvoiceType = exports.InvoiceTypeEnum_InvoiceType || (exports.InvoiceTypeEnum_InvoiceType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class InvoiceTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.InvoiceTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.InvoiceTypeEnum
 */
exports.InvoiceTypeEnum = new InvoiceTypeEnum$Type();
