// google-ads-api-client/dist/protos/bindings/ads/enums/job_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobPlaceholderFieldEnum = exports.JobPlaceholderFieldEnum_JobPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Job placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.JobPlaceholderFieldEnum.JobPlaceholderField
 */
var JobPlaceholderFieldEnum_JobPlaceholderField;
(function (JobPlaceholderFieldEnum_JobPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Required. If only JOB_ID is specified, then it must be
     * unique. If both JOB_ID and LOCATION_ID are specified, then the
     * pair must be unique.
     * ID) pair must be unique.
     *
     * @generated from protobuf enum value: JOB_ID = 2;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["JOB_ID"] = 2] = "JOB_ID";
    /**
     * Data Type: STRING. Combination of JOB_ID and LOCATION_ID must be unique
     * per offer.
     *
     * @generated from protobuf enum value: LOCATION_ID = 3;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["LOCATION_ID"] = 3] = "LOCATION_ID";
    /**
     * Data Type: STRING. Required. Main headline with job title to be shown in
     * dynamic ad.
     *
     * @generated from protobuf enum value: TITLE = 4;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["TITLE"] = 4] = "TITLE";
    /**
     * Data Type: STRING. Job subtitle to be shown in dynamic ad.
     *
     * @generated from protobuf enum value: SUBTITLE = 5;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["SUBTITLE"] = 5] = "SUBTITLE";
    /**
     * Data Type: STRING. Description of job to be shown in dynamic ad.
     *
     * @generated from protobuf enum value: DESCRIPTION = 6;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["DESCRIPTION"] = 6] = "DESCRIPTION";
    /**
     * Data Type: URL. Image to be displayed in the ad. Highly recommended for
     * image ads.
     *
     * @generated from protobuf enum value: IMAGE_URL = 7;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["IMAGE_URL"] = 7] = "IMAGE_URL";
    /**
     * Data Type: STRING. Category of property used to group like items together
     * for recommendation engine.
     *
     * @generated from protobuf enum value: CATEGORY = 8;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["CATEGORY"] = 8] = "CATEGORY";
    /**
     * Data Type: STRING_LIST. Keywords used for product retrieval.
     *
     * @generated from protobuf enum value: CONTEXTUAL_KEYWORDS = 9;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["CONTEXTUAL_KEYWORDS"] = 9] = "CONTEXTUAL_KEYWORDS";
    /**
     * Data Type: STRING. Complete property address, including postal code.
     *
     * @generated from protobuf enum value: ADDRESS = 10;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["ADDRESS"] = 10] = "ADDRESS";
    /**
     * Data Type: STRING. Salary or salary range of job to be shown in dynamic
     * ad.
     *
     * @generated from protobuf enum value: SALARY = 11;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["SALARY"] = 11] = "SALARY";
    /**
     * Data Type: URL_LIST. Required. Final URLs to be used in ad when using
     * Upgraded URLs; the more specific the better (for example, the individual
     * URL of a specific job and its location).
     *
     * @generated from protobuf enum value: FINAL_URLS = 12;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["FINAL_URLS"] = 12] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 14;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["FINAL_MOBILE_URLS"] = 14] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 15;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["TRACKING_URL"] = 15] = "TRACKING_URL";
    /**
     * Data Type: STRING. Android app link. Must be formatted as:
     * android-app://{package_id}/{scheme}/{host_path}.
     * The components are defined as follows:
     * package_id: app ID as specified in Google Play.
     * scheme: the scheme to pass to the application. Can be HTTP, or a custom
     *   scheme.
     * host_path: identifies the specific content within your application.
     *
     * @generated from protobuf enum value: ANDROID_APP_LINK = 16;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["ANDROID_APP_LINK"] = 16] = "ANDROID_APP_LINK";
    /**
     * Data Type: STRING_LIST. List of recommended job IDs to show together with
     * this item.
     *
     * @generated from protobuf enum value: SIMILAR_JOB_IDS = 17;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["SIMILAR_JOB_IDS"] = 17] = "SIMILAR_JOB_IDS";
    /**
     * Data Type: STRING. iOS app link.
     *
     * @generated from protobuf enum value: IOS_APP_LINK = 18;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["IOS_APP_LINK"] = 18] = "IOS_APP_LINK";
    /**
     * Data Type: INT64. iOS app store ID.
     *
     * @generated from protobuf enum value: IOS_APP_STORE_ID = 19;
     */
    JobPlaceholderFieldEnum_JobPlaceholderField[JobPlaceholderFieldEnum_JobPlaceholderField["IOS_APP_STORE_ID"] = 19] = "IOS_APP_STORE_ID";
})(JobPlaceholderFieldEnum_JobPlaceholderField = exports.JobPlaceholderFieldEnum_JobPlaceholderField || (exports.JobPlaceholderFieldEnum_JobPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class JobPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.JobPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.JobPlaceholderFieldEnum
 */
exports.JobPlaceholderFieldEnum = new JobPlaceholderFieldEnum$Type();
