// google-ads-api-client/dist/protos/bindings/ads/enums/keyword_match_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordMatchTypeEnum = exports.KeywordMatchTypeEnum_KeywordMatchType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible Keyword match types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.KeywordMatchTypeEnum.KeywordMatchType
 */
var KeywordMatchTypeEnum_KeywordMatchType;
(function (KeywordMatchTypeEnum_KeywordMatchType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordMatchTypeEnum_KeywordMatchType[KeywordMatchTypeEnum_KeywordMatchType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordMatchTypeEnum_KeywordMatchType[KeywordMatchTypeEnum_KeywordMatchType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Exact match.
     *
     * @generated from protobuf enum value: EXACT = 2;
     */
    KeywordMatchTypeEnum_KeywordMatchType[KeywordMatchTypeEnum_KeywordMatchType["EXACT"] = 2] = "EXACT";
    /**
     * Phrase match.
     *
     * @generated from protobuf enum value: PHRASE = 3;
     */
    KeywordMatchTypeEnum_KeywordMatchType[KeywordMatchTypeEnum_KeywordMatchType["PHRASE"] = 3] = "PHRASE";
    /**
     * Broad match.
     *
     * @generated from protobuf enum value: BROAD = 4;
     */
    KeywordMatchTypeEnum_KeywordMatchType[KeywordMatchTypeEnum_KeywordMatchType["BROAD"] = 4] = "BROAD";
})(KeywordMatchTypeEnum_KeywordMatchType = exports.KeywordMatchTypeEnum_KeywordMatchType || (exports.KeywordMatchTypeEnum_KeywordMatchType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordMatchTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.KeywordMatchTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.KeywordMatchTypeEnum
 */
exports.KeywordMatchTypeEnum = new KeywordMatchTypeEnum$Type();
