// google-ads-api-client/dist/protos/bindings/ads/enums/keyword_plan_keyword_annotation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanKeywordAnnotationEnum = exports.KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates keyword plan annotations that can be requested.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.KeywordPlanKeywordAnnotationEnum.KeywordPlanKeywordAnnotation
 */
var KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation;
(function (KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation[KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation[KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Return the keyword concept and concept group data.
     *
     * @generated from protobuf enum value: KEYWORD_CONCEPT = 2;
     */
    KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation[KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation["KEYWORD_CONCEPT"] = 2] = "KEYWORD_CONCEPT";
})(KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation = exports.KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation || (exports.KeywordPlanKeywordAnnotationEnum_KeywordPlanKeywordAnnotation = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanKeywordAnnotationEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.KeywordPlanKeywordAnnotationEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.KeywordPlanKeywordAnnotationEnum
 */
exports.KeywordPlanKeywordAnnotationEnum = new KeywordPlanKeywordAnnotationEnum$Type();
