// google-ads-api-client/dist/protos/bindings/ads/enums/keyword_plan_network.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanNetworkEnum = exports.KeywordPlanNetworkEnum_KeywordPlanNetwork = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates keyword plan forecastable network types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.KeywordPlanNetworkEnum.KeywordPlanNetwork
 */
var KeywordPlanNetworkEnum_KeywordPlanNetwork;
(function (KeywordPlanNetworkEnum_KeywordPlanNetwork) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanNetworkEnum_KeywordPlanNetwork[KeywordPlanNetworkEnum_KeywordPlanNetwork["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanNetworkEnum_KeywordPlanNetwork[KeywordPlanNetworkEnum_KeywordPlanNetwork["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Google Search.
     *
     * @generated from protobuf enum value: GOOGLE_SEARCH = 2;
     */
    KeywordPlanNetworkEnum_KeywordPlanNetwork[KeywordPlanNetworkEnum_KeywordPlanNetwork["GOOGLE_SEARCH"] = 2] = "GOOGLE_SEARCH";
    /**
     * Google Search + Search partners.
     *
     * @generated from protobuf enum value: GOOGLE_SEARCH_AND_PARTNERS = 3;
     */
    KeywordPlanNetworkEnum_KeywordPlanNetwork[KeywordPlanNetworkEnum_KeywordPlanNetwork["GOOGLE_SEARCH_AND_PARTNERS"] = 3] = "GOOGLE_SEARCH_AND_PARTNERS";
})(KeywordPlanNetworkEnum_KeywordPlanNetwork = exports.KeywordPlanNetworkEnum_KeywordPlanNetwork || (exports.KeywordPlanNetworkEnum_KeywordPlanNetwork = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanNetworkEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.KeywordPlanNetworkEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.KeywordPlanNetworkEnum
 */
exports.KeywordPlanNetworkEnum = new KeywordPlanNetworkEnum$Type();
