// google-ads-api-client/dist/protos/bindings/ads/enums/keyword_plan_concept_group_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanConceptGroupTypeEnum = exports.KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates keyword plan concept group types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.KeywordPlanConceptGroupTypeEnum.KeywordPlanConceptGroupType
 */
var KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType;
(function (KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType) {
    /**
     * The concept group classification different from brand/non-brand.
     * This is a catch all bucket for all classifications that are none of the
     * below.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType[KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType[KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The concept group classification is based on BRAND.
     *
     * @generated from protobuf enum value: BRAND = 2;
     */
    KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType[KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType["BRAND"] = 2] = "BRAND";
    /**
     * The concept group classification based on BRAND, that didn't fit well
     * with the BRAND classifications. These are generally outliers and can have
     * very few keywords in this type of classification.
     *
     * @generated from protobuf enum value: OTHER_BRANDS = 3;
     */
    KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType[KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType["OTHER_BRANDS"] = 3] = "OTHER_BRANDS";
    /**
     * These concept group classification is not based on BRAND. This is
     * returned for generic keywords that don't have a brand association.
     *
     * @generated from protobuf enum value: NON_BRAND = 4;
     */
    KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType[KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType["NON_BRAND"] = 4] = "NON_BRAND";
})(KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType = exports.KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType || (exports.KeywordPlanConceptGroupTypeEnum_KeywordPlanConceptGroupType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanConceptGroupTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.KeywordPlanConceptGroupTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.KeywordPlanConceptGroupTypeEnum
 */
exports.KeywordPlanConceptGroupTypeEnum = new KeywordPlanConceptGroupTypeEnum$Type();
