// google-ads-api-client/dist/protos/bindings/ads/enums/keyword_plan_aggregate_metric_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanAggregateMetricTypeEnum = exports.KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Aggregate fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.KeywordPlanAggregateMetricTypeEnum.KeywordPlanAggregateMetricType
 */
var KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType;
(function (KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType[KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType[KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The device breakdown of aggregate search volume.
     *
     * @generated from protobuf enum value: DEVICE = 2;
     */
    KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType[KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType["DEVICE"] = 2] = "DEVICE";
})(KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType = exports.KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType || (exports.KeywordPlanAggregateMetricTypeEnum_KeywordPlanAggregateMetricType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanAggregateMetricTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.KeywordPlanAggregateMetricTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.KeywordPlanAggregateMetricTypeEnum
 */
exports.KeywordPlanAggregateMetricTypeEnum = new KeywordPlanAggregateMetricTypeEnum$Type();
