// google-ads-api-client/dist/protos/bindings/ads/enums/keyword_plan_forecast_interval.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanForecastIntervalEnum = exports.KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Forecast intervals.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.KeywordPlanForecastIntervalEnum.KeywordPlanForecastInterval
 */
var KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval;
(function (KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval[KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval[KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The next week date range for keyword plan. The next week is based
     * on the default locale of the user's account and is mostly SUN-SAT or
     * MON-SUN.
     * This can be different from next-7 days.
     *
     * @generated from protobuf enum value: NEXT_WEEK = 3;
     */
    KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval[KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval["NEXT_WEEK"] = 3] = "NEXT_WEEK";
    /**
     * The next month date range for keyword plan.
     *
     * @generated from protobuf enum value: NEXT_MONTH = 4;
     */
    KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval[KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval["NEXT_MONTH"] = 4] = "NEXT_MONTH";
    /**
     * The next quarter date range for keyword plan.
     *
     * @generated from protobuf enum value: NEXT_QUARTER = 5;
     */
    KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval[KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval["NEXT_QUARTER"] = 5] = "NEXT_QUARTER";
})(KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval = exports.KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval || (exports.KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanForecastIntervalEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.KeywordPlanForecastIntervalEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.KeywordPlanForecastIntervalEnum
 */
exports.KeywordPlanForecastIntervalEnum = new KeywordPlanForecastIntervalEnum$Type();
