// google-ads-api-client/dist/protos/bindings/ads/enums/keyword_plan_competition_level.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanCompetitionLevelEnum = exports.KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Competition level of a keyword.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.KeywordPlanCompetitionLevelEnum.KeywordPlanCompetitionLevel
 */
var KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel;
(function (KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel[KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel[KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Low competition. The Competition Index range for this is [0, 33].
     *
     * @generated from protobuf enum value: LOW = 2;
     */
    KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel[KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel["LOW"] = 2] = "LOW";
    /**
     * Medium competition. The Competition Index range for this is [34, 66].
     *
     * @generated from protobuf enum value: MEDIUM = 3;
     */
    KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel[KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel["MEDIUM"] = 3] = "MEDIUM";
    /**
     * High competition. The Competition Index range for this is [67, 100].
     *
     * @generated from protobuf enum value: HIGH = 4;
     */
    KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel[KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel["HIGH"] = 4] = "HIGH";
})(KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel = exports.KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel || (exports.KeywordPlanCompetitionLevelEnum_KeywordPlanCompetitionLevel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanCompetitionLevelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.KeywordPlanCompetitionLevelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.KeywordPlanCompetitionLevelEnum
 */
exports.KeywordPlanCompetitionLevelEnum = new KeywordPlanCompetitionLevelEnum$Type();
