// google-ads-api-client/dist/protos/bindings/ads/enums/label_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabelStatusEnum = exports.LabelStatusEnum_LabelStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a label.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LabelStatusEnum.LabelStatus
 */
var LabelStatusEnum_LabelStatus;
(function (LabelStatusEnum_LabelStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LabelStatusEnum_LabelStatus[LabelStatusEnum_LabelStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LabelStatusEnum_LabelStatus[LabelStatusEnum_LabelStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Label is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    LabelStatusEnum_LabelStatus[LabelStatusEnum_LabelStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * Label is removed.
     *
     * @generated from protobuf enum value: REMOVED = 3;
     */
    LabelStatusEnum_LabelStatus[LabelStatusEnum_LabelStatus["REMOVED"] = 3] = "REMOVED";
})(LabelStatusEnum_LabelStatus = exports.LabelStatusEnum_LabelStatus || (exports.LabelStatusEnum_LabelStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LabelStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LabelStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LabelStatusEnum
 */
exports.LabelStatusEnum = new LabelStatusEnum$Type();
