// google-ads-api-client/dist/protos/bindings/ads/enums/lead_form_field_user_input_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LeadFormFieldUserInputTypeEnum = exports.LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the input type of a lead form field.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum.LeadFormFieldUserInputType
 */
var LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType;
(function (LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The user will be asked to fill in their given and family name. This field
     * cannot be set at the same time as GIVEN_NAME or FAMILY_NAME.
     *
     * @generated from protobuf enum value: FULL_NAME = 2;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["FULL_NAME"] = 2] = "FULL_NAME";
    /**
     * The user will be asked to fill in their email address.
     *
     * @generated from protobuf enum value: EMAIL = 3;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["EMAIL"] = 3] = "EMAIL";
    /**
     * The user will be asked to fill in their phone number.
     *
     * @generated from protobuf enum value: PHONE_NUMBER = 4;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PHONE_NUMBER"] = 4] = "PHONE_NUMBER";
    /**
     * The user will be asked to fill in their zip code.
     *
     * @generated from protobuf enum value: POSTAL_CODE = 5;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["POSTAL_CODE"] = 5] = "POSTAL_CODE";
    /**
     * The user will be asked to fill in their street address.
     *
     * @generated from protobuf enum value: STREET_ADDRESS = 8;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["STREET_ADDRESS"] = 8] = "STREET_ADDRESS";
    /**
     * The user will be asked to fill in their city.
     *
     * @generated from protobuf enum value: CITY = 9;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["CITY"] = 9] = "CITY";
    /**
     * The user will be asked to fill in their region part of the address (for
     * example, state for US, province for Canada).
     *
     * @generated from protobuf enum value: REGION = 10;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["REGION"] = 10] = "REGION";
    /**
     * The user will be asked to fill in their country.
     *
     * @generated from protobuf enum value: COUNTRY = 11;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["COUNTRY"] = 11] = "COUNTRY";
    /**
     * The user will be asked to fill in their work email address.
     *
     * @generated from protobuf enum value: WORK_EMAIL = 12;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["WORK_EMAIL"] = 12] = "WORK_EMAIL";
    /**
     * The user will be asked to fill in their company name.
     *
     * @generated from protobuf enum value: COMPANY_NAME = 13;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["COMPANY_NAME"] = 13] = "COMPANY_NAME";
    /**
     * The user will be asked to fill in their work phone.
     *
     * @generated from protobuf enum value: WORK_PHONE = 14;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["WORK_PHONE"] = 14] = "WORK_PHONE";
    /**
     * The user will be asked to fill in their job title.
     *
     * @generated from protobuf enum value: JOB_TITLE = 15;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["JOB_TITLE"] = 15] = "JOB_TITLE";
    /**
     * The user will be asked to fill in their CPF for Brazil users.
     *
     * @generated from protobuf enum value: GOVERNMENT_ISSUED_ID_CPF_BR = 16;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["GOVERNMENT_ISSUED_ID_CPF_BR"] = 16] = "GOVERNMENT_ISSUED_ID_CPF_BR";
    /**
     * The user will be asked to fill in their DNI for Argentina users.
     *
     * @generated from protobuf enum value: GOVERNMENT_ISSUED_ID_DNI_AR = 17;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["GOVERNMENT_ISSUED_ID_DNI_AR"] = 17] = "GOVERNMENT_ISSUED_ID_DNI_AR";
    /**
     * The user will be asked to fill in their DNI for Peru users.
     *
     * @generated from protobuf enum value: GOVERNMENT_ISSUED_ID_DNI_PE = 18;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["GOVERNMENT_ISSUED_ID_DNI_PE"] = 18] = "GOVERNMENT_ISSUED_ID_DNI_PE";
    /**
     * The user will be asked to fill in their RUT for Chile users.
     *
     * @generated from protobuf enum value: GOVERNMENT_ISSUED_ID_RUT_CL = 19;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["GOVERNMENT_ISSUED_ID_RUT_CL"] = 19] = "GOVERNMENT_ISSUED_ID_RUT_CL";
    /**
     * The user will be asked to fill in their CC for Colombia users.
     *
     * @generated from protobuf enum value: GOVERNMENT_ISSUED_ID_CC_CO = 20;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["GOVERNMENT_ISSUED_ID_CC_CO"] = 20] = "GOVERNMENT_ISSUED_ID_CC_CO";
    /**
     * The user will be asked to fill in their CI for Ecuador users.
     *
     * @generated from protobuf enum value: GOVERNMENT_ISSUED_ID_CI_EC = 21;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["GOVERNMENT_ISSUED_ID_CI_EC"] = 21] = "GOVERNMENT_ISSUED_ID_CI_EC";
    /**
     * The user will be asked to fill in their RFC for Mexico users.
     *
     * @generated from protobuf enum value: GOVERNMENT_ISSUED_ID_RFC_MX = 22;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["GOVERNMENT_ISSUED_ID_RFC_MX"] = 22] = "GOVERNMENT_ISSUED_ID_RFC_MX";
    /**
     * The user will be asked to fill in their first name. This
     * field can not be set at the same time as FULL_NAME.
     *
     * @generated from protobuf enum value: FIRST_NAME = 23;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["FIRST_NAME"] = 23] = "FIRST_NAME";
    /**
     * The user will be asked to fill in their last name. This
     * field can not be set at the same time as FULL_NAME.
     *
     * @generated from protobuf enum value: LAST_NAME = 24;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["LAST_NAME"] = 24] = "LAST_NAME";
    /**
     * Question: "Which model are you interested in?"
     * Category: "Auto"
     *
     * @generated from protobuf enum value: VEHICLE_MODEL = 1001;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["VEHICLE_MODEL"] = 1001] = "VEHICLE_MODEL";
    /**
     * Question: "Which type of vehicle are you interested in?"
     * Category: "Auto"
     *
     * @generated from protobuf enum value: VEHICLE_TYPE = 1002;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["VEHICLE_TYPE"] = 1002] = "VEHICLE_TYPE";
    /**
     * Question: "What is your preferred dealership?"
     * Category: "Auto"
     *
     * @generated from protobuf enum value: PREFERRED_DEALERSHIP = 1003;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PREFERRED_DEALERSHIP"] = 1003] = "PREFERRED_DEALERSHIP";
    /**
     * Question: "When do you plan on purchasing a vehicle?"
     * Category: "Auto"
     *
     * @generated from protobuf enum value: VEHICLE_PURCHASE_TIMELINE = 1004;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["VEHICLE_PURCHASE_TIMELINE"] = 1004] = "VEHICLE_PURCHASE_TIMELINE";
    /**
     * Question: "Do you own a vehicle?"
     * Category: "Auto"
     *
     * @generated from protobuf enum value: VEHICLE_OWNERSHIP = 1005;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["VEHICLE_OWNERSHIP"] = 1005] = "VEHICLE_OWNERSHIP";
    /**
     * Question: "What vehicle ownership option are you interested in?"
     * Category: "Auto"
     *
     * @generated from protobuf enum value: VEHICLE_PAYMENT_TYPE = 1009;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["VEHICLE_PAYMENT_TYPE"] = 1009] = "VEHICLE_PAYMENT_TYPE";
    /**
     * Question: "What type of vehicle condition are you interested in?"
     * Category: "Auto"
     *
     * @generated from protobuf enum value: VEHICLE_CONDITION = 1010;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["VEHICLE_CONDITION"] = 1010] = "VEHICLE_CONDITION";
    /**
     * Question: "What size is your company?"
     * Category: "Business"
     *
     * @generated from protobuf enum value: COMPANY_SIZE = 1006;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["COMPANY_SIZE"] = 1006] = "COMPANY_SIZE";
    /**
     * Question: "What is your annual sales volume?"
     * Category: "Business"
     *
     * @generated from protobuf enum value: ANNUAL_SALES = 1007;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["ANNUAL_SALES"] = 1007] = "ANNUAL_SALES";
    /**
     * Question: "How many years have you been in business?"
     * Category: "Business"
     *
     * @generated from protobuf enum value: YEARS_IN_BUSINESS = 1008;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["YEARS_IN_BUSINESS"] = 1008] = "YEARS_IN_BUSINESS";
    /**
     * Question: "What is your job department?"
     * Category: "Business"
     *
     * @generated from protobuf enum value: JOB_DEPARTMENT = 1011;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["JOB_DEPARTMENT"] = 1011] = "JOB_DEPARTMENT";
    /**
     * Question: "What is your job role?"
     * Category: "Business"
     *
     * @generated from protobuf enum value: JOB_ROLE = 1012;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["JOB_ROLE"] = 1012] = "JOB_ROLE";
    /**
     * Question: "Are you over 18 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_18_AGE = 1078;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_18_AGE"] = 1078] = "OVER_18_AGE";
    /**
     * Question: "Are you over 19 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_19_AGE = 1079;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_19_AGE"] = 1079] = "OVER_19_AGE";
    /**
     * Question: "Are you over 20 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_20_AGE = 1080;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_20_AGE"] = 1080] = "OVER_20_AGE";
    /**
     * Question: "Are you over 21 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_21_AGE = 1081;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_21_AGE"] = 1081] = "OVER_21_AGE";
    /**
     * Question: "Are you over 22 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_22_AGE = 1082;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_22_AGE"] = 1082] = "OVER_22_AGE";
    /**
     * Question: "Are you over 23 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_23_AGE = 1083;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_23_AGE"] = 1083] = "OVER_23_AGE";
    /**
     * Question: "Are you over 24 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_24_AGE = 1084;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_24_AGE"] = 1084] = "OVER_24_AGE";
    /**
     * Question: "Are you over 25 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_25_AGE = 1085;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_25_AGE"] = 1085] = "OVER_25_AGE";
    /**
     * Question: "Are you over 26 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_26_AGE = 1086;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_26_AGE"] = 1086] = "OVER_26_AGE";
    /**
     * Question: "Are you over 27 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_27_AGE = 1087;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_27_AGE"] = 1087] = "OVER_27_AGE";
    /**
     * Question: "Are you over 28 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_28_AGE = 1088;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_28_AGE"] = 1088] = "OVER_28_AGE";
    /**
     * Question: "Are you over 29 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_29_AGE = 1089;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_29_AGE"] = 1089] = "OVER_29_AGE";
    /**
     * Question: "Are you over 30 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_30_AGE = 1090;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_30_AGE"] = 1090] = "OVER_30_AGE";
    /**
     * Question: "Are you over 31 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_31_AGE = 1091;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_31_AGE"] = 1091] = "OVER_31_AGE";
    /**
     * Question: "Are you over 32 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_32_AGE = 1092;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_32_AGE"] = 1092] = "OVER_32_AGE";
    /**
     * Question: "Are you over 33 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_33_AGE = 1093;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_33_AGE"] = 1093] = "OVER_33_AGE";
    /**
     * Question: "Are you over 34 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_34_AGE = 1094;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_34_AGE"] = 1094] = "OVER_34_AGE";
    /**
     * Question: "Are you over 35 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_35_AGE = 1095;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_35_AGE"] = 1095] = "OVER_35_AGE";
    /**
     * Question: "Are you over 36 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_36_AGE = 1096;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_36_AGE"] = 1096] = "OVER_36_AGE";
    /**
     * Question: "Are you over 37 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_37_AGE = 1097;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_37_AGE"] = 1097] = "OVER_37_AGE";
    /**
     * Question: "Are you over 38 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_38_AGE = 1098;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_38_AGE"] = 1098] = "OVER_38_AGE";
    /**
     * Question: "Are you over 39 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_39_AGE = 1099;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_39_AGE"] = 1099] = "OVER_39_AGE";
    /**
     * Question: "Are you over 40 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_40_AGE = 1100;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_40_AGE"] = 1100] = "OVER_40_AGE";
    /**
     * Question: "Are you over 41 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_41_AGE = 1101;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_41_AGE"] = 1101] = "OVER_41_AGE";
    /**
     * Question: "Are you over 42 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_42_AGE = 1102;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_42_AGE"] = 1102] = "OVER_42_AGE";
    /**
     * Question: "Are you over 43 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_43_AGE = 1103;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_43_AGE"] = 1103] = "OVER_43_AGE";
    /**
     * Question: "Are you over 44 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_44_AGE = 1104;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_44_AGE"] = 1104] = "OVER_44_AGE";
    /**
     * Question: "Are you over 45 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_45_AGE = 1105;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_45_AGE"] = 1105] = "OVER_45_AGE";
    /**
     * Question: "Are you over 46 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_46_AGE = 1106;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_46_AGE"] = 1106] = "OVER_46_AGE";
    /**
     * Question: "Are you over 47 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_47_AGE = 1107;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_47_AGE"] = 1107] = "OVER_47_AGE";
    /**
     * Question: "Are you over 48 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_48_AGE = 1108;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_48_AGE"] = 1108] = "OVER_48_AGE";
    /**
     * Question: "Are you over 49 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_49_AGE = 1109;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_49_AGE"] = 1109] = "OVER_49_AGE";
    /**
     * Question: "Are you over 50 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_50_AGE = 1110;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_50_AGE"] = 1110] = "OVER_50_AGE";
    /**
     * Question: "Are you over 51 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_51_AGE = 1111;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_51_AGE"] = 1111] = "OVER_51_AGE";
    /**
     * Question: "Are you over 52 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_52_AGE = 1112;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_52_AGE"] = 1112] = "OVER_52_AGE";
    /**
     * Question: "Are you over 53 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_53_AGE = 1113;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_53_AGE"] = 1113] = "OVER_53_AGE";
    /**
     * Question: "Are you over 54 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_54_AGE = 1114;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_54_AGE"] = 1114] = "OVER_54_AGE";
    /**
     * Question: "Are you over 55 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_55_AGE = 1115;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_55_AGE"] = 1115] = "OVER_55_AGE";
    /**
     * Question: "Are you over 56 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_56_AGE = 1116;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_56_AGE"] = 1116] = "OVER_56_AGE";
    /**
     * Question: "Are you over 57 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_57_AGE = 1117;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_57_AGE"] = 1117] = "OVER_57_AGE";
    /**
     * Question: "Are you over 58 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_58_AGE = 1118;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_58_AGE"] = 1118] = "OVER_58_AGE";
    /**
     * Question: "Are you over 59 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_59_AGE = 1119;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_59_AGE"] = 1119] = "OVER_59_AGE";
    /**
     * Question: "Are you over 60 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_60_AGE = 1120;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_60_AGE"] = 1120] = "OVER_60_AGE";
    /**
     * Question: "Are you over 61 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_61_AGE = 1121;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_61_AGE"] = 1121] = "OVER_61_AGE";
    /**
     * Question: "Are you over 62 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_62_AGE = 1122;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_62_AGE"] = 1122] = "OVER_62_AGE";
    /**
     * Question: "Are you over 63 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_63_AGE = 1123;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_63_AGE"] = 1123] = "OVER_63_AGE";
    /**
     * Question: "Are you over 64 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_64_AGE = 1124;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_64_AGE"] = 1124] = "OVER_64_AGE";
    /**
     * Question: "Are you over 65 years of age?"
     * Category: "Demographics"
     *
     * @generated from protobuf enum value: OVER_65_AGE = 1125;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OVER_65_AGE"] = 1125] = "OVER_65_AGE";
    /**
     * Question: "Which program are you interested in?"
     * Category: "Education"
     *
     * @generated from protobuf enum value: EDUCATION_PROGRAM = 1013;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["EDUCATION_PROGRAM"] = 1013] = "EDUCATION_PROGRAM";
    /**
     * Question: "Which course are you interested in?"
     * Category: "Education"
     *
     * @generated from protobuf enum value: EDUCATION_COURSE = 1014;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["EDUCATION_COURSE"] = 1014] = "EDUCATION_COURSE";
    /**
     * Question: "Which product are you interested in?"
     * Category: "General"
     *
     * @generated from protobuf enum value: PRODUCT = 1016;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PRODUCT"] = 1016] = "PRODUCT";
    /**
     * Question: "Which service are you interested in?"
     * Category: "General"
     *
     * @generated from protobuf enum value: SERVICE = 1017;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["SERVICE"] = 1017] = "SERVICE";
    /**
     * Question: "Which offer are you interested in?"
     * Category: "General"
     *
     * @generated from protobuf enum value: OFFER = 1018;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["OFFER"] = 1018] = "OFFER";
    /**
     * Question: "Which category are you interested in?"
     * Category: "General"
     *
     * @generated from protobuf enum value: CATEGORY = 1019;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["CATEGORY"] = 1019] = "CATEGORY";
    /**
     * Question: "What is your preferred method of contact?"
     * Category: "General"
     *
     * @generated from protobuf enum value: PREFERRED_CONTACT_METHOD = 1020;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PREFERRED_CONTACT_METHOD"] = 1020] = "PREFERRED_CONTACT_METHOD";
    /**
     * Question: "What is your preferred location?"
     * Category: "General"
     *
     * @generated from protobuf enum value: PREFERRED_LOCATION = 1021;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PREFERRED_LOCATION"] = 1021] = "PREFERRED_LOCATION";
    /**
     * Question: "What is the best time to contact you?"
     * Category: "General"
     *
     * @generated from protobuf enum value: PREFERRED_CONTACT_TIME = 1022;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PREFERRED_CONTACT_TIME"] = 1022] = "PREFERRED_CONTACT_TIME";
    /**
     * Question: "When are you looking to make a purchase?"
     * Category: "General"
     *
     * @generated from protobuf enum value: PURCHASE_TIMELINE = 1023;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PURCHASE_TIMELINE"] = 1023] = "PURCHASE_TIMELINE";
    /**
     * Question: "How many years of work experience do you have?"
     * Category: "Jobs"
     *
     * @generated from protobuf enum value: YEARS_OF_EXPERIENCE = 1048;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["YEARS_OF_EXPERIENCE"] = 1048] = "YEARS_OF_EXPERIENCE";
    /**
     * Question: "What industry do you work in?"
     * Category: "Jobs"
     *
     * @generated from protobuf enum value: JOB_INDUSTRY = 1049;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["JOB_INDUSTRY"] = 1049] = "JOB_INDUSTRY";
    /**
     * Question: "What is your highest level of education?"
     * Category: "Jobs"
     *
     * @generated from protobuf enum value: LEVEL_OF_EDUCATION = 1050;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["LEVEL_OF_EDUCATION"] = 1050] = "LEVEL_OF_EDUCATION";
    /**
     * Question: "What type of property are you looking for?"
     * Category: "Real Estate"
     *
     * @generated from protobuf enum value: PROPERTY_TYPE = 1024;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PROPERTY_TYPE"] = 1024] = "PROPERTY_TYPE";
    /**
     * Question: "What do you need a realtor's help with?"
     * Category: "Real Estate"
     *
     * @generated from protobuf enum value: REALTOR_HELP_GOAL = 1025;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["REALTOR_HELP_GOAL"] = 1025] = "REALTOR_HELP_GOAL";
    /**
     * Question: "What neighborhood are you interested in?"
     * Category: "Real Estate"
     *
     * @generated from protobuf enum value: PROPERTY_COMMUNITY = 1026;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PROPERTY_COMMUNITY"] = 1026] = "PROPERTY_COMMUNITY";
    /**
     * Question: "What price range are you looking for?"
     * Category: "Real Estate"
     *
     * @generated from protobuf enum value: PRICE_RANGE = 1027;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PRICE_RANGE"] = 1027] = "PRICE_RANGE";
    /**
     * Question: "How many bedrooms are you looking for?"
     * Category: "Real Estate"
     *
     * @generated from protobuf enum value: NUMBER_OF_BEDROOMS = 1028;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["NUMBER_OF_BEDROOMS"] = 1028] = "NUMBER_OF_BEDROOMS";
    /**
     * Question: "Are you looking for a fully furnished property?"
     * Category: "Real Estate"
     *
     * @generated from protobuf enum value: FURNISHED_PROPERTY = 1029;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["FURNISHED_PROPERTY"] = 1029] = "FURNISHED_PROPERTY";
    /**
     * Question: "Are you looking for properties that allow pets?"
     * Category: "Real Estate"
     *
     * @generated from protobuf enum value: PETS_ALLOWED_PROPERTY = 1030;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PETS_ALLOWED_PROPERTY"] = 1030] = "PETS_ALLOWED_PROPERTY";
    /**
     * Question: "What is the next product you plan to purchase?"
     * Category: "Retail"
     *
     * @generated from protobuf enum value: NEXT_PLANNED_PURCHASE = 1031;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["NEXT_PLANNED_PURCHASE"] = 1031] = "NEXT_PLANNED_PURCHASE";
    /**
     * Question: "Would you like to sign up for an event?"
     * Category: "Retail"
     *
     * @generated from protobuf enum value: EVENT_SIGNUP_INTEREST = 1033;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["EVENT_SIGNUP_INTEREST"] = 1033] = "EVENT_SIGNUP_INTEREST";
    /**
     * Question: "Where are you interested in shopping?"
     * Category: "Retail"
     *
     * @generated from protobuf enum value: PREFERRED_SHOPPING_PLACES = 1034;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["PREFERRED_SHOPPING_PLACES"] = 1034] = "PREFERRED_SHOPPING_PLACES";
    /**
     * Question: "What is your favorite brand?"
     * Category: "Retail"
     *
     * @generated from protobuf enum value: FAVORITE_BRAND = 1035;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["FAVORITE_BRAND"] = 1035] = "FAVORITE_BRAND";
    /**
     * Question: "Which type of valid commercial license do you have?"
     * Category: "Transportation"
     *
     * @generated from protobuf enum value: TRANSPORTATION_COMMERCIAL_LICENSE_TYPE = 1036;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["TRANSPORTATION_COMMERCIAL_LICENSE_TYPE"] = 1036] = "TRANSPORTATION_COMMERCIAL_LICENSE_TYPE";
    /**
     * Question: "Interested in booking an event?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: EVENT_BOOKING_INTEREST = 1038;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["EVENT_BOOKING_INTEREST"] = 1038] = "EVENT_BOOKING_INTEREST";
    /**
     * Question: "What is your destination country?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: DESTINATION_COUNTRY = 1039;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["DESTINATION_COUNTRY"] = 1039] = "DESTINATION_COUNTRY";
    /**
     * Question: "What is your destination city?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: DESTINATION_CITY = 1040;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["DESTINATION_CITY"] = 1040] = "DESTINATION_CITY";
    /**
     * Question: "What is your departure country?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: DEPARTURE_COUNTRY = 1041;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["DEPARTURE_COUNTRY"] = 1041] = "DEPARTURE_COUNTRY";
    /**
     * Question: "What is your departure city?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: DEPARTURE_CITY = 1042;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["DEPARTURE_CITY"] = 1042] = "DEPARTURE_CITY";
    /**
     * Question: "What is your departure date?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: DEPARTURE_DATE = 1043;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["DEPARTURE_DATE"] = 1043] = "DEPARTURE_DATE";
    /**
     * Question: "What is your return date?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: RETURN_DATE = 1044;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["RETURN_DATE"] = 1044] = "RETURN_DATE";
    /**
     * Question: "How many people are you traveling with?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: NUMBER_OF_TRAVELERS = 1045;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["NUMBER_OF_TRAVELERS"] = 1045] = "NUMBER_OF_TRAVELERS";
    /**
     * Question: "What is your travel budget?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: TRAVEL_BUDGET = 1046;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["TRAVEL_BUDGET"] = 1046] = "TRAVEL_BUDGET";
    /**
     * Question: "Where do you want to stay during your travel?"
     * Category: "Travel"
     *
     * @generated from protobuf enum value: TRAVEL_ACCOMMODATION = 1047;
     */
    LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType[LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType["TRAVEL_ACCOMMODATION"] = 1047] = "TRAVEL_ACCOMMODATION";
})(LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType = exports.LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType || (exports.LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormFieldUserInputTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum
 */
exports.LeadFormFieldUserInputTypeEnum = new LeadFormFieldUserInputTypeEnum$Type();
