// google-ads-api-client/dist/protos/bindings/ads/enums/lead_form_desired_intent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LeadFormDesiredIntentEnum = exports.LeadFormDesiredIntentEnum_LeadFormDesiredIntent = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the chosen level of intent of generated leads.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LeadFormDesiredIntentEnum.LeadFormDesiredIntent
 */
var LeadFormDesiredIntentEnum_LeadFormDesiredIntent;
(function (LeadFormDesiredIntentEnum_LeadFormDesiredIntent) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LeadFormDesiredIntentEnum_LeadFormDesiredIntent[LeadFormDesiredIntentEnum_LeadFormDesiredIntent["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LeadFormDesiredIntentEnum_LeadFormDesiredIntent[LeadFormDesiredIntentEnum_LeadFormDesiredIntent["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Deliver more leads at a potentially lower quality.
     *
     * @generated from protobuf enum value: LOW_INTENT = 2;
     */
    LeadFormDesiredIntentEnum_LeadFormDesiredIntent[LeadFormDesiredIntentEnum_LeadFormDesiredIntent["LOW_INTENT"] = 2] = "LOW_INTENT";
    /**
     * Deliver leads that are more qualified.
     *
     * @generated from protobuf enum value: HIGH_INTENT = 3;
     */
    LeadFormDesiredIntentEnum_LeadFormDesiredIntent[LeadFormDesiredIntentEnum_LeadFormDesiredIntent["HIGH_INTENT"] = 3] = "HIGH_INTENT";
})(LeadFormDesiredIntentEnum_LeadFormDesiredIntent = exports.LeadFormDesiredIntentEnum_LeadFormDesiredIntent || (exports.LeadFormDesiredIntentEnum_LeadFormDesiredIntent = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormDesiredIntentEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LeadFormDesiredIntentEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LeadFormDesiredIntentEnum
 */
exports.LeadFormDesiredIntentEnum = new LeadFormDesiredIntentEnum$Type();
