// google-ads-api-client/dist/protos/bindings/ads/enums/lead_form_call_to_action_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LeadFormCallToActionTypeEnum = exports.LeadFormCallToActionTypeEnum_LeadFormCallToActionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the type of call-to-action phrases in a lead form.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LeadFormCallToActionTypeEnum.LeadFormCallToActionType
 */
var LeadFormCallToActionTypeEnum_LeadFormCallToActionType;
(function (LeadFormCallToActionTypeEnum_LeadFormCallToActionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Learn more.
     *
     * @generated from protobuf enum value: LEARN_MORE = 2;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["LEARN_MORE"] = 2] = "LEARN_MORE";
    /**
     * Get quote.
     *
     * @generated from protobuf enum value: GET_QUOTE = 3;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["GET_QUOTE"] = 3] = "GET_QUOTE";
    /**
     * Apply now.
     *
     * @generated from protobuf enum value: APPLY_NOW = 4;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["APPLY_NOW"] = 4] = "APPLY_NOW";
    /**
     * Sign Up.
     *
     * @generated from protobuf enum value: SIGN_UP = 5;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["SIGN_UP"] = 5] = "SIGN_UP";
    /**
     * Contact us.
     *
     * @generated from protobuf enum value: CONTACT_US = 6;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["CONTACT_US"] = 6] = "CONTACT_US";
    /**
     * Subscribe.
     *
     * @generated from protobuf enum value: SUBSCRIBE = 7;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["SUBSCRIBE"] = 7] = "SUBSCRIBE";
    /**
     * Download.
     *
     * @generated from protobuf enum value: DOWNLOAD = 8;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["DOWNLOAD"] = 8] = "DOWNLOAD";
    /**
     * Book now.
     *
     * @generated from protobuf enum value: BOOK_NOW = 9;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["BOOK_NOW"] = 9] = "BOOK_NOW";
    /**
     * Get offer.
     *
     * @generated from protobuf enum value: GET_OFFER = 10;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["GET_OFFER"] = 10] = "GET_OFFER";
    /**
     * Register.
     *
     * @generated from protobuf enum value: REGISTER = 11;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["REGISTER"] = 11] = "REGISTER";
    /**
     * Get info.
     *
     * @generated from protobuf enum value: GET_INFO = 12;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["GET_INFO"] = 12] = "GET_INFO";
    /**
     * Request a demo.
     *
     * @generated from protobuf enum value: REQUEST_DEMO = 13;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["REQUEST_DEMO"] = 13] = "REQUEST_DEMO";
    /**
     * Join now.
     *
     * @generated from protobuf enum value: JOIN_NOW = 14;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["JOIN_NOW"] = 14] = "JOIN_NOW";
    /**
     * Get started.
     *
     * @generated from protobuf enum value: GET_STARTED = 15;
     */
    LeadFormCallToActionTypeEnum_LeadFormCallToActionType[LeadFormCallToActionTypeEnum_LeadFormCallToActionType["GET_STARTED"] = 15] = "GET_STARTED";
})(LeadFormCallToActionTypeEnum_LeadFormCallToActionType = exports.LeadFormCallToActionTypeEnum_LeadFormCallToActionType || (exports.LeadFormCallToActionTypeEnum_LeadFormCallToActionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormCallToActionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LeadFormCallToActionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LeadFormCallToActionTypeEnum
 */
exports.LeadFormCallToActionTypeEnum = new LeadFormCallToActionTypeEnum$Type();
