// google-ads-api-client/dist/protos/bindings/ads/enums/lead_form_post_submit_call_to_action_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LeadFormPostSubmitCallToActionTypeEnum = exports.LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the type of post-submit call-to-action phrases for a lead
 * form.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LeadFormPostSubmitCallToActionTypeEnum.LeadFormPostSubmitCallToActionType
 */
var LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType;
(function (LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType[LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType[LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Visit site.
     *
     * @generated from protobuf enum value: VISIT_SITE = 2;
     */
    LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType[LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType["VISIT_SITE"] = 2] = "VISIT_SITE";
    /**
     * Download.
     *
     * @generated from protobuf enum value: DOWNLOAD = 3;
     */
    LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType[LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType["DOWNLOAD"] = 3] = "DOWNLOAD";
    /**
     * Learn more.
     *
     * @generated from protobuf enum value: LEARN_MORE = 4;
     */
    LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType[LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType["LEARN_MORE"] = 4] = "LEARN_MORE";
    /**
     * Shop now.
     *
     * @generated from protobuf enum value: SHOP_NOW = 5;
     */
    LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType[LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType["SHOP_NOW"] = 5] = "SHOP_NOW";
})(LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType = exports.LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType || (exports.LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormPostSubmitCallToActionTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LeadFormPostSubmitCallToActionTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LeadFormPostSubmitCallToActionTypeEnum
 */
exports.LeadFormPostSubmitCallToActionTypeEnum = new LeadFormPostSubmitCallToActionTypeEnum$Type();
