// google-ads-api-client/dist/protos/bindings/ads/enums/legacy_app_install_ad_app_store.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegacyAppInstallAdAppStoreEnum = exports.LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * App store type in a legacy app install ad.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LegacyAppInstallAdAppStoreEnum.LegacyAppInstallAdAppStore
 */
var LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore;
(function (LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore[LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore[LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Apple iTunes.
     *
     * @generated from protobuf enum value: APPLE_APP_STORE = 2;
     */
    LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore[LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore["APPLE_APP_STORE"] = 2] = "APPLE_APP_STORE";
    /**
     * Google Play.
     *
     * @generated from protobuf enum value: GOOGLE_PLAY = 3;
     */
    LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore[LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore["GOOGLE_PLAY"] = 3] = "GOOGLE_PLAY";
    /**
     * Windows Store.
     *
     * @generated from protobuf enum value: WINDOWS_STORE = 4;
     */
    LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore[LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore["WINDOWS_STORE"] = 4] = "WINDOWS_STORE";
    /**
     * Windows Phone Store.
     *
     * @generated from protobuf enum value: WINDOWS_PHONE_STORE = 5;
     */
    LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore[LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore["WINDOWS_PHONE_STORE"] = 5] = "WINDOWS_PHONE_STORE";
    /**
     * The app is hosted in a Chinese app store.
     *
     * @generated from protobuf enum value: CN_APP_STORE = 6;
     */
    LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore[LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore["CN_APP_STORE"] = 6] = "CN_APP_STORE";
})(LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore = exports.LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore || (exports.LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LegacyAppInstallAdAppStoreEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LegacyAppInstallAdAppStoreEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LegacyAppInstallAdAppStoreEnum
 */
exports.LegacyAppInstallAdAppStoreEnum = new LegacyAppInstallAdAppStoreEnum$Type();
