// google-ads-api-client/dist/protos/bindings/ads/enums/linked_account_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkedAccountTypeEnum = exports.LinkedAccountTypeEnum_LinkedAccountType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Describes the possible link types between a Google Ads customer
 * and another account.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LinkedAccountTypeEnum.LinkedAccountType
 */
var LinkedAccountTypeEnum_LinkedAccountType;
(function (LinkedAccountTypeEnum_LinkedAccountType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LinkedAccountTypeEnum_LinkedAccountType[LinkedAccountTypeEnum_LinkedAccountType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LinkedAccountTypeEnum_LinkedAccountType[LinkedAccountTypeEnum_LinkedAccountType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A link to provide third party app analytics data.
     *
     * @generated from protobuf enum value: THIRD_PARTY_APP_ANALYTICS = 2;
     */
    LinkedAccountTypeEnum_LinkedAccountType[LinkedAccountTypeEnum_LinkedAccountType["THIRD_PARTY_APP_ANALYTICS"] = 2] = "THIRD_PARTY_APP_ANALYTICS";
    /**
     * A link to Data partner.
     *
     * @generated from protobuf enum value: DATA_PARTNER = 3;
     */
    LinkedAccountTypeEnum_LinkedAccountType[LinkedAccountTypeEnum_LinkedAccountType["DATA_PARTNER"] = 3] = "DATA_PARTNER";
    /**
     * A link to Google Ads.
     *
     * @generated from protobuf enum value: GOOGLE_ADS = 4;
     */
    LinkedAccountTypeEnum_LinkedAccountType[LinkedAccountTypeEnum_LinkedAccountType["GOOGLE_ADS"] = 4] = "GOOGLE_ADS";
    /**
     * A link to Hotel Center.
     *
     * @generated from protobuf enum value: HOTEL_CENTER = 5;
     */
    LinkedAccountTypeEnum_LinkedAccountType[LinkedAccountTypeEnum_LinkedAccountType["HOTEL_CENTER"] = 5] = "HOTEL_CENTER";
})(LinkedAccountTypeEnum_LinkedAccountType = exports.LinkedAccountTypeEnum_LinkedAccountType || (exports.LinkedAccountTypeEnum_LinkedAccountType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LinkedAccountTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LinkedAccountTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LinkedAccountTypeEnum
 */
exports.LinkedAccountTypeEnum = new LinkedAccountTypeEnum$Type();
