// google-ads-api-client/dist/protos/bindings/ads/enums/listing_group_filter_custom_attribute_index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingGroupFilterCustomAttributeIndexEnum = exports.ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The index of customer attributes.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ListingGroupFilterCustomAttributeIndexEnum.ListingGroupFilterCustomAttributeIndex
 */
var ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex;
(function (ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex[ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex[ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * First listing group filter custom attribute.
     *
     * @generated from protobuf enum value: INDEX0 = 2;
     */
    ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex[ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex["INDEX0"] = 2] = "INDEX0";
    /**
     * Second listing group filter custom attribute.
     *
     * @generated from protobuf enum value: INDEX1 = 3;
     */
    ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex[ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex["INDEX1"] = 3] = "INDEX1";
    /**
     * Third listing group filter custom attribute.
     *
     * @generated from protobuf enum value: INDEX2 = 4;
     */
    ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex[ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex["INDEX2"] = 4] = "INDEX2";
    /**
     * Fourth listing group filter custom attribute.
     *
     * @generated from protobuf enum value: INDEX3 = 5;
     */
    ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex[ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex["INDEX3"] = 5] = "INDEX3";
    /**
     * Fifth listing group filter custom attribute.
     *
     * @generated from protobuf enum value: INDEX4 = 6;
     */
    ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex[ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex["INDEX4"] = 6] = "INDEX4";
})(ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex = exports.ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex || (exports.ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterCustomAttributeIndexEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ListingGroupFilterCustomAttributeIndexEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ListingGroupFilterCustomAttributeIndexEnum
 */
exports.ListingGroupFilterCustomAttributeIndexEnum = new ListingGroupFilterCustomAttributeIndexEnum$Type();
