// google-ads-api-client/dist/protos/bindings/ads/enums/listing_group_filter_product_channel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingGroupFilterProductChannelEnum = exports.ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the locality of a product offer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ListingGroupFilterProductChannelEnum.ListingGroupFilterProductChannel
 */
var ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel;
(function (ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel[ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel[ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The item is sold online.
     *
     * @generated from protobuf enum value: ONLINE = 2;
     */
    ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel[ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel["ONLINE"] = 2] = "ONLINE";
    /**
     * The item is sold in local stores.
     *
     * @generated from protobuf enum value: LOCAL = 3;
     */
    ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel[ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel["LOCAL"] = 3] = "LOCAL";
})(ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel = exports.ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel || (exports.ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterProductChannelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ListingGroupFilterProductChannelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ListingGroupFilterProductChannelEnum
 */
exports.ListingGroupFilterProductChannelEnum = new ListingGroupFilterProductChannelEnum$Type();
