// google-ads-api-client/dist/protos/bindings/ads/enums/listing_group_filter_type_enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingGroupFilterTypeEnum = exports.ListingGroupFilterTypeEnum_ListingGroupFilterType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of the listing group filter.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ListingGroupFilterTypeEnum.ListingGroupFilterType
 */
var ListingGroupFilterTypeEnum_ListingGroupFilterType;
(function (ListingGroupFilterTypeEnum_ListingGroupFilterType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ListingGroupFilterTypeEnum_ListingGroupFilterType[ListingGroupFilterTypeEnum_ListingGroupFilterType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ListingGroupFilterTypeEnum_ListingGroupFilterType[ListingGroupFilterTypeEnum_ListingGroupFilterType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Subdivision of products along some listing dimensions.
     *
     * @generated from protobuf enum value: SUBDIVISION = 2;
     */
    ListingGroupFilterTypeEnum_ListingGroupFilterType[ListingGroupFilterTypeEnum_ListingGroupFilterType["SUBDIVISION"] = 2] = "SUBDIVISION";
    /**
     * An included listing group filter leaf node.
     *
     * @generated from protobuf enum value: UNIT_INCLUDED = 3;
     */
    ListingGroupFilterTypeEnum_ListingGroupFilterType[ListingGroupFilterTypeEnum_ListingGroupFilterType["UNIT_INCLUDED"] = 3] = "UNIT_INCLUDED";
    /**
     * An excluded listing group filter leaf node.
     *
     * @generated from protobuf enum value: UNIT_EXCLUDED = 4;
     */
    ListingGroupFilterTypeEnum_ListingGroupFilterType[ListingGroupFilterTypeEnum_ListingGroupFilterType["UNIT_EXCLUDED"] = 4] = "UNIT_EXCLUDED";
})(ListingGroupFilterTypeEnum_ListingGroupFilterType = exports.ListingGroupFilterTypeEnum_ListingGroupFilterType || (exports.ListingGroupFilterTypeEnum_ListingGroupFilterType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ListingGroupFilterTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ListingGroupFilterTypeEnum
 */
exports.ListingGroupFilterTypeEnum = new ListingGroupFilterTypeEnum$Type();
