// google-ads-api-client/dist/protos/bindings/ads/enums/listing_group_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingGroupTypeEnum = exports.ListingGroupTypeEnum_ListingGroupType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of the listing group.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ListingGroupTypeEnum.ListingGroupType
 */
var ListingGroupTypeEnum_ListingGroupType;
(function (ListingGroupTypeEnum_ListingGroupType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ListingGroupTypeEnum_ListingGroupType[ListingGroupTypeEnum_ListingGroupType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ListingGroupTypeEnum_ListingGroupType[ListingGroupTypeEnum_ListingGroupType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Subdivision of products along some listing dimension. These nodes
     * are not used by serving to target listing entries, but is purely
     * to define the structure of the tree.
     *
     * @generated from protobuf enum value: SUBDIVISION = 2;
     */
    ListingGroupTypeEnum_ListingGroupType[ListingGroupTypeEnum_ListingGroupType["SUBDIVISION"] = 2] = "SUBDIVISION";
    /**
     * Listing group unit that defines a bid.
     *
     * @generated from protobuf enum value: UNIT = 3;
     */
    ListingGroupTypeEnum_ListingGroupType[ListingGroupTypeEnum_ListingGroupType["UNIT"] = 3] = "UNIT";
})(ListingGroupTypeEnum_ListingGroupType = exports.ListingGroupTypeEnum_ListingGroupType || (exports.ListingGroupTypeEnum_ListingGroupType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ListingGroupTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ListingGroupTypeEnum
 */
exports.ListingGroupTypeEnum = new ListingGroupTypeEnum$Type();
