// google-ads-api-client/dist/protos/bindings/ads/enums/listing_group_filter_product_condition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingGroupFilterProductConditionEnum = exports.ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the condition of a product offer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ListingGroupFilterProductConditionEnum.ListingGroupFilterProductCondition
 */
var ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition;
(function (ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition[ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition[ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The product condition is new.
     *
     * @generated from protobuf enum value: NEW = 2;
     */
    ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition[ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition["NEW"] = 2] = "NEW";
    /**
     * The product condition is refurbished.
     *
     * @generated from protobuf enum value: REFURBISHED = 3;
     */
    ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition[ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition["REFURBISHED"] = 3] = "REFURBISHED";
    /**
     * The product condition is used.
     *
     * @generated from protobuf enum value: USED = 4;
     */
    ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition[ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition["USED"] = 4] = "USED";
})(ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition = exports.ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition || (exports.ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterProductConditionEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ListingGroupFilterProductConditionEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ListingGroupFilterProductConditionEnum
 */
exports.ListingGroupFilterProductConditionEnum = new ListingGroupFilterProductConditionEnum$Type();
