// google-ads-api-client/dist/protos/bindings/ads/enums/listing_group_filter_product_type_level.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingGroupFilterProductTypeLevelEnum = exports.ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the level of the type of a product offer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ListingGroupFilterProductTypeLevelEnum.ListingGroupFilterProductTypeLevel
 */
var ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel;
(function (ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel[ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel[ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Level 1.
     *
     * @generated from protobuf enum value: LEVEL1 = 2;
     */
    ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel[ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel["LEVEL1"] = 2] = "LEVEL1";
    /**
     * Level 2.
     *
     * @generated from protobuf enum value: LEVEL2 = 3;
     */
    ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel[ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel["LEVEL2"] = 3] = "LEVEL2";
    /**
     * Level 3.
     *
     * @generated from protobuf enum value: LEVEL3 = 4;
     */
    ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel[ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel["LEVEL3"] = 4] = "LEVEL3";
    /**
     * Level 4.
     *
     * @generated from protobuf enum value: LEVEL4 = 5;
     */
    ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel[ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel["LEVEL4"] = 5] = "LEVEL4";
    /**
     * Level 5.
     *
     * @generated from protobuf enum value: LEVEL5 = 6;
     */
    ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel[ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel["LEVEL5"] = 6] = "LEVEL5";
})(ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel = exports.ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel || (exports.ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterProductTypeLevelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ListingGroupFilterProductTypeLevelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ListingGroupFilterProductTypeLevelEnum
 */
exports.ListingGroupFilterProductTypeLevelEnum = new ListingGroupFilterProductTypeLevelEnum$Type();
