// google-ads-api-client/dist/protos/bindings/ads/enums/location_source_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationSourceTypeEnum = exports.LocationSourceTypeEnum_LocationSourceType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible types of a location source.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LocationSourceTypeEnum.LocationSourceType
 */
var LocationSourceTypeEnum_LocationSourceType;
(function (LocationSourceTypeEnum_LocationSourceType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LocationSourceTypeEnum_LocationSourceType[LocationSourceTypeEnum_LocationSourceType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LocationSourceTypeEnum_LocationSourceType[LocationSourceTypeEnum_LocationSourceType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Locations associated with the customer's linked Business Profile.
     *
     * @generated from protobuf enum value: GOOGLE_MY_BUSINESS = 2;
     */
    LocationSourceTypeEnum_LocationSourceType[LocationSourceTypeEnum_LocationSourceType["GOOGLE_MY_BUSINESS"] = 2] = "GOOGLE_MY_BUSINESS";
    /**
     * Affiliate (chain) store locations. For example, Best Buy store locations.
     *
     * @generated from protobuf enum value: AFFILIATE = 3;
     */
    LocationSourceTypeEnum_LocationSourceType[LocationSourceTypeEnum_LocationSourceType["AFFILIATE"] = 3] = "AFFILIATE";
})(LocationSourceTypeEnum_LocationSourceType = exports.LocationSourceTypeEnum_LocationSourceType || (exports.LocationSourceTypeEnum_LocationSourceType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LocationSourceTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LocationSourceTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LocationSourceTypeEnum
 */
exports.LocationSourceTypeEnum = new LocationSourceTypeEnum$Type();
