// google-ads-api-client/dist/protos/bindings/ads/enums/location_group_radius_units.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationGroupRadiusUnitsEnum = exports.LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The unit of radius distance in location group (for example, MILES)
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LocationGroupRadiusUnitsEnum.LocationGroupRadiusUnits
 */
var LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits;
(function (LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits[LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits[LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Meters
     *
     * @generated from protobuf enum value: METERS = 2;
     */
    LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits[LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits["METERS"] = 2] = "METERS";
    /**
     * Miles
     *
     * @generated from protobuf enum value: MILES = 3;
     */
    LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits[LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits["MILES"] = 3] = "MILES";
    /**
     * Milli Miles
     *
     * @generated from protobuf enum value: MILLI_MILES = 4;
     */
    LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits[LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits["MILLI_MILES"] = 4] = "MILLI_MILES";
})(LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits = exports.LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits || (exports.LocationGroupRadiusUnitsEnum_LocationGroupRadiusUnits = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LocationGroupRadiusUnitsEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LocationGroupRadiusUnitsEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LocationGroupRadiusUnitsEnum
 */
exports.LocationGroupRadiusUnitsEnum = new LocationGroupRadiusUnitsEnum$Type();
