// google-ads-api-client/dist/protos/bindings/ads/enums/location_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationPlaceholderFieldEnum = exports.LocationPlaceholderFieldEnum_LocationPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Location placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LocationPlaceholderFieldEnum.LocationPlaceholderField
 */
var LocationPlaceholderFieldEnum_LocationPlaceholderField;
(function (LocationPlaceholderFieldEnum_LocationPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. The name of the business.
     *
     * @generated from protobuf enum value: BUSINESS_NAME = 2;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["BUSINESS_NAME"] = 2] = "BUSINESS_NAME";
    /**
     * Data Type: STRING. Line 1 of the business address.
     *
     * @generated from protobuf enum value: ADDRESS_LINE_1 = 3;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["ADDRESS_LINE_1"] = 3] = "ADDRESS_LINE_1";
    /**
     * Data Type: STRING. Line 2 of the business address.
     *
     * @generated from protobuf enum value: ADDRESS_LINE_2 = 4;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["ADDRESS_LINE_2"] = 4] = "ADDRESS_LINE_2";
    /**
     * Data Type: STRING. City of the business address.
     *
     * @generated from protobuf enum value: CITY = 5;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["CITY"] = 5] = "CITY";
    /**
     * Data Type: STRING. Province of the business address.
     *
     * @generated from protobuf enum value: PROVINCE = 6;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["PROVINCE"] = 6] = "PROVINCE";
    /**
     * Data Type: STRING. Postal code of the business address.
     *
     * @generated from protobuf enum value: POSTAL_CODE = 7;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["POSTAL_CODE"] = 7] = "POSTAL_CODE";
    /**
     * Data Type: STRING. Country code of the business address.
     *
     * @generated from protobuf enum value: COUNTRY_CODE = 8;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["COUNTRY_CODE"] = 8] = "COUNTRY_CODE";
    /**
     * Data Type: STRING. Phone number of the business.
     *
     * @generated from protobuf enum value: PHONE_NUMBER = 9;
     */
    LocationPlaceholderFieldEnum_LocationPlaceholderField[LocationPlaceholderFieldEnum_LocationPlaceholderField["PHONE_NUMBER"] = 9] = "PHONE_NUMBER";
})(LocationPlaceholderFieldEnum_LocationPlaceholderField = exports.LocationPlaceholderFieldEnum_LocationPlaceholderField || (exports.LocationPlaceholderFieldEnum_LocationPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LocationPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LocationPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LocationPlaceholderFieldEnum
 */
exports.LocationPlaceholderFieldEnum = new LocationPlaceholderFieldEnum$Type();
