// google-ads-api-client/dist/protos/bindings/ads/enums/location_extension_targeting_criterion_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationExtensionTargetingCriterionFieldEnum = exports.LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Location Extension Targeting criterion fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LocationExtensionTargetingCriterionFieldEnum.LocationExtensionTargetingCriterionField
 */
var LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField;
(function (LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField[LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField[LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Line 1 of the business address.
     *
     * @generated from protobuf enum value: ADDRESS_LINE_1 = 2;
     */
    LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField[LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField["ADDRESS_LINE_1"] = 2] = "ADDRESS_LINE_1";
    /**
     * Data Type: STRING. Line 2 of the business address.
     *
     * @generated from protobuf enum value: ADDRESS_LINE_2 = 3;
     */
    LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField[LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField["ADDRESS_LINE_2"] = 3] = "ADDRESS_LINE_2";
    /**
     * Data Type: STRING. City of the business address.
     *
     * @generated from protobuf enum value: CITY = 4;
     */
    LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField[LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField["CITY"] = 4] = "CITY";
    /**
     * Data Type: STRING. Province of the business address.
     *
     * @generated from protobuf enum value: PROVINCE = 5;
     */
    LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField[LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField["PROVINCE"] = 5] = "PROVINCE";
    /**
     * Data Type: STRING. Postal code of the business address.
     *
     * @generated from protobuf enum value: POSTAL_CODE = 6;
     */
    LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField[LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField["POSTAL_CODE"] = 6] = "POSTAL_CODE";
    /**
     * Data Type: STRING. Country code of the business address.
     *
     * @generated from protobuf enum value: COUNTRY_CODE = 7;
     */
    LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField[LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField["COUNTRY_CODE"] = 7] = "COUNTRY_CODE";
})(LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField = exports.LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField || (exports.LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LocationExtensionTargetingCriterionFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LocationExtensionTargetingCriterionFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LocationExtensionTargetingCriterionFieldEnum
 */
exports.LocationExtensionTargetingCriterionFieldEnum = new LocationExtensionTargetingCriterionFieldEnum$Type();
