// google-ads-api-client/dist/protos/bindings/ads/enums/local_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalPlaceholderFieldEnum = exports.LocalPlaceholderFieldEnum_LocalPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Local placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.LocalPlaceholderFieldEnum.LocalPlaceholderField
 */
var LocalPlaceholderFieldEnum_LocalPlaceholderField;
(function (LocalPlaceholderFieldEnum_LocalPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Required. Unique ID.
     *
     * @generated from protobuf enum value: DEAL_ID = 2;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["DEAL_ID"] = 2] = "DEAL_ID";
    /**
     * Data Type: STRING. Required. Main headline with local deal title to be
     * shown in dynamic ad.
     *
     * @generated from protobuf enum value: DEAL_NAME = 3;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["DEAL_NAME"] = 3] = "DEAL_NAME";
    /**
     * Data Type: STRING. Local deal subtitle to be shown in dynamic ad.
     *
     * @generated from protobuf enum value: SUBTITLE = 4;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["SUBTITLE"] = 4] = "SUBTITLE";
    /**
     * Data Type: STRING. Description of local deal to be shown in dynamic ad.
     *
     * @generated from protobuf enum value: DESCRIPTION = 5;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["DESCRIPTION"] = 5] = "DESCRIPTION";
    /**
     * Data Type: STRING. Price to be shown in the ad. Highly recommended for
     * dynamic ads. Example: "100.00 USD"
     *
     * @generated from protobuf enum value: PRICE = 6;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["PRICE"] = 6] = "PRICE";
    /**
     * Data Type: STRING. Formatted price to be shown in the ad.
     * Example: "Starting at $100.00 USD", "$80 - $100"
     *
     * @generated from protobuf enum value: FORMATTED_PRICE = 7;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["FORMATTED_PRICE"] = 7] = "FORMATTED_PRICE";
    /**
     * Data Type: STRING. Sale price to be shown in the ad.
     * Example: "80.00 USD"
     *
     * @generated from protobuf enum value: SALE_PRICE = 8;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["SALE_PRICE"] = 8] = "SALE_PRICE";
    /**
     * Data Type: STRING. Formatted sale price to be shown in the ad.
     * Example: "On sale for $80.00", "$60 - $80"
     *
     * @generated from protobuf enum value: FORMATTED_SALE_PRICE = 9;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["FORMATTED_SALE_PRICE"] = 9] = "FORMATTED_SALE_PRICE";
    /**
     * Data Type: URL. Image to be displayed in the ad.
     *
     * @generated from protobuf enum value: IMAGE_URL = 10;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["IMAGE_URL"] = 10] = "IMAGE_URL";
    /**
     * Data Type: STRING. Complete property address, including postal code.
     *
     * @generated from protobuf enum value: ADDRESS = 11;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["ADDRESS"] = 11] = "ADDRESS";
    /**
     * Data Type: STRING. Category of local deal used to group like items
     * together for recommendation engine.
     *
     * @generated from protobuf enum value: CATEGORY = 12;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["CATEGORY"] = 12] = "CATEGORY";
    /**
     * Data Type: STRING_LIST. Keywords used for product retrieval.
     *
     * @generated from protobuf enum value: CONTEXTUAL_KEYWORDS = 13;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["CONTEXTUAL_KEYWORDS"] = 13] = "CONTEXTUAL_KEYWORDS";
    /**
     * Data Type: URL_LIST. Required. Final URLs to be used in ad when using
     * Upgraded URLs; the more specific the better (for example, the individual
     * URL of a specific local deal and its location).
     *
     * @generated from protobuf enum value: FINAL_URLS = 14;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["FINAL_URLS"] = 14] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 15;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["FINAL_MOBILE_URLS"] = 15] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 16;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["TRACKING_URL"] = 16] = "TRACKING_URL";
    /**
     * Data Type: STRING. Android app link. Must be formatted as:
     * android-app://{package_id}/{scheme}/{host_path}.
     * The components are defined as follows:
     * package_id: app ID as specified in Google Play.
     * scheme: the scheme to pass to the application. Can be HTTP, or a custom
     *   scheme.
     * host_path: identifies the specific content within your application.
     *
     * @generated from protobuf enum value: ANDROID_APP_LINK = 17;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["ANDROID_APP_LINK"] = 17] = "ANDROID_APP_LINK";
    /**
     * Data Type: STRING_LIST. List of recommended local deal IDs to show
     * together with this item.
     *
     * @generated from protobuf enum value: SIMILAR_DEAL_IDS = 18;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["SIMILAR_DEAL_IDS"] = 18] = "SIMILAR_DEAL_IDS";
    /**
     * Data Type: STRING. iOS app link.
     *
     * @generated from protobuf enum value: IOS_APP_LINK = 19;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["IOS_APP_LINK"] = 19] = "IOS_APP_LINK";
    /**
     * Data Type: INT64. iOS app store ID.
     *
     * @generated from protobuf enum value: IOS_APP_STORE_ID = 20;
     */
    LocalPlaceholderFieldEnum_LocalPlaceholderField[LocalPlaceholderFieldEnum_LocalPlaceholderField["IOS_APP_STORE_ID"] = 20] = "IOS_APP_STORE_ID";
})(LocalPlaceholderFieldEnum_LocalPlaceholderField = exports.LocalPlaceholderFieldEnum_LocalPlaceholderField || (exports.LocalPlaceholderFieldEnum_LocalPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LocalPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.LocalPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.LocalPlaceholderFieldEnum
 */
exports.LocalPlaceholderFieldEnum = new LocalPlaceholderFieldEnum$Type();
