// google-ads-api-client/dist/protos/bindings/ads/enums/manager_link_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagerLinkStatusEnum = exports.ManagerLinkStatusEnum_ManagerLinkStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible statuses of a link.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ManagerLinkStatusEnum.ManagerLinkStatus
 */
var ManagerLinkStatusEnum_ManagerLinkStatus;
(function (ManagerLinkStatusEnum_ManagerLinkStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ManagerLinkStatusEnum_ManagerLinkStatus[ManagerLinkStatusEnum_ManagerLinkStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ManagerLinkStatusEnum_ManagerLinkStatus[ManagerLinkStatusEnum_ManagerLinkStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Indicates current in-effect relationship
     *
     * @generated from protobuf enum value: ACTIVE = 2;
     */
    ManagerLinkStatusEnum_ManagerLinkStatus[ManagerLinkStatusEnum_ManagerLinkStatus["ACTIVE"] = 2] = "ACTIVE";
    /**
     * Indicates terminated relationship
     *
     * @generated from protobuf enum value: INACTIVE = 3;
     */
    ManagerLinkStatusEnum_ManagerLinkStatus[ManagerLinkStatusEnum_ManagerLinkStatus["INACTIVE"] = 3] = "INACTIVE";
    /**
     * Indicates relationship has been requested by manager, but the client
     * hasn't accepted yet.
     *
     * @generated from protobuf enum value: PENDING = 4;
     */
    ManagerLinkStatusEnum_ManagerLinkStatus[ManagerLinkStatusEnum_ManagerLinkStatus["PENDING"] = 4] = "PENDING";
    /**
     * Relationship was requested by the manager, but the client has refused.
     *
     * @generated from protobuf enum value: REFUSED = 5;
     */
    ManagerLinkStatusEnum_ManagerLinkStatus[ManagerLinkStatusEnum_ManagerLinkStatus["REFUSED"] = 5] = "REFUSED";
    /**
     * Indicates relationship has been requested by manager, but manager
     * canceled it.
     *
     * @generated from protobuf enum value: CANCELED = 6;
     */
    ManagerLinkStatusEnum_ManagerLinkStatus[ManagerLinkStatusEnum_ManagerLinkStatus["CANCELED"] = 6] = "CANCELED";
})(ManagerLinkStatusEnum_ManagerLinkStatus = exports.ManagerLinkStatusEnum_ManagerLinkStatus || (exports.ManagerLinkStatusEnum_ManagerLinkStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ManagerLinkStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ManagerLinkStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ManagerLinkStatusEnum
 */
exports.ManagerLinkStatusEnum = new ManagerLinkStatusEnum$Type();
