// google-ads-api-client/dist/protos/bindings/ads/enums/matching_function_context_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MatchingFunctionContextTypeEnum = exports.MatchingFunctionContextTypeEnum_MatchingFunctionContextType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible context types for an operand in a matching function.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MatchingFunctionContextTypeEnum.MatchingFunctionContextType
 */
var MatchingFunctionContextTypeEnum_MatchingFunctionContextType;
(function (MatchingFunctionContextTypeEnum_MatchingFunctionContextType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MatchingFunctionContextTypeEnum_MatchingFunctionContextType[MatchingFunctionContextTypeEnum_MatchingFunctionContextType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MatchingFunctionContextTypeEnum_MatchingFunctionContextType[MatchingFunctionContextTypeEnum_MatchingFunctionContextType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Feed item id in the request context.
     *
     * @generated from protobuf enum value: FEED_ITEM_ID = 2;
     */
    MatchingFunctionContextTypeEnum_MatchingFunctionContextType[MatchingFunctionContextTypeEnum_MatchingFunctionContextType["FEED_ITEM_ID"] = 2] = "FEED_ITEM_ID";
    /**
     * The device being used (possible values are 'Desktop' or 'Mobile').
     *
     * @generated from protobuf enum value: DEVICE_NAME = 3;
     */
    MatchingFunctionContextTypeEnum_MatchingFunctionContextType[MatchingFunctionContextTypeEnum_MatchingFunctionContextType["DEVICE_NAME"] = 3] = "DEVICE_NAME";
    /**
     * Feed item set id in the request context.
     *
     * @generated from protobuf enum value: FEED_ITEM_SET_ID = 4;
     */
    MatchingFunctionContextTypeEnum_MatchingFunctionContextType[MatchingFunctionContextTypeEnum_MatchingFunctionContextType["FEED_ITEM_SET_ID"] = 4] = "FEED_ITEM_SET_ID";
})(MatchingFunctionContextTypeEnum_MatchingFunctionContextType = exports.MatchingFunctionContextTypeEnum_MatchingFunctionContextType || (exports.MatchingFunctionContextTypeEnum_MatchingFunctionContextType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MatchingFunctionContextTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MatchingFunctionContextTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MatchingFunctionContextTypeEnum
 */
exports.MatchingFunctionContextTypeEnum = new MatchingFunctionContextTypeEnum$Type();
