// google-ads-api-client/dist/protos/bindings/ads/enums/media_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaTypeEnum = exports.MediaTypeEnum_MediaType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of media.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MediaTypeEnum.MediaType
 */
var MediaTypeEnum_MediaType;
(function (MediaTypeEnum_MediaType) {
    /**
     * The media type has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MediaTypeEnum_MediaType[MediaTypeEnum_MediaType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MediaTypeEnum_MediaType[MediaTypeEnum_MediaType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Static image, used for image ad.
     *
     * @generated from protobuf enum value: IMAGE = 2;
     */
    MediaTypeEnum_MediaType[MediaTypeEnum_MediaType["IMAGE"] = 2] = "IMAGE";
    /**
     * Small image, used for map ad.
     *
     * @generated from protobuf enum value: ICON = 3;
     */
    MediaTypeEnum_MediaType[MediaTypeEnum_MediaType["ICON"] = 3] = "ICON";
    /**
     * ZIP file, used in fields of template ads.
     *
     * @generated from protobuf enum value: MEDIA_BUNDLE = 4;
     */
    MediaTypeEnum_MediaType[MediaTypeEnum_MediaType["MEDIA_BUNDLE"] = 4] = "MEDIA_BUNDLE";
    /**
     * Audio file.
     *
     * @generated from protobuf enum value: AUDIO = 5;
     */
    MediaTypeEnum_MediaType[MediaTypeEnum_MediaType["AUDIO"] = 5] = "AUDIO";
    /**
     * Video file.
     *
     * @generated from protobuf enum value: VIDEO = 6;
     */
    MediaTypeEnum_MediaType[MediaTypeEnum_MediaType["VIDEO"] = 6] = "VIDEO";
    /**
     * Animated image, such as animated GIF.
     *
     * @generated from protobuf enum value: DYNAMIC_IMAGE = 7;
     */
    MediaTypeEnum_MediaType[MediaTypeEnum_MediaType["DYNAMIC_IMAGE"] = 7] = "DYNAMIC_IMAGE";
})(MediaTypeEnum_MediaType = exports.MediaTypeEnum_MediaType || (exports.MediaTypeEnum_MediaType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MediaTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MediaTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MediaTypeEnum
 */
exports.MediaTypeEnum = new MediaTypeEnum$Type();
