// google-ads-api-client/dist/protos/bindings/ads/enums/merchant_center_link_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MerchantCenterLinkStatusEnum = exports.MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Describes the possible statuses for a link between a Google Ads customer
 * and a Google Merchant Center account.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MerchantCenterLinkStatusEnum.MerchantCenterLinkStatus
 */
var MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus;
(function (MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus[MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus[MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The link is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus[MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The link has no effect. It was proposed by the Merchant Center Account
     * owner and hasn't been confirmed by the customer.
     *
     * @generated from protobuf enum value: PENDING = 3;
     */
    MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus[MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus["PENDING"] = 3] = "PENDING";
})(MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus = exports.MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus || (exports.MerchantCenterLinkStatusEnum_MerchantCenterLinkStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MerchantCenterLinkStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MerchantCenterLinkStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MerchantCenterLinkStatusEnum
 */
exports.MerchantCenterLinkStatusEnum = new MerchantCenterLinkStatusEnum$Type();
