// google-ads-api-client/dist/protos/bindings/ads/enums/message_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessagePlaceholderFieldEnum = exports.MessagePlaceholderFieldEnum_MessagePlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Message placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MessagePlaceholderFieldEnum.MessagePlaceholderField
 */
var MessagePlaceholderFieldEnum_MessagePlaceholderField;
(function (MessagePlaceholderFieldEnum_MessagePlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MessagePlaceholderFieldEnum_MessagePlaceholderField[MessagePlaceholderFieldEnum_MessagePlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MessagePlaceholderFieldEnum_MessagePlaceholderField[MessagePlaceholderFieldEnum_MessagePlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. The name of your business.
     *
     * @generated from protobuf enum value: BUSINESS_NAME = 2;
     */
    MessagePlaceholderFieldEnum_MessagePlaceholderField[MessagePlaceholderFieldEnum_MessagePlaceholderField["BUSINESS_NAME"] = 2] = "BUSINESS_NAME";
    /**
     * Data Type: STRING. Country code of phone number.
     *
     * @generated from protobuf enum value: COUNTRY_CODE = 3;
     */
    MessagePlaceholderFieldEnum_MessagePlaceholderField[MessagePlaceholderFieldEnum_MessagePlaceholderField["COUNTRY_CODE"] = 3] = "COUNTRY_CODE";
    /**
     * Data Type: STRING. A phone number that's capable of sending and receiving
     * text messages.
     *
     * @generated from protobuf enum value: PHONE_NUMBER = 4;
     */
    MessagePlaceholderFieldEnum_MessagePlaceholderField[MessagePlaceholderFieldEnum_MessagePlaceholderField["PHONE_NUMBER"] = 4] = "PHONE_NUMBER";
    /**
     * Data Type: STRING. The text that will go in your click-to-message ad.
     *
     * @generated from protobuf enum value: MESSAGE_EXTENSION_TEXT = 5;
     */
    MessagePlaceholderFieldEnum_MessagePlaceholderField[MessagePlaceholderFieldEnum_MessagePlaceholderField["MESSAGE_EXTENSION_TEXT"] = 5] = "MESSAGE_EXTENSION_TEXT";
    /**
     * Data Type: STRING. The message text automatically shows in people's
     * messaging apps when they tap to send you a message.
     *
     * @generated from protobuf enum value: MESSAGE_TEXT = 6;
     */
    MessagePlaceholderFieldEnum_MessagePlaceholderField[MessagePlaceholderFieldEnum_MessagePlaceholderField["MESSAGE_TEXT"] = 6] = "MESSAGE_TEXT";
})(MessagePlaceholderFieldEnum_MessagePlaceholderField = exports.MessagePlaceholderFieldEnum_MessagePlaceholderField || (exports.MessagePlaceholderFieldEnum_MessagePlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MessagePlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MessagePlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MessagePlaceholderFieldEnum
 */
exports.MessagePlaceholderFieldEnum = new MessagePlaceholderFieldEnum$Type();
