// google-ads-api-client/dist/protos/bindings/ads/enums/mime_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MimeTypeEnum = exports.MimeTypeEnum_MimeType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The mime type
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MimeTypeEnum.MimeType
 */
var MimeTypeEnum_MimeType;
(function (MimeTypeEnum_MimeType) {
    /**
     * The mime type has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * MIME type of image/jpeg.
     *
     * @generated from protobuf enum value: IMAGE_JPEG = 2;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["IMAGE_JPEG"] = 2] = "IMAGE_JPEG";
    /**
     * MIME type of image/gif.
     *
     * @generated from protobuf enum value: IMAGE_GIF = 3;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["IMAGE_GIF"] = 3] = "IMAGE_GIF";
    /**
     * MIME type of image/png.
     *
     * @generated from protobuf enum value: IMAGE_PNG = 4;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["IMAGE_PNG"] = 4] = "IMAGE_PNG";
    /**
     * MIME type of application/x-shockwave-flash.
     *
     * @generated from protobuf enum value: FLASH = 5;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["FLASH"] = 5] = "FLASH";
    /**
     * MIME type of text/html.
     *
     * @generated from protobuf enum value: TEXT_HTML = 6;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["TEXT_HTML"] = 6] = "TEXT_HTML";
    /**
     * MIME type of application/pdf.
     *
     * @generated from protobuf enum value: PDF = 7;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["PDF"] = 7] = "PDF";
    /**
     * MIME type of application/msword.
     *
     * @generated from protobuf enum value: MSWORD = 8;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["MSWORD"] = 8] = "MSWORD";
    /**
     * MIME type of application/vnd.ms-excel.
     *
     * @generated from protobuf enum value: MSEXCEL = 9;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["MSEXCEL"] = 9] = "MSEXCEL";
    /**
     * MIME type of application/rtf.
     *
     * @generated from protobuf enum value: RTF = 10;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["RTF"] = 10] = "RTF";
    /**
     * MIME type of audio/wav.
     *
     * @generated from protobuf enum value: AUDIO_WAV = 11;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["AUDIO_WAV"] = 11] = "AUDIO_WAV";
    /**
     * MIME type of audio/mp3.
     *
     * @generated from protobuf enum value: AUDIO_MP3 = 12;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["AUDIO_MP3"] = 12] = "AUDIO_MP3";
    /**
     * MIME type of application/x-html5-ad-zip.
     *
     * @generated from protobuf enum value: HTML5_AD_ZIP = 13;
     */
    MimeTypeEnum_MimeType[MimeTypeEnum_MimeType["HTML5_AD_ZIP"] = 13] = "HTML5_AD_ZIP";
})(MimeTypeEnum_MimeType = exports.MimeTypeEnum_MimeType || (exports.MimeTypeEnum_MimeType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MimeTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MimeTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MimeTypeEnum
 */
exports.MimeTypeEnum = new MimeTypeEnum$Type();
