// google-ads-api-client/dist/protos/bindings/ads/enums/minute_of_hour.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MinuteOfHourEnum = exports.MinuteOfHourEnum_MinuteOfHour = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates of quarter-hours. For example, "FIFTEEN"
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MinuteOfHourEnum.MinuteOfHour
 */
var MinuteOfHourEnum_MinuteOfHour;
(function (MinuteOfHourEnum_MinuteOfHour) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MinuteOfHourEnum_MinuteOfHour[MinuteOfHourEnum_MinuteOfHour["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MinuteOfHourEnum_MinuteOfHour[MinuteOfHourEnum_MinuteOfHour["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Zero minutes past the hour.
     *
     * @generated from protobuf enum value: ZERO = 2;
     */
    MinuteOfHourEnum_MinuteOfHour[MinuteOfHourEnum_MinuteOfHour["ZERO"] = 2] = "ZERO";
    /**
     * Fifteen minutes past the hour.
     *
     * @generated from protobuf enum value: FIFTEEN = 3;
     */
    MinuteOfHourEnum_MinuteOfHour[MinuteOfHourEnum_MinuteOfHour["FIFTEEN"] = 3] = "FIFTEEN";
    /**
     * Thirty minutes past the hour.
     *
     * @generated from protobuf enum value: THIRTY = 4;
     */
    MinuteOfHourEnum_MinuteOfHour[MinuteOfHourEnum_MinuteOfHour["THIRTY"] = 4] = "THIRTY";
    /**
     * Forty-five minutes past the hour.
     *
     * @generated from protobuf enum value: FORTY_FIVE = 5;
     */
    MinuteOfHourEnum_MinuteOfHour[MinuteOfHourEnum_MinuteOfHour["FORTY_FIVE"] = 5] = "FORTY_FIVE";
})(MinuteOfHourEnum_MinuteOfHour = exports.MinuteOfHourEnum_MinuteOfHour || (exports.MinuteOfHourEnum_MinuteOfHour = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MinuteOfHourEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MinuteOfHourEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MinuteOfHourEnum
 */
exports.MinuteOfHourEnum = new MinuteOfHourEnum$Type();
