// google-ads-api-client/dist/protos/bindings/ads/enums/mobile_device_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobileDeviceTypeEnum = exports.MobileDeviceTypeEnum_MobileDeviceType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of mobile device.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MobileDeviceTypeEnum.MobileDeviceType
 */
var MobileDeviceTypeEnum_MobileDeviceType;
(function (MobileDeviceTypeEnum_MobileDeviceType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MobileDeviceTypeEnum_MobileDeviceType[MobileDeviceTypeEnum_MobileDeviceType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MobileDeviceTypeEnum_MobileDeviceType[MobileDeviceTypeEnum_MobileDeviceType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Mobile phones.
     *
     * @generated from protobuf enum value: MOBILE = 2;
     */
    MobileDeviceTypeEnum_MobileDeviceType[MobileDeviceTypeEnum_MobileDeviceType["MOBILE"] = 2] = "MOBILE";
    /**
     * Tablets.
     *
     * @generated from protobuf enum value: TABLET = 3;
     */
    MobileDeviceTypeEnum_MobileDeviceType[MobileDeviceTypeEnum_MobileDeviceType["TABLET"] = 3] = "TABLET";
})(MobileDeviceTypeEnum_MobileDeviceType = exports.MobileDeviceTypeEnum_MobileDeviceType || (exports.MobileDeviceTypeEnum_MobileDeviceType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MobileDeviceTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MobileDeviceTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MobileDeviceTypeEnum
 */
exports.MobileDeviceTypeEnum = new MobileDeviceTypeEnum$Type();
