// google-ads-api-client/dist/protos/bindings/ads/enums/mobile_app_vendor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobileAppVendorEnum = exports.MobileAppVendorEnum_MobileAppVendor = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of mobile app vendor
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor
 */
var MobileAppVendorEnum_MobileAppVendor;
(function (MobileAppVendorEnum_MobileAppVendor) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MobileAppVendorEnum_MobileAppVendor[MobileAppVendorEnum_MobileAppVendor["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MobileAppVendorEnum_MobileAppVendor[MobileAppVendorEnum_MobileAppVendor["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Mobile app vendor for Apple app store.
     *
     * @generated from protobuf enum value: APPLE_APP_STORE = 2;
     */
    MobileAppVendorEnum_MobileAppVendor[MobileAppVendorEnum_MobileAppVendor["APPLE_APP_STORE"] = 2] = "APPLE_APP_STORE";
    /**
     * Mobile app vendor for Google app store.
     *
     * @generated from protobuf enum value: GOOGLE_APP_STORE = 3;
     */
    MobileAppVendorEnum_MobileAppVendor[MobileAppVendorEnum_MobileAppVendor["GOOGLE_APP_STORE"] = 3] = "GOOGLE_APP_STORE";
})(MobileAppVendorEnum_MobileAppVendor = exports.MobileAppVendorEnum_MobileAppVendor || (exports.MobileAppVendorEnum_MobileAppVendor = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MobileAppVendorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MobileAppVendorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MobileAppVendorEnum
 */
exports.MobileAppVendorEnum = new MobileAppVendorEnum$Type();
