// google-ads-api-client/dist/protos/bindings/ads/enums/month_of_year.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonthOfYearEnum = exports.MonthOfYearEnum_MonthOfYear = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates months of the year, for example, "January".
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.MonthOfYearEnum.MonthOfYear
 */
var MonthOfYearEnum_MonthOfYear;
(function (MonthOfYearEnum_MonthOfYear) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * January.
     *
     * @generated from protobuf enum value: JANUARY = 2;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["JANUARY"] = 2] = "JANUARY";
    /**
     * February.
     *
     * @generated from protobuf enum value: FEBRUARY = 3;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["FEBRUARY"] = 3] = "FEBRUARY";
    /**
     * March.
     *
     * @generated from protobuf enum value: MARCH = 4;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["MARCH"] = 4] = "MARCH";
    /**
     * April.
     *
     * @generated from protobuf enum value: APRIL = 5;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["APRIL"] = 5] = "APRIL";
    /**
     * May.
     *
     * @generated from protobuf enum value: MAY = 6;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["MAY"] = 6] = "MAY";
    /**
     * June.
     *
     * @generated from protobuf enum value: JUNE = 7;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["JUNE"] = 7] = "JUNE";
    /**
     * July.
     *
     * @generated from protobuf enum value: JULY = 8;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["JULY"] = 8] = "JULY";
    /**
     * August.
     *
     * @generated from protobuf enum value: AUGUST = 9;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["AUGUST"] = 9] = "AUGUST";
    /**
     * September.
     *
     * @generated from protobuf enum value: SEPTEMBER = 10;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["SEPTEMBER"] = 10] = "SEPTEMBER";
    /**
     * October.
     *
     * @generated from protobuf enum value: OCTOBER = 11;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["OCTOBER"] = 11] = "OCTOBER";
    /**
     * November.
     *
     * @generated from protobuf enum value: NOVEMBER = 12;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["NOVEMBER"] = 12] = "NOVEMBER";
    /**
     * December.
     *
     * @generated from protobuf enum value: DECEMBER = 13;
     */
    MonthOfYearEnum_MonthOfYear[MonthOfYearEnum_MonthOfYear["DECEMBER"] = 13] = "DECEMBER";
})(MonthOfYearEnum_MonthOfYear = exports.MonthOfYearEnum_MonthOfYear || (exports.MonthOfYearEnum_MonthOfYear = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MonthOfYearEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.MonthOfYearEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.MonthOfYearEnum
 */
exports.MonthOfYearEnum = new MonthOfYearEnum$Type();
