// google-ads-api-client/dist/protos/bindings/ads/enums/negative_geo_target_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NegativeGeoTargetTypeEnum = exports.NegativeGeoTargetTypeEnum_NegativeGeoTargetType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible negative geo target types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.NegativeGeoTargetTypeEnum.NegativeGeoTargetType
 */
var NegativeGeoTargetTypeEnum_NegativeGeoTargetType;
(function (NegativeGeoTargetTypeEnum_NegativeGeoTargetType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    NegativeGeoTargetTypeEnum_NegativeGeoTargetType[NegativeGeoTargetTypeEnum_NegativeGeoTargetType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    NegativeGeoTargetTypeEnum_NegativeGeoTargetType[NegativeGeoTargetTypeEnum_NegativeGeoTargetType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Specifies that a user is excluded from seeing the ad if they
     * are in, or show interest in, advertiser's excluded locations.
     *
     * @generated from protobuf enum value: PRESENCE_OR_INTEREST = 4;
     */
    NegativeGeoTargetTypeEnum_NegativeGeoTargetType[NegativeGeoTargetTypeEnum_NegativeGeoTargetType["PRESENCE_OR_INTEREST"] = 4] = "PRESENCE_OR_INTEREST";
    /**
     * Specifies that a user is excluded from seeing the ad if they
     * are in advertiser's excluded locations.
     *
     * @generated from protobuf enum value: PRESENCE = 5;
     */
    NegativeGeoTargetTypeEnum_NegativeGeoTargetType[NegativeGeoTargetTypeEnum_NegativeGeoTargetType["PRESENCE"] = 5] = "PRESENCE";
})(NegativeGeoTargetTypeEnum_NegativeGeoTargetType = exports.NegativeGeoTargetTypeEnum_NegativeGeoTargetType || (exports.NegativeGeoTargetTypeEnum_NegativeGeoTargetType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class NegativeGeoTargetTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.NegativeGeoTargetTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.NegativeGeoTargetTypeEnum
 */
exports.NegativeGeoTargetTypeEnum = new NegativeGeoTargetTypeEnum$Type();
