// google-ads-api-client/dist/protos/bindings/ads/enums/offline_user_data_job_match_rate_range.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OfflineUserDataJobMatchRateRangeEnum = exports.OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The match rate range of an offline user data job.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.OfflineUserDataJobMatchRateRangeEnum.OfflineUserDataJobMatchRateRange
 */
var OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange;
(function (OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Default value for match rate range.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Match rate range for offline data upload entity is between 0% and 19%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_LESS_THAN_20 = 2;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_LESS_THAN_20"] = 2] = "MATCH_RANGE_LESS_THAN_20";
    /**
     * Match rate range for offline data upload entity is between 20% and 30%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_20_TO_30 = 3;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_20_TO_30"] = 3] = "MATCH_RANGE_20_TO_30";
    /**
     * Match rate range for offline data upload entity is between 31% and 40%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_31_TO_40 = 4;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_31_TO_40"] = 4] = "MATCH_RANGE_31_TO_40";
    /**
     * Match rate range for offline data upload entity is between 41% and 50%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_41_TO_50 = 5;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_41_TO_50"] = 5] = "MATCH_RANGE_41_TO_50";
    /**
     * Match rate range for offline data upload entity is between 51% and 60%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_51_TO_60 = 6;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_51_TO_60"] = 6] = "MATCH_RANGE_51_TO_60";
    /**
     * Match rate range for offline data upload entity is between 61% and 70%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_61_TO_70 = 7;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_61_TO_70"] = 7] = "MATCH_RANGE_61_TO_70";
    /**
     * Match rate range for offline data upload entity is between 71% and 80%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_71_TO_80 = 8;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_71_TO_80"] = 8] = "MATCH_RANGE_71_TO_80";
    /**
     * Match rate range for offline data upload entity is between 81% and 90%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_81_TO_90 = 9;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_81_TO_90"] = 9] = "MATCH_RANGE_81_TO_90";
    /**
     * Match rate range for offline data upload entity more than or equal to
     * 91%.
     *
     * @generated from protobuf enum value: MATCH_RANGE_91_TO_100 = 10;
     */
    OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange[OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange["MATCH_RANGE_91_TO_100"] = 10] = "MATCH_RANGE_91_TO_100";
})(OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange = exports.OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange || (exports.OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OfflineUserDataJobMatchRateRangeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.OfflineUserDataJobMatchRateRangeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.OfflineUserDataJobMatchRateRangeEnum
 */
exports.OfflineUserDataJobMatchRateRangeEnum = new OfflineUserDataJobMatchRateRangeEnum$Type();
