// google-ads-api-client/dist/protos/bindings/ads/enums/offline_user_data_job_failure_reason.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OfflineUserDataJobFailureReasonEnum = exports.OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The failure reason of an offline user data job.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.OfflineUserDataJobFailureReasonEnum.OfflineUserDataJobFailureReason
 */
var OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason;
(function (OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason[OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason[OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The matched transactions are insufficient.
     *
     * @generated from protobuf enum value: INSUFFICIENT_MATCHED_TRANSACTIONS = 2;
     */
    OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason[OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason["INSUFFICIENT_MATCHED_TRANSACTIONS"] = 2] = "INSUFFICIENT_MATCHED_TRANSACTIONS";
    /**
     * The uploaded transactions are insufficient.
     *
     * @generated from protobuf enum value: INSUFFICIENT_TRANSACTIONS = 3;
     */
    OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason[OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason["INSUFFICIENT_TRANSACTIONS"] = 3] = "INSUFFICIENT_TRANSACTIONS";
})(OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason = exports.OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason || (exports.OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OfflineUserDataJobFailureReasonEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.OfflineUserDataJobFailureReasonEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.OfflineUserDataJobFailureReasonEnum
 */
exports.OfflineUserDataJobFailureReasonEnum = new OfflineUserDataJobFailureReasonEnum$Type();
