// google-ads-api-client/dist/protos/bindings/ads/enums/offline_user_data_job_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OfflineUserDataJobTypeEnum = exports.OfflineUserDataJobTypeEnum_OfflineUserDataJobType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of an offline user data job.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.OfflineUserDataJobTypeEnum.OfflineUserDataJobType
 */
var OfflineUserDataJobTypeEnum_OfflineUserDataJobType;
(function (OfflineUserDataJobTypeEnum_OfflineUserDataJobType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OfflineUserDataJobTypeEnum_OfflineUserDataJobType[OfflineUserDataJobTypeEnum_OfflineUserDataJobType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OfflineUserDataJobTypeEnum_OfflineUserDataJobType[OfflineUserDataJobTypeEnum_OfflineUserDataJobType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Store Sales Direct data for self service.
     *
     * @generated from protobuf enum value: STORE_SALES_UPLOAD_FIRST_PARTY = 2;
     */
    OfflineUserDataJobTypeEnum_OfflineUserDataJobType[OfflineUserDataJobTypeEnum_OfflineUserDataJobType["STORE_SALES_UPLOAD_FIRST_PARTY"] = 2] = "STORE_SALES_UPLOAD_FIRST_PARTY";
    /**
     * Store Sales Direct data for third party.
     *
     * @generated from protobuf enum value: STORE_SALES_UPLOAD_THIRD_PARTY = 3;
     */
    OfflineUserDataJobTypeEnum_OfflineUserDataJobType[OfflineUserDataJobTypeEnum_OfflineUserDataJobType["STORE_SALES_UPLOAD_THIRD_PARTY"] = 3] = "STORE_SALES_UPLOAD_THIRD_PARTY";
    /**
     * Customer Match user list data.
     *
     * @generated from protobuf enum value: CUSTOMER_MATCH_USER_LIST = 4;
     */
    OfflineUserDataJobTypeEnum_OfflineUserDataJobType[OfflineUserDataJobTypeEnum_OfflineUserDataJobType["CUSTOMER_MATCH_USER_LIST"] = 4] = "CUSTOMER_MATCH_USER_LIST";
    /**
     * Customer Match with attribute data.
     *
     * @generated from protobuf enum value: CUSTOMER_MATCH_WITH_ATTRIBUTES = 5;
     */
    OfflineUserDataJobTypeEnum_OfflineUserDataJobType[OfflineUserDataJobTypeEnum_OfflineUserDataJobType["CUSTOMER_MATCH_WITH_ATTRIBUTES"] = 5] = "CUSTOMER_MATCH_WITH_ATTRIBUTES";
})(OfflineUserDataJobTypeEnum_OfflineUserDataJobType = exports.OfflineUserDataJobTypeEnum_OfflineUserDataJobType || (exports.OfflineUserDataJobTypeEnum_OfflineUserDataJobType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OfflineUserDataJobTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.OfflineUserDataJobTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.OfflineUserDataJobTypeEnum
 */
exports.OfflineUserDataJobTypeEnum = new OfflineUserDataJobTypeEnum$Type();
