// google-ads-api-client/dist/protos/bindings/ads/enums/offline_user_data_job_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OfflineUserDataJobStatusEnum = exports.OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The status of an offline user data job.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.OfflineUserDataJobStatusEnum.OfflineUserDataJobStatus
 */
var OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus;
(function (OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus[OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus[OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The job has been successfully created and pending for uploading.
     *
     * @generated from protobuf enum value: PENDING = 2;
     */
    OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus[OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus["PENDING"] = 2] = "PENDING";
    /**
     * Upload(s) have been accepted and data is being processed.
     *
     * @generated from protobuf enum value: RUNNING = 3;
     */
    OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus[OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus["RUNNING"] = 3] = "RUNNING";
    /**
     * Uploaded data has been successfully processed.
     *
     * @generated from protobuf enum value: SUCCESS = 4;
     */
    OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus[OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus["SUCCESS"] = 4] = "SUCCESS";
    /**
     * Uploaded data has failed to be processed.
     *
     * @generated from protobuf enum value: FAILED = 5;
     */
    OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus[OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus["FAILED"] = 5] = "FAILED";
})(OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus = exports.OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus || (exports.OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OfflineUserDataJobStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.OfflineUserDataJobStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.OfflineUserDataJobStatusEnum
 */
exports.OfflineUserDataJobStatusEnum = new OfflineUserDataJobStatusEnum$Type();
