// google-ads-api-client/dist/protos/bindings/ads/enums/operating_system_version_operator_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperatingSystemVersionOperatorTypeEnum = exports.OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of operating system version.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.OperatingSystemVersionOperatorTypeEnum.OperatingSystemVersionOperatorType
 */
var OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType;
(function (OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType[OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType[OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Equals to the specified version.
     *
     * @generated from protobuf enum value: EQUALS_TO = 2;
     */
    OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType[OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType["EQUALS_TO"] = 2] = "EQUALS_TO";
    /**
     * Greater than or equals to the specified version.
     *
     * @generated from protobuf enum value: GREATER_THAN_EQUALS_TO = 4;
     */
    OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType[OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType["GREATER_THAN_EQUALS_TO"] = 4] = "GREATER_THAN_EQUALS_TO";
})(OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType = exports.OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType || (exports.OperatingSystemVersionOperatorTypeEnum_OperatingSystemVersionOperatorType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OperatingSystemVersionOperatorTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.OperatingSystemVersionOperatorTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.OperatingSystemVersionOperatorTypeEnum
 */
exports.OperatingSystemVersionOperatorTypeEnum = new OperatingSystemVersionOperatorTypeEnum$Type();
