// google-ads-api-client/dist/protos/bindings/ads/enums/optimization_goal_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptimizationGoalTypeEnum = exports.OptimizationGoalTypeEnum_OptimizationGoalType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of optimization goal
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.OptimizationGoalTypeEnum.OptimizationGoalType
 */
var OptimizationGoalTypeEnum_OptimizationGoalType;
(function (OptimizationGoalTypeEnum_OptimizationGoalType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OptimizationGoalTypeEnum_OptimizationGoalType[OptimizationGoalTypeEnum_OptimizationGoalType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used as a return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OptimizationGoalTypeEnum_OptimizationGoalType[OptimizationGoalTypeEnum_OptimizationGoalType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Optimize for call clicks. Call click conversions are times people
     * selected 'Call' to contact a store after viewing an ad.
     *
     * @generated from protobuf enum value: CALL_CLICKS = 2;
     */
    OptimizationGoalTypeEnum_OptimizationGoalType[OptimizationGoalTypeEnum_OptimizationGoalType["CALL_CLICKS"] = 2] = "CALL_CLICKS";
    /**
     * Optimize for driving directions. Driving directions conversions are
     * times people selected 'Get directions' to navigate to a store after
     * viewing an ad.
     *
     * @generated from protobuf enum value: DRIVING_DIRECTIONS = 3;
     */
    OptimizationGoalTypeEnum_OptimizationGoalType[OptimizationGoalTypeEnum_OptimizationGoalType["DRIVING_DIRECTIONS"] = 3] = "DRIVING_DIRECTIONS";
    /**
     * Optimize for pre-registration. Pre-registration conversions are the
     * number of pre-registration signups to receive a notification when the app
     * is released.
     *
     * @generated from protobuf enum value: APP_PRE_REGISTRATION = 4;
     */
    OptimizationGoalTypeEnum_OptimizationGoalType[OptimizationGoalTypeEnum_OptimizationGoalType["APP_PRE_REGISTRATION"] = 4] = "APP_PRE_REGISTRATION";
})(OptimizationGoalTypeEnum_OptimizationGoalType = exports.OptimizationGoalTypeEnum_OptimizationGoalType || (exports.OptimizationGoalTypeEnum_OptimizationGoalType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OptimizationGoalTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.OptimizationGoalTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.OptimizationGoalTypeEnum
 */
exports.OptimizationGoalTypeEnum = new OptimizationGoalTypeEnum$Type();
