// google-ads-api-client/dist/protos/bindings/ads/enums/parental_status_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParentalStatusTypeEnum = exports.ParentalStatusTypeEnum_ParentalStatusType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of parental statuses (for example, not a parent).
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ParentalStatusTypeEnum.ParentalStatusType
 */
var ParentalStatusTypeEnum_ParentalStatusType;
(function (ParentalStatusTypeEnum_ParentalStatusType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ParentalStatusTypeEnum_ParentalStatusType[ParentalStatusTypeEnum_ParentalStatusType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ParentalStatusTypeEnum_ParentalStatusType[ParentalStatusTypeEnum_ParentalStatusType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Parent.
     *
     * @generated from protobuf enum value: PARENT = 300;
     */
    ParentalStatusTypeEnum_ParentalStatusType[ParentalStatusTypeEnum_ParentalStatusType["PARENT"] = 300] = "PARENT";
    /**
     * Not a parent.
     *
     * @generated from protobuf enum value: NOT_A_PARENT = 301;
     */
    ParentalStatusTypeEnum_ParentalStatusType[ParentalStatusTypeEnum_ParentalStatusType["NOT_A_PARENT"] = 301] = "NOT_A_PARENT";
    /**
     * Undetermined parental status.
     *
     * @generated from protobuf enum value: UNDETERMINED = 302;
     */
    ParentalStatusTypeEnum_ParentalStatusType[ParentalStatusTypeEnum_ParentalStatusType["UNDETERMINED"] = 302] = "UNDETERMINED";
})(ParentalStatusTypeEnum_ParentalStatusType = exports.ParentalStatusTypeEnum_ParentalStatusType || (exports.ParentalStatusTypeEnum_ParentalStatusType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ParentalStatusTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ParentalStatusTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ParentalStatusTypeEnum
 */
exports.ParentalStatusTypeEnum = new ParentalStatusTypeEnum$Type();
