// google-ads-api-client/dist/protos/bindings/ads/enums/payment_mode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentModeEnum = exports.PaymentModeEnum_PaymentMode = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible payment modes.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PaymentModeEnum.PaymentMode
 */
var PaymentModeEnum_PaymentMode;
(function (PaymentModeEnum_PaymentMode) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PaymentModeEnum_PaymentMode[PaymentModeEnum_PaymentMode["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PaymentModeEnum_PaymentMode[PaymentModeEnum_PaymentMode["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Pay per click.
     *
     * @generated from protobuf enum value: CLICKS = 4;
     */
    PaymentModeEnum_PaymentMode[PaymentModeEnum_PaymentMode["CLICKS"] = 4] = "CLICKS";
    /**
     * Pay per conversion value. This mode is only supported by campaigns with
     * AdvertisingChannelType.HOTEL, BiddingStrategyType.COMMISSION, and
     * BudgetType.STANDARD.
     *
     * @generated from protobuf enum value: CONVERSION_VALUE = 5;
     */
    PaymentModeEnum_PaymentMode[PaymentModeEnum_PaymentMode["CONVERSION_VALUE"] = 5] = "CONVERSION_VALUE";
    /**
     * Pay per conversion. This mode is only supported by campaigns with
     * AdvertisingChannelType.DISPLAY (excluding
     * AdvertisingChannelSubType.DISPLAY_GMAIL), BiddingStrategyType.TARGET_CPA,
     * and BudgetType.FIXED_CPA. The customer must also be eligible for this
     * mode. See Customer.eligibility_failure_reasons for details.
     *
     * @generated from protobuf enum value: CONVERSIONS = 6;
     */
    PaymentModeEnum_PaymentMode[PaymentModeEnum_PaymentMode["CONVERSIONS"] = 6] = "CONVERSIONS";
    /**
     * Pay per guest stay value. This mode is only supported by campaigns with
     * AdvertisingChannelType.HOTEL, BiddingStrategyType.COMMISSION, and
     * BudgetType.STANDARD.
     *
     * @generated from protobuf enum value: GUEST_STAY = 7;
     */
    PaymentModeEnum_PaymentMode[PaymentModeEnum_PaymentMode["GUEST_STAY"] = 7] = "GUEST_STAY";
})(PaymentModeEnum_PaymentMode = exports.PaymentModeEnum_PaymentMode || (exports.PaymentModeEnum_PaymentMode = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PaymentModeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PaymentModeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PaymentModeEnum
 */
exports.PaymentModeEnum = new PaymentModeEnum$Type();
