// google-ads-api-client/dist/protos/bindings/ads/enums/performance_max_upgrade_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PerformanceMaxUpgradeStatusEnum = exports.PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Performance Max Upgrade status enum for campaign.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PerformanceMaxUpgradeStatusEnum.PerformanceMaxUpgradeStatus
 */
var PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus;
(function (PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus[PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus[PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The campaign is eligible for upgrade to a Performance Max campaign.
     *
     * @generated from protobuf enum value: UPGRADE_ELIBIGLE = 2;
     */
    PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus[PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus["UPGRADE_ELIBIGLE"] = 2] = "UPGRADE_ELIBIGLE";
    /**
     * The upgrade to a Performance Max campaign is in progress.
     *
     * @generated from protobuf enum value: UPGRADE_IN_PROGRESS = 3;
     */
    PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus[PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus["UPGRADE_IN_PROGRESS"] = 3] = "UPGRADE_IN_PROGRESS";
    /**
     * The upgrade to a Performance Max campaign is complete.
     *
     * @generated from protobuf enum value: UPGRADE_COMPLETE = 4;
     */
    PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus[PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus["UPGRADE_COMPLETE"] = 4] = "UPGRADE_COMPLETE";
    /**
     * The upgrade to a Performance Max campaign failed.
     * The campaign will still serve as it was before upgrade was attempted.
     *
     * @generated from protobuf enum value: UPGRADE_FAILED = 5;
     */
    PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus[PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus["UPGRADE_FAILED"] = 5] = "UPGRADE_FAILED";
})(PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus = exports.PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus || (exports.PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PerformanceMaxUpgradeStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PerformanceMaxUpgradeStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PerformanceMaxUpgradeStatusEnum
 */
exports.PerformanceMaxUpgradeStatusEnum = new PerformanceMaxUpgradeStatusEnum$Type();
