// google-ads-api-client/dist/protos/bindings/ads/enums/placement_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlacementTypeEnum = exports.PlacementTypeEnum_PlacementType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible placement types for a feed mapping.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PlacementTypeEnum.PlacementType
 */
var PlacementTypeEnum_PlacementType;
(function (PlacementTypeEnum_PlacementType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PlacementTypeEnum_PlacementType[PlacementTypeEnum_PlacementType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PlacementTypeEnum_PlacementType[PlacementTypeEnum_PlacementType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Websites(for example, 'www.flowers4sale.com').
     *
     * @generated from protobuf enum value: WEBSITE = 2;
     */
    PlacementTypeEnum_PlacementType[PlacementTypeEnum_PlacementType["WEBSITE"] = 2] = "WEBSITE";
    /**
     * Mobile application categories(for example, 'Games').
     *
     * @generated from protobuf enum value: MOBILE_APP_CATEGORY = 3;
     */
    PlacementTypeEnum_PlacementType[PlacementTypeEnum_PlacementType["MOBILE_APP_CATEGORY"] = 3] = "MOBILE_APP_CATEGORY";
    /**
     * mobile applications(for example, 'mobileapp::2-com.whatsthewordanswers').
     *
     * @generated from protobuf enum value: MOBILE_APPLICATION = 4;
     */
    PlacementTypeEnum_PlacementType[PlacementTypeEnum_PlacementType["MOBILE_APPLICATION"] = 4] = "MOBILE_APPLICATION";
    /**
     * YouTube videos(for example, 'youtube.com/video/wtLJPvx7-ys').
     *
     * @generated from protobuf enum value: YOUTUBE_VIDEO = 5;
     */
    PlacementTypeEnum_PlacementType[PlacementTypeEnum_PlacementType["YOUTUBE_VIDEO"] = 5] = "YOUTUBE_VIDEO";
    /**
     * YouTube channels(for example, 'youtube.com::L8ZULXASCc1I_oaOT0NaOQ').
     *
     * @generated from protobuf enum value: YOUTUBE_CHANNEL = 6;
     */
    PlacementTypeEnum_PlacementType[PlacementTypeEnum_PlacementType["YOUTUBE_CHANNEL"] = 6] = "YOUTUBE_CHANNEL";
})(PlacementTypeEnum_PlacementType = exports.PlacementTypeEnum_PlacementType || (exports.PlacementTypeEnum_PlacementType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PlacementTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PlacementTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PlacementTypeEnum
 */
exports.PlacementTypeEnum = new PlacementTypeEnum$Type();
