// google-ads-api-client/dist/protos/bindings/ads/enums/placeholder_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlaceholderTypeEnum = exports.PlaceholderTypeEnum_PlaceholderType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible placeholder types for a feed mapping.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PlaceholderTypeEnum.PlaceholderType
 */
var PlaceholderTypeEnum_PlaceholderType;
(function (PlaceholderTypeEnum_PlaceholderType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Lets you show links in your ad to pages from your website, including the
     * main landing page.
     *
     * @generated from protobuf enum value: SITELINK = 2;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["SITELINK"] = 2] = "SITELINK";
    /**
     * Lets you attach a phone number to an ad, allowing customers to call
     * directly from the ad.
     *
     * @generated from protobuf enum value: CALL = 3;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["CALL"] = 3] = "CALL";
    /**
     * Lets you provide users with a link that points to a mobile app in
     * addition to a website.
     *
     * @generated from protobuf enum value: APP = 4;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["APP"] = 4] = "APP";
    /**
     * Lets you show locations of businesses from your Business Profile
     * in your ad. This helps people find your locations by showing your
     * ads with your address, a map to your location, or the distance to your
     * business. This extension type is useful to draw customers to your
     * brick-and-mortar location.
     *
     * @generated from protobuf enum value: LOCATION = 5;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["LOCATION"] = 5] = "LOCATION";
    /**
     * If you sell your product through retail chains, affiliate location
     * extensions let you show nearby stores that carry your products.
     *
     * @generated from protobuf enum value: AFFILIATE_LOCATION = 6;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["AFFILIATE_LOCATION"] = 6] = "AFFILIATE_LOCATION";
    /**
     * Lets you include additional text with your search ads that provide
     * detailed information about your business, including products and services
     * you offer. Callouts appear in ads at the top and bottom of Google search
     * results.
     *
     * @generated from protobuf enum value: CALLOUT = 7;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["CALLOUT"] = 7] = "CALLOUT";
    /**
     * Lets you add more info to your ad, specific to some predefined categories
     * such as types, brands, styles, etc. A minimum of 3 text (SNIPPETS) values
     * are required.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPET = 8;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["STRUCTURED_SNIPPET"] = 8] = "STRUCTURED_SNIPPET";
    /**
     * Allows users to see your ad, click an icon, and contact you directly by
     * text message. With one tap on your ad, people can contact you to book an
     * appointment, get a quote, ask for information, or request a service.
     *
     * @generated from protobuf enum value: MESSAGE = 9;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["MESSAGE"] = 9] = "MESSAGE";
    /**
     * Lets you display prices for a list of items along with your ads. A price
     * feed is composed of three to eight price table rows.
     *
     * @generated from protobuf enum value: PRICE = 10;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["PRICE"] = 10] = "PRICE";
    /**
     * Lets you highlight sales and other promotions that let users see how
     * they can save by buying now.
     *
     * @generated from protobuf enum value: PROMOTION = 11;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["PROMOTION"] = 11] = "PROMOTION";
    /**
     * Lets you dynamically inject custom data into the title and description
     * of your ads.
     *
     * @generated from protobuf enum value: AD_CUSTOMIZER = 12;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["AD_CUSTOMIZER"] = 12] = "AD_CUSTOMIZER";
    /**
     * Indicates that this feed is for education dynamic remarketing.
     *
     * @generated from protobuf enum value: DYNAMIC_EDUCATION = 13;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["DYNAMIC_EDUCATION"] = 13] = "DYNAMIC_EDUCATION";
    /**
     * Indicates that this feed is for flight dynamic remarketing.
     *
     * @generated from protobuf enum value: DYNAMIC_FLIGHT = 14;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["DYNAMIC_FLIGHT"] = 14] = "DYNAMIC_FLIGHT";
    /**
     * Indicates that this feed is for a custom dynamic remarketing type. Use
     * this only if the other business types don't apply to your products or
     * services.
     *
     * @generated from protobuf enum value: DYNAMIC_CUSTOM = 15;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["DYNAMIC_CUSTOM"] = 15] = "DYNAMIC_CUSTOM";
    /**
     * Indicates that this feed is for hotels and rentals dynamic remarketing.
     *
     * @generated from protobuf enum value: DYNAMIC_HOTEL = 16;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["DYNAMIC_HOTEL"] = 16] = "DYNAMIC_HOTEL";
    /**
     * Indicates that this feed is for real estate dynamic remarketing.
     *
     * @generated from protobuf enum value: DYNAMIC_REAL_ESTATE = 17;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["DYNAMIC_REAL_ESTATE"] = 17] = "DYNAMIC_REAL_ESTATE";
    /**
     * Indicates that this feed is for travel dynamic remarketing.
     *
     * @generated from protobuf enum value: DYNAMIC_TRAVEL = 18;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["DYNAMIC_TRAVEL"] = 18] = "DYNAMIC_TRAVEL";
    /**
     * Indicates that this feed is for local deals dynamic remarketing.
     *
     * @generated from protobuf enum value: DYNAMIC_LOCAL = 19;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["DYNAMIC_LOCAL"] = 19] = "DYNAMIC_LOCAL";
    /**
     * Indicates that this feed is for job dynamic remarketing.
     *
     * @generated from protobuf enum value: DYNAMIC_JOB = 20;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["DYNAMIC_JOB"] = 20] = "DYNAMIC_JOB";
    /**
     * Lets you attach an image to an ad.
     *
     * @generated from protobuf enum value: IMAGE = 21;
     */
    PlaceholderTypeEnum_PlaceholderType[PlaceholderTypeEnum_PlaceholderType["IMAGE"] = 21] = "IMAGE";
})(PlaceholderTypeEnum_PlaceholderType = exports.PlaceholderTypeEnum_PlaceholderType || (exports.PlaceholderTypeEnum_PlaceholderType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PlaceholderTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PlaceholderTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PlaceholderTypeEnum
 */
exports.PlaceholderTypeEnum = new PlaceholderTypeEnum$Type();
