// google-ads-api-client/dist/protos/bindings/ads/enums/policy_review_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyReviewStatusEnum = exports.PolicyReviewStatusEnum_PolicyReviewStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible policy review statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus
 */
var PolicyReviewStatusEnum_PolicyReviewStatus;
(function (PolicyReviewStatusEnum_PolicyReviewStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PolicyReviewStatusEnum_PolicyReviewStatus[PolicyReviewStatusEnum_PolicyReviewStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PolicyReviewStatusEnum_PolicyReviewStatus[PolicyReviewStatusEnum_PolicyReviewStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Currently under review.
     *
     * @generated from protobuf enum value: REVIEW_IN_PROGRESS = 2;
     */
    PolicyReviewStatusEnum_PolicyReviewStatus[PolicyReviewStatusEnum_PolicyReviewStatus["REVIEW_IN_PROGRESS"] = 2] = "REVIEW_IN_PROGRESS";
    /**
     * Primary review complete. Other reviews may be continuing.
     *
     * @generated from protobuf enum value: REVIEWED = 3;
     */
    PolicyReviewStatusEnum_PolicyReviewStatus[PolicyReviewStatusEnum_PolicyReviewStatus["REVIEWED"] = 3] = "REVIEWED";
    /**
     * The resource has been resubmitted for approval or its policy decision has
     * been appealed.
     *
     * @generated from protobuf enum value: UNDER_APPEAL = 4;
     */
    PolicyReviewStatusEnum_PolicyReviewStatus[PolicyReviewStatusEnum_PolicyReviewStatus["UNDER_APPEAL"] = 4] = "UNDER_APPEAL";
    /**
     * The resource is eligible and may be serving but could still undergo
     * further review.
     *
     * @generated from protobuf enum value: ELIGIBLE_MAY_SERVE = 5;
     */
    PolicyReviewStatusEnum_PolicyReviewStatus[PolicyReviewStatusEnum_PolicyReviewStatus["ELIGIBLE_MAY_SERVE"] = 5] = "ELIGIBLE_MAY_SERVE";
})(PolicyReviewStatusEnum_PolicyReviewStatus = exports.PolicyReviewStatusEnum_PolicyReviewStatus || (exports.PolicyReviewStatusEnum_PolicyReviewStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PolicyReviewStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PolicyReviewStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PolicyReviewStatusEnum
 */
exports.PolicyReviewStatusEnum = new PolicyReviewStatusEnum$Type();
