// google-ads-api-client/dist/protos/bindings/ads/enums/policy_approval_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyApprovalStatusEnum = exports.PolicyApprovalStatusEnum_PolicyApprovalStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible policy approval statuses. When there are several approval
 * statuses available the most severe one will be used. The order of severity
 * is DISAPPROVED, AREA_OF_INTEREST_ONLY, APPROVED_LIMITED and APPROVED.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus
 */
var PolicyApprovalStatusEnum_PolicyApprovalStatus;
(function (PolicyApprovalStatusEnum_PolicyApprovalStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PolicyApprovalStatusEnum_PolicyApprovalStatus[PolicyApprovalStatusEnum_PolicyApprovalStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PolicyApprovalStatusEnum_PolicyApprovalStatus[PolicyApprovalStatusEnum_PolicyApprovalStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Will not serve.
     *
     * @generated from protobuf enum value: DISAPPROVED = 2;
     */
    PolicyApprovalStatusEnum_PolicyApprovalStatus[PolicyApprovalStatusEnum_PolicyApprovalStatus["DISAPPROVED"] = 2] = "DISAPPROVED";
    /**
     * Serves with restrictions.
     *
     * @generated from protobuf enum value: APPROVED_LIMITED = 3;
     */
    PolicyApprovalStatusEnum_PolicyApprovalStatus[PolicyApprovalStatusEnum_PolicyApprovalStatus["APPROVED_LIMITED"] = 3] = "APPROVED_LIMITED";
    /**
     * Serves without restrictions.
     *
     * @generated from protobuf enum value: APPROVED = 4;
     */
    PolicyApprovalStatusEnum_PolicyApprovalStatus[PolicyApprovalStatusEnum_PolicyApprovalStatus["APPROVED"] = 4] = "APPROVED";
    /**
     * Will not serve in targeted countries, but may serve for users who are
     * searching for information about the targeted countries.
     *
     * @generated from protobuf enum value: AREA_OF_INTEREST_ONLY = 5;
     */
    PolicyApprovalStatusEnum_PolicyApprovalStatus[PolicyApprovalStatusEnum_PolicyApprovalStatus["AREA_OF_INTEREST_ONLY"] = 5] = "AREA_OF_INTEREST_ONLY";
})(PolicyApprovalStatusEnum_PolicyApprovalStatus = exports.PolicyApprovalStatusEnum_PolicyApprovalStatus || (exports.PolicyApprovalStatusEnum_PolicyApprovalStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PolicyApprovalStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PolicyApprovalStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PolicyApprovalStatusEnum
 */
exports.PolicyApprovalStatusEnum = new PolicyApprovalStatusEnum$Type();
