// google-ads-api-client/dist/protos/bindings/ads/enums/policy_topic_entry_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyTopicEntryTypeEnum = exports.PolicyTopicEntryTypeEnum_PolicyTopicEntryType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible policy topic entry types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PolicyTopicEntryTypeEnum.PolicyTopicEntryType
 */
var PolicyTopicEntryTypeEnum_PolicyTopicEntryType;
(function (PolicyTopicEntryTypeEnum_PolicyTopicEntryType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PolicyTopicEntryTypeEnum_PolicyTopicEntryType[PolicyTopicEntryTypeEnum_PolicyTopicEntryType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PolicyTopicEntryTypeEnum_PolicyTopicEntryType[PolicyTopicEntryTypeEnum_PolicyTopicEntryType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The resource will not be served.
     *
     * @generated from protobuf enum value: PROHIBITED = 2;
     */
    PolicyTopicEntryTypeEnum_PolicyTopicEntryType[PolicyTopicEntryTypeEnum_PolicyTopicEntryType["PROHIBITED"] = 2] = "PROHIBITED";
    /**
     * The resource will not be served under some circumstances.
     *
     * @generated from protobuf enum value: LIMITED = 4;
     */
    PolicyTopicEntryTypeEnum_PolicyTopicEntryType[PolicyTopicEntryTypeEnum_PolicyTopicEntryType["LIMITED"] = 4] = "LIMITED";
    /**
     * The resource cannot serve at all because of the current targeting
     * criteria.
     *
     * @generated from protobuf enum value: FULLY_LIMITED = 8;
     */
    PolicyTopicEntryTypeEnum_PolicyTopicEntryType[PolicyTopicEntryTypeEnum_PolicyTopicEntryType["FULLY_LIMITED"] = 8] = "FULLY_LIMITED";
    /**
     * May be of interest, but does not limit how the resource is served.
     *
     * @generated from protobuf enum value: DESCRIPTIVE = 5;
     */
    PolicyTopicEntryTypeEnum_PolicyTopicEntryType[PolicyTopicEntryTypeEnum_PolicyTopicEntryType["DESCRIPTIVE"] = 5] = "DESCRIPTIVE";
    /**
     * Could increase coverage beyond normal.
     *
     * @generated from protobuf enum value: BROADENING = 6;
     */
    PolicyTopicEntryTypeEnum_PolicyTopicEntryType[PolicyTopicEntryTypeEnum_PolicyTopicEntryType["BROADENING"] = 6] = "BROADENING";
    /**
     * Constrained for all targeted countries, but may serve in other countries
     * through area of interest.
     *
     * @generated from protobuf enum value: AREA_OF_INTEREST_ONLY = 7;
     */
    PolicyTopicEntryTypeEnum_PolicyTopicEntryType[PolicyTopicEntryTypeEnum_PolicyTopicEntryType["AREA_OF_INTEREST_ONLY"] = 7] = "AREA_OF_INTEREST_ONLY";
})(PolicyTopicEntryTypeEnum_PolicyTopicEntryType = exports.PolicyTopicEntryTypeEnum_PolicyTopicEntryType || (exports.PolicyTopicEntryTypeEnum_PolicyTopicEntryType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PolicyTopicEntryTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PolicyTopicEntryTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PolicyTopicEntryTypeEnum
 */
exports.PolicyTopicEntryTypeEnum = new PolicyTopicEntryTypeEnum$Type();
