// google-ads-api-client/dist/protos/bindings/ads/enums/policy_topic_evidence_destination_mismatch_url_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyTopicEvidenceDestinationMismatchUrlTypeEnum = exports.PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible policy topic evidence destination mismatch url types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PolicyTopicEvidenceDestinationMismatchUrlTypeEnum.PolicyTopicEvidenceDestinationMismatchUrlType
 */
var PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType;
(function (PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType[PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType[PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The display url.
     *
     * @generated from protobuf enum value: DISPLAY_URL = 2;
     */
    PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType[PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType["DISPLAY_URL"] = 2] = "DISPLAY_URL";
    /**
     * The final url.
     *
     * @generated from protobuf enum value: FINAL_URL = 3;
     */
    PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType[PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType["FINAL_URL"] = 3] = "FINAL_URL";
    /**
     * The final mobile url.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URL = 4;
     */
    PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType[PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType["FINAL_MOBILE_URL"] = 4] = "FINAL_MOBILE_URL";
    /**
     * The tracking url template, with substituted desktop url.
     *
     * @generated from protobuf enum value: TRACKING_URL = 5;
     */
    PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType[PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType["TRACKING_URL"] = 5] = "TRACKING_URL";
    /**
     * The tracking url template, with substituted mobile url.
     *
     * @generated from protobuf enum value: MOBILE_TRACKING_URL = 6;
     */
    PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType[PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType["MOBILE_TRACKING_URL"] = 6] = "MOBILE_TRACKING_URL";
})(PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType = exports.PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType || (exports.PolicyTopicEvidenceDestinationMismatchUrlTypeEnum_PolicyTopicEvidenceDestinationMismatchUrlType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PolicyTopicEvidenceDestinationMismatchUrlTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PolicyTopicEvidenceDestinationMismatchUrlTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PolicyTopicEvidenceDestinationMismatchUrlTypeEnum
 */
exports.PolicyTopicEvidenceDestinationMismatchUrlTypeEnum = new PolicyTopicEvidenceDestinationMismatchUrlTypeEnum$Type();
