// google-ads-api-client/dist/protos/bindings/ads/enums/positive_geo_target_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PositiveGeoTargetTypeEnum = exports.PositiveGeoTargetTypeEnum_PositiveGeoTargetType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible positive geo target types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PositiveGeoTargetTypeEnum.PositiveGeoTargetType
 */
var PositiveGeoTargetTypeEnum_PositiveGeoTargetType;
(function (PositiveGeoTargetTypeEnum_PositiveGeoTargetType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PositiveGeoTargetTypeEnum_PositiveGeoTargetType[PositiveGeoTargetTypeEnum_PositiveGeoTargetType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PositiveGeoTargetTypeEnum_PositiveGeoTargetType[PositiveGeoTargetTypeEnum_PositiveGeoTargetType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Specifies that an ad is triggered if the user is in,
     * or shows interest in, advertiser's targeted locations.
     *
     * @generated from protobuf enum value: PRESENCE_OR_INTEREST = 5;
     */
    PositiveGeoTargetTypeEnum_PositiveGeoTargetType[PositiveGeoTargetTypeEnum_PositiveGeoTargetType["PRESENCE_OR_INTEREST"] = 5] = "PRESENCE_OR_INTEREST";
    /**
     * Specifies that an ad is triggered if the user
     * searches for advertiser's targeted locations.
     * This can only be used with Search and standard
     * Shopping campaigns.
     *
     * @generated from protobuf enum value: SEARCH_INTEREST = 6;
     */
    PositiveGeoTargetTypeEnum_PositiveGeoTargetType[PositiveGeoTargetTypeEnum_PositiveGeoTargetType["SEARCH_INTEREST"] = 6] = "SEARCH_INTEREST";
    /**
     * Specifies that an ad is triggered if the user is in
     * or regularly in advertiser's targeted locations.
     *
     * @generated from protobuf enum value: PRESENCE = 7;
     */
    PositiveGeoTargetTypeEnum_PositiveGeoTargetType[PositiveGeoTargetTypeEnum_PositiveGeoTargetType["PRESENCE"] = 7] = "PRESENCE";
})(PositiveGeoTargetTypeEnum_PositiveGeoTargetType = exports.PositiveGeoTargetTypeEnum_PositiveGeoTargetType || (exports.PositiveGeoTargetTypeEnum_PositiveGeoTargetType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PositiveGeoTargetTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PositiveGeoTargetTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PositiveGeoTargetTypeEnum
 */
exports.PositiveGeoTargetTypeEnum = new PositiveGeoTargetTypeEnum$Type();
