// google-ads-api-client/dist/protos/bindings/ads/enums/price_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricePlaceholderFieldEnum = exports.PricePlaceholderFieldEnum_PricePlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Price placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PricePlaceholderFieldEnum.PricePlaceholderField
 */
var PricePlaceholderFieldEnum_PricePlaceholderField;
(function (PricePlaceholderFieldEnum_PricePlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. The type of your price feed. Must match one of the
     * predefined price feed type exactly.
     *
     * @generated from protobuf enum value: TYPE = 2;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["TYPE"] = 2] = "TYPE";
    /**
     * Data Type: STRING. The qualifier of each price. Must match one of the
     * predefined price qualifiers exactly.
     *
     * @generated from protobuf enum value: PRICE_QUALIFIER = 3;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["PRICE_QUALIFIER"] = 3] = "PRICE_QUALIFIER";
    /**
     * Data Type: URL. Tracking template for the price feed when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: TRACKING_TEMPLATE = 4;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["TRACKING_TEMPLATE"] = 4] = "TRACKING_TEMPLATE";
    /**
     * Data Type: STRING. Language of the price feed. Must match one of the
     * available available locale codes exactly.
     *
     * @generated from protobuf enum value: LANGUAGE = 5;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["LANGUAGE"] = 5] = "LANGUAGE";
    /**
     * Data Type: STRING. Final URL suffix for the price feed when using
     * parallel tracking.
     *
     * @generated from protobuf enum value: FINAL_URL_SUFFIX = 6;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["FINAL_URL_SUFFIX"] = 6] = "FINAL_URL_SUFFIX";
    /**
     * Data Type: STRING. The header of item 1 of the table.
     *
     * @generated from protobuf enum value: ITEM_1_HEADER = 100;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_1_HEADER"] = 100] = "ITEM_1_HEADER";
    /**
     * Data Type: STRING. The description of item 1 of the table.
     *
     * @generated from protobuf enum value: ITEM_1_DESCRIPTION = 101;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_1_DESCRIPTION"] = 101] = "ITEM_1_DESCRIPTION";
    /**
     * Data Type: MONEY. The price (money with currency) of item 1 of the table,
     * for example, 30 USD. The currency must match one of the available
     * currencies.
     *
     * @generated from protobuf enum value: ITEM_1_PRICE = 102;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_1_PRICE"] = 102] = "ITEM_1_PRICE";
    /**
     * Data Type: STRING. The price unit of item 1 of the table. Must match one
     * of the predefined price units.
     *
     * @generated from protobuf enum value: ITEM_1_UNIT = 103;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_1_UNIT"] = 103] = "ITEM_1_UNIT";
    /**
     * Data Type: URL_LIST. The final URLs of item 1 of the table when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_1_FINAL_URLS = 104;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_1_FINAL_URLS"] = 104] = "ITEM_1_FINAL_URLS";
    /**
     * Data Type: URL_LIST. The final mobile URLs of item 1 of the table when
     * using Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_1_FINAL_MOBILE_URLS = 105;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_1_FINAL_MOBILE_URLS"] = 105] = "ITEM_1_FINAL_MOBILE_URLS";
    /**
     * Data Type: STRING. The header of item 2 of the table.
     *
     * @generated from protobuf enum value: ITEM_2_HEADER = 200;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_2_HEADER"] = 200] = "ITEM_2_HEADER";
    /**
     * Data Type: STRING. The description of item 2 of the table.
     *
     * @generated from protobuf enum value: ITEM_2_DESCRIPTION = 201;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_2_DESCRIPTION"] = 201] = "ITEM_2_DESCRIPTION";
    /**
     * Data Type: MONEY. The price (money with currency) of item 2 of the table,
     * for example, 30 USD. The currency must match one of the available
     * currencies.
     *
     * @generated from protobuf enum value: ITEM_2_PRICE = 202;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_2_PRICE"] = 202] = "ITEM_2_PRICE";
    /**
     * Data Type: STRING. The price unit of item 2 of the table. Must match one
     * of the predefined price units.
     *
     * @generated from protobuf enum value: ITEM_2_UNIT = 203;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_2_UNIT"] = 203] = "ITEM_2_UNIT";
    /**
     * Data Type: URL_LIST. The final URLs of item 2 of the table when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_2_FINAL_URLS = 204;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_2_FINAL_URLS"] = 204] = "ITEM_2_FINAL_URLS";
    /**
     * Data Type: URL_LIST. The final mobile URLs of item 2 of the table when
     * using Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_2_FINAL_MOBILE_URLS = 205;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_2_FINAL_MOBILE_URLS"] = 205] = "ITEM_2_FINAL_MOBILE_URLS";
    /**
     * Data Type: STRING. The header of item 3 of the table.
     *
     * @generated from protobuf enum value: ITEM_3_HEADER = 300;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_3_HEADER"] = 300] = "ITEM_3_HEADER";
    /**
     * Data Type: STRING. The description of item 3 of the table.
     *
     * @generated from protobuf enum value: ITEM_3_DESCRIPTION = 301;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_3_DESCRIPTION"] = 301] = "ITEM_3_DESCRIPTION";
    /**
     * Data Type: MONEY. The price (money with currency) of item 3 of the table,
     * for example, 30 USD. The currency must match one of the available
     * currencies.
     *
     * @generated from protobuf enum value: ITEM_3_PRICE = 302;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_3_PRICE"] = 302] = "ITEM_3_PRICE";
    /**
     * Data Type: STRING. The price unit of item 3 of the table. Must match one
     * of the predefined price units.
     *
     * @generated from protobuf enum value: ITEM_3_UNIT = 303;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_3_UNIT"] = 303] = "ITEM_3_UNIT";
    /**
     * Data Type: URL_LIST. The final URLs of item 3 of the table when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_3_FINAL_URLS = 304;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_3_FINAL_URLS"] = 304] = "ITEM_3_FINAL_URLS";
    /**
     * Data Type: URL_LIST. The final mobile URLs of item 3 of the table when
     * using Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_3_FINAL_MOBILE_URLS = 305;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_3_FINAL_MOBILE_URLS"] = 305] = "ITEM_3_FINAL_MOBILE_URLS";
    /**
     * Data Type: STRING. The header of item 4 of the table.
     *
     * @generated from protobuf enum value: ITEM_4_HEADER = 400;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_4_HEADER"] = 400] = "ITEM_4_HEADER";
    /**
     * Data Type: STRING. The description of item 4 of the table.
     *
     * @generated from protobuf enum value: ITEM_4_DESCRIPTION = 401;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_4_DESCRIPTION"] = 401] = "ITEM_4_DESCRIPTION";
    /**
     * Data Type: MONEY. The price (money with currency) of item 4 of the table,
     * for example, 30 USD. The currency must match one of the available
     * currencies.
     *
     * @generated from protobuf enum value: ITEM_4_PRICE = 402;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_4_PRICE"] = 402] = "ITEM_4_PRICE";
    /**
     * Data Type: STRING. The price unit of item 4 of the table. Must match one
     * of the predefined price units.
     *
     * @generated from protobuf enum value: ITEM_4_UNIT = 403;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_4_UNIT"] = 403] = "ITEM_4_UNIT";
    /**
     * Data Type: URL_LIST. The final URLs of item 4 of the table when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_4_FINAL_URLS = 404;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_4_FINAL_URLS"] = 404] = "ITEM_4_FINAL_URLS";
    /**
     * Data Type: URL_LIST. The final mobile URLs of item 4 of the table when
     * using Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_4_FINAL_MOBILE_URLS = 405;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_4_FINAL_MOBILE_URLS"] = 405] = "ITEM_4_FINAL_MOBILE_URLS";
    /**
     * Data Type: STRING. The header of item 5 of the table.
     *
     * @generated from protobuf enum value: ITEM_5_HEADER = 500;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_5_HEADER"] = 500] = "ITEM_5_HEADER";
    /**
     * Data Type: STRING. The description of item 5 of the table.
     *
     * @generated from protobuf enum value: ITEM_5_DESCRIPTION = 501;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_5_DESCRIPTION"] = 501] = "ITEM_5_DESCRIPTION";
    /**
     * Data Type: MONEY. The price (money with currency) of item 5 of the table,
     * for example, 30 USD. The currency must match one of the available
     * currencies.
     *
     * @generated from protobuf enum value: ITEM_5_PRICE = 502;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_5_PRICE"] = 502] = "ITEM_5_PRICE";
    /**
     * Data Type: STRING. The price unit of item 5 of the table. Must match one
     * of the predefined price units.
     *
     * @generated from protobuf enum value: ITEM_5_UNIT = 503;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_5_UNIT"] = 503] = "ITEM_5_UNIT";
    /**
     * Data Type: URL_LIST. The final URLs of item 5 of the table when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_5_FINAL_URLS = 504;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_5_FINAL_URLS"] = 504] = "ITEM_5_FINAL_URLS";
    /**
     * Data Type: URL_LIST. The final mobile URLs of item 5 of the table when
     * using Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_5_FINAL_MOBILE_URLS = 505;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_5_FINAL_MOBILE_URLS"] = 505] = "ITEM_5_FINAL_MOBILE_URLS";
    /**
     * Data Type: STRING. The header of item 6 of the table.
     *
     * @generated from protobuf enum value: ITEM_6_HEADER = 600;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_6_HEADER"] = 600] = "ITEM_6_HEADER";
    /**
     * Data Type: STRING. The description of item 6 of the table.
     *
     * @generated from protobuf enum value: ITEM_6_DESCRIPTION = 601;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_6_DESCRIPTION"] = 601] = "ITEM_6_DESCRIPTION";
    /**
     * Data Type: MONEY. The price (money with currency) of item 6 of the table,
     * for example, 30 USD. The currency must match one of the available
     * currencies.
     *
     * @generated from protobuf enum value: ITEM_6_PRICE = 602;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_6_PRICE"] = 602] = "ITEM_6_PRICE";
    /**
     * Data Type: STRING. The price unit of item 6 of the table. Must match one
     * of the predefined price units.
     *
     * @generated from protobuf enum value: ITEM_6_UNIT = 603;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_6_UNIT"] = 603] = "ITEM_6_UNIT";
    /**
     * Data Type: URL_LIST. The final URLs of item 6 of the table when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_6_FINAL_URLS = 604;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_6_FINAL_URLS"] = 604] = "ITEM_6_FINAL_URLS";
    /**
     * Data Type: URL_LIST. The final mobile URLs of item 6 of the table when
     * using Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_6_FINAL_MOBILE_URLS = 605;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_6_FINAL_MOBILE_URLS"] = 605] = "ITEM_6_FINAL_MOBILE_URLS";
    /**
     * Data Type: STRING. The header of item 7 of the table.
     *
     * @generated from protobuf enum value: ITEM_7_HEADER = 700;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_7_HEADER"] = 700] = "ITEM_7_HEADER";
    /**
     * Data Type: STRING. The description of item 7 of the table.
     *
     * @generated from protobuf enum value: ITEM_7_DESCRIPTION = 701;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_7_DESCRIPTION"] = 701] = "ITEM_7_DESCRIPTION";
    /**
     * Data Type: MONEY. The price (money with currency) of item 7 of the table,
     * for example, 30 USD. The currency must match one of the available
     * currencies.
     *
     * @generated from protobuf enum value: ITEM_7_PRICE = 702;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_7_PRICE"] = 702] = "ITEM_7_PRICE";
    /**
     * Data Type: STRING. The price unit of item 7 of the table. Must match one
     * of the predefined price units.
     *
     * @generated from protobuf enum value: ITEM_7_UNIT = 703;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_7_UNIT"] = 703] = "ITEM_7_UNIT";
    /**
     * Data Type: URL_LIST. The final URLs of item 7 of the table when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_7_FINAL_URLS = 704;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_7_FINAL_URLS"] = 704] = "ITEM_7_FINAL_URLS";
    /**
     * Data Type: URL_LIST. The final mobile URLs of item 7 of the table when
     * using Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_7_FINAL_MOBILE_URLS = 705;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_7_FINAL_MOBILE_URLS"] = 705] = "ITEM_7_FINAL_MOBILE_URLS";
    /**
     * Data Type: STRING. The header of item 8 of the table.
     *
     * @generated from protobuf enum value: ITEM_8_HEADER = 800;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_8_HEADER"] = 800] = "ITEM_8_HEADER";
    /**
     * Data Type: STRING. The description of item 8 of the table.
     *
     * @generated from protobuf enum value: ITEM_8_DESCRIPTION = 801;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_8_DESCRIPTION"] = 801] = "ITEM_8_DESCRIPTION";
    /**
     * Data Type: MONEY. The price (money with currency) of item 8 of the table,
     * for example, 30 USD. The currency must match one of the available
     * currencies.
     *
     * @generated from protobuf enum value: ITEM_8_PRICE = 802;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_8_PRICE"] = 802] = "ITEM_8_PRICE";
    /**
     * Data Type: STRING. The price unit of item 8 of the table. Must match one
     * of the predefined price units.
     *
     * @generated from protobuf enum value: ITEM_8_UNIT = 803;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_8_UNIT"] = 803] = "ITEM_8_UNIT";
    /**
     * Data Type: URL_LIST. The final URLs of item 8 of the table when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_8_FINAL_URLS = 804;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_8_FINAL_URLS"] = 804] = "ITEM_8_FINAL_URLS";
    /**
     * Data Type: URL_LIST. The final mobile URLs of item 8 of the table when
     * using Upgraded URLs.
     *
     * @generated from protobuf enum value: ITEM_8_FINAL_MOBILE_URLS = 805;
     */
    PricePlaceholderFieldEnum_PricePlaceholderField[PricePlaceholderFieldEnum_PricePlaceholderField["ITEM_8_FINAL_MOBILE_URLS"] = 805] = "ITEM_8_FINAL_MOBILE_URLS";
})(PricePlaceholderFieldEnum_PricePlaceholderField = exports.PricePlaceholderFieldEnum_PricePlaceholderField || (exports.PricePlaceholderFieldEnum_PricePlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PricePlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PricePlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PricePlaceholderFieldEnum
 */
exports.PricePlaceholderFieldEnum = new PricePlaceholderFieldEnum$Type();
