// google-ads-api-client/dist/protos/bindings/ads/enums/price_extension_price_unit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriceExtensionPriceUnitEnum = exports.PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Price extension price unit.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum.PriceExtensionPriceUnit
 */
var PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit;
(function (PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit[PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit[PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Per hour.
     *
     * @generated from protobuf enum value: PER_HOUR = 2;
     */
    PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit[PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit["PER_HOUR"] = 2] = "PER_HOUR";
    /**
     * Per day.
     *
     * @generated from protobuf enum value: PER_DAY = 3;
     */
    PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit[PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit["PER_DAY"] = 3] = "PER_DAY";
    /**
     * Per week.
     *
     * @generated from protobuf enum value: PER_WEEK = 4;
     */
    PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit[PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit["PER_WEEK"] = 4] = "PER_WEEK";
    /**
     * Per month.
     *
     * @generated from protobuf enum value: PER_MONTH = 5;
     */
    PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit[PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit["PER_MONTH"] = 5] = "PER_MONTH";
    /**
     * Per year.
     *
     * @generated from protobuf enum value: PER_YEAR = 6;
     */
    PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit[PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit["PER_YEAR"] = 6] = "PER_YEAR";
    /**
     * Per night.
     *
     * @generated from protobuf enum value: PER_NIGHT = 7;
     */
    PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit[PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit["PER_NIGHT"] = 7] = "PER_NIGHT";
})(PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit = exports.PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit || (exports.PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PriceExtensionPriceUnitEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum
 */
exports.PriceExtensionPriceUnitEnum = new PriceExtensionPriceUnitEnum$Type();
