// google-ads-api-client/dist/protos/bindings/ads/enums/price_extension_price_qualifier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriceExtensionPriceQualifierEnum = exports.PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enums of price extension price qualifier.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum.PriceExtensionPriceQualifier
 */
var PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier;
(function (PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier[PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier[PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * 'From' qualifier for the price.
     *
     * @generated from protobuf enum value: FROM = 2;
     */
    PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier[PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier["FROM"] = 2] = "FROM";
    /**
     * 'Up to' qualifier for the price.
     *
     * @generated from protobuf enum value: UP_TO = 3;
     */
    PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier[PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier["UP_TO"] = 3] = "UP_TO";
    /**
     * 'Average' qualifier for the price.
     *
     * @generated from protobuf enum value: AVERAGE = 4;
     */
    PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier[PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier["AVERAGE"] = 4] = "AVERAGE";
})(PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier = exports.PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier || (exports.PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PriceExtensionPriceQualifierEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum
 */
exports.PriceExtensionPriceQualifierEnum = new PriceExtensionPriceQualifierEnum$Type();
