// google-ads-api-client/dist/protos/bindings/ads/enums/product_channel_exclusivity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductChannelExclusivityEnum = exports.ProductChannelExclusivityEnum_ProductChannelExclusivity = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the availability of a product offer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ProductChannelExclusivityEnum.ProductChannelExclusivity
 */
var ProductChannelExclusivityEnum_ProductChannelExclusivity;
(function (ProductChannelExclusivityEnum_ProductChannelExclusivity) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ProductChannelExclusivityEnum_ProductChannelExclusivity[ProductChannelExclusivityEnum_ProductChannelExclusivity["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ProductChannelExclusivityEnum_ProductChannelExclusivity[ProductChannelExclusivityEnum_ProductChannelExclusivity["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The item is sold through one channel only, either local stores or online
     * as indicated by its ProductChannel.
     *
     * @generated from protobuf enum value: SINGLE_CHANNEL = 2;
     */
    ProductChannelExclusivityEnum_ProductChannelExclusivity[ProductChannelExclusivityEnum_ProductChannelExclusivity["SINGLE_CHANNEL"] = 2] = "SINGLE_CHANNEL";
    /**
     * The item is matched to its online or local stores counterpart, indicating
     * it is available for purchase in both ShoppingProductChannels.
     *
     * @generated from protobuf enum value: MULTI_CHANNEL = 3;
     */
    ProductChannelExclusivityEnum_ProductChannelExclusivity[ProductChannelExclusivityEnum_ProductChannelExclusivity["MULTI_CHANNEL"] = 3] = "MULTI_CHANNEL";
})(ProductChannelExclusivityEnum_ProductChannelExclusivity = exports.ProductChannelExclusivityEnum_ProductChannelExclusivity || (exports.ProductChannelExclusivityEnum_ProductChannelExclusivity = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ProductChannelExclusivityEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ProductChannelExclusivityEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ProductChannelExclusivityEnum
 */
exports.ProductChannelExclusivityEnum = new ProductChannelExclusivityEnum$Type();
