// google-ads-api-client/dist/protos/bindings/ads/enums/product_bidding_category_level.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductBiddingCategoryLevelEnum = exports.ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the level of the product bidding category.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ProductBiddingCategoryLevelEnum.ProductBiddingCategoryLevel
 */
var ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel;
(function (ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel[ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel[ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Level 1.
     *
     * @generated from protobuf enum value: LEVEL1 = 2;
     */
    ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel[ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel["LEVEL1"] = 2] = "LEVEL1";
    /**
     * Level 2.
     *
     * @generated from protobuf enum value: LEVEL2 = 3;
     */
    ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel[ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel["LEVEL2"] = 3] = "LEVEL2";
    /**
     * Level 3.
     *
     * @generated from protobuf enum value: LEVEL3 = 4;
     */
    ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel[ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel["LEVEL3"] = 4] = "LEVEL3";
    /**
     * Level 4.
     *
     * @generated from protobuf enum value: LEVEL4 = 5;
     */
    ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel[ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel["LEVEL4"] = 5] = "LEVEL4";
    /**
     * Level 5.
     *
     * @generated from protobuf enum value: LEVEL5 = 6;
     */
    ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel[ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel["LEVEL5"] = 6] = "LEVEL5";
})(ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel = exports.ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel || (exports.ProductBiddingCategoryLevelEnum_ProductBiddingCategoryLevel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ProductBiddingCategoryLevelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ProductBiddingCategoryLevelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ProductBiddingCategoryLevelEnum
 */
exports.ProductBiddingCategoryLevelEnum = new ProductBiddingCategoryLevelEnum$Type();
