// google-ads-api-client/dist/protos/bindings/ads/enums/product_condition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductConditionEnum = exports.ProductConditionEnum_ProductCondition = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the condition of a product offer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ProductConditionEnum.ProductCondition
 */
var ProductConditionEnum_ProductCondition;
(function (ProductConditionEnum_ProductCondition) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ProductConditionEnum_ProductCondition[ProductConditionEnum_ProductCondition["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ProductConditionEnum_ProductCondition[ProductConditionEnum_ProductCondition["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The product condition is new.
     *
     * @generated from protobuf enum value: NEW = 3;
     */
    ProductConditionEnum_ProductCondition[ProductConditionEnum_ProductCondition["NEW"] = 3] = "NEW";
    /**
     * The product condition is refurbished.
     *
     * @generated from protobuf enum value: REFURBISHED = 4;
     */
    ProductConditionEnum_ProductCondition[ProductConditionEnum_ProductCondition["REFURBISHED"] = 4] = "REFURBISHED";
    /**
     * The product condition is used.
     *
     * @generated from protobuf enum value: USED = 5;
     */
    ProductConditionEnum_ProductCondition[ProductConditionEnum_ProductCondition["USED"] = 5] = "USED";
})(ProductConditionEnum_ProductCondition = exports.ProductConditionEnum_ProductCondition || (exports.ProductConditionEnum_ProductCondition = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ProductConditionEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ProductConditionEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ProductConditionEnum
 */
exports.ProductConditionEnum = new ProductConditionEnum$Type();
