// google-ads-api-client/dist/protos/bindings/ads/enums/product_bidding_category_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductBiddingCategoryStatusEnum = exports.ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the status of the product bidding category.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ProductBiddingCategoryStatusEnum.ProductBiddingCategoryStatus
 */
var ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus;
(function (ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus[ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus[ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The category is active and can be used for bidding.
     *
     * @generated from protobuf enum value: ACTIVE = 2;
     */
    ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus[ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus["ACTIVE"] = 2] = "ACTIVE";
    /**
     * The category is obsolete. Used only for reporting purposes.
     *
     * @generated from protobuf enum value: OBSOLETE = 3;
     */
    ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus[ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus["OBSOLETE"] = 3] = "OBSOLETE";
})(ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus = exports.ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus || (exports.ProductBiddingCategoryStatusEnum_ProductBiddingCategoryStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ProductBiddingCategoryStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ProductBiddingCategoryStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ProductBiddingCategoryStatusEnum
 */
exports.ProductBiddingCategoryStatusEnum = new ProductBiddingCategoryStatusEnum$Type();
