// google-ads-api-client/dist/protos/bindings/ads/enums/product_custom_attribute_index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductCustomAttributeIndexEnum = exports.ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The index of the product custom attribute.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ProductCustomAttributeIndexEnum.ProductCustomAttributeIndex
 */
var ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex;
(function (ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex[ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex[ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * First product custom attribute.
     *
     * @generated from protobuf enum value: INDEX0 = 7;
     */
    ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex[ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex["INDEX0"] = 7] = "INDEX0";
    /**
     * Second product custom attribute.
     *
     * @generated from protobuf enum value: INDEX1 = 8;
     */
    ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex[ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex["INDEX1"] = 8] = "INDEX1";
    /**
     * Third product custom attribute.
     *
     * @generated from protobuf enum value: INDEX2 = 9;
     */
    ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex[ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex["INDEX2"] = 9] = "INDEX2";
    /**
     * Fourth product custom attribute.
     *
     * @generated from protobuf enum value: INDEX3 = 10;
     */
    ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex[ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex["INDEX3"] = 10] = "INDEX3";
    /**
     * Fifth product custom attribute.
     *
     * @generated from protobuf enum value: INDEX4 = 11;
     */
    ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex[ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex["INDEX4"] = 11] = "INDEX4";
})(ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex = exports.ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex || (exports.ProductCustomAttributeIndexEnum_ProductCustomAttributeIndex = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ProductCustomAttributeIndexEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ProductCustomAttributeIndexEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ProductCustomAttributeIndexEnum
 */
exports.ProductCustomAttributeIndexEnum = new ProductCustomAttributeIndexEnum$Type();
