// google-ads-api-client/dist/protos/bindings/ads/enums/product_type_level.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductTypeLevelEnum = exports.ProductTypeLevelEnum_ProductTypeLevel = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the level of the type of a product offer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ProductTypeLevelEnum.ProductTypeLevel
 */
var ProductTypeLevelEnum_ProductTypeLevel;
(function (ProductTypeLevelEnum_ProductTypeLevel) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ProductTypeLevelEnum_ProductTypeLevel[ProductTypeLevelEnum_ProductTypeLevel["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ProductTypeLevelEnum_ProductTypeLevel[ProductTypeLevelEnum_ProductTypeLevel["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Level 1.
     *
     * @generated from protobuf enum value: LEVEL1 = 7;
     */
    ProductTypeLevelEnum_ProductTypeLevel[ProductTypeLevelEnum_ProductTypeLevel["LEVEL1"] = 7] = "LEVEL1";
    /**
     * Level 2.
     *
     * @generated from protobuf enum value: LEVEL2 = 8;
     */
    ProductTypeLevelEnum_ProductTypeLevel[ProductTypeLevelEnum_ProductTypeLevel["LEVEL2"] = 8] = "LEVEL2";
    /**
     * Level 3.
     *
     * @generated from protobuf enum value: LEVEL3 = 9;
     */
    ProductTypeLevelEnum_ProductTypeLevel[ProductTypeLevelEnum_ProductTypeLevel["LEVEL3"] = 9] = "LEVEL3";
    /**
     * Level 4.
     *
     * @generated from protobuf enum value: LEVEL4 = 10;
     */
    ProductTypeLevelEnum_ProductTypeLevel[ProductTypeLevelEnum_ProductTypeLevel["LEVEL4"] = 10] = "LEVEL4";
    /**
     * Level 5.
     *
     * @generated from protobuf enum value: LEVEL5 = 11;
     */
    ProductTypeLevelEnum_ProductTypeLevel[ProductTypeLevelEnum_ProductTypeLevel["LEVEL5"] = 11] = "LEVEL5";
})(ProductTypeLevelEnum_ProductTypeLevel = exports.ProductTypeLevelEnum_ProductTypeLevel || (exports.ProductTypeLevelEnum_ProductTypeLevel = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ProductTypeLevelEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ProductTypeLevelEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ProductTypeLevelEnum
 */
exports.ProductTypeLevelEnum = new ProductTypeLevelEnum$Type();
