// google-ads-api-client/dist/protos/bindings/ads/enums/promotion_extension_discount_modifier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromotionExtensionDiscountModifierEnum = exports.PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * A promotion extension discount modifier.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum.PromotionExtensionDiscountModifier
 */
var PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier;
(function (PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier[PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier[PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * 'Up to'.
     *
     * @generated from protobuf enum value: UP_TO = 2;
     */
    PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier[PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier["UP_TO"] = 2] = "UP_TO";
})(PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier = exports.PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier || (exports.PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PromotionExtensionDiscountModifierEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum
 */
exports.PromotionExtensionDiscountModifierEnum = new PromotionExtensionDiscountModifierEnum$Type();
