// google-ads-api-client/dist/protos/bindings/ads/enums/promotion_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromotionPlaceholderFieldEnum = exports.PromotionPlaceholderFieldEnum_PromotionPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Promotion placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PromotionPlaceholderFieldEnum.PromotionPlaceholderField
 */
var PromotionPlaceholderFieldEnum_PromotionPlaceholderField;
(function (PromotionPlaceholderFieldEnum_PromotionPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. The text that appears on the ad when the extension is
     * shown.
     *
     * @generated from protobuf enum value: PROMOTION_TARGET = 2;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["PROMOTION_TARGET"] = 2] = "PROMOTION_TARGET";
    /**
     * Data Type: STRING. Lets you add "up to" phrase to the promotion,
     * in case you have variable promotion rates.
     *
     * @generated from protobuf enum value: DISCOUNT_MODIFIER = 3;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["DISCOUNT_MODIFIER"] = 3] = "DISCOUNT_MODIFIER";
    /**
     * Data Type: INT64. Takes a value in micros, where 1 million micros
     * represents 1%, and is shown as a percentage when rendered.
     *
     * @generated from protobuf enum value: PERCENT_OFF = 4;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["PERCENT_OFF"] = 4] = "PERCENT_OFF";
    /**
     * Data Type: MONEY. Requires a currency and an amount of money.
     *
     * @generated from protobuf enum value: MONEY_AMOUNT_OFF = 5;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["MONEY_AMOUNT_OFF"] = 5] = "MONEY_AMOUNT_OFF";
    /**
     * Data Type: STRING. A string that the user enters to get the discount.
     *
     * @generated from protobuf enum value: PROMOTION_CODE = 6;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["PROMOTION_CODE"] = 6] = "PROMOTION_CODE";
    /**
     * Data Type: MONEY. A minimum spend before the user qualifies for the
     * promotion.
     *
     * @generated from protobuf enum value: ORDERS_OVER_AMOUNT = 7;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["ORDERS_OVER_AMOUNT"] = 7] = "ORDERS_OVER_AMOUNT";
    /**
     * Data Type: DATE. The start date of the promotion.
     *
     * @generated from protobuf enum value: PROMOTION_START = 8;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["PROMOTION_START"] = 8] = "PROMOTION_START";
    /**
     * Data Type: DATE. The end date of the promotion.
     *
     * @generated from protobuf enum value: PROMOTION_END = 9;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["PROMOTION_END"] = 9] = "PROMOTION_END";
    /**
     * Data Type: STRING. Describes the associated event for the promotion using
     * one of the PromotionExtensionOccasion enum values, for example NEW_YEARS.
     *
     * @generated from protobuf enum value: OCCASION = 10;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["OCCASION"] = 10] = "OCCASION";
    /**
     * Data Type: URL_LIST. Final URLs to be used in the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_URLS = 11;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["FINAL_URLS"] = 11] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 12;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["FINAL_MOBILE_URLS"] = 12] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 13;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["TRACKING_URL"] = 13] = "TRACKING_URL";
    /**
     * Data Type: STRING. A string represented by a language code for the
     * promotion.
     *
     * @generated from protobuf enum value: LANGUAGE = 14;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["LANGUAGE"] = 14] = "LANGUAGE";
    /**
     * Data Type: STRING. Final URL suffix for the ad when using parallel
     * tracking.
     *
     * @generated from protobuf enum value: FINAL_URL_SUFFIX = 15;
     */
    PromotionPlaceholderFieldEnum_PromotionPlaceholderField[PromotionPlaceholderFieldEnum_PromotionPlaceholderField["FINAL_URL_SUFFIX"] = 15] = "FINAL_URL_SUFFIX";
})(PromotionPlaceholderFieldEnum_PromotionPlaceholderField = exports.PromotionPlaceholderFieldEnum_PromotionPlaceholderField || (exports.PromotionPlaceholderFieldEnum_PromotionPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PromotionPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PromotionPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PromotionPlaceholderFieldEnum
 */
exports.PromotionPlaceholderFieldEnum = new PromotionPlaceholderFieldEnum$Type();
