// google-ads-api-client/dist/protos/bindings/ads/enums/promotion_extension_occasion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromotionExtensionOccasionEnum = exports.PromotionExtensionOccasionEnum_PromotionExtensionOccasion = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * A promotion extension occasion.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum.PromotionExtensionOccasion
 */
var PromotionExtensionOccasionEnum_PromotionExtensionOccasion;
(function (PromotionExtensionOccasionEnum_PromotionExtensionOccasion) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * New Year's.
     *
     * @generated from protobuf enum value: NEW_YEARS = 2;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["NEW_YEARS"] = 2] = "NEW_YEARS";
    /**
     * Chinese New Year.
     *
     * @generated from protobuf enum value: CHINESE_NEW_YEAR = 3;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["CHINESE_NEW_YEAR"] = 3] = "CHINESE_NEW_YEAR";
    /**
     * Valentine's Day.
     *
     * @generated from protobuf enum value: VALENTINES_DAY = 4;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["VALENTINES_DAY"] = 4] = "VALENTINES_DAY";
    /**
     * Easter.
     *
     * @generated from protobuf enum value: EASTER = 5;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["EASTER"] = 5] = "EASTER";
    /**
     * Mother's Day.
     *
     * @generated from protobuf enum value: MOTHERS_DAY = 6;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["MOTHERS_DAY"] = 6] = "MOTHERS_DAY";
    /**
     * Father's Day.
     *
     * @generated from protobuf enum value: FATHERS_DAY = 7;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["FATHERS_DAY"] = 7] = "FATHERS_DAY";
    /**
     * Labor Day.
     *
     * @generated from protobuf enum value: LABOR_DAY = 8;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["LABOR_DAY"] = 8] = "LABOR_DAY";
    /**
     * Back To School.
     *
     * @generated from protobuf enum value: BACK_TO_SCHOOL = 9;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["BACK_TO_SCHOOL"] = 9] = "BACK_TO_SCHOOL";
    /**
     * Halloween.
     *
     * @generated from protobuf enum value: HALLOWEEN = 10;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["HALLOWEEN"] = 10] = "HALLOWEEN";
    /**
     * Black Friday.
     *
     * @generated from protobuf enum value: BLACK_FRIDAY = 11;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["BLACK_FRIDAY"] = 11] = "BLACK_FRIDAY";
    /**
     * Cyber Monday.
     *
     * @generated from protobuf enum value: CYBER_MONDAY = 12;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["CYBER_MONDAY"] = 12] = "CYBER_MONDAY";
    /**
     * Christmas.
     *
     * @generated from protobuf enum value: CHRISTMAS = 13;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["CHRISTMAS"] = 13] = "CHRISTMAS";
    /**
     * Boxing Day.
     *
     * @generated from protobuf enum value: BOXING_DAY = 14;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["BOXING_DAY"] = 14] = "BOXING_DAY";
    /**
     * Independence Day in any country.
     *
     * @generated from protobuf enum value: INDEPENDENCE_DAY = 15;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["INDEPENDENCE_DAY"] = 15] = "INDEPENDENCE_DAY";
    /**
     * National Day in any country.
     *
     * @generated from protobuf enum value: NATIONAL_DAY = 16;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["NATIONAL_DAY"] = 16] = "NATIONAL_DAY";
    /**
     * End of any season.
     *
     * @generated from protobuf enum value: END_OF_SEASON = 17;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["END_OF_SEASON"] = 17] = "END_OF_SEASON";
    /**
     * Winter Sale.
     *
     * @generated from protobuf enum value: WINTER_SALE = 18;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["WINTER_SALE"] = 18] = "WINTER_SALE";
    /**
     * Summer sale.
     *
     * @generated from protobuf enum value: SUMMER_SALE = 19;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["SUMMER_SALE"] = 19] = "SUMMER_SALE";
    /**
     * Fall Sale.
     *
     * @generated from protobuf enum value: FALL_SALE = 20;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["FALL_SALE"] = 20] = "FALL_SALE";
    /**
     * Spring Sale.
     *
     * @generated from protobuf enum value: SPRING_SALE = 21;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["SPRING_SALE"] = 21] = "SPRING_SALE";
    /**
     * Ramadan.
     *
     * @generated from protobuf enum value: RAMADAN = 22;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["RAMADAN"] = 22] = "RAMADAN";
    /**
     * Eid al-Fitr.
     *
     * @generated from protobuf enum value: EID_AL_FITR = 23;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["EID_AL_FITR"] = 23] = "EID_AL_FITR";
    /**
     * Eid al-Adha.
     *
     * @generated from protobuf enum value: EID_AL_ADHA = 24;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["EID_AL_ADHA"] = 24] = "EID_AL_ADHA";
    /**
     * Singles Day.
     *
     * @generated from protobuf enum value: SINGLES_DAY = 25;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["SINGLES_DAY"] = 25] = "SINGLES_DAY";
    /**
     * Women's Day.
     *
     * @generated from protobuf enum value: WOMENS_DAY = 26;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["WOMENS_DAY"] = 26] = "WOMENS_DAY";
    /**
     * Holi.
     *
     * @generated from protobuf enum value: HOLI = 27;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["HOLI"] = 27] = "HOLI";
    /**
     * Parent's Day.
     *
     * @generated from protobuf enum value: PARENTS_DAY = 28;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["PARENTS_DAY"] = 28] = "PARENTS_DAY";
    /**
     * St. Nicholas Day.
     *
     * @generated from protobuf enum value: ST_NICHOLAS_DAY = 29;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["ST_NICHOLAS_DAY"] = 29] = "ST_NICHOLAS_DAY";
    /**
     * Carnival.
     *
     * @generated from protobuf enum value: CARNIVAL = 30;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["CARNIVAL"] = 30] = "CARNIVAL";
    /**
     * Epiphany, also known as Three Kings' Day.
     *
     * @generated from protobuf enum value: EPIPHANY = 31;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["EPIPHANY"] = 31] = "EPIPHANY";
    /**
     * Rosh Hashanah.
     *
     * @generated from protobuf enum value: ROSH_HASHANAH = 32;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["ROSH_HASHANAH"] = 32] = "ROSH_HASHANAH";
    /**
     * Passover.
     *
     * @generated from protobuf enum value: PASSOVER = 33;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["PASSOVER"] = 33] = "PASSOVER";
    /**
     * Hanukkah.
     *
     * @generated from protobuf enum value: HANUKKAH = 34;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["HANUKKAH"] = 34] = "HANUKKAH";
    /**
     * Diwali.
     *
     * @generated from protobuf enum value: DIWALI = 35;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["DIWALI"] = 35] = "DIWALI";
    /**
     * Navratri.
     *
     * @generated from protobuf enum value: NAVRATRI = 36;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["NAVRATRI"] = 36] = "NAVRATRI";
    /**
     * Available in Thai: Songkran.
     *
     * @generated from protobuf enum value: SONGKRAN = 37;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["SONGKRAN"] = 37] = "SONGKRAN";
    /**
     * Available in Japanese: Year-end Gift.
     *
     * @generated from protobuf enum value: YEAR_END_GIFT = 38;
     */
    PromotionExtensionOccasionEnum_PromotionExtensionOccasion[PromotionExtensionOccasionEnum_PromotionExtensionOccasion["YEAR_END_GIFT"] = 38] = "YEAR_END_GIFT";
})(PromotionExtensionOccasionEnum_PromotionExtensionOccasion = exports.PromotionExtensionOccasionEnum_PromotionExtensionOccasion || (exports.PromotionExtensionOccasionEnum_PromotionExtensionOccasion = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PromotionExtensionOccasionEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum
 */
exports.PromotionExtensionOccasionEnum = new PromotionExtensionOccasionEnum$Type();
