// google-ads-api-client/dist/protos/bindings/ads/enums/proximity_radius_units.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProximityRadiusUnitsEnum = exports.ProximityRadiusUnitsEnum_ProximityRadiusUnits = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The unit of radius distance in proximity (for example, MILES)
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ProximityRadiusUnitsEnum.ProximityRadiusUnits
 */
var ProximityRadiusUnitsEnum_ProximityRadiusUnits;
(function (ProximityRadiusUnitsEnum_ProximityRadiusUnits) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ProximityRadiusUnitsEnum_ProximityRadiusUnits[ProximityRadiusUnitsEnum_ProximityRadiusUnits["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ProximityRadiusUnitsEnum_ProximityRadiusUnits[ProximityRadiusUnitsEnum_ProximityRadiusUnits["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Miles
     *
     * @generated from protobuf enum value: MILES = 2;
     */
    ProximityRadiusUnitsEnum_ProximityRadiusUnits[ProximityRadiusUnitsEnum_ProximityRadiusUnits["MILES"] = 2] = "MILES";
    /**
     * Kilometers
     *
     * @generated from protobuf enum value: KILOMETERS = 3;
     */
    ProximityRadiusUnitsEnum_ProximityRadiusUnits[ProximityRadiusUnitsEnum_ProximityRadiusUnits["KILOMETERS"] = 3] = "KILOMETERS";
})(ProximityRadiusUnitsEnum_ProximityRadiusUnits = exports.ProximityRadiusUnitsEnum_ProximityRadiusUnits || (exports.ProximityRadiusUnitsEnum_ProximityRadiusUnits = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ProximityRadiusUnitsEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ProximityRadiusUnitsEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ProximityRadiusUnitsEnum
 */
exports.ProximityRadiusUnitsEnum = new ProximityRadiusUnitsEnum$Type();
