// google-ads-api-client/dist/protos/bindings/ads/enums/quality_score_bucket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QualityScoreBucketEnum = exports.QualityScoreBucketEnum_QualityScoreBucket = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the possible quality score buckets.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket
 */
var QualityScoreBucketEnum_QualityScoreBucket;
(function (QualityScoreBucketEnum_QualityScoreBucket) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    QualityScoreBucketEnum_QualityScoreBucket[QualityScoreBucketEnum_QualityScoreBucket["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    QualityScoreBucketEnum_QualityScoreBucket[QualityScoreBucketEnum_QualityScoreBucket["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Quality of the creative is below average.
     *
     * @generated from protobuf enum value: BELOW_AVERAGE = 2;
     */
    QualityScoreBucketEnum_QualityScoreBucket[QualityScoreBucketEnum_QualityScoreBucket["BELOW_AVERAGE"] = 2] = "BELOW_AVERAGE";
    /**
     * Quality of the creative is average.
     *
     * @generated from protobuf enum value: AVERAGE = 3;
     */
    QualityScoreBucketEnum_QualityScoreBucket[QualityScoreBucketEnum_QualityScoreBucket["AVERAGE"] = 3] = "AVERAGE";
    /**
     * Quality of the creative is above average.
     *
     * @generated from protobuf enum value: ABOVE_AVERAGE = 4;
     */
    QualityScoreBucketEnum_QualityScoreBucket[QualityScoreBucketEnum_QualityScoreBucket["ABOVE_AVERAGE"] = 4] = "ABOVE_AVERAGE";
})(QualityScoreBucketEnum_QualityScoreBucket = exports.QualityScoreBucketEnum_QualityScoreBucket || (exports.QualityScoreBucketEnum_QualityScoreBucket = {}));
// @generated message type with reflection information, may provide speed optimized methods
class QualityScoreBucketEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.QualityScoreBucketEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.QualityScoreBucketEnum
 */
exports.QualityScoreBucketEnum = new QualityScoreBucketEnum$Type();
