// google-ads-api-client/dist/protos/bindings/ads/enums/reach_plan_age_range.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReachPlanAgeRangeEnum = exports.ReachPlanAgeRangeEnum_ReachPlanAgeRange = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible plannable age range values.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ReachPlanAgeRangeEnum.ReachPlanAgeRange
 */
var ReachPlanAgeRangeEnum_ReachPlanAgeRange;
(function (ReachPlanAgeRangeEnum_ReachPlanAgeRange) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Between 18 and 24 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_18_24 = 503001;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_18_24"] = 503001] = "AGE_RANGE_18_24";
    /**
     * Between 18 and 34 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_18_34 = 2;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_18_34"] = 2] = "AGE_RANGE_18_34";
    /**
     * Between 18 and 44 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_18_44 = 3;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_18_44"] = 3] = "AGE_RANGE_18_44";
    /**
     * Between 18 and 49 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_18_49 = 4;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_18_49"] = 4] = "AGE_RANGE_18_49";
    /**
     * Between 18 and 54 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_18_54 = 5;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_18_54"] = 5] = "AGE_RANGE_18_54";
    /**
     * Between 18 and 64 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_18_64 = 6;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_18_64"] = 6] = "AGE_RANGE_18_64";
    /**
     * Between 18 and 65+ years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_18_65_UP = 7;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_18_65_UP"] = 7] = "AGE_RANGE_18_65_UP";
    /**
     * Between 21 and 34 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_21_34 = 8;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_21_34"] = 8] = "AGE_RANGE_21_34";
    /**
     * Between 25 and 34 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_25_34 = 503002;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_25_34"] = 503002] = "AGE_RANGE_25_34";
    /**
     * Between 25 and 44 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_25_44 = 9;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_25_44"] = 9] = "AGE_RANGE_25_44";
    /**
     * Between 25 and 49 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_25_49 = 10;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_25_49"] = 10] = "AGE_RANGE_25_49";
    /**
     * Between 25 and 54 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_25_54 = 11;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_25_54"] = 11] = "AGE_RANGE_25_54";
    /**
     * Between 25 and 64 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_25_64 = 12;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_25_64"] = 12] = "AGE_RANGE_25_64";
    /**
     * Between 25 and 65+ years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_25_65_UP = 13;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_25_65_UP"] = 13] = "AGE_RANGE_25_65_UP";
    /**
     * Between 35 and 44 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_35_44 = 503003;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_35_44"] = 503003] = "AGE_RANGE_35_44";
    /**
     * Between 35 and 49 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_35_49 = 14;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_35_49"] = 14] = "AGE_RANGE_35_49";
    /**
     * Between 35 and 54 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_35_54 = 15;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_35_54"] = 15] = "AGE_RANGE_35_54";
    /**
     * Between 35 and 64 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_35_64 = 16;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_35_64"] = 16] = "AGE_RANGE_35_64";
    /**
     * Between 35 and 65+ years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_35_65_UP = 17;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_35_65_UP"] = 17] = "AGE_RANGE_35_65_UP";
    /**
     * Between 45 and 54 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_45_54 = 503004;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_45_54"] = 503004] = "AGE_RANGE_45_54";
    /**
     * Between 45 and 64 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_45_64 = 18;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_45_64"] = 18] = "AGE_RANGE_45_64";
    /**
     * Between 45 and 65+ years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_45_65_UP = 19;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_45_65_UP"] = 19] = "AGE_RANGE_45_65_UP";
    /**
     * Between 50 and 65+ years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_50_65_UP = 20;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_50_65_UP"] = 20] = "AGE_RANGE_50_65_UP";
    /**
     * Between 55 and 64 years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_55_64 = 503005;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_55_64"] = 503005] = "AGE_RANGE_55_64";
    /**
     * Between 55 and 65+ years old.
     *
     * @generated from protobuf enum value: AGE_RANGE_55_65_UP = 21;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_55_65_UP"] = 21] = "AGE_RANGE_55_65_UP";
    /**
     * 65 years old and beyond.
     *
     * @generated from protobuf enum value: AGE_RANGE_65_UP = 503006;
     */
    ReachPlanAgeRangeEnum_ReachPlanAgeRange[ReachPlanAgeRangeEnum_ReachPlanAgeRange["AGE_RANGE_65_UP"] = 503006] = "AGE_RANGE_65_UP";
})(ReachPlanAgeRangeEnum_ReachPlanAgeRange = exports.ReachPlanAgeRangeEnum_ReachPlanAgeRange || (exports.ReachPlanAgeRangeEnum_ReachPlanAgeRange = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ReachPlanAgeRangeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ReachPlanAgeRangeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ReachPlanAgeRangeEnum
 */
exports.ReachPlanAgeRangeEnum = new ReachPlanAgeRangeEnum$Type();
