// google-ads-api-client/dist/protos/bindings/ads/enums/reach_plan_network.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReachPlanNetworkEnum = exports.ReachPlanNetworkEnum_ReachPlanNetwork = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible plannable network values.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ReachPlanNetworkEnum.ReachPlanNetwork
 */
var ReachPlanNetworkEnum_ReachPlanNetwork;
(function (ReachPlanNetworkEnum_ReachPlanNetwork) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ReachPlanNetworkEnum_ReachPlanNetwork[ReachPlanNetworkEnum_ReachPlanNetwork["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used as a return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ReachPlanNetworkEnum_ReachPlanNetwork[ReachPlanNetworkEnum_ReachPlanNetwork["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * YouTube network.
     *
     * @generated from protobuf enum value: YOUTUBE = 2;
     */
    ReachPlanNetworkEnum_ReachPlanNetwork[ReachPlanNetworkEnum_ReachPlanNetwork["YOUTUBE"] = 2] = "YOUTUBE";
    /**
     * Google Video Partners (GVP) network.
     *
     * @generated from protobuf enum value: GOOGLE_VIDEO_PARTNERS = 3;
     */
    ReachPlanNetworkEnum_ReachPlanNetwork[ReachPlanNetworkEnum_ReachPlanNetwork["GOOGLE_VIDEO_PARTNERS"] = 3] = "GOOGLE_VIDEO_PARTNERS";
    /**
     * A combination of the YouTube network and the Google Video Partners
     * network.
     *
     * @generated from protobuf enum value: YOUTUBE_AND_GOOGLE_VIDEO_PARTNERS = 4;
     */
    ReachPlanNetworkEnum_ReachPlanNetwork[ReachPlanNetworkEnum_ReachPlanNetwork["YOUTUBE_AND_GOOGLE_VIDEO_PARTNERS"] = 4] = "YOUTUBE_AND_GOOGLE_VIDEO_PARTNERS";
})(ReachPlanNetworkEnum_ReachPlanNetwork = exports.ReachPlanNetworkEnum_ReachPlanNetwork || (exports.ReachPlanNetworkEnum_ReachPlanNetwork = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ReachPlanNetworkEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ReachPlanNetworkEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ReachPlanNetworkEnum
 */
exports.ReachPlanNetworkEnum = new ReachPlanNetworkEnum$Type();
