// google-ads-api-client/dist/protos/bindings/ads/enums/reach_plan_ad_length.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReachPlanAdLengthEnum = exports.ReachPlanAdLengthEnum_ReachPlanAdLength = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible ad length values.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ReachPlanAdLengthEnum.ReachPlanAdLength
 */
var ReachPlanAdLengthEnum_ReachPlanAdLength;
(function (ReachPlanAdLengthEnum_ReachPlanAdLength) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ReachPlanAdLengthEnum_ReachPlanAdLength[ReachPlanAdLengthEnum_ReachPlanAdLength["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ReachPlanAdLengthEnum_ReachPlanAdLength[ReachPlanAdLengthEnum_ReachPlanAdLength["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * 6 seconds long ad.
     *
     * @generated from protobuf enum value: SIX_SECONDS = 2;
     */
    ReachPlanAdLengthEnum_ReachPlanAdLength[ReachPlanAdLengthEnum_ReachPlanAdLength["SIX_SECONDS"] = 2] = "SIX_SECONDS";
    /**
     * 15 or 20 seconds long ad.
     *
     * @generated from protobuf enum value: FIFTEEN_OR_TWENTY_SECONDS = 3;
     */
    ReachPlanAdLengthEnum_ReachPlanAdLength[ReachPlanAdLengthEnum_ReachPlanAdLength["FIFTEEN_OR_TWENTY_SECONDS"] = 3] = "FIFTEEN_OR_TWENTY_SECONDS";
    /**
     * More than 20 seconds long ad.
     *
     * @generated from protobuf enum value: TWENTY_SECONDS_OR_MORE = 4;
     */
    ReachPlanAdLengthEnum_ReachPlanAdLength[ReachPlanAdLengthEnum_ReachPlanAdLength["TWENTY_SECONDS_OR_MORE"] = 4] = "TWENTY_SECONDS_OR_MORE";
})(ReachPlanAdLengthEnum_ReachPlanAdLength = exports.ReachPlanAdLengthEnum_ReachPlanAdLength || (exports.ReachPlanAdLengthEnum_ReachPlanAdLength = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ReachPlanAdLengthEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ReachPlanAdLengthEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ReachPlanAdLengthEnum
 */
exports.ReachPlanAdLengthEnum = new ReachPlanAdLengthEnum$Type();
