// google-ads-api-client/dist/protos/bindings/ads/enums/real_estate_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RealEstatePlaceholderFieldEnum = exports.RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Real Estate placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.RealEstatePlaceholderFieldEnum.RealEstatePlaceholderField
 */
var RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField;
(function (RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Unique ID.
     *
     * @generated from protobuf enum value: LISTING_ID = 2;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["LISTING_ID"] = 2] = "LISTING_ID";
    /**
     * Data Type: STRING. Main headline with listing name to be shown in dynamic
     * ad.
     *
     * @generated from protobuf enum value: LISTING_NAME = 3;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["LISTING_NAME"] = 3] = "LISTING_NAME";
    /**
     * Data Type: STRING. City name to be shown in dynamic ad.
     *
     * @generated from protobuf enum value: CITY_NAME = 4;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["CITY_NAME"] = 4] = "CITY_NAME";
    /**
     * Data Type: STRING. Description of listing to be shown in dynamic ad.
     *
     * @generated from protobuf enum value: DESCRIPTION = 5;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["DESCRIPTION"] = 5] = "DESCRIPTION";
    /**
     * Data Type: STRING. Complete listing address, including postal code.
     *
     * @generated from protobuf enum value: ADDRESS = 6;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["ADDRESS"] = 6] = "ADDRESS";
    /**
     * Data Type: STRING. Price to be shown in the ad.
     * Example: "100.00 USD"
     *
     * @generated from protobuf enum value: PRICE = 7;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["PRICE"] = 7] = "PRICE";
    /**
     * Data Type: STRING. Formatted price to be shown in the ad.
     * Example: "Starting at $100.00 USD", "$80 - $100"
     *
     * @generated from protobuf enum value: FORMATTED_PRICE = 8;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["FORMATTED_PRICE"] = 8] = "FORMATTED_PRICE";
    /**
     * Data Type: URL. Image to be displayed in the ad.
     *
     * @generated from protobuf enum value: IMAGE_URL = 9;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["IMAGE_URL"] = 9] = "IMAGE_URL";
    /**
     * Data Type: STRING. Type of property (house, condo, apartment, etc.) used
     * to group like items together for recommendation engine.
     *
     * @generated from protobuf enum value: PROPERTY_TYPE = 10;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["PROPERTY_TYPE"] = 10] = "PROPERTY_TYPE";
    /**
     * Data Type: STRING. Type of listing (resale, rental, foreclosure, etc.)
     * used to group like items together for recommendation engine.
     *
     * @generated from protobuf enum value: LISTING_TYPE = 11;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["LISTING_TYPE"] = 11] = "LISTING_TYPE";
    /**
     * Data Type: STRING_LIST. Keywords used for product retrieval.
     *
     * @generated from protobuf enum value: CONTEXTUAL_KEYWORDS = 12;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["CONTEXTUAL_KEYWORDS"] = 12] = "CONTEXTUAL_KEYWORDS";
    /**
     * Data Type: URL_LIST. Final URLs to be used in ad when using Upgraded
     * URLs; the more specific the better (for example, the individual URL of a
     * specific listing and its location).
     *
     * @generated from protobuf enum value: FINAL_URLS = 13;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["FINAL_URLS"] = 13] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 14;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["FINAL_MOBILE_URLS"] = 14] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 15;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["TRACKING_URL"] = 15] = "TRACKING_URL";
    /**
     * Data Type: STRING. Android app link. Must be formatted as:
     * android-app://{package_id}/{scheme}/{host_path}.
     * The components are defined as follows:
     * package_id: app ID as specified in Google Play.
     * scheme: the scheme to pass to the application. Can be HTTP, or a custom
     *   scheme.
     * host_path: identifies the specific content within your application.
     *
     * @generated from protobuf enum value: ANDROID_APP_LINK = 16;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["ANDROID_APP_LINK"] = 16] = "ANDROID_APP_LINK";
    /**
     * Data Type: STRING_LIST. List of recommended listing IDs to show together
     * with this item.
     *
     * @generated from protobuf enum value: SIMILAR_LISTING_IDS = 17;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["SIMILAR_LISTING_IDS"] = 17] = "SIMILAR_LISTING_IDS";
    /**
     * Data Type: STRING. iOS app link.
     *
     * @generated from protobuf enum value: IOS_APP_LINK = 18;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["IOS_APP_LINK"] = 18] = "IOS_APP_LINK";
    /**
     * Data Type: INT64. iOS app store ID.
     *
     * @generated from protobuf enum value: IOS_APP_STORE_ID = 19;
     */
    RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField[RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField["IOS_APP_STORE_ID"] = 19] = "IOS_APP_STORE_ID";
})(RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField = exports.RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField || (exports.RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class RealEstatePlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.RealEstatePlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.RealEstatePlaceholderFieldEnum
 */
exports.RealEstatePlaceholderFieldEnum = new RealEstatePlaceholderFieldEnum$Type();
