// google-ads-api-client/dist/protos/bindings/ads/enums/recommendation_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecommendationTypeEnum = exports.RecommendationTypeEnum_RecommendationType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Types of recommendations.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.RecommendationTypeEnum.RecommendationType
 */
var RecommendationTypeEnum_RecommendationType;
(function (RecommendationTypeEnum_RecommendationType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Budget recommendation for campaigns that are currently budget-constrained
     * (as opposed to the FORECASTING_CAMPAIGN_BUDGET recommendation, which
     * applies to campaigns that are expected to become budget-constrained in
     * the future).
     *
     * @generated from protobuf enum value: CAMPAIGN_BUDGET = 2;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["CAMPAIGN_BUDGET"] = 2] = "CAMPAIGN_BUDGET";
    /**
     * Keyword recommendation.
     *
     * @generated from protobuf enum value: KEYWORD = 3;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["KEYWORD"] = 3] = "KEYWORD";
    /**
     * Recommendation to add a new text ad.
     *
     * @generated from protobuf enum value: TEXT_AD = 4;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["TEXT_AD"] = 4] = "TEXT_AD";
    /**
     * Recommendation to update a campaign to use a Target CPA bidding strategy.
     *
     * @generated from protobuf enum value: TARGET_CPA_OPT_IN = 5;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["TARGET_CPA_OPT_IN"] = 5] = "TARGET_CPA_OPT_IN";
    /**
     * Recommendation to update a campaign to use the Maximize Conversions
     * bidding strategy.
     *
     * @generated from protobuf enum value: MAXIMIZE_CONVERSIONS_OPT_IN = 6;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["MAXIMIZE_CONVERSIONS_OPT_IN"] = 6] = "MAXIMIZE_CONVERSIONS_OPT_IN";
    /**
     * Recommendation to enable Enhanced Cost Per Click for a campaign.
     *
     * @generated from protobuf enum value: ENHANCED_CPC_OPT_IN = 7;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["ENHANCED_CPC_OPT_IN"] = 7] = "ENHANCED_CPC_OPT_IN";
    /**
     * Recommendation to start showing your campaign's ads on Google Search
     * Partners Websites.
     *
     * @generated from protobuf enum value: SEARCH_PARTNERS_OPT_IN = 8;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["SEARCH_PARTNERS_OPT_IN"] = 8] = "SEARCH_PARTNERS_OPT_IN";
    /**
     * Recommendation to update a campaign to use a Maximize Clicks bidding
     * strategy.
     *
     * @generated from protobuf enum value: MAXIMIZE_CLICKS_OPT_IN = 9;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["MAXIMIZE_CLICKS_OPT_IN"] = 9] = "MAXIMIZE_CLICKS_OPT_IN";
    /**
     * Recommendation to start using the "Optimize" ad rotation setting for the
     * given ad group.
     *
     * @generated from protobuf enum value: OPTIMIZE_AD_ROTATION = 10;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["OPTIMIZE_AD_ROTATION"] = 10] = "OPTIMIZE_AD_ROTATION";
    /**
     * Recommendation to add callout extensions to a campaign.
     *
     * @generated from protobuf enum value: CALLOUT_EXTENSION = 11;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["CALLOUT_EXTENSION"] = 11] = "CALLOUT_EXTENSION";
    /**
     * Recommendation to add sitelink extensions to a campaign.
     *
     * @generated from protobuf enum value: SITELINK_EXTENSION = 12;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["SITELINK_EXTENSION"] = 12] = "SITELINK_EXTENSION";
    /**
     * Recommendation to add call extensions to a campaign.
     *
     * @generated from protobuf enum value: CALL_EXTENSION = 13;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["CALL_EXTENSION"] = 13] = "CALL_EXTENSION";
    /**
     * Recommendation to change an existing keyword from one match type to a
     * broader match type.
     *
     * @generated from protobuf enum value: KEYWORD_MATCH_TYPE = 14;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["KEYWORD_MATCH_TYPE"] = 14] = "KEYWORD_MATCH_TYPE";
    /**
     * Recommendation to move unused budget from one budget to a constrained
     * budget.
     *
     * @generated from protobuf enum value: MOVE_UNUSED_BUDGET = 15;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["MOVE_UNUSED_BUDGET"] = 15] = "MOVE_UNUSED_BUDGET";
    /**
     * Budget recommendation for campaigns that are expected to become
     * budget-constrained in the future (as opposed to the CAMPAIGN_BUDGET
     * recommendation, which applies to campaigns that are currently
     * budget-constrained).
     *
     * @generated from protobuf enum value: FORECASTING_CAMPAIGN_BUDGET = 16;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["FORECASTING_CAMPAIGN_BUDGET"] = 16] = "FORECASTING_CAMPAIGN_BUDGET";
    /**
     * Recommendation to update a campaign to use a Target ROAS bidding
     * strategy.
     *
     * @generated from protobuf enum value: TARGET_ROAS_OPT_IN = 17;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["TARGET_ROAS_OPT_IN"] = 17] = "TARGET_ROAS_OPT_IN";
    /**
     * Recommendation to add a new responsive search ad.
     *
     * @generated from protobuf enum value: RESPONSIVE_SEARCH_AD = 18;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["RESPONSIVE_SEARCH_AD"] = 18] = "RESPONSIVE_SEARCH_AD";
    /**
     * Budget recommendation for campaigns whose ROI is predicted to increase
     * with a budget adjustment.
     *
     * @generated from protobuf enum value: MARGINAL_ROI_CAMPAIGN_BUDGET = 19;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["MARGINAL_ROI_CAMPAIGN_BUDGET"] = 19] = "MARGINAL_ROI_CAMPAIGN_BUDGET";
    /**
     * Recommendation to expand keywords to broad match for fully automated
     * conversion-based bidding campaigns.
     *
     * @generated from protobuf enum value: USE_BROAD_MATCH_KEYWORD = 20;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["USE_BROAD_MATCH_KEYWORD"] = 20] = "USE_BROAD_MATCH_KEYWORD";
    /**
     * Recommendation to add new responsive search ad assets.
     *
     * @generated from protobuf enum value: RESPONSIVE_SEARCH_AD_ASSET = 21;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["RESPONSIVE_SEARCH_AD_ASSET"] = 21] = "RESPONSIVE_SEARCH_AD_ASSET";
    /**
     * Recommendation to upgrade a Smart Shopping campaign to a Performance Max
     * campaign.
     *
     * @generated from protobuf enum value: UPGRADE_SMART_SHOPPING_CAMPAIGN_TO_PERFORMANCE_MAX = 22;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["UPGRADE_SMART_SHOPPING_CAMPAIGN_TO_PERFORMANCE_MAX"] = 22] = "UPGRADE_SMART_SHOPPING_CAMPAIGN_TO_PERFORMANCE_MAX";
    /**
     * Recommendation to improve strength of responsive search ad.
     *
     * @generated from protobuf enum value: RESPONSIVE_SEARCH_AD_IMPROVE_AD_STRENGTH = 23;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["RESPONSIVE_SEARCH_AD_IMPROVE_AD_STRENGTH"] = 23] = "RESPONSIVE_SEARCH_AD_IMPROVE_AD_STRENGTH";
    /**
     * Recommendation to update a campaign to use Display Expansion.
     *
     * @generated from protobuf enum value: DISPLAY_EXPANSION_OPT_IN = 24;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["DISPLAY_EXPANSION_OPT_IN"] = 24] = "DISPLAY_EXPANSION_OPT_IN";
    /**
     * Recommendation to upgrade a Local campaign to a Performance Max
     * campaign.
     *
     * @generated from protobuf enum value: UPGRADE_LOCAL_CAMPAIGN_TO_PERFORMANCE_MAX = 25;
     */
    RecommendationTypeEnum_RecommendationType[RecommendationTypeEnum_RecommendationType["UPGRADE_LOCAL_CAMPAIGN_TO_PERFORMANCE_MAX"] = 25] = "UPGRADE_LOCAL_CAMPAIGN_TO_PERFORMANCE_MAX";
})(RecommendationTypeEnum_RecommendationType = exports.RecommendationTypeEnum_RecommendationType || (exports.RecommendationTypeEnum_RecommendationType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class RecommendationTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.RecommendationTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.RecommendationTypeEnum
 */
exports.RecommendationTypeEnum = new RecommendationTypeEnum$Type();
