// google-ads-api-client/dist/protos/bindings/ads/enums/resource_change_operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceChangeOperationEnum = exports.ResourceChangeOperationEnum_ResourceChangeOperation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The operation on the changed resource in change_event resource.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ResourceChangeOperationEnum.ResourceChangeOperation
 */
var ResourceChangeOperationEnum_ResourceChangeOperation;
(function (ResourceChangeOperationEnum_ResourceChangeOperation) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ResourceChangeOperationEnum_ResourceChangeOperation[ResourceChangeOperationEnum_ResourceChangeOperation["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents an unclassified operation unknown
     * in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ResourceChangeOperationEnum_ResourceChangeOperation[ResourceChangeOperationEnum_ResourceChangeOperation["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The resource was created.
     *
     * @generated from protobuf enum value: CREATE = 2;
     */
    ResourceChangeOperationEnum_ResourceChangeOperation[ResourceChangeOperationEnum_ResourceChangeOperation["CREATE"] = 2] = "CREATE";
    /**
     * The resource was modified.
     *
     * @generated from protobuf enum value: UPDATE = 3;
     */
    ResourceChangeOperationEnum_ResourceChangeOperation[ResourceChangeOperationEnum_ResourceChangeOperation["UPDATE"] = 3] = "UPDATE";
    /**
     * The resource was removed.
     *
     * @generated from protobuf enum value: REMOVE = 4;
     */
    ResourceChangeOperationEnum_ResourceChangeOperation[ResourceChangeOperationEnum_ResourceChangeOperation["REMOVE"] = 4] = "REMOVE";
})(ResourceChangeOperationEnum_ResourceChangeOperation = exports.ResourceChangeOperationEnum_ResourceChangeOperation || (exports.ResourceChangeOperationEnum_ResourceChangeOperation = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ResourceChangeOperationEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ResourceChangeOperationEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ResourceChangeOperationEnum
 */
exports.ResourceChangeOperationEnum = new ResourceChangeOperationEnum$Type();
