// google-ads-api-client/dist/protos/bindings/ads/enums/resource_limit_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceLimitTypeEnum = exports.ResourceLimitTypeEnum_ResourceLimitType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Resource limit type.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ResourceLimitTypeEnum.ResourceLimitType
 */
var ResourceLimitTypeEnum_ResourceLimitType;
(function (ResourceLimitTypeEnum_ResourceLimitType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents an unclassified operation unknown
     * in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Number of ENABLED and PAUSED campaigns per customer.
     *
     * @generated from protobuf enum value: CAMPAIGNS_PER_CUSTOMER = 2;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CAMPAIGNS_PER_CUSTOMER"] = 2] = "CAMPAIGNS_PER_CUSTOMER";
    /**
     * Number of ENABLED and PAUSED base campaigns per customer.
     *
     * @generated from protobuf enum value: BASE_CAMPAIGNS_PER_CUSTOMER = 3;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BASE_CAMPAIGNS_PER_CUSTOMER"] = 3] = "BASE_CAMPAIGNS_PER_CUSTOMER";
    /**
     * Number of ENABLED and PAUSED experiment campaigns per customer.
     *
     * @generated from protobuf enum value: EXPERIMENT_CAMPAIGNS_PER_CUSTOMER = 105;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPERIMENT_CAMPAIGNS_PER_CUSTOMER"] = 105] = "EXPERIMENT_CAMPAIGNS_PER_CUSTOMER";
    /**
     * Number of ENABLED and PAUSED Hotel campaigns per customer.
     *
     * @generated from protobuf enum value: HOTEL_CAMPAIGNS_PER_CUSTOMER = 4;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["HOTEL_CAMPAIGNS_PER_CUSTOMER"] = 4] = "HOTEL_CAMPAIGNS_PER_CUSTOMER";
    /**
     * Number of ENABLED and PAUSED Smart Shopping campaigns per customer.
     *
     * @generated from protobuf enum value: SMART_SHOPPING_CAMPAIGNS_PER_CUSTOMER = 5;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["SMART_SHOPPING_CAMPAIGNS_PER_CUSTOMER"] = 5] = "SMART_SHOPPING_CAMPAIGNS_PER_CUSTOMER";
    /**
     * Number of ENABLED ad groups per campaign.
     *
     * @generated from protobuf enum value: AD_GROUPS_PER_CAMPAIGN = 6;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["AD_GROUPS_PER_CAMPAIGN"] = 6] = "AD_GROUPS_PER_CAMPAIGN";
    /**
     * Number of ENABLED ad groups per Shopping campaign.
     *
     * @generated from protobuf enum value: AD_GROUPS_PER_SHOPPING_CAMPAIGN = 8;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["AD_GROUPS_PER_SHOPPING_CAMPAIGN"] = 8] = "AD_GROUPS_PER_SHOPPING_CAMPAIGN";
    /**
     * Number of ENABLED ad groups per Hotel campaign.
     *
     * @generated from protobuf enum value: AD_GROUPS_PER_HOTEL_CAMPAIGN = 9;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["AD_GROUPS_PER_HOTEL_CAMPAIGN"] = 9] = "AD_GROUPS_PER_HOTEL_CAMPAIGN";
    /**
     * Number of ENABLED reporting ad groups per local campaign.
     *
     * @generated from protobuf enum value: REPORTING_AD_GROUPS_PER_LOCAL_CAMPAIGN = 10;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["REPORTING_AD_GROUPS_PER_LOCAL_CAMPAIGN"] = 10] = "REPORTING_AD_GROUPS_PER_LOCAL_CAMPAIGN";
    /**
     * Number of ENABLED reporting ad groups per App campaign. It includes app
     * campaign and app campaign for engagement.
     *
     * @generated from protobuf enum value: REPORTING_AD_GROUPS_PER_APP_CAMPAIGN = 11;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["REPORTING_AD_GROUPS_PER_APP_CAMPAIGN"] = 11] = "REPORTING_AD_GROUPS_PER_APP_CAMPAIGN";
    /**
     * Number of ENABLED managed ad groups per smart campaign.
     *
     * @generated from protobuf enum value: MANAGED_AD_GROUPS_PER_SMART_CAMPAIGN = 52;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["MANAGED_AD_GROUPS_PER_SMART_CAMPAIGN"] = 52] = "MANAGED_AD_GROUPS_PER_SMART_CAMPAIGN";
    /**
     * Number of ENABLED ad group criteria per customer.
     * An ad group criterion is considered as ENABLED if:
     * 1. it's not REMOVED
     * 2. its ad group is not REMOVED
     * 3. its campaign is not REMOVED.
     *
     * @generated from protobuf enum value: AD_GROUP_CRITERIA_PER_CUSTOMER = 12;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["AD_GROUP_CRITERIA_PER_CUSTOMER"] = 12] = "AD_GROUP_CRITERIA_PER_CUSTOMER";
    /**
     * Number of ad group criteria across all base campaigns for a customer.
     *
     * @generated from protobuf enum value: BASE_AD_GROUP_CRITERIA_PER_CUSTOMER = 13;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BASE_AD_GROUP_CRITERIA_PER_CUSTOMER"] = 13] = "BASE_AD_GROUP_CRITERIA_PER_CUSTOMER";
    /**
     * Number of ad group criteria across all experiment campaigns for a
     * customer.
     *
     * @generated from protobuf enum value: EXPERIMENT_AD_GROUP_CRITERIA_PER_CUSTOMER = 107;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPERIMENT_AD_GROUP_CRITERIA_PER_CUSTOMER"] = 107] = "EXPERIMENT_AD_GROUP_CRITERIA_PER_CUSTOMER";
    /**
     * Number of ENABLED ad group criteria per campaign.
     * An ad group criterion is considered as ENABLED if:
     * 1. it's not REMOVED
     * 2. its ad group is not REMOVED.
     *
     * @generated from protobuf enum value: AD_GROUP_CRITERIA_PER_CAMPAIGN = 14;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["AD_GROUP_CRITERIA_PER_CAMPAIGN"] = 14] = "AD_GROUP_CRITERIA_PER_CAMPAIGN";
    /**
     * Number of ENABLED campaign criteria per customer.
     *
     * @generated from protobuf enum value: CAMPAIGN_CRITERIA_PER_CUSTOMER = 15;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CAMPAIGN_CRITERIA_PER_CUSTOMER"] = 15] = "CAMPAIGN_CRITERIA_PER_CUSTOMER";
    /**
     * Number of ENABLED campaign criteria across all base campaigns for a
     * customer.
     *
     * @generated from protobuf enum value: BASE_CAMPAIGN_CRITERIA_PER_CUSTOMER = 16;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BASE_CAMPAIGN_CRITERIA_PER_CUSTOMER"] = 16] = "BASE_CAMPAIGN_CRITERIA_PER_CUSTOMER";
    /**
     * Number of ENABLED campaign criteria across all experiment campaigns for a
     * customer.
     *
     * @generated from protobuf enum value: EXPERIMENT_CAMPAIGN_CRITERIA_PER_CUSTOMER = 108;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPERIMENT_CAMPAIGN_CRITERIA_PER_CUSTOMER"] = 108] = "EXPERIMENT_CAMPAIGN_CRITERIA_PER_CUSTOMER";
    /**
     * Number of ENABLED webpage criteria per customer, including
     * campaign level and ad group level.
     *
     * @generated from protobuf enum value: WEBPAGE_CRITERIA_PER_CUSTOMER = 17;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["WEBPAGE_CRITERIA_PER_CUSTOMER"] = 17] = "WEBPAGE_CRITERIA_PER_CUSTOMER";
    /**
     * Number of ENABLED webpage criteria across all base campaigns for
     * a customer.
     *
     * @generated from protobuf enum value: BASE_WEBPAGE_CRITERIA_PER_CUSTOMER = 18;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BASE_WEBPAGE_CRITERIA_PER_CUSTOMER"] = 18] = "BASE_WEBPAGE_CRITERIA_PER_CUSTOMER";
    /**
     * Meximum number of ENABLED webpage criteria across all experiment
     * campaigns for a customer.
     *
     * @generated from protobuf enum value: EXPERIMENT_WEBPAGE_CRITERIA_PER_CUSTOMER = 19;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPERIMENT_WEBPAGE_CRITERIA_PER_CUSTOMER"] = 19] = "EXPERIMENT_WEBPAGE_CRITERIA_PER_CUSTOMER";
    /**
     * Number of combined audience criteria per ad group.
     *
     * @generated from protobuf enum value: COMBINED_AUDIENCE_CRITERIA_PER_AD_GROUP = 20;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["COMBINED_AUDIENCE_CRITERIA_PER_AD_GROUP"] = 20] = "COMBINED_AUDIENCE_CRITERIA_PER_AD_GROUP";
    /**
     * Limit for placement criterion type group in customer negative criterion.
     *
     * @generated from protobuf enum value: CUSTOMER_NEGATIVE_PLACEMENT_CRITERIA_PER_CUSTOMER = 21;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CUSTOMER_NEGATIVE_PLACEMENT_CRITERIA_PER_CUSTOMER"] = 21] = "CUSTOMER_NEGATIVE_PLACEMENT_CRITERIA_PER_CUSTOMER";
    /**
     * Limit for YouTube TV channels in customer negative criterion.
     *
     * @generated from protobuf enum value: CUSTOMER_NEGATIVE_YOUTUBE_CHANNEL_CRITERIA_PER_CUSTOMER = 22;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CUSTOMER_NEGATIVE_YOUTUBE_CHANNEL_CRITERIA_PER_CUSTOMER"] = 22] = "CUSTOMER_NEGATIVE_YOUTUBE_CHANNEL_CRITERIA_PER_CUSTOMER";
    /**
     * Number of ENABLED criteria per ad group.
     *
     * @generated from protobuf enum value: CRITERIA_PER_AD_GROUP = 23;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CRITERIA_PER_AD_GROUP"] = 23] = "CRITERIA_PER_AD_GROUP";
    /**
     * Number of listing group criteria per ad group.
     *
     * @generated from protobuf enum value: LISTING_GROUPS_PER_AD_GROUP = 24;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LISTING_GROUPS_PER_AD_GROUP"] = 24] = "LISTING_GROUPS_PER_AD_GROUP";
    /**
     * Number of ENABLED explicitly shared budgets per customer.
     *
     * @generated from protobuf enum value: EXPLICITLY_SHARED_BUDGETS_PER_CUSTOMER = 25;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPLICITLY_SHARED_BUDGETS_PER_CUSTOMER"] = 25] = "EXPLICITLY_SHARED_BUDGETS_PER_CUSTOMER";
    /**
     * Number of ENABLED implicitly shared budgets per customer.
     *
     * @generated from protobuf enum value: IMPLICITLY_SHARED_BUDGETS_PER_CUSTOMER = 26;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["IMPLICITLY_SHARED_BUDGETS_PER_CUSTOMER"] = 26] = "IMPLICITLY_SHARED_BUDGETS_PER_CUSTOMER";
    /**
     * Number of combined audience criteria per campaign.
     *
     * @generated from protobuf enum value: COMBINED_AUDIENCE_CRITERIA_PER_CAMPAIGN = 27;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["COMBINED_AUDIENCE_CRITERIA_PER_CAMPAIGN"] = 27] = "COMBINED_AUDIENCE_CRITERIA_PER_CAMPAIGN";
    /**
     * Number of negative keywords per campaign.
     *
     * @generated from protobuf enum value: NEGATIVE_KEYWORDS_PER_CAMPAIGN = 28;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["NEGATIVE_KEYWORDS_PER_CAMPAIGN"] = 28] = "NEGATIVE_KEYWORDS_PER_CAMPAIGN";
    /**
     * Number of excluded campaign criteria in placement dimension, for example,
     * placement, mobile application, YouTube channel, etc. The API criterion
     * type is NOT limited to placement only, and this does not include
     * exclusions at the ad group or other levels.
     *
     * @generated from protobuf enum value: NEGATIVE_PLACEMENTS_PER_CAMPAIGN = 29;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["NEGATIVE_PLACEMENTS_PER_CAMPAIGN"] = 29] = "NEGATIVE_PLACEMENTS_PER_CAMPAIGN";
    /**
     * Number of geo targets per campaign.
     *
     * @generated from protobuf enum value: GEO_TARGETS_PER_CAMPAIGN = 30;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["GEO_TARGETS_PER_CAMPAIGN"] = 30] = "GEO_TARGETS_PER_CAMPAIGN";
    /**
     * Number of negative IP blocks per campaign.
     *
     * @generated from protobuf enum value: NEGATIVE_IP_BLOCKS_PER_CAMPAIGN = 32;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["NEGATIVE_IP_BLOCKS_PER_CAMPAIGN"] = 32] = "NEGATIVE_IP_BLOCKS_PER_CAMPAIGN";
    /**
     * Number of proximity targets per campaign.
     *
     * @generated from protobuf enum value: PROXIMITIES_PER_CAMPAIGN = 33;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["PROXIMITIES_PER_CAMPAIGN"] = 33] = "PROXIMITIES_PER_CAMPAIGN";
    /**
     * Number of listing scopes per Shopping campaign.
     *
     * @generated from protobuf enum value: LISTING_SCOPES_PER_SHOPPING_CAMPAIGN = 34;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LISTING_SCOPES_PER_SHOPPING_CAMPAIGN"] = 34] = "LISTING_SCOPES_PER_SHOPPING_CAMPAIGN";
    /**
     * Number of listing scopes per non-Shopping campaign.
     *
     * @generated from protobuf enum value: LISTING_SCOPES_PER_NON_SHOPPING_CAMPAIGN = 35;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LISTING_SCOPES_PER_NON_SHOPPING_CAMPAIGN"] = 35] = "LISTING_SCOPES_PER_NON_SHOPPING_CAMPAIGN";
    /**
     * Number of criteria per negative keyword shared set.
     *
     * @generated from protobuf enum value: NEGATIVE_KEYWORDS_PER_SHARED_SET = 36;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["NEGATIVE_KEYWORDS_PER_SHARED_SET"] = 36] = "NEGATIVE_KEYWORDS_PER_SHARED_SET";
    /**
     * Number of criteria per negative placement shared set.
     *
     * @generated from protobuf enum value: NEGATIVE_PLACEMENTS_PER_SHARED_SET = 37;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["NEGATIVE_PLACEMENTS_PER_SHARED_SET"] = 37] = "NEGATIVE_PLACEMENTS_PER_SHARED_SET";
    /**
     * Default number of shared sets allowed per type per customer.
     *
     * @generated from protobuf enum value: SHARED_SETS_PER_CUSTOMER_FOR_TYPE_DEFAULT = 40;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["SHARED_SETS_PER_CUSTOMER_FOR_TYPE_DEFAULT"] = 40] = "SHARED_SETS_PER_CUSTOMER_FOR_TYPE_DEFAULT";
    /**
     * Number of shared sets of negative placement list type for a
     * manager customer.
     *
     * @generated from protobuf enum value: SHARED_SETS_PER_CUSTOMER_FOR_NEGATIVE_PLACEMENT_LIST_LOWER = 41;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["SHARED_SETS_PER_CUSTOMER_FOR_NEGATIVE_PLACEMENT_LIST_LOWER"] = 41] = "SHARED_SETS_PER_CUSTOMER_FOR_NEGATIVE_PLACEMENT_LIST_LOWER";
    /**
     * Number of hotel_advance_booking_window bid modifiers per ad group.
     *
     * @generated from protobuf enum value: HOTEL_ADVANCE_BOOKING_WINDOW_BID_MODIFIERS_PER_AD_GROUP = 44;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["HOTEL_ADVANCE_BOOKING_WINDOW_BID_MODIFIERS_PER_AD_GROUP"] = 44] = "HOTEL_ADVANCE_BOOKING_WINDOW_BID_MODIFIERS_PER_AD_GROUP";
    /**
     * Number of ENABLED shared bidding strategies per customer.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGIES_PER_CUSTOMER = 45;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BIDDING_STRATEGIES_PER_CUSTOMER"] = 45] = "BIDDING_STRATEGIES_PER_CUSTOMER";
    /**
     * Number of open basic user lists per customer.
     *
     * @generated from protobuf enum value: BASIC_USER_LISTS_PER_CUSTOMER = 47;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BASIC_USER_LISTS_PER_CUSTOMER"] = 47] = "BASIC_USER_LISTS_PER_CUSTOMER";
    /**
     * Number of open logical user lists per customer.
     *
     * @generated from protobuf enum value: LOGICAL_USER_LISTS_PER_CUSTOMER = 48;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LOGICAL_USER_LISTS_PER_CUSTOMER"] = 48] = "LOGICAL_USER_LISTS_PER_CUSTOMER";
    /**
     * Number of open rule based user lists per customer.
     *
     * @generated from protobuf enum value: RULE_BASED_USER_LISTS_PER_CUSTOMER = 153;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["RULE_BASED_USER_LISTS_PER_CUSTOMER"] = 153] = "RULE_BASED_USER_LISTS_PER_CUSTOMER";
    /**
     * Number of ENABLED and PAUSED ad group ads across all base campaigns for a
     * customer.
     *
     * @generated from protobuf enum value: BASE_AD_GROUP_ADS_PER_CUSTOMER = 53;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BASE_AD_GROUP_ADS_PER_CUSTOMER"] = 53] = "BASE_AD_GROUP_ADS_PER_CUSTOMER";
    /**
     * Number of ENABLED and PAUSED ad group ads across all experiment campaigns
     * for a customer.
     *
     * @generated from protobuf enum value: EXPERIMENT_AD_GROUP_ADS_PER_CUSTOMER = 54;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPERIMENT_AD_GROUP_ADS_PER_CUSTOMER"] = 54] = "EXPERIMENT_AD_GROUP_ADS_PER_CUSTOMER";
    /**
     * Number of ENABLED and PAUSED ad group ads per campaign.
     *
     * @generated from protobuf enum value: AD_GROUP_ADS_PER_CAMPAIGN = 55;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["AD_GROUP_ADS_PER_CAMPAIGN"] = 55] = "AD_GROUP_ADS_PER_CAMPAIGN";
    /**
     * Number of ENABLED ads per ad group that do not fall in to other buckets.
     * Includes text and many other types.
     *
     * @generated from protobuf enum value: TEXT_AND_OTHER_ADS_PER_AD_GROUP = 56;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["TEXT_AND_OTHER_ADS_PER_AD_GROUP"] = 56] = "TEXT_AND_OTHER_ADS_PER_AD_GROUP";
    /**
     * Number of ENABLED image ads per ad group.
     *
     * @generated from protobuf enum value: IMAGE_ADS_PER_AD_GROUP = 57;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["IMAGE_ADS_PER_AD_GROUP"] = 57] = "IMAGE_ADS_PER_AD_GROUP";
    /**
     * Number of ENABLED shopping smart ads per ad group.
     *
     * @generated from protobuf enum value: SHOPPING_SMART_ADS_PER_AD_GROUP = 58;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["SHOPPING_SMART_ADS_PER_AD_GROUP"] = 58] = "SHOPPING_SMART_ADS_PER_AD_GROUP";
    /**
     * Number of ENABLED responsive search ads per ad group.
     *
     * @generated from protobuf enum value: RESPONSIVE_SEARCH_ADS_PER_AD_GROUP = 59;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["RESPONSIVE_SEARCH_ADS_PER_AD_GROUP"] = 59] = "RESPONSIVE_SEARCH_ADS_PER_AD_GROUP";
    /**
     * Number of ENABLED app ads per ad group.
     *
     * @generated from protobuf enum value: APP_ADS_PER_AD_GROUP = 60;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["APP_ADS_PER_AD_GROUP"] = 60] = "APP_ADS_PER_AD_GROUP";
    /**
     * Number of ENABLED app engagement ads per ad group.
     *
     * @generated from protobuf enum value: APP_ENGAGEMENT_ADS_PER_AD_GROUP = 61;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["APP_ENGAGEMENT_ADS_PER_AD_GROUP"] = 61] = "APP_ENGAGEMENT_ADS_PER_AD_GROUP";
    /**
     * Number of ENABLED local ads per ad group.
     *
     * @generated from protobuf enum value: LOCAL_ADS_PER_AD_GROUP = 62;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LOCAL_ADS_PER_AD_GROUP"] = 62] = "LOCAL_ADS_PER_AD_GROUP";
    /**
     * Number of ENABLED video ads per ad group.
     *
     * @generated from protobuf enum value: VIDEO_ADS_PER_AD_GROUP = 63;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["VIDEO_ADS_PER_AD_GROUP"] = 63] = "VIDEO_ADS_PER_AD_GROUP";
    /**
     * Number of ENABLED lead form CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: LEAD_FORM_CAMPAIGN_ASSETS_PER_CAMPAIGN = 143;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LEAD_FORM_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 143] = "LEAD_FORM_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED promotion CustomerAssets per customer.
     *
     * @generated from protobuf enum value: PROMOTION_CUSTOMER_ASSETS_PER_CUSTOMER = 79;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["PROMOTION_CUSTOMER_ASSETS_PER_CUSTOMER"] = 79] = "PROMOTION_CUSTOMER_ASSETS_PER_CUSTOMER";
    /**
     * Number of ENABLED promotion CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: PROMOTION_CAMPAIGN_ASSETS_PER_CAMPAIGN = 80;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["PROMOTION_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 80] = "PROMOTION_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED promotion AdGroupAssets per ad group.
     *
     * @generated from protobuf enum value: PROMOTION_AD_GROUP_ASSETS_PER_AD_GROUP = 81;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["PROMOTION_AD_GROUP_ASSETS_PER_AD_GROUP"] = 81] = "PROMOTION_AD_GROUP_ASSETS_PER_AD_GROUP";
    /**
     * Number of ENABLED callout CustomerAssets per customer.
     *
     * @generated from protobuf enum value: CALLOUT_CUSTOMER_ASSETS_PER_CUSTOMER = 134;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CALLOUT_CUSTOMER_ASSETS_PER_CUSTOMER"] = 134] = "CALLOUT_CUSTOMER_ASSETS_PER_CUSTOMER";
    /**
     * Number of ENABLED callout CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: CALLOUT_CAMPAIGN_ASSETS_PER_CAMPAIGN = 135;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CALLOUT_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 135] = "CALLOUT_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED callout AdGroupAssets per ad group.
     *
     * @generated from protobuf enum value: CALLOUT_AD_GROUP_ASSETS_PER_AD_GROUP = 136;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CALLOUT_AD_GROUP_ASSETS_PER_AD_GROUP"] = 136] = "CALLOUT_AD_GROUP_ASSETS_PER_AD_GROUP";
    /**
     * Number of ENABLED sitelink CustomerAssets per customer.
     *
     * @generated from protobuf enum value: SITELINK_CUSTOMER_ASSETS_PER_CUSTOMER = 137;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["SITELINK_CUSTOMER_ASSETS_PER_CUSTOMER"] = 137] = "SITELINK_CUSTOMER_ASSETS_PER_CUSTOMER";
    /**
     * Number of ENABLED sitelink CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: SITELINK_CAMPAIGN_ASSETS_PER_CAMPAIGN = 138;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["SITELINK_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 138] = "SITELINK_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED sitelink AdGroupAssets per ad group.
     *
     * @generated from protobuf enum value: SITELINK_AD_GROUP_ASSETS_PER_AD_GROUP = 139;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["SITELINK_AD_GROUP_ASSETS_PER_AD_GROUP"] = 139] = "SITELINK_AD_GROUP_ASSETS_PER_AD_GROUP";
    /**
     * Number of ENABLED structured snippet CustomerAssets per customer.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPET_CUSTOMER_ASSETS_PER_CUSTOMER = 140;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["STRUCTURED_SNIPPET_CUSTOMER_ASSETS_PER_CUSTOMER"] = 140] = "STRUCTURED_SNIPPET_CUSTOMER_ASSETS_PER_CUSTOMER";
    /**
     * Number of ENABLED structured snippet CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPET_CAMPAIGN_ASSETS_PER_CAMPAIGN = 141;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["STRUCTURED_SNIPPET_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 141] = "STRUCTURED_SNIPPET_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED structured snippet AdGroupAssets per ad group.
     *
     * @generated from protobuf enum value: STRUCTURED_SNIPPET_AD_GROUP_ASSETS_PER_AD_GROUP = 142;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["STRUCTURED_SNIPPET_AD_GROUP_ASSETS_PER_AD_GROUP"] = 142] = "STRUCTURED_SNIPPET_AD_GROUP_ASSETS_PER_AD_GROUP";
    /**
     * Number of ENABLED mobile app CustomerAssets per customer.
     *
     * @generated from protobuf enum value: MOBILE_APP_CUSTOMER_ASSETS_PER_CUSTOMER = 144;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["MOBILE_APP_CUSTOMER_ASSETS_PER_CUSTOMER"] = 144] = "MOBILE_APP_CUSTOMER_ASSETS_PER_CUSTOMER";
    /**
     * Number of ENABLED mobile app CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: MOBILE_APP_CAMPAIGN_ASSETS_PER_CAMPAIGN = 145;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["MOBILE_APP_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 145] = "MOBILE_APP_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED mobile app AdGroupAssets per ad group.
     *
     * @generated from protobuf enum value: MOBILE_APP_AD_GROUP_ASSETS_PER_AD_GROUP = 146;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["MOBILE_APP_AD_GROUP_ASSETS_PER_AD_GROUP"] = 146] = "MOBILE_APP_AD_GROUP_ASSETS_PER_AD_GROUP";
    /**
     * Number of ENABLED hotel callout CustomerAssets per customer.
     *
     * @generated from protobuf enum value: HOTEL_CALLOUT_CUSTOMER_ASSETS_PER_CUSTOMER = 147;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["HOTEL_CALLOUT_CUSTOMER_ASSETS_PER_CUSTOMER"] = 147] = "HOTEL_CALLOUT_CUSTOMER_ASSETS_PER_CUSTOMER";
    /**
     * Number of ENABLED hotel callout CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: HOTEL_CALLOUT_CAMPAIGN_ASSETS_PER_CAMPAIGN = 148;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["HOTEL_CALLOUT_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 148] = "HOTEL_CALLOUT_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED hotel callout AdGroupAssets per ad group.
     *
     * @generated from protobuf enum value: HOTEL_CALLOUT_AD_GROUP_ASSETS_PER_AD_GROUP = 149;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["HOTEL_CALLOUT_AD_GROUP_ASSETS_PER_AD_GROUP"] = 149] = "HOTEL_CALLOUT_AD_GROUP_ASSETS_PER_AD_GROUP";
    /**
     * Number of ENABLED call CustomerAssets per customer.
     *
     * @generated from protobuf enum value: CALL_CUSTOMER_ASSETS_PER_CUSTOMER = 150;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CALL_CUSTOMER_ASSETS_PER_CUSTOMER"] = 150] = "CALL_CUSTOMER_ASSETS_PER_CUSTOMER";
    /**
     * Number of ENABLED call CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: CALL_CAMPAIGN_ASSETS_PER_CAMPAIGN = 151;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CALL_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 151] = "CALL_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED call AdGroupAssets per ad group.
     *
     * @generated from protobuf enum value: CALL_AD_GROUP_ASSETS_PER_AD_GROUP = 152;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CALL_AD_GROUP_ASSETS_PER_AD_GROUP"] = 152] = "CALL_AD_GROUP_ASSETS_PER_AD_GROUP";
    /**
     * Number of ENABLED price CustomerAssets per customer.
     *
     * @generated from protobuf enum value: PRICE_CUSTOMER_ASSETS_PER_CUSTOMER = 154;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["PRICE_CUSTOMER_ASSETS_PER_CUSTOMER"] = 154] = "PRICE_CUSTOMER_ASSETS_PER_CUSTOMER";
    /**
     * Number of ENABLED price CampaignAssets per campaign.
     *
     * @generated from protobuf enum value: PRICE_CAMPAIGN_ASSETS_PER_CAMPAIGN = 155;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["PRICE_CAMPAIGN_ASSETS_PER_CAMPAIGN"] = 155] = "PRICE_CAMPAIGN_ASSETS_PER_CAMPAIGN";
    /**
     * Number of ENABLED price AdGroupAssets per ad group.
     *
     * @generated from protobuf enum value: PRICE_AD_GROUP_ASSETS_PER_AD_GROUP = 156;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["PRICE_AD_GROUP_ASSETS_PER_AD_GROUP"] = 156] = "PRICE_AD_GROUP_ASSETS_PER_AD_GROUP";
    /**
     * Number of ENABLED page feed asset sets per customer.
     *
     * @generated from protobuf enum value: PAGE_FEED_ASSET_SETS_PER_CUSTOMER = 157;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["PAGE_FEED_ASSET_SETS_PER_CUSTOMER"] = 157] = "PAGE_FEED_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED dynamic education feed asset sets per customer.
     *
     * @generated from protobuf enum value: DYNAMIC_EDUCATION_FEED_ASSET_SETS_PER_CUSTOMER = 158;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["DYNAMIC_EDUCATION_FEED_ASSET_SETS_PER_CUSTOMER"] = 158] = "DYNAMIC_EDUCATION_FEED_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED assets per page feed asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_PAGE_FEED_ASSET_SET = 159;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_PAGE_FEED_ASSET_SET"] = 159] = "ASSETS_PER_PAGE_FEED_ASSET_SET";
    /**
     * Number of ENABLED assets per dynamic education asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_DYNAMIC_EDUCATION_FEED_ASSET_SET = 160;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_DYNAMIC_EDUCATION_FEED_ASSET_SET"] = 160] = "ASSETS_PER_DYNAMIC_EDUCATION_FEED_ASSET_SET";
    /**
     * Number of ENABLED dynamic real estate asset sets per customer.
     *
     * @generated from protobuf enum value: DYNAMIC_REAL_ESTATE_ASSET_SETS_PER_CUSTOMER = 161;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["DYNAMIC_REAL_ESTATE_ASSET_SETS_PER_CUSTOMER"] = 161] = "DYNAMIC_REAL_ESTATE_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED assets per dynamic real estate asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_DYNAMIC_REAL_ESTATE_ASSET_SET = 162;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_DYNAMIC_REAL_ESTATE_ASSET_SET"] = 162] = "ASSETS_PER_DYNAMIC_REAL_ESTATE_ASSET_SET";
    /**
     * Number of ENABLED dynamic custom asset sets per customer.
     *
     * @generated from protobuf enum value: DYNAMIC_CUSTOM_ASSET_SETS_PER_CUSTOMER = 163;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["DYNAMIC_CUSTOM_ASSET_SETS_PER_CUSTOMER"] = 163] = "DYNAMIC_CUSTOM_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED assets per dynamic custom asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_DYNAMIC_CUSTOM_ASSET_SET = 164;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_DYNAMIC_CUSTOM_ASSET_SET"] = 164] = "ASSETS_PER_DYNAMIC_CUSTOM_ASSET_SET";
    /**
     * Number of ENABLED dynamic hotels and rentals asset sets per
     * customer.
     *
     * @generated from protobuf enum value: DYNAMIC_HOTELS_AND_RENTALS_ASSET_SETS_PER_CUSTOMER = 165;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["DYNAMIC_HOTELS_AND_RENTALS_ASSET_SETS_PER_CUSTOMER"] = 165] = "DYNAMIC_HOTELS_AND_RENTALS_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED assets per dynamic hotels and rentals asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_DYNAMIC_HOTELS_AND_RENTALS_ASSET_SET = 166;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_DYNAMIC_HOTELS_AND_RENTALS_ASSET_SET"] = 166] = "ASSETS_PER_DYNAMIC_HOTELS_AND_RENTALS_ASSET_SET";
    /**
     * Number of ENABLED dynamic local asset sets per customer.
     *
     * @generated from protobuf enum value: DYNAMIC_LOCAL_ASSET_SETS_PER_CUSTOMER = 167;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["DYNAMIC_LOCAL_ASSET_SETS_PER_CUSTOMER"] = 167] = "DYNAMIC_LOCAL_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED assets per dynamic local asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_DYNAMIC_LOCAL_ASSET_SET = 168;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_DYNAMIC_LOCAL_ASSET_SET"] = 168] = "ASSETS_PER_DYNAMIC_LOCAL_ASSET_SET";
    /**
     * Number of ENABLED dynamic flights asset sets per customer.
     *
     * @generated from protobuf enum value: DYNAMIC_FLIGHTS_ASSET_SETS_PER_CUSTOMER = 169;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["DYNAMIC_FLIGHTS_ASSET_SETS_PER_CUSTOMER"] = 169] = "DYNAMIC_FLIGHTS_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED assets per dynamic flights asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_DYNAMIC_FLIGHTS_ASSET_SET = 170;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_DYNAMIC_FLIGHTS_ASSET_SET"] = 170] = "ASSETS_PER_DYNAMIC_FLIGHTS_ASSET_SET";
    /**
     * Number of ENABLED dynamic travel asset sets per customer.
     *
     * @generated from protobuf enum value: DYNAMIC_TRAVEL_ASSET_SETS_PER_CUSTOMER = 171;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["DYNAMIC_TRAVEL_ASSET_SETS_PER_CUSTOMER"] = 171] = "DYNAMIC_TRAVEL_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED assets per dynamic travel asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_DYNAMIC_TRAVEL_ASSET_SET = 172;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_DYNAMIC_TRAVEL_ASSET_SET"] = 172] = "ASSETS_PER_DYNAMIC_TRAVEL_ASSET_SET";
    /**
     * Number of ENABLED dynamic jobs asset sets per customer.
     *
     * @generated from protobuf enum value: DYNAMIC_JOBS_ASSET_SETS_PER_CUSTOMER = 173;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["DYNAMIC_JOBS_ASSET_SETS_PER_CUSTOMER"] = 173] = "DYNAMIC_JOBS_ASSET_SETS_PER_CUSTOMER";
    /**
     * Number of ENABLED assets per dynamic jobs asset set.
     *
     * @generated from protobuf enum value: ASSETS_PER_DYNAMIC_JOBS_ASSET_SET = 174;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["ASSETS_PER_DYNAMIC_JOBS_ASSET_SET"] = 174] = "ASSETS_PER_DYNAMIC_JOBS_ASSET_SET";
    /**
     * Number of versions per ad.
     *
     * @generated from protobuf enum value: VERSIONS_PER_AD = 82;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["VERSIONS_PER_AD"] = 82] = "VERSIONS_PER_AD";
    /**
     * Number of ENABLED user feeds per customer.
     *
     * @generated from protobuf enum value: USER_FEEDS_PER_CUSTOMER = 90;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["USER_FEEDS_PER_CUSTOMER"] = 90] = "USER_FEEDS_PER_CUSTOMER";
    /**
     * Number of ENABLED system feeds per customer.
     *
     * @generated from protobuf enum value: SYSTEM_FEEDS_PER_CUSTOMER = 91;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["SYSTEM_FEEDS_PER_CUSTOMER"] = 91] = "SYSTEM_FEEDS_PER_CUSTOMER";
    /**
     * Number of feed attributes per feed.
     *
     * @generated from protobuf enum value: FEED_ATTRIBUTES_PER_FEED = 92;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["FEED_ATTRIBUTES_PER_FEED"] = 92] = "FEED_ATTRIBUTES_PER_FEED";
    /**
     * Number of ENABLED feed items per customer.
     *
     * @generated from protobuf enum value: FEED_ITEMS_PER_CUSTOMER = 94;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["FEED_ITEMS_PER_CUSTOMER"] = 94] = "FEED_ITEMS_PER_CUSTOMER";
    /**
     * Number of ENABLED campaign feeds per customer.
     *
     * @generated from protobuf enum value: CAMPAIGN_FEEDS_PER_CUSTOMER = 95;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CAMPAIGN_FEEDS_PER_CUSTOMER"] = 95] = "CAMPAIGN_FEEDS_PER_CUSTOMER";
    /**
     * Number of ENABLED campaign feeds across all base campaigns for a
     * customer.
     *
     * @generated from protobuf enum value: BASE_CAMPAIGN_FEEDS_PER_CUSTOMER = 96;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BASE_CAMPAIGN_FEEDS_PER_CUSTOMER"] = 96] = "BASE_CAMPAIGN_FEEDS_PER_CUSTOMER";
    /**
     * Number of ENABLED campaign feeds across all experiment campaigns for a
     * customer.
     *
     * @generated from protobuf enum value: EXPERIMENT_CAMPAIGN_FEEDS_PER_CUSTOMER = 109;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPERIMENT_CAMPAIGN_FEEDS_PER_CUSTOMER"] = 109] = "EXPERIMENT_CAMPAIGN_FEEDS_PER_CUSTOMER";
    /**
     * Number of ENABLED ad group feeds per customer.
     *
     * @generated from protobuf enum value: AD_GROUP_FEEDS_PER_CUSTOMER = 97;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["AD_GROUP_FEEDS_PER_CUSTOMER"] = 97] = "AD_GROUP_FEEDS_PER_CUSTOMER";
    /**
     * Number of ENABLED ad group feeds across all base campaigns for a
     * customer.
     *
     * @generated from protobuf enum value: BASE_AD_GROUP_FEEDS_PER_CUSTOMER = 98;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BASE_AD_GROUP_FEEDS_PER_CUSTOMER"] = 98] = "BASE_AD_GROUP_FEEDS_PER_CUSTOMER";
    /**
     * Number of ENABLED ad group feeds across all experiment campaigns for a
     * customer.
     *
     * @generated from protobuf enum value: EXPERIMENT_AD_GROUP_FEEDS_PER_CUSTOMER = 110;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPERIMENT_AD_GROUP_FEEDS_PER_CUSTOMER"] = 110] = "EXPERIMENT_AD_GROUP_FEEDS_PER_CUSTOMER";
    /**
     * Number of ENABLED ad group feeds per campaign.
     *
     * @generated from protobuf enum value: AD_GROUP_FEEDS_PER_CAMPAIGN = 99;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["AD_GROUP_FEEDS_PER_CAMPAIGN"] = 99] = "AD_GROUP_FEEDS_PER_CAMPAIGN";
    /**
     * Number of ENABLED feed items per customer.
     *
     * @generated from protobuf enum value: FEED_ITEM_SETS_PER_CUSTOMER = 100;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["FEED_ITEM_SETS_PER_CUSTOMER"] = 100] = "FEED_ITEM_SETS_PER_CUSTOMER";
    /**
     * Number of feed items per feed item set.
     *
     * @generated from protobuf enum value: FEED_ITEMS_PER_FEED_ITEM_SET = 101;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["FEED_ITEMS_PER_FEED_ITEM_SET"] = 101] = "FEED_ITEMS_PER_FEED_ITEM_SET";
    /**
     * Number of ENABLED campaign experiments per customer.
     *
     * @generated from protobuf enum value: CAMPAIGN_EXPERIMENTS_PER_CUSTOMER = 112;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CAMPAIGN_EXPERIMENTS_PER_CUSTOMER"] = 112] = "CAMPAIGN_EXPERIMENTS_PER_CUSTOMER";
    /**
     * Number of video experiment arms per experiment.
     *
     * @generated from protobuf enum value: EXPERIMENT_ARMS_PER_VIDEO_EXPERIMENT = 113;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["EXPERIMENT_ARMS_PER_VIDEO_EXPERIMENT"] = 113] = "EXPERIMENT_ARMS_PER_VIDEO_EXPERIMENT";
    /**
     * Number of owned labels per customer.
     *
     * @generated from protobuf enum value: OWNED_LABELS_PER_CUSTOMER = 115;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["OWNED_LABELS_PER_CUSTOMER"] = 115] = "OWNED_LABELS_PER_CUSTOMER";
    /**
     * Number of applied labels per campaign.
     *
     * @generated from protobuf enum value: LABELS_PER_CAMPAIGN = 117;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LABELS_PER_CAMPAIGN"] = 117] = "LABELS_PER_CAMPAIGN";
    /**
     * Number of applied labels per ad group.
     *
     * @generated from protobuf enum value: LABELS_PER_AD_GROUP = 118;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LABELS_PER_AD_GROUP"] = 118] = "LABELS_PER_AD_GROUP";
    /**
     * Number of applied labels per ad group ad.
     *
     * @generated from protobuf enum value: LABELS_PER_AD_GROUP_AD = 119;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LABELS_PER_AD_GROUP_AD"] = 119] = "LABELS_PER_AD_GROUP_AD";
    /**
     * Number of applied labels per ad group criterion.
     *
     * @generated from protobuf enum value: LABELS_PER_AD_GROUP_CRITERION = 120;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["LABELS_PER_AD_GROUP_CRITERION"] = 120] = "LABELS_PER_AD_GROUP_CRITERION";
    /**
     * Number of customers with a single label applied.
     *
     * @generated from protobuf enum value: TARGET_CUSTOMERS_PER_LABEL = 121;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["TARGET_CUSTOMERS_PER_LABEL"] = 121] = "TARGET_CUSTOMERS_PER_LABEL";
    /**
     * Number of ENABLED keyword plans per user per customer.
     * The limit is applied per <user, customer> pair because by default a plan
     * is private to a user of a customer. Each user of a customer has their own
     * independent limit.
     *
     * @generated from protobuf enum value: KEYWORD_PLANS_PER_USER_PER_CUSTOMER = 122;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["KEYWORD_PLANS_PER_USER_PER_CUSTOMER"] = 122] = "KEYWORD_PLANS_PER_USER_PER_CUSTOMER";
    /**
     * Number of keyword plan ad group keywords per keyword plan.
     *
     * @generated from protobuf enum value: KEYWORD_PLAN_AD_GROUP_KEYWORDS_PER_KEYWORD_PLAN = 123;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["KEYWORD_PLAN_AD_GROUP_KEYWORDS_PER_KEYWORD_PLAN"] = 123] = "KEYWORD_PLAN_AD_GROUP_KEYWORDS_PER_KEYWORD_PLAN";
    /**
     * Number of keyword plan ad groups per keyword plan.
     *
     * @generated from protobuf enum value: KEYWORD_PLAN_AD_GROUPS_PER_KEYWORD_PLAN = 124;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["KEYWORD_PLAN_AD_GROUPS_PER_KEYWORD_PLAN"] = 124] = "KEYWORD_PLAN_AD_GROUPS_PER_KEYWORD_PLAN";
    /**
     * Number of keyword plan negative keywords (both campaign and ad group) per
     * keyword plan.
     *
     * @generated from protobuf enum value: KEYWORD_PLAN_NEGATIVE_KEYWORDS_PER_KEYWORD_PLAN = 125;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["KEYWORD_PLAN_NEGATIVE_KEYWORDS_PER_KEYWORD_PLAN"] = 125] = "KEYWORD_PLAN_NEGATIVE_KEYWORDS_PER_KEYWORD_PLAN";
    /**
     * Number of keyword plan campaigns per keyword plan.
     *
     * @generated from protobuf enum value: KEYWORD_PLAN_CAMPAIGNS_PER_KEYWORD_PLAN = 126;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["KEYWORD_PLAN_CAMPAIGNS_PER_KEYWORD_PLAN"] = 126] = "KEYWORD_PLAN_CAMPAIGNS_PER_KEYWORD_PLAN";
    /**
     * Number of ENABLED conversion actions per customer.
     *
     * @generated from protobuf enum value: CONVERSION_ACTIONS_PER_CUSTOMER = 128;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["CONVERSION_ACTIONS_PER_CUSTOMER"] = 128] = "CONVERSION_ACTIONS_PER_CUSTOMER";
    /**
     * Number of operations in a single batch job.
     *
     * @generated from protobuf enum value: BATCH_JOB_OPERATIONS_PER_JOB = 130;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BATCH_JOB_OPERATIONS_PER_JOB"] = 130] = "BATCH_JOB_OPERATIONS_PER_JOB";
    /**
     * Number of PENDING or ENABLED batch jobs per customer.
     *
     * @generated from protobuf enum value: BATCH_JOBS_PER_CUSTOMER = 131;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["BATCH_JOBS_PER_CUSTOMER"] = 131] = "BATCH_JOBS_PER_CUSTOMER";
    /**
     * Number of hotel check-in date range bid modifiers per ad agroup.
     *
     * @generated from protobuf enum value: HOTEL_CHECK_IN_DATE_RANGE_BID_MODIFIERS_PER_AD_GROUP = 132;
     */
    ResourceLimitTypeEnum_ResourceLimitType[ResourceLimitTypeEnum_ResourceLimitType["HOTEL_CHECK_IN_DATE_RANGE_BID_MODIFIERS_PER_AD_GROUP"] = 132] = "HOTEL_CHECK_IN_DATE_RANGE_BID_MODIFIERS_PER_AD_GROUP";
})(ResourceLimitTypeEnum_ResourceLimitType = exports.ResourceLimitTypeEnum_ResourceLimitType || (exports.ResourceLimitTypeEnum_ResourceLimitType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ResourceLimitTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ResourceLimitTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ResourceLimitTypeEnum
 */
exports.ResourceLimitTypeEnum = new ResourceLimitTypeEnum$Type();
