// google-ads-api-client/dist/protos/bindings/ads/enums/response_content_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseContentTypeEnum = exports.ResponseContentTypeEnum_ResponseContentType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible response content types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType
 */
var ResponseContentTypeEnum_ResponseContentType;
(function (ResponseContentTypeEnum_ResponseContentType) {
    /**
     * Not specified. Will return the resource name only in the response.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ResponseContentTypeEnum_ResponseContentType[ResponseContentTypeEnum_ResponseContentType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The mutate response will be the resource name.
     *
     * @generated from protobuf enum value: RESOURCE_NAME_ONLY = 1;
     */
    ResponseContentTypeEnum_ResponseContentType[ResponseContentTypeEnum_ResponseContentType["RESOURCE_NAME_ONLY"] = 1] = "RESOURCE_NAME_ONLY";
    /**
     * The mutate response will be the resource name and the resource with
     * all mutable fields.
     *
     * @generated from protobuf enum value: MUTABLE_RESOURCE = 2;
     */
    ResponseContentTypeEnum_ResponseContentType[ResponseContentTypeEnum_ResponseContentType["MUTABLE_RESOURCE"] = 2] = "MUTABLE_RESOURCE";
})(ResponseContentTypeEnum_ResponseContentType = exports.ResponseContentTypeEnum_ResponseContentType || (exports.ResponseContentTypeEnum_ResponseContentType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ResponseContentTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ResponseContentTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ResponseContentTypeEnum
 */
exports.ResponseContentTypeEnum = new ResponseContentTypeEnum$Type();
