// google-ads-api-client/dist/protos/bindings/ads/enums/search_term_match_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchTermMatchTypeEnum = exports.SearchTermMatchTypeEnum_SearchTermMatchType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible match types for a keyword triggering an ad, including variants.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SearchTermMatchTypeEnum.SearchTermMatchType
 */
var SearchTermMatchTypeEnum_SearchTermMatchType;
(function (SearchTermMatchTypeEnum_SearchTermMatchType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SearchTermMatchTypeEnum_SearchTermMatchType[SearchTermMatchTypeEnum_SearchTermMatchType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SearchTermMatchTypeEnum_SearchTermMatchType[SearchTermMatchTypeEnum_SearchTermMatchType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Broad match.
     *
     * @generated from protobuf enum value: BROAD = 2;
     */
    SearchTermMatchTypeEnum_SearchTermMatchType[SearchTermMatchTypeEnum_SearchTermMatchType["BROAD"] = 2] = "BROAD";
    /**
     * Exact match.
     *
     * @generated from protobuf enum value: EXACT = 3;
     */
    SearchTermMatchTypeEnum_SearchTermMatchType[SearchTermMatchTypeEnum_SearchTermMatchType["EXACT"] = 3] = "EXACT";
    /**
     * Phrase match.
     *
     * @generated from protobuf enum value: PHRASE = 4;
     */
    SearchTermMatchTypeEnum_SearchTermMatchType[SearchTermMatchTypeEnum_SearchTermMatchType["PHRASE"] = 4] = "PHRASE";
    /**
     * Exact match (close variant).
     *
     * @generated from protobuf enum value: NEAR_EXACT = 5;
     */
    SearchTermMatchTypeEnum_SearchTermMatchType[SearchTermMatchTypeEnum_SearchTermMatchType["NEAR_EXACT"] = 5] = "NEAR_EXACT";
    /**
     * Phrase match (close variant).
     *
     * @generated from protobuf enum value: NEAR_PHRASE = 6;
     */
    SearchTermMatchTypeEnum_SearchTermMatchType[SearchTermMatchTypeEnum_SearchTermMatchType["NEAR_PHRASE"] = 6] = "NEAR_PHRASE";
})(SearchTermMatchTypeEnum_SearchTermMatchType = exports.SearchTermMatchTypeEnum_SearchTermMatchType || (exports.SearchTermMatchTypeEnum_SearchTermMatchType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SearchTermMatchTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SearchTermMatchTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SearchTermMatchTypeEnum
 */
exports.SearchTermMatchTypeEnum = new SearchTermMatchTypeEnum$Type();
