// google-ads-api-client/dist/protos/bindings/ads/enums/search_engine_results_page_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchEngineResultsPageTypeEnum = exports.SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of the search engine results page.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SearchEngineResultsPageTypeEnum.SearchEngineResultsPageType
 */
var SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType;
(function (SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType[SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType[SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Only ads were contained in the search engine results page.
     *
     * @generated from protobuf enum value: ADS_ONLY = 2;
     */
    SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType[SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType["ADS_ONLY"] = 2] = "ADS_ONLY";
    /**
     * Only organic results were contained in the search engine results page.
     *
     * @generated from protobuf enum value: ORGANIC_ONLY = 3;
     */
    SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType[SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType["ORGANIC_ONLY"] = 3] = "ORGANIC_ONLY";
    /**
     * Both ads and organic results were contained in the search engine results
     * page.
     *
     * @generated from protobuf enum value: ADS_AND_ORGANIC = 4;
     */
    SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType[SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType["ADS_AND_ORGANIC"] = 4] = "ADS_AND_ORGANIC";
})(SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType = exports.SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType || (exports.SearchEngineResultsPageTypeEnum_SearchEngineResultsPageType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SearchEngineResultsPageTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SearchEngineResultsPageTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SearchEngineResultsPageTypeEnum
 */
exports.SearchEngineResultsPageTypeEnum = new SearchEngineResultsPageTypeEnum$Type();
