// google-ads-api-client/dist/protos/bindings/ads/enums/search_term_targeting_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchTermTargetingStatusEnum = exports.SearchTermTargetingStatusEnum_SearchTermTargetingStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Indicates whether the search term is one of your targeted or excluded
 * keywords.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SearchTermTargetingStatusEnum.SearchTermTargetingStatus
 */
var SearchTermTargetingStatusEnum_SearchTermTargetingStatus;
(function (SearchTermTargetingStatusEnum_SearchTermTargetingStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SearchTermTargetingStatusEnum_SearchTermTargetingStatus[SearchTermTargetingStatusEnum_SearchTermTargetingStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SearchTermTargetingStatusEnum_SearchTermTargetingStatus[SearchTermTargetingStatusEnum_SearchTermTargetingStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Search term is added to targeted keywords.
     *
     * @generated from protobuf enum value: ADDED = 2;
     */
    SearchTermTargetingStatusEnum_SearchTermTargetingStatus[SearchTermTargetingStatusEnum_SearchTermTargetingStatus["ADDED"] = 2] = "ADDED";
    /**
     * Search term matches a negative keyword.
     *
     * @generated from protobuf enum value: EXCLUDED = 3;
     */
    SearchTermTargetingStatusEnum_SearchTermTargetingStatus[SearchTermTargetingStatusEnum_SearchTermTargetingStatus["EXCLUDED"] = 3] = "EXCLUDED";
    /**
     * Search term has been both added and excluded.
     *
     * @generated from protobuf enum value: ADDED_EXCLUDED = 4;
     */
    SearchTermTargetingStatusEnum_SearchTermTargetingStatus[SearchTermTargetingStatusEnum_SearchTermTargetingStatus["ADDED_EXCLUDED"] = 4] = "ADDED_EXCLUDED";
    /**
     * Search term is neither targeted nor excluded.
     *
     * @generated from protobuf enum value: NONE = 5;
     */
    SearchTermTargetingStatusEnum_SearchTermTargetingStatus[SearchTermTargetingStatusEnum_SearchTermTargetingStatus["NONE"] = 5] = "NONE";
})(SearchTermTargetingStatusEnum_SearchTermTargetingStatus = exports.SearchTermTargetingStatusEnum_SearchTermTargetingStatus || (exports.SearchTermTargetingStatusEnum_SearchTermTargetingStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SearchTermTargetingStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SearchTermTargetingStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SearchTermTargetingStatusEnum
 */
exports.SearchTermTargetingStatusEnum = new SearchTermTargetingStatusEnum$Type();
