// google-ads-api-client/dist/protos/bindings/ads/enums/seasonality_event_scope.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeasonalityEventScopeEnum = exports.SeasonalityEventScopeEnum_SeasonalityEventScope = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible scopes of a Seasonality Event.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SeasonalityEventScopeEnum.SeasonalityEventScope
 */
var SeasonalityEventScopeEnum_SeasonalityEventScope;
(function (SeasonalityEventScopeEnum_SeasonalityEventScope) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SeasonalityEventScopeEnum_SeasonalityEventScope[SeasonalityEventScopeEnum_SeasonalityEventScope["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SeasonalityEventScopeEnum_SeasonalityEventScope[SeasonalityEventScopeEnum_SeasonalityEventScope["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The seasonality event is applied to all the customer's traffic for
     * supported advertising channel types and device types. The CUSTOMER scope
     * cannot be used in mutates.
     *
     * @generated from protobuf enum value: CUSTOMER = 2;
     */
    SeasonalityEventScopeEnum_SeasonalityEventScope[SeasonalityEventScopeEnum_SeasonalityEventScope["CUSTOMER"] = 2] = "CUSTOMER";
    /**
     * The seasonality event is applied to all specified campaigns.
     *
     * @generated from protobuf enum value: CAMPAIGN = 4;
     */
    SeasonalityEventScopeEnum_SeasonalityEventScope[SeasonalityEventScopeEnum_SeasonalityEventScope["CAMPAIGN"] = 4] = "CAMPAIGN";
    /**
     * The seasonality event is applied to all campaigns that belong to
     * specified channel types.
     *
     * @generated from protobuf enum value: CHANNEL = 5;
     */
    SeasonalityEventScopeEnum_SeasonalityEventScope[SeasonalityEventScopeEnum_SeasonalityEventScope["CHANNEL"] = 5] = "CHANNEL";
})(SeasonalityEventScopeEnum_SeasonalityEventScope = exports.SeasonalityEventScopeEnum_SeasonalityEventScope || (exports.SeasonalityEventScopeEnum_SeasonalityEventScope = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SeasonalityEventScopeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SeasonalityEventScopeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SeasonalityEventScopeEnum
 */
exports.SeasonalityEventScopeEnum = new SeasonalityEventScopeEnum$Type();
