// google-ads-api-client/dist/protos/bindings/ads/enums/seasonality_event_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeasonalityEventStatusEnum = exports.SeasonalityEventStatusEnum_SeasonalityEventStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible statuses of a Seasonality Event.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SeasonalityEventStatusEnum.SeasonalityEventStatus
 */
var SeasonalityEventStatusEnum_SeasonalityEventStatus;
(function (SeasonalityEventStatusEnum_SeasonalityEventStatus) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SeasonalityEventStatusEnum_SeasonalityEventStatus[SeasonalityEventStatusEnum_SeasonalityEventStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SeasonalityEventStatusEnum_SeasonalityEventStatus[SeasonalityEventStatusEnum_SeasonalityEventStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The seasonality event is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    SeasonalityEventStatusEnum_SeasonalityEventStatus[SeasonalityEventStatusEnum_SeasonalityEventStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The seasonality event is removed.
     *
     * @generated from protobuf enum value: REMOVED = 4;
     */
    SeasonalityEventStatusEnum_SeasonalityEventStatus[SeasonalityEventStatusEnum_SeasonalityEventStatus["REMOVED"] = 4] = "REMOVED";
})(SeasonalityEventStatusEnum_SeasonalityEventStatus = exports.SeasonalityEventStatusEnum_SeasonalityEventStatus || (exports.SeasonalityEventStatusEnum_SeasonalityEventStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SeasonalityEventStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SeasonalityEventStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SeasonalityEventStatusEnum
 */
exports.SeasonalityEventStatusEnum = new SeasonalityEventStatusEnum$Type();
