// google-ads-api-client/dist/protos/bindings/ads/enums/served_asset_field_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServedAssetFieldTypeEnum = exports.ServedAssetFieldTypeEnum_ServedAssetFieldType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible asset field types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ServedAssetFieldTypeEnum.ServedAssetFieldType
 */
var ServedAssetFieldTypeEnum_ServedAssetFieldType;
(function (ServedAssetFieldTypeEnum_ServedAssetFieldType) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ServedAssetFieldTypeEnum_ServedAssetFieldType[ServedAssetFieldTypeEnum_ServedAssetFieldType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     *
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ServedAssetFieldTypeEnum_ServedAssetFieldType[ServedAssetFieldTypeEnum_ServedAssetFieldType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The asset is used in headline 1.
     *
     * @generated from protobuf enum value: HEADLINE_1 = 2;
     */
    ServedAssetFieldTypeEnum_ServedAssetFieldType[ServedAssetFieldTypeEnum_ServedAssetFieldType["HEADLINE_1"] = 2] = "HEADLINE_1";
    /**
     * The asset is used in headline 2.
     *
     * @generated from protobuf enum value: HEADLINE_2 = 3;
     */
    ServedAssetFieldTypeEnum_ServedAssetFieldType[ServedAssetFieldTypeEnum_ServedAssetFieldType["HEADLINE_2"] = 3] = "HEADLINE_2";
    /**
     * The asset is used in headline 3.
     *
     * @generated from protobuf enum value: HEADLINE_3 = 4;
     */
    ServedAssetFieldTypeEnum_ServedAssetFieldType[ServedAssetFieldTypeEnum_ServedAssetFieldType["HEADLINE_3"] = 4] = "HEADLINE_3";
    /**
     * The asset is used in description 1.
     *
     * @generated from protobuf enum value: DESCRIPTION_1 = 5;
     */
    ServedAssetFieldTypeEnum_ServedAssetFieldType[ServedAssetFieldTypeEnum_ServedAssetFieldType["DESCRIPTION_1"] = 5] = "DESCRIPTION_1";
    /**
     * The asset is used in description 2.
     *
     * @generated from protobuf enum value: DESCRIPTION_2 = 6;
     */
    ServedAssetFieldTypeEnum_ServedAssetFieldType[ServedAssetFieldTypeEnum_ServedAssetFieldType["DESCRIPTION_2"] = 6] = "DESCRIPTION_2";
})(ServedAssetFieldTypeEnum_ServedAssetFieldType = exports.ServedAssetFieldTypeEnum_ServedAssetFieldType || (exports.ServedAssetFieldTypeEnum_ServedAssetFieldType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ServedAssetFieldTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ServedAssetFieldTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ServedAssetFieldTypeEnum
 */
exports.ServedAssetFieldTypeEnum = new ServedAssetFieldTypeEnum$Type();
