// google-ads-api-client/dist/protos/bindings/ads/enums/shared_set_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedSetTypeEnum = exports.SharedSetTypeEnum_SharedSetType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the possible shared set types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SharedSetTypeEnum.SharedSetType
 */
var SharedSetTypeEnum_SharedSetType;
(function (SharedSetTypeEnum_SharedSetType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SharedSetTypeEnum_SharedSetType[SharedSetTypeEnum_SharedSetType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SharedSetTypeEnum_SharedSetType[SharedSetTypeEnum_SharedSetType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A set of keywords that can be excluded from targeting.
     *
     * @generated from protobuf enum value: NEGATIVE_KEYWORDS = 2;
     */
    SharedSetTypeEnum_SharedSetType[SharedSetTypeEnum_SharedSetType["NEGATIVE_KEYWORDS"] = 2] = "NEGATIVE_KEYWORDS";
    /**
     * A set of placements that can be excluded from targeting.
     *
     * @generated from protobuf enum value: NEGATIVE_PLACEMENTS = 3;
     */
    SharedSetTypeEnum_SharedSetType[SharedSetTypeEnum_SharedSetType["NEGATIVE_PLACEMENTS"] = 3] = "NEGATIVE_PLACEMENTS";
})(SharedSetTypeEnum_SharedSetType = exports.SharedSetTypeEnum_SharedSetType || (exports.SharedSetTypeEnum_SharedSetType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SharedSetTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SharedSetTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SharedSetTypeEnum
 */
exports.SharedSetTypeEnum = new SharedSetTypeEnum$Type();
