// google-ads-api-client/dist/protos/bindings/ads/enums/simulation_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimulationTypeEnum = exports.SimulationTypeEnum_SimulationType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the field a simulation modifies.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType
 */
var SimulationTypeEnum_SimulationType;
(function (SimulationTypeEnum_SimulationType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The simulation is for a CPC bid.
     *
     * @generated from protobuf enum value: CPC_BID = 2;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["CPC_BID"] = 2] = "CPC_BID";
    /**
     * The simulation is for a CPV bid.
     *
     * @generated from protobuf enum value: CPV_BID = 3;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["CPV_BID"] = 3] = "CPV_BID";
    /**
     * The simulation is for a CPA target.
     *
     * @generated from protobuf enum value: TARGET_CPA = 4;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["TARGET_CPA"] = 4] = "TARGET_CPA";
    /**
     * The simulation is for a bid modifier.
     *
     * @generated from protobuf enum value: BID_MODIFIER = 5;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["BID_MODIFIER"] = 5] = "BID_MODIFIER";
    /**
     * The simulation is for a ROAS target.
     *
     * @generated from protobuf enum value: TARGET_ROAS = 6;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["TARGET_ROAS"] = 6] = "TARGET_ROAS";
    /**
     * The simulation is for a percent CPC bid.
     *
     * @generated from protobuf enum value: PERCENT_CPC_BID = 7;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["PERCENT_CPC_BID"] = 7] = "PERCENT_CPC_BID";
    /**
     * The simulation is for an impression share target.
     *
     * @generated from protobuf enum value: TARGET_IMPRESSION_SHARE = 8;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["TARGET_IMPRESSION_SHARE"] = 8] = "TARGET_IMPRESSION_SHARE";
    /**
     * The simulation is for a budget.
     *
     * @generated from protobuf enum value: BUDGET = 9;
     */
    SimulationTypeEnum_SimulationType[SimulationTypeEnum_SimulationType["BUDGET"] = 9] = "BUDGET";
})(SimulationTypeEnum_SimulationType = exports.SimulationTypeEnum_SimulationType || (exports.SimulationTypeEnum_SimulationType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SimulationTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SimulationTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SimulationTypeEnum
 */
exports.SimulationTypeEnum = new SimulationTypeEnum$Type();
