// google-ads-api-client/dist/protos/bindings/ads/enums/simulation_modification_method.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimulationModificationMethodEnum = exports.SimulationModificationMethodEnum_SimulationModificationMethod = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing the method by which a simulation modifies a field.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod
 */
var SimulationModificationMethodEnum_SimulationModificationMethod;
(function (SimulationModificationMethodEnum_SimulationModificationMethod) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SimulationModificationMethodEnum_SimulationModificationMethod[SimulationModificationMethodEnum_SimulationModificationMethod["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SimulationModificationMethodEnum_SimulationModificationMethod[SimulationModificationMethodEnum_SimulationModificationMethod["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The values in a simulation were applied to all children of a given
     * resource uniformly. Overrides on child resources were not respected.
     *
     * @generated from protobuf enum value: UNIFORM = 2;
     */
    SimulationModificationMethodEnum_SimulationModificationMethod[SimulationModificationMethodEnum_SimulationModificationMethod["UNIFORM"] = 2] = "UNIFORM";
    /**
     * The values in a simulation were applied to the given resource.
     * Overrides on child resources were respected, and traffic estimates
     * do not include these resources.
     *
     * @generated from protobuf enum value: DEFAULT = 3;
     */
    SimulationModificationMethodEnum_SimulationModificationMethod[SimulationModificationMethodEnum_SimulationModificationMethod["DEFAULT"] = 3] = "DEFAULT";
    /**
     * The values in a simulation were all scaled by the same factor.
     * For example, in a simulated TargetCpa campaign, the campaign target and
     * all ad group targets were scaled by a factor of X.
     *
     * @generated from protobuf enum value: SCALING = 4;
     */
    SimulationModificationMethodEnum_SimulationModificationMethod[SimulationModificationMethodEnum_SimulationModificationMethod["SCALING"] = 4] = "SCALING";
})(SimulationModificationMethodEnum_SimulationModificationMethod = exports.SimulationModificationMethodEnum_SimulationModificationMethod || (exports.SimulationModificationMethodEnum_SimulationModificationMethod = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SimulationModificationMethodEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SimulationModificationMethodEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SimulationModificationMethodEnum
 */
exports.SimulationModificationMethodEnum = new SimulationModificationMethodEnum$Type();
