// google-ads-api-client/dist/protos/bindings/ads/enums/sitelink_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SitelinkPlaceholderFieldEnum = exports.SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Sitelink placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SitelinkPlaceholderFieldEnum.SitelinkPlaceholderField
 */
var SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField;
(function (SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. The link text for your sitelink.
     *
     * @generated from protobuf enum value: TEXT = 2;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["TEXT"] = 2] = "TEXT";
    /**
     * Data Type: STRING. First line of the sitelink description.
     *
     * @generated from protobuf enum value: LINE_1 = 3;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["LINE_1"] = 3] = "LINE_1";
    /**
     * Data Type: STRING. Second line of the sitelink description.
     *
     * @generated from protobuf enum value: LINE_2 = 4;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["LINE_2"] = 4] = "LINE_2";
    /**
     * Data Type: URL_LIST. Final URLs for the sitelink when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_URLS = 5;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["FINAL_URLS"] = 5] = "FINAL_URLS";
    /**
     * Data Type: URL_LIST. Final Mobile URLs for the sitelink when using
     * Upgraded URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 6;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["FINAL_MOBILE_URLS"] = 6] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the sitelink when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 7;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["TRACKING_URL"] = 7] = "TRACKING_URL";
    /**
     * Data Type: STRING. Final URL suffix for sitelink when using parallel
     * tracking.
     *
     * @generated from protobuf enum value: FINAL_URL_SUFFIX = 8;
     */
    SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField[SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField["FINAL_URL_SUFFIX"] = 8] = "FINAL_URL_SUFFIX";
})(SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField = exports.SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField || (exports.SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SitelinkPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SitelinkPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SitelinkPlaceholderFieldEnum
 */
exports.SitelinkPlaceholderFieldEnum = new SitelinkPlaceholderFieldEnum$Type();
