// google-ads-api-client/dist/protos/bindings/ads/enums/sk_ad_network_attribution_credit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkAdNetworkAttributionCreditEnum = exports.SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates SkAdNetwork attribution credits.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SkAdNetworkAttributionCreditEnum.SkAdNetworkAttributionCredit
 */
var SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit;
(function (SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit) {
    /**
     * Default value. This value is equivalent to null.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit[SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this API version. The true enum value cannot be
     * returned in this API version or is not supported yet.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit[SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The value was not present in the postback or we do not have this data for
     * other reasons.
     *
     * @generated from protobuf enum value: UNAVAILABLE = 2;
     */
    SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit[SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit["UNAVAILABLE"] = 2] = "UNAVAILABLE";
    /**
     * Google was the ad network that won ad attribution.
     *
     * @generated from protobuf enum value: WON = 3;
     */
    SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit[SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit["WON"] = 3] = "WON";
    /**
     * Google qualified for attribution, but didn't win.
     *
     * @generated from protobuf enum value: CONTRIBUTED = 4;
     */
    SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit[SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit["CONTRIBUTED"] = 4] = "CONTRIBUTED";
})(SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit = exports.SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit || (exports.SkAdNetworkAttributionCreditEnum_SkAdNetworkAttributionCredit = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SkAdNetworkAttributionCreditEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SkAdNetworkAttributionCreditEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SkAdNetworkAttributionCreditEnum
 */
exports.SkAdNetworkAttributionCreditEnum = new SkAdNetworkAttributionCreditEnum$Type();
