// google-ads-api-client/dist/protos/bindings/ads/enums/sk_ad_network_ad_event_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkAdNetworkAdEventTypeEnum = exports.SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates SkAdNetwork ad event types
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SkAdNetworkAdEventTypeEnum.SkAdNetworkAdEventType
 */
var SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType;
(function (SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType[SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType[SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The value was not present in the postback or we do not have this data for
     * other reasons.
     *
     * @generated from protobuf enum value: UNAVAILABLE = 2;
     */
    SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType[SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType["UNAVAILABLE"] = 2] = "UNAVAILABLE";
    /**
     * The user interacted with the ad.
     *
     * @generated from protobuf enum value: INTERACTION = 3;
     */
    SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType[SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType["INTERACTION"] = 3] = "INTERACTION";
    /**
     * The user viewed the ad.
     *
     * @generated from protobuf enum value: VIEW = 4;
     */
    SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType[SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType["VIEW"] = 4] = "VIEW";
})(SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType = exports.SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType || (exports.SkAdNetworkAdEventTypeEnum_SkAdNetworkAdEventType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SkAdNetworkAdEventTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SkAdNetworkAdEventTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SkAdNetworkAdEventTypeEnum
 */
exports.SkAdNetworkAdEventTypeEnum = new SkAdNetworkAdEventTypeEnum$Type();
