// google-ads-api-client/dist/protos/bindings/ads/enums/sk_ad_network_user_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkAdNetworkUserTypeEnum = exports.SkAdNetworkUserTypeEnum_SkAdNetworkUserType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enumerates SkAdNetwork user types
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SkAdNetworkUserTypeEnum.SkAdNetworkUserType
 */
var SkAdNetworkUserTypeEnum_SkAdNetworkUserType;
(function (SkAdNetworkUserTypeEnum_SkAdNetworkUserType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SkAdNetworkUserTypeEnum_SkAdNetworkUserType[SkAdNetworkUserTypeEnum_SkAdNetworkUserType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The value is unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SkAdNetworkUserTypeEnum_SkAdNetworkUserType[SkAdNetworkUserTypeEnum_SkAdNetworkUserType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The value was not present in the postback or we do not have this data for
     * other reasons.
     *
     * @generated from protobuf enum value: UNAVAILABLE = 2;
     */
    SkAdNetworkUserTypeEnum_SkAdNetworkUserType[SkAdNetworkUserTypeEnum_SkAdNetworkUserType["UNAVAILABLE"] = 2] = "UNAVAILABLE";
    /**
     * The user installed the app for the first time.
     *
     * @generated from protobuf enum value: NEW_INSTALLER = 3;
     */
    SkAdNetworkUserTypeEnum_SkAdNetworkUserType[SkAdNetworkUserTypeEnum_SkAdNetworkUserType["NEW_INSTALLER"] = 3] = "NEW_INSTALLER";
    /**
     * The user has previously installed the app.
     *
     * @generated from protobuf enum value: REINSTALLER = 4;
     */
    SkAdNetworkUserTypeEnum_SkAdNetworkUserType[SkAdNetworkUserTypeEnum_SkAdNetworkUserType["REINSTALLER"] = 4] = "REINSTALLER";
})(SkAdNetworkUserTypeEnum_SkAdNetworkUserType = exports.SkAdNetworkUserTypeEnum_SkAdNetworkUserType || (exports.SkAdNetworkUserTypeEnum_SkAdNetworkUserType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SkAdNetworkUserTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SkAdNetworkUserTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SkAdNetworkUserTypeEnum
 */
exports.SkAdNetworkUserTypeEnum = new SkAdNetworkUserTypeEnum$Type();
