// google-ads-api-client/dist/protos/bindings/ads/enums/spending_limit_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpendingLimitTypeEnum = exports.SpendingLimitTypeEnum_SpendingLimitType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible spending limit types used by certain resources as an
 * alternative to absolute money values in micros.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType
 */
var SpendingLimitTypeEnum_SpendingLimitType;
(function (SpendingLimitTypeEnum_SpendingLimitType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SpendingLimitTypeEnum_SpendingLimitType[SpendingLimitTypeEnum_SpendingLimitType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SpendingLimitTypeEnum_SpendingLimitType[SpendingLimitTypeEnum_SpendingLimitType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Infinite, indicates unlimited spending power.
     *
     * @generated from protobuf enum value: INFINITE = 2;
     */
    SpendingLimitTypeEnum_SpendingLimitType[SpendingLimitTypeEnum_SpendingLimitType["INFINITE"] = 2] = "INFINITE";
})(SpendingLimitTypeEnum_SpendingLimitType = exports.SpendingLimitTypeEnum_SpendingLimitType || (exports.SpendingLimitTypeEnum_SpendingLimitType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SpendingLimitTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SpendingLimitTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SpendingLimitTypeEnum
 */
exports.SpendingLimitTypeEnum = new SpendingLimitTypeEnum$Type();
