// google-ads-api-client/dist/protos/bindings/ads/enums/structured_snippet_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StructuredSnippetPlaceholderFieldEnum = exports.StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Structured Snippet placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.StructuredSnippetPlaceholderFieldEnum.StructuredSnippetPlaceholderField
 */
var StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField;
(function (StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField[StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField[StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. The category of snippet of your products/services.
     * Must match exactly one of the predefined structured snippets headers.
     * For a list, visit
     * https://developers.google.com/adwords/api/docs/appendix/structured-snippet-headers
     *
     * @generated from protobuf enum value: HEADER = 2;
     */
    StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField[StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField["HEADER"] = 2] = "HEADER";
    /**
     * Data Type: STRING_LIST. Text values that describe your products/services.
     * All text must be family safe. Special or non-ASCII characters are not
     * permitted. A snippet can be at most 25 characters.
     *
     * @generated from protobuf enum value: SNIPPETS = 3;
     */
    StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField[StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField["SNIPPETS"] = 3] = "SNIPPETS";
})(StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField = exports.StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField || (exports.StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class StructuredSnippetPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.StructuredSnippetPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.StructuredSnippetPlaceholderFieldEnum
 */
exports.StructuredSnippetPlaceholderFieldEnum = new StructuredSnippetPlaceholderFieldEnum$Type();
