// google-ads-api-client/dist/protos/bindings/ads/enums/summary_row_setting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummaryRowSettingEnum = exports.SummaryRowSettingEnum_SummaryRowSetting = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing return summary row settings.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.SummaryRowSettingEnum.SummaryRowSetting
 */
var SummaryRowSettingEnum_SummaryRowSetting;
(function (SummaryRowSettingEnum_SummaryRowSetting) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SummaryRowSettingEnum_SummaryRowSetting[SummaryRowSettingEnum_SummaryRowSetting["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Represent unknown values of return summary row.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SummaryRowSettingEnum_SummaryRowSetting[SummaryRowSettingEnum_SummaryRowSetting["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Do not return summary row.
     *
     * @generated from protobuf enum value: NO_SUMMARY_ROW = 2;
     */
    SummaryRowSettingEnum_SummaryRowSetting[SummaryRowSettingEnum_SummaryRowSetting["NO_SUMMARY_ROW"] = 2] = "NO_SUMMARY_ROW";
    /**
     * Return summary row along with results. The summary row will be returned
     * in the last batch alone (last batch will contain no results).
     *
     * @generated from protobuf enum value: SUMMARY_ROW_WITH_RESULTS = 3;
     */
    SummaryRowSettingEnum_SummaryRowSetting[SummaryRowSettingEnum_SummaryRowSetting["SUMMARY_ROW_WITH_RESULTS"] = 3] = "SUMMARY_ROW_WITH_RESULTS";
    /**
     * Return summary row only and return no results.
     *
     * @generated from protobuf enum value: SUMMARY_ROW_ONLY = 4;
     */
    SummaryRowSettingEnum_SummaryRowSetting[SummaryRowSettingEnum_SummaryRowSetting["SUMMARY_ROW_ONLY"] = 4] = "SUMMARY_ROW_ONLY";
})(SummaryRowSettingEnum_SummaryRowSetting = exports.SummaryRowSettingEnum_SummaryRowSetting || (exports.SummaryRowSettingEnum_SummaryRowSetting = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SummaryRowSettingEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.SummaryRowSettingEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.SummaryRowSettingEnum
 */
exports.SummaryRowSettingEnum = new SummaryRowSettingEnum$Type();
