// google-ads-api-client/dist/protos/bindings/ads/enums/target_cpa_opt_in_recommendation_goal.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetCpaOptInRecommendationGoalEnum = exports.TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Goal of TargetCpaOptIn recommendation.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.TargetCpaOptInRecommendationGoalEnum.TargetCpaOptInRecommendationGoal
 */
var TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal;
(function (TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal[TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal[TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Recommendation to set Target CPA to maintain the same cost.
     *
     * @generated from protobuf enum value: SAME_COST = 2;
     */
    TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal[TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal["SAME_COST"] = 2] = "SAME_COST";
    /**
     * Recommendation to set Target CPA to maintain the same conversions.
     *
     * @generated from protobuf enum value: SAME_CONVERSIONS = 3;
     */
    TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal[TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal["SAME_CONVERSIONS"] = 3] = "SAME_CONVERSIONS";
    /**
     * Recommendation to set Target CPA to maintain the same CPA.
     *
     * @generated from protobuf enum value: SAME_CPA = 4;
     */
    TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal[TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal["SAME_CPA"] = 4] = "SAME_CPA";
    /**
     * Recommendation to set Target CPA to a value that is as close as possible
     * to, yet lower than, the actual CPA (computed for past 28 days).
     *
     * @generated from protobuf enum value: CLOSEST_CPA = 5;
     */
    TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal[TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal["CLOSEST_CPA"] = 5] = "CLOSEST_CPA";
})(TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal = exports.TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal || (exports.TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal = {}));
// @generated message type with reflection information, may provide speed optimized methods
class TargetCpaOptInRecommendationGoalEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.TargetCpaOptInRecommendationGoalEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.TargetCpaOptInRecommendationGoalEnum
 */
exports.TargetCpaOptInRecommendationGoalEnum = new TargetCpaOptInRecommendationGoalEnum$Type();
