// google-ads-api-client/dist/protos/bindings/ads/enums/target_impression_share_location.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetImpressionShareLocationEnum = exports.TargetImpressionShareLocationEnum_TargetImpressionShareLocation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible goals.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.TargetImpressionShareLocationEnum.TargetImpressionShareLocation
 */
var TargetImpressionShareLocationEnum_TargetImpressionShareLocation;
(function (TargetImpressionShareLocationEnum_TargetImpressionShareLocation) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    TargetImpressionShareLocationEnum_TargetImpressionShareLocation[TargetImpressionShareLocationEnum_TargetImpressionShareLocation["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    TargetImpressionShareLocationEnum_TargetImpressionShareLocation[TargetImpressionShareLocationEnum_TargetImpressionShareLocation["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Any location on the web page.
     *
     * @generated from protobuf enum value: ANYWHERE_ON_PAGE = 2;
     */
    TargetImpressionShareLocationEnum_TargetImpressionShareLocation[TargetImpressionShareLocationEnum_TargetImpressionShareLocation["ANYWHERE_ON_PAGE"] = 2] = "ANYWHERE_ON_PAGE";
    /**
     * Top box of ads.
     *
     * @generated from protobuf enum value: TOP_OF_PAGE = 3;
     */
    TargetImpressionShareLocationEnum_TargetImpressionShareLocation[TargetImpressionShareLocationEnum_TargetImpressionShareLocation["TOP_OF_PAGE"] = 3] = "TOP_OF_PAGE";
    /**
     * Top slot in the top box of ads.
     *
     * @generated from protobuf enum value: ABSOLUTE_TOP_OF_PAGE = 4;
     */
    TargetImpressionShareLocationEnum_TargetImpressionShareLocation[TargetImpressionShareLocationEnum_TargetImpressionShareLocation["ABSOLUTE_TOP_OF_PAGE"] = 4] = "ABSOLUTE_TOP_OF_PAGE";
})(TargetImpressionShareLocationEnum_TargetImpressionShareLocation = exports.TargetImpressionShareLocationEnum_TargetImpressionShareLocation || (exports.TargetImpressionShareLocationEnum_TargetImpressionShareLocation = {}));
// @generated message type with reflection information, may provide speed optimized methods
class TargetImpressionShareLocationEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.TargetImpressionShareLocationEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.TargetImpressionShareLocationEnum
 */
exports.TargetImpressionShareLocationEnum = new TargetImpressionShareLocationEnum$Type();
