// google-ads-api-client/dist/protos/bindings/ads/enums/targeting_dimension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetingDimensionEnum = exports.TargetingDimensionEnum_TargetingDimension = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible targeting dimensions.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.TargetingDimensionEnum.TargetingDimension
 */
var TargetingDimensionEnum_TargetingDimension;
(function (TargetingDimensionEnum_TargetingDimension) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Keyword criteria, for example, 'mars cruise'. KEYWORD may be used as a
     * custom bid dimension. Keywords are always a targeting dimension, so may
     * not be set as a target "ALL" dimension with TargetRestriction.
     *
     * @generated from protobuf enum value: KEYWORD = 2;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["KEYWORD"] = 2] = "KEYWORD";
    /**
     * Audience criteria, which include user list, user interest, custom
     * affinity,  and custom in market.
     *
     * @generated from protobuf enum value: AUDIENCE = 3;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["AUDIENCE"] = 3] = "AUDIENCE";
    /**
     * Topic criteria for targeting categories of content, for example,
     * 'category::Animals>Pets' Used for Display and Video targeting.
     *
     * @generated from protobuf enum value: TOPIC = 4;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["TOPIC"] = 4] = "TOPIC";
    /**
     * Criteria for targeting gender.
     *
     * @generated from protobuf enum value: GENDER = 5;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["GENDER"] = 5] = "GENDER";
    /**
     * Criteria for targeting age ranges.
     *
     * @generated from protobuf enum value: AGE_RANGE = 6;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["AGE_RANGE"] = 6] = "AGE_RANGE";
    /**
     * Placement criteria, which include websites like 'www.flowers4sale.com',
     * as well as mobile applications, mobile app categories, YouTube videos,
     * and YouTube channels.
     *
     * @generated from protobuf enum value: PLACEMENT = 7;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["PLACEMENT"] = 7] = "PLACEMENT";
    /**
     * Criteria for parental status targeting.
     *
     * @generated from protobuf enum value: PARENTAL_STATUS = 8;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["PARENTAL_STATUS"] = 8] = "PARENTAL_STATUS";
    /**
     * Criteria for income range targeting.
     *
     * @generated from protobuf enum value: INCOME_RANGE = 9;
     */
    TargetingDimensionEnum_TargetingDimension[TargetingDimensionEnum_TargetingDimension["INCOME_RANGE"] = 9] = "INCOME_RANGE";
})(TargetingDimensionEnum_TargetingDimension = exports.TargetingDimensionEnum_TargetingDimension || (exports.TargetingDimensionEnum_TargetingDimension = {}));
// @generated message type with reflection information, may provide speed optimized methods
class TargetingDimensionEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.TargetingDimensionEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.TargetingDimensionEnum
 */
exports.TargetingDimensionEnum = new TargetingDimensionEnum$Type();
