// google-ads-api-client/dist/protos/bindings/ads/enums/time_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeTypeEnum = exports.TimeTypeEnum_TimeType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible time types used by certain resources as an alternative to
 * absolute timestamps.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.TimeTypeEnum.TimeType
 */
var TimeTypeEnum_TimeType;
(function (TimeTypeEnum_TimeType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    TimeTypeEnum_TimeType[TimeTypeEnum_TimeType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    TimeTypeEnum_TimeType[TimeTypeEnum_TimeType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * As soon as possible.
     *
     * @generated from protobuf enum value: NOW = 2;
     */
    TimeTypeEnum_TimeType[TimeTypeEnum_TimeType["NOW"] = 2] = "NOW";
    /**
     * An infinite point in the future.
     *
     * @generated from protobuf enum value: FOREVER = 3;
     */
    TimeTypeEnum_TimeType[TimeTypeEnum_TimeType["FOREVER"] = 3] = "FOREVER";
})(TimeTypeEnum_TimeType = exports.TimeTypeEnum_TimeType || (exports.TimeTypeEnum_TimeType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class TimeTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.TimeTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.TimeTypeEnum
 */
exports.TimeTypeEnum = new TimeTypeEnum$Type();
