// google-ads-api-client/dist/protos/bindings/ads/enums/tracking_code_page_format.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrackingCodePageFormatEnum = exports.TrackingCodePageFormatEnum_TrackingCodePageFormat = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The format of the web page where the tracking tag and snippet will be
 * installed.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.TrackingCodePageFormatEnum.TrackingCodePageFormat
 */
var TrackingCodePageFormatEnum_TrackingCodePageFormat;
(function (TrackingCodePageFormatEnum_TrackingCodePageFormat) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    TrackingCodePageFormatEnum_TrackingCodePageFormat[TrackingCodePageFormatEnum_TrackingCodePageFormat["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    TrackingCodePageFormatEnum_TrackingCodePageFormat[TrackingCodePageFormatEnum_TrackingCodePageFormat["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Standard HTML page format.
     *
     * @generated from protobuf enum value: HTML = 2;
     */
    TrackingCodePageFormatEnum_TrackingCodePageFormat[TrackingCodePageFormatEnum_TrackingCodePageFormat["HTML"] = 2] = "HTML";
    /**
     * Google AMP page format.
     *
     * @generated from protobuf enum value: AMP = 3;
     */
    TrackingCodePageFormatEnum_TrackingCodePageFormat[TrackingCodePageFormatEnum_TrackingCodePageFormat["AMP"] = 3] = "AMP";
})(TrackingCodePageFormatEnum_TrackingCodePageFormat = exports.TrackingCodePageFormatEnum_TrackingCodePageFormat || (exports.TrackingCodePageFormatEnum_TrackingCodePageFormat = {}));
// @generated message type with reflection information, may provide speed optimized methods
class TrackingCodePageFormatEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.TrackingCodePageFormatEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.TrackingCodePageFormatEnum
 */
exports.TrackingCodePageFormatEnum = new TrackingCodePageFormatEnum$Type();
