// google-ads-api-client/dist/protos/bindings/ads/enums/tracking_code_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrackingCodeTypeEnum = exports.TrackingCodeTypeEnum_TrackingCodeType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of the generated tag snippets for tracking conversions.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.TrackingCodeTypeEnum.TrackingCodeType
 */
var TrackingCodeTypeEnum_TrackingCodeType;
(function (TrackingCodeTypeEnum_TrackingCodeType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    TrackingCodeTypeEnum_TrackingCodeType[TrackingCodeTypeEnum_TrackingCodeType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    TrackingCodeTypeEnum_TrackingCodeType[TrackingCodeTypeEnum_TrackingCodeType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The snippet that is fired as a result of a website page loading.
     *
     * @generated from protobuf enum value: WEBPAGE = 2;
     */
    TrackingCodeTypeEnum_TrackingCodeType[TrackingCodeTypeEnum_TrackingCodeType["WEBPAGE"] = 2] = "WEBPAGE";
    /**
     * The snippet contains a JavaScript function which fires the tag. This
     * function is typically called from an onClick handler added to a link or
     * button element on the page.
     *
     * @generated from protobuf enum value: WEBPAGE_ONCLICK = 3;
     */
    TrackingCodeTypeEnum_TrackingCodeType[TrackingCodeTypeEnum_TrackingCodeType["WEBPAGE_ONCLICK"] = 3] = "WEBPAGE_ONCLICK";
    /**
     * For embedding on a mobile webpage. The snippet contains a JavaScript
     * function which fires the tag.
     *
     * @generated from protobuf enum value: CLICK_TO_CALL = 4;
     */
    TrackingCodeTypeEnum_TrackingCodeType[TrackingCodeTypeEnum_TrackingCodeType["CLICK_TO_CALL"] = 4] = "CLICK_TO_CALL";
    /**
     * The snippet that is used to replace the phone number on your website with
     * a Google forwarding number for call tracking purposes.
     *
     * @generated from protobuf enum value: WEBSITE_CALL = 5;
     */
    TrackingCodeTypeEnum_TrackingCodeType[TrackingCodeTypeEnum_TrackingCodeType["WEBSITE_CALL"] = 5] = "WEBSITE_CALL";
})(TrackingCodeTypeEnum_TrackingCodeType = exports.TrackingCodeTypeEnum_TrackingCodeType || (exports.TrackingCodeTypeEnum_TrackingCodeType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class TrackingCodeTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.TrackingCodeTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.TrackingCodeTypeEnum
 */
exports.TrackingCodeTypeEnum = new TrackingCodeTypeEnum$Type();
