// google-ads-api-client/dist/protos/bindings/ads/enums/travel_placeholder_field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TravelPlaceholderFieldEnum = exports.TravelPlaceholderFieldEnum_TravelPlaceholderField = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible values for Travel placeholder fields.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.TravelPlaceholderFieldEnum.TravelPlaceholderField
 */
var TravelPlaceholderFieldEnum_TravelPlaceholderField;
(function (TravelPlaceholderFieldEnum_TravelPlaceholderField) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Data Type: STRING. Required. Destination id. Example: PAR, LON.
     * For feed items that only have destination id, destination id must be a
     * unique key. For feed items that have both destination id and origin id,
     * then the combination must be a unique key.
     *
     * @generated from protobuf enum value: DESTINATION_ID = 2;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["DESTINATION_ID"] = 2] = "DESTINATION_ID";
    /**
     * Data Type: STRING. Origin id. Example: PAR, LON.
     * Combination of DESTINATION_ID and ORIGIN_ID must be
     * unique per offer.
     *
     * @generated from protobuf enum value: ORIGIN_ID = 3;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["ORIGIN_ID"] = 3] = "ORIGIN_ID";
    /**
     * Data Type: STRING. Required. Main headline with name to be shown in
     * dynamic ad.
     *
     * @generated from protobuf enum value: TITLE = 4;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["TITLE"] = 4] = "TITLE";
    /**
     * Data Type: STRING. The destination name. Shorter names are recommended.
     *
     * @generated from protobuf enum value: DESTINATION_NAME = 5;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["DESTINATION_NAME"] = 5] = "DESTINATION_NAME";
    /**
     * Data Type: STRING. Origin name. Shorter names are recommended.
     *
     * @generated from protobuf enum value: ORIGIN_NAME = 6;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["ORIGIN_NAME"] = 6] = "ORIGIN_NAME";
    /**
     * Data Type: STRING. Price to be shown in the ad. Highly recommended for
     * dynamic ads.
     * Example: "100.00 USD"
     *
     * @generated from protobuf enum value: PRICE = 7;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["PRICE"] = 7] = "PRICE";
    /**
     * Data Type: STRING. Formatted price to be shown in the ad.
     * Example: "Starting at $100.00 USD", "$80 - $100"
     *
     * @generated from protobuf enum value: FORMATTED_PRICE = 8;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["FORMATTED_PRICE"] = 8] = "FORMATTED_PRICE";
    /**
     * Data Type: STRING. Sale price to be shown in the ad.
     * Example: "80.00 USD"
     *
     * @generated from protobuf enum value: SALE_PRICE = 9;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["SALE_PRICE"] = 9] = "SALE_PRICE";
    /**
     * Data Type: STRING. Formatted sale price to be shown in the ad.
     * Example: "On sale for $80.00", "$60 - $80"
     *
     * @generated from protobuf enum value: FORMATTED_SALE_PRICE = 10;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["FORMATTED_SALE_PRICE"] = 10] = "FORMATTED_SALE_PRICE";
    /**
     * Data Type: URL. Image to be displayed in the ad.
     *
     * @generated from protobuf enum value: IMAGE_URL = 11;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["IMAGE_URL"] = 11] = "IMAGE_URL";
    /**
     * Data Type: STRING. Category of travel offer used to group like items
     * together for recommendation engine.
     *
     * @generated from protobuf enum value: CATEGORY = 12;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["CATEGORY"] = 12] = "CATEGORY";
    /**
     * Data Type: STRING_LIST. Keywords used for product retrieval.
     *
     * @generated from protobuf enum value: CONTEXTUAL_KEYWORDS = 13;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["CONTEXTUAL_KEYWORDS"] = 13] = "CONTEXTUAL_KEYWORDS";
    /**
     * Data Type: STRING. Address of travel offer, including postal code.
     *
     * @generated from protobuf enum value: DESTINATION_ADDRESS = 14;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["DESTINATION_ADDRESS"] = 14] = "DESTINATION_ADDRESS";
    /**
     * Data Type: URL_LIST. Required. Final URLs to be used in ad, when using
     * Upgraded URLs; the more specific the better (for example, the individual
     * URL of a specific travel offer and its location).
     *
     * @generated from protobuf enum value: FINAL_URL = 15;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["FINAL_URL"] = 15] = "FINAL_URL";
    /**
     * Data Type: URL_LIST. Final mobile URLs for the ad when using Upgraded
     * URLs.
     *
     * @generated from protobuf enum value: FINAL_MOBILE_URLS = 16;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["FINAL_MOBILE_URLS"] = 16] = "FINAL_MOBILE_URLS";
    /**
     * Data Type: URL. Tracking template for the ad when using Upgraded URLs.
     *
     * @generated from protobuf enum value: TRACKING_URL = 17;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["TRACKING_URL"] = 17] = "TRACKING_URL";
    /**
     * Data Type: STRING. Android app link. Must be formatted as:
     * android-app://{package_id}/{scheme}/{host_path}.
     * The components are defined as follows:
     * package_id: app ID as specified in Google Play.
     * scheme: the scheme to pass to the application. Can be HTTP, or a custom
     *   scheme.
     * host_path: identifies the specific content within your application.
     *
     * @generated from protobuf enum value: ANDROID_APP_LINK = 18;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["ANDROID_APP_LINK"] = 18] = "ANDROID_APP_LINK";
    /**
     * Data Type: STRING_LIST. List of recommended destination IDs to show
     * together with this item.
     *
     * @generated from protobuf enum value: SIMILAR_DESTINATION_IDS = 19;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["SIMILAR_DESTINATION_IDS"] = 19] = "SIMILAR_DESTINATION_IDS";
    /**
     * Data Type: STRING. iOS app link.
     *
     * @generated from protobuf enum value: IOS_APP_LINK = 20;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["IOS_APP_LINK"] = 20] = "IOS_APP_LINK";
    /**
     * Data Type: INT64. iOS app store ID.
     *
     * @generated from protobuf enum value: IOS_APP_STORE_ID = 21;
     */
    TravelPlaceholderFieldEnum_TravelPlaceholderField[TravelPlaceholderFieldEnum_TravelPlaceholderField["IOS_APP_STORE_ID"] = 21] = "IOS_APP_STORE_ID";
})(TravelPlaceholderFieldEnum_TravelPlaceholderField = exports.TravelPlaceholderFieldEnum_TravelPlaceholderField || (exports.TravelPlaceholderFieldEnum_TravelPlaceholderField = {}));
// @generated message type with reflection information, may provide speed optimized methods
class TravelPlaceholderFieldEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.TravelPlaceholderFieldEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.TravelPlaceholderFieldEnum
 */
exports.TravelPlaceholderFieldEnum = new TravelPlaceholderFieldEnum$Type();
