// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_access_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListAccessStatusEnum = exports.UserListAccessStatusEnum_UserListAccessStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible user list access statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListAccessStatusEnum.UserListAccessStatus
 */
var UserListAccessStatusEnum_UserListAccessStatus;
(function (UserListAccessStatusEnum_UserListAccessStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListAccessStatusEnum_UserListAccessStatus[UserListAccessStatusEnum_UserListAccessStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListAccessStatusEnum_UserListAccessStatus[UserListAccessStatusEnum_UserListAccessStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The access is enabled.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    UserListAccessStatusEnum_UserListAccessStatus[UserListAccessStatusEnum_UserListAccessStatus["ENABLED"] = 2] = "ENABLED";
    /**
     * The access is disabled.
     *
     * @generated from protobuf enum value: DISABLED = 3;
     */
    UserListAccessStatusEnum_UserListAccessStatus[UserListAccessStatusEnum_UserListAccessStatus["DISABLED"] = 3] = "DISABLED";
})(UserListAccessStatusEnum_UserListAccessStatus = exports.UserListAccessStatusEnum_UserListAccessStatus || (exports.UserListAccessStatusEnum_UserListAccessStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListAccessStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListAccessStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListAccessStatusEnum
 */
exports.UserListAccessStatusEnum = new UserListAccessStatusEnum$Type();
