// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_rule_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListRuleTypeEnum = exports.UserListRuleTypeEnum_UserListRuleType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list rule types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListRuleTypeEnum.UserListRuleType
 */
var UserListRuleTypeEnum_UserListRuleType;
(function (UserListRuleTypeEnum_UserListRuleType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListRuleTypeEnum_UserListRuleType[UserListRuleTypeEnum_UserListRuleType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListRuleTypeEnum_UserListRuleType[UserListRuleTypeEnum_UserListRuleType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Conjunctive normal form.
     *
     * @generated from protobuf enum value: AND_OF_ORS = 2;
     */
    UserListRuleTypeEnum_UserListRuleType[UserListRuleTypeEnum_UserListRuleType["AND_OF_ORS"] = 2] = "AND_OF_ORS";
    /**
     * Disjunctive normal form.
     *
     * @generated from protobuf enum value: OR_OF_ANDS = 3;
     */
    UserListRuleTypeEnum_UserListRuleType[UserListRuleTypeEnum_UserListRuleType["OR_OF_ANDS"] = 3] = "OR_OF_ANDS";
})(UserListRuleTypeEnum_UserListRuleType = exports.UserListRuleTypeEnum_UserListRuleType || (exports.UserListRuleTypeEnum_UserListRuleType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListRuleTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListRuleTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListRuleTypeEnum
 */
exports.UserListRuleTypeEnum = new UserListRuleTypeEnum$Type();
