// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_flexible_rule_operator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListFlexibleRuleOperatorEnum = exports.UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list combined rule operators.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListFlexibleRuleOperatorEnum.UserListFlexibleRuleOperator
 */
var UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator;
(function (UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator[UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator[UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A AND B.
     *
     * @generated from protobuf enum value: AND = 2;
     */
    UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator[UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator["AND"] = 2] = "AND";
    /**
     * A OR B.
     *
     * @generated from protobuf enum value: OR = 3;
     */
    UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator[UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator["OR"] = 3] = "OR";
})(UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator = exports.UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator || (exports.UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListFlexibleRuleOperatorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListFlexibleRuleOperatorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListFlexibleRuleOperatorEnum
 */
exports.UserListFlexibleRuleOperatorEnum = new UserListFlexibleRuleOperatorEnum$Type();
