// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_size_range.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListSizeRangeEnum = exports.UserListSizeRangeEnum_UserListSizeRange = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible user list size ranges.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListSizeRangeEnum.UserListSizeRange
 */
var UserListSizeRangeEnum_UserListSizeRange;
(function (UserListSizeRangeEnum_UserListSizeRange) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * User list has less than 500 users.
     *
     * @generated from protobuf enum value: LESS_THAN_FIVE_HUNDRED = 2;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["LESS_THAN_FIVE_HUNDRED"] = 2] = "LESS_THAN_FIVE_HUNDRED";
    /**
     * User list has number of users in range of 500 to 1000.
     *
     * @generated from protobuf enum value: LESS_THAN_ONE_THOUSAND = 3;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["LESS_THAN_ONE_THOUSAND"] = 3] = "LESS_THAN_ONE_THOUSAND";
    /**
     * User list has number of users in range of 1000 to 10000.
     *
     * @generated from protobuf enum value: ONE_THOUSAND_TO_TEN_THOUSAND = 4;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["ONE_THOUSAND_TO_TEN_THOUSAND"] = 4] = "ONE_THOUSAND_TO_TEN_THOUSAND";
    /**
     * User list has number of users in range of 10000 to 50000.
     *
     * @generated from protobuf enum value: TEN_THOUSAND_TO_FIFTY_THOUSAND = 5;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["TEN_THOUSAND_TO_FIFTY_THOUSAND"] = 5] = "TEN_THOUSAND_TO_FIFTY_THOUSAND";
    /**
     * User list has number of users in range of 50000 to 100000.
     *
     * @generated from protobuf enum value: FIFTY_THOUSAND_TO_ONE_HUNDRED_THOUSAND = 6;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["FIFTY_THOUSAND_TO_ONE_HUNDRED_THOUSAND"] = 6] = "FIFTY_THOUSAND_TO_ONE_HUNDRED_THOUSAND";
    /**
     * User list has number of users in range of 100000 to 300000.
     *
     * @generated from protobuf enum value: ONE_HUNDRED_THOUSAND_TO_THREE_HUNDRED_THOUSAND = 7;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["ONE_HUNDRED_THOUSAND_TO_THREE_HUNDRED_THOUSAND"] = 7] = "ONE_HUNDRED_THOUSAND_TO_THREE_HUNDRED_THOUSAND";
    /**
     * User list has number of users in range of 300000 to 500000.
     *
     * @generated from protobuf enum value: THREE_HUNDRED_THOUSAND_TO_FIVE_HUNDRED_THOUSAND = 8;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["THREE_HUNDRED_THOUSAND_TO_FIVE_HUNDRED_THOUSAND"] = 8] = "THREE_HUNDRED_THOUSAND_TO_FIVE_HUNDRED_THOUSAND";
    /**
     * User list has number of users in range of 500000 to 1 million.
     *
     * @generated from protobuf enum value: FIVE_HUNDRED_THOUSAND_TO_ONE_MILLION = 9;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["FIVE_HUNDRED_THOUSAND_TO_ONE_MILLION"] = 9] = "FIVE_HUNDRED_THOUSAND_TO_ONE_MILLION";
    /**
     * User list has number of users in range of 1 to 2 millions.
     *
     * @generated from protobuf enum value: ONE_MILLION_TO_TWO_MILLION = 10;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["ONE_MILLION_TO_TWO_MILLION"] = 10] = "ONE_MILLION_TO_TWO_MILLION";
    /**
     * User list has number of users in range of 2 to 3 millions.
     *
     * @generated from protobuf enum value: TWO_MILLION_TO_THREE_MILLION = 11;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["TWO_MILLION_TO_THREE_MILLION"] = 11] = "TWO_MILLION_TO_THREE_MILLION";
    /**
     * User list has number of users in range of 3 to 5 millions.
     *
     * @generated from protobuf enum value: THREE_MILLION_TO_FIVE_MILLION = 12;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["THREE_MILLION_TO_FIVE_MILLION"] = 12] = "THREE_MILLION_TO_FIVE_MILLION";
    /**
     * User list has number of users in range of 5 to 10 millions.
     *
     * @generated from protobuf enum value: FIVE_MILLION_TO_TEN_MILLION = 13;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["FIVE_MILLION_TO_TEN_MILLION"] = 13] = "FIVE_MILLION_TO_TEN_MILLION";
    /**
     * User list has number of users in range of 10 to 20 millions.
     *
     * @generated from protobuf enum value: TEN_MILLION_TO_TWENTY_MILLION = 14;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["TEN_MILLION_TO_TWENTY_MILLION"] = 14] = "TEN_MILLION_TO_TWENTY_MILLION";
    /**
     * User list has number of users in range of 20 to 30 millions.
     *
     * @generated from protobuf enum value: TWENTY_MILLION_TO_THIRTY_MILLION = 15;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["TWENTY_MILLION_TO_THIRTY_MILLION"] = 15] = "TWENTY_MILLION_TO_THIRTY_MILLION";
    /**
     * User list has number of users in range of 30 to 50 millions.
     *
     * @generated from protobuf enum value: THIRTY_MILLION_TO_FIFTY_MILLION = 16;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["THIRTY_MILLION_TO_FIFTY_MILLION"] = 16] = "THIRTY_MILLION_TO_FIFTY_MILLION";
    /**
     * User list has over 50 million users.
     *
     * @generated from protobuf enum value: OVER_FIFTY_MILLION = 17;
     */
    UserListSizeRangeEnum_UserListSizeRange[UserListSizeRangeEnum_UserListSizeRange["OVER_FIFTY_MILLION"] = 17] = "OVER_FIFTY_MILLION";
})(UserListSizeRangeEnum_UserListSizeRange = exports.UserListSizeRangeEnum_UserListSizeRange || (exports.UserListSizeRangeEnum_UserListSizeRange = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListSizeRangeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListSizeRangeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListSizeRangeEnum
 */
exports.UserListSizeRangeEnum = new UserListSizeRangeEnum$Type();
