// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_logical_rule_operator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListLogicalRuleOperatorEnum = exports.UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list logical rule operators.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListLogicalRuleOperatorEnum.UserListLogicalRuleOperator
 */
var UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator;
(function (UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator[UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator[UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * And - all of the operands.
     *
     * @generated from protobuf enum value: ALL = 2;
     */
    UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator[UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator["ALL"] = 2] = "ALL";
    /**
     * Or - at least one of the operands.
     *
     * @generated from protobuf enum value: ANY = 3;
     */
    UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator[UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator["ANY"] = 3] = "ANY";
    /**
     * Not - none of the operands.
     *
     * @generated from protobuf enum value: NONE = 4;
     */
    UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator[UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator["NONE"] = 4] = "NONE";
})(UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator = exports.UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator || (exports.UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListLogicalRuleOperatorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListLogicalRuleOperatorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListLogicalRuleOperatorEnum
 */
exports.UserListLogicalRuleOperatorEnum = new UserListLogicalRuleOperatorEnum$Type();
