// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_number_rule_item_operator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListNumberRuleItemOperatorEnum = exports.UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list number rule item operators.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListNumberRuleItemOperatorEnum.UserListNumberRuleItemOperator
 */
var UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator;
(function (UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator[UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator[UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Greater than.
     *
     * @generated from protobuf enum value: GREATER_THAN = 2;
     */
    UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator[UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator["GREATER_THAN"] = 2] = "GREATER_THAN";
    /**
     * Greater than or equal.
     *
     * @generated from protobuf enum value: GREATER_THAN_OR_EQUAL = 3;
     */
    UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator[UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator["GREATER_THAN_OR_EQUAL"] = 3] = "GREATER_THAN_OR_EQUAL";
    /**
     * Equals.
     *
     * @generated from protobuf enum value: EQUALS = 4;
     */
    UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator[UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator["EQUALS"] = 4] = "EQUALS";
    /**
     * Not equals.
     *
     * @generated from protobuf enum value: NOT_EQUALS = 5;
     */
    UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator[UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator["NOT_EQUALS"] = 5] = "NOT_EQUALS";
    /**
     * Less than.
     *
     * @generated from protobuf enum value: LESS_THAN = 6;
     */
    UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator[UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator["LESS_THAN"] = 6] = "LESS_THAN";
    /**
     * Less than or equal.
     *
     * @generated from protobuf enum value: LESS_THAN_OR_EQUAL = 7;
     */
    UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator[UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator["LESS_THAN_OR_EQUAL"] = 7] = "LESS_THAN_OR_EQUAL";
})(UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator = exports.UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator || (exports.UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListNumberRuleItemOperatorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListNumberRuleItemOperatorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListNumberRuleItemOperatorEnum
 */
exports.UserListNumberRuleItemOperatorEnum = new UserListNumberRuleItemOperatorEnum$Type();
