// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListTypeEnum = exports.UserListTypeEnum_UserListType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible user list types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListTypeEnum.UserListType
 */
var UserListTypeEnum_UserListType;
(function (UserListTypeEnum_UserListType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListTypeEnum_UserListType[UserListTypeEnum_UserListType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListTypeEnum_UserListType[UserListTypeEnum_UserListType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * UserList represented as a collection of conversion types.
     *
     * @generated from protobuf enum value: REMARKETING = 2;
     */
    UserListTypeEnum_UserListType[UserListTypeEnum_UserListType["REMARKETING"] = 2] = "REMARKETING";
    /**
     * UserList represented as a combination of other user lists/interests.
     *
     * @generated from protobuf enum value: LOGICAL = 3;
     */
    UserListTypeEnum_UserListType[UserListTypeEnum_UserListType["LOGICAL"] = 3] = "LOGICAL";
    /**
     * UserList created in the Google Ad Manager platform.
     *
     * @generated from protobuf enum value: EXTERNAL_REMARKETING = 4;
     */
    UserListTypeEnum_UserListType[UserListTypeEnum_UserListType["EXTERNAL_REMARKETING"] = 4] = "EXTERNAL_REMARKETING";
    /**
     * UserList associated with a rule.
     *
     * @generated from protobuf enum value: RULE_BASED = 5;
     */
    UserListTypeEnum_UserListType[UserListTypeEnum_UserListType["RULE_BASED"] = 5] = "RULE_BASED";
    /**
     * UserList with users similar to users of another UserList.
     *
     * @generated from protobuf enum value: SIMILAR = 6;
     */
    UserListTypeEnum_UserListType[UserListTypeEnum_UserListType["SIMILAR"] = 6] = "SIMILAR";
    /**
     * UserList of first-party CRM data provided by advertiser in the form of
     * emails or other formats.
     *
     * @generated from protobuf enum value: CRM_BASED = 7;
     */
    UserListTypeEnum_UserListType[UserListTypeEnum_UserListType["CRM_BASED"] = 7] = "CRM_BASED";
})(UserListTypeEnum_UserListType = exports.UserListTypeEnum_UserListType || (exports.UserListTypeEnum_UserListType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListTypeEnum
 */
exports.UserListTypeEnum = new UserListTypeEnum$Type();
