// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_crm_data_source_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListCrmDataSourceTypeEnum = exports.UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list crm data source type.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListCrmDataSourceTypeEnum.UserListCrmDataSourceType
 */
var UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType;
(function (UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType[UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType[UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The uploaded data is first-party data.
     *
     * @generated from protobuf enum value: FIRST_PARTY = 2;
     */
    UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType[UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType["FIRST_PARTY"] = 2] = "FIRST_PARTY";
    /**
     * The uploaded data is from a third-party credit bureau.
     *
     * @generated from protobuf enum value: THIRD_PARTY_CREDIT_BUREAU = 3;
     */
    UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType[UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType["THIRD_PARTY_CREDIT_BUREAU"] = 3] = "THIRD_PARTY_CREDIT_BUREAU";
    /**
     * The uploaded data is from a third-party voter file.
     *
     * @generated from protobuf enum value: THIRD_PARTY_VOTER_FILE = 4;
     */
    UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType[UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType["THIRD_PARTY_VOTER_FILE"] = 4] = "THIRD_PARTY_VOTER_FILE";
})(UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType = exports.UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType || (exports.UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListCrmDataSourceTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListCrmDataSourceTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListCrmDataSourceTypeEnum
 */
exports.UserListCrmDataSourceTypeEnum = new UserListCrmDataSourceTypeEnum$Type();
