// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_date_rule_item_operator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListDateRuleItemOperatorEnum = exports.UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list date rule item operators.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListDateRuleItemOperatorEnum.UserListDateRuleItemOperator
 */
var UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator;
(function (UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator[UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator[UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Equals.
     *
     * @generated from protobuf enum value: EQUALS = 2;
     */
    UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator[UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator["EQUALS"] = 2] = "EQUALS";
    /**
     * Not Equals.
     *
     * @generated from protobuf enum value: NOT_EQUALS = 3;
     */
    UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator[UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator["NOT_EQUALS"] = 3] = "NOT_EQUALS";
    /**
     * Before.
     *
     * @generated from protobuf enum value: BEFORE = 4;
     */
    UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator[UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator["BEFORE"] = 4] = "BEFORE";
    /**
     * After.
     *
     * @generated from protobuf enum value: AFTER = 5;
     */
    UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator[UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator["AFTER"] = 5] = "AFTER";
})(UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator = exports.UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator || (exports.UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListDateRuleItemOperatorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListDateRuleItemOperatorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListDateRuleItemOperatorEnum
 */
exports.UserListDateRuleItemOperatorEnum = new UserListDateRuleItemOperatorEnum$Type();
