// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_membership_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListMembershipStatusEnum = exports.UserListMembershipStatusEnum_UserListMembershipStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing possible user list membership statuses.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListMembershipStatusEnum.UserListMembershipStatus
 */
var UserListMembershipStatusEnum_UserListMembershipStatus;
(function (UserListMembershipStatusEnum_UserListMembershipStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListMembershipStatusEnum_UserListMembershipStatus[UserListMembershipStatusEnum_UserListMembershipStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListMembershipStatusEnum_UserListMembershipStatus[UserListMembershipStatusEnum_UserListMembershipStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Open status - List is accruing members and can be targeted to.
     *
     * @generated from protobuf enum value: OPEN = 2;
     */
    UserListMembershipStatusEnum_UserListMembershipStatus[UserListMembershipStatusEnum_UserListMembershipStatus["OPEN"] = 2] = "OPEN";
    /**
     * Closed status - No new members being added. Cannot be used for targeting.
     *
     * @generated from protobuf enum value: CLOSED = 3;
     */
    UserListMembershipStatusEnum_UserListMembershipStatus[UserListMembershipStatusEnum_UserListMembershipStatus["CLOSED"] = 3] = "CLOSED";
})(UserListMembershipStatusEnum_UserListMembershipStatus = exports.UserListMembershipStatusEnum_UserListMembershipStatus || (exports.UserListMembershipStatusEnum_UserListMembershipStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListMembershipStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListMembershipStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListMembershipStatusEnum
 */
exports.UserListMembershipStatusEnum = new UserListMembershipStatusEnum$Type();
