// google-ads-api-client/dist/protos/bindings/ads/enums/user_identifier_source.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserIdentifierSourceEnum = exports.UserIdentifierSourceEnum_UserIdentifierSource = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The type of user identifier source for offline Store Sales, click
 * conversion, and conversion adjustment uploads.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserIdentifierSourceEnum.UserIdentifierSource
 */
var UserIdentifierSourceEnum_UserIdentifierSource;
(function (UserIdentifierSourceEnum_UserIdentifierSource) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserIdentifierSourceEnum_UserIdentifierSource[UserIdentifierSourceEnum_UserIdentifierSource["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserIdentifierSourceEnum_UserIdentifierSource[UserIdentifierSourceEnum_UserIdentifierSource["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Indicates that the user identifier was provided by the first party
     * (advertiser).
     *
     * @generated from protobuf enum value: FIRST_PARTY = 2;
     */
    UserIdentifierSourceEnum_UserIdentifierSource[UserIdentifierSourceEnum_UserIdentifierSource["FIRST_PARTY"] = 2] = "FIRST_PARTY";
    /**
     * Indicates that the user identifier was provided by the third party
     * (partner).
     *
     * @generated from protobuf enum value: THIRD_PARTY = 3;
     */
    UserIdentifierSourceEnum_UserIdentifierSource[UserIdentifierSourceEnum_UserIdentifierSource["THIRD_PARTY"] = 3] = "THIRD_PARTY";
})(UserIdentifierSourceEnum_UserIdentifierSource = exports.UserIdentifierSourceEnum_UserIdentifierSource || (exports.UserIdentifierSourceEnum_UserIdentifierSource = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserIdentifierSourceEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserIdentifierSourceEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserIdentifierSourceEnum
 */
exports.UserIdentifierSourceEnum = new UserIdentifierSourceEnum$Type();
