// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_string_rule_item_operator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListStringRuleItemOperatorEnum = exports.UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list string rule item operators.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListStringRuleItemOperatorEnum.UserListStringRuleItemOperator
 */
var UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator;
(function (UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Contains.
     *
     * @generated from protobuf enum value: CONTAINS = 2;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["CONTAINS"] = 2] = "CONTAINS";
    /**
     * Equals.
     *
     * @generated from protobuf enum value: EQUALS = 3;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["EQUALS"] = 3] = "EQUALS";
    /**
     * Starts with.
     *
     * @generated from protobuf enum value: STARTS_WITH = 4;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["STARTS_WITH"] = 4] = "STARTS_WITH";
    /**
     * Ends with.
     *
     * @generated from protobuf enum value: ENDS_WITH = 5;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["ENDS_WITH"] = 5] = "ENDS_WITH";
    /**
     * Not equals.
     *
     * @generated from protobuf enum value: NOT_EQUALS = 6;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["NOT_EQUALS"] = 6] = "NOT_EQUALS";
    /**
     * Not contains.
     *
     * @generated from protobuf enum value: NOT_CONTAINS = 7;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["NOT_CONTAINS"] = 7] = "NOT_CONTAINS";
    /**
     * Not starts with.
     *
     * @generated from protobuf enum value: NOT_STARTS_WITH = 8;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["NOT_STARTS_WITH"] = 8] = "NOT_STARTS_WITH";
    /**
     * Not ends with.
     *
     * @generated from protobuf enum value: NOT_ENDS_WITH = 9;
     */
    UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator[UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator["NOT_ENDS_WITH"] = 9] = "NOT_ENDS_WITH";
})(UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator = exports.UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator || (exports.UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListStringRuleItemOperatorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListStringRuleItemOperatorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListStringRuleItemOperatorEnum
 */
exports.UserListStringRuleItemOperatorEnum = new UserListStringRuleItemOperatorEnum$Type();
