// google-ads-api-client/dist/protos/bindings/ads/enums/user_interest_taxonomy_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserInterestTaxonomyTypeEnum = exports.UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum containing the possible UserInterestTaxonomyTypes.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserInterestTaxonomyTypeEnum.UserInterestTaxonomyType
 */
var UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType;
(function (UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType[UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType[UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The affinity for this user interest.
     *
     * @generated from protobuf enum value: AFFINITY = 2;
     */
    UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType[UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType["AFFINITY"] = 2] = "AFFINITY";
    /**
     * The market for this user interest.
     *
     * @generated from protobuf enum value: IN_MARKET = 3;
     */
    UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType[UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType["IN_MARKET"] = 3] = "IN_MARKET";
    /**
     * Users known to have installed applications in the specified categories.
     *
     * @generated from protobuf enum value: MOBILE_APP_INSTALL_USER = 4;
     */
    UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType[UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType["MOBILE_APP_INSTALL_USER"] = 4] = "MOBILE_APP_INSTALL_USER";
    /**
     * The geographical location of the interest-based vertical.
     *
     * @generated from protobuf enum value: VERTICAL_GEO = 5;
     */
    UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType[UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType["VERTICAL_GEO"] = 5] = "VERTICAL_GEO";
    /**
     * User interest criteria for new smart phone users.
     *
     * @generated from protobuf enum value: NEW_SMART_PHONE_USER = 6;
     */
    UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType[UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType["NEW_SMART_PHONE_USER"] = 6] = "NEW_SMART_PHONE_USER";
})(UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType = exports.UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType || (exports.UserInterestTaxonomyTypeEnum_UserInterestTaxonomyType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserInterestTaxonomyTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserInterestTaxonomyTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserInterestTaxonomyTypeEnum
 */
exports.UserInterestTaxonomyTypeEnum = new UserInterestTaxonomyTypeEnum$Type();
