// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_closing_reason.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListClosingReasonEnum = exports.UserListClosingReasonEnum_UserListClosingReason = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list closing reasons.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListClosingReasonEnum.UserListClosingReason
 */
var UserListClosingReasonEnum_UserListClosingReason;
(function (UserListClosingReasonEnum_UserListClosingReason) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListClosingReasonEnum_UserListClosingReason[UserListClosingReasonEnum_UserListClosingReason["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListClosingReasonEnum_UserListClosingReason[UserListClosingReasonEnum_UserListClosingReason["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The userlist was closed because of not being used for over one year.
     *
     * @generated from protobuf enum value: UNUSED = 2;
     */
    UserListClosingReasonEnum_UserListClosingReason[UserListClosingReasonEnum_UserListClosingReason["UNUSED"] = 2] = "UNUSED";
})(UserListClosingReasonEnum_UserListClosingReason = exports.UserListClosingReasonEnum_UserListClosingReason || (exports.UserListClosingReasonEnum_UserListClosingReason = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListClosingReasonEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListClosingReasonEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListClosingReasonEnum
 */
exports.UserListClosingReasonEnum = new UserListClosingReasonEnum$Type();
