// google-ads-api-client/dist/protos/bindings/ads/enums/user_list_prepopulation_status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListPrepopulationStatusEnum = exports.UserListPrepopulationStatusEnum_UserListPrepopulationStatus = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list prepopulation status.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.UserListPrepopulationStatusEnum.UserListPrepopulationStatus
 */
var UserListPrepopulationStatusEnum_UserListPrepopulationStatus;
(function (UserListPrepopulationStatusEnum_UserListPrepopulationStatus) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListPrepopulationStatusEnum_UserListPrepopulationStatus[UserListPrepopulationStatusEnum_UserListPrepopulationStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListPrepopulationStatusEnum_UserListPrepopulationStatus[UserListPrepopulationStatusEnum_UserListPrepopulationStatus["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Prepopoulation is being requested.
     *
     * @generated from protobuf enum value: REQUESTED = 2;
     */
    UserListPrepopulationStatusEnum_UserListPrepopulationStatus[UserListPrepopulationStatusEnum_UserListPrepopulationStatus["REQUESTED"] = 2] = "REQUESTED";
    /**
     * Prepopulation is finished.
     *
     * @generated from protobuf enum value: FINISHED = 3;
     */
    UserListPrepopulationStatusEnum_UserListPrepopulationStatus[UserListPrepopulationStatusEnum_UserListPrepopulationStatus["FINISHED"] = 3] = "FINISHED";
    /**
     * Prepopulation failed.
     *
     * @generated from protobuf enum value: FAILED = 4;
     */
    UserListPrepopulationStatusEnum_UserListPrepopulationStatus[UserListPrepopulationStatusEnum_UserListPrepopulationStatus["FAILED"] = 4] = "FAILED";
})(UserListPrepopulationStatusEnum_UserListPrepopulationStatus = exports.UserListPrepopulationStatusEnum_UserListPrepopulationStatus || (exports.UserListPrepopulationStatusEnum_UserListPrepopulationStatus = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListPrepopulationStatusEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.UserListPrepopulationStatusEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.UserListPrepopulationStatusEnum
 */
exports.UserListPrepopulationStatusEnum = new UserListPrepopulationStatusEnum$Type();
