// google-ads-api-client/dist/protos/bindings/ads/enums/value_rule_geo_location_match_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValueRuleGeoLocationMatchTypeEnum = exports.ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible geographic location matching types.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ValueRuleGeoLocationMatchTypeEnum.ValueRuleGeoLocationMatchType
 */
var ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType;
(function (ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType[ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType[ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Either Area of Interest or Location of Presence can be used to match.
     *
     * @generated from protobuf enum value: ANY = 2;
     */
    ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType[ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType["ANY"] = 2] = "ANY";
    /**
     * Only Location of Presence can be used to match.
     *
     * @generated from protobuf enum value: LOCATION_OF_PRESENCE = 3;
     */
    ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType[ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType["LOCATION_OF_PRESENCE"] = 3] = "LOCATION_OF_PRESENCE";
})(ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType = exports.ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType || (exports.ValueRuleGeoLocationMatchTypeEnum_ValueRuleGeoLocationMatchType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ValueRuleGeoLocationMatchTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ValueRuleGeoLocationMatchTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ValueRuleGeoLocationMatchTypeEnum
 */
exports.ValueRuleGeoLocationMatchTypeEnum = new ValueRuleGeoLocationMatchTypeEnum$Type();
