// google-ads-api-client/dist/protos/bindings/ads/enums/value_rule_set_dimension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValueRuleSetDimensionEnum = exports.ValueRuleSetDimensionEnum_ValueRuleSetDimension = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible dimensions of a conversion value rule set.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ValueRuleSetDimensionEnum.ValueRuleSetDimension
 */
var ValueRuleSetDimensionEnum_ValueRuleSetDimension;
(function (ValueRuleSetDimensionEnum_ValueRuleSetDimension) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ValueRuleSetDimensionEnum_ValueRuleSetDimension[ValueRuleSetDimensionEnum_ValueRuleSetDimension["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ValueRuleSetDimensionEnum_ValueRuleSetDimension[ValueRuleSetDimensionEnum_ValueRuleSetDimension["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Dimension for geo location.
     *
     * @generated from protobuf enum value: GEO_LOCATION = 2;
     */
    ValueRuleSetDimensionEnum_ValueRuleSetDimension[ValueRuleSetDimensionEnum_ValueRuleSetDimension["GEO_LOCATION"] = 2] = "GEO_LOCATION";
    /**
     * Dimension for device type.
     *
     * @generated from protobuf enum value: DEVICE = 3;
     */
    ValueRuleSetDimensionEnum_ValueRuleSetDimension[ValueRuleSetDimensionEnum_ValueRuleSetDimension["DEVICE"] = 3] = "DEVICE";
    /**
     * Dimension for audience.
     *
     * @generated from protobuf enum value: AUDIENCE = 4;
     */
    ValueRuleSetDimensionEnum_ValueRuleSetDimension[ValueRuleSetDimensionEnum_ValueRuleSetDimension["AUDIENCE"] = 4] = "AUDIENCE";
    /**
     * This dimension implies the rule will always apply.
     *
     * @generated from protobuf enum value: NO_CONDITION = 5;
     */
    ValueRuleSetDimensionEnum_ValueRuleSetDimension[ValueRuleSetDimensionEnum_ValueRuleSetDimension["NO_CONDITION"] = 5] = "NO_CONDITION";
})(ValueRuleSetDimensionEnum_ValueRuleSetDimension = exports.ValueRuleSetDimensionEnum_ValueRuleSetDimension || (exports.ValueRuleSetDimensionEnum_ValueRuleSetDimension = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ValueRuleSetDimensionEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ValueRuleSetDimensionEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ValueRuleSetDimensionEnum
 */
exports.ValueRuleSetDimensionEnum = new ValueRuleSetDimensionEnum$Type();
