// google-ads-api-client/dist/protos/bindings/ads/enums/value_rule_device_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValueRuleDeviceTypeEnum = exports.ValueRuleDeviceTypeEnum_ValueRuleDeviceType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible device types used in conversion value rule.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ValueRuleDeviceTypeEnum.ValueRuleDeviceType
 */
var ValueRuleDeviceTypeEnum_ValueRuleDeviceType;
(function (ValueRuleDeviceTypeEnum_ValueRuleDeviceType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ValueRuleDeviceTypeEnum_ValueRuleDeviceType[ValueRuleDeviceTypeEnum_ValueRuleDeviceType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ValueRuleDeviceTypeEnum_ValueRuleDeviceType[ValueRuleDeviceTypeEnum_ValueRuleDeviceType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Mobile device.
     *
     * @generated from protobuf enum value: MOBILE = 2;
     */
    ValueRuleDeviceTypeEnum_ValueRuleDeviceType[ValueRuleDeviceTypeEnum_ValueRuleDeviceType["MOBILE"] = 2] = "MOBILE";
    /**
     * Desktop device.
     *
     * @generated from protobuf enum value: DESKTOP = 3;
     */
    ValueRuleDeviceTypeEnum_ValueRuleDeviceType[ValueRuleDeviceTypeEnum_ValueRuleDeviceType["DESKTOP"] = 3] = "DESKTOP";
    /**
     * Tablet device.
     *
     * @generated from protobuf enum value: TABLET = 4;
     */
    ValueRuleDeviceTypeEnum_ValueRuleDeviceType[ValueRuleDeviceTypeEnum_ValueRuleDeviceType["TABLET"] = 4] = "TABLET";
})(ValueRuleDeviceTypeEnum_ValueRuleDeviceType = exports.ValueRuleDeviceTypeEnum_ValueRuleDeviceType || (exports.ValueRuleDeviceTypeEnum_ValueRuleDeviceType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ValueRuleDeviceTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ValueRuleDeviceTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ValueRuleDeviceTypeEnum
 */
exports.ValueRuleDeviceTypeEnum = new ValueRuleDeviceTypeEnum$Type();
