// google-ads-api-client/dist/protos/bindings/ads/enums/value_rule_set_attachment_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValueRuleSetAttachmentTypeEnum = exports.ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Possible level where a value rule set is attached.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.ValueRuleSetAttachmentTypeEnum.ValueRuleSetAttachmentType
 */
var ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType;
(function (ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType[ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType[ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Attached to the customer.
     *
     * @generated from protobuf enum value: CUSTOMER = 2;
     */
    ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType[ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType["CUSTOMER"] = 2] = "CUSTOMER";
    /**
     * Attached to a campaign.
     *
     * @generated from protobuf enum value: CAMPAIGN = 3;
     */
    ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType[ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType["CAMPAIGN"] = 3] = "CAMPAIGN";
})(ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType = exports.ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType || (exports.ValueRuleSetAttachmentTypeEnum_ValueRuleSetAttachmentType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ValueRuleSetAttachmentTypeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.ValueRuleSetAttachmentTypeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.ValueRuleSetAttachmentTypeEnum
 */
exports.ValueRuleSetAttachmentTypeEnum = new ValueRuleSetAttachmentTypeEnum$Type();
