// google-ads-api-client/dist/protos/bindings/ads/enums/vanity_pharma_text.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VanityPharmaTextEnum = exports.VanityPharmaTextEnum_VanityPharmaText = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible text.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.VanityPharmaTextEnum.VanityPharmaText
 */
var VanityPharmaTextEnum_VanityPharmaText;
(function (VanityPharmaTextEnum_VanityPharmaText) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Prescription treatment website with website content in English.
     *
     * @generated from protobuf enum value: PRESCRIPTION_TREATMENT_WEBSITE_EN = 2;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PRESCRIPTION_TREATMENT_WEBSITE_EN"] = 2] = "PRESCRIPTION_TREATMENT_WEBSITE_EN";
    /**
     * Prescription treatment website with website content in Spanish
     * (Sitio de tratamientos con receta).
     *
     * @generated from protobuf enum value: PRESCRIPTION_TREATMENT_WEBSITE_ES = 3;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PRESCRIPTION_TREATMENT_WEBSITE_ES"] = 3] = "PRESCRIPTION_TREATMENT_WEBSITE_ES";
    /**
     * Prescription device website with website content in English.
     *
     * @generated from protobuf enum value: PRESCRIPTION_DEVICE_WEBSITE_EN = 4;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PRESCRIPTION_DEVICE_WEBSITE_EN"] = 4] = "PRESCRIPTION_DEVICE_WEBSITE_EN";
    /**
     * Prescription device website with website content in Spanish (Sitio de
     * dispositivos con receta).
     *
     * @generated from protobuf enum value: PRESCRIPTION_DEVICE_WEBSITE_ES = 5;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PRESCRIPTION_DEVICE_WEBSITE_ES"] = 5] = "PRESCRIPTION_DEVICE_WEBSITE_ES";
    /**
     * Medical device website with website content in English.
     *
     * @generated from protobuf enum value: MEDICAL_DEVICE_WEBSITE_EN = 6;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["MEDICAL_DEVICE_WEBSITE_EN"] = 6] = "MEDICAL_DEVICE_WEBSITE_EN";
    /**
     * Medical device website with website content in Spanish (Sitio de
     * dispositivos médicos).
     *
     * @generated from protobuf enum value: MEDICAL_DEVICE_WEBSITE_ES = 7;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["MEDICAL_DEVICE_WEBSITE_ES"] = 7] = "MEDICAL_DEVICE_WEBSITE_ES";
    /**
     * Preventative treatment website with website content in English.
     *
     * @generated from protobuf enum value: PREVENTATIVE_TREATMENT_WEBSITE_EN = 8;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PREVENTATIVE_TREATMENT_WEBSITE_EN"] = 8] = "PREVENTATIVE_TREATMENT_WEBSITE_EN";
    /**
     * Preventative treatment website with website content in Spanish (Sitio de
     * tratamientos preventivos).
     *
     * @generated from protobuf enum value: PREVENTATIVE_TREATMENT_WEBSITE_ES = 9;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PREVENTATIVE_TREATMENT_WEBSITE_ES"] = 9] = "PREVENTATIVE_TREATMENT_WEBSITE_ES";
    /**
     * Prescription contraception website with website content in English.
     *
     * @generated from protobuf enum value: PRESCRIPTION_CONTRACEPTION_WEBSITE_EN = 10;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PRESCRIPTION_CONTRACEPTION_WEBSITE_EN"] = 10] = "PRESCRIPTION_CONTRACEPTION_WEBSITE_EN";
    /**
     * Prescription contraception website with website content in Spanish (Sitio
     * de anticonceptivos con receta).
     *
     * @generated from protobuf enum value: PRESCRIPTION_CONTRACEPTION_WEBSITE_ES = 11;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PRESCRIPTION_CONTRACEPTION_WEBSITE_ES"] = 11] = "PRESCRIPTION_CONTRACEPTION_WEBSITE_ES";
    /**
     * Prescription vaccine website with website content in English.
     *
     * @generated from protobuf enum value: PRESCRIPTION_VACCINE_WEBSITE_EN = 12;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PRESCRIPTION_VACCINE_WEBSITE_EN"] = 12] = "PRESCRIPTION_VACCINE_WEBSITE_EN";
    /**
     * Prescription vaccine website with website content in Spanish (Sitio de
     * vacunas con receta).
     *
     * @generated from protobuf enum value: PRESCRIPTION_VACCINE_WEBSITE_ES = 13;
     */
    VanityPharmaTextEnum_VanityPharmaText[VanityPharmaTextEnum_VanityPharmaText["PRESCRIPTION_VACCINE_WEBSITE_ES"] = 13] = "PRESCRIPTION_VACCINE_WEBSITE_ES";
})(VanityPharmaTextEnum_VanityPharmaText = exports.VanityPharmaTextEnum_VanityPharmaText || (exports.VanityPharmaTextEnum_VanityPharmaText = {}));
// @generated message type with reflection information, may provide speed optimized methods
class VanityPharmaTextEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.VanityPharmaTextEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.VanityPharmaTextEnum
 */
exports.VanityPharmaTextEnum = new VanityPharmaTextEnum$Type();
