// google-ads-api-client/dist/protos/bindings/ads/enums/vanity_pharma_display_url_mode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VanityPharmaDisplayUrlModeEnum = exports.VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible display modes for vanity pharma URLs.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.VanityPharmaDisplayUrlModeEnum.VanityPharmaDisplayUrlMode
 */
var VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode;
(function (VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode[VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode[VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Replace vanity pharma URL with manufacturer website url.
     *
     * @generated from protobuf enum value: MANUFACTURER_WEBSITE_URL = 2;
     */
    VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode[VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode["MANUFACTURER_WEBSITE_URL"] = 2] = "MANUFACTURER_WEBSITE_URL";
    /**
     * Replace vanity pharma URL with description of the website.
     *
     * @generated from protobuf enum value: WEBSITE_DESCRIPTION = 3;
     */
    VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode[VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode["WEBSITE_DESCRIPTION"] = 3] = "WEBSITE_DESCRIPTION";
})(VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode = exports.VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode || (exports.VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode = {}));
// @generated message type with reflection information, may provide speed optimized methods
class VanityPharmaDisplayUrlModeEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.VanityPharmaDisplayUrlModeEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.VanityPharmaDisplayUrlModeEnum
 */
exports.VanityPharmaDisplayUrlModeEnum = new VanityPharmaDisplayUrlModeEnum$Type();
