// google-ads-api-client/dist/protos/bindings/ads/enums/video_thumbnail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoThumbnailEnum = exports.VideoThumbnailEnum_VideoThumbnail = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum listing the possible types of a video thumbnail.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.VideoThumbnailEnum.VideoThumbnail
 */
var VideoThumbnailEnum_VideoThumbnail;
(function (VideoThumbnailEnum_VideoThumbnail) {
    /**
     * The type has not been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    VideoThumbnailEnum_VideoThumbnail[VideoThumbnailEnum_VideoThumbnail["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received value is not known in this version.
     * This is a response-only value.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    VideoThumbnailEnum_VideoThumbnail[VideoThumbnailEnum_VideoThumbnail["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The default thumbnail. Can be auto-generated or user-uploaded.
     *
     * @generated from protobuf enum value: DEFAULT_THUMBNAIL = 2;
     */
    VideoThumbnailEnum_VideoThumbnail[VideoThumbnailEnum_VideoThumbnail["DEFAULT_THUMBNAIL"] = 2] = "DEFAULT_THUMBNAIL";
    /**
     * Thumbnail 1, generated from the video.
     *
     * @generated from protobuf enum value: THUMBNAIL_1 = 3;
     */
    VideoThumbnailEnum_VideoThumbnail[VideoThumbnailEnum_VideoThumbnail["THUMBNAIL_1"] = 3] = "THUMBNAIL_1";
    /**
     * Thumbnail 2, generated from the video.
     *
     * @generated from protobuf enum value: THUMBNAIL_2 = 4;
     */
    VideoThumbnailEnum_VideoThumbnail[VideoThumbnailEnum_VideoThumbnail["THUMBNAIL_2"] = 4] = "THUMBNAIL_2";
    /**
     * Thumbnail 3, generated from the video.
     *
     * @generated from protobuf enum value: THUMBNAIL_3 = 5;
     */
    VideoThumbnailEnum_VideoThumbnail[VideoThumbnailEnum_VideoThumbnail["THUMBNAIL_3"] = 5] = "THUMBNAIL_3";
})(VideoThumbnailEnum_VideoThumbnail = exports.VideoThumbnailEnum_VideoThumbnail || (exports.VideoThumbnailEnum_VideoThumbnail = {}));
// @generated message type with reflection information, may provide speed optimized methods
class VideoThumbnailEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.VideoThumbnailEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.VideoThumbnailEnum
 */
exports.VideoThumbnailEnum = new VideoThumbnailEnum$Type();
