// google-ads-api-client/dist/protos/bindings/ads/enums/webpage_condition_operator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebpageConditionOperatorEnum = exports.WebpageConditionOperatorEnum_WebpageConditionOperator = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The webpage condition operator in webpage criterion.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.WebpageConditionOperatorEnum.WebpageConditionOperator
 */
var WebpageConditionOperatorEnum_WebpageConditionOperator;
(function (WebpageConditionOperatorEnum_WebpageConditionOperator) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    WebpageConditionOperatorEnum_WebpageConditionOperator[WebpageConditionOperatorEnum_WebpageConditionOperator["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    WebpageConditionOperatorEnum_WebpageConditionOperator[WebpageConditionOperatorEnum_WebpageConditionOperator["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The argument web condition is equal to the compared web condition.
     *
     * @generated from protobuf enum value: EQUALS = 2;
     */
    WebpageConditionOperatorEnum_WebpageConditionOperator[WebpageConditionOperatorEnum_WebpageConditionOperator["EQUALS"] = 2] = "EQUALS";
    /**
     * The argument web condition is part of the compared web condition.
     *
     * @generated from protobuf enum value: CONTAINS = 3;
     */
    WebpageConditionOperatorEnum_WebpageConditionOperator[WebpageConditionOperatorEnum_WebpageConditionOperator["CONTAINS"] = 3] = "CONTAINS";
})(WebpageConditionOperatorEnum_WebpageConditionOperator = exports.WebpageConditionOperatorEnum_WebpageConditionOperator || (exports.WebpageConditionOperatorEnum_WebpageConditionOperator = {}));
// @generated message type with reflection information, may provide speed optimized methods
class WebpageConditionOperatorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.WebpageConditionOperatorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.WebpageConditionOperatorEnum
 */
exports.WebpageConditionOperatorEnum = new WebpageConditionOperatorEnum$Type();
