// google-ads-api-client/dist/protos/bindings/ads/enums/webpage_condition_operand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebpageConditionOperandEnum = exports.WebpageConditionOperandEnum_WebpageConditionOperand = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The webpage condition operand in webpage criterion.
 *
 * @generated from protobuf enum google.ads.googleads.v11.enums.WebpageConditionOperandEnum.WebpageConditionOperand
 */
var WebpageConditionOperandEnum_WebpageConditionOperand;
(function (WebpageConditionOperandEnum_WebpageConditionOperand) {
    /**
     * Not specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    WebpageConditionOperandEnum_WebpageConditionOperand[WebpageConditionOperandEnum_WebpageConditionOperand["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    WebpageConditionOperandEnum_WebpageConditionOperand[WebpageConditionOperandEnum_WebpageConditionOperand["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Operand denoting a webpage URL targeting condition.
     *
     * @generated from protobuf enum value: URL = 2;
     */
    WebpageConditionOperandEnum_WebpageConditionOperand[WebpageConditionOperandEnum_WebpageConditionOperand["URL"] = 2] = "URL";
    /**
     * Operand denoting a webpage category targeting condition.
     *
     * @generated from protobuf enum value: CATEGORY = 3;
     */
    WebpageConditionOperandEnum_WebpageConditionOperand[WebpageConditionOperandEnum_WebpageConditionOperand["CATEGORY"] = 3] = "CATEGORY";
    /**
     * Operand denoting a webpage title targeting condition.
     *
     * @generated from protobuf enum value: PAGE_TITLE = 4;
     */
    WebpageConditionOperandEnum_WebpageConditionOperand[WebpageConditionOperandEnum_WebpageConditionOperand["PAGE_TITLE"] = 4] = "PAGE_TITLE";
    /**
     * Operand denoting a webpage content targeting condition.
     *
     * @generated from protobuf enum value: PAGE_CONTENT = 5;
     */
    WebpageConditionOperandEnum_WebpageConditionOperand[WebpageConditionOperandEnum_WebpageConditionOperand["PAGE_CONTENT"] = 5] = "PAGE_CONTENT";
    /**
     * Operand denoting a webpage custom label targeting condition.
     *
     * @generated from protobuf enum value: CUSTOM_LABEL = 6;
     */
    WebpageConditionOperandEnum_WebpageConditionOperand[WebpageConditionOperandEnum_WebpageConditionOperand["CUSTOM_LABEL"] = 6] = "CUSTOM_LABEL";
})(WebpageConditionOperandEnum_WebpageConditionOperand = exports.WebpageConditionOperandEnum_WebpageConditionOperand || (exports.WebpageConditionOperandEnum_WebpageConditionOperand = {}));
// @generated message type with reflection information, may provide speed optimized methods
class WebpageConditionOperandEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.enums.WebpageConditionOperandEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.enums.WebpageConditionOperandEnum
 */
exports.WebpageConditionOperandEnum = new WebpageConditionOperandEnum$Type();
