// google-ads-api-client/dist/protos/bindings/google/api/serviceusage/v1beta1/serviceusage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceUsage = exports.GetServiceIdentityMetadata = exports.GetServiceIdentityResponse = exports.GenerateServiceIdentityRequest = exports.DeleteAdminQuotaPolicyMetadata = exports.UpdateAdminQuotaPolicyMetadata = exports.CreateAdminQuotaPolicyMetadata = exports.ImportAdminQuotaPoliciesMetadata = exports.ImportAdminQuotaPoliciesResponse = exports.ImportConsumerOverridesMetadata = exports.ImportConsumerOverridesResponse = exports.ImportConsumerOverridesRequest = exports.BatchCreateConsumerOverridesResponse = exports.ListConsumerOverridesResponse = exports.ListConsumerOverridesRequest = exports.DeleteConsumerOverrideRequest = exports.UpdateConsumerOverrideRequest = exports.CreateConsumerOverrideRequest = exports.ImportAdminOverridesMetadata = exports.ImportAdminOverridesResponse = exports.ImportAdminOverridesRequest = exports.BatchCreateAdminOverridesResponse = exports.ListAdminOverridesResponse = exports.ListAdminOverridesRequest = exports.DeleteAdminOverrideRequest = exports.UpdateAdminOverrideRequest = exports.CreateAdminOverrideRequest = exports.GetConsumerQuotaLimitRequest = exports.GetConsumerQuotaMetricRequest = exports.ListConsumerQuotaMetricsResponse = exports.ListConsumerQuotaMetricsRequest = exports.BatchEnableServicesRequest = exports.ListServicesResponse = exports.ListServicesRequest = exports.GetServiceRequest = exports.DisableServiceRequest = exports.EnableServiceRequest = exports.GetServiceIdentityResponse_IdentityState = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "google/api/serviceusage/v1beta1/serviceusage.proto" (package "google.api.serviceusage.v1beta1", syntax proto3)
// tslint:disable
//
// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const resources_1 = require("./resources");
const operations_1 = require("../../../longrunning/operations");
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const resources_2 = require("./resources");
const resources_3 = require("./resources");
const resources_4 = require("./resources");
const field_mask_1 = require("../../../protobuf/field_mask");
const resources_5 = require("./resources");
const resources_6 = require("./resources");
const resources_7 = require("./resources");
const resources_8 = require("./resources");
const resources_9 = require("./resources");
/**
 * Enum for service identity state.
 *
 * @generated from protobuf enum google.api.serviceusage.v1beta1.GetServiceIdentityResponse.IdentityState
 */
var GetServiceIdentityResponse_IdentityState;
(function (GetServiceIdentityResponse_IdentityState) {
    /**
     * Default service identity state. This value is used if the state is
     * omitted.
     *
     * @generated from protobuf enum value: IDENTITY_STATE_UNSPECIFIED = 0;
     */
    GetServiceIdentityResponse_IdentityState[GetServiceIdentityResponse_IdentityState["IDENTITY_STATE_UNSPECIFIED"] = 0] = "IDENTITY_STATE_UNSPECIFIED";
    /**
     * Service identity has been created and can be used.
     *
     * @generated from protobuf enum value: ACTIVE = 1;
     */
    GetServiceIdentityResponse_IdentityState[GetServiceIdentityResponse_IdentityState["ACTIVE"] = 1] = "ACTIVE";
})(GetServiceIdentityResponse_IdentityState = exports.GetServiceIdentityResponse_IdentityState || (exports.GetServiceIdentityResponse_IdentityState = {}));
// @generated message type with reflection information, may provide speed optimized methods
class EnableServiceRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.EnableServiceRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { name: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.EnableServiceRequest
 */
exports.EnableServiceRequest = new EnableServiceRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DisableServiceRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.DisableServiceRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { name: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.DisableServiceRequest
 */
exports.DisableServiceRequest = new DisableServiceRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GetServiceRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.GetServiceRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { name: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.GetServiceRequest
 */
exports.GetServiceRequest = new GetServiceRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListServicesRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ListServicesRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "page_size", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "filter", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { parent: "", pageSize: 0, pageToken: "", filter: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* int32 page_size */ 2:
                    message.pageSize = reader.int32();
                    break;
                case /* string page_token */ 3:
                    message.pageToken = reader.string();
                    break;
                case /* string filter */ 4:
                    message.filter = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* int32 page_size = 2; */
        if (message.pageSize !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.pageSize);
        /* string page_token = 3; */
        if (message.pageToken !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.pageToken);
        /* string filter = 4; */
        if (message.filter !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.filter);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ListServicesRequest
 */
exports.ListServicesRequest = new ListServicesRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListServicesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ListServicesResponse", [
            { no: 1, name: "services", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_9.Service },
            { no: 2, name: "next_page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { services: [], nextPageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.Service services */ 1:
                    message.services.push(resources_9.Service.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string next_page_token */ 2:
                    message.nextPageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.Service services = 1; */
        for (let i = 0; i < message.services.length; i++)
            resources_9.Service.internalBinaryWrite(message.services[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string next_page_token = 2; */
        if (message.nextPageToken !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.nextPageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ListServicesResponse
 */
exports.ListServicesResponse = new ListServicesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class BatchEnableServicesRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.BatchEnableServicesRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "service_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { parent: "", serviceIds: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* repeated string service_ids */ 2:
                    message.serviceIds.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* repeated string service_ids = 2; */
        for (let i = 0; i < message.serviceIds.length; i++)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.serviceIds[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.BatchEnableServicesRequest
 */
exports.BatchEnableServicesRequest = new BatchEnableServicesRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListConsumerQuotaMetricsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ListConsumerQuotaMetricsRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "page_size", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "view", kind: "enum", T: () => ["google.api.serviceusage.v1beta1.QuotaView", resources_8.QuotaView] }
        ]);
    }
    create(value) {
        const message = { parent: "", pageSize: 0, pageToken: "", view: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* int32 page_size */ 2:
                    message.pageSize = reader.int32();
                    break;
                case /* string page_token */ 3:
                    message.pageToken = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.QuotaView view */ 4:
                    message.view = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* int32 page_size = 2; */
        if (message.pageSize !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.pageSize);
        /* string page_token = 3; */
        if (message.pageToken !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.pageToken);
        /* google.api.serviceusage.v1beta1.QuotaView view = 4; */
        if (message.view !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.view);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ListConsumerQuotaMetricsRequest
 */
exports.ListConsumerQuotaMetricsRequest = new ListConsumerQuotaMetricsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListConsumerQuotaMetricsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ListConsumerQuotaMetricsResponse", [
            { no: 1, name: "metrics", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_7.ConsumerQuotaMetric },
            { no: 2, name: "next_page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { metrics: [], nextPageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.ConsumerQuotaMetric metrics */ 1:
                    message.metrics.push(resources_7.ConsumerQuotaMetric.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string next_page_token */ 2:
                    message.nextPageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.ConsumerQuotaMetric metrics = 1; */
        for (let i = 0; i < message.metrics.length; i++)
            resources_7.ConsumerQuotaMetric.internalBinaryWrite(message.metrics[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string next_page_token = 2; */
        if (message.nextPageToken !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.nextPageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ListConsumerQuotaMetricsResponse
 */
exports.ListConsumerQuotaMetricsResponse = new ListConsumerQuotaMetricsResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GetConsumerQuotaMetricRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.GetConsumerQuotaMetricRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "view", kind: "enum", T: () => ["google.api.serviceusage.v1beta1.QuotaView", resources_8.QuotaView] }
        ]);
    }
    create(value) {
        const message = { name: "", view: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.QuotaView view */ 2:
                    message.view = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.api.serviceusage.v1beta1.QuotaView view = 2; */
        if (message.view !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.view);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.GetConsumerQuotaMetricRequest
 */
exports.GetConsumerQuotaMetricRequest = new GetConsumerQuotaMetricRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GetConsumerQuotaLimitRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.GetConsumerQuotaLimitRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "view", kind: "enum", T: () => ["google.api.serviceusage.v1beta1.QuotaView", resources_8.QuotaView] }
        ]);
    }
    create(value) {
        const message = { name: "", view: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.QuotaView view */ 2:
                    message.view = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.api.serviceusage.v1beta1.QuotaView view = 2; */
        if (message.view !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.view);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.GetConsumerQuotaLimitRequest
 */
exports.GetConsumerQuotaLimitRequest = new GetConsumerQuotaLimitRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CreateAdminOverrideRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.CreateAdminOverrideRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "override", kind: "message", T: () => resources_6.QuotaOverride },
            { no: 3, name: "force", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "force_only", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.serviceusage.v1beta1.QuotaSafetyCheck", resources_5.QuotaSafetyCheck] }
        ]);
    }
    create(value) {
        const message = { parent: "", force: false, forceOnly: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.QuotaOverride override */ 2:
                    message.override = resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options, message.override);
                    break;
                case /* bool force */ 3:
                    message.force = reader.bool();
                    break;
                case /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only */ 4:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forceOnly.push(reader.int32());
                    else
                        message.forceOnly.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* google.api.serviceusage.v1beta1.QuotaOverride override = 2; */
        if (message.override)
            resources_6.QuotaOverride.internalBinaryWrite(message.override, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool force = 3; */
        if (message.force !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.force);
        /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only = 4; */
        if (message.forceOnly.length) {
            writer.tag(4, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.forceOnly.length; i++)
                writer.int32(message.forceOnly[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.CreateAdminOverrideRequest
 */
exports.CreateAdminOverrideRequest = new CreateAdminOverrideRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UpdateAdminOverrideRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.UpdateAdminOverrideRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "override", kind: "message", T: () => resources_6.QuotaOverride },
            { no: 3, name: "force", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "update_mask", kind: "message", T: () => field_mask_1.FieldMask },
            { no: 5, name: "force_only", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.serviceusage.v1beta1.QuotaSafetyCheck", resources_5.QuotaSafetyCheck] }
        ]);
    }
    create(value) {
        const message = { name: "", force: false, forceOnly: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.QuotaOverride override */ 2:
                    message.override = resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options, message.override);
                    break;
                case /* bool force */ 3:
                    message.force = reader.bool();
                    break;
                case /* google.protobuf.FieldMask update_mask */ 4:
                    message.updateMask = field_mask_1.FieldMask.internalBinaryRead(reader, reader.uint32(), options, message.updateMask);
                    break;
                case /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only */ 5:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forceOnly.push(reader.int32());
                    else
                        message.forceOnly.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.api.serviceusage.v1beta1.QuotaOverride override = 2; */
        if (message.override)
            resources_6.QuotaOverride.internalBinaryWrite(message.override, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool force = 3; */
        if (message.force !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.force);
        /* google.protobuf.FieldMask update_mask = 4; */
        if (message.updateMask)
            field_mask_1.FieldMask.internalBinaryWrite(message.updateMask, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only = 5; */
        if (message.forceOnly.length) {
            writer.tag(5, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.forceOnly.length; i++)
                writer.int32(message.forceOnly[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.UpdateAdminOverrideRequest
 */
exports.UpdateAdminOverrideRequest = new UpdateAdminOverrideRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DeleteAdminOverrideRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.DeleteAdminOverrideRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "force", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "force_only", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.serviceusage.v1beta1.QuotaSafetyCheck", resources_5.QuotaSafetyCheck] }
        ]);
    }
    create(value) {
        const message = { name: "", force: false, forceOnly: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* bool force */ 2:
                    message.force = reader.bool();
                    break;
                case /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only */ 3:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forceOnly.push(reader.int32());
                    else
                        message.forceOnly.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* bool force = 2; */
        if (message.force !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.force);
        /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only = 3; */
        if (message.forceOnly.length) {
            writer.tag(3, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.forceOnly.length; i++)
                writer.int32(message.forceOnly[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.DeleteAdminOverrideRequest
 */
exports.DeleteAdminOverrideRequest = new DeleteAdminOverrideRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListAdminOverridesRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ListAdminOverridesRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "page_size", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { parent: "", pageSize: 0, pageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* int32 page_size */ 2:
                    message.pageSize = reader.int32();
                    break;
                case /* string page_token */ 3:
                    message.pageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* int32 page_size = 2; */
        if (message.pageSize !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.pageSize);
        /* string page_token = 3; */
        if (message.pageToken !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.pageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ListAdminOverridesRequest
 */
exports.ListAdminOverridesRequest = new ListAdminOverridesRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListAdminOverridesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ListAdminOverridesResponse", [
            { no: 1, name: "overrides", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_6.QuotaOverride },
            { no: 2, name: "next_page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { overrides: [], nextPageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides */ 1:
                    message.overrides.push(resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string next_page_token */ 2:
                    message.nextPageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides = 1; */
        for (let i = 0; i < message.overrides.length; i++)
            resources_6.QuotaOverride.internalBinaryWrite(message.overrides[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string next_page_token = 2; */
        if (message.nextPageToken !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.nextPageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ListAdminOverridesResponse
 */
exports.ListAdminOverridesResponse = new ListAdminOverridesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class BatchCreateAdminOverridesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.BatchCreateAdminOverridesResponse", [
            { no: 1, name: "overrides", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_6.QuotaOverride }
        ]);
    }
    create(value) {
        const message = { overrides: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides */ 1:
                    message.overrides.push(resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides = 1; */
        for (let i = 0; i < message.overrides.length; i++)
            resources_6.QuotaOverride.internalBinaryWrite(message.overrides[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.BatchCreateAdminOverridesResponse
 */
exports.BatchCreateAdminOverridesResponse = new BatchCreateAdminOverridesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImportAdminOverridesRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ImportAdminOverridesRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "inline_source", kind: "message", oneof: "source", T: () => resources_4.OverrideInlineSource },
            { no: 3, name: "force", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "force_only", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.serviceusage.v1beta1.QuotaSafetyCheck", resources_5.QuotaSafetyCheck] }
        ]);
    }
    create(value) {
        const message = { parent: "", source: { oneofKind: undefined }, force: false, forceOnly: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.OverrideInlineSource inline_source */ 2:
                    message.source = {
                        oneofKind: "inlineSource",
                        inlineSource: resources_4.OverrideInlineSource.internalBinaryRead(reader, reader.uint32(), options, message.source.inlineSource)
                    };
                    break;
                case /* bool force */ 3:
                    message.force = reader.bool();
                    break;
                case /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only */ 4:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forceOnly.push(reader.int32());
                    else
                        message.forceOnly.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* google.api.serviceusage.v1beta1.OverrideInlineSource inline_source = 2; */
        if (message.source.oneofKind === "inlineSource")
            resources_4.OverrideInlineSource.internalBinaryWrite(message.source.inlineSource, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool force = 3; */
        if (message.force !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.force);
        /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only = 4; */
        if (message.forceOnly.length) {
            writer.tag(4, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.forceOnly.length; i++)
                writer.int32(message.forceOnly[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ImportAdminOverridesRequest
 */
exports.ImportAdminOverridesRequest = new ImportAdminOverridesRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImportAdminOverridesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ImportAdminOverridesResponse", [
            { no: 1, name: "overrides", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_6.QuotaOverride }
        ]);
    }
    create(value) {
        const message = { overrides: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides */ 1:
                    message.overrides.push(resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides = 1; */
        for (let i = 0; i < message.overrides.length; i++)
            resources_6.QuotaOverride.internalBinaryWrite(message.overrides[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ImportAdminOverridesResponse
 */
exports.ImportAdminOverridesResponse = new ImportAdminOverridesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImportAdminOverridesMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ImportAdminOverridesMetadata", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ImportAdminOverridesMetadata
 */
exports.ImportAdminOverridesMetadata = new ImportAdminOverridesMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CreateConsumerOverrideRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.CreateConsumerOverrideRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "override", kind: "message", T: () => resources_6.QuotaOverride },
            { no: 3, name: "force", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "force_only", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.serviceusage.v1beta1.QuotaSafetyCheck", resources_5.QuotaSafetyCheck] }
        ]);
    }
    create(value) {
        const message = { parent: "", force: false, forceOnly: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.QuotaOverride override */ 2:
                    message.override = resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options, message.override);
                    break;
                case /* bool force */ 3:
                    message.force = reader.bool();
                    break;
                case /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only */ 4:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forceOnly.push(reader.int32());
                    else
                        message.forceOnly.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* google.api.serviceusage.v1beta1.QuotaOverride override = 2; */
        if (message.override)
            resources_6.QuotaOverride.internalBinaryWrite(message.override, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool force = 3; */
        if (message.force !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.force);
        /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only = 4; */
        if (message.forceOnly.length) {
            writer.tag(4, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.forceOnly.length; i++)
                writer.int32(message.forceOnly[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.CreateConsumerOverrideRequest
 */
exports.CreateConsumerOverrideRequest = new CreateConsumerOverrideRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UpdateConsumerOverrideRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.UpdateConsumerOverrideRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "override", kind: "message", T: () => resources_6.QuotaOverride },
            { no: 3, name: "force", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "update_mask", kind: "message", T: () => field_mask_1.FieldMask },
            { no: 5, name: "force_only", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.serviceusage.v1beta1.QuotaSafetyCheck", resources_5.QuotaSafetyCheck] }
        ]);
    }
    create(value) {
        const message = { name: "", force: false, forceOnly: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.QuotaOverride override */ 2:
                    message.override = resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options, message.override);
                    break;
                case /* bool force */ 3:
                    message.force = reader.bool();
                    break;
                case /* google.protobuf.FieldMask update_mask */ 4:
                    message.updateMask = field_mask_1.FieldMask.internalBinaryRead(reader, reader.uint32(), options, message.updateMask);
                    break;
                case /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only */ 5:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forceOnly.push(reader.int32());
                    else
                        message.forceOnly.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.api.serviceusage.v1beta1.QuotaOverride override = 2; */
        if (message.override)
            resources_6.QuotaOverride.internalBinaryWrite(message.override, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool force = 3; */
        if (message.force !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.force);
        /* google.protobuf.FieldMask update_mask = 4; */
        if (message.updateMask)
            field_mask_1.FieldMask.internalBinaryWrite(message.updateMask, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only = 5; */
        if (message.forceOnly.length) {
            writer.tag(5, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.forceOnly.length; i++)
                writer.int32(message.forceOnly[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.UpdateConsumerOverrideRequest
 */
exports.UpdateConsumerOverrideRequest = new UpdateConsumerOverrideRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DeleteConsumerOverrideRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.DeleteConsumerOverrideRequest", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "force", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "force_only", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.serviceusage.v1beta1.QuotaSafetyCheck", resources_5.QuotaSafetyCheck] }
        ]);
    }
    create(value) {
        const message = { name: "", force: false, forceOnly: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* bool force */ 2:
                    message.force = reader.bool();
                    break;
                case /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only */ 3:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forceOnly.push(reader.int32());
                    else
                        message.forceOnly.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* bool force = 2; */
        if (message.force !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.force);
        /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only = 3; */
        if (message.forceOnly.length) {
            writer.tag(3, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.forceOnly.length; i++)
                writer.int32(message.forceOnly[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.DeleteConsumerOverrideRequest
 */
exports.DeleteConsumerOverrideRequest = new DeleteConsumerOverrideRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListConsumerOverridesRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ListConsumerOverridesRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "page_size", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { parent: "", pageSize: 0, pageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* int32 page_size */ 2:
                    message.pageSize = reader.int32();
                    break;
                case /* string page_token */ 3:
                    message.pageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* int32 page_size = 2; */
        if (message.pageSize !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.pageSize);
        /* string page_token = 3; */
        if (message.pageToken !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.pageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ListConsumerOverridesRequest
 */
exports.ListConsumerOverridesRequest = new ListConsumerOverridesRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListConsumerOverridesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ListConsumerOverridesResponse", [
            { no: 1, name: "overrides", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_6.QuotaOverride },
            { no: 2, name: "next_page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { overrides: [], nextPageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides */ 1:
                    message.overrides.push(resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string next_page_token */ 2:
                    message.nextPageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides = 1; */
        for (let i = 0; i < message.overrides.length; i++)
            resources_6.QuotaOverride.internalBinaryWrite(message.overrides[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string next_page_token = 2; */
        if (message.nextPageToken !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.nextPageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ListConsumerOverridesResponse
 */
exports.ListConsumerOverridesResponse = new ListConsumerOverridesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class BatchCreateConsumerOverridesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.BatchCreateConsumerOverridesResponse", [
            { no: 1, name: "overrides", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_6.QuotaOverride }
        ]);
    }
    create(value) {
        const message = { overrides: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides */ 1:
                    message.overrides.push(resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides = 1; */
        for (let i = 0; i < message.overrides.length; i++)
            resources_6.QuotaOverride.internalBinaryWrite(message.overrides[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.BatchCreateConsumerOverridesResponse
 */
exports.BatchCreateConsumerOverridesResponse = new BatchCreateConsumerOverridesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImportConsumerOverridesRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ImportConsumerOverridesRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "inline_source", kind: "message", oneof: "source", T: () => resources_4.OverrideInlineSource },
            { no: 3, name: "force", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "force_only", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.serviceusage.v1beta1.QuotaSafetyCheck", resources_5.QuotaSafetyCheck] }
        ]);
    }
    create(value) {
        const message = { parent: "", source: { oneofKind: undefined }, force: false, forceOnly: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                case /* google.api.serviceusage.v1beta1.OverrideInlineSource inline_source */ 2:
                    message.source = {
                        oneofKind: "inlineSource",
                        inlineSource: resources_4.OverrideInlineSource.internalBinaryRead(reader, reader.uint32(), options, message.source.inlineSource)
                    };
                    break;
                case /* bool force */ 3:
                    message.force = reader.bool();
                    break;
                case /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only */ 4:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forceOnly.push(reader.int32());
                    else
                        message.forceOnly.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* google.api.serviceusage.v1beta1.OverrideInlineSource inline_source = 2; */
        if (message.source.oneofKind === "inlineSource")
            resources_4.OverrideInlineSource.internalBinaryWrite(message.source.inlineSource, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool force = 3; */
        if (message.force !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.force);
        /* repeated google.api.serviceusage.v1beta1.QuotaSafetyCheck force_only = 4; */
        if (message.forceOnly.length) {
            writer.tag(4, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.forceOnly.length; i++)
                writer.int32(message.forceOnly[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ImportConsumerOverridesRequest
 */
exports.ImportConsumerOverridesRequest = new ImportConsumerOverridesRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImportConsumerOverridesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ImportConsumerOverridesResponse", [
            { no: 1, name: "overrides", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_6.QuotaOverride }
        ]);
    }
    create(value) {
        const message = { overrides: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides */ 1:
                    message.overrides.push(resources_6.QuotaOverride.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.QuotaOverride overrides = 1; */
        for (let i = 0; i < message.overrides.length; i++)
            resources_6.QuotaOverride.internalBinaryWrite(message.overrides[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ImportConsumerOverridesResponse
 */
exports.ImportConsumerOverridesResponse = new ImportConsumerOverridesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImportConsumerOverridesMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ImportConsumerOverridesMetadata", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ImportConsumerOverridesMetadata
 */
exports.ImportConsumerOverridesMetadata = new ImportConsumerOverridesMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImportAdminQuotaPoliciesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ImportAdminQuotaPoliciesResponse", [
            { no: 1, name: "policies", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => resources_3.AdminQuotaPolicy }
        ]);
    }
    create(value) {
        const message = { policies: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.api.serviceusage.v1beta1.AdminQuotaPolicy policies */ 1:
                    message.policies.push(resources_3.AdminQuotaPolicy.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.api.serviceusage.v1beta1.AdminQuotaPolicy policies = 1; */
        for (let i = 0; i < message.policies.length; i++)
            resources_3.AdminQuotaPolicy.internalBinaryWrite(message.policies[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ImportAdminQuotaPoliciesResponse
 */
exports.ImportAdminQuotaPoliciesResponse = new ImportAdminQuotaPoliciesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImportAdminQuotaPoliciesMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.ImportAdminQuotaPoliciesMetadata", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.ImportAdminQuotaPoliciesMetadata
 */
exports.ImportAdminQuotaPoliciesMetadata = new ImportAdminQuotaPoliciesMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CreateAdminQuotaPolicyMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.CreateAdminQuotaPolicyMetadata", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.CreateAdminQuotaPolicyMetadata
 */
exports.CreateAdminQuotaPolicyMetadata = new CreateAdminQuotaPolicyMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UpdateAdminQuotaPolicyMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.UpdateAdminQuotaPolicyMetadata", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.UpdateAdminQuotaPolicyMetadata
 */
exports.UpdateAdminQuotaPolicyMetadata = new UpdateAdminQuotaPolicyMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DeleteAdminQuotaPolicyMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.DeleteAdminQuotaPolicyMetadata", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.DeleteAdminQuotaPolicyMetadata
 */
exports.DeleteAdminQuotaPolicyMetadata = new DeleteAdminQuotaPolicyMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateServiceIdentityRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.GenerateServiceIdentityRequest", [
            { no: 1, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { parent: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string parent */ 1:
                    message.parent = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string parent = 1; */
        if (message.parent !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.parent);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.GenerateServiceIdentityRequest
 */
exports.GenerateServiceIdentityRequest = new GenerateServiceIdentityRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GetServiceIdentityResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.GetServiceIdentityResponse", [
            { no: 1, name: "identity", kind: "message", T: () => resources_2.ServiceIdentity },
            { no: 2, name: "state", kind: "enum", T: () => ["google.api.serviceusage.v1beta1.GetServiceIdentityResponse.IdentityState", GetServiceIdentityResponse_IdentityState] }
        ]);
    }
    create(value) {
        const message = { state: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.api.serviceusage.v1beta1.ServiceIdentity identity */ 1:
                    message.identity = resources_2.ServiceIdentity.internalBinaryRead(reader, reader.uint32(), options, message.identity);
                    break;
                case /* google.api.serviceusage.v1beta1.GetServiceIdentityResponse.IdentityState state */ 2:
                    message.state = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.api.serviceusage.v1beta1.ServiceIdentity identity = 1; */
        if (message.identity)
            resources_2.ServiceIdentity.internalBinaryWrite(message.identity, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.api.serviceusage.v1beta1.GetServiceIdentityResponse.IdentityState state = 2; */
        if (message.state !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.state);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.GetServiceIdentityResponse
 */
exports.GetServiceIdentityResponse = new GetServiceIdentityResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GetServiceIdentityMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1beta1.GetServiceIdentityMetadata", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1beta1.GetServiceIdentityMetadata
 */
exports.GetServiceIdentityMetadata = new GetServiceIdentityMetadata$Type();
/**
 * @generated ServiceType for protobuf service google.api.serviceusage.v1beta1.ServiceUsage
 */
exports.ServiceUsage = new runtime_rpc_1.ServiceType("google.api.serviceusage.v1beta1.ServiceUsage", [
    { name: "EnableService", options: { "google.api.http": { post: "/v1beta1/{name=*/*/services/*}:enable", body: "*" }, "google.longrunning.operation_info": { responseType: "google.protobuf.Empty", metadataType: "OperationMetadata" } }, I: exports.EnableServiceRequest, O: operations_1.Operation },
    { name: "DisableService", options: { "google.api.http": { post: "/v1beta1/{name=*/*/services/*}:disable", body: "*" }, "google.longrunning.operation_info": { responseType: "google.protobuf.Empty", metadataType: "OperationMetadata" } }, I: exports.DisableServiceRequest, O: operations_1.Operation },
    { name: "GetService", options: { "google.api.http": { get: "/v1beta1/{name=*/*/services/*}" } }, I: exports.GetServiceRequest, O: resources_9.Service },
    { name: "ListServices", options: { "google.api.http": { get: "/v1beta1/{parent=*/*}/services" } }, I: exports.ListServicesRequest, O: exports.ListServicesResponse },
    { name: "BatchEnableServices", options: { "google.api.http": { post: "/v1beta1/{parent=*/*}/services:batchEnable", body: "*" }, "google.longrunning.operation_info": { responseType: "google.protobuf.Empty", metadataType: "OperationMetadata" } }, I: exports.BatchEnableServicesRequest, O: operations_1.Operation },
    { name: "ListConsumerQuotaMetrics", options: { "google.api.http": { get: "/v1beta1/{parent=*/*/services/*}/consumerQuotaMetrics" } }, I: exports.ListConsumerQuotaMetricsRequest, O: exports.ListConsumerQuotaMetricsResponse },
    { name: "GetConsumerQuotaMetric", options: { "google.api.http": { get: "/v1beta1/{name=*/*/services/*/consumerQuotaMetrics/*}" } }, I: exports.GetConsumerQuotaMetricRequest, O: resources_7.ConsumerQuotaMetric },
    { name: "GetConsumerQuotaLimit", options: { "google.api.http": { get: "/v1beta1/{name=*/*/services/*/consumerQuotaMetrics/*/limits/*}" } }, I: exports.GetConsumerQuotaLimitRequest, O: resources_1.ConsumerQuotaLimit },
    { name: "CreateAdminOverride", options: { "google.api.http": { post: "/v1beta1/{parent=*/*/services/*/consumerQuotaMetrics/*/limits/*}/adminOverrides", body: "override" }, "google.longrunning.operation_info": { responseType: "QuotaOverride", metadataType: "OperationMetadata" } }, I: exports.CreateAdminOverrideRequest, O: operations_1.Operation },
    { name: "UpdateAdminOverride", options: { "google.api.http": { patch: "/v1beta1/{name=*/*/services/*/consumerQuotaMetrics/*/limits/*/adminOverrides/*}", body: "override" }, "google.longrunning.operation_info": { responseType: "QuotaOverride", metadataType: "OperationMetadata" } }, I: exports.UpdateAdminOverrideRequest, O: operations_1.Operation },
    { name: "DeleteAdminOverride", options: { "google.api.http": { delete: "/v1beta1/{name=*/*/services/*/consumerQuotaMetrics/*/limits/*/adminOverrides/*}" }, "google.longrunning.operation_info": { responseType: "google.protobuf.Empty", metadataType: "OperationMetadata" } }, I: exports.DeleteAdminOverrideRequest, O: operations_1.Operation },
    { name: "ListAdminOverrides", options: { "google.api.http": { get: "/v1beta1/{parent=*/*/services/*/consumerQuotaMetrics/*/limits/*}/adminOverrides" } }, I: exports.ListAdminOverridesRequest, O: exports.ListAdminOverridesResponse },
    { name: "ImportAdminOverrides", options: { "google.api.http": { post: "/v1beta1/{parent=*/*/services/*}/consumerQuotaMetrics:importAdminOverrides", body: "*" }, "google.longrunning.operation_info": { responseType: "ImportAdminOverridesResponse", metadataType: "ImportAdminOverridesMetadata" } }, I: exports.ImportAdminOverridesRequest, O: operations_1.Operation },
    { name: "CreateConsumerOverride", options: { "google.api.http": { post: "/v1beta1/{parent=*/*/services/*/consumerQuotaMetrics/*/limits/*}/consumerOverrides", body: "override" }, "google.longrunning.operation_info": { responseType: "QuotaOverride", metadataType: "OperationMetadata" } }, I: exports.CreateConsumerOverrideRequest, O: operations_1.Operation },
    { name: "UpdateConsumerOverride", options: { "google.api.http": { patch: "/v1beta1/{name=*/*/services/*/consumerQuotaMetrics/*/limits/*/consumerOverrides/*}", body: "override" }, "google.longrunning.operation_info": { responseType: "QuotaOverride", metadataType: "OperationMetadata" } }, I: exports.UpdateConsumerOverrideRequest, O: operations_1.Operation },
    { name: "DeleteConsumerOverride", options: { "google.api.http": { delete: "/v1beta1/{name=*/*/services/*/consumerQuotaMetrics/*/limits/*/consumerOverrides/*}" }, "google.longrunning.operation_info": { responseType: "google.protobuf.Empty", metadataType: "OperationMetadata" } }, I: exports.DeleteConsumerOverrideRequest, O: operations_1.Operation },
    { name: "ListConsumerOverrides", options: { "google.api.http": { get: "/v1beta1/{parent=*/*/services/*/consumerQuotaMetrics/*/limits/*}/consumerOverrides" } }, I: exports.ListConsumerOverridesRequest, O: exports.ListConsumerOverridesResponse },
    { name: "ImportConsumerOverrides", options: { "google.api.http": { post: "/v1beta1/{parent=*/*/services/*}/consumerQuotaMetrics:importConsumerOverrides", body: "*" }, "google.longrunning.operation_info": { responseType: "ImportConsumerOverridesResponse", metadataType: "ImportConsumerOverridesMetadata" } }, I: exports.ImportConsumerOverridesRequest, O: operations_1.Operation },
    { name: "GenerateServiceIdentity", options: { "google.api.http": { post: "/v1beta1/{parent=*/*/services/*}:generateServiceIdentity" }, "google.longrunning.operation_info": { responseType: "ServiceIdentity", metadataType: "google.protobuf.Empty" } }, I: exports.GenerateServiceIdentityRequest, O: operations_1.Operation }
], { "google.api.default_host": "serviceusage.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/cloud-platform,https://www.googleapis.com/auth/cloud-platform.read-only,https://www.googleapis.com/auth/service.management" });
