// google-ads-api-client/dist/protos/bindings/google/api/serviceusage/v1beta1/serviceusage.client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceUsageClient = void 0;
const serviceusage_1 = require("./serviceusage");
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
/**
 * [Service Usage API](https://cloud.google.com/service-usage/docs/overview)
 *
 * @generated from protobuf service google.api.serviceusage.v1beta1.ServiceUsage
 */
class ServiceUsageClient {
    constructor(_transport) {
        this._transport = _transport;
        this.typeName = serviceusage_1.ServiceUsage.typeName;
        this.methods = serviceusage_1.ServiceUsage.methods;
        this.options = serviceusage_1.ServiceUsage.options;
    }
    /**
     * Enables a service so that it can be used with a project.
     *
     * Operation response type: `google.protobuf.Empty`
     *
     * @deprecated
     * @generated from protobuf rpc: EnableService(google.api.serviceusage.v1beta1.EnableServiceRequest) returns (google.longrunning.Operation);
     */
    enableService(input, options) {
        const method = this.methods[0], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Disables a service so that it can no longer be used with a project.
     * This prevents unintended usage that may cause unexpected billing
     * charges or security leaks.
     *
     * It is not valid to call the disable method on a service that is not
     * currently enabled. Callers will receive a `FAILED_PRECONDITION` status if
     * the target service is not currently enabled.
     *
     * Operation response type: `google.protobuf.Empty`
     *
     * @deprecated
     * @generated from protobuf rpc: DisableService(google.api.serviceusage.v1beta1.DisableServiceRequest) returns (google.longrunning.Operation);
     */
    disableService(input, options) {
        const method = this.methods[1], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Returns the service configuration and enabled state for a given service.
     *
     * @deprecated
     * @generated from protobuf rpc: GetService(google.api.serviceusage.v1beta1.GetServiceRequest) returns (google.api.serviceusage.v1beta1.Service);
     */
    getService(input, options) {
        const method = this.methods[2], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Lists all services available to the specified project, and the current
     * state of those services with respect to the project. The list includes
     * all public services, all services for which the calling user has the
     * `servicemanagement.services.bind` permission, and all services that have
     * already been enabled on the project. The list can be filtered to
     * only include services in a specific state, for example to only include
     * services enabled on the project.
     *
     * @deprecated
     * @generated from protobuf rpc: ListServices(google.api.serviceusage.v1beta1.ListServicesRequest) returns (google.api.serviceusage.v1beta1.ListServicesResponse);
     */
    listServices(input, options) {
        const method = this.methods[3], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Enables multiple services on a project. The operation is atomic: if
     * enabling any service fails, then the entire batch fails, and no state
     * changes occur.
     *
     * Operation response type: `google.protobuf.Empty`
     *
     * @deprecated
     * @generated from protobuf rpc: BatchEnableServices(google.api.serviceusage.v1beta1.BatchEnableServicesRequest) returns (google.longrunning.Operation);
     */
    batchEnableServices(input, options) {
        const method = this.methods[4], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Retrieves a summary of all quota information visible to the service
     * consumer, organized by service metric. Each metric includes information
     * about all of its defined limits. Each limit includes the limit
     * configuration (quota unit, preciseness, default value), the current
     * effective limit value, and all of the overrides applied to the limit.
     *
     * @generated from protobuf rpc: ListConsumerQuotaMetrics(google.api.serviceusage.v1beta1.ListConsumerQuotaMetricsRequest) returns (google.api.serviceusage.v1beta1.ListConsumerQuotaMetricsResponse);
     */
    listConsumerQuotaMetrics(input, options) {
        const method = this.methods[5], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Retrieves a summary of quota information for a specific quota metric
     *
     * @generated from protobuf rpc: GetConsumerQuotaMetric(google.api.serviceusage.v1beta1.GetConsumerQuotaMetricRequest) returns (google.api.serviceusage.v1beta1.ConsumerQuotaMetric);
     */
    getConsumerQuotaMetric(input, options) {
        const method = this.methods[6], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Retrieves a summary of quota information for a specific quota limit.
     *
     * @generated from protobuf rpc: GetConsumerQuotaLimit(google.api.serviceusage.v1beta1.GetConsumerQuotaLimitRequest) returns (google.api.serviceusage.v1beta1.ConsumerQuotaLimit);
     */
    getConsumerQuotaLimit(input, options) {
        const method = this.methods[7], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Creates an admin override.
     * An admin override is applied by an administrator of a parent folder or
     * parent organization of the consumer receiving the override. An admin
     * override is intended to limit the amount of quota the consumer can use out
     * of the total quota pool allocated to all children of the folder or
     * organization.
     *
     * @generated from protobuf rpc: CreateAdminOverride(google.api.serviceusage.v1beta1.CreateAdminOverrideRequest) returns (google.longrunning.Operation);
     */
    createAdminOverride(input, options) {
        const method = this.methods[8], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Updates an admin override.
     *
     * @generated from protobuf rpc: UpdateAdminOverride(google.api.serviceusage.v1beta1.UpdateAdminOverrideRequest) returns (google.longrunning.Operation);
     */
    updateAdminOverride(input, options) {
        const method = this.methods[9], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Deletes an admin override.
     *
     * @generated from protobuf rpc: DeleteAdminOverride(google.api.serviceusage.v1beta1.DeleteAdminOverrideRequest) returns (google.longrunning.Operation);
     */
    deleteAdminOverride(input, options) {
        const method = this.methods[10], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Lists all admin overrides on this limit.
     *
     * @generated from protobuf rpc: ListAdminOverrides(google.api.serviceusage.v1beta1.ListAdminOverridesRequest) returns (google.api.serviceusage.v1beta1.ListAdminOverridesResponse);
     */
    listAdminOverrides(input, options) {
        const method = this.methods[11], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Creates or updates multiple admin overrides atomically, all on the
     * same consumer, but on many different metrics or limits.
     * The name field in the quota override message should not be set.
     *
     * @generated from protobuf rpc: ImportAdminOverrides(google.api.serviceusage.v1beta1.ImportAdminOverridesRequest) returns (google.longrunning.Operation);
     */
    importAdminOverrides(input, options) {
        const method = this.methods[12], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Creates a consumer override.
     * A consumer override is applied to the consumer on its own authority to
     * limit its own quota usage. Consumer overrides cannot be used to grant more
     * quota than would be allowed by admin overrides, producer overrides, or the
     * default limit of the service.
     *
     * @generated from protobuf rpc: CreateConsumerOverride(google.api.serviceusage.v1beta1.CreateConsumerOverrideRequest) returns (google.longrunning.Operation);
     */
    createConsumerOverride(input, options) {
        const method = this.methods[13], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Updates a consumer override.
     *
     * @generated from protobuf rpc: UpdateConsumerOverride(google.api.serviceusage.v1beta1.UpdateConsumerOverrideRequest) returns (google.longrunning.Operation);
     */
    updateConsumerOverride(input, options) {
        const method = this.methods[14], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Deletes a consumer override.
     *
     * @generated from protobuf rpc: DeleteConsumerOverride(google.api.serviceusage.v1beta1.DeleteConsumerOverrideRequest) returns (google.longrunning.Operation);
     */
    deleteConsumerOverride(input, options) {
        const method = this.methods[15], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Lists all consumer overrides on this limit.
     *
     * @generated from protobuf rpc: ListConsumerOverrides(google.api.serviceusage.v1beta1.ListConsumerOverridesRequest) returns (google.api.serviceusage.v1beta1.ListConsumerOverridesResponse);
     */
    listConsumerOverrides(input, options) {
        const method = this.methods[16], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Creates or updates multiple consumer overrides atomically, all on the
     * same consumer, but on many different metrics or limits.
     * The name field in the quota override message should not be set.
     *
     * @generated from protobuf rpc: ImportConsumerOverrides(google.api.serviceusage.v1beta1.ImportConsumerOverridesRequest) returns (google.longrunning.Operation);
     */
    importConsumerOverrides(input, options) {
        const method = this.methods[17], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Generates service identity for service.
     *
     * @generated from protobuf rpc: GenerateServiceIdentity(google.api.serviceusage.v1beta1.GenerateServiceIdentityRequest) returns (google.longrunning.Operation);
     */
    generateServiceIdentity(input, options) {
        const method = this.methods[18], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
}
exports.ServiceUsageClient = ServiceUsageClient;
