// google-ads-api-client/dist/protos/bindings/google/api/serviceusage/v1/resources.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationMetadata = exports.ServiceConfig = exports.Service = exports.State = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const monitoring_1 = require("../../monitoring");
const monitored_resource_1 = require("../../monitored_resource");
const endpoint_1 = require("../../endpoint");
const usage_1 = require("../../usage");
const auth_1 = require("../../auth");
const quota_1 = require("../../quota");
const documentation_1 = require("../../documentation");
const api_1 = require("../../../protobuf/api");
/**
 * Whether or not a service has been enabled for use by a consumer.
 *
 * @generated from protobuf enum google.api.serviceusage.v1.State
 */
var State;
(function (State) {
    /**
     * The default value, which indicates that the enabled state of the service
     * is unspecified or not meaningful. Currently, all consumers other than
     * projects (such as folders and organizations) are always in this state.
     *
     * @generated from protobuf enum value: STATE_UNSPECIFIED = 0;
     */
    State[State["STATE_UNSPECIFIED"] = 0] = "STATE_UNSPECIFIED";
    /**
     * The service cannot be used by this consumer. It has either been explicitly
     * disabled, or has never been enabled.
     *
     * @generated from protobuf enum value: DISABLED = 1;
     */
    State[State["DISABLED"] = 1] = "DISABLED";
    /**
     * The service has been explicitly enabled for use by this consumer.
     *
     * @generated from protobuf enum value: ENABLED = 2;
     */
    State[State["ENABLED"] = 2] = "ENABLED";
})(State = exports.State || (exports.State = {}));
// @generated message type with reflection information, may provide speed optimized methods
class Service$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1.Service", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "parent", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "config", kind: "message", T: () => exports.ServiceConfig },
            { no: 4, name: "state", kind: "enum", T: () => ["google.api.serviceusage.v1.State", State] }
        ]);
    }
    create(value) {
        const message = { name: "", parent: "", state: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* string parent */ 5:
                    message.parent = reader.string();
                    break;
                case /* google.api.serviceusage.v1.ServiceConfig config */ 2:
                    message.config = exports.ServiceConfig.internalBinaryRead(reader, reader.uint32(), options, message.config);
                    break;
                case /* google.api.serviceusage.v1.State state */ 4:
                    message.state = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* string parent = 5; */
        if (message.parent !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.parent);
        /* google.api.serviceusage.v1.ServiceConfig config = 2; */
        if (message.config)
            exports.ServiceConfig.internalBinaryWrite(message.config, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.api.serviceusage.v1.State state = 4; */
        if (message.state !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.state);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1.Service
 */
exports.Service = new Service$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ServiceConfig$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1.ServiceConfig", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "title", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "apis", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => api_1.Api },
            { no: 6, name: "documentation", kind: "message", T: () => documentation_1.Documentation },
            { no: 10, name: "quota", kind: "message", T: () => quota_1.Quota },
            { no: 11, name: "authentication", kind: "message", T: () => auth_1.Authentication },
            { no: 15, name: "usage", kind: "message", T: () => usage_1.Usage },
            { no: 18, name: "endpoints", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => endpoint_1.Endpoint },
            { no: 25, name: "monitored_resources", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => monitored_resource_1.MonitoredResourceDescriptor },
            { no: 28, name: "monitoring", kind: "message", T: () => monitoring_1.Monitoring }
        ]);
    }
    create(value) {
        const message = { name: "", title: "", apis: [], endpoints: [], monitoredResources: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* string title */ 2:
                    message.title = reader.string();
                    break;
                case /* repeated google.protobuf.Api apis */ 3:
                    message.apis.push(api_1.Api.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.api.Documentation documentation */ 6:
                    message.documentation = documentation_1.Documentation.internalBinaryRead(reader, reader.uint32(), options, message.documentation);
                    break;
                case /* google.api.Quota quota */ 10:
                    message.quota = quota_1.Quota.internalBinaryRead(reader, reader.uint32(), options, message.quota);
                    break;
                case /* google.api.Authentication authentication */ 11:
                    message.authentication = auth_1.Authentication.internalBinaryRead(reader, reader.uint32(), options, message.authentication);
                    break;
                case /* google.api.Usage usage */ 15:
                    message.usage = usage_1.Usage.internalBinaryRead(reader, reader.uint32(), options, message.usage);
                    break;
                case /* repeated google.api.Endpoint endpoints */ 18:
                    message.endpoints.push(endpoint_1.Endpoint.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.api.MonitoredResourceDescriptor monitored_resources */ 25:
                    message.monitoredResources.push(monitored_resource_1.MonitoredResourceDescriptor.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.api.Monitoring monitoring */ 28:
                    message.monitoring = monitoring_1.Monitoring.internalBinaryRead(reader, reader.uint32(), options, message.monitoring);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* string title = 2; */
        if (message.title !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.title);
        /* repeated google.protobuf.Api apis = 3; */
        for (let i = 0; i < message.apis.length; i++)
            api_1.Api.internalBinaryWrite(message.apis[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.api.Documentation documentation = 6; */
        if (message.documentation)
            documentation_1.Documentation.internalBinaryWrite(message.documentation, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.api.Quota quota = 10; */
        if (message.quota)
            quota_1.Quota.internalBinaryWrite(message.quota, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.api.Authentication authentication = 11; */
        if (message.authentication)
            auth_1.Authentication.internalBinaryWrite(message.authentication, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.api.Usage usage = 15; */
        if (message.usage)
            usage_1.Usage.internalBinaryWrite(message.usage, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.api.Endpoint endpoints = 18; */
        for (let i = 0; i < message.endpoints.length; i++)
            endpoint_1.Endpoint.internalBinaryWrite(message.endpoints[i], writer.tag(18, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.api.MonitoredResourceDescriptor monitored_resources = 25; */
        for (let i = 0; i < message.monitoredResources.length; i++)
            monitored_resource_1.MonitoredResourceDescriptor.internalBinaryWrite(message.monitoredResources[i], writer.tag(25, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.api.Monitoring monitoring = 28; */
        if (message.monitoring)
            monitoring_1.Monitoring.internalBinaryWrite(message.monitoring, writer.tag(28, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1.ServiceConfig
 */
exports.ServiceConfig = new ServiceConfig$Type();
// @generated message type with reflection information, may provide speed optimized methods
class OperationMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.serviceusage.v1.OperationMetadata", [
            { no: 2, name: "resource_names", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { resourceNames: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated string resource_names */ 2:
                    message.resourceNames.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated string resource_names = 2; */
        for (let i = 0; i < message.resourceNames.length; i++)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.resourceNames[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.serviceusage.v1.OperationMetadata
 */
exports.OperationMetadata = new OperationMetadata$Type();
