// google-ads-api-client/dist/protos/bindings/ads/errors/access_invitation_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessInvitationErrorEnum = exports.AccessInvitationErrorEnum_AccessInvitationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible AccessInvitation errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AccessInvitationErrorEnum.AccessInvitationError
 */
var AccessInvitationErrorEnum_AccessInvitationError;
(function (AccessInvitationErrorEnum_AccessInvitationError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The email address is invalid for sending an invitation.
     *
     * @generated from protobuf enum value: INVALID_EMAIL_ADDRESS = 2;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["INVALID_EMAIL_ADDRESS"] = 2] = "INVALID_EMAIL_ADDRESS";
    /**
     * Email address already has access to this customer.
     *
     * @generated from protobuf enum value: EMAIL_ADDRESS_ALREADY_HAS_ACCESS = 3;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["EMAIL_ADDRESS_ALREADY_HAS_ACCESS"] = 3] = "EMAIL_ADDRESS_ALREADY_HAS_ACCESS";
    /**
     * Invalid invitation status for the operation.
     *
     * @generated from protobuf enum value: INVALID_INVITATION_STATUS = 4;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["INVALID_INVITATION_STATUS"] = 4] = "INVALID_INVITATION_STATUS";
    /**
     * Email address cannot be like abc+foo@google.com.
     *
     * @generated from protobuf enum value: GOOGLE_CONSUMER_ACCOUNT_NOT_ALLOWED = 5;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["GOOGLE_CONSUMER_ACCOUNT_NOT_ALLOWED"] = 5] = "GOOGLE_CONSUMER_ACCOUNT_NOT_ALLOWED";
    /**
     * Invalid invitation ID.
     *
     * @generated from protobuf enum value: INVALID_INVITATION_ID = 6;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["INVALID_INVITATION_ID"] = 6] = "INVALID_INVITATION_ID";
    /**
     * Email address already has a pending invitation.
     *
     * @generated from protobuf enum value: EMAIL_ADDRESS_ALREADY_HAS_PENDING_INVITATION = 7;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["EMAIL_ADDRESS_ALREADY_HAS_PENDING_INVITATION"] = 7] = "EMAIL_ADDRESS_ALREADY_HAS_PENDING_INVITATION";
    /**
     * Pending invitation limit exceeded for the customer.
     *
     * @generated from protobuf enum value: PENDING_INVITATIONS_LIMIT_EXCEEDED = 8;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["PENDING_INVITATIONS_LIMIT_EXCEEDED"] = 8] = "PENDING_INVITATIONS_LIMIT_EXCEEDED";
    /**
     * Email address doesn't conform to the email domain policy. See
     * https://support.google.com/google-ads/answer/2375456
     *
     * @generated from protobuf enum value: EMAIL_DOMAIN_POLICY_VIOLATED = 9;
     */
    AccessInvitationErrorEnum_AccessInvitationError[AccessInvitationErrorEnum_AccessInvitationError["EMAIL_DOMAIN_POLICY_VIOLATED"] = 9] = "EMAIL_DOMAIN_POLICY_VIOLATED";
})(AccessInvitationErrorEnum_AccessInvitationError = exports.AccessInvitationErrorEnum_AccessInvitationError || (exports.AccessInvitationErrorEnum_AccessInvitationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccessInvitationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AccessInvitationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AccessInvitationErrorEnum
 */
exports.AccessInvitationErrorEnum = new AccessInvitationErrorEnum$Type();
