// google-ads-api-client/dist/protos/bindings/ads/errors/account_budget_proposal_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountBudgetProposalErrorEnum = exports.AccountBudgetProposalErrorEnum_AccountBudgetProposalError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible account budget proposal errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AccountBudgetProposalErrorEnum.AccountBudgetProposalError
 */
var AccountBudgetProposalErrorEnum_AccountBudgetProposalError;
(function (AccountBudgetProposalErrorEnum_AccountBudgetProposalError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The field mask must be empty for create/end/remove proposals.
     *
     * @generated from protobuf enum value: FIELD_MASK_NOT_ALLOWED = 2;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["FIELD_MASK_NOT_ALLOWED"] = 2] = "FIELD_MASK_NOT_ALLOWED";
    /**
     * The field cannot be set because of the proposal type.
     *
     * @generated from protobuf enum value: IMMUTABLE_FIELD = 3;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["IMMUTABLE_FIELD"] = 3] = "IMMUTABLE_FIELD";
    /**
     * The field is required because of the proposal type.
     *
     * @generated from protobuf enum value: REQUIRED_FIELD_MISSING = 4;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["REQUIRED_FIELD_MISSING"] = 4] = "REQUIRED_FIELD_MISSING";
    /**
     * Proposals that have been approved cannot be cancelled.
     *
     * @generated from protobuf enum value: CANNOT_CANCEL_APPROVED_PROPOSAL = 5;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_CANCEL_APPROVED_PROPOSAL"] = 5] = "CANNOT_CANCEL_APPROVED_PROPOSAL";
    /**
     * Budgets that haven't been approved cannot be removed.
     *
     * @generated from protobuf enum value: CANNOT_REMOVE_UNAPPROVED_BUDGET = 6;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_REMOVE_UNAPPROVED_BUDGET"] = 6] = "CANNOT_REMOVE_UNAPPROVED_BUDGET";
    /**
     * Budgets that are currently running cannot be removed.
     *
     * @generated from protobuf enum value: CANNOT_REMOVE_RUNNING_BUDGET = 7;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_REMOVE_RUNNING_BUDGET"] = 7] = "CANNOT_REMOVE_RUNNING_BUDGET";
    /**
     * Budgets that haven't been approved cannot be truncated.
     *
     * @generated from protobuf enum value: CANNOT_END_UNAPPROVED_BUDGET = 8;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_END_UNAPPROVED_BUDGET"] = 8] = "CANNOT_END_UNAPPROVED_BUDGET";
    /**
     * Only budgets that are currently running can be truncated.
     *
     * @generated from protobuf enum value: CANNOT_END_INACTIVE_BUDGET = 9;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_END_INACTIVE_BUDGET"] = 9] = "CANNOT_END_INACTIVE_BUDGET";
    /**
     * All budgets must have names.
     *
     * @generated from protobuf enum value: BUDGET_NAME_REQUIRED = 10;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["BUDGET_NAME_REQUIRED"] = 10] = "BUDGET_NAME_REQUIRED";
    /**
     * Expired budgets cannot be edited after a sufficient amount of time has
     * passed.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_OLD_BUDGET = 11;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_UPDATE_OLD_BUDGET"] = 11] = "CANNOT_UPDATE_OLD_BUDGET";
    /**
     * It is not permissible a propose a new budget that ends in the past.
     *
     * @generated from protobuf enum value: CANNOT_END_IN_PAST = 12;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_END_IN_PAST"] = 12] = "CANNOT_END_IN_PAST";
    /**
     * An expired budget cannot be extended to overlap with the running budget.
     *
     * @generated from protobuf enum value: CANNOT_EXTEND_END_TIME = 13;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_EXTEND_END_TIME"] = 13] = "CANNOT_EXTEND_END_TIME";
    /**
     * A purchase order number is required.
     *
     * @generated from protobuf enum value: PURCHASE_ORDER_NUMBER_REQUIRED = 14;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["PURCHASE_ORDER_NUMBER_REQUIRED"] = 14] = "PURCHASE_ORDER_NUMBER_REQUIRED";
    /**
     * Budgets that have a pending update cannot be updated.
     *
     * @generated from protobuf enum value: PENDING_UPDATE_PROPOSAL_EXISTS = 15;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["PENDING_UPDATE_PROPOSAL_EXISTS"] = 15] = "PENDING_UPDATE_PROPOSAL_EXISTS";
    /**
     * Cannot propose more than one budget when the corresponding billing setup
     * hasn't been approved.
     *
     * @generated from protobuf enum value: MULTIPLE_BUDGETS_NOT_ALLOWED_FOR_UNAPPROVED_BILLING_SETUP = 16;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["MULTIPLE_BUDGETS_NOT_ALLOWED_FOR_UNAPPROVED_BILLING_SETUP"] = 16] = "MULTIPLE_BUDGETS_NOT_ALLOWED_FOR_UNAPPROVED_BILLING_SETUP";
    /**
     * Cannot update the start time of a budget that has already started.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_START_TIME_FOR_STARTED_BUDGET = 17;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_UPDATE_START_TIME_FOR_STARTED_BUDGET"] = 17] = "CANNOT_UPDATE_START_TIME_FOR_STARTED_BUDGET";
    /**
     * Cannot update the spending limit of a budget with an amount lower than
     * what has already been spent.
     *
     * @generated from protobuf enum value: SPENDING_LIMIT_LOWER_THAN_ACCRUED_COST_NOT_ALLOWED = 18;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["SPENDING_LIMIT_LOWER_THAN_ACCRUED_COST_NOT_ALLOWED"] = 18] = "SPENDING_LIMIT_LOWER_THAN_ACCRUED_COST_NOT_ALLOWED";
    /**
     * Cannot propose a budget update without actually changing any fields.
     *
     * @generated from protobuf enum value: UPDATE_IS_NO_OP = 19;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["UPDATE_IS_NO_OP"] = 19] = "UPDATE_IS_NO_OP";
    /**
     * The end time must come after the start time.
     *
     * @generated from protobuf enum value: END_TIME_MUST_FOLLOW_START_TIME = 20;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["END_TIME_MUST_FOLLOW_START_TIME"] = 20] = "END_TIME_MUST_FOLLOW_START_TIME";
    /**
     * The budget's date range must fall within the date range of its billing
     * setup.
     *
     * @generated from protobuf enum value: BUDGET_DATE_RANGE_INCOMPATIBLE_WITH_BILLING_SETUP = 21;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["BUDGET_DATE_RANGE_INCOMPATIBLE_WITH_BILLING_SETUP"] = 21] = "BUDGET_DATE_RANGE_INCOMPATIBLE_WITH_BILLING_SETUP";
    /**
     * The user is not authorized to mutate budgets for the given billing setup.
     *
     * @generated from protobuf enum value: NOT_AUTHORIZED = 22;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["NOT_AUTHORIZED"] = 22] = "NOT_AUTHORIZED";
    /**
     * Mutates are not allowed for the given billing setup.
     *
     * @generated from protobuf enum value: INVALID_BILLING_SETUP = 23;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["INVALID_BILLING_SETUP"] = 23] = "INVALID_BILLING_SETUP";
    /**
     * Budget creation failed as it overlaps with an pending budget proposal
     * or an approved budget.
     *
     * @generated from protobuf enum value: OVERLAPS_EXISTING_BUDGET = 24;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["OVERLAPS_EXISTING_BUDGET"] = 24] = "OVERLAPS_EXISTING_BUDGET";
    /**
     * The control setting in user's payments profile doesn't allow budget
     * creation through API. Log in to Google Ads to create budget.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_BUDGET_THROUGH_API = 25;
     */
    AccountBudgetProposalErrorEnum_AccountBudgetProposalError[AccountBudgetProposalErrorEnum_AccountBudgetProposalError["CANNOT_CREATE_BUDGET_THROUGH_API"] = 25] = "CANNOT_CREATE_BUDGET_THROUGH_API";
})(AccountBudgetProposalErrorEnum_AccountBudgetProposalError = exports.AccountBudgetProposalErrorEnum_AccountBudgetProposalError || (exports.AccountBudgetProposalErrorEnum_AccountBudgetProposalError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccountBudgetProposalErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AccountBudgetProposalErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AccountBudgetProposalErrorEnum
 */
exports.AccountBudgetProposalErrorEnum = new AccountBudgetProposalErrorEnum$Type();
