// google-ads-api-client/dist/protos/bindings/ads/errors/account_link_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountLinkErrorEnum = exports.AccountLinkErrorEnum_AccountLinkError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible account link errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AccountLinkErrorEnum.AccountLinkError
 */
var AccountLinkErrorEnum_AccountLinkError;
(function (AccountLinkErrorEnum_AccountLinkError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AccountLinkErrorEnum_AccountLinkError[AccountLinkErrorEnum_AccountLinkError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AccountLinkErrorEnum_AccountLinkError[AccountLinkErrorEnum_AccountLinkError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The new link status is invalid.
     *
     * @generated from protobuf enum value: INVALID_STATUS = 2;
     */
    AccountLinkErrorEnum_AccountLinkError[AccountLinkErrorEnum_AccountLinkError["INVALID_STATUS"] = 2] = "INVALID_STATUS";
})(AccountLinkErrorEnum_AccountLinkError = exports.AccountLinkErrorEnum_AccountLinkError || (exports.AccountLinkErrorEnum_AccountLinkError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AccountLinkErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AccountLinkErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AccountLinkErrorEnum
 */
exports.AccountLinkErrorEnum = new AccountLinkErrorEnum$Type();
