// google-ads-api-client/dist/protos/bindings/ads/errors/ad_customizer_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdCustomizerErrorEnum = exports.AdCustomizerErrorEnum_AdCustomizerError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ad customizer errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdCustomizerErrorEnum.AdCustomizerError
 */
var AdCustomizerErrorEnum_AdCustomizerError;
(function (AdCustomizerErrorEnum_AdCustomizerError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdCustomizerErrorEnum_AdCustomizerError[AdCustomizerErrorEnum_AdCustomizerError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdCustomizerErrorEnum_AdCustomizerError[AdCustomizerErrorEnum_AdCustomizerError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Invalid date argument in countdown function.
     *
     * @generated from protobuf enum value: COUNTDOWN_INVALID_DATE_FORMAT = 2;
     */
    AdCustomizerErrorEnum_AdCustomizerError[AdCustomizerErrorEnum_AdCustomizerError["COUNTDOWN_INVALID_DATE_FORMAT"] = 2] = "COUNTDOWN_INVALID_DATE_FORMAT";
    /**
     * Countdown end date is in the past.
     *
     * @generated from protobuf enum value: COUNTDOWN_DATE_IN_PAST = 3;
     */
    AdCustomizerErrorEnum_AdCustomizerError[AdCustomizerErrorEnum_AdCustomizerError["COUNTDOWN_DATE_IN_PAST"] = 3] = "COUNTDOWN_DATE_IN_PAST";
    /**
     * Invalid locale string in countdown function.
     *
     * @generated from protobuf enum value: COUNTDOWN_INVALID_LOCALE = 4;
     */
    AdCustomizerErrorEnum_AdCustomizerError[AdCustomizerErrorEnum_AdCustomizerError["COUNTDOWN_INVALID_LOCALE"] = 4] = "COUNTDOWN_INVALID_LOCALE";
    /**
     * Days-before argument to countdown function is not positive.
     *
     * @generated from protobuf enum value: COUNTDOWN_INVALID_START_DAYS_BEFORE = 5;
     */
    AdCustomizerErrorEnum_AdCustomizerError[AdCustomizerErrorEnum_AdCustomizerError["COUNTDOWN_INVALID_START_DAYS_BEFORE"] = 5] = "COUNTDOWN_INVALID_START_DAYS_BEFORE";
    /**
     * A user list referenced in an IF function does not exist.
     *
     * @generated from protobuf enum value: UNKNOWN_USER_LIST = 6;
     */
    AdCustomizerErrorEnum_AdCustomizerError[AdCustomizerErrorEnum_AdCustomizerError["UNKNOWN_USER_LIST"] = 6] = "UNKNOWN_USER_LIST";
})(AdCustomizerErrorEnum_AdCustomizerError = exports.AdCustomizerErrorEnum_AdCustomizerError || (exports.AdCustomizerErrorEnum_AdCustomizerError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdCustomizerErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdCustomizerErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdCustomizerErrorEnum
 */
exports.AdCustomizerErrorEnum = new AdCustomizerErrorEnum$Type();
