// google-ads-api-client/dist/protos/bindings/ads/errors/ad_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdErrorEnum = exports.AdErrorEnum_AdError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ad errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdErrorEnum.AdError
 */
var AdErrorEnum_AdError;
(function (AdErrorEnum_AdError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Ad customizers are not supported for ad type.
     *
     * @generated from protobuf enum value: AD_CUSTOMIZERS_NOT_SUPPORTED_FOR_AD_TYPE = 2;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["AD_CUSTOMIZERS_NOT_SUPPORTED_FOR_AD_TYPE"] = 2] = "AD_CUSTOMIZERS_NOT_SUPPORTED_FOR_AD_TYPE";
    /**
     * Estimating character sizes the string is too long.
     *
     * @generated from protobuf enum value: APPROXIMATELY_TOO_LONG = 3;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["APPROXIMATELY_TOO_LONG"] = 3] = "APPROXIMATELY_TOO_LONG";
    /**
     * Estimating character sizes the string is too short.
     *
     * @generated from protobuf enum value: APPROXIMATELY_TOO_SHORT = 4;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["APPROXIMATELY_TOO_SHORT"] = 4] = "APPROXIMATELY_TOO_SHORT";
    /**
     * There is a problem with the snippet.
     *
     * @generated from protobuf enum value: BAD_SNIPPET = 5;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["BAD_SNIPPET"] = 5] = "BAD_SNIPPET";
    /**
     * Cannot modify an ad.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_AD = 6;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_MODIFY_AD"] = 6] = "CANNOT_MODIFY_AD";
    /**
     * business name and url cannot be set at the same time
     *
     * @generated from protobuf enum value: CANNOT_SET_BUSINESS_NAME_IF_URL_SET = 7;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_BUSINESS_NAME_IF_URL_SET"] = 7] = "CANNOT_SET_BUSINESS_NAME_IF_URL_SET";
    /**
     * The specified field is incompatible with this ad's type or settings.
     *
     * @generated from protobuf enum value: CANNOT_SET_FIELD = 8;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_FIELD"] = 8] = "CANNOT_SET_FIELD";
    /**
     * Cannot set field when originAdId is set.
     *
     * @generated from protobuf enum value: CANNOT_SET_FIELD_WITH_ORIGIN_AD_ID_SET = 9;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_FIELD_WITH_ORIGIN_AD_ID_SET"] = 9] = "CANNOT_SET_FIELD_WITH_ORIGIN_AD_ID_SET";
    /**
     * Cannot set field when an existing ad id is set for sharing.
     *
     * @generated from protobuf enum value: CANNOT_SET_FIELD_WITH_AD_ID_SET_FOR_SHARING = 10;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_FIELD_WITH_AD_ID_SET_FOR_SHARING"] = 10] = "CANNOT_SET_FIELD_WITH_AD_ID_SET_FOR_SHARING";
    /**
     * Cannot set allowFlexibleColor false if no color is provided by user.
     *
     * @generated from protobuf enum value: CANNOT_SET_ALLOW_FLEXIBLE_COLOR_FALSE = 11;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_ALLOW_FLEXIBLE_COLOR_FALSE"] = 11] = "CANNOT_SET_ALLOW_FLEXIBLE_COLOR_FALSE";
    /**
     * When user select native, no color control is allowed because we will
     * always respect publisher color for native format serving.
     *
     * @generated from protobuf enum value: CANNOT_SET_COLOR_CONTROL_WHEN_NATIVE_FORMAT_SETTING = 12;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_COLOR_CONTROL_WHEN_NATIVE_FORMAT_SETTING"] = 12] = "CANNOT_SET_COLOR_CONTROL_WHEN_NATIVE_FORMAT_SETTING";
    /**
     * Cannot specify a url for the ad type
     *
     * @generated from protobuf enum value: CANNOT_SET_URL = 13;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_URL"] = 13] = "CANNOT_SET_URL";
    /**
     * Cannot specify a tracking or mobile url without also setting final urls
     *
     * @generated from protobuf enum value: CANNOT_SET_WITHOUT_FINAL_URLS = 14;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_WITHOUT_FINAL_URLS"] = 14] = "CANNOT_SET_WITHOUT_FINAL_URLS";
    /**
     * Cannot specify a legacy url and a final url simultaneously
     *
     * @generated from protobuf enum value: CANNOT_SET_WITH_FINAL_URLS = 15;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_WITH_FINAL_URLS"] = 15] = "CANNOT_SET_WITH_FINAL_URLS";
    /**
     * Cannot specify a urls in UrlData and in template fields simultaneously.
     *
     * @generated from protobuf enum value: CANNOT_SET_WITH_URL_DATA = 17;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_WITH_URL_DATA"] = 17] = "CANNOT_SET_WITH_URL_DATA";
    /**
     * This operator cannot be used with a subclass of Ad.
     *
     * @generated from protobuf enum value: CANNOT_USE_AD_SUBCLASS_FOR_OPERATOR = 18;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_USE_AD_SUBCLASS_FOR_OPERATOR"] = 18] = "CANNOT_USE_AD_SUBCLASS_FOR_OPERATOR";
    /**
     * Customer is not approved for mobile ads.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_APPROVED_MOBILEADS = 19;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CUSTOMER_NOT_APPROVED_MOBILEADS"] = 19] = "CUSTOMER_NOT_APPROVED_MOBILEADS";
    /**
     * Customer is not approved for 3PAS richmedia ads.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_APPROVED_THIRDPARTY_ADS = 20;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CUSTOMER_NOT_APPROVED_THIRDPARTY_ADS"] = 20] = "CUSTOMER_NOT_APPROVED_THIRDPARTY_ADS";
    /**
     * Customer is not approved for 3PAS redirect richmedia (Ad Exchange) ads.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_APPROVED_THIRDPARTY_REDIRECT_ADS = 21;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CUSTOMER_NOT_APPROVED_THIRDPARTY_REDIRECT_ADS"] = 21] = "CUSTOMER_NOT_APPROVED_THIRDPARTY_REDIRECT_ADS";
    /**
     * Not an eligible customer
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_ELIGIBLE = 22;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CUSTOMER_NOT_ELIGIBLE"] = 22] = "CUSTOMER_NOT_ELIGIBLE";
    /**
     * Customer is not eligible for updating beacon url
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_ELIGIBLE_FOR_UPDATING_BEACON_URL = 23;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CUSTOMER_NOT_ELIGIBLE_FOR_UPDATING_BEACON_URL"] = 23] = "CUSTOMER_NOT_ELIGIBLE_FOR_UPDATING_BEACON_URL";
    /**
     * There already exists an ad with the same dimensions in the union.
     *
     * @generated from protobuf enum value: DIMENSION_ALREADY_IN_UNION = 24;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DIMENSION_ALREADY_IN_UNION"] = 24] = "DIMENSION_ALREADY_IN_UNION";
    /**
     * Ad's dimension must be set before setting union dimension.
     *
     * @generated from protobuf enum value: DIMENSION_MUST_BE_SET = 25;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DIMENSION_MUST_BE_SET"] = 25] = "DIMENSION_MUST_BE_SET";
    /**
     * Ad's dimension must be included in the union dimensions.
     *
     * @generated from protobuf enum value: DIMENSION_NOT_IN_UNION = 26;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DIMENSION_NOT_IN_UNION"] = 26] = "DIMENSION_NOT_IN_UNION";
    /**
     * Display Url cannot be specified (applies to Ad Exchange Ads)
     *
     * @generated from protobuf enum value: DISPLAY_URL_CANNOT_BE_SPECIFIED = 27;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DISPLAY_URL_CANNOT_BE_SPECIFIED"] = 27] = "DISPLAY_URL_CANNOT_BE_SPECIFIED";
    /**
     * Telephone number contains invalid characters or invalid format.
     * Re-enter your number using digits (0-9), dashes (-), and parentheses
     * only.
     *
     * @generated from protobuf enum value: DOMESTIC_PHONE_NUMBER_FORMAT = 28;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DOMESTIC_PHONE_NUMBER_FORMAT"] = 28] = "DOMESTIC_PHONE_NUMBER_FORMAT";
    /**
     * Emergency telephone numbers are not allowed. Enter a valid
     * domestic phone number to connect customers to your business.
     *
     * @generated from protobuf enum value: EMERGENCY_PHONE_NUMBER = 29;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["EMERGENCY_PHONE_NUMBER"] = 29] = "EMERGENCY_PHONE_NUMBER";
    /**
     * A required field was not specified or is an empty string.
     *
     * @generated from protobuf enum value: EMPTY_FIELD = 30;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["EMPTY_FIELD"] = 30] = "EMPTY_FIELD";
    /**
     * A feed attribute referenced in an ad customizer tag is not in the ad
     * customizer mapping for the feed.
     *
     * @generated from protobuf enum value: FEED_ATTRIBUTE_MUST_HAVE_MAPPING_FOR_TYPE_ID = 31;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["FEED_ATTRIBUTE_MUST_HAVE_MAPPING_FOR_TYPE_ID"] = 31] = "FEED_ATTRIBUTE_MUST_HAVE_MAPPING_FOR_TYPE_ID";
    /**
     * The ad customizer field mapping for the feed attribute does not match the
     * expected field type.
     *
     * @generated from protobuf enum value: FEED_ATTRIBUTE_MAPPING_TYPE_MISMATCH = 32;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["FEED_ATTRIBUTE_MAPPING_TYPE_MISMATCH"] = 32] = "FEED_ATTRIBUTE_MAPPING_TYPE_MISMATCH";
    /**
     * The use of ad customizer tags in the ad text is disallowed. Details in
     * trigger.
     *
     * @generated from protobuf enum value: ILLEGAL_AD_CUSTOMIZER_TAG_USE = 33;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["ILLEGAL_AD_CUSTOMIZER_TAG_USE"] = 33] = "ILLEGAL_AD_CUSTOMIZER_TAG_USE";
    /**
     * Tags of the form {PH_x}, where x is a number, are disallowed in ad text.
     *
     * @generated from protobuf enum value: ILLEGAL_TAG_USE = 34;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["ILLEGAL_TAG_USE"] = 34] = "ILLEGAL_TAG_USE";
    /**
     * The dimensions of the ad are specified or derived in multiple ways and
     * are not consistent.
     *
     * @generated from protobuf enum value: INCONSISTENT_DIMENSIONS = 35;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INCONSISTENT_DIMENSIONS"] = 35] = "INCONSISTENT_DIMENSIONS";
    /**
     * The status cannot differ among template ads of the same union.
     *
     * @generated from protobuf enum value: INCONSISTENT_STATUS_IN_TEMPLATE_UNION = 36;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INCONSISTENT_STATUS_IN_TEMPLATE_UNION"] = 36] = "INCONSISTENT_STATUS_IN_TEMPLATE_UNION";
    /**
     * The length of the string is not valid.
     *
     * @generated from protobuf enum value: INCORRECT_LENGTH = 37;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INCORRECT_LENGTH"] = 37] = "INCORRECT_LENGTH";
    /**
     * The ad is ineligible for upgrade.
     *
     * @generated from protobuf enum value: INELIGIBLE_FOR_UPGRADE = 38;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INELIGIBLE_FOR_UPGRADE"] = 38] = "INELIGIBLE_FOR_UPGRADE";
    /**
     * User cannot create mobile ad for countries targeted in specified
     * campaign.
     *
     * @generated from protobuf enum value: INVALID_AD_ADDRESS_CAMPAIGN_TARGET = 39;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_AD_ADDRESS_CAMPAIGN_TARGET"] = 39] = "INVALID_AD_ADDRESS_CAMPAIGN_TARGET";
    /**
     * Invalid Ad type. A specific type of Ad is required.
     *
     * @generated from protobuf enum value: INVALID_AD_TYPE = 40;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_AD_TYPE"] = 40] = "INVALID_AD_TYPE";
    /**
     * Headline, description or phone cannot be present when creating mobile
     * image ad.
     *
     * @generated from protobuf enum value: INVALID_ATTRIBUTES_FOR_MOBILE_IMAGE = 41;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_ATTRIBUTES_FOR_MOBILE_IMAGE"] = 41] = "INVALID_ATTRIBUTES_FOR_MOBILE_IMAGE";
    /**
     * Image cannot be present when creating mobile text ad.
     *
     * @generated from protobuf enum value: INVALID_ATTRIBUTES_FOR_MOBILE_TEXT = 42;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_ATTRIBUTES_FOR_MOBILE_TEXT"] = 42] = "INVALID_ATTRIBUTES_FOR_MOBILE_TEXT";
    /**
     * Invalid call to action text.
     *
     * @generated from protobuf enum value: INVALID_CALL_TO_ACTION_TEXT = 43;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_CALL_TO_ACTION_TEXT"] = 43] = "INVALID_CALL_TO_ACTION_TEXT";
    /**
     * Invalid character in URL.
     *
     * @generated from protobuf enum value: INVALID_CHARACTER_FOR_URL = 44;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_CHARACTER_FOR_URL"] = 44] = "INVALID_CHARACTER_FOR_URL";
    /**
     * Creative's country code is not valid.
     *
     * @generated from protobuf enum value: INVALID_COUNTRY_CODE = 45;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_COUNTRY_CODE"] = 45] = "INVALID_COUNTRY_CODE";
    /**
     * Invalid use of Expanded Dynamic Search Ads tags ({lpurl} etc.)
     *
     * @generated from protobuf enum value: INVALID_EXPANDED_DYNAMIC_SEARCH_AD_TAG = 47;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_EXPANDED_DYNAMIC_SEARCH_AD_TAG"] = 47] = "INVALID_EXPANDED_DYNAMIC_SEARCH_AD_TAG";
    /**
     * An input error whose real reason was not properly mapped (should not
     * happen).
     *
     * @generated from protobuf enum value: INVALID_INPUT = 48;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_INPUT"] = 48] = "INVALID_INPUT";
    /**
     * An invalid markup language was entered.
     *
     * @generated from protobuf enum value: INVALID_MARKUP_LANGUAGE = 49;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_MARKUP_LANGUAGE"] = 49] = "INVALID_MARKUP_LANGUAGE";
    /**
     * An invalid mobile carrier was entered.
     *
     * @generated from protobuf enum value: INVALID_MOBILE_CARRIER = 50;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_MOBILE_CARRIER"] = 50] = "INVALID_MOBILE_CARRIER";
    /**
     * Specified mobile carriers target a country not targeted by the campaign.
     *
     * @generated from protobuf enum value: INVALID_MOBILE_CARRIER_TARGET = 51;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_MOBILE_CARRIER_TARGET"] = 51] = "INVALID_MOBILE_CARRIER_TARGET";
    /**
     * Wrong number of elements for given element type
     *
     * @generated from protobuf enum value: INVALID_NUMBER_OF_ELEMENTS = 52;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_NUMBER_OF_ELEMENTS"] = 52] = "INVALID_NUMBER_OF_ELEMENTS";
    /**
     * The format of the telephone number is incorrect. Re-enter the
     * number using the correct format.
     *
     * @generated from protobuf enum value: INVALID_PHONE_NUMBER_FORMAT = 53;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_PHONE_NUMBER_FORMAT"] = 53] = "INVALID_PHONE_NUMBER_FORMAT";
    /**
     * The certified vendor format id is incorrect.
     *
     * @generated from protobuf enum value: INVALID_RICH_MEDIA_CERTIFIED_VENDOR_FORMAT_ID = 54;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_RICH_MEDIA_CERTIFIED_VENDOR_FORMAT_ID"] = 54] = "INVALID_RICH_MEDIA_CERTIFIED_VENDOR_FORMAT_ID";
    /**
     * The template ad data contains validation errors.
     *
     * @generated from protobuf enum value: INVALID_TEMPLATE_DATA = 55;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_TEMPLATE_DATA"] = 55] = "INVALID_TEMPLATE_DATA";
    /**
     * The template field doesn't have have the correct type.
     *
     * @generated from protobuf enum value: INVALID_TEMPLATE_ELEMENT_FIELD_TYPE = 56;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_TEMPLATE_ELEMENT_FIELD_TYPE"] = 56] = "INVALID_TEMPLATE_ELEMENT_FIELD_TYPE";
    /**
     * Invalid template id.
     *
     * @generated from protobuf enum value: INVALID_TEMPLATE_ID = 57;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_TEMPLATE_ID"] = 57] = "INVALID_TEMPLATE_ID";
    /**
     * After substituting replacement strings, the line is too wide.
     *
     * @generated from protobuf enum value: LINE_TOO_WIDE = 58;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["LINE_TOO_WIDE"] = 58] = "LINE_TOO_WIDE";
    /**
     * The feed referenced must have ad customizer mapping to be used in a
     * customizer tag.
     *
     * @generated from protobuf enum value: MISSING_AD_CUSTOMIZER_MAPPING = 59;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_AD_CUSTOMIZER_MAPPING"] = 59] = "MISSING_AD_CUSTOMIZER_MAPPING";
    /**
     * Missing address component in template element address field.
     *
     * @generated from protobuf enum value: MISSING_ADDRESS_COMPONENT = 60;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_ADDRESS_COMPONENT"] = 60] = "MISSING_ADDRESS_COMPONENT";
    /**
     * An ad name must be entered.
     *
     * @generated from protobuf enum value: MISSING_ADVERTISEMENT_NAME = 61;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_ADVERTISEMENT_NAME"] = 61] = "MISSING_ADVERTISEMENT_NAME";
    /**
     * Business name must be entered.
     *
     * @generated from protobuf enum value: MISSING_BUSINESS_NAME = 62;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_BUSINESS_NAME"] = 62] = "MISSING_BUSINESS_NAME";
    /**
     * Description (line 2) must be entered.
     *
     * @generated from protobuf enum value: MISSING_DESCRIPTION1 = 63;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_DESCRIPTION1"] = 63] = "MISSING_DESCRIPTION1";
    /**
     * Description (line 3) must be entered.
     *
     * @generated from protobuf enum value: MISSING_DESCRIPTION2 = 64;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_DESCRIPTION2"] = 64] = "MISSING_DESCRIPTION2";
    /**
     * The destination url must contain at least one tag (for example, {lpurl})
     *
     * @generated from protobuf enum value: MISSING_DESTINATION_URL_TAG = 65;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_DESTINATION_URL_TAG"] = 65] = "MISSING_DESTINATION_URL_TAG";
    /**
     * The tracking url template of ExpandedDynamicSearchAd must contain at
     * least one tag. (for example, {lpurl})
     *
     * @generated from protobuf enum value: MISSING_LANDING_PAGE_URL_TAG = 66;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_LANDING_PAGE_URL_TAG"] = 66] = "MISSING_LANDING_PAGE_URL_TAG";
    /**
     * A valid dimension must be specified for this ad.
     *
     * @generated from protobuf enum value: MISSING_DIMENSION = 67;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_DIMENSION"] = 67] = "MISSING_DIMENSION";
    /**
     * A display URL must be entered.
     *
     * @generated from protobuf enum value: MISSING_DISPLAY_URL = 68;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_DISPLAY_URL"] = 68] = "MISSING_DISPLAY_URL";
    /**
     * Headline must be entered.
     *
     * @generated from protobuf enum value: MISSING_HEADLINE = 69;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_HEADLINE"] = 69] = "MISSING_HEADLINE";
    /**
     * A height must be entered.
     *
     * @generated from protobuf enum value: MISSING_HEIGHT = 70;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_HEIGHT"] = 70] = "MISSING_HEIGHT";
    /**
     * An image must be entered.
     *
     * @generated from protobuf enum value: MISSING_IMAGE = 71;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_IMAGE"] = 71] = "MISSING_IMAGE";
    /**
     * Marketing image or product videos are required.
     *
     * @generated from protobuf enum value: MISSING_MARKETING_IMAGE_OR_PRODUCT_VIDEOS = 72;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_MARKETING_IMAGE_OR_PRODUCT_VIDEOS"] = 72] = "MISSING_MARKETING_IMAGE_OR_PRODUCT_VIDEOS";
    /**
     * The markup language in which your site is written must be entered.
     *
     * @generated from protobuf enum value: MISSING_MARKUP_LANGUAGES = 73;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_MARKUP_LANGUAGES"] = 73] = "MISSING_MARKUP_LANGUAGES";
    /**
     * A mobile carrier must be entered.
     *
     * @generated from protobuf enum value: MISSING_MOBILE_CARRIER = 74;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_MOBILE_CARRIER"] = 74] = "MISSING_MOBILE_CARRIER";
    /**
     * Phone number must be entered.
     *
     * @generated from protobuf enum value: MISSING_PHONE = 75;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_PHONE"] = 75] = "MISSING_PHONE";
    /**
     * Missing required template fields
     *
     * @generated from protobuf enum value: MISSING_REQUIRED_TEMPLATE_FIELDS = 76;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_REQUIRED_TEMPLATE_FIELDS"] = 76] = "MISSING_REQUIRED_TEMPLATE_FIELDS";
    /**
     * Missing a required field value
     *
     * @generated from protobuf enum value: MISSING_TEMPLATE_FIELD_VALUE = 77;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_TEMPLATE_FIELD_VALUE"] = 77] = "MISSING_TEMPLATE_FIELD_VALUE";
    /**
     * The ad must have text.
     *
     * @generated from protobuf enum value: MISSING_TEXT = 78;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_TEXT"] = 78] = "MISSING_TEXT";
    /**
     * A visible URL must be entered.
     *
     * @generated from protobuf enum value: MISSING_VISIBLE_URL = 79;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_VISIBLE_URL"] = 79] = "MISSING_VISIBLE_URL";
    /**
     * A width must be entered.
     *
     * @generated from protobuf enum value: MISSING_WIDTH = 80;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_WIDTH"] = 80] = "MISSING_WIDTH";
    /**
     * Only 1 feed can be used as the source of ad customizer substitutions in a
     * single ad.
     *
     * @generated from protobuf enum value: MULTIPLE_DISTINCT_FEEDS_UNSUPPORTED = 81;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MULTIPLE_DISTINCT_FEEDS_UNSUPPORTED"] = 81] = "MULTIPLE_DISTINCT_FEEDS_UNSUPPORTED";
    /**
     * TempAdUnionId must be use when adding template ads.
     *
     * @generated from protobuf enum value: MUST_USE_TEMP_AD_UNION_ID_ON_ADD = 82;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MUST_USE_TEMP_AD_UNION_ID_ON_ADD"] = 82] = "MUST_USE_TEMP_AD_UNION_ID_ON_ADD";
    /**
     * The string has too many characters.
     *
     * @generated from protobuf enum value: TOO_LONG = 83;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["TOO_LONG"] = 83] = "TOO_LONG";
    /**
     * The string has too few characters.
     *
     * @generated from protobuf enum value: TOO_SHORT = 84;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["TOO_SHORT"] = 84] = "TOO_SHORT";
    /**
     * Ad union dimensions cannot change for saved ads.
     *
     * @generated from protobuf enum value: UNION_DIMENSIONS_CANNOT_CHANGE = 85;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNION_DIMENSIONS_CANNOT_CHANGE"] = 85] = "UNION_DIMENSIONS_CANNOT_CHANGE";
    /**
     * Address component is not {country, lat, lng}.
     *
     * @generated from protobuf enum value: UNKNOWN_ADDRESS_COMPONENT = 86;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNKNOWN_ADDRESS_COMPONENT"] = 86] = "UNKNOWN_ADDRESS_COMPONENT";
    /**
     * Unknown unique field name
     *
     * @generated from protobuf enum value: UNKNOWN_FIELD_NAME = 87;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNKNOWN_FIELD_NAME"] = 87] = "UNKNOWN_FIELD_NAME";
    /**
     * Unknown unique name (template element type specifier)
     *
     * @generated from protobuf enum value: UNKNOWN_UNIQUE_NAME = 88;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNKNOWN_UNIQUE_NAME"] = 88] = "UNKNOWN_UNIQUE_NAME";
    /**
     * Unsupported ad dimension
     *
     * @generated from protobuf enum value: UNSUPPORTED_DIMENSIONS = 89;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNSUPPORTED_DIMENSIONS"] = 89] = "UNSUPPORTED_DIMENSIONS";
    /**
     * URL starts with an invalid scheme.
     *
     * @generated from protobuf enum value: URL_INVALID_SCHEME = 90;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_INVALID_SCHEME"] = 90] = "URL_INVALID_SCHEME";
    /**
     * URL ends with an invalid top-level domain name.
     *
     * @generated from protobuf enum value: URL_INVALID_TOP_LEVEL_DOMAIN = 91;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_INVALID_TOP_LEVEL_DOMAIN"] = 91] = "URL_INVALID_TOP_LEVEL_DOMAIN";
    /**
     * URL contains illegal characters.
     *
     * @generated from protobuf enum value: URL_MALFORMED = 92;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_MALFORMED"] = 92] = "URL_MALFORMED";
    /**
     * URL must contain a host name.
     *
     * @generated from protobuf enum value: URL_NO_HOST = 93;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_NO_HOST"] = 93] = "URL_NO_HOST";
    /**
     * URL not equivalent during upgrade.
     *
     * @generated from protobuf enum value: URL_NOT_EQUIVALENT = 94;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_NOT_EQUIVALENT"] = 94] = "URL_NOT_EQUIVALENT";
    /**
     * URL host name too long to be stored as visible URL (applies to Ad
     * Exchange ads)
     *
     * @generated from protobuf enum value: URL_HOST_NAME_TOO_LONG = 95;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_HOST_NAME_TOO_LONG"] = 95] = "URL_HOST_NAME_TOO_LONG";
    /**
     * URL must start with a scheme.
     *
     * @generated from protobuf enum value: URL_NO_SCHEME = 96;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_NO_SCHEME"] = 96] = "URL_NO_SCHEME";
    /**
     * URL should end in a valid domain extension, such as .com or .net.
     *
     * @generated from protobuf enum value: URL_NO_TOP_LEVEL_DOMAIN = 97;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_NO_TOP_LEVEL_DOMAIN"] = 97] = "URL_NO_TOP_LEVEL_DOMAIN";
    /**
     * URL must not end with a path.
     *
     * @generated from protobuf enum value: URL_PATH_NOT_ALLOWED = 98;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_PATH_NOT_ALLOWED"] = 98] = "URL_PATH_NOT_ALLOWED";
    /**
     * URL must not specify a port.
     *
     * @generated from protobuf enum value: URL_PORT_NOT_ALLOWED = 99;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_PORT_NOT_ALLOWED"] = 99] = "URL_PORT_NOT_ALLOWED";
    /**
     * URL must not contain a query.
     *
     * @generated from protobuf enum value: URL_QUERY_NOT_ALLOWED = 100;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_QUERY_NOT_ALLOWED"] = 100] = "URL_QUERY_NOT_ALLOWED";
    /**
     * A url scheme is not allowed in front of tag in tracking url template
     * (for example, http://{lpurl})
     *
     * @generated from protobuf enum value: URL_SCHEME_BEFORE_EXPANDED_DYNAMIC_SEARCH_AD_TAG = 102;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["URL_SCHEME_BEFORE_EXPANDED_DYNAMIC_SEARCH_AD_TAG"] = 102] = "URL_SCHEME_BEFORE_EXPANDED_DYNAMIC_SEARCH_AD_TAG";
    /**
     * The user does not have permissions to create a template ad for the given
     * template.
     *
     * @generated from protobuf enum value: USER_DOES_NOT_HAVE_ACCESS_TO_TEMPLATE = 103;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["USER_DOES_NOT_HAVE_ACCESS_TO_TEMPLATE"] = 103] = "USER_DOES_NOT_HAVE_ACCESS_TO_TEMPLATE";
    /**
     * Expandable setting is inconsistent/wrong. For example, an AdX ad is
     * invalid if it has a expandable vendor format but no expanding directions
     * specified, or expanding directions is specified, but the vendor format is
     * not expandable.
     *
     * @generated from protobuf enum value: INCONSISTENT_EXPANDABLE_SETTINGS = 104;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INCONSISTENT_EXPANDABLE_SETTINGS"] = 104] = "INCONSISTENT_EXPANDABLE_SETTINGS";
    /**
     * Format is invalid
     *
     * @generated from protobuf enum value: INVALID_FORMAT = 105;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_FORMAT"] = 105] = "INVALID_FORMAT";
    /**
     * The text of this field did not match a pattern of allowed values.
     *
     * @generated from protobuf enum value: INVALID_FIELD_TEXT = 106;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_FIELD_TEXT"] = 106] = "INVALID_FIELD_TEXT";
    /**
     * Template element is mising
     *
     * @generated from protobuf enum value: ELEMENT_NOT_PRESENT = 107;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["ELEMENT_NOT_PRESENT"] = 107] = "ELEMENT_NOT_PRESENT";
    /**
     * Error occurred during image processing
     *
     * @generated from protobuf enum value: IMAGE_ERROR = 108;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["IMAGE_ERROR"] = 108] = "IMAGE_ERROR";
    /**
     * The value is not within the valid range
     *
     * @generated from protobuf enum value: VALUE_NOT_IN_RANGE = 109;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["VALUE_NOT_IN_RANGE"] = 109] = "VALUE_NOT_IN_RANGE";
    /**
     * Template element field is not present
     *
     * @generated from protobuf enum value: FIELD_NOT_PRESENT = 110;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["FIELD_NOT_PRESENT"] = 110] = "FIELD_NOT_PRESENT";
    /**
     * Address is incomplete
     *
     * @generated from protobuf enum value: ADDRESS_NOT_COMPLETE = 111;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["ADDRESS_NOT_COMPLETE"] = 111] = "ADDRESS_NOT_COMPLETE";
    /**
     * Invalid address
     *
     * @generated from protobuf enum value: ADDRESS_INVALID = 112;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["ADDRESS_INVALID"] = 112] = "ADDRESS_INVALID";
    /**
     * Error retrieving specified video
     *
     * @generated from protobuf enum value: VIDEO_RETRIEVAL_ERROR = 113;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["VIDEO_RETRIEVAL_ERROR"] = 113] = "VIDEO_RETRIEVAL_ERROR";
    /**
     * Error processing audio
     *
     * @generated from protobuf enum value: AUDIO_ERROR = 114;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["AUDIO_ERROR"] = 114] = "AUDIO_ERROR";
    /**
     * Display URL is incorrect for YouTube PYV ads
     *
     * @generated from protobuf enum value: INVALID_YOUTUBE_DISPLAY_URL = 115;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_YOUTUBE_DISPLAY_URL"] = 115] = "INVALID_YOUTUBE_DISPLAY_URL";
    /**
     * Too many product Images in GmailAd
     *
     * @generated from protobuf enum value: TOO_MANY_PRODUCT_IMAGES = 116;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["TOO_MANY_PRODUCT_IMAGES"] = 116] = "TOO_MANY_PRODUCT_IMAGES";
    /**
     * Too many product Videos in GmailAd
     *
     * @generated from protobuf enum value: TOO_MANY_PRODUCT_VIDEOS = 117;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["TOO_MANY_PRODUCT_VIDEOS"] = 117] = "TOO_MANY_PRODUCT_VIDEOS";
    /**
     * The device preference is not compatible with the ad type
     *
     * @generated from protobuf enum value: INCOMPATIBLE_AD_TYPE_AND_DEVICE_PREFERENCE = 118;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INCOMPATIBLE_AD_TYPE_AND_DEVICE_PREFERENCE"] = 118] = "INCOMPATIBLE_AD_TYPE_AND_DEVICE_PREFERENCE";
    /**
     * Call tracking is not supported for specified country.
     *
     * @generated from protobuf enum value: CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY = 119;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY"] = 119] = "CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * Carrier specific short number is not allowed.
     *
     * @generated from protobuf enum value: CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED = 120;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED"] = 120] = "CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED";
    /**
     * Specified phone number type is disallowed.
     *
     * @generated from protobuf enum value: DISALLOWED_NUMBER_TYPE = 121;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DISALLOWED_NUMBER_TYPE"] = 121] = "DISALLOWED_NUMBER_TYPE";
    /**
     * Phone number not supported for country.
     *
     * @generated from protobuf enum value: PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY = 122;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY"] = 122] = "PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * Phone number not supported with call tracking enabled for country.
     *
     * @generated from protobuf enum value: PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY = 123;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY"] = 123] = "PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY";
    /**
     * Premium rate phone number is not allowed.
     *
     * @generated from protobuf enum value: PREMIUM_RATE_NUMBER_NOT_ALLOWED = 124;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["PREMIUM_RATE_NUMBER_NOT_ALLOWED"] = 124] = "PREMIUM_RATE_NUMBER_NOT_ALLOWED";
    /**
     * Vanity phone number is not allowed.
     *
     * @generated from protobuf enum value: VANITY_PHONE_NUMBER_NOT_ALLOWED = 125;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["VANITY_PHONE_NUMBER_NOT_ALLOWED"] = 125] = "VANITY_PHONE_NUMBER_NOT_ALLOWED";
    /**
     * Invalid call conversion type id.
     *
     * @generated from protobuf enum value: INVALID_CALL_CONVERSION_TYPE_ID = 126;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_CALL_CONVERSION_TYPE_ID"] = 126] = "INVALID_CALL_CONVERSION_TYPE_ID";
    /**
     * Cannot disable call conversion and set conversion type id.
     *
     * @generated from protobuf enum value: CANNOT_DISABLE_CALL_CONVERSION_AND_SET_CONVERSION_TYPE_ID = 127;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_DISABLE_CALL_CONVERSION_AND_SET_CONVERSION_TYPE_ID"] = 127] = "CANNOT_DISABLE_CALL_CONVERSION_AND_SET_CONVERSION_TYPE_ID";
    /**
     * Cannot set path2 without path1.
     *
     * @generated from protobuf enum value: CANNOT_SET_PATH2_WITHOUT_PATH1 = 128;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CANNOT_SET_PATH2_WITHOUT_PATH1"] = 128] = "CANNOT_SET_PATH2_WITHOUT_PATH1";
    /**
     * Missing domain name in campaign setting when adding expanded dynamic
     * search ad.
     *
     * @generated from protobuf enum value: MISSING_DYNAMIC_SEARCH_ADS_SETTING_DOMAIN_NAME = 129;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_DYNAMIC_SEARCH_ADS_SETTING_DOMAIN_NAME"] = 129] = "MISSING_DYNAMIC_SEARCH_ADS_SETTING_DOMAIN_NAME";
    /**
     * The associated ad is not compatible with restriction type.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_WITH_RESTRICTION_TYPE = 130;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INCOMPATIBLE_WITH_RESTRICTION_TYPE"] = 130] = "INCOMPATIBLE_WITH_RESTRICTION_TYPE";
    /**
     * Consent for call recording is required for creating/updating call only
     * ads. See https://support.google.com/google-ads/answer/7412639.
     *
     * @generated from protobuf enum value: CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED = 131;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED"] = 131] = "CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED";
    /**
     * Either an image or a media bundle is required in a display upload ad.
     *
     * @generated from protobuf enum value: MISSING_IMAGE_OR_MEDIA_BUNDLE = 132;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_IMAGE_OR_MEDIA_BUNDLE"] = 132] = "MISSING_IMAGE_OR_MEDIA_BUNDLE";
    /**
     * The display upload product type is not supported in this campaign.
     *
     * @generated from protobuf enum value: PRODUCT_TYPE_NOT_SUPPORTED_IN_THIS_CAMPAIGN = 133;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["PRODUCT_TYPE_NOT_SUPPORTED_IN_THIS_CAMPAIGN"] = 133] = "PRODUCT_TYPE_NOT_SUPPORTED_IN_THIS_CAMPAIGN";
    /**
     * The default value of an ad placeholder can not be the empty string.
     *
     * @generated from protobuf enum value: PLACEHOLDER_CANNOT_HAVE_EMPTY_DEFAULT_VALUE = 134;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["PLACEHOLDER_CANNOT_HAVE_EMPTY_DEFAULT_VALUE"] = 134] = "PLACEHOLDER_CANNOT_HAVE_EMPTY_DEFAULT_VALUE";
    /**
     * Ad placeholders with countdown functions must not have a default value.
     *
     * @generated from protobuf enum value: PLACEHOLDER_COUNTDOWN_FUNCTION_CANNOT_HAVE_DEFAULT_VALUE = 135;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["PLACEHOLDER_COUNTDOWN_FUNCTION_CANNOT_HAVE_DEFAULT_VALUE"] = 135] = "PLACEHOLDER_COUNTDOWN_FUNCTION_CANNOT_HAVE_DEFAULT_VALUE";
    /**
     * A previous ad placeholder that had a default value was found which means
     * that all (non-countdown) placeholders must have a default value. This
     * ad placeholder does not have a default value.
     *
     * @generated from protobuf enum value: PLACEHOLDER_DEFAULT_VALUE_MISSING = 136;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["PLACEHOLDER_DEFAULT_VALUE_MISSING"] = 136] = "PLACEHOLDER_DEFAULT_VALUE_MISSING";
    /**
     * A previous ad placeholder that did not have a default value was found
     * which means that no placeholders may have a default value. This
     * ad placeholder does have a default value.
     *
     * @generated from protobuf enum value: UNEXPECTED_PLACEHOLDER_DEFAULT_VALUE = 137;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNEXPECTED_PLACEHOLDER_DEFAULT_VALUE"] = 137] = "UNEXPECTED_PLACEHOLDER_DEFAULT_VALUE";
    /**
     * Two ad customizers may not be directly adjacent in an ad text. They must
     * be separated by at least one character.
     *
     * @generated from protobuf enum value: AD_CUSTOMIZERS_MAY_NOT_BE_ADJACENT = 138;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["AD_CUSTOMIZERS_MAY_NOT_BE_ADJACENT"] = 138] = "AD_CUSTOMIZERS_MAY_NOT_BE_ADJACENT";
    /**
     * The ad is not associated with any enabled AdGroupAd, and cannot be
     * updated.
     *
     * @generated from protobuf enum value: UPDATING_AD_WITH_NO_ENABLED_ASSOCIATION = 139;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UPDATING_AD_WITH_NO_ENABLED_ASSOCIATION"] = 139] = "UPDATING_AD_WITH_NO_ENABLED_ASSOCIATION";
    /**
     * Call Ad verification url and final url don't have same domain.
     *
     * @generated from protobuf enum value: CALL_AD_VERIFICATION_URL_FINAL_URL_DOES_NOT_HAVE_SAME_DOMAIN = 140;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CALL_AD_VERIFICATION_URL_FINAL_URL_DOES_NOT_HAVE_SAME_DOMAIN"] = 140] = "CALL_AD_VERIFICATION_URL_FINAL_URL_DOES_NOT_HAVE_SAME_DOMAIN";
    /**
     * Final url and verification url cannot both be empty for call ads.
     *
     * @generated from protobuf enum value: CALL_AD_FINAL_URL_AND_VERIFICATION_URL_CANNOT_BOTH_BE_EMPTY = 154;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CALL_AD_FINAL_URL_AND_VERIFICATION_URL_CANNOT_BOTH_BE_EMPTY"] = 154] = "CALL_AD_FINAL_URL_AND_VERIFICATION_URL_CANNOT_BOTH_BE_EMPTY";
    /**
     * Too many ad customizers in one asset.
     *
     * @generated from protobuf enum value: TOO_MANY_AD_CUSTOMIZERS = 141;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["TOO_MANY_AD_CUSTOMIZERS"] = 141] = "TOO_MANY_AD_CUSTOMIZERS";
    /**
     * The ad customizer tag is recognized, but the format is invalid.
     *
     * @generated from protobuf enum value: INVALID_AD_CUSTOMIZER_FORMAT = 142;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_AD_CUSTOMIZER_FORMAT"] = 142] = "INVALID_AD_CUSTOMIZER_FORMAT";
    /**
     * Customizer tags cannot be nested.
     *
     * @generated from protobuf enum value: NESTED_AD_CUSTOMIZER_SYNTAX = 143;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["NESTED_AD_CUSTOMIZER_SYNTAX"] = 143] = "NESTED_AD_CUSTOMIZER_SYNTAX";
    /**
     * The ad customizer syntax used in the ad is not supported.
     *
     * @generated from protobuf enum value: UNSUPPORTED_AD_CUSTOMIZER_SYNTAX = 144;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNSUPPORTED_AD_CUSTOMIZER_SYNTAX"] = 144] = "UNSUPPORTED_AD_CUSTOMIZER_SYNTAX";
    /**
     * There exists unpaired brace in the ad customizer tag.
     *
     * @generated from protobuf enum value: UNPAIRED_BRACE_IN_AD_CUSTOMIZER_TAG = 145;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNPAIRED_BRACE_IN_AD_CUSTOMIZER_TAG"] = 145] = "UNPAIRED_BRACE_IN_AD_CUSTOMIZER_TAG";
    /**
     * More than one type of countdown tag exists among all text lines.
     *
     * @generated from protobuf enum value: MORE_THAN_ONE_COUNTDOWN_TAG_TYPE_EXISTS = 146;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MORE_THAN_ONE_COUNTDOWN_TAG_TYPE_EXISTS"] = 146] = "MORE_THAN_ONE_COUNTDOWN_TAG_TYPE_EXISTS";
    /**
     * Date time in the countdown tag is invalid.
     *
     * @generated from protobuf enum value: DATE_TIME_IN_COUNTDOWN_TAG_IS_INVALID = 147;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DATE_TIME_IN_COUNTDOWN_TAG_IS_INVALID"] = 147] = "DATE_TIME_IN_COUNTDOWN_TAG_IS_INVALID";
    /**
     * Date time in the countdown tag is in the past.
     *
     * @generated from protobuf enum value: DATE_TIME_IN_COUNTDOWN_TAG_IS_PAST = 148;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DATE_TIME_IN_COUNTDOWN_TAG_IS_PAST"] = 148] = "DATE_TIME_IN_COUNTDOWN_TAG_IS_PAST";
    /**
     * Cannot recognize the ad customizer tag.
     *
     * @generated from protobuf enum value: UNRECOGNIZED_AD_CUSTOMIZER_TAG_FOUND = 149;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["UNRECOGNIZED_AD_CUSTOMIZER_TAG_FOUND"] = 149] = "UNRECOGNIZED_AD_CUSTOMIZER_TAG_FOUND";
    /**
     * Customizer type forbidden for this field.
     *
     * @generated from protobuf enum value: CUSTOMIZER_TYPE_FORBIDDEN_FOR_FIELD = 150;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["CUSTOMIZER_TYPE_FORBIDDEN_FOR_FIELD"] = 150] = "CUSTOMIZER_TYPE_FORBIDDEN_FOR_FIELD";
    /**
     * Customizer attribute name is invalid.
     *
     * @generated from protobuf enum value: INVALID_CUSTOMIZER_ATTRIBUTE_NAME = 151;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["INVALID_CUSTOMIZER_ATTRIBUTE_NAME"] = 151] = "INVALID_CUSTOMIZER_ATTRIBUTE_NAME";
    /**
     * App store value does not match the value of the app store in the app
     * specified in the campaign.
     *
     * @generated from protobuf enum value: STORE_MISMATCH = 152;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["STORE_MISMATCH"] = 152] = "STORE_MISMATCH";
    /**
     * Missing required image aspect ratio.
     *
     * @generated from protobuf enum value: MISSING_REQUIRED_IMAGE_ASPECT_RATIO = 153;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISSING_REQUIRED_IMAGE_ASPECT_RATIO"] = 153] = "MISSING_REQUIRED_IMAGE_ASPECT_RATIO";
    /**
     * Aspect ratios mismatch between different assets.
     *
     * @generated from protobuf enum value: MISMATCHED_ASPECT_RATIOS = 155;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["MISMATCHED_ASPECT_RATIOS"] = 155] = "MISMATCHED_ASPECT_RATIOS";
    /**
     * Images must be unique between different carousel card assets.
     *
     * @generated from protobuf enum value: DUPLICATE_IMAGE_ACROSS_CAROUSEL_CARDS = 156;
     */
    AdErrorEnum_AdError[AdErrorEnum_AdError["DUPLICATE_IMAGE_ACROSS_CAROUSEL_CARDS"] = 156] = "DUPLICATE_IMAGE_ACROSS_CAROUSEL_CARDS";
})(AdErrorEnum_AdError = exports.AdErrorEnum_AdError || (exports.AdErrorEnum_AdError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdErrorEnum
 */
exports.AdErrorEnum = new AdErrorEnum$Type();
