// google-ads-api-client/dist/protos/bindings/ads/errors/ad_group_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupErrorEnum = exports.AdGroupErrorEnum_AdGroupError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ad group errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdGroupErrorEnum.AdGroupError
 */
var AdGroupErrorEnum_AdGroupError;
(function (AdGroupErrorEnum_AdGroupError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * AdGroup with the same name already exists for the campaign.
     *
     * @generated from protobuf enum value: DUPLICATE_ADGROUP_NAME = 2;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["DUPLICATE_ADGROUP_NAME"] = 2] = "DUPLICATE_ADGROUP_NAME";
    /**
     * AdGroup name is not valid.
     *
     * @generated from protobuf enum value: INVALID_ADGROUP_NAME = 3;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["INVALID_ADGROUP_NAME"] = 3] = "INVALID_ADGROUP_NAME";
    /**
     * Advertiser is not allowed to target sites or set site bids that are not
     * on the Google Search Network.
     *
     * @generated from protobuf enum value: ADVERTISER_NOT_ON_CONTENT_NETWORK = 5;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["ADVERTISER_NOT_ON_CONTENT_NETWORK"] = 5] = "ADVERTISER_NOT_ON_CONTENT_NETWORK";
    /**
     * Bid amount is too big.
     *
     * @generated from protobuf enum value: BID_TOO_BIG = 6;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["BID_TOO_BIG"] = 6] = "BID_TOO_BIG";
    /**
     * AdGroup bid does not match the campaign's bidding strategy.
     *
     * @generated from protobuf enum value: BID_TYPE_AND_BIDDING_STRATEGY_MISMATCH = 7;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["BID_TYPE_AND_BIDDING_STRATEGY_MISMATCH"] = 7] = "BID_TYPE_AND_BIDDING_STRATEGY_MISMATCH";
    /**
     * AdGroup name is required for Add.
     *
     * @generated from protobuf enum value: MISSING_ADGROUP_NAME = 8;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["MISSING_ADGROUP_NAME"] = 8] = "MISSING_ADGROUP_NAME";
    /**
     * No link found between the ad group and the label.
     *
     * @generated from protobuf enum value: ADGROUP_LABEL_DOES_NOT_EXIST = 9;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["ADGROUP_LABEL_DOES_NOT_EXIST"] = 9] = "ADGROUP_LABEL_DOES_NOT_EXIST";
    /**
     * The label has already been attached to the ad group.
     *
     * @generated from protobuf enum value: ADGROUP_LABEL_ALREADY_EXISTS = 10;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["ADGROUP_LABEL_ALREADY_EXISTS"] = 10] = "ADGROUP_LABEL_ALREADY_EXISTS";
    /**
     * The CriterionTypeGroup is not supported for the content bid dimension.
     *
     * @generated from protobuf enum value: INVALID_CONTENT_BID_CRITERION_TYPE_GROUP = 11;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["INVALID_CONTENT_BID_CRITERION_TYPE_GROUP"] = 11] = "INVALID_CONTENT_BID_CRITERION_TYPE_GROUP";
    /**
     * The ad group type is not compatible with the campaign channel type.
     *
     * @generated from protobuf enum value: AD_GROUP_TYPE_NOT_VALID_FOR_ADVERTISING_CHANNEL_TYPE = 12;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["AD_GROUP_TYPE_NOT_VALID_FOR_ADVERTISING_CHANNEL_TYPE"] = 12] = "AD_GROUP_TYPE_NOT_VALID_FOR_ADVERTISING_CHANNEL_TYPE";
    /**
     * The ad group type is not supported in the country of sale of the
     * campaign.
     *
     * @generated from protobuf enum value: ADGROUP_TYPE_NOT_SUPPORTED_FOR_CAMPAIGN_SALES_COUNTRY = 13;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["ADGROUP_TYPE_NOT_SUPPORTED_FOR_CAMPAIGN_SALES_COUNTRY"] = 13] = "ADGROUP_TYPE_NOT_SUPPORTED_FOR_CAMPAIGN_SALES_COUNTRY";
    /**
     * Ad groups of AdGroupType.SEARCH_DYNAMIC_ADS can only be added to
     * campaigns that have DynamicSearchAdsSetting attached.
     *
     * @generated from protobuf enum value: CANNOT_ADD_ADGROUP_OF_TYPE_DSA_TO_CAMPAIGN_WITHOUT_DSA_SETTING = 14;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["CANNOT_ADD_ADGROUP_OF_TYPE_DSA_TO_CAMPAIGN_WITHOUT_DSA_SETTING"] = 14] = "CANNOT_ADD_ADGROUP_OF_TYPE_DSA_TO_CAMPAIGN_WITHOUT_DSA_SETTING";
    /**
     * Promoted hotels ad groups are only available to customers on the
     * allow-list.
     *
     * @generated from protobuf enum value: PROMOTED_HOTEL_AD_GROUPS_NOT_AVAILABLE_FOR_CUSTOMER = 15;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["PROMOTED_HOTEL_AD_GROUPS_NOT_AVAILABLE_FOR_CUSTOMER"] = 15] = "PROMOTED_HOTEL_AD_GROUPS_NOT_AVAILABLE_FOR_CUSTOMER";
    /**
     * The field type cannot be excluded because an active ad group-asset link
     * of this type exists.
     *
     * @generated from protobuf enum value: INVALID_EXCLUDED_PARENT_ASSET_FIELD_TYPE = 16;
     */
    AdGroupErrorEnum_AdGroupError[AdGroupErrorEnum_AdGroupError["INVALID_EXCLUDED_PARENT_ASSET_FIELD_TYPE"] = 16] = "INVALID_EXCLUDED_PARENT_ASSET_FIELD_TYPE";
})(AdGroupErrorEnum_AdGroupError = exports.AdGroupErrorEnum_AdGroupError || (exports.AdGroupErrorEnum_AdGroupError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdGroupErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdGroupErrorEnum
 */
exports.AdGroupErrorEnum = new AdGroupErrorEnum$Type();
