// google-ads-api-client/dist/protos/bindings/ads/errors/ad_group_ad_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupAdErrorEnum = exports.AdGroupAdErrorEnum_AdGroupAdError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ad group ad errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdGroupAdErrorEnum.AdGroupAdError
 */
var AdGroupAdErrorEnum_AdGroupAdError;
(function (AdGroupAdErrorEnum_AdGroupAdError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * No link found between the adgroup ad and the label.
     *
     * @generated from protobuf enum value: AD_GROUP_AD_LABEL_DOES_NOT_EXIST = 2;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["AD_GROUP_AD_LABEL_DOES_NOT_EXIST"] = 2] = "AD_GROUP_AD_LABEL_DOES_NOT_EXIST";
    /**
     * The label has already been attached to the adgroup ad.
     *
     * @generated from protobuf enum value: AD_GROUP_AD_LABEL_ALREADY_EXISTS = 3;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["AD_GROUP_AD_LABEL_ALREADY_EXISTS"] = 3] = "AD_GROUP_AD_LABEL_ALREADY_EXISTS";
    /**
     * The specified ad was not found in the adgroup
     *
     * @generated from protobuf enum value: AD_NOT_UNDER_ADGROUP = 4;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["AD_NOT_UNDER_ADGROUP"] = 4] = "AD_NOT_UNDER_ADGROUP";
    /**
     * Removed ads may not be modified
     *
     * @generated from protobuf enum value: CANNOT_OPERATE_ON_REMOVED_ADGROUPAD = 5;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["CANNOT_OPERATE_ON_REMOVED_ADGROUPAD"] = 5] = "CANNOT_OPERATE_ON_REMOVED_ADGROUPAD";
    /**
     * An ad of this type is deprecated and cannot be created. Only deletions
     * are permitted.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_DEPRECATED_ADS = 6;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["CANNOT_CREATE_DEPRECATED_ADS"] = 6] = "CANNOT_CREATE_DEPRECATED_ADS";
    /**
     * Text ads are deprecated and cannot be created. Use expanded text ads
     * instead.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_TEXT_ADS = 7;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["CANNOT_CREATE_TEXT_ADS"] = 7] = "CANNOT_CREATE_TEXT_ADS";
    /**
     * A required field was not specified or is an empty string.
     *
     * @generated from protobuf enum value: EMPTY_FIELD = 8;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["EMPTY_FIELD"] = 8] = "EMPTY_FIELD";
    /**
     * An ad may only be modified once per call
     *
     * @generated from protobuf enum value: RESOURCE_REFERENCED_IN_MULTIPLE_OPS = 9;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["RESOURCE_REFERENCED_IN_MULTIPLE_OPS"] = 9] = "RESOURCE_REFERENCED_IN_MULTIPLE_OPS";
    /**
     * AdGroupAds with the given ad type cannot be paused.
     *
     * @generated from protobuf enum value: AD_TYPE_CANNOT_BE_PAUSED = 10;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["AD_TYPE_CANNOT_BE_PAUSED"] = 10] = "AD_TYPE_CANNOT_BE_PAUSED";
    /**
     * AdGroupAds with the given ad type cannot be removed.
     *
     * @generated from protobuf enum value: AD_TYPE_CANNOT_BE_REMOVED = 11;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["AD_TYPE_CANNOT_BE_REMOVED"] = 11] = "AD_TYPE_CANNOT_BE_REMOVED";
    /**
     * An ad of this type is deprecated and cannot be updated. Only removals
     * are permitted.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_DEPRECATED_ADS = 12;
     */
    AdGroupAdErrorEnum_AdGroupAdError[AdGroupAdErrorEnum_AdGroupAdError["CANNOT_UPDATE_DEPRECATED_ADS"] = 12] = "CANNOT_UPDATE_DEPRECATED_ADS";
})(AdGroupAdErrorEnum_AdGroupAdError = exports.AdGroupAdErrorEnum_AdGroupAdError || (exports.AdGroupAdErrorEnum_AdGroupAdError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupAdErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdGroupAdErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdGroupAdErrorEnum
 */
exports.AdGroupAdErrorEnum = new AdGroupAdErrorEnum$Type();
