// google-ads-api-client/dist/protos/bindings/ads/errors/ad_group_feed_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupFeedErrorEnum = exports.AdGroupFeedErrorEnum_AdGroupFeedError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ad group feed errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdGroupFeedErrorEnum.AdGroupFeedError
 */
var AdGroupFeedErrorEnum_AdGroupFeedError;
(function (AdGroupFeedErrorEnum_AdGroupFeedError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An active feed already exists for this ad group and place holder type.
     *
     * @generated from protobuf enum value: FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE = 2;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE"] = 2] = "FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE";
    /**
     * The specified feed is removed.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_FOR_REMOVED_FEED = 3;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["CANNOT_CREATE_FOR_REMOVED_FEED"] = 3] = "CANNOT_CREATE_FOR_REMOVED_FEED";
    /**
     * The AdGroupFeed already exists. UPDATE operation should be used to modify
     * the existing AdGroupFeed.
     *
     * @generated from protobuf enum value: ADGROUP_FEED_ALREADY_EXISTS = 4;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["ADGROUP_FEED_ALREADY_EXISTS"] = 4] = "ADGROUP_FEED_ALREADY_EXISTS";
    /**
     * Cannot operate on removed AdGroupFeed.
     *
     * @generated from protobuf enum value: CANNOT_OPERATE_ON_REMOVED_ADGROUP_FEED = 5;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["CANNOT_OPERATE_ON_REMOVED_ADGROUP_FEED"] = 5] = "CANNOT_OPERATE_ON_REMOVED_ADGROUP_FEED";
    /**
     * Invalid placeholder type.
     *
     * @generated from protobuf enum value: INVALID_PLACEHOLDER_TYPE = 6;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["INVALID_PLACEHOLDER_TYPE"] = 6] = "INVALID_PLACEHOLDER_TYPE";
    /**
     * Feed mapping for this placeholder type does not exist.
     *
     * @generated from protobuf enum value: MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE = 7;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE"] = 7] = "MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE";
    /**
     * Location AdGroupFeeds cannot be created unless there is a location
     * CustomerFeed for the specified feed.
     *
     * @generated from protobuf enum value: NO_EXISTING_LOCATION_CUSTOMER_FEED = 8;
     */
    AdGroupFeedErrorEnum_AdGroupFeedError[AdGroupFeedErrorEnum_AdGroupFeedError["NO_EXISTING_LOCATION_CUSTOMER_FEED"] = 8] = "NO_EXISTING_LOCATION_CUSTOMER_FEED";
})(AdGroupFeedErrorEnum_AdGroupFeedError = exports.AdGroupFeedErrorEnum_AdGroupFeedError || (exports.AdGroupFeedErrorEnum_AdGroupFeedError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupFeedErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdGroupFeedErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdGroupFeedErrorEnum
 */
exports.AdGroupFeedErrorEnum = new AdGroupFeedErrorEnum$Type();
