// google-ads-api-client/dist/protos/bindings/ads/errors/ad_group_bid_modifier_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupBidModifierErrorEnum = exports.AdGroupBidModifierErrorEnum_AdGroupBidModifierError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ad group bid modifier errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdGroupBidModifierErrorEnum.AdGroupBidModifierError
 */
var AdGroupBidModifierErrorEnum_AdGroupBidModifierError;
(function (AdGroupBidModifierErrorEnum_AdGroupBidModifierError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupBidModifierErrorEnum_AdGroupBidModifierError[AdGroupBidModifierErrorEnum_AdGroupBidModifierError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupBidModifierErrorEnum_AdGroupBidModifierError[AdGroupBidModifierErrorEnum_AdGroupBidModifierError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The criterion ID does not support bid modification.
     *
     * @generated from protobuf enum value: CRITERION_ID_NOT_SUPPORTED = 2;
     */
    AdGroupBidModifierErrorEnum_AdGroupBidModifierError[AdGroupBidModifierErrorEnum_AdGroupBidModifierError["CRITERION_ID_NOT_SUPPORTED"] = 2] = "CRITERION_ID_NOT_SUPPORTED";
    /**
     * Cannot override the bid modifier for the given criterion ID if the parent
     * campaign is opted out of the same criterion.
     *
     * @generated from protobuf enum value: CANNOT_OVERRIDE_OPTED_OUT_CAMPAIGN_CRITERION_BID_MODIFIER = 3;
     */
    AdGroupBidModifierErrorEnum_AdGroupBidModifierError[AdGroupBidModifierErrorEnum_AdGroupBidModifierError["CANNOT_OVERRIDE_OPTED_OUT_CAMPAIGN_CRITERION_BID_MODIFIER"] = 3] = "CANNOT_OVERRIDE_OPTED_OUT_CAMPAIGN_CRITERION_BID_MODIFIER";
})(AdGroupBidModifierErrorEnum_AdGroupBidModifierError = exports.AdGroupBidModifierErrorEnum_AdGroupBidModifierError || (exports.AdGroupBidModifierErrorEnum_AdGroupBidModifierError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupBidModifierErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdGroupBidModifierErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdGroupBidModifierErrorEnum
 */
exports.AdGroupBidModifierErrorEnum = new AdGroupBidModifierErrorEnum$Type();
