// google-ads-api-client/dist/protos/bindings/ads/errors/ad_group_criterion_customizer_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupCriterionCustomizerErrorEnum = exports.AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ad group criterion customizer errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdGroupCriterionCustomizerErrorEnum.AdGroupCriterionCustomizerError
 */
var AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError;
(function (AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError[AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError[AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Only keyword type criterion is allowed to link customizer attribute.
     *
     * @generated from protobuf enum value: CRITERION_IS_NOT_KEYWORD = 2;
     */
    AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError[AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError["CRITERION_IS_NOT_KEYWORD"] = 2] = "CRITERION_IS_NOT_KEYWORD";
})(AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError = exports.AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError || (exports.AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupCriterionCustomizerErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdGroupCriterionCustomizerErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdGroupCriterionCustomizerErrorEnum
 */
exports.AdGroupCriterionCustomizerErrorEnum = new AdGroupCriterionCustomizerErrorEnum$Type();
