// google-ads-api-client/dist/protos/bindings/ads/errors/ad_sharing_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdSharingErrorEnum = exports.AdSharingErrorEnum_AdSharingError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ad sharing errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdSharingErrorEnum.AdSharingError
 */
var AdSharingErrorEnum_AdSharingError;
(function (AdSharingErrorEnum_AdSharingError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdSharingErrorEnum_AdSharingError[AdSharingErrorEnum_AdSharingError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdSharingErrorEnum_AdSharingError[AdSharingErrorEnum_AdSharingError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Error resulting in attempting to add an Ad to an AdGroup that already
     * contains the Ad.
     *
     * @generated from protobuf enum value: AD_GROUP_ALREADY_CONTAINS_AD = 2;
     */
    AdSharingErrorEnum_AdSharingError[AdSharingErrorEnum_AdSharingError["AD_GROUP_ALREADY_CONTAINS_AD"] = 2] = "AD_GROUP_ALREADY_CONTAINS_AD";
    /**
     * Ad is not compatible with the AdGroup it is being shared with.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_AD_UNDER_AD_GROUP = 3;
     */
    AdSharingErrorEnum_AdSharingError[AdSharingErrorEnum_AdSharingError["INCOMPATIBLE_AD_UNDER_AD_GROUP"] = 3] = "INCOMPATIBLE_AD_UNDER_AD_GROUP";
    /**
     * Cannot add AdGroupAd on inactive Ad.
     *
     * @generated from protobuf enum value: CANNOT_SHARE_INACTIVE_AD = 4;
     */
    AdSharingErrorEnum_AdSharingError[AdSharingErrorEnum_AdSharingError["CANNOT_SHARE_INACTIVE_AD"] = 4] = "CANNOT_SHARE_INACTIVE_AD";
})(AdSharingErrorEnum_AdSharingError = exports.AdSharingErrorEnum_AdSharingError || (exports.AdSharingErrorEnum_AdSharingError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdSharingErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdSharingErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdSharingErrorEnum
 */
exports.AdSharingErrorEnum = new AdSharingErrorEnum$Type();
