// google-ads-api-client/dist/protos/bindings/ads/errors/adx_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdxErrorEnum = exports.AdxErrorEnum_AdxError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible adx errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AdxErrorEnum.AdxError
 */
var AdxErrorEnum_AdxError;
(function (AdxErrorEnum_AdxError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AdxErrorEnum_AdxError[AdxErrorEnum_AdxError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AdxErrorEnum_AdxError[AdxErrorEnum_AdxError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Attempt to use non-AdX feature by AdX customer.
     *
     * @generated from protobuf enum value: UNSUPPORTED_FEATURE = 2;
     */
    AdxErrorEnum_AdxError[AdxErrorEnum_AdxError["UNSUPPORTED_FEATURE"] = 2] = "UNSUPPORTED_FEATURE";
})(AdxErrorEnum_AdxError = exports.AdxErrorEnum_AdxError || (exports.AdxErrorEnum_AdxError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AdxErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AdxErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AdxErrorEnum
 */
exports.AdxErrorEnum = new AdxErrorEnum$Type();
