// google-ads-api-client/dist/protos/bindings/ads/errors/asset_set_asset_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSetAssetErrorEnum = exports.AssetSetAssetErrorEnum_AssetSetAssetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible asset set asset errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AssetSetAssetErrorEnum.AssetSetAssetError
 */
var AssetSetAssetErrorEnum_AssetSetAssetError;
(function (AssetSetAssetErrorEnum_AssetSetAssetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetSetAssetErrorEnum_AssetSetAssetError[AssetSetAssetErrorEnum_AssetSetAssetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetSetAssetErrorEnum_AssetSetAssetError[AssetSetAssetErrorEnum_AssetSetAssetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The asset type is not eligible to be linked to the specific type of asset
     * set.
     *
     * @generated from protobuf enum value: INVALID_ASSET_TYPE = 2;
     */
    AssetSetAssetErrorEnum_AssetSetAssetError[AssetSetAssetErrorEnum_AssetSetAssetError["INVALID_ASSET_TYPE"] = 2] = "INVALID_ASSET_TYPE";
    /**
     * The asset set type is not eligible to contain the specified type of
     * assets.
     *
     * @generated from protobuf enum value: INVALID_ASSET_SET_TYPE = 3;
     */
    AssetSetAssetErrorEnum_AssetSetAssetError[AssetSetAssetErrorEnum_AssetSetAssetError["INVALID_ASSET_SET_TYPE"] = 3] = "INVALID_ASSET_SET_TYPE";
    /**
     * The asset contains duplicate external key with another asset in the asset
     * set.
     *
     * @generated from protobuf enum value: DUPLICATE_EXTERNAL_KEY = 4;
     */
    AssetSetAssetErrorEnum_AssetSetAssetError[AssetSetAssetErrorEnum_AssetSetAssetError["DUPLICATE_EXTERNAL_KEY"] = 4] = "DUPLICATE_EXTERNAL_KEY";
})(AssetSetAssetErrorEnum_AssetSetAssetError = exports.AssetSetAssetErrorEnum_AssetSetAssetError || (exports.AssetSetAssetErrorEnum_AssetSetAssetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetSetAssetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AssetSetAssetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AssetSetAssetErrorEnum
 */
exports.AssetSetAssetErrorEnum = new AssetSetAssetErrorEnum$Type();
