// google-ads-api-client/dist/protos/bindings/ads/errors/asset_set_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSetErrorEnum = exports.AssetSetErrorEnum_AssetSetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible asset set errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AssetSetErrorEnum.AssetSetError
 */
var AssetSetErrorEnum_AssetSetError;
(function (AssetSetErrorEnum_AssetSetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetSetErrorEnum_AssetSetError[AssetSetErrorEnum_AssetSetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetSetErrorEnum_AssetSetError[AssetSetErrorEnum_AssetSetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The asset set name matches that of another enabled asset set.
     *
     * @generated from protobuf enum value: DUPLICATE_ASSET_SET_NAME = 2;
     */
    AssetSetErrorEnum_AssetSetError[AssetSetErrorEnum_AssetSetError["DUPLICATE_ASSET_SET_NAME"] = 2] = "DUPLICATE_ASSET_SET_NAME";
})(AssetSetErrorEnum_AssetSetError = exports.AssetSetErrorEnum_AssetSetError || (exports.AssetSetErrorEnum_AssetSetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetSetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AssetSetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AssetSetErrorEnum
 */
exports.AssetSetErrorEnum = new AssetSetErrorEnum$Type();
