// google-ads-api-client/dist/protos/bindings/ads/errors/asset_group_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetGroupErrorEnum = exports.AssetGroupErrorEnum_AssetGroupError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible asset group errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AssetGroupErrorEnum.AssetGroupError
 */
var AssetGroupErrorEnum_AssetGroupError;
(function (AssetGroupErrorEnum_AssetGroupError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Each asset group in a single campaign must have a unique name.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 2;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["DUPLICATE_NAME"] = 2] = "DUPLICATE_NAME";
    /**
     * Cannot add asset group for the campaign type.
     *
     * @generated from protobuf enum value: CANNOT_ADD_ASSET_GROUP_FOR_CAMPAIGN_TYPE = 3;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["CANNOT_ADD_ASSET_GROUP_FOR_CAMPAIGN_TYPE"] = 3] = "CANNOT_ADD_ASSET_GROUP_FOR_CAMPAIGN_TYPE";
    /**
     * Not enough headline asset for a valid asset group.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_HEADLINE_ASSET = 4;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["NOT_ENOUGH_HEADLINE_ASSET"] = 4] = "NOT_ENOUGH_HEADLINE_ASSET";
    /**
     * Not enough long headline asset for a valid asset group.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_LONG_HEADLINE_ASSET = 5;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["NOT_ENOUGH_LONG_HEADLINE_ASSET"] = 5] = "NOT_ENOUGH_LONG_HEADLINE_ASSET";
    /**
     * Not enough description headline asset for a valid asset group.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_DESCRIPTION_ASSET = 6;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["NOT_ENOUGH_DESCRIPTION_ASSET"] = 6] = "NOT_ENOUGH_DESCRIPTION_ASSET";
    /**
     * Not enough business name asset for a valid asset group.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_BUSINESS_NAME_ASSET = 7;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["NOT_ENOUGH_BUSINESS_NAME_ASSET"] = 7] = "NOT_ENOUGH_BUSINESS_NAME_ASSET";
    /**
     * Not enough marketing image asset for a valid asset group.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_MARKETING_IMAGE_ASSET = 8;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["NOT_ENOUGH_MARKETING_IMAGE_ASSET"] = 8] = "NOT_ENOUGH_MARKETING_IMAGE_ASSET";
    /**
     * Not enough square marketing image asset for a valid asset group.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_SQUARE_MARKETING_IMAGE_ASSET = 9;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["NOT_ENOUGH_SQUARE_MARKETING_IMAGE_ASSET"] = 9] = "NOT_ENOUGH_SQUARE_MARKETING_IMAGE_ASSET";
    /**
     * Not enough logo asset for a valid asset group.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_LOGO_ASSET = 10;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["NOT_ENOUGH_LOGO_ASSET"] = 10] = "NOT_ENOUGH_LOGO_ASSET";
    /**
     * Final url and shopping merchant url does not have the same domain.
     *
     * @generated from protobuf enum value: FINAL_URL_SHOPPING_MERCHANT_HOME_PAGE_URL_DOMAINS_DIFFER = 11;
     */
    AssetGroupErrorEnum_AssetGroupError[AssetGroupErrorEnum_AssetGroupError["FINAL_URL_SHOPPING_MERCHANT_HOME_PAGE_URL_DOMAINS_DIFFER"] = 11] = "FINAL_URL_SHOPPING_MERCHANT_HOME_PAGE_URL_DOMAINS_DIFFER";
})(AssetGroupErrorEnum_AssetGroupError = exports.AssetGroupErrorEnum_AssetGroupError || (exports.AssetGroupErrorEnum_AssetGroupError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetGroupErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AssetGroupErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AssetGroupErrorEnum
 */
exports.AssetGroupErrorEnum = new AssetGroupErrorEnum$Type();
