// google-ads-api-client/dist/protos/bindings/ads/errors/asset_group_asset_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetGroupAssetErrorEnum = exports.AssetGroupAssetErrorEnum_AssetGroupAssetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible asset group asset errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AssetGroupAssetErrorEnum.AssetGroupAssetError
 */
var AssetGroupAssetErrorEnum_AssetGroupAssetError;
(function (AssetGroupAssetErrorEnum_AssetGroupAssetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetGroupAssetErrorEnum_AssetGroupAssetError[AssetGroupAssetErrorEnum_AssetGroupAssetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetGroupAssetErrorEnum_AssetGroupAssetError[AssetGroupAssetErrorEnum_AssetGroupAssetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Cannot add duplicated asset group asset.
     *
     * @generated from protobuf enum value: DUPLICATE_RESOURCE = 2;
     */
    AssetGroupAssetErrorEnum_AssetGroupAssetError[AssetGroupAssetErrorEnum_AssetGroupAssetError["DUPLICATE_RESOURCE"] = 2] = "DUPLICATE_RESOURCE";
    /**
     * Expandable tags are not allowed in description assets.
     *
     * @generated from protobuf enum value: EXPANDABLE_TAGS_NOT_ALLOWED_IN_DESCRIPTION = 3;
     */
    AssetGroupAssetErrorEnum_AssetGroupAssetError[AssetGroupAssetErrorEnum_AssetGroupAssetError["EXPANDABLE_TAGS_NOT_ALLOWED_IN_DESCRIPTION"] = 3] = "EXPANDABLE_TAGS_NOT_ALLOWED_IN_DESCRIPTION";
    /**
     * Ad customizers are not supported in assetgroup's text assets.
     *
     * @generated from protobuf enum value: AD_CUSTOMIZER_NOT_SUPPORTED = 4;
     */
    AssetGroupAssetErrorEnum_AssetGroupAssetError[AssetGroupAssetErrorEnum_AssetGroupAssetError["AD_CUSTOMIZER_NOT_SUPPORTED"] = 4] = "AD_CUSTOMIZER_NOT_SUPPORTED";
})(AssetGroupAssetErrorEnum_AssetGroupAssetError = exports.AssetGroupAssetErrorEnum_AssetGroupAssetError || (exports.AssetGroupAssetErrorEnum_AssetGroupAssetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetGroupAssetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AssetGroupAssetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AssetGroupAssetErrorEnum
 */
exports.AssetGroupAssetErrorEnum = new AssetGroupAssetErrorEnum$Type();
