// google-ads-api-client/dist/protos/bindings/ads/errors/asset_link_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetLinkErrorEnum = exports.AssetLinkErrorEnum_AssetLinkError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible asset link errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AssetLinkErrorEnum.AssetLinkError
 */
var AssetLinkErrorEnum_AssetLinkError;
(function (AssetLinkErrorEnum_AssetLinkError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Pinning is not supported for the given asset link field.
     *
     * @generated from protobuf enum value: PINNING_UNSUPPORTED = 2;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["PINNING_UNSUPPORTED"] = 2] = "PINNING_UNSUPPORTED";
    /**
     * The given field type is not supported to be added directly through asset
     * links.
     *
     * @generated from protobuf enum value: UNSUPPORTED_FIELD_TYPE = 3;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["UNSUPPORTED_FIELD_TYPE"] = 3] = "UNSUPPORTED_FIELD_TYPE";
    /**
     * The given asset's type and the specified field type are incompatible.
     *
     * @generated from protobuf enum value: FIELD_TYPE_INCOMPATIBLE_WITH_ASSET_TYPE = 4;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["FIELD_TYPE_INCOMPATIBLE_WITH_ASSET_TYPE"] = 4] = "FIELD_TYPE_INCOMPATIBLE_WITH_ASSET_TYPE";
    /**
     * The specified field type is incompatible with the given campaign type.
     *
     * @generated from protobuf enum value: FIELD_TYPE_INCOMPATIBLE_WITH_CAMPAIGN_TYPE = 5;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["FIELD_TYPE_INCOMPATIBLE_WITH_CAMPAIGN_TYPE"] = 5] = "FIELD_TYPE_INCOMPATIBLE_WITH_CAMPAIGN_TYPE";
    /**
     * The campaign advertising channel type cannot be associated with the given
     * asset due to channel-based restrictions on the asset's fields.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_ADVERTISING_CHANNEL_TYPE = 6;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["INCOMPATIBLE_ADVERTISING_CHANNEL_TYPE"] = 6] = "INCOMPATIBLE_ADVERTISING_CHANNEL_TYPE";
    /**
     * The image asset provided is not within the dimension constraints
     * specified for the submitted asset field.
     *
     * @generated from protobuf enum value: IMAGE_NOT_WITHIN_SPECIFIED_DIMENSION_RANGE = 7;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["IMAGE_NOT_WITHIN_SPECIFIED_DIMENSION_RANGE"] = 7] = "IMAGE_NOT_WITHIN_SPECIFIED_DIMENSION_RANGE";
    /**
     * The pinned field is not valid for the submitted asset field.
     *
     * @generated from protobuf enum value: INVALID_PINNED_FIELD = 8;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["INVALID_PINNED_FIELD"] = 8] = "INVALID_PINNED_FIELD";
    /**
     * The media bundle asset provided is too large for the submitted asset
     * field.
     *
     * @generated from protobuf enum value: MEDIA_BUNDLE_ASSET_FILE_SIZE_TOO_LARGE = 9;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["MEDIA_BUNDLE_ASSET_FILE_SIZE_TOO_LARGE"] = 9] = "MEDIA_BUNDLE_ASSET_FILE_SIZE_TOO_LARGE";
    /**
     * Not enough assets are available for use with other fields since other
     * assets are pinned to specific fields.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_AVAILABLE_ASSET_LINKS_FOR_VALID_COMBINATION = 10;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["NOT_ENOUGH_AVAILABLE_ASSET_LINKS_FOR_VALID_COMBINATION"] = 10] = "NOT_ENOUGH_AVAILABLE_ASSET_LINKS_FOR_VALID_COMBINATION";
    /**
     * Not enough assets with fallback are available. When validating the
     * minimum number of assets, assets without fallback (for example, assets
     * that contain location tag without default value "{LOCATION(City)}") will
     * not be counted.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_AVAILABLE_ASSET_LINKS_WITH_FALLBACK = 11;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["NOT_ENOUGH_AVAILABLE_ASSET_LINKS_WITH_FALLBACK"] = 11] = "NOT_ENOUGH_AVAILABLE_ASSET_LINKS_WITH_FALLBACK";
    /**
     * This is a combination of the
     * NOT_ENOUGH_AVAILABLE_ASSET_LINKS_FOR_VALID_COMBINATION and
     * NOT_ENOUGH_AVAILABLE_ASSET_LINKS_WITH_FALLBACK errors. Not enough assets
     * with fallback are available since some assets are pinned.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_AVAILABLE_ASSET_LINKS_WITH_FALLBACK_FOR_VALID_COMBINATION = 12;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["NOT_ENOUGH_AVAILABLE_ASSET_LINKS_WITH_FALLBACK_FOR_VALID_COMBINATION"] = 12] = "NOT_ENOUGH_AVAILABLE_ASSET_LINKS_WITH_FALLBACK_FOR_VALID_COMBINATION";
    /**
     * The YouTube video referenced in the provided asset has been removed.
     *
     * @generated from protobuf enum value: YOUTUBE_VIDEO_REMOVED = 13;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["YOUTUBE_VIDEO_REMOVED"] = 13] = "YOUTUBE_VIDEO_REMOVED";
    /**
     * The YouTube video referenced in the provided asset is too long for the
     * field submitted.
     *
     * @generated from protobuf enum value: YOUTUBE_VIDEO_TOO_LONG = 14;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["YOUTUBE_VIDEO_TOO_LONG"] = 14] = "YOUTUBE_VIDEO_TOO_LONG";
    /**
     * The YouTube video referenced in the provided asset is too short for the
     * field submitted.
     *
     * @generated from protobuf enum value: YOUTUBE_VIDEO_TOO_SHORT = 15;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["YOUTUBE_VIDEO_TOO_SHORT"] = 15] = "YOUTUBE_VIDEO_TOO_SHORT";
    /**
     * The specified field type is excluded for given campaign or ad group.
     *
     * @generated from protobuf enum value: EXCLUDED_PARENT_FIELD_TYPE = 16;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["EXCLUDED_PARENT_FIELD_TYPE"] = 16] = "EXCLUDED_PARENT_FIELD_TYPE";
    /**
     * The status is invalid for the operation specified.
     *
     * @generated from protobuf enum value: INVALID_STATUS = 17;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["INVALID_STATUS"] = 17] = "INVALID_STATUS";
    /**
     * The YouTube video referenced in the provided asset has unknown duration.
     * This might be the case for a livestream video or a video being currently
     * uploaded to YouTube. In both cases, the video duration should eventually
     * get resolved.
     *
     * @generated from protobuf enum value: YOUTUBE_VIDEO_DURATION_NOT_DEFINED = 18;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["YOUTUBE_VIDEO_DURATION_NOT_DEFINED"] = 18] = "YOUTUBE_VIDEO_DURATION_NOT_DEFINED";
    /**
     * User cannot create automatically created links.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_AUTOMATICALLY_CREATED_LINKS = 19;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["CANNOT_CREATE_AUTOMATICALLY_CREATED_LINKS"] = 19] = "CANNOT_CREATE_AUTOMATICALLY_CREATED_LINKS";
    /**
     * Advertiser links cannot link to automatically created asset.
     *
     * @generated from protobuf enum value: CANNOT_LINK_TO_AUTOMATICALLY_CREATED_ASSET = 20;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["CANNOT_LINK_TO_AUTOMATICALLY_CREATED_ASSET"] = 20] = "CANNOT_LINK_TO_AUTOMATICALLY_CREATED_ASSET";
    /**
     * Automatically created links cannot be changed into adveritser links or
     * the reverse.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_ASSET_LINK_SOURCE = 21;
     */
    AssetLinkErrorEnum_AssetLinkError[AssetLinkErrorEnum_AssetLinkError["CANNOT_MODIFY_ASSET_LINK_SOURCE"] = 21] = "CANNOT_MODIFY_ASSET_LINK_SOURCE";
})(AssetLinkErrorEnum_AssetLinkError = exports.AssetLinkErrorEnum_AssetLinkError || (exports.AssetLinkErrorEnum_AssetLinkError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetLinkErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AssetLinkErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AssetLinkErrorEnum
 */
exports.AssetLinkErrorEnum = new AssetLinkErrorEnum$Type();
