// google-ads-api-client/dist/protos/bindings/ads/errors/asset_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetErrorEnum = exports.AssetErrorEnum_AssetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible asset errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AssetErrorEnum.AssetError
 */
var AssetErrorEnum_AssetError;
(function (AssetErrorEnum_AssetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The customer is not is not on the allow-list for this asset type.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_ON_ALLOWLIST_FOR_ASSET_TYPE = 13;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CUSTOMER_NOT_ON_ALLOWLIST_FOR_ASSET_TYPE"] = 13] = "CUSTOMER_NOT_ON_ALLOWLIST_FOR_ASSET_TYPE";
    /**
     * Assets are duplicated across operations.
     *
     * @generated from protobuf enum value: DUPLICATE_ASSET = 3;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["DUPLICATE_ASSET"] = 3] = "DUPLICATE_ASSET";
    /**
     * The asset name is duplicated, either across operations or with an
     * existing asset.
     *
     * @generated from protobuf enum value: DUPLICATE_ASSET_NAME = 4;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["DUPLICATE_ASSET_NAME"] = 4] = "DUPLICATE_ASSET_NAME";
    /**
     * The Asset.asset_data oneof is empty.
     *
     * @generated from protobuf enum value: ASSET_DATA_IS_MISSING = 5;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["ASSET_DATA_IS_MISSING"] = 5] = "ASSET_DATA_IS_MISSING";
    /**
     * The asset has a name which is different from an existing duplicate that
     * represents the same content.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_ASSET_NAME = 6;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CANNOT_MODIFY_ASSET_NAME"] = 6] = "CANNOT_MODIFY_ASSET_NAME";
    /**
     * The field cannot be set for this asset type.
     *
     * @generated from protobuf enum value: FIELD_INCOMPATIBLE_WITH_ASSET_TYPE = 7;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["FIELD_INCOMPATIBLE_WITH_ASSET_TYPE"] = 7] = "FIELD_INCOMPATIBLE_WITH_ASSET_TYPE";
    /**
     * Call to action must come from the list of supported values.
     *
     * @generated from protobuf enum value: INVALID_CALL_TO_ACTION_TEXT = 8;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["INVALID_CALL_TO_ACTION_TEXT"] = 8] = "INVALID_CALL_TO_ACTION_TEXT";
    /**
     * A lead form asset is created with an invalid combination of input fields.
     *
     * @generated from protobuf enum value: LEAD_FORM_INVALID_FIELDS_COMBINATION = 9;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["LEAD_FORM_INVALID_FIELDS_COMBINATION"] = 9] = "LEAD_FORM_INVALID_FIELDS_COMBINATION";
    /**
     * Lead forms require that the Terms of Service have been agreed to before
     * mutates can be executed.
     *
     * @generated from protobuf enum value: LEAD_FORM_MISSING_AGREEMENT = 10;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["LEAD_FORM_MISSING_AGREEMENT"] = 10] = "LEAD_FORM_MISSING_AGREEMENT";
    /**
     * Asset status is invalid in this operation.
     *
     * @generated from protobuf enum value: INVALID_ASSET_STATUS = 11;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["INVALID_ASSET_STATUS"] = 11] = "INVALID_ASSET_STATUS";
    /**
     * The field cannot be modified by this asset type.
     *
     * @generated from protobuf enum value: FIELD_CANNOT_BE_MODIFIED_FOR_ASSET_TYPE = 12;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["FIELD_CANNOT_BE_MODIFIED_FOR_ASSET_TYPE"] = 12] = "FIELD_CANNOT_BE_MODIFIED_FOR_ASSET_TYPE";
    /**
     * Ad schedules for the same asset cannot overlap.
     *
     * @generated from protobuf enum value: SCHEDULES_CANNOT_OVERLAP = 14;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["SCHEDULES_CANNOT_OVERLAP"] = 14] = "SCHEDULES_CANNOT_OVERLAP";
    /**
     * Cannot set both percent off and money amount off fields of promotion
     * asset.
     *
     * @generated from protobuf enum value: PROMOTION_CANNOT_SET_PERCENT_OFF_AND_MONEY_AMOUNT_OFF = 15;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["PROMOTION_CANNOT_SET_PERCENT_OFF_AND_MONEY_AMOUNT_OFF"] = 15] = "PROMOTION_CANNOT_SET_PERCENT_OFF_AND_MONEY_AMOUNT_OFF";
    /**
     * Cannot set both promotion code and orders over amount fields of promotion
     * asset.
     *
     * @generated from protobuf enum value: PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT = 16;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT"] = 16] = "PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT";
    /**
     * The field has too many decimal places specified.
     *
     * @generated from protobuf enum value: TOO_MANY_DECIMAL_PLACES_SPECIFIED = 17;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["TOO_MANY_DECIMAL_PLACES_SPECIFIED"] = 17] = "TOO_MANY_DECIMAL_PLACES_SPECIFIED";
    /**
     * Duplicate assets across operations, which have identical Asset.asset_data
     * oneof, cannot have different asset level fields for asset types which are
     * deduped.
     *
     * @generated from protobuf enum value: DUPLICATE_ASSETS_WITH_DIFFERENT_FIELD_VALUE = 18;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["DUPLICATE_ASSETS_WITH_DIFFERENT_FIELD_VALUE"] = 18] = "DUPLICATE_ASSETS_WITH_DIFFERENT_FIELD_VALUE";
    /**
     * Carrier specific short number is not allowed.
     *
     * @generated from protobuf enum value: CALL_CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED = 19;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED"] = 19] = "CALL_CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED";
    /**
     * Customer consent required for call recording Terms of Service.
     *
     * @generated from protobuf enum value: CALL_CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED = 20;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED"] = 20] = "CALL_CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED";
    /**
     * The type of the specified phone number is not allowed.
     *
     * @generated from protobuf enum value: CALL_DISALLOWED_NUMBER_TYPE = 21;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_DISALLOWED_NUMBER_TYPE"] = 21] = "CALL_DISALLOWED_NUMBER_TYPE";
    /**
     * If the default call_conversion_action is not used, the customer must have
     * a ConversionAction with the same id and the ConversionAction must be call
     * conversion type.
     *
     * @generated from protobuf enum value: CALL_INVALID_CONVERSION_ACTION = 22;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_INVALID_CONVERSION_ACTION"] = 22] = "CALL_INVALID_CONVERSION_ACTION";
    /**
     * The country code of the phone number is invalid.
     *
     * @generated from protobuf enum value: CALL_INVALID_COUNTRY_CODE = 23;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_INVALID_COUNTRY_CODE"] = 23] = "CALL_INVALID_COUNTRY_CODE";
    /**
     * The format of the phone number is incorrect.
     *
     * @generated from protobuf enum value: CALL_INVALID_DOMESTIC_PHONE_NUMBER_FORMAT = 24;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_INVALID_DOMESTIC_PHONE_NUMBER_FORMAT"] = 24] = "CALL_INVALID_DOMESTIC_PHONE_NUMBER_FORMAT";
    /**
     * The input phone number is not a valid phone number.
     *
     * @generated from protobuf enum value: CALL_INVALID_PHONE_NUMBER = 25;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_INVALID_PHONE_NUMBER"] = 25] = "CALL_INVALID_PHONE_NUMBER";
    /**
     * The phone number is not supported for country.
     *
     * @generated from protobuf enum value: CALL_PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY = 26;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY"] = 26] = "CALL_PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * Premium rate phone number is not allowed.
     *
     * @generated from protobuf enum value: CALL_PREMIUM_RATE_NUMBER_NOT_ALLOWED = 27;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_PREMIUM_RATE_NUMBER_NOT_ALLOWED"] = 27] = "CALL_PREMIUM_RATE_NUMBER_NOT_ALLOWED";
    /**
     * Vanity phone number is not allowed.
     *
     * @generated from protobuf enum value: CALL_VANITY_PHONE_NUMBER_NOT_ALLOWED = 28;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CALL_VANITY_PHONE_NUMBER_NOT_ALLOWED"] = 28] = "CALL_VANITY_PHONE_NUMBER_NOT_ALLOWED";
    /**
     * PriceOffering cannot have the same value for header and description.
     *
     * @generated from protobuf enum value: PRICE_HEADER_SAME_AS_DESCRIPTION = 29;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["PRICE_HEADER_SAME_AS_DESCRIPTION"] = 29] = "PRICE_HEADER_SAME_AS_DESCRIPTION";
    /**
     * AppId is invalid.
     *
     * @generated from protobuf enum value: MOBILE_APP_INVALID_APP_ID = 30;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["MOBILE_APP_INVALID_APP_ID"] = 30] = "MOBILE_APP_INVALID_APP_ID";
    /**
     * Invalid App download URL in final URLs.
     *
     * @generated from protobuf enum value: MOBILE_APP_INVALID_FINAL_URL_FOR_APP_DOWNLOAD_URL = 31;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["MOBILE_APP_INVALID_FINAL_URL_FOR_APP_DOWNLOAD_URL"] = 31] = "MOBILE_APP_INVALID_FINAL_URL_FOR_APP_DOWNLOAD_URL";
    /**
     * Asset name is required for the asset type.
     *
     * @generated from protobuf enum value: NAME_REQUIRED_FOR_ASSET_TYPE = 32;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["NAME_REQUIRED_FOR_ASSET_TYPE"] = 32] = "NAME_REQUIRED_FOR_ASSET_TYPE";
    /**
     * Legacy qualifying questions cannot be in the same Lead Form as
     * custom questions.
     *
     * @generated from protobuf enum value: LEAD_FORM_LEGACY_QUALIFYING_QUESTIONS_DISALLOWED = 33;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["LEAD_FORM_LEGACY_QUALIFYING_QUESTIONS_DISALLOWED"] = 33] = "LEAD_FORM_LEGACY_QUALIFYING_QUESTIONS_DISALLOWED";
    /**
     * Unique name is required for this asset type.
     *
     * @generated from protobuf enum value: NAME_CONFLICT_FOR_ASSET_TYPE = 34;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["NAME_CONFLICT_FOR_ASSET_TYPE"] = 34] = "NAME_CONFLICT_FOR_ASSET_TYPE";
    /**
     * Cannot modify asset source.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_ASSET_SOURCE = 35;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CANNOT_MODIFY_ASSET_SOURCE"] = 35] = "CANNOT_MODIFY_ASSET_SOURCE";
    /**
     * User can not modify the automatically created asset.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_AUTOMATICALLY_CREATED_ASSET = 36;
     */
    AssetErrorEnum_AssetError[AssetErrorEnum_AssetError["CANNOT_MODIFY_AUTOMATICALLY_CREATED_ASSET"] = 36] = "CANNOT_MODIFY_AUTOMATICALLY_CREATED_ASSET";
})(AssetErrorEnum_AssetError = exports.AssetErrorEnum_AssetError || (exports.AssetErrorEnum_AssetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AssetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AssetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AssetErrorEnum
 */
exports.AssetErrorEnum = new AssetErrorEnum$Type();
