// google-ads-api-client/dist/protos/bindings/ads/errors/audience_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AudienceErrorEnum = exports.AudienceErrorEnum_AudienceError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible audience errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AudienceErrorEnum.AudienceError
 */
var AudienceErrorEnum_AudienceError;
(function (AudienceErrorEnum_AudienceError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An audience with this name already exists.
     *
     * @generated from protobuf enum value: NAME_ALREADY_IN_USE = 2;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["NAME_ALREADY_IN_USE"] = 2] = "NAME_ALREADY_IN_USE";
    /**
     * A dimension within the audience definition is not valid.
     *
     * @generated from protobuf enum value: DIMENSION_INVALID = 3;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["DIMENSION_INVALID"] = 3] = "DIMENSION_INVALID";
    /**
     * One of the audience segment added is not found.
     *
     * @generated from protobuf enum value: AUDIENCE_SEGMENT_NOT_FOUND = 4;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["AUDIENCE_SEGMENT_NOT_FOUND"] = 4] = "AUDIENCE_SEGMENT_NOT_FOUND";
    /**
     * One of the audience segment type is not supported.
     *
     * @generated from protobuf enum value: AUDIENCE_SEGMENT_TYPE_NOT_SUPPORTED = 5;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["AUDIENCE_SEGMENT_TYPE_NOT_SUPPORTED"] = 5] = "AUDIENCE_SEGMENT_TYPE_NOT_SUPPORTED";
    /**
     * The same segment already exists in this audience.
     *
     * @generated from protobuf enum value: DUPLICATE_AUDIENCE_SEGMENT = 6;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["DUPLICATE_AUDIENCE_SEGMENT"] = 6] = "DUPLICATE_AUDIENCE_SEGMENT";
    /**
     * Audience can't have more than allowed number segments.
     *
     * @generated from protobuf enum value: TOO_MANY_SEGMENTS = 7;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["TOO_MANY_SEGMENTS"] = 7] = "TOO_MANY_SEGMENTS";
    /**
     * Audience can't have multiple dimensions of same type.
     *
     * @generated from protobuf enum value: TOO_MANY_DIMENSIONS_OF_SAME_TYPE = 8;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["TOO_MANY_DIMENSIONS_OF_SAME_TYPE"] = 8] = "TOO_MANY_DIMENSIONS_OF_SAME_TYPE";
    /**
     * The audience cannot be removed, because it is currently used in an
     * ad group criterion or asset group signal in an (enabled or paused)
     * ad group or campaign.
     *
     * @generated from protobuf enum value: IN_USE = 9;
     */
    AudienceErrorEnum_AudienceError[AudienceErrorEnum_AudienceError["IN_USE"] = 9] = "IN_USE";
})(AudienceErrorEnum_AudienceError = exports.AudienceErrorEnum_AudienceError || (exports.AudienceErrorEnum_AudienceError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AudienceErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AudienceErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AudienceErrorEnum
 */
exports.AudienceErrorEnum = new AudienceErrorEnum$Type();
