// google-ads-api-client/dist/protos/bindings/ads/errors/audience_insights_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AudienceInsightsErrorEnum = exports.AudienceInsightsErrorEnum_AudienceInsightsError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible errors from AudienceInsightsService.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AudienceInsightsErrorEnum.AudienceInsightsError
 */
var AudienceInsightsErrorEnum_AudienceInsightsError;
(function (AudienceInsightsErrorEnum_AudienceInsightsError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AudienceInsightsErrorEnum_AudienceInsightsError[AudienceInsightsErrorEnum_AudienceInsightsError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AudienceInsightsErrorEnum_AudienceInsightsError[AudienceInsightsErrorEnum_AudienceInsightsError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The dimensions cannot be used with topic audience combinations.
     *
     * @generated from protobuf enum value: DIMENSION_INCOMPATIBLE_WITH_TOPIC_AUDIENCE_COMBINATIONS = 2;
     */
    AudienceInsightsErrorEnum_AudienceInsightsError[AudienceInsightsErrorEnum_AudienceInsightsError["DIMENSION_INCOMPATIBLE_WITH_TOPIC_AUDIENCE_COMBINATIONS"] = 2] = "DIMENSION_INCOMPATIBLE_WITH_TOPIC_AUDIENCE_COMBINATIONS";
})(AudienceInsightsErrorEnum_AudienceInsightsError = exports.AudienceInsightsErrorEnum_AudienceInsightsError || (exports.AudienceInsightsErrorEnum_AudienceInsightsError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AudienceInsightsErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AudienceInsightsErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AudienceInsightsErrorEnum
 */
exports.AudienceInsightsErrorEnum = new AudienceInsightsErrorEnum$Type();
