// google-ads-api-client/dist/protos/bindings/ads/errors/authorization_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizationErrorEnum = exports.AuthorizationErrorEnum_AuthorizationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible authorization errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AuthorizationErrorEnum.AuthorizationError
 */
var AuthorizationErrorEnum_AuthorizationError;
(function (AuthorizationErrorEnum_AuthorizationError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * User doesn't have permission to access customer. Note: If you're
     * accessing a client customer, the manager's customer ID must be set in the
     * `login-customer-id` header. Learn more at
     * https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid
     *
     * @generated from protobuf enum value: USER_PERMISSION_DENIED = 2;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["USER_PERMISSION_DENIED"] = 2] = "USER_PERMISSION_DENIED";
    /**
     * The developer token is not on the allow-list.
     *
     * @generated from protobuf enum value: DEVELOPER_TOKEN_NOT_ON_ALLOWLIST = 13;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["DEVELOPER_TOKEN_NOT_ON_ALLOWLIST"] = 13] = "DEVELOPER_TOKEN_NOT_ON_ALLOWLIST";
    /**
     * The developer token is not allowed with the project sent in the request.
     *
     * @generated from protobuf enum value: DEVELOPER_TOKEN_PROHIBITED = 4;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["DEVELOPER_TOKEN_PROHIBITED"] = 4] = "DEVELOPER_TOKEN_PROHIBITED";
    /**
     * The Google Cloud project sent in the request does not have permission to
     * access the api.
     *
     * @generated from protobuf enum value: PROJECT_DISABLED = 5;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["PROJECT_DISABLED"] = 5] = "PROJECT_DISABLED";
    /**
     * Authorization of the client failed.
     *
     * @generated from protobuf enum value: AUTHORIZATION_ERROR = 6;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["AUTHORIZATION_ERROR"] = 6] = "AUTHORIZATION_ERROR";
    /**
     * The user does not have permission to perform this action
     * (for example, ADD, UPDATE, REMOVE) on the resource or call a method.
     *
     * @generated from protobuf enum value: ACTION_NOT_PERMITTED = 7;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["ACTION_NOT_PERMITTED"] = 7] = "ACTION_NOT_PERMITTED";
    /**
     * Signup not complete.
     *
     * @generated from protobuf enum value: INCOMPLETE_SIGNUP = 8;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["INCOMPLETE_SIGNUP"] = 8] = "INCOMPLETE_SIGNUP";
    /**
     * The customer can't be used because it isn't enabled.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_ENABLED = 24;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["CUSTOMER_NOT_ENABLED"] = 24] = "CUSTOMER_NOT_ENABLED";
    /**
     * The developer must sign the terms of service. They can be found here:
     * ads.google.com/aw/apicenter
     *
     * @generated from protobuf enum value: MISSING_TOS = 9;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["MISSING_TOS"] = 9] = "MISSING_TOS";
    /**
     * The developer token is not approved. Non-approved developer tokens can
     * only be used with test accounts.
     *
     * @generated from protobuf enum value: DEVELOPER_TOKEN_NOT_APPROVED = 10;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["DEVELOPER_TOKEN_NOT_APPROVED"] = 10] = "DEVELOPER_TOKEN_NOT_APPROVED";
    /**
     * The login customer specified does not have access to the account
     * specified, so the request is invalid.
     *
     * @generated from protobuf enum value: INVALID_LOGIN_CUSTOMER_ID_SERVING_CUSTOMER_ID_COMBINATION = 11;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["INVALID_LOGIN_CUSTOMER_ID_SERVING_CUSTOMER_ID_COMBINATION"] = 11] = "INVALID_LOGIN_CUSTOMER_ID_SERVING_CUSTOMER_ID_COMBINATION";
    /**
     * The developer specified does not have access to the service.
     *
     * @generated from protobuf enum value: SERVICE_ACCESS_DENIED = 12;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["SERVICE_ACCESS_DENIED"] = 12] = "SERVICE_ACCESS_DENIED";
    /**
     * The customer (or login customer) isn't in Google Ads. It belongs to
     * another ads system.
     *
     * @generated from protobuf enum value: ACCESS_DENIED_FOR_ACCOUNT_TYPE = 25;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["ACCESS_DENIED_FOR_ACCOUNT_TYPE"] = 25] = "ACCESS_DENIED_FOR_ACCOUNT_TYPE";
    /**
     * The developer does not have access to the metrics queried.
     *
     * @generated from protobuf enum value: METRIC_ACCESS_DENIED = 26;
     */
    AuthorizationErrorEnum_AuthorizationError[AuthorizationErrorEnum_AuthorizationError["METRIC_ACCESS_DENIED"] = 26] = "METRIC_ACCESS_DENIED";
})(AuthorizationErrorEnum_AuthorizationError = exports.AuthorizationErrorEnum_AuthorizationError || (exports.AuthorizationErrorEnum_AuthorizationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AuthorizationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AuthorizationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AuthorizationErrorEnum
 */
exports.AuthorizationErrorEnum = new AuthorizationErrorEnum$Type();
