// google-ads-api-client/dist/protos/bindings/ads/errors/authentication_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationErrorEnum = exports.AuthenticationErrorEnum_AuthenticationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible authentication errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.AuthenticationErrorEnum.AuthenticationError
 */
var AuthenticationErrorEnum_AuthenticationError;
(function (AuthenticationErrorEnum_AuthenticationError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Authentication of the request failed.
     *
     * @generated from protobuf enum value: AUTHENTICATION_ERROR = 2;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["AUTHENTICATION_ERROR"] = 2] = "AUTHENTICATION_ERROR";
    /**
     * Client customer ID is not a number.
     *
     * @generated from protobuf enum value: CLIENT_CUSTOMER_ID_INVALID = 5;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["CLIENT_CUSTOMER_ID_INVALID"] = 5] = "CLIENT_CUSTOMER_ID_INVALID";
    /**
     * No customer found for the provided customer ID.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_FOUND = 8;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["CUSTOMER_NOT_FOUND"] = 8] = "CUSTOMER_NOT_FOUND";
    /**
     * Client's Google account is deleted.
     *
     * @generated from protobuf enum value: GOOGLE_ACCOUNT_DELETED = 9;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["GOOGLE_ACCOUNT_DELETED"] = 9] = "GOOGLE_ACCOUNT_DELETED";
    /**
     * Google account login token in the cookie is invalid.
     *
     * @generated from protobuf enum value: GOOGLE_ACCOUNT_COOKIE_INVALID = 10;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["GOOGLE_ACCOUNT_COOKIE_INVALID"] = 10] = "GOOGLE_ACCOUNT_COOKIE_INVALID";
    /**
     * A problem occurred during Google account authentication.
     *
     * @generated from protobuf enum value: GOOGLE_ACCOUNT_AUTHENTICATION_FAILED = 25;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["GOOGLE_ACCOUNT_AUTHENTICATION_FAILED"] = 25] = "GOOGLE_ACCOUNT_AUTHENTICATION_FAILED";
    /**
     * The user in the Google account login token does not match the user ID in
     * the cookie.
     *
     * @generated from protobuf enum value: GOOGLE_ACCOUNT_USER_AND_ADS_USER_MISMATCH = 12;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["GOOGLE_ACCOUNT_USER_AND_ADS_USER_MISMATCH"] = 12] = "GOOGLE_ACCOUNT_USER_AND_ADS_USER_MISMATCH";
    /**
     * Login cookie is required for authentication.
     *
     * @generated from protobuf enum value: LOGIN_COOKIE_REQUIRED = 13;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["LOGIN_COOKIE_REQUIRED"] = 13] = "LOGIN_COOKIE_REQUIRED";
    /**
     * User in the cookie is not a valid Ads user.
     *
     * @generated from protobuf enum value: NOT_ADS_USER = 14;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["NOT_ADS_USER"] = 14] = "NOT_ADS_USER";
    /**
     * OAuth token in the header is not valid.
     *
     * @generated from protobuf enum value: OAUTH_TOKEN_INVALID = 15;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["OAUTH_TOKEN_INVALID"] = 15] = "OAUTH_TOKEN_INVALID";
    /**
     * OAuth token in the header has expired.
     *
     * @generated from protobuf enum value: OAUTH_TOKEN_EXPIRED = 16;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["OAUTH_TOKEN_EXPIRED"] = 16] = "OAUTH_TOKEN_EXPIRED";
    /**
     * OAuth token in the header has been disabled.
     *
     * @generated from protobuf enum value: OAUTH_TOKEN_DISABLED = 17;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["OAUTH_TOKEN_DISABLED"] = 17] = "OAUTH_TOKEN_DISABLED";
    /**
     * OAuth token in the header has been revoked.
     *
     * @generated from protobuf enum value: OAUTH_TOKEN_REVOKED = 18;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["OAUTH_TOKEN_REVOKED"] = 18] = "OAUTH_TOKEN_REVOKED";
    /**
     * OAuth token HTTP header is malformed.
     *
     * @generated from protobuf enum value: OAUTH_TOKEN_HEADER_INVALID = 19;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["OAUTH_TOKEN_HEADER_INVALID"] = 19] = "OAUTH_TOKEN_HEADER_INVALID";
    /**
     * Login cookie is not valid.
     *
     * @generated from protobuf enum value: LOGIN_COOKIE_INVALID = 20;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["LOGIN_COOKIE_INVALID"] = 20] = "LOGIN_COOKIE_INVALID";
    /**
     * User ID in the header is not a valid ID.
     *
     * @generated from protobuf enum value: USER_ID_INVALID = 22;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["USER_ID_INVALID"] = 22] = "USER_ID_INVALID";
    /**
     * An account administrator changed this account's authentication settings.
     * To access this Google Ads account, enable 2-Step Verification in your
     * Google account at https://www.google.com/landing/2step.
     *
     * @generated from protobuf enum value: TWO_STEP_VERIFICATION_NOT_ENROLLED = 23;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["TWO_STEP_VERIFICATION_NOT_ENROLLED"] = 23] = "TWO_STEP_VERIFICATION_NOT_ENROLLED";
    /**
     * An account administrator changed this account's authentication settings.
     * To access this Google Ads account, enable Advanced Protection in your
     * Google account at https://landing.google.com/advancedprotection.
     *
     * @generated from protobuf enum value: ADVANCED_PROTECTION_NOT_ENROLLED = 24;
     */
    AuthenticationErrorEnum_AuthenticationError[AuthenticationErrorEnum_AuthenticationError["ADVANCED_PROTECTION_NOT_ENROLLED"] = 24] = "ADVANCED_PROTECTION_NOT_ENROLLED";
})(AuthenticationErrorEnum_AuthenticationError = exports.AuthenticationErrorEnum_AuthenticationError || (exports.AuthenticationErrorEnum_AuthenticationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class AuthenticationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.AuthenticationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.AuthenticationErrorEnum
 */
exports.AuthenticationErrorEnum = new AuthenticationErrorEnum$Type();
