// google-ads-api-client/dist/protos/bindings/ads/errors/bidding_strategy_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BiddingStrategyErrorEnum = exports.BiddingStrategyErrorEnum_BiddingStrategyError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible bidding strategy errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.BiddingStrategyErrorEnum.BiddingStrategyError
 */
var BiddingStrategyErrorEnum_BiddingStrategyError;
(function (BiddingStrategyErrorEnum_BiddingStrategyError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BiddingStrategyErrorEnum_BiddingStrategyError[BiddingStrategyErrorEnum_BiddingStrategyError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BiddingStrategyErrorEnum_BiddingStrategyError[BiddingStrategyErrorEnum_BiddingStrategyError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Each bidding strategy must have a unique name.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 2;
     */
    BiddingStrategyErrorEnum_BiddingStrategyError[BiddingStrategyErrorEnum_BiddingStrategyError["DUPLICATE_NAME"] = 2] = "DUPLICATE_NAME";
    /**
     * Bidding strategy type is immutable.
     *
     * @generated from protobuf enum value: CANNOT_CHANGE_BIDDING_STRATEGY_TYPE = 3;
     */
    BiddingStrategyErrorEnum_BiddingStrategyError[BiddingStrategyErrorEnum_BiddingStrategyError["CANNOT_CHANGE_BIDDING_STRATEGY_TYPE"] = 3] = "CANNOT_CHANGE_BIDDING_STRATEGY_TYPE";
    /**
     * Only bidding strategies not linked to campaigns, adgroups or adgroup
     * criteria can be removed.
     *
     * @generated from protobuf enum value: CANNOT_REMOVE_ASSOCIATED_STRATEGY = 4;
     */
    BiddingStrategyErrorEnum_BiddingStrategyError[BiddingStrategyErrorEnum_BiddingStrategyError["CANNOT_REMOVE_ASSOCIATED_STRATEGY"] = 4] = "CANNOT_REMOVE_ASSOCIATED_STRATEGY";
    /**
     * The specified bidding strategy is not supported.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_NOT_SUPPORTED = 5;
     */
    BiddingStrategyErrorEnum_BiddingStrategyError[BiddingStrategyErrorEnum_BiddingStrategyError["BIDDING_STRATEGY_NOT_SUPPORTED"] = 5] = "BIDDING_STRATEGY_NOT_SUPPORTED";
    /**
     * The bidding strategy is incompatible with the campaign's bidding
     * strategy goal type.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_BIDDING_STRATEGY_AND_BIDDING_STRATEGY_GOAL_TYPE = 6;
     */
    BiddingStrategyErrorEnum_BiddingStrategyError[BiddingStrategyErrorEnum_BiddingStrategyError["INCOMPATIBLE_BIDDING_STRATEGY_AND_BIDDING_STRATEGY_GOAL_TYPE"] = 6] = "INCOMPATIBLE_BIDDING_STRATEGY_AND_BIDDING_STRATEGY_GOAL_TYPE";
})(BiddingStrategyErrorEnum_BiddingStrategyError = exports.BiddingStrategyErrorEnum_BiddingStrategyError || (exports.BiddingStrategyErrorEnum_BiddingStrategyError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BiddingStrategyErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.BiddingStrategyErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.BiddingStrategyErrorEnum
 */
exports.BiddingStrategyErrorEnum = new BiddingStrategyErrorEnum$Type();
