// google-ads-api-client/dist/protos/bindings/ads/errors/bidding_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BiddingErrorEnum = exports.BiddingErrorEnum_BiddingError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible bidding errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.BiddingErrorEnum.BiddingError
 */
var BiddingErrorEnum_BiddingError;
(function (BiddingErrorEnum_BiddingError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Cannot transition to new bidding strategy.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_TRANSITION_NOT_ALLOWED = 2;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_TRANSITION_NOT_ALLOWED"] = 2] = "BIDDING_STRATEGY_TRANSITION_NOT_ALLOWED";
    /**
     * Cannot attach bidding strategy to campaign.
     *
     * @generated from protobuf enum value: CANNOT_ATTACH_BIDDING_STRATEGY_TO_CAMPAIGN = 7;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["CANNOT_ATTACH_BIDDING_STRATEGY_TO_CAMPAIGN"] = 7] = "CANNOT_ATTACH_BIDDING_STRATEGY_TO_CAMPAIGN";
    /**
     * Bidding strategy is not supported or cannot be used as anonymous.
     *
     * @generated from protobuf enum value: INVALID_ANONYMOUS_BIDDING_STRATEGY_TYPE = 10;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["INVALID_ANONYMOUS_BIDDING_STRATEGY_TYPE"] = 10] = "INVALID_ANONYMOUS_BIDDING_STRATEGY_TYPE";
    /**
     * The type does not match the named strategy's type.
     *
     * @generated from protobuf enum value: INVALID_BIDDING_STRATEGY_TYPE = 14;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["INVALID_BIDDING_STRATEGY_TYPE"] = 14] = "INVALID_BIDDING_STRATEGY_TYPE";
    /**
     * The bid is invalid.
     *
     * @generated from protobuf enum value: INVALID_BID = 17;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["INVALID_BID"] = 17] = "INVALID_BID";
    /**
     * Bidding strategy is not available for the account type.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_NOT_AVAILABLE_FOR_ACCOUNT_TYPE = 18;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_NOT_AVAILABLE_FOR_ACCOUNT_TYPE"] = 18] = "BIDDING_STRATEGY_NOT_AVAILABLE_FOR_ACCOUNT_TYPE";
    /**
     * Conversion tracking is not enabled in the campaign that has value-based
     * bidding transitions.
     *
     * @generated from protobuf enum value: CONVERSION_TRACKING_NOT_ENABLED = 19;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["CONVERSION_TRACKING_NOT_ENABLED"] = 19] = "CONVERSION_TRACKING_NOT_ENABLED";
    /**
     * Not enough conversions tracked for value-based bidding transitions.
     *
     * @generated from protobuf enum value: NOT_ENOUGH_CONVERSIONS = 20;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["NOT_ENOUGH_CONVERSIONS"] = 20] = "NOT_ENOUGH_CONVERSIONS";
    /**
     * Campaign can not be created with given bidding strategy. It can be
     * transitioned to the strategy, once eligible.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_CAMPAIGN_WITH_BIDDING_STRATEGY = 21;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["CANNOT_CREATE_CAMPAIGN_WITH_BIDDING_STRATEGY"] = 21] = "CANNOT_CREATE_CAMPAIGN_WITH_BIDDING_STRATEGY";
    /**
     * Cannot target content network only as campaign uses Page One Promoted
     * bidding strategy.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_CONTENT_NETWORK_ONLY_WITH_CAMPAIGN_LEVEL_POP_BIDDING_STRATEGY = 23;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["CANNOT_TARGET_CONTENT_NETWORK_ONLY_WITH_CAMPAIGN_LEVEL_POP_BIDDING_STRATEGY"] = 23] = "CANNOT_TARGET_CONTENT_NETWORK_ONLY_WITH_CAMPAIGN_LEVEL_POP_BIDDING_STRATEGY";
    /**
     * Budget Optimizer and Target Spend bidding strategies are not supported
     * for campaigns with AdSchedule targeting.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_NOT_SUPPORTED_WITH_AD_SCHEDULE = 24;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_NOT_SUPPORTED_WITH_AD_SCHEDULE"] = 24] = "BIDDING_STRATEGY_NOT_SUPPORTED_WITH_AD_SCHEDULE";
    /**
     * Pay per conversion is not available to all the customer, only few
     * customers on the allow-list can use this.
     *
     * @generated from protobuf enum value: PAY_PER_CONVERSION_NOT_AVAILABLE_FOR_CUSTOMER = 25;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["PAY_PER_CONVERSION_NOT_AVAILABLE_FOR_CUSTOMER"] = 25] = "PAY_PER_CONVERSION_NOT_AVAILABLE_FOR_CUSTOMER";
    /**
     * Pay per conversion is not allowed with Target CPA.
     *
     * @generated from protobuf enum value: PAY_PER_CONVERSION_NOT_ALLOWED_WITH_TARGET_CPA = 26;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["PAY_PER_CONVERSION_NOT_ALLOWED_WITH_TARGET_CPA"] = 26] = "PAY_PER_CONVERSION_NOT_ALLOWED_WITH_TARGET_CPA";
    /**
     * Cannot set bidding strategy to Manual CPM for search network only
     * campaigns.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_NOT_ALLOWED_FOR_SEARCH_ONLY_CAMPAIGNS = 27;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_NOT_ALLOWED_FOR_SEARCH_ONLY_CAMPAIGNS"] = 27] = "BIDDING_STRATEGY_NOT_ALLOWED_FOR_SEARCH_ONLY_CAMPAIGNS";
    /**
     * The bidding strategy is not supported for use in drafts or experiments.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_NOT_SUPPORTED_IN_DRAFTS_OR_EXPERIMENTS = 28;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_NOT_SUPPORTED_IN_DRAFTS_OR_EXPERIMENTS"] = 28] = "BIDDING_STRATEGY_NOT_SUPPORTED_IN_DRAFTS_OR_EXPERIMENTS";
    /**
     * Bidding strategy type does not support product type ad group criterion.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_TYPE_DOES_NOT_SUPPORT_PRODUCT_TYPE_ADGROUP_CRITERION = 29;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_TYPE_DOES_NOT_SUPPORT_PRODUCT_TYPE_ADGROUP_CRITERION"] = 29] = "BIDDING_STRATEGY_TYPE_DOES_NOT_SUPPORT_PRODUCT_TYPE_ADGROUP_CRITERION";
    /**
     * Bid amount is too small.
     *
     * @generated from protobuf enum value: BID_TOO_SMALL = 30;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BID_TOO_SMALL"] = 30] = "BID_TOO_SMALL";
    /**
     * Bid amount is too big.
     *
     * @generated from protobuf enum value: BID_TOO_BIG = 31;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BID_TOO_BIG"] = 31] = "BID_TOO_BIG";
    /**
     * Bid has too many fractional digit precision.
     *
     * @generated from protobuf enum value: BID_TOO_MANY_FRACTIONAL_DIGITS = 32;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BID_TOO_MANY_FRACTIONAL_DIGITS"] = 32] = "BID_TOO_MANY_FRACTIONAL_DIGITS";
    /**
     * Invalid domain name specified.
     *
     * @generated from protobuf enum value: INVALID_DOMAIN_NAME = 33;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["INVALID_DOMAIN_NAME"] = 33] = "INVALID_DOMAIN_NAME";
    /**
     * The field is not compatible with the payment mode.
     *
     * @generated from protobuf enum value: NOT_COMPATIBLE_WITH_PAYMENT_MODE = 34;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["NOT_COMPATIBLE_WITH_PAYMENT_MODE"] = 34] = "NOT_COMPATIBLE_WITH_PAYMENT_MODE";
    /**
     * The field is not compatible with the budget type.
     *
     * @generated from protobuf enum value: NOT_COMPATIBLE_WITH_BUDGET_TYPE = 35;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["NOT_COMPATIBLE_WITH_BUDGET_TYPE"] = 35] = "NOT_COMPATIBLE_WITH_BUDGET_TYPE";
    /**
     * The field is not compatible with the bidding strategy type.
     *
     * @generated from protobuf enum value: NOT_COMPATIBLE_WITH_BIDDING_STRATEGY_TYPE = 36;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["NOT_COMPATIBLE_WITH_BIDDING_STRATEGY_TYPE"] = 36] = "NOT_COMPATIBLE_WITH_BIDDING_STRATEGY_TYPE";
    /**
     * Bidding strategy type is incompatible with shared budget.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_TYPE_INCOMPATIBLE_WITH_SHARED_BUDGET = 37;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_TYPE_INCOMPATIBLE_WITH_SHARED_BUDGET"] = 37] = "BIDDING_STRATEGY_TYPE_INCOMPATIBLE_WITH_SHARED_BUDGET";
    /**
     * The attached bidding strategy and budget must be aligned with each other
     * if alignment is specified on either entity.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_AND_BUDGET_MUST_BE_ALIGNED = 38;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_AND_BUDGET_MUST_BE_ALIGNED"] = 38] = "BIDDING_STRATEGY_AND_BUDGET_MUST_BE_ALIGNED";
    /**
     * The attached bidding strategy and budget must be attached to the same
     * campaigns to become aligned.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_AND_BUDGET_MUST_BE_ATTACHED_TO_THE_SAME_CAMPAIGNS_TO_ALIGN = 39;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_AND_BUDGET_MUST_BE_ATTACHED_TO_THE_SAME_CAMPAIGNS_TO_ALIGN"] = 39] = "BIDDING_STRATEGY_AND_BUDGET_MUST_BE_ATTACHED_TO_THE_SAME_CAMPAIGNS_TO_ALIGN";
    /**
     * The aligned bidding strategy and budget must be removed at the same time.
     *
     * @generated from protobuf enum value: BIDDING_STRATEGY_AND_BUDGET_MUST_BE_REMOVED_TOGETHER = 40;
     */
    BiddingErrorEnum_BiddingError[BiddingErrorEnum_BiddingError["BIDDING_STRATEGY_AND_BUDGET_MUST_BE_REMOVED_TOGETHER"] = 40] = "BIDDING_STRATEGY_AND_BUDGET_MUST_BE_REMOVED_TOGETHER";
})(BiddingErrorEnum_BiddingError = exports.BiddingErrorEnum_BiddingError || (exports.BiddingErrorEnum_BiddingError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BiddingErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.BiddingErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.BiddingErrorEnum
 */
exports.BiddingErrorEnum = new BiddingErrorEnum$Type();
