// google-ads-api-client/dist/protos/bindings/ads/errors/billing_setup_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingSetupErrorEnum = exports.BillingSetupErrorEnum_BillingSetupError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible billing setup errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.BillingSetupErrorEnum.BillingSetupError
 */
var BillingSetupErrorEnum_BillingSetupError;
(function (BillingSetupErrorEnum_BillingSetupError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Cannot specify both an existing payments account and a new payments
     * account when setting up billing.
     *
     * @generated from protobuf enum value: CANNOT_USE_EXISTING_AND_NEW_ACCOUNT = 2;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["CANNOT_USE_EXISTING_AND_NEW_ACCOUNT"] = 2] = "CANNOT_USE_EXISTING_AND_NEW_ACCOUNT";
    /**
     * Cannot cancel an approved billing setup whose start time has passed.
     *
     * @generated from protobuf enum value: CANNOT_REMOVE_STARTED_BILLING_SETUP = 3;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["CANNOT_REMOVE_STARTED_BILLING_SETUP"] = 3] = "CANNOT_REMOVE_STARTED_BILLING_SETUP";
    /**
     * Cannot perform a Change of Bill-To (CBT) to the same payments account.
     *
     * @generated from protobuf enum value: CANNOT_CHANGE_BILLING_TO_SAME_PAYMENTS_ACCOUNT = 4;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["CANNOT_CHANGE_BILLING_TO_SAME_PAYMENTS_ACCOUNT"] = 4] = "CANNOT_CHANGE_BILLING_TO_SAME_PAYMENTS_ACCOUNT";
    /**
     * Billing setups can only be used by customers with ENABLED or DRAFT
     * status.
     *
     * @generated from protobuf enum value: BILLING_SETUP_NOT_PERMITTED_FOR_CUSTOMER_STATUS = 5;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["BILLING_SETUP_NOT_PERMITTED_FOR_CUSTOMER_STATUS"] = 5] = "BILLING_SETUP_NOT_PERMITTED_FOR_CUSTOMER_STATUS";
    /**
     * Billing setups must either include a correctly formatted existing
     * payments account id, or a non-empty new payments account name.
     *
     * @generated from protobuf enum value: INVALID_PAYMENTS_ACCOUNT = 6;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["INVALID_PAYMENTS_ACCOUNT"] = 6] = "INVALID_PAYMENTS_ACCOUNT";
    /**
     * Only billable and third-party customers can create billing setups.
     *
     * @generated from protobuf enum value: BILLING_SETUP_NOT_PERMITTED_FOR_CUSTOMER_CATEGORY = 7;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["BILLING_SETUP_NOT_PERMITTED_FOR_CUSTOMER_CATEGORY"] = 7] = "BILLING_SETUP_NOT_PERMITTED_FOR_CUSTOMER_CATEGORY";
    /**
     * Billing setup creations can only use NOW for start time type.
     *
     * @generated from protobuf enum value: INVALID_START_TIME_TYPE = 8;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["INVALID_START_TIME_TYPE"] = 8] = "INVALID_START_TIME_TYPE";
    /**
     * Billing setups can only be created for a third-party customer if they do
     * not already have a setup.
     *
     * @generated from protobuf enum value: THIRD_PARTY_ALREADY_HAS_BILLING = 9;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["THIRD_PARTY_ALREADY_HAS_BILLING"] = 9] = "THIRD_PARTY_ALREADY_HAS_BILLING";
    /**
     * Billing setups cannot be created if there is already a pending billing in
     * progress.
     *
     * @generated from protobuf enum value: BILLING_SETUP_IN_PROGRESS = 10;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["BILLING_SETUP_IN_PROGRESS"] = 10] = "BILLING_SETUP_IN_PROGRESS";
    /**
     * Billing setups can only be created by customers who have permission to
     * setup billings. Users can contact a representative for help setting up
     * permissions.
     *
     * @generated from protobuf enum value: NO_SIGNUP_PERMISSION = 11;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["NO_SIGNUP_PERMISSION"] = 11] = "NO_SIGNUP_PERMISSION";
    /**
     * Billing setups cannot be created if there is already a future-approved
     * billing.
     *
     * @generated from protobuf enum value: CHANGE_OF_BILL_TO_IN_PROGRESS = 12;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["CHANGE_OF_BILL_TO_IN_PROGRESS"] = 12] = "CHANGE_OF_BILL_TO_IN_PROGRESS";
    /**
     * Requested payments profile not found.
     *
     * @generated from protobuf enum value: PAYMENTS_PROFILE_NOT_FOUND = 13;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["PAYMENTS_PROFILE_NOT_FOUND"] = 13] = "PAYMENTS_PROFILE_NOT_FOUND";
    /**
     * Requested payments account not found.
     *
     * @generated from protobuf enum value: PAYMENTS_ACCOUNT_NOT_FOUND = 14;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["PAYMENTS_ACCOUNT_NOT_FOUND"] = 14] = "PAYMENTS_ACCOUNT_NOT_FOUND";
    /**
     * Billing setup creation failed because the payments profile is ineligible.
     *
     * @generated from protobuf enum value: PAYMENTS_PROFILE_INELIGIBLE = 15;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["PAYMENTS_PROFILE_INELIGIBLE"] = 15] = "PAYMENTS_PROFILE_INELIGIBLE";
    /**
     * Billing setup creation failed because the payments account is ineligible.
     *
     * @generated from protobuf enum value: PAYMENTS_ACCOUNT_INELIGIBLE = 16;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["PAYMENTS_ACCOUNT_INELIGIBLE"] = 16] = "PAYMENTS_ACCOUNT_INELIGIBLE";
    /**
     * Billing setup creation failed because the payments profile needs internal
     * approval.
     *
     * @generated from protobuf enum value: CUSTOMER_NEEDS_INTERNAL_APPROVAL = 17;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["CUSTOMER_NEEDS_INTERNAL_APPROVAL"] = 17] = "CUSTOMER_NEEDS_INTERNAL_APPROVAL";
    /**
     * Payments account has different currency code than the current customer
     * and hence cannot be used to setup billing.
     *
     * @generated from protobuf enum value: PAYMENTS_ACCOUNT_INELIGIBLE_CURRENCY_CODE_MISMATCH = 19;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["PAYMENTS_ACCOUNT_INELIGIBLE_CURRENCY_CODE_MISMATCH"] = 19] = "PAYMENTS_ACCOUNT_INELIGIBLE_CURRENCY_CODE_MISMATCH";
    /**
     * A start time in the future cannot be used because there is currently no
     * active billing setup for this customer.
     *
     * @generated from protobuf enum value: FUTURE_START_TIME_PROHIBITED = 20;
     */
    BillingSetupErrorEnum_BillingSetupError[BillingSetupErrorEnum_BillingSetupError["FUTURE_START_TIME_PROHIBITED"] = 20] = "FUTURE_START_TIME_PROHIBITED";
})(BillingSetupErrorEnum_BillingSetupError = exports.BillingSetupErrorEnum_BillingSetupError || (exports.BillingSetupErrorEnum_BillingSetupError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class BillingSetupErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.BillingSetupErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.BillingSetupErrorEnum
 */
exports.BillingSetupErrorEnum = new BillingSetupErrorEnum$Type();
