// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_customizer_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignCustomizerErrorEnum = exports.CampaignCustomizerErrorEnum_CampaignCustomizerError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign customizer errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignCustomizerErrorEnum.CampaignCustomizerError
 */
var CampaignCustomizerErrorEnum_CampaignCustomizerError;
(function (CampaignCustomizerErrorEnum_CampaignCustomizerError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignCustomizerErrorEnum_CampaignCustomizerError[CampaignCustomizerErrorEnum_CampaignCustomizerError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignCustomizerErrorEnum_CampaignCustomizerError[CampaignCustomizerErrorEnum_CampaignCustomizerError["UNKNOWN"] = 1] = "UNKNOWN";
})(CampaignCustomizerErrorEnum_CampaignCustomizerError = exports.CampaignCustomizerErrorEnum_CampaignCustomizerError || (exports.CampaignCustomizerErrorEnum_CampaignCustomizerError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignCustomizerErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignCustomizerErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignCustomizerErrorEnum
 */
exports.CampaignCustomizerErrorEnum = new CampaignCustomizerErrorEnum$Type();
