// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_draft_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignDraftErrorEnum = exports.CampaignDraftErrorEnum_CampaignDraftError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign draft errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignDraftErrorEnum.CampaignDraftError
 */
var CampaignDraftErrorEnum_CampaignDraftError;
(function (CampaignDraftErrorEnum_CampaignDraftError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A draft with this name already exists for this campaign.
     *
     * @generated from protobuf enum value: DUPLICATE_DRAFT_NAME = 2;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["DUPLICATE_DRAFT_NAME"] = 2] = "DUPLICATE_DRAFT_NAME";
    /**
     * The draft is removed and cannot be transitioned to another status.
     *
     * @generated from protobuf enum value: INVALID_STATUS_TRANSITION_FROM_REMOVED = 3;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["INVALID_STATUS_TRANSITION_FROM_REMOVED"] = 3] = "INVALID_STATUS_TRANSITION_FROM_REMOVED";
    /**
     * The draft has been promoted and cannot be transitioned to the specified
     * status.
     *
     * @generated from protobuf enum value: INVALID_STATUS_TRANSITION_FROM_PROMOTED = 4;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["INVALID_STATUS_TRANSITION_FROM_PROMOTED"] = 4] = "INVALID_STATUS_TRANSITION_FROM_PROMOTED";
    /**
     * The draft has failed to be promoted and cannot be transitioned to the
     * specified status.
     *
     * @generated from protobuf enum value: INVALID_STATUS_TRANSITION_FROM_PROMOTE_FAILED = 5;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["INVALID_STATUS_TRANSITION_FROM_PROMOTE_FAILED"] = 5] = "INVALID_STATUS_TRANSITION_FROM_PROMOTE_FAILED";
    /**
     * This customer is not allowed to create drafts.
     *
     * @generated from protobuf enum value: CUSTOMER_CANNOT_CREATE_DRAFT = 6;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["CUSTOMER_CANNOT_CREATE_DRAFT"] = 6] = "CUSTOMER_CANNOT_CREATE_DRAFT";
    /**
     * This campaign is not allowed to create drafts.
     *
     * @generated from protobuf enum value: CAMPAIGN_CANNOT_CREATE_DRAFT = 7;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["CAMPAIGN_CANNOT_CREATE_DRAFT"] = 7] = "CAMPAIGN_CANNOT_CREATE_DRAFT";
    /**
     * This modification cannot be made on a draft.
     *
     * @generated from protobuf enum value: INVALID_DRAFT_CHANGE = 8;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["INVALID_DRAFT_CHANGE"] = 8] = "INVALID_DRAFT_CHANGE";
    /**
     * The draft cannot be transitioned to the specified status from its
     * current status.
     *
     * @generated from protobuf enum value: INVALID_STATUS_TRANSITION = 9;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["INVALID_STATUS_TRANSITION"] = 9] = "INVALID_STATUS_TRANSITION";
    /**
     * The campaign has reached the maximum number of drafts that can be created
     * for a campaign throughout its lifetime. No additional drafts can be
     * created for this campaign. Removed drafts also count towards this limit.
     *
     * @generated from protobuf enum value: MAX_NUMBER_OF_DRAFTS_PER_CAMPAIGN_REACHED = 10;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["MAX_NUMBER_OF_DRAFTS_PER_CAMPAIGN_REACHED"] = 10] = "MAX_NUMBER_OF_DRAFTS_PER_CAMPAIGN_REACHED";
    /**
     * ListAsyncErrors was called without first promoting the draft.
     *
     * @generated from protobuf enum value: LIST_ERRORS_FOR_PROMOTED_DRAFT_ONLY = 11;
     */
    CampaignDraftErrorEnum_CampaignDraftError[CampaignDraftErrorEnum_CampaignDraftError["LIST_ERRORS_FOR_PROMOTED_DRAFT_ONLY"] = 11] = "LIST_ERRORS_FOR_PROMOTED_DRAFT_ONLY";
})(CampaignDraftErrorEnum_CampaignDraftError = exports.CampaignDraftErrorEnum_CampaignDraftError || (exports.CampaignDraftErrorEnum_CampaignDraftError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignDraftErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignDraftErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignDraftErrorEnum
 */
exports.CampaignDraftErrorEnum = new CampaignDraftErrorEnum$Type();
