// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_feed_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignFeedErrorEnum = exports.CampaignFeedErrorEnum_CampaignFeedError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign feed errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignFeedErrorEnum.CampaignFeedError
 */
var CampaignFeedErrorEnum_CampaignFeedError;
(function (CampaignFeedErrorEnum_CampaignFeedError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An active feed already exists for this campaign and placeholder type.
     *
     * @generated from protobuf enum value: FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE = 2;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE"] = 2] = "FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE";
    /**
     * The specified feed is removed.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_FOR_REMOVED_FEED = 4;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["CANNOT_CREATE_FOR_REMOVED_FEED"] = 4] = "CANNOT_CREATE_FOR_REMOVED_FEED";
    /**
     * The CampaignFeed already exists. UPDATE should be used to modify the
     * existing CampaignFeed.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_ALREADY_EXISTING_CAMPAIGN_FEED = 5;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["CANNOT_CREATE_ALREADY_EXISTING_CAMPAIGN_FEED"] = 5] = "CANNOT_CREATE_ALREADY_EXISTING_CAMPAIGN_FEED";
    /**
     * Cannot update removed campaign feed.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_REMOVED_CAMPAIGN_FEED = 6;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["CANNOT_MODIFY_REMOVED_CAMPAIGN_FEED"] = 6] = "CANNOT_MODIFY_REMOVED_CAMPAIGN_FEED";
    /**
     * Invalid placeholder type.
     *
     * @generated from protobuf enum value: INVALID_PLACEHOLDER_TYPE = 7;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["INVALID_PLACEHOLDER_TYPE"] = 7] = "INVALID_PLACEHOLDER_TYPE";
    /**
     * Feed mapping for this placeholder type does not exist.
     *
     * @generated from protobuf enum value: MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE = 8;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE"] = 8] = "MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE";
    /**
     * Location CampaignFeeds cannot be created unless there is a location
     * CustomerFeed for the specified feed.
     *
     * @generated from protobuf enum value: NO_EXISTING_LOCATION_CUSTOMER_FEED = 9;
     */
    CampaignFeedErrorEnum_CampaignFeedError[CampaignFeedErrorEnum_CampaignFeedError["NO_EXISTING_LOCATION_CUSTOMER_FEED"] = 9] = "NO_EXISTING_LOCATION_CUSTOMER_FEED";
})(CampaignFeedErrorEnum_CampaignFeedError = exports.CampaignFeedErrorEnum_CampaignFeedError || (exports.CampaignFeedErrorEnum_CampaignFeedError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignFeedErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignFeedErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignFeedErrorEnum
 */
exports.CampaignFeedErrorEnum = new CampaignFeedErrorEnum$Type();
