// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_shared_set_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignSharedSetErrorEnum = exports.CampaignSharedSetErrorEnum_CampaignSharedSetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign shared set errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignSharedSetErrorEnum.CampaignSharedSetError
 */
var CampaignSharedSetErrorEnum_CampaignSharedSetError;
(function (CampaignSharedSetErrorEnum_CampaignSharedSetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignSharedSetErrorEnum_CampaignSharedSetError[CampaignSharedSetErrorEnum_CampaignSharedSetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignSharedSetErrorEnum_CampaignSharedSetError[CampaignSharedSetErrorEnum_CampaignSharedSetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The shared set belongs to another customer and permission isn't granted.
     *
     * @generated from protobuf enum value: SHARED_SET_ACCESS_DENIED = 2;
     */
    CampaignSharedSetErrorEnum_CampaignSharedSetError[CampaignSharedSetErrorEnum_CampaignSharedSetError["SHARED_SET_ACCESS_DENIED"] = 2] = "SHARED_SET_ACCESS_DENIED";
})(CampaignSharedSetErrorEnum_CampaignSharedSetError = exports.CampaignSharedSetErrorEnum_CampaignSharedSetError || (exports.CampaignSharedSetErrorEnum_CampaignSharedSetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignSharedSetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignSharedSetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignSharedSetErrorEnum
 */
exports.CampaignSharedSetErrorEnum = new CampaignSharedSetErrorEnum$Type();
