// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_conversion_goal_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignConversionGoalErrorEnum = exports.CampaignConversionGoalErrorEnum_CampaignConversionGoalError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign conversion goal errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignConversionGoalErrorEnum.CampaignConversionGoalError
 */
var CampaignConversionGoalErrorEnum_CampaignConversionGoalError;
(function (CampaignConversionGoalErrorEnum_CampaignConversionGoalError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignConversionGoalErrorEnum_CampaignConversionGoalError[CampaignConversionGoalErrorEnum_CampaignConversionGoalError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignConversionGoalErrorEnum_CampaignConversionGoalError[CampaignConversionGoalErrorEnum_CampaignConversionGoalError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Campaign is managed by Search Ads 360 but uses Unified Goal.
     *
     * @generated from protobuf enum value: CANNOT_USE_CAMPAIGN_GOAL_FOR_SEARCH_ADS_360_MANAGED_CAMPAIGN = 2;
     */
    CampaignConversionGoalErrorEnum_CampaignConversionGoalError[CampaignConversionGoalErrorEnum_CampaignConversionGoalError["CANNOT_USE_CAMPAIGN_GOAL_FOR_SEARCH_ADS_360_MANAGED_CAMPAIGN"] = 2] = "CANNOT_USE_CAMPAIGN_GOAL_FOR_SEARCH_ADS_360_MANAGED_CAMPAIGN";
})(CampaignConversionGoalErrorEnum_CampaignConversionGoalError = exports.CampaignConversionGoalErrorEnum_CampaignConversionGoalError || (exports.CampaignConversionGoalErrorEnum_CampaignConversionGoalError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignConversionGoalErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignConversionGoalErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignConversionGoalErrorEnum
 */
exports.CampaignConversionGoalErrorEnum = new CampaignConversionGoalErrorEnum$Type();
