// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_criterion_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignCriterionErrorEnum = exports.CampaignCriterionErrorEnum_CampaignCriterionError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign criterion errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignCriterionErrorEnum.CampaignCriterionError
 */
var CampaignCriterionErrorEnum_CampaignCriterionError;
(function (CampaignCriterionErrorEnum_CampaignCriterionError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Concrete type of criterion (keyword v.s. placement) is required for
     * CREATE and UPDATE operations.
     *
     * @generated from protobuf enum value: CONCRETE_TYPE_REQUIRED = 2;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CONCRETE_TYPE_REQUIRED"] = 2] = "CONCRETE_TYPE_REQUIRED";
    /**
     * Invalid placement URL.
     *
     * @generated from protobuf enum value: INVALID_PLACEMENT_URL = 3;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["INVALID_PLACEMENT_URL"] = 3] = "INVALID_PLACEMENT_URL";
    /**
     * Criteria type can not be excluded for the campaign by the customer. like
     * AOL account type cannot target site type criteria
     *
     * @generated from protobuf enum value: CANNOT_EXCLUDE_CRITERIA_TYPE = 4;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_EXCLUDE_CRITERIA_TYPE"] = 4] = "CANNOT_EXCLUDE_CRITERIA_TYPE";
    /**
     * Cannot set the campaign criterion status for this criteria type.
     *
     * @generated from protobuf enum value: CANNOT_SET_STATUS_FOR_CRITERIA_TYPE = 5;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_SET_STATUS_FOR_CRITERIA_TYPE"] = 5] = "CANNOT_SET_STATUS_FOR_CRITERIA_TYPE";
    /**
     * Cannot set the campaign criterion status for an excluded criteria.
     *
     * @generated from protobuf enum value: CANNOT_SET_STATUS_FOR_EXCLUDED_CRITERIA = 6;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_SET_STATUS_FOR_EXCLUDED_CRITERIA"] = 6] = "CANNOT_SET_STATUS_FOR_EXCLUDED_CRITERIA";
    /**
     * Cannot target and exclude the same criterion.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_AND_EXCLUDE = 7;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_TARGET_AND_EXCLUDE"] = 7] = "CANNOT_TARGET_AND_EXCLUDE";
    /**
     * The mutate contained too many operations.
     *
     * @generated from protobuf enum value: TOO_MANY_OPERATIONS = 8;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["TOO_MANY_OPERATIONS"] = 8] = "TOO_MANY_OPERATIONS";
    /**
     * This operator cannot be applied to a criterion of this type.
     *
     * @generated from protobuf enum value: OPERATOR_NOT_SUPPORTED_FOR_CRITERION_TYPE = 9;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["OPERATOR_NOT_SUPPORTED_FOR_CRITERION_TYPE"] = 9] = "OPERATOR_NOT_SUPPORTED_FOR_CRITERION_TYPE";
    /**
     * The Shopping campaign sales country is not supported for
     * ProductSalesChannel targeting.
     *
     * @generated from protobuf enum value: SHOPPING_CAMPAIGN_SALES_COUNTRY_NOT_SUPPORTED_FOR_SALES_CHANNEL = 10;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["SHOPPING_CAMPAIGN_SALES_COUNTRY_NOT_SUPPORTED_FOR_SALES_CHANNEL"] = 10] = "SHOPPING_CAMPAIGN_SALES_COUNTRY_NOT_SUPPORTED_FOR_SALES_CHANNEL";
    /**
     * The existing field can't be updated with CREATE operation. It can be
     * updated with UPDATE operation only.
     *
     * @generated from protobuf enum value: CANNOT_ADD_EXISTING_FIELD = 11;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_ADD_EXISTING_FIELD"] = 11] = "CANNOT_ADD_EXISTING_FIELD";
    /**
     * Negative criteria are immutable, so updates are not allowed.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_NEGATIVE_CRITERION = 12;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_UPDATE_NEGATIVE_CRITERION"] = 12] = "CANNOT_UPDATE_NEGATIVE_CRITERION";
    /**
     * Only free form names are allowed for negative Smart campaign keyword
     * theme.
     *
     * @generated from protobuf enum value: CANNOT_SET_NEGATIVE_KEYWORD_THEME_CONSTANT_CRITERION = 13;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_SET_NEGATIVE_KEYWORD_THEME_CONSTANT_CRITERION"] = 13] = "CANNOT_SET_NEGATIVE_KEYWORD_THEME_CONSTANT_CRITERION";
    /**
     * Invalid Smart campaign keyword theme constant criterion.
     *
     * @generated from protobuf enum value: INVALID_KEYWORD_THEME_CONSTANT = 14;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["INVALID_KEYWORD_THEME_CONSTANT"] = 14] = "INVALID_KEYWORD_THEME_CONSTANT";
    /**
     * A Smart campaign keyword theme constant or free-form Smart campaign
     * keyword theme is required.
     *
     * @generated from protobuf enum value: MISSING_KEYWORD_THEME_CONSTANT_OR_FREE_FORM_KEYWORD_THEME = 15;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["MISSING_KEYWORD_THEME_CONSTANT_OR_FREE_FORM_KEYWORD_THEME"] = 15] = "MISSING_KEYWORD_THEME_CONSTANT_OR_FREE_FORM_KEYWORD_THEME";
    /**
     * A Smart campaign may not target proximity and location criteria
     * simultaneously.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_BOTH_PROXIMITY_AND_LOCATION_CRITERIA_FOR_SMART_CAMPAIGN = 16;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_TARGET_BOTH_PROXIMITY_AND_LOCATION_CRITERIA_FOR_SMART_CAMPAIGN"] = 16] = "CANNOT_TARGET_BOTH_PROXIMITY_AND_LOCATION_CRITERIA_FOR_SMART_CAMPAIGN";
    /**
     * A Smart campaign may not target multiple proximity criteria.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_MULTIPLE_PROXIMITY_CRITERIA_FOR_SMART_CAMPAIGN = 17;
     */
    CampaignCriterionErrorEnum_CampaignCriterionError[CampaignCriterionErrorEnum_CampaignCriterionError["CANNOT_TARGET_MULTIPLE_PROXIMITY_CRITERIA_FOR_SMART_CAMPAIGN"] = 17] = "CANNOT_TARGET_MULTIPLE_PROXIMITY_CRITERIA_FOR_SMART_CAMPAIGN";
})(CampaignCriterionErrorEnum_CampaignCriterionError = exports.CampaignCriterionErrorEnum_CampaignCriterionError || (exports.CampaignCriterionErrorEnum_CampaignCriterionError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignCriterionErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignCriterionErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignCriterionErrorEnum
 */
exports.CampaignCriterionErrorEnum = new CampaignCriterionErrorEnum$Type();
