// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_experiment_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignExperimentErrorEnum = exports.CampaignExperimentErrorEnum_CampaignExperimentError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign experiment errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignExperimentErrorEnum.CampaignExperimentError
 */
var CampaignExperimentErrorEnum_CampaignExperimentError;
(function (CampaignExperimentErrorEnum_CampaignExperimentError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An active campaign or experiment with this name already exists.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 2;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["DUPLICATE_NAME"] = 2] = "DUPLICATE_NAME";
    /**
     * Experiment cannot be updated from the current state to the
     * requested target state. For example, an experiment can only graduate
     * if its status is ENABLED.
     *
     * @generated from protobuf enum value: INVALID_TRANSITION = 3;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["INVALID_TRANSITION"] = 3] = "INVALID_TRANSITION";
    /**
     * Cannot create an experiment from a campaign using an explicitly shared
     * budget.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_EXPERIMENT_WITH_SHARED_BUDGET = 4;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["CANNOT_CREATE_EXPERIMENT_WITH_SHARED_BUDGET"] = 4] = "CANNOT_CREATE_EXPERIMENT_WITH_SHARED_BUDGET";
    /**
     * Cannot create an experiment for a removed base campaign.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_EXPERIMENT_FOR_REMOVED_BASE_CAMPAIGN = 5;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["CANNOT_CREATE_EXPERIMENT_FOR_REMOVED_BASE_CAMPAIGN"] = 5] = "CANNOT_CREATE_EXPERIMENT_FOR_REMOVED_BASE_CAMPAIGN";
    /**
     * Cannot create an experiment from a draft, which has a status other than
     * proposed.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_EXPERIMENT_FOR_NON_PROPOSED_DRAFT = 6;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["CANNOT_CREATE_EXPERIMENT_FOR_NON_PROPOSED_DRAFT"] = 6] = "CANNOT_CREATE_EXPERIMENT_FOR_NON_PROPOSED_DRAFT";
    /**
     * This customer is not allowed to create an experiment.
     *
     * @generated from protobuf enum value: CUSTOMER_CANNOT_CREATE_EXPERIMENT = 7;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["CUSTOMER_CANNOT_CREATE_EXPERIMENT"] = 7] = "CUSTOMER_CANNOT_CREATE_EXPERIMENT";
    /**
     * This campaign is not allowed to create an experiment.
     *
     * @generated from protobuf enum value: CAMPAIGN_CANNOT_CREATE_EXPERIMENT = 8;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["CAMPAIGN_CANNOT_CREATE_EXPERIMENT"] = 8] = "CAMPAIGN_CANNOT_CREATE_EXPERIMENT";
    /**
     * Trying to set an experiment duration which overlaps with another
     * experiment.
     *
     * @generated from protobuf enum value: EXPERIMENT_DURATIONS_MUST_NOT_OVERLAP = 9;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["EXPERIMENT_DURATIONS_MUST_NOT_OVERLAP"] = 9] = "EXPERIMENT_DURATIONS_MUST_NOT_OVERLAP";
    /**
     * All non-removed experiments must start and end within their campaign's
     * duration.
     *
     * @generated from protobuf enum value: EXPERIMENT_DURATION_MUST_BE_WITHIN_CAMPAIGN_DURATION = 10;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["EXPERIMENT_DURATION_MUST_BE_WITHIN_CAMPAIGN_DURATION"] = 10] = "EXPERIMENT_DURATION_MUST_BE_WITHIN_CAMPAIGN_DURATION";
    /**
     * The experiment cannot be modified because its status is in a terminal
     * state, such as REMOVED.
     *
     * @generated from protobuf enum value: CANNOT_MUTATE_EXPERIMENT_DUE_TO_STATUS = 11;
     */
    CampaignExperimentErrorEnum_CampaignExperimentError[CampaignExperimentErrorEnum_CampaignExperimentError["CANNOT_MUTATE_EXPERIMENT_DUE_TO_STATUS"] = 11] = "CANNOT_MUTATE_EXPERIMENT_DUE_TO_STATUS";
})(CampaignExperimentErrorEnum_CampaignExperimentError = exports.CampaignExperimentErrorEnum_CampaignExperimentError || (exports.CampaignExperimentErrorEnum_CampaignExperimentError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignExperimentErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignExperimentErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignExperimentErrorEnum
 */
exports.CampaignExperimentErrorEnum = new CampaignExperimentErrorEnum$Type();
