// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignErrorEnum = exports.CampaignErrorEnum_CampaignError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignErrorEnum.CampaignError
 */
var CampaignErrorEnum_CampaignError;
(function (CampaignErrorEnum_CampaignError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Cannot target content network.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_CONTENT_NETWORK = 3;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_TARGET_CONTENT_NETWORK"] = 3] = "CANNOT_TARGET_CONTENT_NETWORK";
    /**
     * Cannot target search network.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_SEARCH_NETWORK = 4;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_TARGET_SEARCH_NETWORK"] = 4] = "CANNOT_TARGET_SEARCH_NETWORK";
    /**
     * Cannot cover search network without google search network.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_SEARCH_NETWORK_WITHOUT_GOOGLE_SEARCH = 5;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_TARGET_SEARCH_NETWORK_WITHOUT_GOOGLE_SEARCH"] = 5] = "CANNOT_TARGET_SEARCH_NETWORK_WITHOUT_GOOGLE_SEARCH";
    /**
     * Cannot target Google Search network for a CPM campaign.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_GOOGLE_SEARCH_FOR_CPM_CAMPAIGN = 6;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_TARGET_GOOGLE_SEARCH_FOR_CPM_CAMPAIGN"] = 6] = "CANNOT_TARGET_GOOGLE_SEARCH_FOR_CPM_CAMPAIGN";
    /**
     * Must target at least one network.
     *
     * @generated from protobuf enum value: CAMPAIGN_MUST_TARGET_AT_LEAST_ONE_NETWORK = 7;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CAMPAIGN_MUST_TARGET_AT_LEAST_ONE_NETWORK"] = 7] = "CAMPAIGN_MUST_TARGET_AT_LEAST_ONE_NETWORK";
    /**
     * Only some Google partners are allowed to target partner search network.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_PARTNER_SEARCH_NETWORK = 8;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_TARGET_PARTNER_SEARCH_NETWORK"] = 8] = "CANNOT_TARGET_PARTNER_SEARCH_NETWORK";
    /**
     * Cannot target content network only as campaign has criteria-level bidding
     * strategy.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_CONTENT_NETWORK_ONLY_WITH_CRITERIA_LEVEL_BIDDING_STRATEGY = 9;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_TARGET_CONTENT_NETWORK_ONLY_WITH_CRITERIA_LEVEL_BIDDING_STRATEGY"] = 9] = "CANNOT_TARGET_CONTENT_NETWORK_ONLY_WITH_CRITERIA_LEVEL_BIDDING_STRATEGY";
    /**
     * Cannot modify the start or end date such that the campaign duration would
     * not contain the durations of all runnable trials.
     *
     * @generated from protobuf enum value: CAMPAIGN_DURATION_MUST_CONTAIN_ALL_RUNNABLE_TRIALS = 10;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CAMPAIGN_DURATION_MUST_CONTAIN_ALL_RUNNABLE_TRIALS"] = 10] = "CAMPAIGN_DURATION_MUST_CONTAIN_ALL_RUNNABLE_TRIALS";
    /**
     * Cannot modify dates, budget or status of a trial campaign.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_FOR_TRIAL_CAMPAIGN = 11;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_MODIFY_FOR_TRIAL_CAMPAIGN"] = 11] = "CANNOT_MODIFY_FOR_TRIAL_CAMPAIGN";
    /**
     * Trying to modify the name of an active or paused campaign, where the name
     * is already assigned to another active or paused campaign.
     *
     * @generated from protobuf enum value: DUPLICATE_CAMPAIGN_NAME = 12;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["DUPLICATE_CAMPAIGN_NAME"] = 12] = "DUPLICATE_CAMPAIGN_NAME";
    /**
     * Two fields are in conflicting modes.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_CAMPAIGN_FIELD = 13;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INCOMPATIBLE_CAMPAIGN_FIELD"] = 13] = "INCOMPATIBLE_CAMPAIGN_FIELD";
    /**
     * Campaign name cannot be used.
     *
     * @generated from protobuf enum value: INVALID_CAMPAIGN_NAME = 14;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INVALID_CAMPAIGN_NAME"] = 14] = "INVALID_CAMPAIGN_NAME";
    /**
     * Given status is invalid.
     *
     * @generated from protobuf enum value: INVALID_AD_SERVING_OPTIMIZATION_STATUS = 15;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INVALID_AD_SERVING_OPTIMIZATION_STATUS"] = 15] = "INVALID_AD_SERVING_OPTIMIZATION_STATUS";
    /**
     * Error in the campaign level tracking URL.
     *
     * @generated from protobuf enum value: INVALID_TRACKING_URL = 16;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INVALID_TRACKING_URL"] = 16] = "INVALID_TRACKING_URL";
    /**
     * Cannot set both tracking URL template and tracking setting. A user has
     * to clear legacy tracking setting in order to add tracking URL template.
     *
     * @generated from protobuf enum value: CANNOT_SET_BOTH_TRACKING_URL_TEMPLATE_AND_TRACKING_SETTING = 17;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_SET_BOTH_TRACKING_URL_TEMPLATE_AND_TRACKING_SETTING"] = 17] = "CANNOT_SET_BOTH_TRACKING_URL_TEMPLATE_AND_TRACKING_SETTING";
    /**
     * The maximum number of impressions for Frequency Cap should be an integer
     * greater than 0.
     *
     * @generated from protobuf enum value: MAX_IMPRESSIONS_NOT_IN_RANGE = 18;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["MAX_IMPRESSIONS_NOT_IN_RANGE"] = 18] = "MAX_IMPRESSIONS_NOT_IN_RANGE";
    /**
     * Only the Day, Week and Month time units are supported.
     *
     * @generated from protobuf enum value: TIME_UNIT_NOT_SUPPORTED = 19;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["TIME_UNIT_NOT_SUPPORTED"] = 19] = "TIME_UNIT_NOT_SUPPORTED";
    /**
     * Operation not allowed on a campaign whose serving status has ended
     *
     * @generated from protobuf enum value: INVALID_OPERATION_IF_SERVING_STATUS_HAS_ENDED = 20;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INVALID_OPERATION_IF_SERVING_STATUS_HAS_ENDED"] = 20] = "INVALID_OPERATION_IF_SERVING_STATUS_HAS_ENDED";
    /**
     * This budget is exclusively linked to a Campaign that is using experiments
     * so it cannot be shared.
     *
     * @generated from protobuf enum value: BUDGET_CANNOT_BE_SHARED = 21;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["BUDGET_CANNOT_BE_SHARED"] = 21] = "BUDGET_CANNOT_BE_SHARED";
    /**
     * Campaigns using experiments cannot use a shared budget.
     *
     * @generated from protobuf enum value: CAMPAIGN_CANNOT_USE_SHARED_BUDGET = 22;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CAMPAIGN_CANNOT_USE_SHARED_BUDGET"] = 22] = "CAMPAIGN_CANNOT_USE_SHARED_BUDGET";
    /**
     * A different budget cannot be assigned to a campaign when there are
     * running or scheduled trials.
     *
     * @generated from protobuf enum value: CANNOT_CHANGE_BUDGET_ON_CAMPAIGN_WITH_TRIALS = 23;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_CHANGE_BUDGET_ON_CAMPAIGN_WITH_TRIALS"] = 23] = "CANNOT_CHANGE_BUDGET_ON_CAMPAIGN_WITH_TRIALS";
    /**
     * No link found between the campaign and the label.
     *
     * @generated from protobuf enum value: CAMPAIGN_LABEL_DOES_NOT_EXIST = 24;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CAMPAIGN_LABEL_DOES_NOT_EXIST"] = 24] = "CAMPAIGN_LABEL_DOES_NOT_EXIST";
    /**
     * The label has already been attached to the campaign.
     *
     * @generated from protobuf enum value: CAMPAIGN_LABEL_ALREADY_EXISTS = 25;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CAMPAIGN_LABEL_ALREADY_EXISTS"] = 25] = "CAMPAIGN_LABEL_ALREADY_EXISTS";
    /**
     * A ShoppingSetting was not found when creating a shopping campaign.
     *
     * @generated from protobuf enum value: MISSING_SHOPPING_SETTING = 26;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["MISSING_SHOPPING_SETTING"] = 26] = "MISSING_SHOPPING_SETTING";
    /**
     * The country in shopping setting is not an allowed country.
     *
     * @generated from protobuf enum value: INVALID_SHOPPING_SALES_COUNTRY = 27;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INVALID_SHOPPING_SALES_COUNTRY"] = 27] = "INVALID_SHOPPING_SALES_COUNTRY";
    /**
     * The requested channel type is not available according to the customer's
     * account setting.
     *
     * @generated from protobuf enum value: ADVERTISING_CHANNEL_TYPE_NOT_AVAILABLE_FOR_ACCOUNT_TYPE = 31;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["ADVERTISING_CHANNEL_TYPE_NOT_AVAILABLE_FOR_ACCOUNT_TYPE"] = 31] = "ADVERTISING_CHANNEL_TYPE_NOT_AVAILABLE_FOR_ACCOUNT_TYPE";
    /**
     * The AdvertisingChannelSubType is not a valid subtype of the primary
     * channel type.
     *
     * @generated from protobuf enum value: INVALID_ADVERTISING_CHANNEL_SUB_TYPE = 32;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INVALID_ADVERTISING_CHANNEL_SUB_TYPE"] = 32] = "INVALID_ADVERTISING_CHANNEL_SUB_TYPE";
    /**
     * At least one conversion must be selected.
     *
     * @generated from protobuf enum value: AT_LEAST_ONE_CONVERSION_MUST_BE_SELECTED = 33;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["AT_LEAST_ONE_CONVERSION_MUST_BE_SELECTED"] = 33] = "AT_LEAST_ONE_CONVERSION_MUST_BE_SELECTED";
    /**
     * Setting ad rotation mode for a campaign is not allowed. Ad rotation mode
     * at campaign is deprecated.
     *
     * @generated from protobuf enum value: CANNOT_SET_AD_ROTATION_MODE = 34;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_SET_AD_ROTATION_MODE"] = 34] = "CANNOT_SET_AD_ROTATION_MODE";
    /**
     * Trying to change start date on a campaign that has started.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_START_DATE_IF_ALREADY_STARTED = 35;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_MODIFY_START_DATE_IF_ALREADY_STARTED"] = 35] = "CANNOT_MODIFY_START_DATE_IF_ALREADY_STARTED";
    /**
     * Trying to modify a date into the past.
     *
     * @generated from protobuf enum value: CANNOT_SET_DATE_TO_PAST = 36;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_SET_DATE_TO_PAST"] = 36] = "CANNOT_SET_DATE_TO_PAST";
    /**
     * Hotel center id in the hotel setting does not match any customer links.
     *
     * @generated from protobuf enum value: MISSING_HOTEL_CUSTOMER_LINK = 37;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["MISSING_HOTEL_CUSTOMER_LINK"] = 37] = "MISSING_HOTEL_CUSTOMER_LINK";
    /**
     * Hotel center id in the hotel setting must match an active customer link.
     *
     * @generated from protobuf enum value: INVALID_HOTEL_CUSTOMER_LINK = 38;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INVALID_HOTEL_CUSTOMER_LINK"] = 38] = "INVALID_HOTEL_CUSTOMER_LINK";
    /**
     * Hotel setting was not found when creating a hotel ads campaign.
     *
     * @generated from protobuf enum value: MISSING_HOTEL_SETTING = 39;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["MISSING_HOTEL_SETTING"] = 39] = "MISSING_HOTEL_SETTING";
    /**
     * A Campaign cannot use shared campaign budgets and be part of a campaign
     * group.
     *
     * @generated from protobuf enum value: CANNOT_USE_SHARED_CAMPAIGN_BUDGET_WHILE_PART_OF_CAMPAIGN_GROUP = 40;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_USE_SHARED_CAMPAIGN_BUDGET_WHILE_PART_OF_CAMPAIGN_GROUP"] = 40] = "CANNOT_USE_SHARED_CAMPAIGN_BUDGET_WHILE_PART_OF_CAMPAIGN_GROUP";
    /**
     * The app ID was not found.
     *
     * @generated from protobuf enum value: APP_NOT_FOUND = 41;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["APP_NOT_FOUND"] = 41] = "APP_NOT_FOUND";
    /**
     * Campaign.shopping_setting.enable_local is not supported for the specified
     * campaign type.
     *
     * @generated from protobuf enum value: SHOPPING_ENABLE_LOCAL_NOT_SUPPORTED_FOR_CAMPAIGN_TYPE = 42;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["SHOPPING_ENABLE_LOCAL_NOT_SUPPORTED_FOR_CAMPAIGN_TYPE"] = 42] = "SHOPPING_ENABLE_LOCAL_NOT_SUPPORTED_FOR_CAMPAIGN_TYPE";
    /**
     * The merchant does not support the creation of campaigns for Shopping
     * Comparison Listing Ads.
     *
     * @generated from protobuf enum value: MERCHANT_NOT_ALLOWED_FOR_COMPARISON_LISTING_ADS = 43;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["MERCHANT_NOT_ALLOWED_FOR_COMPARISON_LISTING_ADS"] = 43] = "MERCHANT_NOT_ALLOWED_FOR_COMPARISON_LISTING_ADS";
    /**
     * The App campaign for engagement cannot be created because there aren't
     * enough installs.
     *
     * @generated from protobuf enum value: INSUFFICIENT_APP_INSTALLS_COUNT = 44;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INSUFFICIENT_APP_INSTALLS_COUNT"] = 44] = "INSUFFICIENT_APP_INSTALLS_COUNT";
    /**
     * The App campaign for engagement cannot be created because the app is
     * sensitive.
     *
     * @generated from protobuf enum value: SENSITIVE_CATEGORY_APP = 45;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["SENSITIVE_CATEGORY_APP"] = 45] = "SENSITIVE_CATEGORY_APP";
    /**
     * Customers with Housing, Employment, or Credit ads must accept updated
     * personalized ads policy to continue creating campaigns.
     *
     * @generated from protobuf enum value: HEC_AGREEMENT_REQUIRED = 46;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["HEC_AGREEMENT_REQUIRED"] = 46] = "HEC_AGREEMENT_REQUIRED";
    /**
     * The field is not compatible with view through conversion optimization.
     *
     * @generated from protobuf enum value: NOT_COMPATIBLE_WITH_VIEW_THROUGH_CONVERSION_OPTIMIZATION = 49;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["NOT_COMPATIBLE_WITH_VIEW_THROUGH_CONVERSION_OPTIMIZATION"] = 49] = "NOT_COMPATIBLE_WITH_VIEW_THROUGH_CONVERSION_OPTIMIZATION";
    /**
     * The field type cannot be excluded because an active campaign-asset link
     * of this type exists.
     *
     * @generated from protobuf enum value: INVALID_EXCLUDED_PARENT_ASSET_FIELD_TYPE = 48;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INVALID_EXCLUDED_PARENT_ASSET_FIELD_TYPE"] = 48] = "INVALID_EXCLUDED_PARENT_ASSET_FIELD_TYPE";
    /**
     * The app pre-registration campaign cannot be created for non-Android
     * applications.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_APP_PRE_REGISTRATION_FOR_NON_ANDROID_APP = 50;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["CANNOT_CREATE_APP_PRE_REGISTRATION_FOR_NON_ANDROID_APP"] = 50] = "CANNOT_CREATE_APP_PRE_REGISTRATION_FOR_NON_ANDROID_APP";
    /**
     * The campaign cannot be created since the app is not available for
     * pre-registration in any country.
     *
     * @generated from protobuf enum value: APP_NOT_AVAILABLE_TO_CREATE_APP_PRE_REGISTRATION_CAMPAIGN = 51;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["APP_NOT_AVAILABLE_TO_CREATE_APP_PRE_REGISTRATION_CAMPAIGN"] = 51] = "APP_NOT_AVAILABLE_TO_CREATE_APP_PRE_REGISTRATION_CAMPAIGN";
    /**
     * The type of the Budget is not compatible with this Campaign.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_BUDGET_TYPE = 52;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["INCOMPATIBLE_BUDGET_TYPE"] = 52] = "INCOMPATIBLE_BUDGET_TYPE";
    /**
     * Category bid list in the local services campaign setting contains
     * multiple bids for the same category ID.
     *
     * @generated from protobuf enum value: LOCAL_SERVICES_DUPLICATE_CATEGORY_BID = 53;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["LOCAL_SERVICES_DUPLICATE_CATEGORY_BID"] = 53] = "LOCAL_SERVICES_DUPLICATE_CATEGORY_BID";
    /**
     * Category bid list in the local services campaign setting contains
     * a bid for an invalid category ID.
     *
     * @generated from protobuf enum value: LOCAL_SERVICES_INVALID_CATEGORY_BID = 54;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["LOCAL_SERVICES_INVALID_CATEGORY_BID"] = 54] = "LOCAL_SERVICES_INVALID_CATEGORY_BID";
    /**
     * Category bid list in the local services campaign setting is missing a
     * bid for a category ID that must be present.
     *
     * @generated from protobuf enum value: LOCAL_SERVICES_MISSING_CATEGORY_BID = 55;
     */
    CampaignErrorEnum_CampaignError[CampaignErrorEnum_CampaignError["LOCAL_SERVICES_MISSING_CATEGORY_BID"] = 55] = "LOCAL_SERVICES_MISSING_CATEGORY_BID";
})(CampaignErrorEnum_CampaignError = exports.CampaignErrorEnum_CampaignError || (exports.CampaignErrorEnum_CampaignError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignErrorEnum
 */
exports.CampaignErrorEnum = new CampaignErrorEnum$Type();
