// google-ads-api-client/dist/protos/bindings/ads/errors/campaign_budget_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignBudgetErrorEnum = exports.CampaignBudgetErrorEnum_CampaignBudgetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible campaign budget errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CampaignBudgetErrorEnum.CampaignBudgetError
 */
var CampaignBudgetErrorEnum_CampaignBudgetError;
(function (CampaignBudgetErrorEnum_CampaignBudgetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The campaign budget cannot be shared.
     *
     * @generated from protobuf enum value: CAMPAIGN_BUDGET_CANNOT_BE_SHARED = 17;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CAMPAIGN_BUDGET_CANNOT_BE_SHARED"] = 17] = "CAMPAIGN_BUDGET_CANNOT_BE_SHARED";
    /**
     * The requested campaign budget no longer exists.
     *
     * @generated from protobuf enum value: CAMPAIGN_BUDGET_REMOVED = 2;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CAMPAIGN_BUDGET_REMOVED"] = 2] = "CAMPAIGN_BUDGET_REMOVED";
    /**
     * The campaign budget is associated with at least one campaign, and so the
     * campaign budget cannot be removed.
     *
     * @generated from protobuf enum value: CAMPAIGN_BUDGET_IN_USE = 3;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CAMPAIGN_BUDGET_IN_USE"] = 3] = "CAMPAIGN_BUDGET_IN_USE";
    /**
     * Customer is not on the allow-list for this campaign budget period.
     *
     * @generated from protobuf enum value: CAMPAIGN_BUDGET_PERIOD_NOT_AVAILABLE = 4;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CAMPAIGN_BUDGET_PERIOD_NOT_AVAILABLE"] = 4] = "CAMPAIGN_BUDGET_PERIOD_NOT_AVAILABLE";
    /**
     * This field is not mutable on implicitly shared campaign budgets
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_FIELD_OF_IMPLICITLY_SHARED_CAMPAIGN_BUDGET = 6;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CANNOT_MODIFY_FIELD_OF_IMPLICITLY_SHARED_CAMPAIGN_BUDGET"] = 6] = "CANNOT_MODIFY_FIELD_OF_IMPLICITLY_SHARED_CAMPAIGN_BUDGET";
    /**
     * Cannot change explicitly shared campaign budgets back to implicitly
     * shared ones.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_IMPLICITLY_SHARED = 7;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_IMPLICITLY_SHARED"] = 7] = "CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_IMPLICITLY_SHARED";
    /**
     * An implicit campaign budget without a name cannot be changed to
     * explicitly shared campaign budget.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_EXPLICITLY_SHARED_WITHOUT_NAME = 8;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_EXPLICITLY_SHARED_WITHOUT_NAME"] = 8] = "CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_EXPLICITLY_SHARED_WITHOUT_NAME";
    /**
     * Cannot change an implicitly shared campaign budget to an explicitly
     * shared one.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_EXPLICITLY_SHARED = 9;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_EXPLICITLY_SHARED"] = 9] = "CANNOT_UPDATE_CAMPAIGN_BUDGET_TO_EXPLICITLY_SHARED";
    /**
     * Only explicitly shared campaign budgets can be used with multiple
     * campaigns.
     *
     * @generated from protobuf enum value: CANNOT_USE_IMPLICITLY_SHARED_CAMPAIGN_BUDGET_WITH_MULTIPLE_CAMPAIGNS = 10;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["CANNOT_USE_IMPLICITLY_SHARED_CAMPAIGN_BUDGET_WITH_MULTIPLE_CAMPAIGNS"] = 10] = "CANNOT_USE_IMPLICITLY_SHARED_CAMPAIGN_BUDGET_WITH_MULTIPLE_CAMPAIGNS";
    /**
     * A campaign budget with this name already exists.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 11;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["DUPLICATE_NAME"] = 11] = "DUPLICATE_NAME";
    /**
     * A money amount was not in the expected currency.
     *
     * @generated from protobuf enum value: MONEY_AMOUNT_IN_WRONG_CURRENCY = 12;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["MONEY_AMOUNT_IN_WRONG_CURRENCY"] = 12] = "MONEY_AMOUNT_IN_WRONG_CURRENCY";
    /**
     * A money amount was less than the minimum CPC for currency.
     *
     * @generated from protobuf enum value: MONEY_AMOUNT_LESS_THAN_CURRENCY_MINIMUM_CPC = 13;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["MONEY_AMOUNT_LESS_THAN_CURRENCY_MINIMUM_CPC"] = 13] = "MONEY_AMOUNT_LESS_THAN_CURRENCY_MINIMUM_CPC";
    /**
     * A money amount was greater than the maximum allowed.
     *
     * @generated from protobuf enum value: MONEY_AMOUNT_TOO_LARGE = 14;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["MONEY_AMOUNT_TOO_LARGE"] = 14] = "MONEY_AMOUNT_TOO_LARGE";
    /**
     * A money amount was negative.
     *
     * @generated from protobuf enum value: NEGATIVE_MONEY_AMOUNT = 15;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["NEGATIVE_MONEY_AMOUNT"] = 15] = "NEGATIVE_MONEY_AMOUNT";
    /**
     * A money amount was not a multiple of a minimum unit.
     *
     * @generated from protobuf enum value: NON_MULTIPLE_OF_MINIMUM_CURRENCY_UNIT = 16;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["NON_MULTIPLE_OF_MINIMUM_CURRENCY_UNIT"] = 16] = "NON_MULTIPLE_OF_MINIMUM_CURRENCY_UNIT";
    /**
     * Total budget amount must be unset when BudgetPeriod is DAILY.
     *
     * @generated from protobuf enum value: TOTAL_BUDGET_AMOUNT_MUST_BE_UNSET_FOR_BUDGET_PERIOD_DAILY = 18;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["TOTAL_BUDGET_AMOUNT_MUST_BE_UNSET_FOR_BUDGET_PERIOD_DAILY"] = 18] = "TOTAL_BUDGET_AMOUNT_MUST_BE_UNSET_FOR_BUDGET_PERIOD_DAILY";
    /**
     * The period of the budget is not allowed.
     *
     * @generated from protobuf enum value: INVALID_PERIOD = 19;
     */
    CampaignBudgetErrorEnum_CampaignBudgetError[CampaignBudgetErrorEnum_CampaignBudgetError["INVALID_PERIOD"] = 19] = "INVALID_PERIOD";
})(CampaignBudgetErrorEnum_CampaignBudgetError = exports.CampaignBudgetErrorEnum_CampaignBudgetError || (exports.CampaignBudgetErrorEnum_CampaignBudgetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CampaignBudgetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CampaignBudgetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CampaignBudgetErrorEnum
 */
exports.CampaignBudgetErrorEnum = new CampaignBudgetErrorEnum$Type();
