// google-ads-api-client/dist/protos/bindings/ads/errors/change_event_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeEventErrorEnum = exports.ChangeEventErrorEnum_ChangeEventError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible change event errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ChangeEventErrorEnum.ChangeEventError
 */
var ChangeEventErrorEnum_ChangeEventError;
(function (ChangeEventErrorEnum_ChangeEventError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ChangeEventErrorEnum_ChangeEventError[ChangeEventErrorEnum_ChangeEventError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ChangeEventErrorEnum_ChangeEventError[ChangeEventErrorEnum_ChangeEventError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The requested start date is too old. It cannot be older than 30 days.
     *
     * @generated from protobuf enum value: START_DATE_TOO_OLD = 2;
     */
    ChangeEventErrorEnum_ChangeEventError[ChangeEventErrorEnum_ChangeEventError["START_DATE_TOO_OLD"] = 2] = "START_DATE_TOO_OLD";
    /**
     * The change_event search request must specify a finite range filter
     * on change_date_time.
     *
     * @generated from protobuf enum value: CHANGE_DATE_RANGE_INFINITE = 3;
     */
    ChangeEventErrorEnum_ChangeEventError[ChangeEventErrorEnum_ChangeEventError["CHANGE_DATE_RANGE_INFINITE"] = 3] = "CHANGE_DATE_RANGE_INFINITE";
    /**
     * The change event search request has specified invalid date time filters
     * that can never logically produce any valid results (for example, start
     * time after end time).
     *
     * @generated from protobuf enum value: CHANGE_DATE_RANGE_NEGATIVE = 4;
     */
    ChangeEventErrorEnum_ChangeEventError[ChangeEventErrorEnum_ChangeEventError["CHANGE_DATE_RANGE_NEGATIVE"] = 4] = "CHANGE_DATE_RANGE_NEGATIVE";
    /**
     * The change_event search request must specify a LIMIT.
     *
     * @generated from protobuf enum value: LIMIT_NOT_SPECIFIED = 5;
     */
    ChangeEventErrorEnum_ChangeEventError[ChangeEventErrorEnum_ChangeEventError["LIMIT_NOT_SPECIFIED"] = 5] = "LIMIT_NOT_SPECIFIED";
    /**
     * The LIMIT specified by change_event request should be less than or equal
     * to 10K.
     *
     * @generated from protobuf enum value: INVALID_LIMIT_CLAUSE = 6;
     */
    ChangeEventErrorEnum_ChangeEventError[ChangeEventErrorEnum_ChangeEventError["INVALID_LIMIT_CLAUSE"] = 6] = "INVALID_LIMIT_CLAUSE";
})(ChangeEventErrorEnum_ChangeEventError = exports.ChangeEventErrorEnum_ChangeEventError || (exports.ChangeEventErrorEnum_ChangeEventError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ChangeEventErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ChangeEventErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ChangeEventErrorEnum
 */
exports.ChangeEventErrorEnum = new ChangeEventErrorEnum$Type();
