// google-ads-api-client/dist/protos/bindings/ads/errors/collection_size_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CollectionSizeErrorEnum = exports.CollectionSizeErrorEnum_CollectionSizeError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible collection size errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CollectionSizeErrorEnum.CollectionSizeError
 */
var CollectionSizeErrorEnum_CollectionSizeError;
(function (CollectionSizeErrorEnum_CollectionSizeError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CollectionSizeErrorEnum_CollectionSizeError[CollectionSizeErrorEnum_CollectionSizeError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CollectionSizeErrorEnum_CollectionSizeError[CollectionSizeErrorEnum_CollectionSizeError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Too few.
     *
     * @generated from protobuf enum value: TOO_FEW = 2;
     */
    CollectionSizeErrorEnum_CollectionSizeError[CollectionSizeErrorEnum_CollectionSizeError["TOO_FEW"] = 2] = "TOO_FEW";
    /**
     * Too many.
     *
     * @generated from protobuf enum value: TOO_MANY = 3;
     */
    CollectionSizeErrorEnum_CollectionSizeError[CollectionSizeErrorEnum_CollectionSizeError["TOO_MANY"] = 3] = "TOO_MANY";
})(CollectionSizeErrorEnum_CollectionSizeError = exports.CollectionSizeErrorEnum_CollectionSizeError || (exports.CollectionSizeErrorEnum_CollectionSizeError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CollectionSizeErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CollectionSizeErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CollectionSizeErrorEnum
 */
exports.CollectionSizeErrorEnum = new CollectionSizeErrorEnum$Type();
