// google-ads-api-client/dist/protos/bindings/ads/errors/context_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContextErrorEnum = exports.ContextErrorEnum_ContextError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible context errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ContextErrorEnum.ContextError
 */
var ContextErrorEnum_ContextError;
(function (ContextErrorEnum_ContextError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ContextErrorEnum_ContextError[ContextErrorEnum_ContextError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ContextErrorEnum_ContextError[ContextErrorEnum_ContextError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The operation is not allowed for the given context.
     *
     * @generated from protobuf enum value: OPERATION_NOT_PERMITTED_FOR_CONTEXT = 2;
     */
    ContextErrorEnum_ContextError[ContextErrorEnum_ContextError["OPERATION_NOT_PERMITTED_FOR_CONTEXT"] = 2] = "OPERATION_NOT_PERMITTED_FOR_CONTEXT";
    /**
     * The operation is not allowed for removed resources.
     *
     * @generated from protobuf enum value: OPERATION_NOT_PERMITTED_FOR_REMOVED_RESOURCE = 3;
     */
    ContextErrorEnum_ContextError[ContextErrorEnum_ContextError["OPERATION_NOT_PERMITTED_FOR_REMOVED_RESOURCE"] = 3] = "OPERATION_NOT_PERMITTED_FOR_REMOVED_RESOURCE";
})(ContextErrorEnum_ContextError = exports.ContextErrorEnum_ContextError || (exports.ContextErrorEnum_ContextError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ContextErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ContextErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ContextErrorEnum
 */
exports.ContextErrorEnum = new ContextErrorEnum$Type();
