// google-ads-api-client/dist/protos/bindings/ads/errors/conversion_goal_campaign_config_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionGoalCampaignConfigErrorEnum = exports.ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible conversion goal campaign config errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ConversionGoalCampaignConfigErrorEnum.ConversionGoalCampaignConfigError
 */
var ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError;
(function (ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError[ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError[ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Campaign is managed by Search Ads 360 but uses Unified Goal.
     *
     * @generated from protobuf enum value: CANNOT_USE_CAMPAIGN_GOAL_FOR_SEARCH_ADS_360_MANAGED_CAMPAIGN = 2;
     */
    ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError[ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError["CANNOT_USE_CAMPAIGN_GOAL_FOR_SEARCH_ADS_360_MANAGED_CAMPAIGN"] = 2] = "CANNOT_USE_CAMPAIGN_GOAL_FOR_SEARCH_ADS_360_MANAGED_CAMPAIGN";
    /**
     * The campaign is using a custom goal that does not belong to its Google
     * Ads conversion customer (conversion tracking customer).
     *
     * @generated from protobuf enum value: CUSTOM_GOAL_DOES_NOT_BELONG_TO_GOOGLE_ADS_CONVERSION_CUSTOMER = 3;
     */
    ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError[ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError["CUSTOM_GOAL_DOES_NOT_BELONG_TO_GOOGLE_ADS_CONVERSION_CUSTOMER"] = 3] = "CUSTOM_GOAL_DOES_NOT_BELONG_TO_GOOGLE_ADS_CONVERSION_CUSTOMER";
    /**
     * The campaign is not allowed to use unified goals.
     *
     * @generated from protobuf enum value: CAMPAIGN_CANNOT_USE_UNIFIED_GOALS = 4;
     */
    ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError[ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError["CAMPAIGN_CANNOT_USE_UNIFIED_GOALS"] = 4] = "CAMPAIGN_CANNOT_USE_UNIFIED_GOALS";
})(ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError = exports.ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError || (exports.ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionGoalCampaignConfigErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ConversionGoalCampaignConfigErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ConversionGoalCampaignConfigErrorEnum
 */
exports.ConversionGoalCampaignConfigErrorEnum = new ConversionGoalCampaignConfigErrorEnum$Type();
