// google-ads-api-client/dist/protos/bindings/ads/errors/conversion_action_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionActionErrorEnum = exports.ConversionActionErrorEnum_ConversionActionError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible conversion action errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ConversionActionErrorEnum.ConversionActionError
 */
var ConversionActionErrorEnum_ConversionActionError;
(function (ConversionActionErrorEnum_ConversionActionError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The specified conversion action name already exists.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 2;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["DUPLICATE_NAME"] = 2] = "DUPLICATE_NAME";
    /**
     * Another conversion action with the specified app id already exists.
     *
     * @generated from protobuf enum value: DUPLICATE_APP_ID = 3;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["DUPLICATE_APP_ID"] = 3] = "DUPLICATE_APP_ID";
    /**
     * Android first open action conflicts with Google play codeless download
     * action tracking the same app.
     *
     * @generated from protobuf enum value: TWO_CONVERSION_ACTIONS_BIDDING_ON_SAME_APP_DOWNLOAD = 4;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["TWO_CONVERSION_ACTIONS_BIDDING_ON_SAME_APP_DOWNLOAD"] = 4] = "TWO_CONVERSION_ACTIONS_BIDDING_ON_SAME_APP_DOWNLOAD";
    /**
     * Android first open action conflicts with Google play codeless download
     * action tracking the same app.
     *
     * @generated from protobuf enum value: BIDDING_ON_SAME_APP_DOWNLOAD_AS_GLOBAL_ACTION = 5;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["BIDDING_ON_SAME_APP_DOWNLOAD_AS_GLOBAL_ACTION"] = 5] = "BIDDING_ON_SAME_APP_DOWNLOAD_AS_GLOBAL_ACTION";
    /**
     * The attribution model cannot be set to DATA_DRIVEN because a data-driven
     * model has never been generated.
     *
     * @generated from protobuf enum value: DATA_DRIVEN_MODEL_WAS_NEVER_GENERATED = 6;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["DATA_DRIVEN_MODEL_WAS_NEVER_GENERATED"] = 6] = "DATA_DRIVEN_MODEL_WAS_NEVER_GENERATED";
    /**
     * The attribution model cannot be set to DATA_DRIVEN because the
     * data-driven model is expired.
     *
     * @generated from protobuf enum value: DATA_DRIVEN_MODEL_EXPIRED = 7;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["DATA_DRIVEN_MODEL_EXPIRED"] = 7] = "DATA_DRIVEN_MODEL_EXPIRED";
    /**
     * The attribution model cannot be set to DATA_DRIVEN because the
     * data-driven model is stale.
     *
     * @generated from protobuf enum value: DATA_DRIVEN_MODEL_STALE = 8;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["DATA_DRIVEN_MODEL_STALE"] = 8] = "DATA_DRIVEN_MODEL_STALE";
    /**
     * The attribution model cannot be set to DATA_DRIVEN because the
     * data-driven model is unavailable or the conversion action was newly
     * added.
     *
     * @generated from protobuf enum value: DATA_DRIVEN_MODEL_UNKNOWN = 9;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["DATA_DRIVEN_MODEL_UNKNOWN"] = 9] = "DATA_DRIVEN_MODEL_UNKNOWN";
    /**
     * Creation of this conversion action type isn't supported by Google
     * Ads API.
     *
     * @generated from protobuf enum value: CREATION_NOT_SUPPORTED = 10;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["CREATION_NOT_SUPPORTED"] = 10] = "CREATION_NOT_SUPPORTED";
    /**
     * Update of this conversion action isn't supported by Google Ads API.
     *
     * @generated from protobuf enum value: UPDATE_NOT_SUPPORTED = 11;
     */
    ConversionActionErrorEnum_ConversionActionError[ConversionActionErrorEnum_ConversionActionError["UPDATE_NOT_SUPPORTED"] = 11] = "UPDATE_NOT_SUPPORTED";
})(ConversionActionErrorEnum_ConversionActionError = exports.ConversionActionErrorEnum_ConversionActionError || (exports.ConversionActionErrorEnum_ConversionActionError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionActionErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ConversionActionErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ConversionActionErrorEnum
 */
exports.ConversionActionErrorEnum = new ConversionActionErrorEnum$Type();
