// google-ads-api-client/dist/protos/bindings/ads/errors/conversion_custom_variable_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionCustomVariableErrorEnum = exports.ConversionCustomVariableErrorEnum_ConversionCustomVariableError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible conversion custom variable errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ConversionCustomVariableErrorEnum.ConversionCustomVariableError
 */
var ConversionCustomVariableErrorEnum_ConversionCustomVariableError;
(function (ConversionCustomVariableErrorEnum_ConversionCustomVariableError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionCustomVariableErrorEnum_ConversionCustomVariableError[ConversionCustomVariableErrorEnum_ConversionCustomVariableError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionCustomVariableErrorEnum_ConversionCustomVariableError[ConversionCustomVariableErrorEnum_ConversionCustomVariableError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A conversion custom variable with the specified name already exists.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 2;
     */
    ConversionCustomVariableErrorEnum_ConversionCustomVariableError[ConversionCustomVariableErrorEnum_ConversionCustomVariableError["DUPLICATE_NAME"] = 2] = "DUPLICATE_NAME";
    /**
     * A conversion custom variable with the specified tag already exists.
     *
     * @generated from protobuf enum value: DUPLICATE_TAG = 3;
     */
    ConversionCustomVariableErrorEnum_ConversionCustomVariableError[ConversionCustomVariableErrorEnum_ConversionCustomVariableError["DUPLICATE_TAG"] = 3] = "DUPLICATE_TAG";
    /**
     * A conversion custom variable with the specified tag is reserved for other
     * uses.
     *
     * @generated from protobuf enum value: RESERVED_TAG = 4;
     */
    ConversionCustomVariableErrorEnum_ConversionCustomVariableError[ConversionCustomVariableErrorEnum_ConversionCustomVariableError["RESERVED_TAG"] = 4] = "RESERVED_TAG";
})(ConversionCustomVariableErrorEnum_ConversionCustomVariableError = exports.ConversionCustomVariableErrorEnum_ConversionCustomVariableError || (exports.ConversionCustomVariableErrorEnum_ConversionCustomVariableError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionCustomVariableErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ConversionCustomVariableErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ConversionCustomVariableErrorEnum
 */
exports.ConversionCustomVariableErrorEnum = new ConversionCustomVariableErrorEnum$Type();
