// google-ads-api-client/dist/protos/bindings/ads/errors/conversion_value_rule_set_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionValueRuleSetErrorEnum = exports.ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible conversion value rule set errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ConversionValueRuleSetErrorEnum.ConversionValueRuleSetError
 */
var ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError;
(function (ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Two value rules in this value rule set contain conflicting conditions.
     *
     * @generated from protobuf enum value: CONFLICTING_VALUE_RULE_CONDITIONS = 2;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["CONFLICTING_VALUE_RULE_CONDITIONS"] = 2] = "CONFLICTING_VALUE_RULE_CONDITIONS";
    /**
     * This value rule set includes a value rule that cannot be found, has been
     * permanently removed or belongs to a different customer.
     *
     * @generated from protobuf enum value: INVALID_VALUE_RULE = 3;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["INVALID_VALUE_RULE"] = 3] = "INVALID_VALUE_RULE";
    /**
     * An error that's thrown when a mutate operation is trying to
     * replace/remove some existing elements in the dimensions field. In other
     * words, ADD op is always fine and UPDATE op is fine if it's only appending
     * new elements into dimensions list.
     *
     * @generated from protobuf enum value: DIMENSIONS_UPDATE_ONLY_ALLOW_APPEND = 4;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["DIMENSIONS_UPDATE_ONLY_ALLOW_APPEND"] = 4] = "DIMENSIONS_UPDATE_ONLY_ALLOW_APPEND";
    /**
     * An error that's thrown when a mutate is adding new value rule(s) into a
     * value rule set and the added value rule(s) include conditions that are
     * not specified in the dimensions of the value rule set.
     *
     * @generated from protobuf enum value: CONDITION_TYPE_NOT_ALLOWED = 5;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["CONDITION_TYPE_NOT_ALLOWED"] = 5] = "CONDITION_TYPE_NOT_ALLOWED";
    /**
     * The dimensions field contains duplicate elements.
     *
     * @generated from protobuf enum value: DUPLICATE_DIMENSIONS = 6;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["DUPLICATE_DIMENSIONS"] = 6] = "DUPLICATE_DIMENSIONS";
    /**
     * This value rule set is attached to an invalid campaign id. Either a
     * campaign with this campaign id doesn't exist or it belongs to a different
     * customer.
     *
     * @generated from protobuf enum value: INVALID_CAMPAIGN_ID = 7;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["INVALID_CAMPAIGN_ID"] = 7] = "INVALID_CAMPAIGN_ID";
    /**
     * When a mutate request tries to pause a value rule set, the enabled
     * value rules in this set must be paused in the same command, or this error
     * will be thrown.
     *
     * @generated from protobuf enum value: CANNOT_PAUSE_UNLESS_ALL_VALUE_RULES_ARE_PAUSED = 8;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["CANNOT_PAUSE_UNLESS_ALL_VALUE_RULES_ARE_PAUSED"] = 8] = "CANNOT_PAUSE_UNLESS_ALL_VALUE_RULES_ARE_PAUSED";
    /**
     * When a mutate request tries to pause all the value rules in a value rule
     * set, the value rule set must be paused, or this error will be thrown.
     *
     * @generated from protobuf enum value: SHOULD_PAUSE_WHEN_ALL_VALUE_RULES_ARE_PAUSED = 9;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["SHOULD_PAUSE_WHEN_ALL_VALUE_RULES_ARE_PAUSED"] = 9] = "SHOULD_PAUSE_WHEN_ALL_VALUE_RULES_ARE_PAUSED";
    /**
     * This value rule set is attached to a campaign that does not support value
     * rules. Currently, campaign level value rule sets can only be created on
     * Search, or Display campaigns.
     *
     * @generated from protobuf enum value: VALUE_RULES_NOT_SUPPORTED_FOR_CAMPAIGN_TYPE = 10;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["VALUE_RULES_NOT_SUPPORTED_FOR_CAMPAIGN_TYPE"] = 10] = "VALUE_RULES_NOT_SUPPORTED_FOR_CAMPAIGN_TYPE";
    /**
     * To add a value rule set that applies on Store Visits/Store Sales
     * conversion action categories, the customer must have valid Store Visits/
     * Store Sales conversion actions.
     *
     * @generated from protobuf enum value: INELIGIBLE_CONVERSION_ACTION_CATEGORIES = 11;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["INELIGIBLE_CONVERSION_ACTION_CATEGORIES"] = 11] = "INELIGIBLE_CONVERSION_ACTION_CATEGORIES";
    /**
     * If NO_CONDITION is used as a dimension of a value rule set, it must be
     * the only dimension.
     *
     * @generated from protobuf enum value: DIMENSION_NO_CONDITION_USED_WITH_OTHER_DIMENSIONS = 12;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["DIMENSION_NO_CONDITION_USED_WITH_OTHER_DIMENSIONS"] = 12] = "DIMENSION_NO_CONDITION_USED_WITH_OTHER_DIMENSIONS";
    /**
     * Dimension NO_CONDITION can only be used by Store Visits/Store Sales value
     * rule set.
     *
     * @generated from protobuf enum value: DIMENSION_NO_CONDITION_NOT_ALLOWED = 13;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["DIMENSION_NO_CONDITION_NOT_ALLOWED"] = 13] = "DIMENSION_NO_CONDITION_NOT_ALLOWED";
    /**
     * Value rule sets defined on the specified conversion action categories are
     * not supported. The list of conversion action categories must be an empty
     * list, only STORE_VISIT, or only STORE_SALE.
     *
     * @generated from protobuf enum value: UNSUPPORTED_CONVERSION_ACTION_CATEGORIES = 14;
     */
    ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError[ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError["UNSUPPORTED_CONVERSION_ACTION_CATEGORIES"] = 14] = "UNSUPPORTED_CONVERSION_ACTION_CATEGORIES";
})(ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError = exports.ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError || (exports.ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionValueRuleSetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ConversionValueRuleSetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ConversionValueRuleSetErrorEnum
 */
exports.ConversionValueRuleSetErrorEnum = new ConversionValueRuleSetErrorEnum$Type();
