// google-ads-api-client/dist/protos/bindings/ads/errors/conversion_upload_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionUploadErrorEnum = exports.ConversionUploadErrorEnum_ConversionUploadError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible conversion upload errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ConversionUploadErrorEnum.ConversionUploadError
 */
var ConversionUploadErrorEnum_ConversionUploadError;
(function (ConversionUploadErrorEnum_ConversionUploadError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The request contained more than 2000 conversions.
     *
     * @generated from protobuf enum value: TOO_MANY_CONVERSIONS_IN_REQUEST = 2;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["TOO_MANY_CONVERSIONS_IN_REQUEST"] = 2] = "TOO_MANY_CONVERSIONS_IN_REQUEST";
    /**
     * The specified gclid could not be decoded.
     *
     * @generated from protobuf enum value: UNPARSEABLE_GCLID = 3;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["UNPARSEABLE_GCLID"] = 3] = "UNPARSEABLE_GCLID";
    /**
     * The specified conversion_date_time is before the event time
     * associated with the given identifier or iOS URL parameter.
     *
     * @generated from protobuf enum value: CONVERSION_PRECEDES_EVENT = 42;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CONVERSION_PRECEDES_EVENT"] = 42] = "CONVERSION_PRECEDES_EVENT";
    /**
     * The click associated with the given identifier or iOS URL parameter is
     * either too old to be imported or occurred outside of the click through
     * lookback window for the specified conversion action.
     *
     * @generated from protobuf enum value: EXPIRED_EVENT = 43;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["EXPIRED_EVENT"] = 43] = "EXPIRED_EVENT";
    /**
     * The click associated with the given identifier or iOS URL parameter
     * occurred too recently. Try uploading again after 6 hours have
     * passed since the click occurred.
     *
     * @generated from protobuf enum value: TOO_RECENT_EVENT = 44;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["TOO_RECENT_EVENT"] = 44] = "TOO_RECENT_EVENT";
    /**
     * The click associated with the given identifier or iOS URL parameter could
     * not be found in the system. This can happen if the identifier or iOS URL
     * parameter are collected for non Google Ads clicks.
     *
     * @generated from protobuf enum value: EVENT_NOT_FOUND = 45;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["EVENT_NOT_FOUND"] = 45] = "EVENT_NOT_FOUND";
    /**
     * The click associated with the given identifier or iOS URL parameter is
     * owned by a customer account that the uploading customer does not manage.
     *
     * @generated from protobuf enum value: UNAUTHORIZED_CUSTOMER = 8;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["UNAUTHORIZED_CUSTOMER"] = 8] = "UNAUTHORIZED_CUSTOMER";
    /**
     * No upload eligible conversion action that matches the provided
     * information can be found for the customer.
     *
     * @generated from protobuf enum value: INVALID_CONVERSION_ACTION = 9;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["INVALID_CONVERSION_ACTION"] = 9] = "INVALID_CONVERSION_ACTION";
    /**
     * The specified conversion action was created too recently.
     * Try the upload again after 4-6 hours have passed since the
     * conversion action was created.
     *
     * @generated from protobuf enum value: TOO_RECENT_CONVERSION_ACTION = 10;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["TOO_RECENT_CONVERSION_ACTION"] = 10] = "TOO_RECENT_CONVERSION_ACTION";
    /**
     * The click associated with the given identifier does not contain
     * conversion tracking information.
     *
     * @generated from protobuf enum value: CONVERSION_TRACKING_NOT_ENABLED_AT_IMPRESSION_TIME = 11;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CONVERSION_TRACKING_NOT_ENABLED_AT_IMPRESSION_TIME"] = 11] = "CONVERSION_TRACKING_NOT_ENABLED_AT_IMPRESSION_TIME";
    /**
     * The specified conversion action does not use an external attribution
     * model, but external_attribution_data was set.
     *
     * @generated from protobuf enum value: EXTERNAL_ATTRIBUTION_DATA_SET_FOR_NON_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION = 12;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["EXTERNAL_ATTRIBUTION_DATA_SET_FOR_NON_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION"] = 12] = "EXTERNAL_ATTRIBUTION_DATA_SET_FOR_NON_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION";
    /**
     * The specified conversion action uses an external attribution model, but
     * external_attribution_data or one of its contained fields was not set.
     * Both external_attribution_credit and external_attribution_model must be
     * set for externally attributed conversion actions.
     *
     * @generated from protobuf enum value: EXTERNAL_ATTRIBUTION_DATA_NOT_SET_FOR_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION = 13;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["EXTERNAL_ATTRIBUTION_DATA_NOT_SET_FOR_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION"] = 13] = "EXTERNAL_ATTRIBUTION_DATA_NOT_SET_FOR_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION";
    /**
     * Order IDs are not supported for conversion actions which use an external
     * attribution model.
     *
     * @generated from protobuf enum value: ORDER_ID_NOT_PERMITTED_FOR_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION = 14;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["ORDER_ID_NOT_PERMITTED_FOR_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION"] = 14] = "ORDER_ID_NOT_PERMITTED_FOR_EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION";
    /**
     * A conversion with the same order id and conversion action combination
     * already exists in our system.
     *
     * @generated from protobuf enum value: ORDER_ID_ALREADY_IN_USE = 15;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["ORDER_ID_ALREADY_IN_USE"] = 15] = "ORDER_ID_ALREADY_IN_USE";
    /**
     * The request contained two or more conversions with the same order id and
     * conversion action combination.
     *
     * @generated from protobuf enum value: DUPLICATE_ORDER_ID = 16;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["DUPLICATE_ORDER_ID"] = 16] = "DUPLICATE_ORDER_ID";
    /**
     * The call occurred too recently. Try uploading again after 12 hours
     * have passed since the call occurred.
     *
     * @generated from protobuf enum value: TOO_RECENT_CALL = 17;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["TOO_RECENT_CALL"] = 17] = "TOO_RECENT_CALL";
    /**
     * The click that initiated the call is too old for this conversion to be
     * imported.
     *
     * @generated from protobuf enum value: EXPIRED_CALL = 18;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["EXPIRED_CALL"] = 18] = "EXPIRED_CALL";
    /**
     * The call or the click leading to the call was not found.
     *
     * @generated from protobuf enum value: CALL_NOT_FOUND = 19;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CALL_NOT_FOUND"] = 19] = "CALL_NOT_FOUND";
    /**
     * The specified conversion_date_time is before the call_start_date_time.
     *
     * @generated from protobuf enum value: CONVERSION_PRECEDES_CALL = 20;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CONVERSION_PRECEDES_CALL"] = 20] = "CONVERSION_PRECEDES_CALL";
    /**
     * The click associated with the call does not contain conversion tracking
     * information.
     *
     * @generated from protobuf enum value: CONVERSION_TRACKING_NOT_ENABLED_AT_CALL_TIME = 21;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CONVERSION_TRACKING_NOT_ENABLED_AT_CALL_TIME"] = 21] = "CONVERSION_TRACKING_NOT_ENABLED_AT_CALL_TIME";
    /**
     * The caller's phone number cannot be parsed. It should be formatted either
     * as E.164 "+16502531234", International "+64 3-331 6005" or US national
     * number "6502531234".
     *
     * @generated from protobuf enum value: UNPARSEABLE_CALLERS_PHONE_NUMBER = 22;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["UNPARSEABLE_CALLERS_PHONE_NUMBER"] = 22] = "UNPARSEABLE_CALLERS_PHONE_NUMBER";
    /**
     * A conversion with this timestamp already exists for this click. To upload
     * another conversion, use a different timestamp.
     *
     * @generated from protobuf enum value: CLICK_CONVERSION_ALREADY_EXISTS = 23;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CLICK_CONVERSION_ALREADY_EXISTS"] = 23] = "CLICK_CONVERSION_ALREADY_EXISTS";
    /**
     * A conversion with this timestamp already exists for this call. To upload
     * another conversion, use a different timestamp.
     *
     * @generated from protobuf enum value: CALL_CONVERSION_ALREADY_EXISTS = 24;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CALL_CONVERSION_ALREADY_EXISTS"] = 24] = "CALL_CONVERSION_ALREADY_EXISTS";
    /**
     * This conversion has the same click and timestamp as another conversion in
     * the request. To upload another conversion for this click, use a
     * different timestamp.
     *
     * @generated from protobuf enum value: DUPLICATE_CLICK_CONVERSION_IN_REQUEST = 25;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["DUPLICATE_CLICK_CONVERSION_IN_REQUEST"] = 25] = "DUPLICATE_CLICK_CONVERSION_IN_REQUEST";
    /**
     * This conversion has the same call and timestamp as another conversion in
     * the request. To upload another conversion for this call, use a
     * different timestamp.
     *
     * @generated from protobuf enum value: DUPLICATE_CALL_CONVERSION_IN_REQUEST = 26;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["DUPLICATE_CALL_CONVERSION_IN_REQUEST"] = 26] = "DUPLICATE_CALL_CONVERSION_IN_REQUEST";
    /**
     * The custom variable is not enabled.
     *
     * @generated from protobuf enum value: CUSTOM_VARIABLE_NOT_ENABLED = 28;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CUSTOM_VARIABLE_NOT_ENABLED"] = 28] = "CUSTOM_VARIABLE_NOT_ENABLED";
    /**
     * The value of the custom variable contains personally identifiable
     * information (PII), such as an email address or phone number.
     *
     * @generated from protobuf enum value: CUSTOM_VARIABLE_VALUE_CONTAINS_PII = 29;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CUSTOM_VARIABLE_VALUE_CONTAINS_PII"] = 29] = "CUSTOM_VARIABLE_VALUE_CONTAINS_PII";
    /**
     * The click associated with the given identifier or iOS URL parameter isn't
     * from the account where conversion tracking is set up.
     *
     * @generated from protobuf enum value: INVALID_CUSTOMER_FOR_CLICK = 30;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["INVALID_CUSTOMER_FOR_CLICK"] = 30] = "INVALID_CUSTOMER_FOR_CLICK";
    /**
     * The click associated with the given call isn't from the account where
     * conversion tracking is set up.
     *
     * @generated from protobuf enum value: INVALID_CUSTOMER_FOR_CALL = 31;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["INVALID_CUSTOMER_FOR_CALL"] = 31] = "INVALID_CUSTOMER_FOR_CALL";
    /**
     * The conversion can't be uploaded because the conversion source didn't
     * comply with the App Tracking Transparency (ATT) policy or the person who
     * converted didn't consent to tracking.
     *
     * @generated from protobuf enum value: CONVERSION_NOT_COMPLIANT_WITH_ATT_POLICY = 32;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CONVERSION_NOT_COMPLIANT_WITH_ATT_POLICY"] = 32] = "CONVERSION_NOT_COMPLIANT_WITH_ATT_POLICY";
    /**
     * No click was found for the provided user identifiers. This may be because
     * the conversion did not come from a Google Ads campaign.
     *
     * @generated from protobuf enum value: CLICK_NOT_FOUND = 33;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CLICK_NOT_FOUND"] = 33] = "CLICK_NOT_FOUND";
    /**
     * The provided user identifier is not a SHA-256 hash. It is either unhashed
     * or hashed using a different hash function.
     *
     * @generated from protobuf enum value: INVALID_USER_IDENTIFIER = 34;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["INVALID_USER_IDENTIFIER"] = 34] = "INVALID_USER_IDENTIFIER";
    /**
     * Conversion actions which use an external attribution model cannot be used
     * with UserIdentifier.
     *
     * @generated from protobuf enum value: EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION_NOT_PERMITTED_WITH_USER_IDENTIFIER = 35;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION_NOT_PERMITTED_WITH_USER_IDENTIFIER"] = 35] = "EXTERNALLY_ATTRIBUTED_CONVERSION_ACTION_NOT_PERMITTED_WITH_USER_IDENTIFIER";
    /**
     * The provided user identifier is not supported. ConversionUploadService
     * only supports hashed_email and hashed_phone_number.
     *
     * @generated from protobuf enum value: UNSUPPORTED_USER_IDENTIFIER = 36;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["UNSUPPORTED_USER_IDENTIFIER"] = 36] = "UNSUPPORTED_USER_IDENTIFIER";
    /**
     * gbraid and wbraid are both set in the request. Only one is allowed.
     *
     * @generated from protobuf enum value: GBRAID_WBRAID_BOTH_SET = 38;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["GBRAID_WBRAID_BOTH_SET"] = 38] = "GBRAID_WBRAID_BOTH_SET";
    /**
     * The specified wbraid could not be decoded.
     *
     * @generated from protobuf enum value: UNPARSEABLE_WBRAID = 39;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["UNPARSEABLE_WBRAID"] = 39] = "UNPARSEABLE_WBRAID";
    /**
     * The specified gbraid could not be decoded.
     *
     * @generated from protobuf enum value: UNPARSEABLE_GBRAID = 40;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["UNPARSEABLE_GBRAID"] = 40] = "UNPARSEABLE_GBRAID";
    /**
     * Conversion actions which use the one-per-click counting type cannot be
     * used with gbraid or wbraid.
     *
     * @generated from protobuf enum value: ONE_PER_CLICK_CONVERSION_ACTION_NOT_PERMITTED_WITH_BRAID = 46;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["ONE_PER_CLICK_CONVERSION_ACTION_NOT_PERMITTED_WITH_BRAID"] = 46] = "ONE_PER_CLICK_CONVERSION_ACTION_NOT_PERMITTED_WITH_BRAID";
    /**
     * Per our customer data policies, enhanced conversions have been prohibited
     * in your account. If you have any questions, contact your Google
     * representative.
     *
     * @generated from protobuf enum value: CUSTOMER_DATA_POLICY_PROHIBITS_ENHANCED_CONVERSIONS = 47;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CUSTOMER_DATA_POLICY_PROHIBITS_ENHANCED_CONVERSIONS"] = 47] = "CUSTOMER_DATA_POLICY_PROHIBITS_ENHANCED_CONVERSIONS";
    /**
     * The customer has not accepted the customer data terms in the conversion
     * settings page.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_ACCEPTED_CUSTOMER_DATA_TERMS = 48;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["CUSTOMER_NOT_ACCEPTED_CUSTOMER_DATA_TERMS"] = 48] = "CUSTOMER_NOT_ACCEPTED_CUSTOMER_DATA_TERMS";
    /**
     * The order_id contains personally identifiable information (PII), such as
     * an email address or phone number.
     *
     * @generated from protobuf enum value: ORDER_ID_CONTAINS_PII = 49;
     */
    ConversionUploadErrorEnum_ConversionUploadError[ConversionUploadErrorEnum_ConversionUploadError["ORDER_ID_CONTAINS_PII"] = 49] = "ORDER_ID_CONTAINS_PII";
})(ConversionUploadErrorEnum_ConversionUploadError = exports.ConversionUploadErrorEnum_ConversionUploadError || (exports.ConversionUploadErrorEnum_ConversionUploadError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ConversionUploadErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ConversionUploadErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ConversionUploadErrorEnum
 */
exports.ConversionUploadErrorEnum = new ConversionUploadErrorEnum$Type();
