// google-ads-api-client/dist/protos/bindings/ads/errors/criterion_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CriterionErrorEnum = exports.CriterionErrorEnum_CriterionError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible criterion errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CriterionErrorEnum.CriterionError
 */
var CriterionErrorEnum_CriterionError;
(function (CriterionErrorEnum_CriterionError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Concrete type of criterion is required for CREATE and UPDATE operations.
     *
     * @generated from protobuf enum value: CONCRETE_TYPE_REQUIRED = 2;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CONCRETE_TYPE_REQUIRED"] = 2] = "CONCRETE_TYPE_REQUIRED";
    /**
     * The category requested for exclusion is invalid.
     *
     * @generated from protobuf enum value: INVALID_EXCLUDED_CATEGORY = 3;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_EXCLUDED_CATEGORY"] = 3] = "INVALID_EXCLUDED_CATEGORY";
    /**
     * Invalid keyword criteria text.
     *
     * @generated from protobuf enum value: INVALID_KEYWORD_TEXT = 4;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_KEYWORD_TEXT"] = 4] = "INVALID_KEYWORD_TEXT";
    /**
     * Keyword text should be less than 80 chars.
     *
     * @generated from protobuf enum value: KEYWORD_TEXT_TOO_LONG = 5;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["KEYWORD_TEXT_TOO_LONG"] = 5] = "KEYWORD_TEXT_TOO_LONG";
    /**
     * Keyword text has too many words.
     *
     * @generated from protobuf enum value: KEYWORD_HAS_TOO_MANY_WORDS = 6;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["KEYWORD_HAS_TOO_MANY_WORDS"] = 6] = "KEYWORD_HAS_TOO_MANY_WORDS";
    /**
     * Keyword text has invalid characters or symbols.
     *
     * @generated from protobuf enum value: KEYWORD_HAS_INVALID_CHARS = 7;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["KEYWORD_HAS_INVALID_CHARS"] = 7] = "KEYWORD_HAS_INVALID_CHARS";
    /**
     * Invalid placement URL.
     *
     * @generated from protobuf enum value: INVALID_PLACEMENT_URL = 8;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_PLACEMENT_URL"] = 8] = "INVALID_PLACEMENT_URL";
    /**
     * Invalid user list criterion.
     *
     * @generated from protobuf enum value: INVALID_USER_LIST = 9;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_USER_LIST"] = 9] = "INVALID_USER_LIST";
    /**
     * Invalid user interest criterion.
     *
     * @generated from protobuf enum value: INVALID_USER_INTEREST = 10;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_USER_INTEREST"] = 10] = "INVALID_USER_INTEREST";
    /**
     * Placement URL has wrong format.
     *
     * @generated from protobuf enum value: INVALID_FORMAT_FOR_PLACEMENT_URL = 11;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_FORMAT_FOR_PLACEMENT_URL"] = 11] = "INVALID_FORMAT_FOR_PLACEMENT_URL";
    /**
     * Placement URL is too long.
     *
     * @generated from protobuf enum value: PLACEMENT_URL_IS_TOO_LONG = 12;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["PLACEMENT_URL_IS_TOO_LONG"] = 12] = "PLACEMENT_URL_IS_TOO_LONG";
    /**
     * Indicates the URL contains an illegal character.
     *
     * @generated from protobuf enum value: PLACEMENT_URL_HAS_ILLEGAL_CHAR = 13;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["PLACEMENT_URL_HAS_ILLEGAL_CHAR"] = 13] = "PLACEMENT_URL_HAS_ILLEGAL_CHAR";
    /**
     * Indicates the URL contains multiple comma separated URLs.
     *
     * @generated from protobuf enum value: PLACEMENT_URL_HAS_MULTIPLE_SITES_IN_LINE = 14;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["PLACEMENT_URL_HAS_MULTIPLE_SITES_IN_LINE"] = 14] = "PLACEMENT_URL_HAS_MULTIPLE_SITES_IN_LINE";
    /**
     * Indicates the domain is blocked.
     *
     * @generated from protobuf enum value: PLACEMENT_IS_NOT_AVAILABLE_FOR_TARGETING_OR_EXCLUSION = 15;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["PLACEMENT_IS_NOT_AVAILABLE_FOR_TARGETING_OR_EXCLUSION"] = 15] = "PLACEMENT_IS_NOT_AVAILABLE_FOR_TARGETING_OR_EXCLUSION";
    /**
     * Invalid topic path.
     *
     * @generated from protobuf enum value: INVALID_TOPIC_PATH = 16;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_TOPIC_PATH"] = 16] = "INVALID_TOPIC_PATH";
    /**
     * The YouTube Channel Id is invalid.
     *
     * @generated from protobuf enum value: INVALID_YOUTUBE_CHANNEL_ID = 17;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_YOUTUBE_CHANNEL_ID"] = 17] = "INVALID_YOUTUBE_CHANNEL_ID";
    /**
     * The YouTube Video Id is invalid.
     *
     * @generated from protobuf enum value: INVALID_YOUTUBE_VIDEO_ID = 18;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_YOUTUBE_VIDEO_ID"] = 18] = "INVALID_YOUTUBE_VIDEO_ID";
    /**
     * Indicates the placement is a YouTube vertical channel, which is no longer
     * supported.
     *
     * @generated from protobuf enum value: YOUTUBE_VERTICAL_CHANNEL_DEPRECATED = 19;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["YOUTUBE_VERTICAL_CHANNEL_DEPRECATED"] = 19] = "YOUTUBE_VERTICAL_CHANNEL_DEPRECATED";
    /**
     * Indicates the placement is a YouTube demographic channel, which is no
     * longer supported.
     *
     * @generated from protobuf enum value: YOUTUBE_DEMOGRAPHIC_CHANNEL_DEPRECATED = 20;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["YOUTUBE_DEMOGRAPHIC_CHANNEL_DEPRECATED"] = 20] = "YOUTUBE_DEMOGRAPHIC_CHANNEL_DEPRECATED";
    /**
     * YouTube urls are not supported in Placement criterion. Use YouTubeChannel
     * and YouTubeVideo criterion instead.
     *
     * @generated from protobuf enum value: YOUTUBE_URL_UNSUPPORTED = 21;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["YOUTUBE_URL_UNSUPPORTED"] = 21] = "YOUTUBE_URL_UNSUPPORTED";
    /**
     * Criteria type can not be excluded by the customer, like AOL account type
     * cannot target site type criteria.
     *
     * @generated from protobuf enum value: CANNOT_EXCLUDE_CRITERIA_TYPE = 22;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_EXCLUDE_CRITERIA_TYPE"] = 22] = "CANNOT_EXCLUDE_CRITERIA_TYPE";
    /**
     * Criteria type can not be targeted.
     *
     * @generated from protobuf enum value: CANNOT_ADD_CRITERIA_TYPE = 23;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_ADD_CRITERIA_TYPE"] = 23] = "CANNOT_ADD_CRITERIA_TYPE";
    /**
     * Not allowed to exclude similar user list.
     *
     * @generated from protobuf enum value: CANNOT_EXCLUDE_SIMILAR_USER_LIST = 26;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_EXCLUDE_SIMILAR_USER_LIST"] = 26] = "CANNOT_EXCLUDE_SIMILAR_USER_LIST";
    /**
     * Not allowed to target a closed user list.
     *
     * @generated from protobuf enum value: CANNOT_ADD_CLOSED_USER_LIST = 27;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_ADD_CLOSED_USER_LIST"] = 27] = "CANNOT_ADD_CLOSED_USER_LIST";
    /**
     * Not allowed to add display only UserLists to search only campaigns.
     *
     * @generated from protobuf enum value: CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SEARCH_ONLY_CAMPAIGNS = 28;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SEARCH_ONLY_CAMPAIGNS"] = 28] = "CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SEARCH_ONLY_CAMPAIGNS";
    /**
     * Not allowed to add display only UserLists to search plus campaigns.
     *
     * @generated from protobuf enum value: CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SEARCH_CAMPAIGNS = 29;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SEARCH_CAMPAIGNS"] = 29] = "CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SEARCH_CAMPAIGNS";
    /**
     * Not allowed to add display only UserLists to shopping campaigns.
     *
     * @generated from protobuf enum value: CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SHOPPING_CAMPAIGNS = 30;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SHOPPING_CAMPAIGNS"] = 30] = "CANNOT_ADD_DISPLAY_ONLY_LISTS_TO_SHOPPING_CAMPAIGNS";
    /**
     * Not allowed to add User interests to search only campaigns.
     *
     * @generated from protobuf enum value: CANNOT_ADD_USER_INTERESTS_TO_SEARCH_CAMPAIGNS = 31;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_ADD_USER_INTERESTS_TO_SEARCH_CAMPAIGNS"] = 31] = "CANNOT_ADD_USER_INTERESTS_TO_SEARCH_CAMPAIGNS";
    /**
     * Not allowed to set bids for this criterion type in search campaigns
     *
     * @generated from protobuf enum value: CANNOT_SET_BIDS_ON_CRITERION_TYPE_IN_SEARCH_CAMPAIGNS = 32;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_SET_BIDS_ON_CRITERION_TYPE_IN_SEARCH_CAMPAIGNS"] = 32] = "CANNOT_SET_BIDS_ON_CRITERION_TYPE_IN_SEARCH_CAMPAIGNS";
    /**
     * Final URLs, URL Templates and CustomParameters cannot be set for the
     * criterion types of Gender, AgeRange, UserList, Placement, MobileApp, and
     * MobileAppCategory in search campaigns and shopping campaigns.
     *
     * @generated from protobuf enum value: CANNOT_ADD_URLS_TO_CRITERION_TYPE_FOR_CAMPAIGN_TYPE = 33;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_ADD_URLS_TO_CRITERION_TYPE_FOR_CAMPAIGN_TYPE"] = 33] = "CANNOT_ADD_URLS_TO_CRITERION_TYPE_FOR_CAMPAIGN_TYPE";
    /**
     * Invalid combined audience criterion.
     *
     * @generated from protobuf enum value: INVALID_COMBINED_AUDIENCE = 122;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_COMBINED_AUDIENCE"] = 122] = "INVALID_COMBINED_AUDIENCE";
    /**
     * Invalid custom affinity criterion.
     *
     * @generated from protobuf enum value: INVALID_CUSTOM_AFFINITY = 96;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_CUSTOM_AFFINITY"] = 96] = "INVALID_CUSTOM_AFFINITY";
    /**
     * Invalid custom intent criterion.
     *
     * @generated from protobuf enum value: INVALID_CUSTOM_INTENT = 97;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_CUSTOM_INTENT"] = 97] = "INVALID_CUSTOM_INTENT";
    /**
     * Invalid custom audience criterion.
     *
     * @generated from protobuf enum value: INVALID_CUSTOM_AUDIENCE = 121;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_CUSTOM_AUDIENCE"] = 121] = "INVALID_CUSTOM_AUDIENCE";
    /**
     * IP address is not valid.
     *
     * @generated from protobuf enum value: INVALID_IP_ADDRESS = 34;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_IP_ADDRESS"] = 34] = "INVALID_IP_ADDRESS";
    /**
     * IP format is not valid.
     *
     * @generated from protobuf enum value: INVALID_IP_FORMAT = 35;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_IP_FORMAT"] = 35] = "INVALID_IP_FORMAT";
    /**
     * Mobile application is not valid.
     *
     * @generated from protobuf enum value: INVALID_MOBILE_APP = 36;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_MOBILE_APP"] = 36] = "INVALID_MOBILE_APP";
    /**
     * Mobile application category is not valid.
     *
     * @generated from protobuf enum value: INVALID_MOBILE_APP_CATEGORY = 37;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_MOBILE_APP_CATEGORY"] = 37] = "INVALID_MOBILE_APP_CATEGORY";
    /**
     * The CriterionId does not exist or is of the incorrect type.
     *
     * @generated from protobuf enum value: INVALID_CRITERION_ID = 38;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_CRITERION_ID"] = 38] = "INVALID_CRITERION_ID";
    /**
     * The Criterion is not allowed to be targeted.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_CRITERION = 39;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_TARGET_CRITERION"] = 39] = "CANNOT_TARGET_CRITERION";
    /**
     * The criterion is not allowed to be targeted as it is deprecated.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_OBSOLETE_CRITERION = 40;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_TARGET_OBSOLETE_CRITERION"] = 40] = "CANNOT_TARGET_OBSOLETE_CRITERION";
    /**
     * The CriterionId is not valid for the type.
     *
     * @generated from protobuf enum value: CRITERION_ID_AND_TYPE_MISMATCH = 41;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CRITERION_ID_AND_TYPE_MISMATCH"] = 41] = "CRITERION_ID_AND_TYPE_MISMATCH";
    /**
     * Distance for the radius for the proximity criterion is invalid.
     *
     * @generated from protobuf enum value: INVALID_PROXIMITY_RADIUS = 42;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_PROXIMITY_RADIUS"] = 42] = "INVALID_PROXIMITY_RADIUS";
    /**
     * Units for the distance for the radius for the proximity criterion is
     * invalid.
     *
     * @generated from protobuf enum value: INVALID_PROXIMITY_RADIUS_UNITS = 43;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_PROXIMITY_RADIUS_UNITS"] = 43] = "INVALID_PROXIMITY_RADIUS_UNITS";
    /**
     * Street address in the address is not valid.
     *
     * @generated from protobuf enum value: INVALID_STREETADDRESS_LENGTH = 44;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_STREETADDRESS_LENGTH"] = 44] = "INVALID_STREETADDRESS_LENGTH";
    /**
     * City name in the address is not valid.
     *
     * @generated from protobuf enum value: INVALID_CITYNAME_LENGTH = 45;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_CITYNAME_LENGTH"] = 45] = "INVALID_CITYNAME_LENGTH";
    /**
     * Region code in the address is not valid.
     *
     * @generated from protobuf enum value: INVALID_REGIONCODE_LENGTH = 46;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_REGIONCODE_LENGTH"] = 46] = "INVALID_REGIONCODE_LENGTH";
    /**
     * Region name in the address is not valid.
     *
     * @generated from protobuf enum value: INVALID_REGIONNAME_LENGTH = 47;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_REGIONNAME_LENGTH"] = 47] = "INVALID_REGIONNAME_LENGTH";
    /**
     * Postal code in the address is not valid.
     *
     * @generated from protobuf enum value: INVALID_POSTALCODE_LENGTH = 48;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_POSTALCODE_LENGTH"] = 48] = "INVALID_POSTALCODE_LENGTH";
    /**
     * Country code in the address is not valid.
     *
     * @generated from protobuf enum value: INVALID_COUNTRY_CODE = 49;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_COUNTRY_CODE"] = 49] = "INVALID_COUNTRY_CODE";
    /**
     * Latitude for the GeoPoint is not valid.
     *
     * @generated from protobuf enum value: INVALID_LATITUDE = 50;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_LATITUDE"] = 50] = "INVALID_LATITUDE";
    /**
     * Longitude for the GeoPoint is not valid.
     *
     * @generated from protobuf enum value: INVALID_LONGITUDE = 51;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_LONGITUDE"] = 51] = "INVALID_LONGITUDE";
    /**
     * The Proximity input is not valid. Both address and geoPoint cannot be
     * null.
     *
     * @generated from protobuf enum value: PROXIMITY_GEOPOINT_AND_ADDRESS_BOTH_CANNOT_BE_NULL = 52;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["PROXIMITY_GEOPOINT_AND_ADDRESS_BOTH_CANNOT_BE_NULL"] = 52] = "PROXIMITY_GEOPOINT_AND_ADDRESS_BOTH_CANNOT_BE_NULL";
    /**
     * The Proximity address cannot be geocoded to a valid lat/long.
     *
     * @generated from protobuf enum value: INVALID_PROXIMITY_ADDRESS = 53;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_PROXIMITY_ADDRESS"] = 53] = "INVALID_PROXIMITY_ADDRESS";
    /**
     * User domain name is not valid.
     *
     * @generated from protobuf enum value: INVALID_USER_DOMAIN_NAME = 54;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_USER_DOMAIN_NAME"] = 54] = "INVALID_USER_DOMAIN_NAME";
    /**
     * Length of serialized criterion parameter exceeded size limit.
     *
     * @generated from protobuf enum value: CRITERION_PARAMETER_TOO_LONG = 55;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CRITERION_PARAMETER_TOO_LONG"] = 55] = "CRITERION_PARAMETER_TOO_LONG";
    /**
     * Time interval in the AdSchedule overlaps with another AdSchedule.
     *
     * @generated from protobuf enum value: AD_SCHEDULE_TIME_INTERVALS_OVERLAP = 56;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["AD_SCHEDULE_TIME_INTERVALS_OVERLAP"] = 56] = "AD_SCHEDULE_TIME_INTERVALS_OVERLAP";
    /**
     * AdSchedule time interval cannot span multiple days.
     *
     * @generated from protobuf enum value: AD_SCHEDULE_INTERVAL_CANNOT_SPAN_MULTIPLE_DAYS = 57;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["AD_SCHEDULE_INTERVAL_CANNOT_SPAN_MULTIPLE_DAYS"] = 57] = "AD_SCHEDULE_INTERVAL_CANNOT_SPAN_MULTIPLE_DAYS";
    /**
     * AdSchedule time interval specified is invalid, endTime cannot be earlier
     * than startTime.
     *
     * @generated from protobuf enum value: AD_SCHEDULE_INVALID_TIME_INTERVAL = 58;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["AD_SCHEDULE_INVALID_TIME_INTERVAL"] = 58] = "AD_SCHEDULE_INVALID_TIME_INTERVAL";
    /**
     * The number of AdSchedule entries in a day exceeds the limit.
     *
     * @generated from protobuf enum value: AD_SCHEDULE_EXCEEDED_INTERVALS_PER_DAY_LIMIT = 59;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["AD_SCHEDULE_EXCEEDED_INTERVALS_PER_DAY_LIMIT"] = 59] = "AD_SCHEDULE_EXCEEDED_INTERVALS_PER_DAY_LIMIT";
    /**
     * CriteriaId does not match the interval of the AdSchedule specified.
     *
     * @generated from protobuf enum value: AD_SCHEDULE_CRITERION_ID_MISMATCHING_FIELDS = 60;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["AD_SCHEDULE_CRITERION_ID_MISMATCHING_FIELDS"] = 60] = "AD_SCHEDULE_CRITERION_ID_MISMATCHING_FIELDS";
    /**
     * Cannot set bid modifier for this criterion type.
     *
     * @generated from protobuf enum value: CANNOT_BID_MODIFY_CRITERION_TYPE = 61;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_BID_MODIFY_CRITERION_TYPE"] = 61] = "CANNOT_BID_MODIFY_CRITERION_TYPE";
    /**
     * Cannot bid modify criterion, since it is opted out of the campaign.
     *
     * @generated from protobuf enum value: CANNOT_BID_MODIFY_CRITERION_CAMPAIGN_OPTED_OUT = 62;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_BID_MODIFY_CRITERION_CAMPAIGN_OPTED_OUT"] = 62] = "CANNOT_BID_MODIFY_CRITERION_CAMPAIGN_OPTED_OUT";
    /**
     * Cannot set bid modifier for a negative criterion.
     *
     * @generated from protobuf enum value: CANNOT_BID_MODIFY_NEGATIVE_CRITERION = 63;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_BID_MODIFY_NEGATIVE_CRITERION"] = 63] = "CANNOT_BID_MODIFY_NEGATIVE_CRITERION";
    /**
     * Bid Modifier already exists. Use SET operation to update.
     *
     * @generated from protobuf enum value: BID_MODIFIER_ALREADY_EXISTS = 64;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["BID_MODIFIER_ALREADY_EXISTS"] = 64] = "BID_MODIFIER_ALREADY_EXISTS";
    /**
     * Feed Id is not allowed in these Location Groups.
     *
     * @generated from protobuf enum value: FEED_ID_NOT_ALLOWED = 65;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["FEED_ID_NOT_ALLOWED"] = 65] = "FEED_ID_NOT_ALLOWED";
    /**
     * The account may not use the requested criteria type. For example, some
     * accounts are restricted to keywords only.
     *
     * @generated from protobuf enum value: ACCOUNT_INELIGIBLE_FOR_CRITERIA_TYPE = 66;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["ACCOUNT_INELIGIBLE_FOR_CRITERIA_TYPE"] = 66] = "ACCOUNT_INELIGIBLE_FOR_CRITERIA_TYPE";
    /**
     * The requested criteria type cannot be used with campaign or ad group
     * bidding strategy.
     *
     * @generated from protobuf enum value: CRITERIA_TYPE_INVALID_FOR_BIDDING_STRATEGY = 67;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CRITERIA_TYPE_INVALID_FOR_BIDDING_STRATEGY"] = 67] = "CRITERIA_TYPE_INVALID_FOR_BIDDING_STRATEGY";
    /**
     * The Criterion is not allowed to be excluded.
     *
     * @generated from protobuf enum value: CANNOT_EXCLUDE_CRITERION = 68;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_EXCLUDE_CRITERION"] = 68] = "CANNOT_EXCLUDE_CRITERION";
    /**
     * The criterion is not allowed to be removed. For example, we cannot remove
     * any of the device criterion.
     *
     * @generated from protobuf enum value: CANNOT_REMOVE_CRITERION = 69;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_REMOVE_CRITERION"] = 69] = "CANNOT_REMOVE_CRITERION";
    /**
     * Bidding categories do not form a valid path in the Shopping bidding
     * category taxonomy.
     *
     * @generated from protobuf enum value: INVALID_PRODUCT_BIDDING_CATEGORY = 76;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_PRODUCT_BIDDING_CATEGORY"] = 76] = "INVALID_PRODUCT_BIDDING_CATEGORY";
    /**
     * ShoppingSetting must be added to the campaign before ProductScope
     * criteria can be added.
     *
     * @generated from protobuf enum value: MISSING_SHOPPING_SETTING = 77;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["MISSING_SHOPPING_SETTING"] = 77] = "MISSING_SHOPPING_SETTING";
    /**
     * Matching function is invalid.
     *
     * @generated from protobuf enum value: INVALID_MATCHING_FUNCTION = 78;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_MATCHING_FUNCTION"] = 78] = "INVALID_MATCHING_FUNCTION";
    /**
     * Filter parameters not allowed for location groups targeting.
     *
     * @generated from protobuf enum value: LOCATION_FILTER_NOT_ALLOWED = 79;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LOCATION_FILTER_NOT_ALLOWED"] = 79] = "LOCATION_FILTER_NOT_ALLOWED";
    /**
     * Feed not found, or the feed is not an enabled location feed.
     *
     * @generated from protobuf enum value: INVALID_FEED_FOR_LOCATION_FILTER = 98;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_FEED_FOR_LOCATION_FILTER"] = 98] = "INVALID_FEED_FOR_LOCATION_FILTER";
    /**
     * Given location filter parameter is invalid for location groups targeting.
     *
     * @generated from protobuf enum value: LOCATION_FILTER_INVALID = 80;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LOCATION_FILTER_INVALID"] = 80] = "LOCATION_FILTER_INVALID";
    /**
     * Cannot set geo target constants and feed item sets at the same time.
     *
     * @generated from protobuf enum value: CANNOT_SET_GEO_TARGET_CONSTANTS_WITH_FEED_ITEM_SETS = 123;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_SET_GEO_TARGET_CONSTANTS_WITH_FEED_ITEM_SETS"] = 123] = "CANNOT_SET_GEO_TARGET_CONSTANTS_WITH_FEED_ITEM_SETS";
    /**
     * The location group radius is in the range but not at the valid increment.
     *
     * @generated from protobuf enum value: INVALID_LOCATION_GROUP_RADIUS = 124;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_LOCATION_GROUP_RADIUS"] = 124] = "INVALID_LOCATION_GROUP_RADIUS";
    /**
     * The location group radius unit is invalid.
     *
     * @generated from protobuf enum value: INVALID_LOCATION_GROUP_RADIUS_UNIT = 125;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_LOCATION_GROUP_RADIUS_UNIT"] = 125] = "INVALID_LOCATION_GROUP_RADIUS_UNIT";
    /**
     * Criteria type cannot be associated with a campaign and its ad group(s)
     * simultaneously.
     *
     * @generated from protobuf enum value: CANNOT_ATTACH_CRITERIA_AT_CAMPAIGN_AND_ADGROUP = 81;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_ATTACH_CRITERIA_AT_CAMPAIGN_AND_ADGROUP"] = 81] = "CANNOT_ATTACH_CRITERIA_AT_CAMPAIGN_AND_ADGROUP";
    /**
     * Range represented by hotel length of stay's min nights and max nights
     * overlaps with an existing criterion.
     *
     * @generated from protobuf enum value: HOTEL_LENGTH_OF_STAY_OVERLAPS_WITH_EXISTING_CRITERION = 82;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["HOTEL_LENGTH_OF_STAY_OVERLAPS_WITH_EXISTING_CRITERION"] = 82] = "HOTEL_LENGTH_OF_STAY_OVERLAPS_WITH_EXISTING_CRITERION";
    /**
     * Range represented by hotel advance booking window's min days and max days
     * overlaps with an existing criterion.
     *
     * @generated from protobuf enum value: HOTEL_ADVANCE_BOOKING_WINDOW_OVERLAPS_WITH_EXISTING_CRITERION = 83;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["HOTEL_ADVANCE_BOOKING_WINDOW_OVERLAPS_WITH_EXISTING_CRITERION"] = 83] = "HOTEL_ADVANCE_BOOKING_WINDOW_OVERLAPS_WITH_EXISTING_CRITERION";
    /**
     * The field is not allowed to be set when the negative field is set to
     * true, for example, we don't allow bids in negative ad group or campaign
     * criteria.
     *
     * @generated from protobuf enum value: FIELD_INCOMPATIBLE_WITH_NEGATIVE_TARGETING = 84;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["FIELD_INCOMPATIBLE_WITH_NEGATIVE_TARGETING"] = 84] = "FIELD_INCOMPATIBLE_WITH_NEGATIVE_TARGETING";
    /**
     * The combination of operand and operator in webpage condition is invalid.
     *
     * @generated from protobuf enum value: INVALID_WEBPAGE_CONDITION = 85;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_WEBPAGE_CONDITION"] = 85] = "INVALID_WEBPAGE_CONDITION";
    /**
     * The URL of webpage condition is invalid.
     *
     * @generated from protobuf enum value: INVALID_WEBPAGE_CONDITION_URL = 86;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_WEBPAGE_CONDITION_URL"] = 86] = "INVALID_WEBPAGE_CONDITION_URL";
    /**
     * The URL of webpage condition cannot be empty or contain white space.
     *
     * @generated from protobuf enum value: WEBPAGE_CONDITION_URL_CANNOT_BE_EMPTY = 87;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CONDITION_URL_CANNOT_BE_EMPTY"] = 87] = "WEBPAGE_CONDITION_URL_CANNOT_BE_EMPTY";
    /**
     * The URL of webpage condition contains an unsupported protocol.
     *
     * @generated from protobuf enum value: WEBPAGE_CONDITION_URL_UNSUPPORTED_PROTOCOL = 88;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CONDITION_URL_UNSUPPORTED_PROTOCOL"] = 88] = "WEBPAGE_CONDITION_URL_UNSUPPORTED_PROTOCOL";
    /**
     * The URL of webpage condition cannot be an IP address.
     *
     * @generated from protobuf enum value: WEBPAGE_CONDITION_URL_CANNOT_BE_IP_ADDRESS = 89;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CONDITION_URL_CANNOT_BE_IP_ADDRESS"] = 89] = "WEBPAGE_CONDITION_URL_CANNOT_BE_IP_ADDRESS";
    /**
     * The domain of the URL is not consistent with the domain in campaign
     * setting.
     *
     * @generated from protobuf enum value: WEBPAGE_CONDITION_URL_DOMAIN_NOT_CONSISTENT_WITH_CAMPAIGN_SETTING = 90;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CONDITION_URL_DOMAIN_NOT_CONSISTENT_WITH_CAMPAIGN_SETTING"] = 90] = "WEBPAGE_CONDITION_URL_DOMAIN_NOT_CONSISTENT_WITH_CAMPAIGN_SETTING";
    /**
     * The URL of webpage condition cannot be a public suffix itself.
     *
     * @generated from protobuf enum value: WEBPAGE_CONDITION_URL_CANNOT_BE_PUBLIC_SUFFIX = 91;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CONDITION_URL_CANNOT_BE_PUBLIC_SUFFIX"] = 91] = "WEBPAGE_CONDITION_URL_CANNOT_BE_PUBLIC_SUFFIX";
    /**
     * The URL of webpage condition has an invalid public suffix.
     *
     * @generated from protobuf enum value: WEBPAGE_CONDITION_URL_INVALID_PUBLIC_SUFFIX = 92;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CONDITION_URL_INVALID_PUBLIC_SUFFIX"] = 92] = "WEBPAGE_CONDITION_URL_INVALID_PUBLIC_SUFFIX";
    /**
     * Value track parameter is not supported in webpage condition URL.
     *
     * @generated from protobuf enum value: WEBPAGE_CONDITION_URL_VALUE_TRACK_VALUE_NOT_SUPPORTED = 93;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CONDITION_URL_VALUE_TRACK_VALUE_NOT_SUPPORTED"] = 93] = "WEBPAGE_CONDITION_URL_VALUE_TRACK_VALUE_NOT_SUPPORTED";
    /**
     * Only one URL-EQUALS webpage condition is allowed in a webpage
     * criterion and it cannot be combined with other conditions.
     *
     * @generated from protobuf enum value: WEBPAGE_CRITERION_URL_EQUALS_CAN_HAVE_ONLY_ONE_CONDITION = 94;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CRITERION_URL_EQUALS_CAN_HAVE_ONLY_ONE_CONDITION"] = 94] = "WEBPAGE_CRITERION_URL_EQUALS_CAN_HAVE_ONLY_ONE_CONDITION";
    /**
     * A webpage criterion cannot be added to a non-DSA ad group.
     *
     * @generated from protobuf enum value: WEBPAGE_CRITERION_NOT_SUPPORTED_ON_NON_DSA_AD_GROUP = 95;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["WEBPAGE_CRITERION_NOT_SUPPORTED_ON_NON_DSA_AD_GROUP"] = 95] = "WEBPAGE_CRITERION_NOT_SUPPORTED_ON_NON_DSA_AD_GROUP";
    /**
     * Cannot add positive user list criteria in Smart Display campaigns.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_USER_LIST_FOR_SMART_DISPLAY_CAMPAIGNS = 99;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_TARGET_USER_LIST_FOR_SMART_DISPLAY_CAMPAIGNS"] = 99] = "CANNOT_TARGET_USER_LIST_FOR_SMART_DISPLAY_CAMPAIGNS";
    /**
     * Cannot add positive placement criterion types in search campaigns.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_PLACEMENTS_FOR_SEARCH_CAMPAIGNS = 126;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_TARGET_PLACEMENTS_FOR_SEARCH_CAMPAIGNS"] = 126] = "CANNOT_TARGET_PLACEMENTS_FOR_SEARCH_CAMPAIGNS";
    /**
     * Listing scope contains too many dimension types.
     *
     * @generated from protobuf enum value: LISTING_SCOPE_TOO_MANY_DIMENSION_TYPES = 100;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_SCOPE_TOO_MANY_DIMENSION_TYPES"] = 100] = "LISTING_SCOPE_TOO_MANY_DIMENSION_TYPES";
    /**
     * Listing scope has too many IN operators.
     *
     * @generated from protobuf enum value: LISTING_SCOPE_TOO_MANY_IN_OPERATORS = 101;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_SCOPE_TOO_MANY_IN_OPERATORS"] = 101] = "LISTING_SCOPE_TOO_MANY_IN_OPERATORS";
    /**
     * Listing scope contains IN operator on an unsupported dimension type.
     *
     * @generated from protobuf enum value: LISTING_SCOPE_IN_OPERATOR_NOT_SUPPORTED = 102;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_SCOPE_IN_OPERATOR_NOT_SUPPORTED"] = 102] = "LISTING_SCOPE_IN_OPERATOR_NOT_SUPPORTED";
    /**
     * There are dimensions with duplicate dimension type.
     *
     * @generated from protobuf enum value: DUPLICATE_LISTING_DIMENSION_TYPE = 103;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["DUPLICATE_LISTING_DIMENSION_TYPE"] = 103] = "DUPLICATE_LISTING_DIMENSION_TYPE";
    /**
     * There are dimensions with duplicate dimension value.
     *
     * @generated from protobuf enum value: DUPLICATE_LISTING_DIMENSION_VALUE = 104;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["DUPLICATE_LISTING_DIMENSION_VALUE"] = 104] = "DUPLICATE_LISTING_DIMENSION_VALUE";
    /**
     * Listing group SUBDIVISION nodes cannot have bids.
     *
     * @generated from protobuf enum value: CANNOT_SET_BIDS_ON_LISTING_GROUP_SUBDIVISION = 105;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_SET_BIDS_ON_LISTING_GROUP_SUBDIVISION"] = 105] = "CANNOT_SET_BIDS_ON_LISTING_GROUP_SUBDIVISION";
    /**
     * Ad group is invalid due to the listing groups it contains.
     *
     * @generated from protobuf enum value: INVALID_LISTING_GROUP_HIERARCHY = 106;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_LISTING_GROUP_HIERARCHY"] = 106] = "INVALID_LISTING_GROUP_HIERARCHY";
    /**
     * Listing group unit cannot have children.
     *
     * @generated from protobuf enum value: LISTING_GROUP_UNIT_CANNOT_HAVE_CHILDREN = 107;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_UNIT_CANNOT_HAVE_CHILDREN"] = 107] = "LISTING_GROUP_UNIT_CANNOT_HAVE_CHILDREN";
    /**
     * Subdivided listing groups must have an "others" case.
     *
     * @generated from protobuf enum value: LISTING_GROUP_SUBDIVISION_REQUIRES_OTHERS_CASE = 108;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_SUBDIVISION_REQUIRES_OTHERS_CASE"] = 108] = "LISTING_GROUP_SUBDIVISION_REQUIRES_OTHERS_CASE";
    /**
     * Dimension type of listing group must be the same as that of its siblings.
     *
     * @generated from protobuf enum value: LISTING_GROUP_REQUIRES_SAME_DIMENSION_TYPE_AS_SIBLINGS = 109;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_REQUIRES_SAME_DIMENSION_TYPE_AS_SIBLINGS"] = 109] = "LISTING_GROUP_REQUIRES_SAME_DIMENSION_TYPE_AS_SIBLINGS";
    /**
     * Listing group cannot be added to the ad group because it already exists.
     *
     * @generated from protobuf enum value: LISTING_GROUP_ALREADY_EXISTS = 110;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_ALREADY_EXISTS"] = 110] = "LISTING_GROUP_ALREADY_EXISTS";
    /**
     * Listing group referenced in the operation was not found in the ad group.
     *
     * @generated from protobuf enum value: LISTING_GROUP_DOES_NOT_EXIST = 111;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_DOES_NOT_EXIST"] = 111] = "LISTING_GROUP_DOES_NOT_EXIST";
    /**
     * Recursive removal failed because listing group subdivision is being
     * created or modified in this request.
     *
     * @generated from protobuf enum value: LISTING_GROUP_CANNOT_BE_REMOVED = 112;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_CANNOT_BE_REMOVED"] = 112] = "LISTING_GROUP_CANNOT_BE_REMOVED";
    /**
     * Listing group type is not allowed for specified ad group criterion type.
     *
     * @generated from protobuf enum value: INVALID_LISTING_GROUP_TYPE = 113;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_LISTING_GROUP_TYPE"] = 113] = "INVALID_LISTING_GROUP_TYPE";
    /**
     * Listing group in an ADD operation specifies a non temporary criterion id.
     *
     * @generated from protobuf enum value: LISTING_GROUP_ADD_MAY_ONLY_USE_TEMP_ID = 114;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_ADD_MAY_ONLY_USE_TEMP_ID"] = 114] = "LISTING_GROUP_ADD_MAY_ONLY_USE_TEMP_ID";
    /**
     * The combined length of dimension values of the Listing scope criterion
     * is too long.
     *
     * @generated from protobuf enum value: LISTING_SCOPE_TOO_LONG = 115;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_SCOPE_TOO_LONG"] = 115] = "LISTING_SCOPE_TOO_LONG";
    /**
     * Listing scope contains too many dimensions.
     *
     * @generated from protobuf enum value: LISTING_SCOPE_TOO_MANY_DIMENSIONS = 116;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_SCOPE_TOO_MANY_DIMENSIONS"] = 116] = "LISTING_SCOPE_TOO_MANY_DIMENSIONS";
    /**
     * The combined length of dimension values of the Listing group criterion is
     * too long.
     *
     * @generated from protobuf enum value: LISTING_GROUP_TOO_LONG = 117;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_TOO_LONG"] = 117] = "LISTING_GROUP_TOO_LONG";
    /**
     * Listing group tree is too deep.
     *
     * @generated from protobuf enum value: LISTING_GROUP_TREE_TOO_DEEP = 118;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["LISTING_GROUP_TREE_TOO_DEEP"] = 118] = "LISTING_GROUP_TREE_TOO_DEEP";
    /**
     * Listing dimension is invalid (for example, dimension contains illegal
     * value, dimension type is represented with wrong class, etc). Listing
     * dimension value can not contain "==" or "&+".
     *
     * @generated from protobuf enum value: INVALID_LISTING_DIMENSION = 119;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_LISTING_DIMENSION"] = 119] = "INVALID_LISTING_DIMENSION";
    /**
     * Listing dimension type is either invalid for campaigns of this type or
     * cannot be used in the current context. BIDDING_CATEGORY_Lx and
     * PRODUCT_TYPE_Lx dimensions must be used in ascending order of their
     * levels: L1, L2, L3, L4, L5... The levels must be specified sequentially
     * and start from L1. Furthermore, an "others" Listing group cannot be
     * subdivided with a dimension of the same type but of a higher level
     * ("others" BIDDING_CATEGORY_L3 can be subdivided with BRAND but not with
     * BIDDING_CATEGORY_L4).
     *
     * @generated from protobuf enum value: INVALID_LISTING_DIMENSION_TYPE = 120;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_LISTING_DIMENSION_TYPE"] = 120] = "INVALID_LISTING_DIMENSION_TYPE";
    /**
     * Customer is not on allowlist for composite audience in display campaigns.
     *
     * @generated from protobuf enum value: ADVERTISER_NOT_ON_ALLOWLIST_FOR_COMBINED_AUDIENCE_ON_DISPLAY = 127;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["ADVERTISER_NOT_ON_ALLOWLIST_FOR_COMBINED_AUDIENCE_ON_DISPLAY"] = 127] = "ADVERTISER_NOT_ON_ALLOWLIST_FOR_COMBINED_AUDIENCE_ON_DISPLAY";
    /**
     * Cannot target on a removed combined audience.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_REMOVED_COMBINED_AUDIENCE = 128;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_TARGET_REMOVED_COMBINED_AUDIENCE"] = 128] = "CANNOT_TARGET_REMOVED_COMBINED_AUDIENCE";
    /**
     * Combined audience ID is invalid.
     *
     * @generated from protobuf enum value: INVALID_COMBINED_AUDIENCE_ID = 129;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["INVALID_COMBINED_AUDIENCE_ID"] = 129] = "INVALID_COMBINED_AUDIENCE_ID";
    /**
     * Can not target removed combined audience.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_REMOVED_CUSTOM_AUDIENCE = 130;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_TARGET_REMOVED_CUSTOM_AUDIENCE"] = 130] = "CANNOT_TARGET_REMOVED_CUSTOM_AUDIENCE";
    /**
     * Range represented by hotel check-in date's start date and end date
     * overlaps with an existing criterion.
     *
     * @generated from protobuf enum value: HOTEL_CHECK_IN_DATE_RANGE_OVERLAPS_WITH_EXISTING_CRITERION = 131;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["HOTEL_CHECK_IN_DATE_RANGE_OVERLAPS_WITH_EXISTING_CRITERION"] = 131] = "HOTEL_CHECK_IN_DATE_RANGE_OVERLAPS_WITH_EXISTING_CRITERION";
    /**
     * Start date is earlier than earliest allowed value of yesterday UTC.
     *
     * @generated from protobuf enum value: HOTEL_CHECK_IN_DATE_RANGE_START_DATE_TOO_EARLY = 132;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["HOTEL_CHECK_IN_DATE_RANGE_START_DATE_TOO_EARLY"] = 132] = "HOTEL_CHECK_IN_DATE_RANGE_START_DATE_TOO_EARLY";
    /**
     * End date later is than latest allowed day of 330 days in the future UTC.
     *
     * @generated from protobuf enum value: HOTEL_CHECK_IN_DATE_RANGE_END_DATE_TOO_LATE = 133;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["HOTEL_CHECK_IN_DATE_RANGE_END_DATE_TOO_LATE"] = 133] = "HOTEL_CHECK_IN_DATE_RANGE_END_DATE_TOO_LATE";
    /**
     * Start date is after end date.
     *
     * @generated from protobuf enum value: HOTEL_CHECK_IN_DATE_RANGE_REVERSED = 134;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["HOTEL_CHECK_IN_DATE_RANGE_REVERSED"] = 134] = "HOTEL_CHECK_IN_DATE_RANGE_REVERSED";
    /**
     * Broad match modifier (BMM) keywords can no longer be created. See
     * https://ads-developers.googleblog.com/2021/06/broad-match-modifier-upcoming-changes.html.
     *
     * @generated from protobuf enum value: BROAD_MATCH_MODIFIER_KEYWORD_NOT_ALLOWED = 135;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["BROAD_MATCH_MODIFIER_KEYWORD_NOT_ALLOWED"] = 135] = "BROAD_MATCH_MODIFIER_KEYWORD_NOT_ALLOWED";
    /**
     * Only one audience is allowed in an asset group.
     *
     * @generated from protobuf enum value: ONE_AUDIENCE_ALLOWED_PER_ASSET_GROUP = 136;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["ONE_AUDIENCE_ALLOWED_PER_ASSET_GROUP"] = 136] = "ONE_AUDIENCE_ALLOWED_PER_ASSET_GROUP";
    /**
     * Audience is not supported for the specified campaign type.
     *
     * @generated from protobuf enum value: AUDIENCE_NOT_ELIGIBLE_FOR_CAMPAIGN_TYPE = 137;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["AUDIENCE_NOT_ELIGIBLE_FOR_CAMPAIGN_TYPE"] = 137] = "AUDIENCE_NOT_ELIGIBLE_FOR_CAMPAIGN_TYPE";
    /**
     * Audience is not allowed to attach when use_audience_grouped bit is set to
     * false.
     *
     * @generated from protobuf enum value: AUDIENCE_NOT_ALLOWED_TO_ATTACH_WHEN_AUDIENCE_GROUPED_SET_TO_FALSE = 138;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["AUDIENCE_NOT_ALLOWED_TO_ATTACH_WHEN_AUDIENCE_GROUPED_SET_TO_FALSE"] = 138] = "AUDIENCE_NOT_ALLOWED_TO_ATTACH_WHEN_AUDIENCE_GROUPED_SET_TO_FALSE";
    /**
     * Targeting is not allowed for Customer Match lists as per Customer Match
     * policy. See
     * https://support.google.com/google-ads/answer/6299717.
     *
     * @generated from protobuf enum value: CANNOT_TARGET_CUSTOMER_MATCH_USER_LIST = 139;
     */
    CriterionErrorEnum_CriterionError[CriterionErrorEnum_CriterionError["CANNOT_TARGET_CUSTOMER_MATCH_USER_LIST"] = 139] = "CANNOT_TARGET_CUSTOMER_MATCH_USER_LIST";
})(CriterionErrorEnum_CriterionError = exports.CriterionErrorEnum_CriterionError || (exports.CriterionErrorEnum_CriterionError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CriterionErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CriterionErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CriterionErrorEnum
 */
exports.CriterionErrorEnum = new CriterionErrorEnum$Type();
