// google-ads-api-client/dist/protos/bindings/ads/errors/currency_code_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CurrencyCodeErrorEnum = exports.CurrencyCodeErrorEnum_CurrencyCodeError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible currency code errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CurrencyCodeErrorEnum.CurrencyCodeError
 */
var CurrencyCodeErrorEnum_CurrencyCodeError;
(function (CurrencyCodeErrorEnum_CurrencyCodeError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CurrencyCodeErrorEnum_CurrencyCodeError[CurrencyCodeErrorEnum_CurrencyCodeError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CurrencyCodeErrorEnum_CurrencyCodeError[CurrencyCodeErrorEnum_CurrencyCodeError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The currency code is not supported.
     *
     * @generated from protobuf enum value: UNSUPPORTED = 2;
     */
    CurrencyCodeErrorEnum_CurrencyCodeError[CurrencyCodeErrorEnum_CurrencyCodeError["UNSUPPORTED"] = 2] = "UNSUPPORTED";
})(CurrencyCodeErrorEnum_CurrencyCodeError = exports.CurrencyCodeErrorEnum_CurrencyCodeError || (exports.CurrencyCodeErrorEnum_CurrencyCodeError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CurrencyCodeErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CurrencyCodeErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CurrencyCodeErrorEnum
 */
exports.CurrencyCodeErrorEnum = new CurrencyCodeErrorEnum$Type();
