// google-ads-api-client/dist/protos/bindings/ads/errors/customer_feed_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerFeedErrorEnum = exports.CustomerFeedErrorEnum_CustomerFeedError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible customer feed errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CustomerFeedErrorEnum.CustomerFeedError
 */
var CustomerFeedErrorEnum_CustomerFeedError;
(function (CustomerFeedErrorEnum_CustomerFeedError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An active feed already exists for this customer and place holder type.
     *
     * @generated from protobuf enum value: FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE = 2;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE"] = 2] = "FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE";
    /**
     * The specified feed is removed.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_FOR_REMOVED_FEED = 3;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["CANNOT_CREATE_FOR_REMOVED_FEED"] = 3] = "CANNOT_CREATE_FOR_REMOVED_FEED";
    /**
     * The CustomerFeed already exists. Update should be used to modify the
     * existing CustomerFeed.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_ALREADY_EXISTING_CUSTOMER_FEED = 4;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["CANNOT_CREATE_ALREADY_EXISTING_CUSTOMER_FEED"] = 4] = "CANNOT_CREATE_ALREADY_EXISTING_CUSTOMER_FEED";
    /**
     * Cannot update removed customer feed.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_REMOVED_CUSTOMER_FEED = 5;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["CANNOT_MODIFY_REMOVED_CUSTOMER_FEED"] = 5] = "CANNOT_MODIFY_REMOVED_CUSTOMER_FEED";
    /**
     * Invalid placeholder type.
     *
     * @generated from protobuf enum value: INVALID_PLACEHOLDER_TYPE = 6;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["INVALID_PLACEHOLDER_TYPE"] = 6] = "INVALID_PLACEHOLDER_TYPE";
    /**
     * Feed mapping for this placeholder type does not exist.
     *
     * @generated from protobuf enum value: MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE = 7;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE"] = 7] = "MISSING_FEEDMAPPING_FOR_PLACEHOLDER_TYPE";
    /**
     * Placeholder not allowed at the account level.
     *
     * @generated from protobuf enum value: PLACEHOLDER_TYPE_NOT_ALLOWED_ON_CUSTOMER_FEED = 8;
     */
    CustomerFeedErrorEnum_CustomerFeedError[CustomerFeedErrorEnum_CustomerFeedError["PLACEHOLDER_TYPE_NOT_ALLOWED_ON_CUSTOMER_FEED"] = 8] = "PLACEHOLDER_TYPE_NOT_ALLOWED_ON_CUSTOMER_FEED";
})(CustomerFeedErrorEnum_CustomerFeedError = exports.CustomerFeedErrorEnum_CustomerFeedError || (exports.CustomerFeedErrorEnum_CustomerFeedError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomerFeedErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CustomerFeedErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CustomerFeedErrorEnum
 */
exports.CustomerFeedErrorEnum = new CustomerFeedErrorEnum$Type();
