// google-ads-api-client/dist/protos/bindings/ads/errors/customer_client_link_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerClientLinkErrorEnum = exports.CustomerClientLinkErrorEnum_CustomerClientLinkError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible CustomerClientLink errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CustomerClientLinkErrorEnum.CustomerClientLinkError
 */
var CustomerClientLinkErrorEnum_CustomerClientLinkError;
(function (CustomerClientLinkErrorEnum_CustomerClientLinkError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Trying to manage a client that already in being managed by customer.
     *
     * @generated from protobuf enum value: CLIENT_ALREADY_INVITED_BY_THIS_MANAGER = 2;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["CLIENT_ALREADY_INVITED_BY_THIS_MANAGER"] = 2] = "CLIENT_ALREADY_INVITED_BY_THIS_MANAGER";
    /**
     * Already managed by some other manager in the hierarchy.
     *
     * @generated from protobuf enum value: CLIENT_ALREADY_MANAGED_IN_HIERARCHY = 3;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["CLIENT_ALREADY_MANAGED_IN_HIERARCHY"] = 3] = "CLIENT_ALREADY_MANAGED_IN_HIERARCHY";
    /**
     * Attempt to create a cycle in the hierarchy.
     *
     * @generated from protobuf enum value: CYCLIC_LINK_NOT_ALLOWED = 4;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["CYCLIC_LINK_NOT_ALLOWED"] = 4] = "CYCLIC_LINK_NOT_ALLOWED";
    /**
     * Managed accounts has the maximum number of linked accounts.
     *
     * @generated from protobuf enum value: CUSTOMER_HAS_TOO_MANY_ACCOUNTS = 5;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["CUSTOMER_HAS_TOO_MANY_ACCOUNTS"] = 5] = "CUSTOMER_HAS_TOO_MANY_ACCOUNTS";
    /**
     * Invitor has the maximum pending invitations.
     *
     * @generated from protobuf enum value: CLIENT_HAS_TOO_MANY_INVITATIONS = 6;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["CLIENT_HAS_TOO_MANY_INVITATIONS"] = 6] = "CLIENT_HAS_TOO_MANY_INVITATIONS";
    /**
     * Attempt to change hidden status of a link that is not active.
     *
     * @generated from protobuf enum value: CANNOT_HIDE_OR_UNHIDE_MANAGER_ACCOUNTS = 7;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["CANNOT_HIDE_OR_UNHIDE_MANAGER_ACCOUNTS"] = 7] = "CANNOT_HIDE_OR_UNHIDE_MANAGER_ACCOUNTS";
    /**
     * Parent manager account has the maximum number of linked accounts.
     *
     * @generated from protobuf enum value: CUSTOMER_HAS_TOO_MANY_ACCOUNTS_AT_MANAGER = 8;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["CUSTOMER_HAS_TOO_MANY_ACCOUNTS_AT_MANAGER"] = 8] = "CUSTOMER_HAS_TOO_MANY_ACCOUNTS_AT_MANAGER";
    /**
     * Client has too many managers.
     *
     * @generated from protobuf enum value: CLIENT_HAS_TOO_MANY_MANAGERS = 9;
     */
    CustomerClientLinkErrorEnum_CustomerClientLinkError[CustomerClientLinkErrorEnum_CustomerClientLinkError["CLIENT_HAS_TOO_MANY_MANAGERS"] = 9] = "CLIENT_HAS_TOO_MANY_MANAGERS";
})(CustomerClientLinkErrorEnum_CustomerClientLinkError = exports.CustomerClientLinkErrorEnum_CustomerClientLinkError || (exports.CustomerClientLinkErrorEnum_CustomerClientLinkError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomerClientLinkErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CustomerClientLinkErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CustomerClientLinkErrorEnum
 */
exports.CustomerClientLinkErrorEnum = new CustomerClientLinkErrorEnum$Type();
