// google-ads-api-client/dist/protos/bindings/ads/errors/custom_audience_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAudienceErrorEnum = exports.CustomAudienceErrorEnum_CustomAudienceError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible custom audience errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CustomAudienceErrorEnum.CustomAudienceError
 */
var CustomAudienceErrorEnum_CustomAudienceError;
(function (CustomAudienceErrorEnum_CustomAudienceError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * New name in the custom audience is duplicated ignoring cases.
     *
     * @generated from protobuf enum value: NAME_ALREADY_USED = 2;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["NAME_ALREADY_USED"] = 2] = "NAME_ALREADY_USED";
    /**
     * Cannot remove a custom audience while it's still being used as targeting.
     *
     * @generated from protobuf enum value: CANNOT_REMOVE_WHILE_IN_USE = 3;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["CANNOT_REMOVE_WHILE_IN_USE"] = 3] = "CANNOT_REMOVE_WHILE_IN_USE";
    /**
     * Cannot update or remove a custom audience that is already removed.
     *
     * @generated from protobuf enum value: RESOURCE_ALREADY_REMOVED = 4;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["RESOURCE_ALREADY_REMOVED"] = 4] = "RESOURCE_ALREADY_REMOVED";
    /**
     * The pair of [type, value] already exists in members.
     *
     * @generated from protobuf enum value: MEMBER_TYPE_AND_PARAMETER_ALREADY_EXISTED = 5;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["MEMBER_TYPE_AND_PARAMETER_ALREADY_EXISTED"] = 5] = "MEMBER_TYPE_AND_PARAMETER_ALREADY_EXISTED";
    /**
     * Member type is invalid.
     *
     * @generated from protobuf enum value: INVALID_MEMBER_TYPE = 6;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["INVALID_MEMBER_TYPE"] = 6] = "INVALID_MEMBER_TYPE";
    /**
     * Member type does not have associated value.
     *
     * @generated from protobuf enum value: MEMBER_TYPE_AND_VALUE_DOES_NOT_MATCH = 7;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["MEMBER_TYPE_AND_VALUE_DOES_NOT_MATCH"] = 7] = "MEMBER_TYPE_AND_VALUE_DOES_NOT_MATCH";
    /**
     * Custom audience contains a member that violates policy.
     *
     * @generated from protobuf enum value: POLICY_VIOLATION = 8;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["POLICY_VIOLATION"] = 8] = "POLICY_VIOLATION";
    /**
     * Change in custom audience type is not allowed.
     *
     * @generated from protobuf enum value: INVALID_TYPE_CHANGE = 9;
     */
    CustomAudienceErrorEnum_CustomAudienceError[CustomAudienceErrorEnum_CustomAudienceError["INVALID_TYPE_CHANGE"] = 9] = "INVALID_TYPE_CHANGE";
})(CustomAudienceErrorEnum_CustomAudienceError = exports.CustomAudienceErrorEnum_CustomAudienceError || (exports.CustomAudienceErrorEnum_CustomAudienceError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomAudienceErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CustomAudienceErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CustomAudienceErrorEnum
 */
exports.CustomAudienceErrorEnum = new CustomAudienceErrorEnum$Type();
