// google-ads-api-client/dist/protos/bindings/ads/errors/custom_interest_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomInterestErrorEnum = exports.CustomInterestErrorEnum_CustomInterestError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible custom interest errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CustomInterestErrorEnum.CustomInterestError
 */
var CustomInterestErrorEnum_CustomInterestError;
(function (CustomInterestErrorEnum_CustomInterestError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Duplicate custom interest name ignoring case.
     *
     * @generated from protobuf enum value: NAME_ALREADY_USED = 2;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["NAME_ALREADY_USED"] = 2] = "NAME_ALREADY_USED";
    /**
     * In the remove custom interest member operation, both member ID and
     * pair [type, parameter] are not present.
     *
     * @generated from protobuf enum value: CUSTOM_INTEREST_MEMBER_ID_AND_TYPE_PARAMETER_NOT_PRESENT_IN_REMOVE = 3;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["CUSTOM_INTEREST_MEMBER_ID_AND_TYPE_PARAMETER_NOT_PRESENT_IN_REMOVE"] = 3] = "CUSTOM_INTEREST_MEMBER_ID_AND_TYPE_PARAMETER_NOT_PRESENT_IN_REMOVE";
    /**
     * The pair of [type, parameter] does not exist.
     *
     * @generated from protobuf enum value: TYPE_AND_PARAMETER_NOT_FOUND = 4;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["TYPE_AND_PARAMETER_NOT_FOUND"] = 4] = "TYPE_AND_PARAMETER_NOT_FOUND";
    /**
     * The pair of [type, parameter] already exists.
     *
     * @generated from protobuf enum value: TYPE_AND_PARAMETER_ALREADY_EXISTED = 5;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["TYPE_AND_PARAMETER_ALREADY_EXISTED"] = 5] = "TYPE_AND_PARAMETER_ALREADY_EXISTED";
    /**
     * Unsupported custom interest member type.
     *
     * @generated from protobuf enum value: INVALID_CUSTOM_INTEREST_MEMBER_TYPE = 6;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["INVALID_CUSTOM_INTEREST_MEMBER_TYPE"] = 6] = "INVALID_CUSTOM_INTEREST_MEMBER_TYPE";
    /**
     * Cannot remove a custom interest while it's still being targeted.
     *
     * @generated from protobuf enum value: CANNOT_REMOVE_WHILE_IN_USE = 7;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["CANNOT_REMOVE_WHILE_IN_USE"] = 7] = "CANNOT_REMOVE_WHILE_IN_USE";
    /**
     * Cannot mutate custom interest type.
     *
     * @generated from protobuf enum value: CANNOT_CHANGE_TYPE = 8;
     */
    CustomInterestErrorEnum_CustomInterestError[CustomInterestErrorEnum_CustomInterestError["CANNOT_CHANGE_TYPE"] = 8] = "CANNOT_CHANGE_TYPE";
})(CustomInterestErrorEnum_CustomInterestError = exports.CustomInterestErrorEnum_CustomInterestError || (exports.CustomInterestErrorEnum_CustomInterestError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomInterestErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CustomInterestErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CustomInterestErrorEnum
 */
exports.CustomInterestErrorEnum = new CustomInterestErrorEnum$Type();
