// google-ads-api-client/dist/protos/bindings/ads/errors/customizer_attribute_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomizerAttributeErrorEnum = exports.CustomizerAttributeErrorEnum_CustomizerAttributeError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible customizer attribute errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CustomizerAttributeErrorEnum.CustomizerAttributeError
 */
var CustomizerAttributeErrorEnum_CustomizerAttributeError;
(function (CustomizerAttributeErrorEnum_CustomizerAttributeError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomizerAttributeErrorEnum_CustomizerAttributeError[CustomizerAttributeErrorEnum_CustomizerAttributeError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomizerAttributeErrorEnum_CustomizerAttributeError[CustomizerAttributeErrorEnum_CustomizerAttributeError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * CustomizerAttribute name matches that of another active
     * CustomizerAttribute.
     *
     * @generated from protobuf enum value: DUPLICATE_CUSTOMIZER_ATTRIBUTE_NAME = 2;
     */
    CustomizerAttributeErrorEnum_CustomizerAttributeError[CustomizerAttributeErrorEnum_CustomizerAttributeError["DUPLICATE_CUSTOMIZER_ATTRIBUTE_NAME"] = 2] = "DUPLICATE_CUSTOMIZER_ATTRIBUTE_NAME";
})(CustomizerAttributeErrorEnum_CustomizerAttributeError = exports.CustomizerAttributeErrorEnum_CustomizerAttributeError || (exports.CustomizerAttributeErrorEnum_CustomizerAttributeError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomizerAttributeErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CustomizerAttributeErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CustomizerAttributeErrorEnum
 */
exports.CustomizerAttributeErrorEnum = new CustomizerAttributeErrorEnum$Type();
