// google-ads-api-client/dist/protos/bindings/ads/errors/custom_conversion_goal_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomConversionGoalErrorEnum = exports.CustomConversionGoalErrorEnum_CustomConversionGoalError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible custom conversion goal errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CustomConversionGoalErrorEnum.CustomConversionGoalError
 */
var CustomConversionGoalErrorEnum_CustomConversionGoalError;
(function (CustomConversionGoalErrorEnum_CustomConversionGoalError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomConversionGoalErrorEnum_CustomConversionGoalError[CustomConversionGoalErrorEnum_CustomConversionGoalError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomConversionGoalErrorEnum_CustomConversionGoalError[CustomConversionGoalErrorEnum_CustomConversionGoalError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Cannot find a conversion action with the specified id.
     *
     * @generated from protobuf enum value: INVALID_CONVERSION_ACTION = 2;
     */
    CustomConversionGoalErrorEnum_CustomConversionGoalError[CustomConversionGoalErrorEnum_CustomConversionGoalError["INVALID_CONVERSION_ACTION"] = 2] = "INVALID_CONVERSION_ACTION";
    /**
     * The conversion action is not enabled so it cannot be included in a custom
     * conversion goal.
     *
     * @generated from protobuf enum value: CONVERSION_ACTION_NOT_ENABLED = 3;
     */
    CustomConversionGoalErrorEnum_CustomConversionGoalError[CustomConversionGoalErrorEnum_CustomConversionGoalError["CONVERSION_ACTION_NOT_ENABLED"] = 3] = "CONVERSION_ACTION_NOT_ENABLED";
    /**
     * The custom conversion goal cannot be removed because it's linked to a
     * campaign.
     *
     * @generated from protobuf enum value: CANNOT_REMOVE_LINKED_CUSTOM_CONVERSION_GOAL = 4;
     */
    CustomConversionGoalErrorEnum_CustomConversionGoalError[CustomConversionGoalErrorEnum_CustomConversionGoalError["CANNOT_REMOVE_LINKED_CUSTOM_CONVERSION_GOAL"] = 4] = "CANNOT_REMOVE_LINKED_CUSTOM_CONVERSION_GOAL";
    /**
     * Custom goal with the same name already exists.
     *
     * @generated from protobuf enum value: CUSTOM_GOAL_DUPLICATE_NAME = 5;
     */
    CustomConversionGoalErrorEnum_CustomConversionGoalError[CustomConversionGoalErrorEnum_CustomConversionGoalError["CUSTOM_GOAL_DUPLICATE_NAME"] = 5] = "CUSTOM_GOAL_DUPLICATE_NAME";
    /**
     * Custom goal with the same conversion action list already exists.
     *
     * @generated from protobuf enum value: DUPLICATE_CONVERSION_ACTION_LIST = 6;
     */
    CustomConversionGoalErrorEnum_CustomConversionGoalError[CustomConversionGoalErrorEnum_CustomConversionGoalError["DUPLICATE_CONVERSION_ACTION_LIST"] = 6] = "DUPLICATE_CONVERSION_ACTION_LIST";
})(CustomConversionGoalErrorEnum_CustomConversionGoalError = exports.CustomConversionGoalErrorEnum_CustomConversionGoalError || (exports.CustomConversionGoalErrorEnum_CustomConversionGoalError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomConversionGoalErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CustomConversionGoalErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CustomConversionGoalErrorEnum
 */
exports.CustomConversionGoalErrorEnum = new CustomConversionGoalErrorEnum$Type();
