// google-ads-api-client/dist/protos/bindings/ads/errors/customer_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerErrorEnum = exports.CustomerErrorEnum_CustomerError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Set of errors that are related to requests dealing with Customer.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CustomerErrorEnum.CustomerError
 */
var CustomerErrorEnum_CustomerError;
(function (CustomerErrorEnum_CustomerError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomerErrorEnum_CustomerError[CustomerErrorEnum_CustomerError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomerErrorEnum_CustomerError[CustomerErrorEnum_CustomerError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Customer status is not allowed to be changed from DRAFT and CLOSED.
     * Currency code and at least one of country code and time zone needs to be
     * set when status is changed to ENABLED.
     *
     * @generated from protobuf enum value: STATUS_CHANGE_DISALLOWED = 2;
     */
    CustomerErrorEnum_CustomerError[CustomerErrorEnum_CustomerError["STATUS_CHANGE_DISALLOWED"] = 2] = "STATUS_CHANGE_DISALLOWED";
    /**
     * CustomerService cannot get a customer that has not been fully set up.
     *
     * @generated from protobuf enum value: ACCOUNT_NOT_SET_UP = 3;
     */
    CustomerErrorEnum_CustomerError[CustomerErrorEnum_CustomerError["ACCOUNT_NOT_SET_UP"] = 3] = "ACCOUNT_NOT_SET_UP";
})(CustomerErrorEnum_CustomerError = exports.CustomerErrorEnum_CustomerError || (exports.CustomerErrorEnum_CustomerError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomerErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CustomerErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CustomerErrorEnum
 */
exports.CustomerErrorEnum = new CustomerErrorEnum$Type();
