// google-ads-api-client/dist/protos/bindings/ads/errors/customer_user_access_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerUserAccessErrorEnum = exports.CustomerUserAccessErrorEnum_CustomerUserAccessError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible customer user access errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.CustomerUserAccessErrorEnum.CustomerUserAccessError
 */
var CustomerUserAccessErrorEnum_CustomerUserAccessError;
(function (CustomerUserAccessErrorEnum_CustomerUserAccessError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    CustomerUserAccessErrorEnum_CustomerUserAccessError[CustomerUserAccessErrorEnum_CustomerUserAccessError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    CustomerUserAccessErrorEnum_CustomerUserAccessError[CustomerUserAccessErrorEnum_CustomerUserAccessError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * There is no user associated with the user id specified.
     *
     * @generated from protobuf enum value: INVALID_USER_ID = 2;
     */
    CustomerUserAccessErrorEnum_CustomerUserAccessError[CustomerUserAccessErrorEnum_CustomerUserAccessError["INVALID_USER_ID"] = 2] = "INVALID_USER_ID";
    /**
     * Unable to remove the access between the user and customer.
     *
     * @generated from protobuf enum value: REMOVAL_DISALLOWED = 3;
     */
    CustomerUserAccessErrorEnum_CustomerUserAccessError[CustomerUserAccessErrorEnum_CustomerUserAccessError["REMOVAL_DISALLOWED"] = 3] = "REMOVAL_DISALLOWED";
    /**
     * Unable to add or update the access role as specified.
     *
     * @generated from protobuf enum value: DISALLOWED_ACCESS_ROLE = 4;
     */
    CustomerUserAccessErrorEnum_CustomerUserAccessError[CustomerUserAccessErrorEnum_CustomerUserAccessError["DISALLOWED_ACCESS_ROLE"] = 4] = "DISALLOWED_ACCESS_ROLE";
    /**
     * The user can't remove itself from an active serving customer if it's the
     * last admin user and the customer doesn't have any owner manager
     *
     * @generated from protobuf enum value: LAST_ADMIN_USER_OF_SERVING_CUSTOMER = 5;
     */
    CustomerUserAccessErrorEnum_CustomerUserAccessError[CustomerUserAccessErrorEnum_CustomerUserAccessError["LAST_ADMIN_USER_OF_SERVING_CUSTOMER"] = 5] = "LAST_ADMIN_USER_OF_SERVING_CUSTOMER";
    /**
     * Last admin user cannot be removed from a manager.
     *
     * @generated from protobuf enum value: LAST_ADMIN_USER_OF_MANAGER = 6;
     */
    CustomerUserAccessErrorEnum_CustomerUserAccessError[CustomerUserAccessErrorEnum_CustomerUserAccessError["LAST_ADMIN_USER_OF_MANAGER"] = 6] = "LAST_ADMIN_USER_OF_MANAGER";
})(CustomerUserAccessErrorEnum_CustomerUserAccessError = exports.CustomerUserAccessErrorEnum_CustomerUserAccessError || (exports.CustomerUserAccessErrorEnum_CustomerUserAccessError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CustomerUserAccessErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.CustomerUserAccessErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.CustomerUserAccessErrorEnum
 */
exports.CustomerUserAccessErrorEnum = new CustomerUserAccessErrorEnum$Type();
