// google-ads-api-client/dist/protos/bindings/ads/errors/database_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseErrorEnum = exports.DatabaseErrorEnum_DatabaseError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible database errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.DatabaseErrorEnum.DatabaseError
 */
var DatabaseErrorEnum_DatabaseError;
(function (DatabaseErrorEnum_DatabaseError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DatabaseErrorEnum_DatabaseError[DatabaseErrorEnum_DatabaseError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DatabaseErrorEnum_DatabaseError[DatabaseErrorEnum_DatabaseError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Multiple requests were attempting to modify the same resource at once.
     * Retry the request.
     *
     * @generated from protobuf enum value: CONCURRENT_MODIFICATION = 2;
     */
    DatabaseErrorEnum_DatabaseError[DatabaseErrorEnum_DatabaseError["CONCURRENT_MODIFICATION"] = 2] = "CONCURRENT_MODIFICATION";
    /**
     * The request conflicted with existing data. This error will usually be
     * replaced with a more specific error if the request is retried.
     *
     * @generated from protobuf enum value: DATA_CONSTRAINT_VIOLATION = 3;
     */
    DatabaseErrorEnum_DatabaseError[DatabaseErrorEnum_DatabaseError["DATA_CONSTRAINT_VIOLATION"] = 3] = "DATA_CONSTRAINT_VIOLATION";
    /**
     * The data written is too large. Split the request into smaller
     * requests.
     *
     * @generated from protobuf enum value: REQUEST_TOO_LARGE = 4;
     */
    DatabaseErrorEnum_DatabaseError[DatabaseErrorEnum_DatabaseError["REQUEST_TOO_LARGE"] = 4] = "REQUEST_TOO_LARGE";
})(DatabaseErrorEnum_DatabaseError = exports.DatabaseErrorEnum_DatabaseError || (exports.DatabaseErrorEnum_DatabaseError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DatabaseErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.DatabaseErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.DatabaseErrorEnum
 */
exports.DatabaseErrorEnum = new DatabaseErrorEnum$Type();
