// google-ads-api-client/dist/protos/bindings/ads/errors/date_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DateErrorEnum = exports.DateErrorEnum_DateError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible date errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.DateErrorEnum.DateError
 */
var DateErrorEnum_DateError;
(function (DateErrorEnum_DateError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Given field values do not correspond to a valid date.
     *
     * @generated from protobuf enum value: INVALID_FIELD_VALUES_IN_DATE = 2;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["INVALID_FIELD_VALUES_IN_DATE"] = 2] = "INVALID_FIELD_VALUES_IN_DATE";
    /**
     * Given field values do not correspond to a valid date time.
     *
     * @generated from protobuf enum value: INVALID_FIELD_VALUES_IN_DATE_TIME = 3;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["INVALID_FIELD_VALUES_IN_DATE_TIME"] = 3] = "INVALID_FIELD_VALUES_IN_DATE_TIME";
    /**
     * The string date's format should be yyyy-mm-dd.
     *
     * @generated from protobuf enum value: INVALID_STRING_DATE = 4;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["INVALID_STRING_DATE"] = 4] = "INVALID_STRING_DATE";
    /**
     * The string date time's format should be yyyy-mm-dd hh:mm:ss.ssssss.
     *
     * @generated from protobuf enum value: INVALID_STRING_DATE_TIME_MICROS = 6;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["INVALID_STRING_DATE_TIME_MICROS"] = 6] = "INVALID_STRING_DATE_TIME_MICROS";
    /**
     * The string date time's format should be yyyy-mm-dd hh:mm:ss.
     *
     * @generated from protobuf enum value: INVALID_STRING_DATE_TIME_SECONDS = 11;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["INVALID_STRING_DATE_TIME_SECONDS"] = 11] = "INVALID_STRING_DATE_TIME_SECONDS";
    /**
     * The string date time's format should be yyyy-mm-dd hh:mm:ss+|-hh:mm.
     *
     * @generated from protobuf enum value: INVALID_STRING_DATE_TIME_SECONDS_WITH_OFFSET = 12;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["INVALID_STRING_DATE_TIME_SECONDS_WITH_OFFSET"] = 12] = "INVALID_STRING_DATE_TIME_SECONDS_WITH_OFFSET";
    /**
     * Date is before allowed minimum.
     *
     * @generated from protobuf enum value: EARLIER_THAN_MINIMUM_DATE = 7;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["EARLIER_THAN_MINIMUM_DATE"] = 7] = "EARLIER_THAN_MINIMUM_DATE";
    /**
     * Date is after allowed maximum.
     *
     * @generated from protobuf enum value: LATER_THAN_MAXIMUM_DATE = 8;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["LATER_THAN_MAXIMUM_DATE"] = 8] = "LATER_THAN_MAXIMUM_DATE";
    /**
     * Date range bounds are not in order.
     *
     * @generated from protobuf enum value: DATE_RANGE_MINIMUM_DATE_LATER_THAN_MAXIMUM_DATE = 9;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["DATE_RANGE_MINIMUM_DATE_LATER_THAN_MAXIMUM_DATE"] = 9] = "DATE_RANGE_MINIMUM_DATE_LATER_THAN_MAXIMUM_DATE";
    /**
     * Both dates in range are null.
     *
     * @generated from protobuf enum value: DATE_RANGE_MINIMUM_AND_MAXIMUM_DATES_BOTH_NULL = 10;
     */
    DateErrorEnum_DateError[DateErrorEnum_DateError["DATE_RANGE_MINIMUM_AND_MAXIMUM_DATES_BOTH_NULL"] = 10] = "DATE_RANGE_MINIMUM_AND_MAXIMUM_DATES_BOTH_NULL";
})(DateErrorEnum_DateError = exports.DateErrorEnum_DateError || (exports.DateErrorEnum_DateError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DateErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.DateErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.DateErrorEnum
 */
exports.DateErrorEnum = new DateErrorEnum$Type();
