// google-ads-api-client/dist/protos/bindings/ads/errors/date_range_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DateRangeErrorEnum = exports.DateRangeErrorEnum_DateRangeError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible date range errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.DateRangeErrorEnum.DateRangeError
 */
var DateRangeErrorEnum_DateRangeError;
(function (DateRangeErrorEnum_DateRangeError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DateRangeErrorEnum_DateRangeError[DateRangeErrorEnum_DateRangeError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DateRangeErrorEnum_DateRangeError[DateRangeErrorEnum_DateRangeError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Invalid date.
     *
     * @generated from protobuf enum value: INVALID_DATE = 2;
     */
    DateRangeErrorEnum_DateRangeError[DateRangeErrorEnum_DateRangeError["INVALID_DATE"] = 2] = "INVALID_DATE";
    /**
     * The start date was after the end date.
     *
     * @generated from protobuf enum value: START_DATE_AFTER_END_DATE = 3;
     */
    DateRangeErrorEnum_DateRangeError[DateRangeErrorEnum_DateRangeError["START_DATE_AFTER_END_DATE"] = 3] = "START_DATE_AFTER_END_DATE";
    /**
     * Cannot set date to past time
     *
     * @generated from protobuf enum value: CANNOT_SET_DATE_TO_PAST = 4;
     */
    DateRangeErrorEnum_DateRangeError[DateRangeErrorEnum_DateRangeError["CANNOT_SET_DATE_TO_PAST"] = 4] = "CANNOT_SET_DATE_TO_PAST";
    /**
     * A date was used that is past the system "last" date.
     *
     * @generated from protobuf enum value: AFTER_MAXIMUM_ALLOWABLE_DATE = 5;
     */
    DateRangeErrorEnum_DateRangeError[DateRangeErrorEnum_DateRangeError["AFTER_MAXIMUM_ALLOWABLE_DATE"] = 5] = "AFTER_MAXIMUM_ALLOWABLE_DATE";
    /**
     * Trying to change start date on a resource that has started.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_START_DATE_IF_ALREADY_STARTED = 6;
     */
    DateRangeErrorEnum_DateRangeError[DateRangeErrorEnum_DateRangeError["CANNOT_MODIFY_START_DATE_IF_ALREADY_STARTED"] = 6] = "CANNOT_MODIFY_START_DATE_IF_ALREADY_STARTED";
})(DateRangeErrorEnum_DateRangeError = exports.DateRangeErrorEnum_DateRangeError || (exports.DateRangeErrorEnum_DateRangeError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DateRangeErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.DateRangeErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.DateRangeErrorEnum
 */
exports.DateRangeErrorEnum = new DateRangeErrorEnum$Type();
