// google-ads-api-client/dist/protos/bindings/ads/errors/distinct_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistinctErrorEnum = exports.DistinctErrorEnum_DistinctError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible distinct errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.DistinctErrorEnum.DistinctError
 */
var DistinctErrorEnum_DistinctError;
(function (DistinctErrorEnum_DistinctError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    DistinctErrorEnum_DistinctError[DistinctErrorEnum_DistinctError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    DistinctErrorEnum_DistinctError[DistinctErrorEnum_DistinctError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Duplicate element.
     *
     * @generated from protobuf enum value: DUPLICATE_ELEMENT = 2;
     */
    DistinctErrorEnum_DistinctError[DistinctErrorEnum_DistinctError["DUPLICATE_ELEMENT"] = 2] = "DUPLICATE_ELEMENT";
    /**
     * Duplicate type.
     *
     * @generated from protobuf enum value: DUPLICATE_TYPE = 3;
     */
    DistinctErrorEnum_DistinctError[DistinctErrorEnum_DistinctError["DUPLICATE_TYPE"] = 3] = "DUPLICATE_TYPE";
})(DistinctErrorEnum_DistinctError = exports.DistinctErrorEnum_DistinctError || (exports.DistinctErrorEnum_DistinctError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DistinctErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.DistinctErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.DistinctErrorEnum
 */
exports.DistinctErrorEnum = new DistinctErrorEnum$Type();
