// google-ads-api-client/dist/protos/bindings/ads/errors/enum_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumErrorEnum = exports.EnumErrorEnum_EnumError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible enum errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.EnumErrorEnum.EnumError
 */
var EnumErrorEnum_EnumError;
(function (EnumErrorEnum_EnumError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    EnumErrorEnum_EnumError[EnumErrorEnum_EnumError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    EnumErrorEnum_EnumError[EnumErrorEnum_EnumError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The enum value is not permitted.
     *
     * @generated from protobuf enum value: ENUM_VALUE_NOT_PERMITTED = 3;
     */
    EnumErrorEnum_EnumError[EnumErrorEnum_EnumError["ENUM_VALUE_NOT_PERMITTED"] = 3] = "ENUM_VALUE_NOT_PERMITTED";
})(EnumErrorEnum_EnumError = exports.EnumErrorEnum_EnumError || (exports.EnumErrorEnum_EnumError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class EnumErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.EnumErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.EnumErrorEnum
 */
exports.EnumErrorEnum = new EnumErrorEnum$Type();
