// google-ads-api-client/dist/protos/bindings/ads/errors/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceCountDetails = exports.QuotaErrorDetails = exports.PolicyFindingDetails = exports.PolicyViolationDetails = exports.ErrorDetails = exports.ErrorLocation_FieldPathElement = exports.ErrorLocation = exports.ErrorCode = exports.GoogleAdsError = exports.GoogleAdsFailure = exports.QuotaErrorDetails_QuotaRateScope = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const resource_limit_type_1 = require("../enums/resource_limit_type");
const duration_1 = require("../../google/protobuf/duration");
const policy_1 = require("../common/policy");
const policy_2 = require("../common/policy");
const audience_insights_error_1 = require("./audience_insights_error");
const experiment_arm_error_1 = require("./experiment_arm_error");
const audience_error_1 = require("./audience_error");
const custom_audience_error_1 = require("./custom_audience_error");
const customer_user_access_error_1 = require("./customer_user_access_error");
const third_party_app_analytics_link_error_1 = require("./third_party_app_analytics_link_error");
const account_link_error_1 = require("./account_link_error");
const batch_job_error_1 = require("./batch_job_error");
const user_data_error_1 = require("./user_data_error");
const asset_link_error_1 = require("./asset_link_error");
const time_zone_error_1 = require("./time_zone_error");
const payments_account_error_1 = require("./payments_account_error");
const invoice_error_1 = require("./invoice_error");
const reach_plan_error_1 = require("./reach_plan_error");
const access_invitation_error_1 = require("./access_invitation_error");
const experiment_error_1 = require("./experiment_error");
const currency_code_error_1 = require("./currency_code_error");
const manager_link_error_1 = require("./manager_link_error");
const not_allowlisted_error_1 = require("./not_allowlisted_error");
const offline_user_data_job_error_1 = require("./offline_user_data_job_error");
const size_limit_error_1 = require("./size_limit_error");
const policy_validation_parameter_error_1 = require("./policy_validation_parameter_error");
const partial_failure_error_1 = require("./partial_failure_error");
const policy_violation_error_1 = require("./policy_violation_error");
const feed_item_target_error_1 = require("./feed_item_target_error");
const feed_item_set_link_error_1 = require("./feed_item_set_link_error");
const feed_item_set_error_1 = require("./feed_item_set_error");
const extension_setting_error_1 = require("./extension_setting_error");
const feed_item_validation_error_1 = require("./feed_item_validation_error");
const ad_parameter_error_1 = require("./ad_parameter_error");
const extension_feed_item_error_1 = require("./extension_feed_item_error");
const campaign_experiment_error_1 = require("./campaign_experiment_error");
const custom_interest_error_1 = require("./custom_interest_error");
const campaign_feed_error_1 = require("./campaign_feed_error");
const ad_group_feed_error_1 = require("./ad_group_feed_error");
const customer_feed_error_1 = require("./customer_feed_error");
const feed_mapping_error_1 = require("./feed_mapping_error");
const customer_manager_link_error_1 = require("./customer_manager_link_error");
const customer_client_link_error_1 = require("./customer_client_link_error");
const billing_setup_error_1 = require("./billing_setup_error");
const label_error_1 = require("./label_error");
const feed_item_error_1 = require("./feed_item_error");
const campaign_draft_error_1 = require("./campaign_draft_error");
const geo_target_constant_suggestion_error_1 = require("./geo_target_constant_suggestion_error");
const feed_error_1 = require("./feed_error");
const change_status_error_1 = require("./change_status_error");
const change_event_error_1 = require("./change_event_error");
const user_list_error_1 = require("./user_list_error");
const account_budget_proposal_error_1 = require("./account_budget_proposal_error");
const keyword_plan_idea_error_1 = require("./keyword_plan_idea_error");
const keyword_plan_ad_group_keyword_error_1 = require("./keyword_plan_ad_group_keyword_error");
const keyword_plan_ad_group_error_1 = require("./keyword_plan_ad_group_error");
const keyword_plan_campaign_keyword_error_1 = require("./keyword_plan_campaign_keyword_error");
const keyword_plan_campaign_error_1 = require("./keyword_plan_campaign_error");
const keyword_plan_error_1 = require("./keyword_plan_error");
const enum_error_1 = require("./enum_error");
const policy_finding_error_1 = require("./policy_finding_error");
const database_error_1 = require("./database_error");
const header_error_1 = require("./header_error");
const conversion_value_rule_set_error_1 = require("./conversion_value_rule_set_error");
const conversion_value_rule_error_1 = require("./conversion_value_rule_error");
const conversion_upload_error_1 = require("./conversion_upload_error");
const conversion_custom_variable_error_1 = require("./conversion_custom_variable_error");
const conversion_adjustment_upload_error_1 = require("./conversion_adjustment_upload_error");
const conversion_action_error_1 = require("./conversion_action_error");
const campaign_shared_set_error_1 = require("./campaign_shared_set_error");
const shared_criterion_error_1 = require("./shared_criterion_error");
const shared_set_error_1 = require("./shared_set_error");
const field_error_1 = require("./field_error");
const context_error_1 = require("./context_error");
const ad_group_bid_modifier_error_1 = require("./ad_group_bid_modifier_error");
const youtube_video_registration_error_1 = require("./youtube_video_registration_error");
const resource_count_limit_exceeded_error_1 = require("./resource_count_limit_exceeded_error");
const resource_access_denied_error_1 = require("./resource_access_denied_error");
const operation_access_denied_error_1 = require("./operation_access_denied_error");
const string_length_error_1 = require("./string_length_error");
const string_format_error_1 = require("./string_format_error");
const setting_error_1 = require("./setting_error");
const region_code_error_1 = require("./region_code_error");
const recommendation_error_1 = require("./recommendation_error");
const range_error_1 = require("./range_error");
const operator_error_1 = require("./operator_error");
const null_error_1 = require("./null_error");
const not_empty_error_1 = require("./not_empty_error");
const new_resource_creation_error_1 = require("./new_resource_creation_error");
const multiplier_error_1 = require("./multiplier_error");
const merchant_center_error_1 = require("./merchant_center_error");
const media_file_error_1 = require("./media_file_error");
const media_upload_error_1 = require("./media_upload_error");
const media_bundle_error_1 = require("./media_bundle_error");
const language_code_error_1 = require("./language_code_error");
const image_error_1 = require("./image_error");
const id_error_1 = require("./id_error");
const function_parsing_error_1 = require("./function_parsing_error");
const function_error_1 = require("./function_error");
const feed_attribute_reference_error_1 = require("./feed_attribute_reference_error");
const distinct_error_1 = require("./distinct_error");
const date_range_error_1 = require("./date_range_error");
const date_error_1 = require("./date_error");
const customizer_attribute_error_1 = require("./customizer_attribute_error");
const customer_error_1 = require("./customer_error");
const customer_customizer_error_1 = require("./customer_customizer_error");
const custom_conversion_goal_error_1 = require("./custom_conversion_goal_error");
const criterion_error_1 = require("./criterion_error");
const country_code_error_1 = require("./country_code_error");
const conversion_goal_campaign_config_error_1 = require("./conversion_goal_campaign_config_error");
const collection_size_error_1 = require("./collection_size_error");
const campaign_customizer_error_1 = require("./campaign_customizer_error");
const campaign_conversion_goal_error_1 = require("./campaign_conversion_goal_error");
const campaign_criterion_error_1 = require("./campaign_criterion_error");
const bidding_error_1 = require("./bidding_error");
const asset_set_error_1 = require("./asset_set_error");
const asset_set_link_error_1 = require("./asset_set_link_error");
const asset_set_asset_error_1 = require("./asset_set_asset_error");
const asset_group_error_1 = require("./asset_group_error");
const asset_group_listing_group_filter_error_1 = require("./asset_group_listing_group_filter_error");
const asset_group_asset_error_1 = require("./asset_group_asset_error");
const asset_error_1 = require("./asset_error");
const adx_error_1 = require("./adx_error");
const ad_sharing_error_1 = require("./ad_sharing_error");
const ad_group_ad_error_1 = require("./ad_group_ad_error");
const ad_customizer_error_1 = require("./ad_customizer_error");
const ad_group_customizer_error_1 = require("./ad_group_customizer_error");
const ad_group_criterion_error_1 = require("./ad_group_criterion_error");
const ad_group_criterion_customizer_error_1 = require("./ad_group_criterion_customizer_error");
const authentication_error_1 = require("./authentication_error");
const campaign_error_1 = require("./campaign_error");
const campaign_budget_error_1 = require("./campaign_budget_error");
const ad_group_error_1 = require("./ad_group_error");
const ad_error_1 = require("./ad_error");
const quota_error_1 = require("./quota_error");
const internal_error_1 = require("./internal_error");
const authorization_error_1 = require("./authorization_error");
const field_mask_error_1 = require("./field_mask_error");
const mutate_error_1 = require("./mutate_error");
const query_error_1 = require("./query_error");
const list_operation_error_1 = require("./list_operation_error");
const url_field_error_1 = require("./url_field_error");
const bidding_strategy_error_1 = require("./bidding_strategy_error");
const request_error_1 = require("./request_error");
const value_1 = require("../common/value");
/**
 * Enum of possible scopes that quota buckets belong to.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.QuotaErrorDetails.QuotaRateScope
 */
var QuotaErrorDetails_QuotaRateScope;
(function (QuotaErrorDetails_QuotaRateScope) {
    /**
     * Unspecified enum
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    QuotaErrorDetails_QuotaRateScope[QuotaErrorDetails_QuotaRateScope["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    QuotaErrorDetails_QuotaRateScope[QuotaErrorDetails_QuotaRateScope["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Per customer account quota
     *
     * @generated from protobuf enum value: ACCOUNT = 2;
     */
    QuotaErrorDetails_QuotaRateScope[QuotaErrorDetails_QuotaRateScope["ACCOUNT"] = 2] = "ACCOUNT";
    /**
     * Per project or DevToken quota
     *
     * @generated from protobuf enum value: DEVELOPER = 3;
     */
    QuotaErrorDetails_QuotaRateScope[QuotaErrorDetails_QuotaRateScope["DEVELOPER"] = 3] = "DEVELOPER";
})(QuotaErrorDetails_QuotaRateScope = exports.QuotaErrorDetails_QuotaRateScope || (exports.QuotaErrorDetails_QuotaRateScope = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GoogleAdsFailure$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.GoogleAdsFailure", [
            { no: 1, name: "errors", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.GoogleAdsError },
            { no: 2, name: "request_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { errors: [], requestId: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.errors.GoogleAdsError errors */ 1:
                    message.errors.push(exports.GoogleAdsError.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string request_id */ 2:
                    message.requestId = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.errors.GoogleAdsError errors = 1; */
        for (let i = 0; i < message.errors.length; i++)
            exports.GoogleAdsError.internalBinaryWrite(message.errors[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string request_id = 2; */
        if (message.requestId !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.requestId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.GoogleAdsFailure
 */
exports.GoogleAdsFailure = new GoogleAdsFailure$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GoogleAdsError$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.GoogleAdsError", [
            { no: 1, name: "error_code", kind: "message", T: () => exports.ErrorCode },
            { no: 2, name: "message", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "trigger", kind: "message", T: () => value_1.Value },
            { no: 4, name: "location", kind: "message", T: () => exports.ErrorLocation },
            { no: 5, name: "details", kind: "message", T: () => exports.ErrorDetails }
        ]);
    }
    create(value) {
        const message = { message: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.errors.ErrorCode error_code */ 1:
                    message.errorCode = exports.ErrorCode.internalBinaryRead(reader, reader.uint32(), options, message.errorCode);
                    break;
                case /* string message */ 2:
                    message.message = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.Value trigger */ 3:
                    message.trigger = value_1.Value.internalBinaryRead(reader, reader.uint32(), options, message.trigger);
                    break;
                case /* google.ads.googleads.v11.errors.ErrorLocation location */ 4:
                    message.location = exports.ErrorLocation.internalBinaryRead(reader, reader.uint32(), options, message.location);
                    break;
                case /* google.ads.googleads.v11.errors.ErrorDetails details */ 5:
                    message.details = exports.ErrorDetails.internalBinaryRead(reader, reader.uint32(), options, message.details);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.errors.ErrorCode error_code = 1; */
        if (message.errorCode)
            exports.ErrorCode.internalBinaryWrite(message.errorCode, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string message = 2; */
        if (message.message !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.message);
        /* google.ads.googleads.v11.common.Value trigger = 3; */
        if (message.trigger)
            value_1.Value.internalBinaryWrite(message.trigger, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.errors.ErrorLocation location = 4; */
        if (message.location)
            exports.ErrorLocation.internalBinaryWrite(message.location, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.errors.ErrorDetails details = 5; */
        if (message.details)
            exports.ErrorDetails.internalBinaryWrite(message.details, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.GoogleAdsError
 */
exports.GoogleAdsError = new GoogleAdsError$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ErrorCode$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ErrorCode", [
            { no: 1, name: "request_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.RequestErrorEnum.RequestError", request_error_1.RequestErrorEnum_RequestError] },
            { no: 2, name: "bidding_strategy_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.BiddingStrategyErrorEnum.BiddingStrategyError", bidding_strategy_error_1.BiddingStrategyErrorEnum_BiddingStrategyError] },
            { no: 3, name: "url_field_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.UrlFieldErrorEnum.UrlFieldError", url_field_error_1.UrlFieldErrorEnum_UrlFieldError] },
            { no: 4, name: "list_operation_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ListOperationErrorEnum.ListOperationError", list_operation_error_1.ListOperationErrorEnum_ListOperationError] },
            { no: 5, name: "query_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.QueryErrorEnum.QueryError", query_error_1.QueryErrorEnum_QueryError] },
            { no: 7, name: "mutate_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.MutateErrorEnum.MutateError", mutate_error_1.MutateErrorEnum_MutateError] },
            { no: 8, name: "field_mask_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FieldMaskErrorEnum.FieldMaskError", field_mask_error_1.FieldMaskErrorEnum_FieldMaskError] },
            { no: 9, name: "authorization_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AuthorizationErrorEnum.AuthorizationError", authorization_error_1.AuthorizationErrorEnum_AuthorizationError] },
            { no: 10, name: "internal_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.InternalErrorEnum.InternalError", internal_error_1.InternalErrorEnum_InternalError] },
            { no: 11, name: "quota_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.QuotaErrorEnum.QuotaError", quota_error_1.QuotaErrorEnum_QuotaError] },
            { no: 12, name: "ad_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdErrorEnum.AdError", ad_error_1.AdErrorEnum_AdError] },
            { no: 13, name: "ad_group_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdGroupErrorEnum.AdGroupError", ad_group_error_1.AdGroupErrorEnum_AdGroupError] },
            { no: 14, name: "campaign_budget_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignBudgetErrorEnum.CampaignBudgetError", campaign_budget_error_1.CampaignBudgetErrorEnum_CampaignBudgetError] },
            { no: 15, name: "campaign_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignErrorEnum.CampaignError", campaign_error_1.CampaignErrorEnum_CampaignError] },
            { no: 17, name: "authentication_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AuthenticationErrorEnum.AuthenticationError", authentication_error_1.AuthenticationErrorEnum_AuthenticationError] },
            { no: 161, name: "ad_group_criterion_customizer_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdGroupCriterionCustomizerErrorEnum.AdGroupCriterionCustomizerError", ad_group_criterion_customizer_error_1.AdGroupCriterionCustomizerErrorEnum_AdGroupCriterionCustomizerError] },
            { no: 18, name: "ad_group_criterion_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdGroupCriterionErrorEnum.AdGroupCriterionError", ad_group_criterion_error_1.AdGroupCriterionErrorEnum_AdGroupCriterionError] },
            { no: 159, name: "ad_group_customizer_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdGroupCustomizerErrorEnum.AdGroupCustomizerError", ad_group_customizer_error_1.AdGroupCustomizerErrorEnum_AdGroupCustomizerError] },
            { no: 19, name: "ad_customizer_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdCustomizerErrorEnum.AdCustomizerError", ad_customizer_error_1.AdCustomizerErrorEnum_AdCustomizerError] },
            { no: 21, name: "ad_group_ad_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdGroupAdErrorEnum.AdGroupAdError", ad_group_ad_error_1.AdGroupAdErrorEnum_AdGroupAdError] },
            { no: 24, name: "ad_sharing_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdSharingErrorEnum.AdSharingError", ad_sharing_error_1.AdSharingErrorEnum_AdSharingError] },
            { no: 25, name: "adx_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdxErrorEnum.AdxError", adx_error_1.AdxErrorEnum_AdxError] },
            { no: 107, name: "asset_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AssetErrorEnum.AssetError", asset_error_1.AssetErrorEnum_AssetError] },
            { no: 149, name: "asset_group_asset_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AssetGroupAssetErrorEnum.AssetGroupAssetError", asset_group_asset_error_1.AssetGroupAssetErrorEnum_AssetGroupAssetError] },
            { no: 155, name: "asset_group_listing_group_filter_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AssetGroupListingGroupFilterErrorEnum.AssetGroupListingGroupFilterError", asset_group_listing_group_filter_error_1.AssetGroupListingGroupFilterErrorEnum_AssetGroupListingGroupFilterError] },
            { no: 148, name: "asset_group_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AssetGroupErrorEnum.AssetGroupError", asset_group_error_1.AssetGroupErrorEnum_AssetGroupError] },
            { no: 153, name: "asset_set_asset_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AssetSetAssetErrorEnum.AssetSetAssetError", asset_set_asset_error_1.AssetSetAssetErrorEnum_AssetSetAssetError] },
            { no: 154, name: "asset_set_link_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AssetSetLinkErrorEnum.AssetSetLinkError", asset_set_link_error_1.AssetSetLinkErrorEnum_AssetSetLinkError] },
            { no: 152, name: "asset_set_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AssetSetErrorEnum.AssetSetError", asset_set_error_1.AssetSetErrorEnum_AssetSetError] },
            { no: 26, name: "bidding_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.BiddingErrorEnum.BiddingError", bidding_error_1.BiddingErrorEnum_BiddingError] },
            { no: 29, name: "campaign_criterion_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignCriterionErrorEnum.CampaignCriterionError", campaign_criterion_error_1.CampaignCriterionErrorEnum_CampaignCriterionError] },
            { no: 166, name: "campaign_conversion_goal_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignConversionGoalErrorEnum.CampaignConversionGoalError", campaign_conversion_goal_error_1.CampaignConversionGoalErrorEnum_CampaignConversionGoalError] },
            { no: 160, name: "campaign_customizer_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignCustomizerErrorEnum.CampaignCustomizerError", campaign_customizer_error_1.CampaignCustomizerErrorEnum_CampaignCustomizerError] },
            { no: 31, name: "collection_size_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CollectionSizeErrorEnum.CollectionSizeError", collection_size_error_1.CollectionSizeErrorEnum_CollectionSizeError] },
            { no: 165, name: "conversion_goal_campaign_config_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ConversionGoalCampaignConfigErrorEnum.ConversionGoalCampaignConfigError", conversion_goal_campaign_config_error_1.ConversionGoalCampaignConfigErrorEnum_ConversionGoalCampaignConfigError] },
            { no: 109, name: "country_code_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CountryCodeErrorEnum.CountryCodeError", country_code_error_1.CountryCodeErrorEnum_CountryCodeError] },
            { no: 32, name: "criterion_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CriterionErrorEnum.CriterionError", criterion_error_1.CriterionErrorEnum_CriterionError] },
            { no: 150, name: "custom_conversion_goal_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomConversionGoalErrorEnum.CustomConversionGoalError", custom_conversion_goal_error_1.CustomConversionGoalErrorEnum_CustomConversionGoalError] },
            { no: 158, name: "customer_customizer_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomerCustomizerErrorEnum.CustomerCustomizerError", customer_customizer_error_1.CustomerCustomizerErrorEnum_CustomerCustomizerError] },
            { no: 90, name: "customer_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomerErrorEnum.CustomerError", customer_error_1.CustomerErrorEnum_CustomerError] },
            { no: 151, name: "customizer_attribute_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomizerAttributeErrorEnum.CustomizerAttributeError", customizer_attribute_error_1.CustomizerAttributeErrorEnum_CustomizerAttributeError] },
            { no: 33, name: "date_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.DateErrorEnum.DateError", date_error_1.DateErrorEnum_DateError] },
            { no: 34, name: "date_range_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.DateRangeErrorEnum.DateRangeError", date_range_error_1.DateRangeErrorEnum_DateRangeError] },
            { no: 35, name: "distinct_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.DistinctErrorEnum.DistinctError", distinct_error_1.DistinctErrorEnum_DistinctError] },
            { no: 36, name: "feed_attribute_reference_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FeedAttributeReferenceErrorEnum.FeedAttributeReferenceError", feed_attribute_reference_error_1.FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError] },
            { no: 37, name: "function_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FunctionErrorEnum.FunctionError", function_error_1.FunctionErrorEnum_FunctionError] },
            { no: 38, name: "function_parsing_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FunctionParsingErrorEnum.FunctionParsingError", function_parsing_error_1.FunctionParsingErrorEnum_FunctionParsingError] },
            { no: 39, name: "id_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.IdErrorEnum.IdError", id_error_1.IdErrorEnum_IdError] },
            { no: 40, name: "image_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ImageErrorEnum.ImageError", image_error_1.ImageErrorEnum_ImageError] },
            { no: 110, name: "language_code_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.LanguageCodeErrorEnum.LanguageCodeError", language_code_error_1.LanguageCodeErrorEnum_LanguageCodeError] },
            { no: 42, name: "media_bundle_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.MediaBundleErrorEnum.MediaBundleError", media_bundle_error_1.MediaBundleErrorEnum_MediaBundleError] },
            { no: 116, name: "media_upload_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.MediaUploadErrorEnum.MediaUploadError", media_upload_error_1.MediaUploadErrorEnum_MediaUploadError] },
            { no: 86, name: "media_file_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.MediaFileErrorEnum.MediaFileError", media_file_error_1.MediaFileErrorEnum_MediaFileError] },
            { no: 162, name: "merchant_center_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.MerchantCenterErrorEnum.MerchantCenterError", merchant_center_error_1.MerchantCenterErrorEnum_MerchantCenterError] },
            { no: 44, name: "multiplier_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.MultiplierErrorEnum.MultiplierError", multiplier_error_1.MultiplierErrorEnum_MultiplierError] },
            { no: 45, name: "new_resource_creation_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.NewResourceCreationErrorEnum.NewResourceCreationError", new_resource_creation_error_1.NewResourceCreationErrorEnum_NewResourceCreationError] },
            { no: 46, name: "not_empty_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.NotEmptyErrorEnum.NotEmptyError", not_empty_error_1.NotEmptyErrorEnum_NotEmptyError] },
            { no: 47, name: "null_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.NullErrorEnum.NullError", null_error_1.NullErrorEnum_NullError] },
            { no: 48, name: "operator_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.OperatorErrorEnum.OperatorError", operator_error_1.OperatorErrorEnum_OperatorError] },
            { no: 49, name: "range_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.RangeErrorEnum.RangeError", range_error_1.RangeErrorEnum_RangeError] },
            { no: 58, name: "recommendation_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.RecommendationErrorEnum.RecommendationError", recommendation_error_1.RecommendationErrorEnum_RecommendationError] },
            { no: 51, name: "region_code_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.RegionCodeErrorEnum.RegionCodeError", region_code_error_1.RegionCodeErrorEnum_RegionCodeError] },
            { no: 52, name: "setting_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.SettingErrorEnum.SettingError", setting_error_1.SettingErrorEnum_SettingError] },
            { no: 53, name: "string_format_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.StringFormatErrorEnum.StringFormatError", string_format_error_1.StringFormatErrorEnum_StringFormatError] },
            { no: 54, name: "string_length_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.StringLengthErrorEnum.StringLengthError", string_length_error_1.StringLengthErrorEnum_StringLengthError] },
            { no: 55, name: "operation_access_denied_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.OperationAccessDeniedErrorEnum.OperationAccessDeniedError", operation_access_denied_error_1.OperationAccessDeniedErrorEnum_OperationAccessDeniedError] },
            { no: 56, name: "resource_access_denied_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ResourceAccessDeniedErrorEnum.ResourceAccessDeniedError", resource_access_denied_error_1.ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError] },
            { no: 57, name: "resource_count_limit_exceeded_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ResourceCountLimitExceededErrorEnum.ResourceCountLimitExceededError", resource_count_limit_exceeded_error_1.ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError] },
            { no: 117, name: "youtube_video_registration_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.YoutubeVideoRegistrationErrorEnum.YoutubeVideoRegistrationError", youtube_video_registration_error_1.YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError] },
            { no: 59, name: "ad_group_bid_modifier_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdGroupBidModifierErrorEnum.AdGroupBidModifierError", ad_group_bid_modifier_error_1.AdGroupBidModifierErrorEnum_AdGroupBidModifierError] },
            { no: 60, name: "context_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ContextErrorEnum.ContextError", context_error_1.ContextErrorEnum_ContextError] },
            { no: 61, name: "field_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FieldErrorEnum.FieldError", field_error_1.FieldErrorEnum_FieldError] },
            { no: 62, name: "shared_set_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.SharedSetErrorEnum.SharedSetError", shared_set_error_1.SharedSetErrorEnum_SharedSetError] },
            { no: 63, name: "shared_criterion_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.SharedCriterionErrorEnum.SharedCriterionError", shared_criterion_error_1.SharedCriterionErrorEnum_SharedCriterionError] },
            { no: 64, name: "campaign_shared_set_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignSharedSetErrorEnum.CampaignSharedSetError", campaign_shared_set_error_1.CampaignSharedSetErrorEnum_CampaignSharedSetError] },
            { no: 65, name: "conversion_action_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ConversionActionErrorEnum.ConversionActionError", conversion_action_error_1.ConversionActionErrorEnum_ConversionActionError] },
            { no: 115, name: "conversion_adjustment_upload_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError", conversion_adjustment_upload_error_1.ConversionAdjustmentUploadErrorEnum_ConversionAdjustmentUploadError] },
            { no: 143, name: "conversion_custom_variable_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ConversionCustomVariableErrorEnum.ConversionCustomVariableError", conversion_custom_variable_error_1.ConversionCustomVariableErrorEnum_ConversionCustomVariableError] },
            { no: 111, name: "conversion_upload_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ConversionUploadErrorEnum.ConversionUploadError", conversion_upload_error_1.ConversionUploadErrorEnum_ConversionUploadError] },
            { no: 145, name: "conversion_value_rule_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ConversionValueRuleErrorEnum.ConversionValueRuleError", conversion_value_rule_error_1.ConversionValueRuleErrorEnum_ConversionValueRuleError] },
            { no: 146, name: "conversion_value_rule_set_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ConversionValueRuleSetErrorEnum.ConversionValueRuleSetError", conversion_value_rule_set_error_1.ConversionValueRuleSetErrorEnum_ConversionValueRuleSetError] },
            { no: 66, name: "header_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.HeaderErrorEnum.HeaderError", header_error_1.HeaderErrorEnum_HeaderError] },
            { no: 67, name: "database_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.DatabaseErrorEnum.DatabaseError", database_error_1.DatabaseErrorEnum_DatabaseError] },
            { no: 68, name: "policy_finding_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.PolicyFindingErrorEnum.PolicyFindingError", policy_finding_error_1.PolicyFindingErrorEnum_PolicyFindingError] },
            { no: 70, name: "enum_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.EnumErrorEnum.EnumError", enum_error_1.EnumErrorEnum_EnumError] },
            { no: 71, name: "keyword_plan_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.KeywordPlanErrorEnum.KeywordPlanError", keyword_plan_error_1.KeywordPlanErrorEnum_KeywordPlanError] },
            { no: 72, name: "keyword_plan_campaign_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.KeywordPlanCampaignErrorEnum.KeywordPlanCampaignError", keyword_plan_campaign_error_1.KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError] },
            { no: 132, name: "keyword_plan_campaign_keyword_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.KeywordPlanCampaignKeywordErrorEnum.KeywordPlanCampaignKeywordError", keyword_plan_campaign_keyword_error_1.KeywordPlanCampaignKeywordErrorEnum_KeywordPlanCampaignKeywordError] },
            { no: 74, name: "keyword_plan_ad_group_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.KeywordPlanAdGroupErrorEnum.KeywordPlanAdGroupError", keyword_plan_ad_group_error_1.KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError] },
            { no: 133, name: "keyword_plan_ad_group_keyword_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.KeywordPlanAdGroupKeywordErrorEnum.KeywordPlanAdGroupKeywordError", keyword_plan_ad_group_keyword_error_1.KeywordPlanAdGroupKeywordErrorEnum_KeywordPlanAdGroupKeywordError] },
            { no: 76, name: "keyword_plan_idea_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.KeywordPlanIdeaErrorEnum.KeywordPlanIdeaError", keyword_plan_idea_error_1.KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError] },
            { no: 77, name: "account_budget_proposal_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AccountBudgetProposalErrorEnum.AccountBudgetProposalError", account_budget_proposal_error_1.AccountBudgetProposalErrorEnum_AccountBudgetProposalError] },
            { no: 78, name: "user_list_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.UserListErrorEnum.UserListError", user_list_error_1.UserListErrorEnum_UserListError] },
            { no: 136, name: "change_event_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ChangeEventErrorEnum.ChangeEventError", change_event_error_1.ChangeEventErrorEnum_ChangeEventError] },
            { no: 79, name: "change_status_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ChangeStatusErrorEnum.ChangeStatusError", change_status_error_1.ChangeStatusErrorEnum_ChangeStatusError] },
            { no: 80, name: "feed_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FeedErrorEnum.FeedError", feed_error_1.FeedErrorEnum_FeedError] },
            { no: 81, name: "geo_target_constant_suggestion_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.GeoTargetConstantSuggestionErrorEnum.GeoTargetConstantSuggestionError", geo_target_constant_suggestion_error_1.GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError] },
            { no: 82, name: "campaign_draft_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignDraftErrorEnum.CampaignDraftError", campaign_draft_error_1.CampaignDraftErrorEnum_CampaignDraftError] },
            { no: 83, name: "feed_item_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FeedItemErrorEnum.FeedItemError", feed_item_error_1.FeedItemErrorEnum_FeedItemError] },
            { no: 84, name: "label_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.LabelErrorEnum.LabelError", label_error_1.LabelErrorEnum_LabelError] },
            { no: 87, name: "billing_setup_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.BillingSetupErrorEnum.BillingSetupError", billing_setup_error_1.BillingSetupErrorEnum_BillingSetupError] },
            { no: 88, name: "customer_client_link_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomerClientLinkErrorEnum.CustomerClientLinkError", customer_client_link_error_1.CustomerClientLinkErrorEnum_CustomerClientLinkError] },
            { no: 91, name: "customer_manager_link_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomerManagerLinkErrorEnum.CustomerManagerLinkError", customer_manager_link_error_1.CustomerManagerLinkErrorEnum_CustomerManagerLinkError] },
            { no: 92, name: "feed_mapping_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FeedMappingErrorEnum.FeedMappingError", feed_mapping_error_1.FeedMappingErrorEnum_FeedMappingError] },
            { no: 93, name: "customer_feed_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomerFeedErrorEnum.CustomerFeedError", customer_feed_error_1.CustomerFeedErrorEnum_CustomerFeedError] },
            { no: 94, name: "ad_group_feed_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdGroupFeedErrorEnum.AdGroupFeedError", ad_group_feed_error_1.AdGroupFeedErrorEnum_AdGroupFeedError] },
            { no: 96, name: "campaign_feed_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignFeedErrorEnum.CampaignFeedError", campaign_feed_error_1.CampaignFeedErrorEnum_CampaignFeedError] },
            { no: 97, name: "custom_interest_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomInterestErrorEnum.CustomInterestError", custom_interest_error_1.CustomInterestErrorEnum_CustomInterestError] },
            { no: 98, name: "campaign_experiment_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CampaignExperimentErrorEnum.CampaignExperimentError", campaign_experiment_error_1.CampaignExperimentErrorEnum_CampaignExperimentError] },
            { no: 100, name: "extension_feed_item_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ExtensionFeedItemErrorEnum.ExtensionFeedItemError", extension_feed_item_error_1.ExtensionFeedItemErrorEnum_ExtensionFeedItemError] },
            { no: 101, name: "ad_parameter_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AdParameterErrorEnum.AdParameterError", ad_parameter_error_1.AdParameterErrorEnum_AdParameterError] },
            { no: 102, name: "feed_item_validation_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FeedItemValidationErrorEnum.FeedItemValidationError", feed_item_validation_error_1.FeedItemValidationErrorEnum_FeedItemValidationError] },
            { no: 103, name: "extension_setting_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ExtensionSettingErrorEnum.ExtensionSettingError", extension_setting_error_1.ExtensionSettingErrorEnum_ExtensionSettingError] },
            { no: 140, name: "feed_item_set_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FeedItemSetErrorEnum.FeedItemSetError", feed_item_set_error_1.FeedItemSetErrorEnum_FeedItemSetError] },
            { no: 141, name: "feed_item_set_link_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FeedItemSetLinkErrorEnum.FeedItemSetLinkError", feed_item_set_link_error_1.FeedItemSetLinkErrorEnum_FeedItemSetLinkError] },
            { no: 104, name: "feed_item_target_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.FeedItemTargetErrorEnum.FeedItemTargetError", feed_item_target_error_1.FeedItemTargetErrorEnum_FeedItemTargetError] },
            { no: 105, name: "policy_violation_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.PolicyViolationErrorEnum.PolicyViolationError", policy_violation_error_1.PolicyViolationErrorEnum_PolicyViolationError] },
            { no: 112, name: "partial_failure_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.PartialFailureErrorEnum.PartialFailureError", partial_failure_error_1.PartialFailureErrorEnum_PartialFailureError] },
            { no: 114, name: "policy_validation_parameter_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.PolicyValidationParameterErrorEnum.PolicyValidationParameterError", policy_validation_parameter_error_1.PolicyValidationParameterErrorEnum_PolicyValidationParameterError] },
            { no: 118, name: "size_limit_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.SizeLimitErrorEnum.SizeLimitError", size_limit_error_1.SizeLimitErrorEnum_SizeLimitError] },
            { no: 119, name: "offline_user_data_job_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.OfflineUserDataJobErrorEnum.OfflineUserDataJobError", offline_user_data_job_error_1.OfflineUserDataJobErrorEnum_OfflineUserDataJobError] },
            { no: 137, name: "not_allowlisted_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.NotAllowlistedErrorEnum.NotAllowlistedError", not_allowlisted_error_1.NotAllowlistedErrorEnum_NotAllowlistedError] },
            { no: 121, name: "manager_link_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ManagerLinkErrorEnum.ManagerLinkError", manager_link_error_1.ManagerLinkErrorEnum_ManagerLinkError] },
            { no: 122, name: "currency_code_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CurrencyCodeErrorEnum.CurrencyCodeError", currency_code_error_1.CurrencyCodeErrorEnum_CurrencyCodeError] },
            { no: 123, name: "experiment_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ExperimentErrorEnum.ExperimentError", experiment_error_1.ExperimentErrorEnum_ExperimentError] },
            { no: 124, name: "access_invitation_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AccessInvitationErrorEnum.AccessInvitationError", access_invitation_error_1.AccessInvitationErrorEnum_AccessInvitationError] },
            { no: 125, name: "reach_plan_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ReachPlanErrorEnum.ReachPlanError", reach_plan_error_1.ReachPlanErrorEnum_ReachPlanError] },
            { no: 126, name: "invoice_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.InvoiceErrorEnum.InvoiceError", invoice_error_1.InvoiceErrorEnum_InvoiceError] },
            { no: 127, name: "payments_account_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.PaymentsAccountErrorEnum.PaymentsAccountError", payments_account_error_1.PaymentsAccountErrorEnum_PaymentsAccountError] },
            { no: 128, name: "time_zone_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.TimeZoneErrorEnum.TimeZoneError", time_zone_error_1.TimeZoneErrorEnum_TimeZoneError] },
            { no: 129, name: "asset_link_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AssetLinkErrorEnum.AssetLinkError", asset_link_error_1.AssetLinkErrorEnum_AssetLinkError] },
            { no: 130, name: "user_data_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.UserDataErrorEnum.UserDataError", user_data_error_1.UserDataErrorEnum_UserDataError] },
            { no: 131, name: "batch_job_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.BatchJobErrorEnum.BatchJobError", batch_job_error_1.BatchJobErrorEnum_BatchJobError] },
            { no: 134, name: "account_link_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AccountLinkErrorEnum.AccountLinkError", account_link_error_1.AccountLinkErrorEnum_AccountLinkError] },
            { no: 135, name: "third_party_app_analytics_link_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ThirdPartyAppAnalyticsLinkErrorEnum.ThirdPartyAppAnalyticsLinkError", third_party_app_analytics_link_error_1.ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError] },
            { no: 138, name: "customer_user_access_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomerUserAccessErrorEnum.CustomerUserAccessError", customer_user_access_error_1.CustomerUserAccessErrorEnum_CustomerUserAccessError] },
            { no: 139, name: "custom_audience_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.CustomAudienceErrorEnum.CustomAudienceError", custom_audience_error_1.CustomAudienceErrorEnum_CustomAudienceError] },
            { no: 164, name: "audience_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AudienceErrorEnum.AudienceError", audience_error_1.AudienceErrorEnum_AudienceError] },
            { no: 156, name: "experiment_arm_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.ExperimentArmErrorEnum.ExperimentArmError", experiment_arm_error_1.ExperimentArmErrorEnum_ExperimentArmError] },
            { no: 167, name: "audience_insights_error", kind: "enum", oneof: "errorCode", T: () => ["google.ads.googleads.v11.errors.AudienceInsightsErrorEnum.AudienceInsightsError", audience_insights_error_1.AudienceInsightsErrorEnum_AudienceInsightsError] }
        ]);
    }
    create(value) {
        const message = { errorCode: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.errors.RequestErrorEnum.RequestError request_error */ 1:
                    message.errorCode = {
                        oneofKind: "requestError",
                        requestError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.BiddingStrategyErrorEnum.BiddingStrategyError bidding_strategy_error */ 2:
                    message.errorCode = {
                        oneofKind: "biddingStrategyError",
                        biddingStrategyError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.UrlFieldErrorEnum.UrlFieldError url_field_error */ 3:
                    message.errorCode = {
                        oneofKind: "urlFieldError",
                        urlFieldError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ListOperationErrorEnum.ListOperationError list_operation_error */ 4:
                    message.errorCode = {
                        oneofKind: "listOperationError",
                        listOperationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.QueryErrorEnum.QueryError query_error */ 5:
                    message.errorCode = {
                        oneofKind: "queryError",
                        queryError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.MutateErrorEnum.MutateError mutate_error */ 7:
                    message.errorCode = {
                        oneofKind: "mutateError",
                        mutateError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FieldMaskErrorEnum.FieldMaskError field_mask_error */ 8:
                    message.errorCode = {
                        oneofKind: "fieldMaskError",
                        fieldMaskError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AuthorizationErrorEnum.AuthorizationError authorization_error */ 9:
                    message.errorCode = {
                        oneofKind: "authorizationError",
                        authorizationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.InternalErrorEnum.InternalError internal_error */ 10:
                    message.errorCode = {
                        oneofKind: "internalError",
                        internalError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.QuotaErrorEnum.QuotaError quota_error */ 11:
                    message.errorCode = {
                        oneofKind: "quotaError",
                        quotaError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdErrorEnum.AdError ad_error */ 12:
                    message.errorCode = {
                        oneofKind: "adError",
                        adError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdGroupErrorEnum.AdGroupError ad_group_error */ 13:
                    message.errorCode = {
                        oneofKind: "adGroupError",
                        adGroupError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignBudgetErrorEnum.CampaignBudgetError campaign_budget_error */ 14:
                    message.errorCode = {
                        oneofKind: "campaignBudgetError",
                        campaignBudgetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignErrorEnum.CampaignError campaign_error */ 15:
                    message.errorCode = {
                        oneofKind: "campaignError",
                        campaignError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AuthenticationErrorEnum.AuthenticationError authentication_error */ 17:
                    message.errorCode = {
                        oneofKind: "authenticationError",
                        authenticationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdGroupCriterionCustomizerErrorEnum.AdGroupCriterionCustomizerError ad_group_criterion_customizer_error */ 161:
                    message.errorCode = {
                        oneofKind: "adGroupCriterionCustomizerError",
                        adGroupCriterionCustomizerError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdGroupCriterionErrorEnum.AdGroupCriterionError ad_group_criterion_error */ 18:
                    message.errorCode = {
                        oneofKind: "adGroupCriterionError",
                        adGroupCriterionError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdGroupCustomizerErrorEnum.AdGroupCustomizerError ad_group_customizer_error */ 159:
                    message.errorCode = {
                        oneofKind: "adGroupCustomizerError",
                        adGroupCustomizerError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdCustomizerErrorEnum.AdCustomizerError ad_customizer_error */ 19:
                    message.errorCode = {
                        oneofKind: "adCustomizerError",
                        adCustomizerError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdGroupAdErrorEnum.AdGroupAdError ad_group_ad_error */ 21:
                    message.errorCode = {
                        oneofKind: "adGroupAdError",
                        adGroupAdError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdSharingErrorEnum.AdSharingError ad_sharing_error */ 24:
                    message.errorCode = {
                        oneofKind: "adSharingError",
                        adSharingError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdxErrorEnum.AdxError adx_error */ 25:
                    message.errorCode = {
                        oneofKind: "adxError",
                        adxError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AssetErrorEnum.AssetError asset_error */ 107:
                    message.errorCode = {
                        oneofKind: "assetError",
                        assetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AssetGroupAssetErrorEnum.AssetGroupAssetError asset_group_asset_error */ 149:
                    message.errorCode = {
                        oneofKind: "assetGroupAssetError",
                        assetGroupAssetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AssetGroupListingGroupFilterErrorEnum.AssetGroupListingGroupFilterError asset_group_listing_group_filter_error */ 155:
                    message.errorCode = {
                        oneofKind: "assetGroupListingGroupFilterError",
                        assetGroupListingGroupFilterError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AssetGroupErrorEnum.AssetGroupError asset_group_error */ 148:
                    message.errorCode = {
                        oneofKind: "assetGroupError",
                        assetGroupError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AssetSetAssetErrorEnum.AssetSetAssetError asset_set_asset_error */ 153:
                    message.errorCode = {
                        oneofKind: "assetSetAssetError",
                        assetSetAssetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AssetSetLinkErrorEnum.AssetSetLinkError asset_set_link_error */ 154:
                    message.errorCode = {
                        oneofKind: "assetSetLinkError",
                        assetSetLinkError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AssetSetErrorEnum.AssetSetError asset_set_error */ 152:
                    message.errorCode = {
                        oneofKind: "assetSetError",
                        assetSetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.BiddingErrorEnum.BiddingError bidding_error */ 26:
                    message.errorCode = {
                        oneofKind: "biddingError",
                        biddingError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignCriterionErrorEnum.CampaignCriterionError campaign_criterion_error */ 29:
                    message.errorCode = {
                        oneofKind: "campaignCriterionError",
                        campaignCriterionError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignConversionGoalErrorEnum.CampaignConversionGoalError campaign_conversion_goal_error */ 166:
                    message.errorCode = {
                        oneofKind: "campaignConversionGoalError",
                        campaignConversionGoalError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignCustomizerErrorEnum.CampaignCustomizerError campaign_customizer_error */ 160:
                    message.errorCode = {
                        oneofKind: "campaignCustomizerError",
                        campaignCustomizerError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CollectionSizeErrorEnum.CollectionSizeError collection_size_error */ 31:
                    message.errorCode = {
                        oneofKind: "collectionSizeError",
                        collectionSizeError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ConversionGoalCampaignConfigErrorEnum.ConversionGoalCampaignConfigError conversion_goal_campaign_config_error */ 165:
                    message.errorCode = {
                        oneofKind: "conversionGoalCampaignConfigError",
                        conversionGoalCampaignConfigError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CountryCodeErrorEnum.CountryCodeError country_code_error */ 109:
                    message.errorCode = {
                        oneofKind: "countryCodeError",
                        countryCodeError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CriterionErrorEnum.CriterionError criterion_error */ 32:
                    message.errorCode = {
                        oneofKind: "criterionError",
                        criterionError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomConversionGoalErrorEnum.CustomConversionGoalError custom_conversion_goal_error */ 150:
                    message.errorCode = {
                        oneofKind: "customConversionGoalError",
                        customConversionGoalError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomerCustomizerErrorEnum.CustomerCustomizerError customer_customizer_error */ 158:
                    message.errorCode = {
                        oneofKind: "customerCustomizerError",
                        customerCustomizerError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomerErrorEnum.CustomerError customer_error */ 90:
                    message.errorCode = {
                        oneofKind: "customerError",
                        customerError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomizerAttributeErrorEnum.CustomizerAttributeError customizer_attribute_error */ 151:
                    message.errorCode = {
                        oneofKind: "customizerAttributeError",
                        customizerAttributeError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.DateErrorEnum.DateError date_error */ 33:
                    message.errorCode = {
                        oneofKind: "dateError",
                        dateError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.DateRangeErrorEnum.DateRangeError date_range_error */ 34:
                    message.errorCode = {
                        oneofKind: "dateRangeError",
                        dateRangeError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.DistinctErrorEnum.DistinctError distinct_error */ 35:
                    message.errorCode = {
                        oneofKind: "distinctError",
                        distinctError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FeedAttributeReferenceErrorEnum.FeedAttributeReferenceError feed_attribute_reference_error */ 36:
                    message.errorCode = {
                        oneofKind: "feedAttributeReferenceError",
                        feedAttributeReferenceError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FunctionErrorEnum.FunctionError function_error */ 37:
                    message.errorCode = {
                        oneofKind: "functionError",
                        functionError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FunctionParsingErrorEnum.FunctionParsingError function_parsing_error */ 38:
                    message.errorCode = {
                        oneofKind: "functionParsingError",
                        functionParsingError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.IdErrorEnum.IdError id_error */ 39:
                    message.errorCode = {
                        oneofKind: "idError",
                        idError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ImageErrorEnum.ImageError image_error */ 40:
                    message.errorCode = {
                        oneofKind: "imageError",
                        imageError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.LanguageCodeErrorEnum.LanguageCodeError language_code_error */ 110:
                    message.errorCode = {
                        oneofKind: "languageCodeError",
                        languageCodeError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.MediaBundleErrorEnum.MediaBundleError media_bundle_error */ 42:
                    message.errorCode = {
                        oneofKind: "mediaBundleError",
                        mediaBundleError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.MediaUploadErrorEnum.MediaUploadError media_upload_error */ 116:
                    message.errorCode = {
                        oneofKind: "mediaUploadError",
                        mediaUploadError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.MediaFileErrorEnum.MediaFileError media_file_error */ 86:
                    message.errorCode = {
                        oneofKind: "mediaFileError",
                        mediaFileError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.MerchantCenterErrorEnum.MerchantCenterError merchant_center_error */ 162:
                    message.errorCode = {
                        oneofKind: "merchantCenterError",
                        merchantCenterError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.MultiplierErrorEnum.MultiplierError multiplier_error */ 44:
                    message.errorCode = {
                        oneofKind: "multiplierError",
                        multiplierError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.NewResourceCreationErrorEnum.NewResourceCreationError new_resource_creation_error */ 45:
                    message.errorCode = {
                        oneofKind: "newResourceCreationError",
                        newResourceCreationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.NotEmptyErrorEnum.NotEmptyError not_empty_error */ 46:
                    message.errorCode = {
                        oneofKind: "notEmptyError",
                        notEmptyError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.NullErrorEnum.NullError null_error */ 47:
                    message.errorCode = {
                        oneofKind: "nullError",
                        nullError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.OperatorErrorEnum.OperatorError operator_error */ 48:
                    message.errorCode = {
                        oneofKind: "operatorError",
                        operatorError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.RangeErrorEnum.RangeError range_error */ 49:
                    message.errorCode = {
                        oneofKind: "rangeError",
                        rangeError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.RecommendationErrorEnum.RecommendationError recommendation_error */ 58:
                    message.errorCode = {
                        oneofKind: "recommendationError",
                        recommendationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.RegionCodeErrorEnum.RegionCodeError region_code_error */ 51:
                    message.errorCode = {
                        oneofKind: "regionCodeError",
                        regionCodeError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.SettingErrorEnum.SettingError setting_error */ 52:
                    message.errorCode = {
                        oneofKind: "settingError",
                        settingError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.StringFormatErrorEnum.StringFormatError string_format_error */ 53:
                    message.errorCode = {
                        oneofKind: "stringFormatError",
                        stringFormatError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.StringLengthErrorEnum.StringLengthError string_length_error */ 54:
                    message.errorCode = {
                        oneofKind: "stringLengthError",
                        stringLengthError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.OperationAccessDeniedErrorEnum.OperationAccessDeniedError operation_access_denied_error */ 55:
                    message.errorCode = {
                        oneofKind: "operationAccessDeniedError",
                        operationAccessDeniedError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ResourceAccessDeniedErrorEnum.ResourceAccessDeniedError resource_access_denied_error */ 56:
                    message.errorCode = {
                        oneofKind: "resourceAccessDeniedError",
                        resourceAccessDeniedError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ResourceCountLimitExceededErrorEnum.ResourceCountLimitExceededError resource_count_limit_exceeded_error */ 57:
                    message.errorCode = {
                        oneofKind: "resourceCountLimitExceededError",
                        resourceCountLimitExceededError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.YoutubeVideoRegistrationErrorEnum.YoutubeVideoRegistrationError youtube_video_registration_error */ 117:
                    message.errorCode = {
                        oneofKind: "youtubeVideoRegistrationError",
                        youtubeVideoRegistrationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdGroupBidModifierErrorEnum.AdGroupBidModifierError ad_group_bid_modifier_error */ 59:
                    message.errorCode = {
                        oneofKind: "adGroupBidModifierError",
                        adGroupBidModifierError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ContextErrorEnum.ContextError context_error */ 60:
                    message.errorCode = {
                        oneofKind: "contextError",
                        contextError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FieldErrorEnum.FieldError field_error */ 61:
                    message.errorCode = {
                        oneofKind: "fieldError",
                        fieldError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.SharedSetErrorEnum.SharedSetError shared_set_error */ 62:
                    message.errorCode = {
                        oneofKind: "sharedSetError",
                        sharedSetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.SharedCriterionErrorEnum.SharedCriterionError shared_criterion_error */ 63:
                    message.errorCode = {
                        oneofKind: "sharedCriterionError",
                        sharedCriterionError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignSharedSetErrorEnum.CampaignSharedSetError campaign_shared_set_error */ 64:
                    message.errorCode = {
                        oneofKind: "campaignSharedSetError",
                        campaignSharedSetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ConversionActionErrorEnum.ConversionActionError conversion_action_error */ 65:
                    message.errorCode = {
                        oneofKind: "conversionActionError",
                        conversionActionError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError conversion_adjustment_upload_error */ 115:
                    message.errorCode = {
                        oneofKind: "conversionAdjustmentUploadError",
                        conversionAdjustmentUploadError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ConversionCustomVariableErrorEnum.ConversionCustomVariableError conversion_custom_variable_error */ 143:
                    message.errorCode = {
                        oneofKind: "conversionCustomVariableError",
                        conversionCustomVariableError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ConversionUploadErrorEnum.ConversionUploadError conversion_upload_error */ 111:
                    message.errorCode = {
                        oneofKind: "conversionUploadError",
                        conversionUploadError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ConversionValueRuleErrorEnum.ConversionValueRuleError conversion_value_rule_error */ 145:
                    message.errorCode = {
                        oneofKind: "conversionValueRuleError",
                        conversionValueRuleError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ConversionValueRuleSetErrorEnum.ConversionValueRuleSetError conversion_value_rule_set_error */ 146:
                    message.errorCode = {
                        oneofKind: "conversionValueRuleSetError",
                        conversionValueRuleSetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.HeaderErrorEnum.HeaderError header_error */ 66:
                    message.errorCode = {
                        oneofKind: "headerError",
                        headerError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.DatabaseErrorEnum.DatabaseError database_error */ 67:
                    message.errorCode = {
                        oneofKind: "databaseError",
                        databaseError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.PolicyFindingErrorEnum.PolicyFindingError policy_finding_error */ 68:
                    message.errorCode = {
                        oneofKind: "policyFindingError",
                        policyFindingError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.EnumErrorEnum.EnumError enum_error */ 70:
                    message.errorCode = {
                        oneofKind: "enumError",
                        enumError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.KeywordPlanErrorEnum.KeywordPlanError keyword_plan_error */ 71:
                    message.errorCode = {
                        oneofKind: "keywordPlanError",
                        keywordPlanError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.KeywordPlanCampaignErrorEnum.KeywordPlanCampaignError keyword_plan_campaign_error */ 72:
                    message.errorCode = {
                        oneofKind: "keywordPlanCampaignError",
                        keywordPlanCampaignError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.KeywordPlanCampaignKeywordErrorEnum.KeywordPlanCampaignKeywordError keyword_plan_campaign_keyword_error */ 132:
                    message.errorCode = {
                        oneofKind: "keywordPlanCampaignKeywordError",
                        keywordPlanCampaignKeywordError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.KeywordPlanAdGroupErrorEnum.KeywordPlanAdGroupError keyword_plan_ad_group_error */ 74:
                    message.errorCode = {
                        oneofKind: "keywordPlanAdGroupError",
                        keywordPlanAdGroupError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.KeywordPlanAdGroupKeywordErrorEnum.KeywordPlanAdGroupKeywordError keyword_plan_ad_group_keyword_error */ 133:
                    message.errorCode = {
                        oneofKind: "keywordPlanAdGroupKeywordError",
                        keywordPlanAdGroupKeywordError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.KeywordPlanIdeaErrorEnum.KeywordPlanIdeaError keyword_plan_idea_error */ 76:
                    message.errorCode = {
                        oneofKind: "keywordPlanIdeaError",
                        keywordPlanIdeaError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AccountBudgetProposalErrorEnum.AccountBudgetProposalError account_budget_proposal_error */ 77:
                    message.errorCode = {
                        oneofKind: "accountBudgetProposalError",
                        accountBudgetProposalError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.UserListErrorEnum.UserListError user_list_error */ 78:
                    message.errorCode = {
                        oneofKind: "userListError",
                        userListError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ChangeEventErrorEnum.ChangeEventError change_event_error */ 136:
                    message.errorCode = {
                        oneofKind: "changeEventError",
                        changeEventError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ChangeStatusErrorEnum.ChangeStatusError change_status_error */ 79:
                    message.errorCode = {
                        oneofKind: "changeStatusError",
                        changeStatusError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FeedErrorEnum.FeedError feed_error */ 80:
                    message.errorCode = {
                        oneofKind: "feedError",
                        feedError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.GeoTargetConstantSuggestionErrorEnum.GeoTargetConstantSuggestionError geo_target_constant_suggestion_error */ 81:
                    message.errorCode = {
                        oneofKind: "geoTargetConstantSuggestionError",
                        geoTargetConstantSuggestionError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignDraftErrorEnum.CampaignDraftError campaign_draft_error */ 82:
                    message.errorCode = {
                        oneofKind: "campaignDraftError",
                        campaignDraftError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FeedItemErrorEnum.FeedItemError feed_item_error */ 83:
                    message.errorCode = {
                        oneofKind: "feedItemError",
                        feedItemError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.LabelErrorEnum.LabelError label_error */ 84:
                    message.errorCode = {
                        oneofKind: "labelError",
                        labelError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.BillingSetupErrorEnum.BillingSetupError billing_setup_error */ 87:
                    message.errorCode = {
                        oneofKind: "billingSetupError",
                        billingSetupError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomerClientLinkErrorEnum.CustomerClientLinkError customer_client_link_error */ 88:
                    message.errorCode = {
                        oneofKind: "customerClientLinkError",
                        customerClientLinkError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomerManagerLinkErrorEnum.CustomerManagerLinkError customer_manager_link_error */ 91:
                    message.errorCode = {
                        oneofKind: "customerManagerLinkError",
                        customerManagerLinkError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FeedMappingErrorEnum.FeedMappingError feed_mapping_error */ 92:
                    message.errorCode = {
                        oneofKind: "feedMappingError",
                        feedMappingError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomerFeedErrorEnum.CustomerFeedError customer_feed_error */ 93:
                    message.errorCode = {
                        oneofKind: "customerFeedError",
                        customerFeedError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdGroupFeedErrorEnum.AdGroupFeedError ad_group_feed_error */ 94:
                    message.errorCode = {
                        oneofKind: "adGroupFeedError",
                        adGroupFeedError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignFeedErrorEnum.CampaignFeedError campaign_feed_error */ 96:
                    message.errorCode = {
                        oneofKind: "campaignFeedError",
                        campaignFeedError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomInterestErrorEnum.CustomInterestError custom_interest_error */ 97:
                    message.errorCode = {
                        oneofKind: "customInterestError",
                        customInterestError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CampaignExperimentErrorEnum.CampaignExperimentError campaign_experiment_error */ 98:
                    message.errorCode = {
                        oneofKind: "campaignExperimentError",
                        campaignExperimentError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ExtensionFeedItemErrorEnum.ExtensionFeedItemError extension_feed_item_error */ 100:
                    message.errorCode = {
                        oneofKind: "extensionFeedItemError",
                        extensionFeedItemError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AdParameterErrorEnum.AdParameterError ad_parameter_error */ 101:
                    message.errorCode = {
                        oneofKind: "adParameterError",
                        adParameterError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FeedItemValidationErrorEnum.FeedItemValidationError feed_item_validation_error */ 102:
                    message.errorCode = {
                        oneofKind: "feedItemValidationError",
                        feedItemValidationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ExtensionSettingErrorEnum.ExtensionSettingError extension_setting_error */ 103:
                    message.errorCode = {
                        oneofKind: "extensionSettingError",
                        extensionSettingError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FeedItemSetErrorEnum.FeedItemSetError feed_item_set_error */ 140:
                    message.errorCode = {
                        oneofKind: "feedItemSetError",
                        feedItemSetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FeedItemSetLinkErrorEnum.FeedItemSetLinkError feed_item_set_link_error */ 141:
                    message.errorCode = {
                        oneofKind: "feedItemSetLinkError",
                        feedItemSetLinkError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.FeedItemTargetErrorEnum.FeedItemTargetError feed_item_target_error */ 104:
                    message.errorCode = {
                        oneofKind: "feedItemTargetError",
                        feedItemTargetError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.PolicyViolationErrorEnum.PolicyViolationError policy_violation_error */ 105:
                    message.errorCode = {
                        oneofKind: "policyViolationError",
                        policyViolationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.PartialFailureErrorEnum.PartialFailureError partial_failure_error */ 112:
                    message.errorCode = {
                        oneofKind: "partialFailureError",
                        partialFailureError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.PolicyValidationParameterErrorEnum.PolicyValidationParameterError policy_validation_parameter_error */ 114:
                    message.errorCode = {
                        oneofKind: "policyValidationParameterError",
                        policyValidationParameterError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.SizeLimitErrorEnum.SizeLimitError size_limit_error */ 118:
                    message.errorCode = {
                        oneofKind: "sizeLimitError",
                        sizeLimitError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.OfflineUserDataJobErrorEnum.OfflineUserDataJobError offline_user_data_job_error */ 119:
                    message.errorCode = {
                        oneofKind: "offlineUserDataJobError",
                        offlineUserDataJobError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.NotAllowlistedErrorEnum.NotAllowlistedError not_allowlisted_error */ 137:
                    message.errorCode = {
                        oneofKind: "notAllowlistedError",
                        notAllowlistedError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ManagerLinkErrorEnum.ManagerLinkError manager_link_error */ 121:
                    message.errorCode = {
                        oneofKind: "managerLinkError",
                        managerLinkError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CurrencyCodeErrorEnum.CurrencyCodeError currency_code_error */ 122:
                    message.errorCode = {
                        oneofKind: "currencyCodeError",
                        currencyCodeError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ExperimentErrorEnum.ExperimentError experiment_error */ 123:
                    message.errorCode = {
                        oneofKind: "experimentError",
                        experimentError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AccessInvitationErrorEnum.AccessInvitationError access_invitation_error */ 124:
                    message.errorCode = {
                        oneofKind: "accessInvitationError",
                        accessInvitationError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ReachPlanErrorEnum.ReachPlanError reach_plan_error */ 125:
                    message.errorCode = {
                        oneofKind: "reachPlanError",
                        reachPlanError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.InvoiceErrorEnum.InvoiceError invoice_error */ 126:
                    message.errorCode = {
                        oneofKind: "invoiceError",
                        invoiceError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.PaymentsAccountErrorEnum.PaymentsAccountError payments_account_error */ 127:
                    message.errorCode = {
                        oneofKind: "paymentsAccountError",
                        paymentsAccountError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.TimeZoneErrorEnum.TimeZoneError time_zone_error */ 128:
                    message.errorCode = {
                        oneofKind: "timeZoneError",
                        timeZoneError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AssetLinkErrorEnum.AssetLinkError asset_link_error */ 129:
                    message.errorCode = {
                        oneofKind: "assetLinkError",
                        assetLinkError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.UserDataErrorEnum.UserDataError user_data_error */ 130:
                    message.errorCode = {
                        oneofKind: "userDataError",
                        userDataError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.BatchJobErrorEnum.BatchJobError batch_job_error */ 131:
                    message.errorCode = {
                        oneofKind: "batchJobError",
                        batchJobError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AccountLinkErrorEnum.AccountLinkError account_link_error */ 134:
                    message.errorCode = {
                        oneofKind: "accountLinkError",
                        accountLinkError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ThirdPartyAppAnalyticsLinkErrorEnum.ThirdPartyAppAnalyticsLinkError third_party_app_analytics_link_error */ 135:
                    message.errorCode = {
                        oneofKind: "thirdPartyAppAnalyticsLinkError",
                        thirdPartyAppAnalyticsLinkError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomerUserAccessErrorEnum.CustomerUserAccessError customer_user_access_error */ 138:
                    message.errorCode = {
                        oneofKind: "customerUserAccessError",
                        customerUserAccessError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.CustomAudienceErrorEnum.CustomAudienceError custom_audience_error */ 139:
                    message.errorCode = {
                        oneofKind: "customAudienceError",
                        customAudienceError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AudienceErrorEnum.AudienceError audience_error */ 164:
                    message.errorCode = {
                        oneofKind: "audienceError",
                        audienceError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.ExperimentArmErrorEnum.ExperimentArmError experiment_arm_error */ 156:
                    message.errorCode = {
                        oneofKind: "experimentArmError",
                        experimentArmError: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.errors.AudienceInsightsErrorEnum.AudienceInsightsError audience_insights_error */ 167:
                    message.errorCode = {
                        oneofKind: "audienceInsightsError",
                        audienceInsightsError: reader.int32()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.errors.RequestErrorEnum.RequestError request_error = 1; */
        if (message.errorCode.oneofKind === "requestError")
            writer.tag(1, runtime_1.WireType.Varint).int32(message.errorCode.requestError);
        /* google.ads.googleads.v11.errors.BiddingStrategyErrorEnum.BiddingStrategyError bidding_strategy_error = 2; */
        if (message.errorCode.oneofKind === "biddingStrategyError")
            writer.tag(2, runtime_1.WireType.Varint).int32(message.errorCode.biddingStrategyError);
        /* google.ads.googleads.v11.errors.UrlFieldErrorEnum.UrlFieldError url_field_error = 3; */
        if (message.errorCode.oneofKind === "urlFieldError")
            writer.tag(3, runtime_1.WireType.Varint).int32(message.errorCode.urlFieldError);
        /* google.ads.googleads.v11.errors.ListOperationErrorEnum.ListOperationError list_operation_error = 4; */
        if (message.errorCode.oneofKind === "listOperationError")
            writer.tag(4, runtime_1.WireType.Varint).int32(message.errorCode.listOperationError);
        /* google.ads.googleads.v11.errors.QueryErrorEnum.QueryError query_error = 5; */
        if (message.errorCode.oneofKind === "queryError")
            writer.tag(5, runtime_1.WireType.Varint).int32(message.errorCode.queryError);
        /* google.ads.googleads.v11.errors.MutateErrorEnum.MutateError mutate_error = 7; */
        if (message.errorCode.oneofKind === "mutateError")
            writer.tag(7, runtime_1.WireType.Varint).int32(message.errorCode.mutateError);
        /* google.ads.googleads.v11.errors.FieldMaskErrorEnum.FieldMaskError field_mask_error = 8; */
        if (message.errorCode.oneofKind === "fieldMaskError")
            writer.tag(8, runtime_1.WireType.Varint).int32(message.errorCode.fieldMaskError);
        /* google.ads.googleads.v11.errors.AuthorizationErrorEnum.AuthorizationError authorization_error = 9; */
        if (message.errorCode.oneofKind === "authorizationError")
            writer.tag(9, runtime_1.WireType.Varint).int32(message.errorCode.authorizationError);
        /* google.ads.googleads.v11.errors.InternalErrorEnum.InternalError internal_error = 10; */
        if (message.errorCode.oneofKind === "internalError")
            writer.tag(10, runtime_1.WireType.Varint).int32(message.errorCode.internalError);
        /* google.ads.googleads.v11.errors.QuotaErrorEnum.QuotaError quota_error = 11; */
        if (message.errorCode.oneofKind === "quotaError")
            writer.tag(11, runtime_1.WireType.Varint).int32(message.errorCode.quotaError);
        /* google.ads.googleads.v11.errors.AdErrorEnum.AdError ad_error = 12; */
        if (message.errorCode.oneofKind === "adError")
            writer.tag(12, runtime_1.WireType.Varint).int32(message.errorCode.adError);
        /* google.ads.googleads.v11.errors.AdGroupErrorEnum.AdGroupError ad_group_error = 13; */
        if (message.errorCode.oneofKind === "adGroupError")
            writer.tag(13, runtime_1.WireType.Varint).int32(message.errorCode.adGroupError);
        /* google.ads.googleads.v11.errors.CampaignBudgetErrorEnum.CampaignBudgetError campaign_budget_error = 14; */
        if (message.errorCode.oneofKind === "campaignBudgetError")
            writer.tag(14, runtime_1.WireType.Varint).int32(message.errorCode.campaignBudgetError);
        /* google.ads.googleads.v11.errors.CampaignErrorEnum.CampaignError campaign_error = 15; */
        if (message.errorCode.oneofKind === "campaignError")
            writer.tag(15, runtime_1.WireType.Varint).int32(message.errorCode.campaignError);
        /* google.ads.googleads.v11.errors.AuthenticationErrorEnum.AuthenticationError authentication_error = 17; */
        if (message.errorCode.oneofKind === "authenticationError")
            writer.tag(17, runtime_1.WireType.Varint).int32(message.errorCode.authenticationError);
        /* google.ads.googleads.v11.errors.AdGroupCriterionCustomizerErrorEnum.AdGroupCriterionCustomizerError ad_group_criterion_customizer_error = 161; */
        if (message.errorCode.oneofKind === "adGroupCriterionCustomizerError")
            writer.tag(161, runtime_1.WireType.Varint).int32(message.errorCode.adGroupCriterionCustomizerError);
        /* google.ads.googleads.v11.errors.AdGroupCriterionErrorEnum.AdGroupCriterionError ad_group_criterion_error = 18; */
        if (message.errorCode.oneofKind === "adGroupCriterionError")
            writer.tag(18, runtime_1.WireType.Varint).int32(message.errorCode.adGroupCriterionError);
        /* google.ads.googleads.v11.errors.AdGroupCustomizerErrorEnum.AdGroupCustomizerError ad_group_customizer_error = 159; */
        if (message.errorCode.oneofKind === "adGroupCustomizerError")
            writer.tag(159, runtime_1.WireType.Varint).int32(message.errorCode.adGroupCustomizerError);
        /* google.ads.googleads.v11.errors.AdCustomizerErrorEnum.AdCustomizerError ad_customizer_error = 19; */
        if (message.errorCode.oneofKind === "adCustomizerError")
            writer.tag(19, runtime_1.WireType.Varint).int32(message.errorCode.adCustomizerError);
        /* google.ads.googleads.v11.errors.AdGroupAdErrorEnum.AdGroupAdError ad_group_ad_error = 21; */
        if (message.errorCode.oneofKind === "adGroupAdError")
            writer.tag(21, runtime_1.WireType.Varint).int32(message.errorCode.adGroupAdError);
        /* google.ads.googleads.v11.errors.AdSharingErrorEnum.AdSharingError ad_sharing_error = 24; */
        if (message.errorCode.oneofKind === "adSharingError")
            writer.tag(24, runtime_1.WireType.Varint).int32(message.errorCode.adSharingError);
        /* google.ads.googleads.v11.errors.AdxErrorEnum.AdxError adx_error = 25; */
        if (message.errorCode.oneofKind === "adxError")
            writer.tag(25, runtime_1.WireType.Varint).int32(message.errorCode.adxError);
        /* google.ads.googleads.v11.errors.AssetErrorEnum.AssetError asset_error = 107; */
        if (message.errorCode.oneofKind === "assetError")
            writer.tag(107, runtime_1.WireType.Varint).int32(message.errorCode.assetError);
        /* google.ads.googleads.v11.errors.AssetGroupAssetErrorEnum.AssetGroupAssetError asset_group_asset_error = 149; */
        if (message.errorCode.oneofKind === "assetGroupAssetError")
            writer.tag(149, runtime_1.WireType.Varint).int32(message.errorCode.assetGroupAssetError);
        /* google.ads.googleads.v11.errors.AssetGroupListingGroupFilterErrorEnum.AssetGroupListingGroupFilterError asset_group_listing_group_filter_error = 155; */
        if (message.errorCode.oneofKind === "assetGroupListingGroupFilterError")
            writer.tag(155, runtime_1.WireType.Varint).int32(message.errorCode.assetGroupListingGroupFilterError);
        /* google.ads.googleads.v11.errors.AssetGroupErrorEnum.AssetGroupError asset_group_error = 148; */
        if (message.errorCode.oneofKind === "assetGroupError")
            writer.tag(148, runtime_1.WireType.Varint).int32(message.errorCode.assetGroupError);
        /* google.ads.googleads.v11.errors.AssetSetAssetErrorEnum.AssetSetAssetError asset_set_asset_error = 153; */
        if (message.errorCode.oneofKind === "assetSetAssetError")
            writer.tag(153, runtime_1.WireType.Varint).int32(message.errorCode.assetSetAssetError);
        /* google.ads.googleads.v11.errors.AssetSetLinkErrorEnum.AssetSetLinkError asset_set_link_error = 154; */
        if (message.errorCode.oneofKind === "assetSetLinkError")
            writer.tag(154, runtime_1.WireType.Varint).int32(message.errorCode.assetSetLinkError);
        /* google.ads.googleads.v11.errors.AssetSetErrorEnum.AssetSetError asset_set_error = 152; */
        if (message.errorCode.oneofKind === "assetSetError")
            writer.tag(152, runtime_1.WireType.Varint).int32(message.errorCode.assetSetError);
        /* google.ads.googleads.v11.errors.BiddingErrorEnum.BiddingError bidding_error = 26; */
        if (message.errorCode.oneofKind === "biddingError")
            writer.tag(26, runtime_1.WireType.Varint).int32(message.errorCode.biddingError);
        /* google.ads.googleads.v11.errors.CampaignCriterionErrorEnum.CampaignCriterionError campaign_criterion_error = 29; */
        if (message.errorCode.oneofKind === "campaignCriterionError")
            writer.tag(29, runtime_1.WireType.Varint).int32(message.errorCode.campaignCriterionError);
        /* google.ads.googleads.v11.errors.CampaignConversionGoalErrorEnum.CampaignConversionGoalError campaign_conversion_goal_error = 166; */
        if (message.errorCode.oneofKind === "campaignConversionGoalError")
            writer.tag(166, runtime_1.WireType.Varint).int32(message.errorCode.campaignConversionGoalError);
        /* google.ads.googleads.v11.errors.CampaignCustomizerErrorEnum.CampaignCustomizerError campaign_customizer_error = 160; */
        if (message.errorCode.oneofKind === "campaignCustomizerError")
            writer.tag(160, runtime_1.WireType.Varint).int32(message.errorCode.campaignCustomizerError);
        /* google.ads.googleads.v11.errors.CollectionSizeErrorEnum.CollectionSizeError collection_size_error = 31; */
        if (message.errorCode.oneofKind === "collectionSizeError")
            writer.tag(31, runtime_1.WireType.Varint).int32(message.errorCode.collectionSizeError);
        /* google.ads.googleads.v11.errors.ConversionGoalCampaignConfigErrorEnum.ConversionGoalCampaignConfigError conversion_goal_campaign_config_error = 165; */
        if (message.errorCode.oneofKind === "conversionGoalCampaignConfigError")
            writer.tag(165, runtime_1.WireType.Varint).int32(message.errorCode.conversionGoalCampaignConfigError);
        /* google.ads.googleads.v11.errors.CountryCodeErrorEnum.CountryCodeError country_code_error = 109; */
        if (message.errorCode.oneofKind === "countryCodeError")
            writer.tag(109, runtime_1.WireType.Varint).int32(message.errorCode.countryCodeError);
        /* google.ads.googleads.v11.errors.CriterionErrorEnum.CriterionError criterion_error = 32; */
        if (message.errorCode.oneofKind === "criterionError")
            writer.tag(32, runtime_1.WireType.Varint).int32(message.errorCode.criterionError);
        /* google.ads.googleads.v11.errors.CustomConversionGoalErrorEnum.CustomConversionGoalError custom_conversion_goal_error = 150; */
        if (message.errorCode.oneofKind === "customConversionGoalError")
            writer.tag(150, runtime_1.WireType.Varint).int32(message.errorCode.customConversionGoalError);
        /* google.ads.googleads.v11.errors.CustomerCustomizerErrorEnum.CustomerCustomizerError customer_customizer_error = 158; */
        if (message.errorCode.oneofKind === "customerCustomizerError")
            writer.tag(158, runtime_1.WireType.Varint).int32(message.errorCode.customerCustomizerError);
        /* google.ads.googleads.v11.errors.CustomerErrorEnum.CustomerError customer_error = 90; */
        if (message.errorCode.oneofKind === "customerError")
            writer.tag(90, runtime_1.WireType.Varint).int32(message.errorCode.customerError);
        /* google.ads.googleads.v11.errors.CustomizerAttributeErrorEnum.CustomizerAttributeError customizer_attribute_error = 151; */
        if (message.errorCode.oneofKind === "customizerAttributeError")
            writer.tag(151, runtime_1.WireType.Varint).int32(message.errorCode.customizerAttributeError);
        /* google.ads.googleads.v11.errors.DateErrorEnum.DateError date_error = 33; */
        if (message.errorCode.oneofKind === "dateError")
            writer.tag(33, runtime_1.WireType.Varint).int32(message.errorCode.dateError);
        /* google.ads.googleads.v11.errors.DateRangeErrorEnum.DateRangeError date_range_error = 34; */
        if (message.errorCode.oneofKind === "dateRangeError")
            writer.tag(34, runtime_1.WireType.Varint).int32(message.errorCode.dateRangeError);
        /* google.ads.googleads.v11.errors.DistinctErrorEnum.DistinctError distinct_error = 35; */
        if (message.errorCode.oneofKind === "distinctError")
            writer.tag(35, runtime_1.WireType.Varint).int32(message.errorCode.distinctError);
        /* google.ads.googleads.v11.errors.FeedAttributeReferenceErrorEnum.FeedAttributeReferenceError feed_attribute_reference_error = 36; */
        if (message.errorCode.oneofKind === "feedAttributeReferenceError")
            writer.tag(36, runtime_1.WireType.Varint).int32(message.errorCode.feedAttributeReferenceError);
        /* google.ads.googleads.v11.errors.FunctionErrorEnum.FunctionError function_error = 37; */
        if (message.errorCode.oneofKind === "functionError")
            writer.tag(37, runtime_1.WireType.Varint).int32(message.errorCode.functionError);
        /* google.ads.googleads.v11.errors.FunctionParsingErrorEnum.FunctionParsingError function_parsing_error = 38; */
        if (message.errorCode.oneofKind === "functionParsingError")
            writer.tag(38, runtime_1.WireType.Varint).int32(message.errorCode.functionParsingError);
        /* google.ads.googleads.v11.errors.IdErrorEnum.IdError id_error = 39; */
        if (message.errorCode.oneofKind === "idError")
            writer.tag(39, runtime_1.WireType.Varint).int32(message.errorCode.idError);
        /* google.ads.googleads.v11.errors.ImageErrorEnum.ImageError image_error = 40; */
        if (message.errorCode.oneofKind === "imageError")
            writer.tag(40, runtime_1.WireType.Varint).int32(message.errorCode.imageError);
        /* google.ads.googleads.v11.errors.LanguageCodeErrorEnum.LanguageCodeError language_code_error = 110; */
        if (message.errorCode.oneofKind === "languageCodeError")
            writer.tag(110, runtime_1.WireType.Varint).int32(message.errorCode.languageCodeError);
        /* google.ads.googleads.v11.errors.MediaBundleErrorEnum.MediaBundleError media_bundle_error = 42; */
        if (message.errorCode.oneofKind === "mediaBundleError")
            writer.tag(42, runtime_1.WireType.Varint).int32(message.errorCode.mediaBundleError);
        /* google.ads.googleads.v11.errors.MediaUploadErrorEnum.MediaUploadError media_upload_error = 116; */
        if (message.errorCode.oneofKind === "mediaUploadError")
            writer.tag(116, runtime_1.WireType.Varint).int32(message.errorCode.mediaUploadError);
        /* google.ads.googleads.v11.errors.MediaFileErrorEnum.MediaFileError media_file_error = 86; */
        if (message.errorCode.oneofKind === "mediaFileError")
            writer.tag(86, runtime_1.WireType.Varint).int32(message.errorCode.mediaFileError);
        /* google.ads.googleads.v11.errors.MerchantCenterErrorEnum.MerchantCenterError merchant_center_error = 162; */
        if (message.errorCode.oneofKind === "merchantCenterError")
            writer.tag(162, runtime_1.WireType.Varint).int32(message.errorCode.merchantCenterError);
        /* google.ads.googleads.v11.errors.MultiplierErrorEnum.MultiplierError multiplier_error = 44; */
        if (message.errorCode.oneofKind === "multiplierError")
            writer.tag(44, runtime_1.WireType.Varint).int32(message.errorCode.multiplierError);
        /* google.ads.googleads.v11.errors.NewResourceCreationErrorEnum.NewResourceCreationError new_resource_creation_error = 45; */
        if (message.errorCode.oneofKind === "newResourceCreationError")
            writer.tag(45, runtime_1.WireType.Varint).int32(message.errorCode.newResourceCreationError);
        /* google.ads.googleads.v11.errors.NotEmptyErrorEnum.NotEmptyError not_empty_error = 46; */
        if (message.errorCode.oneofKind === "notEmptyError")
            writer.tag(46, runtime_1.WireType.Varint).int32(message.errorCode.notEmptyError);
        /* google.ads.googleads.v11.errors.NullErrorEnum.NullError null_error = 47; */
        if (message.errorCode.oneofKind === "nullError")
            writer.tag(47, runtime_1.WireType.Varint).int32(message.errorCode.nullError);
        /* google.ads.googleads.v11.errors.OperatorErrorEnum.OperatorError operator_error = 48; */
        if (message.errorCode.oneofKind === "operatorError")
            writer.tag(48, runtime_1.WireType.Varint).int32(message.errorCode.operatorError);
        /* google.ads.googleads.v11.errors.RangeErrorEnum.RangeError range_error = 49; */
        if (message.errorCode.oneofKind === "rangeError")
            writer.tag(49, runtime_1.WireType.Varint).int32(message.errorCode.rangeError);
        /* google.ads.googleads.v11.errors.RecommendationErrorEnum.RecommendationError recommendation_error = 58; */
        if (message.errorCode.oneofKind === "recommendationError")
            writer.tag(58, runtime_1.WireType.Varint).int32(message.errorCode.recommendationError);
        /* google.ads.googleads.v11.errors.RegionCodeErrorEnum.RegionCodeError region_code_error = 51; */
        if (message.errorCode.oneofKind === "regionCodeError")
            writer.tag(51, runtime_1.WireType.Varint).int32(message.errorCode.regionCodeError);
        /* google.ads.googleads.v11.errors.SettingErrorEnum.SettingError setting_error = 52; */
        if (message.errorCode.oneofKind === "settingError")
            writer.tag(52, runtime_1.WireType.Varint).int32(message.errorCode.settingError);
        /* google.ads.googleads.v11.errors.StringFormatErrorEnum.StringFormatError string_format_error = 53; */
        if (message.errorCode.oneofKind === "stringFormatError")
            writer.tag(53, runtime_1.WireType.Varint).int32(message.errorCode.stringFormatError);
        /* google.ads.googleads.v11.errors.StringLengthErrorEnum.StringLengthError string_length_error = 54; */
        if (message.errorCode.oneofKind === "stringLengthError")
            writer.tag(54, runtime_1.WireType.Varint).int32(message.errorCode.stringLengthError);
        /* google.ads.googleads.v11.errors.OperationAccessDeniedErrorEnum.OperationAccessDeniedError operation_access_denied_error = 55; */
        if (message.errorCode.oneofKind === "operationAccessDeniedError")
            writer.tag(55, runtime_1.WireType.Varint).int32(message.errorCode.operationAccessDeniedError);
        /* google.ads.googleads.v11.errors.ResourceAccessDeniedErrorEnum.ResourceAccessDeniedError resource_access_denied_error = 56; */
        if (message.errorCode.oneofKind === "resourceAccessDeniedError")
            writer.tag(56, runtime_1.WireType.Varint).int32(message.errorCode.resourceAccessDeniedError);
        /* google.ads.googleads.v11.errors.ResourceCountLimitExceededErrorEnum.ResourceCountLimitExceededError resource_count_limit_exceeded_error = 57; */
        if (message.errorCode.oneofKind === "resourceCountLimitExceededError")
            writer.tag(57, runtime_1.WireType.Varint).int32(message.errorCode.resourceCountLimitExceededError);
        /* google.ads.googleads.v11.errors.YoutubeVideoRegistrationErrorEnum.YoutubeVideoRegistrationError youtube_video_registration_error = 117; */
        if (message.errorCode.oneofKind === "youtubeVideoRegistrationError")
            writer.tag(117, runtime_1.WireType.Varint).int32(message.errorCode.youtubeVideoRegistrationError);
        /* google.ads.googleads.v11.errors.AdGroupBidModifierErrorEnum.AdGroupBidModifierError ad_group_bid_modifier_error = 59; */
        if (message.errorCode.oneofKind === "adGroupBidModifierError")
            writer.tag(59, runtime_1.WireType.Varint).int32(message.errorCode.adGroupBidModifierError);
        /* google.ads.googleads.v11.errors.ContextErrorEnum.ContextError context_error = 60; */
        if (message.errorCode.oneofKind === "contextError")
            writer.tag(60, runtime_1.WireType.Varint).int32(message.errorCode.contextError);
        /* google.ads.googleads.v11.errors.FieldErrorEnum.FieldError field_error = 61; */
        if (message.errorCode.oneofKind === "fieldError")
            writer.tag(61, runtime_1.WireType.Varint).int32(message.errorCode.fieldError);
        /* google.ads.googleads.v11.errors.SharedSetErrorEnum.SharedSetError shared_set_error = 62; */
        if (message.errorCode.oneofKind === "sharedSetError")
            writer.tag(62, runtime_1.WireType.Varint).int32(message.errorCode.sharedSetError);
        /* google.ads.googleads.v11.errors.SharedCriterionErrorEnum.SharedCriterionError shared_criterion_error = 63; */
        if (message.errorCode.oneofKind === "sharedCriterionError")
            writer.tag(63, runtime_1.WireType.Varint).int32(message.errorCode.sharedCriterionError);
        /* google.ads.googleads.v11.errors.CampaignSharedSetErrorEnum.CampaignSharedSetError campaign_shared_set_error = 64; */
        if (message.errorCode.oneofKind === "campaignSharedSetError")
            writer.tag(64, runtime_1.WireType.Varint).int32(message.errorCode.campaignSharedSetError);
        /* google.ads.googleads.v11.errors.ConversionActionErrorEnum.ConversionActionError conversion_action_error = 65; */
        if (message.errorCode.oneofKind === "conversionActionError")
            writer.tag(65, runtime_1.WireType.Varint).int32(message.errorCode.conversionActionError);
        /* google.ads.googleads.v11.errors.ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError conversion_adjustment_upload_error = 115; */
        if (message.errorCode.oneofKind === "conversionAdjustmentUploadError")
            writer.tag(115, runtime_1.WireType.Varint).int32(message.errorCode.conversionAdjustmentUploadError);
        /* google.ads.googleads.v11.errors.ConversionCustomVariableErrorEnum.ConversionCustomVariableError conversion_custom_variable_error = 143; */
        if (message.errorCode.oneofKind === "conversionCustomVariableError")
            writer.tag(143, runtime_1.WireType.Varint).int32(message.errorCode.conversionCustomVariableError);
        /* google.ads.googleads.v11.errors.ConversionUploadErrorEnum.ConversionUploadError conversion_upload_error = 111; */
        if (message.errorCode.oneofKind === "conversionUploadError")
            writer.tag(111, runtime_1.WireType.Varint).int32(message.errorCode.conversionUploadError);
        /* google.ads.googleads.v11.errors.ConversionValueRuleErrorEnum.ConversionValueRuleError conversion_value_rule_error = 145; */
        if (message.errorCode.oneofKind === "conversionValueRuleError")
            writer.tag(145, runtime_1.WireType.Varint).int32(message.errorCode.conversionValueRuleError);
        /* google.ads.googleads.v11.errors.ConversionValueRuleSetErrorEnum.ConversionValueRuleSetError conversion_value_rule_set_error = 146; */
        if (message.errorCode.oneofKind === "conversionValueRuleSetError")
            writer.tag(146, runtime_1.WireType.Varint).int32(message.errorCode.conversionValueRuleSetError);
        /* google.ads.googleads.v11.errors.HeaderErrorEnum.HeaderError header_error = 66; */
        if (message.errorCode.oneofKind === "headerError")
            writer.tag(66, runtime_1.WireType.Varint).int32(message.errorCode.headerError);
        /* google.ads.googleads.v11.errors.DatabaseErrorEnum.DatabaseError database_error = 67; */
        if (message.errorCode.oneofKind === "databaseError")
            writer.tag(67, runtime_1.WireType.Varint).int32(message.errorCode.databaseError);
        /* google.ads.googleads.v11.errors.PolicyFindingErrorEnum.PolicyFindingError policy_finding_error = 68; */
        if (message.errorCode.oneofKind === "policyFindingError")
            writer.tag(68, runtime_1.WireType.Varint).int32(message.errorCode.policyFindingError);
        /* google.ads.googleads.v11.errors.EnumErrorEnum.EnumError enum_error = 70; */
        if (message.errorCode.oneofKind === "enumError")
            writer.tag(70, runtime_1.WireType.Varint).int32(message.errorCode.enumError);
        /* google.ads.googleads.v11.errors.KeywordPlanErrorEnum.KeywordPlanError keyword_plan_error = 71; */
        if (message.errorCode.oneofKind === "keywordPlanError")
            writer.tag(71, runtime_1.WireType.Varint).int32(message.errorCode.keywordPlanError);
        /* google.ads.googleads.v11.errors.KeywordPlanCampaignErrorEnum.KeywordPlanCampaignError keyword_plan_campaign_error = 72; */
        if (message.errorCode.oneofKind === "keywordPlanCampaignError")
            writer.tag(72, runtime_1.WireType.Varint).int32(message.errorCode.keywordPlanCampaignError);
        /* google.ads.googleads.v11.errors.KeywordPlanCampaignKeywordErrorEnum.KeywordPlanCampaignKeywordError keyword_plan_campaign_keyword_error = 132; */
        if (message.errorCode.oneofKind === "keywordPlanCampaignKeywordError")
            writer.tag(132, runtime_1.WireType.Varint).int32(message.errorCode.keywordPlanCampaignKeywordError);
        /* google.ads.googleads.v11.errors.KeywordPlanAdGroupErrorEnum.KeywordPlanAdGroupError keyword_plan_ad_group_error = 74; */
        if (message.errorCode.oneofKind === "keywordPlanAdGroupError")
            writer.tag(74, runtime_1.WireType.Varint).int32(message.errorCode.keywordPlanAdGroupError);
        /* google.ads.googleads.v11.errors.KeywordPlanAdGroupKeywordErrorEnum.KeywordPlanAdGroupKeywordError keyword_plan_ad_group_keyword_error = 133; */
        if (message.errorCode.oneofKind === "keywordPlanAdGroupKeywordError")
            writer.tag(133, runtime_1.WireType.Varint).int32(message.errorCode.keywordPlanAdGroupKeywordError);
        /* google.ads.googleads.v11.errors.KeywordPlanIdeaErrorEnum.KeywordPlanIdeaError keyword_plan_idea_error = 76; */
        if (message.errorCode.oneofKind === "keywordPlanIdeaError")
            writer.tag(76, runtime_1.WireType.Varint).int32(message.errorCode.keywordPlanIdeaError);
        /* google.ads.googleads.v11.errors.AccountBudgetProposalErrorEnum.AccountBudgetProposalError account_budget_proposal_error = 77; */
        if (message.errorCode.oneofKind === "accountBudgetProposalError")
            writer.tag(77, runtime_1.WireType.Varint).int32(message.errorCode.accountBudgetProposalError);
        /* google.ads.googleads.v11.errors.UserListErrorEnum.UserListError user_list_error = 78; */
        if (message.errorCode.oneofKind === "userListError")
            writer.tag(78, runtime_1.WireType.Varint).int32(message.errorCode.userListError);
        /* google.ads.googleads.v11.errors.ChangeEventErrorEnum.ChangeEventError change_event_error = 136; */
        if (message.errorCode.oneofKind === "changeEventError")
            writer.tag(136, runtime_1.WireType.Varint).int32(message.errorCode.changeEventError);
        /* google.ads.googleads.v11.errors.ChangeStatusErrorEnum.ChangeStatusError change_status_error = 79; */
        if (message.errorCode.oneofKind === "changeStatusError")
            writer.tag(79, runtime_1.WireType.Varint).int32(message.errorCode.changeStatusError);
        /* google.ads.googleads.v11.errors.FeedErrorEnum.FeedError feed_error = 80; */
        if (message.errorCode.oneofKind === "feedError")
            writer.tag(80, runtime_1.WireType.Varint).int32(message.errorCode.feedError);
        /* google.ads.googleads.v11.errors.GeoTargetConstantSuggestionErrorEnum.GeoTargetConstantSuggestionError geo_target_constant_suggestion_error = 81; */
        if (message.errorCode.oneofKind === "geoTargetConstantSuggestionError")
            writer.tag(81, runtime_1.WireType.Varint).int32(message.errorCode.geoTargetConstantSuggestionError);
        /* google.ads.googleads.v11.errors.CampaignDraftErrorEnum.CampaignDraftError campaign_draft_error = 82; */
        if (message.errorCode.oneofKind === "campaignDraftError")
            writer.tag(82, runtime_1.WireType.Varint).int32(message.errorCode.campaignDraftError);
        /* google.ads.googleads.v11.errors.FeedItemErrorEnum.FeedItemError feed_item_error = 83; */
        if (message.errorCode.oneofKind === "feedItemError")
            writer.tag(83, runtime_1.WireType.Varint).int32(message.errorCode.feedItemError);
        /* google.ads.googleads.v11.errors.LabelErrorEnum.LabelError label_error = 84; */
        if (message.errorCode.oneofKind === "labelError")
            writer.tag(84, runtime_1.WireType.Varint).int32(message.errorCode.labelError);
        /* google.ads.googleads.v11.errors.BillingSetupErrorEnum.BillingSetupError billing_setup_error = 87; */
        if (message.errorCode.oneofKind === "billingSetupError")
            writer.tag(87, runtime_1.WireType.Varint).int32(message.errorCode.billingSetupError);
        /* google.ads.googleads.v11.errors.CustomerClientLinkErrorEnum.CustomerClientLinkError customer_client_link_error = 88; */
        if (message.errorCode.oneofKind === "customerClientLinkError")
            writer.tag(88, runtime_1.WireType.Varint).int32(message.errorCode.customerClientLinkError);
        /* google.ads.googleads.v11.errors.CustomerManagerLinkErrorEnum.CustomerManagerLinkError customer_manager_link_error = 91; */
        if (message.errorCode.oneofKind === "customerManagerLinkError")
            writer.tag(91, runtime_1.WireType.Varint).int32(message.errorCode.customerManagerLinkError);
        /* google.ads.googleads.v11.errors.FeedMappingErrorEnum.FeedMappingError feed_mapping_error = 92; */
        if (message.errorCode.oneofKind === "feedMappingError")
            writer.tag(92, runtime_1.WireType.Varint).int32(message.errorCode.feedMappingError);
        /* google.ads.googleads.v11.errors.CustomerFeedErrorEnum.CustomerFeedError customer_feed_error = 93; */
        if (message.errorCode.oneofKind === "customerFeedError")
            writer.tag(93, runtime_1.WireType.Varint).int32(message.errorCode.customerFeedError);
        /* google.ads.googleads.v11.errors.AdGroupFeedErrorEnum.AdGroupFeedError ad_group_feed_error = 94; */
        if (message.errorCode.oneofKind === "adGroupFeedError")
            writer.tag(94, runtime_1.WireType.Varint).int32(message.errorCode.adGroupFeedError);
        /* google.ads.googleads.v11.errors.CampaignFeedErrorEnum.CampaignFeedError campaign_feed_error = 96; */
        if (message.errorCode.oneofKind === "campaignFeedError")
            writer.tag(96, runtime_1.WireType.Varint).int32(message.errorCode.campaignFeedError);
        /* google.ads.googleads.v11.errors.CustomInterestErrorEnum.CustomInterestError custom_interest_error = 97; */
        if (message.errorCode.oneofKind === "customInterestError")
            writer.tag(97, runtime_1.WireType.Varint).int32(message.errorCode.customInterestError);
        /* google.ads.googleads.v11.errors.CampaignExperimentErrorEnum.CampaignExperimentError campaign_experiment_error = 98; */
        if (message.errorCode.oneofKind === "campaignExperimentError")
            writer.tag(98, runtime_1.WireType.Varint).int32(message.errorCode.campaignExperimentError);
        /* google.ads.googleads.v11.errors.ExtensionFeedItemErrorEnum.ExtensionFeedItemError extension_feed_item_error = 100; */
        if (message.errorCode.oneofKind === "extensionFeedItemError")
            writer.tag(100, runtime_1.WireType.Varint).int32(message.errorCode.extensionFeedItemError);
        /* google.ads.googleads.v11.errors.AdParameterErrorEnum.AdParameterError ad_parameter_error = 101; */
        if (message.errorCode.oneofKind === "adParameterError")
            writer.tag(101, runtime_1.WireType.Varint).int32(message.errorCode.adParameterError);
        /* google.ads.googleads.v11.errors.FeedItemValidationErrorEnum.FeedItemValidationError feed_item_validation_error = 102; */
        if (message.errorCode.oneofKind === "feedItemValidationError")
            writer.tag(102, runtime_1.WireType.Varint).int32(message.errorCode.feedItemValidationError);
        /* google.ads.googleads.v11.errors.ExtensionSettingErrorEnum.ExtensionSettingError extension_setting_error = 103; */
        if (message.errorCode.oneofKind === "extensionSettingError")
            writer.tag(103, runtime_1.WireType.Varint).int32(message.errorCode.extensionSettingError);
        /* google.ads.googleads.v11.errors.FeedItemSetErrorEnum.FeedItemSetError feed_item_set_error = 140; */
        if (message.errorCode.oneofKind === "feedItemSetError")
            writer.tag(140, runtime_1.WireType.Varint).int32(message.errorCode.feedItemSetError);
        /* google.ads.googleads.v11.errors.FeedItemSetLinkErrorEnum.FeedItemSetLinkError feed_item_set_link_error = 141; */
        if (message.errorCode.oneofKind === "feedItemSetLinkError")
            writer.tag(141, runtime_1.WireType.Varint).int32(message.errorCode.feedItemSetLinkError);
        /* google.ads.googleads.v11.errors.FeedItemTargetErrorEnum.FeedItemTargetError feed_item_target_error = 104; */
        if (message.errorCode.oneofKind === "feedItemTargetError")
            writer.tag(104, runtime_1.WireType.Varint).int32(message.errorCode.feedItemTargetError);
        /* google.ads.googleads.v11.errors.PolicyViolationErrorEnum.PolicyViolationError policy_violation_error = 105; */
        if (message.errorCode.oneofKind === "policyViolationError")
            writer.tag(105, runtime_1.WireType.Varint).int32(message.errorCode.policyViolationError);
        /* google.ads.googleads.v11.errors.PartialFailureErrorEnum.PartialFailureError partial_failure_error = 112; */
        if (message.errorCode.oneofKind === "partialFailureError")
            writer.tag(112, runtime_1.WireType.Varint).int32(message.errorCode.partialFailureError);
        /* google.ads.googleads.v11.errors.PolicyValidationParameterErrorEnum.PolicyValidationParameterError policy_validation_parameter_error = 114; */
        if (message.errorCode.oneofKind === "policyValidationParameterError")
            writer.tag(114, runtime_1.WireType.Varint).int32(message.errorCode.policyValidationParameterError);
        /* google.ads.googleads.v11.errors.SizeLimitErrorEnum.SizeLimitError size_limit_error = 118; */
        if (message.errorCode.oneofKind === "sizeLimitError")
            writer.tag(118, runtime_1.WireType.Varint).int32(message.errorCode.sizeLimitError);
        /* google.ads.googleads.v11.errors.OfflineUserDataJobErrorEnum.OfflineUserDataJobError offline_user_data_job_error = 119; */
        if (message.errorCode.oneofKind === "offlineUserDataJobError")
            writer.tag(119, runtime_1.WireType.Varint).int32(message.errorCode.offlineUserDataJobError);
        /* google.ads.googleads.v11.errors.NotAllowlistedErrorEnum.NotAllowlistedError not_allowlisted_error = 137; */
        if (message.errorCode.oneofKind === "notAllowlistedError")
            writer.tag(137, runtime_1.WireType.Varint).int32(message.errorCode.notAllowlistedError);
        /* google.ads.googleads.v11.errors.ManagerLinkErrorEnum.ManagerLinkError manager_link_error = 121; */
        if (message.errorCode.oneofKind === "managerLinkError")
            writer.tag(121, runtime_1.WireType.Varint).int32(message.errorCode.managerLinkError);
        /* google.ads.googleads.v11.errors.CurrencyCodeErrorEnum.CurrencyCodeError currency_code_error = 122; */
        if (message.errorCode.oneofKind === "currencyCodeError")
            writer.tag(122, runtime_1.WireType.Varint).int32(message.errorCode.currencyCodeError);
        /* google.ads.googleads.v11.errors.ExperimentErrorEnum.ExperimentError experiment_error = 123; */
        if (message.errorCode.oneofKind === "experimentError")
            writer.tag(123, runtime_1.WireType.Varint).int32(message.errorCode.experimentError);
        /* google.ads.googleads.v11.errors.AccessInvitationErrorEnum.AccessInvitationError access_invitation_error = 124; */
        if (message.errorCode.oneofKind === "accessInvitationError")
            writer.tag(124, runtime_1.WireType.Varint).int32(message.errorCode.accessInvitationError);
        /* google.ads.googleads.v11.errors.ReachPlanErrorEnum.ReachPlanError reach_plan_error = 125; */
        if (message.errorCode.oneofKind === "reachPlanError")
            writer.tag(125, runtime_1.WireType.Varint).int32(message.errorCode.reachPlanError);
        /* google.ads.googleads.v11.errors.InvoiceErrorEnum.InvoiceError invoice_error = 126; */
        if (message.errorCode.oneofKind === "invoiceError")
            writer.tag(126, runtime_1.WireType.Varint).int32(message.errorCode.invoiceError);
        /* google.ads.googleads.v11.errors.PaymentsAccountErrorEnum.PaymentsAccountError payments_account_error = 127; */
        if (message.errorCode.oneofKind === "paymentsAccountError")
            writer.tag(127, runtime_1.WireType.Varint).int32(message.errorCode.paymentsAccountError);
        /* google.ads.googleads.v11.errors.TimeZoneErrorEnum.TimeZoneError time_zone_error = 128; */
        if (message.errorCode.oneofKind === "timeZoneError")
            writer.tag(128, runtime_1.WireType.Varint).int32(message.errorCode.timeZoneError);
        /* google.ads.googleads.v11.errors.AssetLinkErrorEnum.AssetLinkError asset_link_error = 129; */
        if (message.errorCode.oneofKind === "assetLinkError")
            writer.tag(129, runtime_1.WireType.Varint).int32(message.errorCode.assetLinkError);
        /* google.ads.googleads.v11.errors.UserDataErrorEnum.UserDataError user_data_error = 130; */
        if (message.errorCode.oneofKind === "userDataError")
            writer.tag(130, runtime_1.WireType.Varint).int32(message.errorCode.userDataError);
        /* google.ads.googleads.v11.errors.BatchJobErrorEnum.BatchJobError batch_job_error = 131; */
        if (message.errorCode.oneofKind === "batchJobError")
            writer.tag(131, runtime_1.WireType.Varint).int32(message.errorCode.batchJobError);
        /* google.ads.googleads.v11.errors.AccountLinkErrorEnum.AccountLinkError account_link_error = 134; */
        if (message.errorCode.oneofKind === "accountLinkError")
            writer.tag(134, runtime_1.WireType.Varint).int32(message.errorCode.accountLinkError);
        /* google.ads.googleads.v11.errors.ThirdPartyAppAnalyticsLinkErrorEnum.ThirdPartyAppAnalyticsLinkError third_party_app_analytics_link_error = 135; */
        if (message.errorCode.oneofKind === "thirdPartyAppAnalyticsLinkError")
            writer.tag(135, runtime_1.WireType.Varint).int32(message.errorCode.thirdPartyAppAnalyticsLinkError);
        /* google.ads.googleads.v11.errors.CustomerUserAccessErrorEnum.CustomerUserAccessError customer_user_access_error = 138; */
        if (message.errorCode.oneofKind === "customerUserAccessError")
            writer.tag(138, runtime_1.WireType.Varint).int32(message.errorCode.customerUserAccessError);
        /* google.ads.googleads.v11.errors.CustomAudienceErrorEnum.CustomAudienceError custom_audience_error = 139; */
        if (message.errorCode.oneofKind === "customAudienceError")
            writer.tag(139, runtime_1.WireType.Varint).int32(message.errorCode.customAudienceError);
        /* google.ads.googleads.v11.errors.AudienceErrorEnum.AudienceError audience_error = 164; */
        if (message.errorCode.oneofKind === "audienceError")
            writer.tag(164, runtime_1.WireType.Varint).int32(message.errorCode.audienceError);
        /* google.ads.googleads.v11.errors.ExperimentArmErrorEnum.ExperimentArmError experiment_arm_error = 156; */
        if (message.errorCode.oneofKind === "experimentArmError")
            writer.tag(156, runtime_1.WireType.Varint).int32(message.errorCode.experimentArmError);
        /* google.ads.googleads.v11.errors.AudienceInsightsErrorEnum.AudienceInsightsError audience_insights_error = 167; */
        if (message.errorCode.oneofKind === "audienceInsightsError")
            writer.tag(167, runtime_1.WireType.Varint).int32(message.errorCode.audienceInsightsError);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ErrorCode
 */
exports.ErrorCode = new ErrorCode$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ErrorLocation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ErrorLocation", [
            { no: 2, name: "field_path_elements", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.ErrorLocation_FieldPathElement }
        ]);
    }
    create(value) {
        const message = { fieldPathElements: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.errors.ErrorLocation.FieldPathElement field_path_elements */ 2:
                    message.fieldPathElements.push(exports.ErrorLocation_FieldPathElement.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.errors.ErrorLocation.FieldPathElement field_path_elements = 2; */
        for (let i = 0; i < message.fieldPathElements.length; i++)
            exports.ErrorLocation_FieldPathElement.internalBinaryWrite(message.fieldPathElements[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ErrorLocation
 */
exports.ErrorLocation = new ErrorLocation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ErrorLocation_FieldPathElement$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ErrorLocation.FieldPathElement", [
            { no: 1, name: "field_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "index", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = { fieldName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string field_name */ 1:
                    message.fieldName = reader.string();
                    break;
                case /* optional int32 index */ 3:
                    message.index = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string field_name = 1; */
        if (message.fieldName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.fieldName);
        /* optional int32 index = 3; */
        if (message.index !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.index);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ErrorLocation.FieldPathElement
 */
exports.ErrorLocation_FieldPathElement = new ErrorLocation_FieldPathElement$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ErrorDetails$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ErrorDetails", [
            { no: 1, name: "unpublished_error_code", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "policy_violation_details", kind: "message", T: () => exports.PolicyViolationDetails },
            { no: 3, name: "policy_finding_details", kind: "message", T: () => exports.PolicyFindingDetails },
            { no: 4, name: "quota_error_details", kind: "message", T: () => exports.QuotaErrorDetails },
            { no: 5, name: "resource_count_details", kind: "message", T: () => exports.ResourceCountDetails }
        ]);
    }
    create(value) {
        const message = { unpublishedErrorCode: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string unpublished_error_code */ 1:
                    message.unpublishedErrorCode = reader.string();
                    break;
                case /* google.ads.googleads.v11.errors.PolicyViolationDetails policy_violation_details */ 2:
                    message.policyViolationDetails = exports.PolicyViolationDetails.internalBinaryRead(reader, reader.uint32(), options, message.policyViolationDetails);
                    break;
                case /* google.ads.googleads.v11.errors.PolicyFindingDetails policy_finding_details */ 3:
                    message.policyFindingDetails = exports.PolicyFindingDetails.internalBinaryRead(reader, reader.uint32(), options, message.policyFindingDetails);
                    break;
                case /* google.ads.googleads.v11.errors.QuotaErrorDetails quota_error_details */ 4:
                    message.quotaErrorDetails = exports.QuotaErrorDetails.internalBinaryRead(reader, reader.uint32(), options, message.quotaErrorDetails);
                    break;
                case /* google.ads.googleads.v11.errors.ResourceCountDetails resource_count_details */ 5:
                    message.resourceCountDetails = exports.ResourceCountDetails.internalBinaryRead(reader, reader.uint32(), options, message.resourceCountDetails);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string unpublished_error_code = 1; */
        if (message.unpublishedErrorCode !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.unpublishedErrorCode);
        /* google.ads.googleads.v11.errors.PolicyViolationDetails policy_violation_details = 2; */
        if (message.policyViolationDetails)
            exports.PolicyViolationDetails.internalBinaryWrite(message.policyViolationDetails, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.errors.PolicyFindingDetails policy_finding_details = 3; */
        if (message.policyFindingDetails)
            exports.PolicyFindingDetails.internalBinaryWrite(message.policyFindingDetails, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.errors.QuotaErrorDetails quota_error_details = 4; */
        if (message.quotaErrorDetails)
            exports.QuotaErrorDetails.internalBinaryWrite(message.quotaErrorDetails, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.errors.ResourceCountDetails resource_count_details = 5; */
        if (message.resourceCountDetails)
            exports.ResourceCountDetails.internalBinaryWrite(message.resourceCountDetails, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ErrorDetails
 */
exports.ErrorDetails = new ErrorDetails$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PolicyViolationDetails$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.PolicyViolationDetails", [
            { no: 2, name: "external_policy_description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "key", kind: "message", T: () => policy_2.PolicyViolationKey },
            { no: 5, name: "external_policy_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "is_exemptible", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { externalPolicyDescription: "", externalPolicyName: "", isExemptible: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string external_policy_description */ 2:
                    message.externalPolicyDescription = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.PolicyViolationKey key */ 4:
                    message.key = policy_2.PolicyViolationKey.internalBinaryRead(reader, reader.uint32(), options, message.key);
                    break;
                case /* string external_policy_name */ 5:
                    message.externalPolicyName = reader.string();
                    break;
                case /* bool is_exemptible */ 6:
                    message.isExemptible = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string external_policy_description = 2; */
        if (message.externalPolicyDescription !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.externalPolicyDescription);
        /* google.ads.googleads.v11.common.PolicyViolationKey key = 4; */
        if (message.key)
            policy_2.PolicyViolationKey.internalBinaryWrite(message.key, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string external_policy_name = 5; */
        if (message.externalPolicyName !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.externalPolicyName);
        /* bool is_exemptible = 6; */
        if (message.isExemptible !== false)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.isExemptible);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.PolicyViolationDetails
 */
exports.PolicyViolationDetails = new PolicyViolationDetails$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PolicyFindingDetails$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.PolicyFindingDetails", [
            { no: 1, name: "policy_topic_entries", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => policy_1.PolicyTopicEntry }
        ]);
    }
    create(value) {
        const message = { policyTopicEntries: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.PolicyTopicEntry policy_topic_entries */ 1:
                    message.policyTopicEntries.push(policy_1.PolicyTopicEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.PolicyTopicEntry policy_topic_entries = 1; */
        for (let i = 0; i < message.policyTopicEntries.length; i++)
            policy_1.PolicyTopicEntry.internalBinaryWrite(message.policyTopicEntries[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.PolicyFindingDetails
 */
exports.PolicyFindingDetails = new PolicyFindingDetails$Type();
// @generated message type with reflection information, may provide speed optimized methods
class QuotaErrorDetails$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.QuotaErrorDetails", [
            { no: 1, name: "rate_scope", kind: "enum", T: () => ["google.ads.googleads.v11.errors.QuotaErrorDetails.QuotaRateScope", QuotaErrorDetails_QuotaRateScope] },
            { no: 2, name: "rate_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "retry_delay", kind: "message", T: () => duration_1.Duration }
        ]);
    }
    create(value) {
        const message = { rateScope: 0, rateName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.errors.QuotaErrorDetails.QuotaRateScope rate_scope */ 1:
                    message.rateScope = reader.int32();
                    break;
                case /* string rate_name */ 2:
                    message.rateName = reader.string();
                    break;
                case /* google.protobuf.Duration retry_delay */ 3:
                    message.retryDelay = duration_1.Duration.internalBinaryRead(reader, reader.uint32(), options, message.retryDelay);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.errors.QuotaErrorDetails.QuotaRateScope rate_scope = 1; */
        if (message.rateScope !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.rateScope);
        /* string rate_name = 2; */
        if (message.rateName !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.rateName);
        /* google.protobuf.Duration retry_delay = 3; */
        if (message.retryDelay)
            duration_1.Duration.internalBinaryWrite(message.retryDelay, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.QuotaErrorDetails
 */
exports.QuotaErrorDetails = new QuotaErrorDetails$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ResourceCountDetails$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ResourceCountDetails", [
            { no: 1, name: "enclosing_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "enclosing_resource", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "limit", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "limit_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ResourceLimitTypeEnum.ResourceLimitType", resource_limit_type_1.ResourceLimitTypeEnum_ResourceLimitType] },
            { no: 4, name: "existing_count", kind: "scalar", T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = { enclosingId: "", enclosingResource: "", limit: 0, limitType: 0, existingCount: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string enclosing_id */ 1:
                    message.enclosingId = reader.string();
                    break;
                case /* string enclosing_resource */ 5:
                    message.enclosingResource = reader.string();
                    break;
                case /* int32 limit */ 2:
                    message.limit = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.ResourceLimitTypeEnum.ResourceLimitType limit_type */ 3:
                    message.limitType = reader.int32();
                    break;
                case /* int32 existing_count */ 4:
                    message.existingCount = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string enclosing_id = 1; */
        if (message.enclosingId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.enclosingId);
        /* string enclosing_resource = 5; */
        if (message.enclosingResource !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.enclosingResource);
        /* int32 limit = 2; */
        if (message.limit !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.limit);
        /* google.ads.googleads.v11.enums.ResourceLimitTypeEnum.ResourceLimitType limit_type = 3; */
        if (message.limitType !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.limitType);
        /* int32 existing_count = 4; */
        if (message.existingCount !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.existingCount);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ResourceCountDetails
 */
exports.ResourceCountDetails = new ResourceCountDetails$Type();
