// google-ads-api-client/dist/protos/bindings/ads/errors/extension_feed_item_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionFeedItemErrorEnum = exports.ExtensionFeedItemErrorEnum_ExtensionFeedItemError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible extension feed item errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ExtensionFeedItemErrorEnum.ExtensionFeedItemError
 */
var ExtensionFeedItemErrorEnum_ExtensionFeedItemError;
(function (ExtensionFeedItemErrorEnum_ExtensionFeedItemError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Value is not within the accepted range.
     *
     * @generated from protobuf enum value: VALUE_OUT_OF_RANGE = 2;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["VALUE_OUT_OF_RANGE"] = 2] = "VALUE_OUT_OF_RANGE";
    /**
     * Url list is too long.
     *
     * @generated from protobuf enum value: URL_LIST_TOO_LONG = 3;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["URL_LIST_TOO_LONG"] = 3] = "URL_LIST_TOO_LONG";
    /**
     * Cannot have a geo targeting restriction without having geo targeting.
     *
     * @generated from protobuf enum value: CANNOT_HAVE_RESTRICTION_ON_EMPTY_GEO_TARGETING = 4;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CANNOT_HAVE_RESTRICTION_ON_EMPTY_GEO_TARGETING"] = 4] = "CANNOT_HAVE_RESTRICTION_ON_EMPTY_GEO_TARGETING";
    /**
     * Cannot simultaneously set sitelink field with final urls.
     *
     * @generated from protobuf enum value: CANNOT_SET_WITH_FINAL_URLS = 5;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CANNOT_SET_WITH_FINAL_URLS"] = 5] = "CANNOT_SET_WITH_FINAL_URLS";
    /**
     * Must set field with final urls.
     *
     * @generated from protobuf enum value: CANNOT_SET_WITHOUT_FINAL_URLS = 6;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CANNOT_SET_WITHOUT_FINAL_URLS"] = 6] = "CANNOT_SET_WITHOUT_FINAL_URLS";
    /**
     * Phone number for a call extension is invalid.
     *
     * @generated from protobuf enum value: INVALID_PHONE_NUMBER = 7;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INVALID_PHONE_NUMBER"] = 7] = "INVALID_PHONE_NUMBER";
    /**
     * Phone number for a call extension is not supported for the given country
     * code.
     *
     * @generated from protobuf enum value: PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY = 8;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY"] = 8] = "PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * A carrier specific number in short format is not allowed for call
     * extensions.
     *
     * @generated from protobuf enum value: CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED = 9;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED"] = 9] = "CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED";
    /**
     * Premium rate numbers are not allowed for call extensions.
     *
     * @generated from protobuf enum value: PREMIUM_RATE_NUMBER_NOT_ALLOWED = 10;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["PREMIUM_RATE_NUMBER_NOT_ALLOWED"] = 10] = "PREMIUM_RATE_NUMBER_NOT_ALLOWED";
    /**
     * Phone number type for a call extension is not allowed.
     * For example, personal number is not allowed for a call extension in
     * most regions.
     *
     * @generated from protobuf enum value: DISALLOWED_NUMBER_TYPE = 11;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["DISALLOWED_NUMBER_TYPE"] = 11] = "DISALLOWED_NUMBER_TYPE";
    /**
     * Phone number for a call extension does not meet domestic format
     * requirements.
     *
     * @generated from protobuf enum value: INVALID_DOMESTIC_PHONE_NUMBER_FORMAT = 12;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INVALID_DOMESTIC_PHONE_NUMBER_FORMAT"] = 12] = "INVALID_DOMESTIC_PHONE_NUMBER_FORMAT";
    /**
     * Vanity phone numbers (for example, those including letters) are not
     * allowed for call extensions.
     *
     * @generated from protobuf enum value: VANITY_PHONE_NUMBER_NOT_ALLOWED = 13;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["VANITY_PHONE_NUMBER_NOT_ALLOWED"] = 13] = "VANITY_PHONE_NUMBER_NOT_ALLOWED";
    /**
     * Call conversion action provided for a call extension is invalid.
     *
     * @generated from protobuf enum value: INVALID_CALL_CONVERSION_ACTION = 14;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INVALID_CALL_CONVERSION_ACTION"] = 14] = "INVALID_CALL_CONVERSION_ACTION";
    /**
     * For a call extension, the customer is not on the allow-list for call
     * tracking.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_ON_ALLOWLIST_FOR_CALLTRACKING = 47;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CUSTOMER_NOT_ON_ALLOWLIST_FOR_CALLTRACKING"] = 47] = "CUSTOMER_NOT_ON_ALLOWLIST_FOR_CALLTRACKING";
    /**
     * Call tracking is not supported for the given country for a call
     * extension.
     *
     * @generated from protobuf enum value: CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY = 16;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY"] = 16] = "CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * Customer hasn't consented for call recording, which is required for
     * creating/updating call feed items. See
     * https://support.google.com/google-ads/answer/7412639.
     *
     * @generated from protobuf enum value: CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED = 17;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED"] = 17] = "CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED";
    /**
     * App id provided for an app extension is invalid.
     *
     * @generated from protobuf enum value: INVALID_APP_ID = 18;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INVALID_APP_ID"] = 18] = "INVALID_APP_ID";
    /**
     * Quotation marks present in the review text for a review extension.
     *
     * @generated from protobuf enum value: QUOTES_IN_REVIEW_EXTENSION_SNIPPET = 19;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["QUOTES_IN_REVIEW_EXTENSION_SNIPPET"] = 19] = "QUOTES_IN_REVIEW_EXTENSION_SNIPPET";
    /**
     * Hyphen character present in the review text for a review extension.
     *
     * @generated from protobuf enum value: HYPHENS_IN_REVIEW_EXTENSION_SNIPPET = 20;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["HYPHENS_IN_REVIEW_EXTENSION_SNIPPET"] = 20] = "HYPHENS_IN_REVIEW_EXTENSION_SNIPPET";
    /**
     * A denylisted review source name or url was provided for a review
     * extension.
     *
     * @generated from protobuf enum value: REVIEW_EXTENSION_SOURCE_INELIGIBLE = 21;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["REVIEW_EXTENSION_SOURCE_INELIGIBLE"] = 21] = "REVIEW_EXTENSION_SOURCE_INELIGIBLE";
    /**
     * Review source name should not be found in the review text.
     *
     * @generated from protobuf enum value: SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT = 22;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT"] = 22] = "SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT";
    /**
     * Inconsistent currency codes.
     *
     * @generated from protobuf enum value: INCONSISTENT_CURRENCY_CODES = 23;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INCONSISTENT_CURRENCY_CODES"] = 23] = "INCONSISTENT_CURRENCY_CODES";
    /**
     * Price extension cannot have duplicated headers.
     *
     * @generated from protobuf enum value: PRICE_EXTENSION_HAS_DUPLICATED_HEADERS = 24;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["PRICE_EXTENSION_HAS_DUPLICATED_HEADERS"] = 24] = "PRICE_EXTENSION_HAS_DUPLICATED_HEADERS";
    /**
     * Price item cannot have duplicated header and description.
     *
     * @generated from protobuf enum value: PRICE_ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION = 25;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["PRICE_ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION"] = 25] = "PRICE_ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION";
    /**
     * Price extension has too few items.
     *
     * @generated from protobuf enum value: PRICE_EXTENSION_HAS_TOO_FEW_ITEMS = 26;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["PRICE_EXTENSION_HAS_TOO_FEW_ITEMS"] = 26] = "PRICE_EXTENSION_HAS_TOO_FEW_ITEMS";
    /**
     * Price extension has too many items.
     *
     * @generated from protobuf enum value: PRICE_EXTENSION_HAS_TOO_MANY_ITEMS = 27;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["PRICE_EXTENSION_HAS_TOO_MANY_ITEMS"] = 27] = "PRICE_EXTENSION_HAS_TOO_MANY_ITEMS";
    /**
     * The input value is not currently supported.
     *
     * @generated from protobuf enum value: UNSUPPORTED_VALUE = 28;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["UNSUPPORTED_VALUE"] = 28] = "UNSUPPORTED_VALUE";
    /**
     * The input value is not currently supported in the selected language of an
     * extension.
     *
     * @generated from protobuf enum value: UNSUPPORTED_VALUE_IN_SELECTED_LANGUAGE = 29;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["UNSUPPORTED_VALUE_IN_SELECTED_LANGUAGE"] = 29] = "UNSUPPORTED_VALUE_IN_SELECTED_LANGUAGE";
    /**
     * Unknown or unsupported device preference.
     *
     * @generated from protobuf enum value: INVALID_DEVICE_PREFERENCE = 30;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INVALID_DEVICE_PREFERENCE"] = 30] = "INVALID_DEVICE_PREFERENCE";
    /**
     * Invalid feed item schedule end time (for example, endHour = 24 and
     * endMinute != 0).
     *
     * @generated from protobuf enum value: INVALID_SCHEDULE_END = 31;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INVALID_SCHEDULE_END"] = 31] = "INVALID_SCHEDULE_END";
    /**
     * Date time zone does not match the account's time zone.
     *
     * @generated from protobuf enum value: DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE = 32;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE"] = 32] = "DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE";
    /**
     * Invalid structured snippet header.
     *
     * @generated from protobuf enum value: INVALID_SNIPPETS_HEADER = 33;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INVALID_SNIPPETS_HEADER"] = 33] = "INVALID_SNIPPETS_HEADER";
    /**
     * Cannot operate on removed feed item.
     *
     * @generated from protobuf enum value: CANNOT_OPERATE_ON_REMOVED_FEED_ITEM = 34;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CANNOT_OPERATE_ON_REMOVED_FEED_ITEM"] = 34] = "CANNOT_OPERATE_ON_REMOVED_FEED_ITEM";
    /**
     * Phone number not supported when call tracking enabled for country.
     *
     * @generated from protobuf enum value: PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY = 35;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY"] = 35] = "PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY";
    /**
     * Cannot set call_conversion_action while call_conversion_tracking_enabled
     * is set to true.
     *
     * @generated from protobuf enum value: CONFLICTING_CALL_CONVERSION_SETTINGS = 36;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CONFLICTING_CALL_CONVERSION_SETTINGS"] = 36] = "CONFLICTING_CALL_CONVERSION_SETTINGS";
    /**
     * The type of the input extension feed item doesn't match the existing
     * extension feed item.
     *
     * @generated from protobuf enum value: EXTENSION_TYPE_MISMATCH = 37;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["EXTENSION_TYPE_MISMATCH"] = 37] = "EXTENSION_TYPE_MISMATCH";
    /**
     * The oneof field extension for example, subtype of extension feed item is
     * required.
     *
     * @generated from protobuf enum value: EXTENSION_SUBTYPE_REQUIRED = 38;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["EXTENSION_SUBTYPE_REQUIRED"] = 38] = "EXTENSION_SUBTYPE_REQUIRED";
    /**
     * The referenced feed item is not mapped to a supported extension type.
     *
     * @generated from protobuf enum value: EXTENSION_TYPE_UNSUPPORTED = 39;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["EXTENSION_TYPE_UNSUPPORTED"] = 39] = "EXTENSION_TYPE_UNSUPPORTED";
    /**
     * Cannot operate on a Feed with more than one active FeedMapping.
     *
     * @generated from protobuf enum value: CANNOT_OPERATE_ON_FEED_WITH_MULTIPLE_MAPPINGS = 40;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CANNOT_OPERATE_ON_FEED_WITH_MULTIPLE_MAPPINGS"] = 40] = "CANNOT_OPERATE_ON_FEED_WITH_MULTIPLE_MAPPINGS";
    /**
     * Cannot operate on a Feed that has key attributes.
     *
     * @generated from protobuf enum value: CANNOT_OPERATE_ON_FEED_WITH_KEY_ATTRIBUTES = 41;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CANNOT_OPERATE_ON_FEED_WITH_KEY_ATTRIBUTES"] = 41] = "CANNOT_OPERATE_ON_FEED_WITH_KEY_ATTRIBUTES";
    /**
     * Input price is not in a valid format.
     *
     * @generated from protobuf enum value: INVALID_PRICE_FORMAT = 42;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["INVALID_PRICE_FORMAT"] = 42] = "INVALID_PRICE_FORMAT";
    /**
     * The promotion time is invalid.
     *
     * @generated from protobuf enum value: PROMOTION_INVALID_TIME = 43;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["PROMOTION_INVALID_TIME"] = 43] = "PROMOTION_INVALID_TIME";
    /**
     * This field has too many decimal places specified.
     *
     * @generated from protobuf enum value: TOO_MANY_DECIMAL_PLACES_SPECIFIED = 44;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["TOO_MANY_DECIMAL_PLACES_SPECIFIED"] = 44] = "TOO_MANY_DECIMAL_PLACES_SPECIFIED";
    /**
     * Concrete sub type of ExtensionFeedItem is required for this operation.
     *
     * @generated from protobuf enum value: CONCRETE_EXTENSION_TYPE_REQUIRED = 45;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["CONCRETE_EXTENSION_TYPE_REQUIRED"] = 45] = "CONCRETE_EXTENSION_TYPE_REQUIRED";
    /**
     * Feed item schedule end time must be after start time.
     *
     * @generated from protobuf enum value: SCHEDULE_END_NOT_AFTER_START = 46;
     */
    ExtensionFeedItemErrorEnum_ExtensionFeedItemError[ExtensionFeedItemErrorEnum_ExtensionFeedItemError["SCHEDULE_END_NOT_AFTER_START"] = 46] = "SCHEDULE_END_NOT_AFTER_START";
})(ExtensionFeedItemErrorEnum_ExtensionFeedItemError = exports.ExtensionFeedItemErrorEnum_ExtensionFeedItemError || (exports.ExtensionFeedItemErrorEnum_ExtensionFeedItemError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ExtensionFeedItemErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ExtensionFeedItemErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ExtensionFeedItemErrorEnum
 */
exports.ExtensionFeedItemErrorEnum = new ExtensionFeedItemErrorEnum$Type();
