// google-ads-api-client/dist/protos/bindings/ads/errors/extension_setting_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionSettingErrorEnum = exports.ExtensionSettingErrorEnum_ExtensionSettingError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible extension setting errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ExtensionSettingErrorEnum.ExtensionSettingError
 */
var ExtensionSettingErrorEnum_ExtensionSettingError;
(function (ExtensionSettingErrorEnum_ExtensionSettingError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A platform restriction was provided without input extensions or existing
     * extensions.
     *
     * @generated from protobuf enum value: EXTENSIONS_REQUIRED = 2;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["EXTENSIONS_REQUIRED"] = 2] = "EXTENSIONS_REQUIRED";
    /**
     * The provided feed type does not correspond to the provided extensions.
     *
     * @generated from protobuf enum value: FEED_TYPE_EXTENSION_TYPE_MISMATCH = 3;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["FEED_TYPE_EXTENSION_TYPE_MISMATCH"] = 3] = "FEED_TYPE_EXTENSION_TYPE_MISMATCH";
    /**
     * The provided feed type cannot be used.
     *
     * @generated from protobuf enum value: INVALID_FEED_TYPE = 4;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_FEED_TYPE"] = 4] = "INVALID_FEED_TYPE";
    /**
     * The provided feed type cannot be used at the customer level.
     *
     * @generated from protobuf enum value: INVALID_FEED_TYPE_FOR_CUSTOMER_EXTENSION_SETTING = 5;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_FEED_TYPE_FOR_CUSTOMER_EXTENSION_SETTING"] = 5] = "INVALID_FEED_TYPE_FOR_CUSTOMER_EXTENSION_SETTING";
    /**
     * Cannot change a feed item field on a CREATE operation.
     *
     * @generated from protobuf enum value: CANNOT_CHANGE_FEED_ITEM_ON_CREATE = 6;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CANNOT_CHANGE_FEED_ITEM_ON_CREATE"] = 6] = "CANNOT_CHANGE_FEED_ITEM_ON_CREATE";
    /**
     * Cannot update an extension that is not already in this setting.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_NEWLY_CREATED_EXTENSION = 7;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CANNOT_UPDATE_NEWLY_CREATED_EXTENSION"] = 7] = "CANNOT_UPDATE_NEWLY_CREATED_EXTENSION";
    /**
     * There is no existing AdGroupExtensionSetting for this type.
     *
     * @generated from protobuf enum value: NO_EXISTING_AD_GROUP_EXTENSION_SETTING_FOR_TYPE = 8;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["NO_EXISTING_AD_GROUP_EXTENSION_SETTING_FOR_TYPE"] = 8] = "NO_EXISTING_AD_GROUP_EXTENSION_SETTING_FOR_TYPE";
    /**
     * There is no existing CampaignExtensionSetting for this type.
     *
     * @generated from protobuf enum value: NO_EXISTING_CAMPAIGN_EXTENSION_SETTING_FOR_TYPE = 9;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["NO_EXISTING_CAMPAIGN_EXTENSION_SETTING_FOR_TYPE"] = 9] = "NO_EXISTING_CAMPAIGN_EXTENSION_SETTING_FOR_TYPE";
    /**
     * There is no existing CustomerExtensionSetting for this type.
     *
     * @generated from protobuf enum value: NO_EXISTING_CUSTOMER_EXTENSION_SETTING_FOR_TYPE = 10;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["NO_EXISTING_CUSTOMER_EXTENSION_SETTING_FOR_TYPE"] = 10] = "NO_EXISTING_CUSTOMER_EXTENSION_SETTING_FOR_TYPE";
    /**
     * The AdGroupExtensionSetting already exists. UPDATE should be used to
     * modify the existing AdGroupExtensionSetting.
     *
     * @generated from protobuf enum value: AD_GROUP_EXTENSION_SETTING_ALREADY_EXISTS = 11;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["AD_GROUP_EXTENSION_SETTING_ALREADY_EXISTS"] = 11] = "AD_GROUP_EXTENSION_SETTING_ALREADY_EXISTS";
    /**
     * The CampaignExtensionSetting already exists. UPDATE should be used to
     * modify the existing CampaignExtensionSetting.
     *
     * @generated from protobuf enum value: CAMPAIGN_EXTENSION_SETTING_ALREADY_EXISTS = 12;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CAMPAIGN_EXTENSION_SETTING_ALREADY_EXISTS"] = 12] = "CAMPAIGN_EXTENSION_SETTING_ALREADY_EXISTS";
    /**
     * The CustomerExtensionSetting already exists. UPDATE should be used to
     * modify the existing CustomerExtensionSetting.
     *
     * @generated from protobuf enum value: CUSTOMER_EXTENSION_SETTING_ALREADY_EXISTS = 13;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CUSTOMER_EXTENSION_SETTING_ALREADY_EXISTS"] = 13] = "CUSTOMER_EXTENSION_SETTING_ALREADY_EXISTS";
    /**
     * An active ad group feed already exists for this place holder type.
     *
     * @generated from protobuf enum value: AD_GROUP_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE = 14;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["AD_GROUP_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE"] = 14] = "AD_GROUP_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE";
    /**
     * An active campaign feed already exists for this place holder type.
     *
     * @generated from protobuf enum value: CAMPAIGN_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE = 15;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CAMPAIGN_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE"] = 15] = "CAMPAIGN_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE";
    /**
     * An active customer feed already exists for this place holder type.
     *
     * @generated from protobuf enum value: CUSTOMER_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE = 16;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CUSTOMER_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE"] = 16] = "CUSTOMER_FEED_ALREADY_EXISTS_FOR_PLACEHOLDER_TYPE";
    /**
     * Value is not within the accepted range.
     *
     * @generated from protobuf enum value: VALUE_OUT_OF_RANGE = 17;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["VALUE_OUT_OF_RANGE"] = 17] = "VALUE_OUT_OF_RANGE";
    /**
     * Cannot simultaneously set specified field with final urls.
     *
     * @generated from protobuf enum value: CANNOT_SET_FIELD_WITH_FINAL_URLS = 18;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CANNOT_SET_FIELD_WITH_FINAL_URLS"] = 18] = "CANNOT_SET_FIELD_WITH_FINAL_URLS";
    /**
     * Must set field with final urls.
     *
     * @generated from protobuf enum value: FINAL_URLS_NOT_SET = 19;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["FINAL_URLS_NOT_SET"] = 19] = "FINAL_URLS_NOT_SET";
    /**
     * Phone number for a call extension is invalid.
     *
     * @generated from protobuf enum value: INVALID_PHONE_NUMBER = 20;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_PHONE_NUMBER"] = 20] = "INVALID_PHONE_NUMBER";
    /**
     * Phone number for a call extension is not supported for the given country
     * code.
     *
     * @generated from protobuf enum value: PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY = 21;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY"] = 21] = "PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * A carrier specific number in short format is not allowed for call
     * extensions.
     *
     * @generated from protobuf enum value: CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED = 22;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED"] = 22] = "CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED";
    /**
     * Premium rate numbers are not allowed for call extensions.
     *
     * @generated from protobuf enum value: PREMIUM_RATE_NUMBER_NOT_ALLOWED = 23;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PREMIUM_RATE_NUMBER_NOT_ALLOWED"] = 23] = "PREMIUM_RATE_NUMBER_NOT_ALLOWED";
    /**
     * Phone number type for a call extension is not allowed.
     *
     * @generated from protobuf enum value: DISALLOWED_NUMBER_TYPE = 24;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["DISALLOWED_NUMBER_TYPE"] = 24] = "DISALLOWED_NUMBER_TYPE";
    /**
     * Phone number for a call extension does not meet domestic format
     * requirements.
     *
     * @generated from protobuf enum value: INVALID_DOMESTIC_PHONE_NUMBER_FORMAT = 25;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_DOMESTIC_PHONE_NUMBER_FORMAT"] = 25] = "INVALID_DOMESTIC_PHONE_NUMBER_FORMAT";
    /**
     * Vanity phone numbers (for example, those including letters) are not
     * allowed for call extensions.
     *
     * @generated from protobuf enum value: VANITY_PHONE_NUMBER_NOT_ALLOWED = 26;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["VANITY_PHONE_NUMBER_NOT_ALLOWED"] = 26] = "VANITY_PHONE_NUMBER_NOT_ALLOWED";
    /**
     * Country code provided for a call extension is invalid.
     *
     * @generated from protobuf enum value: INVALID_COUNTRY_CODE = 27;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_COUNTRY_CODE"] = 27] = "INVALID_COUNTRY_CODE";
    /**
     * Call conversion type id provided for a call extension is invalid.
     *
     * @generated from protobuf enum value: INVALID_CALL_CONVERSION_TYPE_ID = 28;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_CALL_CONVERSION_TYPE_ID"] = 28] = "INVALID_CALL_CONVERSION_TYPE_ID";
    /**
     * For a call extension, the customer is not on the allow-list for call
     * tracking.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_IN_ALLOWLIST_FOR_CALLTRACKING = 69;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CUSTOMER_NOT_IN_ALLOWLIST_FOR_CALLTRACKING"] = 69] = "CUSTOMER_NOT_IN_ALLOWLIST_FOR_CALLTRACKING";
    /**
     * Call tracking is not supported for the given country for a call
     * extension.
     *
     * @generated from protobuf enum value: CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY = 30;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY"] = 30] = "CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * App id provided for an app extension is invalid.
     *
     * @generated from protobuf enum value: INVALID_APP_ID = 31;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_APP_ID"] = 31] = "INVALID_APP_ID";
    /**
     * Quotation marks present in the review text for a review extension.
     *
     * @generated from protobuf enum value: QUOTES_IN_REVIEW_EXTENSION_SNIPPET = 32;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["QUOTES_IN_REVIEW_EXTENSION_SNIPPET"] = 32] = "QUOTES_IN_REVIEW_EXTENSION_SNIPPET";
    /**
     * Hyphen character present in the review text for a review extension.
     *
     * @generated from protobuf enum value: HYPHENS_IN_REVIEW_EXTENSION_SNIPPET = 33;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["HYPHENS_IN_REVIEW_EXTENSION_SNIPPET"] = 33] = "HYPHENS_IN_REVIEW_EXTENSION_SNIPPET";
    /**
     * A blocked review source name or url was provided for a review
     * extension.
     *
     * @generated from protobuf enum value: REVIEW_EXTENSION_SOURCE_NOT_ELIGIBLE = 34;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["REVIEW_EXTENSION_SOURCE_NOT_ELIGIBLE"] = 34] = "REVIEW_EXTENSION_SOURCE_NOT_ELIGIBLE";
    /**
     * Review source name should not be found in the review text.
     *
     * @generated from protobuf enum value: SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT = 35;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT"] = 35] = "SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT";
    /**
     * Field must be set.
     *
     * @generated from protobuf enum value: MISSING_FIELD = 36;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["MISSING_FIELD"] = 36] = "MISSING_FIELD";
    /**
     * Inconsistent currency codes.
     *
     * @generated from protobuf enum value: INCONSISTENT_CURRENCY_CODES = 37;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INCONSISTENT_CURRENCY_CODES"] = 37] = "INCONSISTENT_CURRENCY_CODES";
    /**
     * Price extension cannot have duplicated headers.
     *
     * @generated from protobuf enum value: PRICE_EXTENSION_HAS_DUPLICATED_HEADERS = 38;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PRICE_EXTENSION_HAS_DUPLICATED_HEADERS"] = 38] = "PRICE_EXTENSION_HAS_DUPLICATED_HEADERS";
    /**
     * Price item cannot have duplicated header and description.
     *
     * @generated from protobuf enum value: PRICE_ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION = 39;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PRICE_ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION"] = 39] = "PRICE_ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION";
    /**
     * Price extension has too few items
     *
     * @generated from protobuf enum value: PRICE_EXTENSION_HAS_TOO_FEW_ITEMS = 40;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PRICE_EXTENSION_HAS_TOO_FEW_ITEMS"] = 40] = "PRICE_EXTENSION_HAS_TOO_FEW_ITEMS";
    /**
     * Price extension has too many items
     *
     * @generated from protobuf enum value: PRICE_EXTENSION_HAS_TOO_MANY_ITEMS = 41;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PRICE_EXTENSION_HAS_TOO_MANY_ITEMS"] = 41] = "PRICE_EXTENSION_HAS_TOO_MANY_ITEMS";
    /**
     * The input value is not currently supported.
     *
     * @generated from protobuf enum value: UNSUPPORTED_VALUE = 42;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["UNSUPPORTED_VALUE"] = 42] = "UNSUPPORTED_VALUE";
    /**
     * Unknown or unsupported device preference.
     *
     * @generated from protobuf enum value: INVALID_DEVICE_PREFERENCE = 43;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_DEVICE_PREFERENCE"] = 43] = "INVALID_DEVICE_PREFERENCE";
    /**
     * Invalid feed item schedule end time (for example, endHour = 24 and
     * endMinute != 0).
     *
     * @generated from protobuf enum value: INVALID_SCHEDULE_END = 45;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_SCHEDULE_END"] = 45] = "INVALID_SCHEDULE_END";
    /**
     * Date time zone does not match the account's time zone.
     *
     * @generated from protobuf enum value: DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE = 47;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE"] = 47] = "DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE";
    /**
     * Overlapping feed item schedule times (for example, 7-10AM and 8-11AM) are
     * not allowed.
     *
     * @generated from protobuf enum value: OVERLAPPING_SCHEDULES_NOT_ALLOWED = 48;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["OVERLAPPING_SCHEDULES_NOT_ALLOWED"] = 48] = "OVERLAPPING_SCHEDULES_NOT_ALLOWED";
    /**
     * Feed item schedule end time must be after start time.
     *
     * @generated from protobuf enum value: SCHEDULE_END_NOT_AFTER_START = 49;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["SCHEDULE_END_NOT_AFTER_START"] = 49] = "SCHEDULE_END_NOT_AFTER_START";
    /**
     * There are too many feed item schedules per day.
     *
     * @generated from protobuf enum value: TOO_MANY_SCHEDULES_PER_DAY = 50;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["TOO_MANY_SCHEDULES_PER_DAY"] = 50] = "TOO_MANY_SCHEDULES_PER_DAY";
    /**
     * Cannot edit the same extension feed item more than once in the same
     * request.
     *
     * @generated from protobuf enum value: DUPLICATE_EXTENSION_FEED_ITEM_EDIT = 51;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["DUPLICATE_EXTENSION_FEED_ITEM_EDIT"] = 51] = "DUPLICATE_EXTENSION_FEED_ITEM_EDIT";
    /**
     * Invalid structured snippet header.
     *
     * @generated from protobuf enum value: INVALID_SNIPPETS_HEADER = 52;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_SNIPPETS_HEADER"] = 52] = "INVALID_SNIPPETS_HEADER";
    /**
     * Phone number with call tracking enabled is not supported for the
     * specified country.
     *
     * @generated from protobuf enum value: PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY = 53;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY"] = 53] = "PHONE_NUMBER_NOT_SUPPORTED_WITH_CALLTRACKING_FOR_COUNTRY";
    /**
     * The targeted adgroup must belong to the targeted campaign.
     *
     * @generated from protobuf enum value: CAMPAIGN_TARGETING_MISMATCH = 54;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CAMPAIGN_TARGETING_MISMATCH"] = 54] = "CAMPAIGN_TARGETING_MISMATCH";
    /**
     * The feed used by the ExtensionSetting is removed and cannot be operated
     * on. Remove the ExtensionSetting to allow a new one to be created using
     * an active feed.
     *
     * @generated from protobuf enum value: CANNOT_OPERATE_ON_REMOVED_FEED = 55;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CANNOT_OPERATE_ON_REMOVED_FEED"] = 55] = "CANNOT_OPERATE_ON_REMOVED_FEED";
    /**
     * The ExtensionFeedItem type is required for this operation.
     *
     * @generated from protobuf enum value: EXTENSION_TYPE_REQUIRED = 56;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["EXTENSION_TYPE_REQUIRED"] = 56] = "EXTENSION_TYPE_REQUIRED";
    /**
     * The matching function that links the extension feed to the customer,
     * campaign, or ad group is not compatible with the ExtensionSetting
     * services.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_UNDERLYING_MATCHING_FUNCTION = 57;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INCOMPATIBLE_UNDERLYING_MATCHING_FUNCTION"] = 57] = "INCOMPATIBLE_UNDERLYING_MATCHING_FUNCTION";
    /**
     * Start date must be before end date.
     *
     * @generated from protobuf enum value: START_DATE_AFTER_END_DATE = 58;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["START_DATE_AFTER_END_DATE"] = 58] = "START_DATE_AFTER_END_DATE";
    /**
     * Input price is not in a valid format.
     *
     * @generated from protobuf enum value: INVALID_PRICE_FORMAT = 59;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_PRICE_FORMAT"] = 59] = "INVALID_PRICE_FORMAT";
    /**
     * The promotion time is invalid.
     *
     * @generated from protobuf enum value: PROMOTION_INVALID_TIME = 60;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PROMOTION_INVALID_TIME"] = 60] = "PROMOTION_INVALID_TIME";
    /**
     * Cannot set both percent discount and money discount fields.
     *
     * @generated from protobuf enum value: PROMOTION_CANNOT_SET_PERCENT_DISCOUNT_AND_MONEY_DISCOUNT = 61;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PROMOTION_CANNOT_SET_PERCENT_DISCOUNT_AND_MONEY_DISCOUNT"] = 61] = "PROMOTION_CANNOT_SET_PERCENT_DISCOUNT_AND_MONEY_DISCOUNT";
    /**
     * Cannot set both promotion code and orders over amount fields.
     *
     * @generated from protobuf enum value: PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT = 62;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT"] = 62] = "PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT";
    /**
     * This field has too many decimal places specified.
     *
     * @generated from protobuf enum value: TOO_MANY_DECIMAL_PLACES_SPECIFIED = 63;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["TOO_MANY_DECIMAL_PLACES_SPECIFIED"] = 63] = "TOO_MANY_DECIMAL_PLACES_SPECIFIED";
    /**
     * The language code is not valid.
     *
     * @generated from protobuf enum value: INVALID_LANGUAGE_CODE = 64;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["INVALID_LANGUAGE_CODE"] = 64] = "INVALID_LANGUAGE_CODE";
    /**
     * The language is not supported.
     *
     * @generated from protobuf enum value: UNSUPPORTED_LANGUAGE = 65;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["UNSUPPORTED_LANGUAGE"] = 65] = "UNSUPPORTED_LANGUAGE";
    /**
     * Customer hasn't consented for call recording, which is required for
     * adding/updating call extensions. See
     * https://support.google.com/google-ads/answer/7412639.
     *
     * @generated from protobuf enum value: CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED = 66;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED"] = 66] = "CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED";
    /**
     * The UPDATE operation does not specify any fields other than the resource
     * name in the update mask.
     *
     * @generated from protobuf enum value: EXTENSION_SETTING_UPDATE_IS_A_NOOP = 67;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["EXTENSION_SETTING_UPDATE_IS_A_NOOP"] = 67] = "EXTENSION_SETTING_UPDATE_IS_A_NOOP";
    /**
     * The extension contains text which has been prohibited on policy grounds.
     *
     * @generated from protobuf enum value: DISALLOWED_TEXT = 68;
     */
    ExtensionSettingErrorEnum_ExtensionSettingError[ExtensionSettingErrorEnum_ExtensionSettingError["DISALLOWED_TEXT"] = 68] = "DISALLOWED_TEXT";
})(ExtensionSettingErrorEnum_ExtensionSettingError = exports.ExtensionSettingErrorEnum_ExtensionSettingError || (exports.ExtensionSettingErrorEnum_ExtensionSettingError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ExtensionSettingErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ExtensionSettingErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ExtensionSettingErrorEnum
 */
exports.ExtensionSettingErrorEnum = new ExtensionSettingErrorEnum$Type();
