// google-ads-api-client/dist/protos/bindings/ads/errors/feed_item_set_link_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemSetLinkErrorEnum = exports.FeedItemSetLinkErrorEnum_FeedItemSetLinkError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible feed item set link errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FeedItemSetLinkErrorEnum.FeedItemSetLinkError
 */
var FeedItemSetLinkErrorEnum_FeedItemSetLinkError;
(function (FeedItemSetLinkErrorEnum_FeedItemSetLinkError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemSetLinkErrorEnum_FeedItemSetLinkError[FeedItemSetLinkErrorEnum_FeedItemSetLinkError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemSetLinkErrorEnum_FeedItemSetLinkError[FeedItemSetLinkErrorEnum_FeedItemSetLinkError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The feed IDs of the FeedItemSet and FeedItem do not match. Only FeedItems
     * in a given Feed can be linked to a FeedItemSet in that Feed.
     *
     * @generated from protobuf enum value: FEED_ID_MISMATCH = 2;
     */
    FeedItemSetLinkErrorEnum_FeedItemSetLinkError[FeedItemSetLinkErrorEnum_FeedItemSetLinkError["FEED_ID_MISMATCH"] = 2] = "FEED_ID_MISMATCH";
    /**
     * Cannot add or remove links to a dynamic set.
     *
     * @generated from protobuf enum value: NO_MUTATE_ALLOWED_FOR_DYNAMIC_SET = 3;
     */
    FeedItemSetLinkErrorEnum_FeedItemSetLinkError[FeedItemSetLinkErrorEnum_FeedItemSetLinkError["NO_MUTATE_ALLOWED_FOR_DYNAMIC_SET"] = 3] = "NO_MUTATE_ALLOWED_FOR_DYNAMIC_SET";
})(FeedItemSetLinkErrorEnum_FeedItemSetLinkError = exports.FeedItemSetLinkErrorEnum_FeedItemSetLinkError || (exports.FeedItemSetLinkErrorEnum_FeedItemSetLinkError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemSetLinkErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FeedItemSetLinkErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FeedItemSetLinkErrorEnum
 */
exports.FeedItemSetLinkErrorEnum = new FeedItemSetLinkErrorEnum$Type();
