// google-ads-api-client/dist/protos/bindings/ads/errors/feed_item_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemErrorEnum = exports.FeedItemErrorEnum_FeedItemError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible feed item errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FeedItemErrorEnum.FeedItemError
 */
var FeedItemErrorEnum_FeedItemError;
(function (FeedItemErrorEnum_FeedItemError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Cannot convert the feed attribute value from string to its real type.
     *
     * @generated from protobuf enum value: CANNOT_CONVERT_ATTRIBUTE_VALUE_FROM_STRING = 2;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["CANNOT_CONVERT_ATTRIBUTE_VALUE_FROM_STRING"] = 2] = "CANNOT_CONVERT_ATTRIBUTE_VALUE_FROM_STRING";
    /**
     * Cannot operate on removed feed item.
     *
     * @generated from protobuf enum value: CANNOT_OPERATE_ON_REMOVED_FEED_ITEM = 3;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["CANNOT_OPERATE_ON_REMOVED_FEED_ITEM"] = 3] = "CANNOT_OPERATE_ON_REMOVED_FEED_ITEM";
    /**
     * Date time zone does not match the account's time zone.
     *
     * @generated from protobuf enum value: DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE = 4;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE"] = 4] = "DATE_TIME_MUST_BE_IN_ACCOUNT_TIME_ZONE";
    /**
     * Feed item with the key attributes could not be found.
     *
     * @generated from protobuf enum value: KEY_ATTRIBUTES_NOT_FOUND = 5;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["KEY_ATTRIBUTES_NOT_FOUND"] = 5] = "KEY_ATTRIBUTES_NOT_FOUND";
    /**
     * Url feed attribute value is not valid.
     *
     * @generated from protobuf enum value: INVALID_URL = 6;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["INVALID_URL"] = 6] = "INVALID_URL";
    /**
     * Some key attributes are missing.
     *
     * @generated from protobuf enum value: MISSING_KEY_ATTRIBUTES = 7;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["MISSING_KEY_ATTRIBUTES"] = 7] = "MISSING_KEY_ATTRIBUTES";
    /**
     * Feed item has same key attributes as another feed item.
     *
     * @generated from protobuf enum value: KEY_ATTRIBUTES_NOT_UNIQUE = 8;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["KEY_ATTRIBUTES_NOT_UNIQUE"] = 8] = "KEY_ATTRIBUTES_NOT_UNIQUE";
    /**
     * Cannot modify key attributes on an existing feed item.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_KEY_ATTRIBUTE_VALUE = 9;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["CANNOT_MODIFY_KEY_ATTRIBUTE_VALUE"] = 9] = "CANNOT_MODIFY_KEY_ATTRIBUTE_VALUE";
    /**
     * The feed attribute value is too large.
     *
     * @generated from protobuf enum value: SIZE_TOO_LARGE_FOR_MULTI_VALUE_ATTRIBUTE = 10;
     */
    FeedItemErrorEnum_FeedItemError[FeedItemErrorEnum_FeedItemError["SIZE_TOO_LARGE_FOR_MULTI_VALUE_ATTRIBUTE"] = 10] = "SIZE_TOO_LARGE_FOR_MULTI_VALUE_ATTRIBUTE";
})(FeedItemErrorEnum_FeedItemError = exports.FeedItemErrorEnum_FeedItemError || (exports.FeedItemErrorEnum_FeedItemError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FeedItemErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FeedItemErrorEnum
 */
exports.FeedItemErrorEnum = new FeedItemErrorEnum$Type();
