// google-ads-api-client/dist/protos/bindings/ads/errors/feed_item_set_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemSetErrorEnum = exports.FeedItemSetErrorEnum_FeedItemSetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible feed item set errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FeedItemSetErrorEnum.FeedItemSetError
 */
var FeedItemSetErrorEnum_FeedItemSetError;
(function (FeedItemSetErrorEnum_FeedItemSetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The given ID refers to a removed FeedItemSet.
     *
     * @generated from protobuf enum value: FEED_ITEM_SET_REMOVED = 2;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["FEED_ITEM_SET_REMOVED"] = 2] = "FEED_ITEM_SET_REMOVED";
    /**
     * The dynamic filter of a feed item set cannot be cleared on UPDATE if it
     * exists. A set is either static or dynamic once added, and that cannot
     * change.
     *
     * @generated from protobuf enum value: CANNOT_CLEAR_DYNAMIC_FILTER = 3;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["CANNOT_CLEAR_DYNAMIC_FILTER"] = 3] = "CANNOT_CLEAR_DYNAMIC_FILTER";
    /**
     * The dynamic filter of a feed item set cannot be created on UPDATE if it
     * does not exist. A set is either static or dynamic once added, and that
     * cannot change.
     *
     * @generated from protobuf enum value: CANNOT_CREATE_DYNAMIC_FILTER = 4;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["CANNOT_CREATE_DYNAMIC_FILTER"] = 4] = "CANNOT_CREATE_DYNAMIC_FILTER";
    /**
     * FeedItemSets can only be made for location or affiliate location feeds.
     *
     * @generated from protobuf enum value: INVALID_FEED_TYPE = 5;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["INVALID_FEED_TYPE"] = 5] = "INVALID_FEED_TYPE";
    /**
     * FeedItemSets duplicate name. Name should be unique within an account.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 6;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["DUPLICATE_NAME"] = 6] = "DUPLICATE_NAME";
    /**
     * The feed type of the parent Feed is not compatible with the type of
     * dynamic filter being set. For example, you can only set
     * dynamic_location_set_filter for LOCATION feed item sets.
     *
     * @generated from protobuf enum value: WRONG_DYNAMIC_FILTER_FOR_FEED_TYPE = 7;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["WRONG_DYNAMIC_FILTER_FOR_FEED_TYPE"] = 7] = "WRONG_DYNAMIC_FILTER_FOR_FEED_TYPE";
    /**
     * Chain ID specified for AffiliateLocationFeedData is invalid.
     *
     * @generated from protobuf enum value: DYNAMIC_FILTER_INVALID_CHAIN_IDS = 8;
     */
    FeedItemSetErrorEnum_FeedItemSetError[FeedItemSetErrorEnum_FeedItemSetError["DYNAMIC_FILTER_INVALID_CHAIN_IDS"] = 8] = "DYNAMIC_FILTER_INVALID_CHAIN_IDS";
})(FeedItemSetErrorEnum_FeedItemSetError = exports.FeedItemSetErrorEnum_FeedItemSetError || (exports.FeedItemSetErrorEnum_FeedItemSetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemSetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FeedItemSetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FeedItemSetErrorEnum
 */
exports.FeedItemSetErrorEnum = new FeedItemSetErrorEnum$Type();
