// google-ads-api-client/dist/protos/bindings/ads/errors/feed_item_validation_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemValidationErrorEnum = exports.FeedItemValidationErrorEnum_FeedItemValidationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * The possible validation errors of a feed item.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FeedItemValidationErrorEnum.FeedItemValidationError
 */
var FeedItemValidationErrorEnum_FeedItemValidationError;
(function (FeedItemValidationErrorEnum_FeedItemValidationError) {
    /**
     * No value has been specified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * String is too short.
     *
     * @generated from protobuf enum value: STRING_TOO_SHORT = 2;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["STRING_TOO_SHORT"] = 2] = "STRING_TOO_SHORT";
    /**
     * String is too long.
     *
     * @generated from protobuf enum value: STRING_TOO_LONG = 3;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["STRING_TOO_LONG"] = 3] = "STRING_TOO_LONG";
    /**
     * Value is not provided.
     *
     * @generated from protobuf enum value: VALUE_NOT_SPECIFIED = 4;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["VALUE_NOT_SPECIFIED"] = 4] = "VALUE_NOT_SPECIFIED";
    /**
     * Phone number format is invalid for region.
     *
     * @generated from protobuf enum value: INVALID_DOMESTIC_PHONE_NUMBER_FORMAT = 5;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_DOMESTIC_PHONE_NUMBER_FORMAT"] = 5] = "INVALID_DOMESTIC_PHONE_NUMBER_FORMAT";
    /**
     * String does not represent a phone number.
     *
     * @generated from protobuf enum value: INVALID_PHONE_NUMBER = 6;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_PHONE_NUMBER"] = 6] = "INVALID_PHONE_NUMBER";
    /**
     * Phone number format is not compatible with country code.
     *
     * @generated from protobuf enum value: PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY = 7;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY"] = 7] = "PHONE_NUMBER_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * Premium rate number is not allowed.
     *
     * @generated from protobuf enum value: PREMIUM_RATE_NUMBER_NOT_ALLOWED = 8;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["PREMIUM_RATE_NUMBER_NOT_ALLOWED"] = 8] = "PREMIUM_RATE_NUMBER_NOT_ALLOWED";
    /**
     * Phone number type is not allowed.
     *
     * @generated from protobuf enum value: DISALLOWED_NUMBER_TYPE = 9;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["DISALLOWED_NUMBER_TYPE"] = 9] = "DISALLOWED_NUMBER_TYPE";
    /**
     * Specified value is outside of the valid range.
     *
     * @generated from protobuf enum value: VALUE_OUT_OF_RANGE = 10;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["VALUE_OUT_OF_RANGE"] = 10] = "VALUE_OUT_OF_RANGE";
    /**
     * Call tracking is not supported in the selected country.
     *
     * @generated from protobuf enum value: CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY = 11;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY"] = 11] = "CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * Customer is not on the allow-list for call tracking.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_IN_ALLOWLIST_FOR_CALLTRACKING = 99;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["CUSTOMER_NOT_IN_ALLOWLIST_FOR_CALLTRACKING"] = 99] = "CUSTOMER_NOT_IN_ALLOWLIST_FOR_CALLTRACKING";
    /**
     * Country code is invalid.
     *
     * @generated from protobuf enum value: INVALID_COUNTRY_CODE = 13;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_COUNTRY_CODE"] = 13] = "INVALID_COUNTRY_CODE";
    /**
     * The specified mobile app id is invalid.
     *
     * @generated from protobuf enum value: INVALID_APP_ID = 14;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_APP_ID"] = 14] = "INVALID_APP_ID";
    /**
     * Some required field attributes are missing.
     *
     * @generated from protobuf enum value: MISSING_ATTRIBUTES_FOR_FIELDS = 15;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_ATTRIBUTES_FOR_FIELDS"] = 15] = "MISSING_ATTRIBUTES_FOR_FIELDS";
    /**
     * Invalid email button type for email extension.
     *
     * @generated from protobuf enum value: INVALID_TYPE_ID = 16;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_TYPE_ID"] = 16] = "INVALID_TYPE_ID";
    /**
     * Email address is invalid.
     *
     * @generated from protobuf enum value: INVALID_EMAIL_ADDRESS = 17;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_EMAIL_ADDRESS"] = 17] = "INVALID_EMAIL_ADDRESS";
    /**
     * The HTTPS URL in email extension is invalid.
     *
     * @generated from protobuf enum value: INVALID_HTTPS_URL = 18;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_HTTPS_URL"] = 18] = "INVALID_HTTPS_URL";
    /**
     * Delivery address is missing from email extension.
     *
     * @generated from protobuf enum value: MISSING_DELIVERY_ADDRESS = 19;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_DELIVERY_ADDRESS"] = 19] = "MISSING_DELIVERY_ADDRESS";
    /**
     * FeedItem scheduling start date comes after end date.
     *
     * @generated from protobuf enum value: START_DATE_AFTER_END_DATE = 20;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["START_DATE_AFTER_END_DATE"] = 20] = "START_DATE_AFTER_END_DATE";
    /**
     * FeedItem scheduling start time is missing.
     *
     * @generated from protobuf enum value: MISSING_FEED_ITEM_START_TIME = 21;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_FEED_ITEM_START_TIME"] = 21] = "MISSING_FEED_ITEM_START_TIME";
    /**
     * FeedItem scheduling end time is missing.
     *
     * @generated from protobuf enum value: MISSING_FEED_ITEM_END_TIME = 22;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_FEED_ITEM_END_TIME"] = 22] = "MISSING_FEED_ITEM_END_TIME";
    /**
     * Cannot compute system attributes on a FeedItem that has no FeedItemId.
     *
     * @generated from protobuf enum value: MISSING_FEED_ITEM_ID = 23;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_FEED_ITEM_ID"] = 23] = "MISSING_FEED_ITEM_ID";
    /**
     * Call extension vanity phone numbers are not supported.
     *
     * @generated from protobuf enum value: VANITY_PHONE_NUMBER_NOT_ALLOWED = 24;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["VANITY_PHONE_NUMBER_NOT_ALLOWED"] = 24] = "VANITY_PHONE_NUMBER_NOT_ALLOWED";
    /**
     * Invalid review text.
     *
     * @generated from protobuf enum value: INVALID_REVIEW_EXTENSION_SNIPPET = 25;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_REVIEW_EXTENSION_SNIPPET"] = 25] = "INVALID_REVIEW_EXTENSION_SNIPPET";
    /**
     * Invalid format for numeric value in ad parameter.
     *
     * @generated from protobuf enum value: INVALID_NUMBER_FORMAT = 26;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_NUMBER_FORMAT"] = 26] = "INVALID_NUMBER_FORMAT";
    /**
     * Invalid format for date value in ad parameter.
     *
     * @generated from protobuf enum value: INVALID_DATE_FORMAT = 27;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_DATE_FORMAT"] = 27] = "INVALID_DATE_FORMAT";
    /**
     * Invalid format for price value in ad parameter.
     *
     * @generated from protobuf enum value: INVALID_PRICE_FORMAT = 28;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_PRICE_FORMAT"] = 28] = "INVALID_PRICE_FORMAT";
    /**
     * Unrecognized type given for value in ad parameter.
     *
     * @generated from protobuf enum value: UNKNOWN_PLACEHOLDER_FIELD = 29;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["UNKNOWN_PLACEHOLDER_FIELD"] = 29] = "UNKNOWN_PLACEHOLDER_FIELD";
    /**
     * Enhanced sitelinks must have both description lines specified.
     *
     * @generated from protobuf enum value: MISSING_ENHANCED_SITELINK_DESCRIPTION_LINE = 30;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_ENHANCED_SITELINK_DESCRIPTION_LINE"] = 30] = "MISSING_ENHANCED_SITELINK_DESCRIPTION_LINE";
    /**
     * Review source is ineligible.
     *
     * @generated from protobuf enum value: REVIEW_EXTENSION_SOURCE_INELIGIBLE = 31;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["REVIEW_EXTENSION_SOURCE_INELIGIBLE"] = 31] = "REVIEW_EXTENSION_SOURCE_INELIGIBLE";
    /**
     * Review text cannot contain hyphens or dashes.
     *
     * @generated from protobuf enum value: HYPHENS_IN_REVIEW_EXTENSION_SNIPPET = 32;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["HYPHENS_IN_REVIEW_EXTENSION_SNIPPET"] = 32] = "HYPHENS_IN_REVIEW_EXTENSION_SNIPPET";
    /**
     * Review text cannot contain double quote characters.
     *
     * @generated from protobuf enum value: DOUBLE_QUOTES_IN_REVIEW_EXTENSION_SNIPPET = 33;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["DOUBLE_QUOTES_IN_REVIEW_EXTENSION_SNIPPET"] = 33] = "DOUBLE_QUOTES_IN_REVIEW_EXTENSION_SNIPPET";
    /**
     * Review text cannot contain quote characters.
     *
     * @generated from protobuf enum value: QUOTES_IN_REVIEW_EXTENSION_SNIPPET = 34;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["QUOTES_IN_REVIEW_EXTENSION_SNIPPET"] = 34] = "QUOTES_IN_REVIEW_EXTENSION_SNIPPET";
    /**
     * Parameters are encoded in the wrong format.
     *
     * @generated from protobuf enum value: INVALID_FORM_ENCODED_PARAMS = 35;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_FORM_ENCODED_PARAMS"] = 35] = "INVALID_FORM_ENCODED_PARAMS";
    /**
     * URL parameter name must contain only letters, numbers, underscores, and
     * dashes.
     *
     * @generated from protobuf enum value: INVALID_URL_PARAMETER_NAME = 36;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_URL_PARAMETER_NAME"] = 36] = "INVALID_URL_PARAMETER_NAME";
    /**
     * Cannot find address location.
     *
     * @generated from protobuf enum value: NO_GEOCODING_RESULT = 37;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["NO_GEOCODING_RESULT"] = 37] = "NO_GEOCODING_RESULT";
    /**
     * Review extension text has source name.
     *
     * @generated from protobuf enum value: SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT = 38;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT"] = 38] = "SOURCE_NAME_IN_REVIEW_EXTENSION_TEXT";
    /**
     * Some phone numbers can be shorter than usual. Some of these short numbers
     * are carrier-specific, and we disallow those in ad extensions because they
     * will not be available to all users.
     *
     * @generated from protobuf enum value: CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED = 39;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED"] = 39] = "CARRIER_SPECIFIC_SHORT_NUMBER_NOT_ALLOWED";
    /**
     * Triggered when a request references a placeholder field id that does not
     * exist.
     *
     * @generated from protobuf enum value: INVALID_PLACEHOLDER_FIELD_ID = 40;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_PLACEHOLDER_FIELD_ID"] = 40] = "INVALID_PLACEHOLDER_FIELD_ID";
    /**
     * URL contains invalid ValueTrack tags or format.
     *
     * @generated from protobuf enum value: INVALID_URL_TAG = 41;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_URL_TAG"] = 41] = "INVALID_URL_TAG";
    /**
     * Provided list exceeds acceptable size.
     *
     * @generated from protobuf enum value: LIST_TOO_LONG = 42;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["LIST_TOO_LONG"] = 42] = "LIST_TOO_LONG";
    /**
     * Certain combinations of attributes aren't allowed to be specified in the
     * same feed item.
     *
     * @generated from protobuf enum value: INVALID_ATTRIBUTES_COMBINATION = 43;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_ATTRIBUTES_COMBINATION"] = 43] = "INVALID_ATTRIBUTES_COMBINATION";
    /**
     * An attribute has the same value repeatedly.
     *
     * @generated from protobuf enum value: DUPLICATE_VALUES = 44;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["DUPLICATE_VALUES"] = 44] = "DUPLICATE_VALUES";
    /**
     * Advertisers can link a conversion action with a phone number to indicate
     * that sufficiently long calls forwarded to that phone number should be
     * counted as conversions of the specified type.  This is an error message
     * indicating that the conversion action specified is invalid (for example,
     * the conversion action does not exist within the appropriate Google Ads
     * account, or it is a type of conversion not appropriate to phone call
     * conversions).
     *
     * @generated from protobuf enum value: INVALID_CALL_CONVERSION_ACTION_ID = 45;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_CALL_CONVERSION_ACTION_ID"] = 45] = "INVALID_CALL_CONVERSION_ACTION_ID";
    /**
     * Tracking template requires final url to be set.
     *
     * @generated from protobuf enum value: CANNOT_SET_WITHOUT_FINAL_URLS = 46;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["CANNOT_SET_WITHOUT_FINAL_URLS"] = 46] = "CANNOT_SET_WITHOUT_FINAL_URLS";
    /**
     * An app id was provided that doesn't exist in the given app store.
     *
     * @generated from protobuf enum value: APP_ID_DOESNT_EXIST_IN_APP_STORE = 47;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["APP_ID_DOESNT_EXIST_IN_APP_STORE"] = 47] = "APP_ID_DOESNT_EXIST_IN_APP_STORE";
    /**
     * Invalid U2 final url.
     *
     * @generated from protobuf enum value: INVALID_FINAL_URL = 48;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_FINAL_URL"] = 48] = "INVALID_FINAL_URL";
    /**
     * Invalid U2 tracking url.
     *
     * @generated from protobuf enum value: INVALID_TRACKING_URL = 49;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_TRACKING_URL"] = 49] = "INVALID_TRACKING_URL";
    /**
     * Final URL should start from App download URL.
     *
     * @generated from protobuf enum value: INVALID_FINAL_URL_FOR_APP_DOWNLOAD_URL = 50;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_FINAL_URL_FOR_APP_DOWNLOAD_URL"] = 50] = "INVALID_FINAL_URL_FOR_APP_DOWNLOAD_URL";
    /**
     * List provided is too short.
     *
     * @generated from protobuf enum value: LIST_TOO_SHORT = 51;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["LIST_TOO_SHORT"] = 51] = "LIST_TOO_SHORT";
    /**
     * User Action field has invalid value.
     *
     * @generated from protobuf enum value: INVALID_USER_ACTION = 52;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_USER_ACTION"] = 52] = "INVALID_USER_ACTION";
    /**
     * Type field has invalid value.
     *
     * @generated from protobuf enum value: INVALID_TYPE_NAME = 53;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_TYPE_NAME"] = 53] = "INVALID_TYPE_NAME";
    /**
     * Change status for event is invalid.
     *
     * @generated from protobuf enum value: INVALID_EVENT_CHANGE_STATUS = 54;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_EVENT_CHANGE_STATUS"] = 54] = "INVALID_EVENT_CHANGE_STATUS";
    /**
     * The header of a structured snippets extension is not one of the valid
     * headers.
     *
     * @generated from protobuf enum value: INVALID_SNIPPETS_HEADER = 55;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_SNIPPETS_HEADER"] = 55] = "INVALID_SNIPPETS_HEADER";
    /**
     * Android app link is not formatted correctly
     *
     * @generated from protobuf enum value: INVALID_ANDROID_APP_LINK = 56;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_ANDROID_APP_LINK"] = 56] = "INVALID_ANDROID_APP_LINK";
    /**
     * Phone number incompatible with call tracking for country.
     *
     * @generated from protobuf enum value: NUMBER_TYPE_WITH_CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY = 57;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["NUMBER_TYPE_WITH_CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY"] = 57] = "NUMBER_TYPE_WITH_CALLTRACKING_NOT_SUPPORTED_FOR_COUNTRY";
    /**
     * The input is identical to a reserved keyword
     *
     * @generated from protobuf enum value: RESERVED_KEYWORD_OTHER = 58;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["RESERVED_KEYWORD_OTHER"] = 58] = "RESERVED_KEYWORD_OTHER";
    /**
     * Each option label in the message extension must be unique.
     *
     * @generated from protobuf enum value: DUPLICATE_OPTION_LABELS = 59;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["DUPLICATE_OPTION_LABELS"] = 59] = "DUPLICATE_OPTION_LABELS";
    /**
     * Each option prefill in the message extension must be unique.
     *
     * @generated from protobuf enum value: DUPLICATE_OPTION_PREFILLS = 60;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["DUPLICATE_OPTION_PREFILLS"] = 60] = "DUPLICATE_OPTION_PREFILLS";
    /**
     * In message extensions, the number of optional labels and optional
     * prefills must be the same.
     *
     * @generated from protobuf enum value: UNEQUAL_LIST_LENGTHS = 61;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["UNEQUAL_LIST_LENGTHS"] = 61] = "UNEQUAL_LIST_LENGTHS";
    /**
     * All currency codes in an ad extension must be the same.
     *
     * @generated from protobuf enum value: INCONSISTENT_CURRENCY_CODES = 62;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INCONSISTENT_CURRENCY_CODES"] = 62] = "INCONSISTENT_CURRENCY_CODES";
    /**
     * Headers in price extension are not unique.
     *
     * @generated from protobuf enum value: PRICE_EXTENSION_HAS_DUPLICATED_HEADERS = 63;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["PRICE_EXTENSION_HAS_DUPLICATED_HEADERS"] = 63] = "PRICE_EXTENSION_HAS_DUPLICATED_HEADERS";
    /**
     * Header and description in an item are the same.
     *
     * @generated from protobuf enum value: ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION = 64;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION"] = 64] = "ITEM_HAS_DUPLICATED_HEADER_AND_DESCRIPTION";
    /**
     * Price extension has too few items.
     *
     * @generated from protobuf enum value: PRICE_EXTENSION_HAS_TOO_FEW_ITEMS = 65;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["PRICE_EXTENSION_HAS_TOO_FEW_ITEMS"] = 65] = "PRICE_EXTENSION_HAS_TOO_FEW_ITEMS";
    /**
     * The given value is not supported.
     *
     * @generated from protobuf enum value: UNSUPPORTED_VALUE = 66;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["UNSUPPORTED_VALUE"] = 66] = "UNSUPPORTED_VALUE";
    /**
     * Invalid final mobile url.
     *
     * @generated from protobuf enum value: INVALID_FINAL_MOBILE_URL = 67;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_FINAL_MOBILE_URL"] = 67] = "INVALID_FINAL_MOBILE_URL";
    /**
     * The given string value of Label contains invalid characters
     *
     * @generated from protobuf enum value: INVALID_KEYWORDLESS_AD_RULE_LABEL = 68;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_KEYWORDLESS_AD_RULE_LABEL"] = 68] = "INVALID_KEYWORDLESS_AD_RULE_LABEL";
    /**
     * The given URL contains value track parameters.
     *
     * @generated from protobuf enum value: VALUE_TRACK_PARAMETER_NOT_SUPPORTED = 69;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["VALUE_TRACK_PARAMETER_NOT_SUPPORTED"] = 69] = "VALUE_TRACK_PARAMETER_NOT_SUPPORTED";
    /**
     * The given value is not supported in the selected language of an
     * extension.
     *
     * @generated from protobuf enum value: UNSUPPORTED_VALUE_IN_SELECTED_LANGUAGE = 70;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["UNSUPPORTED_VALUE_IN_SELECTED_LANGUAGE"] = 70] = "UNSUPPORTED_VALUE_IN_SELECTED_LANGUAGE";
    /**
     * The iOS app link is not formatted correctly.
     *
     * @generated from protobuf enum value: INVALID_IOS_APP_LINK = 71;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_IOS_APP_LINK"] = 71] = "INVALID_IOS_APP_LINK";
    /**
     * iOS app link or iOS app store id is missing.
     *
     * @generated from protobuf enum value: MISSING_IOS_APP_LINK_OR_IOS_APP_STORE_ID = 72;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_IOS_APP_LINK_OR_IOS_APP_STORE_ID"] = 72] = "MISSING_IOS_APP_LINK_OR_IOS_APP_STORE_ID";
    /**
     * Promotion time is invalid.
     *
     * @generated from protobuf enum value: PROMOTION_INVALID_TIME = 73;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["PROMOTION_INVALID_TIME"] = 73] = "PROMOTION_INVALID_TIME";
    /**
     * Both the percent off and money amount off fields are set.
     *
     * @generated from protobuf enum value: PROMOTION_CANNOT_SET_PERCENT_OFF_AND_MONEY_AMOUNT_OFF = 74;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["PROMOTION_CANNOT_SET_PERCENT_OFF_AND_MONEY_AMOUNT_OFF"] = 74] = "PROMOTION_CANNOT_SET_PERCENT_OFF_AND_MONEY_AMOUNT_OFF";
    /**
     * Both the promotion code and orders over amount fields are set.
     *
     * @generated from protobuf enum value: PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT = 75;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT"] = 75] = "PROMOTION_CANNOT_SET_PROMOTION_CODE_AND_ORDERS_OVER_AMOUNT";
    /**
     * Too many decimal places are specified.
     *
     * @generated from protobuf enum value: TOO_MANY_DECIMAL_PLACES_SPECIFIED = 76;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["TOO_MANY_DECIMAL_PLACES_SPECIFIED"] = 76] = "TOO_MANY_DECIMAL_PLACES_SPECIFIED";
    /**
     * Ad Customizers are present and not allowed.
     *
     * @generated from protobuf enum value: AD_CUSTOMIZERS_NOT_ALLOWED = 77;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["AD_CUSTOMIZERS_NOT_ALLOWED"] = 77] = "AD_CUSTOMIZERS_NOT_ALLOWED";
    /**
     * Language code is not valid.
     *
     * @generated from protobuf enum value: INVALID_LANGUAGE_CODE = 78;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_LANGUAGE_CODE"] = 78] = "INVALID_LANGUAGE_CODE";
    /**
     * Language is not supported.
     *
     * @generated from protobuf enum value: UNSUPPORTED_LANGUAGE = 79;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["UNSUPPORTED_LANGUAGE"] = 79] = "UNSUPPORTED_LANGUAGE";
    /**
     * IF Function is present and not allowed.
     *
     * @generated from protobuf enum value: IF_FUNCTION_NOT_ALLOWED = 80;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["IF_FUNCTION_NOT_ALLOWED"] = 80] = "IF_FUNCTION_NOT_ALLOWED";
    /**
     * Final url suffix is not valid.
     *
     * @generated from protobuf enum value: INVALID_FINAL_URL_SUFFIX = 81;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_FINAL_URL_SUFFIX"] = 81] = "INVALID_FINAL_URL_SUFFIX";
    /**
     * Final url suffix contains an invalid tag.
     *
     * @generated from protobuf enum value: INVALID_TAG_IN_FINAL_URL_SUFFIX = 82;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_TAG_IN_FINAL_URL_SUFFIX"] = 82] = "INVALID_TAG_IN_FINAL_URL_SUFFIX";
    /**
     * Final url suffix is formatted incorrectly.
     *
     * @generated from protobuf enum value: INVALID_FINAL_URL_SUFFIX_FORMAT = 83;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_FINAL_URL_SUFFIX_FORMAT"] = 83] = "INVALID_FINAL_URL_SUFFIX_FORMAT";
    /**
     * Consent for call recording, which is required for the use of call
     * extensions, was not provided by the advertiser. See
     * https://support.google.com/google-ads/answer/7412639.
     *
     * @generated from protobuf enum value: CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED = 84;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED"] = 84] = "CUSTOMER_CONSENT_FOR_CALL_RECORDING_REQUIRED";
    /**
     * Multiple message delivery options are set.
     *
     * @generated from protobuf enum value: ONLY_ONE_DELIVERY_OPTION_IS_ALLOWED = 85;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["ONLY_ONE_DELIVERY_OPTION_IS_ALLOWED"] = 85] = "ONLY_ONE_DELIVERY_OPTION_IS_ALLOWED";
    /**
     * No message delivery option is set.
     *
     * @generated from protobuf enum value: NO_DELIVERY_OPTION_IS_SET = 86;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["NO_DELIVERY_OPTION_IS_SET"] = 86] = "NO_DELIVERY_OPTION_IS_SET";
    /**
     * String value of conversion reporting state field is not valid.
     *
     * @generated from protobuf enum value: INVALID_CONVERSION_REPORTING_STATE = 87;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_CONVERSION_REPORTING_STATE"] = 87] = "INVALID_CONVERSION_REPORTING_STATE";
    /**
     * Image size is not right.
     *
     * @generated from protobuf enum value: IMAGE_SIZE_WRONG = 88;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["IMAGE_SIZE_WRONG"] = 88] = "IMAGE_SIZE_WRONG";
    /**
     * Email delivery is not supported in the country specified in the country
     * code field.
     *
     * @generated from protobuf enum value: EMAIL_DELIVERY_NOT_AVAILABLE_IN_COUNTRY = 89;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["EMAIL_DELIVERY_NOT_AVAILABLE_IN_COUNTRY"] = 89] = "EMAIL_DELIVERY_NOT_AVAILABLE_IN_COUNTRY";
    /**
     * Auto reply is not supported in the country specified in the country code
     * field.
     *
     * @generated from protobuf enum value: AUTO_REPLY_NOT_AVAILABLE_IN_COUNTRY = 90;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["AUTO_REPLY_NOT_AVAILABLE_IN_COUNTRY"] = 90] = "AUTO_REPLY_NOT_AVAILABLE_IN_COUNTRY";
    /**
     * Invalid value specified for latitude.
     *
     * @generated from protobuf enum value: INVALID_LATITUDE_VALUE = 91;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_LATITUDE_VALUE"] = 91] = "INVALID_LATITUDE_VALUE";
    /**
     * Invalid value specified for longitude.
     *
     * @generated from protobuf enum value: INVALID_LONGITUDE_VALUE = 92;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_LONGITUDE_VALUE"] = 92] = "INVALID_LONGITUDE_VALUE";
    /**
     * Too many label fields provided.
     *
     * @generated from protobuf enum value: TOO_MANY_LABELS = 93;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["TOO_MANY_LABELS"] = 93] = "TOO_MANY_LABELS";
    /**
     * Invalid image url.
     *
     * @generated from protobuf enum value: INVALID_IMAGE_URL = 94;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_IMAGE_URL"] = 94] = "INVALID_IMAGE_URL";
    /**
     * Latitude value is missing.
     *
     * @generated from protobuf enum value: MISSING_LATITUDE_VALUE = 95;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_LATITUDE_VALUE"] = 95] = "MISSING_LATITUDE_VALUE";
    /**
     * Longitude value is missing.
     *
     * @generated from protobuf enum value: MISSING_LONGITUDE_VALUE = 96;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["MISSING_LONGITUDE_VALUE"] = 96] = "MISSING_LONGITUDE_VALUE";
    /**
     * Unable to find address.
     *
     * @generated from protobuf enum value: ADDRESS_NOT_FOUND = 97;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["ADDRESS_NOT_FOUND"] = 97] = "ADDRESS_NOT_FOUND";
    /**
     * Cannot target provided address.
     *
     * @generated from protobuf enum value: ADDRESS_NOT_TARGETABLE = 98;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["ADDRESS_NOT_TARGETABLE"] = 98] = "ADDRESS_NOT_TARGETABLE";
    /**
     * The specified asset ID does not exist.
     *
     * @generated from protobuf enum value: INVALID_ASSET_ID = 100;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INVALID_ASSET_ID"] = 100] = "INVALID_ASSET_ID";
    /**
     * The asset type cannot be set for the field.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_ASSET_TYPE = 101;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["INCOMPATIBLE_ASSET_TYPE"] = 101] = "INCOMPATIBLE_ASSET_TYPE";
    /**
     * The image has unexpected size.
     *
     * @generated from protobuf enum value: IMAGE_ERROR_UNEXPECTED_SIZE = 102;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["IMAGE_ERROR_UNEXPECTED_SIZE"] = 102] = "IMAGE_ERROR_UNEXPECTED_SIZE";
    /**
     * The image aspect ratio is not allowed.
     *
     * @generated from protobuf enum value: IMAGE_ERROR_ASPECT_RATIO_NOT_ALLOWED = 103;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["IMAGE_ERROR_ASPECT_RATIO_NOT_ALLOWED"] = 103] = "IMAGE_ERROR_ASPECT_RATIO_NOT_ALLOWED";
    /**
     * The image file is too large.
     *
     * @generated from protobuf enum value: IMAGE_ERROR_FILE_TOO_LARGE = 104;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["IMAGE_ERROR_FILE_TOO_LARGE"] = 104] = "IMAGE_ERROR_FILE_TOO_LARGE";
    /**
     * The image format is unsupported.
     *
     * @generated from protobuf enum value: IMAGE_ERROR_FORMAT_NOT_ALLOWED = 105;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["IMAGE_ERROR_FORMAT_NOT_ALLOWED"] = 105] = "IMAGE_ERROR_FORMAT_NOT_ALLOWED";
    /**
     * Image violates constraints without more details.
     *
     * @generated from protobuf enum value: IMAGE_ERROR_CONSTRAINTS_VIOLATED = 106;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["IMAGE_ERROR_CONSTRAINTS_VIOLATED"] = 106] = "IMAGE_ERROR_CONSTRAINTS_VIOLATED";
    /**
     * An error occurred when validating image.
     *
     * @generated from protobuf enum value: IMAGE_ERROR_SERVER_ERROR = 107;
     */
    FeedItemValidationErrorEnum_FeedItemValidationError[FeedItemValidationErrorEnum_FeedItemValidationError["IMAGE_ERROR_SERVER_ERROR"] = 107] = "IMAGE_ERROR_SERVER_ERROR";
})(FeedItemValidationErrorEnum_FeedItemValidationError = exports.FeedItemValidationErrorEnum_FeedItemValidationError || (exports.FeedItemValidationErrorEnum_FeedItemValidationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemValidationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FeedItemValidationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FeedItemValidationErrorEnum
 */
exports.FeedItemValidationErrorEnum = new FeedItemValidationErrorEnum$Type();
