// google-ads-api-client/dist/protos/bindings/ads/errors/feed_item_target_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemTargetErrorEnum = exports.FeedItemTargetErrorEnum_FeedItemTargetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible feed item target errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FeedItemTargetErrorEnum.FeedItemTargetError
 */
var FeedItemTargetErrorEnum_FeedItemTargetError;
(function (FeedItemTargetErrorEnum_FeedItemTargetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * On CREATE, the FeedItemTarget must have a populated field in the oneof
     * target.
     *
     * @generated from protobuf enum value: MUST_SET_TARGET_ONEOF_ON_CREATE = 2;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["MUST_SET_TARGET_ONEOF_ON_CREATE"] = 2] = "MUST_SET_TARGET_ONEOF_ON_CREATE";
    /**
     * The specified feed item target already exists, so it cannot be added.
     *
     * @generated from protobuf enum value: FEED_ITEM_TARGET_ALREADY_EXISTS = 3;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["FEED_ITEM_TARGET_ALREADY_EXISTS"] = 3] = "FEED_ITEM_TARGET_ALREADY_EXISTS";
    /**
     * The schedules for a given feed item cannot overlap.
     *
     * @generated from protobuf enum value: FEED_ITEM_SCHEDULES_CANNOT_OVERLAP = 4;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["FEED_ITEM_SCHEDULES_CANNOT_OVERLAP"] = 4] = "FEED_ITEM_SCHEDULES_CANNOT_OVERLAP";
    /**
     * Too many targets of a given type were added for a single feed item.
     *
     * @generated from protobuf enum value: TARGET_LIMIT_EXCEEDED_FOR_GIVEN_TYPE = 5;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["TARGET_LIMIT_EXCEEDED_FOR_GIVEN_TYPE"] = 5] = "TARGET_LIMIT_EXCEEDED_FOR_GIVEN_TYPE";
    /**
     * Too many AdSchedules are enabled for the feed item for the given day.
     *
     * @generated from protobuf enum value: TOO_MANY_SCHEDULES_PER_DAY = 6;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["TOO_MANY_SCHEDULES_PER_DAY"] = 6] = "TOO_MANY_SCHEDULES_PER_DAY";
    /**
     * A feed item may either have an enabled campaign target or an enabled ad
     * group target.
     *
     * @generated from protobuf enum value: CANNOT_HAVE_ENABLED_CAMPAIGN_AND_ENABLED_AD_GROUP_TARGETS = 7;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["CANNOT_HAVE_ENABLED_CAMPAIGN_AND_ENABLED_AD_GROUP_TARGETS"] = 7] = "CANNOT_HAVE_ENABLED_CAMPAIGN_AND_ENABLED_AD_GROUP_TARGETS";
    /**
     * Duplicate ad schedules aren't allowed.
     *
     * @generated from protobuf enum value: DUPLICATE_AD_SCHEDULE = 8;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["DUPLICATE_AD_SCHEDULE"] = 8] = "DUPLICATE_AD_SCHEDULE";
    /**
     * Duplicate keywords aren't allowed.
     *
     * @generated from protobuf enum value: DUPLICATE_KEYWORD = 9;
     */
    FeedItemTargetErrorEnum_FeedItemTargetError[FeedItemTargetErrorEnum_FeedItemTargetError["DUPLICATE_KEYWORD"] = 9] = "DUPLICATE_KEYWORD";
})(FeedItemTargetErrorEnum_FeedItemTargetError = exports.FeedItemTargetErrorEnum_FeedItemTargetError || (exports.FeedItemTargetErrorEnum_FeedItemTargetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemTargetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FeedItemTargetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FeedItemTargetErrorEnum
 */
exports.FeedItemTargetErrorEnum = new FeedItemTargetErrorEnum$Type();
