// google-ads-api-client/dist/protos/bindings/ads/errors/feed_attribute_reference_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedAttributeReferenceErrorEnum = exports.FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible feed attribute reference errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FeedAttributeReferenceErrorEnum.FeedAttributeReferenceError
 */
var FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError;
(function (FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError[FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError[FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A feed referenced by ID has been removed.
     *
     * @generated from protobuf enum value: CANNOT_REFERENCE_REMOVED_FEED = 2;
     */
    FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError[FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError["CANNOT_REFERENCE_REMOVED_FEED"] = 2] = "CANNOT_REFERENCE_REMOVED_FEED";
    /**
     * There is no enabled feed with the given name.
     *
     * @generated from protobuf enum value: INVALID_FEED_NAME = 3;
     */
    FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError[FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError["INVALID_FEED_NAME"] = 3] = "INVALID_FEED_NAME";
    /**
     * There is no feed attribute in an enabled feed with the given name.
     *
     * @generated from protobuf enum value: INVALID_FEED_ATTRIBUTE_NAME = 4;
     */
    FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError[FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError["INVALID_FEED_ATTRIBUTE_NAME"] = 4] = "INVALID_FEED_ATTRIBUTE_NAME";
})(FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError = exports.FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError || (exports.FeedAttributeReferenceErrorEnum_FeedAttributeReferenceError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedAttributeReferenceErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FeedAttributeReferenceErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FeedAttributeReferenceErrorEnum
 */
exports.FeedAttributeReferenceErrorEnum = new FeedAttributeReferenceErrorEnum$Type();
