// google-ads-api-client/dist/protos/bindings/ads/errors/feed_mapping_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedMappingErrorEnum = exports.FeedMappingErrorEnum_FeedMappingError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible feed item errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FeedMappingErrorEnum.FeedMappingError
 */
var FeedMappingErrorEnum_FeedMappingError;
(function (FeedMappingErrorEnum_FeedMappingError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The given placeholder field does not exist.
     *
     * @generated from protobuf enum value: INVALID_PLACEHOLDER_FIELD = 2;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["INVALID_PLACEHOLDER_FIELD"] = 2] = "INVALID_PLACEHOLDER_FIELD";
    /**
     * The given criterion field does not exist.
     *
     * @generated from protobuf enum value: INVALID_CRITERION_FIELD = 3;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["INVALID_CRITERION_FIELD"] = 3] = "INVALID_CRITERION_FIELD";
    /**
     * The given placeholder type does not exist.
     *
     * @generated from protobuf enum value: INVALID_PLACEHOLDER_TYPE = 4;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["INVALID_PLACEHOLDER_TYPE"] = 4] = "INVALID_PLACEHOLDER_TYPE";
    /**
     * The given criterion type does not exist.
     *
     * @generated from protobuf enum value: INVALID_CRITERION_TYPE = 5;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["INVALID_CRITERION_TYPE"] = 5] = "INVALID_CRITERION_TYPE";
    /**
     * A feed mapping must contain at least one attribute field mapping.
     *
     * @generated from protobuf enum value: NO_ATTRIBUTE_FIELD_MAPPINGS = 7;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["NO_ATTRIBUTE_FIELD_MAPPINGS"] = 7] = "NO_ATTRIBUTE_FIELD_MAPPINGS";
    /**
     * The type of the feed attribute referenced in the attribute field mapping
     * must match the type of the placeholder field.
     *
     * @generated from protobuf enum value: FEED_ATTRIBUTE_TYPE_MISMATCH = 8;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["FEED_ATTRIBUTE_TYPE_MISMATCH"] = 8] = "FEED_ATTRIBUTE_TYPE_MISMATCH";
    /**
     * A feed mapping for a system generated feed cannot be operated on.
     *
     * @generated from protobuf enum value: CANNOT_OPERATE_ON_MAPPINGS_FOR_SYSTEM_GENERATED_FEED = 9;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["CANNOT_OPERATE_ON_MAPPINGS_FOR_SYSTEM_GENERATED_FEED"] = 9] = "CANNOT_OPERATE_ON_MAPPINGS_FOR_SYSTEM_GENERATED_FEED";
    /**
     * Only one feed mapping for a placeholder type is allowed per feed or
     * customer (depending on the placeholder type).
     *
     * @generated from protobuf enum value: MULTIPLE_MAPPINGS_FOR_PLACEHOLDER_TYPE = 10;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["MULTIPLE_MAPPINGS_FOR_PLACEHOLDER_TYPE"] = 10] = "MULTIPLE_MAPPINGS_FOR_PLACEHOLDER_TYPE";
    /**
     * Only one feed mapping for a criterion type is allowed per customer.
     *
     * @generated from protobuf enum value: MULTIPLE_MAPPINGS_FOR_CRITERION_TYPE = 11;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["MULTIPLE_MAPPINGS_FOR_CRITERION_TYPE"] = 11] = "MULTIPLE_MAPPINGS_FOR_CRITERION_TYPE";
    /**
     * Only one feed attribute mapping for a placeholder field is allowed
     * (depending on the placeholder type).
     *
     * @generated from protobuf enum value: MULTIPLE_MAPPINGS_FOR_PLACEHOLDER_FIELD = 12;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["MULTIPLE_MAPPINGS_FOR_PLACEHOLDER_FIELD"] = 12] = "MULTIPLE_MAPPINGS_FOR_PLACEHOLDER_FIELD";
    /**
     * Only one feed attribute mapping for a criterion field is allowed
     * (depending on the criterion type).
     *
     * @generated from protobuf enum value: MULTIPLE_MAPPINGS_FOR_CRITERION_FIELD = 13;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["MULTIPLE_MAPPINGS_FOR_CRITERION_FIELD"] = 13] = "MULTIPLE_MAPPINGS_FOR_CRITERION_FIELD";
    /**
     * This feed mapping may not contain any explicit attribute field mappings.
     *
     * @generated from protobuf enum value: UNEXPECTED_ATTRIBUTE_FIELD_MAPPINGS = 14;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["UNEXPECTED_ATTRIBUTE_FIELD_MAPPINGS"] = 14] = "UNEXPECTED_ATTRIBUTE_FIELD_MAPPINGS";
    /**
     * Location placeholder feed mappings can only be created for Places feeds.
     *
     * @generated from protobuf enum value: LOCATION_PLACEHOLDER_ONLY_FOR_PLACES_FEEDS = 15;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["LOCATION_PLACEHOLDER_ONLY_FOR_PLACES_FEEDS"] = 15] = "LOCATION_PLACEHOLDER_ONLY_FOR_PLACES_FEEDS";
    /**
     * Mappings for typed feeds cannot be modified.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_MAPPINGS_FOR_TYPED_FEED = 16;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["CANNOT_MODIFY_MAPPINGS_FOR_TYPED_FEED"] = 16] = "CANNOT_MODIFY_MAPPINGS_FOR_TYPED_FEED";
    /**
     * The given placeholder type can only be mapped to system generated feeds.
     *
     * @generated from protobuf enum value: INVALID_PLACEHOLDER_TYPE_FOR_NON_SYSTEM_GENERATED_FEED = 17;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["INVALID_PLACEHOLDER_TYPE_FOR_NON_SYSTEM_GENERATED_FEED"] = 17] = "INVALID_PLACEHOLDER_TYPE_FOR_NON_SYSTEM_GENERATED_FEED";
    /**
     * The given placeholder type cannot be mapped to a system generated feed
     * with the given type.
     *
     * @generated from protobuf enum value: INVALID_PLACEHOLDER_TYPE_FOR_SYSTEM_GENERATED_FEED_TYPE = 18;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["INVALID_PLACEHOLDER_TYPE_FOR_SYSTEM_GENERATED_FEED_TYPE"] = 18] = "INVALID_PLACEHOLDER_TYPE_FOR_SYSTEM_GENERATED_FEED_TYPE";
    /**
     * The "field" oneof was not set in an AttributeFieldMapping.
     *
     * @generated from protobuf enum value: ATTRIBUTE_FIELD_MAPPING_MISSING_FIELD = 19;
     */
    FeedMappingErrorEnum_FeedMappingError[FeedMappingErrorEnum_FeedMappingError["ATTRIBUTE_FIELD_MAPPING_MISSING_FIELD"] = 19] = "ATTRIBUTE_FIELD_MAPPING_MISSING_FIELD";
})(FeedMappingErrorEnum_FeedMappingError = exports.FeedMappingErrorEnum_FeedMappingError || (exports.FeedMappingErrorEnum_FeedMappingError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedMappingErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FeedMappingErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FeedMappingErrorEnum
 */
exports.FeedMappingErrorEnum = new FeedMappingErrorEnum$Type();
