// google-ads-api-client/dist/protos/bindings/ads/errors/feed_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedErrorEnum = exports.FeedErrorEnum_FeedError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible feed errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FeedErrorEnum.FeedError
 */
var FeedErrorEnum_FeedError;
(function (FeedErrorEnum_FeedError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The names of the FeedAttributes must be unique.
     *
     * @generated from protobuf enum value: ATTRIBUTE_NAMES_NOT_UNIQUE = 2;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["ATTRIBUTE_NAMES_NOT_UNIQUE"] = 2] = "ATTRIBUTE_NAMES_NOT_UNIQUE";
    /**
     * The attribute list must be an exact copy of the existing list if the
     * attribute ID's are present.
     *
     * @generated from protobuf enum value: ATTRIBUTES_DO_NOT_MATCH_EXISTING_ATTRIBUTES = 3;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["ATTRIBUTES_DO_NOT_MATCH_EXISTING_ATTRIBUTES"] = 3] = "ATTRIBUTES_DO_NOT_MATCH_EXISTING_ATTRIBUTES";
    /**
     * Cannot specify USER origin for a system generated feed.
     *
     * @generated from protobuf enum value: CANNOT_SPECIFY_USER_ORIGIN_FOR_SYSTEM_FEED = 4;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["CANNOT_SPECIFY_USER_ORIGIN_FOR_SYSTEM_FEED"] = 4] = "CANNOT_SPECIFY_USER_ORIGIN_FOR_SYSTEM_FEED";
    /**
     * Cannot specify GOOGLE origin for a non-system generated feed.
     *
     * @generated from protobuf enum value: CANNOT_SPECIFY_GOOGLE_ORIGIN_FOR_NON_SYSTEM_FEED = 5;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["CANNOT_SPECIFY_GOOGLE_ORIGIN_FOR_NON_SYSTEM_FEED"] = 5] = "CANNOT_SPECIFY_GOOGLE_ORIGIN_FOR_NON_SYSTEM_FEED";
    /**
     * Cannot specify feed attributes for system feed.
     *
     * @generated from protobuf enum value: CANNOT_SPECIFY_FEED_ATTRIBUTES_FOR_SYSTEM_FEED = 6;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["CANNOT_SPECIFY_FEED_ATTRIBUTES_FOR_SYSTEM_FEED"] = 6] = "CANNOT_SPECIFY_FEED_ATTRIBUTES_FOR_SYSTEM_FEED";
    /**
     * Cannot update FeedAttributes on feed with origin GOOGLE.
     *
     * @generated from protobuf enum value: CANNOT_UPDATE_FEED_ATTRIBUTES_WITH_ORIGIN_GOOGLE = 7;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["CANNOT_UPDATE_FEED_ATTRIBUTES_WITH_ORIGIN_GOOGLE"] = 7] = "CANNOT_UPDATE_FEED_ATTRIBUTES_WITH_ORIGIN_GOOGLE";
    /**
     * The given ID refers to a removed Feed. Removed Feeds are immutable.
     *
     * @generated from protobuf enum value: FEED_REMOVED = 8;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["FEED_REMOVED"] = 8] = "FEED_REMOVED";
    /**
     * The origin of the feed is not valid for the client.
     *
     * @generated from protobuf enum value: INVALID_ORIGIN_VALUE = 9;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["INVALID_ORIGIN_VALUE"] = 9] = "INVALID_ORIGIN_VALUE";
    /**
     * A user can only create and modify feeds with USER origin.
     *
     * @generated from protobuf enum value: FEED_ORIGIN_IS_NOT_USER = 10;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["FEED_ORIGIN_IS_NOT_USER"] = 10] = "FEED_ORIGIN_IS_NOT_USER";
    /**
     * Invalid auth token for the given email.
     *
     * @generated from protobuf enum value: INVALID_AUTH_TOKEN_FOR_EMAIL = 11;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["INVALID_AUTH_TOKEN_FOR_EMAIL"] = 11] = "INVALID_AUTH_TOKEN_FOR_EMAIL";
    /**
     * Invalid email specified.
     *
     * @generated from protobuf enum value: INVALID_EMAIL = 12;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["INVALID_EMAIL"] = 12] = "INVALID_EMAIL";
    /**
     * Feed name matches that of another active Feed.
     *
     * @generated from protobuf enum value: DUPLICATE_FEED_NAME = 13;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["DUPLICATE_FEED_NAME"] = 13] = "DUPLICATE_FEED_NAME";
    /**
     * Name of feed is not allowed.
     *
     * @generated from protobuf enum value: INVALID_FEED_NAME = 14;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["INVALID_FEED_NAME"] = 14] = "INVALID_FEED_NAME";
    /**
     * Missing OAuthInfo.
     *
     * @generated from protobuf enum value: MISSING_OAUTH_INFO = 15;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["MISSING_OAUTH_INFO"] = 15] = "MISSING_OAUTH_INFO";
    /**
     * New FeedAttributes must not affect the unique key.
     *
     * @generated from protobuf enum value: NEW_ATTRIBUTE_CANNOT_BE_PART_OF_UNIQUE_KEY = 16;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["NEW_ATTRIBUTE_CANNOT_BE_PART_OF_UNIQUE_KEY"] = 16] = "NEW_ATTRIBUTE_CANNOT_BE_PART_OF_UNIQUE_KEY";
    /**
     * Too many FeedAttributes for a Feed.
     *
     * @generated from protobuf enum value: TOO_MANY_ATTRIBUTES = 17;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["TOO_MANY_ATTRIBUTES"] = 17] = "TOO_MANY_ATTRIBUTES";
    /**
     * The business account is not valid.
     *
     * @generated from protobuf enum value: INVALID_BUSINESS_ACCOUNT = 18;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["INVALID_BUSINESS_ACCOUNT"] = 18] = "INVALID_BUSINESS_ACCOUNT";
    /**
     * Business account cannot access Business Profile.
     *
     * @generated from protobuf enum value: BUSINESS_ACCOUNT_CANNOT_ACCESS_LOCATION_ACCOUNT = 19;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["BUSINESS_ACCOUNT_CANNOT_ACCESS_LOCATION_ACCOUNT"] = 19] = "BUSINESS_ACCOUNT_CANNOT_ACCESS_LOCATION_ACCOUNT";
    /**
     * Invalid chain ID provided for affiliate location feed.
     *
     * @generated from protobuf enum value: INVALID_AFFILIATE_CHAIN_ID = 20;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["INVALID_AFFILIATE_CHAIN_ID"] = 20] = "INVALID_AFFILIATE_CHAIN_ID";
    /**
     * There is already a feed with the given system feed generation data.
     *
     * @generated from protobuf enum value: DUPLICATE_SYSTEM_FEED = 21;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["DUPLICATE_SYSTEM_FEED"] = 21] = "DUPLICATE_SYSTEM_FEED";
    /**
     * An error occurred accessing Business Profile.
     *
     * @generated from protobuf enum value: GMB_ACCESS_ERROR = 22;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["GMB_ACCESS_ERROR"] = 22] = "GMB_ACCESS_ERROR";
    /**
     * A customer cannot have both LOCATION and AFFILIATE_LOCATION feeds.
     *
     * @generated from protobuf enum value: CANNOT_HAVE_LOCATION_AND_AFFILIATE_LOCATION_FEEDS = 23;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["CANNOT_HAVE_LOCATION_AND_AFFILIATE_LOCATION_FEEDS"] = 23] = "CANNOT_HAVE_LOCATION_AND_AFFILIATE_LOCATION_FEEDS";
    /**
     * Feed-based extension is read-only for this extension type.
     *
     * @generated from protobuf enum value: LEGACY_EXTENSION_TYPE_READ_ONLY = 24;
     */
    FeedErrorEnum_FeedError[FeedErrorEnum_FeedError["LEGACY_EXTENSION_TYPE_READ_ONLY"] = 24] = "LEGACY_EXTENSION_TYPE_READ_ONLY";
})(FeedErrorEnum_FeedError = exports.FeedErrorEnum_FeedError || (exports.FeedErrorEnum_FeedError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FeedErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FeedErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FeedErrorEnum
 */
exports.FeedErrorEnum = new FeedErrorEnum$Type();
