// google-ads-api-client/dist/protos/bindings/ads/errors/function_parsing_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FunctionParsingErrorEnum = exports.FunctionParsingErrorEnum_FunctionParsingError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible function parsing errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FunctionParsingErrorEnum.FunctionParsingError
 */
var FunctionParsingErrorEnum_FunctionParsingError;
(function (FunctionParsingErrorEnum_FunctionParsingError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Unexpected end of function string.
     *
     * @generated from protobuf enum value: NO_MORE_INPUT = 2;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["NO_MORE_INPUT"] = 2] = "NO_MORE_INPUT";
    /**
     * Could not find an expected character.
     *
     * @generated from protobuf enum value: EXPECTED_CHARACTER = 3;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["EXPECTED_CHARACTER"] = 3] = "EXPECTED_CHARACTER";
    /**
     * Unexpected separator character.
     *
     * @generated from protobuf enum value: UNEXPECTED_SEPARATOR = 4;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["UNEXPECTED_SEPARATOR"] = 4] = "UNEXPECTED_SEPARATOR";
    /**
     * Unmatched left bracket or parenthesis.
     *
     * @generated from protobuf enum value: UNMATCHED_LEFT_BRACKET = 5;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["UNMATCHED_LEFT_BRACKET"] = 5] = "UNMATCHED_LEFT_BRACKET";
    /**
     * Unmatched right bracket or parenthesis.
     *
     * @generated from protobuf enum value: UNMATCHED_RIGHT_BRACKET = 6;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["UNMATCHED_RIGHT_BRACKET"] = 6] = "UNMATCHED_RIGHT_BRACKET";
    /**
     * Functions are nested too deeply.
     *
     * @generated from protobuf enum value: TOO_MANY_NESTED_FUNCTIONS = 7;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["TOO_MANY_NESTED_FUNCTIONS"] = 7] = "TOO_MANY_NESTED_FUNCTIONS";
    /**
     * Missing right-hand-side operand.
     *
     * @generated from protobuf enum value: MISSING_RIGHT_HAND_OPERAND = 8;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["MISSING_RIGHT_HAND_OPERAND"] = 8] = "MISSING_RIGHT_HAND_OPERAND";
    /**
     * Invalid operator/function name.
     *
     * @generated from protobuf enum value: INVALID_OPERATOR_NAME = 9;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["INVALID_OPERATOR_NAME"] = 9] = "INVALID_OPERATOR_NAME";
    /**
     * Feed attribute operand's argument is not an integer.
     *
     * @generated from protobuf enum value: FEED_ATTRIBUTE_OPERAND_ARGUMENT_NOT_INTEGER = 10;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["FEED_ATTRIBUTE_OPERAND_ARGUMENT_NOT_INTEGER"] = 10] = "FEED_ATTRIBUTE_OPERAND_ARGUMENT_NOT_INTEGER";
    /**
     * Missing function operands.
     *
     * @generated from protobuf enum value: NO_OPERANDS = 11;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["NO_OPERANDS"] = 11] = "NO_OPERANDS";
    /**
     * Function had too many operands.
     *
     * @generated from protobuf enum value: TOO_MANY_OPERANDS = 12;
     */
    FunctionParsingErrorEnum_FunctionParsingError[FunctionParsingErrorEnum_FunctionParsingError["TOO_MANY_OPERANDS"] = 12] = "TOO_MANY_OPERANDS";
})(FunctionParsingErrorEnum_FunctionParsingError = exports.FunctionParsingErrorEnum_FunctionParsingError || (exports.FunctionParsingErrorEnum_FunctionParsingError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FunctionParsingErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FunctionParsingErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FunctionParsingErrorEnum
 */
exports.FunctionParsingErrorEnum = new FunctionParsingErrorEnum$Type();
