// google-ads-api-client/dist/protos/bindings/ads/errors/function_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FunctionErrorEnum = exports.FunctionErrorEnum_FunctionError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible function errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.FunctionErrorEnum.FunctionError
 */
var FunctionErrorEnum_FunctionError;
(function (FunctionErrorEnum_FunctionError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The format of the function is not recognized as a supported function
     * format.
     *
     * @generated from protobuf enum value: INVALID_FUNCTION_FORMAT = 2;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_FUNCTION_FORMAT"] = 2] = "INVALID_FUNCTION_FORMAT";
    /**
     * Operand data types do not match.
     *
     * @generated from protobuf enum value: DATA_TYPE_MISMATCH = 3;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["DATA_TYPE_MISMATCH"] = 3] = "DATA_TYPE_MISMATCH";
    /**
     * The operands cannot be used together in a conjunction.
     *
     * @generated from protobuf enum value: INVALID_CONJUNCTION_OPERANDS = 4;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_CONJUNCTION_OPERANDS"] = 4] = "INVALID_CONJUNCTION_OPERANDS";
    /**
     * Invalid numer of Operands.
     *
     * @generated from protobuf enum value: INVALID_NUMBER_OF_OPERANDS = 5;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_NUMBER_OF_OPERANDS"] = 5] = "INVALID_NUMBER_OF_OPERANDS";
    /**
     * Operand Type not supported.
     *
     * @generated from protobuf enum value: INVALID_OPERAND_TYPE = 6;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_OPERAND_TYPE"] = 6] = "INVALID_OPERAND_TYPE";
    /**
     * Operator not supported.
     *
     * @generated from protobuf enum value: INVALID_OPERATOR = 7;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_OPERATOR"] = 7] = "INVALID_OPERATOR";
    /**
     * Request context type not supported.
     *
     * @generated from protobuf enum value: INVALID_REQUEST_CONTEXT_TYPE = 8;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_REQUEST_CONTEXT_TYPE"] = 8] = "INVALID_REQUEST_CONTEXT_TYPE";
    /**
     * The matching function is not allowed for call placeholders
     *
     * @generated from protobuf enum value: INVALID_FUNCTION_FOR_CALL_PLACEHOLDER = 9;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_FUNCTION_FOR_CALL_PLACEHOLDER"] = 9] = "INVALID_FUNCTION_FOR_CALL_PLACEHOLDER";
    /**
     * The matching function is not allowed for the specified placeholder
     *
     * @generated from protobuf enum value: INVALID_FUNCTION_FOR_PLACEHOLDER = 10;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_FUNCTION_FOR_PLACEHOLDER"] = 10] = "INVALID_FUNCTION_FOR_PLACEHOLDER";
    /**
     * Invalid operand.
     *
     * @generated from protobuf enum value: INVALID_OPERAND = 11;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_OPERAND"] = 11] = "INVALID_OPERAND";
    /**
     * Missing value for the constant operand.
     *
     * @generated from protobuf enum value: MISSING_CONSTANT_OPERAND_VALUE = 12;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["MISSING_CONSTANT_OPERAND_VALUE"] = 12] = "MISSING_CONSTANT_OPERAND_VALUE";
    /**
     * The value of the constant operand is invalid.
     *
     * @generated from protobuf enum value: INVALID_CONSTANT_OPERAND_VALUE = 13;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_CONSTANT_OPERAND_VALUE"] = 13] = "INVALID_CONSTANT_OPERAND_VALUE";
    /**
     * Invalid function nesting.
     *
     * @generated from protobuf enum value: INVALID_NESTING = 14;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_NESTING"] = 14] = "INVALID_NESTING";
    /**
     * The Feed ID was different from another Feed ID in the same function.
     *
     * @generated from protobuf enum value: MULTIPLE_FEED_IDS_NOT_SUPPORTED = 15;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["MULTIPLE_FEED_IDS_NOT_SUPPORTED"] = 15] = "MULTIPLE_FEED_IDS_NOT_SUPPORTED";
    /**
     * The matching function is invalid for use with a feed with a fixed schema.
     *
     * @generated from protobuf enum value: INVALID_FUNCTION_FOR_FEED_WITH_FIXED_SCHEMA = 16;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_FUNCTION_FOR_FEED_WITH_FIXED_SCHEMA"] = 16] = "INVALID_FUNCTION_FOR_FEED_WITH_FIXED_SCHEMA";
    /**
     * Invalid attribute name.
     *
     * @generated from protobuf enum value: INVALID_ATTRIBUTE_NAME = 17;
     */
    FunctionErrorEnum_FunctionError[FunctionErrorEnum_FunctionError["INVALID_ATTRIBUTE_NAME"] = 17] = "INVALID_ATTRIBUTE_NAME";
})(FunctionErrorEnum_FunctionError = exports.FunctionErrorEnum_FunctionError || (exports.FunctionErrorEnum_FunctionError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class FunctionErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.FunctionErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.FunctionErrorEnum
 */
exports.FunctionErrorEnum = new FunctionErrorEnum$Type();
