// google-ads-api-client/dist/protos/bindings/ads/errors/geo_target_constant_suggestion_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeoTargetConstantSuggestionErrorEnum = exports.GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible geo target constant suggestion errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.GeoTargetConstantSuggestionErrorEnum.GeoTargetConstantSuggestionError
 */
var GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError;
(function (GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError[GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError[GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A location name cannot be greater than 300 characters.
     *
     * @generated from protobuf enum value: LOCATION_NAME_SIZE_LIMIT = 2;
     */
    GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError[GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError["LOCATION_NAME_SIZE_LIMIT"] = 2] = "LOCATION_NAME_SIZE_LIMIT";
    /**
     * At most 25 location names can be specified in a SuggestGeoTargetConstants
     * method.
     *
     * @generated from protobuf enum value: LOCATION_NAME_LIMIT = 3;
     */
    GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError[GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError["LOCATION_NAME_LIMIT"] = 3] = "LOCATION_NAME_LIMIT";
    /**
     * The country code is invalid.
     *
     * @generated from protobuf enum value: INVALID_COUNTRY_CODE = 4;
     */
    GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError[GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError["INVALID_COUNTRY_CODE"] = 4] = "INVALID_COUNTRY_CODE";
    /**
     * Geo target constant resource names or location names must be provided in
     * the request.
     *
     * @generated from protobuf enum value: REQUEST_PARAMETERS_UNSET = 5;
     */
    GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError[GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError["REQUEST_PARAMETERS_UNSET"] = 5] = "REQUEST_PARAMETERS_UNSET";
})(GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError = exports.GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError || (exports.GeoTargetConstantSuggestionErrorEnum_GeoTargetConstantSuggestionError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class GeoTargetConstantSuggestionErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.GeoTargetConstantSuggestionErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.GeoTargetConstantSuggestionErrorEnum
 */
exports.GeoTargetConstantSuggestionErrorEnum = new GeoTargetConstantSuggestionErrorEnum$Type();
