// google-ads-api-client/dist/protos/bindings/ads/errors/header_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeaderErrorEnum = exports.HeaderErrorEnum_HeaderError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible header errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.HeaderErrorEnum.HeaderError
 */
var HeaderErrorEnum_HeaderError;
(function (HeaderErrorEnum_HeaderError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    HeaderErrorEnum_HeaderError[HeaderErrorEnum_HeaderError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    HeaderErrorEnum_HeaderError[HeaderErrorEnum_HeaderError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The login customer ID could not be validated.
     *
     * @generated from protobuf enum value: INVALID_LOGIN_CUSTOMER_ID = 3;
     */
    HeaderErrorEnum_HeaderError[HeaderErrorEnum_HeaderError["INVALID_LOGIN_CUSTOMER_ID"] = 3] = "INVALID_LOGIN_CUSTOMER_ID";
    /**
     * The linked customer ID could not be validated.
     *
     * @generated from protobuf enum value: INVALID_LINKED_CUSTOMER_ID = 7;
     */
    HeaderErrorEnum_HeaderError[HeaderErrorEnum_HeaderError["INVALID_LINKED_CUSTOMER_ID"] = 7] = "INVALID_LINKED_CUSTOMER_ID";
})(HeaderErrorEnum_HeaderError = exports.HeaderErrorEnum_HeaderError || (exports.HeaderErrorEnum_HeaderError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class HeaderErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.HeaderErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.HeaderErrorEnum
 */
exports.HeaderErrorEnum = new HeaderErrorEnum$Type();
