// google-ads-api-client/dist/protos/bindings/ads/errors/id_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdErrorEnum = exports.IdErrorEnum_IdError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible ID errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.IdErrorEnum.IdError
 */
var IdErrorEnum_IdError;
(function (IdErrorEnum_IdError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    IdErrorEnum_IdError[IdErrorEnum_IdError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    IdErrorEnum_IdError[IdErrorEnum_IdError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * ID not found
     *
     * @generated from protobuf enum value: NOT_FOUND = 2;
     */
    IdErrorEnum_IdError[IdErrorEnum_IdError["NOT_FOUND"] = 2] = "NOT_FOUND";
})(IdErrorEnum_IdError = exports.IdErrorEnum_IdError || (exports.IdErrorEnum_IdError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class IdErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.IdErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.IdErrorEnum
 */
exports.IdErrorEnum = new IdErrorEnum$Type();
