// google-ads-api-client/dist/protos/bindings/ads/errors/image_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageErrorEnum = exports.ImageErrorEnum_ImageError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible image errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ImageErrorEnum.ImageError
 */
var ImageErrorEnum_ImageError;
(function (ImageErrorEnum_ImageError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The image is not valid.
     *
     * @generated from protobuf enum value: INVALID_IMAGE = 2;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["INVALID_IMAGE"] = 2] = "INVALID_IMAGE";
    /**
     * The image could not be stored.
     *
     * @generated from protobuf enum value: STORAGE_ERROR = 3;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["STORAGE_ERROR"] = 3] = "STORAGE_ERROR";
    /**
     * There was a problem with the request.
     *
     * @generated from protobuf enum value: BAD_REQUEST = 4;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["BAD_REQUEST"] = 4] = "BAD_REQUEST";
    /**
     * The image is not of legal dimensions.
     *
     * @generated from protobuf enum value: UNEXPECTED_SIZE = 5;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["UNEXPECTED_SIZE"] = 5] = "UNEXPECTED_SIZE";
    /**
     * Animated image are not permitted.
     *
     * @generated from protobuf enum value: ANIMATED_NOT_ALLOWED = 6;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["ANIMATED_NOT_ALLOWED"] = 6] = "ANIMATED_NOT_ALLOWED";
    /**
     * Animation is too long.
     *
     * @generated from protobuf enum value: ANIMATION_TOO_LONG = 7;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["ANIMATION_TOO_LONG"] = 7] = "ANIMATION_TOO_LONG";
    /**
     * There was an error on the server.
     *
     * @generated from protobuf enum value: SERVER_ERROR = 8;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["SERVER_ERROR"] = 8] = "SERVER_ERROR";
    /**
     * Image cannot be in CMYK color format.
     *
     * @generated from protobuf enum value: CMYK_JPEG_NOT_ALLOWED = 9;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["CMYK_JPEG_NOT_ALLOWED"] = 9] = "CMYK_JPEG_NOT_ALLOWED";
    /**
     * Flash images are not permitted.
     *
     * @generated from protobuf enum value: FLASH_NOT_ALLOWED = 10;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_NOT_ALLOWED"] = 10] = "FLASH_NOT_ALLOWED";
    /**
     * Flash images must support clickTag.
     *
     * @generated from protobuf enum value: FLASH_WITHOUT_CLICKTAG = 11;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_WITHOUT_CLICKTAG"] = 11] = "FLASH_WITHOUT_CLICKTAG";
    /**
     * A flash error has occurred after fixing the click tag.
     *
     * @generated from protobuf enum value: FLASH_ERROR_AFTER_FIXING_CLICK_TAG = 12;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_ERROR_AFTER_FIXING_CLICK_TAG"] = 12] = "FLASH_ERROR_AFTER_FIXING_CLICK_TAG";
    /**
     * Unacceptable visual effects.
     *
     * @generated from protobuf enum value: ANIMATED_VISUAL_EFFECT = 13;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["ANIMATED_VISUAL_EFFECT"] = 13] = "ANIMATED_VISUAL_EFFECT";
    /**
     * There was a problem with the flash image.
     *
     * @generated from protobuf enum value: FLASH_ERROR = 14;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_ERROR"] = 14] = "FLASH_ERROR";
    /**
     * Incorrect image layout.
     *
     * @generated from protobuf enum value: LAYOUT_PROBLEM = 15;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["LAYOUT_PROBLEM"] = 15] = "LAYOUT_PROBLEM";
    /**
     * There was a problem reading the image file.
     *
     * @generated from protobuf enum value: PROBLEM_READING_IMAGE_FILE = 16;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["PROBLEM_READING_IMAGE_FILE"] = 16] = "PROBLEM_READING_IMAGE_FILE";
    /**
     * There was an error storing the image.
     *
     * @generated from protobuf enum value: ERROR_STORING_IMAGE = 17;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["ERROR_STORING_IMAGE"] = 17] = "ERROR_STORING_IMAGE";
    /**
     * The aspect ratio of the image is not allowed.
     *
     * @generated from protobuf enum value: ASPECT_RATIO_NOT_ALLOWED = 18;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["ASPECT_RATIO_NOT_ALLOWED"] = 18] = "ASPECT_RATIO_NOT_ALLOWED";
    /**
     * Flash cannot have network objects.
     *
     * @generated from protobuf enum value: FLASH_HAS_NETWORK_OBJECTS = 19;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_HAS_NETWORK_OBJECTS"] = 19] = "FLASH_HAS_NETWORK_OBJECTS";
    /**
     * Flash cannot have network methods.
     *
     * @generated from protobuf enum value: FLASH_HAS_NETWORK_METHODS = 20;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_HAS_NETWORK_METHODS"] = 20] = "FLASH_HAS_NETWORK_METHODS";
    /**
     * Flash cannot have a Url.
     *
     * @generated from protobuf enum value: FLASH_HAS_URL = 21;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_HAS_URL"] = 21] = "FLASH_HAS_URL";
    /**
     * Flash cannot use mouse tracking.
     *
     * @generated from protobuf enum value: FLASH_HAS_MOUSE_TRACKING = 22;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_HAS_MOUSE_TRACKING"] = 22] = "FLASH_HAS_MOUSE_TRACKING";
    /**
     * Flash cannot have a random number.
     *
     * @generated from protobuf enum value: FLASH_HAS_RANDOM_NUM = 23;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_HAS_RANDOM_NUM"] = 23] = "FLASH_HAS_RANDOM_NUM";
    /**
     * Ad click target cannot be '_self'.
     *
     * @generated from protobuf enum value: FLASH_SELF_TARGETS = 24;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_SELF_TARGETS"] = 24] = "FLASH_SELF_TARGETS";
    /**
     * GetUrl method should only use '_blank'.
     *
     * @generated from protobuf enum value: FLASH_BAD_GETURL_TARGET = 25;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_BAD_GETURL_TARGET"] = 25] = "FLASH_BAD_GETURL_TARGET";
    /**
     * Flash version is not supported.
     *
     * @generated from protobuf enum value: FLASH_VERSION_NOT_SUPPORTED = 26;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_VERSION_NOT_SUPPORTED"] = 26] = "FLASH_VERSION_NOT_SUPPORTED";
    /**
     * Flash movies need to have hard coded click URL or clickTAG
     *
     * @generated from protobuf enum value: FLASH_WITHOUT_HARD_CODED_CLICK_URL = 27;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_WITHOUT_HARD_CODED_CLICK_URL"] = 27] = "FLASH_WITHOUT_HARD_CODED_CLICK_URL";
    /**
     * Uploaded flash file is corrupted.
     *
     * @generated from protobuf enum value: INVALID_FLASH_FILE = 28;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["INVALID_FLASH_FILE"] = 28] = "INVALID_FLASH_FILE";
    /**
     * Uploaded flash file can be parsed, but the click tag can not be fixed
     * properly.
     *
     * @generated from protobuf enum value: FAILED_TO_FIX_CLICK_TAG_IN_FLASH = 29;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FAILED_TO_FIX_CLICK_TAG_IN_FLASH"] = 29] = "FAILED_TO_FIX_CLICK_TAG_IN_FLASH";
    /**
     * Flash movie accesses network resources
     *
     * @generated from protobuf enum value: FLASH_ACCESSES_NETWORK_RESOURCES = 30;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_ACCESSES_NETWORK_RESOURCES"] = 30] = "FLASH_ACCESSES_NETWORK_RESOURCES";
    /**
     * Flash movie attempts to call external javascript code
     *
     * @generated from protobuf enum value: FLASH_EXTERNAL_JS_CALL = 31;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_EXTERNAL_JS_CALL"] = 31] = "FLASH_EXTERNAL_JS_CALL";
    /**
     * Flash movie attempts to call flash system commands
     *
     * @generated from protobuf enum value: FLASH_EXTERNAL_FS_CALL = 32;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FLASH_EXTERNAL_FS_CALL"] = 32] = "FLASH_EXTERNAL_FS_CALL";
    /**
     * Image file is too large.
     *
     * @generated from protobuf enum value: FILE_TOO_LARGE = 33;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FILE_TOO_LARGE"] = 33] = "FILE_TOO_LARGE";
    /**
     * Image data is too large.
     *
     * @generated from protobuf enum value: IMAGE_DATA_TOO_LARGE = 34;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["IMAGE_DATA_TOO_LARGE"] = 34] = "IMAGE_DATA_TOO_LARGE";
    /**
     * Error while processing the image.
     *
     * @generated from protobuf enum value: IMAGE_PROCESSING_ERROR = 35;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["IMAGE_PROCESSING_ERROR"] = 35] = "IMAGE_PROCESSING_ERROR";
    /**
     * Image is too small.
     *
     * @generated from protobuf enum value: IMAGE_TOO_SMALL = 36;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["IMAGE_TOO_SMALL"] = 36] = "IMAGE_TOO_SMALL";
    /**
     * Input was invalid.
     *
     * @generated from protobuf enum value: INVALID_INPUT = 37;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["INVALID_INPUT"] = 37] = "INVALID_INPUT";
    /**
     * There was a problem reading the image file.
     *
     * @generated from protobuf enum value: PROBLEM_READING_FILE = 38;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["PROBLEM_READING_FILE"] = 38] = "PROBLEM_READING_FILE";
    /**
     * Image constraints are violated, but details like ASPECT_RATIO_NOT_ALLOWED
     * can't be provided. This happens when asset spec contains more than one
     * constraint and different criteria of different constraints are violated.
     *
     * @generated from protobuf enum value: IMAGE_CONSTRAINTS_VIOLATED = 39;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["IMAGE_CONSTRAINTS_VIOLATED"] = 39] = "IMAGE_CONSTRAINTS_VIOLATED";
    /**
     * Image format is not allowed.
     *
     * @generated from protobuf enum value: FORMAT_NOT_ALLOWED = 40;
     */
    ImageErrorEnum_ImageError[ImageErrorEnum_ImageError["FORMAT_NOT_ALLOWED"] = 40] = "FORMAT_NOT_ALLOWED";
})(ImageErrorEnum_ImageError = exports.ImageErrorEnum_ImageError || (exports.ImageErrorEnum_ImageError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ImageErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ImageErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ImageErrorEnum
 */
exports.ImageErrorEnum = new ImageErrorEnum$Type();
