// google-ads-api-client/dist/protos/bindings/ads/errors/internal_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InternalErrorEnum = exports.InternalErrorEnum_InternalError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible internal errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.InternalErrorEnum.InternalError
 */
var InternalErrorEnum_InternalError;
(function (InternalErrorEnum_InternalError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    InternalErrorEnum_InternalError[InternalErrorEnum_InternalError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    InternalErrorEnum_InternalError[InternalErrorEnum_InternalError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Google Ads API encountered unexpected internal error.
     *
     * @generated from protobuf enum value: INTERNAL_ERROR = 2;
     */
    InternalErrorEnum_InternalError[InternalErrorEnum_InternalError["INTERNAL_ERROR"] = 2] = "INTERNAL_ERROR";
    /**
     * The intended error code doesn't exist in specified API version. It will
     * be released in a future API version.
     *
     * @generated from protobuf enum value: ERROR_CODE_NOT_PUBLISHED = 3;
     */
    InternalErrorEnum_InternalError[InternalErrorEnum_InternalError["ERROR_CODE_NOT_PUBLISHED"] = 3] = "ERROR_CODE_NOT_PUBLISHED";
    /**
     * Google Ads API encountered an unexpected transient error. The user
     * should retry their request in these cases.
     *
     * @generated from protobuf enum value: TRANSIENT_ERROR = 4;
     */
    InternalErrorEnum_InternalError[InternalErrorEnum_InternalError["TRANSIENT_ERROR"] = 4] = "TRANSIENT_ERROR";
    /**
     * The request took longer than a deadline.
     *
     * @generated from protobuf enum value: DEADLINE_EXCEEDED = 5;
     */
    InternalErrorEnum_InternalError[InternalErrorEnum_InternalError["DEADLINE_EXCEEDED"] = 5] = "DEADLINE_EXCEEDED";
})(InternalErrorEnum_InternalError = exports.InternalErrorEnum_InternalError || (exports.InternalErrorEnum_InternalError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class InternalErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.InternalErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.InternalErrorEnum
 */
exports.InternalErrorEnum = new InternalErrorEnum$Type();
