// google-ads-api-client/dist/protos/bindings/ads/errors/invoice_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceErrorEnum = exports.InvoiceErrorEnum_InvoiceError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible invoice errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.InvoiceErrorEnum.InvoiceError
 */
var InvoiceErrorEnum_InvoiceError;
(function (InvoiceErrorEnum_InvoiceError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    InvoiceErrorEnum_InvoiceError[InvoiceErrorEnum_InvoiceError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    InvoiceErrorEnum_InvoiceError[InvoiceErrorEnum_InvoiceError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Cannot request invoices issued before 2019-01-01.
     *
     * @generated from protobuf enum value: YEAR_MONTH_TOO_OLD = 2;
     */
    InvoiceErrorEnum_InvoiceError[InvoiceErrorEnum_InvoiceError["YEAR_MONTH_TOO_OLD"] = 2] = "YEAR_MONTH_TOO_OLD";
    /**
     * Cannot request invoices for customer who doesn't receive invoices.
     *
     * @generated from protobuf enum value: NOT_INVOICED_CUSTOMER = 3;
     */
    InvoiceErrorEnum_InvoiceError[InvoiceErrorEnum_InvoiceError["NOT_INVOICED_CUSTOMER"] = 3] = "NOT_INVOICED_CUSTOMER";
    /**
     * Cannot request invoices for a non approved billing setup.
     *
     * @generated from protobuf enum value: BILLING_SETUP_NOT_APPROVED = 4;
     */
    InvoiceErrorEnum_InvoiceError[InvoiceErrorEnum_InvoiceError["BILLING_SETUP_NOT_APPROVED"] = 4] = "BILLING_SETUP_NOT_APPROVED";
    /**
     * Cannot request invoices for a billing setup that is not on monthly
     * invoicing.
     *
     * @generated from protobuf enum value: BILLING_SETUP_NOT_ON_MONTHLY_INVOICING = 5;
     */
    InvoiceErrorEnum_InvoiceError[InvoiceErrorEnum_InvoiceError["BILLING_SETUP_NOT_ON_MONTHLY_INVOICING"] = 5] = "BILLING_SETUP_NOT_ON_MONTHLY_INVOICING";
    /**
     * Cannot request invoices for a non serving customer.
     *
     * @generated from protobuf enum value: NON_SERVING_CUSTOMER = 6;
     */
    InvoiceErrorEnum_InvoiceError[InvoiceErrorEnum_InvoiceError["NON_SERVING_CUSTOMER"] = 6] = "NON_SERVING_CUSTOMER";
})(InvoiceErrorEnum_InvoiceError = exports.InvoiceErrorEnum_InvoiceError || (exports.InvoiceErrorEnum_InvoiceError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class InvoiceErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.InvoiceErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.InvoiceErrorEnum
 */
exports.InvoiceErrorEnum = new InvoiceErrorEnum$Type();
