// google-ads-api-client/dist/protos/bindings/ads/errors/keyword_plan_campaign_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanCampaignErrorEnum = exports.KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible errors from applying a keyword plan campaign.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.KeywordPlanCampaignErrorEnum.KeywordPlanCampaignError
 */
var KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError;
(function (KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError[KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError[KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A keyword plan campaign name is missing, empty, longer than allowed limit
     * or contains invalid chars.
     *
     * @generated from protobuf enum value: INVALID_NAME = 2;
     */
    KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError[KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError["INVALID_NAME"] = 2] = "INVALID_NAME";
    /**
     * A keyword plan campaign contains one or more untargetable languages.
     *
     * @generated from protobuf enum value: INVALID_LANGUAGES = 3;
     */
    KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError[KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError["INVALID_LANGUAGES"] = 3] = "INVALID_LANGUAGES";
    /**
     * A keyword plan campaign contains one or more invalid geo targets.
     *
     * @generated from protobuf enum value: INVALID_GEOS = 4;
     */
    KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError[KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError["INVALID_GEOS"] = 4] = "INVALID_GEOS";
    /**
     * The keyword plan campaign name is duplicate to an existing keyword plan
     * campaign name or other keyword plan campaign name in the request.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 5;
     */
    KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError[KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError["DUPLICATE_NAME"] = 5] = "DUPLICATE_NAME";
    /**
     * The number of geo targets in the keyword plan campaign exceeds limits.
     *
     * @generated from protobuf enum value: MAX_GEOS_EXCEEDED = 6;
     */
    KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError[KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError["MAX_GEOS_EXCEEDED"] = 6] = "MAX_GEOS_EXCEEDED";
    /**
     * The number of languages in the keyword plan campaign exceeds limits.
     *
     * @generated from protobuf enum value: MAX_LANGUAGES_EXCEEDED = 7;
     */
    KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError[KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError["MAX_LANGUAGES_EXCEEDED"] = 7] = "MAX_LANGUAGES_EXCEEDED";
})(KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError = exports.KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError || (exports.KeywordPlanCampaignErrorEnum_KeywordPlanCampaignError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanCampaignErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.KeywordPlanCampaignErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.KeywordPlanCampaignErrorEnum
 */
exports.KeywordPlanCampaignErrorEnum = new KeywordPlanCampaignErrorEnum$Type();
