// google-ads-api-client/dist/protos/bindings/ads/errors/keyword_plan_idea_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanIdeaErrorEnum = exports.KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible errors from KeywordPlanIdeaService.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.KeywordPlanIdeaErrorEnum.KeywordPlanIdeaError
 */
var KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError;
(function (KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError[KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError[KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Error when crawling the input URL.
     *
     * @generated from protobuf enum value: URL_CRAWL_ERROR = 2;
     */
    KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError[KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError["URL_CRAWL_ERROR"] = 2] = "URL_CRAWL_ERROR";
    /**
     * The input has an invalid value.
     *
     * @generated from protobuf enum value: INVALID_VALUE = 3;
     */
    KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError[KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError["INVALID_VALUE"] = 3] = "INVALID_VALUE";
})(KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError = exports.KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError || (exports.KeywordPlanIdeaErrorEnum_KeywordPlanIdeaError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanIdeaErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.KeywordPlanIdeaErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.KeywordPlanIdeaErrorEnum
 */
exports.KeywordPlanIdeaErrorEnum = new KeywordPlanIdeaErrorEnum$Type();
