// google-ads-api-client/dist/protos/bindings/ads/errors/keyword_plan_ad_group_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanAdGroupErrorEnum = exports.KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible errors from applying a keyword plan ad group.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.KeywordPlanAdGroupErrorEnum.KeywordPlanAdGroupError
 */
var KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError;
(function (KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError[KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError[KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The keyword plan ad group name is missing, empty, longer than allowed
     * limit or contains invalid chars.
     *
     * @generated from protobuf enum value: INVALID_NAME = 2;
     */
    KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError[KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError["INVALID_NAME"] = 2] = "INVALID_NAME";
    /**
     * The keyword plan ad group name is duplicate to an existing keyword plan
     * AdGroup name or other keyword plan AdGroup name in the request.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 3;
     */
    KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError[KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError["DUPLICATE_NAME"] = 3] = "DUPLICATE_NAME";
})(KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError = exports.KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError || (exports.KeywordPlanAdGroupErrorEnum_KeywordPlanAdGroupError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanAdGroupErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.KeywordPlanAdGroupErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.KeywordPlanAdGroupErrorEnum
 */
exports.KeywordPlanAdGroupErrorEnum = new KeywordPlanAdGroupErrorEnum$Type();
