// google-ads-api-client/dist/protos/bindings/ads/errors/label_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabelErrorEnum = exports.LabelErrorEnum_LabelError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible label errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.LabelErrorEnum.LabelError
 */
var LabelErrorEnum_LabelError;
(function (LabelErrorEnum_LabelError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * An inactive label cannot be applied.
     *
     * @generated from protobuf enum value: CANNOT_APPLY_INACTIVE_LABEL = 2;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["CANNOT_APPLY_INACTIVE_LABEL"] = 2] = "CANNOT_APPLY_INACTIVE_LABEL";
    /**
     * A label cannot be applied to a disabled ad group criterion.
     *
     * @generated from protobuf enum value: CANNOT_APPLY_LABEL_TO_DISABLED_AD_GROUP_CRITERION = 3;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["CANNOT_APPLY_LABEL_TO_DISABLED_AD_GROUP_CRITERION"] = 3] = "CANNOT_APPLY_LABEL_TO_DISABLED_AD_GROUP_CRITERION";
    /**
     * A label cannot be applied to a negative ad group criterion.
     *
     * @generated from protobuf enum value: CANNOT_APPLY_LABEL_TO_NEGATIVE_AD_GROUP_CRITERION = 4;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["CANNOT_APPLY_LABEL_TO_NEGATIVE_AD_GROUP_CRITERION"] = 4] = "CANNOT_APPLY_LABEL_TO_NEGATIVE_AD_GROUP_CRITERION";
    /**
     * Cannot apply more than 50 labels per resource.
     *
     * @generated from protobuf enum value: EXCEEDED_LABEL_LIMIT_PER_TYPE = 5;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["EXCEEDED_LABEL_LIMIT_PER_TYPE"] = 5] = "EXCEEDED_LABEL_LIMIT_PER_TYPE";
    /**
     * Labels from a manager account cannot be applied to campaign, ad group,
     * ad group ad, or ad group criterion resources.
     *
     * @generated from protobuf enum value: INVALID_RESOURCE_FOR_MANAGER_LABEL = 6;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["INVALID_RESOURCE_FOR_MANAGER_LABEL"] = 6] = "INVALID_RESOURCE_FOR_MANAGER_LABEL";
    /**
     * Label names must be unique.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 7;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["DUPLICATE_NAME"] = 7] = "DUPLICATE_NAME";
    /**
     * Label names cannot be empty.
     *
     * @generated from protobuf enum value: INVALID_LABEL_NAME = 8;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["INVALID_LABEL_NAME"] = 8] = "INVALID_LABEL_NAME";
    /**
     * Labels cannot be applied to a draft.
     *
     * @generated from protobuf enum value: CANNOT_ATTACH_LABEL_TO_DRAFT = 9;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["CANNOT_ATTACH_LABEL_TO_DRAFT"] = 9] = "CANNOT_ATTACH_LABEL_TO_DRAFT";
    /**
     * Labels not from a manager account cannot be applied to the customer
     * resource.
     *
     * @generated from protobuf enum value: CANNOT_ATTACH_NON_MANAGER_LABEL_TO_CUSTOMER = 10;
     */
    LabelErrorEnum_LabelError[LabelErrorEnum_LabelError["CANNOT_ATTACH_NON_MANAGER_LABEL_TO_CUSTOMER"] = 10] = "CANNOT_ATTACH_NON_MANAGER_LABEL_TO_CUSTOMER";
})(LabelErrorEnum_LabelError = exports.LabelErrorEnum_LabelError || (exports.LabelErrorEnum_LabelError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LabelErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.LabelErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.LabelErrorEnum
 */
exports.LabelErrorEnum = new LabelErrorEnum$Type();
