// google-ads-api-client/dist/protos/bindings/ads/errors/language_code_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LanguageCodeErrorEnum = exports.LanguageCodeErrorEnum_LanguageCodeError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing language code errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.LanguageCodeErrorEnum.LanguageCodeError
 */
var LanguageCodeErrorEnum_LanguageCodeError;
(function (LanguageCodeErrorEnum_LanguageCodeError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    LanguageCodeErrorEnum_LanguageCodeError[LanguageCodeErrorEnum_LanguageCodeError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    LanguageCodeErrorEnum_LanguageCodeError[LanguageCodeErrorEnum_LanguageCodeError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The input language code is not recognized.
     *
     * @generated from protobuf enum value: LANGUAGE_CODE_NOT_FOUND = 2;
     */
    LanguageCodeErrorEnum_LanguageCodeError[LanguageCodeErrorEnum_LanguageCodeError["LANGUAGE_CODE_NOT_FOUND"] = 2] = "LANGUAGE_CODE_NOT_FOUND";
    /**
     * The language code is not supported.
     *
     * @generated from protobuf enum value: INVALID_LANGUAGE_CODE = 3;
     */
    LanguageCodeErrorEnum_LanguageCodeError[LanguageCodeErrorEnum_LanguageCodeError["INVALID_LANGUAGE_CODE"] = 3] = "INVALID_LANGUAGE_CODE";
})(LanguageCodeErrorEnum_LanguageCodeError = exports.LanguageCodeErrorEnum_LanguageCodeError || (exports.LanguageCodeErrorEnum_LanguageCodeError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class LanguageCodeErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.LanguageCodeErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.LanguageCodeErrorEnum
 */
exports.LanguageCodeErrorEnum = new LanguageCodeErrorEnum$Type();
