// google-ads-api-client/dist/protos/bindings/ads/errors/list_operation_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListOperationErrorEnum = exports.ListOperationErrorEnum_ListOperationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible list operation errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ListOperationErrorEnum.ListOperationError
 */
var ListOperationErrorEnum_ListOperationError;
(function (ListOperationErrorEnum_ListOperationError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ListOperationErrorEnum_ListOperationError[ListOperationErrorEnum_ListOperationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ListOperationErrorEnum_ListOperationError[ListOperationErrorEnum_ListOperationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Field required in value is missing.
     *
     * @generated from protobuf enum value: REQUIRED_FIELD_MISSING = 7;
     */
    ListOperationErrorEnum_ListOperationError[ListOperationErrorEnum_ListOperationError["REQUIRED_FIELD_MISSING"] = 7] = "REQUIRED_FIELD_MISSING";
    /**
     * Duplicate or identical value is sent in multiple list operations.
     *
     * @generated from protobuf enum value: DUPLICATE_VALUES = 8;
     */
    ListOperationErrorEnum_ListOperationError[ListOperationErrorEnum_ListOperationError["DUPLICATE_VALUES"] = 8] = "DUPLICATE_VALUES";
})(ListOperationErrorEnum_ListOperationError = exports.ListOperationErrorEnum_ListOperationError || (exports.ListOperationErrorEnum_ListOperationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ListOperationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ListOperationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ListOperationErrorEnum
 */
exports.ListOperationErrorEnum = new ListOperationErrorEnum$Type();
