// google-ads-api-client/dist/protos/bindings/ads/errors/media_upload_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaUploadErrorEnum = exports.MediaUploadErrorEnum_MediaUploadError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible media uploading errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.MediaUploadErrorEnum.MediaUploadError
 */
var MediaUploadErrorEnum_MediaUploadError;
(function (MediaUploadErrorEnum_MediaUploadError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The uploaded file is too big.
     *
     * @generated from protobuf enum value: FILE_TOO_BIG = 2;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["FILE_TOO_BIG"] = 2] = "FILE_TOO_BIG";
    /**
     * Image data is unparseable.
     *
     * @generated from protobuf enum value: UNPARSEABLE_IMAGE = 3;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["UNPARSEABLE_IMAGE"] = 3] = "UNPARSEABLE_IMAGE";
    /**
     * Animated images are not allowed.
     *
     * @generated from protobuf enum value: ANIMATED_IMAGE_NOT_ALLOWED = 4;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["ANIMATED_IMAGE_NOT_ALLOWED"] = 4] = "ANIMATED_IMAGE_NOT_ALLOWED";
    /**
     * The image or media bundle format is not allowed.
     *
     * @generated from protobuf enum value: FORMAT_NOT_ALLOWED = 5;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["FORMAT_NOT_ALLOWED"] = 5] = "FORMAT_NOT_ALLOWED";
    /**
     * Cannot reference URL external to the media bundle.
     *
     * @generated from protobuf enum value: EXTERNAL_URL_NOT_ALLOWED = 6;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["EXTERNAL_URL_NOT_ALLOWED"] = 6] = "EXTERNAL_URL_NOT_ALLOWED";
    /**
     * HTML5 ad is trying to reference an asset not in .ZIP file.
     *
     * @generated from protobuf enum value: INVALID_URL_REFERENCE = 7;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["INVALID_URL_REFERENCE"] = 7] = "INVALID_URL_REFERENCE";
    /**
     * The media bundle contains no primary entry.
     *
     * @generated from protobuf enum value: MISSING_PRIMARY_MEDIA_BUNDLE_ENTRY = 8;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["MISSING_PRIMARY_MEDIA_BUNDLE_ENTRY"] = 8] = "MISSING_PRIMARY_MEDIA_BUNDLE_ENTRY";
    /**
     * Animation has disallowed visual effects.
     *
     * @generated from protobuf enum value: ANIMATED_VISUAL_EFFECT = 9;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["ANIMATED_VISUAL_EFFECT"] = 9] = "ANIMATED_VISUAL_EFFECT";
    /**
     * Animation longer than the allowed 30 second limit.
     *
     * @generated from protobuf enum value: ANIMATION_TOO_LONG = 10;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["ANIMATION_TOO_LONG"] = 10] = "ANIMATION_TOO_LONG";
    /**
     * The aspect ratio of the image does not match the expected aspect ratios
     * provided in the asset spec.
     *
     * @generated from protobuf enum value: ASPECT_RATIO_NOT_ALLOWED = 11;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["ASPECT_RATIO_NOT_ALLOWED"] = 11] = "ASPECT_RATIO_NOT_ALLOWED";
    /**
     * Audio files are not allowed in bundle.
     *
     * @generated from protobuf enum value: AUDIO_NOT_ALLOWED_IN_MEDIA_BUNDLE = 12;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["AUDIO_NOT_ALLOWED_IN_MEDIA_BUNDLE"] = 12] = "AUDIO_NOT_ALLOWED_IN_MEDIA_BUNDLE";
    /**
     * CMYK jpegs are not supported.
     *
     * @generated from protobuf enum value: CMYK_JPEG_NOT_ALLOWED = 13;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["CMYK_JPEG_NOT_ALLOWED"] = 13] = "CMYK_JPEG_NOT_ALLOWED";
    /**
     * Flash movies are not allowed.
     *
     * @generated from protobuf enum value: FLASH_NOT_ALLOWED = 14;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["FLASH_NOT_ALLOWED"] = 14] = "FLASH_NOT_ALLOWED";
    /**
     * The frame rate of the video is higher than the allowed 5fps.
     *
     * @generated from protobuf enum value: FRAME_RATE_TOO_HIGH = 15;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["FRAME_RATE_TOO_HIGH"] = 15] = "FRAME_RATE_TOO_HIGH";
    /**
     * ZIP file from Google Web Designer is not published.
     *
     * @generated from protobuf enum value: GOOGLE_WEB_DESIGNER_ZIP_FILE_NOT_PUBLISHED = 16;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["GOOGLE_WEB_DESIGNER_ZIP_FILE_NOT_PUBLISHED"] = 16] = "GOOGLE_WEB_DESIGNER_ZIP_FILE_NOT_PUBLISHED";
    /**
     * Image constraints are violated, but more details (like
     * DIMENSIONS_NOT_ALLOWED or ASPECT_RATIO_NOT_ALLOWED) can not be provided.
     * This happens when asset spec contains more than one constraint and
     * criteria of different constraints are violated.
     *
     * @generated from protobuf enum value: IMAGE_CONSTRAINTS_VIOLATED = 17;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["IMAGE_CONSTRAINTS_VIOLATED"] = 17] = "IMAGE_CONSTRAINTS_VIOLATED";
    /**
     * Media bundle data is unrecognizable.
     *
     * @generated from protobuf enum value: INVALID_MEDIA_BUNDLE = 18;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["INVALID_MEDIA_BUNDLE"] = 18] = "INVALID_MEDIA_BUNDLE";
    /**
     * There was a problem with one or more of the media bundle entries.
     *
     * @generated from protobuf enum value: INVALID_MEDIA_BUNDLE_ENTRY = 19;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["INVALID_MEDIA_BUNDLE_ENTRY"] = 19] = "INVALID_MEDIA_BUNDLE_ENTRY";
    /**
     * The asset has an invalid mime type.
     *
     * @generated from protobuf enum value: INVALID_MIME_TYPE = 20;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["INVALID_MIME_TYPE"] = 20] = "INVALID_MIME_TYPE";
    /**
     * The media bundle contains an invalid asset path.
     *
     * @generated from protobuf enum value: INVALID_PATH = 21;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["INVALID_PATH"] = 21] = "INVALID_PATH";
    /**
     * Image has layout problem.
     *
     * @generated from protobuf enum value: LAYOUT_PROBLEM = 22;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["LAYOUT_PROBLEM"] = 22] = "LAYOUT_PROBLEM";
    /**
     * An asset had a URL reference that is malformed per RFC 1738 convention.
     *
     * @generated from protobuf enum value: MALFORMED_URL = 23;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["MALFORMED_URL"] = 23] = "MALFORMED_URL";
    /**
     * The uploaded media bundle format is not allowed.
     *
     * @generated from protobuf enum value: MEDIA_BUNDLE_NOT_ALLOWED = 24;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["MEDIA_BUNDLE_NOT_ALLOWED"] = 24] = "MEDIA_BUNDLE_NOT_ALLOWED";
    /**
     * The media bundle is not compatible with the asset spec product type.
     * (For example, Gmail, dynamic remarketing, etc.)
     *
     * @generated from protobuf enum value: MEDIA_BUNDLE_NOT_COMPATIBLE_TO_PRODUCT_TYPE = 25;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["MEDIA_BUNDLE_NOT_COMPATIBLE_TO_PRODUCT_TYPE"] = 25] = "MEDIA_BUNDLE_NOT_COMPATIBLE_TO_PRODUCT_TYPE";
    /**
     * A bundle being uploaded that is incompatible with multiple assets for
     * different reasons.
     *
     * @generated from protobuf enum value: MEDIA_BUNDLE_REJECTED_BY_MULTIPLE_ASSET_SPECS = 26;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["MEDIA_BUNDLE_REJECTED_BY_MULTIPLE_ASSET_SPECS"] = 26] = "MEDIA_BUNDLE_REJECTED_BY_MULTIPLE_ASSET_SPECS";
    /**
     * The media bundle contains too many files.
     *
     * @generated from protobuf enum value: TOO_MANY_FILES_IN_MEDIA_BUNDLE = 27;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["TOO_MANY_FILES_IN_MEDIA_BUNDLE"] = 27] = "TOO_MANY_FILES_IN_MEDIA_BUNDLE";
    /**
     * Google Web Designer not created for "Google Ads" environment.
     *
     * @generated from protobuf enum value: UNSUPPORTED_GOOGLE_WEB_DESIGNER_ENVIRONMENT = 28;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["UNSUPPORTED_GOOGLE_WEB_DESIGNER_ENVIRONMENT"] = 28] = "UNSUPPORTED_GOOGLE_WEB_DESIGNER_ENVIRONMENT";
    /**
     * Unsupported HTML5 feature in HTML5 asset.
     *
     * @generated from protobuf enum value: UNSUPPORTED_HTML5_FEATURE = 29;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["UNSUPPORTED_HTML5_FEATURE"] = 29] = "UNSUPPORTED_HTML5_FEATURE";
    /**
     * URL in HTML5 entry is not SSL compliant.
     *
     * @generated from protobuf enum value: URL_IN_MEDIA_BUNDLE_NOT_SSL_COMPLIANT = 30;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["URL_IN_MEDIA_BUNDLE_NOT_SSL_COMPLIANT"] = 30] = "URL_IN_MEDIA_BUNDLE_NOT_SSL_COMPLIANT";
    /**
     * Video file name is longer than the 50 allowed characters.
     *
     * @generated from protobuf enum value: VIDEO_FILE_NAME_TOO_LONG = 31;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["VIDEO_FILE_NAME_TOO_LONG"] = 31] = "VIDEO_FILE_NAME_TOO_LONG";
    /**
     * Multiple videos with same name in a bundle.
     *
     * @generated from protobuf enum value: VIDEO_MULTIPLE_FILES_WITH_SAME_NAME = 32;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["VIDEO_MULTIPLE_FILES_WITH_SAME_NAME"] = 32] = "VIDEO_MULTIPLE_FILES_WITH_SAME_NAME";
    /**
     * Videos are not allowed in media bundle.
     *
     * @generated from protobuf enum value: VIDEO_NOT_ALLOWED_IN_MEDIA_BUNDLE = 33;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["VIDEO_NOT_ALLOWED_IN_MEDIA_BUNDLE"] = 33] = "VIDEO_NOT_ALLOWED_IN_MEDIA_BUNDLE";
    /**
     * This type of media cannot be uploaded through the Google Ads API.
     *
     * @generated from protobuf enum value: CANNOT_UPLOAD_MEDIA_TYPE_THROUGH_API = 34;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["CANNOT_UPLOAD_MEDIA_TYPE_THROUGH_API"] = 34] = "CANNOT_UPLOAD_MEDIA_TYPE_THROUGH_API";
    /**
     * The dimensions of the image are not allowed.
     *
     * @generated from protobuf enum value: DIMENSIONS_NOT_ALLOWED = 35;
     */
    MediaUploadErrorEnum_MediaUploadError[MediaUploadErrorEnum_MediaUploadError["DIMENSIONS_NOT_ALLOWED"] = 35] = "DIMENSIONS_NOT_ALLOWED";
})(MediaUploadErrorEnum_MediaUploadError = exports.MediaUploadErrorEnum_MediaUploadError || (exports.MediaUploadErrorEnum_MediaUploadError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MediaUploadErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.MediaUploadErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.MediaUploadErrorEnum
 */
exports.MediaUploadErrorEnum = new MediaUploadErrorEnum$Type();
