// google-ads-api-client/dist/protos/bindings/ads/errors/media_bundle_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaBundleErrorEnum = exports.MediaBundleErrorEnum_MediaBundleError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible media bundle errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.MediaBundleErrorEnum.MediaBundleError
 */
var MediaBundleErrorEnum_MediaBundleError;
(function (MediaBundleErrorEnum_MediaBundleError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * There was a problem with the request.
     *
     * @generated from protobuf enum value: BAD_REQUEST = 3;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["BAD_REQUEST"] = 3] = "BAD_REQUEST";
    /**
     * HTML5 ads using DoubleClick Studio created ZIP files are not supported.
     *
     * @generated from protobuf enum value: DOUBLECLICK_BUNDLE_NOT_ALLOWED = 4;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["DOUBLECLICK_BUNDLE_NOT_ALLOWED"] = 4] = "DOUBLECLICK_BUNDLE_NOT_ALLOWED";
    /**
     * Cannot reference URL external to the media bundle.
     *
     * @generated from protobuf enum value: EXTERNAL_URL_NOT_ALLOWED = 5;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["EXTERNAL_URL_NOT_ALLOWED"] = 5] = "EXTERNAL_URL_NOT_ALLOWED";
    /**
     * Media bundle file is too large.
     *
     * @generated from protobuf enum value: FILE_TOO_LARGE = 6;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["FILE_TOO_LARGE"] = 6] = "FILE_TOO_LARGE";
    /**
     * ZIP file from Google Web Designer is not published.
     *
     * @generated from protobuf enum value: GOOGLE_WEB_DESIGNER_ZIP_FILE_NOT_PUBLISHED = 7;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["GOOGLE_WEB_DESIGNER_ZIP_FILE_NOT_PUBLISHED"] = 7] = "GOOGLE_WEB_DESIGNER_ZIP_FILE_NOT_PUBLISHED";
    /**
     * Input was invalid.
     *
     * @generated from protobuf enum value: INVALID_INPUT = 8;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["INVALID_INPUT"] = 8] = "INVALID_INPUT";
    /**
     * There was a problem with the media bundle.
     *
     * @generated from protobuf enum value: INVALID_MEDIA_BUNDLE = 9;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["INVALID_MEDIA_BUNDLE"] = 9] = "INVALID_MEDIA_BUNDLE";
    /**
     * There was a problem with one or more of the media bundle entries.
     *
     * @generated from protobuf enum value: INVALID_MEDIA_BUNDLE_ENTRY = 10;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["INVALID_MEDIA_BUNDLE_ENTRY"] = 10] = "INVALID_MEDIA_BUNDLE_ENTRY";
    /**
     * The media bundle contains a file with an unknown mime type
     *
     * @generated from protobuf enum value: INVALID_MIME_TYPE = 11;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["INVALID_MIME_TYPE"] = 11] = "INVALID_MIME_TYPE";
    /**
     * The media bundle contain an invalid asset path.
     *
     * @generated from protobuf enum value: INVALID_PATH = 12;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["INVALID_PATH"] = 12] = "INVALID_PATH";
    /**
     * HTML5 ad is trying to reference an asset not in .ZIP file
     *
     * @generated from protobuf enum value: INVALID_URL_REFERENCE = 13;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["INVALID_URL_REFERENCE"] = 13] = "INVALID_URL_REFERENCE";
    /**
     * Media data is too large.
     *
     * @generated from protobuf enum value: MEDIA_DATA_TOO_LARGE = 14;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["MEDIA_DATA_TOO_LARGE"] = 14] = "MEDIA_DATA_TOO_LARGE";
    /**
     * The media bundle contains no primary entry.
     *
     * @generated from protobuf enum value: MISSING_PRIMARY_MEDIA_BUNDLE_ENTRY = 15;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["MISSING_PRIMARY_MEDIA_BUNDLE_ENTRY"] = 15] = "MISSING_PRIMARY_MEDIA_BUNDLE_ENTRY";
    /**
     * There was an error on the server.
     *
     * @generated from protobuf enum value: SERVER_ERROR = 16;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["SERVER_ERROR"] = 16] = "SERVER_ERROR";
    /**
     * The image could not be stored.
     *
     * @generated from protobuf enum value: STORAGE_ERROR = 17;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["STORAGE_ERROR"] = 17] = "STORAGE_ERROR";
    /**
     * Media bundle created with the Swiffy tool is not allowed.
     *
     * @generated from protobuf enum value: SWIFFY_BUNDLE_NOT_ALLOWED = 18;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["SWIFFY_BUNDLE_NOT_ALLOWED"] = 18] = "SWIFFY_BUNDLE_NOT_ALLOWED";
    /**
     * The media bundle contains too many files.
     *
     * @generated from protobuf enum value: TOO_MANY_FILES = 19;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["TOO_MANY_FILES"] = 19] = "TOO_MANY_FILES";
    /**
     * The media bundle is not of legal dimensions.
     *
     * @generated from protobuf enum value: UNEXPECTED_SIZE = 20;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["UNEXPECTED_SIZE"] = 20] = "UNEXPECTED_SIZE";
    /**
     * Google Web Designer not created for "Google Ads" environment.
     *
     * @generated from protobuf enum value: UNSUPPORTED_GOOGLE_WEB_DESIGNER_ENVIRONMENT = 21;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["UNSUPPORTED_GOOGLE_WEB_DESIGNER_ENVIRONMENT"] = 21] = "UNSUPPORTED_GOOGLE_WEB_DESIGNER_ENVIRONMENT";
    /**
     * Unsupported HTML5 feature in HTML5 asset.
     *
     * @generated from protobuf enum value: UNSUPPORTED_HTML5_FEATURE = 22;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["UNSUPPORTED_HTML5_FEATURE"] = 22] = "UNSUPPORTED_HTML5_FEATURE";
    /**
     * URL in HTML5 entry is not ssl compliant.
     *
     * @generated from protobuf enum value: URL_IN_MEDIA_BUNDLE_NOT_SSL_COMPLIANT = 23;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["URL_IN_MEDIA_BUNDLE_NOT_SSL_COMPLIANT"] = 23] = "URL_IN_MEDIA_BUNDLE_NOT_SSL_COMPLIANT";
    /**
     * Custom exits not allowed in HTML5 entry.
     *
     * @generated from protobuf enum value: CUSTOM_EXIT_NOT_ALLOWED = 24;
     */
    MediaBundleErrorEnum_MediaBundleError[MediaBundleErrorEnum_MediaBundleError["CUSTOM_EXIT_NOT_ALLOWED"] = 24] = "CUSTOM_EXIT_NOT_ALLOWED";
})(MediaBundleErrorEnum_MediaBundleError = exports.MediaBundleErrorEnum_MediaBundleError || (exports.MediaBundleErrorEnum_MediaBundleError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MediaBundleErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.MediaBundleErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.MediaBundleErrorEnum
 */
exports.MediaBundleErrorEnum = new MediaBundleErrorEnum$Type();
