// google-ads-api-client/dist/protos/bindings/ads/errors/multiplier_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiplierErrorEnum = exports.MultiplierErrorEnum_MultiplierError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible multiplier errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.MultiplierErrorEnum.MultiplierError
 */
var MultiplierErrorEnum_MultiplierError;
(function (MultiplierErrorEnum_MultiplierError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Multiplier value is too high
     *
     * @generated from protobuf enum value: MULTIPLIER_TOO_HIGH = 2;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_TOO_HIGH"] = 2] = "MULTIPLIER_TOO_HIGH";
    /**
     * Multiplier value is too low
     *
     * @generated from protobuf enum value: MULTIPLIER_TOO_LOW = 3;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_TOO_LOW"] = 3] = "MULTIPLIER_TOO_LOW";
    /**
     * Too many fractional digits
     *
     * @generated from protobuf enum value: TOO_MANY_FRACTIONAL_DIGITS = 4;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["TOO_MANY_FRACTIONAL_DIGITS"] = 4] = "TOO_MANY_FRACTIONAL_DIGITS";
    /**
     * A multiplier cannot be set for this bidding strategy
     *
     * @generated from protobuf enum value: MULTIPLIER_NOT_ALLOWED_FOR_BIDDING_STRATEGY = 5;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_NOT_ALLOWED_FOR_BIDDING_STRATEGY"] = 5] = "MULTIPLIER_NOT_ALLOWED_FOR_BIDDING_STRATEGY";
    /**
     * A multiplier cannot be set when there is no base bid (for example,
     * content max cpc)
     *
     * @generated from protobuf enum value: MULTIPLIER_NOT_ALLOWED_WHEN_BASE_BID_IS_MISSING = 6;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_NOT_ALLOWED_WHEN_BASE_BID_IS_MISSING"] = 6] = "MULTIPLIER_NOT_ALLOWED_WHEN_BASE_BID_IS_MISSING";
    /**
     * A bid multiplier must be specified
     *
     * @generated from protobuf enum value: NO_MULTIPLIER_SPECIFIED = 7;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["NO_MULTIPLIER_SPECIFIED"] = 7] = "NO_MULTIPLIER_SPECIFIED";
    /**
     * Multiplier causes bid to exceed daily budget
     *
     * @generated from protobuf enum value: MULTIPLIER_CAUSES_BID_TO_EXCEED_DAILY_BUDGET = 8;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_CAUSES_BID_TO_EXCEED_DAILY_BUDGET"] = 8] = "MULTIPLIER_CAUSES_BID_TO_EXCEED_DAILY_BUDGET";
    /**
     * Multiplier causes bid to exceed monthly budget
     *
     * @generated from protobuf enum value: MULTIPLIER_CAUSES_BID_TO_EXCEED_MONTHLY_BUDGET = 9;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_CAUSES_BID_TO_EXCEED_MONTHLY_BUDGET"] = 9] = "MULTIPLIER_CAUSES_BID_TO_EXCEED_MONTHLY_BUDGET";
    /**
     * Multiplier causes bid to exceed custom budget
     *
     * @generated from protobuf enum value: MULTIPLIER_CAUSES_BID_TO_EXCEED_CUSTOM_BUDGET = 10;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_CAUSES_BID_TO_EXCEED_CUSTOM_BUDGET"] = 10] = "MULTIPLIER_CAUSES_BID_TO_EXCEED_CUSTOM_BUDGET";
    /**
     * Multiplier causes bid to exceed maximum allowed bid
     *
     * @generated from protobuf enum value: MULTIPLIER_CAUSES_BID_TO_EXCEED_MAX_ALLOWED_BID = 11;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_CAUSES_BID_TO_EXCEED_MAX_ALLOWED_BID"] = 11] = "MULTIPLIER_CAUSES_BID_TO_EXCEED_MAX_ALLOWED_BID";
    /**
     * Multiplier causes bid to become less than the minimum bid allowed
     *
     * @generated from protobuf enum value: BID_LESS_THAN_MIN_ALLOWED_BID_WITH_MULTIPLIER = 12;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["BID_LESS_THAN_MIN_ALLOWED_BID_WITH_MULTIPLIER"] = 12] = "BID_LESS_THAN_MIN_ALLOWED_BID_WITH_MULTIPLIER";
    /**
     * Multiplier type (cpc versus cpm) needs to match campaign's bidding
     * strategy
     *
     * @generated from protobuf enum value: MULTIPLIER_AND_BIDDING_STRATEGY_TYPE_MISMATCH = 13;
     */
    MultiplierErrorEnum_MultiplierError[MultiplierErrorEnum_MultiplierError["MULTIPLIER_AND_BIDDING_STRATEGY_TYPE_MISMATCH"] = 13] = "MULTIPLIER_AND_BIDDING_STRATEGY_TYPE_MISMATCH";
})(MultiplierErrorEnum_MultiplierError = exports.MultiplierErrorEnum_MultiplierError || (exports.MultiplierErrorEnum_MultiplierError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MultiplierErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.MultiplierErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.MultiplierErrorEnum
 */
exports.MultiplierErrorEnum = new MultiplierErrorEnum$Type();
