// google-ads-api-client/dist/protos/bindings/ads/errors/mutate_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MutateErrorEnum = exports.MutateErrorEnum_MutateError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible mutate errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.MutateErrorEnum.MutateError
 */
var MutateErrorEnum_MutateError;
(function (MutateErrorEnum_MutateError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Requested resource was not found.
     *
     * @generated from protobuf enum value: RESOURCE_NOT_FOUND = 3;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["RESOURCE_NOT_FOUND"] = 3] = "RESOURCE_NOT_FOUND";
    /**
     * Cannot mutate the same resource twice in one request.
     *
     * @generated from protobuf enum value: ID_EXISTS_IN_MULTIPLE_MUTATES = 7;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["ID_EXISTS_IN_MULTIPLE_MUTATES"] = 7] = "ID_EXISTS_IN_MULTIPLE_MUTATES";
    /**
     * The field's contents don't match another field that represents the same
     * data.
     *
     * @generated from protobuf enum value: INCONSISTENT_FIELD_VALUES = 8;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["INCONSISTENT_FIELD_VALUES"] = 8] = "INCONSISTENT_FIELD_VALUES";
    /**
     * Mutates are not allowed for the requested resource.
     *
     * @generated from protobuf enum value: MUTATE_NOT_ALLOWED = 9;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["MUTATE_NOT_ALLOWED"] = 9] = "MUTATE_NOT_ALLOWED";
    /**
     * The resource isn't in Google Ads. It belongs to another ads system.
     *
     * @generated from protobuf enum value: RESOURCE_NOT_IN_GOOGLE_ADS = 10;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["RESOURCE_NOT_IN_GOOGLE_ADS"] = 10] = "RESOURCE_NOT_IN_GOOGLE_ADS";
    /**
     * The resource being created already exists.
     *
     * @generated from protobuf enum value: RESOURCE_ALREADY_EXISTS = 11;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["RESOURCE_ALREADY_EXISTS"] = 11] = "RESOURCE_ALREADY_EXISTS";
    /**
     * This resource cannot be used with "validate_only".
     *
     * @generated from protobuf enum value: RESOURCE_DOES_NOT_SUPPORT_VALIDATE_ONLY = 12;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["RESOURCE_DOES_NOT_SUPPORT_VALIDATE_ONLY"] = 12] = "RESOURCE_DOES_NOT_SUPPORT_VALIDATE_ONLY";
    /**
     * This operation cannot be used with "partial_failure".
     *
     * @generated from protobuf enum value: OPERATION_DOES_NOT_SUPPORT_PARTIAL_FAILURE = 16;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["OPERATION_DOES_NOT_SUPPORT_PARTIAL_FAILURE"] = 16] = "OPERATION_DOES_NOT_SUPPORT_PARTIAL_FAILURE";
    /**
     * Attempt to write to read-only fields.
     *
     * @generated from protobuf enum value: RESOURCE_READ_ONLY = 13;
     */
    MutateErrorEnum_MutateError[MutateErrorEnum_MutateError["RESOURCE_READ_ONLY"] = 13] = "RESOURCE_READ_ONLY";
})(MutateErrorEnum_MutateError = exports.MutateErrorEnum_MutateError || (exports.MutateErrorEnum_MutateError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class MutateErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.MutateErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.MutateErrorEnum
 */
exports.MutateErrorEnum = new MutateErrorEnum$Type();
