// google-ads-api-client/dist/protos/bindings/ads/errors/new_resource_creation_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewResourceCreationErrorEnum = exports.NewResourceCreationErrorEnum_NewResourceCreationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible new resource creation errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.NewResourceCreationErrorEnum.NewResourceCreationError
 */
var NewResourceCreationErrorEnum_NewResourceCreationError;
(function (NewResourceCreationErrorEnum_NewResourceCreationError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    NewResourceCreationErrorEnum_NewResourceCreationError[NewResourceCreationErrorEnum_NewResourceCreationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    NewResourceCreationErrorEnum_NewResourceCreationError[NewResourceCreationErrorEnum_NewResourceCreationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Do not set the id field while creating new resources.
     *
     * @generated from protobuf enum value: CANNOT_SET_ID_FOR_CREATE = 2;
     */
    NewResourceCreationErrorEnum_NewResourceCreationError[NewResourceCreationErrorEnum_NewResourceCreationError["CANNOT_SET_ID_FOR_CREATE"] = 2] = "CANNOT_SET_ID_FOR_CREATE";
    /**
     * Creating more than one resource with the same temp ID is not allowed.
     *
     * @generated from protobuf enum value: DUPLICATE_TEMP_IDS = 3;
     */
    NewResourceCreationErrorEnum_NewResourceCreationError[NewResourceCreationErrorEnum_NewResourceCreationError["DUPLICATE_TEMP_IDS"] = 3] = "DUPLICATE_TEMP_IDS";
    /**
     * Parent resource with specified temp ID failed validation, so no
     * validation will be done for this child resource.
     *
     * @generated from protobuf enum value: TEMP_ID_RESOURCE_HAD_ERRORS = 4;
     */
    NewResourceCreationErrorEnum_NewResourceCreationError[NewResourceCreationErrorEnum_NewResourceCreationError["TEMP_ID_RESOURCE_HAD_ERRORS"] = 4] = "TEMP_ID_RESOURCE_HAD_ERRORS";
})(NewResourceCreationErrorEnum_NewResourceCreationError = exports.NewResourceCreationErrorEnum_NewResourceCreationError || (exports.NewResourceCreationErrorEnum_NewResourceCreationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class NewResourceCreationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.NewResourceCreationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.NewResourceCreationErrorEnum
 */
exports.NewResourceCreationErrorEnum = new NewResourceCreationErrorEnum$Type();
