// google-ads-api-client/dist/protos/bindings/ads/errors/not_allowlisted_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotAllowlistedErrorEnum = exports.NotAllowlistedErrorEnum_NotAllowlistedError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible not allowlisted errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.NotAllowlistedErrorEnum.NotAllowlistedError
 */
var NotAllowlistedErrorEnum_NotAllowlistedError;
(function (NotAllowlistedErrorEnum_NotAllowlistedError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    NotAllowlistedErrorEnum_NotAllowlistedError[NotAllowlistedErrorEnum_NotAllowlistedError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    NotAllowlistedErrorEnum_NotAllowlistedError[NotAllowlistedErrorEnum_NotAllowlistedError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Customer is not allowlisted for accessing this feature.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_ALLOWLISTED_FOR_THIS_FEATURE = 2;
     */
    NotAllowlistedErrorEnum_NotAllowlistedError[NotAllowlistedErrorEnum_NotAllowlistedError["CUSTOMER_NOT_ALLOWLISTED_FOR_THIS_FEATURE"] = 2] = "CUSTOMER_NOT_ALLOWLISTED_FOR_THIS_FEATURE";
})(NotAllowlistedErrorEnum_NotAllowlistedError = exports.NotAllowlistedErrorEnum_NotAllowlistedError || (exports.NotAllowlistedErrorEnum_NotAllowlistedError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class NotAllowlistedErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.NotAllowlistedErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.NotAllowlistedErrorEnum
 */
exports.NotAllowlistedErrorEnum = new NotAllowlistedErrorEnum$Type();
