// google-ads-api-client/dist/protos/bindings/ads/errors/not_empty_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotEmptyErrorEnum = exports.NotEmptyErrorEnum_NotEmptyError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible not empty errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.NotEmptyErrorEnum.NotEmptyError
 */
var NotEmptyErrorEnum_NotEmptyError;
(function (NotEmptyErrorEnum_NotEmptyError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    NotEmptyErrorEnum_NotEmptyError[NotEmptyErrorEnum_NotEmptyError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    NotEmptyErrorEnum_NotEmptyError[NotEmptyErrorEnum_NotEmptyError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Empty list.
     *
     * @generated from protobuf enum value: EMPTY_LIST = 2;
     */
    NotEmptyErrorEnum_NotEmptyError[NotEmptyErrorEnum_NotEmptyError["EMPTY_LIST"] = 2] = "EMPTY_LIST";
})(NotEmptyErrorEnum_NotEmptyError = exports.NotEmptyErrorEnum_NotEmptyError || (exports.NotEmptyErrorEnum_NotEmptyError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class NotEmptyErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.NotEmptyErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.NotEmptyErrorEnum
 */
exports.NotEmptyErrorEnum = new NotEmptyErrorEnum$Type();
