// google-ads-api-client/dist/protos/bindings/ads/errors/null_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NullErrorEnum = exports.NullErrorEnum_NullError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible null errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.NullErrorEnum.NullError
 */
var NullErrorEnum_NullError;
(function (NullErrorEnum_NullError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    NullErrorEnum_NullError[NullErrorEnum_NullError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    NullErrorEnum_NullError[NullErrorEnum_NullError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Specified list/container must not contain any null elements
     *
     * @generated from protobuf enum value: NULL_CONTENT = 2;
     */
    NullErrorEnum_NullError[NullErrorEnum_NullError["NULL_CONTENT"] = 2] = "NULL_CONTENT";
})(NullErrorEnum_NullError = exports.NullErrorEnum_NullError || (exports.NullErrorEnum_NullError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class NullErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.NullErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.NullErrorEnum
 */
exports.NullErrorEnum = new NullErrorEnum$Type();
