// google-ads-api-client/dist/protos/bindings/ads/errors/offline_user_data_job_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OfflineUserDataJobErrorEnum = exports.OfflineUserDataJobErrorEnum_OfflineUserDataJobError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible request errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.OfflineUserDataJobErrorEnum.OfflineUserDataJobError
 */
var OfflineUserDataJobErrorEnum_OfflineUserDataJobError;
(function (OfflineUserDataJobErrorEnum_OfflineUserDataJobError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The user list ID provided for the job is invalid.
     *
     * @generated from protobuf enum value: INVALID_USER_LIST_ID = 3;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_USER_LIST_ID"] = 3] = "INVALID_USER_LIST_ID";
    /**
     * Type of the user list is not applicable for the job.
     *
     * @generated from protobuf enum value: INVALID_USER_LIST_TYPE = 4;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_USER_LIST_TYPE"] = 4] = "INVALID_USER_LIST_TYPE";
    /**
     * Customer is not allowisted for using user ID in upload data.
     *
     * @generated from protobuf enum value: NOT_ON_ALLOWLIST_FOR_USER_ID = 33;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["NOT_ON_ALLOWLIST_FOR_USER_ID"] = 33] = "NOT_ON_ALLOWLIST_FOR_USER_ID";
    /**
     * Upload data is not compatible with the upload key type of the associated
     * user list.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_UPLOAD_KEY_TYPE = 6;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INCOMPATIBLE_UPLOAD_KEY_TYPE"] = 6] = "INCOMPATIBLE_UPLOAD_KEY_TYPE";
    /**
     * The user identifier is missing valid data.
     *
     * @generated from protobuf enum value: MISSING_USER_IDENTIFIER = 7;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["MISSING_USER_IDENTIFIER"] = 7] = "MISSING_USER_IDENTIFIER";
    /**
     * The mobile ID is malformed.
     *
     * @generated from protobuf enum value: INVALID_MOBILE_ID_FORMAT = 8;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_MOBILE_ID_FORMAT"] = 8] = "INVALID_MOBILE_ID_FORMAT";
    /**
     * Maximum number of user identifiers allowed per request is 100,000 and per
     * operation is 20.
     *
     * @generated from protobuf enum value: TOO_MANY_USER_IDENTIFIERS = 9;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["TOO_MANY_USER_IDENTIFIERS"] = 9] = "TOO_MANY_USER_IDENTIFIERS";
    /**
     * Customer is not on the allow-list for store sales direct data.
     *
     * @generated from protobuf enum value: NOT_ON_ALLOWLIST_FOR_STORE_SALES_DIRECT = 31;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["NOT_ON_ALLOWLIST_FOR_STORE_SALES_DIRECT"] = 31] = "NOT_ON_ALLOWLIST_FOR_STORE_SALES_DIRECT";
    /**
     * Customer is not on the allow-list for unified store sales data.
     *
     * @generated from protobuf enum value: NOT_ON_ALLOWLIST_FOR_UNIFIED_STORE_SALES = 32;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["NOT_ON_ALLOWLIST_FOR_UNIFIED_STORE_SALES"] = 32] = "NOT_ON_ALLOWLIST_FOR_UNIFIED_STORE_SALES";
    /**
     * The partner ID in store sales direct metadata is invalid.
     *
     * @generated from protobuf enum value: INVALID_PARTNER_ID = 11;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_PARTNER_ID"] = 11] = "INVALID_PARTNER_ID";
    /**
     * The data in user identifier should not be encoded.
     *
     * @generated from protobuf enum value: INVALID_ENCODING = 12;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_ENCODING"] = 12] = "INVALID_ENCODING";
    /**
     * The country code is invalid.
     *
     * @generated from protobuf enum value: INVALID_COUNTRY_CODE = 13;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_COUNTRY_CODE"] = 13] = "INVALID_COUNTRY_CODE";
    /**
     * Incompatible user identifier when using third_party_user_id for store
     * sales direct first party data or not using third_party_user_id for store
     * sales third party data.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_USER_IDENTIFIER = 14;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INCOMPATIBLE_USER_IDENTIFIER"] = 14] = "INCOMPATIBLE_USER_IDENTIFIER";
    /**
     * A transaction time in the future is not allowed.
     *
     * @generated from protobuf enum value: FUTURE_TRANSACTION_TIME = 15;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["FUTURE_TRANSACTION_TIME"] = 15] = "FUTURE_TRANSACTION_TIME";
    /**
     * The conversion_action specified in transaction_attributes is used to
     * report conversions to a conversion action configured in Google Ads. This
     * error indicates there is no such conversion action in the account.
     *
     * @generated from protobuf enum value: INVALID_CONVERSION_ACTION = 16;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_CONVERSION_ACTION"] = 16] = "INVALID_CONVERSION_ACTION";
    /**
     * Mobile ID is not supported for store sales direct data.
     *
     * @generated from protobuf enum value: MOBILE_ID_NOT_SUPPORTED = 17;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["MOBILE_ID_NOT_SUPPORTED"] = 17] = "MOBILE_ID_NOT_SUPPORTED";
    /**
     * When a remove-all operation is provided, it has to be the first operation
     * of the operation list.
     *
     * @generated from protobuf enum value: INVALID_OPERATION_ORDER = 18;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_OPERATION_ORDER"] = 18] = "INVALID_OPERATION_ORDER";
    /**
     * Mixing creation and removal of offline data in the same job is not
     * allowed.
     *
     * @generated from protobuf enum value: CONFLICTING_OPERATION = 19;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["CONFLICTING_OPERATION"] = 19] = "CONFLICTING_OPERATION";
    /**
     * The external update ID already exists.
     *
     * @generated from protobuf enum value: EXTERNAL_UPDATE_ID_ALREADY_EXISTS = 21;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["EXTERNAL_UPDATE_ID_ALREADY_EXISTS"] = 21] = "EXTERNAL_UPDATE_ID_ALREADY_EXISTS";
    /**
     * Once the upload job is started, new operations cannot be added.
     *
     * @generated from protobuf enum value: JOB_ALREADY_STARTED = 22;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["JOB_ALREADY_STARTED"] = 22] = "JOB_ALREADY_STARTED";
    /**
     * Remove operation is not allowed for store sales direct updates.
     *
     * @generated from protobuf enum value: REMOVE_NOT_SUPPORTED = 23;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["REMOVE_NOT_SUPPORTED"] = 23] = "REMOVE_NOT_SUPPORTED";
    /**
     * Remove-all is not supported for certain offline user data job types.
     *
     * @generated from protobuf enum value: REMOVE_ALL_NOT_SUPPORTED = 24;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["REMOVE_ALL_NOT_SUPPORTED"] = 24] = "REMOVE_ALL_NOT_SUPPORTED";
    /**
     * The SHA256 encoded value is malformed.
     *
     * @generated from protobuf enum value: INVALID_SHA256_FORMAT = 25;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_SHA256_FORMAT"] = 25] = "INVALID_SHA256_FORMAT";
    /**
     * The custom key specified is not enabled for the unified store sales
     * upload.
     *
     * @generated from protobuf enum value: CUSTOM_KEY_DISABLED = 26;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["CUSTOM_KEY_DISABLED"] = 26] = "CUSTOM_KEY_DISABLED";
    /**
     * The custom key specified is not predefined through the Google Ads UI.
     *
     * @generated from protobuf enum value: CUSTOM_KEY_NOT_PREDEFINED = 27;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["CUSTOM_KEY_NOT_PREDEFINED"] = 27] = "CUSTOM_KEY_NOT_PREDEFINED";
    /**
     * The custom key specified is not set in the upload.
     *
     * @generated from protobuf enum value: CUSTOM_KEY_NOT_SET = 29;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["CUSTOM_KEY_NOT_SET"] = 29] = "CUSTOM_KEY_NOT_SET";
    /**
     * The customer has not accepted the customer data terms in the conversion
     * settings page.
     *
     * @generated from protobuf enum value: CUSTOMER_NOT_ACCEPTED_CUSTOMER_DATA_TERMS = 30;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["CUSTOMER_NOT_ACCEPTED_CUSTOMER_DATA_TERMS"] = 30] = "CUSTOMER_NOT_ACCEPTED_CUSTOMER_DATA_TERMS";
    /**
     * User attributes cannot be uploaded into a user list.
     *
     * @generated from protobuf enum value: ATTRIBUTES_NOT_APPLICABLE_FOR_CUSTOMER_MATCH_USER_LIST = 34;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["ATTRIBUTES_NOT_APPLICABLE_FOR_CUSTOMER_MATCH_USER_LIST"] = 34] = "ATTRIBUTES_NOT_APPLICABLE_FOR_CUSTOMER_MATCH_USER_LIST";
    /**
     * Lifetime bucket value must be a number from 0 to 10; 0 is only accepted
     * for remove operations
     *
     * @generated from protobuf enum value: LIFETIME_VALUE_BUCKET_NOT_IN_RANGE = 35;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["LIFETIME_VALUE_BUCKET_NOT_IN_RANGE"] = 35] = "LIFETIME_VALUE_BUCKET_NOT_IN_RANGE";
    /**
     * Identifiers not supported for Customer Match attributes. User attributes
     * can only be provided with contact info (email, phone, address) user
     * identifiers.
     *
     * @generated from protobuf enum value: INCOMPATIBLE_USER_IDENTIFIER_FOR_ATTRIBUTES = 36;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INCOMPATIBLE_USER_IDENTIFIER_FOR_ATTRIBUTES"] = 36] = "INCOMPATIBLE_USER_IDENTIFIER_FOR_ATTRIBUTES";
    /**
     * A time in the future is not allowed.
     *
     * @generated from protobuf enum value: FUTURE_TIME_NOT_ALLOWED = 37;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["FUTURE_TIME_NOT_ALLOWED"] = 37] = "FUTURE_TIME_NOT_ALLOWED";
    /**
     * Last purchase date time cannot be less than acquisition date time.
     *
     * @generated from protobuf enum value: LAST_PURCHASE_TIME_LESS_THAN_ACQUISITION_TIME = 38;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["LAST_PURCHASE_TIME_LESS_THAN_ACQUISITION_TIME"] = 38] = "LAST_PURCHASE_TIME_LESS_THAN_ACQUISITION_TIME";
    /**
     * Only emails are accepted as user identifiers for shopping loyalty match.
     * {-- api.dev/not-precedent: The identifier is not limited to ids, but
     * also include other user info eg. phone numbers.}
     *
     * @generated from protobuf enum value: CUSTOMER_IDENTIFIER_NOT_ALLOWED = 39;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["CUSTOMER_IDENTIFIER_NOT_ALLOWED"] = 39] = "CUSTOMER_IDENTIFIER_NOT_ALLOWED";
    /**
     * Provided item ID is invalid.
     *
     * @generated from protobuf enum value: INVALID_ITEM_ID = 40;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_ITEM_ID"] = 40] = "INVALID_ITEM_ID";
    /**
     * First purchase date time cannot be greater than the last purchase date
     * time.
     *
     * @generated from protobuf enum value: FIRST_PURCHASE_TIME_GREATER_THAN_LAST_PURCHASE_TIME = 42;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["FIRST_PURCHASE_TIME_GREATER_THAN_LAST_PURCHASE_TIME"] = 42] = "FIRST_PURCHASE_TIME_GREATER_THAN_LAST_PURCHASE_TIME";
    /**
     * Provided lifecycle stage is invalid.
     *
     * @generated from protobuf enum value: INVALID_LIFECYCLE_STAGE = 43;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_LIFECYCLE_STAGE"] = 43] = "INVALID_LIFECYCLE_STAGE";
    /**
     * The event value of the Customer Match user attribute is invalid.
     *
     * @generated from protobuf enum value: INVALID_EVENT_VALUE = 44;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["INVALID_EVENT_VALUE"] = 44] = "INVALID_EVENT_VALUE";
    /**
     * All the fields are not present in the EventAttribute of the Customer
     * Match.
     *
     * @generated from protobuf enum value: EVENT_ATTRIBUTE_ALL_FIELDS_ARE_REQUIRED = 45;
     */
    OfflineUserDataJobErrorEnum_OfflineUserDataJobError[OfflineUserDataJobErrorEnum_OfflineUserDataJobError["EVENT_ATTRIBUTE_ALL_FIELDS_ARE_REQUIRED"] = 45] = "EVENT_ATTRIBUTE_ALL_FIELDS_ARE_REQUIRED";
})(OfflineUserDataJobErrorEnum_OfflineUserDataJobError = exports.OfflineUserDataJobErrorEnum_OfflineUserDataJobError || (exports.OfflineUserDataJobErrorEnum_OfflineUserDataJobError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OfflineUserDataJobErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.OfflineUserDataJobErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.OfflineUserDataJobErrorEnum
 */
exports.OfflineUserDataJobErrorEnum = new OfflineUserDataJobErrorEnum$Type();
