// google-ads-api-client/dist/protos/bindings/ads/errors/operator_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperatorErrorEnum = exports.OperatorErrorEnum_OperatorError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible operator errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.OperatorErrorEnum.OperatorError
 */
var OperatorErrorEnum_OperatorError;
(function (OperatorErrorEnum_OperatorError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OperatorErrorEnum_OperatorError[OperatorErrorEnum_OperatorError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OperatorErrorEnum_OperatorError[OperatorErrorEnum_OperatorError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Operator not supported.
     *
     * @generated from protobuf enum value: OPERATOR_NOT_SUPPORTED = 2;
     */
    OperatorErrorEnum_OperatorError[OperatorErrorEnum_OperatorError["OPERATOR_NOT_SUPPORTED"] = 2] = "OPERATOR_NOT_SUPPORTED";
})(OperatorErrorEnum_OperatorError = exports.OperatorErrorEnum_OperatorError || (exports.OperatorErrorEnum_OperatorError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OperatorErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.OperatorErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.OperatorErrorEnum
 */
exports.OperatorErrorEnum = new OperatorErrorEnum$Type();
