// google-ads-api-client/dist/protos/bindings/ads/errors/operation_access_denied_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationAccessDeniedErrorEnum = exports.OperationAccessDeniedErrorEnum_OperationAccessDeniedError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible operation access denied errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.OperationAccessDeniedErrorEnum.OperationAccessDeniedError
 */
var OperationAccessDeniedErrorEnum_OperationAccessDeniedError;
(function (OperationAccessDeniedErrorEnum_OperationAccessDeniedError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Unauthorized invocation of a service's method (get, mutate, etc.)
     *
     * @generated from protobuf enum value: ACTION_NOT_PERMITTED = 2;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["ACTION_NOT_PERMITTED"] = 2] = "ACTION_NOT_PERMITTED";
    /**
     * Unauthorized CREATE operation in invoking a service's mutate method.
     *
     * @generated from protobuf enum value: CREATE_OPERATION_NOT_PERMITTED = 3;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["CREATE_OPERATION_NOT_PERMITTED"] = 3] = "CREATE_OPERATION_NOT_PERMITTED";
    /**
     * Unauthorized REMOVE operation in invoking a service's mutate method.
     *
     * @generated from protobuf enum value: REMOVE_OPERATION_NOT_PERMITTED = 4;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["REMOVE_OPERATION_NOT_PERMITTED"] = 4] = "REMOVE_OPERATION_NOT_PERMITTED";
    /**
     * Unauthorized UPDATE operation in invoking a service's mutate method.
     *
     * @generated from protobuf enum value: UPDATE_OPERATION_NOT_PERMITTED = 5;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["UPDATE_OPERATION_NOT_PERMITTED"] = 5] = "UPDATE_OPERATION_NOT_PERMITTED";
    /**
     * A mutate action is not allowed on this resource, from this client.
     *
     * @generated from protobuf enum value: MUTATE_ACTION_NOT_PERMITTED_FOR_CLIENT = 6;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["MUTATE_ACTION_NOT_PERMITTED_FOR_CLIENT"] = 6] = "MUTATE_ACTION_NOT_PERMITTED_FOR_CLIENT";
    /**
     * This operation is not permitted on this campaign type
     *
     * @generated from protobuf enum value: OPERATION_NOT_PERMITTED_FOR_CAMPAIGN_TYPE = 7;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["OPERATION_NOT_PERMITTED_FOR_CAMPAIGN_TYPE"] = 7] = "OPERATION_NOT_PERMITTED_FOR_CAMPAIGN_TYPE";
    /**
     * A CREATE operation may not set status to REMOVED.
     *
     * @generated from protobuf enum value: CREATE_AS_REMOVED_NOT_PERMITTED = 8;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["CREATE_AS_REMOVED_NOT_PERMITTED"] = 8] = "CREATE_AS_REMOVED_NOT_PERMITTED";
    /**
     * This operation is not allowed because the resource is removed.
     *
     * @generated from protobuf enum value: OPERATION_NOT_PERMITTED_FOR_REMOVED_RESOURCE = 9;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["OPERATION_NOT_PERMITTED_FOR_REMOVED_RESOURCE"] = 9] = "OPERATION_NOT_PERMITTED_FOR_REMOVED_RESOURCE";
    /**
     * This operation is not permitted on this ad group type.
     *
     * @generated from protobuf enum value: OPERATION_NOT_PERMITTED_FOR_AD_GROUP_TYPE = 10;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["OPERATION_NOT_PERMITTED_FOR_AD_GROUP_TYPE"] = 10] = "OPERATION_NOT_PERMITTED_FOR_AD_GROUP_TYPE";
    /**
     * The mutate is not allowed for this customer.
     *
     * @generated from protobuf enum value: MUTATE_NOT_PERMITTED_FOR_CUSTOMER = 11;
     */
    OperationAccessDeniedErrorEnum_OperationAccessDeniedError[OperationAccessDeniedErrorEnum_OperationAccessDeniedError["MUTATE_NOT_PERMITTED_FOR_CUSTOMER"] = 11] = "MUTATE_NOT_PERMITTED_FOR_CUSTOMER";
})(OperationAccessDeniedErrorEnum_OperationAccessDeniedError = exports.OperationAccessDeniedErrorEnum_OperationAccessDeniedError || (exports.OperationAccessDeniedErrorEnum_OperationAccessDeniedError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class OperationAccessDeniedErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.OperationAccessDeniedErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.OperationAccessDeniedErrorEnum
 */
exports.OperationAccessDeniedErrorEnum = new OperationAccessDeniedErrorEnum$Type();
