// google-ads-api-client/dist/protos/bindings/ads/errors/partial_failure_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartialFailureErrorEnum = exports.PartialFailureErrorEnum_PartialFailureError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible partial failure errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.PartialFailureErrorEnum.PartialFailureError
 */
var PartialFailureErrorEnum_PartialFailureError;
(function (PartialFailureErrorEnum_PartialFailureError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PartialFailureErrorEnum_PartialFailureError[PartialFailureErrorEnum_PartialFailureError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PartialFailureErrorEnum_PartialFailureError[PartialFailureErrorEnum_PartialFailureError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The partial failure field was false in the request.
     * This method requires this field be set to true.
     *
     * @generated from protobuf enum value: PARTIAL_FAILURE_MODE_REQUIRED = 2;
     */
    PartialFailureErrorEnum_PartialFailureError[PartialFailureErrorEnum_PartialFailureError["PARTIAL_FAILURE_MODE_REQUIRED"] = 2] = "PARTIAL_FAILURE_MODE_REQUIRED";
})(PartialFailureErrorEnum_PartialFailureError = exports.PartialFailureErrorEnum_PartialFailureError || (exports.PartialFailureErrorEnum_PartialFailureError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PartialFailureErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.PartialFailureErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.PartialFailureErrorEnum
 */
exports.PartialFailureErrorEnum = new PartialFailureErrorEnum$Type();
