// google-ads-api-client/dist/protos/bindings/ads/errors/policy_violation_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyViolationErrorEnum = exports.PolicyViolationErrorEnum_PolicyViolationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible policy violation errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.PolicyViolationErrorEnum.PolicyViolationError
 */
var PolicyViolationErrorEnum_PolicyViolationError;
(function (PolicyViolationErrorEnum_PolicyViolationError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PolicyViolationErrorEnum_PolicyViolationError[PolicyViolationErrorEnum_PolicyViolationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PolicyViolationErrorEnum_PolicyViolationError[PolicyViolationErrorEnum_PolicyViolationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * A policy was violated. See PolicyViolationDetails for more detail.
     *
     * @generated from protobuf enum value: POLICY_ERROR = 2;
     */
    PolicyViolationErrorEnum_PolicyViolationError[PolicyViolationErrorEnum_PolicyViolationError["POLICY_ERROR"] = 2] = "POLICY_ERROR";
})(PolicyViolationErrorEnum_PolicyViolationError = exports.PolicyViolationErrorEnum_PolicyViolationError || (exports.PolicyViolationErrorEnum_PolicyViolationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PolicyViolationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.PolicyViolationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.PolicyViolationErrorEnum
 */
exports.PolicyViolationErrorEnum = new PolicyViolationErrorEnum$Type();
