// google-ads-api-client/dist/protos/bindings/ads/errors/policy_finding_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyFindingErrorEnum = exports.PolicyFindingErrorEnum_PolicyFindingError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible policy finding errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.PolicyFindingErrorEnum.PolicyFindingError
 */
var PolicyFindingErrorEnum_PolicyFindingError;
(function (PolicyFindingErrorEnum_PolicyFindingError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PolicyFindingErrorEnum_PolicyFindingError[PolicyFindingErrorEnum_PolicyFindingError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PolicyFindingErrorEnum_PolicyFindingError[PolicyFindingErrorEnum_PolicyFindingError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The resource has been disapproved since the policy summary includes
     * policy topics of type PROHIBITED.
     *
     * @generated from protobuf enum value: POLICY_FINDING = 2;
     */
    PolicyFindingErrorEnum_PolicyFindingError[PolicyFindingErrorEnum_PolicyFindingError["POLICY_FINDING"] = 2] = "POLICY_FINDING";
    /**
     * The given policy topic does not exist.
     *
     * @generated from protobuf enum value: POLICY_TOPIC_NOT_FOUND = 3;
     */
    PolicyFindingErrorEnum_PolicyFindingError[PolicyFindingErrorEnum_PolicyFindingError["POLICY_TOPIC_NOT_FOUND"] = 3] = "POLICY_TOPIC_NOT_FOUND";
})(PolicyFindingErrorEnum_PolicyFindingError = exports.PolicyFindingErrorEnum_PolicyFindingError || (exports.PolicyFindingErrorEnum_PolicyFindingError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PolicyFindingErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.PolicyFindingErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.PolicyFindingErrorEnum
 */
exports.PolicyFindingErrorEnum = new PolicyFindingErrorEnum$Type();
