// google-ads-api-client/dist/protos/bindings/ads/errors/policy_validation_parameter_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyValidationParameterErrorEnum = exports.PolicyValidationParameterErrorEnum_PolicyValidationParameterError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible policy validation parameter errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.PolicyValidationParameterErrorEnum.PolicyValidationParameterError
 */
var PolicyValidationParameterErrorEnum_PolicyValidationParameterError;
(function (PolicyValidationParameterErrorEnum_PolicyValidationParameterError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    PolicyValidationParameterErrorEnum_PolicyValidationParameterError[PolicyValidationParameterErrorEnum_PolicyValidationParameterError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    PolicyValidationParameterErrorEnum_PolicyValidationParameterError[PolicyValidationParameterErrorEnum_PolicyValidationParameterError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Ignorable policy topics are not supported for the ad type.
     *
     * @generated from protobuf enum value: UNSUPPORTED_AD_TYPE_FOR_IGNORABLE_POLICY_TOPICS = 2;
     */
    PolicyValidationParameterErrorEnum_PolicyValidationParameterError[PolicyValidationParameterErrorEnum_PolicyValidationParameterError["UNSUPPORTED_AD_TYPE_FOR_IGNORABLE_POLICY_TOPICS"] = 2] = "UNSUPPORTED_AD_TYPE_FOR_IGNORABLE_POLICY_TOPICS";
    /**
     * Exempt policy violation keys are not supported for the ad type.
     *
     * @generated from protobuf enum value: UNSUPPORTED_AD_TYPE_FOR_EXEMPT_POLICY_VIOLATION_KEYS = 3;
     */
    PolicyValidationParameterErrorEnum_PolicyValidationParameterError[PolicyValidationParameterErrorEnum_PolicyValidationParameterError["UNSUPPORTED_AD_TYPE_FOR_EXEMPT_POLICY_VIOLATION_KEYS"] = 3] = "UNSUPPORTED_AD_TYPE_FOR_EXEMPT_POLICY_VIOLATION_KEYS";
    /**
     * Cannot set ignorable policy topics and exempt policy violation keys in
     * the same policy violation parameter.
     *
     * @generated from protobuf enum value: CANNOT_SET_BOTH_IGNORABLE_POLICY_TOPICS_AND_EXEMPT_POLICY_VIOLATION_KEYS = 4;
     */
    PolicyValidationParameterErrorEnum_PolicyValidationParameterError[PolicyValidationParameterErrorEnum_PolicyValidationParameterError["CANNOT_SET_BOTH_IGNORABLE_POLICY_TOPICS_AND_EXEMPT_POLICY_VIOLATION_KEYS"] = 4] = "CANNOT_SET_BOTH_IGNORABLE_POLICY_TOPICS_AND_EXEMPT_POLICY_VIOLATION_KEYS";
})(PolicyValidationParameterErrorEnum_PolicyValidationParameterError = exports.PolicyValidationParameterErrorEnum_PolicyValidationParameterError || (exports.PolicyValidationParameterErrorEnum_PolicyValidationParameterError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class PolicyValidationParameterErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.PolicyValidationParameterErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.PolicyValidationParameterErrorEnum
 */
exports.PolicyValidationParameterErrorEnum = new PolicyValidationParameterErrorEnum$Type();
