// google-ads-api-client/dist/protos/bindings/ads/errors/quota_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QuotaErrorEnum = exports.QuotaErrorEnum_QuotaError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible quota errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.QuotaErrorEnum.QuotaError
 */
var QuotaErrorEnum_QuotaError;
(function (QuotaErrorEnum_QuotaError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    QuotaErrorEnum_QuotaError[QuotaErrorEnum_QuotaError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    QuotaErrorEnum_QuotaError[QuotaErrorEnum_QuotaError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Too many requests.
     *
     * @generated from protobuf enum value: RESOURCE_EXHAUSTED = 2;
     */
    QuotaErrorEnum_QuotaError[QuotaErrorEnum_QuotaError["RESOURCE_EXHAUSTED"] = 2] = "RESOURCE_EXHAUSTED";
    /**
     * Access is prohibited.
     *
     * @generated from protobuf enum value: ACCESS_PROHIBITED = 3;
     */
    QuotaErrorEnum_QuotaError[QuotaErrorEnum_QuotaError["ACCESS_PROHIBITED"] = 3] = "ACCESS_PROHIBITED";
    /**
     * Too many requests in a short amount of time.
     *
     * @generated from protobuf enum value: RESOURCE_TEMPORARILY_EXHAUSTED = 4;
     */
    QuotaErrorEnum_QuotaError[QuotaErrorEnum_QuotaError["RESOURCE_TEMPORARILY_EXHAUSTED"] = 4] = "RESOURCE_TEMPORARILY_EXHAUSTED";
})(QuotaErrorEnum_QuotaError = exports.QuotaErrorEnum_QuotaError || (exports.QuotaErrorEnum_QuotaError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class QuotaErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.QuotaErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.QuotaErrorEnum
 */
exports.QuotaErrorEnum = new QuotaErrorEnum$Type();
