// google-ads-api-client/dist/protos/bindings/ads/errors/range_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RangeErrorEnum = exports.RangeErrorEnum_RangeError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible range errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.RangeErrorEnum.RangeError
 */
var RangeErrorEnum_RangeError;
(function (RangeErrorEnum_RangeError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    RangeErrorEnum_RangeError[RangeErrorEnum_RangeError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    RangeErrorEnum_RangeError[RangeErrorEnum_RangeError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Too low.
     *
     * @generated from protobuf enum value: TOO_LOW = 2;
     */
    RangeErrorEnum_RangeError[RangeErrorEnum_RangeError["TOO_LOW"] = 2] = "TOO_LOW";
    /**
     * Too high.
     *
     * @generated from protobuf enum value: TOO_HIGH = 3;
     */
    RangeErrorEnum_RangeError[RangeErrorEnum_RangeError["TOO_HIGH"] = 3] = "TOO_HIGH";
})(RangeErrorEnum_RangeError = exports.RangeErrorEnum_RangeError || (exports.RangeErrorEnum_RangeError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class RangeErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.RangeErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.RangeErrorEnum
 */
exports.RangeErrorEnum = new RangeErrorEnum$Type();
