// google-ads-api-client/dist/protos/bindings/ads/errors/reach_plan_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReachPlanErrorEnum = exports.ReachPlanErrorEnum_ReachPlanError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible errors from ReachPlanService.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ReachPlanErrorEnum.ReachPlanError
 */
var ReachPlanErrorEnum_ReachPlanError;
(function (ReachPlanErrorEnum_ReachPlanError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ReachPlanErrorEnum_ReachPlanError[ReachPlanErrorEnum_ReachPlanError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ReachPlanErrorEnum_ReachPlanError[ReachPlanErrorEnum_ReachPlanError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Not forecastable due to missing rate card data.
     *
     * @generated from protobuf enum value: NOT_FORECASTABLE_MISSING_RATE = 2;
     */
    ReachPlanErrorEnum_ReachPlanError[ReachPlanErrorEnum_ReachPlanError["NOT_FORECASTABLE_MISSING_RATE"] = 2] = "NOT_FORECASTABLE_MISSING_RATE";
})(ReachPlanErrorEnum_ReachPlanError = exports.ReachPlanErrorEnum_ReachPlanError || (exports.ReachPlanErrorEnum_ReachPlanError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ReachPlanErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ReachPlanErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ReachPlanErrorEnum
 */
exports.ReachPlanErrorEnum = new ReachPlanErrorEnum$Type();
