// google-ads-api-client/dist/protos/bindings/ads/errors/recommendation_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecommendationErrorEnum = exports.RecommendationErrorEnum_RecommendationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible errors from applying a recommendation.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.RecommendationErrorEnum.RecommendationError
 */
var RecommendationErrorEnum_RecommendationError;
(function (RecommendationErrorEnum_RecommendationError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The specified budget amount is too low for example, lower than minimum
     * currency unit or lower than ad group minimum cost-per-click.
     *
     * @generated from protobuf enum value: BUDGET_AMOUNT_TOO_SMALL = 2;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["BUDGET_AMOUNT_TOO_SMALL"] = 2] = "BUDGET_AMOUNT_TOO_SMALL";
    /**
     * The specified budget amount is too large.
     *
     * @generated from protobuf enum value: BUDGET_AMOUNT_TOO_LARGE = 3;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["BUDGET_AMOUNT_TOO_LARGE"] = 3] = "BUDGET_AMOUNT_TOO_LARGE";
    /**
     * The specified budget amount is not a valid amount, for example, not a
     * multiple of minimum currency unit.
     *
     * @generated from protobuf enum value: INVALID_BUDGET_AMOUNT = 4;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["INVALID_BUDGET_AMOUNT"] = 4] = "INVALID_BUDGET_AMOUNT";
    /**
     * The specified keyword or ad violates ad policy.
     *
     * @generated from protobuf enum value: POLICY_ERROR = 5;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["POLICY_ERROR"] = 5] = "POLICY_ERROR";
    /**
     * The specified bid amount is not valid, for example, too many fractional
     * digits, or negative amount.
     *
     * @generated from protobuf enum value: INVALID_BID_AMOUNT = 6;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["INVALID_BID_AMOUNT"] = 6] = "INVALID_BID_AMOUNT";
    /**
     * The number of keywords in ad group have reached the maximum allowed.
     *
     * @generated from protobuf enum value: ADGROUP_KEYWORD_LIMIT = 7;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["ADGROUP_KEYWORD_LIMIT"] = 7] = "ADGROUP_KEYWORD_LIMIT";
    /**
     * The recommendation requested to apply has already been applied.
     *
     * @generated from protobuf enum value: RECOMMENDATION_ALREADY_APPLIED = 8;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["RECOMMENDATION_ALREADY_APPLIED"] = 8] = "RECOMMENDATION_ALREADY_APPLIED";
    /**
     * The recommendation requested to apply has been invalidated.
     *
     * @generated from protobuf enum value: RECOMMENDATION_INVALIDATED = 9;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["RECOMMENDATION_INVALIDATED"] = 9] = "RECOMMENDATION_INVALIDATED";
    /**
     * The number of operations in a single request exceeds the maximum allowed.
     *
     * @generated from protobuf enum value: TOO_MANY_OPERATIONS = 10;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["TOO_MANY_OPERATIONS"] = 10] = "TOO_MANY_OPERATIONS";
    /**
     * There are no operations in the request.
     *
     * @generated from protobuf enum value: NO_OPERATIONS = 11;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["NO_OPERATIONS"] = 11] = "NO_OPERATIONS";
    /**
     * Operations with multiple recommendation types are not supported when
     * partial failure mode is not enabled.
     *
     * @generated from protobuf enum value: DIFFERENT_TYPES_NOT_SUPPORTED = 12;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["DIFFERENT_TYPES_NOT_SUPPORTED"] = 12] = "DIFFERENT_TYPES_NOT_SUPPORTED";
    /**
     * Request contains multiple operations with the same resource_name.
     *
     * @generated from protobuf enum value: DUPLICATE_RESOURCE_NAME = 13;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["DUPLICATE_RESOURCE_NAME"] = 13] = "DUPLICATE_RESOURCE_NAME";
    /**
     * The recommendation requested to dismiss has already been dismissed.
     *
     * @generated from protobuf enum value: RECOMMENDATION_ALREADY_DISMISSED = 14;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["RECOMMENDATION_ALREADY_DISMISSED"] = 14] = "RECOMMENDATION_ALREADY_DISMISSED";
    /**
     * The recommendation apply request was malformed and invalid.
     *
     * @generated from protobuf enum value: INVALID_APPLY_REQUEST = 15;
     */
    RecommendationErrorEnum_RecommendationError[RecommendationErrorEnum_RecommendationError["INVALID_APPLY_REQUEST"] = 15] = "INVALID_APPLY_REQUEST";
})(RecommendationErrorEnum_RecommendationError = exports.RecommendationErrorEnum_RecommendationError || (exports.RecommendationErrorEnum_RecommendationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class RecommendationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.RecommendationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.RecommendationErrorEnum
 */
exports.RecommendationErrorEnum = new RecommendationErrorEnum$Type();
