// google-ads-api-client/dist/protos/bindings/ads/errors/region_code_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegionCodeErrorEnum = exports.RegionCodeErrorEnum_RegionCodeError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible region code errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.RegionCodeErrorEnum.RegionCodeError
 */
var RegionCodeErrorEnum_RegionCodeError;
(function (RegionCodeErrorEnum_RegionCodeError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    RegionCodeErrorEnum_RegionCodeError[RegionCodeErrorEnum_RegionCodeError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    RegionCodeErrorEnum_RegionCodeError[RegionCodeErrorEnum_RegionCodeError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Invalid region code.
     *
     * @generated from protobuf enum value: INVALID_REGION_CODE = 2;
     */
    RegionCodeErrorEnum_RegionCodeError[RegionCodeErrorEnum_RegionCodeError["INVALID_REGION_CODE"] = 2] = "INVALID_REGION_CODE";
})(RegionCodeErrorEnum_RegionCodeError = exports.RegionCodeErrorEnum_RegionCodeError || (exports.RegionCodeErrorEnum_RegionCodeError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class RegionCodeErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.RegionCodeErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.RegionCodeErrorEnum
 */
exports.RegionCodeErrorEnum = new RegionCodeErrorEnum$Type();
