// google-ads-api-client/dist/protos/bindings/ads/errors/request_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestErrorEnum = exports.RequestErrorEnum_RequestError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible request errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.RequestErrorEnum.RequestError
 */
var RequestErrorEnum_RequestError;
(function (RequestErrorEnum_RequestError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Resource name is required for this request.
     *
     * @generated from protobuf enum value: RESOURCE_NAME_MISSING = 3;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["RESOURCE_NAME_MISSING"] = 3] = "RESOURCE_NAME_MISSING";
    /**
     * Resource name provided is malformed.
     *
     * @generated from protobuf enum value: RESOURCE_NAME_MALFORMED = 4;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["RESOURCE_NAME_MALFORMED"] = 4] = "RESOURCE_NAME_MALFORMED";
    /**
     * Resource name provided is malformed.
     *
     * @generated from protobuf enum value: BAD_RESOURCE_ID = 17;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["BAD_RESOURCE_ID"] = 17] = "BAD_RESOURCE_ID";
    /**
     * Customer ID is invalid.
     *
     * @generated from protobuf enum value: INVALID_CUSTOMER_ID = 16;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["INVALID_CUSTOMER_ID"] = 16] = "INVALID_CUSTOMER_ID";
    /**
     * Mutate operation should have either create, update, or remove specified.
     *
     * @generated from protobuf enum value: OPERATION_REQUIRED = 5;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["OPERATION_REQUIRED"] = 5] = "OPERATION_REQUIRED";
    /**
     * Requested resource not found.
     *
     * @generated from protobuf enum value: RESOURCE_NOT_FOUND = 6;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["RESOURCE_NOT_FOUND"] = 6] = "RESOURCE_NOT_FOUND";
    /**
     * Next page token specified in user request is invalid.
     *
     * @generated from protobuf enum value: INVALID_PAGE_TOKEN = 7;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["INVALID_PAGE_TOKEN"] = 7] = "INVALID_PAGE_TOKEN";
    /**
     * Next page token specified in user request has expired.
     *
     * @generated from protobuf enum value: EXPIRED_PAGE_TOKEN = 8;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["EXPIRED_PAGE_TOKEN"] = 8] = "EXPIRED_PAGE_TOKEN";
    /**
     * Page size specified in user request is invalid.
     *
     * @generated from protobuf enum value: INVALID_PAGE_SIZE = 22;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["INVALID_PAGE_SIZE"] = 22] = "INVALID_PAGE_SIZE";
    /**
     * Required field is missing.
     *
     * @generated from protobuf enum value: REQUIRED_FIELD_MISSING = 9;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["REQUIRED_FIELD_MISSING"] = 9] = "REQUIRED_FIELD_MISSING";
    /**
     * The field cannot be modified because it's immutable. It's also possible
     * that the field can be modified using 'create' operation but not 'update'.
     *
     * @generated from protobuf enum value: IMMUTABLE_FIELD = 11;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["IMMUTABLE_FIELD"] = 11] = "IMMUTABLE_FIELD";
    /**
     * Received too many entries in request.
     *
     * @generated from protobuf enum value: TOO_MANY_MUTATE_OPERATIONS = 13;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["TOO_MANY_MUTATE_OPERATIONS"] = 13] = "TOO_MANY_MUTATE_OPERATIONS";
    /**
     * Request cannot be executed by a manager account.
     *
     * @generated from protobuf enum value: CANNOT_BE_EXECUTED_BY_MANAGER_ACCOUNT = 14;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["CANNOT_BE_EXECUTED_BY_MANAGER_ACCOUNT"] = 14] = "CANNOT_BE_EXECUTED_BY_MANAGER_ACCOUNT";
    /**
     * Mutate request was attempting to modify a readonly field.
     * For instance, Budget fields can be requested for Ad Group,
     * but are read-only for adGroups:mutate.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_FOREIGN_FIELD = 15;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["CANNOT_MODIFY_FOREIGN_FIELD"] = 15] = "CANNOT_MODIFY_FOREIGN_FIELD";
    /**
     * Enum value is not permitted.
     *
     * @generated from protobuf enum value: INVALID_ENUM_VALUE = 18;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["INVALID_ENUM_VALUE"] = 18] = "INVALID_ENUM_VALUE";
    /**
     * The developer-token parameter is required for all requests.
     *
     * @generated from protobuf enum value: DEVELOPER_TOKEN_PARAMETER_MISSING = 19;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["DEVELOPER_TOKEN_PARAMETER_MISSING"] = 19] = "DEVELOPER_TOKEN_PARAMETER_MISSING";
    /**
     * The login-customer-id parameter is required for this request.
     *
     * @generated from protobuf enum value: LOGIN_CUSTOMER_ID_PARAMETER_MISSING = 20;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["LOGIN_CUSTOMER_ID_PARAMETER_MISSING"] = 20] = "LOGIN_CUSTOMER_ID_PARAMETER_MISSING";
    /**
     * page_token is set in the validate only request
     *
     * @generated from protobuf enum value: VALIDATE_ONLY_REQUEST_HAS_PAGE_TOKEN = 21;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["VALIDATE_ONLY_REQUEST_HAS_PAGE_TOKEN"] = 21] = "VALIDATE_ONLY_REQUEST_HAS_PAGE_TOKEN";
    /**
     * return_summary_row cannot be enabled if request did not select any
     * metrics field.
     *
     * @generated from protobuf enum value: CANNOT_RETURN_SUMMARY_ROW_FOR_REQUEST_WITHOUT_METRICS = 29;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["CANNOT_RETURN_SUMMARY_ROW_FOR_REQUEST_WITHOUT_METRICS"] = 29] = "CANNOT_RETURN_SUMMARY_ROW_FOR_REQUEST_WITHOUT_METRICS";
    /**
     * return_summary_row should not be enabled for validate only requests.
     *
     * @generated from protobuf enum value: CANNOT_RETURN_SUMMARY_ROW_FOR_VALIDATE_ONLY_REQUESTS = 30;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["CANNOT_RETURN_SUMMARY_ROW_FOR_VALIDATE_ONLY_REQUESTS"] = 30] = "CANNOT_RETURN_SUMMARY_ROW_FOR_VALIDATE_ONLY_REQUESTS";
    /**
     * return_summary_row parameter value should be the same between requests
     * with page_token field set and their original request.
     *
     * @generated from protobuf enum value: INCONSISTENT_RETURN_SUMMARY_ROW_VALUE = 31;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["INCONSISTENT_RETURN_SUMMARY_ROW_VALUE"] = 31] = "INCONSISTENT_RETURN_SUMMARY_ROW_VALUE";
    /**
     * The total results count cannot be returned if it was not requested in the
     * original request.
     *
     * @generated from protobuf enum value: TOTAL_RESULTS_COUNT_NOT_ORIGINALLY_REQUESTED = 32;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["TOTAL_RESULTS_COUNT_NOT_ORIGINALLY_REQUESTED"] = 32] = "TOTAL_RESULTS_COUNT_NOT_ORIGINALLY_REQUESTED";
    /**
     * Deadline specified by the client was too short.
     *
     * @generated from protobuf enum value: RPC_DEADLINE_TOO_SHORT = 33;
     */
    RequestErrorEnum_RequestError[RequestErrorEnum_RequestError["RPC_DEADLINE_TOO_SHORT"] = 33] = "RPC_DEADLINE_TOO_SHORT";
})(RequestErrorEnum_RequestError = exports.RequestErrorEnum_RequestError || (exports.RequestErrorEnum_RequestError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class RequestErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.RequestErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.RequestErrorEnum
 */
exports.RequestErrorEnum = new RequestErrorEnum$Type();
