// google-ads-api-client/dist/protos/bindings/ads/errors/resource_access_denied_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceAccessDeniedErrorEnum = exports.ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible resource access denied errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ResourceAccessDeniedErrorEnum.ResourceAccessDeniedError
 */
var ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError;
(function (ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError[ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError[ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * User did not have write access.
     *
     * @generated from protobuf enum value: WRITE_ACCESS_DENIED = 3;
     */
    ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError[ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError["WRITE_ACCESS_DENIED"] = 3] = "WRITE_ACCESS_DENIED";
})(ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError = exports.ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError || (exports.ResourceAccessDeniedErrorEnum_ResourceAccessDeniedError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ResourceAccessDeniedErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ResourceAccessDeniedErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ResourceAccessDeniedErrorEnum
 */
exports.ResourceAccessDeniedErrorEnum = new ResourceAccessDeniedErrorEnum$Type();
