// google-ads-api-client/dist/protos/bindings/ads/errors/resource_count_limit_exceeded_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceCountLimitExceededErrorEnum = exports.ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible resource count limit exceeded errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ResourceCountLimitExceededErrorEnum.ResourceCountLimitExceededError
 */
var ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError;
(function (ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Indicates that this request would exceed the number of allowed resources
     * for the Google Ads account. The exact resource type and limit being
     * checked can be inferred from accountLimitType.
     *
     * @generated from protobuf enum value: ACCOUNT_LIMIT = 2;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["ACCOUNT_LIMIT"] = 2] = "ACCOUNT_LIMIT";
    /**
     * Indicates that this request would exceed the number of allowed resources
     * in a Campaign. The exact resource type and limit being checked can be
     * inferred from accountLimitType, and the numeric id of the
     * Campaign involved is given by enclosingId.
     *
     * @generated from protobuf enum value: CAMPAIGN_LIMIT = 3;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["CAMPAIGN_LIMIT"] = 3] = "CAMPAIGN_LIMIT";
    /**
     * Indicates that this request would exceed the number of allowed resources
     * in an ad group. The exact resource type and limit being checked can be
     * inferred from accountLimitType, and the numeric id of the
     * ad group involved is given by enclosingId.
     *
     * @generated from protobuf enum value: ADGROUP_LIMIT = 4;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["ADGROUP_LIMIT"] = 4] = "ADGROUP_LIMIT";
    /**
     * Indicates that this request would exceed the number of allowed resources
     * in an ad group ad. The exact resource type and limit being checked can
     * be inferred from accountLimitType, and the enclosingId
     * contains the ad group id followed by the ad id, separated by a single
     * comma (,).
     *
     * @generated from protobuf enum value: AD_GROUP_AD_LIMIT = 5;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["AD_GROUP_AD_LIMIT"] = 5] = "AD_GROUP_AD_LIMIT";
    /**
     * Indicates that this request would exceed the number of allowed resources
     * in an ad group criterion. The exact resource type and limit being checked
     * can be inferred from accountLimitType, and the
     * enclosingId contains the ad group id followed by the
     * criterion id, separated by a single comma (,).
     *
     * @generated from protobuf enum value: AD_GROUP_CRITERION_LIMIT = 6;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["AD_GROUP_CRITERION_LIMIT"] = 6] = "AD_GROUP_CRITERION_LIMIT";
    /**
     * Indicates that this request would exceed the number of allowed resources
     * in this shared set. The exact resource type and limit being checked can
     * be inferred from accountLimitType, and the numeric id of the
     * shared set involved is given by enclosingId.
     *
     * @generated from protobuf enum value: SHARED_SET_LIMIT = 7;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["SHARED_SET_LIMIT"] = 7] = "SHARED_SET_LIMIT";
    /**
     * Exceeds a limit related to a matching function.
     *
     * @generated from protobuf enum value: MATCHING_FUNCTION_LIMIT = 8;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["MATCHING_FUNCTION_LIMIT"] = 8] = "MATCHING_FUNCTION_LIMIT";
    /**
     * The response for this request would exceed the maximum number of rows
     * that can be returned.
     *
     * @generated from protobuf enum value: RESPONSE_ROW_LIMIT_EXCEEDED = 9;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["RESPONSE_ROW_LIMIT_EXCEEDED"] = 9] = "RESPONSE_ROW_LIMIT_EXCEEDED";
    /**
     * This request would exceed a limit on the number of allowed resources.
     * The details of which type of limit was exceeded will eventually be
     * returned in ErrorDetails.
     *
     * @generated from protobuf enum value: RESOURCE_LIMIT = 10;
     */
    ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError[ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError["RESOURCE_LIMIT"] = 10] = "RESOURCE_LIMIT";
})(ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError = exports.ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError || (exports.ResourceCountLimitExceededErrorEnum_ResourceCountLimitExceededError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ResourceCountLimitExceededErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ResourceCountLimitExceededErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ResourceCountLimitExceededErrorEnum
 */
exports.ResourceCountLimitExceededErrorEnum = new ResourceCountLimitExceededErrorEnum$Type();
