// google-ads-api-client/dist/protos/bindings/ads/errors/setting_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingErrorEnum = exports.SettingErrorEnum_SettingError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible setting errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.SettingErrorEnum.SettingError
 */
var SettingErrorEnum_SettingError;
(function (SettingErrorEnum_SettingError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The campaign setting is not available for this Google Ads account.
     *
     * @generated from protobuf enum value: SETTING_TYPE_IS_NOT_AVAILABLE = 3;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["SETTING_TYPE_IS_NOT_AVAILABLE"] = 3] = "SETTING_TYPE_IS_NOT_AVAILABLE";
    /**
     * The setting is not compatible with the campaign.
     *
     * @generated from protobuf enum value: SETTING_TYPE_IS_NOT_COMPATIBLE_WITH_CAMPAIGN = 4;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["SETTING_TYPE_IS_NOT_COMPATIBLE_WITH_CAMPAIGN"] = 4] = "SETTING_TYPE_IS_NOT_COMPATIBLE_WITH_CAMPAIGN";
    /**
     * The supplied TargetingSetting contains an invalid CriterionTypeGroup. See
     * CriterionTypeGroup documentation for CriterionTypeGroups allowed
     * in Campaign or AdGroup TargetingSettings.
     *
     * @generated from protobuf enum value: TARGETING_SETTING_CONTAINS_INVALID_CRITERION_TYPE_GROUP = 5;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["TARGETING_SETTING_CONTAINS_INVALID_CRITERION_TYPE_GROUP"] = 5] = "TARGETING_SETTING_CONTAINS_INVALID_CRITERION_TYPE_GROUP";
    /**
     * TargetingSetting must not explicitly
     * set any of the Demographic CriterionTypeGroups (AGE_RANGE, GENDER,
     * PARENT, INCOME_RANGE) to false (it's okay to not set them at all, in
     * which case the system will set them to true automatically).
     *
     * @generated from protobuf enum value: TARGETING_SETTING_DEMOGRAPHIC_CRITERION_TYPE_GROUPS_MUST_BE_SET_TO_TARGET_ALL = 6;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["TARGETING_SETTING_DEMOGRAPHIC_CRITERION_TYPE_GROUPS_MUST_BE_SET_TO_TARGET_ALL"] = 6] = "TARGETING_SETTING_DEMOGRAPHIC_CRITERION_TYPE_GROUPS_MUST_BE_SET_TO_TARGET_ALL";
    /**
     * TargetingSetting cannot change any of
     * the Demographic CriterionTypeGroups (AGE_RANGE, GENDER, PARENT,
     * INCOME_RANGE) from true to false.
     *
     * @generated from protobuf enum value: TARGETING_SETTING_CANNOT_CHANGE_TARGET_ALL_TO_FALSE_FOR_DEMOGRAPHIC_CRITERION_TYPE_GROUP = 7;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["TARGETING_SETTING_CANNOT_CHANGE_TARGET_ALL_TO_FALSE_FOR_DEMOGRAPHIC_CRITERION_TYPE_GROUP"] = 7] = "TARGETING_SETTING_CANNOT_CHANGE_TARGET_ALL_TO_FALSE_FOR_DEMOGRAPHIC_CRITERION_TYPE_GROUP";
    /**
     * At least one feed id should be present.
     *
     * @generated from protobuf enum value: DYNAMIC_SEARCH_ADS_SETTING_AT_LEAST_ONE_FEED_ID_MUST_BE_PRESENT = 8;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["DYNAMIC_SEARCH_ADS_SETTING_AT_LEAST_ONE_FEED_ID_MUST_BE_PRESENT"] = 8] = "DYNAMIC_SEARCH_ADS_SETTING_AT_LEAST_ONE_FEED_ID_MUST_BE_PRESENT";
    /**
     * The supplied DynamicSearchAdsSetting contains an invalid domain name.
     *
     * @generated from protobuf enum value: DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_INVALID_DOMAIN_NAME = 9;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_INVALID_DOMAIN_NAME"] = 9] = "DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_INVALID_DOMAIN_NAME";
    /**
     * The supplied DynamicSearchAdsSetting contains a subdomain name.
     *
     * @generated from protobuf enum value: DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_SUBDOMAIN_NAME = 10;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_SUBDOMAIN_NAME"] = 10] = "DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_SUBDOMAIN_NAME";
    /**
     * The supplied DynamicSearchAdsSetting contains an invalid language code.
     *
     * @generated from protobuf enum value: DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_INVALID_LANGUAGE_CODE = 11;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_INVALID_LANGUAGE_CODE"] = 11] = "DYNAMIC_SEARCH_ADS_SETTING_CONTAINS_INVALID_LANGUAGE_CODE";
    /**
     * TargetingSettings in search campaigns should not have
     * CriterionTypeGroup.PLACEMENT set to targetAll.
     *
     * @generated from protobuf enum value: TARGET_ALL_IS_NOT_ALLOWED_FOR_PLACEMENT_IN_SEARCH_CAMPAIGN = 12;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["TARGET_ALL_IS_NOT_ALLOWED_FOR_PLACEMENT_IN_SEARCH_CAMPAIGN"] = 12] = "TARGET_ALL_IS_NOT_ALLOWED_FOR_PLACEMENT_IN_SEARCH_CAMPAIGN";
    /**
     * The setting value is not compatible with the campaign type.
     *
     * @generated from protobuf enum value: SETTING_VALUE_NOT_COMPATIBLE_WITH_CAMPAIGN = 20;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["SETTING_VALUE_NOT_COMPATIBLE_WITH_CAMPAIGN"] = 20] = "SETTING_VALUE_NOT_COMPATIBLE_WITH_CAMPAIGN";
    /**
     * Switching from observation setting to targeting setting is not allowed
     * for Customer Match lists. See
     * https://support.google.com/google-ads/answer/6299717.
     *
     * @generated from protobuf enum value: BID_ONLY_IS_NOT_ALLOWED_TO_BE_MODIFIED_WITH_CUSTOMER_MATCH_TARGETING = 21;
     */
    SettingErrorEnum_SettingError[SettingErrorEnum_SettingError["BID_ONLY_IS_NOT_ALLOWED_TO_BE_MODIFIED_WITH_CUSTOMER_MATCH_TARGETING"] = 21] = "BID_ONLY_IS_NOT_ALLOWED_TO_BE_MODIFIED_WITH_CUSTOMER_MATCH_TARGETING";
})(SettingErrorEnum_SettingError = exports.SettingErrorEnum_SettingError || (exports.SettingErrorEnum_SettingError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SettingErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.SettingErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.SettingErrorEnum
 */
exports.SettingErrorEnum = new SettingErrorEnum$Type();
