// google-ads-api-client/dist/protos/bindings/ads/errors/shared_set_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedSetErrorEnum = exports.SharedSetErrorEnum_SharedSetError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible shared set errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.SharedSetErrorEnum.SharedSetError
 */
var SharedSetErrorEnum_SharedSetError;
(function (SharedSetErrorEnum_SharedSetError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SharedSetErrorEnum_SharedSetError[SharedSetErrorEnum_SharedSetError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SharedSetErrorEnum_SharedSetError[SharedSetErrorEnum_SharedSetError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The customer cannot create this type of shared set.
     *
     * @generated from protobuf enum value: CUSTOMER_CANNOT_CREATE_SHARED_SET_OF_THIS_TYPE = 2;
     */
    SharedSetErrorEnum_SharedSetError[SharedSetErrorEnum_SharedSetError["CUSTOMER_CANNOT_CREATE_SHARED_SET_OF_THIS_TYPE"] = 2] = "CUSTOMER_CANNOT_CREATE_SHARED_SET_OF_THIS_TYPE";
    /**
     * A shared set with this name already exists.
     *
     * @generated from protobuf enum value: DUPLICATE_NAME = 3;
     */
    SharedSetErrorEnum_SharedSetError[SharedSetErrorEnum_SharedSetError["DUPLICATE_NAME"] = 3] = "DUPLICATE_NAME";
    /**
     * Removed shared sets cannot be mutated.
     *
     * @generated from protobuf enum value: SHARED_SET_REMOVED = 4;
     */
    SharedSetErrorEnum_SharedSetError[SharedSetErrorEnum_SharedSetError["SHARED_SET_REMOVED"] = 4] = "SHARED_SET_REMOVED";
    /**
     * The shared set cannot be removed because it is in use.
     *
     * @generated from protobuf enum value: SHARED_SET_IN_USE = 5;
     */
    SharedSetErrorEnum_SharedSetError[SharedSetErrorEnum_SharedSetError["SHARED_SET_IN_USE"] = 5] = "SHARED_SET_IN_USE";
})(SharedSetErrorEnum_SharedSetError = exports.SharedSetErrorEnum_SharedSetError || (exports.SharedSetErrorEnum_SharedSetError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SharedSetErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.SharedSetErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.SharedSetErrorEnum
 */
exports.SharedSetErrorEnum = new SharedSetErrorEnum$Type();
