// google-ads-api-client/dist/protos/bindings/ads/errors/shared_criterion_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedCriterionErrorEnum = exports.SharedCriterionErrorEnum_SharedCriterionError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible shared criterion errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.SharedCriterionErrorEnum.SharedCriterionError
 */
var SharedCriterionErrorEnum_SharedCriterionError;
(function (SharedCriterionErrorEnum_SharedCriterionError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SharedCriterionErrorEnum_SharedCriterionError[SharedCriterionErrorEnum_SharedCriterionError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SharedCriterionErrorEnum_SharedCriterionError[SharedCriterionErrorEnum_SharedCriterionError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The criterion is not appropriate for the shared set type.
     *
     * @generated from protobuf enum value: CRITERION_TYPE_NOT_ALLOWED_FOR_SHARED_SET_TYPE = 2;
     */
    SharedCriterionErrorEnum_SharedCriterionError[SharedCriterionErrorEnum_SharedCriterionError["CRITERION_TYPE_NOT_ALLOWED_FOR_SHARED_SET_TYPE"] = 2] = "CRITERION_TYPE_NOT_ALLOWED_FOR_SHARED_SET_TYPE";
})(SharedCriterionErrorEnum_SharedCriterionError = exports.SharedCriterionErrorEnum_SharedCriterionError || (exports.SharedCriterionErrorEnum_SharedCriterionError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SharedCriterionErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.SharedCriterionErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.SharedCriterionErrorEnum
 */
exports.SharedCriterionErrorEnum = new SharedCriterionErrorEnum$Type();
