// google-ads-api-client/dist/protos/bindings/ads/errors/size_limit_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SizeLimitErrorEnum = exports.SizeLimitErrorEnum_SizeLimitError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible size limit errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.SizeLimitErrorEnum.SizeLimitError
 */
var SizeLimitErrorEnum_SizeLimitError;
(function (SizeLimitErrorEnum_SizeLimitError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    SizeLimitErrorEnum_SizeLimitError[SizeLimitErrorEnum_SizeLimitError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    SizeLimitErrorEnum_SizeLimitError[SizeLimitErrorEnum_SizeLimitError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The number of entries in the request exceeds the system limit, or the
     * contents of the operations exceed transaction limits due to their size
     * or complexity. Try reducing the number of entries per request.
     *
     * @generated from protobuf enum value: REQUEST_SIZE_LIMIT_EXCEEDED = 2;
     */
    SizeLimitErrorEnum_SizeLimitError[SizeLimitErrorEnum_SizeLimitError["REQUEST_SIZE_LIMIT_EXCEEDED"] = 2] = "REQUEST_SIZE_LIMIT_EXCEEDED";
    /**
     * The number of entries in the response exceeds the system limit.
     *
     * @generated from protobuf enum value: RESPONSE_SIZE_LIMIT_EXCEEDED = 3;
     */
    SizeLimitErrorEnum_SizeLimitError[SizeLimitErrorEnum_SizeLimitError["RESPONSE_SIZE_LIMIT_EXCEEDED"] = 3] = "RESPONSE_SIZE_LIMIT_EXCEEDED";
})(SizeLimitErrorEnum_SizeLimitError = exports.SizeLimitErrorEnum_SizeLimitError || (exports.SizeLimitErrorEnum_SizeLimitError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class SizeLimitErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.SizeLimitErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.SizeLimitErrorEnum
 */
exports.SizeLimitErrorEnum = new SizeLimitErrorEnum$Type();
