// google-ads-api-client/dist/protos/bindings/ads/errors/string_length_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StringLengthErrorEnum = exports.StringLengthErrorEnum_StringLengthError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible string length errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.StringLengthErrorEnum.StringLengthError
 */
var StringLengthErrorEnum_StringLengthError;
(function (StringLengthErrorEnum_StringLengthError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    StringLengthErrorEnum_StringLengthError[StringLengthErrorEnum_StringLengthError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    StringLengthErrorEnum_StringLengthError[StringLengthErrorEnum_StringLengthError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The specified field should have a least one non-whitespace character in
     * it.
     *
     * @generated from protobuf enum value: EMPTY = 4;
     */
    StringLengthErrorEnum_StringLengthError[StringLengthErrorEnum_StringLengthError["EMPTY"] = 4] = "EMPTY";
    /**
     * Too short.
     *
     * @generated from protobuf enum value: TOO_SHORT = 2;
     */
    StringLengthErrorEnum_StringLengthError[StringLengthErrorEnum_StringLengthError["TOO_SHORT"] = 2] = "TOO_SHORT";
    /**
     * Too long.
     *
     * @generated from protobuf enum value: TOO_LONG = 3;
     */
    StringLengthErrorEnum_StringLengthError[StringLengthErrorEnum_StringLengthError["TOO_LONG"] = 3] = "TOO_LONG";
})(StringLengthErrorEnum_StringLengthError = exports.StringLengthErrorEnum_StringLengthError || (exports.StringLengthErrorEnum_StringLengthError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class StringLengthErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.StringLengthErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.StringLengthErrorEnum
 */
exports.StringLengthErrorEnum = new StringLengthErrorEnum$Type();
