// google-ads-api-client/dist/protos/bindings/ads/errors/third_party_app_analytics_link_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThirdPartyAppAnalyticsLinkErrorEnum = exports.ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible third party app analytics link errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.ThirdPartyAppAnalyticsLinkErrorEnum.ThirdPartyAppAnalyticsLinkError
 */
var ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError;
(function (ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError[ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError[ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The provided analytics provider ID is invalid.
     *
     * @generated from protobuf enum value: INVALID_ANALYTICS_PROVIDER_ID = 2;
     */
    ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError[ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError["INVALID_ANALYTICS_PROVIDER_ID"] = 2] = "INVALID_ANALYTICS_PROVIDER_ID";
    /**
     * The provided mobile app ID is invalid.
     *
     * @generated from protobuf enum value: INVALID_MOBILE_APP_ID = 3;
     */
    ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError[ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError["INVALID_MOBILE_APP_ID"] = 3] = "INVALID_MOBILE_APP_ID";
    /**
     * The mobile app corresponding to the provided app ID is not
     * active/enabled.
     *
     * @generated from protobuf enum value: MOBILE_APP_IS_NOT_ENABLED = 4;
     */
    ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError[ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError["MOBILE_APP_IS_NOT_ENABLED"] = 4] = "MOBILE_APP_IS_NOT_ENABLED";
    /**
     * Regenerating shareable link ID is only allowed on active links
     *
     * @generated from protobuf enum value: CANNOT_REGENERATE_SHAREABLE_LINK_ID_FOR_REMOVED_LINK = 5;
     */
    ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError[ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError["CANNOT_REGENERATE_SHAREABLE_LINK_ID_FOR_REMOVED_LINK"] = 5] = "CANNOT_REGENERATE_SHAREABLE_LINK_ID_FOR_REMOVED_LINK";
})(ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError = exports.ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError || (exports.ThirdPartyAppAnalyticsLinkErrorEnum_ThirdPartyAppAnalyticsLinkError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class ThirdPartyAppAnalyticsLinkErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.ThirdPartyAppAnalyticsLinkErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.ThirdPartyAppAnalyticsLinkErrorEnum
 */
exports.ThirdPartyAppAnalyticsLinkErrorEnum = new ThirdPartyAppAnalyticsLinkErrorEnum$Type();
