// google-ads-api-client/dist/protos/bindings/ads/errors/time_zone_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeZoneErrorEnum = exports.TimeZoneErrorEnum_TimeZoneError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible currency code errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.TimeZoneErrorEnum.TimeZoneError
 */
var TimeZoneErrorEnum_TimeZoneError;
(function (TimeZoneErrorEnum_TimeZoneError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    TimeZoneErrorEnum_TimeZoneError[TimeZoneErrorEnum_TimeZoneError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    TimeZoneErrorEnum_TimeZoneError[TimeZoneErrorEnum_TimeZoneError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Time zone is not valid.
     *
     * @generated from protobuf enum value: INVALID_TIME_ZONE = 5;
     */
    TimeZoneErrorEnum_TimeZoneError[TimeZoneErrorEnum_TimeZoneError["INVALID_TIME_ZONE"] = 5] = "INVALID_TIME_ZONE";
})(TimeZoneErrorEnum_TimeZoneError = exports.TimeZoneErrorEnum_TimeZoneError || (exports.TimeZoneErrorEnum_TimeZoneError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class TimeZoneErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.TimeZoneErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.TimeZoneErrorEnum
 */
exports.TimeZoneErrorEnum = new TimeZoneErrorEnum$Type();
