// google-ads-api-client/dist/protos/bindings/ads/errors/url_field_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UrlFieldErrorEnum = exports.UrlFieldErrorEnum_UrlFieldError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible url field errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.UrlFieldErrorEnum.UrlFieldError
 */
var UrlFieldErrorEnum_UrlFieldError;
(function (UrlFieldErrorEnum_UrlFieldError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * The tracking url template is invalid.
     *
     * @generated from protobuf enum value: INVALID_TRACKING_URL_TEMPLATE = 2;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TRACKING_URL_TEMPLATE"] = 2] = "INVALID_TRACKING_URL_TEMPLATE";
    /**
     * The tracking url template contains invalid tag.
     *
     * @generated from protobuf enum value: INVALID_TAG_IN_TRACKING_URL_TEMPLATE = 3;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TAG_IN_TRACKING_URL_TEMPLATE"] = 3] = "INVALID_TAG_IN_TRACKING_URL_TEMPLATE";
    /**
     * The tracking url template must contain at least one tag (for example,
     * {lpurl}), This applies only to tracking url template associated with
     * website ads or product ads.
     *
     * @generated from protobuf enum value: MISSING_TRACKING_URL_TEMPLATE_TAG = 4;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_TRACKING_URL_TEMPLATE_TAG"] = 4] = "MISSING_TRACKING_URL_TEMPLATE_TAG";
    /**
     * The tracking url template must start with a valid protocol (or lpurl
     * tag).
     *
     * @generated from protobuf enum value: MISSING_PROTOCOL_IN_TRACKING_URL_TEMPLATE = 5;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_PROTOCOL_IN_TRACKING_URL_TEMPLATE"] = 5] = "MISSING_PROTOCOL_IN_TRACKING_URL_TEMPLATE";
    /**
     * The tracking url template starts with an invalid protocol.
     *
     * @generated from protobuf enum value: INVALID_PROTOCOL_IN_TRACKING_URL_TEMPLATE = 6;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_PROTOCOL_IN_TRACKING_URL_TEMPLATE"] = 6] = "INVALID_PROTOCOL_IN_TRACKING_URL_TEMPLATE";
    /**
     * The tracking url template contains illegal characters.
     *
     * @generated from protobuf enum value: MALFORMED_TRACKING_URL_TEMPLATE = 7;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MALFORMED_TRACKING_URL_TEMPLATE"] = 7] = "MALFORMED_TRACKING_URL_TEMPLATE";
    /**
     * The tracking url template must contain a host name (or lpurl tag).
     *
     * @generated from protobuf enum value: MISSING_HOST_IN_TRACKING_URL_TEMPLATE = 8;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_HOST_IN_TRACKING_URL_TEMPLATE"] = 8] = "MISSING_HOST_IN_TRACKING_URL_TEMPLATE";
    /**
     * The tracking url template has an invalid or missing top level domain
     * extension.
     *
     * @generated from protobuf enum value: INVALID_TLD_IN_TRACKING_URL_TEMPLATE = 9;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TLD_IN_TRACKING_URL_TEMPLATE"] = 9] = "INVALID_TLD_IN_TRACKING_URL_TEMPLATE";
    /**
     * The tracking url template contains nested occurrences of the same
     * conditional tag (for example, {ifmobile:{ifmobile:x}}).
     *
     * @generated from protobuf enum value: REDUNDANT_NESTED_TRACKING_URL_TEMPLATE_TAG = 10;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["REDUNDANT_NESTED_TRACKING_URL_TEMPLATE_TAG"] = 10] = "REDUNDANT_NESTED_TRACKING_URL_TEMPLATE_TAG";
    /**
     * The final url is invalid.
     *
     * @generated from protobuf enum value: INVALID_FINAL_URL = 11;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_FINAL_URL"] = 11] = "INVALID_FINAL_URL";
    /**
     * The final url contains invalid tag.
     *
     * @generated from protobuf enum value: INVALID_TAG_IN_FINAL_URL = 12;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TAG_IN_FINAL_URL"] = 12] = "INVALID_TAG_IN_FINAL_URL";
    /**
     * The final url contains nested occurrences of the same conditional tag
     * (for example, {ifmobile:{ifmobile:x}}).
     *
     * @generated from protobuf enum value: REDUNDANT_NESTED_FINAL_URL_TAG = 13;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["REDUNDANT_NESTED_FINAL_URL_TAG"] = 13] = "REDUNDANT_NESTED_FINAL_URL_TAG";
    /**
     * The final url must start with a valid protocol.
     *
     * @generated from protobuf enum value: MISSING_PROTOCOL_IN_FINAL_URL = 14;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_PROTOCOL_IN_FINAL_URL"] = 14] = "MISSING_PROTOCOL_IN_FINAL_URL";
    /**
     * The final url starts with an invalid protocol.
     *
     * @generated from protobuf enum value: INVALID_PROTOCOL_IN_FINAL_URL = 15;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_PROTOCOL_IN_FINAL_URL"] = 15] = "INVALID_PROTOCOL_IN_FINAL_URL";
    /**
     * The final url contains illegal characters.
     *
     * @generated from protobuf enum value: MALFORMED_FINAL_URL = 16;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MALFORMED_FINAL_URL"] = 16] = "MALFORMED_FINAL_URL";
    /**
     * The final url must contain a host name.
     *
     * @generated from protobuf enum value: MISSING_HOST_IN_FINAL_URL = 17;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_HOST_IN_FINAL_URL"] = 17] = "MISSING_HOST_IN_FINAL_URL";
    /**
     * The tracking url template has an invalid or missing top level domain
     * extension.
     *
     * @generated from protobuf enum value: INVALID_TLD_IN_FINAL_URL = 18;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TLD_IN_FINAL_URL"] = 18] = "INVALID_TLD_IN_FINAL_URL";
    /**
     * The final mobile url is invalid.
     *
     * @generated from protobuf enum value: INVALID_FINAL_MOBILE_URL = 19;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_FINAL_MOBILE_URL"] = 19] = "INVALID_FINAL_MOBILE_URL";
    /**
     * The final mobile url contains invalid tag.
     *
     * @generated from protobuf enum value: INVALID_TAG_IN_FINAL_MOBILE_URL = 20;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TAG_IN_FINAL_MOBILE_URL"] = 20] = "INVALID_TAG_IN_FINAL_MOBILE_URL";
    /**
     * The final mobile url contains nested occurrences of the same conditional
     * tag (for example, {ifmobile:{ifmobile:x}}).
     *
     * @generated from protobuf enum value: REDUNDANT_NESTED_FINAL_MOBILE_URL_TAG = 21;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["REDUNDANT_NESTED_FINAL_MOBILE_URL_TAG"] = 21] = "REDUNDANT_NESTED_FINAL_MOBILE_URL_TAG";
    /**
     * The final mobile url must start with a valid protocol.
     *
     * @generated from protobuf enum value: MISSING_PROTOCOL_IN_FINAL_MOBILE_URL = 22;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_PROTOCOL_IN_FINAL_MOBILE_URL"] = 22] = "MISSING_PROTOCOL_IN_FINAL_MOBILE_URL";
    /**
     * The final mobile url starts with an invalid protocol.
     *
     * @generated from protobuf enum value: INVALID_PROTOCOL_IN_FINAL_MOBILE_URL = 23;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_PROTOCOL_IN_FINAL_MOBILE_URL"] = 23] = "INVALID_PROTOCOL_IN_FINAL_MOBILE_URL";
    /**
     * The final mobile url contains illegal characters.
     *
     * @generated from protobuf enum value: MALFORMED_FINAL_MOBILE_URL = 24;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MALFORMED_FINAL_MOBILE_URL"] = 24] = "MALFORMED_FINAL_MOBILE_URL";
    /**
     * The final mobile url must contain a host name.
     *
     * @generated from protobuf enum value: MISSING_HOST_IN_FINAL_MOBILE_URL = 25;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_HOST_IN_FINAL_MOBILE_URL"] = 25] = "MISSING_HOST_IN_FINAL_MOBILE_URL";
    /**
     * The tracking url template has an invalid or missing top level domain
     * extension.
     *
     * @generated from protobuf enum value: INVALID_TLD_IN_FINAL_MOBILE_URL = 26;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TLD_IN_FINAL_MOBILE_URL"] = 26] = "INVALID_TLD_IN_FINAL_MOBILE_URL";
    /**
     * The final app url is invalid.
     *
     * @generated from protobuf enum value: INVALID_FINAL_APP_URL = 27;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_FINAL_APP_URL"] = 27] = "INVALID_FINAL_APP_URL";
    /**
     * The final app url contains invalid tag.
     *
     * @generated from protobuf enum value: INVALID_TAG_IN_FINAL_APP_URL = 28;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TAG_IN_FINAL_APP_URL"] = 28] = "INVALID_TAG_IN_FINAL_APP_URL";
    /**
     * The final app url contains nested occurrences of the same conditional tag
     * (for example, {ifmobile:{ifmobile:x}}).
     *
     * @generated from protobuf enum value: REDUNDANT_NESTED_FINAL_APP_URL_TAG = 29;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["REDUNDANT_NESTED_FINAL_APP_URL_TAG"] = 29] = "REDUNDANT_NESTED_FINAL_APP_URL_TAG";
    /**
     * More than one app url found for the same OS type.
     *
     * @generated from protobuf enum value: MULTIPLE_APP_URLS_FOR_OSTYPE = 30;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MULTIPLE_APP_URLS_FOR_OSTYPE"] = 30] = "MULTIPLE_APP_URLS_FOR_OSTYPE";
    /**
     * The OS type given for an app url is not valid.
     *
     * @generated from protobuf enum value: INVALID_OSTYPE = 31;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_OSTYPE"] = 31] = "INVALID_OSTYPE";
    /**
     * The protocol given for an app url is not valid. (For example,
     * "android-app://")
     *
     * @generated from protobuf enum value: INVALID_PROTOCOL_FOR_APP_URL = 32;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_PROTOCOL_FOR_APP_URL"] = 32] = "INVALID_PROTOCOL_FOR_APP_URL";
    /**
     * The package id (app id) given for an app url is not valid.
     *
     * @generated from protobuf enum value: INVALID_PACKAGE_ID_FOR_APP_URL = 33;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_PACKAGE_ID_FOR_APP_URL"] = 33] = "INVALID_PACKAGE_ID_FOR_APP_URL";
    /**
     * The number of url custom parameters for an resource exceeds the maximum
     * limit allowed.
     *
     * @generated from protobuf enum value: URL_CUSTOM_PARAMETERS_COUNT_EXCEEDS_LIMIT = 34;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["URL_CUSTOM_PARAMETERS_COUNT_EXCEEDS_LIMIT"] = 34] = "URL_CUSTOM_PARAMETERS_COUNT_EXCEEDS_LIMIT";
    /**
     * An invalid character appears in the parameter key.
     *
     * @generated from protobuf enum value: INVALID_CHARACTERS_IN_URL_CUSTOM_PARAMETER_KEY = 39;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_CHARACTERS_IN_URL_CUSTOM_PARAMETER_KEY"] = 39] = "INVALID_CHARACTERS_IN_URL_CUSTOM_PARAMETER_KEY";
    /**
     * An invalid character appears in the parameter value.
     *
     * @generated from protobuf enum value: INVALID_CHARACTERS_IN_URL_CUSTOM_PARAMETER_VALUE = 40;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_CHARACTERS_IN_URL_CUSTOM_PARAMETER_VALUE"] = 40] = "INVALID_CHARACTERS_IN_URL_CUSTOM_PARAMETER_VALUE";
    /**
     * The url custom parameter value fails url tag validation.
     *
     * @generated from protobuf enum value: INVALID_TAG_IN_URL_CUSTOM_PARAMETER_VALUE = 41;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TAG_IN_URL_CUSTOM_PARAMETER_VALUE"] = 41] = "INVALID_TAG_IN_URL_CUSTOM_PARAMETER_VALUE";
    /**
     * The custom parameter contains nested occurrences of the same conditional
     * tag (for example, {ifmobile:{ifmobile:x}}).
     *
     * @generated from protobuf enum value: REDUNDANT_NESTED_URL_CUSTOM_PARAMETER_TAG = 42;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["REDUNDANT_NESTED_URL_CUSTOM_PARAMETER_TAG"] = 42] = "REDUNDANT_NESTED_URL_CUSTOM_PARAMETER_TAG";
    /**
     * The protocol (http:// or https://) is missing.
     *
     * @generated from protobuf enum value: MISSING_PROTOCOL = 43;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_PROTOCOL"] = 43] = "MISSING_PROTOCOL";
    /**
     * Unsupported protocol in URL. Only http and https are supported.
     *
     * @generated from protobuf enum value: INVALID_PROTOCOL = 52;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_PROTOCOL"] = 52] = "INVALID_PROTOCOL";
    /**
     * The url is invalid.
     *
     * @generated from protobuf enum value: INVALID_URL = 44;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_URL"] = 44] = "INVALID_URL";
    /**
     * Destination Url is deprecated.
     *
     * @generated from protobuf enum value: DESTINATION_URL_DEPRECATED = 45;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["DESTINATION_URL_DEPRECATED"] = 45] = "DESTINATION_URL_DEPRECATED";
    /**
     * The url contains invalid tag.
     *
     * @generated from protobuf enum value: INVALID_TAG_IN_URL = 46;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TAG_IN_URL"] = 46] = "INVALID_TAG_IN_URL";
    /**
     * The url must contain at least one tag (for example, {lpurl}).
     *
     * @generated from protobuf enum value: MISSING_URL_TAG = 47;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_URL_TAG"] = 47] = "MISSING_URL_TAG";
    /**
     * Duplicate url id.
     *
     * @generated from protobuf enum value: DUPLICATE_URL_ID = 48;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["DUPLICATE_URL_ID"] = 48] = "DUPLICATE_URL_ID";
    /**
     * Invalid url id.
     *
     * @generated from protobuf enum value: INVALID_URL_ID = 49;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_URL_ID"] = 49] = "INVALID_URL_ID";
    /**
     * The final url suffix cannot begin with '?' or '&' characters and must be
     * a valid query string.
     *
     * @generated from protobuf enum value: FINAL_URL_SUFFIX_MALFORMED = 50;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["FINAL_URL_SUFFIX_MALFORMED"] = 50] = "FINAL_URL_SUFFIX_MALFORMED";
    /**
     * The final url suffix cannot contain {lpurl} related or {ignore} tags.
     *
     * @generated from protobuf enum value: INVALID_TAG_IN_FINAL_URL_SUFFIX = 51;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TAG_IN_FINAL_URL_SUFFIX"] = 51] = "INVALID_TAG_IN_FINAL_URL_SUFFIX";
    /**
     * The top level domain is invalid, for example, not a public top level
     * domain listed in publicsuffix.org.
     *
     * @generated from protobuf enum value: INVALID_TOP_LEVEL_DOMAIN = 53;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["INVALID_TOP_LEVEL_DOMAIN"] = 53] = "INVALID_TOP_LEVEL_DOMAIN";
    /**
     * Malformed top level domain in URL.
     *
     * @generated from protobuf enum value: MALFORMED_TOP_LEVEL_DOMAIN = 54;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MALFORMED_TOP_LEVEL_DOMAIN"] = 54] = "MALFORMED_TOP_LEVEL_DOMAIN";
    /**
     * Malformed URL.
     *
     * @generated from protobuf enum value: MALFORMED_URL = 55;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MALFORMED_URL"] = 55] = "MALFORMED_URL";
    /**
     * No host found in URL.
     *
     * @generated from protobuf enum value: MISSING_HOST = 56;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["MISSING_HOST"] = 56] = "MISSING_HOST";
    /**
     * Custom parameter value cannot be null.
     *
     * @generated from protobuf enum value: NULL_CUSTOM_PARAMETER_VALUE = 57;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["NULL_CUSTOM_PARAMETER_VALUE"] = 57] = "NULL_CUSTOM_PARAMETER_VALUE";
    /**
     * Track parameter is not supported.
     *
     * @generated from protobuf enum value: VALUE_TRACK_PARAMETER_NOT_SUPPORTED = 58;
     */
    UrlFieldErrorEnum_UrlFieldError[UrlFieldErrorEnum_UrlFieldError["VALUE_TRACK_PARAMETER_NOT_SUPPORTED"] = 58] = "VALUE_TRACK_PARAMETER_NOT_SUPPORTED";
})(UrlFieldErrorEnum_UrlFieldError = exports.UrlFieldErrorEnum_UrlFieldError || (exports.UrlFieldErrorEnum_UrlFieldError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UrlFieldErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.UrlFieldErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.UrlFieldErrorEnum
 */
exports.UrlFieldErrorEnum = new UrlFieldErrorEnum$Type();
