// google-ads-api-client/dist/protos/bindings/ads/errors/user_data_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserDataErrorEnum = exports.UserDataErrorEnum_UserDataError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible request errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.UserDataErrorEnum.UserDataError
 */
var UserDataErrorEnum_UserDataError;
(function (UserDataErrorEnum_UserDataError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserDataErrorEnum_UserDataError[UserDataErrorEnum_UserDataError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserDataErrorEnum_UserDataError[UserDataErrorEnum_UserDataError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Customer is not allowed to perform operations related to Customer Match.
     *
     * @generated from protobuf enum value: OPERATIONS_FOR_CUSTOMER_MATCH_NOT_ALLOWED = 2;
     */
    UserDataErrorEnum_UserDataError[UserDataErrorEnum_UserDataError["OPERATIONS_FOR_CUSTOMER_MATCH_NOT_ALLOWED"] = 2] = "OPERATIONS_FOR_CUSTOMER_MATCH_NOT_ALLOWED";
    /**
     * Maximum number of user identifiers allowed for each request is 100 and
     * for each operation is 20.
     *
     * @generated from protobuf enum value: TOO_MANY_USER_IDENTIFIERS = 3;
     */
    UserDataErrorEnum_UserDataError[UserDataErrorEnum_UserDataError["TOO_MANY_USER_IDENTIFIERS"] = 3] = "TOO_MANY_USER_IDENTIFIERS";
    /**
     * Current user list is not applicable for the given customer.
     *
     * @generated from protobuf enum value: USER_LIST_NOT_APPLICABLE = 4;
     */
    UserDataErrorEnum_UserDataError[UserDataErrorEnum_UserDataError["USER_LIST_NOT_APPLICABLE"] = 4] = "USER_LIST_NOT_APPLICABLE";
})(UserDataErrorEnum_UserDataError = exports.UserDataErrorEnum_UserDataError || (exports.UserDataErrorEnum_UserDataError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserDataErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.UserDataErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.UserDataErrorEnum
 */
exports.UserDataErrorEnum = new UserDataErrorEnum$Type();
