// google-ads-api-client/dist/protos/bindings/ads/errors/user_list_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListErrorEnum = exports.UserListErrorEnum_UserListError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing possible user list errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.UserListErrorEnum.UserListError
 */
var UserListErrorEnum_UserListError;
(function (UserListErrorEnum_UserListError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Creating and updating external remarketing user lists is not supported.
     *
     * @generated from protobuf enum value: EXTERNAL_REMARKETING_USER_LIST_MUTATE_NOT_SUPPORTED = 2;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["EXTERNAL_REMARKETING_USER_LIST_MUTATE_NOT_SUPPORTED"] = 2] = "EXTERNAL_REMARKETING_USER_LIST_MUTATE_NOT_SUPPORTED";
    /**
     * Concrete type of user list is required.
     *
     * @generated from protobuf enum value: CONCRETE_TYPE_REQUIRED = 3;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["CONCRETE_TYPE_REQUIRED"] = 3] = "CONCRETE_TYPE_REQUIRED";
    /**
     * Creating/updating user list conversion types requires specifying the
     * conversion type Id.
     *
     * @generated from protobuf enum value: CONVERSION_TYPE_ID_REQUIRED = 4;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["CONVERSION_TYPE_ID_REQUIRED"] = 4] = "CONVERSION_TYPE_ID_REQUIRED";
    /**
     * Remarketing user list cannot have duplicate conversion types.
     *
     * @generated from protobuf enum value: DUPLICATE_CONVERSION_TYPES = 5;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["DUPLICATE_CONVERSION_TYPES"] = 5] = "DUPLICATE_CONVERSION_TYPES";
    /**
     * Conversion type is invalid/unknown.
     *
     * @generated from protobuf enum value: INVALID_CONVERSION_TYPE = 6;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["INVALID_CONVERSION_TYPE"] = 6] = "INVALID_CONVERSION_TYPE";
    /**
     * User list description is empty or invalid.
     *
     * @generated from protobuf enum value: INVALID_DESCRIPTION = 7;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["INVALID_DESCRIPTION"] = 7] = "INVALID_DESCRIPTION";
    /**
     * User list name is empty or invalid.
     *
     * @generated from protobuf enum value: INVALID_NAME = 8;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["INVALID_NAME"] = 8] = "INVALID_NAME";
    /**
     * Type of the UserList does not match.
     *
     * @generated from protobuf enum value: INVALID_TYPE = 9;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["INVALID_TYPE"] = 9] = "INVALID_TYPE";
    /**
     * Embedded logical user lists are not allowed.
     *
     * @generated from protobuf enum value: CAN_NOT_ADD_LOGICAL_LIST_AS_LOGICAL_LIST_OPERAND = 10;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["CAN_NOT_ADD_LOGICAL_LIST_AS_LOGICAL_LIST_OPERAND"] = 10] = "CAN_NOT_ADD_LOGICAL_LIST_AS_LOGICAL_LIST_OPERAND";
    /**
     * User list rule operand is invalid.
     *
     * @generated from protobuf enum value: INVALID_USER_LIST_LOGICAL_RULE_OPERAND = 11;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["INVALID_USER_LIST_LOGICAL_RULE_OPERAND"] = 11] = "INVALID_USER_LIST_LOGICAL_RULE_OPERAND";
    /**
     * Name is already being used for another user list for the account.
     *
     * @generated from protobuf enum value: NAME_ALREADY_USED = 12;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["NAME_ALREADY_USED"] = 12] = "NAME_ALREADY_USED";
    /**
     * Name is required when creating a new conversion type.
     *
     * @generated from protobuf enum value: NEW_CONVERSION_TYPE_NAME_REQUIRED = 13;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["NEW_CONVERSION_TYPE_NAME_REQUIRED"] = 13] = "NEW_CONVERSION_TYPE_NAME_REQUIRED";
    /**
     * The given conversion type name has been used.
     *
     * @generated from protobuf enum value: CONVERSION_TYPE_NAME_ALREADY_USED = 14;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["CONVERSION_TYPE_NAME_ALREADY_USED"] = 14] = "CONVERSION_TYPE_NAME_ALREADY_USED";
    /**
     * Only an owner account may edit a user list.
     *
     * @generated from protobuf enum value: OWNERSHIP_REQUIRED_FOR_SET = 15;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["OWNERSHIP_REQUIRED_FOR_SET"] = 15] = "OWNERSHIP_REQUIRED_FOR_SET";
    /**
     * Creating user list without setting type in oneof user_list field, or
     * creating/updating read-only user list types is not allowed.
     *
     * @generated from protobuf enum value: USER_LIST_MUTATE_NOT_SUPPORTED = 16;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["USER_LIST_MUTATE_NOT_SUPPORTED"] = 16] = "USER_LIST_MUTATE_NOT_SUPPORTED";
    /**
     * Rule is invalid.
     *
     * @generated from protobuf enum value: INVALID_RULE = 17;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["INVALID_RULE"] = 17] = "INVALID_RULE";
    /**
     * The specified date range is empty.
     *
     * @generated from protobuf enum value: INVALID_DATE_RANGE = 27;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["INVALID_DATE_RANGE"] = 27] = "INVALID_DATE_RANGE";
    /**
     * A UserList which is privacy sensitive or legal rejected cannot be mutated
     * by external users.
     *
     * @generated from protobuf enum value: CAN_NOT_MUTATE_SENSITIVE_USERLIST = 28;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["CAN_NOT_MUTATE_SENSITIVE_USERLIST"] = 28] = "CAN_NOT_MUTATE_SENSITIVE_USERLIST";
    /**
     * Maximum number of rulebased user lists a customer can have.
     *
     * @generated from protobuf enum value: MAX_NUM_RULEBASED_USERLISTS = 29;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["MAX_NUM_RULEBASED_USERLISTS"] = 29] = "MAX_NUM_RULEBASED_USERLISTS";
    /**
     * BasicUserList's billable record field cannot be modified once it is set.
     *
     * @generated from protobuf enum value: CANNOT_MODIFY_BILLABLE_RECORD_COUNT = 30;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["CANNOT_MODIFY_BILLABLE_RECORD_COUNT"] = 30] = "CANNOT_MODIFY_BILLABLE_RECORD_COUNT";
    /**
     * crm_based_user_list.app_id field must be set when upload_key_type is
     * MOBILE_ADVERTISING_ID.
     *
     * @generated from protobuf enum value: APP_ID_NOT_SET = 31;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["APP_ID_NOT_SET"] = 31] = "APP_ID_NOT_SET";
    /**
     * Name of the user list is reserved for system generated lists and cannot
     * be used.
     *
     * @generated from protobuf enum value: USERLIST_NAME_IS_RESERVED_FOR_SYSTEM_LIST = 32;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["USERLIST_NAME_IS_RESERVED_FOR_SYSTEM_LIST"] = 32] = "USERLIST_NAME_IS_RESERVED_FOR_SYSTEM_LIST";
    /**
     * Advertiser needs to be on the allow-list to use remarketing lists created
     * from advertiser uploaded data (for example, Customer Match lists).
     *
     * @generated from protobuf enum value: ADVERTISER_NOT_ON_ALLOWLIST_FOR_USING_UPLOADED_DATA = 37;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["ADVERTISER_NOT_ON_ALLOWLIST_FOR_USING_UPLOADED_DATA"] = 37] = "ADVERTISER_NOT_ON_ALLOWLIST_FOR_USING_UPLOADED_DATA";
    /**
     * The provided rule_type is not supported for the user list.
     *
     * @generated from protobuf enum value: RULE_TYPE_IS_NOT_SUPPORTED = 34;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["RULE_TYPE_IS_NOT_SUPPORTED"] = 34] = "RULE_TYPE_IS_NOT_SUPPORTED";
    /**
     * Similar user list cannot be used as a logical user list operand.
     *
     * @generated from protobuf enum value: CAN_NOT_ADD_A_SIMILAR_USERLIST_AS_LOGICAL_LIST_OPERAND = 35;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["CAN_NOT_ADD_A_SIMILAR_USERLIST_AS_LOGICAL_LIST_OPERAND"] = 35] = "CAN_NOT_ADD_A_SIMILAR_USERLIST_AS_LOGICAL_LIST_OPERAND";
    /**
     * Logical user list should not have a mix of CRM based user list and other
     * types of lists in its rules.
     *
     * @generated from protobuf enum value: CAN_NOT_MIX_CRM_BASED_IN_LOGICAL_LIST_WITH_OTHER_LISTS = 36;
     */
    UserListErrorEnum_UserListError[UserListErrorEnum_UserListError["CAN_NOT_MIX_CRM_BASED_IN_LOGICAL_LIST_WITH_OTHER_LISTS"] = 36] = "CAN_NOT_MIX_CRM_BASED_IN_LOGICAL_LIST_WITH_OTHER_LISTS";
})(UserListErrorEnum_UserListError = exports.UserListErrorEnum_UserListError || (exports.UserListErrorEnum_UserListError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class UserListErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.UserListErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.UserListErrorEnum
 */
exports.UserListErrorEnum = new UserListErrorEnum$Type();
