// google-ads-api-client/dist/protos/bindings/ads/errors/youtube_video_registration_error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.YoutubeVideoRegistrationErrorEnum = exports.YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
/**
 * Enum describing YouTube video registration errors.
 *
 * @generated from protobuf enum google.ads.googleads.v11.errors.YoutubeVideoRegistrationErrorEnum.YoutubeVideoRegistrationError
 */
var YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError;
(function (YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError) {
    /**
     * Enum unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError[YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * The received error code is not known in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError[YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Video to be registered wasn't found.
     *
     * @generated from protobuf enum value: VIDEO_NOT_FOUND = 2;
     */
    YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError[YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError["VIDEO_NOT_FOUND"] = 2] = "VIDEO_NOT_FOUND";
    /**
     * Video to be registered is not accessible (for example, private).
     *
     * @generated from protobuf enum value: VIDEO_NOT_ACCESSIBLE = 3;
     */
    YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError[YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError["VIDEO_NOT_ACCESSIBLE"] = 3] = "VIDEO_NOT_ACCESSIBLE";
    /**
     * Video to be registered is not eligible (for example, mature content).
     *
     * @generated from protobuf enum value: VIDEO_NOT_ELIGIBLE = 4;
     */
    YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError[YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError["VIDEO_NOT_ELIGIBLE"] = 4] = "VIDEO_NOT_ELIGIBLE";
})(YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError = exports.YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError || (exports.YoutubeVideoRegistrationErrorEnum_YoutubeVideoRegistrationError = {}));
// @generated message type with reflection information, may provide speed optimized methods
class YoutubeVideoRegistrationErrorEnum$Type extends runtime_4.MessageType {
    constructor() {
        super("google.ads.googleads.v11.errors.YoutubeVideoRegistrationErrorEnum", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_3.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_2.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.errors.YoutubeVideoRegistrationErrorEnum
 */
exports.YoutubeVideoRegistrationErrorEnum = new YoutubeVideoRegistrationErrorEnum$Type();
