// google-ads-api-client/dist/protos/bindings/ads/services/account_link_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountLinkService = exports.MutateAccountLinkResult = exports.MutateAccountLinkResponse = exports.AccountLinkOperation = exports.MutateAccountLinkRequest = exports.CreateAccountLinkResponse = exports.CreateAccountLinkRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/account_link_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const status_1 = require("../../google/rpc/status");
const field_mask_1 = require("../../google/protobuf/field_mask");
const account_link_1 = require("../resources/account_link");
// @generated message type with reflection information, may provide speed optimized methods
class CreateAccountLinkRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.CreateAccountLinkRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "account_link", kind: "message", T: () => account_link_1.AccountLink, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { customerId: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.AccountLink account_link */ 2:
                    message.accountLink = account_link_1.AccountLink.internalBinaryRead(reader, reader.uint32(), options, message.accountLink);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* google.ads.googleads.v11.resources.AccountLink account_link = 2; */
        if (message.accountLink)
            account_link_1.AccountLink.internalBinaryWrite(message.accountLink, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.CreateAccountLinkRequest
 */
exports.CreateAccountLinkRequest = new CreateAccountLinkRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CreateAccountLinkResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.CreateAccountLinkResponse", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/AccountLink" } } }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.CreateAccountLinkResponse
 */
exports.CreateAccountLinkResponse = new CreateAccountLinkResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateAccountLinkRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateAccountLinkRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "operation", kind: "message", T: () => exports.AccountLinkOperation, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "partial_failure", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { customerId: "", partialFailure: false, validateOnly: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.AccountLinkOperation operation */ 2:
                    message.operation = exports.AccountLinkOperation.internalBinaryRead(reader, reader.uint32(), options, message.operation);
                    break;
                case /* bool partial_failure */ 3:
                    message.partialFailure = reader.bool();
                    break;
                case /* bool validate_only */ 4:
                    message.validateOnly = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* google.ads.googleads.v11.services.AccountLinkOperation operation = 2; */
        if (message.operation)
            exports.AccountLinkOperation.internalBinaryWrite(message.operation, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool partial_failure = 3; */
        if (message.partialFailure !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.partialFailure);
        /* bool validate_only = 4; */
        if (message.validateOnly !== false)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.validateOnly);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateAccountLinkRequest
 */
exports.MutateAccountLinkRequest = new MutateAccountLinkRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AccountLinkOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.AccountLinkOperation", [
            { no: 4, name: "update_mask", kind: "message", T: () => field_mask_1.FieldMask },
            { no: 2, name: "update", kind: "message", oneof: "operation", T: () => account_link_1.AccountLink },
            { no: 3, name: "remove", kind: "scalar", oneof: "operation", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/AccountLink" } } }
        ]);
    }
    create(value) {
        const message = { operation: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.protobuf.FieldMask update_mask */ 4:
                    message.updateMask = field_mask_1.FieldMask.internalBinaryRead(reader, reader.uint32(), options, message.updateMask);
                    break;
                case /* google.ads.googleads.v11.resources.AccountLink update */ 2:
                    message.operation = {
                        oneofKind: "update",
                        update: account_link_1.AccountLink.internalBinaryRead(reader, reader.uint32(), options, message.operation.update)
                    };
                    break;
                case /* string remove */ 3:
                    message.operation = {
                        oneofKind: "remove",
                        remove: reader.string()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.protobuf.FieldMask update_mask = 4; */
        if (message.updateMask)
            field_mask_1.FieldMask.internalBinaryWrite(message.updateMask, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AccountLink update = 2; */
        if (message.operation.oneofKind === "update")
            account_link_1.AccountLink.internalBinaryWrite(message.operation.update, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string remove = 3; */
        if (message.operation.oneofKind === "remove")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.operation.remove);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.AccountLinkOperation
 */
exports.AccountLinkOperation = new AccountLinkOperation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateAccountLinkResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateAccountLinkResponse", [
            { no: 1, name: "result", kind: "message", T: () => exports.MutateAccountLinkResult },
            { no: 2, name: "partial_failure_error", kind: "message", T: () => status_1.Status }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.services.MutateAccountLinkResult result */ 1:
                    message.result = exports.MutateAccountLinkResult.internalBinaryRead(reader, reader.uint32(), options, message.result);
                    break;
                case /* google.rpc.Status partial_failure_error */ 2:
                    message.partialFailureError = status_1.Status.internalBinaryRead(reader, reader.uint32(), options, message.partialFailureError);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.services.MutateAccountLinkResult result = 1; */
        if (message.result)
            exports.MutateAccountLinkResult.internalBinaryWrite(message.result, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.rpc.Status partial_failure_error = 2; */
        if (message.partialFailureError)
            status_1.Status.internalBinaryWrite(message.partialFailureError, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateAccountLinkResponse
 */
exports.MutateAccountLinkResponse = new MutateAccountLinkResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateAccountLinkResult$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateAccountLinkResult", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/AccountLink" } } }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateAccountLinkResult
 */
exports.MutateAccountLinkResult = new MutateAccountLinkResult$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.AccountLinkService
 */
exports.AccountLinkService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.AccountLinkService", [
    { name: "CreateAccountLink", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}/accountLinks:create", body: "*" }, "google.api.method_signature": ["customer_id,account_link"] }, I: exports.CreateAccountLinkRequest, O: exports.CreateAccountLinkResponse },
    { name: "MutateAccountLink", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}/accountLinks:mutate", body: "*" }, "google.api.method_signature": ["customer_id,operation"] }, I: exports.MutateAccountLinkRequest, O: exports.MutateAccountLinkResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
