// google-ads-api-client/dist/protos/bindings/ads/services/account_link_service.client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountLinkServiceClient = void 0;
const account_link_service_1 = require("./account_link_service");
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
/**
 * This service allows management of links between Google Ads accounts and other
 * accounts.
 *
 * @generated from protobuf service google.ads.googleads.v11.services.AccountLinkService
 */
class AccountLinkServiceClient {
    constructor(_transport) {
        this._transport = _transport;
        this.typeName = account_link_service_1.AccountLinkService.typeName;
        this.methods = account_link_service_1.AccountLinkService.methods;
        this.options = account_link_service_1.AccountLinkService.options;
    }
    /**
     * Creates an account link.
     *
     * List of thrown errors:
     *   [AuthenticationError]()
     *   [AuthorizationError]()
     *   [DatabaseError]()
     *   [FieldError]()
     *   [HeaderError]()
     *   [InternalError]()
     *   [MutateError]()
     *   [QuotaError]()
     *   [RequestError]()
     *   [ThirdPartyAppAnalyticsLinkError]()
     *
     * @generated from protobuf rpc: CreateAccountLink(google.ads.googleads.v11.services.CreateAccountLinkRequest) returns (google.ads.googleads.v11.services.CreateAccountLinkResponse);
     */
    createAccountLink(input, options) {
        const method = this.methods[0], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Creates or removes an account link.
     * From V5, create is not supported through
     * AccountLinkService.MutateAccountLink. Use
     * AccountLinkService.CreateAccountLink instead.
     *
     * List of thrown errors:
     *   [AccountLinkError]()
     *   [AuthenticationError]()
     *   [AuthorizationError]()
     *   [FieldMaskError]()
     *   [HeaderError]()
     *   [InternalError]()
     *   [MutateError]()
     *   [QuotaError]()
     *   [RequestError]()
     *
     * @generated from protobuf rpc: MutateAccountLink(google.ads.googleads.v11.services.MutateAccountLinkRequest) returns (google.ads.googleads.v11.services.MutateAccountLinkResponse);
     */
    mutateAccountLink(input, options) {
        const method = this.methods[1], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
}
exports.AccountLinkServiceClient = AccountLinkServiceClient;
