// google-ads-api-client/dist/protos/bindings/ads/services/ad_group_ad_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupAdService = exports.MutateAdGroupAdResult = exports.MutateAdGroupAdsResponse = exports.AdGroupAdOperation = exports.MutateAdGroupAdsRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/ad_group_ad_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const status_1 = require("../../google/rpc/status");
const ad_group_ad_1 = require("../resources/ad_group_ad");
const policy_1 = require("../common/policy");
const field_mask_1 = require("../../google/protobuf/field_mask");
const response_content_type_1 = require("../enums/response_content_type");
// @generated message type with reflection information, may provide speed optimized methods
class MutateAdGroupAdsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateAdGroupAdsRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "operations", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.AdGroupAdOperation, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "partial_failure", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 5, name: "response_content_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType", response_content_type_1.ResponseContentTypeEnum_ResponseContentType] }
        ]);
    }
    create(value) {
        const message = { customerId: "", operations: [], partialFailure: false, validateOnly: false, responseContentType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.services.AdGroupAdOperation operations */ 2:
                    message.operations.push(exports.AdGroupAdOperation.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* bool partial_failure */ 3:
                    message.partialFailure = reader.bool();
                    break;
                case /* bool validate_only */ 4:
                    message.validateOnly = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType response_content_type */ 5:
                    message.responseContentType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* repeated google.ads.googleads.v11.services.AdGroupAdOperation operations = 2; */
        for (let i = 0; i < message.operations.length; i++)
            exports.AdGroupAdOperation.internalBinaryWrite(message.operations[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool partial_failure = 3; */
        if (message.partialFailure !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.partialFailure);
        /* bool validate_only = 4; */
        if (message.validateOnly !== false)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.validateOnly);
        /* google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType response_content_type = 5; */
        if (message.responseContentType !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.responseContentType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateAdGroupAdsRequest
 */
exports.MutateAdGroupAdsRequest = new MutateAdGroupAdsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupAdOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.AdGroupAdOperation", [
            { no: 4, name: "update_mask", kind: "message", T: () => field_mask_1.FieldMask },
            { no: 5, name: "policy_validation_parameter", kind: "message", T: () => policy_1.PolicyValidationParameter },
            { no: 1, name: "create", kind: "message", oneof: "operation", T: () => ad_group_ad_1.AdGroupAd },
            { no: 2, name: "update", kind: "message", oneof: "operation", T: () => ad_group_ad_1.AdGroupAd },
            { no: 3, name: "remove", kind: "scalar", oneof: "operation", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupAd" } } }
        ]);
    }
    create(value) {
        const message = { operation: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.protobuf.FieldMask update_mask */ 4:
                    message.updateMask = field_mask_1.FieldMask.internalBinaryRead(reader, reader.uint32(), options, message.updateMask);
                    break;
                case /* google.ads.googleads.v11.common.PolicyValidationParameter policy_validation_parameter */ 5:
                    message.policyValidationParameter = policy_1.PolicyValidationParameter.internalBinaryRead(reader, reader.uint32(), options, message.policyValidationParameter);
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupAd create */ 1:
                    message.operation = {
                        oneofKind: "create",
                        create: ad_group_ad_1.AdGroupAd.internalBinaryRead(reader, reader.uint32(), options, message.operation.create)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupAd update */ 2:
                    message.operation = {
                        oneofKind: "update",
                        update: ad_group_ad_1.AdGroupAd.internalBinaryRead(reader, reader.uint32(), options, message.operation.update)
                    };
                    break;
                case /* string remove */ 3:
                    message.operation = {
                        oneofKind: "remove",
                        remove: reader.string()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.protobuf.FieldMask update_mask = 4; */
        if (message.updateMask)
            field_mask_1.FieldMask.internalBinaryWrite(message.updateMask, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PolicyValidationParameter policy_validation_parameter = 5; */
        if (message.policyValidationParameter)
            policy_1.PolicyValidationParameter.internalBinaryWrite(message.policyValidationParameter, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroupAd create = 1; */
        if (message.operation.oneofKind === "create")
            ad_group_ad_1.AdGroupAd.internalBinaryWrite(message.operation.create, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroupAd update = 2; */
        if (message.operation.oneofKind === "update")
            ad_group_ad_1.AdGroupAd.internalBinaryWrite(message.operation.update, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string remove = 3; */
        if (message.operation.oneofKind === "remove")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.operation.remove);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.AdGroupAdOperation
 */
exports.AdGroupAdOperation = new AdGroupAdOperation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateAdGroupAdsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateAdGroupAdsResponse", [
            { no: 3, name: "partial_failure_error", kind: "message", T: () => status_1.Status },
            { no: 2, name: "results", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.MutateAdGroupAdResult }
        ]);
    }
    create(value) {
        const message = { results: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.rpc.Status partial_failure_error */ 3:
                    message.partialFailureError = status_1.Status.internalBinaryRead(reader, reader.uint32(), options, message.partialFailureError);
                    break;
                case /* repeated google.ads.googleads.v11.services.MutateAdGroupAdResult results */ 2:
                    message.results.push(exports.MutateAdGroupAdResult.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.rpc.Status partial_failure_error = 3; */
        if (message.partialFailureError)
            status_1.Status.internalBinaryWrite(message.partialFailureError, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.services.MutateAdGroupAdResult results = 2; */
        for (let i = 0; i < message.results.length; i++)
            exports.MutateAdGroupAdResult.internalBinaryWrite(message.results[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateAdGroupAdsResponse
 */
exports.MutateAdGroupAdsResponse = new MutateAdGroupAdsResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateAdGroupAdResult$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateAdGroupAdResult", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupAd" } } },
            { no: 2, name: "ad_group_ad", kind: "message", T: () => ad_group_ad_1.AdGroupAd }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupAd ad_group_ad */ 2:
                    message.adGroupAd = ad_group_ad_1.AdGroupAd.internalBinaryRead(reader, reader.uint32(), options, message.adGroupAd);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* google.ads.googleads.v11.resources.AdGroupAd ad_group_ad = 2; */
        if (message.adGroupAd)
            ad_group_ad_1.AdGroupAd.internalBinaryWrite(message.adGroupAd, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateAdGroupAdResult
 */
exports.MutateAdGroupAdResult = new MutateAdGroupAdResult$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.AdGroupAdService
 */
exports.AdGroupAdService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.AdGroupAdService", [
    { name: "MutateAdGroupAds", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}/adGroupAds:mutate", body: "*" }, "google.api.method_signature": ["customer_id,operations"] }, I: exports.MutateAdGroupAdsRequest, O: exports.MutateAdGroupAdsResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
