// google-ads-api-client/dist/protos/bindings/ads/services/audience_insights_service.client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AudienceInsightsServiceClient = void 0;
const audience_insights_service_1 = require("./audience_insights_service");
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
// Proto file describing the audience insights service.
/**
 * Audience Insights Service helps users find information about groups of
 * people and how they can be reached with Google Ads.
 *
 * @generated from protobuf service google.ads.googleads.v11.services.AudienceInsightsService
 */
class AudienceInsightsServiceClient {
    constructor(_transport) {
        this._transport = _transport;
        this.typeName = audience_insights_service_1.AudienceInsightsService.typeName;
        this.methods = audience_insights_service_1.AudienceInsightsService.methods;
        this.options = audience_insights_service_1.AudienceInsightsService.options;
    }
    /**
     * Creates a saved report that can be viewed in the Insights Finder tool.
     *
     * List of thrown errors:
     *   [AuthenticationError]()
     *   [AuthorizationError]()
     *   [FieldError]()
     *   [HeaderError]()
     *   [InternalError]()
     *   [QuotaError]()
     *   [RangeError]()
     *   [RequestError]()
     *
     * @generated from protobuf rpc: GenerateInsightsFinderReport(google.ads.googleads.v11.services.GenerateInsightsFinderReportRequest) returns (google.ads.googleads.v11.services.GenerateInsightsFinderReportResponse);
     */
    generateInsightsFinderReport(input, options) {
        const method = this.methods[0], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Searches for audience attributes that can be used to generate insights.
     *
     * List of thrown errors:
     *   [AuthenticationError]()
     *   [AuthorizationError]()
     *   [FieldError]()
     *   [HeaderError]()
     *   [InternalError]()
     *   [QuotaError]()
     *   [RangeError]()
     *   [RequestError]()
     *
     * @generated from protobuf rpc: ListAudienceInsightsAttributes(google.ads.googleads.v11.services.ListAudienceInsightsAttributesRequest) returns (google.ads.googleads.v11.services.ListAudienceInsightsAttributesResponse);
     */
    listAudienceInsightsAttributes(input, options) {
        const method = this.methods[1], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Returns a collection of attributes that are represented in an audience of
     * interest, with metrics that compare each attribute's share of the audience
     * with its share of a baseline audience.
     *
     * List of thrown errors:
     *   [AudienceInsightsError]()
     *   [AuthenticationError]()
     *   [AuthorizationError]()
     *   [FieldError]()
     *   [HeaderError]()
     *   [InternalError]()
     *   [QuotaError]()
     *   [RangeError]()
     *   [RequestError]()
     *
     * @generated from protobuf rpc: GenerateAudienceCompositionInsights(google.ads.googleads.v11.services.GenerateAudienceCompositionInsightsRequest) returns (google.ads.googleads.v11.services.GenerateAudienceCompositionInsightsResponse);
     */
    generateAudienceCompositionInsights(input, options) {
        const method = this.methods[2], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
}
exports.AudienceInsightsServiceClient = AudienceInsightsServiceClient;
