// google-ads-api-client/dist/protos/bindings/ads/services/campaign_experiment_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignExperimentService = exports.ListCampaignExperimentAsyncErrorsResponse = exports.ListCampaignExperimentAsyncErrorsRequest = exports.EndCampaignExperimentRequest = exports.PromoteCampaignExperimentRequest = exports.GraduateCampaignExperimentResponse = exports.GraduateCampaignExperimentRequest = exports.CreateCampaignExperimentMetadata = exports.CreateCampaignExperimentRequest = exports.MutateCampaignExperimentResult = exports.MutateCampaignExperimentsResponse = exports.CampaignExperimentOperation = exports.MutateCampaignExperimentsRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/campaign_experiment_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const empty_1 = require("../../google/protobuf/empty");
const operations_1 = require("../../google/longrunning/operations");
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const status_1 = require("../../google/rpc/status");
const campaign_experiment_1 = require("../resources/campaign_experiment");
const field_mask_1 = require("../../google/protobuf/field_mask");
const response_content_type_1 = require("../enums/response_content_type");
// @generated message type with reflection information, may provide speed optimized methods
class MutateCampaignExperimentsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateCampaignExperimentsRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "operations", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.CampaignExperimentOperation, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "partial_failure", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 5, name: "response_content_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType", response_content_type_1.ResponseContentTypeEnum_ResponseContentType] }
        ]);
    }
    create(value) {
        const message = { customerId: "", operations: [], partialFailure: false, validateOnly: false, responseContentType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.services.CampaignExperimentOperation operations */ 2:
                    message.operations.push(exports.CampaignExperimentOperation.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* bool partial_failure */ 3:
                    message.partialFailure = reader.bool();
                    break;
                case /* bool validate_only */ 4:
                    message.validateOnly = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType response_content_type */ 5:
                    message.responseContentType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* repeated google.ads.googleads.v11.services.CampaignExperimentOperation operations = 2; */
        for (let i = 0; i < message.operations.length; i++)
            exports.CampaignExperimentOperation.internalBinaryWrite(message.operations[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool partial_failure = 3; */
        if (message.partialFailure !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.partialFailure);
        /* bool validate_only = 4; */
        if (message.validateOnly !== false)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.validateOnly);
        /* google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType response_content_type = 5; */
        if (message.responseContentType !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.responseContentType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateCampaignExperimentsRequest
 */
exports.MutateCampaignExperimentsRequest = new MutateCampaignExperimentsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CampaignExperimentOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.CampaignExperimentOperation", [
            { no: 3, name: "update_mask", kind: "message", T: () => field_mask_1.FieldMask },
            { no: 1, name: "update", kind: "message", oneof: "operation", T: () => campaign_experiment_1.CampaignExperiment },
            { no: 2, name: "remove", kind: "scalar", oneof: "operation", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignExperiment" } } }
        ]);
    }
    create(value) {
        const message = { operation: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.protobuf.FieldMask update_mask */ 3:
                    message.updateMask = field_mask_1.FieldMask.internalBinaryRead(reader, reader.uint32(), options, message.updateMask);
                    break;
                case /* google.ads.googleads.v11.resources.CampaignExperiment update */ 1:
                    message.operation = {
                        oneofKind: "update",
                        update: campaign_experiment_1.CampaignExperiment.internalBinaryRead(reader, reader.uint32(), options, message.operation.update)
                    };
                    break;
                case /* string remove */ 2:
                    message.operation = {
                        oneofKind: "remove",
                        remove: reader.string()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.protobuf.FieldMask update_mask = 3; */
        if (message.updateMask)
            field_mask_1.FieldMask.internalBinaryWrite(message.updateMask, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.CampaignExperiment update = 1; */
        if (message.operation.oneofKind === "update")
            campaign_experiment_1.CampaignExperiment.internalBinaryWrite(message.operation.update, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string remove = 2; */
        if (message.operation.oneofKind === "remove")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.operation.remove);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.CampaignExperimentOperation
 */
exports.CampaignExperimentOperation = new CampaignExperimentOperation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateCampaignExperimentsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateCampaignExperimentsResponse", [
            { no: 3, name: "partial_failure_error", kind: "message", T: () => status_1.Status },
            { no: 2, name: "results", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.MutateCampaignExperimentResult }
        ]);
    }
    create(value) {
        const message = { results: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.rpc.Status partial_failure_error */ 3:
                    message.partialFailureError = status_1.Status.internalBinaryRead(reader, reader.uint32(), options, message.partialFailureError);
                    break;
                case /* repeated google.ads.googleads.v11.services.MutateCampaignExperimentResult results */ 2:
                    message.results.push(exports.MutateCampaignExperimentResult.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.rpc.Status partial_failure_error = 3; */
        if (message.partialFailureError)
            status_1.Status.internalBinaryWrite(message.partialFailureError, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.services.MutateCampaignExperimentResult results = 2; */
        for (let i = 0; i < message.results.length; i++)
            exports.MutateCampaignExperimentResult.internalBinaryWrite(message.results[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateCampaignExperimentsResponse
 */
exports.MutateCampaignExperimentsResponse = new MutateCampaignExperimentsResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateCampaignExperimentResult$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateCampaignExperimentResult", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignExperiment" } } },
            { no: 2, name: "campaign_experiment", kind: "message", T: () => campaign_experiment_1.CampaignExperiment }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.CampaignExperiment campaign_experiment */ 2:
                    message.campaignExperiment = campaign_experiment_1.CampaignExperiment.internalBinaryRead(reader, reader.uint32(), options, message.campaignExperiment);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* google.ads.googleads.v11.resources.CampaignExperiment campaign_experiment = 2; */
        if (message.campaignExperiment)
            campaign_experiment_1.CampaignExperiment.internalBinaryWrite(message.campaignExperiment, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateCampaignExperimentResult
 */
exports.MutateCampaignExperimentResult = new MutateCampaignExperimentResult$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CreateCampaignExperimentRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.CreateCampaignExperimentRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "campaign_experiment", kind: "message", T: () => campaign_experiment_1.CampaignExperiment, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { customerId: "", validateOnly: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.CampaignExperiment campaign_experiment */ 2:
                    message.campaignExperiment = campaign_experiment_1.CampaignExperiment.internalBinaryRead(reader, reader.uint32(), options, message.campaignExperiment);
                    break;
                case /* bool validate_only */ 3:
                    message.validateOnly = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* google.ads.googleads.v11.resources.CampaignExperiment campaign_experiment = 2; */
        if (message.campaignExperiment)
            campaign_experiment_1.CampaignExperiment.internalBinaryWrite(message.campaignExperiment, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool validate_only = 3; */
        if (message.validateOnly !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.validateOnly);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.CreateCampaignExperimentRequest
 */
exports.CreateCampaignExperimentRequest = new CreateCampaignExperimentRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CreateCampaignExperimentMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.CreateCampaignExperimentMetadata", [
            { no: 1, name: "campaign_experiment", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { campaignExperiment: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string campaign_experiment */ 1:
                    message.campaignExperiment = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string campaign_experiment = 1; */
        if (message.campaignExperiment !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.campaignExperiment);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.CreateCampaignExperimentMetadata
 */
exports.CreateCampaignExperimentMetadata = new CreateCampaignExperimentMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GraduateCampaignExperimentRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GraduateCampaignExperimentRequest", [
            { no: 1, name: "campaign_experiment", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignExperiment" } } },
            { no: 2, name: "campaign_budget", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { campaignExperiment: "", campaignBudget: "", validateOnly: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string campaign_experiment */ 1:
                    message.campaignExperiment = reader.string();
                    break;
                case /* string campaign_budget */ 2:
                    message.campaignBudget = reader.string();
                    break;
                case /* bool validate_only */ 3:
                    message.validateOnly = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string campaign_experiment = 1; */
        if (message.campaignExperiment !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.campaignExperiment);
        /* string campaign_budget = 2; */
        if (message.campaignBudget !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.campaignBudget);
        /* bool validate_only = 3; */
        if (message.validateOnly !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.validateOnly);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GraduateCampaignExperimentRequest
 */
exports.GraduateCampaignExperimentRequest = new GraduateCampaignExperimentRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GraduateCampaignExperimentResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GraduateCampaignExperimentResponse", [
            { no: 1, name: "graduated_campaign", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { graduatedCampaign: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string graduated_campaign */ 1:
                    message.graduatedCampaign = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string graduated_campaign = 1; */
        if (message.graduatedCampaign !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.graduatedCampaign);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GraduateCampaignExperimentResponse
 */
exports.GraduateCampaignExperimentResponse = new GraduateCampaignExperimentResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PromoteCampaignExperimentRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.PromoteCampaignExperimentRequest", [
            { no: 1, name: "campaign_experiment", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignExperiment" } } },
            { no: 2, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { campaignExperiment: "", validateOnly: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string campaign_experiment */ 1:
                    message.campaignExperiment = reader.string();
                    break;
                case /* bool validate_only */ 2:
                    message.validateOnly = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string campaign_experiment = 1; */
        if (message.campaignExperiment !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.campaignExperiment);
        /* bool validate_only = 2; */
        if (message.validateOnly !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.validateOnly);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.PromoteCampaignExperimentRequest
 */
exports.PromoteCampaignExperimentRequest = new PromoteCampaignExperimentRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class EndCampaignExperimentRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.EndCampaignExperimentRequest", [
            { no: 1, name: "campaign_experiment", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignExperiment" } } },
            { no: 2, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { campaignExperiment: "", validateOnly: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string campaign_experiment */ 1:
                    message.campaignExperiment = reader.string();
                    break;
                case /* bool validate_only */ 2:
                    message.validateOnly = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string campaign_experiment = 1; */
        if (message.campaignExperiment !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.campaignExperiment);
        /* bool validate_only = 2; */
        if (message.validateOnly !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.validateOnly);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.EndCampaignExperimentRequest
 */
exports.EndCampaignExperimentRequest = new EndCampaignExperimentRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListCampaignExperimentAsyncErrorsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ListCampaignExperimentAsyncErrorsRequest", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignExperiment" } } },
            { no: 2, name: "page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "page_size", kind: "scalar", T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = { resourceName: "", pageToken: "", pageSize: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* string page_token */ 2:
                    message.pageToken = reader.string();
                    break;
                case /* int32 page_size */ 3:
                    message.pageSize = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* string page_token = 2; */
        if (message.pageToken !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.pageToken);
        /* int32 page_size = 3; */
        if (message.pageSize !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.pageSize);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ListCampaignExperimentAsyncErrorsRequest
 */
exports.ListCampaignExperimentAsyncErrorsRequest = new ListCampaignExperimentAsyncErrorsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListCampaignExperimentAsyncErrorsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ListCampaignExperimentAsyncErrorsResponse", [
            { no: 1, name: "errors", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => status_1.Status },
            { no: 2, name: "next_page_token", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { errors: [], nextPageToken: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.rpc.Status errors */ 1:
                    message.errors.push(status_1.Status.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string next_page_token */ 2:
                    message.nextPageToken = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.rpc.Status errors = 1; */
        for (let i = 0; i < message.errors.length; i++)
            status_1.Status.internalBinaryWrite(message.errors[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string next_page_token = 2; */
        if (message.nextPageToken !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.nextPageToken);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ListCampaignExperimentAsyncErrorsResponse
 */
exports.ListCampaignExperimentAsyncErrorsResponse = new ListCampaignExperimentAsyncErrorsResponse$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.CampaignExperimentService
 */
exports.CampaignExperimentService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.CampaignExperimentService", [
    { name: "CreateCampaignExperiment", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}/campaignExperiments:create", body: "*" }, "google.api.method_signature": ["customer_id,campaign_experiment"], "google.longrunning.operation_info": { responseType: "google.protobuf.Empty", metadataType: "google.ads.googleads.v11.services.CreateCampaignExperimentMetadata" } }, I: exports.CreateCampaignExperimentRequest, O: operations_1.Operation },
    { name: "MutateCampaignExperiments", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}/campaignExperiments:mutate", body: "*" }, "google.api.method_signature": ["customer_id,operations"] }, I: exports.MutateCampaignExperimentsRequest, O: exports.MutateCampaignExperimentsResponse },
    { name: "GraduateCampaignExperiment", options: { "google.api.http": { post: "/v11/{campaign_experiment=customers/*/campaignExperiments/*}:graduate", body: "*" }, "google.api.method_signature": ["campaign_experiment,campaign_budget"] }, I: exports.GraduateCampaignExperimentRequest, O: exports.GraduateCampaignExperimentResponse },
    { name: "PromoteCampaignExperiment", options: { "google.api.http": { post: "/v11/{campaign_experiment=customers/*/campaignExperiments/*}:promote", body: "*" }, "google.api.method_signature": ["campaign_experiment"], "google.longrunning.operation_info": { responseType: "google.protobuf.Empty", metadataType: "google.protobuf.Empty" } }, I: exports.PromoteCampaignExperimentRequest, O: operations_1.Operation },
    { name: "EndCampaignExperiment", options: { "google.api.http": { post: "/v11/{campaign_experiment=customers/*/campaignExperiments/*}:end", body: "*" }, "google.api.method_signature": ["campaign_experiment"] }, I: exports.EndCampaignExperimentRequest, O: empty_1.Empty },
    { name: "ListCampaignExperimentAsyncErrors", options: { "google.api.http": { get: "/v11/{resource_name=customers/*/campaignExperiments/*}:listAsyncErrors" }, "google.api.method_signature": ["resource_name"] }, I: exports.ListCampaignExperimentAsyncErrorsRequest, O: exports.ListCampaignExperimentAsyncErrorsResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
