// google-ads-api-client/dist/protos/bindings/ads/services/customer_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerService = exports.ListAccessibleCustomersResponse = exports.ListAccessibleCustomersRequest = exports.MutateCustomerResult = exports.MutateCustomerResponse = exports.CreateCustomerClientResponse = exports.CustomerOperation = exports.CreateCustomerClientRequest = exports.MutateCustomerRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/customer_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const field_mask_1 = require("../../google/protobuf/field_mask");
const access_role_1 = require("../enums/access_role");
const customer_1 = require("../resources/customer");
const response_content_type_1 = require("../enums/response_content_type");
// @generated message type with reflection information, may provide speed optimized methods
class MutateCustomerRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateCustomerRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "operation", kind: "message", T: () => exports.CustomerOperation, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 5, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 6, name: "response_content_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType", response_content_type_1.ResponseContentTypeEnum_ResponseContentType] }
        ]);
    }
    create(value) {
        const message = { customerId: "", validateOnly: false, responseContentType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.CustomerOperation operation */ 4:
                    message.operation = exports.CustomerOperation.internalBinaryRead(reader, reader.uint32(), options, message.operation);
                    break;
                case /* bool validate_only */ 5:
                    message.validateOnly = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType response_content_type */ 6:
                    message.responseContentType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* google.ads.googleads.v11.services.CustomerOperation operation = 4; */
        if (message.operation)
            exports.CustomerOperation.internalBinaryWrite(message.operation, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool validate_only = 5; */
        if (message.validateOnly !== false)
            writer.tag(5, runtime_1.WireType.Varint).bool(message.validateOnly);
        /* google.ads.googleads.v11.enums.ResponseContentTypeEnum.ResponseContentType response_content_type = 6; */
        if (message.responseContentType !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.responseContentType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateCustomerRequest
 */
exports.MutateCustomerRequest = new MutateCustomerRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CreateCustomerClientRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.CreateCustomerClientRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "customer_client", kind: "message", T: () => customer_1.Customer, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 5, name: "email_address", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "access_role", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AccessRoleEnum.AccessRole", access_role_1.AccessRoleEnum_AccessRole] },
            { no: 6, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { customerId: "", accessRole: 0, validateOnly: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.Customer customer_client */ 2:
                    message.customerClient = customer_1.Customer.internalBinaryRead(reader, reader.uint32(), options, message.customerClient);
                    break;
                case /* optional string email_address */ 5:
                    message.emailAddress = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AccessRoleEnum.AccessRole access_role */ 4:
                    message.accessRole = reader.int32();
                    break;
                case /* bool validate_only */ 6:
                    message.validateOnly = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* google.ads.googleads.v11.resources.Customer customer_client = 2; */
        if (message.customerClient)
            customer_1.Customer.internalBinaryWrite(message.customerClient, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string email_address = 5; */
        if (message.emailAddress !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.emailAddress);
        /* google.ads.googleads.v11.enums.AccessRoleEnum.AccessRole access_role = 4; */
        if (message.accessRole !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.accessRole);
        /* bool validate_only = 6; */
        if (message.validateOnly !== false)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.validateOnly);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.CreateCustomerClientRequest
 */
exports.CreateCustomerClientRequest = new CreateCustomerClientRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CustomerOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.CustomerOperation", [
            { no: 1, name: "update", kind: "message", T: () => customer_1.Customer },
            { no: 2, name: "update_mask", kind: "message", T: () => field_mask_1.FieldMask }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Customer update */ 1:
                    message.update = customer_1.Customer.internalBinaryRead(reader, reader.uint32(), options, message.update);
                    break;
                case /* google.protobuf.FieldMask update_mask */ 2:
                    message.updateMask = field_mask_1.FieldMask.internalBinaryRead(reader, reader.uint32(), options, message.updateMask);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Customer update = 1; */
        if (message.update)
            customer_1.Customer.internalBinaryWrite(message.update, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.protobuf.FieldMask update_mask = 2; */
        if (message.updateMask)
            field_mask_1.FieldMask.internalBinaryWrite(message.updateMask, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.CustomerOperation
 */
exports.CustomerOperation = new CustomerOperation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CreateCustomerClientResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.CreateCustomerClientResponse", [
            { no: 2, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/Customer" } } },
            { no: 3, name: "invitation_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { resourceName: "", invitationLink: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 2:
                    message.resourceName = reader.string();
                    break;
                case /* string invitation_link */ 3:
                    message.invitationLink = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 2; */
        if (message.resourceName !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* string invitation_link = 3; */
        if (message.invitationLink !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.invitationLink);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.CreateCustomerClientResponse
 */
exports.CreateCustomerClientResponse = new CreateCustomerClientResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateCustomerResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateCustomerResponse", [
            { no: 2, name: "result", kind: "message", T: () => exports.MutateCustomerResult }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.services.MutateCustomerResult result */ 2:
                    message.result = exports.MutateCustomerResult.internalBinaryRead(reader, reader.uint32(), options, message.result);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.services.MutateCustomerResult result = 2; */
        if (message.result)
            exports.MutateCustomerResult.internalBinaryWrite(message.result, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateCustomerResponse
 */
exports.MutateCustomerResponse = new MutateCustomerResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateCustomerResult$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateCustomerResult", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/Customer" } } },
            { no: 2, name: "customer", kind: "message", T: () => customer_1.Customer }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.Customer customer */ 2:
                    message.customer = customer_1.Customer.internalBinaryRead(reader, reader.uint32(), options, message.customer);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* google.ads.googleads.v11.resources.Customer customer = 2; */
        if (message.customer)
            customer_1.Customer.internalBinaryWrite(message.customer, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateCustomerResult
 */
exports.MutateCustomerResult = new MutateCustomerResult$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListAccessibleCustomersRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ListAccessibleCustomersRequest", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ListAccessibleCustomersRequest
 */
exports.ListAccessibleCustomersRequest = new ListAccessibleCustomersRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListAccessibleCustomersResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ListAccessibleCustomersResponse", [
            { no: 1, name: "resource_names", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { resourceNames: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated string resource_names */ 1:
                    message.resourceNames.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated string resource_names = 1; */
        for (let i = 0; i < message.resourceNames.length; i++)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceNames[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ListAccessibleCustomersResponse
 */
exports.ListAccessibleCustomersResponse = new ListAccessibleCustomersResponse$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.CustomerService
 */
exports.CustomerService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.CustomerService", [
    { name: "MutateCustomer", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}:mutate", body: "*" }, "google.api.method_signature": ["customer_id,operation"] }, I: exports.MutateCustomerRequest, O: exports.MutateCustomerResponse },
    { name: "ListAccessibleCustomers", options: { "google.api.http": { get: "/v11/customers:listAccessibleCustomers" } }, I: exports.ListAccessibleCustomersRequest, O: exports.ListAccessibleCustomersResponse },
    { name: "CreateCustomerClient", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}:createCustomerClient", body: "*" }, "google.api.method_signature": ["customer_id,customer_client"] }, I: exports.CreateCustomerClientRequest, O: exports.CreateCustomerClientResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
