// google-ads-api-client/dist/protos/bindings/ads/services/keyword_plan_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanService = exports.KeywordPlanKeywordHistoricalMetrics = exports.GenerateHistoricalMetricsResponse = exports.GenerateHistoricalMetricsRequest = exports.ForecastMetrics = exports.KeywordPlanWeeklyForecast = exports.KeywordPlanWeeklyTimeSeriesForecast = exports.KeywordPlanMaxCpcBidForecast = exports.KeywordPlanMaxCpcBidForecastCurve = exports.KeywordPlanCampaignForecastCurve = exports.KeywordPlanKeywordForecast = exports.KeywordPlanAdGroupForecast = exports.KeywordPlanCampaignForecast = exports.GenerateForecastMetricsResponse = exports.GenerateForecastMetricsRequest = exports.GenerateForecastTimeSeriesResponse = exports.GenerateForecastTimeSeriesRequest = exports.GenerateForecastCurveResponse = exports.GenerateForecastCurveRequest = exports.MutateKeywordPlansResult = exports.MutateKeywordPlansResponse = exports.KeywordPlanOperation = exports.MutateKeywordPlansRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/keyword_plan_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const keyword_plan_common_1 = require("../common/keyword_plan_common");
const keyword_plan_common_2 = require("../common/keyword_plan_common");
const keyword_plan_common_3 = require("../common/keyword_plan_common");
const keyword_plan_common_4 = require("../common/keyword_plan_common");
const status_1 = require("../../google/rpc/status");
const keyword_plan_1 = require("../resources/keyword_plan");
const field_mask_1 = require("../../google/protobuf/field_mask");
// @generated message type with reflection information, may provide speed optimized methods
class MutateKeywordPlansRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateKeywordPlansRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "operations", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanOperation, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "partial_failure", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "validate_only", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { customerId: "", operations: [], partialFailure: false, validateOnly: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.services.KeywordPlanOperation operations */ 2:
                    message.operations.push(exports.KeywordPlanOperation.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* bool partial_failure */ 3:
                    message.partialFailure = reader.bool();
                    break;
                case /* bool validate_only */ 4:
                    message.validateOnly = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* repeated google.ads.googleads.v11.services.KeywordPlanOperation operations = 2; */
        for (let i = 0; i < message.operations.length; i++)
            exports.KeywordPlanOperation.internalBinaryWrite(message.operations[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool partial_failure = 3; */
        if (message.partialFailure !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.partialFailure);
        /* bool validate_only = 4; */
        if (message.validateOnly !== false)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.validateOnly);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateKeywordPlansRequest
 */
exports.MutateKeywordPlansRequest = new MutateKeywordPlansRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanOperation", [
            { no: 4, name: "update_mask", kind: "message", T: () => field_mask_1.FieldMask },
            { no: 1, name: "create", kind: "message", oneof: "operation", T: () => keyword_plan_1.KeywordPlan },
            { no: 2, name: "update", kind: "message", oneof: "operation", T: () => keyword_plan_1.KeywordPlan },
            { no: 3, name: "remove", kind: "scalar", oneof: "operation", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlan" } } }
        ]);
    }
    create(value) {
        const message = { operation: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.protobuf.FieldMask update_mask */ 4:
                    message.updateMask = field_mask_1.FieldMask.internalBinaryRead(reader, reader.uint32(), options, message.updateMask);
                    break;
                case /* google.ads.googleads.v11.resources.KeywordPlan create */ 1:
                    message.operation = {
                        oneofKind: "create",
                        create: keyword_plan_1.KeywordPlan.internalBinaryRead(reader, reader.uint32(), options, message.operation.create)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.KeywordPlan update */ 2:
                    message.operation = {
                        oneofKind: "update",
                        update: keyword_plan_1.KeywordPlan.internalBinaryRead(reader, reader.uint32(), options, message.operation.update)
                    };
                    break;
                case /* string remove */ 3:
                    message.operation = {
                        oneofKind: "remove",
                        remove: reader.string()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.protobuf.FieldMask update_mask = 4; */
        if (message.updateMask)
            field_mask_1.FieldMask.internalBinaryWrite(message.updateMask, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.KeywordPlan create = 1; */
        if (message.operation.oneofKind === "create")
            keyword_plan_1.KeywordPlan.internalBinaryWrite(message.operation.create, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.KeywordPlan update = 2; */
        if (message.operation.oneofKind === "update")
            keyword_plan_1.KeywordPlan.internalBinaryWrite(message.operation.update, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string remove = 3; */
        if (message.operation.oneofKind === "remove")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.operation.remove);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanOperation
 */
exports.KeywordPlanOperation = new KeywordPlanOperation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateKeywordPlansResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateKeywordPlansResponse", [
            { no: 3, name: "partial_failure_error", kind: "message", T: () => status_1.Status },
            { no: 2, name: "results", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.MutateKeywordPlansResult }
        ]);
    }
    create(value) {
        const message = { results: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.rpc.Status partial_failure_error */ 3:
                    message.partialFailureError = status_1.Status.internalBinaryRead(reader, reader.uint32(), options, message.partialFailureError);
                    break;
                case /* repeated google.ads.googleads.v11.services.MutateKeywordPlansResult results */ 2:
                    message.results.push(exports.MutateKeywordPlansResult.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.rpc.Status partial_failure_error = 3; */
        if (message.partialFailureError)
            status_1.Status.internalBinaryWrite(message.partialFailureError, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.services.MutateKeywordPlansResult results = 2; */
        for (let i = 0; i < message.results.length; i++)
            exports.MutateKeywordPlansResult.internalBinaryWrite(message.results[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateKeywordPlansResponse
 */
exports.MutateKeywordPlansResponse = new MutateKeywordPlansResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MutateKeywordPlansResult$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.MutateKeywordPlansResult", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlan" } } }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.MutateKeywordPlansResult
 */
exports.MutateKeywordPlansResult = new MutateKeywordPlansResult$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateForecastCurveRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GenerateForecastCurveRequest", [
            { no: 1, name: "keyword_plan", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlan" } } }
        ]);
    }
    create(value) {
        const message = { keywordPlan: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string keyword_plan */ 1:
                    message.keywordPlan = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string keyword_plan = 1; */
        if (message.keywordPlan !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.keywordPlan);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GenerateForecastCurveRequest
 */
exports.GenerateForecastCurveRequest = new GenerateForecastCurveRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateForecastCurveResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GenerateForecastCurveResponse", [
            { no: 1, name: "campaign_forecast_curves", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanCampaignForecastCurve }
        ]);
    }
    create(value) {
        const message = { campaignForecastCurves: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.services.KeywordPlanCampaignForecastCurve campaign_forecast_curves */ 1:
                    message.campaignForecastCurves.push(exports.KeywordPlanCampaignForecastCurve.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.services.KeywordPlanCampaignForecastCurve campaign_forecast_curves = 1; */
        for (let i = 0; i < message.campaignForecastCurves.length; i++)
            exports.KeywordPlanCampaignForecastCurve.internalBinaryWrite(message.campaignForecastCurves[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GenerateForecastCurveResponse
 */
exports.GenerateForecastCurveResponse = new GenerateForecastCurveResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateForecastTimeSeriesRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GenerateForecastTimeSeriesRequest", [
            { no: 1, name: "keyword_plan", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlan" } } }
        ]);
    }
    create(value) {
        const message = { keywordPlan: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string keyword_plan */ 1:
                    message.keywordPlan = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string keyword_plan = 1; */
        if (message.keywordPlan !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.keywordPlan);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GenerateForecastTimeSeriesRequest
 */
exports.GenerateForecastTimeSeriesRequest = new GenerateForecastTimeSeriesRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateForecastTimeSeriesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GenerateForecastTimeSeriesResponse", [
            { no: 1, name: "weekly_time_series_forecasts", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanWeeklyTimeSeriesForecast }
        ]);
    }
    create(value) {
        const message = { weeklyTimeSeriesForecasts: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.services.KeywordPlanWeeklyTimeSeriesForecast weekly_time_series_forecasts */ 1:
                    message.weeklyTimeSeriesForecasts.push(exports.KeywordPlanWeeklyTimeSeriesForecast.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.services.KeywordPlanWeeklyTimeSeriesForecast weekly_time_series_forecasts = 1; */
        for (let i = 0; i < message.weeklyTimeSeriesForecasts.length; i++)
            exports.KeywordPlanWeeklyTimeSeriesForecast.internalBinaryWrite(message.weeklyTimeSeriesForecasts[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GenerateForecastTimeSeriesResponse
 */
exports.GenerateForecastTimeSeriesResponse = new GenerateForecastTimeSeriesResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateForecastMetricsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GenerateForecastMetricsRequest", [
            { no: 1, name: "keyword_plan", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlan" } } }
        ]);
    }
    create(value) {
        const message = { keywordPlan: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string keyword_plan */ 1:
                    message.keywordPlan = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string keyword_plan = 1; */
        if (message.keywordPlan !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.keywordPlan);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GenerateForecastMetricsRequest
 */
exports.GenerateForecastMetricsRequest = new GenerateForecastMetricsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateForecastMetricsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GenerateForecastMetricsResponse", [
            { no: 1, name: "campaign_forecasts", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanCampaignForecast },
            { no: 2, name: "ad_group_forecasts", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanAdGroupForecast },
            { no: 3, name: "keyword_forecasts", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanKeywordForecast }
        ]);
    }
    create(value) {
        const message = { campaignForecasts: [], adGroupForecasts: [], keywordForecasts: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.services.KeywordPlanCampaignForecast campaign_forecasts */ 1:
                    message.campaignForecasts.push(exports.KeywordPlanCampaignForecast.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.services.KeywordPlanAdGroupForecast ad_group_forecasts */ 2:
                    message.adGroupForecasts.push(exports.KeywordPlanAdGroupForecast.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.services.KeywordPlanKeywordForecast keyword_forecasts */ 3:
                    message.keywordForecasts.push(exports.KeywordPlanKeywordForecast.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.services.KeywordPlanCampaignForecast campaign_forecasts = 1; */
        for (let i = 0; i < message.campaignForecasts.length; i++)
            exports.KeywordPlanCampaignForecast.internalBinaryWrite(message.campaignForecasts[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.services.KeywordPlanAdGroupForecast ad_group_forecasts = 2; */
        for (let i = 0; i < message.adGroupForecasts.length; i++)
            exports.KeywordPlanAdGroupForecast.internalBinaryWrite(message.adGroupForecasts[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.services.KeywordPlanKeywordForecast keyword_forecasts = 3; */
        for (let i = 0; i < message.keywordForecasts.length; i++)
            exports.KeywordPlanKeywordForecast.internalBinaryWrite(message.keywordForecasts[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GenerateForecastMetricsResponse
 */
exports.GenerateForecastMetricsResponse = new GenerateForecastMetricsResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanCampaignForecast$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanCampaignForecast", [
            { no: 3, name: "keyword_plan_campaign", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "campaign_forecast", kind: "message", T: () => exports.ForecastMetrics }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string keyword_plan_campaign */ 3:
                    message.keywordPlanCampaign = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.ForecastMetrics campaign_forecast */ 2:
                    message.campaignForecast = exports.ForecastMetrics.internalBinaryRead(reader, reader.uint32(), options, message.campaignForecast);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string keyword_plan_campaign = 3; */
        if (message.keywordPlanCampaign !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.keywordPlanCampaign);
        /* google.ads.googleads.v11.services.ForecastMetrics campaign_forecast = 2; */
        if (message.campaignForecast)
            exports.ForecastMetrics.internalBinaryWrite(message.campaignForecast, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanCampaignForecast
 */
exports.KeywordPlanCampaignForecast = new KeywordPlanCampaignForecast$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanAdGroupForecast$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanAdGroupForecast", [
            { no: 3, name: "keyword_plan_ad_group", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "ad_group_forecast", kind: "message", T: () => exports.ForecastMetrics }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string keyword_plan_ad_group */ 3:
                    message.keywordPlanAdGroup = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.ForecastMetrics ad_group_forecast */ 2:
                    message.adGroupForecast = exports.ForecastMetrics.internalBinaryRead(reader, reader.uint32(), options, message.adGroupForecast);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string keyword_plan_ad_group = 3; */
        if (message.keywordPlanAdGroup !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.keywordPlanAdGroup);
        /* google.ads.googleads.v11.services.ForecastMetrics ad_group_forecast = 2; */
        if (message.adGroupForecast)
            exports.ForecastMetrics.internalBinaryWrite(message.adGroupForecast, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanAdGroupForecast
 */
exports.KeywordPlanAdGroupForecast = new KeywordPlanAdGroupForecast$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanKeywordForecast$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanKeywordForecast", [
            { no: 3, name: "keyword_plan_ad_group_keyword", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "keyword_forecast", kind: "message", T: () => exports.ForecastMetrics }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string keyword_plan_ad_group_keyword */ 3:
                    message.keywordPlanAdGroupKeyword = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.ForecastMetrics keyword_forecast */ 2:
                    message.keywordForecast = exports.ForecastMetrics.internalBinaryRead(reader, reader.uint32(), options, message.keywordForecast);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string keyword_plan_ad_group_keyword = 3; */
        if (message.keywordPlanAdGroupKeyword !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.keywordPlanAdGroupKeyword);
        /* google.ads.googleads.v11.services.ForecastMetrics keyword_forecast = 2; */
        if (message.keywordForecast)
            exports.ForecastMetrics.internalBinaryWrite(message.keywordForecast, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanKeywordForecast
 */
exports.KeywordPlanKeywordForecast = new KeywordPlanKeywordForecast$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanCampaignForecastCurve$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanCampaignForecastCurve", [
            { no: 3, name: "keyword_plan_campaign", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "max_cpc_bid_forecast_curve", kind: "message", T: () => exports.KeywordPlanMaxCpcBidForecastCurve }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string keyword_plan_campaign */ 3:
                    message.keywordPlanCampaign = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.KeywordPlanMaxCpcBidForecastCurve max_cpc_bid_forecast_curve */ 2:
                    message.maxCpcBidForecastCurve = exports.KeywordPlanMaxCpcBidForecastCurve.internalBinaryRead(reader, reader.uint32(), options, message.maxCpcBidForecastCurve);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string keyword_plan_campaign = 3; */
        if (message.keywordPlanCampaign !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.keywordPlanCampaign);
        /* google.ads.googleads.v11.services.KeywordPlanMaxCpcBidForecastCurve max_cpc_bid_forecast_curve = 2; */
        if (message.maxCpcBidForecastCurve)
            exports.KeywordPlanMaxCpcBidForecastCurve.internalBinaryWrite(message.maxCpcBidForecastCurve, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanCampaignForecastCurve
 */
exports.KeywordPlanCampaignForecastCurve = new KeywordPlanCampaignForecastCurve$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanMaxCpcBidForecastCurve$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanMaxCpcBidForecastCurve", [
            { no: 1, name: "max_cpc_bid_forecasts", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanMaxCpcBidForecast }
        ]);
    }
    create(value) {
        const message = { maxCpcBidForecasts: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.services.KeywordPlanMaxCpcBidForecast max_cpc_bid_forecasts */ 1:
                    message.maxCpcBidForecasts.push(exports.KeywordPlanMaxCpcBidForecast.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.services.KeywordPlanMaxCpcBidForecast max_cpc_bid_forecasts = 1; */
        for (let i = 0; i < message.maxCpcBidForecasts.length; i++)
            exports.KeywordPlanMaxCpcBidForecast.internalBinaryWrite(message.maxCpcBidForecasts[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanMaxCpcBidForecastCurve
 */
exports.KeywordPlanMaxCpcBidForecastCurve = new KeywordPlanMaxCpcBidForecastCurve$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanMaxCpcBidForecast$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanMaxCpcBidForecast", [
            { no: 3, name: "max_cpc_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 2, name: "max_cpc_bid_forecast", kind: "message", T: () => exports.ForecastMetrics }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 max_cpc_bid_micros */ 3:
                    message.maxCpcBidMicros = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.services.ForecastMetrics max_cpc_bid_forecast */ 2:
                    message.maxCpcBidForecast = exports.ForecastMetrics.internalBinaryRead(reader, reader.uint32(), options, message.maxCpcBidForecast);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 max_cpc_bid_micros = 3; */
        if (message.maxCpcBidMicros !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.maxCpcBidMicros);
        /* google.ads.googleads.v11.services.ForecastMetrics max_cpc_bid_forecast = 2; */
        if (message.maxCpcBidForecast)
            exports.ForecastMetrics.internalBinaryWrite(message.maxCpcBidForecast, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanMaxCpcBidForecast
 */
exports.KeywordPlanMaxCpcBidForecast = new KeywordPlanMaxCpcBidForecast$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanWeeklyTimeSeriesForecast$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanWeeklyTimeSeriesForecast", [
            { no: 1, name: "keyword_plan_campaign", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "weekly_forecasts", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanWeeklyForecast }
        ]);
    }
    create(value) {
        const message = { weeklyForecasts: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string keyword_plan_campaign */ 1:
                    message.keywordPlanCampaign = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.services.KeywordPlanWeeklyForecast weekly_forecasts */ 2:
                    message.weeklyForecasts.push(exports.KeywordPlanWeeklyForecast.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string keyword_plan_campaign = 1; */
        if (message.keywordPlanCampaign !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.keywordPlanCampaign);
        /* repeated google.ads.googleads.v11.services.KeywordPlanWeeklyForecast weekly_forecasts = 2; */
        for (let i = 0; i < message.weeklyForecasts.length; i++)
            exports.KeywordPlanWeeklyForecast.internalBinaryWrite(message.weeklyForecasts[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanWeeklyTimeSeriesForecast
 */
exports.KeywordPlanWeeklyTimeSeriesForecast = new KeywordPlanWeeklyTimeSeriesForecast$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanWeeklyForecast$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanWeeklyForecast", [
            { no: 1, name: "start_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "forecast", kind: "message", T: () => exports.ForecastMetrics }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string start_date */ 1:
                    message.startDate = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.ForecastMetrics forecast */ 2:
                    message.forecast = exports.ForecastMetrics.internalBinaryRead(reader, reader.uint32(), options, message.forecast);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string start_date = 1; */
        if (message.startDate !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* google.ads.googleads.v11.services.ForecastMetrics forecast = 2; */
        if (message.forecast)
            exports.ForecastMetrics.internalBinaryWrite(message.forecast, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanWeeklyForecast
 */
exports.KeywordPlanWeeklyForecast = new KeywordPlanWeeklyForecast$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ForecastMetrics$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ForecastMetrics", [
            { no: 7, name: "impressions", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 8, name: "ctr", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 9, name: "average_cpc", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 10, name: "clicks", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 11, name: "cost_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional double impressions */ 7:
                    message.impressions = reader.double();
                    break;
                case /* optional double ctr */ 8:
                    message.ctr = reader.double();
                    break;
                case /* optional int64 average_cpc */ 9:
                    message.averageCpc = reader.int64().toBigInt();
                    break;
                case /* optional double clicks */ 10:
                    message.clicks = reader.double();
                    break;
                case /* optional int64 cost_micros */ 11:
                    message.costMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional double impressions = 7; */
        if (message.impressions !== undefined)
            writer.tag(7, runtime_1.WireType.Bit64).double(message.impressions);
        /* optional double ctr = 8; */
        if (message.ctr !== undefined)
            writer.tag(8, runtime_1.WireType.Bit64).double(message.ctr);
        /* optional int64 average_cpc = 9; */
        if (message.averageCpc !== undefined)
            writer.tag(9, runtime_1.WireType.Varint).int64(message.averageCpc);
        /* optional double clicks = 10; */
        if (message.clicks !== undefined)
            writer.tag(10, runtime_1.WireType.Bit64).double(message.clicks);
        /* optional int64 cost_micros = 11; */
        if (message.costMicros !== undefined)
            writer.tag(11, runtime_1.WireType.Varint).int64(message.costMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ForecastMetrics
 */
exports.ForecastMetrics = new ForecastMetrics$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateHistoricalMetricsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GenerateHistoricalMetricsRequest", [
            { no: 1, name: "keyword_plan", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlan" } } },
            { no: 2, name: "aggregate_metrics", kind: "message", T: () => keyword_plan_common_4.KeywordPlanAggregateMetrics },
            { no: 3, name: "historical_metrics_options", kind: "message", T: () => keyword_plan_common_3.HistoricalMetricsOptions }
        ]);
    }
    create(value) {
        const message = { keywordPlan: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string keyword_plan */ 1:
                    message.keywordPlan = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.KeywordPlanAggregateMetrics aggregate_metrics */ 2:
                    message.aggregateMetrics = keyword_plan_common_4.KeywordPlanAggregateMetrics.internalBinaryRead(reader, reader.uint32(), options, message.aggregateMetrics);
                    break;
                case /* google.ads.googleads.v11.common.HistoricalMetricsOptions historical_metrics_options */ 3:
                    message.historicalMetricsOptions = keyword_plan_common_3.HistoricalMetricsOptions.internalBinaryRead(reader, reader.uint32(), options, message.historicalMetricsOptions);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string keyword_plan = 1; */
        if (message.keywordPlan !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.keywordPlan);
        /* google.ads.googleads.v11.common.KeywordPlanAggregateMetrics aggregate_metrics = 2; */
        if (message.aggregateMetrics)
            keyword_plan_common_4.KeywordPlanAggregateMetrics.internalBinaryWrite(message.aggregateMetrics, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HistoricalMetricsOptions historical_metrics_options = 3; */
        if (message.historicalMetricsOptions)
            keyword_plan_common_3.HistoricalMetricsOptions.internalBinaryWrite(message.historicalMetricsOptions, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GenerateHistoricalMetricsRequest
 */
exports.GenerateHistoricalMetricsRequest = new GenerateHistoricalMetricsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenerateHistoricalMetricsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.GenerateHistoricalMetricsResponse", [
            { no: 1, name: "metrics", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanKeywordHistoricalMetrics },
            { no: 2, name: "aggregate_metric_results", kind: "message", T: () => keyword_plan_common_2.KeywordPlanAggregateMetricResults }
        ]);
    }
    create(value) {
        const message = { metrics: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.services.KeywordPlanKeywordHistoricalMetrics metrics */ 1:
                    message.metrics.push(exports.KeywordPlanKeywordHistoricalMetrics.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.common.KeywordPlanAggregateMetricResults aggregate_metric_results */ 2:
                    message.aggregateMetricResults = keyword_plan_common_2.KeywordPlanAggregateMetricResults.internalBinaryRead(reader, reader.uint32(), options, message.aggregateMetricResults);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.services.KeywordPlanKeywordHistoricalMetrics metrics = 1; */
        for (let i = 0; i < message.metrics.length; i++)
            exports.KeywordPlanKeywordHistoricalMetrics.internalBinaryWrite(message.metrics[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.KeywordPlanAggregateMetricResults aggregate_metric_results = 2; */
        if (message.aggregateMetricResults)
            keyword_plan_common_2.KeywordPlanAggregateMetricResults.internalBinaryWrite(message.aggregateMetricResults, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.GenerateHistoricalMetricsResponse
 */
exports.GenerateHistoricalMetricsResponse = new GenerateHistoricalMetricsResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanKeywordHistoricalMetrics$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.KeywordPlanKeywordHistoricalMetrics", [
            { no: 4, name: "search_query", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "close_variants", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "keyword_metrics", kind: "message", T: () => keyword_plan_common_1.KeywordPlanHistoricalMetrics }
        ]);
    }
    create(value) {
        const message = { closeVariants: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string search_query */ 4:
                    message.searchQuery = reader.string();
                    break;
                case /* repeated string close_variants */ 3:
                    message.closeVariants.push(reader.string());
                    break;
                case /* google.ads.googleads.v11.common.KeywordPlanHistoricalMetrics keyword_metrics */ 2:
                    message.keywordMetrics = keyword_plan_common_1.KeywordPlanHistoricalMetrics.internalBinaryRead(reader, reader.uint32(), options, message.keywordMetrics);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string search_query = 4; */
        if (message.searchQuery !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.searchQuery);
        /* repeated string close_variants = 3; */
        for (let i = 0; i < message.closeVariants.length; i++)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.closeVariants[i]);
        /* google.ads.googleads.v11.common.KeywordPlanHistoricalMetrics keyword_metrics = 2; */
        if (message.keywordMetrics)
            keyword_plan_common_1.KeywordPlanHistoricalMetrics.internalBinaryWrite(message.keywordMetrics, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.KeywordPlanKeywordHistoricalMetrics
 */
exports.KeywordPlanKeywordHistoricalMetrics = new KeywordPlanKeywordHistoricalMetrics$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.KeywordPlanService
 */
exports.KeywordPlanService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.KeywordPlanService", [
    { name: "MutateKeywordPlans", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}/keywordPlans:mutate", body: "*" }, "google.api.method_signature": ["customer_id,operations"] }, I: exports.MutateKeywordPlansRequest, O: exports.MutateKeywordPlansResponse },
    { name: "GenerateForecastCurve", options: { "google.api.http": { post: "/v11/{keyword_plan=customers/*/keywordPlans/*}:generateForecastCurve", body: "*" }, "google.api.method_signature": ["keyword_plan"] }, I: exports.GenerateForecastCurveRequest, O: exports.GenerateForecastCurveResponse },
    { name: "GenerateForecastTimeSeries", options: { "google.api.http": { post: "/v11/{keyword_plan=customers/*/keywordPlans/*}:generateForecastTimeSeries", body: "*" }, "google.api.method_signature": ["keyword_plan"] }, I: exports.GenerateForecastTimeSeriesRequest, O: exports.GenerateForecastTimeSeriesResponse },
    { name: "GenerateForecastMetrics", options: { "google.api.http": { post: "/v11/{keyword_plan=customers/*/keywordPlans/*}:generateForecastMetrics", body: "*" }, "google.api.method_signature": ["keyword_plan"] }, I: exports.GenerateForecastMetricsRequest, O: exports.GenerateForecastMetricsResponse },
    { name: "GenerateHistoricalMetrics", options: { "google.api.http": { post: "/v11/{keyword_plan=customers/*/keywordPlans/*}:generateHistoricalMetrics", body: "*" }, "google.api.method_signature": ["keyword_plan"] }, I: exports.GenerateHistoricalMetricsRequest, O: exports.GenerateHistoricalMetricsResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
