// google-ads-api-client/dist/protos/bindings/ads/services/keyword_theme_constant_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordThemeConstantService = exports.SuggestKeywordThemeConstantsResponse = exports.SuggestKeywordThemeConstantsRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/keyword_theme_constant_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const keyword_theme_constant_1 = require("../resources/keyword_theme_constant");
// @generated message type with reflection information, may provide speed optimized methods
class SuggestKeywordThemeConstantsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestKeywordThemeConstantsRequest", [
            { no: 1, name: "query_text", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "country_code", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "language_code", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { queryText: "", countryCode: "", languageCode: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string query_text */ 1:
                    message.queryText = reader.string();
                    break;
                case /* string country_code */ 2:
                    message.countryCode = reader.string();
                    break;
                case /* string language_code */ 3:
                    message.languageCode = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string query_text = 1; */
        if (message.queryText !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.queryText);
        /* string country_code = 2; */
        if (message.countryCode !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* string language_code = 3; */
        if (message.languageCode !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestKeywordThemeConstantsRequest
 */
exports.SuggestKeywordThemeConstantsRequest = new SuggestKeywordThemeConstantsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SuggestKeywordThemeConstantsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestKeywordThemeConstantsResponse", [
            { no: 1, name: "keyword_theme_constants", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => keyword_theme_constant_1.KeywordThemeConstant }
        ]);
    }
    create(value) {
        const message = { keywordThemeConstants: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.resources.KeywordThemeConstant keyword_theme_constants */ 1:
                    message.keywordThemeConstants.push(keyword_theme_constant_1.KeywordThemeConstant.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.resources.KeywordThemeConstant keyword_theme_constants = 1; */
        for (let i = 0; i < message.keywordThemeConstants.length; i++)
            keyword_theme_constant_1.KeywordThemeConstant.internalBinaryWrite(message.keywordThemeConstants[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestKeywordThemeConstantsResponse
 */
exports.SuggestKeywordThemeConstantsResponse = new SuggestKeywordThemeConstantsResponse$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.KeywordThemeConstantService
 */
exports.KeywordThemeConstantService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.KeywordThemeConstantService", [
    { name: "SuggestKeywordThemeConstants", options: { "google.api.http": { post: "/v11/keywordThemeConstants:suggest", body: "*" } }, I: exports.SuggestKeywordThemeConstantsRequest, O: exports.SuggestKeywordThemeConstantsResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
