// google-ads-api-client/dist/protos/bindings/ads/services/recommendation_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecommendationService = exports.DismissRecommendationResponse_DismissRecommendationResult = exports.DismissRecommendationResponse = exports.DismissRecommendationRequest_DismissRecommendationOperation = exports.DismissRecommendationRequest = exports.ApplyRecommendationResult = exports.ApplyRecommendationResponse = exports.ApplyRecommendationOperation_UseBroadMatchKeywordParameters = exports.ApplyRecommendationOperation_ResponsiveSearchAdParameters = exports.ApplyRecommendationOperation_ResponsiveSearchAdImproveAdStrengthParameters = exports.ApplyRecommendationOperation_ResponsiveSearchAdAssetParameters = exports.ApplyRecommendationOperation_MoveUnusedBudgetParameters = exports.ApplyRecommendationOperation_SitelinkExtensionParameters = exports.ApplyRecommendationOperation_CallExtensionParameters = exports.ApplyRecommendationOperation_CalloutExtensionParameters = exports.ApplyRecommendationOperation_TargetRoasOptInParameters = exports.ApplyRecommendationOperation_TargetCpaOptInParameters = exports.ApplyRecommendationOperation_KeywordParameters = exports.ApplyRecommendationOperation_TextAdParameters = exports.ApplyRecommendationOperation_CampaignBudgetParameters = exports.ApplyRecommendationOperation = exports.ApplyRecommendationRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/recommendation_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const status_1 = require("../../google/rpc/status");
const extensions_1 = require("../common/extensions");
const extensions_2 = require("../common/extensions");
const extensions_3 = require("../common/extensions");
const keyword_match_type_1 = require("../enums/keyword_match_type");
const ad_1 = require("../resources/ad");
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "operations", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.ApplyRecommendationOperation, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "partial_failure", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { customerId: "", operations: [], partialFailure: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.services.ApplyRecommendationOperation operations */ 2:
                    message.operations.push(exports.ApplyRecommendationOperation.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* bool partial_failure */ 3:
                    message.partialFailure = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* repeated google.ads.googleads.v11.services.ApplyRecommendationOperation operations = 2; */
        for (let i = 0; i < message.operations.length; i++)
            exports.ApplyRecommendationOperation.internalBinaryWrite(message.operations[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool partial_failure = 3; */
        if (message.partialFailure !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.partialFailure);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationRequest
 */
exports.ApplyRecommendationRequest = new ApplyRecommendationRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/Recommendation" } } },
            { no: 2, name: "campaign_budget", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_CampaignBudgetParameters },
            { no: 3, name: "text_ad", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_TextAdParameters },
            { no: 4, name: "keyword", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_KeywordParameters },
            { no: 5, name: "target_cpa_opt_in", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_TargetCpaOptInParameters },
            { no: 10, name: "target_roas_opt_in", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_TargetRoasOptInParameters },
            { no: 6, name: "callout_extension", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_CalloutExtensionParameters },
            { no: 7, name: "call_extension", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_CallExtensionParameters },
            { no: 8, name: "sitelink_extension", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_SitelinkExtensionParameters },
            { no: 9, name: "move_unused_budget", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_MoveUnusedBudgetParameters },
            { no: 11, name: "responsive_search_ad", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_ResponsiveSearchAdParameters },
            { no: 12, name: "use_broad_match_keyword", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_UseBroadMatchKeywordParameters },
            { no: 13, name: "responsive_search_ad_asset", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_ResponsiveSearchAdAssetParameters },
            { no: 14, name: "responsive_search_ad_improve_ad_strength", kind: "message", oneof: "applyParameters", T: () => exports.ApplyRecommendationOperation_ResponsiveSearchAdImproveAdStrengthParameters }
        ]);
    }
    create(value) {
        const message = { resourceName: "", applyParameters: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.CampaignBudgetParameters campaign_budget */ 2:
                    message.applyParameters = {
                        oneofKind: "campaignBudget",
                        campaignBudget: exports.ApplyRecommendationOperation_CampaignBudgetParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.campaignBudget)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.TextAdParameters text_ad */ 3:
                    message.applyParameters = {
                        oneofKind: "textAd",
                        textAd: exports.ApplyRecommendationOperation_TextAdParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.textAd)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.KeywordParameters keyword */ 4:
                    message.applyParameters = {
                        oneofKind: "keyword",
                        keyword: exports.ApplyRecommendationOperation_KeywordParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.keyword)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.TargetCpaOptInParameters target_cpa_opt_in */ 5:
                    message.applyParameters = {
                        oneofKind: "targetCpaOptIn",
                        targetCpaOptIn: exports.ApplyRecommendationOperation_TargetCpaOptInParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.targetCpaOptIn)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.TargetRoasOptInParameters target_roas_opt_in */ 10:
                    message.applyParameters = {
                        oneofKind: "targetRoasOptIn",
                        targetRoasOptIn: exports.ApplyRecommendationOperation_TargetRoasOptInParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.targetRoasOptIn)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.CalloutExtensionParameters callout_extension */ 6:
                    message.applyParameters = {
                        oneofKind: "calloutExtension",
                        calloutExtension: exports.ApplyRecommendationOperation_CalloutExtensionParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.calloutExtension)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.CallExtensionParameters call_extension */ 7:
                    message.applyParameters = {
                        oneofKind: "callExtension",
                        callExtension: exports.ApplyRecommendationOperation_CallExtensionParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.callExtension)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.SitelinkExtensionParameters sitelink_extension */ 8:
                    message.applyParameters = {
                        oneofKind: "sitelinkExtension",
                        sitelinkExtension: exports.ApplyRecommendationOperation_SitelinkExtensionParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.sitelinkExtension)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.MoveUnusedBudgetParameters move_unused_budget */ 9:
                    message.applyParameters = {
                        oneofKind: "moveUnusedBudget",
                        moveUnusedBudget: exports.ApplyRecommendationOperation_MoveUnusedBudgetParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.moveUnusedBudget)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdParameters responsive_search_ad */ 11:
                    message.applyParameters = {
                        oneofKind: "responsiveSearchAd",
                        responsiveSearchAd: exports.ApplyRecommendationOperation_ResponsiveSearchAdParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.responsiveSearchAd)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.UseBroadMatchKeywordParameters use_broad_match_keyword */ 12:
                    message.applyParameters = {
                        oneofKind: "useBroadMatchKeyword",
                        useBroadMatchKeyword: exports.ApplyRecommendationOperation_UseBroadMatchKeywordParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.useBroadMatchKeyword)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdAssetParameters responsive_search_ad_asset */ 13:
                    message.applyParameters = {
                        oneofKind: "responsiveSearchAdAsset",
                        responsiveSearchAdAsset: exports.ApplyRecommendationOperation_ResponsiveSearchAdAssetParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.responsiveSearchAdAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdImproveAdStrengthParameters responsive_search_ad_improve_ad_strength */ 14:
                    message.applyParameters = {
                        oneofKind: "responsiveSearchAdImproveAdStrength",
                        responsiveSearchAdImproveAdStrength: exports.ApplyRecommendationOperation_ResponsiveSearchAdImproveAdStrengthParameters.internalBinaryRead(reader, reader.uint32(), options, message.applyParameters.responsiveSearchAdImproveAdStrength)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.CampaignBudgetParameters campaign_budget = 2; */
        if (message.applyParameters.oneofKind === "campaignBudget")
            exports.ApplyRecommendationOperation_CampaignBudgetParameters.internalBinaryWrite(message.applyParameters.campaignBudget, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.TextAdParameters text_ad = 3; */
        if (message.applyParameters.oneofKind === "textAd")
            exports.ApplyRecommendationOperation_TextAdParameters.internalBinaryWrite(message.applyParameters.textAd, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.KeywordParameters keyword = 4; */
        if (message.applyParameters.oneofKind === "keyword")
            exports.ApplyRecommendationOperation_KeywordParameters.internalBinaryWrite(message.applyParameters.keyword, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.TargetCpaOptInParameters target_cpa_opt_in = 5; */
        if (message.applyParameters.oneofKind === "targetCpaOptIn")
            exports.ApplyRecommendationOperation_TargetCpaOptInParameters.internalBinaryWrite(message.applyParameters.targetCpaOptIn, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.TargetRoasOptInParameters target_roas_opt_in = 10; */
        if (message.applyParameters.oneofKind === "targetRoasOptIn")
            exports.ApplyRecommendationOperation_TargetRoasOptInParameters.internalBinaryWrite(message.applyParameters.targetRoasOptIn, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.CalloutExtensionParameters callout_extension = 6; */
        if (message.applyParameters.oneofKind === "calloutExtension")
            exports.ApplyRecommendationOperation_CalloutExtensionParameters.internalBinaryWrite(message.applyParameters.calloutExtension, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.CallExtensionParameters call_extension = 7; */
        if (message.applyParameters.oneofKind === "callExtension")
            exports.ApplyRecommendationOperation_CallExtensionParameters.internalBinaryWrite(message.applyParameters.callExtension, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.SitelinkExtensionParameters sitelink_extension = 8; */
        if (message.applyParameters.oneofKind === "sitelinkExtension")
            exports.ApplyRecommendationOperation_SitelinkExtensionParameters.internalBinaryWrite(message.applyParameters.sitelinkExtension, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.MoveUnusedBudgetParameters move_unused_budget = 9; */
        if (message.applyParameters.oneofKind === "moveUnusedBudget")
            exports.ApplyRecommendationOperation_MoveUnusedBudgetParameters.internalBinaryWrite(message.applyParameters.moveUnusedBudget, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdParameters responsive_search_ad = 11; */
        if (message.applyParameters.oneofKind === "responsiveSearchAd")
            exports.ApplyRecommendationOperation_ResponsiveSearchAdParameters.internalBinaryWrite(message.applyParameters.responsiveSearchAd, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.UseBroadMatchKeywordParameters use_broad_match_keyword = 12; */
        if (message.applyParameters.oneofKind === "useBroadMatchKeyword")
            exports.ApplyRecommendationOperation_UseBroadMatchKeywordParameters.internalBinaryWrite(message.applyParameters.useBroadMatchKeyword, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdAssetParameters responsive_search_ad_asset = 13; */
        if (message.applyParameters.oneofKind === "responsiveSearchAdAsset")
            exports.ApplyRecommendationOperation_ResponsiveSearchAdAssetParameters.internalBinaryWrite(message.applyParameters.responsiveSearchAdAsset, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdImproveAdStrengthParameters responsive_search_ad_improve_ad_strength = 14; */
        if (message.applyParameters.oneofKind === "responsiveSearchAdImproveAdStrength")
            exports.ApplyRecommendationOperation_ResponsiveSearchAdImproveAdStrengthParameters.internalBinaryWrite(message.applyParameters.responsiveSearchAdImproveAdStrength, writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation
 */
exports.ApplyRecommendationOperation = new ApplyRecommendationOperation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_CampaignBudgetParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.CampaignBudgetParameters", [
            { no: 2, name: "new_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 new_budget_amount_micros */ 2:
                    message.newBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 new_budget_amount_micros = 2; */
        if (message.newBudgetAmountMicros !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.newBudgetAmountMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.CampaignBudgetParameters
 */
exports.ApplyRecommendationOperation_CampaignBudgetParameters = new ApplyRecommendationOperation_CampaignBudgetParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_TextAdParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.TextAdParameters", [
            { no: 1, name: "ad", kind: "message", T: () => ad_1.Ad }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad ad */ 1:
                    message.ad = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.ad);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad ad = 1; */
        if (message.ad)
            ad_1.Ad.internalBinaryWrite(message.ad, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.TextAdParameters
 */
exports.ApplyRecommendationOperation_TextAdParameters = new ApplyRecommendationOperation_TextAdParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_KeywordParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.KeywordParameters", [
            { no: 4, name: "ad_group", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "match_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.KeywordMatchTypeEnum.KeywordMatchType", keyword_match_type_1.KeywordMatchTypeEnum_KeywordMatchType] },
            { no: 5, name: "cpc_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { matchType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string ad_group */ 4:
                    message.adGroup = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.KeywordMatchTypeEnum.KeywordMatchType match_type */ 2:
                    message.matchType = reader.int32();
                    break;
                case /* optional int64 cpc_bid_micros */ 5:
                    message.cpcBidMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string ad_group = 4; */
        if (message.adGroup !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.adGroup);
        /* google.ads.googleads.v11.enums.KeywordMatchTypeEnum.KeywordMatchType match_type = 2; */
        if (message.matchType !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.matchType);
        /* optional int64 cpc_bid_micros = 5; */
        if (message.cpcBidMicros !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.cpcBidMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.KeywordParameters
 */
exports.ApplyRecommendationOperation_KeywordParameters = new ApplyRecommendationOperation_KeywordParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_TargetCpaOptInParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.TargetCpaOptInParameters", [
            { no: 3, name: "target_cpa_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 4, name: "new_campaign_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 target_cpa_micros */ 3:
                    message.targetCpaMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 new_campaign_budget_amount_micros */ 4:
                    message.newCampaignBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 target_cpa_micros = 3; */
        if (message.targetCpaMicros !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.targetCpaMicros);
        /* optional int64 new_campaign_budget_amount_micros = 4; */
        if (message.newCampaignBudgetAmountMicros !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).int64(message.newCampaignBudgetAmountMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.TargetCpaOptInParameters
 */
exports.ApplyRecommendationOperation_TargetCpaOptInParameters = new ApplyRecommendationOperation_TargetCpaOptInParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_TargetRoasOptInParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.TargetRoasOptInParameters", [
            { no: 1, name: "target_roas", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 2, name: "new_campaign_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional double target_roas */ 1:
                    message.targetRoas = reader.double();
                    break;
                case /* optional int64 new_campaign_budget_amount_micros */ 2:
                    message.newCampaignBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional double target_roas = 1; */
        if (message.targetRoas !== undefined)
            writer.tag(1, runtime_1.WireType.Bit64).double(message.targetRoas);
        /* optional int64 new_campaign_budget_amount_micros = 2; */
        if (message.newCampaignBudgetAmountMicros !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.newCampaignBudgetAmountMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.TargetRoasOptInParameters
 */
exports.ApplyRecommendationOperation_TargetRoasOptInParameters = new ApplyRecommendationOperation_TargetRoasOptInParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_CalloutExtensionParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.CalloutExtensionParameters", [
            { no: 1, name: "callout_extensions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => extensions_3.CalloutFeedItem }
        ]);
    }
    create(value) {
        const message = { calloutExtensions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.CalloutFeedItem callout_extensions */ 1:
                    message.calloutExtensions.push(extensions_3.CalloutFeedItem.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.CalloutFeedItem callout_extensions = 1; */
        for (let i = 0; i < message.calloutExtensions.length; i++)
            extensions_3.CalloutFeedItem.internalBinaryWrite(message.calloutExtensions[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.CalloutExtensionParameters
 */
exports.ApplyRecommendationOperation_CalloutExtensionParameters = new ApplyRecommendationOperation_CalloutExtensionParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_CallExtensionParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.CallExtensionParameters", [
            { no: 1, name: "call_extensions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => extensions_2.CallFeedItem }
        ]);
    }
    create(value) {
        const message = { callExtensions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.CallFeedItem call_extensions */ 1:
                    message.callExtensions.push(extensions_2.CallFeedItem.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.CallFeedItem call_extensions = 1; */
        for (let i = 0; i < message.callExtensions.length; i++)
            extensions_2.CallFeedItem.internalBinaryWrite(message.callExtensions[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.CallExtensionParameters
 */
exports.ApplyRecommendationOperation_CallExtensionParameters = new ApplyRecommendationOperation_CallExtensionParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_SitelinkExtensionParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.SitelinkExtensionParameters", [
            { no: 1, name: "sitelink_extensions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => extensions_1.SitelinkFeedItem }
        ]);
    }
    create(value) {
        const message = { sitelinkExtensions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.SitelinkFeedItem sitelink_extensions */ 1:
                    message.sitelinkExtensions.push(extensions_1.SitelinkFeedItem.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.SitelinkFeedItem sitelink_extensions = 1; */
        for (let i = 0; i < message.sitelinkExtensions.length; i++)
            extensions_1.SitelinkFeedItem.internalBinaryWrite(message.sitelinkExtensions[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.SitelinkExtensionParameters
 */
exports.ApplyRecommendationOperation_SitelinkExtensionParameters = new ApplyRecommendationOperation_SitelinkExtensionParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_MoveUnusedBudgetParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.MoveUnusedBudgetParameters", [
            { no: 2, name: "budget_micros_to_move", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 budget_micros_to_move */ 2:
                    message.budgetMicrosToMove = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 budget_micros_to_move = 2; */
        if (message.budgetMicrosToMove !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.budgetMicrosToMove);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.MoveUnusedBudgetParameters
 */
exports.ApplyRecommendationOperation_MoveUnusedBudgetParameters = new ApplyRecommendationOperation_MoveUnusedBudgetParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_ResponsiveSearchAdAssetParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdAssetParameters", [
            { no: 1, name: "updated_ad", kind: "message", T: () => ad_1.Ad }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad updated_ad */ 1:
                    message.updatedAd = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.updatedAd);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad updated_ad = 1; */
        if (message.updatedAd)
            ad_1.Ad.internalBinaryWrite(message.updatedAd, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdAssetParameters
 */
exports.ApplyRecommendationOperation_ResponsiveSearchAdAssetParameters = new ApplyRecommendationOperation_ResponsiveSearchAdAssetParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_ResponsiveSearchAdImproveAdStrengthParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdImproveAdStrengthParameters", [
            { no: 1, name: "updated_ad", kind: "message", T: () => ad_1.Ad }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad updated_ad */ 1:
                    message.updatedAd = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.updatedAd);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad updated_ad = 1; */
        if (message.updatedAd)
            ad_1.Ad.internalBinaryWrite(message.updatedAd, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdImproveAdStrengthParameters
 */
exports.ApplyRecommendationOperation_ResponsiveSearchAdImproveAdStrengthParameters = new ApplyRecommendationOperation_ResponsiveSearchAdImproveAdStrengthParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_ResponsiveSearchAdParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdParameters", [
            { no: 1, name: "ad", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad ad */ 1:
                    message.ad = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.ad);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad ad = 1; */
        if (message.ad)
            ad_1.Ad.internalBinaryWrite(message.ad, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.ResponsiveSearchAdParameters
 */
exports.ApplyRecommendationOperation_ResponsiveSearchAdParameters = new ApplyRecommendationOperation_ResponsiveSearchAdParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationOperation_UseBroadMatchKeywordParameters$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationOperation.UseBroadMatchKeywordParameters", [
            { no: 1, name: "new_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 new_budget_amount_micros */ 1:
                    message.newBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 new_budget_amount_micros = 1; */
        if (message.newBudgetAmountMicros !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.newBudgetAmountMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationOperation.UseBroadMatchKeywordParameters
 */
exports.ApplyRecommendationOperation_UseBroadMatchKeywordParameters = new ApplyRecommendationOperation_UseBroadMatchKeywordParameters$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationResponse", [
            { no: 1, name: "results", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.ApplyRecommendationResult },
            { no: 2, name: "partial_failure_error", kind: "message", T: () => status_1.Status }
        ]);
    }
    create(value) {
        const message = { results: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.services.ApplyRecommendationResult results */ 1:
                    message.results.push(exports.ApplyRecommendationResult.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.rpc.Status partial_failure_error */ 2:
                    message.partialFailureError = status_1.Status.internalBinaryRead(reader, reader.uint32(), options, message.partialFailureError);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.services.ApplyRecommendationResult results = 1; */
        for (let i = 0; i < message.results.length; i++)
            exports.ApplyRecommendationResult.internalBinaryWrite(message.results[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.rpc.Status partial_failure_error = 2; */
        if (message.partialFailureError)
            status_1.Status.internalBinaryWrite(message.partialFailureError, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationResponse
 */
exports.ApplyRecommendationResponse = new ApplyRecommendationResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyRecommendationResult$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.ApplyRecommendationResult", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/Recommendation" } } }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.ApplyRecommendationResult
 */
exports.ApplyRecommendationResult = new ApplyRecommendationResult$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DismissRecommendationRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.DismissRecommendationRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "operations", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.DismissRecommendationRequest_DismissRecommendationOperation, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "partial_failure", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { customerId: "", operations: [], partialFailure: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.services.DismissRecommendationRequest.DismissRecommendationOperation operations */ 3:
                    message.operations.push(exports.DismissRecommendationRequest_DismissRecommendationOperation.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* bool partial_failure */ 2:
                    message.partialFailure = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* repeated google.ads.googleads.v11.services.DismissRecommendationRequest.DismissRecommendationOperation operations = 3; */
        for (let i = 0; i < message.operations.length; i++)
            exports.DismissRecommendationRequest_DismissRecommendationOperation.internalBinaryWrite(message.operations[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool partial_failure = 2; */
        if (message.partialFailure !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.partialFailure);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.DismissRecommendationRequest
 */
exports.DismissRecommendationRequest = new DismissRecommendationRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DismissRecommendationRequest_DismissRecommendationOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.DismissRecommendationRequest.DismissRecommendationOperation", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.DismissRecommendationRequest.DismissRecommendationOperation
 */
exports.DismissRecommendationRequest_DismissRecommendationOperation = new DismissRecommendationRequest_DismissRecommendationOperation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DismissRecommendationResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.DismissRecommendationResponse", [
            { no: 1, name: "results", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.DismissRecommendationResponse_DismissRecommendationResult },
            { no: 2, name: "partial_failure_error", kind: "message", T: () => status_1.Status }
        ]);
    }
    create(value) {
        const message = { results: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.services.DismissRecommendationResponse.DismissRecommendationResult results */ 1:
                    message.results.push(exports.DismissRecommendationResponse_DismissRecommendationResult.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.rpc.Status partial_failure_error */ 2:
                    message.partialFailureError = status_1.Status.internalBinaryRead(reader, reader.uint32(), options, message.partialFailureError);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.services.DismissRecommendationResponse.DismissRecommendationResult results = 1; */
        for (let i = 0; i < message.results.length; i++)
            exports.DismissRecommendationResponse_DismissRecommendationResult.internalBinaryWrite(message.results[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.rpc.Status partial_failure_error = 2; */
        if (message.partialFailureError)
            status_1.Status.internalBinaryWrite(message.partialFailureError, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.DismissRecommendationResponse
 */
exports.DismissRecommendationResponse = new DismissRecommendationResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DismissRecommendationResponse_DismissRecommendationResult$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.DismissRecommendationResponse.DismissRecommendationResult", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.DismissRecommendationResponse.DismissRecommendationResult
 */
exports.DismissRecommendationResponse_DismissRecommendationResult = new DismissRecommendationResponse_DismissRecommendationResult$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.RecommendationService
 */
exports.RecommendationService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.RecommendationService", [
    { name: "ApplyRecommendation", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}/recommendations:apply", body: "*" }, "google.api.method_signature": ["customer_id,operations"] }, I: exports.ApplyRecommendationRequest, O: exports.ApplyRecommendationResponse },
    { name: "DismissRecommendation", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}/recommendations:dismiss", body: "*" }, "google.api.method_signature": ["customer_id,operations"] }, I: exports.DismissRecommendationRequest, O: exports.DismissRecommendationResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
