// google-ads-api-client/dist/protos/bindings/ads/services/smart_campaign_suggest_service.client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmartCampaignSuggestServiceClient = void 0;
const smart_campaign_suggest_service_1 = require("./smart_campaign_suggest_service");
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
/**
 * Service to get suggestions for Smart Campaigns.
 *
 * @generated from protobuf service google.ads.googleads.v11.services.SmartCampaignSuggestService
 */
class SmartCampaignSuggestServiceClient {
    constructor(_transport) {
        this._transport = _transport;
        this.typeName = smart_campaign_suggest_service_1.SmartCampaignSuggestService.typeName;
        this.methods = smart_campaign_suggest_service_1.SmartCampaignSuggestService.methods;
        this.options = smart_campaign_suggest_service_1.SmartCampaignSuggestService.options;
    }
    /**
     * Returns BudgetOption suggestions.
     *
     * @generated from protobuf rpc: SuggestSmartCampaignBudgetOptions(google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsRequest) returns (google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse);
     */
    suggestSmartCampaignBudgetOptions(input, options) {
        const method = this.methods[0], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Suggests a Smart campaign ad compatible with the Ad family of resources,
     * based on data points such as targeting and the business to advertise.
     *
     * @generated from protobuf rpc: SuggestSmartCampaignAd(google.ads.googleads.v11.services.SuggestSmartCampaignAdRequest) returns (google.ads.googleads.v11.services.SuggestSmartCampaignAdResponse);
     */
    suggestSmartCampaignAd(input, options) {
        const method = this.methods[1], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
    /**
     * Suggests keyword themes to advertise on.
     *
     * @generated from protobuf rpc: SuggestKeywordThemes(google.ads.googleads.v11.services.SuggestKeywordThemesRequest) returns (google.ads.googleads.v11.services.SuggestKeywordThemesResponse);
     */
    suggestKeywordThemes(input, options) {
        const method = this.methods[2], opt = this._transport.mergeOptions(options);
        return (0, runtime_rpc_1.stackIntercept)("unary", this._transport, method, opt, input);
    }
}
exports.SmartCampaignSuggestServiceClient = SmartCampaignSuggestServiceClient;
