// google-ads-api-client/dist/protos/bindings/ads/services/smart_campaign_suggest_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmartCampaignSuggestService = exports.SuggestKeywordThemesResponse = exports.SuggestKeywordThemesRequest = exports.SuggestSmartCampaignAdResponse = exports.SuggestSmartCampaignAdRequest = exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption = exports.SuggestSmartCampaignBudgetOptionsResponse_Metrics = exports.SuggestSmartCampaignBudgetOptionsResponse = exports.SmartCampaignSuggestionInfo_BusinessContext = exports.SmartCampaignSuggestionInfo_LocationList = exports.SmartCampaignSuggestionInfo = exports.SuggestSmartCampaignBudgetOptionsRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/smart_campaign_suggest_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const keyword_theme_constant_1 = require("../resources/keyword_theme_constant");
const ad_type_infos_1 = require("../common/ad_type_infos");
const criteria_1 = require("../common/criteria");
const criteria_2 = require("../common/criteria");
const criteria_3 = require("../common/criteria");
const criteria_4 = require("../common/criteria");
// @generated message type with reflection information, may provide speed optimized methods
class SuggestSmartCampaignBudgetOptionsRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "campaign", kind: "scalar", oneof: "suggestionData", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 3, name: "suggestion_info", kind: "message", oneof: "suggestionData", T: () => exports.SmartCampaignSuggestionInfo, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { customerId: "", suggestionData: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* string campaign */ 2:
                    message.suggestionData = {
                        oneofKind: "campaign",
                        campaign: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo suggestion_info */ 3:
                    message.suggestionData = {
                        oneofKind: "suggestionInfo",
                        suggestionInfo: exports.SmartCampaignSuggestionInfo.internalBinaryRead(reader, reader.uint32(), options, message.suggestionData.suggestionInfo)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* string campaign = 2; */
        if (message.suggestionData.oneofKind === "campaign")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.suggestionData.campaign);
        /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo suggestion_info = 3; */
        if (message.suggestionData.oneofKind === "suggestionInfo")
            exports.SmartCampaignSuggestionInfo.internalBinaryWrite(message.suggestionData.suggestionInfo, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsRequest
 */
exports.SuggestSmartCampaignBudgetOptionsRequest = new SuggestSmartCampaignBudgetOptionsRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SmartCampaignSuggestionInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SmartCampaignSuggestionInfo", [
            { no: 1, name: "final_url", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 3, name: "language_code", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 6, name: "ad_schedules", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_4.AdScheduleInfo, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 7, name: "keyword_themes", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_3.KeywordThemeInfo, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 8, name: "business_context", kind: "message", oneof: "businessSetting", T: () => exports.SmartCampaignSuggestionInfo_BusinessContext, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 9, name: "business_profile_location", kind: "scalar", oneof: "businessSetting", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 4, name: "location_list", kind: "message", oneof: "geoTarget", T: () => exports.SmartCampaignSuggestionInfo_LocationList, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 5, name: "proximity", kind: "message", oneof: "geoTarget", T: () => criteria_2.ProximityInfo, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = { finalUrl: "", languageCode: "", adSchedules: [], keywordThemes: [], businessSetting: { oneofKind: undefined }, geoTarget: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string final_url */ 1:
                    message.finalUrl = reader.string();
                    break;
                case /* string language_code */ 3:
                    message.languageCode = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedules */ 6:
                    message.adSchedules.push(criteria_4.AdScheduleInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.KeywordThemeInfo keyword_themes */ 7:
                    message.keywordThemes.push(criteria_3.KeywordThemeInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo.BusinessContext business_context */ 8:
                    message.businessSetting = {
                        oneofKind: "businessContext",
                        businessContext: exports.SmartCampaignSuggestionInfo_BusinessContext.internalBinaryRead(reader, reader.uint32(), options, message.businessSetting.businessContext)
                    };
                    break;
                case /* string business_profile_location */ 9:
                    message.businessSetting = {
                        oneofKind: "businessProfileLocation",
                        businessProfileLocation: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo.LocationList location_list */ 4:
                    message.geoTarget = {
                        oneofKind: "locationList",
                        locationList: exports.SmartCampaignSuggestionInfo_LocationList.internalBinaryRead(reader, reader.uint32(), options, message.geoTarget.locationList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ProximityInfo proximity */ 5:
                    message.geoTarget = {
                        oneofKind: "proximity",
                        proximity: criteria_2.ProximityInfo.internalBinaryRead(reader, reader.uint32(), options, message.geoTarget.proximity)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string final_url = 1; */
        if (message.finalUrl !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.finalUrl);
        /* string language_code = 3; */
        if (message.languageCode !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedules = 6; */
        for (let i = 0; i < message.adSchedules.length; i++)
            criteria_4.AdScheduleInfo.internalBinaryWrite(message.adSchedules[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.KeywordThemeInfo keyword_themes = 7; */
        for (let i = 0; i < message.keywordThemes.length; i++)
            criteria_3.KeywordThemeInfo.internalBinaryWrite(message.keywordThemes[i], writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo.BusinessContext business_context = 8; */
        if (message.businessSetting.oneofKind === "businessContext")
            exports.SmartCampaignSuggestionInfo_BusinessContext.internalBinaryWrite(message.businessSetting.businessContext, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string business_profile_location = 9; */
        if (message.businessSetting.oneofKind === "businessProfileLocation")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.businessSetting.businessProfileLocation);
        /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo.LocationList location_list = 4; */
        if (message.geoTarget.oneofKind === "locationList")
            exports.SmartCampaignSuggestionInfo_LocationList.internalBinaryWrite(message.geoTarget.locationList, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ProximityInfo proximity = 5; */
        if (message.geoTarget.oneofKind === "proximity")
            criteria_2.ProximityInfo.internalBinaryWrite(message.geoTarget.proximity, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SmartCampaignSuggestionInfo
 */
exports.SmartCampaignSuggestionInfo = new SmartCampaignSuggestionInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SmartCampaignSuggestionInfo_LocationList$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SmartCampaignSuggestionInfo.LocationList", [
            { no: 1, name: "locations", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_1.LocationInfo, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { locations: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.LocationInfo locations */ 1:
                    message.locations.push(criteria_1.LocationInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.LocationInfo locations = 1; */
        for (let i = 0; i < message.locations.length; i++)
            criteria_1.LocationInfo.internalBinaryWrite(message.locations[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SmartCampaignSuggestionInfo.LocationList
 */
exports.SmartCampaignSuggestionInfo_LocationList = new SmartCampaignSuggestionInfo_LocationList$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SmartCampaignSuggestionInfo_BusinessContext$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SmartCampaignSuggestionInfo.BusinessContext", [
            { no: 1, name: "business_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = { businessName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string business_name */ 1:
                    message.businessName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string business_name = 1; */
        if (message.businessName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.businessName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SmartCampaignSuggestionInfo.BusinessContext
 */
exports.SmartCampaignSuggestionInfo_BusinessContext = new SmartCampaignSuggestionInfo_BusinessContext$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SuggestSmartCampaignBudgetOptionsResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse", [
            { no: 1, name: "low", kind: "message", T: () => exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 2, name: "recommended", kind: "message", T: () => exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 3, name: "high", kind: "message", T: () => exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.BudgetOption low */ 1:
                    message.low = exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption.internalBinaryRead(reader, reader.uint32(), options, message.low);
                    break;
                case /* optional google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.BudgetOption recommended */ 2:
                    message.recommended = exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption.internalBinaryRead(reader, reader.uint32(), options, message.recommended);
                    break;
                case /* optional google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.BudgetOption high */ 3:
                    message.high = exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption.internalBinaryRead(reader, reader.uint32(), options, message.high);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.BudgetOption low = 1; */
        if (message.low)
            exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption.internalBinaryWrite(message.low, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.BudgetOption recommended = 2; */
        if (message.recommended)
            exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption.internalBinaryWrite(message.recommended, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.BudgetOption high = 3; */
        if (message.high)
            exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption.internalBinaryWrite(message.high, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse
 */
exports.SuggestSmartCampaignBudgetOptionsResponse = new SuggestSmartCampaignBudgetOptionsResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SuggestSmartCampaignBudgetOptionsResponse_Metrics$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.Metrics", [
            { no: 1, name: "min_daily_clicks", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 2, name: "max_daily_clicks", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { minDailyClicks: 0n, maxDailyClicks: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int64 min_daily_clicks */ 1:
                    message.minDailyClicks = reader.int64().toBigInt();
                    break;
                case /* int64 max_daily_clicks */ 2:
                    message.maxDailyClicks = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int64 min_daily_clicks = 1; */
        if (message.minDailyClicks !== 0n)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.minDailyClicks);
        /* int64 max_daily_clicks = 2; */
        if (message.maxDailyClicks !== 0n)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.maxDailyClicks);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.Metrics
 */
exports.SuggestSmartCampaignBudgetOptionsResponse_Metrics = new SuggestSmartCampaignBudgetOptionsResponse_Metrics$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SuggestSmartCampaignBudgetOptionsResponse_BudgetOption$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.BudgetOption", [
            { no: 1, name: "daily_amount_micros", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 2, name: "metrics", kind: "message", T: () => exports.SuggestSmartCampaignBudgetOptionsResponse_Metrics }
        ]);
    }
    create(value) {
        const message = { dailyAmountMicros: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int64 daily_amount_micros */ 1:
                    message.dailyAmountMicros = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.Metrics metrics */ 2:
                    message.metrics = exports.SuggestSmartCampaignBudgetOptionsResponse_Metrics.internalBinaryRead(reader, reader.uint32(), options, message.metrics);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int64 daily_amount_micros = 1; */
        if (message.dailyAmountMicros !== 0n)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.dailyAmountMicros);
        /* google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.Metrics metrics = 2; */
        if (message.metrics)
            exports.SuggestSmartCampaignBudgetOptionsResponse_Metrics.internalBinaryWrite(message.metrics, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestSmartCampaignBudgetOptionsResponse.BudgetOption
 */
exports.SuggestSmartCampaignBudgetOptionsResponse_BudgetOption = new SuggestSmartCampaignBudgetOptionsResponse_BudgetOption$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SuggestSmartCampaignAdRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestSmartCampaignAdRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "suggestion_info", kind: "message", T: () => exports.SmartCampaignSuggestionInfo, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { customerId: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo suggestion_info */ 2:
                    message.suggestionInfo = exports.SmartCampaignSuggestionInfo.internalBinaryRead(reader, reader.uint32(), options, message.suggestionInfo);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo suggestion_info = 2; */
        if (message.suggestionInfo)
            exports.SmartCampaignSuggestionInfo.internalBinaryWrite(message.suggestionInfo, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestSmartCampaignAdRequest
 */
exports.SuggestSmartCampaignAdRequest = new SuggestSmartCampaignAdRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SuggestSmartCampaignAdResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestSmartCampaignAdResponse", [
            { no: 1, name: "ad_info", kind: "message", T: () => ad_type_infos_1.SmartCampaignAdInfo, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.SmartCampaignAdInfo ad_info */ 1:
                    message.adInfo = ad_type_infos_1.SmartCampaignAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adInfo);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.SmartCampaignAdInfo ad_info = 1; */
        if (message.adInfo)
            ad_type_infos_1.SmartCampaignAdInfo.internalBinaryWrite(message.adInfo, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestSmartCampaignAdResponse
 */
exports.SuggestSmartCampaignAdResponse = new SuggestSmartCampaignAdResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SuggestKeywordThemesRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestKeywordThemesRequest", [
            { no: 1, name: "customer_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "suggestion_info", kind: "message", T: () => exports.SmartCampaignSuggestionInfo, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { customerId: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string customer_id */ 1:
                    message.customerId = reader.string();
                    break;
                case /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo suggestion_info */ 2:
                    message.suggestionInfo = exports.SmartCampaignSuggestionInfo.internalBinaryRead(reader, reader.uint32(), options, message.suggestionInfo);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string customer_id = 1; */
        if (message.customerId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customerId);
        /* google.ads.googleads.v11.services.SmartCampaignSuggestionInfo suggestion_info = 2; */
        if (message.suggestionInfo)
            exports.SmartCampaignSuggestionInfo.internalBinaryWrite(message.suggestionInfo, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestKeywordThemesRequest
 */
exports.SuggestKeywordThemesRequest = new SuggestKeywordThemesRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SuggestKeywordThemesResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.SuggestKeywordThemesResponse", [
            { no: 1, name: "keyword_themes", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => keyword_theme_constant_1.KeywordThemeConstant }
        ]);
    }
    create(value) {
        const message = { keywordThemes: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.resources.KeywordThemeConstant keyword_themes */ 1:
                    message.keywordThemes.push(keyword_theme_constant_1.KeywordThemeConstant.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.resources.KeywordThemeConstant keyword_themes = 1; */
        for (let i = 0; i < message.keywordThemes.length; i++)
            keyword_theme_constant_1.KeywordThemeConstant.internalBinaryWrite(message.keywordThemes[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.SuggestKeywordThemesResponse
 */
exports.SuggestKeywordThemesResponse = new SuggestKeywordThemesResponse$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.SmartCampaignSuggestService
 */
exports.SmartCampaignSuggestService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.SmartCampaignSuggestService", [
    { name: "SuggestSmartCampaignBudgetOptions", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}:suggestSmartCampaignBudgetOptions", body: "*" } }, I: exports.SuggestSmartCampaignBudgetOptionsRequest, O: exports.SuggestSmartCampaignBudgetOptionsResponse },
    { name: "SuggestSmartCampaignAd", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}:suggestSmartCampaignAd", body: "*" } }, I: exports.SuggestSmartCampaignAdRequest, O: exports.SuggestSmartCampaignAdResponse },
    { name: "SuggestKeywordThemes", options: { "google.api.http": { post: "/v11/customers/{customer_id=*}:suggestKeywordThemes", body: "*" } }, I: exports.SuggestKeywordThemesRequest, O: exports.SuggestKeywordThemesResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
