// google-ads-api-client/dist/protos/bindings/ads/services/third_party_app_analytics_link_service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThirdPartyAppAnalyticsLinkService = exports.RegenerateShareableLinkIdResponse = exports.RegenerateShareableLinkIdRequest = void 0;
// @generated by protobuf-ts 2.8.1
// @generated from protobuf file "ads/services/third_party_app_analytics_link_service.proto" (package "google.ads.googleads.v11.services", syntax proto3)
// tslint:disable
//
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
const runtime_rpc_1 = require("@protobuf-ts/runtime-rpc");
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
// @generated message type with reflection information, may provide speed optimized methods
class RegenerateShareableLinkIdRequest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.RegenerateShareableLinkIdRequest", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/ThirdPartyAppAnalyticsLink" } } }
        ]);
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.RegenerateShareableLinkIdRequest
 */
exports.RegenerateShareableLinkIdRequest = new RegenerateShareableLinkIdRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class RegenerateShareableLinkIdResponse$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.services.RegenerateShareableLinkIdResponse", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.services.RegenerateShareableLinkIdResponse
 */
exports.RegenerateShareableLinkIdResponse = new RegenerateShareableLinkIdResponse$Type();
/**
 * @generated ServiceType for protobuf service google.ads.googleads.v11.services.ThirdPartyAppAnalyticsLinkService
 */
exports.ThirdPartyAppAnalyticsLinkService = new runtime_rpc_1.ServiceType("google.ads.googleads.v11.services.ThirdPartyAppAnalyticsLinkService", [
    { name: "RegenerateShareableLinkId", options: { "google.api.http": { post: "/v11/{resource_name=customers/*/thirdPartyAppAnalyticsLinks/*}:regenerateShareableLinkId", body: "*" } }, I: exports.RegenerateShareableLinkIdRequest, O: exports.RegenerateShareableLinkIdResponse }
], { "google.api.default_host": "googleads.googleapis.com", "google.api.oauth_scopes": "https://www.googleapis.com/auth/adwords" });
