// google-ads-api-client/dist/protos/bindings/ads/common/ad_asset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdDiscoveryCarouselCardAsset = exports.AdMediaBundleAsset = exports.AdVideoAsset = exports.AdImageAsset = exports.AdTextAsset = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const asset_policy_1 = require("./asset_policy");
const asset_performance_label_1 = require("../enums/asset_performance_label");
const served_asset_field_type_1 = require("../enums/served_asset_field_type");
// @generated message type with reflection information, may provide speed optimized methods
class AdTextAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AdTextAsset", [
            { no: 4, name: "text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "pinned_field", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ServedAssetFieldTypeEnum.ServedAssetFieldType", served_asset_field_type_1.ServedAssetFieldTypeEnum_ServedAssetFieldType] },
            { no: 5, name: "asset_performance_label", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AssetPerformanceLabelEnum.AssetPerformanceLabel", asset_performance_label_1.AssetPerformanceLabelEnum_AssetPerformanceLabel] },
            { no: 6, name: "policy_summary_info", kind: "message", T: () => asset_policy_1.AdAssetPolicySummary }
        ]);
    }
    create(value) {
        const message = { pinnedField: 0, assetPerformanceLabel: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string text */ 4:
                    message.text = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.ServedAssetFieldTypeEnum.ServedAssetFieldType pinned_field */ 2:
                    message.pinnedField = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.AssetPerformanceLabelEnum.AssetPerformanceLabel asset_performance_label */ 5:
                    message.assetPerformanceLabel = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.AdAssetPolicySummary policy_summary_info */ 6:
                    message.policySummaryInfo = asset_policy_1.AdAssetPolicySummary.internalBinaryRead(reader, reader.uint32(), options, message.policySummaryInfo);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string text = 4; */
        if (message.text !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.text);
        /* google.ads.googleads.v11.enums.ServedAssetFieldTypeEnum.ServedAssetFieldType pinned_field = 2; */
        if (message.pinnedField !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.pinnedField);
        /* google.ads.googleads.v11.enums.AssetPerformanceLabelEnum.AssetPerformanceLabel asset_performance_label = 5; */
        if (message.assetPerformanceLabel !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.assetPerformanceLabel);
        /* google.ads.googleads.v11.common.AdAssetPolicySummary policy_summary_info = 6; */
        if (message.policySummaryInfo)
            asset_policy_1.AdAssetPolicySummary.internalBinaryWrite(message.policySummaryInfo, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AdTextAsset
 */
exports.AdTextAsset = new AdTextAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AdImageAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AdImageAsset", [
            { no: 2, name: "asset", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string asset */ 2:
                    message.asset = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string asset = 2; */
        if (message.asset !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.asset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AdImageAsset
 */
exports.AdImageAsset = new AdImageAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AdVideoAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AdVideoAsset", [
            { no: 2, name: "asset", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string asset */ 2:
                    message.asset = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string asset = 2; */
        if (message.asset !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.asset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AdVideoAsset
 */
exports.AdVideoAsset = new AdVideoAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AdMediaBundleAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AdMediaBundleAsset", [
            { no: 2, name: "asset", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string asset */ 2:
                    message.asset = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string asset = 2; */
        if (message.asset !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.asset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AdMediaBundleAsset
 */
exports.AdMediaBundleAsset = new AdMediaBundleAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AdDiscoveryCarouselCardAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AdDiscoveryCarouselCardAsset", [
            { no: 1, name: "asset", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string asset */ 1:
                    message.asset = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string asset = 1; */
        if (message.asset !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.asset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AdDiscoveryCarouselCardAsset
 */
exports.AdDiscoveryCarouselCardAsset = new AdDiscoveryCarouselCardAsset$Type();
