// google-ads-api-client/dist/protos/bindings/ads/common/ad_type_infos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryCarouselAdInfo = exports.DiscoveryMultiAssetAdInfo = exports.CallAdInfo = exports.SmartCampaignAdInfo = exports.ResponsiveDisplayAdControlSpec = exports.DisplayUploadAdInfo = exports.LocalAdInfo = exports.ResponsiveDisplayAdInfo = exports.LegacyAppInstallAdInfo = exports.AppPreRegistrationAdInfo = exports.AppEngagementAdInfo = exports.AppAdInfo = exports.LegacyResponsiveDisplayAdInfo = exports.ResponsiveSearchAdInfo = exports.VideoResponsiveAdInfo = exports.VideoAdInfo = exports.InFeedVideoAdInfo = exports.VideoOutstreamAdInfo = exports.VideoTrueViewInStreamAdInfo = exports.VideoNonSkippableInStreamAdInfo = exports.VideoBumperInStreamAdInfo = exports.ImageAdInfo = exports.ProductVideo = exports.ProductImage = exports.DisplayCallToAction = exports.GmailTeaser = exports.GmailAdInfo = exports.ShoppingComparisonListingAdInfo = exports.ShoppingProductAdInfo = exports.ShoppingSmartAdInfo = exports.HotelAdInfo = exports.ExpandedDynamicSearchAdInfo = exports.ExpandedTextAdInfo = exports.TextAdInfo = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const ad_asset_1 = require("./ad_asset");
const call_conversion_reporting_state_1 = require("../enums/call_conversion_reporting_state");
const display_upload_product_type_1 = require("../enums/display_upload_product_type");
const legacy_app_install_ad_app_store_1 = require("../enums/legacy_app_install_ad_app_store");
const ad_asset_2 = require("./ad_asset");
const display_ad_format_setting_1 = require("../enums/display_ad_format_setting");
const ad_asset_3 = require("./ad_asset");
const ad_asset_4 = require("./ad_asset");
const video_thumbnail_1 = require("../enums/video_thumbnail");
const ad_asset_5 = require("./ad_asset");
const mime_type_1 = require("../enums/mime_type");
// @generated message type with reflection information, may provide speed optimized methods
class TextAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.TextAdInfo", [
            { no: 4, name: "headline", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "description1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "description2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string headline */ 4:
                    message.headline = reader.string();
                    break;
                case /* optional string description1 */ 5:
                    message.description1 = reader.string();
                    break;
                case /* optional string description2 */ 6:
                    message.description2 = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string headline = 4; */
        if (message.headline !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.headline);
        /* optional string description1 = 5; */
        if (message.description1 !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.description1);
        /* optional string description2 = 6; */
        if (message.description2 !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.description2);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.TextAdInfo
 */
exports.TextAdInfo = new TextAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ExpandedTextAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ExpandedTextAdInfo", [
            { no: 8, name: "headline_part1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "headline_part2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "headline_part3", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "description2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "path1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "path2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string headline_part1 */ 8:
                    message.headlinePart1 = reader.string();
                    break;
                case /* optional string headline_part2 */ 9:
                    message.headlinePart2 = reader.string();
                    break;
                case /* optional string headline_part3 */ 10:
                    message.headlinePart3 = reader.string();
                    break;
                case /* optional string description */ 11:
                    message.description = reader.string();
                    break;
                case /* optional string description2 */ 12:
                    message.description2 = reader.string();
                    break;
                case /* optional string path1 */ 13:
                    message.path1 = reader.string();
                    break;
                case /* optional string path2 */ 14:
                    message.path2 = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string headline_part1 = 8; */
        if (message.headlinePart1 !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.headlinePart1);
        /* optional string headline_part2 = 9; */
        if (message.headlinePart2 !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.headlinePart2);
        /* optional string headline_part3 = 10; */
        if (message.headlinePart3 !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.headlinePart3);
        /* optional string description = 11; */
        if (message.description !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.description);
        /* optional string description2 = 12; */
        if (message.description2 !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.description2);
        /* optional string path1 = 13; */
        if (message.path1 !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.path1);
        /* optional string path2 = 14; */
        if (message.path2 !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.path2);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ExpandedTextAdInfo
 */
exports.ExpandedTextAdInfo = new ExpandedTextAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ExpandedDynamicSearchAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ExpandedDynamicSearchAdInfo", [
            { no: 3, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "description2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string description */ 3:
                    message.description = reader.string();
                    break;
                case /* optional string description2 */ 4:
                    message.description2 = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string description = 3; */
        if (message.description !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.description);
        /* optional string description2 = 4; */
        if (message.description2 !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.description2);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ExpandedDynamicSearchAdInfo
 */
exports.ExpandedDynamicSearchAdInfo = new ExpandedDynamicSearchAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class HotelAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.HotelAdInfo", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.HotelAdInfo
 */
exports.HotelAdInfo = new HotelAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ShoppingSmartAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ShoppingSmartAdInfo", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ShoppingSmartAdInfo
 */
exports.ShoppingSmartAdInfo = new ShoppingSmartAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ShoppingProductAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ShoppingProductAdInfo", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ShoppingProductAdInfo
 */
exports.ShoppingProductAdInfo = new ShoppingProductAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ShoppingComparisonListingAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ShoppingComparisonListingAdInfo", [
            { no: 2, name: "headline", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string headline */ 2:
                    message.headline = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string headline = 2; */
        if (message.headline !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.headline);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ShoppingComparisonListingAdInfo
 */
exports.ShoppingComparisonListingAdInfo = new ShoppingComparisonListingAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GmailAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.GmailAdInfo", [
            { no: 1, name: "teaser", kind: "message", T: () => exports.GmailTeaser },
            { no: 10, name: "header_image", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "marketing_image", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "marketing_image_headline", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "marketing_image_description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "marketing_image_display_call_to_action", kind: "message", T: () => exports.DisplayCallToAction },
            { no: 7, name: "product_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.ProductImage },
            { no: 8, name: "product_videos", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.ProductVideo }
        ]);
    }
    create(value) {
        const message = { productImages: [], productVideos: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.GmailTeaser teaser */ 1:
                    message.teaser = exports.GmailTeaser.internalBinaryRead(reader, reader.uint32(), options, message.teaser);
                    break;
                case /* optional string header_image */ 10:
                    message.headerImage = reader.string();
                    break;
                case /* optional string marketing_image */ 11:
                    message.marketingImage = reader.string();
                    break;
                case /* optional string marketing_image_headline */ 12:
                    message.marketingImageHeadline = reader.string();
                    break;
                case /* optional string marketing_image_description */ 13:
                    message.marketingImageDescription = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.DisplayCallToAction marketing_image_display_call_to_action */ 6:
                    message.marketingImageDisplayCallToAction = exports.DisplayCallToAction.internalBinaryRead(reader, reader.uint32(), options, message.marketingImageDisplayCallToAction);
                    break;
                case /* repeated google.ads.googleads.v11.common.ProductImage product_images */ 7:
                    message.productImages.push(exports.ProductImage.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.ProductVideo product_videos */ 8:
                    message.productVideos.push(exports.ProductVideo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.GmailTeaser teaser = 1; */
        if (message.teaser)
            exports.GmailTeaser.internalBinaryWrite(message.teaser, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string header_image = 10; */
        if (message.headerImage !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.headerImage);
        /* optional string marketing_image = 11; */
        if (message.marketingImage !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.marketingImage);
        /* optional string marketing_image_headline = 12; */
        if (message.marketingImageHeadline !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.marketingImageHeadline);
        /* optional string marketing_image_description = 13; */
        if (message.marketingImageDescription !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.marketingImageDescription);
        /* google.ads.googleads.v11.common.DisplayCallToAction marketing_image_display_call_to_action = 6; */
        if (message.marketingImageDisplayCallToAction)
            exports.DisplayCallToAction.internalBinaryWrite(message.marketingImageDisplayCallToAction, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.ProductImage product_images = 7; */
        for (let i = 0; i < message.productImages.length; i++)
            exports.ProductImage.internalBinaryWrite(message.productImages[i], writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.ProductVideo product_videos = 8; */
        for (let i = 0; i < message.productVideos.length; i++)
            exports.ProductVideo.internalBinaryWrite(message.productVideos[i], writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.GmailAdInfo
 */
exports.GmailAdInfo = new GmailAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GmailTeaser$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.GmailTeaser", [
            { no: 5, name: "headline", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "business_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "logo_image", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string headline */ 5:
                    message.headline = reader.string();
                    break;
                case /* optional string description */ 6:
                    message.description = reader.string();
                    break;
                case /* optional string business_name */ 7:
                    message.businessName = reader.string();
                    break;
                case /* optional string logo_image */ 8:
                    message.logoImage = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string headline = 5; */
        if (message.headline !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.headline);
        /* optional string description = 6; */
        if (message.description !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.description);
        /* optional string business_name = 7; */
        if (message.businessName !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* optional string logo_image = 8; */
        if (message.logoImage !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.logoImage);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.GmailTeaser
 */
exports.GmailTeaser = new GmailTeaser$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DisplayCallToAction$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DisplayCallToAction", [
            { no: 5, name: "text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "text_color", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "url_collection_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string text */ 5:
                    message.text = reader.string();
                    break;
                case /* optional string text_color */ 6:
                    message.textColor = reader.string();
                    break;
                case /* optional string url_collection_id */ 7:
                    message.urlCollectionId = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string text = 5; */
        if (message.text !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.text);
        /* optional string text_color = 6; */
        if (message.textColor !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.textColor);
        /* optional string url_collection_id = 7; */
        if (message.urlCollectionId !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.urlCollectionId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DisplayCallToAction
 */
exports.DisplayCallToAction = new DisplayCallToAction$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ProductImage$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ProductImage", [
            { no: 4, name: "product_image", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "display_call_to_action", kind: "message", T: () => exports.DisplayCallToAction }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string product_image */ 4:
                    message.productImage = reader.string();
                    break;
                case /* optional string description */ 5:
                    message.description = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.DisplayCallToAction display_call_to_action */ 3:
                    message.displayCallToAction = exports.DisplayCallToAction.internalBinaryRead(reader, reader.uint32(), options, message.displayCallToAction);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string product_image = 4; */
        if (message.productImage !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.productImage);
        /* optional string description = 5; */
        if (message.description !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.description);
        /* google.ads.googleads.v11.common.DisplayCallToAction display_call_to_action = 3; */
        if (message.displayCallToAction)
            exports.DisplayCallToAction.internalBinaryWrite(message.displayCallToAction, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ProductImage
 */
exports.ProductImage = new ProductImage$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ProductVideo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ProductVideo", [
            { no: 2, name: "product_video", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string product_video */ 2:
                    message.productVideo = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string product_video = 2; */
        if (message.productVideo !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.productVideo);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ProductVideo
 */
exports.ProductVideo = new ProductVideo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImageAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ImageAdInfo", [
            { no: 15, name: "pixel_width", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 16, name: "pixel_height", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 17, name: "image_url", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 18, name: "preview_pixel_width", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 19, name: "preview_pixel_height", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 20, name: "preview_image_url", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "mime_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.MimeTypeEnum.MimeType", mime_type_1.MimeTypeEnum_MimeType] },
            { no: 21, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "media_file", kind: "scalar", oneof: "image", T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "data", kind: "scalar", oneof: "image", T: 12 /*ScalarType.BYTES*/ },
            { no: 14, name: "ad_id_to_copy_image_from", kind: "scalar", oneof: "image", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { mimeType: 0, image: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 pixel_width */ 15:
                    message.pixelWidth = reader.int64().toBigInt();
                    break;
                case /* optional int64 pixel_height */ 16:
                    message.pixelHeight = reader.int64().toBigInt();
                    break;
                case /* optional string image_url */ 17:
                    message.imageUrl = reader.string();
                    break;
                case /* optional int64 preview_pixel_width */ 18:
                    message.previewPixelWidth = reader.int64().toBigInt();
                    break;
                case /* optional int64 preview_pixel_height */ 19:
                    message.previewPixelHeight = reader.int64().toBigInt();
                    break;
                case /* optional string preview_image_url */ 20:
                    message.previewImageUrl = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.MimeTypeEnum.MimeType mime_type */ 10:
                    message.mimeType = reader.int32();
                    break;
                case /* optional string name */ 21:
                    message.name = reader.string();
                    break;
                case /* string media_file */ 12:
                    message.image = {
                        oneofKind: "mediaFile",
                        mediaFile: reader.string()
                    };
                    break;
                case /* bytes data */ 13:
                    message.image = {
                        oneofKind: "data",
                        data: reader.bytes()
                    };
                    break;
                case /* int64 ad_id_to_copy_image_from */ 14:
                    message.image = {
                        oneofKind: "adIdToCopyImageFrom",
                        adIdToCopyImageFrom: reader.int64().toBigInt()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 pixel_width = 15; */
        if (message.pixelWidth !== undefined)
            writer.tag(15, runtime_1.WireType.Varint).int64(message.pixelWidth);
        /* optional int64 pixel_height = 16; */
        if (message.pixelHeight !== undefined)
            writer.tag(16, runtime_1.WireType.Varint).int64(message.pixelHeight);
        /* optional string image_url = 17; */
        if (message.imageUrl !== undefined)
            writer.tag(17, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        /* optional int64 preview_pixel_width = 18; */
        if (message.previewPixelWidth !== undefined)
            writer.tag(18, runtime_1.WireType.Varint).int64(message.previewPixelWidth);
        /* optional int64 preview_pixel_height = 19; */
        if (message.previewPixelHeight !== undefined)
            writer.tag(19, runtime_1.WireType.Varint).int64(message.previewPixelHeight);
        /* optional string preview_image_url = 20; */
        if (message.previewImageUrl !== undefined)
            writer.tag(20, runtime_1.WireType.LengthDelimited).string(message.previewImageUrl);
        /* google.ads.googleads.v11.enums.MimeTypeEnum.MimeType mime_type = 10; */
        if (message.mimeType !== 0)
            writer.tag(10, runtime_1.WireType.Varint).int32(message.mimeType);
        /* optional string name = 21; */
        if (message.name !== undefined)
            writer.tag(21, runtime_1.WireType.LengthDelimited).string(message.name);
        /* string media_file = 12; */
        if (message.image.oneofKind === "mediaFile")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.image.mediaFile);
        /* bytes data = 13; */
        if (message.image.oneofKind === "data")
            writer.tag(13, runtime_1.WireType.LengthDelimited).bytes(message.image.data);
        /* int64 ad_id_to_copy_image_from = 14; */
        if (message.image.oneofKind === "adIdToCopyImageFrom")
            writer.tag(14, runtime_1.WireType.Varint).int64(message.image.adIdToCopyImageFrom);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ImageAdInfo
 */
exports.ImageAdInfo = new ImageAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class VideoBumperInStreamAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.VideoBumperInStreamAdInfo", [
            { no: 3, name: "companion_banner", kind: "message", T: () => ad_asset_5.AdImageAsset }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.AdImageAsset companion_banner */ 3:
                    message.companionBanner = ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options, message.companionBanner);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.AdImageAsset companion_banner = 3; */
        if (message.companionBanner)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.companionBanner, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.VideoBumperInStreamAdInfo
 */
exports.VideoBumperInStreamAdInfo = new VideoBumperInStreamAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class VideoNonSkippableInStreamAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.VideoNonSkippableInStreamAdInfo", [
            { no: 5, name: "companion_banner", kind: "message", T: () => ad_asset_5.AdImageAsset },
            { no: 3, name: "action_button_label", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "action_headline", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { actionButtonLabel: "", actionHeadline: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.AdImageAsset companion_banner */ 5:
                    message.companionBanner = ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options, message.companionBanner);
                    break;
                case /* string action_button_label */ 3:
                    message.actionButtonLabel = reader.string();
                    break;
                case /* string action_headline */ 4:
                    message.actionHeadline = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.AdImageAsset companion_banner = 5; */
        if (message.companionBanner)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.companionBanner, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string action_button_label = 3; */
        if (message.actionButtonLabel !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.actionButtonLabel);
        /* string action_headline = 4; */
        if (message.actionHeadline !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.actionHeadline);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.VideoNonSkippableInStreamAdInfo
 */
exports.VideoNonSkippableInStreamAdInfo = new VideoNonSkippableInStreamAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class VideoTrueViewInStreamAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.VideoTrueViewInStreamAdInfo", [
            { no: 4, name: "action_button_label", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "action_headline", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "companion_banner", kind: "message", T: () => ad_asset_5.AdImageAsset }
        ]);
    }
    create(value) {
        const message = { actionButtonLabel: "", actionHeadline: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string action_button_label */ 4:
                    message.actionButtonLabel = reader.string();
                    break;
                case /* string action_headline */ 5:
                    message.actionHeadline = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.AdImageAsset companion_banner */ 7:
                    message.companionBanner = ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options, message.companionBanner);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string action_button_label = 4; */
        if (message.actionButtonLabel !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.actionButtonLabel);
        /* string action_headline = 5; */
        if (message.actionHeadline !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.actionHeadline);
        /* google.ads.googleads.v11.common.AdImageAsset companion_banner = 7; */
        if (message.companionBanner)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.companionBanner, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.VideoTrueViewInStreamAdInfo
 */
exports.VideoTrueViewInStreamAdInfo = new VideoTrueViewInStreamAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class VideoOutstreamAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.VideoOutstreamAdInfo", [
            { no: 3, name: "headline", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { headline: "", description: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string headline */ 3:
                    message.headline = reader.string();
                    break;
                case /* string description */ 4:
                    message.description = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string headline = 3; */
        if (message.headline !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.headline);
        /* string description = 4; */
        if (message.description !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.description);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.VideoOutstreamAdInfo
 */
exports.VideoOutstreamAdInfo = new VideoOutstreamAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class InFeedVideoAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.InFeedVideoAdInfo", [
            { no: 1, name: "headline", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "description1", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "description2", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "thumbnail", kind: "enum", T: () => ["google.ads.googleads.v11.enums.VideoThumbnailEnum.VideoThumbnail", video_thumbnail_1.VideoThumbnailEnum_VideoThumbnail] }
        ]);
    }
    create(value) {
        const message = { headline: "", description1: "", description2: "", thumbnail: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string headline */ 1:
                    message.headline = reader.string();
                    break;
                case /* string description1 */ 2:
                    message.description1 = reader.string();
                    break;
                case /* string description2 */ 3:
                    message.description2 = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.VideoThumbnailEnum.VideoThumbnail thumbnail */ 4:
                    message.thumbnail = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string headline = 1; */
        if (message.headline !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.headline);
        /* string description1 = 2; */
        if (message.description1 !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.description1);
        /* string description2 = 3; */
        if (message.description2 !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.description2);
        /* google.ads.googleads.v11.enums.VideoThumbnailEnum.VideoThumbnail thumbnail = 4; */
        if (message.thumbnail !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.thumbnail);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.InFeedVideoAdInfo
 */
exports.InFeedVideoAdInfo = new InFeedVideoAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class VideoAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.VideoAdInfo", [
            { no: 8, name: "video", kind: "message", T: () => ad_asset_4.AdVideoAsset },
            { no: 2, name: "in_stream", kind: "message", oneof: "format", T: () => exports.VideoTrueViewInStreamAdInfo },
            { no: 3, name: "bumper", kind: "message", oneof: "format", T: () => exports.VideoBumperInStreamAdInfo },
            { no: 4, name: "out_stream", kind: "message", oneof: "format", T: () => exports.VideoOutstreamAdInfo },
            { no: 5, name: "non_skippable", kind: "message", oneof: "format", T: () => exports.VideoNonSkippableInStreamAdInfo },
            { no: 9, name: "in_feed", kind: "message", oneof: "format", T: () => exports.InFeedVideoAdInfo }
        ]);
    }
    create(value) {
        const message = { format: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.AdVideoAsset video */ 8:
                    message.video = ad_asset_4.AdVideoAsset.internalBinaryRead(reader, reader.uint32(), options, message.video);
                    break;
                case /* google.ads.googleads.v11.common.VideoTrueViewInStreamAdInfo in_stream */ 2:
                    message.format = {
                        oneofKind: "inStream",
                        inStream: exports.VideoTrueViewInStreamAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.format.inStream)
                    };
                    break;
                case /* google.ads.googleads.v11.common.VideoBumperInStreamAdInfo bumper */ 3:
                    message.format = {
                        oneofKind: "bumper",
                        bumper: exports.VideoBumperInStreamAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.format.bumper)
                    };
                    break;
                case /* google.ads.googleads.v11.common.VideoOutstreamAdInfo out_stream */ 4:
                    message.format = {
                        oneofKind: "outStream",
                        outStream: exports.VideoOutstreamAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.format.outStream)
                    };
                    break;
                case /* google.ads.googleads.v11.common.VideoNonSkippableInStreamAdInfo non_skippable */ 5:
                    message.format = {
                        oneofKind: "nonSkippable",
                        nonSkippable: exports.VideoNonSkippableInStreamAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.format.nonSkippable)
                    };
                    break;
                case /* google.ads.googleads.v11.common.InFeedVideoAdInfo in_feed */ 9:
                    message.format = {
                        oneofKind: "inFeed",
                        inFeed: exports.InFeedVideoAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.format.inFeed)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.AdVideoAsset video = 8; */
        if (message.video)
            ad_asset_4.AdVideoAsset.internalBinaryWrite(message.video, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.VideoTrueViewInStreamAdInfo in_stream = 2; */
        if (message.format.oneofKind === "inStream")
            exports.VideoTrueViewInStreamAdInfo.internalBinaryWrite(message.format.inStream, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.VideoBumperInStreamAdInfo bumper = 3; */
        if (message.format.oneofKind === "bumper")
            exports.VideoBumperInStreamAdInfo.internalBinaryWrite(message.format.bumper, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.VideoOutstreamAdInfo out_stream = 4; */
        if (message.format.oneofKind === "outStream")
            exports.VideoOutstreamAdInfo.internalBinaryWrite(message.format.outStream, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.VideoNonSkippableInStreamAdInfo non_skippable = 5; */
        if (message.format.oneofKind === "nonSkippable")
            exports.VideoNonSkippableInStreamAdInfo.internalBinaryWrite(message.format.nonSkippable, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.InFeedVideoAdInfo in_feed = 9; */
        if (message.format.oneofKind === "inFeed")
            exports.InFeedVideoAdInfo.internalBinaryWrite(message.format.inFeed, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.VideoAdInfo
 */
exports.VideoAdInfo = new VideoAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class VideoResponsiveAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.VideoResponsiveAdInfo", [
            { no: 1, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 2, name: "long_headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 3, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 4, name: "call_to_actions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 5, name: "videos", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_4.AdVideoAsset },
            { no: 6, name: "companion_banners", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 7, name: "breadcrumb1", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "breadcrumb2", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { headlines: [], longHeadlines: [], descriptions: [], callToActions: [], videos: [], companionBanners: [], breadcrumb1: "", breadcrumb2: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 1:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset long_headlines */ 2:
                    message.longHeadlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 3:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset call_to_actions */ 4:
                    message.callToActions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdVideoAsset videos */ 5:
                    message.videos.push(ad_asset_4.AdVideoAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset companion_banners */ 6:
                    message.companionBanners.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string breadcrumb1 */ 7:
                    message.breadcrumb1 = reader.string();
                    break;
                case /* string breadcrumb2 */ 8:
                    message.breadcrumb2 = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 1; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset long_headlines = 2; */
        for (let i = 0; i < message.longHeadlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.longHeadlines[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 3; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset call_to_actions = 4; */
        for (let i = 0; i < message.callToActions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.callToActions[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdVideoAsset videos = 5; */
        for (let i = 0; i < message.videos.length; i++)
            ad_asset_4.AdVideoAsset.internalBinaryWrite(message.videos[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset companion_banners = 6; */
        for (let i = 0; i < message.companionBanners.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.companionBanners[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string breadcrumb1 = 7; */
        if (message.breadcrumb1 !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.breadcrumb1);
        /* string breadcrumb2 = 8; */
        if (message.breadcrumb2 !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.breadcrumb2);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.VideoResponsiveAdInfo
 */
exports.VideoResponsiveAdInfo = new VideoResponsiveAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ResponsiveSearchAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ResponsiveSearchAdInfo", [
            { no: 1, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 2, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 5, name: "path1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "path2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { headlines: [], descriptions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 1:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 2:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string path1 */ 5:
                    message.path1 = reader.string();
                    break;
                case /* optional string path2 */ 6:
                    message.path2 = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 1; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 2; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string path1 = 5; */
        if (message.path1 !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.path1);
        /* optional string path2 = 6; */
        if (message.path2 !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.path2);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ResponsiveSearchAdInfo
 */
exports.ResponsiveSearchAdInfo = new ResponsiveSearchAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LegacyResponsiveDisplayAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LegacyResponsiveDisplayAdInfo", [
            { no: 16, name: "short_headline", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 17, name: "long_headline", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 18, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 19, name: "business_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 20, name: "allow_flexible_color", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 21, name: "accent_color", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 22, name: "main_color", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 23, name: "call_to_action_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 24, name: "logo_image", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 25, name: "square_logo_image", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 26, name: "marketing_image", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 27, name: "square_marketing_image", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "format_setting", kind: "enum", T: () => ["google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum.DisplayAdFormatSetting", display_ad_format_setting_1.DisplayAdFormatSettingEnum_DisplayAdFormatSetting] },
            { no: 28, name: "price_prefix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 29, name: "promo_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { formatSetting: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string short_headline */ 16:
                    message.shortHeadline = reader.string();
                    break;
                case /* optional string long_headline */ 17:
                    message.longHeadline = reader.string();
                    break;
                case /* optional string description */ 18:
                    message.description = reader.string();
                    break;
                case /* optional string business_name */ 19:
                    message.businessName = reader.string();
                    break;
                case /* optional bool allow_flexible_color */ 20:
                    message.allowFlexibleColor = reader.bool();
                    break;
                case /* optional string accent_color */ 21:
                    message.accentColor = reader.string();
                    break;
                case /* optional string main_color */ 22:
                    message.mainColor = reader.string();
                    break;
                case /* optional string call_to_action_text */ 23:
                    message.callToActionText = reader.string();
                    break;
                case /* optional string logo_image */ 24:
                    message.logoImage = reader.string();
                    break;
                case /* optional string square_logo_image */ 25:
                    message.squareLogoImage = reader.string();
                    break;
                case /* optional string marketing_image */ 26:
                    message.marketingImage = reader.string();
                    break;
                case /* optional string square_marketing_image */ 27:
                    message.squareMarketingImage = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum.DisplayAdFormatSetting format_setting */ 13:
                    message.formatSetting = reader.int32();
                    break;
                case /* optional string price_prefix */ 28:
                    message.pricePrefix = reader.string();
                    break;
                case /* optional string promo_text */ 29:
                    message.promoText = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string short_headline = 16; */
        if (message.shortHeadline !== undefined)
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.shortHeadline);
        /* optional string long_headline = 17; */
        if (message.longHeadline !== undefined)
            writer.tag(17, runtime_1.WireType.LengthDelimited).string(message.longHeadline);
        /* optional string description = 18; */
        if (message.description !== undefined)
            writer.tag(18, runtime_1.WireType.LengthDelimited).string(message.description);
        /* optional string business_name = 19; */
        if (message.businessName !== undefined)
            writer.tag(19, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* optional bool allow_flexible_color = 20; */
        if (message.allowFlexibleColor !== undefined)
            writer.tag(20, runtime_1.WireType.Varint).bool(message.allowFlexibleColor);
        /* optional string accent_color = 21; */
        if (message.accentColor !== undefined)
            writer.tag(21, runtime_1.WireType.LengthDelimited).string(message.accentColor);
        /* optional string main_color = 22; */
        if (message.mainColor !== undefined)
            writer.tag(22, runtime_1.WireType.LengthDelimited).string(message.mainColor);
        /* optional string call_to_action_text = 23; */
        if (message.callToActionText !== undefined)
            writer.tag(23, runtime_1.WireType.LengthDelimited).string(message.callToActionText);
        /* optional string logo_image = 24; */
        if (message.logoImage !== undefined)
            writer.tag(24, runtime_1.WireType.LengthDelimited).string(message.logoImage);
        /* optional string square_logo_image = 25; */
        if (message.squareLogoImage !== undefined)
            writer.tag(25, runtime_1.WireType.LengthDelimited).string(message.squareLogoImage);
        /* optional string marketing_image = 26; */
        if (message.marketingImage !== undefined)
            writer.tag(26, runtime_1.WireType.LengthDelimited).string(message.marketingImage);
        /* optional string square_marketing_image = 27; */
        if (message.squareMarketingImage !== undefined)
            writer.tag(27, runtime_1.WireType.LengthDelimited).string(message.squareMarketingImage);
        /* google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum.DisplayAdFormatSetting format_setting = 13; */
        if (message.formatSetting !== 0)
            writer.tag(13, runtime_1.WireType.Varint).int32(message.formatSetting);
        /* optional string price_prefix = 28; */
        if (message.pricePrefix !== undefined)
            writer.tag(28, runtime_1.WireType.LengthDelimited).string(message.pricePrefix);
        /* optional string promo_text = 29; */
        if (message.promoText !== undefined)
            writer.tag(29, runtime_1.WireType.LengthDelimited).string(message.promoText);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LegacyResponsiveDisplayAdInfo
 */
exports.LegacyResponsiveDisplayAdInfo = new LegacyResponsiveDisplayAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AppAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AppAdInfo", [
            { no: 1, name: "mandatory_ad_text", kind: "message", T: () => ad_asset_3.AdTextAsset },
            { no: 2, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 3, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 4, name: "images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 5, name: "youtube_videos", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_4.AdVideoAsset },
            { no: 6, name: "html5_media_bundles", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_2.AdMediaBundleAsset }
        ]);
    }
    create(value) {
        const message = { headlines: [], descriptions: [], images: [], youtubeVideos: [], html5MediaBundles: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.AdTextAsset mandatory_ad_text */ 1:
                    message.mandatoryAdText = ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options, message.mandatoryAdText);
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 2:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 3:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset images */ 4:
                    message.images.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdVideoAsset youtube_videos */ 5:
                    message.youtubeVideos.push(ad_asset_4.AdVideoAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdMediaBundleAsset html5_media_bundles */ 6:
                    message.html5MediaBundles.push(ad_asset_2.AdMediaBundleAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.AdTextAsset mandatory_ad_text = 1; */
        if (message.mandatoryAdText)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.mandatoryAdText, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 2; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 3; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset images = 4; */
        for (let i = 0; i < message.images.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.images[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdVideoAsset youtube_videos = 5; */
        for (let i = 0; i < message.youtubeVideos.length; i++)
            ad_asset_4.AdVideoAsset.internalBinaryWrite(message.youtubeVideos[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdMediaBundleAsset html5_media_bundles = 6; */
        for (let i = 0; i < message.html5MediaBundles.length; i++)
            ad_asset_2.AdMediaBundleAsset.internalBinaryWrite(message.html5MediaBundles[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AppAdInfo
 */
exports.AppAdInfo = new AppAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AppEngagementAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AppEngagementAdInfo", [
            { no: 1, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 2, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 3, name: "images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 4, name: "videos", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_4.AdVideoAsset }
        ]);
    }
    create(value) {
        const message = { headlines: [], descriptions: [], images: [], videos: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 1:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 2:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset images */ 3:
                    message.images.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdVideoAsset videos */ 4:
                    message.videos.push(ad_asset_4.AdVideoAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 1; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 2; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset images = 3; */
        for (let i = 0; i < message.images.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.images[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdVideoAsset videos = 4; */
        for (let i = 0; i < message.videos.length; i++)
            ad_asset_4.AdVideoAsset.internalBinaryWrite(message.videos[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AppEngagementAdInfo
 */
exports.AppEngagementAdInfo = new AppEngagementAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AppPreRegistrationAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AppPreRegistrationAdInfo", [
            { no: 1, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 2, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 3, name: "images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 4, name: "youtube_videos", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_4.AdVideoAsset }
        ]);
    }
    create(value) {
        const message = { headlines: [], descriptions: [], images: [], youtubeVideos: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 1:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 2:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset images */ 3:
                    message.images.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdVideoAsset youtube_videos */ 4:
                    message.youtubeVideos.push(ad_asset_4.AdVideoAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 1; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 2; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset images = 3; */
        for (let i = 0; i < message.images.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.images[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdVideoAsset youtube_videos = 4; */
        for (let i = 0; i < message.youtubeVideos.length; i++)
            ad_asset_4.AdVideoAsset.internalBinaryWrite(message.youtubeVideos[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AppPreRegistrationAdInfo
 */
exports.AppPreRegistrationAdInfo = new AppPreRegistrationAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LegacyAppInstallAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LegacyAppInstallAdInfo", [
            { no: 6, name: "app_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "app_store", kind: "enum", T: () => ["google.ads.googleads.v11.enums.LegacyAppInstallAdAppStoreEnum.LegacyAppInstallAdAppStore", legacy_app_install_ad_app_store_1.LegacyAppInstallAdAppStoreEnum_LegacyAppInstallAdAppStore] },
            { no: 7, name: "headline", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "description1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "description2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { appStore: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string app_id */ 6:
                    message.appId = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.LegacyAppInstallAdAppStoreEnum.LegacyAppInstallAdAppStore app_store */ 2:
                    message.appStore = reader.int32();
                    break;
                case /* optional string headline */ 7:
                    message.headline = reader.string();
                    break;
                case /* optional string description1 */ 8:
                    message.description1 = reader.string();
                    break;
                case /* optional string description2 */ 9:
                    message.description2 = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string app_id = 6; */
        if (message.appId !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.appId);
        /* google.ads.googleads.v11.enums.LegacyAppInstallAdAppStoreEnum.LegacyAppInstallAdAppStore app_store = 2; */
        if (message.appStore !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.appStore);
        /* optional string headline = 7; */
        if (message.headline !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.headline);
        /* optional string description1 = 8; */
        if (message.description1 !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.description1);
        /* optional string description2 = 9; */
        if (message.description2 !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.description2);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LegacyAppInstallAdInfo
 */
exports.LegacyAppInstallAdInfo = new LegacyAppInstallAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ResponsiveDisplayAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ResponsiveDisplayAdInfo", [
            { no: 1, name: "marketing_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 2, name: "square_marketing_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 3, name: "logo_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 4, name: "square_logo_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 5, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 6, name: "long_headline", kind: "message", T: () => ad_asset_3.AdTextAsset },
            { no: 7, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 8, name: "youtube_videos", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_4.AdVideoAsset },
            { no: 17, name: "business_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 18, name: "main_color", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 19, name: "accent_color", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 20, name: "allow_flexible_color", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 21, name: "call_to_action_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 22, name: "price_prefix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 23, name: "promo_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "format_setting", kind: "enum", T: () => ["google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum.DisplayAdFormatSetting", display_ad_format_setting_1.DisplayAdFormatSettingEnum_DisplayAdFormatSetting] },
            { no: 24, name: "control_spec", kind: "message", T: () => exports.ResponsiveDisplayAdControlSpec }
        ]);
    }
    create(value) {
        const message = { marketingImages: [], squareMarketingImages: [], logoImages: [], squareLogoImages: [], headlines: [], descriptions: [], youtubeVideos: [], formatSetting: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AdImageAsset marketing_images */ 1:
                    message.marketingImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset square_marketing_images */ 2:
                    message.squareMarketingImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset logo_images */ 3:
                    message.logoImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset square_logo_images */ 4:
                    message.squareLogoImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 5:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.common.AdTextAsset long_headline */ 6:
                    message.longHeadline = ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options, message.longHeadline);
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 7:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdVideoAsset youtube_videos */ 8:
                    message.youtubeVideos.push(ad_asset_4.AdVideoAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string business_name */ 17:
                    message.businessName = reader.string();
                    break;
                case /* optional string main_color */ 18:
                    message.mainColor = reader.string();
                    break;
                case /* optional string accent_color */ 19:
                    message.accentColor = reader.string();
                    break;
                case /* optional bool allow_flexible_color */ 20:
                    message.allowFlexibleColor = reader.bool();
                    break;
                case /* optional string call_to_action_text */ 21:
                    message.callToActionText = reader.string();
                    break;
                case /* optional string price_prefix */ 22:
                    message.pricePrefix = reader.string();
                    break;
                case /* optional string promo_text */ 23:
                    message.promoText = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum.DisplayAdFormatSetting format_setting */ 16:
                    message.formatSetting = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.ResponsiveDisplayAdControlSpec control_spec */ 24:
                    message.controlSpec = exports.ResponsiveDisplayAdControlSpec.internalBinaryRead(reader, reader.uint32(), options, message.controlSpec);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AdImageAsset marketing_images = 1; */
        for (let i = 0; i < message.marketingImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.marketingImages[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset square_marketing_images = 2; */
        for (let i = 0; i < message.squareMarketingImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.squareMarketingImages[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset logo_images = 3; */
        for (let i = 0; i < message.logoImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.logoImages[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset square_logo_images = 4; */
        for (let i = 0; i < message.squareLogoImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.squareLogoImages[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 5; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AdTextAsset long_headline = 6; */
        if (message.longHeadline)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.longHeadline, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 7; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdVideoAsset youtube_videos = 8; */
        for (let i = 0; i < message.youtubeVideos.length; i++)
            ad_asset_4.AdVideoAsset.internalBinaryWrite(message.youtubeVideos[i], writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string business_name = 17; */
        if (message.businessName !== undefined)
            writer.tag(17, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* optional string main_color = 18; */
        if (message.mainColor !== undefined)
            writer.tag(18, runtime_1.WireType.LengthDelimited).string(message.mainColor);
        /* optional string accent_color = 19; */
        if (message.accentColor !== undefined)
            writer.tag(19, runtime_1.WireType.LengthDelimited).string(message.accentColor);
        /* optional bool allow_flexible_color = 20; */
        if (message.allowFlexibleColor !== undefined)
            writer.tag(20, runtime_1.WireType.Varint).bool(message.allowFlexibleColor);
        /* optional string call_to_action_text = 21; */
        if (message.callToActionText !== undefined)
            writer.tag(21, runtime_1.WireType.LengthDelimited).string(message.callToActionText);
        /* optional string price_prefix = 22; */
        if (message.pricePrefix !== undefined)
            writer.tag(22, runtime_1.WireType.LengthDelimited).string(message.pricePrefix);
        /* optional string promo_text = 23; */
        if (message.promoText !== undefined)
            writer.tag(23, runtime_1.WireType.LengthDelimited).string(message.promoText);
        /* google.ads.googleads.v11.enums.DisplayAdFormatSettingEnum.DisplayAdFormatSetting format_setting = 16; */
        if (message.formatSetting !== 0)
            writer.tag(16, runtime_1.WireType.Varint).int32(message.formatSetting);
        /* google.ads.googleads.v11.common.ResponsiveDisplayAdControlSpec control_spec = 24; */
        if (message.controlSpec)
            exports.ResponsiveDisplayAdControlSpec.internalBinaryWrite(message.controlSpec, writer.tag(24, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ResponsiveDisplayAdInfo
 */
exports.ResponsiveDisplayAdInfo = new ResponsiveDisplayAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LocalAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LocalAdInfo", [
            { no: 1, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 2, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 3, name: "call_to_actions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 4, name: "marketing_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 5, name: "logo_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 6, name: "videos", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_4.AdVideoAsset },
            { no: 9, name: "path1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "path2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { headlines: [], descriptions: [], callToActions: [], marketingImages: [], logoImages: [], videos: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 1:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 2:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset call_to_actions */ 3:
                    message.callToActions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset marketing_images */ 4:
                    message.marketingImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset logo_images */ 5:
                    message.logoImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdVideoAsset videos */ 6:
                    message.videos.push(ad_asset_4.AdVideoAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string path1 */ 9:
                    message.path1 = reader.string();
                    break;
                case /* optional string path2 */ 10:
                    message.path2 = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 1; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 2; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset call_to_actions = 3; */
        for (let i = 0; i < message.callToActions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.callToActions[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset marketing_images = 4; */
        for (let i = 0; i < message.marketingImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.marketingImages[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset logo_images = 5; */
        for (let i = 0; i < message.logoImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.logoImages[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdVideoAsset videos = 6; */
        for (let i = 0; i < message.videos.length; i++)
            ad_asset_4.AdVideoAsset.internalBinaryWrite(message.videos[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string path1 = 9; */
        if (message.path1 !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.path1);
        /* optional string path2 = 10; */
        if (message.path2 !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.path2);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LocalAdInfo
 */
exports.LocalAdInfo = new LocalAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DisplayUploadAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DisplayUploadAdInfo", [
            { no: 1, name: "display_upload_product_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.DisplayUploadProductTypeEnum.DisplayUploadProductType", display_upload_product_type_1.DisplayUploadProductTypeEnum_DisplayUploadProductType] },
            { no: 2, name: "media_bundle", kind: "message", oneof: "mediaAsset", T: () => ad_asset_2.AdMediaBundleAsset }
        ]);
    }
    create(value) {
        const message = { displayUploadProductType: 0, mediaAsset: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.DisplayUploadProductTypeEnum.DisplayUploadProductType display_upload_product_type */ 1:
                    message.displayUploadProductType = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.AdMediaBundleAsset media_bundle */ 2:
                    message.mediaAsset = {
                        oneofKind: "mediaBundle",
                        mediaBundle: ad_asset_2.AdMediaBundleAsset.internalBinaryRead(reader, reader.uint32(), options, message.mediaAsset.mediaBundle)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.DisplayUploadProductTypeEnum.DisplayUploadProductType display_upload_product_type = 1; */
        if (message.displayUploadProductType !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.displayUploadProductType);
        /* google.ads.googleads.v11.common.AdMediaBundleAsset media_bundle = 2; */
        if (message.mediaAsset.oneofKind === "mediaBundle")
            ad_asset_2.AdMediaBundleAsset.internalBinaryWrite(message.mediaAsset.mediaBundle, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DisplayUploadAdInfo
 */
exports.DisplayUploadAdInfo = new DisplayUploadAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ResponsiveDisplayAdControlSpec$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ResponsiveDisplayAdControlSpec", [
            { no: 1, name: "enable_asset_enhancements", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 2, name: "enable_autogen_video", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { enableAssetEnhancements: false, enableAutogenVideo: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* bool enable_asset_enhancements */ 1:
                    message.enableAssetEnhancements = reader.bool();
                    break;
                case /* bool enable_autogen_video */ 2:
                    message.enableAutogenVideo = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* bool enable_asset_enhancements = 1; */
        if (message.enableAssetEnhancements !== false)
            writer.tag(1, runtime_1.WireType.Varint).bool(message.enableAssetEnhancements);
        /* bool enable_autogen_video = 2; */
        if (message.enableAutogenVideo !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.enableAutogenVideo);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ResponsiveDisplayAdControlSpec
 */
exports.ResponsiveDisplayAdControlSpec = new ResponsiveDisplayAdControlSpec$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SmartCampaignAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.SmartCampaignAdInfo", [
            { no: 1, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 2, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset }
        ]);
    }
    create(value) {
        const message = { headlines: [], descriptions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 1:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 2:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 1; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 2; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.SmartCampaignAdInfo
 */
exports.SmartCampaignAdInfo = new SmartCampaignAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CallAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CallAdInfo", [
            { no: 1, name: "country_code", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "phone_number", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "business_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "headline1", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "headline2", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "description1", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "description2", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "call_tracked", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "disable_call_conversion", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 8, name: "phone_number_verification_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "conversion_action", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "conversion_reporting_state", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState", call_conversion_reporting_state_1.CallConversionReportingStateEnum_CallConversionReportingState] },
            { no: 13, name: "path1", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "path2", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { countryCode: "", phoneNumber: "", businessName: "", headline1: "", headline2: "", description1: "", description2: "", callTracked: false, disableCallConversion: false, phoneNumberVerificationUrl: "", conversionAction: "", conversionReportingState: 0, path1: "", path2: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string country_code */ 1:
                    message.countryCode = reader.string();
                    break;
                case /* string phone_number */ 2:
                    message.phoneNumber = reader.string();
                    break;
                case /* string business_name */ 3:
                    message.businessName = reader.string();
                    break;
                case /* string headline1 */ 11:
                    message.headline1 = reader.string();
                    break;
                case /* string headline2 */ 12:
                    message.headline2 = reader.string();
                    break;
                case /* string description1 */ 4:
                    message.description1 = reader.string();
                    break;
                case /* string description2 */ 5:
                    message.description2 = reader.string();
                    break;
                case /* bool call_tracked */ 6:
                    message.callTracked = reader.bool();
                    break;
                case /* bool disable_call_conversion */ 7:
                    message.disableCallConversion = reader.bool();
                    break;
                case /* string phone_number_verification_url */ 8:
                    message.phoneNumberVerificationUrl = reader.string();
                    break;
                case /* string conversion_action */ 9:
                    message.conversionAction = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState conversion_reporting_state */ 10:
                    message.conversionReportingState = reader.int32();
                    break;
                case /* string path1 */ 13:
                    message.path1 = reader.string();
                    break;
                case /* string path2 */ 14:
                    message.path2 = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string country_code = 1; */
        if (message.countryCode !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* string phone_number = 2; */
        if (message.phoneNumber !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.phoneNumber);
        /* string business_name = 3; */
        if (message.businessName !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* string headline1 = 11; */
        if (message.headline1 !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.headline1);
        /* string headline2 = 12; */
        if (message.headline2 !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.headline2);
        /* string description1 = 4; */
        if (message.description1 !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.description1);
        /* string description2 = 5; */
        if (message.description2 !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.description2);
        /* bool call_tracked = 6; */
        if (message.callTracked !== false)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.callTracked);
        /* bool disable_call_conversion = 7; */
        if (message.disableCallConversion !== false)
            writer.tag(7, runtime_1.WireType.Varint).bool(message.disableCallConversion);
        /* string phone_number_verification_url = 8; */
        if (message.phoneNumberVerificationUrl !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.phoneNumberVerificationUrl);
        /* string conversion_action = 9; */
        if (message.conversionAction !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.conversionAction);
        /* google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState conversion_reporting_state = 10; */
        if (message.conversionReportingState !== 0)
            writer.tag(10, runtime_1.WireType.Varint).int32(message.conversionReportingState);
        /* string path1 = 13; */
        if (message.path1 !== "")
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.path1);
        /* string path2 = 14; */
        if (message.path2 !== "")
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.path2);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CallAdInfo
 */
exports.CallAdInfo = new CallAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DiscoveryMultiAssetAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DiscoveryMultiAssetAdInfo", [
            { no: 1, name: "marketing_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 2, name: "square_marketing_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 3, name: "portrait_marketing_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 4, name: "logo_images", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_5.AdImageAsset },
            { no: 5, name: "headlines", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 6, name: "descriptions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_3.AdTextAsset },
            { no: 7, name: "business_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "call_to_action_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "lead_form_only", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { marketingImages: [], squareMarketingImages: [], portraitMarketingImages: [], logoImages: [], headlines: [], descriptions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AdImageAsset marketing_images */ 1:
                    message.marketingImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset square_marketing_images */ 2:
                    message.squareMarketingImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset portrait_marketing_images */ 3:
                    message.portraitMarketingImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdImageAsset logo_images */ 4:
                    message.logoImages.push(ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset headlines */ 5:
                    message.headlines.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions */ 6:
                    message.descriptions.push(ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string business_name */ 7:
                    message.businessName = reader.string();
                    break;
                case /* optional string call_to_action_text */ 8:
                    message.callToActionText = reader.string();
                    break;
                case /* optional bool lead_form_only */ 9:
                    message.leadFormOnly = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AdImageAsset marketing_images = 1; */
        for (let i = 0; i < message.marketingImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.marketingImages[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset square_marketing_images = 2; */
        for (let i = 0; i < message.squareMarketingImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.squareMarketingImages[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset portrait_marketing_images = 3; */
        for (let i = 0; i < message.portraitMarketingImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.portraitMarketingImages[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdImageAsset logo_images = 4; */
        for (let i = 0; i < message.logoImages.length; i++)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.logoImages[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset headlines = 5; */
        for (let i = 0; i < message.headlines.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headlines[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.AdTextAsset descriptions = 6; */
        for (let i = 0; i < message.descriptions.length; i++)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.descriptions[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string business_name = 7; */
        if (message.businessName !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* optional string call_to_action_text = 8; */
        if (message.callToActionText !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.callToActionText);
        /* optional bool lead_form_only = 9; */
        if (message.leadFormOnly !== undefined)
            writer.tag(9, runtime_1.WireType.Varint).bool(message.leadFormOnly);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DiscoveryMultiAssetAdInfo
 */
exports.DiscoveryMultiAssetAdInfo = new DiscoveryMultiAssetAdInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DiscoveryCarouselAdInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DiscoveryCarouselAdInfo", [
            { no: 1, name: "business_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "logo_image", kind: "message", T: () => ad_asset_5.AdImageAsset, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "headline", kind: "message", T: () => ad_asset_3.AdTextAsset, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "description", kind: "message", T: () => ad_asset_3.AdTextAsset, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 5, name: "call_to_action_text", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "carousel_cards", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ad_asset_1.AdDiscoveryCarouselCardAsset, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { businessName: "", callToActionText: "", carouselCards: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string business_name */ 1:
                    message.businessName = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.AdImageAsset logo_image */ 2:
                    message.logoImage = ad_asset_5.AdImageAsset.internalBinaryRead(reader, reader.uint32(), options, message.logoImage);
                    break;
                case /* google.ads.googleads.v11.common.AdTextAsset headline */ 3:
                    message.headline = ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options, message.headline);
                    break;
                case /* google.ads.googleads.v11.common.AdTextAsset description */ 4:
                    message.description = ad_asset_3.AdTextAsset.internalBinaryRead(reader, reader.uint32(), options, message.description);
                    break;
                case /* string call_to_action_text */ 5:
                    message.callToActionText = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.AdDiscoveryCarouselCardAsset carousel_cards */ 6:
                    message.carouselCards.push(ad_asset_1.AdDiscoveryCarouselCardAsset.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string business_name = 1; */
        if (message.businessName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* google.ads.googleads.v11.common.AdImageAsset logo_image = 2; */
        if (message.logoImage)
            ad_asset_5.AdImageAsset.internalBinaryWrite(message.logoImage, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AdTextAsset headline = 3; */
        if (message.headline)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.headline, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AdTextAsset description = 4; */
        if (message.description)
            ad_asset_3.AdTextAsset.internalBinaryWrite(message.description, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string call_to_action_text = 5; */
        if (message.callToActionText !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.callToActionText);
        /* repeated google.ads.googleads.v11.common.AdDiscoveryCarouselCardAsset carousel_cards = 6; */
        for (let i = 0; i < message.carouselCards.length; i++)
            ad_asset_1.AdDiscoveryCarouselCardAsset.internalBinaryWrite(message.carouselCards[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DiscoveryCarouselAdInfo
 */
exports.DiscoveryCarouselAdInfo = new DiscoveryCarouselAdInfo$Type();
