// google-ads-api-client/dist/protos/bindings/ads/common/asset_types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DynamicJobsAsset = exports.DynamicLocalAsset = exports.DynamicTravelAsset = exports.DiscoveryCarouselCardAsset = exports.DynamicFlightsAsset = exports.DynamicHotelsAndRentalsAsset = exports.DynamicCustomAsset = exports.DynamicRealEstateAsset = exports.CallToActionAsset = exports.PriceOffering = exports.PriceAsset = exports.CallAsset = exports.HotelCalloutAsset = exports.MobileAppAsset = exports.DynamicEducationAsset = exports.PageFeedAsset = exports.SitelinkAsset = exports.StructuredSnippetAsset = exports.CalloutAsset = exports.PromotionAsset = exports.BookOnGoogleAsset = exports.WebhookDelivery = exports.LeadFormDeliveryMethod = exports.LeadFormSingleChoiceAnswers = exports.LeadFormCustomQuestionField = exports.LeadFormField = exports.LeadFormAsset = exports.TextAsset = exports.ImageDimension = exports.ImageAsset = exports.MediaBundleAsset = exports.YoutubeVideoAsset = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const call_to_action_type_1 = require("../enums/call_to_action_type");
const price_extension_price_unit_1 = require("../enums/price_extension_price_unit");
const price_extension_price_qualifier_1 = require("../enums/price_extension_price_qualifier");
const price_extension_type_1 = require("../enums/price_extension_type");
const call_conversion_reporting_state_1 = require("../enums/call_conversion_reporting_state");
const mobile_app_vendor_1 = require("../enums/mobile_app_vendor");
const feed_common_1 = require("./feed_common");
const criteria_1 = require("./criteria");
const promotion_extension_occasion_1 = require("../enums/promotion_extension_occasion");
const promotion_extension_discount_modifier_1 = require("../enums/promotion_extension_discount_modifier");
const lead_form_field_user_input_type_1 = require("../enums/lead_form_field_user_input_type");
const lead_form_desired_intent_1 = require("../enums/lead_form_desired_intent");
const lead_form_post_submit_call_to_action_type_1 = require("../enums/lead_form_post_submit_call_to_action_type");
const lead_form_call_to_action_type_1 = require("../enums/lead_form_call_to_action_type");
const mime_type_1 = require("../enums/mime_type");
// @generated message type with reflection information, may provide speed optimized methods
class YoutubeVideoAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.YoutubeVideoAsset", [
            { no: 2, name: "youtube_video_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "youtube_video_title", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { youtubeVideoTitle: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string youtube_video_id */ 2:
                    message.youtubeVideoId = reader.string();
                    break;
                case /* string youtube_video_title */ 3:
                    message.youtubeVideoTitle = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string youtube_video_id = 2; */
        if (message.youtubeVideoId !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.youtubeVideoId);
        /* string youtube_video_title = 3; */
        if (message.youtubeVideoTitle !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.youtubeVideoTitle);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.YoutubeVideoAsset
 */
exports.YoutubeVideoAsset = new YoutubeVideoAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MediaBundleAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.MediaBundleAsset", [
            { no: 2, name: "data", kind: "scalar", opt: true, T: 12 /*ScalarType.BYTES*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional bytes data */ 2:
                    message.data = reader.bytes();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional bytes data = 2; */
        if (message.data !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).bytes(message.data);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.MediaBundleAsset
 */
exports.MediaBundleAsset = new MediaBundleAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImageAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ImageAsset", [
            { no: 5, name: "data", kind: "scalar", opt: true, T: 12 /*ScalarType.BYTES*/ },
            { no: 6, name: "file_size", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 3, name: "mime_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.MimeTypeEnum.MimeType", mime_type_1.MimeTypeEnum_MimeType] },
            { no: 4, name: "full_size", kind: "message", T: () => exports.ImageDimension }
        ]);
    }
    create(value) {
        const message = { mimeType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional bytes data */ 5:
                    message.data = reader.bytes();
                    break;
                case /* optional int64 file_size */ 6:
                    message.fileSize = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.MimeTypeEnum.MimeType mime_type */ 3:
                    message.mimeType = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.ImageDimension full_size */ 4:
                    message.fullSize = exports.ImageDimension.internalBinaryRead(reader, reader.uint32(), options, message.fullSize);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional bytes data = 5; */
        if (message.data !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).bytes(message.data);
        /* optional int64 file_size = 6; */
        if (message.fileSize !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).int64(message.fileSize);
        /* google.ads.googleads.v11.enums.MimeTypeEnum.MimeType mime_type = 3; */
        if (message.mimeType !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.mimeType);
        /* google.ads.googleads.v11.common.ImageDimension full_size = 4; */
        if (message.fullSize)
            exports.ImageDimension.internalBinaryWrite(message.fullSize, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ImageAsset
 */
exports.ImageAsset = new ImageAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImageDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ImageDimension", [
            { no: 4, name: "height_pixels", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 5, name: "width_pixels", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 6, name: "url", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 height_pixels */ 4:
                    message.heightPixels = reader.int64().toBigInt();
                    break;
                case /* optional int64 width_pixels */ 5:
                    message.widthPixels = reader.int64().toBigInt();
                    break;
                case /* optional string url */ 6:
                    message.url = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 height_pixels = 4; */
        if (message.heightPixels !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).int64(message.heightPixels);
        /* optional int64 width_pixels = 5; */
        if (message.widthPixels !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.widthPixels);
        /* optional string url = 6; */
        if (message.url !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.url);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ImageDimension
 */
exports.ImageDimension = new ImageDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TextAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.TextAsset", [
            { no: 2, name: "text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string text */ 2:
                    message.text = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string text = 2; */
        if (message.text !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.text);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.TextAsset
 */
exports.TextAsset = new TextAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LeadFormAsset", [
            { no: 10, name: "business_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 17, name: "call_to_action_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.LeadFormCallToActionTypeEnum.LeadFormCallToActionType", lead_form_call_to_action_type_1.LeadFormCallToActionTypeEnum_LeadFormCallToActionType], options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 18, name: "call_to_action_description", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 12, name: "headline", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 13, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 14, name: "privacy_policy_url", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 15, name: "post_submit_headline", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "post_submit_description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "fields", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.LeadFormField },
            { no: 23, name: "custom_question_fields", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.LeadFormCustomQuestionField },
            { no: 9, name: "delivery_methods", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.LeadFormDeliveryMethod },
            { no: 19, name: "post_submit_call_to_action_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.LeadFormPostSubmitCallToActionTypeEnum.LeadFormPostSubmitCallToActionType", lead_form_post_submit_call_to_action_type_1.LeadFormPostSubmitCallToActionTypeEnum_LeadFormPostSubmitCallToActionType] },
            { no: 20, name: "background_image_asset", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 21, name: "desired_intent", kind: "enum", T: () => ["google.ads.googleads.v11.enums.LeadFormDesiredIntentEnum.LeadFormDesiredIntent", lead_form_desired_intent_1.LeadFormDesiredIntentEnum_LeadFormDesiredIntent] },
            { no: 22, name: "custom_disclosure", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { businessName: "", callToActionType: 0, callToActionDescription: "", headline: "", description: "", privacyPolicyUrl: "", fields: [], customQuestionFields: [], deliveryMethods: [], postSubmitCallToActionType: 0, desiredIntent: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string business_name */ 10:
                    message.businessName = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.LeadFormCallToActionTypeEnum.LeadFormCallToActionType call_to_action_type */ 17:
                    message.callToActionType = reader.int32();
                    break;
                case /* string call_to_action_description */ 18:
                    message.callToActionDescription = reader.string();
                    break;
                case /* string headline */ 12:
                    message.headline = reader.string();
                    break;
                case /* string description */ 13:
                    message.description = reader.string();
                    break;
                case /* string privacy_policy_url */ 14:
                    message.privacyPolicyUrl = reader.string();
                    break;
                case /* optional string post_submit_headline */ 15:
                    message.postSubmitHeadline = reader.string();
                    break;
                case /* optional string post_submit_description */ 16:
                    message.postSubmitDescription = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.LeadFormField fields */ 8:
                    message.fields.push(exports.LeadFormField.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.LeadFormCustomQuestionField custom_question_fields */ 23:
                    message.customQuestionFields.push(exports.LeadFormCustomQuestionField.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.LeadFormDeliveryMethod delivery_methods */ 9:
                    message.deliveryMethods.push(exports.LeadFormDeliveryMethod.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.LeadFormPostSubmitCallToActionTypeEnum.LeadFormPostSubmitCallToActionType post_submit_call_to_action_type */ 19:
                    message.postSubmitCallToActionType = reader.int32();
                    break;
                case /* optional string background_image_asset */ 20:
                    message.backgroundImageAsset = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.LeadFormDesiredIntentEnum.LeadFormDesiredIntent desired_intent */ 21:
                    message.desiredIntent = reader.int32();
                    break;
                case /* optional string custom_disclosure */ 22:
                    message.customDisclosure = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string business_name = 10; */
        if (message.businessName !== "")
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* google.ads.googleads.v11.enums.LeadFormCallToActionTypeEnum.LeadFormCallToActionType call_to_action_type = 17; */
        if (message.callToActionType !== 0)
            writer.tag(17, runtime_1.WireType.Varint).int32(message.callToActionType);
        /* string call_to_action_description = 18; */
        if (message.callToActionDescription !== "")
            writer.tag(18, runtime_1.WireType.LengthDelimited).string(message.callToActionDescription);
        /* string headline = 12; */
        if (message.headline !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.headline);
        /* string description = 13; */
        if (message.description !== "")
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.description);
        /* string privacy_policy_url = 14; */
        if (message.privacyPolicyUrl !== "")
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.privacyPolicyUrl);
        /* optional string post_submit_headline = 15; */
        if (message.postSubmitHeadline !== undefined)
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.postSubmitHeadline);
        /* optional string post_submit_description = 16; */
        if (message.postSubmitDescription !== undefined)
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.postSubmitDescription);
        /* repeated google.ads.googleads.v11.common.LeadFormField fields = 8; */
        for (let i = 0; i < message.fields.length; i++)
            exports.LeadFormField.internalBinaryWrite(message.fields[i], writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.LeadFormCustomQuestionField custom_question_fields = 23; */
        for (let i = 0; i < message.customQuestionFields.length; i++)
            exports.LeadFormCustomQuestionField.internalBinaryWrite(message.customQuestionFields[i], writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.LeadFormDeliveryMethod delivery_methods = 9; */
        for (let i = 0; i < message.deliveryMethods.length; i++)
            exports.LeadFormDeliveryMethod.internalBinaryWrite(message.deliveryMethods[i], writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.LeadFormPostSubmitCallToActionTypeEnum.LeadFormPostSubmitCallToActionType post_submit_call_to_action_type = 19; */
        if (message.postSubmitCallToActionType !== 0)
            writer.tag(19, runtime_1.WireType.Varint).int32(message.postSubmitCallToActionType);
        /* optional string background_image_asset = 20; */
        if (message.backgroundImageAsset !== undefined)
            writer.tag(20, runtime_1.WireType.LengthDelimited).string(message.backgroundImageAsset);
        /* google.ads.googleads.v11.enums.LeadFormDesiredIntentEnum.LeadFormDesiredIntent desired_intent = 21; */
        if (message.desiredIntent !== 0)
            writer.tag(21, runtime_1.WireType.Varint).int32(message.desiredIntent);
        /* optional string custom_disclosure = 22; */
        if (message.customDisclosure !== undefined)
            writer.tag(22, runtime_1.WireType.LengthDelimited).string(message.customDisclosure);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LeadFormAsset
 */
exports.LeadFormAsset = new LeadFormAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormField$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LeadFormField", [
            { no: 1, name: "input_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum.LeadFormFieldUserInputType", lead_form_field_user_input_type_1.LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType] },
            { no: 2, name: "single_choice_answers", kind: "message", oneof: "answers", T: () => exports.LeadFormSingleChoiceAnswers }
        ]);
    }
    create(value) {
        const message = { inputType: 0, answers: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum.LeadFormFieldUserInputType input_type */ 1:
                    message.inputType = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.LeadFormSingleChoiceAnswers single_choice_answers */ 2:
                    message.answers = {
                        oneofKind: "singleChoiceAnswers",
                        singleChoiceAnswers: exports.LeadFormSingleChoiceAnswers.internalBinaryRead(reader, reader.uint32(), options, message.answers.singleChoiceAnswers)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum.LeadFormFieldUserInputType input_type = 1; */
        if (message.inputType !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.inputType);
        /* google.ads.googleads.v11.common.LeadFormSingleChoiceAnswers single_choice_answers = 2; */
        if (message.answers.oneofKind === "singleChoiceAnswers")
            exports.LeadFormSingleChoiceAnswers.internalBinaryWrite(message.answers.singleChoiceAnswers, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LeadFormField
 */
exports.LeadFormField = new LeadFormField$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormCustomQuestionField$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LeadFormCustomQuestionField", [
            { no: 1, name: "custom_question_text", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "single_choice_answers", kind: "message", oneof: "answers", T: () => exports.LeadFormSingleChoiceAnswers }
        ]);
    }
    create(value) {
        const message = { customQuestionText: "", answers: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string custom_question_text */ 1:
                    message.customQuestionText = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.LeadFormSingleChoiceAnswers single_choice_answers */ 2:
                    message.answers = {
                        oneofKind: "singleChoiceAnswers",
                        singleChoiceAnswers: exports.LeadFormSingleChoiceAnswers.internalBinaryRead(reader, reader.uint32(), options, message.answers.singleChoiceAnswers)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string custom_question_text = 1; */
        if (message.customQuestionText !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customQuestionText);
        /* google.ads.googleads.v11.common.LeadFormSingleChoiceAnswers single_choice_answers = 2; */
        if (message.answers.oneofKind === "singleChoiceAnswers")
            exports.LeadFormSingleChoiceAnswers.internalBinaryWrite(message.answers.singleChoiceAnswers, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LeadFormCustomQuestionField
 */
exports.LeadFormCustomQuestionField = new LeadFormCustomQuestionField$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormSingleChoiceAnswers$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LeadFormSingleChoiceAnswers", [
            { no: 1, name: "answers", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { answers: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated string answers */ 1:
                    message.answers.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated string answers = 1; */
        for (let i = 0; i < message.answers.length; i++)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.answers[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LeadFormSingleChoiceAnswers
 */
exports.LeadFormSingleChoiceAnswers = new LeadFormSingleChoiceAnswers$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormDeliveryMethod$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LeadFormDeliveryMethod", [
            { no: 1, name: "webhook", kind: "message", oneof: "deliveryDetails", T: () => exports.WebhookDelivery }
        ]);
    }
    create(value) {
        const message = { deliveryDetails: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.WebhookDelivery webhook */ 1:
                    message.deliveryDetails = {
                        oneofKind: "webhook",
                        webhook: exports.WebhookDelivery.internalBinaryRead(reader, reader.uint32(), options, message.deliveryDetails.webhook)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.WebhookDelivery webhook = 1; */
        if (message.deliveryDetails.oneofKind === "webhook")
            exports.WebhookDelivery.internalBinaryWrite(message.deliveryDetails.webhook, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LeadFormDeliveryMethod
 */
exports.LeadFormDeliveryMethod = new LeadFormDeliveryMethod$Type();
// @generated message type with reflection information, may provide speed optimized methods
class WebhookDelivery$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.WebhookDelivery", [
            { no: 4, name: "advertiser_webhook_url", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "google_secret", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "payload_schema_version", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string advertiser_webhook_url */ 4:
                    message.advertiserWebhookUrl = reader.string();
                    break;
                case /* optional string google_secret */ 5:
                    message.googleSecret = reader.string();
                    break;
                case /* optional int64 payload_schema_version */ 6:
                    message.payloadSchemaVersion = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string advertiser_webhook_url = 4; */
        if (message.advertiserWebhookUrl !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.advertiserWebhookUrl);
        /* optional string google_secret = 5; */
        if (message.googleSecret !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.googleSecret);
        /* optional int64 payload_schema_version = 6; */
        if (message.payloadSchemaVersion !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).int64(message.payloadSchemaVersion);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.WebhookDelivery
 */
exports.WebhookDelivery = new WebhookDelivery$Type();
// @generated message type with reflection information, may provide speed optimized methods
class BookOnGoogleAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.BookOnGoogleAsset", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.BookOnGoogleAsset
 */
exports.BookOnGoogleAsset = new BookOnGoogleAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PromotionAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.PromotionAsset", [
            { no: 1, name: "promotion_target", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "discount_modifier", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum.PromotionExtensionDiscountModifier", promotion_extension_discount_modifier_1.PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier] },
            { no: 7, name: "redemption_start_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "redemption_end_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "occasion", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum.PromotionExtensionOccasion", promotion_extension_occasion_1.PromotionExtensionOccasionEnum_PromotionExtensionOccasion] },
            { no: 10, name: "language_code", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "start_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "end_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "ad_schedule_targets", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_1.AdScheduleInfo },
            { no: 3, name: "percent_off", kind: "scalar", oneof: "discountType", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 4, name: "money_amount_off", kind: "message", oneof: "discountType", T: () => feed_common_1.Money },
            { no: 5, name: "promotion_code", kind: "scalar", oneof: "promotionTrigger", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "orders_over_amount", kind: "message", oneof: "promotionTrigger", T: () => feed_common_1.Money }
        ]);
    }
    create(value) {
        const message = { promotionTarget: "", discountModifier: 0, redemptionStartDate: "", redemptionEndDate: "", occasion: 0, languageCode: "", startDate: "", endDate: "", adScheduleTargets: [], discountType: { oneofKind: undefined }, promotionTrigger: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string promotion_target */ 1:
                    message.promotionTarget = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum.PromotionExtensionDiscountModifier discount_modifier */ 2:
                    message.discountModifier = reader.int32();
                    break;
                case /* string redemption_start_date */ 7:
                    message.redemptionStartDate = reader.string();
                    break;
                case /* string redemption_end_date */ 8:
                    message.redemptionEndDate = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum.PromotionExtensionOccasion occasion */ 9:
                    message.occasion = reader.int32();
                    break;
                case /* string language_code */ 10:
                    message.languageCode = reader.string();
                    break;
                case /* string start_date */ 11:
                    message.startDate = reader.string();
                    break;
                case /* string end_date */ 12:
                    message.endDate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedule_targets */ 13:
                    message.adScheduleTargets.push(criteria_1.AdScheduleInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* int64 percent_off */ 3:
                    message.discountType = {
                        oneofKind: "percentOff",
                        percentOff: reader.int64().toBigInt()
                    };
                    break;
                case /* google.ads.googleads.v11.common.Money money_amount_off */ 4:
                    message.discountType = {
                        oneofKind: "moneyAmountOff",
                        moneyAmountOff: feed_common_1.Money.internalBinaryRead(reader, reader.uint32(), options, message.discountType.moneyAmountOff)
                    };
                    break;
                case /* string promotion_code */ 5:
                    message.promotionTrigger = {
                        oneofKind: "promotionCode",
                        promotionCode: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.common.Money orders_over_amount */ 6:
                    message.promotionTrigger = {
                        oneofKind: "ordersOverAmount",
                        ordersOverAmount: feed_common_1.Money.internalBinaryRead(reader, reader.uint32(), options, message.promotionTrigger.ordersOverAmount)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string promotion_target = 1; */
        if (message.promotionTarget !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.promotionTarget);
        /* google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum.PromotionExtensionDiscountModifier discount_modifier = 2; */
        if (message.discountModifier !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.discountModifier);
        /* string redemption_start_date = 7; */
        if (message.redemptionStartDate !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.redemptionStartDate);
        /* string redemption_end_date = 8; */
        if (message.redemptionEndDate !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.redemptionEndDate);
        /* google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum.PromotionExtensionOccasion occasion = 9; */
        if (message.occasion !== 0)
            writer.tag(9, runtime_1.WireType.Varint).int32(message.occasion);
        /* string language_code = 10; */
        if (message.languageCode !== "")
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        /* string start_date = 11; */
        if (message.startDate !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* string end_date = 12; */
        if (message.endDate !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.endDate);
        /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedule_targets = 13; */
        for (let i = 0; i < message.adScheduleTargets.length; i++)
            criteria_1.AdScheduleInfo.internalBinaryWrite(message.adScheduleTargets[i], writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* int64 percent_off = 3; */
        if (message.discountType.oneofKind === "percentOff")
            writer.tag(3, runtime_1.WireType.Varint).int64(message.discountType.percentOff);
        /* google.ads.googleads.v11.common.Money money_amount_off = 4; */
        if (message.discountType.oneofKind === "moneyAmountOff")
            feed_common_1.Money.internalBinaryWrite(message.discountType.moneyAmountOff, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string promotion_code = 5; */
        if (message.promotionTrigger.oneofKind === "promotionCode")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.promotionTrigger.promotionCode);
        /* google.ads.googleads.v11.common.Money orders_over_amount = 6; */
        if (message.promotionTrigger.oneofKind === "ordersOverAmount")
            feed_common_1.Money.internalBinaryWrite(message.promotionTrigger.ordersOverAmount, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.PromotionAsset
 */
exports.PromotionAsset = new PromotionAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CalloutAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CalloutAsset", [
            { no: 1, name: "callout_text", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "start_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "end_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "ad_schedule_targets", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_1.AdScheduleInfo }
        ]);
    }
    create(value) {
        const message = { calloutText: "", startDate: "", endDate: "", adScheduleTargets: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string callout_text */ 1:
                    message.calloutText = reader.string();
                    break;
                case /* string start_date */ 2:
                    message.startDate = reader.string();
                    break;
                case /* string end_date */ 3:
                    message.endDate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedule_targets */ 4:
                    message.adScheduleTargets.push(criteria_1.AdScheduleInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string callout_text = 1; */
        if (message.calloutText !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.calloutText);
        /* string start_date = 2; */
        if (message.startDate !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* string end_date = 3; */
        if (message.endDate !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.endDate);
        /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedule_targets = 4; */
        for (let i = 0; i < message.adScheduleTargets.length; i++)
            criteria_1.AdScheduleInfo.internalBinaryWrite(message.adScheduleTargets[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CalloutAsset
 */
exports.CalloutAsset = new CalloutAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StructuredSnippetAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.StructuredSnippetAsset", [
            { no: 1, name: "header", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "values", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { header: "", values: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string header */ 1:
                    message.header = reader.string();
                    break;
                case /* repeated string values */ 2:
                    message.values.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string header = 1; */
        if (message.header !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.header);
        /* repeated string values = 2; */
        for (let i = 0; i < message.values.length; i++)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.values[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.StructuredSnippetAsset
 */
exports.StructuredSnippetAsset = new StructuredSnippetAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SitelinkAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.SitelinkAsset", [
            { no: 1, name: "link_text", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "description1", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "description2", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "start_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "end_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "ad_schedule_targets", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_1.AdScheduleInfo }
        ]);
    }
    create(value) {
        const message = { linkText: "", description1: "", description2: "", startDate: "", endDate: "", adScheduleTargets: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string link_text */ 1:
                    message.linkText = reader.string();
                    break;
                case /* string description1 */ 2:
                    message.description1 = reader.string();
                    break;
                case /* string description2 */ 3:
                    message.description2 = reader.string();
                    break;
                case /* string start_date */ 4:
                    message.startDate = reader.string();
                    break;
                case /* string end_date */ 5:
                    message.endDate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedule_targets */ 6:
                    message.adScheduleTargets.push(criteria_1.AdScheduleInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string link_text = 1; */
        if (message.linkText !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.linkText);
        /* string description1 = 2; */
        if (message.description1 !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.description1);
        /* string description2 = 3; */
        if (message.description2 !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.description2);
        /* string start_date = 4; */
        if (message.startDate !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* string end_date = 5; */
        if (message.endDate !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.endDate);
        /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedule_targets = 6; */
        for (let i = 0; i < message.adScheduleTargets.length; i++)
            criteria_1.AdScheduleInfo.internalBinaryWrite(message.adScheduleTargets[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.SitelinkAsset
 */
exports.SitelinkAsset = new SitelinkAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PageFeedAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.PageFeedAsset", [
            { no: 1, name: "page_url", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "labels", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { pageUrl: "", labels: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string page_url */ 1:
                    message.pageUrl = reader.string();
                    break;
                case /* repeated string labels */ 2:
                    message.labels.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string page_url = 1; */
        if (message.pageUrl !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.pageUrl);
        /* repeated string labels = 2; */
        for (let i = 0; i < message.labels.length; i++)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.labels[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.PageFeedAsset
 */
exports.PageFeedAsset = new PageFeedAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicEducationAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicEducationAsset", [
            { no: 1, name: "program_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "location_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "program_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "subject", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "program_description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "school_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "address", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "contextual_keywords", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "android_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "similar_program_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "ios_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "ios_app_store_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 13, name: "thumbnail_image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { programId: "", locationId: "", programName: "", subject: "", programDescription: "", schoolName: "", address: "", contextualKeywords: [], androidAppLink: "", similarProgramIds: [], iosAppLink: "", iosAppStoreId: 0n, thumbnailImageUrl: "", imageUrl: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string program_id */ 1:
                    message.programId = reader.string();
                    break;
                case /* string location_id */ 2:
                    message.locationId = reader.string();
                    break;
                case /* string program_name */ 3:
                    message.programName = reader.string();
                    break;
                case /* string subject */ 4:
                    message.subject = reader.string();
                    break;
                case /* string program_description */ 5:
                    message.programDescription = reader.string();
                    break;
                case /* string school_name */ 6:
                    message.schoolName = reader.string();
                    break;
                case /* string address */ 7:
                    message.address = reader.string();
                    break;
                case /* repeated string contextual_keywords */ 8:
                    message.contextualKeywords.push(reader.string());
                    break;
                case /* string android_app_link */ 9:
                    message.androidAppLink = reader.string();
                    break;
                case /* repeated string similar_program_ids */ 10:
                    message.similarProgramIds.push(reader.string());
                    break;
                case /* string ios_app_link */ 11:
                    message.iosAppLink = reader.string();
                    break;
                case /* int64 ios_app_store_id */ 12:
                    message.iosAppStoreId = reader.int64().toBigInt();
                    break;
                case /* string thumbnail_image_url */ 13:
                    message.thumbnailImageUrl = reader.string();
                    break;
                case /* string image_url */ 14:
                    message.imageUrl = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string program_id = 1; */
        if (message.programId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.programId);
        /* string location_id = 2; */
        if (message.locationId !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.locationId);
        /* string program_name = 3; */
        if (message.programName !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.programName);
        /* string subject = 4; */
        if (message.subject !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.subject);
        /* string program_description = 5; */
        if (message.programDescription !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.programDescription);
        /* string school_name = 6; */
        if (message.schoolName !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.schoolName);
        /* string address = 7; */
        if (message.address !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.address);
        /* repeated string contextual_keywords = 8; */
        for (let i = 0; i < message.contextualKeywords.length; i++)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.contextualKeywords[i]);
        /* string android_app_link = 9; */
        if (message.androidAppLink !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.androidAppLink);
        /* repeated string similar_program_ids = 10; */
        for (let i = 0; i < message.similarProgramIds.length; i++)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.similarProgramIds[i]);
        /* string ios_app_link = 11; */
        if (message.iosAppLink !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.iosAppLink);
        /* int64 ios_app_store_id = 12; */
        if (message.iosAppStoreId !== 0n)
            writer.tag(12, runtime_1.WireType.Varint).int64(message.iosAppStoreId);
        /* string thumbnail_image_url = 13; */
        if (message.thumbnailImageUrl !== "")
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.thumbnailImageUrl);
        /* string image_url = 14; */
        if (message.imageUrl !== "")
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicEducationAsset
 */
exports.DynamicEducationAsset = new DynamicEducationAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MobileAppAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.MobileAppAsset", [
            { no: 1, name: "app_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "app_store", kind: "enum", T: () => ["google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor", mobile_app_vendor_1.MobileAppVendorEnum_MobileAppVendor], options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "link_text", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "start_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "end_date", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { appId: "", appStore: 0, linkText: "", startDate: "", endDate: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string app_id */ 1:
                    message.appId = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor app_store */ 2:
                    message.appStore = reader.int32();
                    break;
                case /* string link_text */ 3:
                    message.linkText = reader.string();
                    break;
                case /* string start_date */ 4:
                    message.startDate = reader.string();
                    break;
                case /* string end_date */ 5:
                    message.endDate = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string app_id = 1; */
        if (message.appId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.appId);
        /* google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor app_store = 2; */
        if (message.appStore !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.appStore);
        /* string link_text = 3; */
        if (message.linkText !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.linkText);
        /* string start_date = 4; */
        if (message.startDate !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* string end_date = 5; */
        if (message.endDate !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.endDate);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.MobileAppAsset
 */
exports.MobileAppAsset = new MobileAppAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class HotelCalloutAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.HotelCalloutAsset", [
            { no: 1, name: "text", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "language_code", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { text: "", languageCode: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string text */ 1:
                    message.text = reader.string();
                    break;
                case /* string language_code */ 2:
                    message.languageCode = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string text = 1; */
        if (message.text !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.text);
        /* string language_code = 2; */
        if (message.languageCode !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.HotelCalloutAsset
 */
exports.HotelCalloutAsset = new HotelCalloutAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CallAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CallAsset", [
            { no: 1, name: "country_code", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "phone_number", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "call_conversion_reporting_state", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState", call_conversion_reporting_state_1.CallConversionReportingStateEnum_CallConversionReportingState] },
            { no: 4, name: "call_conversion_action", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/ConversionAction" } } },
            { no: 5, name: "ad_schedule_targets", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_1.AdScheduleInfo }
        ]);
    }
    create(value) {
        const message = { countryCode: "", phoneNumber: "", callConversionReportingState: 0, callConversionAction: "", adScheduleTargets: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string country_code */ 1:
                    message.countryCode = reader.string();
                    break;
                case /* string phone_number */ 2:
                    message.phoneNumber = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState call_conversion_reporting_state */ 3:
                    message.callConversionReportingState = reader.int32();
                    break;
                case /* string call_conversion_action */ 4:
                    message.callConversionAction = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedule_targets */ 5:
                    message.adScheduleTargets.push(criteria_1.AdScheduleInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string country_code = 1; */
        if (message.countryCode !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* string phone_number = 2; */
        if (message.phoneNumber !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.phoneNumber);
        /* google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState call_conversion_reporting_state = 3; */
        if (message.callConversionReportingState !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.callConversionReportingState);
        /* string call_conversion_action = 4; */
        if (message.callConversionAction !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.callConversionAction);
        /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedule_targets = 5; */
        for (let i = 0; i < message.adScheduleTargets.length; i++)
            criteria_1.AdScheduleInfo.internalBinaryWrite(message.adScheduleTargets[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CallAsset
 */
exports.CallAsset = new CallAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PriceAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.PriceAsset", [
            { no: 1, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PriceExtensionTypeEnum.PriceExtensionType", price_extension_type_1.PriceExtensionTypeEnum_PriceExtensionType], options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "price_qualifier", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum.PriceExtensionPriceQualifier", price_extension_price_qualifier_1.PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier] },
            { no: 3, name: "language_code", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "price_offerings", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.PriceOffering }
        ]);
    }
    create(value) {
        const message = { type: 0, priceQualifier: 0, languageCode: "", priceOfferings: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.PriceExtensionTypeEnum.PriceExtensionType type */ 1:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum.PriceExtensionPriceQualifier price_qualifier */ 2:
                    message.priceQualifier = reader.int32();
                    break;
                case /* string language_code */ 3:
                    message.languageCode = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.PriceOffering price_offerings */ 4:
                    message.priceOfferings.push(exports.PriceOffering.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.PriceExtensionTypeEnum.PriceExtensionType type = 1; */
        if (message.type !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum.PriceExtensionPriceQualifier price_qualifier = 2; */
        if (message.priceQualifier !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.priceQualifier);
        /* string language_code = 3; */
        if (message.languageCode !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        /* repeated google.ads.googleads.v11.common.PriceOffering price_offerings = 4; */
        for (let i = 0; i < message.priceOfferings.length; i++)
            exports.PriceOffering.internalBinaryWrite(message.priceOfferings[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.PriceAsset
 */
exports.PriceAsset = new PriceAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PriceOffering$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.PriceOffering", [
            { no: 1, name: "header", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "price", kind: "message", T: () => feed_common_1.Money, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "unit", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum.PriceExtensionPriceUnit", price_extension_price_unit_1.PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit] },
            { no: 5, name: "final_url", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 6, name: "final_mobile_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { header: "", description: "", unit: 0, finalUrl: "", finalMobileUrl: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string header */ 1:
                    message.header = reader.string();
                    break;
                case /* string description */ 2:
                    message.description = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.Money price */ 3:
                    message.price = feed_common_1.Money.internalBinaryRead(reader, reader.uint32(), options, message.price);
                    break;
                case /* google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum.PriceExtensionPriceUnit unit */ 4:
                    message.unit = reader.int32();
                    break;
                case /* string final_url */ 5:
                    message.finalUrl = reader.string();
                    break;
                case /* string final_mobile_url */ 6:
                    message.finalMobileUrl = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string header = 1; */
        if (message.header !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.header);
        /* string description = 2; */
        if (message.description !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.description);
        /* google.ads.googleads.v11.common.Money price = 3; */
        if (message.price)
            feed_common_1.Money.internalBinaryWrite(message.price, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum.PriceExtensionPriceUnit unit = 4; */
        if (message.unit !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.unit);
        /* string final_url = 5; */
        if (message.finalUrl !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.finalUrl);
        /* string final_mobile_url = 6; */
        if (message.finalMobileUrl !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.finalMobileUrl);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.PriceOffering
 */
exports.PriceOffering = new PriceOffering$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CallToActionAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CallToActionAsset", [
            { no: 1, name: "call_to_action", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CallToActionTypeEnum.CallToActionType", call_to_action_type_1.CallToActionTypeEnum_CallToActionType] }
        ]);
    }
    create(value) {
        const message = { callToAction: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.CallToActionTypeEnum.CallToActionType call_to_action */ 1:
                    message.callToAction = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.CallToActionTypeEnum.CallToActionType call_to_action = 1; */
        if (message.callToAction !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.callToAction);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CallToActionAsset
 */
exports.CallToActionAsset = new CallToActionAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicRealEstateAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicRealEstateAsset", [
            { no: 1, name: "listing_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "listing_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "city_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "address", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "property_type", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "listing_type", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "contextual_keywords", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "formatted_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "android_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "ios_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "ios_app_store_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 15, name: "similar_listing_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { listingId: "", listingName: "", cityName: "", description: "", address: "", price: "", imageUrl: "", propertyType: "", listingType: "", contextualKeywords: [], formattedPrice: "", androidAppLink: "", iosAppLink: "", iosAppStoreId: 0n, similarListingIds: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string listing_id */ 1:
                    message.listingId = reader.string();
                    break;
                case /* string listing_name */ 2:
                    message.listingName = reader.string();
                    break;
                case /* string city_name */ 3:
                    message.cityName = reader.string();
                    break;
                case /* string description */ 4:
                    message.description = reader.string();
                    break;
                case /* string address */ 5:
                    message.address = reader.string();
                    break;
                case /* string price */ 6:
                    message.price = reader.string();
                    break;
                case /* string image_url */ 7:
                    message.imageUrl = reader.string();
                    break;
                case /* string property_type */ 8:
                    message.propertyType = reader.string();
                    break;
                case /* string listing_type */ 9:
                    message.listingType = reader.string();
                    break;
                case /* repeated string contextual_keywords */ 10:
                    message.contextualKeywords.push(reader.string());
                    break;
                case /* string formatted_price */ 11:
                    message.formattedPrice = reader.string();
                    break;
                case /* string android_app_link */ 12:
                    message.androidAppLink = reader.string();
                    break;
                case /* string ios_app_link */ 13:
                    message.iosAppLink = reader.string();
                    break;
                case /* int64 ios_app_store_id */ 14:
                    message.iosAppStoreId = reader.int64().toBigInt();
                    break;
                case /* repeated string similar_listing_ids */ 15:
                    message.similarListingIds.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string listing_id = 1; */
        if (message.listingId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.listingId);
        /* string listing_name = 2; */
        if (message.listingName !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.listingName);
        /* string city_name = 3; */
        if (message.cityName !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.cityName);
        /* string description = 4; */
        if (message.description !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.description);
        /* string address = 5; */
        if (message.address !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.address);
        /* string price = 6; */
        if (message.price !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.price);
        /* string image_url = 7; */
        if (message.imageUrl !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        /* string property_type = 8; */
        if (message.propertyType !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.propertyType);
        /* string listing_type = 9; */
        if (message.listingType !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.listingType);
        /* repeated string contextual_keywords = 10; */
        for (let i = 0; i < message.contextualKeywords.length; i++)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.contextualKeywords[i]);
        /* string formatted_price = 11; */
        if (message.formattedPrice !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.formattedPrice);
        /* string android_app_link = 12; */
        if (message.androidAppLink !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.androidAppLink);
        /* string ios_app_link = 13; */
        if (message.iosAppLink !== "")
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.iosAppLink);
        /* int64 ios_app_store_id = 14; */
        if (message.iosAppStoreId !== 0n)
            writer.tag(14, runtime_1.WireType.Varint).int64(message.iosAppStoreId);
        /* repeated string similar_listing_ids = 15; */
        for (let i = 0; i < message.similarListingIds.length; i++)
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.similarListingIds[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicRealEstateAsset
 */
exports.DynamicRealEstateAsset = new DynamicRealEstateAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicCustomAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicCustomAsset", [
            { no: 1, name: "id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "id2", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "item_title", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "item_subtitle", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "item_description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "item_address", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "item_category", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "formatted_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "formatted_sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "contextual_keywords", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "android_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "ios_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 17, name: "ios_app_store_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 15, name: "similar_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { id: "", id2: "", itemTitle: "", itemSubtitle: "", itemDescription: "", itemAddress: "", itemCategory: "", price: "", salePrice: "", formattedPrice: "", formattedSalePrice: "", imageUrl: "", contextualKeywords: [], androidAppLink: "", iosAppLink: "", iosAppStoreId: 0n, similarIds: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string id */ 1:
                    message.id = reader.string();
                    break;
                case /* string id2 */ 2:
                    message.id2 = reader.string();
                    break;
                case /* string item_title */ 3:
                    message.itemTitle = reader.string();
                    break;
                case /* string item_subtitle */ 4:
                    message.itemSubtitle = reader.string();
                    break;
                case /* string item_description */ 5:
                    message.itemDescription = reader.string();
                    break;
                case /* string item_address */ 6:
                    message.itemAddress = reader.string();
                    break;
                case /* string item_category */ 7:
                    message.itemCategory = reader.string();
                    break;
                case /* string price */ 8:
                    message.price = reader.string();
                    break;
                case /* string sale_price */ 9:
                    message.salePrice = reader.string();
                    break;
                case /* string formatted_price */ 10:
                    message.formattedPrice = reader.string();
                    break;
                case /* string formatted_sale_price */ 11:
                    message.formattedSalePrice = reader.string();
                    break;
                case /* string image_url */ 12:
                    message.imageUrl = reader.string();
                    break;
                case /* repeated string contextual_keywords */ 13:
                    message.contextualKeywords.push(reader.string());
                    break;
                case /* string android_app_link */ 14:
                    message.androidAppLink = reader.string();
                    break;
                case /* string ios_app_link */ 16:
                    message.iosAppLink = reader.string();
                    break;
                case /* int64 ios_app_store_id */ 17:
                    message.iosAppStoreId = reader.int64().toBigInt();
                    break;
                case /* repeated string similar_ids */ 15:
                    message.similarIds.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string id = 1; */
        if (message.id !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.id);
        /* string id2 = 2; */
        if (message.id2 !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.id2);
        /* string item_title = 3; */
        if (message.itemTitle !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.itemTitle);
        /* string item_subtitle = 4; */
        if (message.itemSubtitle !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.itemSubtitle);
        /* string item_description = 5; */
        if (message.itemDescription !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.itemDescription);
        /* string item_address = 6; */
        if (message.itemAddress !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.itemAddress);
        /* string item_category = 7; */
        if (message.itemCategory !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.itemCategory);
        /* string price = 8; */
        if (message.price !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.price);
        /* string sale_price = 9; */
        if (message.salePrice !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.salePrice);
        /* string formatted_price = 10; */
        if (message.formattedPrice !== "")
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.formattedPrice);
        /* string formatted_sale_price = 11; */
        if (message.formattedSalePrice !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.formattedSalePrice);
        /* string image_url = 12; */
        if (message.imageUrl !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        /* repeated string contextual_keywords = 13; */
        for (let i = 0; i < message.contextualKeywords.length; i++)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.contextualKeywords[i]);
        /* string android_app_link = 14; */
        if (message.androidAppLink !== "")
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.androidAppLink);
        /* string ios_app_link = 16; */
        if (message.iosAppLink !== "")
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.iosAppLink);
        /* int64 ios_app_store_id = 17; */
        if (message.iosAppStoreId !== 0n)
            writer.tag(17, runtime_1.WireType.Varint).int64(message.iosAppStoreId);
        /* repeated string similar_ids = 15; */
        for (let i = 0; i < message.similarIds.length; i++)
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.similarIds[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicCustomAsset
 */
exports.DynamicCustomAsset = new DynamicCustomAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicHotelsAndRentalsAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicHotelsAndRentalsAsset", [
            { no: 1, name: "property_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "property_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "destination_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "star_rating", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 9, name: "category", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "contextual_keywords", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "address", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "android_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "ios_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "ios_app_store_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 15, name: "formatted_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "formatted_sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 17, name: "similar_property_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { propertyId: "", propertyName: "", imageUrl: "", destinationName: "", description: "", price: "", salePrice: "", starRating: 0n, category: "", contextualKeywords: [], address: "", androidAppLink: "", iosAppLink: "", iosAppStoreId: 0n, formattedPrice: "", formattedSalePrice: "", similarPropertyIds: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string property_id */ 1:
                    message.propertyId = reader.string();
                    break;
                case /* string property_name */ 2:
                    message.propertyName = reader.string();
                    break;
                case /* string image_url */ 3:
                    message.imageUrl = reader.string();
                    break;
                case /* string destination_name */ 4:
                    message.destinationName = reader.string();
                    break;
                case /* string description */ 5:
                    message.description = reader.string();
                    break;
                case /* string price */ 6:
                    message.price = reader.string();
                    break;
                case /* string sale_price */ 7:
                    message.salePrice = reader.string();
                    break;
                case /* int64 star_rating */ 8:
                    message.starRating = reader.int64().toBigInt();
                    break;
                case /* string category */ 9:
                    message.category = reader.string();
                    break;
                case /* repeated string contextual_keywords */ 10:
                    message.contextualKeywords.push(reader.string());
                    break;
                case /* string address */ 11:
                    message.address = reader.string();
                    break;
                case /* string android_app_link */ 12:
                    message.androidAppLink = reader.string();
                    break;
                case /* string ios_app_link */ 13:
                    message.iosAppLink = reader.string();
                    break;
                case /* int64 ios_app_store_id */ 14:
                    message.iosAppStoreId = reader.int64().toBigInt();
                    break;
                case /* string formatted_price */ 15:
                    message.formattedPrice = reader.string();
                    break;
                case /* string formatted_sale_price */ 16:
                    message.formattedSalePrice = reader.string();
                    break;
                case /* repeated string similar_property_ids */ 17:
                    message.similarPropertyIds.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string property_id = 1; */
        if (message.propertyId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.propertyId);
        /* string property_name = 2; */
        if (message.propertyName !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.propertyName);
        /* string image_url = 3; */
        if (message.imageUrl !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        /* string destination_name = 4; */
        if (message.destinationName !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.destinationName);
        /* string description = 5; */
        if (message.description !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.description);
        /* string price = 6; */
        if (message.price !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.price);
        /* string sale_price = 7; */
        if (message.salePrice !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.salePrice);
        /* int64 star_rating = 8; */
        if (message.starRating !== 0n)
            writer.tag(8, runtime_1.WireType.Varint).int64(message.starRating);
        /* string category = 9; */
        if (message.category !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.category);
        /* repeated string contextual_keywords = 10; */
        for (let i = 0; i < message.contextualKeywords.length; i++)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.contextualKeywords[i]);
        /* string address = 11; */
        if (message.address !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.address);
        /* string android_app_link = 12; */
        if (message.androidAppLink !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.androidAppLink);
        /* string ios_app_link = 13; */
        if (message.iosAppLink !== "")
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.iosAppLink);
        /* int64 ios_app_store_id = 14; */
        if (message.iosAppStoreId !== 0n)
            writer.tag(14, runtime_1.WireType.Varint).int64(message.iosAppStoreId);
        /* string formatted_price = 15; */
        if (message.formattedPrice !== "")
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.formattedPrice);
        /* string formatted_sale_price = 16; */
        if (message.formattedSalePrice !== "")
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.formattedSalePrice);
        /* repeated string similar_property_ids = 17; */
        for (let i = 0; i < message.similarPropertyIds.length; i++)
            writer.tag(17, runtime_1.WireType.LengthDelimited).string(message.similarPropertyIds[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicHotelsAndRentalsAsset
 */
exports.DynamicHotelsAndRentalsAsset = new DynamicHotelsAndRentalsAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicFlightsAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicFlightsAsset", [
            { no: 1, name: "destination_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "origin_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "flight_description", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "destination_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "origin_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "flight_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "flight_sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "formatted_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "formatted_sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "android_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "ios_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "ios_app_store_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 14, name: "similar_destination_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 15, name: "custom_mapping", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { destinationId: "", originId: "", flightDescription: "", imageUrl: "", destinationName: "", originName: "", flightPrice: "", flightSalePrice: "", formattedPrice: "", formattedSalePrice: "", androidAppLink: "", iosAppLink: "", iosAppStoreId: 0n, similarDestinationIds: [], customMapping: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string destination_id */ 1:
                    message.destinationId = reader.string();
                    break;
                case /* string origin_id */ 2:
                    message.originId = reader.string();
                    break;
                case /* string flight_description */ 3:
                    message.flightDescription = reader.string();
                    break;
                case /* string image_url */ 4:
                    message.imageUrl = reader.string();
                    break;
                case /* string destination_name */ 5:
                    message.destinationName = reader.string();
                    break;
                case /* string origin_name */ 6:
                    message.originName = reader.string();
                    break;
                case /* string flight_price */ 7:
                    message.flightPrice = reader.string();
                    break;
                case /* string flight_sale_price */ 8:
                    message.flightSalePrice = reader.string();
                    break;
                case /* string formatted_price */ 9:
                    message.formattedPrice = reader.string();
                    break;
                case /* string formatted_sale_price */ 10:
                    message.formattedSalePrice = reader.string();
                    break;
                case /* string android_app_link */ 11:
                    message.androidAppLink = reader.string();
                    break;
                case /* string ios_app_link */ 12:
                    message.iosAppLink = reader.string();
                    break;
                case /* int64 ios_app_store_id */ 13:
                    message.iosAppStoreId = reader.int64().toBigInt();
                    break;
                case /* repeated string similar_destination_ids */ 14:
                    message.similarDestinationIds.push(reader.string());
                    break;
                case /* string custom_mapping */ 15:
                    message.customMapping = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string destination_id = 1; */
        if (message.destinationId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.destinationId);
        /* string origin_id = 2; */
        if (message.originId !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.originId);
        /* string flight_description = 3; */
        if (message.flightDescription !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.flightDescription);
        /* string image_url = 4; */
        if (message.imageUrl !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        /* string destination_name = 5; */
        if (message.destinationName !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.destinationName);
        /* string origin_name = 6; */
        if (message.originName !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.originName);
        /* string flight_price = 7; */
        if (message.flightPrice !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.flightPrice);
        /* string flight_sale_price = 8; */
        if (message.flightSalePrice !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.flightSalePrice);
        /* string formatted_price = 9; */
        if (message.formattedPrice !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.formattedPrice);
        /* string formatted_sale_price = 10; */
        if (message.formattedSalePrice !== "")
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.formattedSalePrice);
        /* string android_app_link = 11; */
        if (message.androidAppLink !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.androidAppLink);
        /* string ios_app_link = 12; */
        if (message.iosAppLink !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.iosAppLink);
        /* int64 ios_app_store_id = 13; */
        if (message.iosAppStoreId !== 0n)
            writer.tag(13, runtime_1.WireType.Varint).int64(message.iosAppStoreId);
        /* repeated string similar_destination_ids = 14; */
        for (let i = 0; i < message.similarDestinationIds.length; i++)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.similarDestinationIds[i]);
        /* string custom_mapping = 15; */
        if (message.customMapping !== "")
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.customMapping);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicFlightsAsset
 */
exports.DynamicFlightsAsset = new DynamicFlightsAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DiscoveryCarouselCardAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DiscoveryCarouselCardAsset", [
            { no: 1, name: "marketing_image_asset", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "square_marketing_image_asset", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "portrait_marketing_image_asset", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "headline", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 5, name: "call_to_action_text", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { marketingImageAsset: "", squareMarketingImageAsset: "", portraitMarketingImageAsset: "", headline: "", callToActionText: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string marketing_image_asset */ 1:
                    message.marketingImageAsset = reader.string();
                    break;
                case /* string square_marketing_image_asset */ 2:
                    message.squareMarketingImageAsset = reader.string();
                    break;
                case /* string portrait_marketing_image_asset */ 3:
                    message.portraitMarketingImageAsset = reader.string();
                    break;
                case /* string headline */ 4:
                    message.headline = reader.string();
                    break;
                case /* string call_to_action_text */ 5:
                    message.callToActionText = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string marketing_image_asset = 1; */
        if (message.marketingImageAsset !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.marketingImageAsset);
        /* string square_marketing_image_asset = 2; */
        if (message.squareMarketingImageAsset !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.squareMarketingImageAsset);
        /* string portrait_marketing_image_asset = 3; */
        if (message.portraitMarketingImageAsset !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.portraitMarketingImageAsset);
        /* string headline = 4; */
        if (message.headline !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.headline);
        /* string call_to_action_text = 5; */
        if (message.callToActionText !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.callToActionText);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DiscoveryCarouselCardAsset
 */
exports.DiscoveryCarouselCardAsset = new DiscoveryCarouselCardAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicTravelAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicTravelAsset", [
            { no: 1, name: "destination_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "origin_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "title", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "destination_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "destination_address", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "origin_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "formatted_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "formatted_sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "category", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "contextual_keywords", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "similar_destination_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 15, name: "android_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "ios_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 17, name: "ios_app_store_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { destinationId: "", originId: "", title: "", destinationName: "", destinationAddress: "", originName: "", price: "", salePrice: "", formattedPrice: "", formattedSalePrice: "", category: "", contextualKeywords: [], similarDestinationIds: [], imageUrl: "", androidAppLink: "", iosAppLink: "", iosAppStoreId: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string destination_id */ 1:
                    message.destinationId = reader.string();
                    break;
                case /* string origin_id */ 2:
                    message.originId = reader.string();
                    break;
                case /* string title */ 3:
                    message.title = reader.string();
                    break;
                case /* string destination_name */ 4:
                    message.destinationName = reader.string();
                    break;
                case /* string destination_address */ 5:
                    message.destinationAddress = reader.string();
                    break;
                case /* string origin_name */ 6:
                    message.originName = reader.string();
                    break;
                case /* string price */ 7:
                    message.price = reader.string();
                    break;
                case /* string sale_price */ 8:
                    message.salePrice = reader.string();
                    break;
                case /* string formatted_price */ 9:
                    message.formattedPrice = reader.string();
                    break;
                case /* string formatted_sale_price */ 10:
                    message.formattedSalePrice = reader.string();
                    break;
                case /* string category */ 11:
                    message.category = reader.string();
                    break;
                case /* repeated string contextual_keywords */ 12:
                    message.contextualKeywords.push(reader.string());
                    break;
                case /* repeated string similar_destination_ids */ 13:
                    message.similarDestinationIds.push(reader.string());
                    break;
                case /* string image_url */ 14:
                    message.imageUrl = reader.string();
                    break;
                case /* string android_app_link */ 15:
                    message.androidAppLink = reader.string();
                    break;
                case /* string ios_app_link */ 16:
                    message.iosAppLink = reader.string();
                    break;
                case /* int64 ios_app_store_id */ 17:
                    message.iosAppStoreId = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string destination_id = 1; */
        if (message.destinationId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.destinationId);
        /* string origin_id = 2; */
        if (message.originId !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.originId);
        /* string title = 3; */
        if (message.title !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.title);
        /* string destination_name = 4; */
        if (message.destinationName !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.destinationName);
        /* string destination_address = 5; */
        if (message.destinationAddress !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.destinationAddress);
        /* string origin_name = 6; */
        if (message.originName !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.originName);
        /* string price = 7; */
        if (message.price !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.price);
        /* string sale_price = 8; */
        if (message.salePrice !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.salePrice);
        /* string formatted_price = 9; */
        if (message.formattedPrice !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.formattedPrice);
        /* string formatted_sale_price = 10; */
        if (message.formattedSalePrice !== "")
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.formattedSalePrice);
        /* string category = 11; */
        if (message.category !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.category);
        /* repeated string contextual_keywords = 12; */
        for (let i = 0; i < message.contextualKeywords.length; i++)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.contextualKeywords[i]);
        /* repeated string similar_destination_ids = 13; */
        for (let i = 0; i < message.similarDestinationIds.length; i++)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.similarDestinationIds[i]);
        /* string image_url = 14; */
        if (message.imageUrl !== "")
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        /* string android_app_link = 15; */
        if (message.androidAppLink !== "")
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.androidAppLink);
        /* string ios_app_link = 16; */
        if (message.iosAppLink !== "")
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.iosAppLink);
        /* int64 ios_app_store_id = 17; */
        if (message.iosAppStoreId !== 0n)
            writer.tag(17, runtime_1.WireType.Varint).int64(message.iosAppStoreId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicTravelAsset
 */
exports.DynamicTravelAsset = new DynamicTravelAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicLocalAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicLocalAsset", [
            { no: 1, name: "deal_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "deal_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "subtitle", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "address", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "category", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "contextual_keywords", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "formatted_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "formatted_sale_price", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "android_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "similar_deal_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 15, name: "ios_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "ios_app_store_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { dealId: "", dealName: "", subtitle: "", description: "", price: "", salePrice: "", imageUrl: "", address: "", category: "", contextualKeywords: [], formattedPrice: "", formattedSalePrice: "", androidAppLink: "", similarDealIds: [], iosAppLink: "", iosAppStoreId: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string deal_id */ 1:
                    message.dealId = reader.string();
                    break;
                case /* string deal_name */ 2:
                    message.dealName = reader.string();
                    break;
                case /* string subtitle */ 3:
                    message.subtitle = reader.string();
                    break;
                case /* string description */ 4:
                    message.description = reader.string();
                    break;
                case /* string price */ 5:
                    message.price = reader.string();
                    break;
                case /* string sale_price */ 6:
                    message.salePrice = reader.string();
                    break;
                case /* string image_url */ 7:
                    message.imageUrl = reader.string();
                    break;
                case /* string address */ 8:
                    message.address = reader.string();
                    break;
                case /* string category */ 9:
                    message.category = reader.string();
                    break;
                case /* repeated string contextual_keywords */ 10:
                    message.contextualKeywords.push(reader.string());
                    break;
                case /* string formatted_price */ 11:
                    message.formattedPrice = reader.string();
                    break;
                case /* string formatted_sale_price */ 12:
                    message.formattedSalePrice = reader.string();
                    break;
                case /* string android_app_link */ 13:
                    message.androidAppLink = reader.string();
                    break;
                case /* repeated string similar_deal_ids */ 14:
                    message.similarDealIds.push(reader.string());
                    break;
                case /* string ios_app_link */ 15:
                    message.iosAppLink = reader.string();
                    break;
                case /* int64 ios_app_store_id */ 16:
                    message.iosAppStoreId = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string deal_id = 1; */
        if (message.dealId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.dealId);
        /* string deal_name = 2; */
        if (message.dealName !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.dealName);
        /* string subtitle = 3; */
        if (message.subtitle !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.subtitle);
        /* string description = 4; */
        if (message.description !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.description);
        /* string price = 5; */
        if (message.price !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.price);
        /* string sale_price = 6; */
        if (message.salePrice !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.salePrice);
        /* string image_url = 7; */
        if (message.imageUrl !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        /* string address = 8; */
        if (message.address !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.address);
        /* string category = 9; */
        if (message.category !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.category);
        /* repeated string contextual_keywords = 10; */
        for (let i = 0; i < message.contextualKeywords.length; i++)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.contextualKeywords[i]);
        /* string formatted_price = 11; */
        if (message.formattedPrice !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.formattedPrice);
        /* string formatted_sale_price = 12; */
        if (message.formattedSalePrice !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.formattedSalePrice);
        /* string android_app_link = 13; */
        if (message.androidAppLink !== "")
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.androidAppLink);
        /* repeated string similar_deal_ids = 14; */
        for (let i = 0; i < message.similarDealIds.length; i++)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.similarDealIds[i]);
        /* string ios_app_link = 15; */
        if (message.iosAppLink !== "")
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.iosAppLink);
        /* int64 ios_app_store_id = 16; */
        if (message.iosAppStoreId !== 0n)
            writer.tag(16, runtime_1.WireType.Varint).int64(message.iosAppStoreId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicLocalAsset
 */
exports.DynamicLocalAsset = new DynamicLocalAsset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicJobsAsset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicJobsAsset", [
            { no: 1, name: "job_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "location_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "job_title", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 4, name: "job_subtitle", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "image_url", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "job_category", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "contextual_keywords", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "address", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "salary", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "android_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "similar_job_ids", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "ios_app_link", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "ios_app_store_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { jobId: "", locationId: "", jobTitle: "", jobSubtitle: "", description: "", imageUrl: "", jobCategory: "", contextualKeywords: [], address: "", salary: "", androidAppLink: "", similarJobIds: [], iosAppLink: "", iosAppStoreId: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string job_id */ 1:
                    message.jobId = reader.string();
                    break;
                case /* string location_id */ 2:
                    message.locationId = reader.string();
                    break;
                case /* string job_title */ 3:
                    message.jobTitle = reader.string();
                    break;
                case /* string job_subtitle */ 4:
                    message.jobSubtitle = reader.string();
                    break;
                case /* string description */ 5:
                    message.description = reader.string();
                    break;
                case /* string image_url */ 6:
                    message.imageUrl = reader.string();
                    break;
                case /* string job_category */ 7:
                    message.jobCategory = reader.string();
                    break;
                case /* repeated string contextual_keywords */ 8:
                    message.contextualKeywords.push(reader.string());
                    break;
                case /* string address */ 9:
                    message.address = reader.string();
                    break;
                case /* string salary */ 10:
                    message.salary = reader.string();
                    break;
                case /* string android_app_link */ 11:
                    message.androidAppLink = reader.string();
                    break;
                case /* repeated string similar_job_ids */ 12:
                    message.similarJobIds.push(reader.string());
                    break;
                case /* string ios_app_link */ 13:
                    message.iosAppLink = reader.string();
                    break;
                case /* int64 ios_app_store_id */ 14:
                    message.iosAppStoreId = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string job_id = 1; */
        if (message.jobId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.jobId);
        /* string location_id = 2; */
        if (message.locationId !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.locationId);
        /* string job_title = 3; */
        if (message.jobTitle !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.jobTitle);
        /* string job_subtitle = 4; */
        if (message.jobSubtitle !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.jobSubtitle);
        /* string description = 5; */
        if (message.description !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.description);
        /* string image_url = 6; */
        if (message.imageUrl !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.imageUrl);
        /* string job_category = 7; */
        if (message.jobCategory !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.jobCategory);
        /* repeated string contextual_keywords = 8; */
        for (let i = 0; i < message.contextualKeywords.length; i++)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.contextualKeywords[i]);
        /* string address = 9; */
        if (message.address !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.address);
        /* string salary = 10; */
        if (message.salary !== "")
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.salary);
        /* string android_app_link = 11; */
        if (message.androidAppLink !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.androidAppLink);
        /* repeated string similar_job_ids = 12; */
        for (let i = 0; i < message.similarJobIds.length; i++)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.similarJobIds[i]);
        /* string ios_app_link = 13; */
        if (message.iosAppLink !== "")
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.iosAppLink);
        /* int64 ios_app_store_id = 14; */
        if (message.iosAppStoreId !== 0n)
            writer.tag(14, runtime_1.WireType.Varint).int64(message.iosAppStoreId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicJobsAsset
 */
exports.DynamicJobsAsset = new DynamicJobsAsset$Type();
