// google-ads-api-client/dist/protos/bindings/ads/common/audiences.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAudienceSegment = exports.DetailedDemographicSegment = exports.LifeEventSegment = exports.UserInterestSegment = exports.UserListSegment = exports.AudienceSegment = exports.AudienceSegmentDimension = exports.ParentalStatusDimension = exports.HouseholdIncomeDimension = exports.GenderDimension = exports.AgeSegment = exports.AgeDimension = exports.ExclusionSegment = exports.AudienceExclusionDimension = exports.AudienceDimension = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const parental_status_type_1 = require("../enums/parental_status_type");
const income_range_type_1 = require("../enums/income_range_type");
const gender_type_1 = require("../enums/gender_type");
// @generated message type with reflection information, may provide speed optimized methods
class AudienceDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AudienceDimension", [
            { no: 1, name: "age", kind: "message", oneof: "dimension", T: () => exports.AgeDimension },
            { no: 2, name: "gender", kind: "message", oneof: "dimension", T: () => exports.GenderDimension },
            { no: 3, name: "household_income", kind: "message", oneof: "dimension", T: () => exports.HouseholdIncomeDimension },
            { no: 4, name: "parental_status", kind: "message", oneof: "dimension", T: () => exports.ParentalStatusDimension },
            { no: 5, name: "audience_segments", kind: "message", oneof: "dimension", T: () => exports.AudienceSegmentDimension }
        ]);
    }
    create(value) {
        const message = { dimension: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.AgeDimension age */ 1:
                    message.dimension = {
                        oneofKind: "age",
                        age: exports.AgeDimension.internalBinaryRead(reader, reader.uint32(), options, message.dimension.age)
                    };
                    break;
                case /* google.ads.googleads.v11.common.GenderDimension gender */ 2:
                    message.dimension = {
                        oneofKind: "gender",
                        gender: exports.GenderDimension.internalBinaryRead(reader, reader.uint32(), options, message.dimension.gender)
                    };
                    break;
                case /* google.ads.googleads.v11.common.HouseholdIncomeDimension household_income */ 3:
                    message.dimension = {
                        oneofKind: "householdIncome",
                        householdIncome: exports.HouseholdIncomeDimension.internalBinaryRead(reader, reader.uint32(), options, message.dimension.householdIncome)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ParentalStatusDimension parental_status */ 4:
                    message.dimension = {
                        oneofKind: "parentalStatus",
                        parentalStatus: exports.ParentalStatusDimension.internalBinaryRead(reader, reader.uint32(), options, message.dimension.parentalStatus)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AudienceSegmentDimension audience_segments */ 5:
                    message.dimension = {
                        oneofKind: "audienceSegments",
                        audienceSegments: exports.AudienceSegmentDimension.internalBinaryRead(reader, reader.uint32(), options, message.dimension.audienceSegments)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.AgeDimension age = 1; */
        if (message.dimension.oneofKind === "age")
            exports.AgeDimension.internalBinaryWrite(message.dimension.age, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.GenderDimension gender = 2; */
        if (message.dimension.oneofKind === "gender")
            exports.GenderDimension.internalBinaryWrite(message.dimension.gender, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HouseholdIncomeDimension household_income = 3; */
        if (message.dimension.oneofKind === "householdIncome")
            exports.HouseholdIncomeDimension.internalBinaryWrite(message.dimension.householdIncome, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ParentalStatusDimension parental_status = 4; */
        if (message.dimension.oneofKind === "parentalStatus")
            exports.ParentalStatusDimension.internalBinaryWrite(message.dimension.parentalStatus, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AudienceSegmentDimension audience_segments = 5; */
        if (message.dimension.oneofKind === "audienceSegments")
            exports.AudienceSegmentDimension.internalBinaryWrite(message.dimension.audienceSegments, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AudienceDimension
 */
exports.AudienceDimension = new AudienceDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AudienceExclusionDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AudienceExclusionDimension", [
            { no: 1, name: "exclusions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.ExclusionSegment }
        ]);
    }
    create(value) {
        const message = { exclusions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.ExclusionSegment exclusions */ 1:
                    message.exclusions.push(exports.ExclusionSegment.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.ExclusionSegment exclusions = 1; */
        for (let i = 0; i < message.exclusions.length; i++)
            exports.ExclusionSegment.internalBinaryWrite(message.exclusions[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AudienceExclusionDimension
 */
exports.AudienceExclusionDimension = new AudienceExclusionDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ExclusionSegment$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ExclusionSegment", [
            { no: 1, name: "user_list", kind: "message", oneof: "segment", T: () => exports.UserListSegment }
        ]);
    }
    create(value) {
        const message = { segment: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.UserListSegment user_list */ 1:
                    message.segment = {
                        oneofKind: "userList",
                        userList: exports.UserListSegment.internalBinaryRead(reader, reader.uint32(), options, message.segment.userList)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.UserListSegment user_list = 1; */
        if (message.segment.oneofKind === "userList")
            exports.UserListSegment.internalBinaryWrite(message.segment.userList, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ExclusionSegment
 */
exports.ExclusionSegment = new ExclusionSegment$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AgeDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AgeDimension", [
            { no: 1, name: "age_ranges", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.AgeSegment },
            { no: 2, name: "include_undetermined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { ageRanges: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AgeSegment age_ranges */ 1:
                    message.ageRanges.push(exports.AgeSegment.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool include_undetermined */ 2:
                    message.includeUndetermined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AgeSegment age_ranges = 1; */
        for (let i = 0; i < message.ageRanges.length; i++)
            exports.AgeSegment.internalBinaryWrite(message.ageRanges[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool include_undetermined = 2; */
        if (message.includeUndetermined !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.includeUndetermined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AgeDimension
 */
exports.AgeDimension = new AgeDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AgeSegment$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AgeSegment", [
            { no: 1, name: "min_age", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 2, name: "max_age", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int32 min_age */ 1:
                    message.minAge = reader.int32();
                    break;
                case /* optional int32 max_age */ 2:
                    message.maxAge = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int32 min_age = 1; */
        if (message.minAge !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.minAge);
        /* optional int32 max_age = 2; */
        if (message.maxAge !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.maxAge);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AgeSegment
 */
exports.AgeSegment = new AgeSegment$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GenderDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.GenderDimension", [
            { no: 1, name: "genders", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.ads.googleads.v11.enums.GenderTypeEnum.GenderType", gender_type_1.GenderTypeEnum_GenderType] },
            { no: 2, name: "include_undetermined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { genders: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.enums.GenderTypeEnum.GenderType genders */ 1:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.genders.push(reader.int32());
                    else
                        message.genders.push(reader.int32());
                    break;
                case /* optional bool include_undetermined */ 2:
                    message.includeUndetermined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.enums.GenderTypeEnum.GenderType genders = 1; */
        if (message.genders.length) {
            writer.tag(1, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.genders.length; i++)
                writer.int32(message.genders[i]);
            writer.join();
        }
        /* optional bool include_undetermined = 2; */
        if (message.includeUndetermined !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.includeUndetermined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.GenderDimension
 */
exports.GenderDimension = new GenderDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class HouseholdIncomeDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.HouseholdIncomeDimension", [
            { no: 1, name: "income_ranges", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.ads.googleads.v11.enums.IncomeRangeTypeEnum.IncomeRangeType", income_range_type_1.IncomeRangeTypeEnum_IncomeRangeType] },
            { no: 2, name: "include_undetermined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { incomeRanges: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.enums.IncomeRangeTypeEnum.IncomeRangeType income_ranges */ 1:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.incomeRanges.push(reader.int32());
                    else
                        message.incomeRanges.push(reader.int32());
                    break;
                case /* optional bool include_undetermined */ 2:
                    message.includeUndetermined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.enums.IncomeRangeTypeEnum.IncomeRangeType income_ranges = 1; */
        if (message.incomeRanges.length) {
            writer.tag(1, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.incomeRanges.length; i++)
                writer.int32(message.incomeRanges[i]);
            writer.join();
        }
        /* optional bool include_undetermined = 2; */
        if (message.includeUndetermined !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.includeUndetermined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.HouseholdIncomeDimension
 */
exports.HouseholdIncomeDimension = new HouseholdIncomeDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ParentalStatusDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ParentalStatusDimension", [
            { no: 1, name: "parental_statuses", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.ads.googleads.v11.enums.ParentalStatusTypeEnum.ParentalStatusType", parental_status_type_1.ParentalStatusTypeEnum_ParentalStatusType] },
            { no: 2, name: "include_undetermined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { parentalStatuses: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.enums.ParentalStatusTypeEnum.ParentalStatusType parental_statuses */ 1:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.parentalStatuses.push(reader.int32());
                    else
                        message.parentalStatuses.push(reader.int32());
                    break;
                case /* optional bool include_undetermined */ 2:
                    message.includeUndetermined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.enums.ParentalStatusTypeEnum.ParentalStatusType parental_statuses = 1; */
        if (message.parentalStatuses.length) {
            writer.tag(1, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.parentalStatuses.length; i++)
                writer.int32(message.parentalStatuses[i]);
            writer.join();
        }
        /* optional bool include_undetermined = 2; */
        if (message.includeUndetermined !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.includeUndetermined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ParentalStatusDimension
 */
exports.ParentalStatusDimension = new ParentalStatusDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AudienceSegmentDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AudienceSegmentDimension", [
            { no: 1, name: "segments", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.AudienceSegment }
        ]);
    }
    create(value) {
        const message = { segments: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.AudienceSegment segments */ 1:
                    message.segments.push(exports.AudienceSegment.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.AudienceSegment segments = 1; */
        for (let i = 0; i < message.segments.length; i++)
            exports.AudienceSegment.internalBinaryWrite(message.segments[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AudienceSegmentDimension
 */
exports.AudienceSegmentDimension = new AudienceSegmentDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AudienceSegment$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AudienceSegment", [
            { no: 1, name: "user_list", kind: "message", oneof: "segment", T: () => exports.UserListSegment },
            { no: 2, name: "user_interest", kind: "message", oneof: "segment", T: () => exports.UserInterestSegment },
            { no: 3, name: "life_event", kind: "message", oneof: "segment", T: () => exports.LifeEventSegment },
            { no: 4, name: "detailed_demographic", kind: "message", oneof: "segment", T: () => exports.DetailedDemographicSegment },
            { no: 5, name: "custom_audience", kind: "message", oneof: "segment", T: () => exports.CustomAudienceSegment }
        ]);
    }
    create(value) {
        const message = { segment: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.UserListSegment user_list */ 1:
                    message.segment = {
                        oneofKind: "userList",
                        userList: exports.UserListSegment.internalBinaryRead(reader, reader.uint32(), options, message.segment.userList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.UserInterestSegment user_interest */ 2:
                    message.segment = {
                        oneofKind: "userInterest",
                        userInterest: exports.UserInterestSegment.internalBinaryRead(reader, reader.uint32(), options, message.segment.userInterest)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LifeEventSegment life_event */ 3:
                    message.segment = {
                        oneofKind: "lifeEvent",
                        lifeEvent: exports.LifeEventSegment.internalBinaryRead(reader, reader.uint32(), options, message.segment.lifeEvent)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DetailedDemographicSegment detailed_demographic */ 4:
                    message.segment = {
                        oneofKind: "detailedDemographic",
                        detailedDemographic: exports.DetailedDemographicSegment.internalBinaryRead(reader, reader.uint32(), options, message.segment.detailedDemographic)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CustomAudienceSegment custom_audience */ 5:
                    message.segment = {
                        oneofKind: "customAudience",
                        customAudience: exports.CustomAudienceSegment.internalBinaryRead(reader, reader.uint32(), options, message.segment.customAudience)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.UserListSegment user_list = 1; */
        if (message.segment.oneofKind === "userList")
            exports.UserListSegment.internalBinaryWrite(message.segment.userList, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserInterestSegment user_interest = 2; */
        if (message.segment.oneofKind === "userInterest")
            exports.UserInterestSegment.internalBinaryWrite(message.segment.userInterest, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LifeEventSegment life_event = 3; */
        if (message.segment.oneofKind === "lifeEvent")
            exports.LifeEventSegment.internalBinaryWrite(message.segment.lifeEvent, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DetailedDemographicSegment detailed_demographic = 4; */
        if (message.segment.oneofKind === "detailedDemographic")
            exports.DetailedDemographicSegment.internalBinaryWrite(message.segment.detailedDemographic, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CustomAudienceSegment custom_audience = 5; */
        if (message.segment.oneofKind === "customAudience")
            exports.CustomAudienceSegment.internalBinaryWrite(message.segment.customAudience, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AudienceSegment
 */
exports.AudienceSegment = new AudienceSegment$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListSegment$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListSegment", [
            { no: 1, name: "user_list", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string user_list */ 1:
                    message.userList = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string user_list = 1; */
        if (message.userList !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.userList);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListSegment
 */
exports.UserListSegment = new UserListSegment$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserInterestSegment$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserInterestSegment", [
            { no: 1, name: "user_interest_category", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string user_interest_category */ 1:
                    message.userInterestCategory = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string user_interest_category = 1; */
        if (message.userInterestCategory !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.userInterestCategory);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserInterestSegment
 */
exports.UserInterestSegment = new UserInterestSegment$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LifeEventSegment$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LifeEventSegment", [
            { no: 1, name: "life_event", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string life_event */ 1:
                    message.lifeEvent = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string life_event = 1; */
        if (message.lifeEvent !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.lifeEvent);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LifeEventSegment
 */
exports.LifeEventSegment = new LifeEventSegment$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DetailedDemographicSegment$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DetailedDemographicSegment", [
            { no: 1, name: "detailed_demographic", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string detailed_demographic */ 1:
                    message.detailedDemographic = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string detailed_demographic = 1; */
        if (message.detailedDemographic !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.detailedDemographic);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DetailedDemographicSegment
 */
exports.DetailedDemographicSegment = new DetailedDemographicSegment$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CustomAudienceSegment$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CustomAudienceSegment", [
            { no: 1, name: "custom_audience", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string custom_audience */ 1:
                    message.customAudience = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string custom_audience = 1; */
        if (message.customAudience !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.customAudience);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CustomAudienceSegment
 */
exports.CustomAudienceSegment = new CustomAudienceSegment$Type();
