// google-ads-api-client/dist/protos/bindings/ads/common/dates.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.YearMonth = exports.YearMonthRange = exports.DateRange = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const month_of_year_1 = require("../enums/month_of_year");
// @generated message type with reflection information, may provide speed optimized methods
class DateRange$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DateRange", [
            { no: 3, name: "start_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "end_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string start_date */ 3:
                    message.startDate = reader.string();
                    break;
                case /* optional string end_date */ 4:
                    message.endDate = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string start_date = 3; */
        if (message.startDate !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* optional string end_date = 4; */
        if (message.endDate !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.endDate);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DateRange
 */
exports.DateRange = new DateRange$Type();
// @generated message type with reflection information, may provide speed optimized methods
class YearMonthRange$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.YearMonthRange", [
            { no: 1, name: "start", kind: "message", T: () => exports.YearMonth },
            { no: 2, name: "end", kind: "message", T: () => exports.YearMonth }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.YearMonth start */ 1:
                    message.start = exports.YearMonth.internalBinaryRead(reader, reader.uint32(), options, message.start);
                    break;
                case /* google.ads.googleads.v11.common.YearMonth end */ 2:
                    message.end = exports.YearMonth.internalBinaryRead(reader, reader.uint32(), options, message.end);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.YearMonth start = 1; */
        if (message.start)
            exports.YearMonth.internalBinaryWrite(message.start, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.YearMonth end = 2; */
        if (message.end)
            exports.YearMonth.internalBinaryWrite(message.end, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.YearMonthRange
 */
exports.YearMonthRange = new YearMonthRange$Type();
// @generated message type with reflection information, may provide speed optimized methods
class YearMonth$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.YearMonth", [
            { no: 1, name: "year", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 2, name: "month", kind: "enum", T: () => ["google.ads.googleads.v11.enums.MonthOfYearEnum.MonthOfYear", month_of_year_1.MonthOfYearEnum_MonthOfYear] }
        ]);
    }
    create(value) {
        const message = { year: 0n, month: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int64 year */ 1:
                    message.year = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.MonthOfYearEnum.MonthOfYear month */ 2:
                    message.month = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int64 year = 1; */
        if (message.year !== 0n)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.year);
        /* google.ads.googleads.v11.enums.MonthOfYearEnum.MonthOfYear month = 2; */
        if (message.month !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.month);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.YearMonth
 */
exports.YearMonth = new YearMonth$Type();
