// google-ads-api-client/dist/protos/bindings/ads/common/extensions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageFeedItem = exports.HotelCalloutFeedItem = exports.SitelinkFeedItem = exports.StructuredSnippetFeedItem = exports.PromotionFeedItem = exports.PriceOffer = exports.PriceFeedItem = exports.TextMessageFeedItem = exports.AffiliateLocationFeedItem = exports.LocationFeedItem = exports.CalloutFeedItem = exports.CallFeedItem = exports.AppFeedItem = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const promotion_extension_occasion_1 = require("../enums/promotion_extension_occasion");
const promotion_extension_discount_modifier_1 = require("../enums/promotion_extension_discount_modifier");
const price_extension_price_unit_1 = require("../enums/price_extension_price_unit");
const feed_common_1 = require("./feed_common");
const price_extension_price_qualifier_1 = require("../enums/price_extension_price_qualifier");
const price_extension_type_1 = require("../enums/price_extension_type");
const call_conversion_reporting_state_1 = require("../enums/call_conversion_reporting_state");
const custom_parameter_1 = require("./custom_parameter");
const app_store_1 = require("../enums/app_store");
// @generated message type with reflection information, may provide speed optimized methods
class AppFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AppFeedItem", [
            { no: 9, name: "link_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "app_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "app_store", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AppStoreEnum.AppStore", app_store_1.AppStoreEnum_AppStore] },
            { no: 11, name: "final_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "final_mobile_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "url_custom_parameters", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => custom_parameter_1.CustomParameter },
            { no: 14, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { appStore: 0, finalUrls: [], finalMobileUrls: [], urlCustomParameters: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string link_text */ 9:
                    message.linkText = reader.string();
                    break;
                case /* optional string app_id */ 10:
                    message.appId = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AppStoreEnum.AppStore app_store */ 3:
                    message.appStore = reader.int32();
                    break;
                case /* repeated string final_urls */ 11:
                    message.finalUrls.push(reader.string());
                    break;
                case /* repeated string final_mobile_urls */ 12:
                    message.finalMobileUrls.push(reader.string());
                    break;
                case /* optional string tracking_url_template */ 13:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters */ 7:
                    message.urlCustomParameters.push(custom_parameter_1.CustomParameter.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string final_url_suffix */ 14:
                    message.finalUrlSuffix = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string link_text = 9; */
        if (message.linkText !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.linkText);
        /* optional string app_id = 10; */
        if (message.appId !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.appId);
        /* google.ads.googleads.v11.enums.AppStoreEnum.AppStore app_store = 3; */
        if (message.appStore !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.appStore);
        /* repeated string final_urls = 11; */
        for (let i = 0; i < message.finalUrls.length; i++)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.finalUrls[i]);
        /* repeated string final_mobile_urls = 12; */
        for (let i = 0; i < message.finalMobileUrls.length; i++)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.finalMobileUrls[i]);
        /* optional string tracking_url_template = 13; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters = 7; */
        for (let i = 0; i < message.urlCustomParameters.length; i++)
            custom_parameter_1.CustomParameter.internalBinaryWrite(message.urlCustomParameters[i], writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string final_url_suffix = 14; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.finalUrlSuffix);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AppFeedItem
 */
exports.AppFeedItem = new AppFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CallFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CallFeedItem", [
            { no: 7, name: "phone_number", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "country_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "call_tracking_enabled", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 10, name: "call_conversion_action", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "call_conversion_tracking_disabled", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 6, name: "call_conversion_reporting_state", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState", call_conversion_reporting_state_1.CallConversionReportingStateEnum_CallConversionReportingState] }
        ]);
    }
    create(value) {
        const message = { callConversionReportingState: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string phone_number */ 7:
                    message.phoneNumber = reader.string();
                    break;
                case /* optional string country_code */ 8:
                    message.countryCode = reader.string();
                    break;
                case /* optional bool call_tracking_enabled */ 9:
                    message.callTrackingEnabled = reader.bool();
                    break;
                case /* optional string call_conversion_action */ 10:
                    message.callConversionAction = reader.string();
                    break;
                case /* optional bool call_conversion_tracking_disabled */ 11:
                    message.callConversionTrackingDisabled = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState call_conversion_reporting_state */ 6:
                    message.callConversionReportingState = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string phone_number = 7; */
        if (message.phoneNumber !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.phoneNumber);
        /* optional string country_code = 8; */
        if (message.countryCode !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* optional bool call_tracking_enabled = 9; */
        if (message.callTrackingEnabled !== undefined)
            writer.tag(9, runtime_1.WireType.Varint).bool(message.callTrackingEnabled);
        /* optional string call_conversion_action = 10; */
        if (message.callConversionAction !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.callConversionAction);
        /* optional bool call_conversion_tracking_disabled = 11; */
        if (message.callConversionTrackingDisabled !== undefined)
            writer.tag(11, runtime_1.WireType.Varint).bool(message.callConversionTrackingDisabled);
        /* google.ads.googleads.v11.enums.CallConversionReportingStateEnum.CallConversionReportingState call_conversion_reporting_state = 6; */
        if (message.callConversionReportingState !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.callConversionReportingState);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CallFeedItem
 */
exports.CallFeedItem = new CallFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CalloutFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CalloutFeedItem", [
            { no: 2, name: "callout_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string callout_text */ 2:
                    message.calloutText = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string callout_text = 2; */
        if (message.calloutText !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.calloutText);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CalloutFeedItem
 */
exports.CalloutFeedItem = new CalloutFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LocationFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LocationFeedItem", [
            { no: 9, name: "business_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "address_line_1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "address_line_2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "city", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "province", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "postal_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 15, name: "country_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "phone_number", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string business_name */ 9:
                    message.businessName = reader.string();
                    break;
                case /* optional string address_line_1 */ 10:
                    message.addressLine1 = reader.string();
                    break;
                case /* optional string address_line_2 */ 11:
                    message.addressLine2 = reader.string();
                    break;
                case /* optional string city */ 12:
                    message.city = reader.string();
                    break;
                case /* optional string province */ 13:
                    message.province = reader.string();
                    break;
                case /* optional string postal_code */ 14:
                    message.postalCode = reader.string();
                    break;
                case /* optional string country_code */ 15:
                    message.countryCode = reader.string();
                    break;
                case /* optional string phone_number */ 16:
                    message.phoneNumber = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string business_name = 9; */
        if (message.businessName !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* optional string address_line_1 = 10; */
        if (message.addressLine1 !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.addressLine1);
        /* optional string address_line_2 = 11; */
        if (message.addressLine2 !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.addressLine2);
        /* optional string city = 12; */
        if (message.city !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.city);
        /* optional string province = 13; */
        if (message.province !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.province);
        /* optional string postal_code = 14; */
        if (message.postalCode !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.postalCode);
        /* optional string country_code = 15; */
        if (message.countryCode !== undefined)
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* optional string phone_number = 16; */
        if (message.phoneNumber !== undefined)
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.phoneNumber);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LocationFeedItem
 */
exports.LocationFeedItem = new LocationFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AffiliateLocationFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.AffiliateLocationFeedItem", [
            { no: 11, name: "business_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "address_line_1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "address_line_2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "city", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 15, name: "province", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "postal_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 17, name: "country_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 18, name: "phone_number", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 19, name: "chain_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 20, name: "chain_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string business_name */ 11:
                    message.businessName = reader.string();
                    break;
                case /* optional string address_line_1 */ 12:
                    message.addressLine1 = reader.string();
                    break;
                case /* optional string address_line_2 */ 13:
                    message.addressLine2 = reader.string();
                    break;
                case /* optional string city */ 14:
                    message.city = reader.string();
                    break;
                case /* optional string province */ 15:
                    message.province = reader.string();
                    break;
                case /* optional string postal_code */ 16:
                    message.postalCode = reader.string();
                    break;
                case /* optional string country_code */ 17:
                    message.countryCode = reader.string();
                    break;
                case /* optional string phone_number */ 18:
                    message.phoneNumber = reader.string();
                    break;
                case /* optional int64 chain_id */ 19:
                    message.chainId = reader.int64().toBigInt();
                    break;
                case /* optional string chain_name */ 20:
                    message.chainName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string business_name = 11; */
        if (message.businessName !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* optional string address_line_1 = 12; */
        if (message.addressLine1 !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.addressLine1);
        /* optional string address_line_2 = 13; */
        if (message.addressLine2 !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.addressLine2);
        /* optional string city = 14; */
        if (message.city !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.city);
        /* optional string province = 15; */
        if (message.province !== undefined)
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.province);
        /* optional string postal_code = 16; */
        if (message.postalCode !== undefined)
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.postalCode);
        /* optional string country_code = 17; */
        if (message.countryCode !== undefined)
            writer.tag(17, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* optional string phone_number = 18; */
        if (message.phoneNumber !== undefined)
            writer.tag(18, runtime_1.WireType.LengthDelimited).string(message.phoneNumber);
        /* optional int64 chain_id = 19; */
        if (message.chainId !== undefined)
            writer.tag(19, runtime_1.WireType.Varint).int64(message.chainId);
        /* optional string chain_name = 20; */
        if (message.chainName !== undefined)
            writer.tag(20, runtime_1.WireType.LengthDelimited).string(message.chainName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.AffiliateLocationFeedItem
 */
exports.AffiliateLocationFeedItem = new AffiliateLocationFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TextMessageFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.TextMessageFeedItem", [
            { no: 6, name: "business_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "country_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "phone_number", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "extension_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string business_name */ 6:
                    message.businessName = reader.string();
                    break;
                case /* optional string country_code */ 7:
                    message.countryCode = reader.string();
                    break;
                case /* optional string phone_number */ 8:
                    message.phoneNumber = reader.string();
                    break;
                case /* optional string text */ 9:
                    message.text = reader.string();
                    break;
                case /* optional string extension_text */ 10:
                    message.extensionText = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string business_name = 6; */
        if (message.businessName !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* optional string country_code = 7; */
        if (message.countryCode !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* optional string phone_number = 8; */
        if (message.phoneNumber !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.phoneNumber);
        /* optional string text = 9; */
        if (message.text !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.text);
        /* optional string extension_text = 10; */
        if (message.extensionText !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.extensionText);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.TextMessageFeedItem
 */
exports.TextMessageFeedItem = new TextMessageFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PriceFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.PriceFeedItem", [
            { no: 1, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PriceExtensionTypeEnum.PriceExtensionType", price_extension_type_1.PriceExtensionTypeEnum_PriceExtensionType] },
            { no: 2, name: "price_qualifier", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum.PriceExtensionPriceQualifier", price_extension_price_qualifier_1.PriceExtensionPriceQualifierEnum_PriceExtensionPriceQualifier] },
            { no: 7, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "language_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "price_offerings", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.PriceOffer },
            { no: 9, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { type: 0, priceQualifier: 0, priceOfferings: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.PriceExtensionTypeEnum.PriceExtensionType type */ 1:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum.PriceExtensionPriceQualifier price_qualifier */ 2:
                    message.priceQualifier = reader.int32();
                    break;
                case /* optional string tracking_url_template */ 7:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* optional string language_code */ 8:
                    message.languageCode = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.PriceOffer price_offerings */ 5:
                    message.priceOfferings.push(exports.PriceOffer.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string final_url_suffix */ 9:
                    message.finalUrlSuffix = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.PriceExtensionTypeEnum.PriceExtensionType type = 1; */
        if (message.type !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.PriceExtensionPriceQualifierEnum.PriceExtensionPriceQualifier price_qualifier = 2; */
        if (message.priceQualifier !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.priceQualifier);
        /* optional string tracking_url_template = 7; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* optional string language_code = 8; */
        if (message.languageCode !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        /* repeated google.ads.googleads.v11.common.PriceOffer price_offerings = 5; */
        for (let i = 0; i < message.priceOfferings.length; i++)
            exports.PriceOffer.internalBinaryWrite(message.priceOfferings[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string final_url_suffix = 9; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.finalUrlSuffix);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.PriceFeedItem
 */
exports.PriceFeedItem = new PriceFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PriceOffer$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.PriceOffer", [
            { no: 7, name: "header", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "price", kind: "message", T: () => feed_common_1.Money },
            { no: 4, name: "unit", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum.PriceExtensionPriceUnit", price_extension_price_unit_1.PriceExtensionPriceUnitEnum_PriceExtensionPriceUnit] },
            { no: 9, name: "final_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "final_mobile_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { unit: 0, finalUrls: [], finalMobileUrls: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string header */ 7:
                    message.header = reader.string();
                    break;
                case /* optional string description */ 8:
                    message.description = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.Money price */ 3:
                    message.price = feed_common_1.Money.internalBinaryRead(reader, reader.uint32(), options, message.price);
                    break;
                case /* google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum.PriceExtensionPriceUnit unit */ 4:
                    message.unit = reader.int32();
                    break;
                case /* repeated string final_urls */ 9:
                    message.finalUrls.push(reader.string());
                    break;
                case /* repeated string final_mobile_urls */ 10:
                    message.finalMobileUrls.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string header = 7; */
        if (message.header !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.header);
        /* optional string description = 8; */
        if (message.description !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.description);
        /* google.ads.googleads.v11.common.Money price = 3; */
        if (message.price)
            feed_common_1.Money.internalBinaryWrite(message.price, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.PriceExtensionPriceUnitEnum.PriceExtensionPriceUnit unit = 4; */
        if (message.unit !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.unit);
        /* repeated string final_urls = 9; */
        for (let i = 0; i < message.finalUrls.length; i++)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.finalUrls[i]);
        /* repeated string final_mobile_urls = 10; */
        for (let i = 0; i < message.finalMobileUrls.length; i++)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.finalMobileUrls[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.PriceOffer
 */
exports.PriceOffer = new PriceOffer$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PromotionFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.PromotionFeedItem", [
            { no: 16, name: "promotion_target", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "discount_modifier", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum.PromotionExtensionDiscountModifier", promotion_extension_discount_modifier_1.PromotionExtensionDiscountModifierEnum_PromotionExtensionDiscountModifier] },
            { no: 19, name: "promotion_start_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 20, name: "promotion_end_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "occasion", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum.PromotionExtensionOccasion", promotion_extension_occasion_1.PromotionExtensionOccasionEnum_PromotionExtensionOccasion] },
            { no: 21, name: "final_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 22, name: "final_mobile_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 23, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "url_custom_parameters", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => custom_parameter_1.CustomParameter },
            { no: 24, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 25, name: "language_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 17, name: "percent_off", kind: "scalar", oneof: "discountType", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 4, name: "money_amount_off", kind: "message", oneof: "discountType", T: () => feed_common_1.Money },
            { no: 18, name: "promotion_code", kind: "scalar", oneof: "promotionTrigger", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "orders_over_amount", kind: "message", oneof: "promotionTrigger", T: () => feed_common_1.Money }
        ]);
    }
    create(value) {
        const message = { discountModifier: 0, occasion: 0, finalUrls: [], finalMobileUrls: [], urlCustomParameters: [], discountType: { oneofKind: undefined }, promotionTrigger: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string promotion_target */ 16:
                    message.promotionTarget = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum.PromotionExtensionDiscountModifier discount_modifier */ 2:
                    message.discountModifier = reader.int32();
                    break;
                case /* optional string promotion_start_date */ 19:
                    message.promotionStartDate = reader.string();
                    break;
                case /* optional string promotion_end_date */ 20:
                    message.promotionEndDate = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum.PromotionExtensionOccasion occasion */ 9:
                    message.occasion = reader.int32();
                    break;
                case /* repeated string final_urls */ 21:
                    message.finalUrls.push(reader.string());
                    break;
                case /* repeated string final_mobile_urls */ 22:
                    message.finalMobileUrls.push(reader.string());
                    break;
                case /* optional string tracking_url_template */ 23:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters */ 13:
                    message.urlCustomParameters.push(custom_parameter_1.CustomParameter.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string final_url_suffix */ 24:
                    message.finalUrlSuffix = reader.string();
                    break;
                case /* optional string language_code */ 25:
                    message.languageCode = reader.string();
                    break;
                case /* int64 percent_off */ 17:
                    message.discountType = {
                        oneofKind: "percentOff",
                        percentOff: reader.int64().toBigInt()
                    };
                    break;
                case /* google.ads.googleads.v11.common.Money money_amount_off */ 4:
                    message.discountType = {
                        oneofKind: "moneyAmountOff",
                        moneyAmountOff: feed_common_1.Money.internalBinaryRead(reader, reader.uint32(), options, message.discountType.moneyAmountOff)
                    };
                    break;
                case /* string promotion_code */ 18:
                    message.promotionTrigger = {
                        oneofKind: "promotionCode",
                        promotionCode: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.common.Money orders_over_amount */ 6:
                    message.promotionTrigger = {
                        oneofKind: "ordersOverAmount",
                        ordersOverAmount: feed_common_1.Money.internalBinaryRead(reader, reader.uint32(), options, message.promotionTrigger.ordersOverAmount)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string promotion_target = 16; */
        if (message.promotionTarget !== undefined)
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.promotionTarget);
        /* google.ads.googleads.v11.enums.PromotionExtensionDiscountModifierEnum.PromotionExtensionDiscountModifier discount_modifier = 2; */
        if (message.discountModifier !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.discountModifier);
        /* optional string promotion_start_date = 19; */
        if (message.promotionStartDate !== undefined)
            writer.tag(19, runtime_1.WireType.LengthDelimited).string(message.promotionStartDate);
        /* optional string promotion_end_date = 20; */
        if (message.promotionEndDate !== undefined)
            writer.tag(20, runtime_1.WireType.LengthDelimited).string(message.promotionEndDate);
        /* google.ads.googleads.v11.enums.PromotionExtensionOccasionEnum.PromotionExtensionOccasion occasion = 9; */
        if (message.occasion !== 0)
            writer.tag(9, runtime_1.WireType.Varint).int32(message.occasion);
        /* repeated string final_urls = 21; */
        for (let i = 0; i < message.finalUrls.length; i++)
            writer.tag(21, runtime_1.WireType.LengthDelimited).string(message.finalUrls[i]);
        /* repeated string final_mobile_urls = 22; */
        for (let i = 0; i < message.finalMobileUrls.length; i++)
            writer.tag(22, runtime_1.WireType.LengthDelimited).string(message.finalMobileUrls[i]);
        /* optional string tracking_url_template = 23; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(23, runtime_1.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters = 13; */
        for (let i = 0; i < message.urlCustomParameters.length; i++)
            custom_parameter_1.CustomParameter.internalBinaryWrite(message.urlCustomParameters[i], writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string final_url_suffix = 24; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(24, runtime_1.WireType.LengthDelimited).string(message.finalUrlSuffix);
        /* optional string language_code = 25; */
        if (message.languageCode !== undefined)
            writer.tag(25, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        /* int64 percent_off = 17; */
        if (message.discountType.oneofKind === "percentOff")
            writer.tag(17, runtime_1.WireType.Varint).int64(message.discountType.percentOff);
        /* google.ads.googleads.v11.common.Money money_amount_off = 4; */
        if (message.discountType.oneofKind === "moneyAmountOff")
            feed_common_1.Money.internalBinaryWrite(message.discountType.moneyAmountOff, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string promotion_code = 18; */
        if (message.promotionTrigger.oneofKind === "promotionCode")
            writer.tag(18, runtime_1.WireType.LengthDelimited).string(message.promotionTrigger.promotionCode);
        /* google.ads.googleads.v11.common.Money orders_over_amount = 6; */
        if (message.promotionTrigger.oneofKind === "ordersOverAmount")
            feed_common_1.Money.internalBinaryWrite(message.promotionTrigger.ordersOverAmount, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.PromotionFeedItem
 */
exports.PromotionFeedItem = new PromotionFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StructuredSnippetFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.StructuredSnippetFeedItem", [
            { no: 3, name: "header", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "values", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { values: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string header */ 3:
                    message.header = reader.string();
                    break;
                case /* repeated string values */ 4:
                    message.values.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string header = 3; */
        if (message.header !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.header);
        /* repeated string values = 4; */
        for (let i = 0; i < message.values.length; i++)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.values[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.StructuredSnippetFeedItem
 */
exports.StructuredSnippetFeedItem = new StructuredSnippetFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SitelinkFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.SitelinkFeedItem", [
            { no: 9, name: "link_text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "line1", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "line2", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "final_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "final_mobile_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "url_custom_parameters", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => custom_parameter_1.CustomParameter },
            { no: 15, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { finalUrls: [], finalMobileUrls: [], urlCustomParameters: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string link_text */ 9:
                    message.linkText = reader.string();
                    break;
                case /* optional string line1 */ 10:
                    message.line1 = reader.string();
                    break;
                case /* optional string line2 */ 11:
                    message.line2 = reader.string();
                    break;
                case /* repeated string final_urls */ 12:
                    message.finalUrls.push(reader.string());
                    break;
                case /* repeated string final_mobile_urls */ 13:
                    message.finalMobileUrls.push(reader.string());
                    break;
                case /* optional string tracking_url_template */ 14:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters */ 7:
                    message.urlCustomParameters.push(custom_parameter_1.CustomParameter.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string final_url_suffix */ 15:
                    message.finalUrlSuffix = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string link_text = 9; */
        if (message.linkText !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.linkText);
        /* optional string line1 = 10; */
        if (message.line1 !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.line1);
        /* optional string line2 = 11; */
        if (message.line2 !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.line2);
        /* repeated string final_urls = 12; */
        for (let i = 0; i < message.finalUrls.length; i++)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.finalUrls[i]);
        /* repeated string final_mobile_urls = 13; */
        for (let i = 0; i < message.finalMobileUrls.length; i++)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.finalMobileUrls[i]);
        /* optional string tracking_url_template = 14; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters = 7; */
        for (let i = 0; i < message.urlCustomParameters.length; i++)
            custom_parameter_1.CustomParameter.internalBinaryWrite(message.urlCustomParameters[i], writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string final_url_suffix = 15; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.finalUrlSuffix);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.SitelinkFeedItem
 */
exports.SitelinkFeedItem = new SitelinkFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class HotelCalloutFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.HotelCalloutFeedItem", [
            { no: 3, name: "text", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "language_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string text */ 3:
                    message.text = reader.string();
                    break;
                case /* optional string language_code */ 4:
                    message.languageCode = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string text = 3; */
        if (message.text !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.text);
        /* optional string language_code = 4; */
        if (message.languageCode !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.HotelCalloutFeedItem
 */
exports.HotelCalloutFeedItem = new HotelCalloutFeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ImageFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ImageFeedItem", [
            { no: 1, name: "image_asset", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"], "google.api.resource_reference": { type: "googleads.googleapis.com/Asset" } } }
        ]);
    }
    create(value) {
        const message = { imageAsset: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string image_asset */ 1:
                    message.imageAsset = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string image_asset = 1; */
        if (message.imageAsset !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.imageAsset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ImageFeedItem
 */
exports.ImageFeedItem = new ImageFeedItem$Type();
