// google-ads-api-client/dist/protos/bindings/ads/common/feed_item_set_filter_type_infos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DynamicAffiliateLocationSetFilter = exports.BusinessNameFilter = exports.DynamicLocationSetFilter = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const feed_item_set_string_filter_type_1 = require("../enums/feed_item_set_string_filter_type");
// @generated message type with reflection information, may provide speed optimized methods
class DynamicLocationSetFilter$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicLocationSetFilter", [
            { no: 1, name: "labels", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "business_name_filter", kind: "message", T: () => exports.BusinessNameFilter }
        ]);
    }
    create(value) {
        const message = { labels: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated string labels */ 1:
                    message.labels.push(reader.string());
                    break;
                case /* google.ads.googleads.v11.common.BusinessNameFilter business_name_filter */ 2:
                    message.businessNameFilter = exports.BusinessNameFilter.internalBinaryRead(reader, reader.uint32(), options, message.businessNameFilter);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated string labels = 1; */
        for (let i = 0; i < message.labels.length; i++)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.labels[i]);
        /* google.ads.googleads.v11.common.BusinessNameFilter business_name_filter = 2; */
        if (message.businessNameFilter)
            exports.BusinessNameFilter.internalBinaryWrite(message.businessNameFilter, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicLocationSetFilter
 */
exports.DynamicLocationSetFilter = new DynamicLocationSetFilter$Type();
// @generated message type with reflection information, may provide speed optimized methods
class BusinessNameFilter$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.BusinessNameFilter", [
            { no: 1, name: "business_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "filter_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemSetStringFilterTypeEnum.FeedItemSetStringFilterType", feed_item_set_string_filter_type_1.FeedItemSetStringFilterTypeEnum_FeedItemSetStringFilterType] }
        ]);
    }
    create(value) {
        const message = { businessName: "", filterType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string business_name */ 1:
                    message.businessName = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemSetStringFilterTypeEnum.FeedItemSetStringFilterType filter_type */ 2:
                    message.filterType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string business_name = 1; */
        if (message.businessName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.businessName);
        /* google.ads.googleads.v11.enums.FeedItemSetStringFilterTypeEnum.FeedItemSetStringFilterType filter_type = 2; */
        if (message.filterType !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.filterType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.BusinessNameFilter
 */
exports.BusinessNameFilter = new BusinessNameFilter$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DynamicAffiliateLocationSetFilter$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.DynamicAffiliateLocationSetFilter", [
            { no: 1, name: "chain_ids", kind: "scalar", repeat: 1 /*RepeatType.PACKED*/, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { chainIds: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated int64 chain_ids */ 1:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.chainIds.push(reader.int64().toBigInt());
                    else
                        message.chainIds.push(reader.int64().toBigInt());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated int64 chain_ids = 1; */
        if (message.chainIds.length) {
            writer.tag(1, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.chainIds.length; i++)
                writer.int64(message.chainIds[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.DynamicAffiliateLocationSetFilter
 */
exports.DynamicAffiliateLocationSetFilter = new DynamicAffiliateLocationSetFilter$Type();
