// google-ads-api-client/dist/protos/bindings/ads/common/matching_function.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Operand_RequestContextOperand = exports.Operand_FunctionOperand = exports.Operand_FeedAttributeOperand = exports.Operand_ConstantOperand = exports.Operand = exports.MatchingFunction = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const matching_function_context_type_1 = require("../enums/matching_function_context_type");
const matching_function_operator_1 = require("../enums/matching_function_operator");
// @generated message type with reflection information, may provide speed optimized methods
class MatchingFunction$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.MatchingFunction", [
            { no: 5, name: "function_string", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "operator", kind: "enum", T: () => ["google.ads.googleads.v11.enums.MatchingFunctionOperatorEnum.MatchingFunctionOperator", matching_function_operator_1.MatchingFunctionOperatorEnum_MatchingFunctionOperator] },
            { no: 2, name: "left_operands", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.Operand },
            { no: 3, name: "right_operands", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.Operand }
        ]);
    }
    create(value) {
        const message = { operator: 0, leftOperands: [], rightOperands: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string function_string */ 5:
                    message.functionString = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.MatchingFunctionOperatorEnum.MatchingFunctionOperator operator */ 4:
                    message.operator = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.common.Operand left_operands */ 2:
                    message.leftOperands.push(exports.Operand.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.Operand right_operands */ 3:
                    message.rightOperands.push(exports.Operand.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string function_string = 5; */
        if (message.functionString !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.functionString);
        /* google.ads.googleads.v11.enums.MatchingFunctionOperatorEnum.MatchingFunctionOperator operator = 4; */
        if (message.operator !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.operator);
        /* repeated google.ads.googleads.v11.common.Operand left_operands = 2; */
        for (let i = 0; i < message.leftOperands.length; i++)
            exports.Operand.internalBinaryWrite(message.leftOperands[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.Operand right_operands = 3; */
        for (let i = 0; i < message.rightOperands.length; i++)
            exports.Operand.internalBinaryWrite(message.rightOperands[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.MatchingFunction
 */
exports.MatchingFunction = new MatchingFunction$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Operand$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.Operand", [
            { no: 1, name: "constant_operand", kind: "message", oneof: "functionArgumentOperand", T: () => exports.Operand_ConstantOperand },
            { no: 2, name: "feed_attribute_operand", kind: "message", oneof: "functionArgumentOperand", T: () => exports.Operand_FeedAttributeOperand },
            { no: 3, name: "function_operand", kind: "message", oneof: "functionArgumentOperand", T: () => exports.Operand_FunctionOperand },
            { no: 4, name: "request_context_operand", kind: "message", oneof: "functionArgumentOperand", T: () => exports.Operand_RequestContextOperand }
        ]);
    }
    create(value) {
        const message = { functionArgumentOperand: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.Operand.ConstantOperand constant_operand */ 1:
                    message.functionArgumentOperand = {
                        oneofKind: "constantOperand",
                        constantOperand: exports.Operand_ConstantOperand.internalBinaryRead(reader, reader.uint32(), options, message.functionArgumentOperand.constantOperand)
                    };
                    break;
                case /* google.ads.googleads.v11.common.Operand.FeedAttributeOperand feed_attribute_operand */ 2:
                    message.functionArgumentOperand = {
                        oneofKind: "feedAttributeOperand",
                        feedAttributeOperand: exports.Operand_FeedAttributeOperand.internalBinaryRead(reader, reader.uint32(), options, message.functionArgumentOperand.feedAttributeOperand)
                    };
                    break;
                case /* google.ads.googleads.v11.common.Operand.FunctionOperand function_operand */ 3:
                    message.functionArgumentOperand = {
                        oneofKind: "functionOperand",
                        functionOperand: exports.Operand_FunctionOperand.internalBinaryRead(reader, reader.uint32(), options, message.functionArgumentOperand.functionOperand)
                    };
                    break;
                case /* google.ads.googleads.v11.common.Operand.RequestContextOperand request_context_operand */ 4:
                    message.functionArgumentOperand = {
                        oneofKind: "requestContextOperand",
                        requestContextOperand: exports.Operand_RequestContextOperand.internalBinaryRead(reader, reader.uint32(), options, message.functionArgumentOperand.requestContextOperand)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.Operand.ConstantOperand constant_operand = 1; */
        if (message.functionArgumentOperand.oneofKind === "constantOperand")
            exports.Operand_ConstantOperand.internalBinaryWrite(message.functionArgumentOperand.constantOperand, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.Operand.FeedAttributeOperand feed_attribute_operand = 2; */
        if (message.functionArgumentOperand.oneofKind === "feedAttributeOperand")
            exports.Operand_FeedAttributeOperand.internalBinaryWrite(message.functionArgumentOperand.feedAttributeOperand, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.Operand.FunctionOperand function_operand = 3; */
        if (message.functionArgumentOperand.oneofKind === "functionOperand")
            exports.Operand_FunctionOperand.internalBinaryWrite(message.functionArgumentOperand.functionOperand, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.Operand.RequestContextOperand request_context_operand = 4; */
        if (message.functionArgumentOperand.oneofKind === "requestContextOperand")
            exports.Operand_RequestContextOperand.internalBinaryWrite(message.functionArgumentOperand.requestContextOperand, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.Operand
 */
exports.Operand = new Operand$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Operand_ConstantOperand$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.Operand.ConstantOperand", [
            { no: 5, name: "string_value", kind: "scalar", oneof: "constantOperandValue", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "long_value", kind: "scalar", oneof: "constantOperandValue", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 7, name: "boolean_value", kind: "scalar", oneof: "constantOperandValue", T: 8 /*ScalarType.BOOL*/ },
            { no: 8, name: "double_value", kind: "scalar", oneof: "constantOperandValue", T: 1 /*ScalarType.DOUBLE*/ }
        ]);
    }
    create(value) {
        const message = { constantOperandValue: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string string_value */ 5:
                    message.constantOperandValue = {
                        oneofKind: "stringValue",
                        stringValue: reader.string()
                    };
                    break;
                case /* int64 long_value */ 6:
                    message.constantOperandValue = {
                        oneofKind: "longValue",
                        longValue: reader.int64().toBigInt()
                    };
                    break;
                case /* bool boolean_value */ 7:
                    message.constantOperandValue = {
                        oneofKind: "booleanValue",
                        booleanValue: reader.bool()
                    };
                    break;
                case /* double double_value */ 8:
                    message.constantOperandValue = {
                        oneofKind: "doubleValue",
                        doubleValue: reader.double()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string string_value = 5; */
        if (message.constantOperandValue.oneofKind === "stringValue")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.constantOperandValue.stringValue);
        /* int64 long_value = 6; */
        if (message.constantOperandValue.oneofKind === "longValue")
            writer.tag(6, runtime_1.WireType.Varint).int64(message.constantOperandValue.longValue);
        /* bool boolean_value = 7; */
        if (message.constantOperandValue.oneofKind === "booleanValue")
            writer.tag(7, runtime_1.WireType.Varint).bool(message.constantOperandValue.booleanValue);
        /* double double_value = 8; */
        if (message.constantOperandValue.oneofKind === "doubleValue")
            writer.tag(8, runtime_1.WireType.Bit64).double(message.constantOperandValue.doubleValue);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.Operand.ConstantOperand
 */
exports.Operand_ConstantOperand = new Operand_ConstantOperand$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Operand_FeedAttributeOperand$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.Operand.FeedAttributeOperand", [
            { no: 3, name: "feed_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 4, name: "feed_attribute_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 feed_id */ 3:
                    message.feedId = reader.int64().toBigInt();
                    break;
                case /* optional int64 feed_attribute_id */ 4:
                    message.feedAttributeId = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 feed_id = 3; */
        if (message.feedId !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.feedId);
        /* optional int64 feed_attribute_id = 4; */
        if (message.feedAttributeId !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).int64(message.feedAttributeId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.Operand.FeedAttributeOperand
 */
exports.Operand_FeedAttributeOperand = new Operand_FeedAttributeOperand$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Operand_FunctionOperand$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.Operand.FunctionOperand", [
            { no: 1, name: "matching_function", kind: "message", T: () => exports.MatchingFunction }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.MatchingFunction matching_function */ 1:
                    message.matchingFunction = exports.MatchingFunction.internalBinaryRead(reader, reader.uint32(), options, message.matchingFunction);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.MatchingFunction matching_function = 1; */
        if (message.matchingFunction)
            exports.MatchingFunction.internalBinaryWrite(message.matchingFunction, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.Operand.FunctionOperand
 */
exports.Operand_FunctionOperand = new Operand_FunctionOperand$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Operand_RequestContextOperand$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.Operand.RequestContextOperand", [
            { no: 1, name: "context_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.MatchingFunctionContextTypeEnum.MatchingFunctionContextType", matching_function_context_type_1.MatchingFunctionContextTypeEnum_MatchingFunctionContextType] }
        ]);
    }
    create(value) {
        const message = { contextType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.MatchingFunctionContextTypeEnum.MatchingFunctionContextType context_type */ 1:
                    message.contextType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.MatchingFunctionContextTypeEnum.MatchingFunctionContextType context_type = 1; */
        if (message.contextType !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.contextType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.Operand.RequestContextOperand
 */
exports.Operand_RequestContextOperand = new Operand_RequestContextOperand$Type();
