// google-ads-api-client/dist/protos/bindings/ads/common/offline_user_data.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoreSalesThirdPartyMetadata = exports.StoreSalesMetadata = exports.CustomerMatchUserListMetadata = exports.ShoppingLoyalty = exports.EventItemAttribute = exports.EventAttribute = exports.UserAttribute = exports.UserData = exports.ItemAttribute = exports.StoreAttribute = exports.TransactionAttribute = exports.UserIdentifier = exports.OfflineUserAddressInfo = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const user_identifier_source_1 = require("../enums/user_identifier_source");
// @generated message type with reflection information, may provide speed optimized methods
class OfflineUserAddressInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.OfflineUserAddressInfo", [
            { no: 7, name: "hashed_first_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "hashed_last_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "city", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "state", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "country_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "postal_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 13, name: "hashed_street_address", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string hashed_first_name */ 7:
                    message.hashedFirstName = reader.string();
                    break;
                case /* optional string hashed_last_name */ 8:
                    message.hashedLastName = reader.string();
                    break;
                case /* optional string city */ 9:
                    message.city = reader.string();
                    break;
                case /* optional string state */ 10:
                    message.state = reader.string();
                    break;
                case /* optional string country_code */ 11:
                    message.countryCode = reader.string();
                    break;
                case /* optional string postal_code */ 12:
                    message.postalCode = reader.string();
                    break;
                case /* optional string hashed_street_address */ 13:
                    message.hashedStreetAddress = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string hashed_first_name = 7; */
        if (message.hashedFirstName !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.hashedFirstName);
        /* optional string hashed_last_name = 8; */
        if (message.hashedLastName !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.hashedLastName);
        /* optional string city = 9; */
        if (message.city !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.city);
        /* optional string state = 10; */
        if (message.state !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.state);
        /* optional string country_code = 11; */
        if (message.countryCode !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* optional string postal_code = 12; */
        if (message.postalCode !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.postalCode);
        /* optional string hashed_street_address = 13; */
        if (message.hashedStreetAddress !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.hashedStreetAddress);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.OfflineUserAddressInfo
 */
exports.OfflineUserAddressInfo = new OfflineUserAddressInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserIdentifier$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserIdentifier", [
            { no: 6, name: "user_identifier_source", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserIdentifierSourceEnum.UserIdentifierSource", user_identifier_source_1.UserIdentifierSourceEnum_UserIdentifierSource] },
            { no: 7, name: "hashed_email", kind: "scalar", oneof: "identifier", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "hashed_phone_number", kind: "scalar", oneof: "identifier", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "mobile_id", kind: "scalar", oneof: "identifier", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "third_party_user_id", kind: "scalar", oneof: "identifier", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "address_info", kind: "message", oneof: "identifier", T: () => exports.OfflineUserAddressInfo }
        ]);
    }
    create(value) {
        const message = { userIdentifierSource: 0, identifier: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.UserIdentifierSourceEnum.UserIdentifierSource user_identifier_source */ 6:
                    message.userIdentifierSource = reader.int32();
                    break;
                case /* string hashed_email */ 7:
                    message.identifier = {
                        oneofKind: "hashedEmail",
                        hashedEmail: reader.string()
                    };
                    break;
                case /* string hashed_phone_number */ 8:
                    message.identifier = {
                        oneofKind: "hashedPhoneNumber",
                        hashedPhoneNumber: reader.string()
                    };
                    break;
                case /* string mobile_id */ 9:
                    message.identifier = {
                        oneofKind: "mobileId",
                        mobileId: reader.string()
                    };
                    break;
                case /* string third_party_user_id */ 10:
                    message.identifier = {
                        oneofKind: "thirdPartyUserId",
                        thirdPartyUserId: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.common.OfflineUserAddressInfo address_info */ 5:
                    message.identifier = {
                        oneofKind: "addressInfo",
                        addressInfo: exports.OfflineUserAddressInfo.internalBinaryRead(reader, reader.uint32(), options, message.identifier.addressInfo)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.UserIdentifierSourceEnum.UserIdentifierSource user_identifier_source = 6; */
        if (message.userIdentifierSource !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.userIdentifierSource);
        /* string hashed_email = 7; */
        if (message.identifier.oneofKind === "hashedEmail")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.identifier.hashedEmail);
        /* string hashed_phone_number = 8; */
        if (message.identifier.oneofKind === "hashedPhoneNumber")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.identifier.hashedPhoneNumber);
        /* string mobile_id = 9; */
        if (message.identifier.oneofKind === "mobileId")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.identifier.mobileId);
        /* string third_party_user_id = 10; */
        if (message.identifier.oneofKind === "thirdPartyUserId")
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.identifier.thirdPartyUserId);
        /* google.ads.googleads.v11.common.OfflineUserAddressInfo address_info = 5; */
        if (message.identifier.oneofKind === "addressInfo")
            exports.OfflineUserAddressInfo.internalBinaryWrite(message.identifier.addressInfo, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserIdentifier
 */
exports.UserIdentifier = new UserIdentifier$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TransactionAttribute$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.TransactionAttribute", [
            { no: 8, name: "transaction_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "transaction_amount_micros", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 10, name: "currency_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "conversion_action", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "order_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "store_attribute", kind: "message", T: () => exports.StoreAttribute },
            { no: 13, name: "custom_value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "item_attribute", kind: "message", T: () => exports.ItemAttribute }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string transaction_date_time */ 8:
                    message.transactionDateTime = reader.string();
                    break;
                case /* optional double transaction_amount_micros */ 9:
                    message.transactionAmountMicros = reader.double();
                    break;
                case /* optional string currency_code */ 10:
                    message.currencyCode = reader.string();
                    break;
                case /* optional string conversion_action */ 11:
                    message.conversionAction = reader.string();
                    break;
                case /* optional string order_id */ 12:
                    message.orderId = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.StoreAttribute store_attribute */ 6:
                    message.storeAttribute = exports.StoreAttribute.internalBinaryRead(reader, reader.uint32(), options, message.storeAttribute);
                    break;
                case /* optional string custom_value */ 13:
                    message.customValue = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.ItemAttribute item_attribute */ 14:
                    message.itemAttribute = exports.ItemAttribute.internalBinaryRead(reader, reader.uint32(), options, message.itemAttribute);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string transaction_date_time = 8; */
        if (message.transactionDateTime !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.transactionDateTime);
        /* optional double transaction_amount_micros = 9; */
        if (message.transactionAmountMicros !== undefined)
            writer.tag(9, runtime_1.WireType.Bit64).double(message.transactionAmountMicros);
        /* optional string currency_code = 10; */
        if (message.currencyCode !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.currencyCode);
        /* optional string conversion_action = 11; */
        if (message.conversionAction !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.conversionAction);
        /* optional string order_id = 12; */
        if (message.orderId !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.orderId);
        /* google.ads.googleads.v11.common.StoreAttribute store_attribute = 6; */
        if (message.storeAttribute)
            exports.StoreAttribute.internalBinaryWrite(message.storeAttribute, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string custom_value = 13; */
        if (message.customValue !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.customValue);
        /* google.ads.googleads.v11.common.ItemAttribute item_attribute = 14; */
        if (message.itemAttribute)
            exports.ItemAttribute.internalBinaryWrite(message.itemAttribute, writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.TransactionAttribute
 */
exports.TransactionAttribute = new TransactionAttribute$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StoreAttribute$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.StoreAttribute", [
            { no: 2, name: "store_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string store_code */ 2:
                    message.storeCode = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string store_code = 2; */
        if (message.storeCode !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.storeCode);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.StoreAttribute
 */
exports.StoreAttribute = new StoreAttribute$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ItemAttribute$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ItemAttribute", [
            { no: 1, name: "item_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "merchant_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 3, name: "country_code", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "language_code", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "quantity", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { itemId: "", countryCode: "", languageCode: "", quantity: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string item_id */ 1:
                    message.itemId = reader.string();
                    break;
                case /* optional int64 merchant_id */ 2:
                    message.merchantId = reader.int64().toBigInt();
                    break;
                case /* string country_code */ 3:
                    message.countryCode = reader.string();
                    break;
                case /* string language_code */ 4:
                    message.languageCode = reader.string();
                    break;
                case /* int64 quantity */ 5:
                    message.quantity = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string item_id = 1; */
        if (message.itemId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.itemId);
        /* optional int64 merchant_id = 2; */
        if (message.merchantId !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.merchantId);
        /* string country_code = 3; */
        if (message.countryCode !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        /* string language_code = 4; */
        if (message.languageCode !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.languageCode);
        /* int64 quantity = 5; */
        if (message.quantity !== 0n)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.quantity);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ItemAttribute
 */
exports.ItemAttribute = new ItemAttribute$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserData$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserData", [
            { no: 1, name: "user_identifiers", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.UserIdentifier },
            { no: 2, name: "transaction_attribute", kind: "message", T: () => exports.TransactionAttribute },
            { no: 3, name: "user_attribute", kind: "message", T: () => exports.UserAttribute }
        ]);
    }
    create(value) {
        const message = { userIdentifiers: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.UserIdentifier user_identifiers */ 1:
                    message.userIdentifiers.push(exports.UserIdentifier.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.common.TransactionAttribute transaction_attribute */ 2:
                    message.transactionAttribute = exports.TransactionAttribute.internalBinaryRead(reader, reader.uint32(), options, message.transactionAttribute);
                    break;
                case /* google.ads.googleads.v11.common.UserAttribute user_attribute */ 3:
                    message.userAttribute = exports.UserAttribute.internalBinaryRead(reader, reader.uint32(), options, message.userAttribute);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.UserIdentifier user_identifiers = 1; */
        for (let i = 0; i < message.userIdentifiers.length; i++)
            exports.UserIdentifier.internalBinaryWrite(message.userIdentifiers[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TransactionAttribute transaction_attribute = 2; */
        if (message.transactionAttribute)
            exports.TransactionAttribute.internalBinaryWrite(message.transactionAttribute, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserAttribute user_attribute = 3; */
        if (message.userAttribute)
            exports.UserAttribute.internalBinaryWrite(message.userAttribute, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserData
 */
exports.UserData = new UserData$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserAttribute$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserAttribute", [
            { no: 1, name: "lifetime_value_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 2, name: "lifetime_value_bucket", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "last_purchase_date_time", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "average_purchase_count", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 5, name: "average_purchase_value_micros", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 6, name: "acquisition_date_time", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "shopping_loyalty", kind: "message", T: () => exports.ShoppingLoyalty },
            { no: 8, name: "lifecycle_stage", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 9, name: "first_purchase_date_time", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 10, name: "event_attribute", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.EventAttribute, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = { lastPurchaseDateTime: "", averagePurchaseCount: 0, averagePurchaseValueMicros: 0n, acquisitionDateTime: "", lifecycleStage: "", firstPurchaseDateTime: "", eventAttribute: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 lifetime_value_micros */ 1:
                    message.lifetimeValueMicros = reader.int64().toBigInt();
                    break;
                case /* optional int32 lifetime_value_bucket */ 2:
                    message.lifetimeValueBucket = reader.int32();
                    break;
                case /* string last_purchase_date_time */ 3:
                    message.lastPurchaseDateTime = reader.string();
                    break;
                case /* int32 average_purchase_count */ 4:
                    message.averagePurchaseCount = reader.int32();
                    break;
                case /* int64 average_purchase_value_micros */ 5:
                    message.averagePurchaseValueMicros = reader.int64().toBigInt();
                    break;
                case /* string acquisition_date_time */ 6:
                    message.acquisitionDateTime = reader.string();
                    break;
                case /* optional google.ads.googleads.v11.common.ShoppingLoyalty shopping_loyalty */ 7:
                    message.shoppingLoyalty = exports.ShoppingLoyalty.internalBinaryRead(reader, reader.uint32(), options, message.shoppingLoyalty);
                    break;
                case /* string lifecycle_stage */ 8:
                    message.lifecycleStage = reader.string();
                    break;
                case /* string first_purchase_date_time */ 9:
                    message.firstPurchaseDateTime = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.EventAttribute event_attribute */ 10:
                    message.eventAttribute.push(exports.EventAttribute.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 lifetime_value_micros = 1; */
        if (message.lifetimeValueMicros !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.lifetimeValueMicros);
        /* optional int32 lifetime_value_bucket = 2; */
        if (message.lifetimeValueBucket !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.lifetimeValueBucket);
        /* string last_purchase_date_time = 3; */
        if (message.lastPurchaseDateTime !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.lastPurchaseDateTime);
        /* int32 average_purchase_count = 4; */
        if (message.averagePurchaseCount !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.averagePurchaseCount);
        /* int64 average_purchase_value_micros = 5; */
        if (message.averagePurchaseValueMicros !== 0n)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.averagePurchaseValueMicros);
        /* string acquisition_date_time = 6; */
        if (message.acquisitionDateTime !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.acquisitionDateTime);
        /* optional google.ads.googleads.v11.common.ShoppingLoyalty shopping_loyalty = 7; */
        if (message.shoppingLoyalty)
            exports.ShoppingLoyalty.internalBinaryWrite(message.shoppingLoyalty, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string lifecycle_stage = 8; */
        if (message.lifecycleStage !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.lifecycleStage);
        /* string first_purchase_date_time = 9; */
        if (message.firstPurchaseDateTime !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.firstPurchaseDateTime);
        /* repeated google.ads.googleads.v11.common.EventAttribute event_attribute = 10; */
        for (let i = 0; i < message.eventAttribute.length; i++)
            exports.EventAttribute.internalBinaryWrite(message.eventAttribute[i], writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserAttribute
 */
exports.UserAttribute = new UserAttribute$Type();
// @generated message type with reflection information, may provide speed optimized methods
class EventAttribute$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.EventAttribute", [
            { no: 1, name: "event", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "event_date_time", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "item_attribute", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.EventItemAttribute, options: { "google.api.field_behavior": ["REQUIRED"] } }
        ]);
    }
    create(value) {
        const message = { event: "", eventDateTime: "", itemAttribute: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string event */ 1:
                    message.event = reader.string();
                    break;
                case /* string event_date_time */ 2:
                    message.eventDateTime = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.EventItemAttribute item_attribute */ 3:
                    message.itemAttribute.push(exports.EventItemAttribute.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string event = 1; */
        if (message.event !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.event);
        /* string event_date_time = 2; */
        if (message.eventDateTime !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.eventDateTime);
        /* repeated google.ads.googleads.v11.common.EventItemAttribute item_attribute = 3; */
        for (let i = 0; i < message.itemAttribute.length; i++)
            exports.EventItemAttribute.internalBinaryWrite(message.itemAttribute[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.EventAttribute
 */
exports.EventAttribute = new EventAttribute$Type();
// @generated message type with reflection information, may provide speed optimized methods
class EventItemAttribute$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.EventItemAttribute", [
            { no: 1, name: "item_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = { itemId: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string item_id */ 1:
                    message.itemId = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string item_id = 1; */
        if (message.itemId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.itemId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.EventItemAttribute
 */
exports.EventItemAttribute = new EventItemAttribute$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ShoppingLoyalty$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ShoppingLoyalty", [
            { no: 1, name: "loyalty_tier", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string loyalty_tier */ 1:
                    message.loyaltyTier = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string loyalty_tier = 1; */
        if (message.loyaltyTier !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.loyaltyTier);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ShoppingLoyalty
 */
exports.ShoppingLoyalty = new ShoppingLoyalty$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CustomerMatchUserListMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CustomerMatchUserListMetadata", [
            { no: 2, name: "user_list", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string user_list */ 2:
                    message.userList = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string user_list = 2; */
        if (message.userList !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.userList);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CustomerMatchUserListMetadata
 */
exports.CustomerMatchUserListMetadata = new CustomerMatchUserListMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StoreSalesMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.StoreSalesMetadata", [
            { no: 5, name: "loyalty_fraction", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 6, name: "transaction_upload_fraction", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 7, name: "custom_key", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "third_party_metadata", kind: "message", T: () => exports.StoreSalesThirdPartyMetadata }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional double loyalty_fraction */ 5:
                    message.loyaltyFraction = reader.double();
                    break;
                case /* optional double transaction_upload_fraction */ 6:
                    message.transactionUploadFraction = reader.double();
                    break;
                case /* optional string custom_key */ 7:
                    message.customKey = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.StoreSalesThirdPartyMetadata third_party_metadata */ 3:
                    message.thirdPartyMetadata = exports.StoreSalesThirdPartyMetadata.internalBinaryRead(reader, reader.uint32(), options, message.thirdPartyMetadata);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional double loyalty_fraction = 5; */
        if (message.loyaltyFraction !== undefined)
            writer.tag(5, runtime_1.WireType.Bit64).double(message.loyaltyFraction);
        /* optional double transaction_upload_fraction = 6; */
        if (message.transactionUploadFraction !== undefined)
            writer.tag(6, runtime_1.WireType.Bit64).double(message.transactionUploadFraction);
        /* optional string custom_key = 7; */
        if (message.customKey !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.customKey);
        /* google.ads.googleads.v11.common.StoreSalesThirdPartyMetadata third_party_metadata = 3; */
        if (message.thirdPartyMetadata)
            exports.StoreSalesThirdPartyMetadata.internalBinaryWrite(message.thirdPartyMetadata, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.StoreSalesMetadata
 */
exports.StoreSalesMetadata = new StoreSalesMetadata$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StoreSalesThirdPartyMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.StoreSalesThirdPartyMetadata", [
            { no: 7, name: "advertiser_upload_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "valid_transaction_fraction", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 9, name: "partner_match_fraction", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 10, name: "partner_upload_fraction", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 11, name: "bridge_map_version_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "partner_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string advertiser_upload_date_time */ 7:
                    message.advertiserUploadDateTime = reader.string();
                    break;
                case /* optional double valid_transaction_fraction */ 8:
                    message.validTransactionFraction = reader.double();
                    break;
                case /* optional double partner_match_fraction */ 9:
                    message.partnerMatchFraction = reader.double();
                    break;
                case /* optional double partner_upload_fraction */ 10:
                    message.partnerUploadFraction = reader.double();
                    break;
                case /* optional string bridge_map_version_id */ 11:
                    message.bridgeMapVersionId = reader.string();
                    break;
                case /* optional int64 partner_id */ 12:
                    message.partnerId = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string advertiser_upload_date_time = 7; */
        if (message.advertiserUploadDateTime !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.advertiserUploadDateTime);
        /* optional double valid_transaction_fraction = 8; */
        if (message.validTransactionFraction !== undefined)
            writer.tag(8, runtime_1.WireType.Bit64).double(message.validTransactionFraction);
        /* optional double partner_match_fraction = 9; */
        if (message.partnerMatchFraction !== undefined)
            writer.tag(9, runtime_1.WireType.Bit64).double(message.partnerMatchFraction);
        /* optional double partner_upload_fraction = 10; */
        if (message.partnerUploadFraction !== undefined)
            writer.tag(10, runtime_1.WireType.Bit64).double(message.partnerUploadFraction);
        /* optional string bridge_map_version_id = 11; */
        if (message.bridgeMapVersionId !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.bridgeMapVersionId);
        /* optional int64 partner_id = 12; */
        if (message.partnerId !== undefined)
            writer.tag(12, runtime_1.WireType.Varint).int64(message.partnerId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.StoreSalesThirdPartyMetadata
 */
exports.StoreSalesThirdPartyMetadata = new StoreSalesThirdPartyMetadata$Type();
