// google-ads-api-client/dist/protos/bindings/ads/common/user_lists.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListActionInfo = exports.BasicUserListInfo = exports.LogicalUserListOperandInfo = exports.UserListLogicalRuleInfo = exports.LogicalUserListInfo = exports.RuleBasedUserListInfo = exports.FlexibleRuleUserListInfo = exports.FlexibleRuleOperandInfo = exports.ExpressionRuleUserListInfo = exports.CombinedRuleUserListInfo = exports.UserListStringRuleItemInfo = exports.UserListNumberRuleItemInfo = exports.UserListDateRuleItemInfo = exports.UserListRuleItemInfo = exports.UserListRuleItemGroupInfo = exports.UserListRuleInfo = exports.CrmBasedUserListInfo = exports.SimilarUserListInfo = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const user_list_logical_rule_operator_1 = require("../enums/user_list_logical_rule_operator");
const user_list_prepopulation_status_1 = require("../enums/user_list_prepopulation_status");
const user_list_flexible_rule_operator_1 = require("../enums/user_list_flexible_rule_operator");
const user_list_combined_rule_operator_1 = require("../enums/user_list_combined_rule_operator");
const user_list_string_rule_item_operator_1 = require("../enums/user_list_string_rule_item_operator");
const user_list_number_rule_item_operator_1 = require("../enums/user_list_number_rule_item_operator");
const user_list_date_rule_item_operator_1 = require("../enums/user_list_date_rule_item_operator");
const user_list_rule_type_1 = require("../enums/user_list_rule_type");
const user_list_crm_data_source_type_1 = require("../enums/user_list_crm_data_source_type");
const customer_match_upload_key_type_1 = require("../enums/customer_match_upload_key_type");
// @generated message type with reflection information, may provide speed optimized methods
class SimilarUserListInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.SimilarUserListInfo", [
            { no: 2, name: "seed_user_list", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string seed_user_list */ 2:
                    message.seedUserList = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string seed_user_list = 2; */
        if (message.seedUserList !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.seedUserList);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.SimilarUserListInfo
 */
exports.SimilarUserListInfo = new SimilarUserListInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CrmBasedUserListInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CrmBasedUserListInfo", [
            { no: 4, name: "app_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "upload_key_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CustomerMatchUploadKeyTypeEnum.CustomerMatchUploadKeyType", customer_match_upload_key_type_1.CustomerMatchUploadKeyTypeEnum_CustomerMatchUploadKeyType] },
            { no: 3, name: "data_source_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListCrmDataSourceTypeEnum.UserListCrmDataSourceType", user_list_crm_data_source_type_1.UserListCrmDataSourceTypeEnum_UserListCrmDataSourceType] }
        ]);
    }
    create(value) {
        const message = { uploadKeyType: 0, dataSourceType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string app_id */ 4:
                    message.appId = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.CustomerMatchUploadKeyTypeEnum.CustomerMatchUploadKeyType upload_key_type */ 2:
                    message.uploadKeyType = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.UserListCrmDataSourceTypeEnum.UserListCrmDataSourceType data_source_type */ 3:
                    message.dataSourceType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string app_id = 4; */
        if (message.appId !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.appId);
        /* google.ads.googleads.v11.enums.CustomerMatchUploadKeyTypeEnum.CustomerMatchUploadKeyType upload_key_type = 2; */
        if (message.uploadKeyType !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.uploadKeyType);
        /* google.ads.googleads.v11.enums.UserListCrmDataSourceTypeEnum.UserListCrmDataSourceType data_source_type = 3; */
        if (message.dataSourceType !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.dataSourceType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CrmBasedUserListInfo
 */
exports.CrmBasedUserListInfo = new CrmBasedUserListInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListRuleInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListRuleInfo", [
            { no: 1, name: "rule_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListRuleTypeEnum.UserListRuleType", user_list_rule_type_1.UserListRuleTypeEnum_UserListRuleType] },
            { no: 2, name: "rule_item_groups", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.UserListRuleItemGroupInfo }
        ]);
    }
    create(value) {
        const message = { ruleType: 0, ruleItemGroups: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.UserListRuleTypeEnum.UserListRuleType rule_type */ 1:
                    message.ruleType = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.common.UserListRuleItemGroupInfo rule_item_groups */ 2:
                    message.ruleItemGroups.push(exports.UserListRuleItemGroupInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.UserListRuleTypeEnum.UserListRuleType rule_type = 1; */
        if (message.ruleType !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.ruleType);
        /* repeated google.ads.googleads.v11.common.UserListRuleItemGroupInfo rule_item_groups = 2; */
        for (let i = 0; i < message.ruleItemGroups.length; i++)
            exports.UserListRuleItemGroupInfo.internalBinaryWrite(message.ruleItemGroups[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListRuleInfo
 */
exports.UserListRuleInfo = new UserListRuleInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListRuleItemGroupInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListRuleItemGroupInfo", [
            { no: 1, name: "rule_items", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.UserListRuleItemInfo }
        ]);
    }
    create(value) {
        const message = { ruleItems: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.UserListRuleItemInfo rule_items */ 1:
                    message.ruleItems.push(exports.UserListRuleItemInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.UserListRuleItemInfo rule_items = 1; */
        for (let i = 0; i < message.ruleItems.length; i++)
            exports.UserListRuleItemInfo.internalBinaryWrite(message.ruleItems[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListRuleItemGroupInfo
 */
exports.UserListRuleItemGroupInfo = new UserListRuleItemGroupInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListRuleItemInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListRuleItemInfo", [
            { no: 5, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "number_rule_item", kind: "message", oneof: "ruleItem", T: () => exports.UserListNumberRuleItemInfo },
            { no: 3, name: "string_rule_item", kind: "message", oneof: "ruleItem", T: () => exports.UserListStringRuleItemInfo },
            { no: 4, name: "date_rule_item", kind: "message", oneof: "ruleItem", T: () => exports.UserListDateRuleItemInfo }
        ]);
    }
    create(value) {
        const message = { ruleItem: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string name */ 5:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.UserListNumberRuleItemInfo number_rule_item */ 2:
                    message.ruleItem = {
                        oneofKind: "numberRuleItem",
                        numberRuleItem: exports.UserListNumberRuleItemInfo.internalBinaryRead(reader, reader.uint32(), options, message.ruleItem.numberRuleItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.UserListStringRuleItemInfo string_rule_item */ 3:
                    message.ruleItem = {
                        oneofKind: "stringRuleItem",
                        stringRuleItem: exports.UserListStringRuleItemInfo.internalBinaryRead(reader, reader.uint32(), options, message.ruleItem.stringRuleItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.UserListDateRuleItemInfo date_rule_item */ 4:
                    message.ruleItem = {
                        oneofKind: "dateRuleItem",
                        dateRuleItem: exports.UserListDateRuleItemInfo.internalBinaryRead(reader, reader.uint32(), options, message.ruleItem.dateRuleItem)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string name = 5; */
        if (message.name !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.common.UserListNumberRuleItemInfo number_rule_item = 2; */
        if (message.ruleItem.oneofKind === "numberRuleItem")
            exports.UserListNumberRuleItemInfo.internalBinaryWrite(message.ruleItem.numberRuleItem, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserListStringRuleItemInfo string_rule_item = 3; */
        if (message.ruleItem.oneofKind === "stringRuleItem")
            exports.UserListStringRuleItemInfo.internalBinaryWrite(message.ruleItem.stringRuleItem, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserListDateRuleItemInfo date_rule_item = 4; */
        if (message.ruleItem.oneofKind === "dateRuleItem")
            exports.UserListDateRuleItemInfo.internalBinaryWrite(message.ruleItem.dateRuleItem, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListRuleItemInfo
 */
exports.UserListRuleItemInfo = new UserListRuleItemInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListDateRuleItemInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListDateRuleItemInfo", [
            { no: 1, name: "operator", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListDateRuleItemOperatorEnum.UserListDateRuleItemOperator", user_list_date_rule_item_operator_1.UserListDateRuleItemOperatorEnum_UserListDateRuleItemOperator] },
            { no: 4, name: "value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "offset_in_days", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = { operator: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.UserListDateRuleItemOperatorEnum.UserListDateRuleItemOperator operator */ 1:
                    message.operator = reader.int32();
                    break;
                case /* optional string value */ 4:
                    message.value = reader.string();
                    break;
                case /* optional int64 offset_in_days */ 5:
                    message.offsetInDays = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.UserListDateRuleItemOperatorEnum.UserListDateRuleItemOperator operator = 1; */
        if (message.operator !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.operator);
        /* optional string value = 4; */
        if (message.value !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.value);
        /* optional int64 offset_in_days = 5; */
        if (message.offsetInDays !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.offsetInDays);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListDateRuleItemInfo
 */
exports.UserListDateRuleItemInfo = new UserListDateRuleItemInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListNumberRuleItemInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListNumberRuleItemInfo", [
            { no: 1, name: "operator", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListNumberRuleItemOperatorEnum.UserListNumberRuleItemOperator", user_list_number_rule_item_operator_1.UserListNumberRuleItemOperatorEnum_UserListNumberRuleItemOperator] },
            { no: 3, name: "value", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ }
        ]);
    }
    create(value) {
        const message = { operator: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.UserListNumberRuleItemOperatorEnum.UserListNumberRuleItemOperator operator */ 1:
                    message.operator = reader.int32();
                    break;
                case /* optional double value */ 3:
                    message.value = reader.double();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.UserListNumberRuleItemOperatorEnum.UserListNumberRuleItemOperator operator = 1; */
        if (message.operator !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.operator);
        /* optional double value = 3; */
        if (message.value !== undefined)
            writer.tag(3, runtime_1.WireType.Bit64).double(message.value);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListNumberRuleItemInfo
 */
exports.UserListNumberRuleItemInfo = new UserListNumberRuleItemInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListStringRuleItemInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListStringRuleItemInfo", [
            { no: 1, name: "operator", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListStringRuleItemOperatorEnum.UserListStringRuleItemOperator", user_list_string_rule_item_operator_1.UserListStringRuleItemOperatorEnum_UserListStringRuleItemOperator] },
            { no: 3, name: "value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { operator: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.UserListStringRuleItemOperatorEnum.UserListStringRuleItemOperator operator */ 1:
                    message.operator = reader.int32();
                    break;
                case /* optional string value */ 3:
                    message.value = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.UserListStringRuleItemOperatorEnum.UserListStringRuleItemOperator operator = 1; */
        if (message.operator !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.operator);
        /* optional string value = 3; */
        if (message.value !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.value);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListStringRuleItemInfo
 */
exports.UserListStringRuleItemInfo = new UserListStringRuleItemInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CombinedRuleUserListInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.CombinedRuleUserListInfo", [
            { no: 1, name: "left_operand", kind: "message", T: () => exports.UserListRuleInfo },
            { no: 2, name: "right_operand", kind: "message", T: () => exports.UserListRuleInfo },
            { no: 3, name: "rule_operator", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListCombinedRuleOperatorEnum.UserListCombinedRuleOperator", user_list_combined_rule_operator_1.UserListCombinedRuleOperatorEnum_UserListCombinedRuleOperator] }
        ]);
    }
    create(value) {
        const message = { ruleOperator: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.UserListRuleInfo left_operand */ 1:
                    message.leftOperand = exports.UserListRuleInfo.internalBinaryRead(reader, reader.uint32(), options, message.leftOperand);
                    break;
                case /* google.ads.googleads.v11.common.UserListRuleInfo right_operand */ 2:
                    message.rightOperand = exports.UserListRuleInfo.internalBinaryRead(reader, reader.uint32(), options, message.rightOperand);
                    break;
                case /* google.ads.googleads.v11.enums.UserListCombinedRuleOperatorEnum.UserListCombinedRuleOperator rule_operator */ 3:
                    message.ruleOperator = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.UserListRuleInfo left_operand = 1; */
        if (message.leftOperand)
            exports.UserListRuleInfo.internalBinaryWrite(message.leftOperand, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserListRuleInfo right_operand = 2; */
        if (message.rightOperand)
            exports.UserListRuleInfo.internalBinaryWrite(message.rightOperand, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.UserListCombinedRuleOperatorEnum.UserListCombinedRuleOperator rule_operator = 3; */
        if (message.ruleOperator !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.ruleOperator);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.CombinedRuleUserListInfo
 */
exports.CombinedRuleUserListInfo = new CombinedRuleUserListInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ExpressionRuleUserListInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.ExpressionRuleUserListInfo", [
            { no: 1, name: "rule", kind: "message", T: () => exports.UserListRuleInfo }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.UserListRuleInfo rule */ 1:
                    message.rule = exports.UserListRuleInfo.internalBinaryRead(reader, reader.uint32(), options, message.rule);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.UserListRuleInfo rule = 1; */
        if (message.rule)
            exports.UserListRuleInfo.internalBinaryWrite(message.rule, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.ExpressionRuleUserListInfo
 */
exports.ExpressionRuleUserListInfo = new ExpressionRuleUserListInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class FlexibleRuleOperandInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.FlexibleRuleOperandInfo", [
            { no: 1, name: "rule", kind: "message", T: () => exports.UserListRuleInfo },
            { no: 2, name: "lookback_window_days", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.UserListRuleInfo rule */ 1:
                    message.rule = exports.UserListRuleInfo.internalBinaryRead(reader, reader.uint32(), options, message.rule);
                    break;
                case /* optional int64 lookback_window_days */ 2:
                    message.lookbackWindowDays = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.UserListRuleInfo rule = 1; */
        if (message.rule)
            exports.UserListRuleInfo.internalBinaryWrite(message.rule, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional int64 lookback_window_days = 2; */
        if (message.lookbackWindowDays !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.lookbackWindowDays);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.FlexibleRuleOperandInfo
 */
exports.FlexibleRuleOperandInfo = new FlexibleRuleOperandInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class FlexibleRuleUserListInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.FlexibleRuleUserListInfo", [
            { no: 1, name: "inclusive_rule_operator", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListFlexibleRuleOperatorEnum.UserListFlexibleRuleOperator", user_list_flexible_rule_operator_1.UserListFlexibleRuleOperatorEnum_UserListFlexibleRuleOperator] },
            { no: 2, name: "inclusive_operands", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.FlexibleRuleOperandInfo },
            { no: 3, name: "exclusive_operands", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.FlexibleRuleOperandInfo }
        ]);
    }
    create(value) {
        const message = { inclusiveRuleOperator: 0, inclusiveOperands: [], exclusiveOperands: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.UserListFlexibleRuleOperatorEnum.UserListFlexibleRuleOperator inclusive_rule_operator */ 1:
                    message.inclusiveRuleOperator = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.common.FlexibleRuleOperandInfo inclusive_operands */ 2:
                    message.inclusiveOperands.push(exports.FlexibleRuleOperandInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.common.FlexibleRuleOperandInfo exclusive_operands */ 3:
                    message.exclusiveOperands.push(exports.FlexibleRuleOperandInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.UserListFlexibleRuleOperatorEnum.UserListFlexibleRuleOperator inclusive_rule_operator = 1; */
        if (message.inclusiveRuleOperator !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.inclusiveRuleOperator);
        /* repeated google.ads.googleads.v11.common.FlexibleRuleOperandInfo inclusive_operands = 2; */
        for (let i = 0; i < message.inclusiveOperands.length; i++)
            exports.FlexibleRuleOperandInfo.internalBinaryWrite(message.inclusiveOperands[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.FlexibleRuleOperandInfo exclusive_operands = 3; */
        for (let i = 0; i < message.exclusiveOperands.length; i++)
            exports.FlexibleRuleOperandInfo.internalBinaryWrite(message.exclusiveOperands[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.FlexibleRuleUserListInfo
 */
exports.FlexibleRuleUserListInfo = new FlexibleRuleUserListInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class RuleBasedUserListInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.RuleBasedUserListInfo", [
            { no: 1, name: "prepopulation_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListPrepopulationStatusEnum.UserListPrepopulationStatus", user_list_prepopulation_status_1.UserListPrepopulationStatusEnum_UserListPrepopulationStatus] },
            { no: 5, name: "flexible_rule_user_list", kind: "message", T: () => exports.FlexibleRuleUserListInfo },
            { no: 2, name: "combined_rule_user_list", kind: "message", oneof: "ruleBasedUserList", T: () => exports.CombinedRuleUserListInfo },
            { no: 4, name: "expression_rule_user_list", kind: "message", oneof: "ruleBasedUserList", T: () => exports.ExpressionRuleUserListInfo }
        ]);
    }
    create(value) {
        const message = { prepopulationStatus: 0, ruleBasedUserList: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.UserListPrepopulationStatusEnum.UserListPrepopulationStatus prepopulation_status */ 1:
                    message.prepopulationStatus = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.FlexibleRuleUserListInfo flexible_rule_user_list */ 5:
                    message.flexibleRuleUserList = exports.FlexibleRuleUserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.flexibleRuleUserList);
                    break;
                case /* google.ads.googleads.v11.common.CombinedRuleUserListInfo combined_rule_user_list */ 2:
                    message.ruleBasedUserList = {
                        oneofKind: "combinedRuleUserList",
                        combinedRuleUserList: exports.CombinedRuleUserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.ruleBasedUserList.combinedRuleUserList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ExpressionRuleUserListInfo expression_rule_user_list */ 4:
                    message.ruleBasedUserList = {
                        oneofKind: "expressionRuleUserList",
                        expressionRuleUserList: exports.ExpressionRuleUserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.ruleBasedUserList.expressionRuleUserList)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.UserListPrepopulationStatusEnum.UserListPrepopulationStatus prepopulation_status = 1; */
        if (message.prepopulationStatus !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.prepopulationStatus);
        /* google.ads.googleads.v11.common.FlexibleRuleUserListInfo flexible_rule_user_list = 5; */
        if (message.flexibleRuleUserList)
            exports.FlexibleRuleUserListInfo.internalBinaryWrite(message.flexibleRuleUserList, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CombinedRuleUserListInfo combined_rule_user_list = 2; */
        if (message.ruleBasedUserList.oneofKind === "combinedRuleUserList")
            exports.CombinedRuleUserListInfo.internalBinaryWrite(message.ruleBasedUserList.combinedRuleUserList, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ExpressionRuleUserListInfo expression_rule_user_list = 4; */
        if (message.ruleBasedUserList.oneofKind === "expressionRuleUserList")
            exports.ExpressionRuleUserListInfo.internalBinaryWrite(message.ruleBasedUserList.expressionRuleUserList, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.RuleBasedUserListInfo
 */
exports.RuleBasedUserListInfo = new RuleBasedUserListInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LogicalUserListInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LogicalUserListInfo", [
            { no: 1, name: "rules", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.UserListLogicalRuleInfo }
        ]);
    }
    create(value) {
        const message = { rules: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.UserListLogicalRuleInfo rules */ 1:
                    message.rules.push(exports.UserListLogicalRuleInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.UserListLogicalRuleInfo rules = 1; */
        for (let i = 0; i < message.rules.length; i++)
            exports.UserListLogicalRuleInfo.internalBinaryWrite(message.rules[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LogicalUserListInfo
 */
exports.LogicalUserListInfo = new LogicalUserListInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListLogicalRuleInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListLogicalRuleInfo", [
            { no: 1, name: "operator", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListLogicalRuleOperatorEnum.UserListLogicalRuleOperator", user_list_logical_rule_operator_1.UserListLogicalRuleOperatorEnum_UserListLogicalRuleOperator] },
            { no: 2, name: "rule_operands", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.LogicalUserListOperandInfo }
        ]);
    }
    create(value) {
        const message = { operator: 0, ruleOperands: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.UserListLogicalRuleOperatorEnum.UserListLogicalRuleOperator operator */ 1:
                    message.operator = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.common.LogicalUserListOperandInfo rule_operands */ 2:
                    message.ruleOperands.push(exports.LogicalUserListOperandInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.UserListLogicalRuleOperatorEnum.UserListLogicalRuleOperator operator = 1; */
        if (message.operator !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.operator);
        /* repeated google.ads.googleads.v11.common.LogicalUserListOperandInfo rule_operands = 2; */
        for (let i = 0; i < message.ruleOperands.length; i++)
            exports.LogicalUserListOperandInfo.internalBinaryWrite(message.ruleOperands[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListLogicalRuleInfo
 */
exports.UserListLogicalRuleInfo = new UserListLogicalRuleInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LogicalUserListOperandInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.LogicalUserListOperandInfo", [
            { no: 2, name: "user_list", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string user_list */ 2:
                    message.userList = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string user_list = 2; */
        if (message.userList !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.userList);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.LogicalUserListOperandInfo
 */
exports.LogicalUserListOperandInfo = new LogicalUserListOperandInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class BasicUserListInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.BasicUserListInfo", [
            { no: 1, name: "actions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.UserListActionInfo }
        ]);
    }
    create(value) {
        const message = { actions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.UserListActionInfo actions */ 1:
                    message.actions.push(exports.UserListActionInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.UserListActionInfo actions = 1; */
        for (let i = 0; i < message.actions.length; i++)
            exports.UserListActionInfo.internalBinaryWrite(message.actions[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.BasicUserListInfo
 */
exports.BasicUserListInfo = new BasicUserListInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UserListActionInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.common.UserListActionInfo", [
            { no: 3, name: "conversion_action", kind: "scalar", oneof: "userListAction", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "remarketing_action", kind: "scalar", oneof: "userListAction", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { userListAction: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string conversion_action */ 3:
                    message.userListAction = {
                        oneofKind: "conversionAction",
                        conversionAction: reader.string()
                    };
                    break;
                case /* string remarketing_action */ 4:
                    message.userListAction = {
                        oneofKind: "remarketingAction",
                        remarketingAction: reader.string()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string conversion_action = 3; */
        if (message.userListAction.oneofKind === "conversionAction")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.userListAction.conversionAction);
        /* string remarketing_action = 4; */
        if (message.userListAction.oneofKind === "remarketingAction")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.userListAction.remarketingAction);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.common.UserListActionInfo
 */
exports.UserListActionInfo = new UserListActionInfo$Type();
