// google-ads-api-client/dist/protos/bindings/ads/resources/account_link.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleAdsLinkIdentifier = exports.HotelCenterLinkIdentifier = exports.DataPartnerLinkIdentifier = exports.ThirdPartyAppAnalyticsLinkIdentifier = exports.AccountLink = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const mobile_app_vendor_1 = require("../enums/mobile_app_vendor");
const linked_account_type_1 = require("../enums/linked_account_type");
const account_link_status_1 = require("../enums/account_link_status");
// @generated message type with reflection information, may provide speed optimized methods
class AccountLink$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AccountLink", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AccountLink" } } },
            { no: 8, name: "account_link_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AccountLinkStatusEnum.AccountLinkStatus", account_link_status_1.AccountLinkStatusEnum_AccountLinkStatus] },
            { no: 4, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.LinkedAccountTypeEnum.LinkedAccountType", linked_account_type_1.LinkedAccountTypeEnum_LinkedAccountType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "third_party_app_analytics", kind: "message", oneof: "linkedAccount", T: () => exports.ThirdPartyAppAnalyticsLinkIdentifier, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "data_partner", kind: "message", oneof: "linkedAccount", T: () => exports.DataPartnerLinkIdentifier, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "google_ads", kind: "message", oneof: "linkedAccount", T: () => exports.GoogleAdsLinkIdentifier, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "hotel_center", kind: "message", oneof: "linkedAccount", T: () => exports.HotelCenterLinkIdentifier, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/AccountLink", pattern: ["customers/{customer_id}/accountLinks/{account_link_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", status: 0, type: 0, linkedAccount: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 account_link_id */ 8:
                    message.accountLinkId = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.AccountLinkStatusEnum.AccountLinkStatus status */ 3:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.LinkedAccountTypeEnum.LinkedAccountType type */ 4:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.ThirdPartyAppAnalyticsLinkIdentifier third_party_app_analytics */ 5:
                    message.linkedAccount = {
                        oneofKind: "thirdPartyAppAnalytics",
                        thirdPartyAppAnalytics: exports.ThirdPartyAppAnalyticsLinkIdentifier.internalBinaryRead(reader, reader.uint32(), options, message.linkedAccount.thirdPartyAppAnalytics)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.DataPartnerLinkIdentifier data_partner */ 6:
                    message.linkedAccount = {
                        oneofKind: "dataPartner",
                        dataPartner: exports.DataPartnerLinkIdentifier.internalBinaryRead(reader, reader.uint32(), options, message.linkedAccount.dataPartner)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.GoogleAdsLinkIdentifier google_ads */ 7:
                    message.linkedAccount = {
                        oneofKind: "googleAds",
                        googleAds: exports.GoogleAdsLinkIdentifier.internalBinaryRead(reader, reader.uint32(), options, message.linkedAccount.googleAds)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.HotelCenterLinkIdentifier hotel_center */ 9:
                    message.linkedAccount = {
                        oneofKind: "hotelCenter",
                        hotelCenter: exports.HotelCenterLinkIdentifier.internalBinaryRead(reader, reader.uint32(), options, message.linkedAccount.hotelCenter)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 account_link_id = 8; */
        if (message.accountLinkId !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).int64(message.accountLinkId);
        /* google.ads.googleads.v11.enums.AccountLinkStatusEnum.AccountLinkStatus status = 3; */
        if (message.status !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.enums.LinkedAccountTypeEnum.LinkedAccountType type = 4; */
        if (message.type !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.resources.ThirdPartyAppAnalyticsLinkIdentifier third_party_app_analytics = 5; */
        if (message.linkedAccount.oneofKind === "thirdPartyAppAnalytics")
            exports.ThirdPartyAppAnalyticsLinkIdentifier.internalBinaryWrite(message.linkedAccount.thirdPartyAppAnalytics, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.DataPartnerLinkIdentifier data_partner = 6; */
        if (message.linkedAccount.oneofKind === "dataPartner")
            exports.DataPartnerLinkIdentifier.internalBinaryWrite(message.linkedAccount.dataPartner, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.GoogleAdsLinkIdentifier google_ads = 7; */
        if (message.linkedAccount.oneofKind === "googleAds")
            exports.GoogleAdsLinkIdentifier.internalBinaryWrite(message.linkedAccount.googleAds, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.HotelCenterLinkIdentifier hotel_center = 9; */
        if (message.linkedAccount.oneofKind === "hotelCenter")
            exports.HotelCenterLinkIdentifier.internalBinaryWrite(message.linkedAccount.hotelCenter, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AccountLink
 */
exports.AccountLink = new AccountLink$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThirdPartyAppAnalyticsLinkIdentifier$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ThirdPartyAppAnalyticsLinkIdentifier", [
            { no: 4, name: "app_analytics_provider_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 5, name: "app_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 3, name: "app_vendor", kind: "enum", T: () => ["google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor", mobile_app_vendor_1.MobileAppVendorEnum_MobileAppVendor], options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ]);
    }
    create(value) {
        const message = { appVendor: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 app_analytics_provider_id */ 4:
                    message.appAnalyticsProviderId = reader.int64().toBigInt();
                    break;
                case /* optional string app_id */ 5:
                    message.appId = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor app_vendor */ 3:
                    message.appVendor = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 app_analytics_provider_id = 4; */
        if (message.appAnalyticsProviderId !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).int64(message.appAnalyticsProviderId);
        /* optional string app_id = 5; */
        if (message.appId !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.appId);
        /* google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor app_vendor = 3; */
        if (message.appVendor !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.appVendor);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ThirdPartyAppAnalyticsLinkIdentifier
 */
exports.ThirdPartyAppAnalyticsLinkIdentifier = new ThirdPartyAppAnalyticsLinkIdentifier$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DataPartnerLinkIdentifier$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.DataPartnerLinkIdentifier", [
            { no: 1, name: "data_partner_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 data_partner_id */ 1:
                    message.dataPartnerId = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 data_partner_id = 1; */
        if (message.dataPartnerId !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.dataPartnerId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.DataPartnerLinkIdentifier
 */
exports.DataPartnerLinkIdentifier = new DataPartnerLinkIdentifier$Type();
// @generated message type with reflection information, may provide speed optimized methods
class HotelCenterLinkIdentifier$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.HotelCenterLinkIdentifier", [
            { no: 1, name: "hotel_center_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { hotelCenterId: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int64 hotel_center_id */ 1:
                    message.hotelCenterId = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int64 hotel_center_id = 1; */
        if (message.hotelCenterId !== 0n)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.hotelCenterId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.HotelCenterLinkIdentifier
 */
exports.HotelCenterLinkIdentifier = new HotelCenterLinkIdentifier$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GoogleAdsLinkIdentifier$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.GoogleAdsLinkIdentifier", [
            { no: 3, name: "customer", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Customer" } } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string customer */ 3:
                    message.customer = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string customer = 3; */
        if (message.customer !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.customer);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.GoogleAdsLinkIdentifier
 */
exports.GoogleAdsLinkIdentifier = new GoogleAdsLinkIdentifier$Type();
