// google-ads-api-client/dist/protos/bindings/ads/resources/account_budget.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountBudget_PendingAccountBudgetProposal = exports.AccountBudget = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const account_budget_proposal_type_1 = require("../enums/account_budget_proposal_type");
const spending_limit_type_1 = require("../enums/spending_limit_type");
const time_type_1 = require("../enums/time_type");
const account_budget_status_1 = require("../enums/account_budget_status");
// @generated message type with reflection information, may provide speed optimized methods
class AccountBudget$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AccountBudget", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AccountBudget" } } },
            { no: 23, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 24, name: "billing_setup", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/BillingSetup" } } },
            { no: 4, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AccountBudgetStatusEnum.AccountBudgetStatus", account_budget_status_1.AccountBudgetStatusEnum_AccountBudgetStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 25, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 26, name: "proposed_start_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 27, name: "approved_start_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 33, name: "total_adjustments_micros", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 34, name: "amount_served_micros", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 35, name: "purchase_order_number", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 36, name: "notes", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 22, name: "pending_proposal", kind: "message", T: () => exports.AccountBudget_PendingAccountBudgetProposal, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 28, name: "proposed_end_date_time", kind: "scalar", oneof: "proposedEndTime", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "proposed_end_time_type", kind: "enum", oneof: "proposedEndTime", T: () => ["google.ads.googleads.v11.enums.TimeTypeEnum.TimeType", time_type_1.TimeTypeEnum_TimeType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 29, name: "approved_end_date_time", kind: "scalar", oneof: "approvedEndTime", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "approved_end_time_type", kind: "enum", oneof: "approvedEndTime", T: () => ["google.ads.googleads.v11.enums.TimeTypeEnum.TimeType", time_type_1.TimeTypeEnum_TimeType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 30, name: "proposed_spending_limit_micros", kind: "scalar", oneof: "proposedSpendingLimit", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "proposed_spending_limit_type", kind: "enum", oneof: "proposedSpendingLimit", T: () => ["google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType", spending_limit_type_1.SpendingLimitTypeEnum_SpendingLimitType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 31, name: "approved_spending_limit_micros", kind: "scalar", oneof: "approvedSpendingLimit", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 15, name: "approved_spending_limit_type", kind: "enum", oneof: "approvedSpendingLimit", T: () => ["google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType", spending_limit_type_1.SpendingLimitTypeEnum_SpendingLimitType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 32, name: "adjusted_spending_limit_micros", kind: "scalar", oneof: "adjustedSpendingLimit", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 17, name: "adjusted_spending_limit_type", kind: "enum", oneof: "adjustedSpendingLimit", T: () => ["google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType", spending_limit_type_1.SpendingLimitTypeEnum_SpendingLimitType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/AccountBudget", pattern: ["customers/{customer_id}/accountBudgets/{account_budget_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", status: 0, totalAdjustmentsMicros: 0n, amountServedMicros: 0n, proposedEndTime: { oneofKind: undefined }, approvedEndTime: { oneofKind: undefined }, proposedSpendingLimit: { oneofKind: undefined }, approvedSpendingLimit: { oneofKind: undefined }, adjustedSpendingLimit: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 23:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string billing_setup */ 24:
                    message.billingSetup = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AccountBudgetStatusEnum.AccountBudgetStatus status */ 4:
                    message.status = reader.int32();
                    break;
                case /* optional string name */ 25:
                    message.name = reader.string();
                    break;
                case /* optional string proposed_start_date_time */ 26:
                    message.proposedStartDateTime = reader.string();
                    break;
                case /* optional string approved_start_date_time */ 27:
                    message.approvedStartDateTime = reader.string();
                    break;
                case /* int64 total_adjustments_micros */ 33:
                    message.totalAdjustmentsMicros = reader.int64().toBigInt();
                    break;
                case /* int64 amount_served_micros */ 34:
                    message.amountServedMicros = reader.int64().toBigInt();
                    break;
                case /* optional string purchase_order_number */ 35:
                    message.purchaseOrderNumber = reader.string();
                    break;
                case /* optional string notes */ 36:
                    message.notes = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.AccountBudget.PendingAccountBudgetProposal pending_proposal */ 22:
                    message.pendingProposal = exports.AccountBudget_PendingAccountBudgetProposal.internalBinaryRead(reader, reader.uint32(), options, message.pendingProposal);
                    break;
                case /* string proposed_end_date_time */ 28:
                    message.proposedEndTime = {
                        oneofKind: "proposedEndDateTime",
                        proposedEndDateTime: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.TimeTypeEnum.TimeType proposed_end_time_type */ 9:
                    message.proposedEndTime = {
                        oneofKind: "proposedEndTimeType",
                        proposedEndTimeType: reader.int32()
                    };
                    break;
                case /* string approved_end_date_time */ 29:
                    message.approvedEndTime = {
                        oneofKind: "approvedEndDateTime",
                        approvedEndDateTime: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.TimeTypeEnum.TimeType approved_end_time_type */ 11:
                    message.approvedEndTime = {
                        oneofKind: "approvedEndTimeType",
                        approvedEndTimeType: reader.int32()
                    };
                    break;
                case /* int64 proposed_spending_limit_micros */ 30:
                    message.proposedSpendingLimit = {
                        oneofKind: "proposedSpendingLimitMicros",
                        proposedSpendingLimitMicros: reader.int64().toBigInt()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType proposed_spending_limit_type */ 13:
                    message.proposedSpendingLimit = {
                        oneofKind: "proposedSpendingLimitType",
                        proposedSpendingLimitType: reader.int32()
                    };
                    break;
                case /* int64 approved_spending_limit_micros */ 31:
                    message.approvedSpendingLimit = {
                        oneofKind: "approvedSpendingLimitMicros",
                        approvedSpendingLimitMicros: reader.int64().toBigInt()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType approved_spending_limit_type */ 15:
                    message.approvedSpendingLimit = {
                        oneofKind: "approvedSpendingLimitType",
                        approvedSpendingLimitType: reader.int32()
                    };
                    break;
                case /* int64 adjusted_spending_limit_micros */ 32:
                    message.adjustedSpendingLimit = {
                        oneofKind: "adjustedSpendingLimitMicros",
                        adjustedSpendingLimitMicros: reader.int64().toBigInt()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType adjusted_spending_limit_type */ 17:
                    message.adjustedSpendingLimit = {
                        oneofKind: "adjustedSpendingLimitType",
                        adjustedSpendingLimitType: reader.int32()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 23; */
        if (message.id !== undefined)
            writer.tag(23, runtime_1.WireType.Varint).int64(message.id);
        /* optional string billing_setup = 24; */
        if (message.billingSetup !== undefined)
            writer.tag(24, runtime_1.WireType.LengthDelimited).string(message.billingSetup);
        /* google.ads.googleads.v11.enums.AccountBudgetStatusEnum.AccountBudgetStatus status = 4; */
        if (message.status !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.status);
        /* optional string name = 25; */
        if (message.name !== undefined)
            writer.tag(25, runtime_1.WireType.LengthDelimited).string(message.name);
        /* optional string proposed_start_date_time = 26; */
        if (message.proposedStartDateTime !== undefined)
            writer.tag(26, runtime_1.WireType.LengthDelimited).string(message.proposedStartDateTime);
        /* optional string approved_start_date_time = 27; */
        if (message.approvedStartDateTime !== undefined)
            writer.tag(27, runtime_1.WireType.LengthDelimited).string(message.approvedStartDateTime);
        /* int64 total_adjustments_micros = 33; */
        if (message.totalAdjustmentsMicros !== 0n)
            writer.tag(33, runtime_1.WireType.Varint).int64(message.totalAdjustmentsMicros);
        /* int64 amount_served_micros = 34; */
        if (message.amountServedMicros !== 0n)
            writer.tag(34, runtime_1.WireType.Varint).int64(message.amountServedMicros);
        /* optional string purchase_order_number = 35; */
        if (message.purchaseOrderNumber !== undefined)
            writer.tag(35, runtime_1.WireType.LengthDelimited).string(message.purchaseOrderNumber);
        /* optional string notes = 36; */
        if (message.notes !== undefined)
            writer.tag(36, runtime_1.WireType.LengthDelimited).string(message.notes);
        /* google.ads.googleads.v11.resources.AccountBudget.PendingAccountBudgetProposal pending_proposal = 22; */
        if (message.pendingProposal)
            exports.AccountBudget_PendingAccountBudgetProposal.internalBinaryWrite(message.pendingProposal, writer.tag(22, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string proposed_end_date_time = 28; */
        if (message.proposedEndTime.oneofKind === "proposedEndDateTime")
            writer.tag(28, runtime_1.WireType.LengthDelimited).string(message.proposedEndTime.proposedEndDateTime);
        /* google.ads.googleads.v11.enums.TimeTypeEnum.TimeType proposed_end_time_type = 9; */
        if (message.proposedEndTime.oneofKind === "proposedEndTimeType")
            writer.tag(9, runtime_1.WireType.Varint).int32(message.proposedEndTime.proposedEndTimeType);
        /* string approved_end_date_time = 29; */
        if (message.approvedEndTime.oneofKind === "approvedEndDateTime")
            writer.tag(29, runtime_1.WireType.LengthDelimited).string(message.approvedEndTime.approvedEndDateTime);
        /* google.ads.googleads.v11.enums.TimeTypeEnum.TimeType approved_end_time_type = 11; */
        if (message.approvedEndTime.oneofKind === "approvedEndTimeType")
            writer.tag(11, runtime_1.WireType.Varint).int32(message.approvedEndTime.approvedEndTimeType);
        /* int64 proposed_spending_limit_micros = 30; */
        if (message.proposedSpendingLimit.oneofKind === "proposedSpendingLimitMicros")
            writer.tag(30, runtime_1.WireType.Varint).int64(message.proposedSpendingLimit.proposedSpendingLimitMicros);
        /* google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType proposed_spending_limit_type = 13; */
        if (message.proposedSpendingLimit.oneofKind === "proposedSpendingLimitType")
            writer.tag(13, runtime_1.WireType.Varint).int32(message.proposedSpendingLimit.proposedSpendingLimitType);
        /* int64 approved_spending_limit_micros = 31; */
        if (message.approvedSpendingLimit.oneofKind === "approvedSpendingLimitMicros")
            writer.tag(31, runtime_1.WireType.Varint).int64(message.approvedSpendingLimit.approvedSpendingLimitMicros);
        /* google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType approved_spending_limit_type = 15; */
        if (message.approvedSpendingLimit.oneofKind === "approvedSpendingLimitType")
            writer.tag(15, runtime_1.WireType.Varint).int32(message.approvedSpendingLimit.approvedSpendingLimitType);
        /* int64 adjusted_spending_limit_micros = 32; */
        if (message.adjustedSpendingLimit.oneofKind === "adjustedSpendingLimitMicros")
            writer.tag(32, runtime_1.WireType.Varint).int64(message.adjustedSpendingLimit.adjustedSpendingLimitMicros);
        /* google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType adjusted_spending_limit_type = 17; */
        if (message.adjustedSpendingLimit.oneofKind === "adjustedSpendingLimitType")
            writer.tag(17, runtime_1.WireType.Varint).int32(message.adjustedSpendingLimit.adjustedSpendingLimitType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AccountBudget
 */
exports.AccountBudget = new AccountBudget$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AccountBudget_PendingAccountBudgetProposal$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AccountBudget.PendingAccountBudgetProposal", [
            { no: 12, name: "account_budget_proposal", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AccountBudgetProposal" } } },
            { no: 2, name: "proposal_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AccountBudgetProposalTypeEnum.AccountBudgetProposalType", account_budget_proposal_type_1.AccountBudgetProposalTypeEnum_AccountBudgetProposalType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 14, name: "start_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 17, name: "purchase_order_number", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 18, name: "notes", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 19, name: "creation_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 15, name: "end_date_time", kind: "scalar", oneof: "endTime", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "end_time_type", kind: "enum", oneof: "endTime", T: () => ["google.ads.googleads.v11.enums.TimeTypeEnum.TimeType", time_type_1.TimeTypeEnum_TimeType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 16, name: "spending_limit_micros", kind: "scalar", oneof: "spendingLimit", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "spending_limit_type", kind: "enum", oneof: "spendingLimit", T: () => ["google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType", spending_limit_type_1.SpendingLimitTypeEnum_SpendingLimitType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { proposalType: 0, endTime: { oneofKind: undefined }, spendingLimit: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string account_budget_proposal */ 12:
                    message.accountBudgetProposal = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AccountBudgetProposalTypeEnum.AccountBudgetProposalType proposal_type */ 2:
                    message.proposalType = reader.int32();
                    break;
                case /* optional string name */ 13:
                    message.name = reader.string();
                    break;
                case /* optional string start_date_time */ 14:
                    message.startDateTime = reader.string();
                    break;
                case /* optional string purchase_order_number */ 17:
                    message.purchaseOrderNumber = reader.string();
                    break;
                case /* optional string notes */ 18:
                    message.notes = reader.string();
                    break;
                case /* optional string creation_date_time */ 19:
                    message.creationDateTime = reader.string();
                    break;
                case /* string end_date_time */ 15:
                    message.endTime = {
                        oneofKind: "endDateTime",
                        endDateTime: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.TimeTypeEnum.TimeType end_time_type */ 6:
                    message.endTime = {
                        oneofKind: "endTimeType",
                        endTimeType: reader.int32()
                    };
                    break;
                case /* int64 spending_limit_micros */ 16:
                    message.spendingLimit = {
                        oneofKind: "spendingLimitMicros",
                        spendingLimitMicros: reader.int64().toBigInt()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType spending_limit_type */ 8:
                    message.spendingLimit = {
                        oneofKind: "spendingLimitType",
                        spendingLimitType: reader.int32()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string account_budget_proposal = 12; */
        if (message.accountBudgetProposal !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.accountBudgetProposal);
        /* google.ads.googleads.v11.enums.AccountBudgetProposalTypeEnum.AccountBudgetProposalType proposal_type = 2; */
        if (message.proposalType !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.proposalType);
        /* optional string name = 13; */
        if (message.name !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.name);
        /* optional string start_date_time = 14; */
        if (message.startDateTime !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.startDateTime);
        /* optional string purchase_order_number = 17; */
        if (message.purchaseOrderNumber !== undefined)
            writer.tag(17, runtime_1.WireType.LengthDelimited).string(message.purchaseOrderNumber);
        /* optional string notes = 18; */
        if (message.notes !== undefined)
            writer.tag(18, runtime_1.WireType.LengthDelimited).string(message.notes);
        /* optional string creation_date_time = 19; */
        if (message.creationDateTime !== undefined)
            writer.tag(19, runtime_1.WireType.LengthDelimited).string(message.creationDateTime);
        /* string end_date_time = 15; */
        if (message.endTime.oneofKind === "endDateTime")
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.endTime.endDateTime);
        /* google.ads.googleads.v11.enums.TimeTypeEnum.TimeType end_time_type = 6; */
        if (message.endTime.oneofKind === "endTimeType")
            writer.tag(6, runtime_1.WireType.Varint).int32(message.endTime.endTimeType);
        /* int64 spending_limit_micros = 16; */
        if (message.spendingLimit.oneofKind === "spendingLimitMicros")
            writer.tag(16, runtime_1.WireType.Varint).int64(message.spendingLimit.spendingLimitMicros);
        /* google.ads.googleads.v11.enums.SpendingLimitTypeEnum.SpendingLimitType spending_limit_type = 8; */
        if (message.spendingLimit.oneofKind === "spendingLimitType")
            writer.tag(8, runtime_1.WireType.Varint).int32(message.spendingLimit.spendingLimitType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AccountBudget.PendingAccountBudgetProposal
 */
exports.AccountBudget_PendingAccountBudgetProposal = new AccountBudget_PendingAccountBudgetProposal$Type();
