// google-ads-api-client/dist/protos/bindings/ads/resources/ad.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Ad = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const ad_type_infos_1 = require("../common/ad_type_infos");
const ad_type_infos_2 = require("../common/ad_type_infos");
const ad_type_infos_3 = require("../common/ad_type_infos");
const ad_type_infos_4 = require("../common/ad_type_infos");
const ad_type_infos_5 = require("../common/ad_type_infos");
const ad_type_infos_6 = require("../common/ad_type_infos");
const ad_type_infos_7 = require("../common/ad_type_infos");
const ad_type_infos_8 = require("../common/ad_type_infos");
const ad_type_infos_9 = require("../common/ad_type_infos");
const ad_type_infos_10 = require("../common/ad_type_infos");
const ad_type_infos_11 = require("../common/ad_type_infos");
const ad_type_infos_12 = require("../common/ad_type_infos");
const ad_type_infos_13 = require("../common/ad_type_infos");
const ad_type_infos_14 = require("../common/ad_type_infos");
const ad_type_infos_15 = require("../common/ad_type_infos");
const ad_type_infos_16 = require("../common/ad_type_infos");
const ad_type_infos_17 = require("../common/ad_type_infos");
const ad_type_infos_18 = require("../common/ad_type_infos");
const ad_type_infos_19 = require("../common/ad_type_infos");
const ad_type_infos_20 = require("../common/ad_type_infos");
const ad_type_infos_21 = require("../common/ad_type_infos");
const ad_type_infos_22 = require("../common/ad_type_infos");
const ad_type_infos_23 = require("../common/ad_type_infos");
const ad_type_infos_24 = require("../common/ad_type_infos");
const system_managed_entity_source_1 = require("../enums/system_managed_entity_source");
const url_collection_1 = require("../common/url_collection");
const device_1 = require("../enums/device");
const ad_type_1 = require("../enums/ad_type");
const custom_parameter_1 = require("../common/custom_parameter");
const final_app_url_1 = require("../common/final_app_url");
// @generated message type with reflection information, may provide speed optimized methods
class Ad$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Ad", [
            { no: 37, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Ad" } } },
            { no: 40, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 41, name: "final_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 35, name: "final_app_urls", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => final_app_url_1.FinalAppUrl },
            { no: 42, name: "final_mobile_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 43, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 44, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "url_custom_parameters", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => custom_parameter_1.CustomParameter },
            { no: 45, name: "display_url", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AdTypeEnum.AdType", ad_type_1.AdTypeEnum_AdType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 46, name: "added_by_google_ads", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 20, name: "device_preference", kind: "enum", T: () => ["google.ads.googleads.v11.enums.DeviceEnum.Device", device_1.DeviceEnum_Device] },
            { no: 26, name: "url_collections", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => url_collection_1.UrlCollection },
            { no: 47, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 27, name: "system_managed_resource_source", kind: "enum", T: () => ["google.ads.googleads.v11.enums.SystemManagedResourceSourceEnum.SystemManagedResourceSource", system_managed_entity_source_1.SystemManagedResourceSourceEnum_SystemManagedResourceSource], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "text_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_24.TextAdInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 7, name: "expanded_text_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_23.ExpandedTextAdInfo },
            { no: 49, name: "call_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_22.CallAdInfo },
            { no: 14, name: "expanded_dynamic_search_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_21.ExpandedDynamicSearchAdInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 15, name: "hotel_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_20.HotelAdInfo },
            { no: 17, name: "shopping_smart_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_19.ShoppingSmartAdInfo },
            { no: 18, name: "shopping_product_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_18.ShoppingProductAdInfo },
            { no: 21, name: "gmail_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_17.GmailAdInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 22, name: "image_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_16.ImageAdInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 24, name: "video_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_15.VideoAdInfo },
            { no: 39, name: "video_responsive_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_14.VideoResponsiveAdInfo },
            { no: 25, name: "responsive_search_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_13.ResponsiveSearchAdInfo },
            { no: 28, name: "legacy_responsive_display_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_12.LegacyResponsiveDisplayAdInfo },
            { no: 29, name: "app_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_11.AppAdInfo },
            { no: 30, name: "legacy_app_install_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_10.LegacyAppInstallAdInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 31, name: "responsive_display_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_9.ResponsiveDisplayAdInfo },
            { no: 32, name: "local_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_8.LocalAdInfo },
            { no: 33, name: "display_upload_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_7.DisplayUploadAdInfo },
            { no: 34, name: "app_engagement_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_6.AppEngagementAdInfo },
            { no: 36, name: "shopping_comparison_listing_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_5.ShoppingComparisonListingAdInfo },
            { no: 48, name: "smart_campaign_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_4.SmartCampaignAdInfo },
            { no: 50, name: "app_pre_registration_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_3.AppPreRegistrationAdInfo },
            { no: 51, name: "discovery_multi_asset_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_2.DiscoveryMultiAssetAdInfo },
            { no: 52, name: "discovery_carousel_ad", kind: "message", oneof: "adData", T: () => ad_type_infos_1.DiscoveryCarouselAdInfo }
        ], { "google.api.resource": { type: "googleads.googleapis.com/Ad", pattern: ["customers/{customer_id}/ads/{ad_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", finalUrls: [], finalAppUrls: [], finalMobileUrls: [], urlCustomParameters: [], type: 0, devicePreference: 0, urlCollections: [], systemManagedResourceSource: 0, adData: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 37:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 40:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* repeated string final_urls */ 41:
                    message.finalUrls.push(reader.string());
                    break;
                case /* repeated google.ads.googleads.v11.common.FinalAppUrl final_app_urls */ 35:
                    message.finalAppUrls.push(final_app_url_1.FinalAppUrl.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated string final_mobile_urls */ 42:
                    message.finalMobileUrls.push(reader.string());
                    break;
                case /* optional string tracking_url_template */ 43:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* optional string final_url_suffix */ 44:
                    message.finalUrlSuffix = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters */ 10:
                    message.urlCustomParameters.push(custom_parameter_1.CustomParameter.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string display_url */ 45:
                    message.displayUrl = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AdTypeEnum.AdType type */ 5:
                    message.type = reader.int32();
                    break;
                case /* optional bool added_by_google_ads */ 46:
                    message.addedByGoogleAds = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.DeviceEnum.Device device_preference */ 20:
                    message.devicePreference = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.common.UrlCollection url_collections */ 26:
                    message.urlCollections.push(url_collection_1.UrlCollection.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string name */ 47:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.SystemManagedResourceSourceEnum.SystemManagedResourceSource system_managed_resource_source */ 27:
                    message.systemManagedResourceSource = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.TextAdInfo text_ad */ 6:
                    message.adData = {
                        oneofKind: "textAd",
                        textAd: ad_type_infos_24.TextAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.textAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ExpandedTextAdInfo expanded_text_ad */ 7:
                    message.adData = {
                        oneofKind: "expandedTextAd",
                        expandedTextAd: ad_type_infos_23.ExpandedTextAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.expandedTextAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CallAdInfo call_ad */ 49:
                    message.adData = {
                        oneofKind: "callAd",
                        callAd: ad_type_infos_22.CallAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.callAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ExpandedDynamicSearchAdInfo expanded_dynamic_search_ad */ 14:
                    message.adData = {
                        oneofKind: "expandedDynamicSearchAd",
                        expandedDynamicSearchAd: ad_type_infos_21.ExpandedDynamicSearchAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.expandedDynamicSearchAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.HotelAdInfo hotel_ad */ 15:
                    message.adData = {
                        oneofKind: "hotelAd",
                        hotelAd: ad_type_infos_20.HotelAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.hotelAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ShoppingSmartAdInfo shopping_smart_ad */ 17:
                    message.adData = {
                        oneofKind: "shoppingSmartAd",
                        shoppingSmartAd: ad_type_infos_19.ShoppingSmartAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.shoppingSmartAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ShoppingProductAdInfo shopping_product_ad */ 18:
                    message.adData = {
                        oneofKind: "shoppingProductAd",
                        shoppingProductAd: ad_type_infos_18.ShoppingProductAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.shoppingProductAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.GmailAdInfo gmail_ad */ 21:
                    message.adData = {
                        oneofKind: "gmailAd",
                        gmailAd: ad_type_infos_17.GmailAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.gmailAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ImageAdInfo image_ad */ 22:
                    message.adData = {
                        oneofKind: "imageAd",
                        imageAd: ad_type_infos_16.ImageAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.imageAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.VideoAdInfo video_ad */ 24:
                    message.adData = {
                        oneofKind: "videoAd",
                        videoAd: ad_type_infos_15.VideoAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.videoAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.VideoResponsiveAdInfo video_responsive_ad */ 39:
                    message.adData = {
                        oneofKind: "videoResponsiveAd",
                        videoResponsiveAd: ad_type_infos_14.VideoResponsiveAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.videoResponsiveAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ResponsiveSearchAdInfo responsive_search_ad */ 25:
                    message.adData = {
                        oneofKind: "responsiveSearchAd",
                        responsiveSearchAd: ad_type_infos_13.ResponsiveSearchAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.responsiveSearchAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LegacyResponsiveDisplayAdInfo legacy_responsive_display_ad */ 28:
                    message.adData = {
                        oneofKind: "legacyResponsiveDisplayAd",
                        legacyResponsiveDisplayAd: ad_type_infos_12.LegacyResponsiveDisplayAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.legacyResponsiveDisplayAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AppAdInfo app_ad */ 29:
                    message.adData = {
                        oneofKind: "appAd",
                        appAd: ad_type_infos_11.AppAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.appAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LegacyAppInstallAdInfo legacy_app_install_ad */ 30:
                    message.adData = {
                        oneofKind: "legacyAppInstallAd",
                        legacyAppInstallAd: ad_type_infos_10.LegacyAppInstallAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.legacyAppInstallAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ResponsiveDisplayAdInfo responsive_display_ad */ 31:
                    message.adData = {
                        oneofKind: "responsiveDisplayAd",
                        responsiveDisplayAd: ad_type_infos_9.ResponsiveDisplayAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.responsiveDisplayAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LocalAdInfo local_ad */ 32:
                    message.adData = {
                        oneofKind: "localAd",
                        localAd: ad_type_infos_8.LocalAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.localAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DisplayUploadAdInfo display_upload_ad */ 33:
                    message.adData = {
                        oneofKind: "displayUploadAd",
                        displayUploadAd: ad_type_infos_7.DisplayUploadAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.displayUploadAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AppEngagementAdInfo app_engagement_ad */ 34:
                    message.adData = {
                        oneofKind: "appEngagementAd",
                        appEngagementAd: ad_type_infos_6.AppEngagementAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.appEngagementAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ShoppingComparisonListingAdInfo shopping_comparison_listing_ad */ 36:
                    message.adData = {
                        oneofKind: "shoppingComparisonListingAd",
                        shoppingComparisonListingAd: ad_type_infos_5.ShoppingComparisonListingAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.shoppingComparisonListingAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.SmartCampaignAdInfo smart_campaign_ad */ 48:
                    message.adData = {
                        oneofKind: "smartCampaignAd",
                        smartCampaignAd: ad_type_infos_4.SmartCampaignAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.smartCampaignAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AppPreRegistrationAdInfo app_pre_registration_ad */ 50:
                    message.adData = {
                        oneofKind: "appPreRegistrationAd",
                        appPreRegistrationAd: ad_type_infos_3.AppPreRegistrationAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.appPreRegistrationAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DiscoveryMultiAssetAdInfo discovery_multi_asset_ad */ 51:
                    message.adData = {
                        oneofKind: "discoveryMultiAssetAd",
                        discoveryMultiAssetAd: ad_type_infos_2.DiscoveryMultiAssetAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.discoveryMultiAssetAd)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DiscoveryCarouselAdInfo discovery_carousel_ad */ 52:
                    message.adData = {
                        oneofKind: "discoveryCarouselAd",
                        discoveryCarouselAd: ad_type_infos_1.DiscoveryCarouselAdInfo.internalBinaryRead(reader, reader.uint32(), options, message.adData.discoveryCarouselAd)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 37; */
        if (message.resourceName !== "")
            writer.tag(37, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 40; */
        if (message.id !== undefined)
            writer.tag(40, runtime_1.WireType.Varint).int64(message.id);
        /* repeated string final_urls = 41; */
        for (let i = 0; i < message.finalUrls.length; i++)
            writer.tag(41, runtime_1.WireType.LengthDelimited).string(message.finalUrls[i]);
        /* repeated google.ads.googleads.v11.common.FinalAppUrl final_app_urls = 35; */
        for (let i = 0; i < message.finalAppUrls.length; i++)
            final_app_url_1.FinalAppUrl.internalBinaryWrite(message.finalAppUrls[i], writer.tag(35, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated string final_mobile_urls = 42; */
        for (let i = 0; i < message.finalMobileUrls.length; i++)
            writer.tag(42, runtime_1.WireType.LengthDelimited).string(message.finalMobileUrls[i]);
        /* optional string tracking_url_template = 43; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(43, runtime_1.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* optional string final_url_suffix = 44; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(44, runtime_1.WireType.LengthDelimited).string(message.finalUrlSuffix);
        /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters = 10; */
        for (let i = 0; i < message.urlCustomParameters.length; i++)
            custom_parameter_1.CustomParameter.internalBinaryWrite(message.urlCustomParameters[i], writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string display_url = 45; */
        if (message.displayUrl !== undefined)
            writer.tag(45, runtime_1.WireType.LengthDelimited).string(message.displayUrl);
        /* google.ads.googleads.v11.enums.AdTypeEnum.AdType type = 5; */
        if (message.type !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.type);
        /* optional bool added_by_google_ads = 46; */
        if (message.addedByGoogleAds !== undefined)
            writer.tag(46, runtime_1.WireType.Varint).bool(message.addedByGoogleAds);
        /* google.ads.googleads.v11.enums.DeviceEnum.Device device_preference = 20; */
        if (message.devicePreference !== 0)
            writer.tag(20, runtime_1.WireType.Varint).int32(message.devicePreference);
        /* repeated google.ads.googleads.v11.common.UrlCollection url_collections = 26; */
        for (let i = 0; i < message.urlCollections.length; i++)
            url_collection_1.UrlCollection.internalBinaryWrite(message.urlCollections[i], writer.tag(26, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string name = 47; */
        if (message.name !== undefined)
            writer.tag(47, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.enums.SystemManagedResourceSourceEnum.SystemManagedResourceSource system_managed_resource_source = 27; */
        if (message.systemManagedResourceSource !== 0)
            writer.tag(27, runtime_1.WireType.Varint).int32(message.systemManagedResourceSource);
        /* google.ads.googleads.v11.common.TextAdInfo text_ad = 6; */
        if (message.adData.oneofKind === "textAd")
            ad_type_infos_24.TextAdInfo.internalBinaryWrite(message.adData.textAd, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ExpandedTextAdInfo expanded_text_ad = 7; */
        if (message.adData.oneofKind === "expandedTextAd")
            ad_type_infos_23.ExpandedTextAdInfo.internalBinaryWrite(message.adData.expandedTextAd, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CallAdInfo call_ad = 49; */
        if (message.adData.oneofKind === "callAd")
            ad_type_infos_22.CallAdInfo.internalBinaryWrite(message.adData.callAd, writer.tag(49, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ExpandedDynamicSearchAdInfo expanded_dynamic_search_ad = 14; */
        if (message.adData.oneofKind === "expandedDynamicSearchAd")
            ad_type_infos_21.ExpandedDynamicSearchAdInfo.internalBinaryWrite(message.adData.expandedDynamicSearchAd, writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HotelAdInfo hotel_ad = 15; */
        if (message.adData.oneofKind === "hotelAd")
            ad_type_infos_20.HotelAdInfo.internalBinaryWrite(message.adData.hotelAd, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ShoppingSmartAdInfo shopping_smart_ad = 17; */
        if (message.adData.oneofKind === "shoppingSmartAd")
            ad_type_infos_19.ShoppingSmartAdInfo.internalBinaryWrite(message.adData.shoppingSmartAd, writer.tag(17, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ShoppingProductAdInfo shopping_product_ad = 18; */
        if (message.adData.oneofKind === "shoppingProductAd")
            ad_type_infos_18.ShoppingProductAdInfo.internalBinaryWrite(message.adData.shoppingProductAd, writer.tag(18, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.GmailAdInfo gmail_ad = 21; */
        if (message.adData.oneofKind === "gmailAd")
            ad_type_infos_17.GmailAdInfo.internalBinaryWrite(message.adData.gmailAd, writer.tag(21, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ImageAdInfo image_ad = 22; */
        if (message.adData.oneofKind === "imageAd")
            ad_type_infos_16.ImageAdInfo.internalBinaryWrite(message.adData.imageAd, writer.tag(22, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.VideoAdInfo video_ad = 24; */
        if (message.adData.oneofKind === "videoAd")
            ad_type_infos_15.VideoAdInfo.internalBinaryWrite(message.adData.videoAd, writer.tag(24, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.VideoResponsiveAdInfo video_responsive_ad = 39; */
        if (message.adData.oneofKind === "videoResponsiveAd")
            ad_type_infos_14.VideoResponsiveAdInfo.internalBinaryWrite(message.adData.videoResponsiveAd, writer.tag(39, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ResponsiveSearchAdInfo responsive_search_ad = 25; */
        if (message.adData.oneofKind === "responsiveSearchAd")
            ad_type_infos_13.ResponsiveSearchAdInfo.internalBinaryWrite(message.adData.responsiveSearchAd, writer.tag(25, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LegacyResponsiveDisplayAdInfo legacy_responsive_display_ad = 28; */
        if (message.adData.oneofKind === "legacyResponsiveDisplayAd")
            ad_type_infos_12.LegacyResponsiveDisplayAdInfo.internalBinaryWrite(message.adData.legacyResponsiveDisplayAd, writer.tag(28, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AppAdInfo app_ad = 29; */
        if (message.adData.oneofKind === "appAd")
            ad_type_infos_11.AppAdInfo.internalBinaryWrite(message.adData.appAd, writer.tag(29, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LegacyAppInstallAdInfo legacy_app_install_ad = 30; */
        if (message.adData.oneofKind === "legacyAppInstallAd")
            ad_type_infos_10.LegacyAppInstallAdInfo.internalBinaryWrite(message.adData.legacyAppInstallAd, writer.tag(30, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ResponsiveDisplayAdInfo responsive_display_ad = 31; */
        if (message.adData.oneofKind === "responsiveDisplayAd")
            ad_type_infos_9.ResponsiveDisplayAdInfo.internalBinaryWrite(message.adData.responsiveDisplayAd, writer.tag(31, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LocalAdInfo local_ad = 32; */
        if (message.adData.oneofKind === "localAd")
            ad_type_infos_8.LocalAdInfo.internalBinaryWrite(message.adData.localAd, writer.tag(32, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DisplayUploadAdInfo display_upload_ad = 33; */
        if (message.adData.oneofKind === "displayUploadAd")
            ad_type_infos_7.DisplayUploadAdInfo.internalBinaryWrite(message.adData.displayUploadAd, writer.tag(33, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AppEngagementAdInfo app_engagement_ad = 34; */
        if (message.adData.oneofKind === "appEngagementAd")
            ad_type_infos_6.AppEngagementAdInfo.internalBinaryWrite(message.adData.appEngagementAd, writer.tag(34, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ShoppingComparisonListingAdInfo shopping_comparison_listing_ad = 36; */
        if (message.adData.oneofKind === "shoppingComparisonListingAd")
            ad_type_infos_5.ShoppingComparisonListingAdInfo.internalBinaryWrite(message.adData.shoppingComparisonListingAd, writer.tag(36, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.SmartCampaignAdInfo smart_campaign_ad = 48; */
        if (message.adData.oneofKind === "smartCampaignAd")
            ad_type_infos_4.SmartCampaignAdInfo.internalBinaryWrite(message.adData.smartCampaignAd, writer.tag(48, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AppPreRegistrationAdInfo app_pre_registration_ad = 50; */
        if (message.adData.oneofKind === "appPreRegistrationAd")
            ad_type_infos_3.AppPreRegistrationAdInfo.internalBinaryWrite(message.adData.appPreRegistrationAd, writer.tag(50, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DiscoveryMultiAssetAdInfo discovery_multi_asset_ad = 51; */
        if (message.adData.oneofKind === "discoveryMultiAssetAd")
            ad_type_infos_2.DiscoveryMultiAssetAdInfo.internalBinaryWrite(message.adData.discoveryMultiAssetAd, writer.tag(51, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DiscoveryCarouselAdInfo discovery_carousel_ad = 52; */
        if (message.adData.oneofKind === "discoveryCarouselAd")
            ad_type_infos_1.DiscoveryCarouselAdInfo.internalBinaryWrite(message.adData.discoveryCarouselAd, writer.tag(52, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Ad
 */
exports.Ad = new Ad$Type();
