// google-ads-api-client/dist/protos/bindings/ads/resources/ad_group_bid_modifier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupBidModifier = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const criteria_1 = require("../common/criteria");
const criteria_2 = require("../common/criteria");
const criteria_3 = require("../common/criteria");
const criteria_4 = require("../common/criteria");
const criteria_5 = require("../common/criteria");
const criteria_6 = require("../common/criteria");
const criteria_7 = require("../common/criteria");
const bid_modifier_source_1 = require("../enums/bid_modifier_source");
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupBidModifier$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AdGroupBidModifier", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupBidModifier" } } },
            { no: 13, name: "ad_group", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } },
            { no: 14, name: "criterion_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 15, name: "bid_modifier", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 16, name: "base_ad_group", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } },
            { no: 10, name: "bid_modifier_source", kind: "enum", T: () => ["google.ads.googleads.v11.enums.BidModifierSourceEnum.BidModifierSource", bid_modifier_source_1.BidModifierSourceEnum_BidModifierSource], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "hotel_date_selection_type", kind: "message", oneof: "criterion", T: () => criteria_7.HotelDateSelectionTypeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "hotel_advance_booking_window", kind: "message", oneof: "criterion", T: () => criteria_6.HotelAdvanceBookingWindowInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 7, name: "hotel_length_of_stay", kind: "message", oneof: "criterion", T: () => criteria_5.HotelLengthOfStayInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 8, name: "hotel_check_in_day", kind: "message", oneof: "criterion", T: () => criteria_4.HotelCheckInDayInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 11, name: "device", kind: "message", oneof: "criterion", T: () => criteria_3.DeviceInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 12, name: "preferred_content", kind: "message", oneof: "criterion", T: () => criteria_2.PreferredContentInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 17, name: "hotel_check_in_date_range", kind: "message", oneof: "criterion", T: () => criteria_1.HotelCheckInDateRangeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/AdGroupBidModifier", pattern: ["customers/{customer_id}/adGroupBidModifiers/{ad_group_id}~{criterion_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", bidModifierSource: 0, criterion: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional string ad_group */ 13:
                    message.adGroup = reader.string();
                    break;
                case /* optional int64 criterion_id */ 14:
                    message.criterionId = reader.int64().toBigInt();
                    break;
                case /* optional double bid_modifier */ 15:
                    message.bidModifier = reader.double();
                    break;
                case /* optional string base_ad_group */ 16:
                    message.baseAdGroup = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.BidModifierSourceEnum.BidModifierSource bid_modifier_source */ 10:
                    message.bidModifierSource = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.HotelDateSelectionTypeInfo hotel_date_selection_type */ 5:
                    message.criterion = {
                        oneofKind: "hotelDateSelectionType",
                        hotelDateSelectionType: criteria_7.HotelDateSelectionTypeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.hotelDateSelectionType)
                    };
                    break;
                case /* google.ads.googleads.v11.common.HotelAdvanceBookingWindowInfo hotel_advance_booking_window */ 6:
                    message.criterion = {
                        oneofKind: "hotelAdvanceBookingWindow",
                        hotelAdvanceBookingWindow: criteria_6.HotelAdvanceBookingWindowInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.hotelAdvanceBookingWindow)
                    };
                    break;
                case /* google.ads.googleads.v11.common.HotelLengthOfStayInfo hotel_length_of_stay */ 7:
                    message.criterion = {
                        oneofKind: "hotelLengthOfStay",
                        hotelLengthOfStay: criteria_5.HotelLengthOfStayInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.hotelLengthOfStay)
                    };
                    break;
                case /* google.ads.googleads.v11.common.HotelCheckInDayInfo hotel_check_in_day */ 8:
                    message.criterion = {
                        oneofKind: "hotelCheckInDay",
                        hotelCheckInDay: criteria_4.HotelCheckInDayInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.hotelCheckInDay)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DeviceInfo device */ 11:
                    message.criterion = {
                        oneofKind: "device",
                        device: criteria_3.DeviceInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.device)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PreferredContentInfo preferred_content */ 12:
                    message.criterion = {
                        oneofKind: "preferredContent",
                        preferredContent: criteria_2.PreferredContentInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.preferredContent)
                    };
                    break;
                case /* google.ads.googleads.v11.common.HotelCheckInDateRangeInfo hotel_check_in_date_range */ 17:
                    message.criterion = {
                        oneofKind: "hotelCheckInDateRange",
                        hotelCheckInDateRange: criteria_1.HotelCheckInDateRangeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.hotelCheckInDateRange)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional string ad_group = 13; */
        if (message.adGroup !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.adGroup);
        /* optional int64 criterion_id = 14; */
        if (message.criterionId !== undefined)
            writer.tag(14, runtime_1.WireType.Varint).int64(message.criterionId);
        /* optional double bid_modifier = 15; */
        if (message.bidModifier !== undefined)
            writer.tag(15, runtime_1.WireType.Bit64).double(message.bidModifier);
        /* optional string base_ad_group = 16; */
        if (message.baseAdGroup !== undefined)
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.baseAdGroup);
        /* google.ads.googleads.v11.enums.BidModifierSourceEnum.BidModifierSource bid_modifier_source = 10; */
        if (message.bidModifierSource !== 0)
            writer.tag(10, runtime_1.WireType.Varint).int32(message.bidModifierSource);
        /* google.ads.googleads.v11.common.HotelDateSelectionTypeInfo hotel_date_selection_type = 5; */
        if (message.criterion.oneofKind === "hotelDateSelectionType")
            criteria_7.HotelDateSelectionTypeInfo.internalBinaryWrite(message.criterion.hotelDateSelectionType, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HotelAdvanceBookingWindowInfo hotel_advance_booking_window = 6; */
        if (message.criterion.oneofKind === "hotelAdvanceBookingWindow")
            criteria_6.HotelAdvanceBookingWindowInfo.internalBinaryWrite(message.criterion.hotelAdvanceBookingWindow, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HotelLengthOfStayInfo hotel_length_of_stay = 7; */
        if (message.criterion.oneofKind === "hotelLengthOfStay")
            criteria_5.HotelLengthOfStayInfo.internalBinaryWrite(message.criterion.hotelLengthOfStay, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HotelCheckInDayInfo hotel_check_in_day = 8; */
        if (message.criterion.oneofKind === "hotelCheckInDay")
            criteria_4.HotelCheckInDayInfo.internalBinaryWrite(message.criterion.hotelCheckInDay, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DeviceInfo device = 11; */
        if (message.criterion.oneofKind === "device")
            criteria_3.DeviceInfo.internalBinaryWrite(message.criterion.device, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PreferredContentInfo preferred_content = 12; */
        if (message.criterion.oneofKind === "preferredContent")
            criteria_2.PreferredContentInfo.internalBinaryWrite(message.criterion.preferredContent, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HotelCheckInDateRangeInfo hotel_check_in_date_range = 17; */
        if (message.criterion.oneofKind === "hotelCheckInDateRange")
            criteria_1.HotelCheckInDateRangeInfo.internalBinaryWrite(message.criterion.hotelCheckInDateRange, writer.tag(17, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AdGroupBidModifier
 */
exports.AdGroupBidModifier = new AdGroupBidModifier$Type();
