// google-ads-api-client/dist/protos/bindings/ads/resources/ad_group_criterion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupCriterion_PositionEstimates = exports.AdGroupCriterion_QualityInfo = exports.AdGroupCriterion = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const quality_score_bucket_1 = require("../enums/quality_score_bucket");
const criteria_1 = require("../common/criteria");
const criteria_2 = require("../common/criteria");
const criteria_3 = require("../common/criteria");
const criteria_4 = require("../common/criteria");
const criteria_5 = require("../common/criteria");
const criteria_6 = require("../common/criteria");
const criteria_7 = require("../common/criteria");
const criteria_8 = require("../common/criteria");
const criteria_9 = require("../common/criteria");
const criteria_10 = require("../common/criteria");
const criteria_11 = require("../common/criteria");
const criteria_12 = require("../common/criteria");
const criteria_13 = require("../common/criteria");
const criteria_14 = require("../common/criteria");
const criteria_15 = require("../common/criteria");
const criteria_16 = require("../common/criteria");
const criteria_17 = require("../common/criteria");
const criteria_18 = require("../common/criteria");
const criteria_19 = require("../common/criteria");
const criteria_20 = require("../common/criteria");
const criteria_21 = require("../common/criteria");
const custom_parameter_1 = require("../common/custom_parameter");
const bidding_source_1 = require("../enums/bidding_source");
const ad_group_criterion_approval_status_1 = require("../enums/ad_group_criterion_approval_status");
const criterion_system_serving_status_1 = require("../enums/criterion_system_serving_status");
const criterion_type_1 = require("../enums/criterion_type");
const ad_group_criterion_status_1 = require("../enums/ad_group_criterion_status");
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupCriterion$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AdGroupCriterion", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupCriterion" } } },
            { no: 56, name: "criterion_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 77, name: "display_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AdGroupCriterionStatusEnum.AdGroupCriterionStatus", ad_group_criterion_status_1.AdGroupCriterionStatusEnum_AdGroupCriterionStatus] },
            { no: 4, name: "quality_info", kind: "message", T: () => exports.AdGroupCriterion_QualityInfo, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 57, name: "ad_group", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } },
            { no: 25, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CriterionTypeEnum.CriterionType", criterion_type_1.CriterionTypeEnum_CriterionType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 58, name: "negative", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 52, name: "system_serving_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CriterionSystemServingStatusEnum.CriterionSystemServingStatus", criterion_system_serving_status_1.CriterionSystemServingStatusEnum_CriterionSystemServingStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 53, name: "approval_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AdGroupCriterionApprovalStatusEnum.AdGroupCriterionApprovalStatus", ad_group_criterion_approval_status_1.AdGroupCriterionApprovalStatusEnum_AdGroupCriterionApprovalStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 59, name: "disapproval_reasons", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 60, name: "labels", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupCriterionLabel" } } },
            { no: 61, name: "bid_modifier", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 62, name: "cpc_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 63, name: "cpm_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 64, name: "cpv_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 65, name: "percent_cpc_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 66, name: "effective_cpc_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 67, name: "effective_cpm_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 68, name: "effective_cpv_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 69, name: "effective_percent_cpc_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 21, name: "effective_cpc_bid_source", kind: "enum", T: () => ["google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource", bidding_source_1.BiddingSourceEnum_BiddingSource], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 22, name: "effective_cpm_bid_source", kind: "enum", T: () => ["google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource", bidding_source_1.BiddingSourceEnum_BiddingSource], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 23, name: "effective_cpv_bid_source", kind: "enum", T: () => ["google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource", bidding_source_1.BiddingSourceEnum_BiddingSource], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 35, name: "effective_percent_cpc_bid_source", kind: "enum", T: () => ["google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource", bidding_source_1.BiddingSourceEnum_BiddingSource], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "position_estimates", kind: "message", T: () => exports.AdGroupCriterion_PositionEstimates, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 70, name: "final_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 71, name: "final_mobile_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 72, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 73, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "url_custom_parameters", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => custom_parameter_1.CustomParameter },
            { no: 27, name: "keyword", kind: "message", oneof: "criterion", T: () => criteria_21.KeywordInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 28, name: "placement", kind: "message", oneof: "criterion", T: () => criteria_20.PlacementInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 29, name: "mobile_app_category", kind: "message", oneof: "criterion", T: () => criteria_19.MobileAppCategoryInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 30, name: "mobile_application", kind: "message", oneof: "criterion", T: () => criteria_18.MobileApplicationInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 32, name: "listing_group", kind: "message", oneof: "criterion", T: () => criteria_17.ListingGroupInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 36, name: "age_range", kind: "message", oneof: "criterion", T: () => criteria_16.AgeRangeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 37, name: "gender", kind: "message", oneof: "criterion", T: () => criteria_15.GenderInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 38, name: "income_range", kind: "message", oneof: "criterion", T: () => criteria_14.IncomeRangeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 39, name: "parental_status", kind: "message", oneof: "criterion", T: () => criteria_13.ParentalStatusInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 42, name: "user_list", kind: "message", oneof: "criterion", T: () => criteria_12.UserListInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 40, name: "youtube_video", kind: "message", oneof: "criterion", T: () => criteria_11.YouTubeVideoInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 41, name: "youtube_channel", kind: "message", oneof: "criterion", T: () => criteria_10.YouTubeChannelInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 43, name: "topic", kind: "message", oneof: "criterion", T: () => criteria_9.TopicInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 45, name: "user_interest", kind: "message", oneof: "criterion", T: () => criteria_8.UserInterestInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 46, name: "webpage", kind: "message", oneof: "criterion", T: () => criteria_7.WebpageInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 47, name: "app_payment_model", kind: "message", oneof: "criterion", T: () => criteria_6.AppPaymentModelInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 48, name: "custom_affinity", kind: "message", oneof: "criterion", T: () => criteria_5.CustomAffinityInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 49, name: "custom_intent", kind: "message", oneof: "criterion", T: () => criteria_4.CustomIntentInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 74, name: "custom_audience", kind: "message", oneof: "criterion", T: () => criteria_3.CustomAudienceInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 75, name: "combined_audience", kind: "message", oneof: "criterion", T: () => criteria_2.CombinedAudienceInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 79, name: "audience", kind: "message", oneof: "criterion", T: () => criteria_1.AudienceInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/AdGroupCriterion", pattern: ["customers/{customer_id}/adGroupCriteria/{ad_group_id}~{criterion_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", displayName: "", status: 0, type: 0, systemServingStatus: 0, approvalStatus: 0, disapprovalReasons: [], labels: [], effectiveCpcBidSource: 0, effectiveCpmBidSource: 0, effectiveCpvBidSource: 0, effectivePercentCpcBidSource: 0, finalUrls: [], finalMobileUrls: [], urlCustomParameters: [], criterion: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 criterion_id */ 56:
                    message.criterionId = reader.int64().toBigInt();
                    break;
                case /* string display_name */ 77:
                    message.displayName = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AdGroupCriterionStatusEnum.AdGroupCriterionStatus status */ 3:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupCriterion.QualityInfo quality_info */ 4:
                    message.qualityInfo = exports.AdGroupCriterion_QualityInfo.internalBinaryRead(reader, reader.uint32(), options, message.qualityInfo);
                    break;
                case /* optional string ad_group */ 57:
                    message.adGroup = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.CriterionTypeEnum.CriterionType type */ 25:
                    message.type = reader.int32();
                    break;
                case /* optional bool negative */ 58:
                    message.negative = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.CriterionSystemServingStatusEnum.CriterionSystemServingStatus system_serving_status */ 52:
                    message.systemServingStatus = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.AdGroupCriterionApprovalStatusEnum.AdGroupCriterionApprovalStatus approval_status */ 53:
                    message.approvalStatus = reader.int32();
                    break;
                case /* repeated string disapproval_reasons */ 59:
                    message.disapprovalReasons.push(reader.string());
                    break;
                case /* repeated string labels */ 60:
                    message.labels.push(reader.string());
                    break;
                case /* optional double bid_modifier */ 61:
                    message.bidModifier = reader.double();
                    break;
                case /* optional int64 cpc_bid_micros */ 62:
                    message.cpcBidMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 cpm_bid_micros */ 63:
                    message.cpmBidMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 cpv_bid_micros */ 64:
                    message.cpvBidMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 percent_cpc_bid_micros */ 65:
                    message.percentCpcBidMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 effective_cpc_bid_micros */ 66:
                    message.effectiveCpcBidMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 effective_cpm_bid_micros */ 67:
                    message.effectiveCpmBidMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 effective_cpv_bid_micros */ 68:
                    message.effectiveCpvBidMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 effective_percent_cpc_bid_micros */ 69:
                    message.effectivePercentCpcBidMicros = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource effective_cpc_bid_source */ 21:
                    message.effectiveCpcBidSource = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource effective_cpm_bid_source */ 22:
                    message.effectiveCpmBidSource = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource effective_cpv_bid_source */ 23:
                    message.effectiveCpvBidSource = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource effective_percent_cpc_bid_source */ 35:
                    message.effectivePercentCpcBidSource = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupCriterion.PositionEstimates position_estimates */ 10:
                    message.positionEstimates = exports.AdGroupCriterion_PositionEstimates.internalBinaryRead(reader, reader.uint32(), options, message.positionEstimates);
                    break;
                case /* repeated string final_urls */ 70:
                    message.finalUrls.push(reader.string());
                    break;
                case /* repeated string final_mobile_urls */ 71:
                    message.finalMobileUrls.push(reader.string());
                    break;
                case /* optional string final_url_suffix */ 72:
                    message.finalUrlSuffix = reader.string();
                    break;
                case /* optional string tracking_url_template */ 73:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters */ 14:
                    message.urlCustomParameters.push(custom_parameter_1.CustomParameter.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.common.KeywordInfo keyword */ 27:
                    message.criterion = {
                        oneofKind: "keyword",
                        keyword: criteria_21.KeywordInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.keyword)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PlacementInfo placement */ 28:
                    message.criterion = {
                        oneofKind: "placement",
                        placement: criteria_20.PlacementInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.placement)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MobileAppCategoryInfo mobile_app_category */ 29:
                    message.criterion = {
                        oneofKind: "mobileAppCategory",
                        mobileAppCategory: criteria_19.MobileAppCategoryInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.mobileAppCategory)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MobileApplicationInfo mobile_application */ 30:
                    message.criterion = {
                        oneofKind: "mobileApplication",
                        mobileApplication: criteria_18.MobileApplicationInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.mobileApplication)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ListingGroupInfo listing_group */ 32:
                    message.criterion = {
                        oneofKind: "listingGroup",
                        listingGroup: criteria_17.ListingGroupInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.listingGroup)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AgeRangeInfo age_range */ 36:
                    message.criterion = {
                        oneofKind: "ageRange",
                        ageRange: criteria_16.AgeRangeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.ageRange)
                    };
                    break;
                case /* google.ads.googleads.v11.common.GenderInfo gender */ 37:
                    message.criterion = {
                        oneofKind: "gender",
                        gender: criteria_15.GenderInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.gender)
                    };
                    break;
                case /* google.ads.googleads.v11.common.IncomeRangeInfo income_range */ 38:
                    message.criterion = {
                        oneofKind: "incomeRange",
                        incomeRange: criteria_14.IncomeRangeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.incomeRange)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ParentalStatusInfo parental_status */ 39:
                    message.criterion = {
                        oneofKind: "parentalStatus",
                        parentalStatus: criteria_13.ParentalStatusInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.parentalStatus)
                    };
                    break;
                case /* google.ads.googleads.v11.common.UserListInfo user_list */ 42:
                    message.criterion = {
                        oneofKind: "userList",
                        userList: criteria_12.UserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.userList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.YouTubeVideoInfo youtube_video */ 40:
                    message.criterion = {
                        oneofKind: "youtubeVideo",
                        youtubeVideo: criteria_11.YouTubeVideoInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.youtubeVideo)
                    };
                    break;
                case /* google.ads.googleads.v11.common.YouTubeChannelInfo youtube_channel */ 41:
                    message.criterion = {
                        oneofKind: "youtubeChannel",
                        youtubeChannel: criteria_10.YouTubeChannelInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.youtubeChannel)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TopicInfo topic */ 43:
                    message.criterion = {
                        oneofKind: "topic",
                        topic: criteria_9.TopicInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.topic)
                    };
                    break;
                case /* google.ads.googleads.v11.common.UserInterestInfo user_interest */ 45:
                    message.criterion = {
                        oneofKind: "userInterest",
                        userInterest: criteria_8.UserInterestInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.userInterest)
                    };
                    break;
                case /* google.ads.googleads.v11.common.WebpageInfo webpage */ 46:
                    message.criterion = {
                        oneofKind: "webpage",
                        webpage: criteria_7.WebpageInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.webpage)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AppPaymentModelInfo app_payment_model */ 47:
                    message.criterion = {
                        oneofKind: "appPaymentModel",
                        appPaymentModel: criteria_6.AppPaymentModelInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.appPaymentModel)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CustomAffinityInfo custom_affinity */ 48:
                    message.criterion = {
                        oneofKind: "customAffinity",
                        customAffinity: criteria_5.CustomAffinityInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.customAffinity)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CustomIntentInfo custom_intent */ 49:
                    message.criterion = {
                        oneofKind: "customIntent",
                        customIntent: criteria_4.CustomIntentInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.customIntent)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CustomAudienceInfo custom_audience */ 74:
                    message.criterion = {
                        oneofKind: "customAudience",
                        customAudience: criteria_3.CustomAudienceInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.customAudience)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CombinedAudienceInfo combined_audience */ 75:
                    message.criterion = {
                        oneofKind: "combinedAudience",
                        combinedAudience: criteria_2.CombinedAudienceInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.combinedAudience)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AudienceInfo audience */ 79:
                    message.criterion = {
                        oneofKind: "audience",
                        audience: criteria_1.AudienceInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.audience)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 criterion_id = 56; */
        if (message.criterionId !== undefined)
            writer.tag(56, runtime_1.WireType.Varint).int64(message.criterionId);
        /* string display_name = 77; */
        if (message.displayName !== "")
            writer.tag(77, runtime_1.WireType.LengthDelimited).string(message.displayName);
        /* google.ads.googleads.v11.enums.AdGroupCriterionStatusEnum.AdGroupCriterionStatus status = 3; */
        if (message.status !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.resources.AdGroupCriterion.QualityInfo quality_info = 4; */
        if (message.qualityInfo)
            exports.AdGroupCriterion_QualityInfo.internalBinaryWrite(message.qualityInfo, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string ad_group = 57; */
        if (message.adGroup !== undefined)
            writer.tag(57, runtime_1.WireType.LengthDelimited).string(message.adGroup);
        /* google.ads.googleads.v11.enums.CriterionTypeEnum.CriterionType type = 25; */
        if (message.type !== 0)
            writer.tag(25, runtime_1.WireType.Varint).int32(message.type);
        /* optional bool negative = 58; */
        if (message.negative !== undefined)
            writer.tag(58, runtime_1.WireType.Varint).bool(message.negative);
        /* google.ads.googleads.v11.enums.CriterionSystemServingStatusEnum.CriterionSystemServingStatus system_serving_status = 52; */
        if (message.systemServingStatus !== 0)
            writer.tag(52, runtime_1.WireType.Varint).int32(message.systemServingStatus);
        /* google.ads.googleads.v11.enums.AdGroupCriterionApprovalStatusEnum.AdGroupCriterionApprovalStatus approval_status = 53; */
        if (message.approvalStatus !== 0)
            writer.tag(53, runtime_1.WireType.Varint).int32(message.approvalStatus);
        /* repeated string disapproval_reasons = 59; */
        for (let i = 0; i < message.disapprovalReasons.length; i++)
            writer.tag(59, runtime_1.WireType.LengthDelimited).string(message.disapprovalReasons[i]);
        /* repeated string labels = 60; */
        for (let i = 0; i < message.labels.length; i++)
            writer.tag(60, runtime_1.WireType.LengthDelimited).string(message.labels[i]);
        /* optional double bid_modifier = 61; */
        if (message.bidModifier !== undefined)
            writer.tag(61, runtime_1.WireType.Bit64).double(message.bidModifier);
        /* optional int64 cpc_bid_micros = 62; */
        if (message.cpcBidMicros !== undefined)
            writer.tag(62, runtime_1.WireType.Varint).int64(message.cpcBidMicros);
        /* optional int64 cpm_bid_micros = 63; */
        if (message.cpmBidMicros !== undefined)
            writer.tag(63, runtime_1.WireType.Varint).int64(message.cpmBidMicros);
        /* optional int64 cpv_bid_micros = 64; */
        if (message.cpvBidMicros !== undefined)
            writer.tag(64, runtime_1.WireType.Varint).int64(message.cpvBidMicros);
        /* optional int64 percent_cpc_bid_micros = 65; */
        if (message.percentCpcBidMicros !== undefined)
            writer.tag(65, runtime_1.WireType.Varint).int64(message.percentCpcBidMicros);
        /* optional int64 effective_cpc_bid_micros = 66; */
        if (message.effectiveCpcBidMicros !== undefined)
            writer.tag(66, runtime_1.WireType.Varint).int64(message.effectiveCpcBidMicros);
        /* optional int64 effective_cpm_bid_micros = 67; */
        if (message.effectiveCpmBidMicros !== undefined)
            writer.tag(67, runtime_1.WireType.Varint).int64(message.effectiveCpmBidMicros);
        /* optional int64 effective_cpv_bid_micros = 68; */
        if (message.effectiveCpvBidMicros !== undefined)
            writer.tag(68, runtime_1.WireType.Varint).int64(message.effectiveCpvBidMicros);
        /* optional int64 effective_percent_cpc_bid_micros = 69; */
        if (message.effectivePercentCpcBidMicros !== undefined)
            writer.tag(69, runtime_1.WireType.Varint).int64(message.effectivePercentCpcBidMicros);
        /* google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource effective_cpc_bid_source = 21; */
        if (message.effectiveCpcBidSource !== 0)
            writer.tag(21, runtime_1.WireType.Varint).int32(message.effectiveCpcBidSource);
        /* google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource effective_cpm_bid_source = 22; */
        if (message.effectiveCpmBidSource !== 0)
            writer.tag(22, runtime_1.WireType.Varint).int32(message.effectiveCpmBidSource);
        /* google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource effective_cpv_bid_source = 23; */
        if (message.effectiveCpvBidSource !== 0)
            writer.tag(23, runtime_1.WireType.Varint).int32(message.effectiveCpvBidSource);
        /* google.ads.googleads.v11.enums.BiddingSourceEnum.BiddingSource effective_percent_cpc_bid_source = 35; */
        if (message.effectivePercentCpcBidSource !== 0)
            writer.tag(35, runtime_1.WireType.Varint).int32(message.effectivePercentCpcBidSource);
        /* google.ads.googleads.v11.resources.AdGroupCriterion.PositionEstimates position_estimates = 10; */
        if (message.positionEstimates)
            exports.AdGroupCriterion_PositionEstimates.internalBinaryWrite(message.positionEstimates, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated string final_urls = 70; */
        for (let i = 0; i < message.finalUrls.length; i++)
            writer.tag(70, runtime_1.WireType.LengthDelimited).string(message.finalUrls[i]);
        /* repeated string final_mobile_urls = 71; */
        for (let i = 0; i < message.finalMobileUrls.length; i++)
            writer.tag(71, runtime_1.WireType.LengthDelimited).string(message.finalMobileUrls[i]);
        /* optional string final_url_suffix = 72; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(72, runtime_1.WireType.LengthDelimited).string(message.finalUrlSuffix);
        /* optional string tracking_url_template = 73; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(73, runtime_1.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters = 14; */
        for (let i = 0; i < message.urlCustomParameters.length; i++)
            custom_parameter_1.CustomParameter.internalBinaryWrite(message.urlCustomParameters[i], writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.KeywordInfo keyword = 27; */
        if (message.criterion.oneofKind === "keyword")
            criteria_21.KeywordInfo.internalBinaryWrite(message.criterion.keyword, writer.tag(27, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PlacementInfo placement = 28; */
        if (message.criterion.oneofKind === "placement")
            criteria_20.PlacementInfo.internalBinaryWrite(message.criterion.placement, writer.tag(28, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MobileAppCategoryInfo mobile_app_category = 29; */
        if (message.criterion.oneofKind === "mobileAppCategory")
            criteria_19.MobileAppCategoryInfo.internalBinaryWrite(message.criterion.mobileAppCategory, writer.tag(29, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MobileApplicationInfo mobile_application = 30; */
        if (message.criterion.oneofKind === "mobileApplication")
            criteria_18.MobileApplicationInfo.internalBinaryWrite(message.criterion.mobileApplication, writer.tag(30, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ListingGroupInfo listing_group = 32; */
        if (message.criterion.oneofKind === "listingGroup")
            criteria_17.ListingGroupInfo.internalBinaryWrite(message.criterion.listingGroup, writer.tag(32, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AgeRangeInfo age_range = 36; */
        if (message.criterion.oneofKind === "ageRange")
            criteria_16.AgeRangeInfo.internalBinaryWrite(message.criterion.ageRange, writer.tag(36, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.GenderInfo gender = 37; */
        if (message.criterion.oneofKind === "gender")
            criteria_15.GenderInfo.internalBinaryWrite(message.criterion.gender, writer.tag(37, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.IncomeRangeInfo income_range = 38; */
        if (message.criterion.oneofKind === "incomeRange")
            criteria_14.IncomeRangeInfo.internalBinaryWrite(message.criterion.incomeRange, writer.tag(38, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ParentalStatusInfo parental_status = 39; */
        if (message.criterion.oneofKind === "parentalStatus")
            criteria_13.ParentalStatusInfo.internalBinaryWrite(message.criterion.parentalStatus, writer.tag(39, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserListInfo user_list = 42; */
        if (message.criterion.oneofKind === "userList")
            criteria_12.UserListInfo.internalBinaryWrite(message.criterion.userList, writer.tag(42, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.YouTubeVideoInfo youtube_video = 40; */
        if (message.criterion.oneofKind === "youtubeVideo")
            criteria_11.YouTubeVideoInfo.internalBinaryWrite(message.criterion.youtubeVideo, writer.tag(40, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.YouTubeChannelInfo youtube_channel = 41; */
        if (message.criterion.oneofKind === "youtubeChannel")
            criteria_10.YouTubeChannelInfo.internalBinaryWrite(message.criterion.youtubeChannel, writer.tag(41, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TopicInfo topic = 43; */
        if (message.criterion.oneofKind === "topic")
            criteria_9.TopicInfo.internalBinaryWrite(message.criterion.topic, writer.tag(43, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserInterestInfo user_interest = 45; */
        if (message.criterion.oneofKind === "userInterest")
            criteria_8.UserInterestInfo.internalBinaryWrite(message.criterion.userInterest, writer.tag(45, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.WebpageInfo webpage = 46; */
        if (message.criterion.oneofKind === "webpage")
            criteria_7.WebpageInfo.internalBinaryWrite(message.criterion.webpage, writer.tag(46, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AppPaymentModelInfo app_payment_model = 47; */
        if (message.criterion.oneofKind === "appPaymentModel")
            criteria_6.AppPaymentModelInfo.internalBinaryWrite(message.criterion.appPaymentModel, writer.tag(47, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CustomAffinityInfo custom_affinity = 48; */
        if (message.criterion.oneofKind === "customAffinity")
            criteria_5.CustomAffinityInfo.internalBinaryWrite(message.criterion.customAffinity, writer.tag(48, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CustomIntentInfo custom_intent = 49; */
        if (message.criterion.oneofKind === "customIntent")
            criteria_4.CustomIntentInfo.internalBinaryWrite(message.criterion.customIntent, writer.tag(49, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CustomAudienceInfo custom_audience = 74; */
        if (message.criterion.oneofKind === "customAudience")
            criteria_3.CustomAudienceInfo.internalBinaryWrite(message.criterion.customAudience, writer.tag(74, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CombinedAudienceInfo combined_audience = 75; */
        if (message.criterion.oneofKind === "combinedAudience")
            criteria_2.CombinedAudienceInfo.internalBinaryWrite(message.criterion.combinedAudience, writer.tag(75, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AudienceInfo audience = 79; */
        if (message.criterion.oneofKind === "audience")
            criteria_1.AudienceInfo.internalBinaryWrite(message.criterion.audience, writer.tag(79, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AdGroupCriterion
 */
exports.AdGroupCriterion = new AdGroupCriterion$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupCriterion_QualityInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AdGroupCriterion.QualityInfo", [
            { no: 5, name: "quality_score", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "creative_quality_score", kind: "enum", T: () => ["google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket", quality_score_bucket_1.QualityScoreBucketEnum_QualityScoreBucket], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "post_click_quality_score", kind: "enum", T: () => ["google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket", quality_score_bucket_1.QualityScoreBucketEnum_QualityScoreBucket], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "search_predicted_ctr", kind: "enum", T: () => ["google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket", quality_score_bucket_1.QualityScoreBucketEnum_QualityScoreBucket], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { creativeQualityScore: 0, postClickQualityScore: 0, searchPredictedCtr: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int32 quality_score */ 5:
                    message.qualityScore = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket creative_quality_score */ 2:
                    message.creativeQualityScore = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket post_click_quality_score */ 3:
                    message.postClickQualityScore = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket search_predicted_ctr */ 4:
                    message.searchPredictedCtr = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int32 quality_score = 5; */
        if (message.qualityScore !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.qualityScore);
        /* google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket creative_quality_score = 2; */
        if (message.creativeQualityScore !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.creativeQualityScore);
        /* google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket post_click_quality_score = 3; */
        if (message.postClickQualityScore !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.postClickQualityScore);
        /* google.ads.googleads.v11.enums.QualityScoreBucketEnum.QualityScoreBucket search_predicted_ctr = 4; */
        if (message.searchPredictedCtr !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.searchPredictedCtr);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AdGroupCriterion.QualityInfo
 */
exports.AdGroupCriterion_QualityInfo = new AdGroupCriterion_QualityInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupCriterion_PositionEstimates$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AdGroupCriterion.PositionEstimates", [
            { no: 6, name: "first_page_cpc_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "first_position_cpc_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "top_of_page_cpc_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "estimated_add_clicks_at_first_position_cpc", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "estimated_add_cost_at_first_position_cpc", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 first_page_cpc_micros */ 6:
                    message.firstPageCpcMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 first_position_cpc_micros */ 7:
                    message.firstPositionCpcMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 top_of_page_cpc_micros */ 8:
                    message.topOfPageCpcMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 estimated_add_clicks_at_first_position_cpc */ 9:
                    message.estimatedAddClicksAtFirstPositionCpc = reader.int64().toBigInt();
                    break;
                case /* optional int64 estimated_add_cost_at_first_position_cpc */ 10:
                    message.estimatedAddCostAtFirstPositionCpc = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 first_page_cpc_micros = 6; */
        if (message.firstPageCpcMicros !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).int64(message.firstPageCpcMicros);
        /* optional int64 first_position_cpc_micros = 7; */
        if (message.firstPositionCpcMicros !== undefined)
            writer.tag(7, runtime_1.WireType.Varint).int64(message.firstPositionCpcMicros);
        /* optional int64 top_of_page_cpc_micros = 8; */
        if (message.topOfPageCpcMicros !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).int64(message.topOfPageCpcMicros);
        /* optional int64 estimated_add_clicks_at_first_position_cpc = 9; */
        if (message.estimatedAddClicksAtFirstPositionCpc !== undefined)
            writer.tag(9, runtime_1.WireType.Varint).int64(message.estimatedAddClicksAtFirstPositionCpc);
        /* optional int64 estimated_add_cost_at_first_position_cpc = 10; */
        if (message.estimatedAddCostAtFirstPositionCpc !== undefined)
            writer.tag(10, runtime_1.WireType.Varint).int64(message.estimatedAddCostAtFirstPositionCpc);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AdGroupCriterion.PositionEstimates
 */
exports.AdGroupCriterion_PositionEstimates = new AdGroupCriterion_PositionEstimates$Type();
