// google-ads-api-client/dist/protos/bindings/ads/resources/ad_group_simulation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupSimulation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const simulation_1 = require("../common/simulation");
const simulation_2 = require("../common/simulation");
const simulation_3 = require("../common/simulation");
const simulation_4 = require("../common/simulation");
const simulation_modification_method_1 = require("../enums/simulation_modification_method");
const simulation_type_1 = require("../enums/simulation_type");
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupSimulation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AdGroupSimulation", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupSimulation" } } },
            { no: 12, name: "ad_group_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType", simulation_type_1.SimulationTypeEnum_SimulationType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "modification_method", kind: "enum", T: () => ["google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod", simulation_modification_method_1.SimulationModificationMethodEnum_SimulationModificationMethod], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "start_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 14, name: "end_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "cpc_bid_point_list", kind: "message", oneof: "pointList", T: () => simulation_4.CpcBidSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "cpv_bid_point_list", kind: "message", oneof: "pointList", T: () => simulation_3.CpvBidSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "target_cpa_point_list", kind: "message", oneof: "pointList", T: () => simulation_2.TargetCpaSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "target_roas_point_list", kind: "message", oneof: "pointList", T: () => simulation_1.TargetRoasSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/AdGroupSimulation", pattern: ["customers/{customer_id}/adGroupSimulations/{ad_group_id}~{type}~{modification_method}~{start_date}~{end_date}"] } });
    }
    create(value) {
        const message = { resourceName: "", type: 0, modificationMethod: 0, pointList: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 ad_group_id */ 12:
                    message.adGroupId = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType type */ 3:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod modification_method */ 4:
                    message.modificationMethod = reader.int32();
                    break;
                case /* optional string start_date */ 13:
                    message.startDate = reader.string();
                    break;
                case /* optional string end_date */ 14:
                    message.endDate = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.CpcBidSimulationPointList cpc_bid_point_list */ 8:
                    message.pointList = {
                        oneofKind: "cpcBidPointList",
                        cpcBidPointList: simulation_4.CpcBidSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.cpcBidPointList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CpvBidSimulationPointList cpv_bid_point_list */ 10:
                    message.pointList = {
                        oneofKind: "cpvBidPointList",
                        cpvBidPointList: simulation_3.CpvBidSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.cpvBidPointList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetCpaSimulationPointList target_cpa_point_list */ 9:
                    message.pointList = {
                        oneofKind: "targetCpaPointList",
                        targetCpaPointList: simulation_2.TargetCpaSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.targetCpaPointList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetRoasSimulationPointList target_roas_point_list */ 11:
                    message.pointList = {
                        oneofKind: "targetRoasPointList",
                        targetRoasPointList: simulation_1.TargetRoasSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.targetRoasPointList)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 ad_group_id = 12; */
        if (message.adGroupId !== undefined)
            writer.tag(12, runtime_1.WireType.Varint).int64(message.adGroupId);
        /* google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType type = 3; */
        if (message.type !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod modification_method = 4; */
        if (message.modificationMethod !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.modificationMethod);
        /* optional string start_date = 13; */
        if (message.startDate !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* optional string end_date = 14; */
        if (message.endDate !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.endDate);
        /* google.ads.googleads.v11.common.CpcBidSimulationPointList cpc_bid_point_list = 8; */
        if (message.pointList.oneofKind === "cpcBidPointList")
            simulation_4.CpcBidSimulationPointList.internalBinaryWrite(message.pointList.cpcBidPointList, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CpvBidSimulationPointList cpv_bid_point_list = 10; */
        if (message.pointList.oneofKind === "cpvBidPointList")
            simulation_3.CpvBidSimulationPointList.internalBinaryWrite(message.pointList.cpvBidPointList, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetCpaSimulationPointList target_cpa_point_list = 9; */
        if (message.pointList.oneofKind === "targetCpaPointList")
            simulation_2.TargetCpaSimulationPointList.internalBinaryWrite(message.pointList.targetCpaPointList, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetRoasSimulationPointList target_roas_point_list = 11; */
        if (message.pointList.oneofKind === "targetRoasPointList")
            simulation_1.TargetRoasSimulationPointList.internalBinaryWrite(message.pointList.targetRoasPointList, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AdGroupSimulation
 */
exports.AdGroupSimulation = new AdGroupSimulation$Type();
