// google-ads-api-client/dist/protos/bindings/ads/resources/ad_group_ad.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdGroupAdPolicySummary = exports.AdGroupAd = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const policy_approval_status_1 = require("../enums/policy_approval_status");
const policy_review_status_1 = require("../enums/policy_review_status");
const policy_1 = require("../common/policy");
const ad_strength_1 = require("../enums/ad_strength");
const ad_1 = require("./ad");
const ad_group_ad_status_1 = require("../enums/ad_group_ad_status");
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupAd$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AdGroupAd", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupAd" } } },
            { no: 3, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AdGroupAdStatusEnum.AdGroupAdStatus", ad_group_ad_status_1.AdGroupAdStatusEnum_AdGroupAdStatus] },
            { no: 9, name: "ad_group", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } },
            { no: 5, name: "ad", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "policy_summary", kind: "message", T: () => exports.AdGroupAdPolicySummary, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "ad_strength", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AdStrengthEnum.AdStrength", ad_strength_1.AdStrengthEnum_AdStrength], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "action_items", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "labels", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupAdLabel" } } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/AdGroupAd", pattern: ["customers/{customer_id}/adGroupAds/{ad_group_id}~{ad_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", status: 0, adStrength: 0, actionItems: [], labels: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AdGroupAdStatusEnum.AdGroupAdStatus status */ 3:
                    message.status = reader.int32();
                    break;
                case /* optional string ad_group */ 9:
                    message.adGroup = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.Ad ad */ 5:
                    message.ad = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.ad);
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupAdPolicySummary policy_summary */ 6:
                    message.policySummary = exports.AdGroupAdPolicySummary.internalBinaryRead(reader, reader.uint32(), options, message.policySummary);
                    break;
                case /* google.ads.googleads.v11.enums.AdStrengthEnum.AdStrength ad_strength */ 7:
                    message.adStrength = reader.int32();
                    break;
                case /* repeated string action_items */ 13:
                    message.actionItems.push(reader.string());
                    break;
                case /* repeated string labels */ 10:
                    message.labels.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* google.ads.googleads.v11.enums.AdGroupAdStatusEnum.AdGroupAdStatus status = 3; */
        if (message.status !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.status);
        /* optional string ad_group = 9; */
        if (message.adGroup !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.adGroup);
        /* google.ads.googleads.v11.resources.Ad ad = 5; */
        if (message.ad)
            ad_1.Ad.internalBinaryWrite(message.ad, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroupAdPolicySummary policy_summary = 6; */
        if (message.policySummary)
            exports.AdGroupAdPolicySummary.internalBinaryWrite(message.policySummary, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.AdStrengthEnum.AdStrength ad_strength = 7; */
        if (message.adStrength !== 0)
            writer.tag(7, runtime_1.WireType.Varint).int32(message.adStrength);
        /* repeated string action_items = 13; */
        for (let i = 0; i < message.actionItems.length; i++)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.actionItems[i]);
        /* repeated string labels = 10; */
        for (let i = 0; i < message.labels.length; i++)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.labels[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AdGroupAd
 */
exports.AdGroupAd = new AdGroupAd$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AdGroupAdPolicySummary$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AdGroupAdPolicySummary", [
            { no: 1, name: "policy_topic_entries", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => policy_1.PolicyTopicEntry, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "review_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus", policy_review_status_1.PolicyReviewStatusEnum_PolicyReviewStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "approval_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus", policy_approval_status_1.PolicyApprovalStatusEnum_PolicyApprovalStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { policyTopicEntries: [], reviewStatus: 0, approvalStatus: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.PolicyTopicEntry policy_topic_entries */ 1:
                    message.policyTopicEntries.push(policy_1.PolicyTopicEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus review_status */ 2:
                    message.reviewStatus = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus approval_status */ 3:
                    message.approvalStatus = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.PolicyTopicEntry policy_topic_entries = 1; */
        for (let i = 0; i < message.policyTopicEntries.length; i++)
            policy_1.PolicyTopicEntry.internalBinaryWrite(message.policyTopicEntries[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus review_status = 2; */
        if (message.reviewStatus !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.reviewStatus);
        /* google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus approval_status = 3; */
        if (message.approvalStatus !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.approvalStatus);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AdGroupAdPolicySummary
 */
exports.AdGroupAdPolicySummary = new AdGroupAdPolicySummary$Type();
