// google-ads-api-client/dist/protos/bindings/ads/resources/asset_group_listing_group_filter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingGroupFilterDimension_ProductType = exports.ListingGroupFilterDimension_ProductItemId = exports.ListingGroupFilterDimension_ProductCustomAttribute = exports.ListingGroupFilterDimension_ProductCondition = exports.ListingGroupFilterDimension_ProductChannel = exports.ListingGroupFilterDimension_ProductBrand = exports.ListingGroupFilterDimension_ProductBiddingCategory = exports.ListingGroupFilterDimension = exports.AssetGroupListingGroupFilter = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const listing_group_filter_product_type_level_1 = require("../enums/listing_group_filter_product_type_level");
const listing_group_filter_custom_attribute_index_1 = require("../enums/listing_group_filter_custom_attribute_index");
const listing_group_filter_product_condition_1 = require("../enums/listing_group_filter_product_condition");
const listing_group_filter_product_channel_1 = require("../enums/listing_group_filter_product_channel");
const listing_group_filter_bidding_category_level_1 = require("../enums/listing_group_filter_bidding_category_level");
const listing_group_filter_vertical_1 = require("../enums/listing_group_filter_vertical");
const listing_group_filter_type_enum_1 = require("../enums/listing_group_filter_type_enum");
// @generated message type with reflection information, may provide speed optimized methods
class AssetGroupListingGroupFilter$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AssetGroupListingGroupFilter", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AssetGroupListingGroupFilter" } } },
            { no: 2, name: "asset_group", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AssetGroup" } } },
            { no: 3, name: "id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ListingGroupFilterTypeEnum.ListingGroupFilterType", listing_group_filter_type_enum_1.ListingGroupFilterTypeEnum_ListingGroupFilterType], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 5, name: "vertical", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ListingGroupFilterVerticalEnum.ListingGroupFilterVertical", listing_group_filter_vertical_1.ListingGroupFilterVerticalEnum_ListingGroupFilterVertical], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "case_value", kind: "message", T: () => exports.ListingGroupFilterDimension },
            { no: 7, name: "parent_listing_group_filter", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AssetGroupListingGroupFilter" } } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/AssetGroupListingGroupFilter", pattern: ["customers/{customer_id}/assetGroupListingGroupFilters/{asset_group_id}~{listing_group_filter_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", assetGroup: "", id: 0n, type: 0, vertical: 0, parentListingGroupFilter: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* string asset_group */ 2:
                    message.assetGroup = reader.string();
                    break;
                case /* int64 id */ 3:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.ListingGroupFilterTypeEnum.ListingGroupFilterType type */ 4:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.ListingGroupFilterVerticalEnum.ListingGroupFilterVertical vertical */ 5:
                    message.vertical = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.ListingGroupFilterDimension case_value */ 6:
                    message.caseValue = exports.ListingGroupFilterDimension.internalBinaryRead(reader, reader.uint32(), options, message.caseValue);
                    break;
                case /* string parent_listing_group_filter */ 7:
                    message.parentListingGroupFilter = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* string asset_group = 2; */
        if (message.assetGroup !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.assetGroup);
        /* int64 id = 3; */
        if (message.id !== 0n)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.id);
        /* google.ads.googleads.v11.enums.ListingGroupFilterTypeEnum.ListingGroupFilterType type = 4; */
        if (message.type !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.ListingGroupFilterVerticalEnum.ListingGroupFilterVertical vertical = 5; */
        if (message.vertical !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.vertical);
        /* google.ads.googleads.v11.resources.ListingGroupFilterDimension case_value = 6; */
        if (message.caseValue)
            exports.ListingGroupFilterDimension.internalBinaryWrite(message.caseValue, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string parent_listing_group_filter = 7; */
        if (message.parentListingGroupFilter !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.parentListingGroupFilter);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AssetGroupListingGroupFilter
 */
exports.AssetGroupListingGroupFilter = new AssetGroupListingGroupFilter$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterDimension$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ListingGroupFilterDimension", [
            { no: 1, name: "product_bidding_category", kind: "message", oneof: "dimension", T: () => exports.ListingGroupFilterDimension_ProductBiddingCategory },
            { no: 2, name: "product_brand", kind: "message", oneof: "dimension", T: () => exports.ListingGroupFilterDimension_ProductBrand },
            { no: 3, name: "product_channel", kind: "message", oneof: "dimension", T: () => exports.ListingGroupFilterDimension_ProductChannel },
            { no: 4, name: "product_condition", kind: "message", oneof: "dimension", T: () => exports.ListingGroupFilterDimension_ProductCondition },
            { no: 5, name: "product_custom_attribute", kind: "message", oneof: "dimension", T: () => exports.ListingGroupFilterDimension_ProductCustomAttribute },
            { no: 6, name: "product_item_id", kind: "message", oneof: "dimension", T: () => exports.ListingGroupFilterDimension_ProductItemId },
            { no: 7, name: "product_type", kind: "message", oneof: "dimension", T: () => exports.ListingGroupFilterDimension_ProductType }
        ]);
    }
    create(value) {
        const message = { dimension: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductBiddingCategory product_bidding_category */ 1:
                    message.dimension = {
                        oneofKind: "productBiddingCategory",
                        productBiddingCategory: exports.ListingGroupFilterDimension_ProductBiddingCategory.internalBinaryRead(reader, reader.uint32(), options, message.dimension.productBiddingCategory)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductBrand product_brand */ 2:
                    message.dimension = {
                        oneofKind: "productBrand",
                        productBrand: exports.ListingGroupFilterDimension_ProductBrand.internalBinaryRead(reader, reader.uint32(), options, message.dimension.productBrand)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductChannel product_channel */ 3:
                    message.dimension = {
                        oneofKind: "productChannel",
                        productChannel: exports.ListingGroupFilterDimension_ProductChannel.internalBinaryRead(reader, reader.uint32(), options, message.dimension.productChannel)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductCondition product_condition */ 4:
                    message.dimension = {
                        oneofKind: "productCondition",
                        productCondition: exports.ListingGroupFilterDimension_ProductCondition.internalBinaryRead(reader, reader.uint32(), options, message.dimension.productCondition)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductCustomAttribute product_custom_attribute */ 5:
                    message.dimension = {
                        oneofKind: "productCustomAttribute",
                        productCustomAttribute: exports.ListingGroupFilterDimension_ProductCustomAttribute.internalBinaryRead(reader, reader.uint32(), options, message.dimension.productCustomAttribute)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductItemId product_item_id */ 6:
                    message.dimension = {
                        oneofKind: "productItemId",
                        productItemId: exports.ListingGroupFilterDimension_ProductItemId.internalBinaryRead(reader, reader.uint32(), options, message.dimension.productItemId)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductType product_type */ 7:
                    message.dimension = {
                        oneofKind: "productType",
                        productType: exports.ListingGroupFilterDimension_ProductType.internalBinaryRead(reader, reader.uint32(), options, message.dimension.productType)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductBiddingCategory product_bidding_category = 1; */
        if (message.dimension.oneofKind === "productBiddingCategory")
            exports.ListingGroupFilterDimension_ProductBiddingCategory.internalBinaryWrite(message.dimension.productBiddingCategory, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductBrand product_brand = 2; */
        if (message.dimension.oneofKind === "productBrand")
            exports.ListingGroupFilterDimension_ProductBrand.internalBinaryWrite(message.dimension.productBrand, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductChannel product_channel = 3; */
        if (message.dimension.oneofKind === "productChannel")
            exports.ListingGroupFilterDimension_ProductChannel.internalBinaryWrite(message.dimension.productChannel, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductCondition product_condition = 4; */
        if (message.dimension.oneofKind === "productCondition")
            exports.ListingGroupFilterDimension_ProductCondition.internalBinaryWrite(message.dimension.productCondition, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductCustomAttribute product_custom_attribute = 5; */
        if (message.dimension.oneofKind === "productCustomAttribute")
            exports.ListingGroupFilterDimension_ProductCustomAttribute.internalBinaryWrite(message.dimension.productCustomAttribute, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductItemId product_item_id = 6; */
        if (message.dimension.oneofKind === "productItemId")
            exports.ListingGroupFilterDimension_ProductItemId.internalBinaryWrite(message.dimension.productItemId, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductType product_type = 7; */
        if (message.dimension.oneofKind === "productType")
            exports.ListingGroupFilterDimension_ProductType.internalBinaryWrite(message.dimension.productType, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ListingGroupFilterDimension
 */
exports.ListingGroupFilterDimension = new ListingGroupFilterDimension$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterDimension_ProductBiddingCategory$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductBiddingCategory", [
            { no: 1, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 2, name: "level", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ListingGroupFilterBiddingCategoryLevelEnum.ListingGroupFilterBiddingCategoryLevel", listing_group_filter_bidding_category_level_1.ListingGroupFilterBiddingCategoryLevelEnum_ListingGroupFilterBiddingCategoryLevel] }
        ]);
    }
    create(value) {
        const message = { level: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 id */ 1:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.ListingGroupFilterBiddingCategoryLevelEnum.ListingGroupFilterBiddingCategoryLevel level */ 2:
                    message.level = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 id = 1; */
        if (message.id !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.id);
        /* google.ads.googleads.v11.enums.ListingGroupFilterBiddingCategoryLevelEnum.ListingGroupFilterBiddingCategoryLevel level = 2; */
        if (message.level !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.level);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductBiddingCategory
 */
exports.ListingGroupFilterDimension_ProductBiddingCategory = new ListingGroupFilterDimension_ProductBiddingCategory$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterDimension_ProductBrand$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductBrand", [
            { no: 1, name: "value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string value */ 1:
                    message.value = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string value = 1; */
        if (message.value !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.value);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductBrand
 */
exports.ListingGroupFilterDimension_ProductBrand = new ListingGroupFilterDimension_ProductBrand$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterDimension_ProductChannel$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductChannel", [
            { no: 1, name: "channel", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ListingGroupFilterProductChannelEnum.ListingGroupFilterProductChannel", listing_group_filter_product_channel_1.ListingGroupFilterProductChannelEnum_ListingGroupFilterProductChannel] }
        ]);
    }
    create(value) {
        const message = { channel: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.ListingGroupFilterProductChannelEnum.ListingGroupFilterProductChannel channel */ 1:
                    message.channel = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.ListingGroupFilterProductChannelEnum.ListingGroupFilterProductChannel channel = 1; */
        if (message.channel !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.channel);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductChannel
 */
exports.ListingGroupFilterDimension_ProductChannel = new ListingGroupFilterDimension_ProductChannel$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterDimension_ProductCondition$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductCondition", [
            { no: 1, name: "condition", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ListingGroupFilterProductConditionEnum.ListingGroupFilterProductCondition", listing_group_filter_product_condition_1.ListingGroupFilterProductConditionEnum_ListingGroupFilterProductCondition] }
        ]);
    }
    create(value) {
        const message = { condition: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.ListingGroupFilterProductConditionEnum.ListingGroupFilterProductCondition condition */ 1:
                    message.condition = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.ListingGroupFilterProductConditionEnum.ListingGroupFilterProductCondition condition = 1; */
        if (message.condition !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.condition);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductCondition
 */
exports.ListingGroupFilterDimension_ProductCondition = new ListingGroupFilterDimension_ProductCondition$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterDimension_ProductCustomAttribute$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductCustomAttribute", [
            { no: 1, name: "value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "index", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ListingGroupFilterCustomAttributeIndexEnum.ListingGroupFilterCustomAttributeIndex", listing_group_filter_custom_attribute_index_1.ListingGroupFilterCustomAttributeIndexEnum_ListingGroupFilterCustomAttributeIndex] }
        ]);
    }
    create(value) {
        const message = { index: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string value */ 1:
                    message.value = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.ListingGroupFilterCustomAttributeIndexEnum.ListingGroupFilterCustomAttributeIndex index */ 2:
                    message.index = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string value = 1; */
        if (message.value !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.value);
        /* google.ads.googleads.v11.enums.ListingGroupFilterCustomAttributeIndexEnum.ListingGroupFilterCustomAttributeIndex index = 2; */
        if (message.index !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.index);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductCustomAttribute
 */
exports.ListingGroupFilterDimension_ProductCustomAttribute = new ListingGroupFilterDimension_ProductCustomAttribute$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterDimension_ProductItemId$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductItemId", [
            { no: 1, name: "value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string value */ 1:
                    message.value = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string value = 1; */
        if (message.value !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.value);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductItemId
 */
exports.ListingGroupFilterDimension_ProductItemId = new ListingGroupFilterDimension_ProductItemId$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ListingGroupFilterDimension_ProductType$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductType", [
            { no: 1, name: "value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "level", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ListingGroupFilterProductTypeLevelEnum.ListingGroupFilterProductTypeLevel", listing_group_filter_product_type_level_1.ListingGroupFilterProductTypeLevelEnum_ListingGroupFilterProductTypeLevel] }
        ]);
    }
    create(value) {
        const message = { level: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string value */ 1:
                    message.value = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.ListingGroupFilterProductTypeLevelEnum.ListingGroupFilterProductTypeLevel level */ 2:
                    message.level = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string value = 1; */
        if (message.value !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.value);
        /* google.ads.googleads.v11.enums.ListingGroupFilterProductTypeLevelEnum.ListingGroupFilterProductTypeLevel level = 2; */
        if (message.level !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.level);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ListingGroupFilterDimension.ProductType
 */
exports.ListingGroupFilterDimension_ProductType = new ListingGroupFilterDimension_ProductType$Type();
