// google-ads-api-client/dist/protos/bindings/ads/resources/asset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetPolicySummary = exports.Asset = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const policy_approval_status_1 = require("../enums/policy_approval_status");
const policy_review_status_1 = require("../enums/policy_review_status");
const policy_1 = require("../common/policy");
const asset_types_1 = require("../common/asset_types");
const asset_types_2 = require("../common/asset_types");
const asset_types_3 = require("../common/asset_types");
const asset_types_4 = require("../common/asset_types");
const asset_types_5 = require("../common/asset_types");
const asset_types_6 = require("../common/asset_types");
const asset_types_7 = require("../common/asset_types");
const asset_types_8 = require("../common/asset_types");
const asset_types_9 = require("../common/asset_types");
const asset_types_10 = require("../common/asset_types");
const asset_types_11 = require("../common/asset_types");
const asset_types_12 = require("../common/asset_types");
const asset_types_13 = require("../common/asset_types");
const asset_types_14 = require("../common/asset_types");
const asset_types_15 = require("../common/asset_types");
const asset_types_16 = require("../common/asset_types");
const asset_types_17 = require("../common/asset_types");
const asset_types_18 = require("../common/asset_types");
const asset_types_19 = require("../common/asset_types");
const asset_types_20 = require("../common/asset_types");
const asset_types_21 = require("../common/asset_types");
const asset_types_22 = require("../common/asset_types");
const asset_types_23 = require("../common/asset_types");
const asset_types_24 = require("../common/asset_types");
const asset_types_25 = require("../common/asset_types");
const asset_source_1 = require("../enums/asset_source");
const custom_parameter_1 = require("../common/custom_parameter");
const asset_type_1 = require("../enums/asset_type");
// @generated message type with reflection information, may provide speed optimized methods
class Asset$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Asset", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Asset" } } },
            { no: 11, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 12, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AssetTypeEnum.AssetType", asset_type_1.AssetTypeEnum_AssetType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 14, name: "final_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "final_mobile_urls", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 17, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 18, name: "url_custom_parameters", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => custom_parameter_1.CustomParameter },
            { no: 19, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 38, name: "source", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AssetSourceEnum.AssetSource", asset_source_1.AssetSourceEnum_AssetSource], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "policy_summary", kind: "message", T: () => exports.AssetPolicySummary, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "youtube_video_asset", kind: "message", oneof: "assetData", T: () => asset_types_25.YoutubeVideoAsset, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "media_bundle_asset", kind: "message", oneof: "assetData", T: () => asset_types_24.MediaBundleAsset, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 7, name: "image_asset", kind: "message", oneof: "assetData", T: () => asset_types_23.ImageAsset, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "text_asset", kind: "message", oneof: "assetData", T: () => asset_types_22.TextAsset, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 9, name: "lead_form_asset", kind: "message", oneof: "assetData", T: () => asset_types_21.LeadFormAsset },
            { no: 10, name: "book_on_google_asset", kind: "message", oneof: "assetData", T: () => asset_types_20.BookOnGoogleAsset },
            { no: 15, name: "promotion_asset", kind: "message", oneof: "assetData", T: () => asset_types_19.PromotionAsset },
            { no: 20, name: "callout_asset", kind: "message", oneof: "assetData", T: () => asset_types_18.CalloutAsset },
            { no: 21, name: "structured_snippet_asset", kind: "message", oneof: "assetData", T: () => asset_types_17.StructuredSnippetAsset },
            { no: 22, name: "sitelink_asset", kind: "message", oneof: "assetData", T: () => asset_types_16.SitelinkAsset },
            { no: 23, name: "page_feed_asset", kind: "message", oneof: "assetData", T: () => asset_types_15.PageFeedAsset },
            { no: 24, name: "dynamic_education_asset", kind: "message", oneof: "assetData", T: () => asset_types_14.DynamicEducationAsset },
            { no: 25, name: "mobile_app_asset", kind: "message", oneof: "assetData", T: () => asset_types_13.MobileAppAsset },
            { no: 26, name: "hotel_callout_asset", kind: "message", oneof: "assetData", T: () => asset_types_12.HotelCalloutAsset },
            { no: 27, name: "call_asset", kind: "message", oneof: "assetData", T: () => asset_types_11.CallAsset },
            { no: 28, name: "price_asset", kind: "message", oneof: "assetData", T: () => asset_types_10.PriceAsset },
            { no: 29, name: "call_to_action_asset", kind: "message", oneof: "assetData", T: () => asset_types_9.CallToActionAsset, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 30, name: "dynamic_real_estate_asset", kind: "message", oneof: "assetData", T: () => asset_types_8.DynamicRealEstateAsset },
            { no: 31, name: "dynamic_custom_asset", kind: "message", oneof: "assetData", T: () => asset_types_7.DynamicCustomAsset },
            { no: 32, name: "dynamic_hotels_and_rentals_asset", kind: "message", oneof: "assetData", T: () => asset_types_6.DynamicHotelsAndRentalsAsset },
            { no: 33, name: "dynamic_flights_asset", kind: "message", oneof: "assetData", T: () => asset_types_5.DynamicFlightsAsset },
            { no: 34, name: "discovery_carousel_card_asset", kind: "message", oneof: "assetData", T: () => asset_types_4.DiscoveryCarouselCardAsset, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 35, name: "dynamic_travel_asset", kind: "message", oneof: "assetData", T: () => asset_types_3.DynamicTravelAsset },
            { no: 36, name: "dynamic_local_asset", kind: "message", oneof: "assetData", T: () => asset_types_2.DynamicLocalAsset },
            { no: 37, name: "dynamic_jobs_asset", kind: "message", oneof: "assetData", T: () => asset_types_1.DynamicJobsAsset }
        ], { "google.api.resource": { type: "googleads.googleapis.com/Asset", pattern: ["customers/{customer_id}/assets/{asset_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", type: 0, finalUrls: [], finalMobileUrls: [], urlCustomParameters: [], source: 0, assetData: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 11:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string name */ 12:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AssetTypeEnum.AssetType type */ 4:
                    message.type = reader.int32();
                    break;
                case /* repeated string final_urls */ 14:
                    message.finalUrls.push(reader.string());
                    break;
                case /* repeated string final_mobile_urls */ 16:
                    message.finalMobileUrls.push(reader.string());
                    break;
                case /* optional string tracking_url_template */ 17:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters */ 18:
                    message.urlCustomParameters.push(custom_parameter_1.CustomParameter.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string final_url_suffix */ 19:
                    message.finalUrlSuffix = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AssetSourceEnum.AssetSource source */ 38:
                    message.source = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.AssetPolicySummary policy_summary */ 13:
                    message.policySummary = exports.AssetPolicySummary.internalBinaryRead(reader, reader.uint32(), options, message.policySummary);
                    break;
                case /* google.ads.googleads.v11.common.YoutubeVideoAsset youtube_video_asset */ 5:
                    message.assetData = {
                        oneofKind: "youtubeVideoAsset",
                        youtubeVideoAsset: asset_types_25.YoutubeVideoAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.youtubeVideoAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MediaBundleAsset media_bundle_asset */ 6:
                    message.assetData = {
                        oneofKind: "mediaBundleAsset",
                        mediaBundleAsset: asset_types_24.MediaBundleAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.mediaBundleAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ImageAsset image_asset */ 7:
                    message.assetData = {
                        oneofKind: "imageAsset",
                        imageAsset: asset_types_23.ImageAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.imageAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TextAsset text_asset */ 8:
                    message.assetData = {
                        oneofKind: "textAsset",
                        textAsset: asset_types_22.TextAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.textAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LeadFormAsset lead_form_asset */ 9:
                    message.assetData = {
                        oneofKind: "leadFormAsset",
                        leadFormAsset: asset_types_21.LeadFormAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.leadFormAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.BookOnGoogleAsset book_on_google_asset */ 10:
                    message.assetData = {
                        oneofKind: "bookOnGoogleAsset",
                        bookOnGoogleAsset: asset_types_20.BookOnGoogleAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.bookOnGoogleAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PromotionAsset promotion_asset */ 15:
                    message.assetData = {
                        oneofKind: "promotionAsset",
                        promotionAsset: asset_types_19.PromotionAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.promotionAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CalloutAsset callout_asset */ 20:
                    message.assetData = {
                        oneofKind: "calloutAsset",
                        calloutAsset: asset_types_18.CalloutAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.calloutAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.StructuredSnippetAsset structured_snippet_asset */ 21:
                    message.assetData = {
                        oneofKind: "structuredSnippetAsset",
                        structuredSnippetAsset: asset_types_17.StructuredSnippetAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.structuredSnippetAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.SitelinkAsset sitelink_asset */ 22:
                    message.assetData = {
                        oneofKind: "sitelinkAsset",
                        sitelinkAsset: asset_types_16.SitelinkAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.sitelinkAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PageFeedAsset page_feed_asset */ 23:
                    message.assetData = {
                        oneofKind: "pageFeedAsset",
                        pageFeedAsset: asset_types_15.PageFeedAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.pageFeedAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicEducationAsset dynamic_education_asset */ 24:
                    message.assetData = {
                        oneofKind: "dynamicEducationAsset",
                        dynamicEducationAsset: asset_types_14.DynamicEducationAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.dynamicEducationAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MobileAppAsset mobile_app_asset */ 25:
                    message.assetData = {
                        oneofKind: "mobileAppAsset",
                        mobileAppAsset: asset_types_13.MobileAppAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.mobileAppAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.HotelCalloutAsset hotel_callout_asset */ 26:
                    message.assetData = {
                        oneofKind: "hotelCalloutAsset",
                        hotelCalloutAsset: asset_types_12.HotelCalloutAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.hotelCalloutAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CallAsset call_asset */ 27:
                    message.assetData = {
                        oneofKind: "callAsset",
                        callAsset: asset_types_11.CallAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.callAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PriceAsset price_asset */ 28:
                    message.assetData = {
                        oneofKind: "priceAsset",
                        priceAsset: asset_types_10.PriceAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.priceAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CallToActionAsset call_to_action_asset */ 29:
                    message.assetData = {
                        oneofKind: "callToActionAsset",
                        callToActionAsset: asset_types_9.CallToActionAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.callToActionAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicRealEstateAsset dynamic_real_estate_asset */ 30:
                    message.assetData = {
                        oneofKind: "dynamicRealEstateAsset",
                        dynamicRealEstateAsset: asset_types_8.DynamicRealEstateAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.dynamicRealEstateAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicCustomAsset dynamic_custom_asset */ 31:
                    message.assetData = {
                        oneofKind: "dynamicCustomAsset",
                        dynamicCustomAsset: asset_types_7.DynamicCustomAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.dynamicCustomAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicHotelsAndRentalsAsset dynamic_hotels_and_rentals_asset */ 32:
                    message.assetData = {
                        oneofKind: "dynamicHotelsAndRentalsAsset",
                        dynamicHotelsAndRentalsAsset: asset_types_6.DynamicHotelsAndRentalsAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.dynamicHotelsAndRentalsAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicFlightsAsset dynamic_flights_asset */ 33:
                    message.assetData = {
                        oneofKind: "dynamicFlightsAsset",
                        dynamicFlightsAsset: asset_types_5.DynamicFlightsAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.dynamicFlightsAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DiscoveryCarouselCardAsset discovery_carousel_card_asset */ 34:
                    message.assetData = {
                        oneofKind: "discoveryCarouselCardAsset",
                        discoveryCarouselCardAsset: asset_types_4.DiscoveryCarouselCardAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.discoveryCarouselCardAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicTravelAsset dynamic_travel_asset */ 35:
                    message.assetData = {
                        oneofKind: "dynamicTravelAsset",
                        dynamicTravelAsset: asset_types_3.DynamicTravelAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.dynamicTravelAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicLocalAsset dynamic_local_asset */ 36:
                    message.assetData = {
                        oneofKind: "dynamicLocalAsset",
                        dynamicLocalAsset: asset_types_2.DynamicLocalAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.dynamicLocalAsset)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicJobsAsset dynamic_jobs_asset */ 37:
                    message.assetData = {
                        oneofKind: "dynamicJobsAsset",
                        dynamicJobsAsset: asset_types_1.DynamicJobsAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetData.dynamicJobsAsset)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 11; */
        if (message.id !== undefined)
            writer.tag(11, runtime_1.WireType.Varint).int64(message.id);
        /* optional string name = 12; */
        if (message.name !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.enums.AssetTypeEnum.AssetType type = 4; */
        if (message.type !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.type);
        /* repeated string final_urls = 14; */
        for (let i = 0; i < message.finalUrls.length; i++)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.finalUrls[i]);
        /* repeated string final_mobile_urls = 16; */
        for (let i = 0; i < message.finalMobileUrls.length; i++)
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.finalMobileUrls[i]);
        /* optional string tracking_url_template = 17; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(17, runtime_1.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters = 18; */
        for (let i = 0; i < message.urlCustomParameters.length; i++)
            custom_parameter_1.CustomParameter.internalBinaryWrite(message.urlCustomParameters[i], writer.tag(18, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string final_url_suffix = 19; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(19, runtime_1.WireType.LengthDelimited).string(message.finalUrlSuffix);
        /* google.ads.googleads.v11.enums.AssetSourceEnum.AssetSource source = 38; */
        if (message.source !== 0)
            writer.tag(38, runtime_1.WireType.Varint).int32(message.source);
        /* google.ads.googleads.v11.resources.AssetPolicySummary policy_summary = 13; */
        if (message.policySummary)
            exports.AssetPolicySummary.internalBinaryWrite(message.policySummary, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.YoutubeVideoAsset youtube_video_asset = 5; */
        if (message.assetData.oneofKind === "youtubeVideoAsset")
            asset_types_25.YoutubeVideoAsset.internalBinaryWrite(message.assetData.youtubeVideoAsset, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MediaBundleAsset media_bundle_asset = 6; */
        if (message.assetData.oneofKind === "mediaBundleAsset")
            asset_types_24.MediaBundleAsset.internalBinaryWrite(message.assetData.mediaBundleAsset, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ImageAsset image_asset = 7; */
        if (message.assetData.oneofKind === "imageAsset")
            asset_types_23.ImageAsset.internalBinaryWrite(message.assetData.imageAsset, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TextAsset text_asset = 8; */
        if (message.assetData.oneofKind === "textAsset")
            asset_types_22.TextAsset.internalBinaryWrite(message.assetData.textAsset, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LeadFormAsset lead_form_asset = 9; */
        if (message.assetData.oneofKind === "leadFormAsset")
            asset_types_21.LeadFormAsset.internalBinaryWrite(message.assetData.leadFormAsset, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.BookOnGoogleAsset book_on_google_asset = 10; */
        if (message.assetData.oneofKind === "bookOnGoogleAsset")
            asset_types_20.BookOnGoogleAsset.internalBinaryWrite(message.assetData.bookOnGoogleAsset, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PromotionAsset promotion_asset = 15; */
        if (message.assetData.oneofKind === "promotionAsset")
            asset_types_19.PromotionAsset.internalBinaryWrite(message.assetData.promotionAsset, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CalloutAsset callout_asset = 20; */
        if (message.assetData.oneofKind === "calloutAsset")
            asset_types_18.CalloutAsset.internalBinaryWrite(message.assetData.calloutAsset, writer.tag(20, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.StructuredSnippetAsset structured_snippet_asset = 21; */
        if (message.assetData.oneofKind === "structuredSnippetAsset")
            asset_types_17.StructuredSnippetAsset.internalBinaryWrite(message.assetData.structuredSnippetAsset, writer.tag(21, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.SitelinkAsset sitelink_asset = 22; */
        if (message.assetData.oneofKind === "sitelinkAsset")
            asset_types_16.SitelinkAsset.internalBinaryWrite(message.assetData.sitelinkAsset, writer.tag(22, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PageFeedAsset page_feed_asset = 23; */
        if (message.assetData.oneofKind === "pageFeedAsset")
            asset_types_15.PageFeedAsset.internalBinaryWrite(message.assetData.pageFeedAsset, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicEducationAsset dynamic_education_asset = 24; */
        if (message.assetData.oneofKind === "dynamicEducationAsset")
            asset_types_14.DynamicEducationAsset.internalBinaryWrite(message.assetData.dynamicEducationAsset, writer.tag(24, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MobileAppAsset mobile_app_asset = 25; */
        if (message.assetData.oneofKind === "mobileAppAsset")
            asset_types_13.MobileAppAsset.internalBinaryWrite(message.assetData.mobileAppAsset, writer.tag(25, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HotelCalloutAsset hotel_callout_asset = 26; */
        if (message.assetData.oneofKind === "hotelCalloutAsset")
            asset_types_12.HotelCalloutAsset.internalBinaryWrite(message.assetData.hotelCalloutAsset, writer.tag(26, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CallAsset call_asset = 27; */
        if (message.assetData.oneofKind === "callAsset")
            asset_types_11.CallAsset.internalBinaryWrite(message.assetData.callAsset, writer.tag(27, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PriceAsset price_asset = 28; */
        if (message.assetData.oneofKind === "priceAsset")
            asset_types_10.PriceAsset.internalBinaryWrite(message.assetData.priceAsset, writer.tag(28, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CallToActionAsset call_to_action_asset = 29; */
        if (message.assetData.oneofKind === "callToActionAsset")
            asset_types_9.CallToActionAsset.internalBinaryWrite(message.assetData.callToActionAsset, writer.tag(29, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicRealEstateAsset dynamic_real_estate_asset = 30; */
        if (message.assetData.oneofKind === "dynamicRealEstateAsset")
            asset_types_8.DynamicRealEstateAsset.internalBinaryWrite(message.assetData.dynamicRealEstateAsset, writer.tag(30, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicCustomAsset dynamic_custom_asset = 31; */
        if (message.assetData.oneofKind === "dynamicCustomAsset")
            asset_types_7.DynamicCustomAsset.internalBinaryWrite(message.assetData.dynamicCustomAsset, writer.tag(31, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicHotelsAndRentalsAsset dynamic_hotels_and_rentals_asset = 32; */
        if (message.assetData.oneofKind === "dynamicHotelsAndRentalsAsset")
            asset_types_6.DynamicHotelsAndRentalsAsset.internalBinaryWrite(message.assetData.dynamicHotelsAndRentalsAsset, writer.tag(32, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicFlightsAsset dynamic_flights_asset = 33; */
        if (message.assetData.oneofKind === "dynamicFlightsAsset")
            asset_types_5.DynamicFlightsAsset.internalBinaryWrite(message.assetData.dynamicFlightsAsset, writer.tag(33, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DiscoveryCarouselCardAsset discovery_carousel_card_asset = 34; */
        if (message.assetData.oneofKind === "discoveryCarouselCardAsset")
            asset_types_4.DiscoveryCarouselCardAsset.internalBinaryWrite(message.assetData.discoveryCarouselCardAsset, writer.tag(34, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicTravelAsset dynamic_travel_asset = 35; */
        if (message.assetData.oneofKind === "dynamicTravelAsset")
            asset_types_3.DynamicTravelAsset.internalBinaryWrite(message.assetData.dynamicTravelAsset, writer.tag(35, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicLocalAsset dynamic_local_asset = 36; */
        if (message.assetData.oneofKind === "dynamicLocalAsset")
            asset_types_2.DynamicLocalAsset.internalBinaryWrite(message.assetData.dynamicLocalAsset, writer.tag(36, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicJobsAsset dynamic_jobs_asset = 37; */
        if (message.assetData.oneofKind === "dynamicJobsAsset")
            asset_types_1.DynamicJobsAsset.internalBinaryWrite(message.assetData.dynamicJobsAsset, writer.tag(37, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Asset
 */
exports.Asset = new Asset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AssetPolicySummary$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AssetPolicySummary", [
            { no: 1, name: "policy_topic_entries", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => policy_1.PolicyTopicEntry, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "review_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus", policy_review_status_1.PolicyReviewStatusEnum_PolicyReviewStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "approval_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus", policy_approval_status_1.PolicyApprovalStatusEnum_PolicyApprovalStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { policyTopicEntries: [], reviewStatus: 0, approvalStatus: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.PolicyTopicEntry policy_topic_entries */ 1:
                    message.policyTopicEntries.push(policy_1.PolicyTopicEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus review_status */ 2:
                    message.reviewStatus = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus approval_status */ 3:
                    message.approvalStatus = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.PolicyTopicEntry policy_topic_entries = 1; */
        for (let i = 0; i < message.policyTopicEntries.length; i++)
            policy_1.PolicyTopicEntry.internalBinaryWrite(message.policyTopicEntries[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus review_status = 2; */
        if (message.reviewStatus !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.reviewStatus);
        /* google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus approval_status = 3; */
        if (message.approvalStatus !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.approvalStatus);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AssetPolicySummary
 */
exports.AssetPolicySummary = new AssetPolicySummary$Type();
