// google-ads-api-client/dist/protos/bindings/ads/resources/asset_set.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSet_MerchantCenterFeed = exports.AssetSet = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const asset_set_status_1 = require("../enums/asset_set_status");
const asset_set_type_1 = require("../enums/asset_set_type");
// @generated message type with reflection information, may provide speed optimized methods
class AssetSet$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AssetSet", [
            { no: 6, name: "id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AssetSet" } } },
            { no: 2, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 3, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AssetSetTypeEnum.AssetSetType", asset_set_type_1.AssetSetTypeEnum_AssetSetType], options: { "google.api.field_behavior": ["REQUIRED", "IMMUTABLE"] } },
            { no: 4, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AssetSetStatusEnum.AssetSetStatus", asset_set_status_1.AssetSetStatusEnum_AssetSetStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "merchant_center_feed", kind: "message", T: () => exports.AssetSet_MerchantCenterFeed }
        ], { "google.api.resource": { type: "googleads.googleapis.com/AssetSet", pattern: ["customers/{customer_id}/assetSets/{asset_set_id}"] } });
    }
    create(value) {
        const message = { id: 0n, resourceName: "", name: "", type: 0, status: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int64 id */ 6:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* string name */ 2:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AssetSetTypeEnum.AssetSetType type */ 3:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.AssetSetStatusEnum.AssetSetStatus status */ 4:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.AssetSet.MerchantCenterFeed merchant_center_feed */ 5:
                    message.merchantCenterFeed = exports.AssetSet_MerchantCenterFeed.internalBinaryRead(reader, reader.uint32(), options, message.merchantCenterFeed);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int64 id = 6; */
        if (message.id !== 0n)
            writer.tag(6, runtime_1.WireType.Varint).int64(message.id);
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* string name = 2; */
        if (message.name !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.enums.AssetSetTypeEnum.AssetSetType type = 3; */
        if (message.type !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.AssetSetStatusEnum.AssetSetStatus status = 4; */
        if (message.status !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.resources.AssetSet.MerchantCenterFeed merchant_center_feed = 5; */
        if (message.merchantCenterFeed)
            exports.AssetSet_MerchantCenterFeed.internalBinaryWrite(message.merchantCenterFeed, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AssetSet
 */
exports.AssetSet = new AssetSet$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AssetSet_MerchantCenterFeed$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AssetSet.MerchantCenterFeed", [
            { no: 1, name: "merchant_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "feed_label", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = { merchantId: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int64 merchant_id */ 1:
                    message.merchantId = reader.int64().toBigInt();
                    break;
                case /* optional string feed_label */ 2:
                    message.feedLabel = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int64 merchant_id = 1; */
        if (message.merchantId !== 0n)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.merchantId);
        /* optional string feed_label = 2; */
        if (message.feedLabel !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.feedLabel);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AssetSet.MerchantCenterFeed
 */
exports.AssetSet_MerchantCenterFeed = new AssetSet_MerchantCenterFeed$Type();
