// google-ads-api-client/dist/protos/bindings/ads/resources/campaign.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Campaign_CategoryBid = exports.Campaign_LocalServicesCampaignSettings = exports.Campaign_AudienceSetting = exports.Campaign_OptimizationGoalSetting = exports.Campaign_SelectiveOptimization = exports.Campaign_VanityPharma = exports.Campaign_AppCampaignSetting = exports.Campaign_LocalCampaignSetting = exports.Campaign_GeoTargetTypeSetting = exports.Campaign_TrackingSetting = exports.Campaign_ShoppingSetting = exports.Campaign_DynamicSearchAdsSetting = exports.Campaign_HotelSettingInfo = exports.Campaign_NetworkSettings = exports.Campaign_PerformanceMaxUpgrade = exports.Campaign = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const optimization_goal_type_1 = require("../enums/optimization_goal_type");
const vanity_pharma_text_1 = require("../enums/vanity_pharma_text");
const vanity_pharma_display_url_mode_1 = require("../enums/vanity_pharma_display_url_mode");
const app_campaign_app_store_1 = require("../enums/app_campaign_app_store");
const app_campaign_bidding_strategy_goal_type_1 = require("../enums/app_campaign_bidding_strategy_goal_type");
const location_source_type_1 = require("../enums/location_source_type");
const negative_geo_target_type_1 = require("../enums/negative_geo_target_type");
const positive_geo_target_type_1 = require("../enums/positive_geo_target_type");
const performance_max_upgrade_status_1 = require("../enums/performance_max_upgrade_status");
const bidding_1 = require("../common/bidding");
const bidding_2 = require("../common/bidding");
const bidding_3 = require("../common/bidding");
const bidding_4 = require("../common/bidding");
const bidding_5 = require("../common/bidding");
const bidding_6 = require("../common/bidding");
const bidding_7 = require("../common/bidding");
const bidding_8 = require("../common/bidding");
const bidding_9 = require("../common/bidding");
const bidding_10 = require("../common/bidding");
const bidding_11 = require("../common/bidding");
const bidding_12 = require("../common/bidding");
const bidding_13 = require("../common/bidding");
const asset_field_type_1 = require("../enums/asset_field_type");
const payment_mode_1 = require("../enums/payment_mode");
const brand_safety_suitability_1 = require("../enums/brand_safety_suitability");
const frequency_cap_1 = require("../common/frequency_cap");
const bidding_strategy_type_1 = require("../enums/bidding_strategy_type");
const campaign_experiment_type_1 = require("../enums/campaign_experiment_type");
const targeting_setting_1 = require("../common/targeting_setting");
const real_time_bidding_setting_1 = require("../common/real_time_bidding_setting");
const custom_parameter_1 = require("../common/custom_parameter");
const advertising_channel_sub_type_1 = require("../enums/advertising_channel_sub_type");
const advertising_channel_type_1 = require("../enums/advertising_channel_type");
const ad_serving_optimization_status_1 = require("../enums/ad_serving_optimization_status");
const bidding_strategy_system_status_1 = require("../enums/bidding_strategy_system_status");
const campaign_serving_status_1 = require("../enums/campaign_serving_status");
const campaign_status_1 = require("../enums/campaign_status");
// @generated message type with reflection information, may provide speed optimized methods
class Campaign$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 59, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 58, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CampaignStatusEnum.CampaignStatus", campaign_status_1.CampaignStatusEnum_CampaignStatus] },
            { no: 21, name: "serving_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CampaignServingStatusEnum.CampaignServingStatus", campaign_serving_status_1.CampaignServingStatusEnum_CampaignServingStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 78, name: "bidding_strategy_system_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.BiddingStrategySystemStatusEnum.BiddingStrategySystemStatus", bidding_strategy_system_status_1.BiddingStrategySystemStatusEnum_BiddingStrategySystemStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "ad_serving_optimization_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AdServingOptimizationStatusEnum.AdServingOptimizationStatus", ad_serving_optimization_status_1.AdServingOptimizationStatusEnum_AdServingOptimizationStatus] },
            { no: 9, name: "advertising_channel_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AdvertisingChannelTypeEnum.AdvertisingChannelType", advertising_channel_type_1.AdvertisingChannelTypeEnum_AdvertisingChannelType], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 10, name: "advertising_channel_sub_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AdvertisingChannelSubTypeEnum.AdvertisingChannelSubType", advertising_channel_sub_type_1.AdvertisingChannelSubTypeEnum_AdvertisingChannelSubType], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 60, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "url_custom_parameters", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => custom_parameter_1.CustomParameter },
            { no: 75, name: "local_services_campaign_settings", kind: "message", T: () => exports.Campaign_LocalServicesCampaignSettings },
            { no: 39, name: "real_time_bidding_setting", kind: "message", T: () => real_time_bidding_setting_1.RealTimeBiddingSetting },
            { no: 14, name: "network_settings", kind: "message", T: () => exports.Campaign_NetworkSettings },
            { no: 32, name: "hotel_setting", kind: "message", T: () => exports.Campaign_HotelSettingInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 33, name: "dynamic_search_ads_setting", kind: "message", T: () => exports.Campaign_DynamicSearchAdsSetting },
            { no: 36, name: "shopping_setting", kind: "message", T: () => exports.Campaign_ShoppingSetting },
            { no: 43, name: "targeting_setting", kind: "message", T: () => targeting_setting_1.TargetingSetting },
            { no: 73, name: "audience_setting", kind: "message", T: () => exports.Campaign_AudienceSetting, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 47, name: "geo_target_type_setting", kind: "message", T: () => exports.Campaign_GeoTargetTypeSetting },
            { no: 50, name: "local_campaign_setting", kind: "message", T: () => exports.Campaign_LocalCampaignSetting },
            { no: 51, name: "app_campaign_setting", kind: "message", T: () => exports.Campaign_AppCampaignSetting },
            { no: 61, name: "labels", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignLabel" } } },
            { no: 17, name: "experiment_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CampaignExperimentTypeEnum.CampaignExperimentType", campaign_experiment_type_1.CampaignExperimentTypeEnum_CampaignExperimentType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 56, name: "base_campaign", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 62, name: "campaign_budget", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignBudget" } } },
            { no: 22, name: "bidding_strategy_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.BiddingStrategyTypeEnum.BiddingStrategyType", bidding_strategy_type_1.BiddingStrategyTypeEnum_BiddingStrategyType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 71, name: "accessible_bidding_strategy", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AccessibleBiddingStrategy" } } },
            { no: 63, name: "start_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 76, name: "campaign_group", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignGroup" } } },
            { no: 64, name: "end_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 65, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 40, name: "frequency_caps", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => frequency_cap_1.FrequencyCapEntry },
            { no: 42, name: "video_brand_safety_suitability", kind: "enum", T: () => ["google.ads.googleads.v11.enums.BrandSafetySuitabilityEnum.BrandSafetySuitability", brand_safety_suitability_1.BrandSafetySuitabilityEnum_BrandSafetySuitability], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 44, name: "vanity_pharma", kind: "message", T: () => exports.Campaign_VanityPharma },
            { no: 45, name: "selective_optimization", kind: "message", T: () => exports.Campaign_SelectiveOptimization },
            { no: 54, name: "optimization_goal_setting", kind: "message", T: () => exports.Campaign_OptimizationGoalSetting },
            { no: 46, name: "tracking_setting", kind: "message", T: () => exports.Campaign_TrackingSetting, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 52, name: "payment_mode", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PaymentModeEnum.PaymentMode", payment_mode_1.PaymentModeEnum_PaymentMode] },
            { no: 66, name: "optimization_score", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 69, name: "excluded_parent_asset_field_types", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.ads.googleads.v11.enums.AssetFieldTypeEnum.AssetFieldType", asset_field_type_1.AssetFieldTypeEnum_AssetFieldType] },
            { no: 72, name: "url_expansion_opt_out", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 77, name: "performance_max_upgrade", kind: "message", T: () => exports.Campaign_PerformanceMaxUpgrade, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 67, name: "bidding_strategy", kind: "scalar", oneof: "campaignBiddingStrategy", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/BiddingStrategy" } } },
            { no: 49, name: "commission", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_13.Commission },
            { no: 74, name: "manual_cpa", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_12.ManualCpa },
            { no: 24, name: "manual_cpc", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_11.ManualCpc },
            { no: 25, name: "manual_cpm", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_10.ManualCpm },
            { no: 37, name: "manual_cpv", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_9.ManualCpv, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 30, name: "maximize_conversions", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_8.MaximizeConversions },
            { no: 31, name: "maximize_conversion_value", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_7.MaximizeConversionValue },
            { no: 26, name: "target_cpa", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_6.TargetCpa },
            { no: 48, name: "target_impression_share", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_5.TargetImpressionShare },
            { no: 29, name: "target_roas", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_4.TargetRoas },
            { no: 27, name: "target_spend", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_3.TargetSpend },
            { no: 34, name: "percent_cpc", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_2.PercentCpc },
            { no: 41, name: "target_cpm", kind: "message", oneof: "campaignBiddingStrategy", T: () => bidding_1.TargetCpm }
        ], { "google.api.resource": { type: "googleads.googleapis.com/Campaign", pattern: ["customers/{customer_id}/campaigns/{campaign_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", status: 0, servingStatus: 0, biddingStrategySystemStatus: 0, adServingOptimizationStatus: 0, advertisingChannelType: 0, advertisingChannelSubType: 0, urlCustomParameters: [], labels: [], experimentType: 0, biddingStrategyType: 0, accessibleBiddingStrategy: "", frequencyCaps: [], videoBrandSafetySuitability: 0, paymentMode: 0, excludedParentAssetFieldTypes: [], campaignBiddingStrategy: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 59:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string name */ 58:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.CampaignStatusEnum.CampaignStatus status */ 5:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.CampaignServingStatusEnum.CampaignServingStatus serving_status */ 21:
                    message.servingStatus = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.BiddingStrategySystemStatusEnum.BiddingStrategySystemStatus bidding_strategy_system_status */ 78:
                    message.biddingStrategySystemStatus = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.AdServingOptimizationStatusEnum.AdServingOptimizationStatus ad_serving_optimization_status */ 8:
                    message.adServingOptimizationStatus = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.AdvertisingChannelTypeEnum.AdvertisingChannelType advertising_channel_type */ 9:
                    message.advertisingChannelType = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.AdvertisingChannelSubTypeEnum.AdvertisingChannelSubType advertising_channel_sub_type */ 10:
                    message.advertisingChannelSubType = reader.int32();
                    break;
                case /* optional string tracking_url_template */ 60:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters */ 12:
                    message.urlCustomParameters.push(custom_parameter_1.CustomParameter.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.LocalServicesCampaignSettings local_services_campaign_settings */ 75:
                    message.localServicesCampaignSettings = exports.Campaign_LocalServicesCampaignSettings.internalBinaryRead(reader, reader.uint32(), options, message.localServicesCampaignSettings);
                    break;
                case /* google.ads.googleads.v11.common.RealTimeBiddingSetting real_time_bidding_setting */ 39:
                    message.realTimeBiddingSetting = real_time_bidding_setting_1.RealTimeBiddingSetting.internalBinaryRead(reader, reader.uint32(), options, message.realTimeBiddingSetting);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.NetworkSettings network_settings */ 14:
                    message.networkSettings = exports.Campaign_NetworkSettings.internalBinaryRead(reader, reader.uint32(), options, message.networkSettings);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.HotelSettingInfo hotel_setting */ 32:
                    message.hotelSetting = exports.Campaign_HotelSettingInfo.internalBinaryRead(reader, reader.uint32(), options, message.hotelSetting);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.DynamicSearchAdsSetting dynamic_search_ads_setting */ 33:
                    message.dynamicSearchAdsSetting = exports.Campaign_DynamicSearchAdsSetting.internalBinaryRead(reader, reader.uint32(), options, message.dynamicSearchAdsSetting);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.ShoppingSetting shopping_setting */ 36:
                    message.shoppingSetting = exports.Campaign_ShoppingSetting.internalBinaryRead(reader, reader.uint32(), options, message.shoppingSetting);
                    break;
                case /* google.ads.googleads.v11.common.TargetingSetting targeting_setting */ 43:
                    message.targetingSetting = targeting_setting_1.TargetingSetting.internalBinaryRead(reader, reader.uint32(), options, message.targetingSetting);
                    break;
                case /* optional google.ads.googleads.v11.resources.Campaign.AudienceSetting audience_setting */ 73:
                    message.audienceSetting = exports.Campaign_AudienceSetting.internalBinaryRead(reader, reader.uint32(), options, message.audienceSetting);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.GeoTargetTypeSetting geo_target_type_setting */ 47:
                    message.geoTargetTypeSetting = exports.Campaign_GeoTargetTypeSetting.internalBinaryRead(reader, reader.uint32(), options, message.geoTargetTypeSetting);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.LocalCampaignSetting local_campaign_setting */ 50:
                    message.localCampaignSetting = exports.Campaign_LocalCampaignSetting.internalBinaryRead(reader, reader.uint32(), options, message.localCampaignSetting);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.AppCampaignSetting app_campaign_setting */ 51:
                    message.appCampaignSetting = exports.Campaign_AppCampaignSetting.internalBinaryRead(reader, reader.uint32(), options, message.appCampaignSetting);
                    break;
                case /* repeated string labels */ 61:
                    message.labels.push(reader.string());
                    break;
                case /* google.ads.googleads.v11.enums.CampaignExperimentTypeEnum.CampaignExperimentType experiment_type */ 17:
                    message.experimentType = reader.int32();
                    break;
                case /* optional string base_campaign */ 56:
                    message.baseCampaign = reader.string();
                    break;
                case /* optional string campaign_budget */ 62:
                    message.campaignBudget = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.BiddingStrategyTypeEnum.BiddingStrategyType bidding_strategy_type */ 22:
                    message.biddingStrategyType = reader.int32();
                    break;
                case /* string accessible_bidding_strategy */ 71:
                    message.accessibleBiddingStrategy = reader.string();
                    break;
                case /* optional string start_date */ 63:
                    message.startDate = reader.string();
                    break;
                case /* optional string campaign_group */ 76:
                    message.campaignGroup = reader.string();
                    break;
                case /* optional string end_date */ 64:
                    message.endDate = reader.string();
                    break;
                case /* optional string final_url_suffix */ 65:
                    message.finalUrlSuffix = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.FrequencyCapEntry frequency_caps */ 40:
                    message.frequencyCaps.push(frequency_cap_1.FrequencyCapEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.BrandSafetySuitabilityEnum.BrandSafetySuitability video_brand_safety_suitability */ 42:
                    message.videoBrandSafetySuitability = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.VanityPharma vanity_pharma */ 44:
                    message.vanityPharma = exports.Campaign_VanityPharma.internalBinaryRead(reader, reader.uint32(), options, message.vanityPharma);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.SelectiveOptimization selective_optimization */ 45:
                    message.selectiveOptimization = exports.Campaign_SelectiveOptimization.internalBinaryRead(reader, reader.uint32(), options, message.selectiveOptimization);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.OptimizationGoalSetting optimization_goal_setting */ 54:
                    message.optimizationGoalSetting = exports.Campaign_OptimizationGoalSetting.internalBinaryRead(reader, reader.uint32(), options, message.optimizationGoalSetting);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.TrackingSetting tracking_setting */ 46:
                    message.trackingSetting = exports.Campaign_TrackingSetting.internalBinaryRead(reader, reader.uint32(), options, message.trackingSetting);
                    break;
                case /* google.ads.googleads.v11.enums.PaymentModeEnum.PaymentMode payment_mode */ 52:
                    message.paymentMode = reader.int32();
                    break;
                case /* optional double optimization_score */ 66:
                    message.optimizationScore = reader.double();
                    break;
                case /* repeated google.ads.googleads.v11.enums.AssetFieldTypeEnum.AssetFieldType excluded_parent_asset_field_types */ 69:
                    if (wireType === runtime_2.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.excludedParentAssetFieldTypes.push(reader.int32());
                    else
                        message.excludedParentAssetFieldTypes.push(reader.int32());
                    break;
                case /* optional bool url_expansion_opt_out */ 72:
                    message.urlExpansionOptOut = reader.bool();
                    break;
                case /* google.ads.googleads.v11.resources.Campaign.PerformanceMaxUpgrade performance_max_upgrade */ 77:
                    message.performanceMaxUpgrade = exports.Campaign_PerformanceMaxUpgrade.internalBinaryRead(reader, reader.uint32(), options, message.performanceMaxUpgrade);
                    break;
                case /* string bidding_strategy */ 67:
                    message.campaignBiddingStrategy = {
                        oneofKind: "biddingStrategy",
                        biddingStrategy: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.common.Commission commission */ 49:
                    message.campaignBiddingStrategy = {
                        oneofKind: "commission",
                        commission: bidding_13.Commission.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.commission)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ManualCpa manual_cpa */ 74:
                    message.campaignBiddingStrategy = {
                        oneofKind: "manualCpa",
                        manualCpa: bidding_12.ManualCpa.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.manualCpa)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ManualCpc manual_cpc */ 24:
                    message.campaignBiddingStrategy = {
                        oneofKind: "manualCpc",
                        manualCpc: bidding_11.ManualCpc.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.manualCpc)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ManualCpm manual_cpm */ 25:
                    message.campaignBiddingStrategy = {
                        oneofKind: "manualCpm",
                        manualCpm: bidding_10.ManualCpm.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.manualCpm)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ManualCpv manual_cpv */ 37:
                    message.campaignBiddingStrategy = {
                        oneofKind: "manualCpv",
                        manualCpv: bidding_9.ManualCpv.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.manualCpv)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MaximizeConversions maximize_conversions */ 30:
                    message.campaignBiddingStrategy = {
                        oneofKind: "maximizeConversions",
                        maximizeConversions: bidding_8.MaximizeConversions.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.maximizeConversions)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MaximizeConversionValue maximize_conversion_value */ 31:
                    message.campaignBiddingStrategy = {
                        oneofKind: "maximizeConversionValue",
                        maximizeConversionValue: bidding_7.MaximizeConversionValue.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.maximizeConversionValue)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetCpa target_cpa */ 26:
                    message.campaignBiddingStrategy = {
                        oneofKind: "targetCpa",
                        targetCpa: bidding_6.TargetCpa.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.targetCpa)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetImpressionShare target_impression_share */ 48:
                    message.campaignBiddingStrategy = {
                        oneofKind: "targetImpressionShare",
                        targetImpressionShare: bidding_5.TargetImpressionShare.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.targetImpressionShare)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetRoas target_roas */ 29:
                    message.campaignBiddingStrategy = {
                        oneofKind: "targetRoas",
                        targetRoas: bidding_4.TargetRoas.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.targetRoas)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetSpend target_spend */ 27:
                    message.campaignBiddingStrategy = {
                        oneofKind: "targetSpend",
                        targetSpend: bidding_3.TargetSpend.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.targetSpend)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PercentCpc percent_cpc */ 34:
                    message.campaignBiddingStrategy = {
                        oneofKind: "percentCpc",
                        percentCpc: bidding_2.PercentCpc.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.percentCpc)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetCpm target_cpm */ 41:
                    message.campaignBiddingStrategy = {
                        oneofKind: "targetCpm",
                        targetCpm: bidding_1.TargetCpm.internalBinaryRead(reader, reader.uint32(), options, message.campaignBiddingStrategy.targetCpm)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_2.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 59; */
        if (message.id !== undefined)
            writer.tag(59, runtime_2.WireType.Varint).int64(message.id);
        /* optional string name = 58; */
        if (message.name !== undefined)
            writer.tag(58, runtime_2.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.enums.CampaignStatusEnum.CampaignStatus status = 5; */
        if (message.status !== 0)
            writer.tag(5, runtime_2.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.enums.CampaignServingStatusEnum.CampaignServingStatus serving_status = 21; */
        if (message.servingStatus !== 0)
            writer.tag(21, runtime_2.WireType.Varint).int32(message.servingStatus);
        /* google.ads.googleads.v11.enums.BiddingStrategySystemStatusEnum.BiddingStrategySystemStatus bidding_strategy_system_status = 78; */
        if (message.biddingStrategySystemStatus !== 0)
            writer.tag(78, runtime_2.WireType.Varint).int32(message.biddingStrategySystemStatus);
        /* google.ads.googleads.v11.enums.AdServingOptimizationStatusEnum.AdServingOptimizationStatus ad_serving_optimization_status = 8; */
        if (message.adServingOptimizationStatus !== 0)
            writer.tag(8, runtime_2.WireType.Varint).int32(message.adServingOptimizationStatus);
        /* google.ads.googleads.v11.enums.AdvertisingChannelTypeEnum.AdvertisingChannelType advertising_channel_type = 9; */
        if (message.advertisingChannelType !== 0)
            writer.tag(9, runtime_2.WireType.Varint).int32(message.advertisingChannelType);
        /* google.ads.googleads.v11.enums.AdvertisingChannelSubTypeEnum.AdvertisingChannelSubType advertising_channel_sub_type = 10; */
        if (message.advertisingChannelSubType !== 0)
            writer.tag(10, runtime_2.WireType.Varint).int32(message.advertisingChannelSubType);
        /* optional string tracking_url_template = 60; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(60, runtime_2.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters = 12; */
        for (let i = 0; i < message.urlCustomParameters.length; i++)
            custom_parameter_1.CustomParameter.internalBinaryWrite(message.urlCustomParameters[i], writer.tag(12, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.LocalServicesCampaignSettings local_services_campaign_settings = 75; */
        if (message.localServicesCampaignSettings)
            exports.Campaign_LocalServicesCampaignSettings.internalBinaryWrite(message.localServicesCampaignSettings, writer.tag(75, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.RealTimeBiddingSetting real_time_bidding_setting = 39; */
        if (message.realTimeBiddingSetting)
            real_time_bidding_setting_1.RealTimeBiddingSetting.internalBinaryWrite(message.realTimeBiddingSetting, writer.tag(39, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.NetworkSettings network_settings = 14; */
        if (message.networkSettings)
            exports.Campaign_NetworkSettings.internalBinaryWrite(message.networkSettings, writer.tag(14, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.HotelSettingInfo hotel_setting = 32; */
        if (message.hotelSetting)
            exports.Campaign_HotelSettingInfo.internalBinaryWrite(message.hotelSetting, writer.tag(32, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.DynamicSearchAdsSetting dynamic_search_ads_setting = 33; */
        if (message.dynamicSearchAdsSetting)
            exports.Campaign_DynamicSearchAdsSetting.internalBinaryWrite(message.dynamicSearchAdsSetting, writer.tag(33, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.ShoppingSetting shopping_setting = 36; */
        if (message.shoppingSetting)
            exports.Campaign_ShoppingSetting.internalBinaryWrite(message.shoppingSetting, writer.tag(36, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetingSetting targeting_setting = 43; */
        if (message.targetingSetting)
            targeting_setting_1.TargetingSetting.internalBinaryWrite(message.targetingSetting, writer.tag(43, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* optional google.ads.googleads.v11.resources.Campaign.AudienceSetting audience_setting = 73; */
        if (message.audienceSetting)
            exports.Campaign_AudienceSetting.internalBinaryWrite(message.audienceSetting, writer.tag(73, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.GeoTargetTypeSetting geo_target_type_setting = 47; */
        if (message.geoTargetTypeSetting)
            exports.Campaign_GeoTargetTypeSetting.internalBinaryWrite(message.geoTargetTypeSetting, writer.tag(47, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.LocalCampaignSetting local_campaign_setting = 50; */
        if (message.localCampaignSetting)
            exports.Campaign_LocalCampaignSetting.internalBinaryWrite(message.localCampaignSetting, writer.tag(50, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.AppCampaignSetting app_campaign_setting = 51; */
        if (message.appCampaignSetting)
            exports.Campaign_AppCampaignSetting.internalBinaryWrite(message.appCampaignSetting, writer.tag(51, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* repeated string labels = 61; */
        for (let i = 0; i < message.labels.length; i++)
            writer.tag(61, runtime_2.WireType.LengthDelimited).string(message.labels[i]);
        /* google.ads.googleads.v11.enums.CampaignExperimentTypeEnum.CampaignExperimentType experiment_type = 17; */
        if (message.experimentType !== 0)
            writer.tag(17, runtime_2.WireType.Varint).int32(message.experimentType);
        /* optional string base_campaign = 56; */
        if (message.baseCampaign !== undefined)
            writer.tag(56, runtime_2.WireType.LengthDelimited).string(message.baseCampaign);
        /* optional string campaign_budget = 62; */
        if (message.campaignBudget !== undefined)
            writer.tag(62, runtime_2.WireType.LengthDelimited).string(message.campaignBudget);
        /* google.ads.googleads.v11.enums.BiddingStrategyTypeEnum.BiddingStrategyType bidding_strategy_type = 22; */
        if (message.biddingStrategyType !== 0)
            writer.tag(22, runtime_2.WireType.Varint).int32(message.biddingStrategyType);
        /* string accessible_bidding_strategy = 71; */
        if (message.accessibleBiddingStrategy !== "")
            writer.tag(71, runtime_2.WireType.LengthDelimited).string(message.accessibleBiddingStrategy);
        /* optional string start_date = 63; */
        if (message.startDate !== undefined)
            writer.tag(63, runtime_2.WireType.LengthDelimited).string(message.startDate);
        /* optional string campaign_group = 76; */
        if (message.campaignGroup !== undefined)
            writer.tag(76, runtime_2.WireType.LengthDelimited).string(message.campaignGroup);
        /* optional string end_date = 64; */
        if (message.endDate !== undefined)
            writer.tag(64, runtime_2.WireType.LengthDelimited).string(message.endDate);
        /* optional string final_url_suffix = 65; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(65, runtime_2.WireType.LengthDelimited).string(message.finalUrlSuffix);
        /* repeated google.ads.googleads.v11.common.FrequencyCapEntry frequency_caps = 40; */
        for (let i = 0; i < message.frequencyCaps.length; i++)
            frequency_cap_1.FrequencyCapEntry.internalBinaryWrite(message.frequencyCaps[i], writer.tag(40, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.BrandSafetySuitabilityEnum.BrandSafetySuitability video_brand_safety_suitability = 42; */
        if (message.videoBrandSafetySuitability !== 0)
            writer.tag(42, runtime_2.WireType.Varint).int32(message.videoBrandSafetySuitability);
        /* google.ads.googleads.v11.resources.Campaign.VanityPharma vanity_pharma = 44; */
        if (message.vanityPharma)
            exports.Campaign_VanityPharma.internalBinaryWrite(message.vanityPharma, writer.tag(44, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.SelectiveOptimization selective_optimization = 45; */
        if (message.selectiveOptimization)
            exports.Campaign_SelectiveOptimization.internalBinaryWrite(message.selectiveOptimization, writer.tag(45, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.OptimizationGoalSetting optimization_goal_setting = 54; */
        if (message.optimizationGoalSetting)
            exports.Campaign_OptimizationGoalSetting.internalBinaryWrite(message.optimizationGoalSetting, writer.tag(54, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign.TrackingSetting tracking_setting = 46; */
        if (message.trackingSetting)
            exports.Campaign_TrackingSetting.internalBinaryWrite(message.trackingSetting, writer.tag(46, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.PaymentModeEnum.PaymentMode payment_mode = 52; */
        if (message.paymentMode !== 0)
            writer.tag(52, runtime_2.WireType.Varint).int32(message.paymentMode);
        /* optional double optimization_score = 66; */
        if (message.optimizationScore !== undefined)
            writer.tag(66, runtime_2.WireType.Bit64).double(message.optimizationScore);
        /* repeated google.ads.googleads.v11.enums.AssetFieldTypeEnum.AssetFieldType excluded_parent_asset_field_types = 69; */
        if (message.excludedParentAssetFieldTypes.length) {
            writer.tag(69, runtime_2.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.excludedParentAssetFieldTypes.length; i++)
                writer.int32(message.excludedParentAssetFieldTypes[i]);
            writer.join();
        }
        /* optional bool url_expansion_opt_out = 72; */
        if (message.urlExpansionOptOut !== undefined)
            writer.tag(72, runtime_2.WireType.Varint).bool(message.urlExpansionOptOut);
        /* google.ads.googleads.v11.resources.Campaign.PerformanceMaxUpgrade performance_max_upgrade = 77; */
        if (message.performanceMaxUpgrade)
            exports.Campaign_PerformanceMaxUpgrade.internalBinaryWrite(message.performanceMaxUpgrade, writer.tag(77, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* string bidding_strategy = 67; */
        if (message.campaignBiddingStrategy.oneofKind === "biddingStrategy")
            writer.tag(67, runtime_2.WireType.LengthDelimited).string(message.campaignBiddingStrategy.biddingStrategy);
        /* google.ads.googleads.v11.common.Commission commission = 49; */
        if (message.campaignBiddingStrategy.oneofKind === "commission")
            bidding_13.Commission.internalBinaryWrite(message.campaignBiddingStrategy.commission, writer.tag(49, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ManualCpa manual_cpa = 74; */
        if (message.campaignBiddingStrategy.oneofKind === "manualCpa")
            bidding_12.ManualCpa.internalBinaryWrite(message.campaignBiddingStrategy.manualCpa, writer.tag(74, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ManualCpc manual_cpc = 24; */
        if (message.campaignBiddingStrategy.oneofKind === "manualCpc")
            bidding_11.ManualCpc.internalBinaryWrite(message.campaignBiddingStrategy.manualCpc, writer.tag(24, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ManualCpm manual_cpm = 25; */
        if (message.campaignBiddingStrategy.oneofKind === "manualCpm")
            bidding_10.ManualCpm.internalBinaryWrite(message.campaignBiddingStrategy.manualCpm, writer.tag(25, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ManualCpv manual_cpv = 37; */
        if (message.campaignBiddingStrategy.oneofKind === "manualCpv")
            bidding_9.ManualCpv.internalBinaryWrite(message.campaignBiddingStrategy.manualCpv, writer.tag(37, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MaximizeConversions maximize_conversions = 30; */
        if (message.campaignBiddingStrategy.oneofKind === "maximizeConversions")
            bidding_8.MaximizeConversions.internalBinaryWrite(message.campaignBiddingStrategy.maximizeConversions, writer.tag(30, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MaximizeConversionValue maximize_conversion_value = 31; */
        if (message.campaignBiddingStrategy.oneofKind === "maximizeConversionValue")
            bidding_7.MaximizeConversionValue.internalBinaryWrite(message.campaignBiddingStrategy.maximizeConversionValue, writer.tag(31, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetCpa target_cpa = 26; */
        if (message.campaignBiddingStrategy.oneofKind === "targetCpa")
            bidding_6.TargetCpa.internalBinaryWrite(message.campaignBiddingStrategy.targetCpa, writer.tag(26, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetImpressionShare target_impression_share = 48; */
        if (message.campaignBiddingStrategy.oneofKind === "targetImpressionShare")
            bidding_5.TargetImpressionShare.internalBinaryWrite(message.campaignBiddingStrategy.targetImpressionShare, writer.tag(48, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetRoas target_roas = 29; */
        if (message.campaignBiddingStrategy.oneofKind === "targetRoas")
            bidding_4.TargetRoas.internalBinaryWrite(message.campaignBiddingStrategy.targetRoas, writer.tag(29, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetSpend target_spend = 27; */
        if (message.campaignBiddingStrategy.oneofKind === "targetSpend")
            bidding_3.TargetSpend.internalBinaryWrite(message.campaignBiddingStrategy.targetSpend, writer.tag(27, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PercentCpc percent_cpc = 34; */
        if (message.campaignBiddingStrategy.oneofKind === "percentCpc")
            bidding_2.PercentCpc.internalBinaryWrite(message.campaignBiddingStrategy.percentCpc, writer.tag(34, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetCpm target_cpm = 41; */
        if (message.campaignBiddingStrategy.oneofKind === "targetCpm")
            bidding_1.TargetCpm.internalBinaryWrite(message.campaignBiddingStrategy.targetCpm, writer.tag(41, runtime_2.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign
 */
exports.Campaign = new Campaign$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_PerformanceMaxUpgrade$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.PerformanceMaxUpgrade", [
            { no: 1, name: "performance_max_campaign", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 2, name: "pre_upgrade_campaign", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 3, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PerformanceMaxUpgradeStatusEnum.PerformanceMaxUpgradeStatus", performance_max_upgrade_status_1.PerformanceMaxUpgradeStatusEnum_PerformanceMaxUpgradeStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { performanceMaxCampaign: "", preUpgradeCampaign: "", status: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string performance_max_campaign */ 1:
                    message.performanceMaxCampaign = reader.string();
                    break;
                case /* string pre_upgrade_campaign */ 2:
                    message.preUpgradeCampaign = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.PerformanceMaxUpgradeStatusEnum.PerformanceMaxUpgradeStatus status */ 3:
                    message.status = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string performance_max_campaign = 1; */
        if (message.performanceMaxCampaign !== "")
            writer.tag(1, runtime_2.WireType.LengthDelimited).string(message.performanceMaxCampaign);
        /* string pre_upgrade_campaign = 2; */
        if (message.preUpgradeCampaign !== "")
            writer.tag(2, runtime_2.WireType.LengthDelimited).string(message.preUpgradeCampaign);
        /* google.ads.googleads.v11.enums.PerformanceMaxUpgradeStatusEnum.PerformanceMaxUpgradeStatus status = 3; */
        if (message.status !== 0)
            writer.tag(3, runtime_2.WireType.Varint).int32(message.status);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.PerformanceMaxUpgrade
 */
exports.Campaign_PerformanceMaxUpgrade = new Campaign_PerformanceMaxUpgrade$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_NetworkSettings$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.NetworkSettings", [
            { no: 5, name: "target_google_search", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 6, name: "target_search_network", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "target_content_network", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 8, name: "target_partner_search_network", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional bool target_google_search */ 5:
                    message.targetGoogleSearch = reader.bool();
                    break;
                case /* optional bool target_search_network */ 6:
                    message.targetSearchNetwork = reader.bool();
                    break;
                case /* optional bool target_content_network */ 7:
                    message.targetContentNetwork = reader.bool();
                    break;
                case /* optional bool target_partner_search_network */ 8:
                    message.targetPartnerSearchNetwork = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional bool target_google_search = 5; */
        if (message.targetGoogleSearch !== undefined)
            writer.tag(5, runtime_2.WireType.Varint).bool(message.targetGoogleSearch);
        /* optional bool target_search_network = 6; */
        if (message.targetSearchNetwork !== undefined)
            writer.tag(6, runtime_2.WireType.Varint).bool(message.targetSearchNetwork);
        /* optional bool target_content_network = 7; */
        if (message.targetContentNetwork !== undefined)
            writer.tag(7, runtime_2.WireType.Varint).bool(message.targetContentNetwork);
        /* optional bool target_partner_search_network = 8; */
        if (message.targetPartnerSearchNetwork !== undefined)
            writer.tag(8, runtime_2.WireType.Varint).bool(message.targetPartnerSearchNetwork);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.NetworkSettings
 */
exports.Campaign_NetworkSettings = new Campaign_NetworkSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_HotelSettingInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.HotelSettingInfo", [
            { no: 2, name: "hotel_center_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 hotel_center_id */ 2:
                    message.hotelCenterId = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 hotel_center_id = 2; */
        if (message.hotelCenterId !== undefined)
            writer.tag(2, runtime_2.WireType.Varint).int64(message.hotelCenterId);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.HotelSettingInfo
 */
exports.Campaign_HotelSettingInfo = new Campaign_HotelSettingInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_DynamicSearchAdsSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.DynamicSearchAdsSetting", [
            { no: 6, name: "domain_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 7, name: "language_code", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 8, name: "use_supplied_urls_only", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 9, name: "feeds", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/Feed" } } }
        ]);
    }
    create(value) {
        const message = { domainName: "", languageCode: "", feeds: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string domain_name */ 6:
                    message.domainName = reader.string();
                    break;
                case /* string language_code */ 7:
                    message.languageCode = reader.string();
                    break;
                case /* optional bool use_supplied_urls_only */ 8:
                    message.useSuppliedUrlsOnly = reader.bool();
                    break;
                case /* repeated string feeds */ 9:
                    message.feeds.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string domain_name = 6; */
        if (message.domainName !== "")
            writer.tag(6, runtime_2.WireType.LengthDelimited).string(message.domainName);
        /* string language_code = 7; */
        if (message.languageCode !== "")
            writer.tag(7, runtime_2.WireType.LengthDelimited).string(message.languageCode);
        /* optional bool use_supplied_urls_only = 8; */
        if (message.useSuppliedUrlsOnly !== undefined)
            writer.tag(8, runtime_2.WireType.Varint).bool(message.useSuppliedUrlsOnly);
        /* repeated string feeds = 9; */
        for (let i = 0; i < message.feeds.length; i++)
            writer.tag(9, runtime_2.WireType.LengthDelimited).string(message.feeds[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.DynamicSearchAdsSetting
 */
exports.Campaign_DynamicSearchAdsSetting = new Campaign_DynamicSearchAdsSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_ShoppingSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.ShoppingSetting", [
            { no: 5, name: "merchant_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "sales_country", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "feed_label", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "campaign_priority", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 8, name: "enable_local", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 9, name: "use_vehicle_inventory", kind: "scalar", T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ]);
    }
    create(value) {
        const message = { feedLabel: "", useVehicleInventory: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 merchant_id */ 5:
                    message.merchantId = reader.int64().toBigInt();
                    break;
                case /* optional string sales_country */ 6:
                    message.salesCountry = reader.string();
                    break;
                case /* string feed_label */ 10:
                    message.feedLabel = reader.string();
                    break;
                case /* optional int32 campaign_priority */ 7:
                    message.campaignPriority = reader.int32();
                    break;
                case /* optional bool enable_local */ 8:
                    message.enableLocal = reader.bool();
                    break;
                case /* bool use_vehicle_inventory */ 9:
                    message.useVehicleInventory = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 merchant_id = 5; */
        if (message.merchantId !== undefined)
            writer.tag(5, runtime_2.WireType.Varint).int64(message.merchantId);
        /* optional string sales_country = 6; */
        if (message.salesCountry !== undefined)
            writer.tag(6, runtime_2.WireType.LengthDelimited).string(message.salesCountry);
        /* string feed_label = 10; */
        if (message.feedLabel !== "")
            writer.tag(10, runtime_2.WireType.LengthDelimited).string(message.feedLabel);
        /* optional int32 campaign_priority = 7; */
        if (message.campaignPriority !== undefined)
            writer.tag(7, runtime_2.WireType.Varint).int32(message.campaignPriority);
        /* optional bool enable_local = 8; */
        if (message.enableLocal !== undefined)
            writer.tag(8, runtime_2.WireType.Varint).bool(message.enableLocal);
        /* bool use_vehicle_inventory = 9; */
        if (message.useVehicleInventory !== false)
            writer.tag(9, runtime_2.WireType.Varint).bool(message.useVehicleInventory);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.ShoppingSetting
 */
exports.Campaign_ShoppingSetting = new Campaign_ShoppingSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_TrackingSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.TrackingSetting", [
            { no: 2, name: "tracking_url", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string tracking_url */ 2:
                    message.trackingUrl = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string tracking_url = 2; */
        if (message.trackingUrl !== undefined)
            writer.tag(2, runtime_2.WireType.LengthDelimited).string(message.trackingUrl);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.TrackingSetting
 */
exports.Campaign_TrackingSetting = new Campaign_TrackingSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_GeoTargetTypeSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.GeoTargetTypeSetting", [
            { no: 1, name: "positive_geo_target_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PositiveGeoTargetTypeEnum.PositiveGeoTargetType", positive_geo_target_type_1.PositiveGeoTargetTypeEnum_PositiveGeoTargetType] },
            { no: 2, name: "negative_geo_target_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.NegativeGeoTargetTypeEnum.NegativeGeoTargetType", negative_geo_target_type_1.NegativeGeoTargetTypeEnum_NegativeGeoTargetType] }
        ]);
    }
    create(value) {
        const message = { positiveGeoTargetType: 0, negativeGeoTargetType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.PositiveGeoTargetTypeEnum.PositiveGeoTargetType positive_geo_target_type */ 1:
                    message.positiveGeoTargetType = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.NegativeGeoTargetTypeEnum.NegativeGeoTargetType negative_geo_target_type */ 2:
                    message.negativeGeoTargetType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.PositiveGeoTargetTypeEnum.PositiveGeoTargetType positive_geo_target_type = 1; */
        if (message.positiveGeoTargetType !== 0)
            writer.tag(1, runtime_2.WireType.Varint).int32(message.positiveGeoTargetType);
        /* google.ads.googleads.v11.enums.NegativeGeoTargetTypeEnum.NegativeGeoTargetType negative_geo_target_type = 2; */
        if (message.negativeGeoTargetType !== 0)
            writer.tag(2, runtime_2.WireType.Varint).int32(message.negativeGeoTargetType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.GeoTargetTypeSetting
 */
exports.Campaign_GeoTargetTypeSetting = new Campaign_GeoTargetTypeSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_LocalCampaignSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.LocalCampaignSetting", [
            { no: 1, name: "location_source_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.LocationSourceTypeEnum.LocationSourceType", location_source_type_1.LocationSourceTypeEnum_LocationSourceType] }
        ]);
    }
    create(value) {
        const message = { locationSourceType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.LocationSourceTypeEnum.LocationSourceType location_source_type */ 1:
                    message.locationSourceType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.LocationSourceTypeEnum.LocationSourceType location_source_type = 1; */
        if (message.locationSourceType !== 0)
            writer.tag(1, runtime_2.WireType.Varint).int32(message.locationSourceType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.LocalCampaignSetting
 */
exports.Campaign_LocalCampaignSetting = new Campaign_LocalCampaignSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_AppCampaignSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.AppCampaignSetting", [
            { no: 1, name: "bidding_strategy_goal_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AppCampaignBiddingStrategyGoalTypeEnum.AppCampaignBiddingStrategyGoalType", app_campaign_bidding_strategy_goal_type_1.AppCampaignBiddingStrategyGoalTypeEnum_AppCampaignBiddingStrategyGoalType] },
            { no: 4, name: "app_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 3, name: "app_store", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AppCampaignAppStoreEnum.AppCampaignAppStore", app_campaign_app_store_1.AppCampaignAppStoreEnum_AppCampaignAppStore], options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ]);
    }
    create(value) {
        const message = { biddingStrategyGoalType: 0, appStore: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.AppCampaignBiddingStrategyGoalTypeEnum.AppCampaignBiddingStrategyGoalType bidding_strategy_goal_type */ 1:
                    message.biddingStrategyGoalType = reader.int32();
                    break;
                case /* optional string app_id */ 4:
                    message.appId = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.AppCampaignAppStoreEnum.AppCampaignAppStore app_store */ 3:
                    message.appStore = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.AppCampaignBiddingStrategyGoalTypeEnum.AppCampaignBiddingStrategyGoalType bidding_strategy_goal_type = 1; */
        if (message.biddingStrategyGoalType !== 0)
            writer.tag(1, runtime_2.WireType.Varint).int32(message.biddingStrategyGoalType);
        /* optional string app_id = 4; */
        if (message.appId !== undefined)
            writer.tag(4, runtime_2.WireType.LengthDelimited).string(message.appId);
        /* google.ads.googleads.v11.enums.AppCampaignAppStoreEnum.AppCampaignAppStore app_store = 3; */
        if (message.appStore !== 0)
            writer.tag(3, runtime_2.WireType.Varint).int32(message.appStore);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.AppCampaignSetting
 */
exports.Campaign_AppCampaignSetting = new Campaign_AppCampaignSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_VanityPharma$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.VanityPharma", [
            { no: 1, name: "vanity_pharma_display_url_mode", kind: "enum", T: () => ["google.ads.googleads.v11.enums.VanityPharmaDisplayUrlModeEnum.VanityPharmaDisplayUrlMode", vanity_pharma_display_url_mode_1.VanityPharmaDisplayUrlModeEnum_VanityPharmaDisplayUrlMode] },
            { no: 2, name: "vanity_pharma_text", kind: "enum", T: () => ["google.ads.googleads.v11.enums.VanityPharmaTextEnum.VanityPharmaText", vanity_pharma_text_1.VanityPharmaTextEnum_VanityPharmaText] }
        ]);
    }
    create(value) {
        const message = { vanityPharmaDisplayUrlMode: 0, vanityPharmaText: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.VanityPharmaDisplayUrlModeEnum.VanityPharmaDisplayUrlMode vanity_pharma_display_url_mode */ 1:
                    message.vanityPharmaDisplayUrlMode = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.VanityPharmaTextEnum.VanityPharmaText vanity_pharma_text */ 2:
                    message.vanityPharmaText = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.VanityPharmaDisplayUrlModeEnum.VanityPharmaDisplayUrlMode vanity_pharma_display_url_mode = 1; */
        if (message.vanityPharmaDisplayUrlMode !== 0)
            writer.tag(1, runtime_2.WireType.Varint).int32(message.vanityPharmaDisplayUrlMode);
        /* google.ads.googleads.v11.enums.VanityPharmaTextEnum.VanityPharmaText vanity_pharma_text = 2; */
        if (message.vanityPharmaText !== 0)
            writer.tag(2, runtime_2.WireType.Varint).int32(message.vanityPharmaText);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.VanityPharma
 */
exports.Campaign_VanityPharma = new Campaign_VanityPharma$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_SelectiveOptimization$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.SelectiveOptimization", [
            { no: 2, name: "conversion_actions", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/ConversionAction" } } }
        ]);
    }
    create(value) {
        const message = { conversionActions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated string conversion_actions */ 2:
                    message.conversionActions.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated string conversion_actions = 2; */
        for (let i = 0; i < message.conversionActions.length; i++)
            writer.tag(2, runtime_2.WireType.LengthDelimited).string(message.conversionActions[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.SelectiveOptimization
 */
exports.Campaign_SelectiveOptimization = new Campaign_SelectiveOptimization$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_OptimizationGoalSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.OptimizationGoalSetting", [
            { no: 1, name: "optimization_goal_types", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.ads.googleads.v11.enums.OptimizationGoalTypeEnum.OptimizationGoalType", optimization_goal_type_1.OptimizationGoalTypeEnum_OptimizationGoalType] }
        ]);
    }
    create(value) {
        const message = { optimizationGoalTypes: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.enums.OptimizationGoalTypeEnum.OptimizationGoalType optimization_goal_types */ 1:
                    if (wireType === runtime_2.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.optimizationGoalTypes.push(reader.int32());
                    else
                        message.optimizationGoalTypes.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.enums.OptimizationGoalTypeEnum.OptimizationGoalType optimization_goal_types = 1; */
        if (message.optimizationGoalTypes.length) {
            writer.tag(1, runtime_2.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.optimizationGoalTypes.length; i++)
                writer.int32(message.optimizationGoalTypes[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.OptimizationGoalSetting
 */
exports.Campaign_OptimizationGoalSetting = new Campaign_OptimizationGoalSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_AudienceSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.AudienceSetting", [
            { no: 1, name: "use_audience_grouped", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional bool use_audience_grouped */ 1:
                    message.useAudienceGrouped = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional bool use_audience_grouped = 1; */
        if (message.useAudienceGrouped !== undefined)
            writer.tag(1, runtime_2.WireType.Varint).bool(message.useAudienceGrouped);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.AudienceSetting
 */
exports.Campaign_AudienceSetting = new Campaign_AudienceSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_LocalServicesCampaignSettings$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.LocalServicesCampaignSettings", [
            { no: 1, name: "category_bids", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.Campaign_CategoryBid }
        ]);
    }
    create(value) {
        const message = { categoryBids: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.resources.Campaign.CategoryBid category_bids */ 1:
                    message.categoryBids.push(exports.Campaign_CategoryBid.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.resources.Campaign.CategoryBid category_bids = 1; */
        for (let i = 0; i < message.categoryBids.length; i++)
            exports.Campaign_CategoryBid.internalBinaryWrite(message.categoryBids[i], writer.tag(1, runtime_2.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.LocalServicesCampaignSettings
 */
exports.Campaign_LocalServicesCampaignSettings = new Campaign_LocalServicesCampaignSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Campaign_CategoryBid$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Campaign.CategoryBid", [
            { no: 1, name: "category_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "manual_cpa_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string category_id */ 1:
                    message.categoryId = reader.string();
                    break;
                case /* optional int64 manual_cpa_bid_micros */ 2:
                    message.manualCpaBidMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string category_id = 1; */
        if (message.categoryId !== undefined)
            writer.tag(1, runtime_2.WireType.LengthDelimited).string(message.categoryId);
        /* optional int64 manual_cpa_bid_micros = 2; */
        if (message.manualCpaBidMicros !== undefined)
            writer.tag(2, runtime_2.WireType.Varint).int64(message.manualCpaBidMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Campaign.CategoryBid
 */
exports.Campaign_CategoryBid = new Campaign_CategoryBid$Type();
