// google-ads-api-client/dist/protos/bindings/ads/resources/campaign_criterion_simulation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignCriterionSimulation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const simulation_1 = require("../common/simulation");
const simulation_modification_method_1 = require("../enums/simulation_modification_method");
const simulation_type_1 = require("../enums/simulation_type");
// @generated message type with reflection information, may provide speed optimized methods
class CampaignCriterionSimulation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CampaignCriterionSimulation", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignCriterionSimulation" } } },
            { no: 9, name: "campaign_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "criterion_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType", simulation_type_1.SimulationTypeEnum_SimulationType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "modification_method", kind: "enum", T: () => ["google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod", simulation_modification_method_1.SimulationModificationMethodEnum_SimulationModificationMethod], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "start_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 12, name: "end_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "bid_modifier_point_list", kind: "message", oneof: "pointList", T: () => simulation_1.BidModifierSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/CampaignCriterionSimulation", pattern: ["customers/{customer_id}/campaignCriterionSimulations/{campaign_id}~{criterion_id}~{type}~{modification_method}~{start_date}~{end_date}"] } });
    }
    create(value) {
        const message = { resourceName: "", type: 0, modificationMethod: 0, pointList: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 campaign_id */ 9:
                    message.campaignId = reader.int64().toBigInt();
                    break;
                case /* optional int64 criterion_id */ 10:
                    message.criterionId = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType type */ 4:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod modification_method */ 5:
                    message.modificationMethod = reader.int32();
                    break;
                case /* optional string start_date */ 11:
                    message.startDate = reader.string();
                    break;
                case /* optional string end_date */ 12:
                    message.endDate = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.BidModifierSimulationPointList bid_modifier_point_list */ 8:
                    message.pointList = {
                        oneofKind: "bidModifierPointList",
                        bidModifierPointList: simulation_1.BidModifierSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.bidModifierPointList)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 campaign_id = 9; */
        if (message.campaignId !== undefined)
            writer.tag(9, runtime_1.WireType.Varint).int64(message.campaignId);
        /* optional int64 criterion_id = 10; */
        if (message.criterionId !== undefined)
            writer.tag(10, runtime_1.WireType.Varint).int64(message.criterionId);
        /* google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType type = 4; */
        if (message.type !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod modification_method = 5; */
        if (message.modificationMethod !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.modificationMethod);
        /* optional string start_date = 11; */
        if (message.startDate !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* optional string end_date = 12; */
        if (message.endDate !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.endDate);
        /* google.ads.googleads.v11.common.BidModifierSimulationPointList bid_modifier_point_list = 8; */
        if (message.pointList.oneofKind === "bidModifierPointList")
            simulation_1.BidModifierSimulationPointList.internalBinaryWrite(message.pointList.bidModifierPointList, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CampaignCriterionSimulation
 */
exports.CampaignCriterionSimulation = new CampaignCriterionSimulation$Type();
