// google-ads-api-client/dist/protos/bindings/ads/resources/campaign_simulation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignSimulation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const simulation_1 = require("../common/simulation");
const simulation_2 = require("../common/simulation");
const simulation_3 = require("../common/simulation");
const simulation_4 = require("../common/simulation");
const simulation_5 = require("../common/simulation");
const simulation_modification_method_1 = require("../enums/simulation_modification_method");
const simulation_type_1 = require("../enums/simulation_type");
// @generated message type with reflection information, may provide speed optimized methods
class CampaignSimulation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CampaignSimulation", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignSimulation" } } },
            { no: 2, name: "campaign_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType", simulation_type_1.SimulationTypeEnum_SimulationType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "modification_method", kind: "enum", T: () => ["google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod", simulation_modification_method_1.SimulationModificationMethodEnum_SimulationModificationMethod], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "start_date", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "end_date", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "cpc_bid_point_list", kind: "message", oneof: "pointList", T: () => simulation_5.CpcBidSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "target_cpa_point_list", kind: "message", oneof: "pointList", T: () => simulation_4.TargetCpaSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "target_roas_point_list", kind: "message", oneof: "pointList", T: () => simulation_3.TargetRoasSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "target_impression_share_point_list", kind: "message", oneof: "pointList", T: () => simulation_2.TargetImpressionShareSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "budget_point_list", kind: "message", oneof: "pointList", T: () => simulation_1.BudgetSimulationPointList, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/CampaignSimulation", pattern: ["customers/{customer_id}/campaignSimulations/{campaign_id}~{type}~{modification_method}~{start_date}~{end_date}"] } });
    }
    create(value) {
        const message = { resourceName: "", campaignId: 0n, type: 0, modificationMethod: 0, startDate: "", endDate: "", pointList: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* int64 campaign_id */ 2:
                    message.campaignId = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType type */ 3:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod modification_method */ 4:
                    message.modificationMethod = reader.int32();
                    break;
                case /* string start_date */ 5:
                    message.startDate = reader.string();
                    break;
                case /* string end_date */ 6:
                    message.endDate = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.CpcBidSimulationPointList cpc_bid_point_list */ 7:
                    message.pointList = {
                        oneofKind: "cpcBidPointList",
                        cpcBidPointList: simulation_5.CpcBidSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.cpcBidPointList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetCpaSimulationPointList target_cpa_point_list */ 8:
                    message.pointList = {
                        oneofKind: "targetCpaPointList",
                        targetCpaPointList: simulation_4.TargetCpaSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.targetCpaPointList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetRoasSimulationPointList target_roas_point_list */ 9:
                    message.pointList = {
                        oneofKind: "targetRoasPointList",
                        targetRoasPointList: simulation_3.TargetRoasSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.targetRoasPointList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TargetImpressionShareSimulationPointList target_impression_share_point_list */ 10:
                    message.pointList = {
                        oneofKind: "targetImpressionSharePointList",
                        targetImpressionSharePointList: simulation_2.TargetImpressionShareSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.targetImpressionSharePointList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.BudgetSimulationPointList budget_point_list */ 11:
                    message.pointList = {
                        oneofKind: "budgetPointList",
                        budgetPointList: simulation_1.BudgetSimulationPointList.internalBinaryRead(reader, reader.uint32(), options, message.pointList.budgetPointList)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* int64 campaign_id = 2; */
        if (message.campaignId !== 0n)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.campaignId);
        /* google.ads.googleads.v11.enums.SimulationTypeEnum.SimulationType type = 3; */
        if (message.type !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.SimulationModificationMethodEnum.SimulationModificationMethod modification_method = 4; */
        if (message.modificationMethod !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.modificationMethod);
        /* string start_date = 5; */
        if (message.startDate !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.startDate);
        /* string end_date = 6; */
        if (message.endDate !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.endDate);
        /* google.ads.googleads.v11.common.CpcBidSimulationPointList cpc_bid_point_list = 7; */
        if (message.pointList.oneofKind === "cpcBidPointList")
            simulation_5.CpcBidSimulationPointList.internalBinaryWrite(message.pointList.cpcBidPointList, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetCpaSimulationPointList target_cpa_point_list = 8; */
        if (message.pointList.oneofKind === "targetCpaPointList")
            simulation_4.TargetCpaSimulationPointList.internalBinaryWrite(message.pointList.targetCpaPointList, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetRoasSimulationPointList target_roas_point_list = 9; */
        if (message.pointList.oneofKind === "targetRoasPointList")
            simulation_3.TargetRoasSimulationPointList.internalBinaryWrite(message.pointList.targetRoasPointList, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TargetImpressionShareSimulationPointList target_impression_share_point_list = 10; */
        if (message.pointList.oneofKind === "targetImpressionSharePointList")
            simulation_2.TargetImpressionShareSimulationPointList.internalBinaryWrite(message.pointList.targetImpressionSharePointList, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.BudgetSimulationPointList budget_point_list = 11; */
        if (message.pointList.oneofKind === "budgetPointList")
            simulation_1.BudgetSimulationPointList.internalBinaryWrite(message.pointList.budgetPointList, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CampaignSimulation
 */
exports.CampaignSimulation = new CampaignSimulation$Type();
