// google-ads-api-client/dist/protos/bindings/ads/resources/campaign_bid_modifier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignBidModifier = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const criteria_1 = require("../common/criteria");
// @generated message type with reflection information, may provide speed optimized methods
class CampaignBidModifier$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CampaignBidModifier", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignBidModifier" } } },
            { no: 6, name: "campaign", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 7, name: "criterion_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "bid_modifier", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 5, name: "interaction_type", kind: "message", oneof: "criterion", T: () => criteria_1.InteractionTypeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/CampaignBidModifier", pattern: ["customers/{customer_id}/campaignBidModifiers/{campaign_id}~{criterion_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", criterion: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional string campaign */ 6:
                    message.campaign = reader.string();
                    break;
                case /* optional int64 criterion_id */ 7:
                    message.criterionId = reader.int64().toBigInt();
                    break;
                case /* optional double bid_modifier */ 8:
                    message.bidModifier = reader.double();
                    break;
                case /* google.ads.googleads.v11.common.InteractionTypeInfo interaction_type */ 5:
                    message.criterion = {
                        oneofKind: "interactionType",
                        interactionType: criteria_1.InteractionTypeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.interactionType)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional string campaign = 6; */
        if (message.campaign !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.campaign);
        /* optional int64 criterion_id = 7; */
        if (message.criterionId !== undefined)
            writer.tag(7, runtime_1.WireType.Varint).int64(message.criterionId);
        /* optional double bid_modifier = 8; */
        if (message.bidModifier !== undefined)
            writer.tag(8, runtime_1.WireType.Bit64).double(message.bidModifier);
        /* google.ads.googleads.v11.common.InteractionTypeInfo interaction_type = 5; */
        if (message.criterion.oneofKind === "interactionType")
            criteria_1.InteractionTypeInfo.internalBinaryWrite(message.criterion.interactionType, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CampaignBidModifier
 */
exports.CampaignBidModifier = new CampaignBidModifier$Type();
