// google-ads-api-client/dist/protos/bindings/ads/resources/campaign_criterion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignCriterion = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const criteria_1 = require("../common/criteria");
const criteria_2 = require("../common/criteria");
const criteria_3 = require("../common/criteria");
const criteria_4 = require("../common/criteria");
const criteria_5 = require("../common/criteria");
const criteria_6 = require("../common/criteria");
const criteria_7 = require("../common/criteria");
const criteria_8 = require("../common/criteria");
const criteria_9 = require("../common/criteria");
const criteria_10 = require("../common/criteria");
const criteria_11 = require("../common/criteria");
const criteria_12 = require("../common/criteria");
const criteria_13 = require("../common/criteria");
const criteria_14 = require("../common/criteria");
const criteria_15 = require("../common/criteria");
const criteria_16 = require("../common/criteria");
const criteria_17 = require("../common/criteria");
const criteria_18 = require("../common/criteria");
const criteria_19 = require("../common/criteria");
const criteria_20 = require("../common/criteria");
const criteria_21 = require("../common/criteria");
const criteria_22 = require("../common/criteria");
const criteria_23 = require("../common/criteria");
const criteria_24 = require("../common/criteria");
const criteria_25 = require("../common/criteria");
const criteria_26 = require("../common/criteria");
const criteria_27 = require("../common/criteria");
const criteria_28 = require("../common/criteria");
const criteria_29 = require("../common/criteria");
const criteria_30 = require("../common/criteria");
const campaign_criterion_status_1 = require("../enums/campaign_criterion_status");
const criterion_type_1 = require("../enums/criterion_type");
// @generated message type with reflection information, may provide speed optimized methods
class CampaignCriterion$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CampaignCriterion", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignCriterion" } } },
            { no: 37, name: "campaign", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 38, name: "criterion_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 43, name: "display_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 39, name: "bid_modifier", kind: "scalar", opt: true, T: 2 /*ScalarType.FLOAT*/ },
            { no: 40, name: "negative", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CriterionTypeEnum.CriterionType", criterion_type_1.CriterionTypeEnum_CriterionType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 35, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CampaignCriterionStatusEnum.CampaignCriterionStatus", campaign_criterion_status_1.CampaignCriterionStatusEnum_CampaignCriterionStatus] },
            { no: 8, name: "keyword", kind: "message", oneof: "criterion", T: () => criteria_30.KeywordInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 9, name: "placement", kind: "message", oneof: "criterion", T: () => criteria_29.PlacementInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 10, name: "mobile_app_category", kind: "message", oneof: "criterion", T: () => criteria_28.MobileAppCategoryInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 11, name: "mobile_application", kind: "message", oneof: "criterion", T: () => criteria_27.MobileApplicationInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 12, name: "location", kind: "message", oneof: "criterion", T: () => criteria_26.LocationInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 13, name: "device", kind: "message", oneof: "criterion", T: () => criteria_25.DeviceInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 15, name: "ad_schedule", kind: "message", oneof: "criterion", T: () => criteria_24.AdScheduleInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 16, name: "age_range", kind: "message", oneof: "criterion", T: () => criteria_23.AgeRangeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 17, name: "gender", kind: "message", oneof: "criterion", T: () => criteria_22.GenderInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 18, name: "income_range", kind: "message", oneof: "criterion", T: () => criteria_21.IncomeRangeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 19, name: "parental_status", kind: "message", oneof: "criterion", T: () => criteria_20.ParentalStatusInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 22, name: "user_list", kind: "message", oneof: "criterion", T: () => criteria_19.UserListInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 20, name: "youtube_video", kind: "message", oneof: "criterion", T: () => criteria_18.YouTubeVideoInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 21, name: "youtube_channel", kind: "message", oneof: "criterion", T: () => criteria_17.YouTubeChannelInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 23, name: "proximity", kind: "message", oneof: "criterion", T: () => criteria_16.ProximityInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 24, name: "topic", kind: "message", oneof: "criterion", T: () => criteria_15.TopicInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 25, name: "listing_scope", kind: "message", oneof: "criterion", T: () => criteria_14.ListingScopeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 26, name: "language", kind: "message", oneof: "criterion", T: () => criteria_13.LanguageInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 27, name: "ip_block", kind: "message", oneof: "criterion", T: () => criteria_12.IpBlockInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 28, name: "content_label", kind: "message", oneof: "criterion", T: () => criteria_11.ContentLabelInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 29, name: "carrier", kind: "message", oneof: "criterion", T: () => criteria_10.CarrierInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 30, name: "user_interest", kind: "message", oneof: "criterion", T: () => criteria_9.UserInterestInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 31, name: "webpage", kind: "message", oneof: "criterion", T: () => criteria_8.WebpageInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 32, name: "operating_system_version", kind: "message", oneof: "criterion", T: () => criteria_7.OperatingSystemVersionInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 33, name: "mobile_device", kind: "message", oneof: "criterion", T: () => criteria_6.MobileDeviceInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 34, name: "location_group", kind: "message", oneof: "criterion", T: () => criteria_5.LocationGroupInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 36, name: "custom_affinity", kind: "message", oneof: "criterion", T: () => criteria_4.CustomAffinityInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 41, name: "custom_audience", kind: "message", oneof: "criterion", T: () => criteria_3.CustomAudienceInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 42, name: "combined_audience", kind: "message", oneof: "criterion", T: () => criteria_2.CombinedAudienceInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 45, name: "keyword_theme", kind: "message", oneof: "criterion", T: () => criteria_1.KeywordThemeInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/CampaignCriterion", pattern: ["customers/{customer_id}/campaignCriteria/{campaign_id}~{criterion_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", displayName: "", type: 0, status: 0, criterion: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional string campaign */ 37:
                    message.campaign = reader.string();
                    break;
                case /* optional int64 criterion_id */ 38:
                    message.criterionId = reader.int64().toBigInt();
                    break;
                case /* string display_name */ 43:
                    message.displayName = reader.string();
                    break;
                case /* optional float bid_modifier */ 39:
                    message.bidModifier = reader.float();
                    break;
                case /* optional bool negative */ 40:
                    message.negative = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.CriterionTypeEnum.CriterionType type */ 6:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.CampaignCriterionStatusEnum.CampaignCriterionStatus status */ 35:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.KeywordInfo keyword */ 8:
                    message.criterion = {
                        oneofKind: "keyword",
                        keyword: criteria_30.KeywordInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.keyword)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PlacementInfo placement */ 9:
                    message.criterion = {
                        oneofKind: "placement",
                        placement: criteria_29.PlacementInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.placement)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MobileAppCategoryInfo mobile_app_category */ 10:
                    message.criterion = {
                        oneofKind: "mobileAppCategory",
                        mobileAppCategory: criteria_28.MobileAppCategoryInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.mobileAppCategory)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MobileApplicationInfo mobile_application */ 11:
                    message.criterion = {
                        oneofKind: "mobileApplication",
                        mobileApplication: criteria_27.MobileApplicationInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.mobileApplication)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LocationInfo location */ 12:
                    message.criterion = {
                        oneofKind: "location",
                        location: criteria_26.LocationInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.location)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DeviceInfo device */ 13:
                    message.criterion = {
                        oneofKind: "device",
                        device: criteria_25.DeviceInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.device)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AdScheduleInfo ad_schedule */ 15:
                    message.criterion = {
                        oneofKind: "adSchedule",
                        adSchedule: criteria_24.AdScheduleInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.adSchedule)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AgeRangeInfo age_range */ 16:
                    message.criterion = {
                        oneofKind: "ageRange",
                        ageRange: criteria_23.AgeRangeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.ageRange)
                    };
                    break;
                case /* google.ads.googleads.v11.common.GenderInfo gender */ 17:
                    message.criterion = {
                        oneofKind: "gender",
                        gender: criteria_22.GenderInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.gender)
                    };
                    break;
                case /* google.ads.googleads.v11.common.IncomeRangeInfo income_range */ 18:
                    message.criterion = {
                        oneofKind: "incomeRange",
                        incomeRange: criteria_21.IncomeRangeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.incomeRange)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ParentalStatusInfo parental_status */ 19:
                    message.criterion = {
                        oneofKind: "parentalStatus",
                        parentalStatus: criteria_20.ParentalStatusInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.parentalStatus)
                    };
                    break;
                case /* google.ads.googleads.v11.common.UserListInfo user_list */ 22:
                    message.criterion = {
                        oneofKind: "userList",
                        userList: criteria_19.UserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.userList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.YouTubeVideoInfo youtube_video */ 20:
                    message.criterion = {
                        oneofKind: "youtubeVideo",
                        youtubeVideo: criteria_18.YouTubeVideoInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.youtubeVideo)
                    };
                    break;
                case /* google.ads.googleads.v11.common.YouTubeChannelInfo youtube_channel */ 21:
                    message.criterion = {
                        oneofKind: "youtubeChannel",
                        youtubeChannel: criteria_17.YouTubeChannelInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.youtubeChannel)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ProximityInfo proximity */ 23:
                    message.criterion = {
                        oneofKind: "proximity",
                        proximity: criteria_16.ProximityInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.proximity)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TopicInfo topic */ 24:
                    message.criterion = {
                        oneofKind: "topic",
                        topic: criteria_15.TopicInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.topic)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ListingScopeInfo listing_scope */ 25:
                    message.criterion = {
                        oneofKind: "listingScope",
                        listingScope: criteria_14.ListingScopeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.listingScope)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LanguageInfo language */ 26:
                    message.criterion = {
                        oneofKind: "language",
                        language: criteria_13.LanguageInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.language)
                    };
                    break;
                case /* google.ads.googleads.v11.common.IpBlockInfo ip_block */ 27:
                    message.criterion = {
                        oneofKind: "ipBlock",
                        ipBlock: criteria_12.IpBlockInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.ipBlock)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ContentLabelInfo content_label */ 28:
                    message.criterion = {
                        oneofKind: "contentLabel",
                        contentLabel: criteria_11.ContentLabelInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.contentLabel)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CarrierInfo carrier */ 29:
                    message.criterion = {
                        oneofKind: "carrier",
                        carrier: criteria_10.CarrierInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.carrier)
                    };
                    break;
                case /* google.ads.googleads.v11.common.UserInterestInfo user_interest */ 30:
                    message.criterion = {
                        oneofKind: "userInterest",
                        userInterest: criteria_9.UserInterestInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.userInterest)
                    };
                    break;
                case /* google.ads.googleads.v11.common.WebpageInfo webpage */ 31:
                    message.criterion = {
                        oneofKind: "webpage",
                        webpage: criteria_8.WebpageInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.webpage)
                    };
                    break;
                case /* google.ads.googleads.v11.common.OperatingSystemVersionInfo operating_system_version */ 32:
                    message.criterion = {
                        oneofKind: "operatingSystemVersion",
                        operatingSystemVersion: criteria_7.OperatingSystemVersionInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.operatingSystemVersion)
                    };
                    break;
                case /* google.ads.googleads.v11.common.MobileDeviceInfo mobile_device */ 33:
                    message.criterion = {
                        oneofKind: "mobileDevice",
                        mobileDevice: criteria_6.MobileDeviceInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.mobileDevice)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LocationGroupInfo location_group */ 34:
                    message.criterion = {
                        oneofKind: "locationGroup",
                        locationGroup: criteria_5.LocationGroupInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.locationGroup)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CustomAffinityInfo custom_affinity */ 36:
                    message.criterion = {
                        oneofKind: "customAffinity",
                        customAffinity: criteria_4.CustomAffinityInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.customAffinity)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CustomAudienceInfo custom_audience */ 41:
                    message.criterion = {
                        oneofKind: "customAudience",
                        customAudience: criteria_3.CustomAudienceInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.customAudience)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CombinedAudienceInfo combined_audience */ 42:
                    message.criterion = {
                        oneofKind: "combinedAudience",
                        combinedAudience: criteria_2.CombinedAudienceInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.combinedAudience)
                    };
                    break;
                case /* google.ads.googleads.v11.common.KeywordThemeInfo keyword_theme */ 45:
                    message.criterion = {
                        oneofKind: "keywordTheme",
                        keywordTheme: criteria_1.KeywordThemeInfo.internalBinaryRead(reader, reader.uint32(), options, message.criterion.keywordTheme)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional string campaign = 37; */
        if (message.campaign !== undefined)
            writer.tag(37, runtime_1.WireType.LengthDelimited).string(message.campaign);
        /* optional int64 criterion_id = 38; */
        if (message.criterionId !== undefined)
            writer.tag(38, runtime_1.WireType.Varint).int64(message.criterionId);
        /* string display_name = 43; */
        if (message.displayName !== "")
            writer.tag(43, runtime_1.WireType.LengthDelimited).string(message.displayName);
        /* optional float bid_modifier = 39; */
        if (message.bidModifier !== undefined)
            writer.tag(39, runtime_1.WireType.Bit32).float(message.bidModifier);
        /* optional bool negative = 40; */
        if (message.negative !== undefined)
            writer.tag(40, runtime_1.WireType.Varint).bool(message.negative);
        /* google.ads.googleads.v11.enums.CriterionTypeEnum.CriterionType type = 6; */
        if (message.type !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.CampaignCriterionStatusEnum.CampaignCriterionStatus status = 35; */
        if (message.status !== 0)
            writer.tag(35, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.common.KeywordInfo keyword = 8; */
        if (message.criterion.oneofKind === "keyword")
            criteria_30.KeywordInfo.internalBinaryWrite(message.criterion.keyword, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PlacementInfo placement = 9; */
        if (message.criterion.oneofKind === "placement")
            criteria_29.PlacementInfo.internalBinaryWrite(message.criterion.placement, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MobileAppCategoryInfo mobile_app_category = 10; */
        if (message.criterion.oneofKind === "mobileAppCategory")
            criteria_28.MobileAppCategoryInfo.internalBinaryWrite(message.criterion.mobileAppCategory, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MobileApplicationInfo mobile_application = 11; */
        if (message.criterion.oneofKind === "mobileApplication")
            criteria_27.MobileApplicationInfo.internalBinaryWrite(message.criterion.mobileApplication, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LocationInfo location = 12; */
        if (message.criterion.oneofKind === "location")
            criteria_26.LocationInfo.internalBinaryWrite(message.criterion.location, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DeviceInfo device = 13; */
        if (message.criterion.oneofKind === "device")
            criteria_25.DeviceInfo.internalBinaryWrite(message.criterion.device, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AdScheduleInfo ad_schedule = 15; */
        if (message.criterion.oneofKind === "adSchedule")
            criteria_24.AdScheduleInfo.internalBinaryWrite(message.criterion.adSchedule, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AgeRangeInfo age_range = 16; */
        if (message.criterion.oneofKind === "ageRange")
            criteria_23.AgeRangeInfo.internalBinaryWrite(message.criterion.ageRange, writer.tag(16, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.GenderInfo gender = 17; */
        if (message.criterion.oneofKind === "gender")
            criteria_22.GenderInfo.internalBinaryWrite(message.criterion.gender, writer.tag(17, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.IncomeRangeInfo income_range = 18; */
        if (message.criterion.oneofKind === "incomeRange")
            criteria_21.IncomeRangeInfo.internalBinaryWrite(message.criterion.incomeRange, writer.tag(18, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ParentalStatusInfo parental_status = 19; */
        if (message.criterion.oneofKind === "parentalStatus")
            criteria_20.ParentalStatusInfo.internalBinaryWrite(message.criterion.parentalStatus, writer.tag(19, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserListInfo user_list = 22; */
        if (message.criterion.oneofKind === "userList")
            criteria_19.UserListInfo.internalBinaryWrite(message.criterion.userList, writer.tag(22, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.YouTubeVideoInfo youtube_video = 20; */
        if (message.criterion.oneofKind === "youtubeVideo")
            criteria_18.YouTubeVideoInfo.internalBinaryWrite(message.criterion.youtubeVideo, writer.tag(20, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.YouTubeChannelInfo youtube_channel = 21; */
        if (message.criterion.oneofKind === "youtubeChannel")
            criteria_17.YouTubeChannelInfo.internalBinaryWrite(message.criterion.youtubeChannel, writer.tag(21, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ProximityInfo proximity = 23; */
        if (message.criterion.oneofKind === "proximity")
            criteria_16.ProximityInfo.internalBinaryWrite(message.criterion.proximity, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TopicInfo topic = 24; */
        if (message.criterion.oneofKind === "topic")
            criteria_15.TopicInfo.internalBinaryWrite(message.criterion.topic, writer.tag(24, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ListingScopeInfo listing_scope = 25; */
        if (message.criterion.oneofKind === "listingScope")
            criteria_14.ListingScopeInfo.internalBinaryWrite(message.criterion.listingScope, writer.tag(25, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LanguageInfo language = 26; */
        if (message.criterion.oneofKind === "language")
            criteria_13.LanguageInfo.internalBinaryWrite(message.criterion.language, writer.tag(26, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.IpBlockInfo ip_block = 27; */
        if (message.criterion.oneofKind === "ipBlock")
            criteria_12.IpBlockInfo.internalBinaryWrite(message.criterion.ipBlock, writer.tag(27, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ContentLabelInfo content_label = 28; */
        if (message.criterion.oneofKind === "contentLabel")
            criteria_11.ContentLabelInfo.internalBinaryWrite(message.criterion.contentLabel, writer.tag(28, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CarrierInfo carrier = 29; */
        if (message.criterion.oneofKind === "carrier")
            criteria_10.CarrierInfo.internalBinaryWrite(message.criterion.carrier, writer.tag(29, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.UserInterestInfo user_interest = 30; */
        if (message.criterion.oneofKind === "userInterest")
            criteria_9.UserInterestInfo.internalBinaryWrite(message.criterion.userInterest, writer.tag(30, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.WebpageInfo webpage = 31; */
        if (message.criterion.oneofKind === "webpage")
            criteria_8.WebpageInfo.internalBinaryWrite(message.criterion.webpage, writer.tag(31, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.OperatingSystemVersionInfo operating_system_version = 32; */
        if (message.criterion.oneofKind === "operatingSystemVersion")
            criteria_7.OperatingSystemVersionInfo.internalBinaryWrite(message.criterion.operatingSystemVersion, writer.tag(32, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.MobileDeviceInfo mobile_device = 33; */
        if (message.criterion.oneofKind === "mobileDevice")
            criteria_6.MobileDeviceInfo.internalBinaryWrite(message.criterion.mobileDevice, writer.tag(33, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LocationGroupInfo location_group = 34; */
        if (message.criterion.oneofKind === "locationGroup")
            criteria_5.LocationGroupInfo.internalBinaryWrite(message.criterion.locationGroup, writer.tag(34, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CustomAffinityInfo custom_affinity = 36; */
        if (message.criterion.oneofKind === "customAffinity")
            criteria_4.CustomAffinityInfo.internalBinaryWrite(message.criterion.customAffinity, writer.tag(36, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CustomAudienceInfo custom_audience = 41; */
        if (message.criterion.oneofKind === "customAudience")
            criteria_3.CustomAudienceInfo.internalBinaryWrite(message.criterion.customAudience, writer.tag(41, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CombinedAudienceInfo combined_audience = 42; */
        if (message.criterion.oneofKind === "combinedAudience")
            criteria_2.CombinedAudienceInfo.internalBinaryWrite(message.criterion.combinedAudience, writer.tag(42, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.KeywordThemeInfo keyword_theme = 45; */
        if (message.criterion.oneofKind === "keywordTheme")
            criteria_1.KeywordThemeInfo.internalBinaryWrite(message.criterion.keywordTheme, writer.tag(45, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CampaignCriterion
 */
exports.CampaignCriterion = new CampaignCriterion$Type();
