// google-ads-api-client/dist/protos/bindings/ads/resources/change_event.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeEvent_ChangedResource = exports.ChangeEvent = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const campaign_asset_set_1 = require("./campaign_asset_set");
const asset_set_asset_1 = require("./asset_set_asset");
const asset_set_1 = require("./asset_set");
const ad_group_asset_1 = require("./ad_group_asset");
const campaign_asset_1 = require("./campaign_asset");
const customer_asset_1 = require("./customer_asset");
const asset_1 = require("./asset");
const ad_group_ad_1 = require("./ad_group_ad");
const ad_group_feed_1 = require("./ad_group_feed");
const campaign_feed_1 = require("./campaign_feed");
const feed_item_1 = require("./feed_item");
const feed_1 = require("./feed");
const campaign_criterion_1 = require("./campaign_criterion");
const ad_group_bid_modifier_1 = require("./ad_group_bid_modifier");
const campaign_budget_1 = require("./campaign_budget");
const campaign_1 = require("./campaign");
const ad_group_criterion_1 = require("./ad_group_criterion");
const ad_group_1 = require("./ad_group");
const ad_1 = require("./ad");
const field_mask_1 = require("../../google/protobuf/field_mask");
const resource_change_operation_1 = require("../enums/resource_change_operation");
const change_client_type_1 = require("../enums/change_client_type");
const change_event_resource_type_1 = require("../enums/change_event_resource_type");
// @generated message type with reflection information, may provide speed optimized methods
class ChangeEvent$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ChangeEvent", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/ChangeEvent" } } },
            { no: 2, name: "change_date_time", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "change_resource_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ChangeEventResourceTypeEnum.ChangeEventResourceType", change_event_resource_type_1.ChangeEventResourceTypeEnum_ChangeEventResourceType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "change_resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "client_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ChangeClientTypeEnum.ChangeClientType", change_client_type_1.ChangeClientTypeEnum_ChangeClientType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "user_email", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "old_resource", kind: "message", T: () => exports.ChangeEvent_ChangedResource, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "new_resource", kind: "message", T: () => exports.ChangeEvent_ChangedResource, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "resource_change_operation", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ResourceChangeOperationEnum.ResourceChangeOperation", resource_change_operation_1.ResourceChangeOperationEnum_ResourceChangeOperation], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "changed_fields", kind: "message", T: () => field_mask_1.FieldMask, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "campaign", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 12, name: "ad_group", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } },
            { no: 13, name: "feed", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Feed" } } },
            { no: 14, name: "feed_item", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/FeedItem" } } },
            { no: 20, name: "asset", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Asset" } } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/ChangeEvent", pattern: ["customers/{customer_id}/changeEvents/{timestamp_micros}~{command_index}~{mutate_index}"] } });
    }
    create(value) {
        const message = { resourceName: "", changeDateTime: "", changeResourceType: 0, changeResourceName: "", clientType: 0, userEmail: "", resourceChangeOperation: 0, campaign: "", adGroup: "", feed: "", feedItem: "", asset: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* string change_date_time */ 2:
                    message.changeDateTime = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.ChangeEventResourceTypeEnum.ChangeEventResourceType change_resource_type */ 3:
                    message.changeResourceType = reader.int32();
                    break;
                case /* string change_resource_name */ 4:
                    message.changeResourceName = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.ChangeClientTypeEnum.ChangeClientType client_type */ 5:
                    message.clientType = reader.int32();
                    break;
                case /* string user_email */ 6:
                    message.userEmail = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.ChangeEvent.ChangedResource old_resource */ 7:
                    message.oldResource = exports.ChangeEvent_ChangedResource.internalBinaryRead(reader, reader.uint32(), options, message.oldResource);
                    break;
                case /* google.ads.googleads.v11.resources.ChangeEvent.ChangedResource new_resource */ 8:
                    message.newResource = exports.ChangeEvent_ChangedResource.internalBinaryRead(reader, reader.uint32(), options, message.newResource);
                    break;
                case /* google.ads.googleads.v11.enums.ResourceChangeOperationEnum.ResourceChangeOperation resource_change_operation */ 9:
                    message.resourceChangeOperation = reader.int32();
                    break;
                case /* google.protobuf.FieldMask changed_fields */ 10:
                    message.changedFields = field_mask_1.FieldMask.internalBinaryRead(reader, reader.uint32(), options, message.changedFields);
                    break;
                case /* string campaign */ 11:
                    message.campaign = reader.string();
                    break;
                case /* string ad_group */ 12:
                    message.adGroup = reader.string();
                    break;
                case /* string feed */ 13:
                    message.feed = reader.string();
                    break;
                case /* string feed_item */ 14:
                    message.feedItem = reader.string();
                    break;
                case /* string asset */ 20:
                    message.asset = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* string change_date_time = 2; */
        if (message.changeDateTime !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.changeDateTime);
        /* google.ads.googleads.v11.enums.ChangeEventResourceTypeEnum.ChangeEventResourceType change_resource_type = 3; */
        if (message.changeResourceType !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.changeResourceType);
        /* string change_resource_name = 4; */
        if (message.changeResourceName !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.changeResourceName);
        /* google.ads.googleads.v11.enums.ChangeClientTypeEnum.ChangeClientType client_type = 5; */
        if (message.clientType !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.clientType);
        /* string user_email = 6; */
        if (message.userEmail !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.userEmail);
        /* google.ads.googleads.v11.resources.ChangeEvent.ChangedResource old_resource = 7; */
        if (message.oldResource)
            exports.ChangeEvent_ChangedResource.internalBinaryWrite(message.oldResource, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ChangeEvent.ChangedResource new_resource = 8; */
        if (message.newResource)
            exports.ChangeEvent_ChangedResource.internalBinaryWrite(message.newResource, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.ResourceChangeOperationEnum.ResourceChangeOperation resource_change_operation = 9; */
        if (message.resourceChangeOperation !== 0)
            writer.tag(9, runtime_1.WireType.Varint).int32(message.resourceChangeOperation);
        /* google.protobuf.FieldMask changed_fields = 10; */
        if (message.changedFields)
            field_mask_1.FieldMask.internalBinaryWrite(message.changedFields, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string campaign = 11; */
        if (message.campaign !== "")
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.campaign);
        /* string ad_group = 12; */
        if (message.adGroup !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.adGroup);
        /* string feed = 13; */
        if (message.feed !== "")
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.feed);
        /* string feed_item = 14; */
        if (message.feedItem !== "")
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.feedItem);
        /* string asset = 20; */
        if (message.asset !== "")
            writer.tag(20, runtime_1.WireType.LengthDelimited).string(message.asset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ChangeEvent
 */
exports.ChangeEvent = new ChangeEvent$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChangeEvent_ChangedResource$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ChangeEvent.ChangedResource", [
            { no: 1, name: "ad", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "ad_group", kind: "message", T: () => ad_group_1.AdGroup, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "ad_group_criterion", kind: "message", T: () => ad_group_criterion_1.AdGroupCriterion, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "campaign", kind: "message", T: () => campaign_1.Campaign, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "campaign_budget", kind: "message", T: () => campaign_budget_1.CampaignBudget, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "ad_group_bid_modifier", kind: "message", T: () => ad_group_bid_modifier_1.AdGroupBidModifier, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "campaign_criterion", kind: "message", T: () => campaign_criterion_1.CampaignCriterion, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "feed", kind: "message", T: () => feed_1.Feed, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "feed_item", kind: "message", T: () => feed_item_1.FeedItem, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "campaign_feed", kind: "message", T: () => campaign_feed_1.CampaignFeed, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "ad_group_feed", kind: "message", T: () => ad_group_feed_1.AdGroupFeed, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 12, name: "ad_group_ad", kind: "message", T: () => ad_group_ad_1.AdGroupAd, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "asset", kind: "message", T: () => asset_1.Asset, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 14, name: "customer_asset", kind: "message", T: () => customer_asset_1.CustomerAsset, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 15, name: "campaign_asset", kind: "message", T: () => campaign_asset_1.CampaignAsset, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 16, name: "ad_group_asset", kind: "message", T: () => ad_group_asset_1.AdGroupAsset, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 17, name: "asset_set", kind: "message", T: () => asset_set_1.AssetSet, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 18, name: "asset_set_asset", kind: "message", T: () => asset_set_asset_1.AssetSetAsset, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 19, name: "campaign_asset_set", kind: "message", T: () => campaign_asset_set_1.CampaignAssetSet, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad ad */ 1:
                    message.ad = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.ad);
                    break;
                case /* google.ads.googleads.v11.resources.AdGroup ad_group */ 2:
                    message.adGroup = ad_group_1.AdGroup.internalBinaryRead(reader, reader.uint32(), options, message.adGroup);
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupCriterion ad_group_criterion */ 3:
                    message.adGroupCriterion = ad_group_criterion_1.AdGroupCriterion.internalBinaryRead(reader, reader.uint32(), options, message.adGroupCriterion);
                    break;
                case /* google.ads.googleads.v11.resources.Campaign campaign */ 4:
                    message.campaign = campaign_1.Campaign.internalBinaryRead(reader, reader.uint32(), options, message.campaign);
                    break;
                case /* google.ads.googleads.v11.resources.CampaignBudget campaign_budget */ 5:
                    message.campaignBudget = campaign_budget_1.CampaignBudget.internalBinaryRead(reader, reader.uint32(), options, message.campaignBudget);
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupBidModifier ad_group_bid_modifier */ 6:
                    message.adGroupBidModifier = ad_group_bid_modifier_1.AdGroupBidModifier.internalBinaryRead(reader, reader.uint32(), options, message.adGroupBidModifier);
                    break;
                case /* google.ads.googleads.v11.resources.CampaignCriterion campaign_criterion */ 7:
                    message.campaignCriterion = campaign_criterion_1.CampaignCriterion.internalBinaryRead(reader, reader.uint32(), options, message.campaignCriterion);
                    break;
                case /* google.ads.googleads.v11.resources.Feed feed */ 8:
                    message.feed = feed_1.Feed.internalBinaryRead(reader, reader.uint32(), options, message.feed);
                    break;
                case /* google.ads.googleads.v11.resources.FeedItem feed_item */ 9:
                    message.feedItem = feed_item_1.FeedItem.internalBinaryRead(reader, reader.uint32(), options, message.feedItem);
                    break;
                case /* google.ads.googleads.v11.resources.CampaignFeed campaign_feed */ 10:
                    message.campaignFeed = campaign_feed_1.CampaignFeed.internalBinaryRead(reader, reader.uint32(), options, message.campaignFeed);
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupFeed ad_group_feed */ 11:
                    message.adGroupFeed = ad_group_feed_1.AdGroupFeed.internalBinaryRead(reader, reader.uint32(), options, message.adGroupFeed);
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupAd ad_group_ad */ 12:
                    message.adGroupAd = ad_group_ad_1.AdGroupAd.internalBinaryRead(reader, reader.uint32(), options, message.adGroupAd);
                    break;
                case /* google.ads.googleads.v11.resources.Asset asset */ 13:
                    message.asset = asset_1.Asset.internalBinaryRead(reader, reader.uint32(), options, message.asset);
                    break;
                case /* google.ads.googleads.v11.resources.CustomerAsset customer_asset */ 14:
                    message.customerAsset = customer_asset_1.CustomerAsset.internalBinaryRead(reader, reader.uint32(), options, message.customerAsset);
                    break;
                case /* google.ads.googleads.v11.resources.CampaignAsset campaign_asset */ 15:
                    message.campaignAsset = campaign_asset_1.CampaignAsset.internalBinaryRead(reader, reader.uint32(), options, message.campaignAsset);
                    break;
                case /* google.ads.googleads.v11.resources.AdGroupAsset ad_group_asset */ 16:
                    message.adGroupAsset = ad_group_asset_1.AdGroupAsset.internalBinaryRead(reader, reader.uint32(), options, message.adGroupAsset);
                    break;
                case /* google.ads.googleads.v11.resources.AssetSet asset_set */ 17:
                    message.assetSet = asset_set_1.AssetSet.internalBinaryRead(reader, reader.uint32(), options, message.assetSet);
                    break;
                case /* google.ads.googleads.v11.resources.AssetSetAsset asset_set_asset */ 18:
                    message.assetSetAsset = asset_set_asset_1.AssetSetAsset.internalBinaryRead(reader, reader.uint32(), options, message.assetSetAsset);
                    break;
                case /* google.ads.googleads.v11.resources.CampaignAssetSet campaign_asset_set */ 19:
                    message.campaignAssetSet = campaign_asset_set_1.CampaignAssetSet.internalBinaryRead(reader, reader.uint32(), options, message.campaignAssetSet);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad ad = 1; */
        if (message.ad)
            ad_1.Ad.internalBinaryWrite(message.ad, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroup ad_group = 2; */
        if (message.adGroup)
            ad_group_1.AdGroup.internalBinaryWrite(message.adGroup, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroupCriterion ad_group_criterion = 3; */
        if (message.adGroupCriterion)
            ad_group_criterion_1.AdGroupCriterion.internalBinaryWrite(message.adGroupCriterion, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Campaign campaign = 4; */
        if (message.campaign)
            campaign_1.Campaign.internalBinaryWrite(message.campaign, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.CampaignBudget campaign_budget = 5; */
        if (message.campaignBudget)
            campaign_budget_1.CampaignBudget.internalBinaryWrite(message.campaignBudget, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroupBidModifier ad_group_bid_modifier = 6; */
        if (message.adGroupBidModifier)
            ad_group_bid_modifier_1.AdGroupBidModifier.internalBinaryWrite(message.adGroupBidModifier, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.CampaignCriterion campaign_criterion = 7; */
        if (message.campaignCriterion)
            campaign_criterion_1.CampaignCriterion.internalBinaryWrite(message.campaignCriterion, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Feed feed = 8; */
        if (message.feed)
            feed_1.Feed.internalBinaryWrite(message.feed, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.FeedItem feed_item = 9; */
        if (message.feedItem)
            feed_item_1.FeedItem.internalBinaryWrite(message.feedItem, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.CampaignFeed campaign_feed = 10; */
        if (message.campaignFeed)
            campaign_feed_1.CampaignFeed.internalBinaryWrite(message.campaignFeed, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroupFeed ad_group_feed = 11; */
        if (message.adGroupFeed)
            ad_group_feed_1.AdGroupFeed.internalBinaryWrite(message.adGroupFeed, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroupAd ad_group_ad = 12; */
        if (message.adGroupAd)
            ad_group_ad_1.AdGroupAd.internalBinaryWrite(message.adGroupAd, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Asset asset = 13; */
        if (message.asset)
            asset_1.Asset.internalBinaryWrite(message.asset, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.CustomerAsset customer_asset = 14; */
        if (message.customerAsset)
            customer_asset_1.CustomerAsset.internalBinaryWrite(message.customerAsset, writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.CampaignAsset campaign_asset = 15; */
        if (message.campaignAsset)
            campaign_asset_1.CampaignAsset.internalBinaryWrite(message.campaignAsset, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AdGroupAsset ad_group_asset = 16; */
        if (message.adGroupAsset)
            ad_group_asset_1.AdGroupAsset.internalBinaryWrite(message.adGroupAsset, writer.tag(16, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AssetSet asset_set = 17; */
        if (message.assetSet)
            asset_set_1.AssetSet.internalBinaryWrite(message.assetSet, writer.tag(17, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.AssetSetAsset asset_set_asset = 18; */
        if (message.assetSetAsset)
            asset_set_asset_1.AssetSetAsset.internalBinaryWrite(message.assetSetAsset, writer.tag(18, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.CampaignAssetSet campaign_asset_set = 19; */
        if (message.campaignAssetSet)
            campaign_asset_set_1.CampaignAssetSet.internalBinaryWrite(message.campaignAssetSet, writer.tag(19, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ChangeEvent.ChangedResource
 */
exports.ChangeEvent_ChangedResource = new ChangeEvent_ChangedResource$Type();
