// google-ads-api-client/dist/protos/bindings/ads/resources/conversion_action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionAction_FirebaseSettings = exports.ConversionAction_ThirdPartyAppAnalyticsSettings = exports.ConversionAction_ValueSettings = exports.ConversionAction_AttributionModelSettings = exports.ConversionAction = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const data_driven_model_status_1 = require("../enums/data_driven_model_status");
const attribution_model_1 = require("../enums/attribution_model");
const mobile_app_vendor_1 = require("../enums/mobile_app_vendor");
const tag_snippet_1 = require("../common/tag_snippet");
const conversion_action_counting_type_1 = require("../enums/conversion_action_counting_type");
const conversion_action_category_1 = require("../enums/conversion_action_category");
const conversion_origin_1 = require("../enums/conversion_origin");
const conversion_action_type_1 = require("../enums/conversion_action_type");
const conversion_action_status_1 = require("../enums/conversion_action_status");
// @generated message type with reflection information, may provide speed optimized methods
class ConversionAction$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ConversionAction", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/ConversionAction" } } },
            { no: 21, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 22, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ConversionActionStatusEnum.ConversionActionStatus", conversion_action_status_1.ConversionActionStatusEnum_ConversionActionStatus] },
            { no: 5, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ConversionActionTypeEnum.ConversionActionType", conversion_action_type_1.ConversionActionTypeEnum_ConversionActionType], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 30, name: "origin", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ConversionOriginEnum.ConversionOrigin", conversion_origin_1.ConversionOriginEnum_ConversionOrigin], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 31, name: "primary_for_goal", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 6, name: "category", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ConversionActionCategoryEnum.ConversionActionCategory", conversion_action_category_1.ConversionActionCategoryEnum_ConversionActionCategory] },
            { no: 23, name: "owner_customer", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Customer" } } },
            { no: 24, name: "include_in_conversions_metric", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 25, name: "click_through_lookback_window_days", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 26, name: "view_through_lookback_window_days", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 11, name: "value_settings", kind: "message", T: () => exports.ConversionAction_ValueSettings },
            { no: 12, name: "counting_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ConversionActionCountingTypeEnum.ConversionActionCountingType", conversion_action_counting_type_1.ConversionActionCountingTypeEnum_ConversionActionCountingType] },
            { no: 13, name: "attribution_model_settings", kind: "message", T: () => exports.ConversionAction_AttributionModelSettings },
            { no: 14, name: "tag_snippets", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => tag_snippet_1.TagSnippet, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 27, name: "phone_call_duration_seconds", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 28, name: "app_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 17, name: "mobile_app_vendor", kind: "enum", T: () => ["google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor", mobile_app_vendor_1.MobileAppVendorEnum_MobileAppVendor], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 18, name: "firebase_settings", kind: "message", T: () => exports.ConversionAction_FirebaseSettings, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 19, name: "third_party_app_analytics_settings", kind: "message", T: () => exports.ConversionAction_ThirdPartyAppAnalyticsSettings, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/ConversionAction", pattern: ["customers/{customer_id}/conversionActions/{conversion_action_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", status: 0, type: 0, origin: 0, category: 0, countingType: 0, tagSnippets: [], mobileAppVendor: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 21:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string name */ 22:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.ConversionActionStatusEnum.ConversionActionStatus status */ 4:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.ConversionActionTypeEnum.ConversionActionType type */ 5:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.ConversionOriginEnum.ConversionOrigin origin */ 30:
                    message.origin = reader.int32();
                    break;
                case /* optional bool primary_for_goal */ 31:
                    message.primaryForGoal = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.ConversionActionCategoryEnum.ConversionActionCategory category */ 6:
                    message.category = reader.int32();
                    break;
                case /* optional string owner_customer */ 23:
                    message.ownerCustomer = reader.string();
                    break;
                case /* optional bool include_in_conversions_metric */ 24:
                    message.includeInConversionsMetric = reader.bool();
                    break;
                case /* optional int64 click_through_lookback_window_days */ 25:
                    message.clickThroughLookbackWindowDays = reader.int64().toBigInt();
                    break;
                case /* optional int64 view_through_lookback_window_days */ 26:
                    message.viewThroughLookbackWindowDays = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.resources.ConversionAction.ValueSettings value_settings */ 11:
                    message.valueSettings = exports.ConversionAction_ValueSettings.internalBinaryRead(reader, reader.uint32(), options, message.valueSettings);
                    break;
                case /* google.ads.googleads.v11.enums.ConversionActionCountingTypeEnum.ConversionActionCountingType counting_type */ 12:
                    message.countingType = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.ConversionAction.AttributionModelSettings attribution_model_settings */ 13:
                    message.attributionModelSettings = exports.ConversionAction_AttributionModelSettings.internalBinaryRead(reader, reader.uint32(), options, message.attributionModelSettings);
                    break;
                case /* repeated google.ads.googleads.v11.common.TagSnippet tag_snippets */ 14:
                    message.tagSnippets.push(tag_snippet_1.TagSnippet.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional int64 phone_call_duration_seconds */ 27:
                    message.phoneCallDurationSeconds = reader.int64().toBigInt();
                    break;
                case /* optional string app_id */ 28:
                    message.appId = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor mobile_app_vendor */ 17:
                    message.mobileAppVendor = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.ConversionAction.FirebaseSettings firebase_settings */ 18:
                    message.firebaseSettings = exports.ConversionAction_FirebaseSettings.internalBinaryRead(reader, reader.uint32(), options, message.firebaseSettings);
                    break;
                case /* google.ads.googleads.v11.resources.ConversionAction.ThirdPartyAppAnalyticsSettings third_party_app_analytics_settings */ 19:
                    message.thirdPartyAppAnalyticsSettings = exports.ConversionAction_ThirdPartyAppAnalyticsSettings.internalBinaryRead(reader, reader.uint32(), options, message.thirdPartyAppAnalyticsSettings);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 21; */
        if (message.id !== undefined)
            writer.tag(21, runtime_1.WireType.Varint).int64(message.id);
        /* optional string name = 22; */
        if (message.name !== undefined)
            writer.tag(22, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.enums.ConversionActionStatusEnum.ConversionActionStatus status = 4; */
        if (message.status !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.enums.ConversionActionTypeEnum.ConversionActionType type = 5; */
        if (message.type !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.ConversionOriginEnum.ConversionOrigin origin = 30; */
        if (message.origin !== 0)
            writer.tag(30, runtime_1.WireType.Varint).int32(message.origin);
        /* optional bool primary_for_goal = 31; */
        if (message.primaryForGoal !== undefined)
            writer.tag(31, runtime_1.WireType.Varint).bool(message.primaryForGoal);
        /* google.ads.googleads.v11.enums.ConversionActionCategoryEnum.ConversionActionCategory category = 6; */
        if (message.category !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.category);
        /* optional string owner_customer = 23; */
        if (message.ownerCustomer !== undefined)
            writer.tag(23, runtime_1.WireType.LengthDelimited).string(message.ownerCustomer);
        /* optional bool include_in_conversions_metric = 24; */
        if (message.includeInConversionsMetric !== undefined)
            writer.tag(24, runtime_1.WireType.Varint).bool(message.includeInConversionsMetric);
        /* optional int64 click_through_lookback_window_days = 25; */
        if (message.clickThroughLookbackWindowDays !== undefined)
            writer.tag(25, runtime_1.WireType.Varint).int64(message.clickThroughLookbackWindowDays);
        /* optional int64 view_through_lookback_window_days = 26; */
        if (message.viewThroughLookbackWindowDays !== undefined)
            writer.tag(26, runtime_1.WireType.Varint).int64(message.viewThroughLookbackWindowDays);
        /* google.ads.googleads.v11.resources.ConversionAction.ValueSettings value_settings = 11; */
        if (message.valueSettings)
            exports.ConversionAction_ValueSettings.internalBinaryWrite(message.valueSettings, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.ConversionActionCountingTypeEnum.ConversionActionCountingType counting_type = 12; */
        if (message.countingType !== 0)
            writer.tag(12, runtime_1.WireType.Varint).int32(message.countingType);
        /* google.ads.googleads.v11.resources.ConversionAction.AttributionModelSettings attribution_model_settings = 13; */
        if (message.attributionModelSettings)
            exports.ConversionAction_AttributionModelSettings.internalBinaryWrite(message.attributionModelSettings, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.common.TagSnippet tag_snippets = 14; */
        for (let i = 0; i < message.tagSnippets.length; i++)
            tag_snippet_1.TagSnippet.internalBinaryWrite(message.tagSnippets[i], writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional int64 phone_call_duration_seconds = 27; */
        if (message.phoneCallDurationSeconds !== undefined)
            writer.tag(27, runtime_1.WireType.Varint).int64(message.phoneCallDurationSeconds);
        /* optional string app_id = 28; */
        if (message.appId !== undefined)
            writer.tag(28, runtime_1.WireType.LengthDelimited).string(message.appId);
        /* google.ads.googleads.v11.enums.MobileAppVendorEnum.MobileAppVendor mobile_app_vendor = 17; */
        if (message.mobileAppVendor !== 0)
            writer.tag(17, runtime_1.WireType.Varint).int32(message.mobileAppVendor);
        /* google.ads.googleads.v11.resources.ConversionAction.FirebaseSettings firebase_settings = 18; */
        if (message.firebaseSettings)
            exports.ConversionAction_FirebaseSettings.internalBinaryWrite(message.firebaseSettings, writer.tag(18, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ConversionAction.ThirdPartyAppAnalyticsSettings third_party_app_analytics_settings = 19; */
        if (message.thirdPartyAppAnalyticsSettings)
            exports.ConversionAction_ThirdPartyAppAnalyticsSettings.internalBinaryWrite(message.thirdPartyAppAnalyticsSettings, writer.tag(19, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ConversionAction
 */
exports.ConversionAction = new ConversionAction$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ConversionAction_AttributionModelSettings$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ConversionAction.AttributionModelSettings", [
            { no: 1, name: "attribution_model", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AttributionModelEnum.AttributionModel", attribution_model_1.AttributionModelEnum_AttributionModel] },
            { no: 2, name: "data_driven_model_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.DataDrivenModelStatusEnum.DataDrivenModelStatus", data_driven_model_status_1.DataDrivenModelStatusEnum_DataDrivenModelStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { attributionModel: 0, dataDrivenModelStatus: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.AttributionModelEnum.AttributionModel attribution_model */ 1:
                    message.attributionModel = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.DataDrivenModelStatusEnum.DataDrivenModelStatus data_driven_model_status */ 2:
                    message.dataDrivenModelStatus = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.AttributionModelEnum.AttributionModel attribution_model = 1; */
        if (message.attributionModel !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.attributionModel);
        /* google.ads.googleads.v11.enums.DataDrivenModelStatusEnum.DataDrivenModelStatus data_driven_model_status = 2; */
        if (message.dataDrivenModelStatus !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.dataDrivenModelStatus);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ConversionAction.AttributionModelSettings
 */
exports.ConversionAction_AttributionModelSettings = new ConversionAction_AttributionModelSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ConversionAction_ValueSettings$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ConversionAction.ValueSettings", [
            { no: 4, name: "default_value", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 5, name: "default_currency_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "always_use_default_value", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional double default_value */ 4:
                    message.defaultValue = reader.double();
                    break;
                case /* optional string default_currency_code */ 5:
                    message.defaultCurrencyCode = reader.string();
                    break;
                case /* optional bool always_use_default_value */ 6:
                    message.alwaysUseDefaultValue = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional double default_value = 4; */
        if (message.defaultValue !== undefined)
            writer.tag(4, runtime_1.WireType.Bit64).double(message.defaultValue);
        /* optional string default_currency_code = 5; */
        if (message.defaultCurrencyCode !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.defaultCurrencyCode);
        /* optional bool always_use_default_value = 6; */
        if (message.alwaysUseDefaultValue !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.alwaysUseDefaultValue);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ConversionAction.ValueSettings
 */
exports.ConversionAction_ValueSettings = new ConversionAction_ValueSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ConversionAction_ThirdPartyAppAnalyticsSettings$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ConversionAction.ThirdPartyAppAnalyticsSettings", [
            { no: 2, name: "event_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "provider_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { providerName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string event_name */ 2:
                    message.eventName = reader.string();
                    break;
                case /* string provider_name */ 3:
                    message.providerName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string event_name = 2; */
        if (message.eventName !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.eventName);
        /* string provider_name = 3; */
        if (message.providerName !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.providerName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ConversionAction.ThirdPartyAppAnalyticsSettings
 */
exports.ConversionAction_ThirdPartyAppAnalyticsSettings = new ConversionAction_ThirdPartyAppAnalyticsSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ConversionAction_FirebaseSettings$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ConversionAction.FirebaseSettings", [
            { no: 3, name: "event_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "project_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "property_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "property_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { propertyId: 0n, propertyName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string event_name */ 3:
                    message.eventName = reader.string();
                    break;
                case /* optional string project_id */ 4:
                    message.projectId = reader.string();
                    break;
                case /* int64 property_id */ 5:
                    message.propertyId = reader.int64().toBigInt();
                    break;
                case /* string property_name */ 6:
                    message.propertyName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string event_name = 3; */
        if (message.eventName !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.eventName);
        /* optional string project_id = 4; */
        if (message.projectId !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.projectId);
        /* int64 property_id = 5; */
        if (message.propertyId !== 0n)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.propertyId);
        /* string property_name = 6; */
        if (message.propertyName !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.propertyName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ConversionAction.FirebaseSettings
 */
exports.ConversionAction_FirebaseSettings = new ConversionAction_FirebaseSettings$Type();
