// google-ads-api-client/dist/protos/bindings/ads/resources/custom_audience.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAudienceMember = exports.CustomAudience = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const custom_audience_member_type_1 = require("../enums/custom_audience_member_type");
const custom_audience_type_1 = require("../enums/custom_audience_type");
const custom_audience_status_1 = require("../enums/custom_audience_status");
// @generated message type with reflection information, may provide speed optimized methods
class CustomAudience$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CustomAudience", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/CustomAudience" } } },
            { no: 2, name: "id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CustomAudienceStatusEnum.CustomAudienceStatus", custom_audience_status_1.CustomAudienceStatusEnum_CustomAudienceStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CustomAudienceTypeEnum.CustomAudienceType", custom_audience_type_1.CustomAudienceTypeEnum_CustomAudienceType] },
            { no: 6, name: "description", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "members", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.CustomAudienceMember }
        ], { "google.api.resource": { type: "googleads.googleapis.com/CustomAudience", pattern: ["customers/{customer_id}/customAudiences/{custom_audience_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", id: 0n, status: 0, name: "", type: 0, description: "", members: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* int64 id */ 2:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.CustomAudienceStatusEnum.CustomAudienceStatus status */ 3:
                    message.status = reader.int32();
                    break;
                case /* string name */ 4:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.CustomAudienceTypeEnum.CustomAudienceType type */ 5:
                    message.type = reader.int32();
                    break;
                case /* string description */ 6:
                    message.description = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.resources.CustomAudienceMember members */ 7:
                    message.members.push(exports.CustomAudienceMember.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* int64 id = 2; */
        if (message.id !== 0n)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.id);
        /* google.ads.googleads.v11.enums.CustomAudienceStatusEnum.CustomAudienceStatus status = 3; */
        if (message.status !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.status);
        /* string name = 4; */
        if (message.name !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.enums.CustomAudienceTypeEnum.CustomAudienceType type = 5; */
        if (message.type !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.type);
        /* string description = 6; */
        if (message.description !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.description);
        /* repeated google.ads.googleads.v11.resources.CustomAudienceMember members = 7; */
        for (let i = 0; i < message.members.length; i++)
            exports.CustomAudienceMember.internalBinaryWrite(message.members[i], writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CustomAudience
 */
exports.CustomAudience = new CustomAudience$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CustomAudienceMember$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CustomAudienceMember", [
            { no: 1, name: "member_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CustomAudienceMemberTypeEnum.CustomAudienceMemberType", custom_audience_member_type_1.CustomAudienceMemberTypeEnum_CustomAudienceMemberType] },
            { no: 2, name: "keyword", kind: "scalar", oneof: "value", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "url", kind: "scalar", oneof: "value", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "place_category", kind: "scalar", oneof: "value", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 5, name: "app", kind: "scalar", oneof: "value", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { memberType: 0, value: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.CustomAudienceMemberTypeEnum.CustomAudienceMemberType member_type */ 1:
                    message.memberType = reader.int32();
                    break;
                case /* string keyword */ 2:
                    message.value = {
                        oneofKind: "keyword",
                        keyword: reader.string()
                    };
                    break;
                case /* string url */ 3:
                    message.value = {
                        oneofKind: "url",
                        url: reader.string()
                    };
                    break;
                case /* int64 place_category */ 4:
                    message.value = {
                        oneofKind: "placeCategory",
                        placeCategory: reader.int64().toBigInt()
                    };
                    break;
                case /* string app */ 5:
                    message.value = {
                        oneofKind: "app",
                        app: reader.string()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.CustomAudienceMemberTypeEnum.CustomAudienceMemberType member_type = 1; */
        if (message.memberType !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.memberType);
        /* string keyword = 2; */
        if (message.value.oneofKind === "keyword")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.value.keyword);
        /* string url = 3; */
        if (message.value.oneofKind === "url")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.value.url);
        /* int64 place_category = 4; */
        if (message.value.oneofKind === "placeCategory")
            writer.tag(4, runtime_1.WireType.Varint).int64(message.value.placeCategory);
        /* string app = 5; */
        if (message.value.oneofKind === "app")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.value.app);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CustomAudienceMember
 */
exports.CustomAudienceMember = new CustomAudienceMember$Type();
