// google-ads-api-client/dist/protos/bindings/ads/resources/customer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemarketingSetting = exports.ConversionTrackingSetting = exports.CallReportingSetting = exports.Customer = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const conversion_tracking_status_enum_1 = require("../enums/conversion_tracking_status_enum");
const customer_status_1 = require("../enums/customer_status");
const customer_pay_per_conversion_eligibility_failure_reason_1 = require("../enums/customer_pay_per_conversion_eligibility_failure_reason");
// @generated message type with reflection information, may provide speed optimized methods
class Customer$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Customer", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Customer" } } },
            { no: 19, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 20, name: "descriptive_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 21, name: "currency_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 22, name: "time_zone", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 23, name: "tracking_url_template", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 24, name: "final_url_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 25, name: "auto_tagging_enabled", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 26, name: "has_partners_badge", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 27, name: "manager", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 28, name: "test_account", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "call_reporting_setting", kind: "message", T: () => exports.CallReportingSetting },
            { no: 14, name: "conversion_tracking_setting", kind: "message", T: () => exports.ConversionTrackingSetting, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 15, name: "remarketing_setting", kind: "message", T: () => exports.RemarketingSetting, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 16, name: "pay_per_conversion_eligibility_failure_reasons", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.ads.googleads.v11.enums.CustomerPayPerConversionEligibilityFailureReasonEnum.CustomerPayPerConversionEligibilityFailureReason", customer_pay_per_conversion_eligibility_failure_reason_1.CustomerPayPerConversionEligibilityFailureReasonEnum_CustomerPayPerConversionEligibilityFailureReason], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 29, name: "optimization_score", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 30, name: "optimization_score_weight", kind: "scalar", T: 1 /*ScalarType.DOUBLE*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 36, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CustomerStatusEnum.CustomerStatus", customer_status_1.CustomerStatusEnum_CustomerStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/Customer", pattern: ["customers/{customer_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", payPerConversionEligibilityFailureReasons: [], optimizationScoreWeight: 0, status: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 19:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string descriptive_name */ 20:
                    message.descriptiveName = reader.string();
                    break;
                case /* optional string currency_code */ 21:
                    message.currencyCode = reader.string();
                    break;
                case /* optional string time_zone */ 22:
                    message.timeZone = reader.string();
                    break;
                case /* optional string tracking_url_template */ 23:
                    message.trackingUrlTemplate = reader.string();
                    break;
                case /* optional string final_url_suffix */ 24:
                    message.finalUrlSuffix = reader.string();
                    break;
                case /* optional bool auto_tagging_enabled */ 25:
                    message.autoTaggingEnabled = reader.bool();
                    break;
                case /* optional bool has_partners_badge */ 26:
                    message.hasPartnersBadge = reader.bool();
                    break;
                case /* optional bool manager */ 27:
                    message.manager = reader.bool();
                    break;
                case /* optional bool test_account */ 28:
                    message.testAccount = reader.bool();
                    break;
                case /* google.ads.googleads.v11.resources.CallReportingSetting call_reporting_setting */ 10:
                    message.callReportingSetting = exports.CallReportingSetting.internalBinaryRead(reader, reader.uint32(), options, message.callReportingSetting);
                    break;
                case /* google.ads.googleads.v11.resources.ConversionTrackingSetting conversion_tracking_setting */ 14:
                    message.conversionTrackingSetting = exports.ConversionTrackingSetting.internalBinaryRead(reader, reader.uint32(), options, message.conversionTrackingSetting);
                    break;
                case /* google.ads.googleads.v11.resources.RemarketingSetting remarketing_setting */ 15:
                    message.remarketingSetting = exports.RemarketingSetting.internalBinaryRead(reader, reader.uint32(), options, message.remarketingSetting);
                    break;
                case /* repeated google.ads.googleads.v11.enums.CustomerPayPerConversionEligibilityFailureReasonEnum.CustomerPayPerConversionEligibilityFailureReason pay_per_conversion_eligibility_failure_reasons */ 16:
                    if (wireType === runtime_2.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.payPerConversionEligibilityFailureReasons.push(reader.int32());
                    else
                        message.payPerConversionEligibilityFailureReasons.push(reader.int32());
                    break;
                case /* optional double optimization_score */ 29:
                    message.optimizationScore = reader.double();
                    break;
                case /* double optimization_score_weight */ 30:
                    message.optimizationScoreWeight = reader.double();
                    break;
                case /* google.ads.googleads.v11.enums.CustomerStatusEnum.CustomerStatus status */ 36:
                    message.status = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_2.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 19; */
        if (message.id !== undefined)
            writer.tag(19, runtime_2.WireType.Varint).int64(message.id);
        /* optional string descriptive_name = 20; */
        if (message.descriptiveName !== undefined)
            writer.tag(20, runtime_2.WireType.LengthDelimited).string(message.descriptiveName);
        /* optional string currency_code = 21; */
        if (message.currencyCode !== undefined)
            writer.tag(21, runtime_2.WireType.LengthDelimited).string(message.currencyCode);
        /* optional string time_zone = 22; */
        if (message.timeZone !== undefined)
            writer.tag(22, runtime_2.WireType.LengthDelimited).string(message.timeZone);
        /* optional string tracking_url_template = 23; */
        if (message.trackingUrlTemplate !== undefined)
            writer.tag(23, runtime_2.WireType.LengthDelimited).string(message.trackingUrlTemplate);
        /* optional string final_url_suffix = 24; */
        if (message.finalUrlSuffix !== undefined)
            writer.tag(24, runtime_2.WireType.LengthDelimited).string(message.finalUrlSuffix);
        /* optional bool auto_tagging_enabled = 25; */
        if (message.autoTaggingEnabled !== undefined)
            writer.tag(25, runtime_2.WireType.Varint).bool(message.autoTaggingEnabled);
        /* optional bool has_partners_badge = 26; */
        if (message.hasPartnersBadge !== undefined)
            writer.tag(26, runtime_2.WireType.Varint).bool(message.hasPartnersBadge);
        /* optional bool manager = 27; */
        if (message.manager !== undefined)
            writer.tag(27, runtime_2.WireType.Varint).bool(message.manager);
        /* optional bool test_account = 28; */
        if (message.testAccount !== undefined)
            writer.tag(28, runtime_2.WireType.Varint).bool(message.testAccount);
        /* google.ads.googleads.v11.resources.CallReportingSetting call_reporting_setting = 10; */
        if (message.callReportingSetting)
            exports.CallReportingSetting.internalBinaryWrite(message.callReportingSetting, writer.tag(10, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.ConversionTrackingSetting conversion_tracking_setting = 14; */
        if (message.conversionTrackingSetting)
            exports.ConversionTrackingSetting.internalBinaryWrite(message.conversionTrackingSetting, writer.tag(14, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.RemarketingSetting remarketing_setting = 15; */
        if (message.remarketingSetting)
            exports.RemarketingSetting.internalBinaryWrite(message.remarketingSetting, writer.tag(15, runtime_2.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.enums.CustomerPayPerConversionEligibilityFailureReasonEnum.CustomerPayPerConversionEligibilityFailureReason pay_per_conversion_eligibility_failure_reasons = 16; */
        if (message.payPerConversionEligibilityFailureReasons.length) {
            writer.tag(16, runtime_2.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.payPerConversionEligibilityFailureReasons.length; i++)
                writer.int32(message.payPerConversionEligibilityFailureReasons[i]);
            writer.join();
        }
        /* optional double optimization_score = 29; */
        if (message.optimizationScore !== undefined)
            writer.tag(29, runtime_2.WireType.Bit64).double(message.optimizationScore);
        /* double optimization_score_weight = 30; */
        if (message.optimizationScoreWeight !== 0)
            writer.tag(30, runtime_2.WireType.Bit64).double(message.optimizationScoreWeight);
        /* google.ads.googleads.v11.enums.CustomerStatusEnum.CustomerStatus status = 36; */
        if (message.status !== 0)
            writer.tag(36, runtime_2.WireType.Varint).int32(message.status);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Customer
 */
exports.Customer = new Customer$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CallReportingSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CallReportingSetting", [
            { no: 10, name: "call_reporting_enabled", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 11, name: "call_conversion_reporting_enabled", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 12, name: "call_conversion_action", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/ConversionAction" } } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional bool call_reporting_enabled */ 10:
                    message.callReportingEnabled = reader.bool();
                    break;
                case /* optional bool call_conversion_reporting_enabled */ 11:
                    message.callConversionReportingEnabled = reader.bool();
                    break;
                case /* optional string call_conversion_action */ 12:
                    message.callConversionAction = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional bool call_reporting_enabled = 10; */
        if (message.callReportingEnabled !== undefined)
            writer.tag(10, runtime_2.WireType.Varint).bool(message.callReportingEnabled);
        /* optional bool call_conversion_reporting_enabled = 11; */
        if (message.callConversionReportingEnabled !== undefined)
            writer.tag(11, runtime_2.WireType.Varint).bool(message.callConversionReportingEnabled);
        /* optional string call_conversion_action = 12; */
        if (message.callConversionAction !== undefined)
            writer.tag(12, runtime_2.WireType.LengthDelimited).string(message.callConversionAction);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CallReportingSetting
 */
exports.CallReportingSetting = new CallReportingSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ConversionTrackingSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ConversionTrackingSetting", [
            { no: 3, name: "conversion_tracking_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "cross_account_conversion_tracking_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "accepted_customer_data_terms", kind: "scalar", T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "conversion_tracking_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ConversionTrackingStatusEnum.ConversionTrackingStatus", conversion_tracking_status_enum_1.ConversionTrackingStatusEnum_ConversionTrackingStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "enhanced_conversions_for_leads_enabled", kind: "scalar", T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "google_ads_conversion_customer", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { acceptedCustomerDataTerms: false, conversionTrackingStatus: 0, enhancedConversionsForLeadsEnabled: false, googleAdsConversionCustomer: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 conversion_tracking_id */ 3:
                    message.conversionTrackingId = reader.int64().toBigInt();
                    break;
                case /* optional int64 cross_account_conversion_tracking_id */ 4:
                    message.crossAccountConversionTrackingId = reader.int64().toBigInt();
                    break;
                case /* bool accepted_customer_data_terms */ 5:
                    message.acceptedCustomerDataTerms = reader.bool();
                    break;
                case /* google.ads.googleads.v11.enums.ConversionTrackingStatusEnum.ConversionTrackingStatus conversion_tracking_status */ 6:
                    message.conversionTrackingStatus = reader.int32();
                    break;
                case /* bool enhanced_conversions_for_leads_enabled */ 7:
                    message.enhancedConversionsForLeadsEnabled = reader.bool();
                    break;
                case /* string google_ads_conversion_customer */ 8:
                    message.googleAdsConversionCustomer = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 conversion_tracking_id = 3; */
        if (message.conversionTrackingId !== undefined)
            writer.tag(3, runtime_2.WireType.Varint).int64(message.conversionTrackingId);
        /* optional int64 cross_account_conversion_tracking_id = 4; */
        if (message.crossAccountConversionTrackingId !== undefined)
            writer.tag(4, runtime_2.WireType.Varint).int64(message.crossAccountConversionTrackingId);
        /* bool accepted_customer_data_terms = 5; */
        if (message.acceptedCustomerDataTerms !== false)
            writer.tag(5, runtime_2.WireType.Varint).bool(message.acceptedCustomerDataTerms);
        /* google.ads.googleads.v11.enums.ConversionTrackingStatusEnum.ConversionTrackingStatus conversion_tracking_status = 6; */
        if (message.conversionTrackingStatus !== 0)
            writer.tag(6, runtime_2.WireType.Varint).int32(message.conversionTrackingStatus);
        /* bool enhanced_conversions_for_leads_enabled = 7; */
        if (message.enhancedConversionsForLeadsEnabled !== false)
            writer.tag(7, runtime_2.WireType.Varint).bool(message.enhancedConversionsForLeadsEnabled);
        /* string google_ads_conversion_customer = 8; */
        if (message.googleAdsConversionCustomer !== "")
            writer.tag(8, runtime_2.WireType.LengthDelimited).string(message.googleAdsConversionCustomer);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ConversionTrackingSetting
 */
exports.ConversionTrackingSetting = new ConversionTrackingSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class RemarketingSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.RemarketingSetting", [
            { no: 2, name: "google_global_site_tag", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string google_global_site_tag */ 2:
                    message.googleGlobalSiteTag = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string google_global_site_tag = 2; */
        if (message.googleGlobalSiteTag !== undefined)
            writer.tag(2, runtime_2.WireType.LengthDelimited).string(message.googleGlobalSiteTag);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.RemarketingSetting
 */
exports.RemarketingSetting = new RemarketingSetting$Type();
