// google-ads-api-client/dist/protos/bindings/ads/resources/custom_interest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomInterestMember = exports.CustomInterest = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const custom_interest_member_type_1 = require("../enums/custom_interest_member_type");
const custom_interest_type_1 = require("../enums/custom_interest_type");
const custom_interest_status_1 = require("../enums/custom_interest_status");
// @generated message type with reflection information, may provide speed optimized methods
class CustomInterest$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CustomInterest", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/CustomInterest" } } },
            { no: 8, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CustomInterestStatusEnum.CustomInterestStatus", custom_interest_status_1.CustomInterestStatusEnum_CustomInterestStatus] },
            { no: 9, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CustomInterestTypeEnum.CustomInterestType", custom_interest_type_1.CustomInterestTypeEnum_CustomInterestType] },
            { no: 10, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "members", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.CustomInterestMember }
        ], { "google.api.resource": { type: "googleads.googleapis.com/CustomInterest", pattern: ["customers/{customer_id}/customInterests/{custom_interest_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", status: 0, type: 0, members: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 8:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.CustomInterestStatusEnum.CustomInterestStatus status */ 3:
                    message.status = reader.int32();
                    break;
                case /* optional string name */ 9:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.CustomInterestTypeEnum.CustomInterestType type */ 5:
                    message.type = reader.int32();
                    break;
                case /* optional string description */ 10:
                    message.description = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.resources.CustomInterestMember members */ 7:
                    message.members.push(exports.CustomInterestMember.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 8; */
        if (message.id !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).int64(message.id);
        /* google.ads.googleads.v11.enums.CustomInterestStatusEnum.CustomInterestStatus status = 3; */
        if (message.status !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.status);
        /* optional string name = 9; */
        if (message.name !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.enums.CustomInterestTypeEnum.CustomInterestType type = 5; */
        if (message.type !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.type);
        /* optional string description = 10; */
        if (message.description !== undefined)
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.description);
        /* repeated google.ads.googleads.v11.resources.CustomInterestMember members = 7; */
        for (let i = 0; i < message.members.length; i++)
            exports.CustomInterestMember.internalBinaryWrite(message.members[i], writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CustomInterest
 */
exports.CustomInterest = new CustomInterest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CustomInterestMember$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.CustomInterestMember", [
            { no: 1, name: "member_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.CustomInterestMemberTypeEnum.CustomInterestMemberType", custom_interest_member_type_1.CustomInterestMemberTypeEnum_CustomInterestMemberType] },
            { no: 3, name: "parameter", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { memberType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.CustomInterestMemberTypeEnum.CustomInterestMemberType member_type */ 1:
                    message.memberType = reader.int32();
                    break;
                case /* optional string parameter */ 3:
                    message.parameter = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.CustomInterestMemberTypeEnum.CustomInterestMemberType member_type = 1; */
        if (message.memberType !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.memberType);
        /* optional string parameter = 3; */
        if (message.parameter !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.parameter);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.CustomInterestMember
 */
exports.CustomInterestMember = new CustomInterestMember$Type();
