// google-ads-api-client/dist/protos/bindings/ads/resources/extension_feed_item.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionFeedItem = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const extensions_1 = require("../common/extensions");
const extensions_2 = require("../common/extensions");
const extensions_3 = require("../common/extensions");
const extensions_4 = require("../common/extensions");
const extensions_5 = require("../common/extensions");
const extensions_6 = require("../common/extensions");
const extensions_7 = require("../common/extensions");
const extensions_8 = require("../common/extensions");
const extensions_9 = require("../common/extensions");
const extensions_10 = require("../common/extensions");
const extensions_11 = require("../common/extensions");
const extensions_12 = require("../common/extensions");
const feed_item_status_1 = require("../enums/feed_item_status");
const criteria_1 = require("../common/criteria");
const feed_item_target_device_1 = require("../enums/feed_item_target_device");
const criteria_2 = require("../common/criteria");
const extension_type_1 = require("../enums/extension_type");
// @generated message type with reflection information, may provide speed optimized methods
class ExtensionFeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.ExtensionFeedItem", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/ExtensionFeedItem" } } },
            { no: 25, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "extension_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.ExtensionTypeEnum.ExtensionType", extension_type_1.ExtensionTypeEnum_ExtensionType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 26, name: "start_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 27, name: "end_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 16, name: "ad_schedules", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_2.AdScheduleInfo },
            { no: 17, name: "device", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum.FeedItemTargetDevice", feed_item_target_device_1.FeedItemTargetDeviceEnum_FeedItemTargetDevice] },
            { no: 30, name: "targeted_geo_target_constant", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/GeoTargetConstant" } } },
            { no: 22, name: "targeted_keyword", kind: "message", T: () => criteria_1.KeywordInfo },
            { no: 4, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemStatusEnum.FeedItemStatus", feed_item_status_1.FeedItemStatusEnum_FeedItemStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "sitelink_feed_item", kind: "message", oneof: "extension", T: () => extensions_12.SitelinkFeedItem },
            { no: 3, name: "structured_snippet_feed_item", kind: "message", oneof: "extension", T: () => extensions_11.StructuredSnippetFeedItem },
            { no: 7, name: "app_feed_item", kind: "message", oneof: "extension", T: () => extensions_10.AppFeedItem },
            { no: 8, name: "call_feed_item", kind: "message", oneof: "extension", T: () => extensions_9.CallFeedItem },
            { no: 9, name: "callout_feed_item", kind: "message", oneof: "extension", T: () => extensions_8.CalloutFeedItem },
            { no: 10, name: "text_message_feed_item", kind: "message", oneof: "extension", T: () => extensions_7.TextMessageFeedItem },
            { no: 11, name: "price_feed_item", kind: "message", oneof: "extension", T: () => extensions_6.PriceFeedItem },
            { no: 12, name: "promotion_feed_item", kind: "message", oneof: "extension", T: () => extensions_5.PromotionFeedItem },
            { no: 14, name: "location_feed_item", kind: "message", oneof: "extension", T: () => extensions_4.LocationFeedItem, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 15, name: "affiliate_location_feed_item", kind: "message", oneof: "extension", T: () => extensions_3.AffiliateLocationFeedItem, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 23, name: "hotel_callout_feed_item", kind: "message", oneof: "extension", T: () => extensions_2.HotelCalloutFeedItem },
            { no: 31, name: "image_feed_item", kind: "message", oneof: "extension", T: () => extensions_1.ImageFeedItem, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 28, name: "targeted_campaign", kind: "scalar", oneof: "servingResourceTargeting", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 29, name: "targeted_ad_group", kind: "scalar", oneof: "servingResourceTargeting", T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/ExtensionFeedItem", pattern: ["customers/{customer_id}/extensionFeedItems/{feed_item_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", extensionType: 0, adSchedules: [], device: 0, status: 0, extension: { oneofKind: undefined }, servingResourceTargeting: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 25:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.ExtensionTypeEnum.ExtensionType extension_type */ 13:
                    message.extensionType = reader.int32();
                    break;
                case /* optional string start_date_time */ 26:
                    message.startDateTime = reader.string();
                    break;
                case /* optional string end_date_time */ 27:
                    message.endDateTime = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedules */ 16:
                    message.adSchedules.push(criteria_2.AdScheduleInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum.FeedItemTargetDevice device */ 17:
                    message.device = reader.int32();
                    break;
                case /* optional string targeted_geo_target_constant */ 30:
                    message.targetedGeoTargetConstant = reader.string();
                    break;
                case /* google.ads.googleads.v11.common.KeywordInfo targeted_keyword */ 22:
                    message.targetedKeyword = criteria_1.KeywordInfo.internalBinaryRead(reader, reader.uint32(), options, message.targetedKeyword);
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemStatusEnum.FeedItemStatus status */ 4:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.SitelinkFeedItem sitelink_feed_item */ 2:
                    message.extension = {
                        oneofKind: "sitelinkFeedItem",
                        sitelinkFeedItem: extensions_12.SitelinkFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.sitelinkFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.StructuredSnippetFeedItem structured_snippet_feed_item */ 3:
                    message.extension = {
                        oneofKind: "structuredSnippetFeedItem",
                        structuredSnippetFeedItem: extensions_11.StructuredSnippetFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.structuredSnippetFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AppFeedItem app_feed_item */ 7:
                    message.extension = {
                        oneofKind: "appFeedItem",
                        appFeedItem: extensions_10.AppFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.appFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CallFeedItem call_feed_item */ 8:
                    message.extension = {
                        oneofKind: "callFeedItem",
                        callFeedItem: extensions_9.CallFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.callFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.CalloutFeedItem callout_feed_item */ 9:
                    message.extension = {
                        oneofKind: "calloutFeedItem",
                        calloutFeedItem: extensions_8.CalloutFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.calloutFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.TextMessageFeedItem text_message_feed_item */ 10:
                    message.extension = {
                        oneofKind: "textMessageFeedItem",
                        textMessageFeedItem: extensions_7.TextMessageFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.textMessageFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PriceFeedItem price_feed_item */ 11:
                    message.extension = {
                        oneofKind: "priceFeedItem",
                        priceFeedItem: extensions_6.PriceFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.priceFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.PromotionFeedItem promotion_feed_item */ 12:
                    message.extension = {
                        oneofKind: "promotionFeedItem",
                        promotionFeedItem: extensions_5.PromotionFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.promotionFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LocationFeedItem location_feed_item */ 14:
                    message.extension = {
                        oneofKind: "locationFeedItem",
                        locationFeedItem: extensions_4.LocationFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.locationFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.AffiliateLocationFeedItem affiliate_location_feed_item */ 15:
                    message.extension = {
                        oneofKind: "affiliateLocationFeedItem",
                        affiliateLocationFeedItem: extensions_3.AffiliateLocationFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.affiliateLocationFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.HotelCalloutFeedItem hotel_callout_feed_item */ 23:
                    message.extension = {
                        oneofKind: "hotelCalloutFeedItem",
                        hotelCalloutFeedItem: extensions_2.HotelCalloutFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.hotelCalloutFeedItem)
                    };
                    break;
                case /* google.ads.googleads.v11.common.ImageFeedItem image_feed_item */ 31:
                    message.extension = {
                        oneofKind: "imageFeedItem",
                        imageFeedItem: extensions_1.ImageFeedItem.internalBinaryRead(reader, reader.uint32(), options, message.extension.imageFeedItem)
                    };
                    break;
                case /* string targeted_campaign */ 28:
                    message.servingResourceTargeting = {
                        oneofKind: "targetedCampaign",
                        targetedCampaign: reader.string()
                    };
                    break;
                case /* string targeted_ad_group */ 29:
                    message.servingResourceTargeting = {
                        oneofKind: "targetedAdGroup",
                        targetedAdGroup: reader.string()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 25; */
        if (message.id !== undefined)
            writer.tag(25, runtime_1.WireType.Varint).int64(message.id);
        /* google.ads.googleads.v11.enums.ExtensionTypeEnum.ExtensionType extension_type = 13; */
        if (message.extensionType !== 0)
            writer.tag(13, runtime_1.WireType.Varint).int32(message.extensionType);
        /* optional string start_date_time = 26; */
        if (message.startDateTime !== undefined)
            writer.tag(26, runtime_1.WireType.LengthDelimited).string(message.startDateTime);
        /* optional string end_date_time = 27; */
        if (message.endDateTime !== undefined)
            writer.tag(27, runtime_1.WireType.LengthDelimited).string(message.endDateTime);
        /* repeated google.ads.googleads.v11.common.AdScheduleInfo ad_schedules = 16; */
        for (let i = 0; i < message.adSchedules.length; i++)
            criteria_2.AdScheduleInfo.internalBinaryWrite(message.adSchedules[i], writer.tag(16, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum.FeedItemTargetDevice device = 17; */
        if (message.device !== 0)
            writer.tag(17, runtime_1.WireType.Varint).int32(message.device);
        /* optional string targeted_geo_target_constant = 30; */
        if (message.targetedGeoTargetConstant !== undefined)
            writer.tag(30, runtime_1.WireType.LengthDelimited).string(message.targetedGeoTargetConstant);
        /* google.ads.googleads.v11.common.KeywordInfo targeted_keyword = 22; */
        if (message.targetedKeyword)
            criteria_1.KeywordInfo.internalBinaryWrite(message.targetedKeyword, writer.tag(22, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.FeedItemStatusEnum.FeedItemStatus status = 4; */
        if (message.status !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.common.SitelinkFeedItem sitelink_feed_item = 2; */
        if (message.extension.oneofKind === "sitelinkFeedItem")
            extensions_12.SitelinkFeedItem.internalBinaryWrite(message.extension.sitelinkFeedItem, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.StructuredSnippetFeedItem structured_snippet_feed_item = 3; */
        if (message.extension.oneofKind === "structuredSnippetFeedItem")
            extensions_11.StructuredSnippetFeedItem.internalBinaryWrite(message.extension.structuredSnippetFeedItem, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AppFeedItem app_feed_item = 7; */
        if (message.extension.oneofKind === "appFeedItem")
            extensions_10.AppFeedItem.internalBinaryWrite(message.extension.appFeedItem, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CallFeedItem call_feed_item = 8; */
        if (message.extension.oneofKind === "callFeedItem")
            extensions_9.CallFeedItem.internalBinaryWrite(message.extension.callFeedItem, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CalloutFeedItem callout_feed_item = 9; */
        if (message.extension.oneofKind === "calloutFeedItem")
            extensions_8.CalloutFeedItem.internalBinaryWrite(message.extension.calloutFeedItem, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.TextMessageFeedItem text_message_feed_item = 10; */
        if (message.extension.oneofKind === "textMessageFeedItem")
            extensions_7.TextMessageFeedItem.internalBinaryWrite(message.extension.textMessageFeedItem, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PriceFeedItem price_feed_item = 11; */
        if (message.extension.oneofKind === "priceFeedItem")
            extensions_6.PriceFeedItem.internalBinaryWrite(message.extension.priceFeedItem, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.PromotionFeedItem promotion_feed_item = 12; */
        if (message.extension.oneofKind === "promotionFeedItem")
            extensions_5.PromotionFeedItem.internalBinaryWrite(message.extension.promotionFeedItem, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LocationFeedItem location_feed_item = 14; */
        if (message.extension.oneofKind === "locationFeedItem")
            extensions_4.LocationFeedItem.internalBinaryWrite(message.extension.locationFeedItem, writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.AffiliateLocationFeedItem affiliate_location_feed_item = 15; */
        if (message.extension.oneofKind === "affiliateLocationFeedItem")
            extensions_3.AffiliateLocationFeedItem.internalBinaryWrite(message.extension.affiliateLocationFeedItem, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.HotelCalloutFeedItem hotel_callout_feed_item = 23; */
        if (message.extension.oneofKind === "hotelCalloutFeedItem")
            extensions_2.HotelCalloutFeedItem.internalBinaryWrite(message.extension.hotelCalloutFeedItem, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.ImageFeedItem image_feed_item = 31; */
        if (message.extension.oneofKind === "imageFeedItem")
            extensions_1.ImageFeedItem.internalBinaryWrite(message.extension.imageFeedItem, writer.tag(31, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string targeted_campaign = 28; */
        if (message.servingResourceTargeting.oneofKind === "targetedCampaign")
            writer.tag(28, runtime_1.WireType.LengthDelimited).string(message.servingResourceTargeting.targetedCampaign);
        /* string targeted_ad_group = 29; */
        if (message.servingResourceTargeting.oneofKind === "targetedAdGroup")
            writer.tag(29, runtime_1.WireType.LengthDelimited).string(message.servingResourceTargeting.targetedAdGroup);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.ExtensionFeedItem
 */
exports.ExtensionFeedItem = new ExtensionFeedItem$Type();
