// google-ads-api-client/dist/protos/bindings/ads/resources/feed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedAttributeOperation = exports.FeedAttribute = exports.Feed_AffiliateLocationFeedData = exports.Feed_PlacesLocationFeedData_OAuthInfo = exports.Feed_PlacesLocationFeedData = exports.Feed = exports.FeedAttributeOperation_Operator = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const feed_attribute_type_1 = require("../enums/feed_attribute_type");
const affiliate_location_feed_relationship_type_1 = require("../enums/affiliate_location_feed_relationship_type");
const feed_status_1 = require("../enums/feed_status");
const feed_origin_1 = require("../enums/feed_origin");
/**
 * The operator.
 *
 * @generated from protobuf enum google.ads.googleads.v11.resources.FeedAttributeOperation.Operator
 */
var FeedAttributeOperation_Operator;
(function (FeedAttributeOperation_Operator) {
    /**
     * Unspecified.
     *
     * @generated from protobuf enum value: UNSPECIFIED = 0;
     */
    FeedAttributeOperation_Operator[FeedAttributeOperation_Operator["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * Used for return value only. Represents value unknown in this version.
     *
     * @generated from protobuf enum value: UNKNOWN = 1;
     */
    FeedAttributeOperation_Operator[FeedAttributeOperation_Operator["UNKNOWN"] = 1] = "UNKNOWN";
    /**
     * Add the attribute to the existing attributes.
     *
     * @generated from protobuf enum value: ADD = 2;
     */
    FeedAttributeOperation_Operator[FeedAttributeOperation_Operator["ADD"] = 2] = "ADD";
})(FeedAttributeOperation_Operator = exports.FeedAttributeOperation_Operator || (exports.FeedAttributeOperation_Operator = {}));
// @generated message type with reflection information, may provide speed optimized methods
class Feed$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Feed", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Feed" } } },
            { no: 11, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 12, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 4, name: "attributes", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.FeedAttribute },
            { no: 9, name: "attribute_operations", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.FeedAttributeOperation },
            { no: 5, name: "origin", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedOriginEnum.FeedOrigin", feed_origin_1.FeedOriginEnum_FeedOrigin], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 8, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedStatusEnum.FeedStatus", feed_status_1.FeedStatusEnum_FeedStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "places_location_feed_data", kind: "message", oneof: "systemFeedGenerationData", T: () => exports.Feed_PlacesLocationFeedData },
            { no: 7, name: "affiliate_location_feed_data", kind: "message", oneof: "systemFeedGenerationData", T: () => exports.Feed_AffiliateLocationFeedData }
        ], { "google.api.resource": { type: "googleads.googleapis.com/Feed", pattern: ["customers/{customer_id}/feeds/{feed_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", attributes: [], attributeOperations: [], origin: 0, status: 0, systemFeedGenerationData: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 11:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string name */ 12:
                    message.name = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.resources.FeedAttribute attributes */ 4:
                    message.attributes.push(exports.FeedAttribute.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated google.ads.googleads.v11.resources.FeedAttributeOperation attribute_operations */ 9:
                    message.attributeOperations.push(exports.FeedAttributeOperation.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.FeedOriginEnum.FeedOrigin origin */ 5:
                    message.origin = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.FeedStatusEnum.FeedStatus status */ 8:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.Feed.PlacesLocationFeedData places_location_feed_data */ 6:
                    message.systemFeedGenerationData = {
                        oneofKind: "placesLocationFeedData",
                        placesLocationFeedData: exports.Feed_PlacesLocationFeedData.internalBinaryRead(reader, reader.uint32(), options, message.systemFeedGenerationData.placesLocationFeedData)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Feed.AffiliateLocationFeedData affiliate_location_feed_data */ 7:
                    message.systemFeedGenerationData = {
                        oneofKind: "affiliateLocationFeedData",
                        affiliateLocationFeedData: exports.Feed_AffiliateLocationFeedData.internalBinaryRead(reader, reader.uint32(), options, message.systemFeedGenerationData.affiliateLocationFeedData)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 11; */
        if (message.id !== undefined)
            writer.tag(11, runtime_1.WireType.Varint).int64(message.id);
        /* optional string name = 12; */
        if (message.name !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.name);
        /* repeated google.ads.googleads.v11.resources.FeedAttribute attributes = 4; */
        for (let i = 0; i < message.attributes.length; i++)
            exports.FeedAttribute.internalBinaryWrite(message.attributes[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated google.ads.googleads.v11.resources.FeedAttributeOperation attribute_operations = 9; */
        for (let i = 0; i < message.attributeOperations.length; i++)
            exports.FeedAttributeOperation.internalBinaryWrite(message.attributeOperations[i], writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.FeedOriginEnum.FeedOrigin origin = 5; */
        if (message.origin !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.origin);
        /* google.ads.googleads.v11.enums.FeedStatusEnum.FeedStatus status = 8; */
        if (message.status !== 0)
            writer.tag(8, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.resources.Feed.PlacesLocationFeedData places_location_feed_data = 6; */
        if (message.systemFeedGenerationData.oneofKind === "placesLocationFeedData")
            exports.Feed_PlacesLocationFeedData.internalBinaryWrite(message.systemFeedGenerationData.placesLocationFeedData, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Feed.AffiliateLocationFeedData affiliate_location_feed_data = 7; */
        if (message.systemFeedGenerationData.oneofKind === "affiliateLocationFeedData")
            exports.Feed_AffiliateLocationFeedData.internalBinaryWrite(message.systemFeedGenerationData.affiliateLocationFeedData, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Feed
 */
exports.Feed = new Feed$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Feed_PlacesLocationFeedData$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Feed.PlacesLocationFeedData", [
            { no: 1, name: "oauth_info", kind: "message", T: () => exports.Feed_PlacesLocationFeedData_OAuthInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 7, name: "email_address", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "business_account_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "business_name_filter", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 11, name: "category_filters", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "label_filters", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = { businessAccountId: "", categoryFilters: [], labelFilters: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Feed.PlacesLocationFeedData.OAuthInfo oauth_info */ 1:
                    message.oauthInfo = exports.Feed_PlacesLocationFeedData_OAuthInfo.internalBinaryRead(reader, reader.uint32(), options, message.oauthInfo);
                    break;
                case /* optional string email_address */ 7:
                    message.emailAddress = reader.string();
                    break;
                case /* string business_account_id */ 8:
                    message.businessAccountId = reader.string();
                    break;
                case /* optional string business_name_filter */ 9:
                    message.businessNameFilter = reader.string();
                    break;
                case /* repeated string category_filters */ 11:
                    message.categoryFilters.push(reader.string());
                    break;
                case /* repeated string label_filters */ 12:
                    message.labelFilters.push(reader.string());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Feed.PlacesLocationFeedData.OAuthInfo oauth_info = 1; */
        if (message.oauthInfo)
            exports.Feed_PlacesLocationFeedData_OAuthInfo.internalBinaryWrite(message.oauthInfo, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string email_address = 7; */
        if (message.emailAddress !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.emailAddress);
        /* string business_account_id = 8; */
        if (message.businessAccountId !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.businessAccountId);
        /* optional string business_name_filter = 9; */
        if (message.businessNameFilter !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.businessNameFilter);
        /* repeated string category_filters = 11; */
        for (let i = 0; i < message.categoryFilters.length; i++)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.categoryFilters[i]);
        /* repeated string label_filters = 12; */
        for (let i = 0; i < message.labelFilters.length; i++)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.labelFilters[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Feed.PlacesLocationFeedData
 */
exports.Feed_PlacesLocationFeedData = new Feed_PlacesLocationFeedData$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Feed_PlacesLocationFeedData_OAuthInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Feed.PlacesLocationFeedData.OAuthInfo", [
            { no: 4, name: "http_method", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "http_request_url", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "http_authorization_header", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string http_method */ 4:
                    message.httpMethod = reader.string();
                    break;
                case /* optional string http_request_url */ 5:
                    message.httpRequestUrl = reader.string();
                    break;
                case /* optional string http_authorization_header */ 6:
                    message.httpAuthorizationHeader = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string http_method = 4; */
        if (message.httpMethod !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.httpMethod);
        /* optional string http_request_url = 5; */
        if (message.httpRequestUrl !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.httpRequestUrl);
        /* optional string http_authorization_header = 6; */
        if (message.httpAuthorizationHeader !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.httpAuthorizationHeader);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Feed.PlacesLocationFeedData.OAuthInfo
 */
exports.Feed_PlacesLocationFeedData_OAuthInfo = new Feed_PlacesLocationFeedData_OAuthInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Feed_AffiliateLocationFeedData$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Feed.AffiliateLocationFeedData", [
            { no: 3, name: "chain_ids", kind: "scalar", repeat: 1 /*RepeatType.PACKED*/, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 2, name: "relationship_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AffiliateLocationFeedRelationshipTypeEnum.AffiliateLocationFeedRelationshipType", affiliate_location_feed_relationship_type_1.AffiliateLocationFeedRelationshipTypeEnum_AffiliateLocationFeedRelationshipType] }
        ]);
    }
    create(value) {
        const message = { chainIds: [], relationshipType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated int64 chain_ids */ 3:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.chainIds.push(reader.int64().toBigInt());
                    else
                        message.chainIds.push(reader.int64().toBigInt());
                    break;
                case /* google.ads.googleads.v11.enums.AffiliateLocationFeedRelationshipTypeEnum.AffiliateLocationFeedRelationshipType relationship_type */ 2:
                    message.relationshipType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated int64 chain_ids = 3; */
        if (message.chainIds.length) {
            writer.tag(3, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.chainIds.length; i++)
                writer.int64(message.chainIds[i]);
            writer.join();
        }
        /* google.ads.googleads.v11.enums.AffiliateLocationFeedRelationshipTypeEnum.AffiliateLocationFeedRelationshipType relationship_type = 2; */
        if (message.relationshipType !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.relationshipType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Feed.AffiliateLocationFeedData
 */
exports.Feed_AffiliateLocationFeedData = new Feed_AffiliateLocationFeedData$Type();
// @generated message type with reflection information, may provide speed optimized methods
class FeedAttribute$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedAttribute", [
            { no: 5, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 6, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedAttributeTypeEnum.FeedAttributeType", feed_attribute_type_1.FeedAttributeTypeEnum_FeedAttributeType] },
            { no: 7, name: "is_part_of_key", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { type: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 id */ 5:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string name */ 6:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.FeedAttributeTypeEnum.FeedAttributeType type */ 3:
                    message.type = reader.int32();
                    break;
                case /* optional bool is_part_of_key */ 7:
                    message.isPartOfKey = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 id = 5; */
        if (message.id !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.id);
        /* optional string name = 6; */
        if (message.name !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.enums.FeedAttributeTypeEnum.FeedAttributeType type = 3; */
        if (message.type !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.type);
        /* optional bool is_part_of_key = 7; */
        if (message.isPartOfKey !== undefined)
            writer.tag(7, runtime_1.WireType.Varint).bool(message.isPartOfKey);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedAttribute
 */
exports.FeedAttribute = new FeedAttribute$Type();
// @generated message type with reflection information, may provide speed optimized methods
class FeedAttributeOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedAttributeOperation", [
            { no: 1, name: "operator", kind: "enum", T: () => ["google.ads.googleads.v11.resources.FeedAttributeOperation.Operator", FeedAttributeOperation_Operator], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "value", kind: "message", T: () => exports.FeedAttribute, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { operator: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.FeedAttributeOperation.Operator operator */ 1:
                    message.operator = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.FeedAttribute value */ 2:
                    message.value = exports.FeedAttribute.internalBinaryRead(reader, reader.uint32(), options, message.value);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.FeedAttributeOperation.Operator operator = 1; */
        if (message.operator !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.operator);
        /* google.ads.googleads.v11.resources.FeedAttribute value = 2; */
        if (message.value)
            exports.FeedAttribute.internalBinaryWrite(message.value, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedAttributeOperation
 */
exports.FeedAttributeOperation = new FeedAttributeOperation$Type();
