// google-ads-api-client/dist/protos/bindings/ads/resources/feed_item_target.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemTarget = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const criteria_1 = require("../common/criteria");
const feed_item_target_device_1 = require("../enums/feed_item_target_device");
const criteria_2 = require("../common/criteria");
const feed_item_target_status_1 = require("../enums/feed_item_target_status");
const feed_item_target_type_1 = require("../enums/feed_item_target_type");
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemTarget$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedItemTarget", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/FeedItemTarget" } } },
            { no: 12, name: "feed_item", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/FeedItem" } } },
            { no: 3, name: "feed_item_target_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemTargetTypeEnum.FeedItemTargetType", feed_item_target_type_1.FeedItemTargetTypeEnum_FeedItemTargetType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "feed_item_target_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemTargetStatusEnum.FeedItemTargetStatus", feed_item_target_status_1.FeedItemTargetStatusEnum_FeedItemTargetStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 14, name: "campaign", kind: "scalar", oneof: "target", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 15, name: "ad_group", kind: "scalar", oneof: "target", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } },
            { no: 7, name: "keyword", kind: "message", oneof: "target", T: () => criteria_2.KeywordInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 16, name: "geo_target_constant", kind: "scalar", oneof: "target", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/GeoTargetConstant" } } },
            { no: 9, name: "device", kind: "enum", oneof: "target", T: () => ["google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum.FeedItemTargetDevice", feed_item_target_device_1.FeedItemTargetDeviceEnum_FeedItemTargetDevice], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 10, name: "ad_schedule", kind: "message", oneof: "target", T: () => criteria_1.AdScheduleInfo, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/FeedItemTarget", pattern: ["customers/{customer_id}/feedItemTargets/{feed_id}~{feed_item_id}~{feed_item_target_type}~{feed_item_target_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", feedItemTargetType: 0, status: 0, target: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional string feed_item */ 12:
                    message.feedItem = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemTargetTypeEnum.FeedItemTargetType feed_item_target_type */ 3:
                    message.feedItemTargetType = reader.int32();
                    break;
                case /* optional int64 feed_item_target_id */ 13:
                    message.feedItemTargetId = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemTargetStatusEnum.FeedItemTargetStatus status */ 11:
                    message.status = reader.int32();
                    break;
                case /* string campaign */ 14:
                    message.target = {
                        oneofKind: "campaign",
                        campaign: reader.string()
                    };
                    break;
                case /* string ad_group */ 15:
                    message.target = {
                        oneofKind: "adGroup",
                        adGroup: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.common.KeywordInfo keyword */ 7:
                    message.target = {
                        oneofKind: "keyword",
                        keyword: criteria_2.KeywordInfo.internalBinaryRead(reader, reader.uint32(), options, message.target.keyword)
                    };
                    break;
                case /* string geo_target_constant */ 16:
                    message.target = {
                        oneofKind: "geoTargetConstant",
                        geoTargetConstant: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum.FeedItemTargetDevice device */ 9:
                    message.target = {
                        oneofKind: "device",
                        device: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.common.AdScheduleInfo ad_schedule */ 10:
                    message.target = {
                        oneofKind: "adSchedule",
                        adSchedule: criteria_1.AdScheduleInfo.internalBinaryRead(reader, reader.uint32(), options, message.target.adSchedule)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional string feed_item = 12; */
        if (message.feedItem !== undefined)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.feedItem);
        /* google.ads.googleads.v11.enums.FeedItemTargetTypeEnum.FeedItemTargetType feed_item_target_type = 3; */
        if (message.feedItemTargetType !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.feedItemTargetType);
        /* optional int64 feed_item_target_id = 13; */
        if (message.feedItemTargetId !== undefined)
            writer.tag(13, runtime_1.WireType.Varint).int64(message.feedItemTargetId);
        /* google.ads.googleads.v11.enums.FeedItemTargetStatusEnum.FeedItemTargetStatus status = 11; */
        if (message.status !== 0)
            writer.tag(11, runtime_1.WireType.Varint).int32(message.status);
        /* string campaign = 14; */
        if (message.target.oneofKind === "campaign")
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.target.campaign);
        /* string ad_group = 15; */
        if (message.target.oneofKind === "adGroup")
            writer.tag(15, runtime_1.WireType.LengthDelimited).string(message.target.adGroup);
        /* google.ads.googleads.v11.common.KeywordInfo keyword = 7; */
        if (message.target.oneofKind === "keyword")
            criteria_2.KeywordInfo.internalBinaryWrite(message.target.keyword, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string geo_target_constant = 16; */
        if (message.target.oneofKind === "geoTargetConstant")
            writer.tag(16, runtime_1.WireType.LengthDelimited).string(message.target.geoTargetConstant);
        /* google.ads.googleads.v11.enums.FeedItemTargetDeviceEnum.FeedItemTargetDevice device = 9; */
        if (message.target.oneofKind === "device")
            writer.tag(9, runtime_1.WireType.Varint).int32(message.target.device);
        /* google.ads.googleads.v11.common.AdScheduleInfo ad_schedule = 10; */
        if (message.target.oneofKind === "adSchedule")
            criteria_1.AdScheduleInfo.internalBinaryWrite(message.target.adSchedule, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedItemTarget
 */
exports.FeedItemTarget = new FeedItemTarget$Type();
