// google-ads-api-client/dist/protos/bindings/ads/resources/feed_item.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemValidationError = exports.FeedItemPlaceholderPolicyInfo = exports.FeedItemAttributeValue = exports.FeedItem = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const feed_item_validation_error_1 = require("../errors/feed_item_validation_error");
const feed_item_quality_disapproval_reason_1 = require("../enums/feed_item_quality_disapproval_reason");
const feed_item_quality_approval_status_1 = require("../enums/feed_item_quality_approval_status");
const feed_item_validation_status_1 = require("../enums/feed_item_validation_status");
const policy_1 = require("../common/policy");
const policy_approval_status_1 = require("../enums/policy_approval_status");
const policy_review_status_1 = require("../enums/policy_review_status");
const placeholder_type_1 = require("../enums/placeholder_type");
const feed_common_1 = require("../common/feed_common");
const feed_item_status_1 = require("../enums/feed_item_status");
const custom_parameter_1 = require("../common/custom_parameter");
const geo_targeting_restriction_1 = require("../enums/geo_targeting_restriction");
// @generated message type with reflection information, may provide speed optimized methods
class FeedItem$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedItem", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/FeedItem" } } },
            { no: 11, name: "feed", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Feed" } } },
            { no: 12, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "start_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 14, name: "end_date_time", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "attribute_values", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.FeedItemAttributeValue },
            { no: 7, name: "geo_targeting_restriction", kind: "enum", T: () => ["google.ads.googleads.v11.enums.GeoTargetingRestrictionEnum.GeoTargetingRestriction", geo_targeting_restriction_1.GeoTargetingRestrictionEnum_GeoTargetingRestriction] },
            { no: 8, name: "url_custom_parameters", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => custom_parameter_1.CustomParameter },
            { no: 9, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemStatusEnum.FeedItemStatus", feed_item_status_1.FeedItemStatusEnum_FeedItemStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "policy_infos", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.FeedItemPlaceholderPolicyInfo, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/FeedItem", pattern: ["customers/{customer_id}/feedItems/{feed_id}~{feed_item_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", attributeValues: [], geoTargetingRestriction: 0, urlCustomParameters: [], status: 0, policyInfos: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional string feed */ 11:
                    message.feed = reader.string();
                    break;
                case /* optional int64 id */ 12:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string start_date_time */ 13:
                    message.startDateTime = reader.string();
                    break;
                case /* optional string end_date_time */ 14:
                    message.endDateTime = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.resources.FeedItemAttributeValue attribute_values */ 6:
                    message.attributeValues.push(exports.FeedItemAttributeValue.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.GeoTargetingRestrictionEnum.GeoTargetingRestriction geo_targeting_restriction */ 7:
                    message.geoTargetingRestriction = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters */ 8:
                    message.urlCustomParameters.push(custom_parameter_1.CustomParameter.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemStatusEnum.FeedItemStatus status */ 9:
                    message.status = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.resources.FeedItemPlaceholderPolicyInfo policy_infos */ 10:
                    message.policyInfos.push(exports.FeedItemPlaceholderPolicyInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional string feed = 11; */
        if (message.feed !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.feed);
        /* optional int64 id = 12; */
        if (message.id !== undefined)
            writer.tag(12, runtime_1.WireType.Varint).int64(message.id);
        /* optional string start_date_time = 13; */
        if (message.startDateTime !== undefined)
            writer.tag(13, runtime_1.WireType.LengthDelimited).string(message.startDateTime);
        /* optional string end_date_time = 14; */
        if (message.endDateTime !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.endDateTime);
        /* repeated google.ads.googleads.v11.resources.FeedItemAttributeValue attribute_values = 6; */
        for (let i = 0; i < message.attributeValues.length; i++)
            exports.FeedItemAttributeValue.internalBinaryWrite(message.attributeValues[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.GeoTargetingRestrictionEnum.GeoTargetingRestriction geo_targeting_restriction = 7; */
        if (message.geoTargetingRestriction !== 0)
            writer.tag(7, runtime_1.WireType.Varint).int32(message.geoTargetingRestriction);
        /* repeated google.ads.googleads.v11.common.CustomParameter url_custom_parameters = 8; */
        for (let i = 0; i < message.urlCustomParameters.length; i++)
            custom_parameter_1.CustomParameter.internalBinaryWrite(message.urlCustomParameters[i], writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.FeedItemStatusEnum.FeedItemStatus status = 9; */
        if (message.status !== 0)
            writer.tag(9, runtime_1.WireType.Varint).int32(message.status);
        /* repeated google.ads.googleads.v11.resources.FeedItemPlaceholderPolicyInfo policy_infos = 10; */
        for (let i = 0; i < message.policyInfos.length; i++)
            exports.FeedItemPlaceholderPolicyInfo.internalBinaryWrite(message.policyInfos[i], writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedItem
 */
exports.FeedItem = new FeedItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemAttributeValue$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedItemAttributeValue", [
            { no: 11, name: "feed_attribute_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 12, name: "integer_value", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 13, name: "boolean_value", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 14, name: "string_value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 15, name: "double_value", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 6, name: "price_value", kind: "message", T: () => feed_common_1.Money },
            { no: 16, name: "integer_values", kind: "scalar", repeat: 1 /*RepeatType.PACKED*/, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 17, name: "boolean_values", kind: "scalar", repeat: 1 /*RepeatType.PACKED*/, T: 8 /*ScalarType.BOOL*/ },
            { no: 18, name: "string_values", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 19, name: "double_values", kind: "scalar", repeat: 1 /*RepeatType.PACKED*/, T: 1 /*ScalarType.DOUBLE*/ }
        ]);
    }
    create(value) {
        const message = { integerValues: [], booleanValues: [], stringValues: [], doubleValues: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 feed_attribute_id */ 11:
                    message.feedAttributeId = reader.int64().toBigInt();
                    break;
                case /* optional int64 integer_value */ 12:
                    message.integerValue = reader.int64().toBigInt();
                    break;
                case /* optional bool boolean_value */ 13:
                    message.booleanValue = reader.bool();
                    break;
                case /* optional string string_value */ 14:
                    message.stringValue = reader.string();
                    break;
                case /* optional double double_value */ 15:
                    message.doubleValue = reader.double();
                    break;
                case /* google.ads.googleads.v11.common.Money price_value */ 6:
                    message.priceValue = feed_common_1.Money.internalBinaryRead(reader, reader.uint32(), options, message.priceValue);
                    break;
                case /* repeated int64 integer_values */ 16:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.integerValues.push(reader.int64().toBigInt());
                    else
                        message.integerValues.push(reader.int64().toBigInt());
                    break;
                case /* repeated bool boolean_values */ 17:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.booleanValues.push(reader.bool());
                    else
                        message.booleanValues.push(reader.bool());
                    break;
                case /* repeated string string_values */ 18:
                    message.stringValues.push(reader.string());
                    break;
                case /* repeated double double_values */ 19:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.doubleValues.push(reader.double());
                    else
                        message.doubleValues.push(reader.double());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 feed_attribute_id = 11; */
        if (message.feedAttributeId !== undefined)
            writer.tag(11, runtime_1.WireType.Varint).int64(message.feedAttributeId);
        /* optional int64 integer_value = 12; */
        if (message.integerValue !== undefined)
            writer.tag(12, runtime_1.WireType.Varint).int64(message.integerValue);
        /* optional bool boolean_value = 13; */
        if (message.booleanValue !== undefined)
            writer.tag(13, runtime_1.WireType.Varint).bool(message.booleanValue);
        /* optional string string_value = 14; */
        if (message.stringValue !== undefined)
            writer.tag(14, runtime_1.WireType.LengthDelimited).string(message.stringValue);
        /* optional double double_value = 15; */
        if (message.doubleValue !== undefined)
            writer.tag(15, runtime_1.WireType.Bit64).double(message.doubleValue);
        /* google.ads.googleads.v11.common.Money price_value = 6; */
        if (message.priceValue)
            feed_common_1.Money.internalBinaryWrite(message.priceValue, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated int64 integer_values = 16; */
        if (message.integerValues.length) {
            writer.tag(16, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.integerValues.length; i++)
                writer.int64(message.integerValues[i]);
            writer.join();
        }
        /* repeated bool boolean_values = 17; */
        if (message.booleanValues.length) {
            writer.tag(17, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.booleanValues.length; i++)
                writer.bool(message.booleanValues[i]);
            writer.join();
        }
        /* repeated string string_values = 18; */
        for (let i = 0; i < message.stringValues.length; i++)
            writer.tag(18, runtime_1.WireType.LengthDelimited).string(message.stringValues[i]);
        /* repeated double double_values = 19; */
        if (message.doubleValues.length) {
            writer.tag(19, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.doubleValues.length; i++)
                writer.double(message.doubleValues[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedItemAttributeValue
 */
exports.FeedItemAttributeValue = new FeedItemAttributeValue$Type();
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemPlaceholderPolicyInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedItemPlaceholderPolicyInfo", [
            { no: 10, name: "placeholder_type_enum", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PlaceholderTypeEnum.PlaceholderType", placeholder_type_1.PlaceholderTypeEnum_PlaceholderType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "feed_mapping_resource_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "review_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus", policy_review_status_1.PolicyReviewStatusEnum_PolicyReviewStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "approval_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus", policy_approval_status_1.PolicyApprovalStatusEnum_PolicyApprovalStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "policy_topic_entries", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => policy_1.PolicyTopicEntry, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "validation_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemValidationStatusEnum.FeedItemValidationStatus", feed_item_validation_status_1.FeedItemValidationStatusEnum_FeedItemValidationStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "validation_errors", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.FeedItemValidationError, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "quality_approval_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemQualityApprovalStatusEnum.FeedItemQualityApprovalStatus", feed_item_quality_approval_status_1.FeedItemQualityApprovalStatusEnum_FeedItemQualityApprovalStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "quality_disapproval_reasons", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.ads.googleads.v11.enums.FeedItemQualityDisapprovalReasonEnum.FeedItemQualityDisapprovalReason", feed_item_quality_disapproval_reason_1.FeedItemQualityDisapprovalReasonEnum_FeedItemQualityDisapprovalReason], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { placeholderTypeEnum: 0, reviewStatus: 0, approvalStatus: 0, policyTopicEntries: [], validationStatus: 0, validationErrors: [], qualityApprovalStatus: 0, qualityDisapprovalReasons: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.PlaceholderTypeEnum.PlaceholderType placeholder_type_enum */ 10:
                    message.placeholderTypeEnum = reader.int32();
                    break;
                case /* optional string feed_mapping_resource_name */ 11:
                    message.feedMappingResourceName = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus review_status */ 3:
                    message.reviewStatus = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus approval_status */ 4:
                    message.approvalStatus = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.common.PolicyTopicEntry policy_topic_entries */ 5:
                    message.policyTopicEntries.push(policy_1.PolicyTopicEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemValidationStatusEnum.FeedItemValidationStatus validation_status */ 6:
                    message.validationStatus = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.resources.FeedItemValidationError validation_errors */ 7:
                    message.validationErrors.push(exports.FeedItemValidationError.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemQualityApprovalStatusEnum.FeedItemQualityApprovalStatus quality_approval_status */ 8:
                    message.qualityApprovalStatus = reader.int32();
                    break;
                case /* repeated google.ads.googleads.v11.enums.FeedItemQualityDisapprovalReasonEnum.FeedItemQualityDisapprovalReason quality_disapproval_reasons */ 9:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.qualityDisapprovalReasons.push(reader.int32());
                    else
                        message.qualityDisapprovalReasons.push(reader.int32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.PlaceholderTypeEnum.PlaceholderType placeholder_type_enum = 10; */
        if (message.placeholderTypeEnum !== 0)
            writer.tag(10, runtime_1.WireType.Varint).int32(message.placeholderTypeEnum);
        /* optional string feed_mapping_resource_name = 11; */
        if (message.feedMappingResourceName !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.feedMappingResourceName);
        /* google.ads.googleads.v11.enums.PolicyReviewStatusEnum.PolicyReviewStatus review_status = 3; */
        if (message.reviewStatus !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.reviewStatus);
        /* google.ads.googleads.v11.enums.PolicyApprovalStatusEnum.PolicyApprovalStatus approval_status = 4; */
        if (message.approvalStatus !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.approvalStatus);
        /* repeated google.ads.googleads.v11.common.PolicyTopicEntry policy_topic_entries = 5; */
        for (let i = 0; i < message.policyTopicEntries.length; i++)
            policy_1.PolicyTopicEntry.internalBinaryWrite(message.policyTopicEntries[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.FeedItemValidationStatusEnum.FeedItemValidationStatus validation_status = 6; */
        if (message.validationStatus !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.validationStatus);
        /* repeated google.ads.googleads.v11.resources.FeedItemValidationError validation_errors = 7; */
        for (let i = 0; i < message.validationErrors.length; i++)
            exports.FeedItemValidationError.internalBinaryWrite(message.validationErrors[i], writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.FeedItemQualityApprovalStatusEnum.FeedItemQualityApprovalStatus quality_approval_status = 8; */
        if (message.qualityApprovalStatus !== 0)
            writer.tag(8, runtime_1.WireType.Varint).int32(message.qualityApprovalStatus);
        /* repeated google.ads.googleads.v11.enums.FeedItemQualityDisapprovalReasonEnum.FeedItemQualityDisapprovalReason quality_disapproval_reasons = 9; */
        if (message.qualityDisapprovalReasons.length) {
            writer.tag(9, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.qualityDisapprovalReasons.length; i++)
                writer.int32(message.qualityDisapprovalReasons[i]);
            writer.join();
        }
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedItemPlaceholderPolicyInfo
 */
exports.FeedItemPlaceholderPolicyInfo = new FeedItemPlaceholderPolicyInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemValidationError$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedItemValidationError", [
            { no: 1, name: "validation_error", kind: "enum", T: () => ["google.ads.googleads.v11.errors.FeedItemValidationErrorEnum.FeedItemValidationError", feed_item_validation_error_1.FeedItemValidationErrorEnum_FeedItemValidationError], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "feed_attribute_ids", kind: "scalar", repeat: 1 /*RepeatType.PACKED*/, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "extra_info", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { validationError: 0, feedAttributeIds: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.errors.FeedItemValidationErrorEnum.FeedItemValidationError validation_error */ 1:
                    message.validationError = reader.int32();
                    break;
                case /* optional string description */ 6:
                    message.description = reader.string();
                    break;
                case /* repeated int64 feed_attribute_ids */ 7:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.feedAttributeIds.push(reader.int64().toBigInt());
                    else
                        message.feedAttributeIds.push(reader.int64().toBigInt());
                    break;
                case /* optional string extra_info */ 8:
                    message.extraInfo = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.errors.FeedItemValidationErrorEnum.FeedItemValidationError validation_error = 1; */
        if (message.validationError !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.validationError);
        /* optional string description = 6; */
        if (message.description !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.description);
        /* repeated int64 feed_attribute_ids = 7; */
        if (message.feedAttributeIds.length) {
            writer.tag(7, runtime_1.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.feedAttributeIds.length; i++)
                writer.int64(message.feedAttributeIds[i]);
            writer.join();
        }
        /* optional string extra_info = 8; */
        if (message.extraInfo !== undefined)
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.extraInfo);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedItemValidationError
 */
exports.FeedItemValidationError = new FeedItemValidationError$Type();
