// google-ads-api-client/dist/protos/bindings/ads/resources/feed_mapping.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttributeFieldMapping = exports.FeedMapping = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const image_placeholder_field_1 = require("../enums/image_placeholder_field");
const job_placeholder_field_1 = require("../enums/job_placeholder_field");
const local_placeholder_field_1 = require("../enums/local_placeholder_field");
const travel_placeholder_field_1 = require("../enums/travel_placeholder_field");
const real_estate_placeholder_field_1 = require("../enums/real_estate_placeholder_field");
const hotel_placeholder_field_1 = require("../enums/hotel_placeholder_field");
const custom_placeholder_field_1 = require("../enums/custom_placeholder_field");
const flight_placeholder_field_1 = require("../enums/flight_placeholder_field");
const education_placeholder_field_1 = require("../enums/education_placeholder_field");
const location_extension_targeting_criterion_field_1 = require("../enums/location_extension_targeting_criterion_field");
const dsa_page_feed_criterion_field_1 = require("../enums/dsa_page_feed_criterion_field");
const ad_customizer_placeholder_field_1 = require("../enums/ad_customizer_placeholder_field");
const promotion_placeholder_field_1 = require("../enums/promotion_placeholder_field");
const price_placeholder_field_1 = require("../enums/price_placeholder_field");
const message_placeholder_field_1 = require("../enums/message_placeholder_field");
const structured_snippet_placeholder_field_1 = require("../enums/structured_snippet_placeholder_field");
const callout_placeholder_field_1 = require("../enums/callout_placeholder_field");
const affiliate_location_placeholder_field_1 = require("../enums/affiliate_location_placeholder_field");
const location_placeholder_field_1 = require("../enums/location_placeholder_field");
const app_placeholder_field_1 = require("../enums/app_placeholder_field");
const call_placeholder_field_1 = require("../enums/call_placeholder_field");
const sitelink_placeholder_field_1 = require("../enums/sitelink_placeholder_field");
const feed_mapping_criterion_type_1 = require("../enums/feed_mapping_criterion_type");
const placeholder_type_1 = require("../enums/placeholder_type");
const feed_mapping_status_1 = require("../enums/feed_mapping_status");
// @generated message type with reflection information, may provide speed optimized methods
class FeedMapping$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedMapping", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/FeedMapping" } } },
            { no: 7, name: "feed", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Feed" } } },
            { no: 5, name: "attribute_field_mappings", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.AttributeFieldMapping, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedMappingStatusEnum.FeedMappingStatus", feed_mapping_status_1.FeedMappingStatusEnum_FeedMappingStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "placeholder_type", kind: "enum", oneof: "target", T: () => ["google.ads.googleads.v11.enums.PlaceholderTypeEnum.PlaceholderType", placeholder_type_1.PlaceholderTypeEnum_PlaceholderType], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 4, name: "criterion_type", kind: "enum", oneof: "target", T: () => ["google.ads.googleads.v11.enums.FeedMappingCriterionTypeEnum.FeedMappingCriterionType", feed_mapping_criterion_type_1.FeedMappingCriterionTypeEnum_FeedMappingCriterionType], options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/FeedMapping", pattern: ["customers/{customer_id}/feedMappings/{feed_id}~{feed_mapping_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", attributeFieldMappings: [], status: 0, target: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional string feed */ 7:
                    message.feed = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.resources.AttributeFieldMapping attribute_field_mappings */ 5:
                    message.attributeFieldMappings.push(exports.AttributeFieldMapping.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* google.ads.googleads.v11.enums.FeedMappingStatusEnum.FeedMappingStatus status */ 6:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.PlaceholderTypeEnum.PlaceholderType placeholder_type */ 3:
                    message.target = {
                        oneofKind: "placeholderType",
                        placeholderType: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.FeedMappingCriterionTypeEnum.FeedMappingCriterionType criterion_type */ 4:
                    message.target = {
                        oneofKind: "criterionType",
                        criterionType: reader.int32()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional string feed = 7; */
        if (message.feed !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.feed);
        /* repeated google.ads.googleads.v11.resources.AttributeFieldMapping attribute_field_mappings = 5; */
        for (let i = 0; i < message.attributeFieldMappings.length; i++)
            exports.AttributeFieldMapping.internalBinaryWrite(message.attributeFieldMappings[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.FeedMappingStatusEnum.FeedMappingStatus status = 6; */
        if (message.status !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.enums.PlaceholderTypeEnum.PlaceholderType placeholder_type = 3; */
        if (message.target.oneofKind === "placeholderType")
            writer.tag(3, runtime_1.WireType.Varint).int32(message.target.placeholderType);
        /* google.ads.googleads.v11.enums.FeedMappingCriterionTypeEnum.FeedMappingCriterionType criterion_type = 4; */
        if (message.target.oneofKind === "criterionType")
            writer.tag(4, runtime_1.WireType.Varint).int32(message.target.criterionType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedMapping
 */
exports.FeedMapping = new FeedMapping$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AttributeFieldMapping$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.AttributeFieldMapping", [
            { no: 24, name: "feed_attribute_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 25, name: "field_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "sitelink_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.SitelinkPlaceholderFieldEnum.SitelinkPlaceholderField", sitelink_placeholder_field_1.SitelinkPlaceholderFieldEnum_SitelinkPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 4, name: "call_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.CallPlaceholderFieldEnum.CallPlaceholderField", call_placeholder_field_1.CallPlaceholderFieldEnum_CallPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 5, name: "app_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.AppPlaceholderFieldEnum.AppPlaceholderField", app_placeholder_field_1.AppPlaceholderFieldEnum_AppPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 6, name: "location_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.LocationPlaceholderFieldEnum.LocationPlaceholderField", location_placeholder_field_1.LocationPlaceholderFieldEnum_LocationPlaceholderField], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "affiliate_location_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.AffiliateLocationPlaceholderFieldEnum.AffiliateLocationPlaceholderField", affiliate_location_placeholder_field_1.AffiliateLocationPlaceholderFieldEnum_AffiliateLocationPlaceholderField], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "callout_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.CalloutPlaceholderFieldEnum.CalloutPlaceholderField", callout_placeholder_field_1.CalloutPlaceholderFieldEnum_CalloutPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 9, name: "structured_snippet_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.StructuredSnippetPlaceholderFieldEnum.StructuredSnippetPlaceholderField", structured_snippet_placeholder_field_1.StructuredSnippetPlaceholderFieldEnum_StructuredSnippetPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 10, name: "message_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.MessagePlaceholderFieldEnum.MessagePlaceholderField", message_placeholder_field_1.MessagePlaceholderFieldEnum_MessagePlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 11, name: "price_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.PricePlaceholderFieldEnum.PricePlaceholderField", price_placeholder_field_1.PricePlaceholderFieldEnum_PricePlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 12, name: "promotion_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.PromotionPlaceholderFieldEnum.PromotionPlaceholderField", promotion_placeholder_field_1.PromotionPlaceholderFieldEnum_PromotionPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 13, name: "ad_customizer_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.AdCustomizerPlaceholderFieldEnum.AdCustomizerPlaceholderField", ad_customizer_placeholder_field_1.AdCustomizerPlaceholderFieldEnum_AdCustomizerPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 14, name: "dsa_page_feed_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.DsaPageFeedCriterionFieldEnum.DsaPageFeedCriterionField", dsa_page_feed_criterion_field_1.DsaPageFeedCriterionFieldEnum_DsaPageFeedCriterionField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 15, name: "location_extension_targeting_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.LocationExtensionTargetingCriterionFieldEnum.LocationExtensionTargetingCriterionField", location_extension_targeting_criterion_field_1.LocationExtensionTargetingCriterionFieldEnum_LocationExtensionTargetingCriterionField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 16, name: "education_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.EducationPlaceholderFieldEnum.EducationPlaceholderField", education_placeholder_field_1.EducationPlaceholderFieldEnum_EducationPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 17, name: "flight_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.FlightPlaceholderFieldEnum.FlightPlaceholderField", flight_placeholder_field_1.FlightPlaceholderFieldEnum_FlightPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 18, name: "custom_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.CustomPlaceholderFieldEnum.CustomPlaceholderField", custom_placeholder_field_1.CustomPlaceholderFieldEnum_CustomPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 19, name: "hotel_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.HotelPlaceholderFieldEnum.HotelPlaceholderField", hotel_placeholder_field_1.HotelPlaceholderFieldEnum_HotelPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 20, name: "real_estate_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.RealEstatePlaceholderFieldEnum.RealEstatePlaceholderField", real_estate_placeholder_field_1.RealEstatePlaceholderFieldEnum_RealEstatePlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 21, name: "travel_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.TravelPlaceholderFieldEnum.TravelPlaceholderField", travel_placeholder_field_1.TravelPlaceholderFieldEnum_TravelPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 22, name: "local_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.LocalPlaceholderFieldEnum.LocalPlaceholderField", local_placeholder_field_1.LocalPlaceholderFieldEnum_LocalPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 23, name: "job_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.JobPlaceholderFieldEnum.JobPlaceholderField", job_placeholder_field_1.JobPlaceholderFieldEnum_JobPlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 26, name: "image_field", kind: "enum", oneof: "field", T: () => ["google.ads.googleads.v11.enums.ImagePlaceholderFieldEnum.ImagePlaceholderField", image_placeholder_field_1.ImagePlaceholderFieldEnum_ImagePlaceholderField], options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ]);
    }
    create(value) {
        const message = { field: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 feed_attribute_id */ 24:
                    message.feedAttributeId = reader.int64().toBigInt();
                    break;
                case /* optional int64 field_id */ 25:
                    message.fieldId = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.SitelinkPlaceholderFieldEnum.SitelinkPlaceholderField sitelink_field */ 3:
                    message.field = {
                        oneofKind: "sitelinkField",
                        sitelinkField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.CallPlaceholderFieldEnum.CallPlaceholderField call_field */ 4:
                    message.field = {
                        oneofKind: "callField",
                        callField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.AppPlaceholderFieldEnum.AppPlaceholderField app_field */ 5:
                    message.field = {
                        oneofKind: "appField",
                        appField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.LocationPlaceholderFieldEnum.LocationPlaceholderField location_field */ 6:
                    message.field = {
                        oneofKind: "locationField",
                        locationField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.AffiliateLocationPlaceholderFieldEnum.AffiliateLocationPlaceholderField affiliate_location_field */ 7:
                    message.field = {
                        oneofKind: "affiliateLocationField",
                        affiliateLocationField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.CalloutPlaceholderFieldEnum.CalloutPlaceholderField callout_field */ 8:
                    message.field = {
                        oneofKind: "calloutField",
                        calloutField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.StructuredSnippetPlaceholderFieldEnum.StructuredSnippetPlaceholderField structured_snippet_field */ 9:
                    message.field = {
                        oneofKind: "structuredSnippetField",
                        structuredSnippetField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.MessagePlaceholderFieldEnum.MessagePlaceholderField message_field */ 10:
                    message.field = {
                        oneofKind: "messageField",
                        messageField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.PricePlaceholderFieldEnum.PricePlaceholderField price_field */ 11:
                    message.field = {
                        oneofKind: "priceField",
                        priceField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.PromotionPlaceholderFieldEnum.PromotionPlaceholderField promotion_field */ 12:
                    message.field = {
                        oneofKind: "promotionField",
                        promotionField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.AdCustomizerPlaceholderFieldEnum.AdCustomizerPlaceholderField ad_customizer_field */ 13:
                    message.field = {
                        oneofKind: "adCustomizerField",
                        adCustomizerField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.DsaPageFeedCriterionFieldEnum.DsaPageFeedCriterionField dsa_page_feed_field */ 14:
                    message.field = {
                        oneofKind: "dsaPageFeedField",
                        dsaPageFeedField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.LocationExtensionTargetingCriterionFieldEnum.LocationExtensionTargetingCriterionField location_extension_targeting_field */ 15:
                    message.field = {
                        oneofKind: "locationExtensionTargetingField",
                        locationExtensionTargetingField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.EducationPlaceholderFieldEnum.EducationPlaceholderField education_field */ 16:
                    message.field = {
                        oneofKind: "educationField",
                        educationField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.FlightPlaceholderFieldEnum.FlightPlaceholderField flight_field */ 17:
                    message.field = {
                        oneofKind: "flightField",
                        flightField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.CustomPlaceholderFieldEnum.CustomPlaceholderField custom_field */ 18:
                    message.field = {
                        oneofKind: "customField",
                        customField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.HotelPlaceholderFieldEnum.HotelPlaceholderField hotel_field */ 19:
                    message.field = {
                        oneofKind: "hotelField",
                        hotelField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.RealEstatePlaceholderFieldEnum.RealEstatePlaceholderField real_estate_field */ 20:
                    message.field = {
                        oneofKind: "realEstateField",
                        realEstateField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.TravelPlaceholderFieldEnum.TravelPlaceholderField travel_field */ 21:
                    message.field = {
                        oneofKind: "travelField",
                        travelField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.LocalPlaceholderFieldEnum.LocalPlaceholderField local_field */ 22:
                    message.field = {
                        oneofKind: "localField",
                        localField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.JobPlaceholderFieldEnum.JobPlaceholderField job_field */ 23:
                    message.field = {
                        oneofKind: "jobField",
                        jobField: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.enums.ImagePlaceholderFieldEnum.ImagePlaceholderField image_field */ 26:
                    message.field = {
                        oneofKind: "imageField",
                        imageField: reader.int32()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 feed_attribute_id = 24; */
        if (message.feedAttributeId !== undefined)
            writer.tag(24, runtime_1.WireType.Varint).int64(message.feedAttributeId);
        /* optional int64 field_id = 25; */
        if (message.fieldId !== undefined)
            writer.tag(25, runtime_1.WireType.Varint).int64(message.fieldId);
        /* google.ads.googleads.v11.enums.SitelinkPlaceholderFieldEnum.SitelinkPlaceholderField sitelink_field = 3; */
        if (message.field.oneofKind === "sitelinkField")
            writer.tag(3, runtime_1.WireType.Varint).int32(message.field.sitelinkField);
        /* google.ads.googleads.v11.enums.CallPlaceholderFieldEnum.CallPlaceholderField call_field = 4; */
        if (message.field.oneofKind === "callField")
            writer.tag(4, runtime_1.WireType.Varint).int32(message.field.callField);
        /* google.ads.googleads.v11.enums.AppPlaceholderFieldEnum.AppPlaceholderField app_field = 5; */
        if (message.field.oneofKind === "appField")
            writer.tag(5, runtime_1.WireType.Varint).int32(message.field.appField);
        /* google.ads.googleads.v11.enums.LocationPlaceholderFieldEnum.LocationPlaceholderField location_field = 6; */
        if (message.field.oneofKind === "locationField")
            writer.tag(6, runtime_1.WireType.Varint).int32(message.field.locationField);
        /* google.ads.googleads.v11.enums.AffiliateLocationPlaceholderFieldEnum.AffiliateLocationPlaceholderField affiliate_location_field = 7; */
        if (message.field.oneofKind === "affiliateLocationField")
            writer.tag(7, runtime_1.WireType.Varint).int32(message.field.affiliateLocationField);
        /* google.ads.googleads.v11.enums.CalloutPlaceholderFieldEnum.CalloutPlaceholderField callout_field = 8; */
        if (message.field.oneofKind === "calloutField")
            writer.tag(8, runtime_1.WireType.Varint).int32(message.field.calloutField);
        /* google.ads.googleads.v11.enums.StructuredSnippetPlaceholderFieldEnum.StructuredSnippetPlaceholderField structured_snippet_field = 9; */
        if (message.field.oneofKind === "structuredSnippetField")
            writer.tag(9, runtime_1.WireType.Varint).int32(message.field.structuredSnippetField);
        /* google.ads.googleads.v11.enums.MessagePlaceholderFieldEnum.MessagePlaceholderField message_field = 10; */
        if (message.field.oneofKind === "messageField")
            writer.tag(10, runtime_1.WireType.Varint).int32(message.field.messageField);
        /* google.ads.googleads.v11.enums.PricePlaceholderFieldEnum.PricePlaceholderField price_field = 11; */
        if (message.field.oneofKind === "priceField")
            writer.tag(11, runtime_1.WireType.Varint).int32(message.field.priceField);
        /* google.ads.googleads.v11.enums.PromotionPlaceholderFieldEnum.PromotionPlaceholderField promotion_field = 12; */
        if (message.field.oneofKind === "promotionField")
            writer.tag(12, runtime_1.WireType.Varint).int32(message.field.promotionField);
        /* google.ads.googleads.v11.enums.AdCustomizerPlaceholderFieldEnum.AdCustomizerPlaceholderField ad_customizer_field = 13; */
        if (message.field.oneofKind === "adCustomizerField")
            writer.tag(13, runtime_1.WireType.Varint).int32(message.field.adCustomizerField);
        /* google.ads.googleads.v11.enums.DsaPageFeedCriterionFieldEnum.DsaPageFeedCriterionField dsa_page_feed_field = 14; */
        if (message.field.oneofKind === "dsaPageFeedField")
            writer.tag(14, runtime_1.WireType.Varint).int32(message.field.dsaPageFeedField);
        /* google.ads.googleads.v11.enums.LocationExtensionTargetingCriterionFieldEnum.LocationExtensionTargetingCriterionField location_extension_targeting_field = 15; */
        if (message.field.oneofKind === "locationExtensionTargetingField")
            writer.tag(15, runtime_1.WireType.Varint).int32(message.field.locationExtensionTargetingField);
        /* google.ads.googleads.v11.enums.EducationPlaceholderFieldEnum.EducationPlaceholderField education_field = 16; */
        if (message.field.oneofKind === "educationField")
            writer.tag(16, runtime_1.WireType.Varint).int32(message.field.educationField);
        /* google.ads.googleads.v11.enums.FlightPlaceholderFieldEnum.FlightPlaceholderField flight_field = 17; */
        if (message.field.oneofKind === "flightField")
            writer.tag(17, runtime_1.WireType.Varint).int32(message.field.flightField);
        /* google.ads.googleads.v11.enums.CustomPlaceholderFieldEnum.CustomPlaceholderField custom_field = 18; */
        if (message.field.oneofKind === "customField")
            writer.tag(18, runtime_1.WireType.Varint).int32(message.field.customField);
        /* google.ads.googleads.v11.enums.HotelPlaceholderFieldEnum.HotelPlaceholderField hotel_field = 19; */
        if (message.field.oneofKind === "hotelField")
            writer.tag(19, runtime_1.WireType.Varint).int32(message.field.hotelField);
        /* google.ads.googleads.v11.enums.RealEstatePlaceholderFieldEnum.RealEstatePlaceholderField real_estate_field = 20; */
        if (message.field.oneofKind === "realEstateField")
            writer.tag(20, runtime_1.WireType.Varint).int32(message.field.realEstateField);
        /* google.ads.googleads.v11.enums.TravelPlaceholderFieldEnum.TravelPlaceholderField travel_field = 21; */
        if (message.field.oneofKind === "travelField")
            writer.tag(21, runtime_1.WireType.Varint).int32(message.field.travelField);
        /* google.ads.googleads.v11.enums.LocalPlaceholderFieldEnum.LocalPlaceholderField local_field = 22; */
        if (message.field.oneofKind === "localField")
            writer.tag(22, runtime_1.WireType.Varint).int32(message.field.localField);
        /* google.ads.googleads.v11.enums.JobPlaceholderFieldEnum.JobPlaceholderField job_field = 23; */
        if (message.field.oneofKind === "jobField")
            writer.tag(23, runtime_1.WireType.Varint).int32(message.field.jobField);
        /* google.ads.googleads.v11.enums.ImagePlaceholderFieldEnum.ImagePlaceholderField image_field = 26; */
        if (message.field.oneofKind === "imageField")
            writer.tag(26, runtime_1.WireType.Varint).int32(message.field.imageField);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.AttributeFieldMapping
 */
exports.AttributeFieldMapping = new AttributeFieldMapping$Type();
