// google-ads-api-client/dist/protos/bindings/ads/resources/feed_item_set.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedItemSet = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const feed_item_set_filter_type_infos_1 = require("../common/feed_item_set_filter_type_infos");
const feed_item_set_filter_type_infos_2 = require("../common/feed_item_set_filter_type_infos");
const feed_item_set_status_1 = require("../enums/feed_item_set_status");
// @generated message type with reflection information, may provide speed optimized methods
class FeedItemSet$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.FeedItemSet", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/FeedItemSet" } } },
            { no: 2, name: "feed", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Feed" } } },
            { no: 3, name: "feed_item_set_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "display_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.FeedItemSetStatusEnum.FeedItemSetStatus", feed_item_set_status_1.FeedItemSetStatusEnum_FeedItemSetStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "dynamic_location_set_filter", kind: "message", oneof: "dynamicSetFilter", T: () => feed_item_set_filter_type_infos_2.DynamicLocationSetFilter },
            { no: 6, name: "dynamic_affiliate_location_set_filter", kind: "message", oneof: "dynamicSetFilter", T: () => feed_item_set_filter_type_infos_1.DynamicAffiliateLocationSetFilter }
        ], { "google.api.resource": { type: "googleads.googleapis.com/FeedItemSet", pattern: ["customers/{customer_id}/feedItemSets/{feed_id}~{feed_item_set_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", feed: "", feedItemSetId: 0n, displayName: "", status: 0, dynamicSetFilter: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* string feed */ 2:
                    message.feed = reader.string();
                    break;
                case /* int64 feed_item_set_id */ 3:
                    message.feedItemSetId = reader.int64().toBigInt();
                    break;
                case /* string display_name */ 4:
                    message.displayName = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.FeedItemSetStatusEnum.FeedItemSetStatus status */ 8:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.DynamicLocationSetFilter dynamic_location_set_filter */ 5:
                    message.dynamicSetFilter = {
                        oneofKind: "dynamicLocationSetFilter",
                        dynamicLocationSetFilter: feed_item_set_filter_type_infos_2.DynamicLocationSetFilter.internalBinaryRead(reader, reader.uint32(), options, message.dynamicSetFilter.dynamicLocationSetFilter)
                    };
                    break;
                case /* google.ads.googleads.v11.common.DynamicAffiliateLocationSetFilter dynamic_affiliate_location_set_filter */ 6:
                    message.dynamicSetFilter = {
                        oneofKind: "dynamicAffiliateLocationSetFilter",
                        dynamicAffiliateLocationSetFilter: feed_item_set_filter_type_infos_1.DynamicAffiliateLocationSetFilter.internalBinaryRead(reader, reader.uint32(), options, message.dynamicSetFilter.dynamicAffiliateLocationSetFilter)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* string feed = 2; */
        if (message.feed !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.feed);
        /* int64 feed_item_set_id = 3; */
        if (message.feedItemSetId !== 0n)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.feedItemSetId);
        /* string display_name = 4; */
        if (message.displayName !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.displayName);
        /* google.ads.googleads.v11.enums.FeedItemSetStatusEnum.FeedItemSetStatus status = 8; */
        if (message.status !== 0)
            writer.tag(8, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.common.DynamicLocationSetFilter dynamic_location_set_filter = 5; */
        if (message.dynamicSetFilter.oneofKind === "dynamicLocationSetFilter")
            feed_item_set_filter_type_infos_2.DynamicLocationSetFilter.internalBinaryWrite(message.dynamicSetFilter.dynamicLocationSetFilter, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.DynamicAffiliateLocationSetFilter dynamic_affiliate_location_set_filter = 6; */
        if (message.dynamicSetFilter.oneofKind === "dynamicAffiliateLocationSetFilter")
            feed_item_set_filter_type_infos_1.DynamicAffiliateLocationSetFilter.internalBinaryWrite(message.dynamicSetFilter.dynamicAffiliateLocationSetFilter, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.FeedItemSet
 */
exports.FeedItemSet = new FeedItemSet$Type();
