// google-ads-api-client/dist/protos/bindings/ads/resources/keyword_plan.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanForecastPeriod = exports.KeywordPlan = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const dates_1 = require("../common/dates");
const keyword_plan_forecast_interval_1 = require("../enums/keyword_plan_forecast_interval");
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlan$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.KeywordPlan", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlan" } } },
            { no: 5, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "forecast_period", kind: "message", T: () => exports.KeywordPlanForecastPeriod }
        ], { "google.api.resource": { type: "googleads.googleapis.com/KeywordPlan", pattern: ["customers/{customer_id}/keywordPlans/{keyword_plan_id}"] } });
    }
    create(value) {
        const message = { resourceName: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 5:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string name */ 6:
                    message.name = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.KeywordPlanForecastPeriod forecast_period */ 4:
                    message.forecastPeriod = exports.KeywordPlanForecastPeriod.internalBinaryRead(reader, reader.uint32(), options, message.forecastPeriod);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 5; */
        if (message.id !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.id);
        /* optional string name = 6; */
        if (message.name !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.name);
        /* google.ads.googleads.v11.resources.KeywordPlanForecastPeriod forecast_period = 4; */
        if (message.forecastPeriod)
            exports.KeywordPlanForecastPeriod.internalBinaryWrite(message.forecastPeriod, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.KeywordPlan
 */
exports.KeywordPlan = new KeywordPlan$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanForecastPeriod$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.KeywordPlanForecastPeriod", [
            { no: 1, name: "date_interval", kind: "enum", oneof: "interval", T: () => ["google.ads.googleads.v11.enums.KeywordPlanForecastIntervalEnum.KeywordPlanForecastInterval", keyword_plan_forecast_interval_1.KeywordPlanForecastIntervalEnum_KeywordPlanForecastInterval] },
            { no: 2, name: "date_range", kind: "message", oneof: "interval", T: () => dates_1.DateRange }
        ]);
    }
    create(value) {
        const message = { interval: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.KeywordPlanForecastIntervalEnum.KeywordPlanForecastInterval date_interval */ 1:
                    message.interval = {
                        oneofKind: "dateInterval",
                        dateInterval: reader.int32()
                    };
                    break;
                case /* google.ads.googleads.v11.common.DateRange date_range */ 2:
                    message.interval = {
                        oneofKind: "dateRange",
                        dateRange: dates_1.DateRange.internalBinaryRead(reader, reader.uint32(), options, message.interval.dateRange)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.KeywordPlanForecastIntervalEnum.KeywordPlanForecastInterval date_interval = 1; */
        if (message.interval.oneofKind === "dateInterval")
            writer.tag(1, runtime_1.WireType.Varint).int32(message.interval.dateInterval);
        /* google.ads.googleads.v11.common.DateRange date_range = 2; */
        if (message.interval.oneofKind === "dateRange")
            dates_1.DateRange.internalBinaryWrite(message.interval.dateRange, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.KeywordPlanForecastPeriod
 */
exports.KeywordPlanForecastPeriod = new KeywordPlanForecastPeriod$Type();
