// google-ads-api-client/dist/protos/bindings/ads/resources/keyword_plan_campaign.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeywordPlanGeoTarget = exports.KeywordPlanCampaign = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const keyword_plan_network_1 = require("../enums/keyword_plan_network");
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanCampaign$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.KeywordPlanCampaign", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlanCampaign" } } },
            { no: 9, name: "keyword_plan", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/KeywordPlan" } } },
            { no: 10, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 12, name: "language_constants", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/LanguageConstant" } } },
            { no: 6, name: "keyword_plan_network", kind: "enum", T: () => ["google.ads.googleads.v11.enums.KeywordPlanNetworkEnum.KeywordPlanNetwork", keyword_plan_network_1.KeywordPlanNetworkEnum_KeywordPlanNetwork] },
            { no: 13, name: "cpc_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 8, name: "geo_targets", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.KeywordPlanGeoTarget }
        ], { "google.api.resource": { type: "googleads.googleapis.com/KeywordPlanCampaign", pattern: ["customers/{customer_id}/keywordPlanCampaigns/{keyword_plan_campaign_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", languageConstants: [], keywordPlanNetwork: 0, geoTargets: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional string keyword_plan */ 9:
                    message.keywordPlan = reader.string();
                    break;
                case /* optional int64 id */ 10:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional string name */ 11:
                    message.name = reader.string();
                    break;
                case /* repeated string language_constants */ 12:
                    message.languageConstants.push(reader.string());
                    break;
                case /* google.ads.googleads.v11.enums.KeywordPlanNetworkEnum.KeywordPlanNetwork keyword_plan_network */ 6:
                    message.keywordPlanNetwork = reader.int32();
                    break;
                case /* optional int64 cpc_bid_micros */ 13:
                    message.cpcBidMicros = reader.int64().toBigInt();
                    break;
                case /* repeated google.ads.googleads.v11.resources.KeywordPlanGeoTarget geo_targets */ 8:
                    message.geoTargets.push(exports.KeywordPlanGeoTarget.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional string keyword_plan = 9; */
        if (message.keywordPlan !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.keywordPlan);
        /* optional int64 id = 10; */
        if (message.id !== undefined)
            writer.tag(10, runtime_1.WireType.Varint).int64(message.id);
        /* optional string name = 11; */
        if (message.name !== undefined)
            writer.tag(11, runtime_1.WireType.LengthDelimited).string(message.name);
        /* repeated string language_constants = 12; */
        for (let i = 0; i < message.languageConstants.length; i++)
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.languageConstants[i]);
        /* google.ads.googleads.v11.enums.KeywordPlanNetworkEnum.KeywordPlanNetwork keyword_plan_network = 6; */
        if (message.keywordPlanNetwork !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.keywordPlanNetwork);
        /* optional int64 cpc_bid_micros = 13; */
        if (message.cpcBidMicros !== undefined)
            writer.tag(13, runtime_1.WireType.Varint).int64(message.cpcBidMicros);
        /* repeated google.ads.googleads.v11.resources.KeywordPlanGeoTarget geo_targets = 8; */
        for (let i = 0; i < message.geoTargets.length; i++)
            exports.KeywordPlanGeoTarget.internalBinaryWrite(message.geoTargets[i], writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.KeywordPlanCampaign
 */
exports.KeywordPlanCampaign = new KeywordPlanCampaign$Type();
// @generated message type with reflection information, may provide speed optimized methods
class KeywordPlanGeoTarget$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.KeywordPlanGeoTarget", [
            { no: 2, name: "geo_target_constant", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.resource_reference": { type: "googleads.googleapis.com/GeoTargetConstant" } } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string geo_target_constant */ 2:
                    message.geoTargetConstant = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string geo_target_constant = 2; */
        if (message.geoTargetConstant !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.geoTargetConstant);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.KeywordPlanGeoTarget
 */
exports.KeywordPlanGeoTarget = new KeywordPlanGeoTarget$Type();
