// google-ads-api-client/dist/protos/bindings/ads/resources/lead_form_submission_data.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LeadFormSubmissionField = exports.LeadFormSubmissionData = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const lead_form_field_user_input_type_1 = require("../enums/lead_form_field_user_input_type");
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormSubmissionData$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.LeadFormSubmissionData", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/LeadFormSubmissionData" } } },
            { no: 2, name: "id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "asset", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Asset" } } },
            { no: 4, name: "campaign", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 5, name: "lead_form_submission_fields", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.LeadFormSubmissionField, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "ad_group", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } },
            { no: 7, name: "ad_group_ad", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroupAd" } } },
            { no: 8, name: "gclid", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "submission_date_time", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/LeadFormSubmissionData", pattern: ["customers/{customer_id}/leadFormSubmissionData/{lead_form_user_submission_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", id: "", asset: "", campaign: "", leadFormSubmissionFields: [], adGroup: "", adGroupAd: "", gclid: "", submissionDateTime: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* string id */ 2:
                    message.id = reader.string();
                    break;
                case /* string asset */ 3:
                    message.asset = reader.string();
                    break;
                case /* string campaign */ 4:
                    message.campaign = reader.string();
                    break;
                case /* repeated google.ads.googleads.v11.resources.LeadFormSubmissionField lead_form_submission_fields */ 5:
                    message.leadFormSubmissionFields.push(exports.LeadFormSubmissionField.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* string ad_group */ 6:
                    message.adGroup = reader.string();
                    break;
                case /* string ad_group_ad */ 7:
                    message.adGroupAd = reader.string();
                    break;
                case /* string gclid */ 8:
                    message.gclid = reader.string();
                    break;
                case /* string submission_date_time */ 9:
                    message.submissionDateTime = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* string id = 2; */
        if (message.id !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.id);
        /* string asset = 3; */
        if (message.asset !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.asset);
        /* string campaign = 4; */
        if (message.campaign !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.campaign);
        /* repeated google.ads.googleads.v11.resources.LeadFormSubmissionField lead_form_submission_fields = 5; */
        for (let i = 0; i < message.leadFormSubmissionFields.length; i++)
            exports.LeadFormSubmissionField.internalBinaryWrite(message.leadFormSubmissionFields[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string ad_group = 6; */
        if (message.adGroup !== "")
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.adGroup);
        /* string ad_group_ad = 7; */
        if (message.adGroupAd !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.adGroupAd);
        /* string gclid = 8; */
        if (message.gclid !== "")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.gclid);
        /* string submission_date_time = 9; */
        if (message.submissionDateTime !== "")
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.submissionDateTime);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.LeadFormSubmissionData
 */
exports.LeadFormSubmissionData = new LeadFormSubmissionData$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LeadFormSubmissionField$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.LeadFormSubmissionField", [
            { no: 1, name: "field_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum.LeadFormFieldUserInputType", lead_form_field_user_input_type_1.LeadFormFieldUserInputTypeEnum_LeadFormFieldUserInputType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "field_value", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { fieldType: 0, fieldValue: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum.LeadFormFieldUserInputType field_type */ 1:
                    message.fieldType = reader.int32();
                    break;
                case /* string field_value */ 2:
                    message.fieldValue = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.LeadFormFieldUserInputTypeEnum.LeadFormFieldUserInputType field_type = 1; */
        if (message.fieldType !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.fieldType);
        /* string field_value = 2; */
        if (message.fieldValue !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.fieldValue);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.LeadFormSubmissionField
 */
exports.LeadFormSubmissionField = new LeadFormSubmissionField$Type();
