// google-ads-api-client/dist/protos/bindings/ads/resources/offline_user_data_job.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OfflineUserDataJobMetadata = exports.OfflineUserDataJob = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const offline_user_data_job_match_rate_range_1 = require("../enums/offline_user_data_job_match_rate_range");
const offline_user_data_1 = require("../common/offline_user_data");
const offline_user_data_2 = require("../common/offline_user_data");
const offline_user_data_job_failure_reason_1 = require("../enums/offline_user_data_job_failure_reason");
const offline_user_data_job_status_1 = require("../enums/offline_user_data_job_status");
const offline_user_data_job_type_1 = require("../enums/offline_user_data_job_type");
// @generated message type with reflection information, may provide speed optimized methods
class OfflineUserDataJob$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.OfflineUserDataJob", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/OfflineUserDataJob" } } },
            { no: 9, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "external_id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 4, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.OfflineUserDataJobTypeEnum.OfflineUserDataJobType", offline_user_data_job_type_1.OfflineUserDataJobTypeEnum_OfflineUserDataJobType], options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 5, name: "status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.OfflineUserDataJobStatusEnum.OfflineUserDataJobStatus", offline_user_data_job_status_1.OfflineUserDataJobStatusEnum_OfflineUserDataJobStatus], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "failure_reason", kind: "enum", T: () => ["google.ads.googleads.v11.enums.OfflineUserDataJobFailureReasonEnum.OfflineUserDataJobFailureReason", offline_user_data_job_failure_reason_1.OfflineUserDataJobFailureReasonEnum_OfflineUserDataJobFailureReason], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "operation_metadata", kind: "message", T: () => exports.OfflineUserDataJobMetadata, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "customer_match_user_list_metadata", kind: "message", oneof: "metadata", T: () => offline_user_data_2.CustomerMatchUserListMetadata, options: { "google.api.field_behavior": ["IMMUTABLE"] } },
            { no: 8, name: "store_sales_metadata", kind: "message", oneof: "metadata", T: () => offline_user_data_1.StoreSalesMetadata, options: { "google.api.field_behavior": ["IMMUTABLE"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/OfflineUserDataJob", pattern: ["customers/{customer_id}/offlineUserDataJobs/{offline_user_data_update_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", type: 0, status: 0, failureReason: 0, metadata: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 9:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional int64 external_id */ 10:
                    message.externalId = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.OfflineUserDataJobTypeEnum.OfflineUserDataJobType type */ 4:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.OfflineUserDataJobStatusEnum.OfflineUserDataJobStatus status */ 5:
                    message.status = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.OfflineUserDataJobFailureReasonEnum.OfflineUserDataJobFailureReason failure_reason */ 6:
                    message.failureReason = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.OfflineUserDataJobMetadata operation_metadata */ 11:
                    message.operationMetadata = exports.OfflineUserDataJobMetadata.internalBinaryRead(reader, reader.uint32(), options, message.operationMetadata);
                    break;
                case /* google.ads.googleads.v11.common.CustomerMatchUserListMetadata customer_match_user_list_metadata */ 7:
                    message.metadata = {
                        oneofKind: "customerMatchUserListMetadata",
                        customerMatchUserListMetadata: offline_user_data_2.CustomerMatchUserListMetadata.internalBinaryRead(reader, reader.uint32(), options, message.metadata.customerMatchUserListMetadata)
                    };
                    break;
                case /* google.ads.googleads.v11.common.StoreSalesMetadata store_sales_metadata */ 8:
                    message.metadata = {
                        oneofKind: "storeSalesMetadata",
                        storeSalesMetadata: offline_user_data_1.StoreSalesMetadata.internalBinaryRead(reader, reader.uint32(), options, message.metadata.storeSalesMetadata)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 9; */
        if (message.id !== undefined)
            writer.tag(9, runtime_1.WireType.Varint).int64(message.id);
        /* optional int64 external_id = 10; */
        if (message.externalId !== undefined)
            writer.tag(10, runtime_1.WireType.Varint).int64(message.externalId);
        /* google.ads.googleads.v11.enums.OfflineUserDataJobTypeEnum.OfflineUserDataJobType type = 4; */
        if (message.type !== 0)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.OfflineUserDataJobStatusEnum.OfflineUserDataJobStatus status = 5; */
        if (message.status !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.status);
        /* google.ads.googleads.v11.enums.OfflineUserDataJobFailureReasonEnum.OfflineUserDataJobFailureReason failure_reason = 6; */
        if (message.failureReason !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.failureReason);
        /* google.ads.googleads.v11.resources.OfflineUserDataJobMetadata operation_metadata = 11; */
        if (message.operationMetadata)
            exports.OfflineUserDataJobMetadata.internalBinaryWrite(message.operationMetadata, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.CustomerMatchUserListMetadata customer_match_user_list_metadata = 7; */
        if (message.metadata.oneofKind === "customerMatchUserListMetadata")
            offline_user_data_2.CustomerMatchUserListMetadata.internalBinaryWrite(message.metadata.customerMatchUserListMetadata, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.StoreSalesMetadata store_sales_metadata = 8; */
        if (message.metadata.oneofKind === "storeSalesMetadata")
            offline_user_data_1.StoreSalesMetadata.internalBinaryWrite(message.metadata.storeSalesMetadata, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.OfflineUserDataJob
 */
exports.OfflineUserDataJob = new OfflineUserDataJob$Type();
// @generated message type with reflection information, may provide speed optimized methods
class OfflineUserDataJobMetadata$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.OfflineUserDataJobMetadata", [
            { no: 1, name: "match_rate_range", kind: "enum", T: () => ["google.ads.googleads.v11.enums.OfflineUserDataJobMatchRateRangeEnum.OfflineUserDataJobMatchRateRange", offline_user_data_job_match_rate_range_1.OfflineUserDataJobMatchRateRangeEnum_OfflineUserDataJobMatchRateRange], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { matchRateRange: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.OfflineUserDataJobMatchRateRangeEnum.OfflineUserDataJobMatchRateRange match_rate_range */ 1:
                    message.matchRateRange = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.OfflineUserDataJobMatchRateRangeEnum.OfflineUserDataJobMatchRateRange match_rate_range = 1; */
        if (message.matchRateRange !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.matchRateRange);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.OfflineUserDataJobMetadata
 */
exports.OfflineUserDataJobMetadata = new OfflineUserDataJobMetadata$Type();
