// google-ads-api-client/dist/protos/bindings/ads/resources/recommendation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Recommendation_UpgradeLocalCampaignToPerformanceMaxRecommendation = exports.Recommendation_DisplayExpansionOptInRecommendation = exports.Recommendation_UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation = exports.Recommendation_UseBroadMatchKeywordRecommendation = exports.Recommendation_ResponsiveSearchAdRecommendation = exports.Recommendation_ResponsiveSearchAdImproveAdStrengthRecommendation = exports.Recommendation_ResponsiveSearchAdAssetRecommendation = exports.Recommendation_TargetRoasOptInRecommendation = exports.Recommendation_MoveUnusedBudgetRecommendation = exports.Recommendation_KeywordMatchTypeRecommendation = exports.Recommendation_CallExtensionRecommendation = exports.Recommendation_SitelinkExtensionRecommendation = exports.Recommendation_CalloutExtensionRecommendation = exports.Recommendation_OptimizeAdRotationRecommendation = exports.Recommendation_MaximizeClicksOptInRecommendation = exports.Recommendation_SearchPartnersOptInRecommendation = exports.Recommendation_EnhancedCpcOptInRecommendation = exports.Recommendation_MaximizeConversionsOptInRecommendation = exports.Recommendation_TargetCpaOptInRecommendation_TargetCpaOptInRecommendationOption = exports.Recommendation_TargetCpaOptInRecommendation = exports.Recommendation_TextAdRecommendation = exports.Recommendation_KeywordRecommendation = exports.Recommendation_CampaignBudgetRecommendation_CampaignBudgetRecommendationOption = exports.Recommendation_CampaignBudgetRecommendation = exports.Recommendation_RecommendationMetrics = exports.Recommendation_RecommendationImpact = exports.Recommendation = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const keyword_match_type_1 = require("../enums/keyword_match_type");
const extensions_1 = require("../common/extensions");
const extensions_2 = require("../common/extensions");
const extensions_3 = require("../common/extensions");
const target_cpa_opt_in_recommendation_goal_1 = require("../enums/target_cpa_opt_in_recommendation_goal");
const ad_1 = require("./ad");
const criteria_1 = require("../common/criteria");
const recommendation_type_1 = require("../enums/recommendation_type");
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/Recommendation" } } },
            { no: 2, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.RecommendationTypeEnum.RecommendationType", recommendation_type_1.RecommendationTypeEnum_RecommendationType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "impact", kind: "message", T: () => exports.Recommendation_RecommendationImpact, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 24, name: "campaign_budget", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/CampaignBudget" } } },
            { no: 25, name: "campaign", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 26, name: "ad_group", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/AdGroup" } } },
            { no: 27, name: "dismissed", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "campaign_budget_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_CampaignBudgetRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 22, name: "forecasting_campaign_budget_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_CampaignBudgetRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "keyword_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_KeywordRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "text_ad_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_TextAdRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "target_cpa_opt_in_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_TargetCpaOptInRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 11, name: "maximize_conversions_opt_in_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_MaximizeConversionsOptInRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 12, name: "enhanced_cpc_opt_in_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_EnhancedCpcOptInRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 14, name: "search_partners_opt_in_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_SearchPartnersOptInRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 15, name: "maximize_clicks_opt_in_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_MaximizeClicksOptInRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 16, name: "optimize_ad_rotation_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_OptimizeAdRotationRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 17, name: "callout_extension_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_CalloutExtensionRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 18, name: "sitelink_extension_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_SitelinkExtensionRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 19, name: "call_extension_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_CallExtensionRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 20, name: "keyword_match_type_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_KeywordMatchTypeRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 21, name: "move_unused_budget_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_MoveUnusedBudgetRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 23, name: "target_roas_opt_in_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_TargetRoasOptInRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 28, name: "responsive_search_ad_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_ResponsiveSearchAdRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 29, name: "marginal_roi_campaign_budget_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_CampaignBudgetRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 30, name: "use_broad_match_keyword_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_UseBroadMatchKeywordRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 31, name: "responsive_search_ad_asset_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_ResponsiveSearchAdAssetRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 32, name: "upgrade_smart_shopping_campaign_to_performance_max_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 33, name: "responsive_search_ad_improve_ad_strength_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_ResponsiveSearchAdImproveAdStrengthRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 34, name: "display_expansion_opt_in_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_DisplayExpansionOptInRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 35, name: "upgrade_local_campaign_to_performance_max_recommendation", kind: "message", oneof: "recommendation", T: () => exports.Recommendation_UpgradeLocalCampaignToPerformanceMaxRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ], { "google.api.resource": { type: "googleads.googleapis.com/Recommendation", pattern: ["customers/{customer_id}/recommendations/{recommendation_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", type: 0, recommendation: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.RecommendationTypeEnum.RecommendationType type */ 2:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.RecommendationImpact impact */ 3:
                    message.impact = exports.Recommendation_RecommendationImpact.internalBinaryRead(reader, reader.uint32(), options, message.impact);
                    break;
                case /* optional string campaign_budget */ 24:
                    message.campaignBudget = reader.string();
                    break;
                case /* optional string campaign */ 25:
                    message.campaign = reader.string();
                    break;
                case /* optional string ad_group */ 26:
                    message.adGroup = reader.string();
                    break;
                case /* optional bool dismissed */ 27:
                    message.dismissed = reader.bool();
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation campaign_budget_recommendation */ 4:
                    message.recommendation = {
                        oneofKind: "campaignBudgetRecommendation",
                        campaignBudgetRecommendation: exports.Recommendation_CampaignBudgetRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.campaignBudgetRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation forecasting_campaign_budget_recommendation */ 22:
                    message.recommendation = {
                        oneofKind: "forecastingCampaignBudgetRecommendation",
                        forecastingCampaignBudgetRecommendation: exports.Recommendation_CampaignBudgetRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.forecastingCampaignBudgetRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.KeywordRecommendation keyword_recommendation */ 8:
                    message.recommendation = {
                        oneofKind: "keywordRecommendation",
                        keywordRecommendation: exports.Recommendation_KeywordRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.keywordRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.TextAdRecommendation text_ad_recommendation */ 9:
                    message.recommendation = {
                        oneofKind: "textAdRecommendation",
                        textAdRecommendation: exports.Recommendation_TextAdRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.textAdRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.TargetCpaOptInRecommendation target_cpa_opt_in_recommendation */ 10:
                    message.recommendation = {
                        oneofKind: "targetCpaOptInRecommendation",
                        targetCpaOptInRecommendation: exports.Recommendation_TargetCpaOptInRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.targetCpaOptInRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.MaximizeConversionsOptInRecommendation maximize_conversions_opt_in_recommendation */ 11:
                    message.recommendation = {
                        oneofKind: "maximizeConversionsOptInRecommendation",
                        maximizeConversionsOptInRecommendation: exports.Recommendation_MaximizeConversionsOptInRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.maximizeConversionsOptInRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.EnhancedCpcOptInRecommendation enhanced_cpc_opt_in_recommendation */ 12:
                    message.recommendation = {
                        oneofKind: "enhancedCpcOptInRecommendation",
                        enhancedCpcOptInRecommendation: exports.Recommendation_EnhancedCpcOptInRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.enhancedCpcOptInRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.SearchPartnersOptInRecommendation search_partners_opt_in_recommendation */ 14:
                    message.recommendation = {
                        oneofKind: "searchPartnersOptInRecommendation",
                        searchPartnersOptInRecommendation: exports.Recommendation_SearchPartnersOptInRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.searchPartnersOptInRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.MaximizeClicksOptInRecommendation maximize_clicks_opt_in_recommendation */ 15:
                    message.recommendation = {
                        oneofKind: "maximizeClicksOptInRecommendation",
                        maximizeClicksOptInRecommendation: exports.Recommendation_MaximizeClicksOptInRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.maximizeClicksOptInRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.OptimizeAdRotationRecommendation optimize_ad_rotation_recommendation */ 16:
                    message.recommendation = {
                        oneofKind: "optimizeAdRotationRecommendation",
                        optimizeAdRotationRecommendation: exports.Recommendation_OptimizeAdRotationRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.optimizeAdRotationRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.CalloutExtensionRecommendation callout_extension_recommendation */ 17:
                    message.recommendation = {
                        oneofKind: "calloutExtensionRecommendation",
                        calloutExtensionRecommendation: exports.Recommendation_CalloutExtensionRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.calloutExtensionRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.SitelinkExtensionRecommendation sitelink_extension_recommendation */ 18:
                    message.recommendation = {
                        oneofKind: "sitelinkExtensionRecommendation",
                        sitelinkExtensionRecommendation: exports.Recommendation_SitelinkExtensionRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.sitelinkExtensionRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.CallExtensionRecommendation call_extension_recommendation */ 19:
                    message.recommendation = {
                        oneofKind: "callExtensionRecommendation",
                        callExtensionRecommendation: exports.Recommendation_CallExtensionRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.callExtensionRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.KeywordMatchTypeRecommendation keyword_match_type_recommendation */ 20:
                    message.recommendation = {
                        oneofKind: "keywordMatchTypeRecommendation",
                        keywordMatchTypeRecommendation: exports.Recommendation_KeywordMatchTypeRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.keywordMatchTypeRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.MoveUnusedBudgetRecommendation move_unused_budget_recommendation */ 21:
                    message.recommendation = {
                        oneofKind: "moveUnusedBudgetRecommendation",
                        moveUnusedBudgetRecommendation: exports.Recommendation_MoveUnusedBudgetRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.moveUnusedBudgetRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.TargetRoasOptInRecommendation target_roas_opt_in_recommendation */ 23:
                    message.recommendation = {
                        oneofKind: "targetRoasOptInRecommendation",
                        targetRoasOptInRecommendation: exports.Recommendation_TargetRoasOptInRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.targetRoasOptInRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdRecommendation responsive_search_ad_recommendation */ 28:
                    message.recommendation = {
                        oneofKind: "responsiveSearchAdRecommendation",
                        responsiveSearchAdRecommendation: exports.Recommendation_ResponsiveSearchAdRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.responsiveSearchAdRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation marginal_roi_campaign_budget_recommendation */ 29:
                    message.recommendation = {
                        oneofKind: "marginalRoiCampaignBudgetRecommendation",
                        marginalRoiCampaignBudgetRecommendation: exports.Recommendation_CampaignBudgetRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.marginalRoiCampaignBudgetRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.UseBroadMatchKeywordRecommendation use_broad_match_keyword_recommendation */ 30:
                    message.recommendation = {
                        oneofKind: "useBroadMatchKeywordRecommendation",
                        useBroadMatchKeywordRecommendation: exports.Recommendation_UseBroadMatchKeywordRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.useBroadMatchKeywordRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdAssetRecommendation responsive_search_ad_asset_recommendation */ 31:
                    message.recommendation = {
                        oneofKind: "responsiveSearchAdAssetRecommendation",
                        responsiveSearchAdAssetRecommendation: exports.Recommendation_ResponsiveSearchAdAssetRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.responsiveSearchAdAssetRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation upgrade_smart_shopping_campaign_to_performance_max_recommendation */ 32:
                    message.recommendation = {
                        oneofKind: "upgradeSmartShoppingCampaignToPerformanceMaxRecommendation",
                        upgradeSmartShoppingCampaignToPerformanceMaxRecommendation: exports.Recommendation_UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.upgradeSmartShoppingCampaignToPerformanceMaxRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdImproveAdStrengthRecommendation responsive_search_ad_improve_ad_strength_recommendation */ 33:
                    message.recommendation = {
                        oneofKind: "responsiveSearchAdImproveAdStrengthRecommendation",
                        responsiveSearchAdImproveAdStrengthRecommendation: exports.Recommendation_ResponsiveSearchAdImproveAdStrengthRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.responsiveSearchAdImproveAdStrengthRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.DisplayExpansionOptInRecommendation display_expansion_opt_in_recommendation */ 34:
                    message.recommendation = {
                        oneofKind: "displayExpansionOptInRecommendation",
                        displayExpansionOptInRecommendation: exports.Recommendation_DisplayExpansionOptInRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.displayExpansionOptInRecommendation)
                    };
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.UpgradeLocalCampaignToPerformanceMaxRecommendation upgrade_local_campaign_to_performance_max_recommendation */ 35:
                    message.recommendation = {
                        oneofKind: "upgradeLocalCampaignToPerformanceMaxRecommendation",
                        upgradeLocalCampaignToPerformanceMaxRecommendation: exports.Recommendation_UpgradeLocalCampaignToPerformanceMaxRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.recommendation.upgradeLocalCampaignToPerformanceMaxRecommendation)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* google.ads.googleads.v11.enums.RecommendationTypeEnum.RecommendationType type = 2; */
        if (message.type !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.resources.Recommendation.RecommendationImpact impact = 3; */
        if (message.impact)
            exports.Recommendation_RecommendationImpact.internalBinaryWrite(message.impact, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string campaign_budget = 24; */
        if (message.campaignBudget !== undefined)
            writer.tag(24, runtime_1.WireType.LengthDelimited).string(message.campaignBudget);
        /* optional string campaign = 25; */
        if (message.campaign !== undefined)
            writer.tag(25, runtime_1.WireType.LengthDelimited).string(message.campaign);
        /* optional string ad_group = 26; */
        if (message.adGroup !== undefined)
            writer.tag(26, runtime_1.WireType.LengthDelimited).string(message.adGroup);
        /* optional bool dismissed = 27; */
        if (message.dismissed !== undefined)
            writer.tag(27, runtime_1.WireType.Varint).bool(message.dismissed);
        /* google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation campaign_budget_recommendation = 4; */
        if (message.recommendation.oneofKind === "campaignBudgetRecommendation")
            exports.Recommendation_CampaignBudgetRecommendation.internalBinaryWrite(message.recommendation.campaignBudgetRecommendation, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation forecasting_campaign_budget_recommendation = 22; */
        if (message.recommendation.oneofKind === "forecastingCampaignBudgetRecommendation")
            exports.Recommendation_CampaignBudgetRecommendation.internalBinaryWrite(message.recommendation.forecastingCampaignBudgetRecommendation, writer.tag(22, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.KeywordRecommendation keyword_recommendation = 8; */
        if (message.recommendation.oneofKind === "keywordRecommendation")
            exports.Recommendation_KeywordRecommendation.internalBinaryWrite(message.recommendation.keywordRecommendation, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.TextAdRecommendation text_ad_recommendation = 9; */
        if (message.recommendation.oneofKind === "textAdRecommendation")
            exports.Recommendation_TextAdRecommendation.internalBinaryWrite(message.recommendation.textAdRecommendation, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.TargetCpaOptInRecommendation target_cpa_opt_in_recommendation = 10; */
        if (message.recommendation.oneofKind === "targetCpaOptInRecommendation")
            exports.Recommendation_TargetCpaOptInRecommendation.internalBinaryWrite(message.recommendation.targetCpaOptInRecommendation, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.MaximizeConversionsOptInRecommendation maximize_conversions_opt_in_recommendation = 11; */
        if (message.recommendation.oneofKind === "maximizeConversionsOptInRecommendation")
            exports.Recommendation_MaximizeConversionsOptInRecommendation.internalBinaryWrite(message.recommendation.maximizeConversionsOptInRecommendation, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.EnhancedCpcOptInRecommendation enhanced_cpc_opt_in_recommendation = 12; */
        if (message.recommendation.oneofKind === "enhancedCpcOptInRecommendation")
            exports.Recommendation_EnhancedCpcOptInRecommendation.internalBinaryWrite(message.recommendation.enhancedCpcOptInRecommendation, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.SearchPartnersOptInRecommendation search_partners_opt_in_recommendation = 14; */
        if (message.recommendation.oneofKind === "searchPartnersOptInRecommendation")
            exports.Recommendation_SearchPartnersOptInRecommendation.internalBinaryWrite(message.recommendation.searchPartnersOptInRecommendation, writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.MaximizeClicksOptInRecommendation maximize_clicks_opt_in_recommendation = 15; */
        if (message.recommendation.oneofKind === "maximizeClicksOptInRecommendation")
            exports.Recommendation_MaximizeClicksOptInRecommendation.internalBinaryWrite(message.recommendation.maximizeClicksOptInRecommendation, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.OptimizeAdRotationRecommendation optimize_ad_rotation_recommendation = 16; */
        if (message.recommendation.oneofKind === "optimizeAdRotationRecommendation")
            exports.Recommendation_OptimizeAdRotationRecommendation.internalBinaryWrite(message.recommendation.optimizeAdRotationRecommendation, writer.tag(16, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.CalloutExtensionRecommendation callout_extension_recommendation = 17; */
        if (message.recommendation.oneofKind === "calloutExtensionRecommendation")
            exports.Recommendation_CalloutExtensionRecommendation.internalBinaryWrite(message.recommendation.calloutExtensionRecommendation, writer.tag(17, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.SitelinkExtensionRecommendation sitelink_extension_recommendation = 18; */
        if (message.recommendation.oneofKind === "sitelinkExtensionRecommendation")
            exports.Recommendation_SitelinkExtensionRecommendation.internalBinaryWrite(message.recommendation.sitelinkExtensionRecommendation, writer.tag(18, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.CallExtensionRecommendation call_extension_recommendation = 19; */
        if (message.recommendation.oneofKind === "callExtensionRecommendation")
            exports.Recommendation_CallExtensionRecommendation.internalBinaryWrite(message.recommendation.callExtensionRecommendation, writer.tag(19, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.KeywordMatchTypeRecommendation keyword_match_type_recommendation = 20; */
        if (message.recommendation.oneofKind === "keywordMatchTypeRecommendation")
            exports.Recommendation_KeywordMatchTypeRecommendation.internalBinaryWrite(message.recommendation.keywordMatchTypeRecommendation, writer.tag(20, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.MoveUnusedBudgetRecommendation move_unused_budget_recommendation = 21; */
        if (message.recommendation.oneofKind === "moveUnusedBudgetRecommendation")
            exports.Recommendation_MoveUnusedBudgetRecommendation.internalBinaryWrite(message.recommendation.moveUnusedBudgetRecommendation, writer.tag(21, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.TargetRoasOptInRecommendation target_roas_opt_in_recommendation = 23; */
        if (message.recommendation.oneofKind === "targetRoasOptInRecommendation")
            exports.Recommendation_TargetRoasOptInRecommendation.internalBinaryWrite(message.recommendation.targetRoasOptInRecommendation, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdRecommendation responsive_search_ad_recommendation = 28; */
        if (message.recommendation.oneofKind === "responsiveSearchAdRecommendation")
            exports.Recommendation_ResponsiveSearchAdRecommendation.internalBinaryWrite(message.recommendation.responsiveSearchAdRecommendation, writer.tag(28, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation marginal_roi_campaign_budget_recommendation = 29; */
        if (message.recommendation.oneofKind === "marginalRoiCampaignBudgetRecommendation")
            exports.Recommendation_CampaignBudgetRecommendation.internalBinaryWrite(message.recommendation.marginalRoiCampaignBudgetRecommendation, writer.tag(29, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.UseBroadMatchKeywordRecommendation use_broad_match_keyword_recommendation = 30; */
        if (message.recommendation.oneofKind === "useBroadMatchKeywordRecommendation")
            exports.Recommendation_UseBroadMatchKeywordRecommendation.internalBinaryWrite(message.recommendation.useBroadMatchKeywordRecommendation, writer.tag(30, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdAssetRecommendation responsive_search_ad_asset_recommendation = 31; */
        if (message.recommendation.oneofKind === "responsiveSearchAdAssetRecommendation")
            exports.Recommendation_ResponsiveSearchAdAssetRecommendation.internalBinaryWrite(message.recommendation.responsiveSearchAdAssetRecommendation, writer.tag(31, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation upgrade_smart_shopping_campaign_to_performance_max_recommendation = 32; */
        if (message.recommendation.oneofKind === "upgradeSmartShoppingCampaignToPerformanceMaxRecommendation")
            exports.Recommendation_UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation.internalBinaryWrite(message.recommendation.upgradeSmartShoppingCampaignToPerformanceMaxRecommendation, writer.tag(32, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdImproveAdStrengthRecommendation responsive_search_ad_improve_ad_strength_recommendation = 33; */
        if (message.recommendation.oneofKind === "responsiveSearchAdImproveAdStrengthRecommendation")
            exports.Recommendation_ResponsiveSearchAdImproveAdStrengthRecommendation.internalBinaryWrite(message.recommendation.responsiveSearchAdImproveAdStrengthRecommendation, writer.tag(33, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.DisplayExpansionOptInRecommendation display_expansion_opt_in_recommendation = 34; */
        if (message.recommendation.oneofKind === "displayExpansionOptInRecommendation")
            exports.Recommendation_DisplayExpansionOptInRecommendation.internalBinaryWrite(message.recommendation.displayExpansionOptInRecommendation, writer.tag(34, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.UpgradeLocalCampaignToPerformanceMaxRecommendation upgrade_local_campaign_to_performance_max_recommendation = 35; */
        if (message.recommendation.oneofKind === "upgradeLocalCampaignToPerformanceMaxRecommendation")
            exports.Recommendation_UpgradeLocalCampaignToPerformanceMaxRecommendation.internalBinaryWrite(message.recommendation.upgradeLocalCampaignToPerformanceMaxRecommendation, writer.tag(35, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation
 */
exports.Recommendation = new Recommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_RecommendationImpact$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.RecommendationImpact", [
            { no: 1, name: "base_metrics", kind: "message", T: () => exports.Recommendation_RecommendationMetrics, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "potential_metrics", kind: "message", T: () => exports.Recommendation_RecommendationMetrics, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Recommendation.RecommendationMetrics base_metrics */ 1:
                    message.baseMetrics = exports.Recommendation_RecommendationMetrics.internalBinaryRead(reader, reader.uint32(), options, message.baseMetrics);
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.RecommendationMetrics potential_metrics */ 2:
                    message.potentialMetrics = exports.Recommendation_RecommendationMetrics.internalBinaryRead(reader, reader.uint32(), options, message.potentialMetrics);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Recommendation.RecommendationMetrics base_metrics = 1; */
        if (message.baseMetrics)
            exports.Recommendation_RecommendationMetrics.internalBinaryWrite(message.baseMetrics, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Recommendation.RecommendationMetrics potential_metrics = 2; */
        if (message.potentialMetrics)
            exports.Recommendation_RecommendationMetrics.internalBinaryWrite(message.potentialMetrics, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.RecommendationImpact
 */
exports.Recommendation_RecommendationImpact = new Recommendation_RecommendationImpact$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_RecommendationMetrics$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.RecommendationMetrics", [
            { no: 6, name: "impressions", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 7, name: "clicks", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "cost_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 9, name: "conversions", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "video_views", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional double impressions */ 6:
                    message.impressions = reader.double();
                    break;
                case /* optional double clicks */ 7:
                    message.clicks = reader.double();
                    break;
                case /* optional int64 cost_micros */ 8:
                    message.costMicros = reader.int64().toBigInt();
                    break;
                case /* optional double conversions */ 9:
                    message.conversions = reader.double();
                    break;
                case /* optional double video_views */ 10:
                    message.videoViews = reader.double();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional double impressions = 6; */
        if (message.impressions !== undefined)
            writer.tag(6, runtime_1.WireType.Bit64).double(message.impressions);
        /* optional double clicks = 7; */
        if (message.clicks !== undefined)
            writer.tag(7, runtime_1.WireType.Bit64).double(message.clicks);
        /* optional int64 cost_micros = 8; */
        if (message.costMicros !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).int64(message.costMicros);
        /* optional double conversions = 9; */
        if (message.conversions !== undefined)
            writer.tag(9, runtime_1.WireType.Bit64).double(message.conversions);
        /* optional double video_views = 10; */
        if (message.videoViews !== undefined)
            writer.tag(10, runtime_1.WireType.Bit64).double(message.videoViews);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.RecommendationMetrics
 */
exports.Recommendation_RecommendationMetrics = new Recommendation_RecommendationMetrics$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_CampaignBudgetRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation", [
            { no: 7, name: "current_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 8, name: "recommended_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "budget_options", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.Recommendation_CampaignBudgetRecommendation_CampaignBudgetRecommendationOption, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { budgetOptions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 current_budget_amount_micros */ 7:
                    message.currentBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 recommended_budget_amount_micros */ 8:
                    message.recommendedBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                case /* repeated google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation.CampaignBudgetRecommendationOption budget_options */ 3:
                    message.budgetOptions.push(exports.Recommendation_CampaignBudgetRecommendation_CampaignBudgetRecommendationOption.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 current_budget_amount_micros = 7; */
        if (message.currentBudgetAmountMicros !== undefined)
            writer.tag(7, runtime_1.WireType.Varint).int64(message.currentBudgetAmountMicros);
        /* optional int64 recommended_budget_amount_micros = 8; */
        if (message.recommendedBudgetAmountMicros !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).int64(message.recommendedBudgetAmountMicros);
        /* repeated google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation.CampaignBudgetRecommendationOption budget_options = 3; */
        for (let i = 0; i < message.budgetOptions.length; i++)
            exports.Recommendation_CampaignBudgetRecommendation_CampaignBudgetRecommendationOption.internalBinaryWrite(message.budgetOptions[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation
 */
exports.Recommendation_CampaignBudgetRecommendation = new Recommendation_CampaignBudgetRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_CampaignBudgetRecommendation_CampaignBudgetRecommendationOption$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation.CampaignBudgetRecommendationOption", [
            { no: 3, name: "budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "impact", kind: "message", T: () => exports.Recommendation_RecommendationImpact, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 budget_amount_micros */ 3:
                    message.budgetAmountMicros = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.RecommendationImpact impact */ 2:
                    message.impact = exports.Recommendation_RecommendationImpact.internalBinaryRead(reader, reader.uint32(), options, message.impact);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 budget_amount_micros = 3; */
        if (message.budgetAmountMicros !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.budgetAmountMicros);
        /* google.ads.googleads.v11.resources.Recommendation.RecommendationImpact impact = 2; */
        if (message.impact)
            exports.Recommendation_RecommendationImpact.internalBinaryWrite(message.impact, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation.CampaignBudgetRecommendationOption
 */
exports.Recommendation_CampaignBudgetRecommendation_CampaignBudgetRecommendationOption = new Recommendation_CampaignBudgetRecommendation_CampaignBudgetRecommendationOption$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_KeywordRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.KeywordRecommendation", [
            { no: 1, name: "keyword", kind: "message", T: () => criteria_1.KeywordInfo, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "recommended_cpc_bid_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.KeywordInfo keyword */ 1:
                    message.keyword = criteria_1.KeywordInfo.internalBinaryRead(reader, reader.uint32(), options, message.keyword);
                    break;
                case /* optional int64 recommended_cpc_bid_micros */ 3:
                    message.recommendedCpcBidMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.KeywordInfo keyword = 1; */
        if (message.keyword)
            criteria_1.KeywordInfo.internalBinaryWrite(message.keyword, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional int64 recommended_cpc_bid_micros = 3; */
        if (message.recommendedCpcBidMicros !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.recommendedCpcBidMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.KeywordRecommendation
 */
exports.Recommendation_KeywordRecommendation = new Recommendation_KeywordRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_TextAdRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.TextAdRecommendation", [
            { no: 1, name: "ad", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "creation_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "auto_apply_date", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad ad */ 1:
                    message.ad = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.ad);
                    break;
                case /* optional string creation_date */ 4:
                    message.creationDate = reader.string();
                    break;
                case /* optional string auto_apply_date */ 5:
                    message.autoApplyDate = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad ad = 1; */
        if (message.ad)
            ad_1.Ad.internalBinaryWrite(message.ad, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string creation_date = 4; */
        if (message.creationDate !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.creationDate);
        /* optional string auto_apply_date = 5; */
        if (message.autoApplyDate !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.autoApplyDate);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.TextAdRecommendation
 */
exports.Recommendation_TextAdRecommendation = new Recommendation_TextAdRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_TargetCpaOptInRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.TargetCpaOptInRecommendation", [
            { no: 1, name: "options", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => exports.Recommendation_TargetCpaOptInRecommendation_TargetCpaOptInRecommendationOption, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "recommended_target_cpa_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { options: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.resources.Recommendation.TargetCpaOptInRecommendation.TargetCpaOptInRecommendationOption options */ 1:
                    message.options.push(exports.Recommendation_TargetCpaOptInRecommendation_TargetCpaOptInRecommendationOption.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional int64 recommended_target_cpa_micros */ 3:
                    message.recommendedTargetCpaMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.resources.Recommendation.TargetCpaOptInRecommendation.TargetCpaOptInRecommendationOption options = 1; */
        for (let i = 0; i < message.options.length; i++)
            exports.Recommendation_TargetCpaOptInRecommendation_TargetCpaOptInRecommendationOption.internalBinaryWrite(message.options[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional int64 recommended_target_cpa_micros = 3; */
        if (message.recommendedTargetCpaMicros !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.recommendedTargetCpaMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.TargetCpaOptInRecommendation
 */
exports.Recommendation_TargetCpaOptInRecommendation = new Recommendation_TargetCpaOptInRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_TargetCpaOptInRecommendation_TargetCpaOptInRecommendationOption$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.TargetCpaOptInRecommendation.TargetCpaOptInRecommendationOption", [
            { no: 1, name: "goal", kind: "enum", T: () => ["google.ads.googleads.v11.enums.TargetCpaOptInRecommendationGoalEnum.TargetCpaOptInRecommendationGoal", target_cpa_opt_in_recommendation_goal_1.TargetCpaOptInRecommendationGoalEnum_TargetCpaOptInRecommendationGoal], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "target_cpa_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 6, name: "required_campaign_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "impact", kind: "message", T: () => exports.Recommendation_RecommendationImpact, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { goal: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.enums.TargetCpaOptInRecommendationGoalEnum.TargetCpaOptInRecommendationGoal goal */ 1:
                    message.goal = reader.int32();
                    break;
                case /* optional int64 target_cpa_micros */ 5:
                    message.targetCpaMicros = reader.int64().toBigInt();
                    break;
                case /* optional int64 required_campaign_budget_amount_micros */ 6:
                    message.requiredCampaignBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.RecommendationImpact impact */ 4:
                    message.impact = exports.Recommendation_RecommendationImpact.internalBinaryRead(reader, reader.uint32(), options, message.impact);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.enums.TargetCpaOptInRecommendationGoalEnum.TargetCpaOptInRecommendationGoal goal = 1; */
        if (message.goal !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.goal);
        /* optional int64 target_cpa_micros = 5; */
        if (message.targetCpaMicros !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.targetCpaMicros);
        /* optional int64 required_campaign_budget_amount_micros = 6; */
        if (message.requiredCampaignBudgetAmountMicros !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).int64(message.requiredCampaignBudgetAmountMicros);
        /* google.ads.googleads.v11.resources.Recommendation.RecommendationImpact impact = 4; */
        if (message.impact)
            exports.Recommendation_RecommendationImpact.internalBinaryWrite(message.impact, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.TargetCpaOptInRecommendation.TargetCpaOptInRecommendationOption
 */
exports.Recommendation_TargetCpaOptInRecommendation_TargetCpaOptInRecommendationOption = new Recommendation_TargetCpaOptInRecommendation_TargetCpaOptInRecommendationOption$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_MaximizeConversionsOptInRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.MaximizeConversionsOptInRecommendation", [
            { no: 2, name: "recommended_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 recommended_budget_amount_micros */ 2:
                    message.recommendedBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 recommended_budget_amount_micros = 2; */
        if (message.recommendedBudgetAmountMicros !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.recommendedBudgetAmountMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.MaximizeConversionsOptInRecommendation
 */
exports.Recommendation_MaximizeConversionsOptInRecommendation = new Recommendation_MaximizeConversionsOptInRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_EnhancedCpcOptInRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.EnhancedCpcOptInRecommendation", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.EnhancedCpcOptInRecommendation
 */
exports.Recommendation_EnhancedCpcOptInRecommendation = new Recommendation_EnhancedCpcOptInRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_SearchPartnersOptInRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.SearchPartnersOptInRecommendation", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.SearchPartnersOptInRecommendation
 */
exports.Recommendation_SearchPartnersOptInRecommendation = new Recommendation_SearchPartnersOptInRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_MaximizeClicksOptInRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.MaximizeClicksOptInRecommendation", [
            { no: 2, name: "recommended_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int64 recommended_budget_amount_micros */ 2:
                    message.recommendedBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int64 recommended_budget_amount_micros = 2; */
        if (message.recommendedBudgetAmountMicros !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.recommendedBudgetAmountMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.MaximizeClicksOptInRecommendation
 */
exports.Recommendation_MaximizeClicksOptInRecommendation = new Recommendation_MaximizeClicksOptInRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_OptimizeAdRotationRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.OptimizeAdRotationRecommendation", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.OptimizeAdRotationRecommendation
 */
exports.Recommendation_OptimizeAdRotationRecommendation = new Recommendation_OptimizeAdRotationRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_CalloutExtensionRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.CalloutExtensionRecommendation", [
            { no: 1, name: "recommended_extensions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => extensions_3.CalloutFeedItem, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { recommendedExtensions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.CalloutFeedItem recommended_extensions */ 1:
                    message.recommendedExtensions.push(extensions_3.CalloutFeedItem.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.CalloutFeedItem recommended_extensions = 1; */
        for (let i = 0; i < message.recommendedExtensions.length; i++)
            extensions_3.CalloutFeedItem.internalBinaryWrite(message.recommendedExtensions[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.CalloutExtensionRecommendation
 */
exports.Recommendation_CalloutExtensionRecommendation = new Recommendation_CalloutExtensionRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_SitelinkExtensionRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.SitelinkExtensionRecommendation", [
            { no: 1, name: "recommended_extensions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => extensions_2.SitelinkFeedItem, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { recommendedExtensions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.SitelinkFeedItem recommended_extensions */ 1:
                    message.recommendedExtensions.push(extensions_2.SitelinkFeedItem.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.SitelinkFeedItem recommended_extensions = 1; */
        for (let i = 0; i < message.recommendedExtensions.length; i++)
            extensions_2.SitelinkFeedItem.internalBinaryWrite(message.recommendedExtensions[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.SitelinkExtensionRecommendation
 */
exports.Recommendation_SitelinkExtensionRecommendation = new Recommendation_SitelinkExtensionRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_CallExtensionRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.CallExtensionRecommendation", [
            { no: 1, name: "recommended_extensions", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => extensions_1.CallFeedItem, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { recommendedExtensions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.CallFeedItem recommended_extensions */ 1:
                    message.recommendedExtensions.push(extensions_1.CallFeedItem.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.CallFeedItem recommended_extensions = 1; */
        for (let i = 0; i < message.recommendedExtensions.length; i++)
            extensions_1.CallFeedItem.internalBinaryWrite(message.recommendedExtensions[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.CallExtensionRecommendation
 */
exports.Recommendation_CallExtensionRecommendation = new Recommendation_CallExtensionRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_KeywordMatchTypeRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.KeywordMatchTypeRecommendation", [
            { no: 1, name: "keyword", kind: "message", T: () => criteria_1.KeywordInfo, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "recommended_match_type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.KeywordMatchTypeEnum.KeywordMatchType", keyword_match_type_1.KeywordMatchTypeEnum_KeywordMatchType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { recommendedMatchType: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.common.KeywordInfo keyword */ 1:
                    message.keyword = criteria_1.KeywordInfo.internalBinaryRead(reader, reader.uint32(), options, message.keyword);
                    break;
                case /* google.ads.googleads.v11.enums.KeywordMatchTypeEnum.KeywordMatchType recommended_match_type */ 2:
                    message.recommendedMatchType = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.common.KeywordInfo keyword = 1; */
        if (message.keyword)
            criteria_1.KeywordInfo.internalBinaryWrite(message.keyword, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.enums.KeywordMatchTypeEnum.KeywordMatchType recommended_match_type = 2; */
        if (message.recommendedMatchType !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.recommendedMatchType);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.KeywordMatchTypeRecommendation
 */
exports.Recommendation_KeywordMatchTypeRecommendation = new Recommendation_KeywordMatchTypeRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_MoveUnusedBudgetRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.MoveUnusedBudgetRecommendation", [
            { no: 3, name: "excess_campaign_budget", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "budget_recommendation", kind: "message", T: () => exports.Recommendation_CampaignBudgetRecommendation, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string excess_campaign_budget */ 3:
                    message.excessCampaignBudget = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation budget_recommendation */ 2:
                    message.budgetRecommendation = exports.Recommendation_CampaignBudgetRecommendation.internalBinaryRead(reader, reader.uint32(), options, message.budgetRecommendation);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string excess_campaign_budget = 3; */
        if (message.excessCampaignBudget !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.excessCampaignBudget);
        /* google.ads.googleads.v11.resources.Recommendation.CampaignBudgetRecommendation budget_recommendation = 2; */
        if (message.budgetRecommendation)
            exports.Recommendation_CampaignBudgetRecommendation.internalBinaryWrite(message.budgetRecommendation, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.MoveUnusedBudgetRecommendation
 */
exports.Recommendation_MoveUnusedBudgetRecommendation = new Recommendation_MoveUnusedBudgetRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_TargetRoasOptInRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.TargetRoasOptInRecommendation", [
            { no: 1, name: "recommended_target_roas", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "required_campaign_budget_amount_micros", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional double recommended_target_roas */ 1:
                    message.recommendedTargetRoas = reader.double();
                    break;
                case /* optional int64 required_campaign_budget_amount_micros */ 2:
                    message.requiredCampaignBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional double recommended_target_roas = 1; */
        if (message.recommendedTargetRoas !== undefined)
            writer.tag(1, runtime_1.WireType.Bit64).double(message.recommendedTargetRoas);
        /* optional int64 required_campaign_budget_amount_micros = 2; */
        if (message.requiredCampaignBudgetAmountMicros !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.requiredCampaignBudgetAmountMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.TargetRoasOptInRecommendation
 */
exports.Recommendation_TargetRoasOptInRecommendation = new Recommendation_TargetRoasOptInRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_ResponsiveSearchAdAssetRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdAssetRecommendation", [
            { no: 1, name: "current_ad", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "recommended_assets", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad current_ad */ 1:
                    message.currentAd = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.currentAd);
                    break;
                case /* google.ads.googleads.v11.resources.Ad recommended_assets */ 2:
                    message.recommendedAssets = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.recommendedAssets);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad current_ad = 1; */
        if (message.currentAd)
            ad_1.Ad.internalBinaryWrite(message.currentAd, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Ad recommended_assets = 2; */
        if (message.recommendedAssets)
            ad_1.Ad.internalBinaryWrite(message.recommendedAssets, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdAssetRecommendation
 */
exports.Recommendation_ResponsiveSearchAdAssetRecommendation = new Recommendation_ResponsiveSearchAdAssetRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_ResponsiveSearchAdImproveAdStrengthRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdImproveAdStrengthRecommendation", [
            { no: 1, name: "current_ad", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "recommended_ad", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad current_ad */ 1:
                    message.currentAd = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.currentAd);
                    break;
                case /* google.ads.googleads.v11.resources.Ad recommended_ad */ 2:
                    message.recommendedAd = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.recommendedAd);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad current_ad = 1; */
        if (message.currentAd)
            ad_1.Ad.internalBinaryWrite(message.currentAd, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.resources.Ad recommended_ad = 2; */
        if (message.recommendedAd)
            ad_1.Ad.internalBinaryWrite(message.recommendedAd, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdImproveAdStrengthRecommendation
 */
exports.Recommendation_ResponsiveSearchAdImproveAdStrengthRecommendation = new Recommendation_ResponsiveSearchAdImproveAdStrengthRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_ResponsiveSearchAdRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdRecommendation", [
            { no: 1, name: "ad", kind: "message", T: () => ad_1.Ad, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* google.ads.googleads.v11.resources.Ad ad */ 1:
                    message.ad = ad_1.Ad.internalBinaryRead(reader, reader.uint32(), options, message.ad);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* google.ads.googleads.v11.resources.Ad ad = 1; */
        if (message.ad)
            ad_1.Ad.internalBinaryWrite(message.ad, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.ResponsiveSearchAdRecommendation
 */
exports.Recommendation_ResponsiveSearchAdRecommendation = new Recommendation_ResponsiveSearchAdRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_UseBroadMatchKeywordRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.UseBroadMatchKeywordRecommendation", [
            { no: 1, name: "keyword", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => criteria_1.KeywordInfo, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "suggested_keywords_count", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 3, name: "campaign_keywords_count", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 4, name: "campaign_uses_shared_budget", kind: "scalar", T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 5, name: "required_campaign_budget_amount_micros", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { keyword: [], suggestedKeywordsCount: 0n, campaignKeywordsCount: 0n, campaignUsesSharedBudget: false, requiredCampaignBudgetAmountMicros: 0n };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated google.ads.googleads.v11.common.KeywordInfo keyword */ 1:
                    message.keyword.push(criteria_1.KeywordInfo.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* int64 suggested_keywords_count */ 2:
                    message.suggestedKeywordsCount = reader.int64().toBigInt();
                    break;
                case /* int64 campaign_keywords_count */ 3:
                    message.campaignKeywordsCount = reader.int64().toBigInt();
                    break;
                case /* bool campaign_uses_shared_budget */ 4:
                    message.campaignUsesSharedBudget = reader.bool();
                    break;
                case /* int64 required_campaign_budget_amount_micros */ 5:
                    message.requiredCampaignBudgetAmountMicros = reader.int64().toBigInt();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated google.ads.googleads.v11.common.KeywordInfo keyword = 1; */
        for (let i = 0; i < message.keyword.length; i++)
            criteria_1.KeywordInfo.internalBinaryWrite(message.keyword[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* int64 suggested_keywords_count = 2; */
        if (message.suggestedKeywordsCount !== 0n)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.suggestedKeywordsCount);
        /* int64 campaign_keywords_count = 3; */
        if (message.campaignKeywordsCount !== 0n)
            writer.tag(3, runtime_1.WireType.Varint).int64(message.campaignKeywordsCount);
        /* bool campaign_uses_shared_budget = 4; */
        if (message.campaignUsesSharedBudget !== false)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.campaignUsesSharedBudget);
        /* int64 required_campaign_budget_amount_micros = 5; */
        if (message.requiredCampaignBudgetAmountMicros !== 0n)
            writer.tag(5, runtime_1.WireType.Varint).int64(message.requiredCampaignBudgetAmountMicros);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.UseBroadMatchKeywordRecommendation
 */
exports.Recommendation_UseBroadMatchKeywordRecommendation = new Recommendation_UseBroadMatchKeywordRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation", [
            { no: 1, name: "merchant_id", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 2, name: "sales_country_code", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } }
        ]);
    }
    create(value) {
        const message = { merchantId: 0n, salesCountryCode: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int64 merchant_id */ 1:
                    message.merchantId = reader.int64().toBigInt();
                    break;
                case /* string sales_country_code */ 2:
                    message.salesCountryCode = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int64 merchant_id = 1; */
        if (message.merchantId !== 0n)
            writer.tag(1, runtime_1.WireType.Varint).int64(message.merchantId);
        /* string sales_country_code = 2; */
        if (message.salesCountryCode !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.salesCountryCode);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation
 */
exports.Recommendation_UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation = new Recommendation_UpgradeSmartShoppingCampaignToPerformanceMaxRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_DisplayExpansionOptInRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.DisplayExpansionOptInRecommendation", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.DisplayExpansionOptInRecommendation
 */
exports.Recommendation_DisplayExpansionOptInRecommendation = new Recommendation_DisplayExpansionOptInRecommendation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Recommendation_UpgradeLocalCampaignToPerformanceMaxRecommendation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.Recommendation.UpgradeLocalCampaignToPerformanceMaxRecommendation", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.Recommendation.UpgradeLocalCampaignToPerformanceMaxRecommendation
 */
exports.Recommendation_UpgradeLocalCampaignToPerformanceMaxRecommendation = new Recommendation_UpgradeLocalCampaignToPerformanceMaxRecommendation$Type();
