// google-ads-api-client/dist/protos/bindings/ads/resources/smart_campaign_setting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmartCampaignSetting_AdOptimizedBusinessProfileSetting = exports.SmartCampaignSetting_PhoneNumber = exports.SmartCampaignSetting = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
// @generated message type with reflection information, may provide speed optimized methods
class SmartCampaignSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.SmartCampaignSetting", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/SmartCampaignSetting" } } },
            { no: 2, name: "campaign", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"], "google.api.resource_reference": { type: "googleads.googleapis.com/Campaign" } } },
            { no: 3, name: "phone_number", kind: "message", T: () => exports.SmartCampaignSetting_PhoneNumber },
            { no: 7, name: "advertising_language_code", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "final_url", kind: "scalar", oneof: "landingPage", T: 9 /*ScalarType.STRING*/ },
            { no: 9, name: "ad_optimized_business_profile_setting", kind: "message", oneof: "landingPage", T: () => exports.SmartCampaignSetting_AdOptimizedBusinessProfileSetting },
            { no: 5, name: "business_name", kind: "scalar", oneof: "businessSetting", T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "business_profile_location", kind: "scalar", oneof: "businessSetting", T: 9 /*ScalarType.STRING*/ }
        ], { "google.api.resource": { type: "googleads.googleapis.com/SmartCampaignSetting", pattern: ["customers/{customer_id}/smartCampaignSettings/{campaign_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", campaign: "", advertisingLanguageCode: "", landingPage: { oneofKind: undefined }, businessSetting: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* string campaign */ 2:
                    message.campaign = reader.string();
                    break;
                case /* google.ads.googleads.v11.resources.SmartCampaignSetting.PhoneNumber phone_number */ 3:
                    message.phoneNumber = exports.SmartCampaignSetting_PhoneNumber.internalBinaryRead(reader, reader.uint32(), options, message.phoneNumber);
                    break;
                case /* string advertising_language_code */ 7:
                    message.advertisingLanguageCode = reader.string();
                    break;
                case /* string final_url */ 8:
                    message.landingPage = {
                        oneofKind: "finalUrl",
                        finalUrl: reader.string()
                    };
                    break;
                case /* google.ads.googleads.v11.resources.SmartCampaignSetting.AdOptimizedBusinessProfileSetting ad_optimized_business_profile_setting */ 9:
                    message.landingPage = {
                        oneofKind: "adOptimizedBusinessProfileSetting",
                        adOptimizedBusinessProfileSetting: exports.SmartCampaignSetting_AdOptimizedBusinessProfileSetting.internalBinaryRead(reader, reader.uint32(), options, message.landingPage.adOptimizedBusinessProfileSetting)
                    };
                    break;
                case /* string business_name */ 5:
                    message.businessSetting = {
                        oneofKind: "businessName",
                        businessName: reader.string()
                    };
                    break;
                case /* string business_profile_location */ 10:
                    message.businessSetting = {
                        oneofKind: "businessProfileLocation",
                        businessProfileLocation: reader.string()
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* string campaign = 2; */
        if (message.campaign !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.campaign);
        /* google.ads.googleads.v11.resources.SmartCampaignSetting.PhoneNumber phone_number = 3; */
        if (message.phoneNumber)
            exports.SmartCampaignSetting_PhoneNumber.internalBinaryWrite(message.phoneNumber, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string advertising_language_code = 7; */
        if (message.advertisingLanguageCode !== "")
            writer.tag(7, runtime_1.WireType.LengthDelimited).string(message.advertisingLanguageCode);
        /* string final_url = 8; */
        if (message.landingPage.oneofKind === "finalUrl")
            writer.tag(8, runtime_1.WireType.LengthDelimited).string(message.landingPage.finalUrl);
        /* google.ads.googleads.v11.resources.SmartCampaignSetting.AdOptimizedBusinessProfileSetting ad_optimized_business_profile_setting = 9; */
        if (message.landingPage.oneofKind === "adOptimizedBusinessProfileSetting")
            exports.SmartCampaignSetting_AdOptimizedBusinessProfileSetting.internalBinaryWrite(message.landingPage.adOptimizedBusinessProfileSetting, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string business_name = 5; */
        if (message.businessSetting.oneofKind === "businessName")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.businessSetting.businessName);
        /* string business_profile_location = 10; */
        if (message.businessSetting.oneofKind === "businessProfileLocation")
            writer.tag(10, runtime_1.WireType.LengthDelimited).string(message.businessSetting.businessProfileLocation);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.SmartCampaignSetting
 */
exports.SmartCampaignSetting = new SmartCampaignSetting$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SmartCampaignSetting_PhoneNumber$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.SmartCampaignSetting.PhoneNumber", [
            { no: 1, name: "phone_number", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "country_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string phone_number */ 1:
                    message.phoneNumber = reader.string();
                    break;
                case /* optional string country_code */ 2:
                    message.countryCode = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string phone_number = 1; */
        if (message.phoneNumber !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.phoneNumber);
        /* optional string country_code = 2; */
        if (message.countryCode !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.countryCode);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.SmartCampaignSetting.PhoneNumber
 */
exports.SmartCampaignSetting_PhoneNumber = new SmartCampaignSetting_PhoneNumber$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SmartCampaignSetting_AdOptimizedBusinessProfileSetting$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.SmartCampaignSetting.AdOptimizedBusinessProfileSetting", [
            { no: 1, name: "include_lead_form", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { includeLeadForm: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* bool include_lead_form */ 1:
                    message.includeLeadForm = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* bool include_lead_form = 1; */
        if (message.includeLeadForm !== false)
            writer.tag(1, runtime_1.WireType.Varint).bool(message.includeLeadForm);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.SmartCampaignSetting.AdOptimizedBusinessProfileSetting
 */
exports.SmartCampaignSetting_AdOptimizedBusinessProfileSetting = new SmartCampaignSetting_AdOptimizedBusinessProfileSetting$Type();
