// google-ads-api-client/dist/protos/bindings/ads/resources/user_list.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserList = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const user_lists_1 = require("../common/user_lists");
const user_lists_2 = require("../common/user_lists");
const user_lists_3 = require("../common/user_lists");
const user_lists_4 = require("../common/user_lists");
const user_lists_5 = require("../common/user_lists");
const user_list_access_status_1 = require("../enums/user_list_access_status");
const access_reason_1 = require("../enums/access_reason");
const user_list_closing_reason_1 = require("../enums/user_list_closing_reason");
const user_list_type_1 = require("../enums/user_list_type");
const user_list_size_range_1 = require("../enums/user_list_size_range");
const user_list_membership_status_1 = require("../enums/user_list_membership_status");
// @generated message type with reflection information, may provide speed optimized methods
class UserList$Type extends runtime_5.MessageType {
    constructor() {
        super("google.ads.googleads.v11.resources.UserList", [
            { no: 1, name: "resource_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["IMMUTABLE"], "google.api.resource_reference": { type: "googleads.googleapis.com/UserList" } } },
            { no: 25, name: "id", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 26, name: "read_only", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 27, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 28, name: "description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "membership_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListMembershipStatusEnum.UserListMembershipStatus", user_list_membership_status_1.UserListMembershipStatusEnum_UserListMembershipStatus] },
            { no: 29, name: "integration_code", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 30, name: "membership_life_span", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 31, name: "size_for_display", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "size_range_for_display", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListSizeRangeEnum.UserListSizeRange", user_list_size_range_1.UserListSizeRangeEnum_UserListSizeRange], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 32, name: "size_for_search", kind: "scalar", opt: true, T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 12, name: "size_range_for_search", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListSizeRangeEnum.UserListSizeRange", user_list_size_range_1.UserListSizeRangeEnum_UserListSizeRange], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 13, name: "type", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListTypeEnum.UserListType", user_list_type_1.UserListTypeEnum_UserListType], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 14, name: "closing_reason", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListClosingReasonEnum.UserListClosingReason", user_list_closing_reason_1.UserListClosingReasonEnum_UserListClosingReason] },
            { no: 15, name: "access_reason", kind: "enum", T: () => ["google.ads.googleads.v11.enums.AccessReasonEnum.AccessReason", access_reason_1.AccessReasonEnum_AccessReason], options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 16, name: "account_user_list_status", kind: "enum", T: () => ["google.ads.googleads.v11.enums.UserListAccessStatusEnum.UserListAccessStatus", user_list_access_status_1.UserListAccessStatusEnum_UserListAccessStatus] },
            { no: 33, name: "eligible_for_search", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 34, name: "eligible_for_display", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 24, name: "match_rate_percentage", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 19, name: "crm_based_user_list", kind: "message", oneof: "userList", T: () => user_lists_5.CrmBasedUserListInfo },
            { no: 20, name: "similar_user_list", kind: "message", oneof: "userList", T: () => user_lists_4.SimilarUserListInfo, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 21, name: "rule_based_user_list", kind: "message", oneof: "userList", T: () => user_lists_3.RuleBasedUserListInfo },
            { no: 22, name: "logical_user_list", kind: "message", oneof: "userList", T: () => user_lists_2.LogicalUserListInfo },
            { no: 23, name: "basic_user_list", kind: "message", oneof: "userList", T: () => user_lists_1.BasicUserListInfo }
        ], { "google.api.resource": { type: "googleads.googleapis.com/UserList", pattern: ["customers/{customer_id}/userLists/{user_list_id}"] } });
    }
    create(value) {
        const message = { resourceName: "", membershipStatus: 0, sizeRangeForDisplay: 0, sizeRangeForSearch: 0, type: 0, closingReason: 0, accessReason: 0, accountUserListStatus: 0, userList: { oneofKind: undefined } };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string resource_name */ 1:
                    message.resourceName = reader.string();
                    break;
                case /* optional int64 id */ 25:
                    message.id = reader.int64().toBigInt();
                    break;
                case /* optional bool read_only */ 26:
                    message.readOnly = reader.bool();
                    break;
                case /* optional string name */ 27:
                    message.name = reader.string();
                    break;
                case /* optional string description */ 28:
                    message.description = reader.string();
                    break;
                case /* google.ads.googleads.v11.enums.UserListMembershipStatusEnum.UserListMembershipStatus membership_status */ 6:
                    message.membershipStatus = reader.int32();
                    break;
                case /* optional string integration_code */ 29:
                    message.integrationCode = reader.string();
                    break;
                case /* optional int64 membership_life_span */ 30:
                    message.membershipLifeSpan = reader.int64().toBigInt();
                    break;
                case /* optional int64 size_for_display */ 31:
                    message.sizeForDisplay = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.UserListSizeRangeEnum.UserListSizeRange size_range_for_display */ 10:
                    message.sizeRangeForDisplay = reader.int32();
                    break;
                case /* optional int64 size_for_search */ 32:
                    message.sizeForSearch = reader.int64().toBigInt();
                    break;
                case /* google.ads.googleads.v11.enums.UserListSizeRangeEnum.UserListSizeRange size_range_for_search */ 12:
                    message.sizeRangeForSearch = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.UserListTypeEnum.UserListType type */ 13:
                    message.type = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.UserListClosingReasonEnum.UserListClosingReason closing_reason */ 14:
                    message.closingReason = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.AccessReasonEnum.AccessReason access_reason */ 15:
                    message.accessReason = reader.int32();
                    break;
                case /* google.ads.googleads.v11.enums.UserListAccessStatusEnum.UserListAccessStatus account_user_list_status */ 16:
                    message.accountUserListStatus = reader.int32();
                    break;
                case /* optional bool eligible_for_search */ 33:
                    message.eligibleForSearch = reader.bool();
                    break;
                case /* optional bool eligible_for_display */ 34:
                    message.eligibleForDisplay = reader.bool();
                    break;
                case /* optional int32 match_rate_percentage */ 24:
                    message.matchRatePercentage = reader.int32();
                    break;
                case /* google.ads.googleads.v11.common.CrmBasedUserListInfo crm_based_user_list */ 19:
                    message.userList = {
                        oneofKind: "crmBasedUserList",
                        crmBasedUserList: user_lists_5.CrmBasedUserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.userList.crmBasedUserList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.SimilarUserListInfo similar_user_list */ 20:
                    message.userList = {
                        oneofKind: "similarUserList",
                        similarUserList: user_lists_4.SimilarUserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.userList.similarUserList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.RuleBasedUserListInfo rule_based_user_list */ 21:
                    message.userList = {
                        oneofKind: "ruleBasedUserList",
                        ruleBasedUserList: user_lists_3.RuleBasedUserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.userList.ruleBasedUserList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.LogicalUserListInfo logical_user_list */ 22:
                    message.userList = {
                        oneofKind: "logicalUserList",
                        logicalUserList: user_lists_2.LogicalUserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.userList.logicalUserList)
                    };
                    break;
                case /* google.ads.googleads.v11.common.BasicUserListInfo basic_user_list */ 23:
                    message.userList = {
                        oneofKind: "basicUserList",
                        basicUserList: user_lists_1.BasicUserListInfo.internalBinaryRead(reader, reader.uint32(), options, message.userList.basicUserList)
                    };
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string resource_name = 1; */
        if (message.resourceName !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.resourceName);
        /* optional int64 id = 25; */
        if (message.id !== undefined)
            writer.tag(25, runtime_1.WireType.Varint).int64(message.id);
        /* optional bool read_only = 26; */
        if (message.readOnly !== undefined)
            writer.tag(26, runtime_1.WireType.Varint).bool(message.readOnly);
        /* optional string name = 27; */
        if (message.name !== undefined)
            writer.tag(27, runtime_1.WireType.LengthDelimited).string(message.name);
        /* optional string description = 28; */
        if (message.description !== undefined)
            writer.tag(28, runtime_1.WireType.LengthDelimited).string(message.description);
        /* google.ads.googleads.v11.enums.UserListMembershipStatusEnum.UserListMembershipStatus membership_status = 6; */
        if (message.membershipStatus !== 0)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.membershipStatus);
        /* optional string integration_code = 29; */
        if (message.integrationCode !== undefined)
            writer.tag(29, runtime_1.WireType.LengthDelimited).string(message.integrationCode);
        /* optional int64 membership_life_span = 30; */
        if (message.membershipLifeSpan !== undefined)
            writer.tag(30, runtime_1.WireType.Varint).int64(message.membershipLifeSpan);
        /* optional int64 size_for_display = 31; */
        if (message.sizeForDisplay !== undefined)
            writer.tag(31, runtime_1.WireType.Varint).int64(message.sizeForDisplay);
        /* google.ads.googleads.v11.enums.UserListSizeRangeEnum.UserListSizeRange size_range_for_display = 10; */
        if (message.sizeRangeForDisplay !== 0)
            writer.tag(10, runtime_1.WireType.Varint).int32(message.sizeRangeForDisplay);
        /* optional int64 size_for_search = 32; */
        if (message.sizeForSearch !== undefined)
            writer.tag(32, runtime_1.WireType.Varint).int64(message.sizeForSearch);
        /* google.ads.googleads.v11.enums.UserListSizeRangeEnum.UserListSizeRange size_range_for_search = 12; */
        if (message.sizeRangeForSearch !== 0)
            writer.tag(12, runtime_1.WireType.Varint).int32(message.sizeRangeForSearch);
        /* google.ads.googleads.v11.enums.UserListTypeEnum.UserListType type = 13; */
        if (message.type !== 0)
            writer.tag(13, runtime_1.WireType.Varint).int32(message.type);
        /* google.ads.googleads.v11.enums.UserListClosingReasonEnum.UserListClosingReason closing_reason = 14; */
        if (message.closingReason !== 0)
            writer.tag(14, runtime_1.WireType.Varint).int32(message.closingReason);
        /* google.ads.googleads.v11.enums.AccessReasonEnum.AccessReason access_reason = 15; */
        if (message.accessReason !== 0)
            writer.tag(15, runtime_1.WireType.Varint).int32(message.accessReason);
        /* google.ads.googleads.v11.enums.UserListAccessStatusEnum.UserListAccessStatus account_user_list_status = 16; */
        if (message.accountUserListStatus !== 0)
            writer.tag(16, runtime_1.WireType.Varint).int32(message.accountUserListStatus);
        /* optional bool eligible_for_search = 33; */
        if (message.eligibleForSearch !== undefined)
            writer.tag(33, runtime_1.WireType.Varint).bool(message.eligibleForSearch);
        /* optional bool eligible_for_display = 34; */
        if (message.eligibleForDisplay !== undefined)
            writer.tag(34, runtime_1.WireType.Varint).bool(message.eligibleForDisplay);
        /* optional int32 match_rate_percentage = 24; */
        if (message.matchRatePercentage !== undefined)
            writer.tag(24, runtime_1.WireType.Varint).int32(message.matchRatePercentage);
        /* google.ads.googleads.v11.common.CrmBasedUserListInfo crm_based_user_list = 19; */
        if (message.userList.oneofKind === "crmBasedUserList")
            user_lists_5.CrmBasedUserListInfo.internalBinaryWrite(message.userList.crmBasedUserList, writer.tag(19, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.SimilarUserListInfo similar_user_list = 20; */
        if (message.userList.oneofKind === "similarUserList")
            user_lists_4.SimilarUserListInfo.internalBinaryWrite(message.userList.similarUserList, writer.tag(20, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.RuleBasedUserListInfo rule_based_user_list = 21; */
        if (message.userList.oneofKind === "ruleBasedUserList")
            user_lists_3.RuleBasedUserListInfo.internalBinaryWrite(message.userList.ruleBasedUserList, writer.tag(21, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.LogicalUserListInfo logical_user_list = 22; */
        if (message.userList.oneofKind === "logicalUserList")
            user_lists_2.LogicalUserListInfo.internalBinaryWrite(message.userList.logicalUserList, writer.tag(22, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.ads.googleads.v11.common.BasicUserListInfo basic_user_list = 23; */
        if (message.userList.oneofKind === "basicUserList")
            user_lists_1.BasicUserListInfo.internalBinaryWrite(message.userList.basicUserList, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.ads.googleads.v11.resources.UserList
 */
exports.UserList = new UserList$Type();
