// google-ads-api-client/dist/protos/bindings/google/api/expr/v1beta1/source.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourcePosition = exports.SourceInfo = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
// @generated message type with reflection information, may provide speed optimized methods
class SourceInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.expr.v1beta1.SourceInfo", [
            { no: 2, name: "location", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "line_offsets", kind: "scalar", repeat: 1 /*RepeatType.PACKED*/, T: 5 /*ScalarType.INT32*/ },
            { no: 4, name: "positions", kind: "map", K: 5 /*ScalarType.INT32*/, V: { kind: "scalar", T: 5 /*ScalarType.INT32*/ } }
        ]);
    }
    create(value) {
        const message = { location: "", lineOffsets: [], positions: {} };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string location */ 2:
                    message.location = reader.string();
                    break;
                case /* repeated int32 line_offsets */ 3:
                    if (wireType === runtime_2.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.lineOffsets.push(reader.int32());
                    else
                        message.lineOffsets.push(reader.int32());
                    break;
                case /* map<int32, int32> positions */ 4:
                    this.binaryReadMap4(message.positions, reader, options);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    binaryReadMap4(map, reader, options) {
        let len = reader.uint32(), end = reader.pos + len, key, val;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case 1:
                    key = reader.int32();
                    break;
                case 2:
                    val = reader.int32();
                    break;
                default: throw new globalThis.Error("unknown map entry field for field google.api.expr.v1beta1.SourceInfo.positions");
            }
        }
        map[key ?? 0] = val ?? 0;
    }
    internalBinaryWrite(message, writer, options) {
        /* string location = 2; */
        if (message.location !== "")
            writer.tag(2, runtime_2.WireType.LengthDelimited).string(message.location);
        /* repeated int32 line_offsets = 3; */
        if (message.lineOffsets.length) {
            writer.tag(3, runtime_2.WireType.LengthDelimited).fork();
            for (let i = 0; i < message.lineOffsets.length; i++)
                writer.int32(message.lineOffsets[i]);
            writer.join();
        }
        /* map<int32, int32> positions = 4; */
        for (let k of Object.keys(message.positions))
            writer.tag(4, runtime_2.WireType.LengthDelimited).fork().tag(1, runtime_2.WireType.Varint).int32(parseInt(k)).tag(2, runtime_2.WireType.Varint).int32(message.positions[k]).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.expr.v1beta1.SourceInfo
 */
exports.SourceInfo = new SourceInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SourcePosition$Type extends runtime_5.MessageType {
    constructor() {
        super("google.api.expr.v1beta1.SourcePosition", [
            { no: 1, name: "location", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "offset", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "line", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 4, name: "column", kind: "scalar", T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = { location: "", offset: 0, line: 0, column: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string location */ 1:
                    message.location = reader.string();
                    break;
                case /* int32 offset */ 2:
                    message.offset = reader.int32();
                    break;
                case /* int32 line */ 3:
                    message.line = reader.int32();
                    break;
                case /* int32 column */ 4:
                    message.column = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_1.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string location = 1; */
        if (message.location !== "")
            writer.tag(1, runtime_2.WireType.LengthDelimited).string(message.location);
        /* int32 offset = 2; */
        if (message.offset !== 0)
            writer.tag(2, runtime_2.WireType.Varint).int32(message.offset);
        /* int32 line = 3; */
        if (message.line !== 0)
            writer.tag(3, runtime_2.WireType.Varint).int32(message.line);
        /* int32 column = 4; */
        if (message.column !== 0)
            writer.tag(4, runtime_2.WireType.Varint).int32(message.column);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_1.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.api.expr.v1beta1.SourcePosition
 */
exports.SourcePosition = new SourcePosition$Type();
