// google-ads-api-client/dist/protos/bindings/google/logging/v2/log_entry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogSplit = exports.LogEntrySourceLocation = exports.LogEntryOperation = exports.LogEntry = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const http_request_1 = require("../type/http_request");
const log_severity_1 = require("../type/log_severity");
const timestamp_1 = require("../../protobuf/timestamp");
const struct_1 = require("../../protobuf/struct");
const any_1 = require("../../protobuf/any");
const monitored_resource_1 = require("../../api/monitored_resource");
// @generated message type with reflection information, may provide speed optimized methods
class LogEntry$Type extends runtime_5.MessageType {
    constructor() {
        super("google.logging.v2.LogEntry", [
            { no: 12, name: "log_name", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 8, name: "resource", kind: "message", T: () => monitored_resource_1.MonitoredResource, options: { "google.api.field_behavior": ["REQUIRED"] } },
            { no: 2, name: "proto_payload", kind: "message", oneof: "payload", T: () => any_1.Any },
            { no: 3, name: "text_payload", kind: "scalar", oneof: "payload", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "json_payload", kind: "message", oneof: "payload", T: () => struct_1.Struct },
            { no: 9, name: "timestamp", kind: "message", T: () => timestamp_1.Timestamp, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 24, name: "receive_timestamp", kind: "message", T: () => timestamp_1.Timestamp, options: { "google.api.field_behavior": ["OUTPUT_ONLY"] } },
            { no: 10, name: "severity", kind: "enum", T: () => ["google.logging.type.LogSeverity", log_severity_1.LogSeverity], options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 4, name: "insert_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 7, name: "http_request", kind: "message", T: () => http_request_1.HttpRequest, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 11, name: "labels", kind: "map", K: 9 /*ScalarType.STRING*/, V: { kind: "scalar", T: 9 /*ScalarType.STRING*/ }, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 15, name: "operation", kind: "message", T: () => exports.LogEntryOperation, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 22, name: "trace", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 27, name: "span_id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 30, name: "trace_sampled", kind: "scalar", T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 23, name: "source_location", kind: "message", T: () => exports.LogEntrySourceLocation, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 35, name: "split", kind: "message", T: () => exports.LogSplit, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ], { "google.api.resource": { type: "logging.googleapis.com/Log", pattern: ["projects/{project}/logs/{log}", "organizations/{organization}/logs/{log}", "folders/{folder}/logs/{log}", "billingAccounts/{billing_account}/logs/{log}"], nameField: "log_name" } });
    }
    create(value) {
        const message = { logName: "", payload: { oneofKind: undefined }, severity: 0, insertId: "", labels: {}, trace: "", spanId: "", traceSampled: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string log_name */ 12:
                    message.logName = reader.string();
                    break;
                case /* google.api.MonitoredResource resource */ 8:
                    message.resource = monitored_resource_1.MonitoredResource.internalBinaryRead(reader, reader.uint32(), options, message.resource);
                    break;
                case /* google.protobuf.Any proto_payload */ 2:
                    message.payload = {
                        oneofKind: "protoPayload",
                        protoPayload: any_1.Any.internalBinaryRead(reader, reader.uint32(), options, message.payload.protoPayload)
                    };
                    break;
                case /* string text_payload */ 3:
                    message.payload = {
                        oneofKind: "textPayload",
                        textPayload: reader.string()
                    };
                    break;
                case /* google.protobuf.Struct json_payload */ 6:
                    message.payload = {
                        oneofKind: "jsonPayload",
                        jsonPayload: struct_1.Struct.internalBinaryRead(reader, reader.uint32(), options, message.payload.jsonPayload)
                    };
                    break;
                case /* google.protobuf.Timestamp timestamp */ 9:
                    message.timestamp = timestamp_1.Timestamp.internalBinaryRead(reader, reader.uint32(), options, message.timestamp);
                    break;
                case /* google.protobuf.Timestamp receive_timestamp */ 24:
                    message.receiveTimestamp = timestamp_1.Timestamp.internalBinaryRead(reader, reader.uint32(), options, message.receiveTimestamp);
                    break;
                case /* google.logging.type.LogSeverity severity */ 10:
                    message.severity = reader.int32();
                    break;
                case /* string insert_id */ 4:
                    message.insertId = reader.string();
                    break;
                case /* google.logging.type.HttpRequest http_request */ 7:
                    message.httpRequest = http_request_1.HttpRequest.internalBinaryRead(reader, reader.uint32(), options, message.httpRequest);
                    break;
                case /* map<string, string> labels */ 11:
                    this.binaryReadMap11(message.labels, reader, options);
                    break;
                case /* google.logging.v2.LogEntryOperation operation */ 15:
                    message.operation = exports.LogEntryOperation.internalBinaryRead(reader, reader.uint32(), options, message.operation);
                    break;
                case /* string trace */ 22:
                    message.trace = reader.string();
                    break;
                case /* string span_id */ 27:
                    message.spanId = reader.string();
                    break;
                case /* bool trace_sampled */ 30:
                    message.traceSampled = reader.bool();
                    break;
                case /* google.logging.v2.LogEntrySourceLocation source_location */ 23:
                    message.sourceLocation = exports.LogEntrySourceLocation.internalBinaryRead(reader, reader.uint32(), options, message.sourceLocation);
                    break;
                case /* google.logging.v2.LogSplit split */ 35:
                    message.split = exports.LogSplit.internalBinaryRead(reader, reader.uint32(), options, message.split);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    binaryReadMap11(map, reader, options) {
        let len = reader.uint32(), end = reader.pos + len, key, val;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case 1:
                    key = reader.string();
                    break;
                case 2:
                    val = reader.string();
                    break;
                default: throw new globalThis.Error("unknown map entry field for field google.logging.v2.LogEntry.labels");
            }
        }
        map[key ?? ""] = val ?? "";
    }
    internalBinaryWrite(message, writer, options) {
        /* string log_name = 12; */
        if (message.logName !== "")
            writer.tag(12, runtime_1.WireType.LengthDelimited).string(message.logName);
        /* google.api.MonitoredResource resource = 8; */
        if (message.resource)
            monitored_resource_1.MonitoredResource.internalBinaryWrite(message.resource, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.protobuf.Any proto_payload = 2; */
        if (message.payload.oneofKind === "protoPayload")
            any_1.Any.internalBinaryWrite(message.payload.protoPayload, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string text_payload = 3; */
        if (message.payload.oneofKind === "textPayload")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.payload.textPayload);
        /* google.protobuf.Struct json_payload = 6; */
        if (message.payload.oneofKind === "jsonPayload")
            struct_1.Struct.internalBinaryWrite(message.payload.jsonPayload, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.protobuf.Timestamp timestamp = 9; */
        if (message.timestamp)
            timestamp_1.Timestamp.internalBinaryWrite(message.timestamp, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.protobuf.Timestamp receive_timestamp = 24; */
        if (message.receiveTimestamp)
            timestamp_1.Timestamp.internalBinaryWrite(message.receiveTimestamp, writer.tag(24, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.logging.type.LogSeverity severity = 10; */
        if (message.severity !== 0)
            writer.tag(10, runtime_1.WireType.Varint).int32(message.severity);
        /* string insert_id = 4; */
        if (message.insertId !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.insertId);
        /* google.logging.type.HttpRequest http_request = 7; */
        if (message.httpRequest)
            http_request_1.HttpRequest.internalBinaryWrite(message.httpRequest, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* map<string, string> labels = 11; */
        for (let k of Object.keys(message.labels))
            writer.tag(11, runtime_1.WireType.LengthDelimited).fork().tag(1, runtime_1.WireType.LengthDelimited).string(k).tag(2, runtime_1.WireType.LengthDelimited).string(message.labels[k]).join();
        /* google.logging.v2.LogEntryOperation operation = 15; */
        if (message.operation)
            exports.LogEntryOperation.internalBinaryWrite(message.operation, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string trace = 22; */
        if (message.trace !== "")
            writer.tag(22, runtime_1.WireType.LengthDelimited).string(message.trace);
        /* string span_id = 27; */
        if (message.spanId !== "")
            writer.tag(27, runtime_1.WireType.LengthDelimited).string(message.spanId);
        /* bool trace_sampled = 30; */
        if (message.traceSampled !== false)
            writer.tag(30, runtime_1.WireType.Varint).bool(message.traceSampled);
        /* google.logging.v2.LogEntrySourceLocation source_location = 23; */
        if (message.sourceLocation)
            exports.LogEntrySourceLocation.internalBinaryWrite(message.sourceLocation, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* google.logging.v2.LogSplit split = 35; */
        if (message.split)
            exports.LogSplit.internalBinaryWrite(message.split, writer.tag(35, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.logging.v2.LogEntry
 */
exports.LogEntry = new LogEntry$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LogEntryOperation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.logging.v2.LogEntryOperation", [
            { no: 1, name: "id", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 2, name: "producer", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 3, name: "first", kind: "scalar", T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 4, name: "last", kind: "scalar", T: 8 /*ScalarType.BOOL*/, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = { id: "", producer: "", first: false, last: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string id */ 1:
                    message.id = reader.string();
                    break;
                case /* string producer */ 2:
                    message.producer = reader.string();
                    break;
                case /* bool first */ 3:
                    message.first = reader.bool();
                    break;
                case /* bool last */ 4:
                    message.last = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string id = 1; */
        if (message.id !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.id);
        /* string producer = 2; */
        if (message.producer !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.producer);
        /* bool first = 3; */
        if (message.first !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.first);
        /* bool last = 4; */
        if (message.last !== false)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.last);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.logging.v2.LogEntryOperation
 */
exports.LogEntryOperation = new LogEntryOperation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LogEntrySourceLocation$Type extends runtime_5.MessageType {
    constructor() {
        super("google.logging.v2.LogEntrySourceLocation", [
            { no: 1, name: "file", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 2, name: "line", kind: "scalar", T: 3 /*ScalarType.INT64*/, L: 0 /*LongType.BIGINT*/, options: { "google.api.field_behavior": ["OPTIONAL"] } },
            { no: 3, name: "function", kind: "scalar", T: 9 /*ScalarType.STRING*/, options: { "google.api.field_behavior": ["OPTIONAL"] } }
        ]);
    }
    create(value) {
        const message = { file: "", line: 0n, function: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string file */ 1:
                    message.file = reader.string();
                    break;
                case /* int64 line */ 2:
                    message.line = reader.int64().toBigInt();
                    break;
                case /* string function */ 3:
                    message.function = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string file = 1; */
        if (message.file !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.file);
        /* int64 line = 2; */
        if (message.line !== 0n)
            writer.tag(2, runtime_1.WireType.Varint).int64(message.line);
        /* string function = 3; */
        if (message.function !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.function);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.logging.v2.LogEntrySourceLocation
 */
exports.LogEntrySourceLocation = new LogEntrySourceLocation$Type();
// @generated message type with reflection information, may provide speed optimized methods
class LogSplit$Type extends runtime_5.MessageType {
    constructor() {
        super("google.logging.v2.LogSplit", [
            { no: 1, name: "uid", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "index", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "total_splits", kind: "scalar", T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = { uid: "", index: 0, totalSplits: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string uid */ 1:
                    message.uid = reader.string();
                    break;
                case /* int32 index */ 2:
                    message.index = reader.int32();
                    break;
                case /* int32 total_splits */ 3:
                    message.totalSplits = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string uid = 1; */
        if (message.uid !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.uid);
        /* int32 index = 2; */
        if (message.index !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.index);
        /* int32 total_splits = 3; */
        if (message.totalSplits !== 0)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.totalSplits);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message google.logging.v2.LogSplit
 */
exports.LogSplit = new LogSplit$Type();
